
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="NWChem Website">
      
      
      
        <link rel="canonical" href="https://nwchemgit.github.io/print_page.html">
      
      
      
      
      <link rel="icon" href="assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.0, mkdocs-material-9.5.26">
    
    
      
        <title>Print Site - NWChem</title>
      
    
    
      <link rel="stylesheet" href="assets/stylesheets/main.6543a935.min.css">
      
        
        <link rel="stylesheet" href="assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
    
    
      <link rel="stylesheet" href="stylesheets/extra.css">
    
    <script>__md_scope=new URL("/",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  
        <script type="text/javascript">
        document.addEventListener('DOMContentLoaded', function () {
            remove_material_navigation();remove_mkdocs_theme_navigation();generate_toc();
        })
        </script>
        </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#index" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="index.html" title="NWChem" class="md-header__button md-logo" aria-label="NWChem" data-md-component="logo">
      
  <img src="assets/MS3logo.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            NWChem
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Print Site
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var media,input,key,value,palette=__md_get("__palette");if(palette&&palette.color){"(prefers-color-scheme)"===palette.color.media&&(media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']"),palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent"));for([key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08Z"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/nwchemgit/nwchem-wiki" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.5.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="index.html" title="NWChem" class="md-nav__button md-logo" aria-label="NWChem" data-md-component="logo">
      
  <img src="assets/MS3logo.png" alt="logo">

    </a>
    NWChem
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/nwchemgit/nwchem-wiki" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.5.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="index.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NWChem Homepage
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="Home.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NWChem Manual
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    How to get NWChem
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            How to get NWChem
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Download.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    How to download and install NWChem
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Compiling-NWChem.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Compiling NWChem from source
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Introduction 
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Introduction 
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Getting-Started.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Getting Started
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="NWChem-Architecture.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NWChem Architecture
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="Capabilities.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Functionalities
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="Top-level.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Top-level Directives
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="Geometry.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Geometries
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="Basis.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Basis sets
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="ECP.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Effective Core Potentials
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="Relativistic-All-electron-Approximations.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Relativistic all-electron approximations
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_11" >
        
          
          <label class="md-nav__link" for="__nav_11" id="__nav_11_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Quantum-Mechanical-Methods
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_11_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_11">
            <span class="md-nav__icon md-icon"></span>
            Quantum-Mechanical-Methods
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Hartree-Fock-Theory-for-Molecules.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Hartree-Fock
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Density-Functional-Theory-for-Molecules.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Density Functional Theory (DFT)
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Excited-State-Calculations.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    CIS, TDHF, TDDFT
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="RT-TDDFT.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Real-time TDDFT
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Plane-Wave-Density-Functional-Theory.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Plane-Wave Density Functional Theory
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="TCE.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Tensor Contraction Engine Module: CI, MBPT, and CC
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="MP2.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    MP2
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="CCSD.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Coupled Cluster Calculations
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Multiconfiguration_SCF.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    MCSCF
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="GW.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Molecular GW
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="XTB.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    XTB
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_12" >
        
          
          <label class="md-nav__link" for="__nav_12" id="__nav_12_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Quantum Molecular Dynamics
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_12_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_12">
            <span class="md-nav__icon md-icon"></span>
            Quantum Molecular Dynamics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Plane-Wave-Density-Functional-Theory.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Plane-Wave Density Functional Theory
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Gaussian-Basis-AIMD.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gaussian Basis AIMD
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_13" >
        
          
          <label class="md-nav__link" for="__nav_13" id="__nav_13_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Hybrid Approaches
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_13_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_13">
            <span class="md-nav__icon md-icon"></span>
            Hybrid Approaches
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Solvation-Models.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Solvation Models
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="VEM-Model.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    VEM (Vertical Excitation or Emission) Model
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="ONIOM.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Hybrid Calculations with ONIOM
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_13_4" >
        
          
          <label class="md-nav__link" for="__nav_13_4" id="__nav_13_4_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    QMMM
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_13_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_13_4">
            <span class="md-nav__icon md-icon"></span>
            QMMM
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="qmmm_introduction.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    QMMM Introduction
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="QMMM.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    QMMM table of contents
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_13_4_3" >
        
          
          <label class="md-nav__link" for="__nav_13_4_3" id="__nav_13_4_3_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    QMMM Restart and Topology Files
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_13_4_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_13_4_3">
            <span class="md-nav__icon md-icon"></span>
            QMMM Restart and Topology Files
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="QMMM_Restart_and_Topology_Files.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Overview
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="QMMM_Preparation_Prerequisites.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    QMMM Preparation Prerequisites
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Qmmm_preparation_basic.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Qmmm preparation basic
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Qmmm_preparation_solvation.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Qmmm preparation solvation
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Qmmm_preparation_constraints.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Qmmm preparation constraints
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_13_4_4" >
        
          
          <label class="md-nav__link" for="__nav_13_4_4" id="__nav_13_4_4_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    QMMM Input File
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_13_4_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_13_4_4">
            <span class="md-nav__icon md-icon"></span>
            QMMM Input File
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="QMMM_Input_File.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Overview
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="QMMM_QM_Parameters.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    QM_Parameters
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="QMMM_MM_Parameters.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    QMMM MM Parameters
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="QMMM_QMMM_Parameters.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    QM/MM parameters
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_13_4_5" >
        
          
          <label class="md-nav__link" for="__nav_13_4_5" id="__nav_13_4_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    QMMMM Single Point Calculations
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_13_4_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_13_4_5">
            <span class="md-nav__icon md-icon"></span>
            QMMMM Single Point Calculations
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="qmmm_sp_energy.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    QMMM Ground State Energy and Gradient
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="QMMM_Excited_States.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    QMMM Excited State Energy
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="qmmm_sp_property.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    QMMM Properties
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="QMMM_ESP.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    QMMM ESP Charge Analysis
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_13_4_6" >
        
          
          <label class="md-nav__link" for="__nav_13_4_6" id="__nav_13_4_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    QMMM Potential Energy Surface Analysis
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_13_4_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_13_4_6">
            <span class="md-nav__icon md-icon"></span>
            QMMM Potential Energy Surface Analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Qmmm_optimization.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    QMMM Optimization
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="QMMM_Transition_States.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    QMMM Transition States
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Qmmm_freq.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    QMMM Hessians and Frequency
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Qmmm_NEB_Calculations.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    QMMM Reaction Pathway Calculations with NEB
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="QMMM_Dynamics.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    QMMM Dynamics
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="QMMM_Free_Energy.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    QMMM Free Energy
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="QMMM_Appendix.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    QMMM Apppendix
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Bq.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Point charges
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="1D-RISM.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    1-D RISM
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_14" >
        
          
          <label class="md-nav__link" for="__nav_14" id="__nav_14_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Potential Energy Surface Analysis
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_14_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_14">
            <span class="md-nav__icon md-icon"></span>
            Potential Energy Surface Analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Geometry-Optimization.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Geometry Optimization
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Hessians-and-Vibrational-Frequencies.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Hessians
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Constraints.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Constraints
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Nudged-Elastic-Band-and-Zero-Temperature-String-Methods.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Nudged Elastic Band (NEB) method
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_15" >
        
          
          <label class="md-nav__link" for="__nav_15" id="__nav_15_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Electronic Structure Analysis
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_15_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_15">
            <span class="md-nav__icon md-icon"></span>
            Electronic Structure Analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Properties.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Properties
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="ESP.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Electrostatic potentials
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="DPLOT.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    DPLOT
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="FCIDUMP.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    FCIDUMP
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_16" >
        
          
          <label class="md-nav__link" for="__nav_16" id="__nav_16_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Other-Capabilities
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_16_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_16">
            <span class="md-nav__icon md-icon"></span>
            Other-Capabilities
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Electron-Transfer.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Electron Transfer
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Python.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Controlling NWChem with Python
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Quantum.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Quantum
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="VSCF.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Vibrational SCF (VSCF)
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="CCCA_method.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Correlation consistent Composite Approach (ccCA)
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Interface.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Interfaces to Other Programs
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_17" >
        
          
          <label class="md-nav__link" for="__nav_17" id="__nav_17_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Molecular Mechanics
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_17_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_17">
            <span class="md-nav__icon md-icon"></span>
            Molecular Mechanics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Prepare.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Prepare
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="MD.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Molecular dynamics
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Analysis.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Analysis
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Format_of_MD_Files.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Format of MD files
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="Python.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Controlling NWChem with Python
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_19" >
        
          
          <label class="md-nav__link" for="__nav_19" id="__nav_19_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    NWChem Examples
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_19_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_19">
            <span class="md-nav__icon md-icon"></span>
            NWChem Examples
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Sample.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Sample input files
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="Geometry-examples.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Examples of geometries using symmetry
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="Benchmarks.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Benchmarks performed with NWChem
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="Known-Bugs.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Known Bugs
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="FAQ.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    FAQ
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="Tutorials.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NWChem Tutorials
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="Forum.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Forum
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="Archived-Forum.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Archived Forum
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="EMSL_Arrows.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    EMSL Arrows
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_27" >
        
          
          <label class="md-nav__link" for="__nav_27" id="__nav_27_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Supplementary Information
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_27_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_27">
            <span class="md-nav__icon md-icon"></span>
            Supplementary Information
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="https://hpc.pnl.gov/globalarrays/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Global Arrays
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="ARMCI.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Choosing the ARMCI Library
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="Interfaces-with-External-Software.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Interfaces with External Software
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="Software-supporting-NWChem.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Software supporting NWChem
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="Containers.html" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Containers
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="https://nwchemgit.github.io/nwchemusermanual.pdf" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Documentation in PDF format
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#index" class="md-nav__link">
    <span class="md-ellipsis">
      1 NWChem Homepage
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-3" class="md-nav__link">
    <span class="md-ellipsis">
      3 How to get NWChem
    </span>
  </a>
  
    <nav class="md-nav" aria-label="3 How to get NWChem">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#download" class="md-nav__link">
    <span class="md-ellipsis">
      3.1 How to download and install NWChem
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#compiling-nwchem" class="md-nav__link">
    <span class="md-ellipsis">
      3.2 Compiling NWChem from source
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-4" class="md-nav__link">
    <span class="md-ellipsis">
      4 Introduction 
    </span>
  </a>
  
    <nav class="md-nav" aria-label="4 Introduction">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#getting-started" class="md-nav__link">
    <span class="md-ellipsis">
      4.1 Getting Started
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nwchem-architecture" class="md-nav__link">
    <span class="md-ellipsis">
      4.2 NWChem Architecture
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#capabilities" class="md-nav__link">
    <span class="md-ellipsis">
      5 Functionalities
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#top-level" class="md-nav__link">
    <span class="md-ellipsis">
      6 Top-level Directives
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#geometry" class="md-nav__link">
    <span class="md-ellipsis">
      7 Geometries
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#basis" class="md-nav__link">
    <span class="md-ellipsis">
      8 Basis sets
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ecp" class="md-nav__link">
    <span class="md-ellipsis">
      9 Effective Core Potentials
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#relativistic-all-electron-approximations" class="md-nav__link">
    <span class="md-ellipsis">
      10 Relativistic all-electron approximations
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-11" class="md-nav__link">
    <span class="md-ellipsis">
      11 Quantum-Mechanical-Methods
    </span>
  </a>
  
    <nav class="md-nav" aria-label="11 Quantum-Mechanical-Methods">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#hartree-fock-theory-for-molecules" class="md-nav__link">
    <span class="md-ellipsis">
      11.1 Hartree-Fock
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#density-functional-theory-for-molecules" class="md-nav__link">
    <span class="md-ellipsis">
      11.2 Density Functional Theory (DFT)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#excited-state-calculations" class="md-nav__link">
    <span class="md-ellipsis">
      11.3 CIS, TDHF, TDDFT
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#rt-tddft" class="md-nav__link">
    <span class="md-ellipsis">
      11.4 Real-time TDDFT
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#plane-wave-density-functional-theory" class="md-nav__link">
    <span class="md-ellipsis">
      11.5 Plane-Wave Density Functional Theory
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tce" class="md-nav__link">
    <span class="md-ellipsis">
      11.6 Tensor Contraction Engine Module: CI, MBPT, and CC
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mp2" class="md-nav__link">
    <span class="md-ellipsis">
      11.7 MP2
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ccsd" class="md-nav__link">
    <span class="md-ellipsis">
      11.8 Coupled Cluster Calculations
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#multiconfiguration_scf" class="md-nav__link">
    <span class="md-ellipsis">
      11.9 MCSCF
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gw" class="md-nav__link">
    <span class="md-ellipsis">
      11.10 Molecular GW
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#xtb" class="md-nav__link">
    <span class="md-ellipsis">
      11.11 XTB
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-12" class="md-nav__link">
    <span class="md-ellipsis">
      12 Quantum Molecular Dynamics
    </span>
  </a>
  
    <nav class="md-nav" aria-label="12 Quantum Molecular Dynamics">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#plane-wave-density-functional-theory" class="md-nav__link">
    <span class="md-ellipsis">
      12.1 Plane-Wave Density Functional Theory
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gaussian-basis-aimd" class="md-nav__link">
    <span class="md-ellipsis">
      12.2 Gaussian Basis AIMD
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-13" class="md-nav__link">
    <span class="md-ellipsis">
      13 Hybrid Approaches
    </span>
  </a>
  
    <nav class="md-nav" aria-label="13 Hybrid Approaches">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#solvation-models" class="md-nav__link">
    <span class="md-ellipsis">
      13.1 Solvation Models
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#vem-model" class="md-nav__link">
    <span class="md-ellipsis">
      13.2 VEM (Vertical Excitation or Emission) Model
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#oniom" class="md-nav__link">
    <span class="md-ellipsis">
      13.3 Hybrid Calculations with ONIOM
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#section-13-4" class="md-nav__link">
    <span class="md-ellipsis">
      13.4 QMMM
    </span>
  </a>
  
    <nav class="md-nav" aria-label="13.4 QMMM">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#qmmm_introduction" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.1 QMMM Introduction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qmmm" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.2 QMMM table of contents
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#section-13-4-3" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.3 QMMM Restart and Topology Files
    </span>
  </a>
  
    <nav class="md-nav" aria-label="13.4.3 QMMM Restart and Topology Files">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#qmmm_restart_and_topology_files" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.3.1 Overview
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qmmm_preparation_prerequisites" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.3.2 QMMM Preparation Prerequisites
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qmmm_preparation_basic" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.3.3 Qmmm preparation basic
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qmmm_preparation_solvation" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.3.4 Qmmm preparation solvation
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qmmm_preparation_constraints" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.3.5 Qmmm preparation constraints
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#section-13-4-4" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.4 QMMM Input File
    </span>
  </a>
  
    <nav class="md-nav" aria-label="13.4.4 QMMM Input File">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#qmmm_input_file" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.4.1 Overview
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qmmm_qm_parameters" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.4.2 QM_Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qmmm_mm_parameters" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.4.3 QMMM MM Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qmmm_qmmm_parameters" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.4.4 QM/MM parameters
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#section-13-4-5" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.5 QMMMM Single Point Calculations
    </span>
  </a>
  
    <nav class="md-nav" aria-label="13.4.5 QMMMM Single Point Calculations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#qmmm_sp_energy" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.5.1 QMMM Ground State Energy and Gradient
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qmmm_excited_states" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.5.2 QMMM Excited State Energy
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qmmm_sp_property" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.5.3 QMMM Properties
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qmmm_esp" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.5.4 QMMM ESP Charge Analysis
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#section-13-4-6" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.6 QMMM Potential Energy Surface Analysis
    </span>
  </a>
  
    <nav class="md-nav" aria-label="13.4.6 QMMM Potential Energy Surface Analysis">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#qmmm_optimization" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.6.1 QMMM Optimization
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qmmm_transition_states" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.6.2 QMMM Transition States
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qmmm_freq" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.6.3 QMMM Hessians and Frequency
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qmmm_neb_calculations" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.6.4 QMMM Reaction Pathway Calculations with NEB
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qmmm_dynamics" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.7 QMMM Dynamics
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qmmm_free_energy" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.8 QMMM Free Energy
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qmmm_appendix" class="md-nav__link">
    <span class="md-ellipsis">
      13.4.9 QMMM Apppendix
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bq" class="md-nav__link">
    <span class="md-ellipsis">
      13.5 Point charges
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#1d-rism" class="md-nav__link">
    <span class="md-ellipsis">
      13.6 1-D RISM
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-14" class="md-nav__link">
    <span class="md-ellipsis">
      14 Potential Energy Surface Analysis
    </span>
  </a>
  
    <nav class="md-nav" aria-label="14 Potential Energy Surface Analysis">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#geometry-optimization" class="md-nav__link">
    <span class="md-ellipsis">
      14.1 Geometry Optimization
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hessians-and-vibrational-frequencies" class="md-nav__link">
    <span class="md-ellipsis">
      14.2 Hessians
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#constraints" class="md-nav__link">
    <span class="md-ellipsis">
      14.3 Constraints
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nudged-elastic-band-and-zero-temperature-string-methods" class="md-nav__link">
    <span class="md-ellipsis">
      14.4 Nudged Elastic Band (NEB) method
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-15" class="md-nav__link">
    <span class="md-ellipsis">
      15 Electronic Structure Analysis
    </span>
  </a>
  
    <nav class="md-nav" aria-label="15 Electronic Structure Analysis">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#properties" class="md-nav__link">
    <span class="md-ellipsis">
      15.1 Properties
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#esp" class="md-nav__link">
    <span class="md-ellipsis">
      15.2 Electrostatic potentials
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#dplot" class="md-nav__link">
    <span class="md-ellipsis">
      15.3 DPLOT
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-16" class="md-nav__link">
    <span class="md-ellipsis">
      16 Other-Capabilities
    </span>
  </a>
  
    <nav class="md-nav" aria-label="16 Other-Capabilities">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#electron-transfer" class="md-nav__link">
    <span class="md-ellipsis">
      16.1 Electron Transfer
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#python" class="md-nav__link">
    <span class="md-ellipsis">
      16.2 Controlling NWChem with Python
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quantum" class="md-nav__link">
    <span class="md-ellipsis">
      16.3 Quantum
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#vscf" class="md-nav__link">
    <span class="md-ellipsis">
      16.4 Vibrational SCF (VSCF)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ccca_method" class="md-nav__link">
    <span class="md-ellipsis">
      16.5 Correlation consistent Composite Approach (ccCA)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#interface" class="md-nav__link">
    <span class="md-ellipsis">
      16.6 Interfaces to Other Programs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-17" class="md-nav__link">
    <span class="md-ellipsis">
      17 Molecular Mechanics
    </span>
  </a>
  
    <nav class="md-nav" aria-label="17 Molecular Mechanics">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#prepare" class="md-nav__link">
    <span class="md-ellipsis">
      17.1 Prepare
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#md" class="md-nav__link">
    <span class="md-ellipsis">
      17.2 Molecular dynamics
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#analysis" class="md-nav__link">
    <span class="md-ellipsis">
      17.3 Analysis
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#format_of_md_files" class="md-nav__link">
    <span class="md-ellipsis">
      17.4 Format of MD files
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#python" class="md-nav__link">
    <span class="md-ellipsis">
      18 Controlling NWChem with Python
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-19" class="md-nav__link">
    <span class="md-ellipsis">
      19 NWChem Examples
    </span>
  </a>
  
    <nav class="md-nav" aria-label="19 NWChem Examples">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#sample" class="md-nav__link">
    <span class="md-ellipsis">
      19.1 Sample input files
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#geometry-examples" class="md-nav__link">
    <span class="md-ellipsis">
      19.2 Examples of geometries using symmetry
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#benchmarks" class="md-nav__link">
    <span class="md-ellipsis">
      20 Benchmarks performed with NWChem
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#known-bugs" class="md-nav__link">
    <span class="md-ellipsis">
      21 Known Bugs
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#faq" class="md-nav__link">
    <span class="md-ellipsis">
      22 FAQ
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#tutorials" class="md-nav__link">
    <span class="md-ellipsis">
      23 NWChem Tutorials
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#forum" class="md-nav__link">
    <span class="md-ellipsis">
      24 Forum
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-27" class="md-nav__link">
    <span class="md-ellipsis">
      27 Supplementary Information
    </span>
  </a>
  
    <nav class="md-nav" aria-label="27 Supplementary Information">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#armci" class="md-nav__link">
    <span class="md-ellipsis">
      27.2 Choosing the ARMCI Library
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#interfaces-with-external-software" class="md-nav__link">
    <span class="md-ellipsis">
      28 Interfaces with External Software
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#software-supporting-nwchem" class="md-nav__link">
    <span class="md-ellipsis">
      29 Software supporting NWChem
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#containers" class="md-nav__link">
    <span class="md-ellipsis">
      30 Containers
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<div id="print-site-page" class="print-site-enumerate-headings print-site-enumerate-figures">
        <section id="print-site-cover-page">
            <center>
<h1>NWChem User Manual</h1>
<br>
<h1>Release 7.2.0</h1>
<br>
<img src="assets/MS3logo.png" />

<h3>W.R. Wiley Environmental Molecular Sciences Laboratory</h3>
<h3>Pacific Northwest National Laboratory</h3>
<h3>P.O. Box 999, Richland, WA 99352</h3>
<br>

<h2><div id="current_date"></p>
<script>
date = new Date();
year = date.getFullYear();
  month = date.toLocaleString('default', { month: 'long' });
  day = date.getDate();

document.getElementById("current_date").innerHTML = month + " " + day + " " + year;
</script></h2>

<h2>Educational Community License, Version 2.0 (ECL-2.0)</h2>
</center
<br>
<hr>
<br>
<div style="break-after:always"></div>
<hr>
        </section>
        
        <section class="print-page">
            <div id="print-page-toc" data-toc-depth="6">
                <nav role='navigation' class='print-page-toc-nav'>
                <h1 class='print-page-toc-title'>Table of Contents</h1>
                </nav>
            </div>
        </section>
        <section class="print-page" id="index" heading-number="1"><h1 id="index-nwchem-open-source-high-performance-computational-chemistry">NWChem: Open Source High-Performance Computational Chemistry<a class="headerlink" href="#index-nwchem-open-source-high-performance-computational-chemistry" title="Permanent link">&para;</a></h1>
<p>The NWChem software provides computational chemistry tools that can accurately tackle complex scientific problems.
The software has been develop to efficiently use the aggregate resources available on supercomputers or computer clusters.</p>
<p>NWChem can handle:
<img align="right" alt="" width="72" height="175" src="mol1andmol2.webp"></p>
<!--![](mol1andmol2.webp){ align=right}-->

<ul>
<li>Biomolecules, nanostructures, and solid-state</li>
<li>From quantum to classical, and all combinations</li>
<li>Ground and excited-states</li>
<li>Gaussian basis functions or plane-waves</li>
<li>Scaling from one to thousands of processors</li>
<li>Properties and relativistic effects</li>
</ul>
<p><img align="right" alt="" width="103" height="43" src="EMSL_light_60pct.svg"></p>
<!--![](EMSL_light_60pct.svg){ align=right }-->

<p>NWChem is actively developed by a consortium of developers and maintained by<br />
<a href="https://www.emsl.pnnl.gov">The Environmental Molecular Sciences Laboratory (EMSL)</a> located at the Pacific<br />
Northwest National Laboratory (<a href="https://www.pnl.gov">PNNL</a>) in Washington State. Researchers interested<br />
in contributing to NWChem should review
the <a href="#developer">Developers page</a>. The
code is distributed as open-source under the terms of the <a href="https://opensource.org/license/ecl-2-0">Educational
Community License
version 2.0</a> (ECL 2.0).</p>
<p>The NWChem development strategy is focused on providing new and
essential scientific capabilities to its users in the areas of kinetics
and dynamics of chemical transformations, chemistry at interfaces and in
the condensed phase, and enabling innovative and integrated research at
EMSL. At the same time continued development is needed to enable NWChem
to effectively utilize architectures of tens of petaflops and beyond.</p>
<h2 id="index-latest-nwchem-release">Latest NWChem release<a class="headerlink" href="#index-latest-nwchem-release" title="Permanent link">&para;</a></h2>
<p>NWChem version 7.2.3 is the latest release available for download from the link <a href="https://github.com/nwchemgit/nwchem/releases">https://github.com/nwchemgit/nwchem/releases</a>.</p>
<h2 id="index-emsl-arrows">EMSL Arrows<a class="headerlink" href="#index-emsl-arrows" title="Permanent link">&para;</a></h2>
<p><img align="right" alt="EMSLArrows" src="myarrowmovie.webp" /></p>
<!-- img align="right" alt=""  width="200" height="150" src="myarrowmovie.webp"-->
<!--![](myarrowmovie.webp){ align=right }-->
<p>Are you just
learning how to use NWChem and would like to have an easy way to generate input
decks, check your output decks against a large database of calculations,
perform simple thermochemistry calculations, calculate the NMR and IR
spectra of a modest size molecule, or just try out nwchem before
installing it? EMSL Arrows scientific service can help. A web api to
EMSL Arrows is now available for alpha testing.</p>
<p>For more information see
<a href="#emsl_arrows">EMSL Arrows - an easier way to use nwchem</a></p>
<p><span style="background: #82BA57; border: solid 1px #a9a9a9; padding: 8px">
<a href="https://arrows.emsl.pnnl.gov/api/">EMSL Arrows API</a> </span></p>
<h2 id="index-nwchem-documentation"><a href="#home">NWChem Documentation</a><a class="headerlink" href="#index-nwchem-documentation" title="Permanent link">&para;</a></h2>
<h2 id="index-nwchem-citation">NWChem Citation<a class="headerlink" href="#index-nwchem-citation" title="Permanent link">&para;</a></h2>
<p>Please cite the following reference when publishing results obtained
with NWChem:</p>
<p>E. Apr, E. J. Bylaska, W. A. de Jong, N. Govind, K. Kowalski, T. P. Straatsma, M. Valiev, H. J. J. van Dam, Y. Alexeev, J. Anchell, V. Anisimov, F. W. Aquino, R. Atta-Fynn, J. Autschbach, N. P. Bauman, J. C. Becca, D. E. Bernholdt, K. Bhaskaran-Nair, S. Bogatko, P. Borowski, J. Boschen, J. Brabec, A. Bruner, E. Caut, Y. Chen, G. N. Chuev, C. J. Cramer, J. Daily, M. J. O. Deegan, T. H. Dunning Jr., M. Dupuis, K. G. Dyall, G. I. Fann, S. A. Fischer, A. Fonari, H. Frchtl, L. Gagliardi, J. Garza, N. Gawande, S. Ghosh, K. Glaesemann, A. W. Gtz, J. Hammond, V. Helms, E. D. Hermes, K. Hirao, S. Hirata, M. Jacquelin, L. Jensen, B. G. Johnson, H. Jnsson, R. A. Kendall, M. Klemm, R. Kobayashi, V. Konkov, S. Krishnamoorthy, M. Krishnan, Z. Lin, R. D. Lins, R. J. Littlefield, A. J. Logsdail, K. Lopata, W. Ma, A. V. Marenich, J. Martin del Campo, D. Mejia-Rodriguez, J. E. Moore, J. M. Mullin, T. Nakajima, D. R. Nascimento, J. A. Nichols, P. J. Nichols, J. Nieplocha, A. Otero-de-la-Roza, B. Palmer, A. Panyala, T. Pirojsirikul, B. Peng, R. Peverati, J. Pittner, L. Pollack, R. M. Richard, P. Sadayappan, G. C. Schatz, W. A. Shelton, D. W. Silverstein, D. M. A. Smith, T. A. Soares, D. Song, M. Swart, H. L. Taylor, G. S. Thomas, V. Tipparaju, D. G. Truhlar, K. Tsemekhman, T. Van Voorhis, . Vzquez-Mayagoitia, P. Verma, O. Villa, A. Vishnu, K. D. Vogiatzis, D. Wang, J. H. Weare, M. J. Williamson, T. L. Windus, K. Woliski, A. T. Wong, Q. Wu, C. Yang, Q. Yu, M. Zacharias, Z. Zhang, Y. Zhao, and R. J. Harrison,
&ldquo;NWChem: Past, present, and future&rdquo;, <em>The Journal of Chemical Physics</em> <strong>152</strong>, 184102 (2020).
DOI: <a href="http://dx.doi.org/10.1063/5.0004997">10.1063/5.0004997</a>  </p></section>
                    <section class='print-page md-section' id='section-3' heading-number='3'>
                        <h1>How to get NWChem<a class='headerlink' href='#section-3' title='Permanent link'></a>
                        </h1>
                    <section class="print-page" id="download" heading-number="3.1"><h1 id="download-how-to-download-and-install-nwchem">How to download and install NWChem<a class="headerlink" href="#download-how-to-download-and-install-nwchem" title="Permanent link">&para;</a></h1>
<h2 id="download-source-download">Source Download<a class="headerlink" href="#download-source-download" title="Permanent link">&para;</a></h2>
<p>The NWChem source is available for download from
<a href="https://github.com/nwchemgit/nwchem/releases">https://github.com/nwchemgit/nwchem/releases</a></p>
<p>Compilation instructions can be found at this <a href="#compiling-nwchem">link</a></p>
<h2 id="download-nwchem-availability-in-linux-distributions">NWChem availability in Linux distributions<a class="headerlink" href="#download-nwchem-availability-in-linux-distributions" title="Permanent link">&para;</a></h2>
<p>Debian: <a href="https://packages.debian.org/search?keywords=nwchem">https://packages.debian.org/search?keywords=nwchem</a></p>
<p>Ubuntu: <a href="https://launchpad.net/ubuntu/+source/nwchem">https://launchpad.net/ubuntu/+source/nwchem</a></p>
<p>Fedora and EPEL: <a href="https://admin.fedoraproject.org/updates/search/nwchem">https://admin.fedoraproject.org/updates/search/nwchem</a></p>
<p>Good search engine for NWChem Linux packages: <a href="https://pkgs.org/search/?q=nwchem">https://pkgs.org/search/?q=nwchem</a></p>
<h3 id="download-example-of-nwchem-installation-on-debianubuntu">Example of NWChem installation on Debian/Ubuntu<a class="headerlink" href="#download-example-of-nwchem-installation-on-debianubuntu" title="Permanent link">&para;</a></h3>
<pre><code>sudo apt-get  install nwchem
</code></pre>
<p>Parallel runs (using more than one process) can be performed with the command</p>
<pre><code>/usr/bin/mpirun.openmpi -np 2 nwchem n2.nw
</code></pre>
<h3 id="download-example-of-nwchem-rpm-installation-under-redhatcentos-7-x86_64">Example of NWChem RPM installation under Redhat/Centos 7 x86_64<a class="headerlink" href="#download-example-of-nwchem-rpm-installation-under-redhatcentos-7-x86_64" title="Permanent link">&para;</a></h3>
<pre><code>sudo yum -y install https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
sudo yum update
sudo yum install nwchem nwchem-openmpi Lmod
</code></pre>
<p>In order to run NWChem, you must type</p>
<pre><code>module load mpi/openmpi-x86_64
</code></pre>
<p>The name of the NWChem executable is</p>
<pre><code>nwchem_openmpi
</code></pre>
<p>Serial runs, using a single process, (on a input file named <code>n2.nw</code> in the following example) can be performed with the command</p>
<pre><code>nwchem_openmpi n2.nw
</code></pre>
<p>Parallel runs (using more than one process) can be performed with the command</p>
<pre><code>mpirun -np 2 nwchem_openmpi n2.nw
</code></pre>
<h2 id="download-nwchem-availability-on-macos">NWChem availability on macOS<a class="headerlink" href="#download-nwchem-availability-on-macos" title="Permanent link">&para;</a></h2>
<p>NWChem can be installed from <a href="https://brew.sh/">Homebrew</a>, by executing the following commands  </p>
<pre><code>bin/bash -c &quot;$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)&quot;

brew install nwchem
</code></pre>
<h2 id="download-nwchem-installation-on-conda">NWChem installation on Conda<a class="headerlink" href="#download-nwchem-installation-on-conda" title="Permanent link">&para;</a></h2>
<p>NWChem can be installed  on Linux or MacOS from the <a href="https://conda-forge.org/">conda-forge</a> channel of <a href="https://docs.conda.io">Conda</a> with the commands</p>
<pre><code>conda install -c conda-forge micromamba
micromamba install -c conda-forge nwchem
</code></pre>
<p>More details at <a href="https://github.com/conda-forge/nwchem-feedstock">https://github.com/conda-forge/nwchem-feedstock</a></p></section><section class="print-page" id="compiling-nwchem" heading-number="3.2"><h1 id="compiling-nwchem-compiling-nwchem-from-source">Compiling NWChem from source<a class="headerlink" href="#compiling-nwchem-compiling-nwchem-from-source" title="Permanent link">&para;</a></h1>
<p>On this page, a step-by-step description of the build process and
necessary and optional environment variables is outlined. In addition,
based on the experiences of developers and users how-to&rsquo;s for various
platforms have been created. These how-to&rsquo;s will be updated with
additional platforms and better environment variables over time.  </p>
<p>Download of the NWChem source is a step needed before compilation. Details for downloading
as well as instructions for installing pre-compiled version of NWChem are available at the
<a href="#download">Download page</a>.</p>
<h2 id="compiling-nwchem-setting-up-the-proper-environment-variables">Setting up the proper environment variables<a class="headerlink" href="#compiling-nwchem-setting-up-the-proper-environment-variables" title="Permanent link">&para;</a></h2>
<ul>
<li><code>$NWCHEM_TOP</code> defines the top directory of the NWChem source tree,
    e.g.</li>
</ul>
<p>When dealing with source from a <strong><em>NWChem release</em></strong> (7.2.3 in this example)</p>
<pre><code>export NWCHEM_TOP=&lt;your path&gt;/nwchem-7.2.3
</code></pre>
<ul>
<li><code>$NWCHEM_TARGET</code> defines your target platform, e.g.</li>
</ul>
<pre><code>export NWCHEM_TARGET=LINUX64
</code></pre>
<p>The following platforms are available:</p>
<table>
<thead>
<tr>
<th>NWCHEM_TARGET</th>
<th style="text-align: left;">Platform</th>
<th>OS</th>
<th>Compilers</th>
</tr>
</thead>
<tbody>
<tr>
<td>LINUX</td>
<td style="text-align: left;">x86</td>
<td>Linux</td>
<td>GNU, Intel</td>
</tr>
<tr>
<td></td>
<td style="text-align: left;">ppc</td>
<td>Linux</td>
<td>GNU, IBM</td>
</tr>
<tr>
<td></td>
<td style="text-align: left;">arm</td>
<td>Linux</td>
<td>GNU</td>
</tr>
<tr>
<td>LINUX64</td>
<td style="text-align: left;">x86_64</td>
<td>Linux</td>
<td>GNU, Intel,  flang</td>
</tr>
<tr>
<td></td>
<td style="text-align: left;">ppc64le</td>
<td>Linux</td>
<td>GNU, IBM</td>
</tr>
<tr>
<td></td>
<td style="text-align: left;">aarch64</td>
<td>Linux</td>
<td>GNU, flang</td>
</tr>
<tr>
<td>MACX</td>
<td style="text-align: left;">x86</td>
<td>Darwin</td>
<td>GNU</td>
</tr>
<tr>
<td>MACX64</td>
<td style="text-align: left;">x86_64</td>
<td>Darwin</td>
<td>GNU</td>
</tr>
<tr>
<td></td>
<td style="text-align: left;">arm64</td>
<td>Darwin</td>
<td>GNU</td>
</tr>
</tbody>
</table>
<p><br></p>
<ul>
<li><code>$ARMCI_NETWORK</code> must be defined in order to achieve best
    performance on high performance networks, e.g.</li>
</ul>
<pre><code>export ARMCI_NETWORK=MPI-PR
</code></pre>
<p>For a single processor system, this environment variable does not have
to be defined.
Supported combination of ARMCI_NETWORK and NWCHEM_TARGET variables:  </p>
<table>
<thead>
<tr>
<th>ARMCI_NETWORK</th>
<th>NWCHEM_TARGET</th>
<th>Network</th>
<th>Protocol</th>
</tr>
</thead>
<tbody>
<tr>
<td>OPENIB</td>
<td>LINUX, LINUX64</td>
<td>Mellanox InfiniBand</td>
<td>Verbs</td>
</tr>
<tr>
<td>MPI-PR</td>
<td>LINUX64</td>
<td>Any network</td>
<td>MPI</td>
</tr>
<tr>
<td>MPI-MT<br />MPI-SPAWN</td>
<td>LINUX64</td>
<td>MPI supporting<br />multi-threading multiple</td>
<td>MPI-2</td>
</tr>
<tr>
<td>MPI-TS<br /> MPI-PT</td>
<td>any</td>
<td>any network with MPI</td>
<td>MPI</td>
</tr>
<tr>
<td>BGMLMPI</td>
<td>BGL</td>
<td>IBM Blue Gene/L</td>
<td>BGLMPI</td>
</tr>
<tr>
<td>DC        MFMPI</td>
<td>BGP</td>
<td>IBM Blue Gene/P</td>
<td>DCMF,MPI</td>
</tr>
</tbody>
</table>
<p><br></p>
<p>Please see <a href="#armci">Choosing the ARMCI Library</a> for
additional information on choosing the right network
options.</p>
<h3 id="compiling-nwchem-mpi-variables">MPI variables<a class="headerlink" href="#compiling-nwchem-mpi-variables" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Variable</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>USE_MPI</code></td>
<td>Set to &ldquo;y&rdquo; to indicate that NWChem should be compiled with MPI</td>
</tr>
<tr>
<td><code>USE_MPIF</code></td>
<td>Set to &ldquo;y&rdquo; for the NWPW module to use fortran-bindings of MPI. <br /> (Generally set when USE_MPI is set)</td>
</tr>
<tr>
<td><code>USE_MPIF4</code></td>
<td>Set to &ldquo;y&rdquo; for the NWPW module to use Integer*4 fortran-bindings of MPI. <br /> (Generally set when USE_MPI is set on most platforms)</td>
</tr>
<tr>
<td><code>LIBMPI</code> (deprecated)</td>
<td>Name of the MPI library that should be linked with -l</td>
</tr>
<tr>
<td><code>MPI_LIB</code> (deprecated)</td>
<td>Directory where the MPI library resides</td>
</tr>
<tr>
<td><code>MPI_INCLUDE</code> (deprecated)</td>
<td>Directory where the MPI include files reside</td>
</tr>
</tbody>
</table>
<p><br>  </p>
<h4 id="compiling-nwchem-automatic-detection-of-mpi-variables-with-mpif90"><span style="color:red;">Automatic detection of MPI variables with mpif90</span><a class="headerlink" href="#compiling-nwchem-automatic-detection-of-mpi-variables-with-mpif90" title="Permanent link">&para;</a></h4>
<p><strong><em>New in NWChem 6.6</em></strong>: If the
location of the <code>mpif90</code> command is part of your <code>PATH</code> env. variable,
NWChem will figure out the values of <code>LIBMPI</code>, <code>MPI_LIB</code> and <code>MPI_INCLUDE</code>
(if they are not set). Therefore, we do <strong>NOT</strong> recommend to set <code>LIBMPI</code>,
<code>MPI_LIB</code> and <code>MPI_INCLUDE</code> and add the location of <code>mpif90</code> to the <code>PATH</code>
variable, instead. Therefore, the next section can be considered obsolete.</p>
<h4 id="compiling-nwchem-obsolete-how-to-set-the-mpi-variables">OBSOLETE: How to set the MPI variables<a class="headerlink" href="#compiling-nwchem-obsolete-how-to-set-the-mpi-variables" title="Permanent link">&para;</a></h4>
<p>The information of this section should not be used because of the automatic detection of MPI variables described in the previous 
<a href="#compiling-nwchem-automatic-detection-of-mpi-variables-with-mpif90">section</a>.  </p>
<p><del>The output of the command</del></p>
<pre><code><del>mpif90 -show</del></code></pre>

<p><del>can be used to extract the values of LIBMPI, MPI_LIB and MPI_INCLUDE</del></p>
<p><del>E.g. for MPICH2, this might look like:</del></p>
<pre><code><del>
$ mpif90 -show
f95 -I/usr/local/mpich2.141p1/include -I/usr/local/mpich2.141p1/include -L/usr/local/mpich2.141p1/lib \
-lmpichf90 -lmpichf90 -lmpich -lopa -lmpl -lrt -lpthread
</del></code></pre>
<p><del>The corresponding environment variables are</del></p>
<pre><code><del>
  % export USE_MPI=y
  % export LIBMPI="-lmpich -lopa -lmpl -lpthread -lmpichf90 -lfmpich -lmpich"
  % export MPI_LIB=/usr/local/mpich2.141p1/lib 
  % export MPI_INCLUDE='/usr/local/mpich2.141p1/include
</del></code></pre>

<h4 id="compiling-nwchem-how-to-start-nwchem">How to start NWChem<a class="headerlink" href="#compiling-nwchem-how-to-start-nwchem" title="Permanent link">&para;</a></h4>
<p>When MPI is used, the appropriate MPI run command should be used to
start an NWChem calculation, e.g.</p>
<pre><code>  % mpirun -np 8 $NWCHEM_TOP/bin/$NWCHEM_TARGET/nwchem h2o.nw
</code></pre>
<h3 id="compiling-nwchem-nwchem_modules">NWCHEM_MODULES<a class="headerlink" href="#compiling-nwchem-nwchem_modules" title="Permanent link">&para;</a></h3>
<ul>
<li><code>$NWCHEM_MODULES</code> defines the modules to be compiled, e.g.</li>
</ul>
<pre><code>export NWCHEM_MODULES=&quot;all python&quot;
</code></pre>
<p>The following modules are available:</p>
<table>
<thead>
<tr>
<th>Module</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>all</td>
<td>Everything useful</td>
</tr>
<tr>
<td>all python</td>
<td>Everything useful plus python</td>
</tr>
<tr>
<td>qm</td>
<td>All quantum mechanics modules</td>
</tr>
<tr>
<td>md</td>
<td>MD only build</td>
</tr>
</tbody>
</table>
<p><br>
Note that additional environment variables need to be defined to specify
the location of the Python libraries, when the python module is
compiled. See the optional environmental variables section for
specifics.</p>
<h2 id="compiling-nwchem-adding-optional-environmental-variables">Adding optional environmental variables<a class="headerlink" href="#compiling-nwchem-adding-optional-environmental-variables" title="Permanent link">&para;</a></h2>
<p><strong>USE_NOFSCHECK</strong> can be set to avoid NWChem creating files for each
process when testing the size of the scratch directory (a.k.a. creation
of junk files), e.g.</p>
<pre><code>export USE_NOFSCHECK=TRUE
</code></pre>
<p><strong>USE_NOIO</strong> can be set to avoid NWChem 6.5 doing I/O for the ddscf,
mp2 and ccsd modules (it automatically sets <code>USE_NOFSCHECK</code>, too). It is
strongly recommended on large clusters or supercomputers or any computer
lacking any fast and large local filesystem.</p>
<pre><code>export USE_NOIO=TRUE
</code></pre>
<p><del><strong>LIB_DEFINES</strong> can be set to pass additional defines to the C
preprocessor (for both Fortran and C), e.g.</del></p>
<pre><code>export LIB_DEFINES=-DDFLT_TOT_MEM=16777216
</code></pre>
<p><del>Note: <code>-DDFLT_TOT_MEM</code> sets the default dynamic memory available for
NWChem to run, where the units are in doubles.</del> <br />
However, it is recommended that, instead of manually
defining this environment variable, the <a href="https://github.com/nwchemgit/nwchem/blob/master/contrib/getmem.nwchem">getmem.nwchem</a> script to be executed as described in the related <a href="#compiling-nwchem-setting-the-default-memory-values">section</a></p>
<p><strong>MRCC_METHODS</strong> can be set to request the multireference coupled
cluster capability to be included in the code, e.g.</p>
<pre><code>export MRCC_METHODS=TRUE
</code></pre>
<p><strong>CCSDTQ</strong> can be set to request the CCSDTQ method and its derivatives 
to be included in the code, e.g.</p>
<pre><code>export CCSDTQ=TRUE
</code></pre>
<h3 id="compiling-nwchem-setting-python-environment-variables">Setting Python environment variables<a class="headerlink" href="#compiling-nwchem-setting-python-environment-variables" title="Permanent link">&para;</a></h3>
<p>Python programs may be embedded into the NWChem input and used to
control the execution of NWChem. To build with Python, Python needs to
be available on your machine. The software can be download from
<a href="https://www.python.org">https://www.python.org</a> . Follow the Python instructions for
installation and testing. NWChem has been tested with Python versions
up to 3.11</p>
<p>The following environment variables need to be set when compiling with
Python, together with having the location of your installed python binary part of
the <code>PATH</code> environment variable:</p>
<pre><code>export PYTHONVERSION=3.8
</code></pre>
<p>Note that the third number in the version should not be kept: 3.8.1
should be set as 3.8</p>
<p>You will also need to set PYTHONPATH to include any modules that you are
using in your input. Examples of Python within NWChem are in the
<code>$NWCHEM_TOP/QA/tests/pyqa3</code> and <code>$NWCHEM_TOP/contrib/python</code> directories.</p>
<h3 id="compiling-nwchem-optimized-math-libraries">Optimized math libraries<a class="headerlink" href="#compiling-nwchem-optimized-math-libraries" title="Permanent link">&para;</a></h3>
<p>By default NWChem uses its own basic linear algebra subroutines (BLAS).
To include faster BLAS routines, the environment variable BLASOPT needs
to be set before building the code. For example, with
OpenBLAS</p>
<pre><code>export BLASOPT=&quot;-lopenblas&quot;
</code></pre>
<p>Good choices of optimized BLAS libraries on x86 (e.g. LINUX and LINUX64)
hardware include:  </p>
<table>
<thead>
<tr>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td>BLIS</td>
<td><a href="https://github.com/flame/blis">https://github.com/flame/blis</a></td>
</tr>
<tr>
<td>OpenBLAS</td>
<td><a href="https://github.com/xianyi/OpenBLAS">https://github.com/xianyi/OpenBLAS</a></td>
</tr>
<tr>
<td>Intel MKL</td>
<td><a href="https://www.intel.com/content/www/us/en/developer/tools/oneapi/onemkl.html">https://www.intel.com/content/www/us/en/developer/tools/oneapi/onemkl.html</a></td>
</tr>
<tr>
<td>Cray LibSci</td>
<td>Available only on Cray hardware, it is automatically linked when compiling on Cray computers.</td>
</tr>
<tr>
<td>IBM ESSL</td>
<td>Available only on IBM hardware <a href="https://www.ibm.com/docs/en/essl/6.3">https://www.ibm.com/docs/en/essl/6.3</a></td>
</tr>
</tbody>
</table>
<p><br>  </p>
<p><em><strong>New since release 7.0.0</strong></em> (after commit <a href="https://github.com/nwchemgit/nwchem/commit/6b0a971207e776f43dec81974014e86caf8cee61#diff-1750a4dcc9a0a9b1773d275e96c46a1e">6b0a971</a>):  If <code>BLASOPT</code> is defined, the <code>LAPACK_LIB</code> environment variable must be set up, too.  <code>LAPACK_LIB</code> must provide the location of the library containing the LAPACK routines. For example, OpenBLAS provides the full suite of LAPACK routines, therefore, in this case, <code>LAPACK_LIB</code> can be set to the same value as BLASOPT  </p>
<pre><code>export BLASOPT=-lopenblas  
export LAPACK_LIB=-lopenblas  
</code></pre>
<p>NWChem can also take advantage of the <a href="http://netlib.org/scalapack/">ScaLAPACK
library</a> if it is installed on your
system. The following environment variables need to be set:</p>
<pre><code>export USE_SCALAPACK=y

export SCALAPACK=&quot;location of Scalapack and BLACS library&quot;
</code></pre>
<h3 id="compiling-nwchem-how-to-deal-with-integer-size-of-linear-algebra-libraries">How to deal with integer size of Linear Algebra libraries<a class="headerlink" href="#compiling-nwchem-how-to-deal-with-integer-size-of-linear-algebra-libraries" title="Permanent link">&para;</a></h3>
<p>In the case of 64-bit platforms, most vendors optimized BLAS
libraries cannot be used. This is due to the fact that while NWChem uses
64-bit integers (i.e. integer*8) on 64-bit platforms, most of the
vendors optimized BLAS libraries used 32-bit integers. The same holds
for the ScaLAPACK libraries, which internally use 32-bit integers.<br />
The BLAS_SIZE environment variable is used at compile time to set the size of integer arguments in BLAS calls.  </p>
<table>
<thead>
<tr>
<th style="text-align: center;">BLAS_SIZE</th>
<th>size of integer arguments in BLAS routines</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: center;">4</td>
<td>32-bit (most common default)</td>
</tr>
<tr>
<td style="text-align: center;">8</td>
<td>64-bit</td>
</tr>
</tbody>
</table>
<p>A method is available to link against the libraries mentioned above,
using the following procedure:</p>
<pre><code>   cd $NWCHEM_TOP/src
   make clean
   make 64_to_32
   make USE_64TO32=y  BLAS_SIZE=4 BLASOPT=&quot; optimized BLAS&quot; SCALAPACK=&quot;location of Scalapack and BLACS library&quot;
</code></pre>
<p>E.g., for IBM64 this looks like</p>
<pre><code>  % make USE_64TO32=y BLAS_SIZE=4 BLASOPT=&quot;-lessl -lmass&quot;
</code></pre>
<p>Notes:</p>
<ul>
<li>OpenBLAS can be installed with 64bit integers. This
    is accomplished by compiling the OpenBLAS library after having
    edited the OpenBLAS Makefile.rule file and un-commenting the line
    containing the INTERFACE64 definition. In other words, the line</li>
</ul>
<pre><code>          #INTERFACE64 = 1
</code></pre>
<p>needs to be changed to</p>
<pre><code>          INTERFACE64 = 1
</code></pre>
<ul>
<li>AOCL and MKL can support 64-bit integers if the appropriate library
    is chosen. For MKL, one can choose the ILP64 Version of Intel MKL
    and the correct recipe can be extracted from the website
    <a href="https://www.intel.com/content/www/us/en/developer/tools/oneapi/onemkl-link-line-advisor.html">https://www.intel.com/content/www/us/en/developer/tools/oneapi/onemkl-link-line-advisor.html</a>.</li>
</ul>
<h3 id="compiling-nwchem-automated-build-of-openblasscalapack"><span style="color:#FF0000">Automated build of OpenBLAS/ScaLAPACK</span><a class="headerlink" href="#compiling-nwchem-automated-build-of-openblasscalapack" title="Permanent link">&para;</a></h3>
<p><strong><em><span style="color:#FF0000">New in NWChem 7.0.2</span></em></strong>:   </p>
<ol>
<li>The environment variable <code>BUILD_OPENBLAS</code> can be used to automatically build the OpenBLAS library during a NWChem compilation (either using <code>BLAS_SIZE=8</code> or <code>BLAS_SIZE=4</code>)  </li>
<li>The environment variable <code>BUILD_SCALAPACK</code> can be used to automatically build the ScaLapack library during a NWChem compilation (either using <code>SCALAPACK_SIZE=8</code> or <code>SCALAPACK_SIZE=4</code>)</li>
</ol>
<p>The following settings are strongly recommended over setting variables pointing to existing installations:</p>
<pre><code>BUILD_OPENBLAS=1
BUILD_SCALAPACK=1
BLAS_SIZE=8
SCALAPACK_SIZE=8
</code></pre>
<h3 id="compiling-nwchem-linking-in-nbo">Linking in NBO<a class="headerlink" href="#compiling-nwchem-linking-in-nbo" title="Permanent link">&para;</a></h3>
<p>The 5.0 (obsolete) version of NBO provides a utility to generate source
code that can be linked into computational chemistry packages such as
NWChem. To utilize this functionality, follow the instructions in the
NBO 5 package to generate an nwnbo.f file. Linking NBO into NWChem can
be done using the following procedure:</p>
<pre><code>  % cd $NWCHEM_TOP/src 
  % cp nwnbo.f $NWCHEM_TOP/src/nbo/.  
  % make nwchem_config NWCHEM_MODULES=&quot;all nbo&quot; 
  % make
</code></pre>
<p>One can now use &ldquo;task nbo&rdquo; and incorporate NBO input into the NWChem
input file directly:</p>
<pre><code> nbo  
   $NBO NRT $END  
   ... 
 end  

 task nbo
</code></pre>
<h2 id="compiling-nwchem-building-the-nwchem-binary">Building the NWChem binary<a class="headerlink" href="#compiling-nwchem-building-the-nwchem-binary" title="Permanent link">&para;</a></h2>
<p>Once all required and optional environment variables have been set,
NWChem can be compiled:</p>
<pre><code>  % cd $NWCHEM_TOP/src  

  % make nwchem_config 

  % make &gt;&amp; make.log
</code></pre>
<p>The make above will use the standard compilers available on your system.
To use compilers different from the default one can either set
environment variables:</p>
<pre><code>  % export FC=&lt;fortran compiler&gt;  
  % export CC=&lt;c compiler&gt;
</code></pre>
<p>Or one can supply the compiler options to the make command (<em>recommended</em> option), e.g:</p>
<pre><code>  % make FC=ifort 
</code></pre>
<p>For example, on Linux FC could be set either equal to ifort, gfortran or pgf90</p>
<p><strong>Nota bene:</strong> NWChem does NOT support usage of the full path in FC and
CC variables. Please provide filenames only as in the examples above!</p>
<p>Note 1: If in a Linux environment, FC is set equal to anything other
than the tested compilers, there is no guarantee of a successful
installation, since the makefile structure has not been tested to
process other settings. In other words, please avoid make FC=&rdquo;ifort -O3 -xhost&rdquo;
and stick to make FC=&rdquo;ifort&rdquo;, instead</p>
<p>Note 2: It&rsquo;s better to avoid redefining CC, since a) NWChem does not
have C source that is a computational bottleneck and b) we typically
test just the default C compiler. In other words, the recommendation is
to compile with make FC=ifort</p>
<p>Note 3: It&rsquo;s better to avoid modifying the values of the FOPTIMIZE and
COPTIMIZE variables. The reason is that the default values for FOPTIMIZE
and COPTIMIZE have been tested by the NWChem developers (using the
internal QA suites, among others), while any modification might produce
incorrect results.</p>
<h3 id="compiling-nwchem-setting-the-default-memory-values">Setting the default memory values<a class="headerlink" href="#compiling-nwchem-setting-the-default-memory-values" title="Permanent link">&para;</a></h3>
<p>It is strongly recommended to use, after a successful compilation, the <a href="https://github.com/nwchemgit/nwchem/blob/master/contrib/getmem.nwchem">getmem.nwchem</a> script in the
<code>$NWCHEM_TOP/contrib</code> directory.  The script will choose the default <a href="#memory">memory</a> settings based on the available physical memory, recompile the appropriate files and relink. Here is an example of its usage:  </p>
<pre><code>cd $NWCHEM_TOP/src
../contrib/getmem.nwchem
</code></pre>
<p>If non default compiler are used, the <code>getmem.nwchem</code> script must be called, using bash shell, by first specifying the compiler environment variable. The example below  uses ifort as Fortran compiler</p>
<pre><code>cd $NWCHEM_TOP/src
FC=ifort ../contrib/getmem.nwchem
</code></pre>
<h2 id="compiling-nwchem-how-to-linux-platforms">How-to: Linux platforms<a class="headerlink" href="#compiling-nwchem-how-to-linux-platforms" title="Permanent link">&para;</a></h2>
<ul>
<li><strong>Common environmental variables for building in serial or in
    parallel with MPI</strong></li>
</ul>
<pre><code> % export NWCHEM_TOP=&lt;your path&gt;/nwchem  
 % export NWCHEM_TARGET=LINUX64  
 % export NWCHEM_MODULES =all
</code></pre>
<ul>
<li><strong>Common environmental variables for building with MPI</strong></li>
</ul>
<p>The following environment variables need to be set when NWChem is
compiled with
MPI:</p>
<pre><code>% export USE_MPI=y
% export USE_MPIF=y
% export USE_MPIF4=y

</code></pre>
<p><strong>New in NWChem 6.6:</strong> If the location of the mpif90 command is part of
your PATH env. variable, NWChem will figure out the values of LIBMPI,
MPI_LIB and MPI_INCLUDE (if they are not set). Therefore, we recommend
not to set LIBMPI, MPI_LIB and MPI_INCLUDE and add the location of
mpif90 to the PATH variable, instead.</p>
<ul>
<li><strong>Compiling the code once all variables are set</strong></li>
</ul>
<pre><code>% cd $NWCHEM_TOP/src
% make nwchem_config
% make FC=gfortran &gt;&amp; make.log
</code></pre>
<h3 id="compiling-nwchem-nwchem-68-on-centos-71fedora-27">NWChem 6.8 on Centos 7.1/Fedora 27<a class="headerlink" href="#compiling-nwchem-nwchem-68-on-centos-71fedora-27" title="Permanent link">&para;</a></h3>
<p>Once you have added the <a href="https://docs.fedoraproject.org/en-US/epel/">EPEL
repository</a> to your Centos/Fedora/RedHat
installation, you can have a more efficient NWChem build. </p>
<pre><code>sudo rpm -Uvh http://download.fedoraproject.org/pub/epel/7/x86_64/Packages/e/epel-release-7-11.noarch.rpm
</code></pre>
<ul>
<li>Packages
required:</li>
</ul>
<pre><code>python-devel gcc-gfortran openblas-devel openblas-serial64 openmpi-devel scalapack-openmpi-devel \
elpa-openmpi-devel tcsh openssh-clients which tar bzip2
</code></pre>
<ul>
<li>Settings</li>
</ul>
<pre><code>export USE_MPI=y
export NWCHEM_TARGET=LINUX64
export USE_PYTHONCONFIG=y
export PYTHONVERSION=2.7
export PYTHONHOME=/usr
export USE_64TO32=y
export BLAS_SIZE=4
export BLASOPT=&quot;-lopenblas -lpthread -lrt&quot;
export LAPACK_LIB=$BLASOPT
export SCALAPACK_SIZE=4
export SCALAPACK=&quot;-L/usr/lib64/openmpi/lib -lscalapack&quot;
export ELPA=&quot;-I/usr/lib64/gfortran/modules/openmpi -L/usr/lib64/openmpi/lib -lelpa&quot;
export LD_LIBRARY_PATH=/usr/lib64/openmpi/lib/:$LD_LIBRARY_PATH
export PATH=/usr/lib64/openmpi/bin/:$PATH
</code></pre>
<ul>
<li>Compilation steps</li>
</ul>
<pre><code>cd $NWCHEM_TOP/src  
make nwchem_config NWCHEM_MODULES=&quot;all python&quot;  
make 64_to_32  
make
</code></pre>
<h2 id="compiling-nwchem-how-to-mac-platforms">How to: Mac platforms<a class="headerlink" href="#compiling-nwchem-how-to-mac-platforms" title="Permanent link">&para;</a></h2>
<h3 id="compiling-nwchem-compilation-of-nwchem-66-on-mac-os-x-1010-yosemite-x86_64">Compilation of NWChem 6.6 on Mac OS X 10.10 (Yosemite) x86_64<a class="headerlink" href="#compiling-nwchem-compilation-of-nwchem-66-on-mac-os-x-1010-yosemite-x86_64" title="Permanent link">&para;</a></h3>
<h4 id="compiling-nwchem-method-1-using-gfortran-and-openmpi-from-brew">Method #1: using gfortran and openmpi from brew<a class="headerlink" href="#compiling-nwchem-method-1-using-gfortran-and-openmpi-from-brew" title="Permanent link">&para;</a></h4>
<ul>
<li>Download and unpack latest NWChem tarball to the directory of your
    choosing, say /Users/johndoe/nwchem</li>
<li>Install Homebrew as described at <a href="http://brew.sh">http://brew.sh</a> (more details at
    <a href="https://docs.brew.sh/Installation.html">https://docs.brew.sh/Installation.html</a>)</li>
</ul>
<!-- end list -->

<pre><code> /usr/bin/ruby -e &quot;$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)&quot; 
</code></pre>
<ul>
<li>Use Homebrew to install open-mpi</li>
</ul>
<pre><code>brew install open-mpi
</code></pre>
<ul>
<li>As usual, set the env. variables</li>
</ul>
<pre><code>export USE_MPI=y  
export NWCHEM_TARGET=MACX64  
export NWCHEM_TOP=/Users/johndoe/nwchem  
export USE_INTERNALBLAS=y
</code></pre>
<ul>
<li><strong>Important</strong>: set the following env. variable (GA will not compile
    otherwise)</li>
</ul>
<pre><code>export CFLAGS_FORGA &quot;-DMPICH_NO_ATTR_TYPE_TAGS&quot;
</code></pre>
<ul>
<li>Go to your source directory, configure, and compile</li>
</ul>
<pre><code> cd /Users/johndoe/nwchem/src  
 make nwchem_config`  
 make
</code></pre>
<p><strong>WARNING:</strong> Please do not use the Mac OS X default BLAS and LAPACK
libraries available (or brew&rsquo;s veclibfort), since they are causing
NWChem to produce erroneous results</p>
<h4 id="compiling-nwchem-method-2-using-intel-compilers-and-mkl">Method #2: using Intel compilers and MKL<a class="headerlink" href="#compiling-nwchem-method-2-using-intel-compilers-and-mkl" title="Permanent link">&para;</a></h4>
<p>The Intel compilers and MKL work just fine on Mac with the following
options:</p>
<pre><code>NWCHEM_TARGET=MACX64
CC=icc
FC=ifort
BLASOPT=&quot;-mkl -openmp&quot;
USE_OPENMP=T
</code></pre>
<p>MPICH and ARMCI-MPI work reliably on Mac. See <a href="#armci">Choosing the ARMCI
Library</a> for details on ARMCI-MPI</p>
<h2 id="compiling-nwchem-how-to-cray-platforms">How-to: Cray platforms<a class="headerlink" href="#compiling-nwchem-how-to-cray-platforms" title="Permanent link">&para;</a></h2>
<p>Common environmental variables for building and running on the Cray XT, XE, XC and XK:</p>
<pre><code>  % export NWCHEM_TOP=&lt;your path&gt;/nwchem  
  % export NWCHEM_TARGET=LINUX64  
  % export NWCHEM_MODULES=all  
  % export USE_MPI=y
  % export USE_MPIF=y 
  % export USE_MPIF4=y 
  % export USE_SCALAPACK=y 
  % export USE_64TO32=y  
  % export LIBMPI=&quot; &quot;
</code></pre>
<ul>
<li><strong>Compiling the code on Cray once all variables (described below)
    are set</strong></li>
</ul>
<pre><code>  % cd $NWCHEM_TOP/src
  % make nwchem_config
  % make 64_to_32
  % make FC=ftn &gt;&amp; make.log
</code></pre>
<p>The step <code>make 64_to_32</code> is required only if either SCALAPACK_SIZE or
BLAS_SIZE are set equal to 4.</p>
<h3 id="compiling-nwchem-method-2-armci_networkmpi-pr">Method #2: ARMCI_NETWORK=MPI-PR<a class="headerlink" href="#compiling-nwchem-method-2-armci_networkmpi-pr" title="Permanent link">&para;</a></h3>
<p>This is a <strong><span style="color:#FF0000">new option available in NWChem
6.6.</span></strong></p>
<p>Set the environmental variables for compilation:</p>
<pre><code>  % export ARMCI_NETWORK=MPI-PR
</code></pre>
<h3 id="compiling-nwchem-example-olcf-titan">Example: OLCF Titan<a class="headerlink" href="#compiling-nwchem-example-olcf-titan" title="Permanent link">&para;</a></h3>
<p>These are variables used for compilation on the <a href="https://www.olcf.ornl.gov/olcf-resources/compute-systems/titan/">OLCF Titan, a Cray
XK7</a> We
assume use of Portland Group compilers programming environment (<code>module
load PrgEnv-pgi</code>)</p>
<pre><code>NWCHEM_TARGET=LINUX64  
ARMCI_NETWORK=MPI-PR  
USE_64TO32=y  
USE_MPI=y  
BLAS_SIZE=4 
LAPACK_SIZE=4  
SCALAPACK_SIZE=4  
SCALAPACK=-lsci_pgi_mp  
BLASOPT=-lsci_pgi_mp
</code></pre>
<p>To enable the GPU part, set</p>
<pre><code>TCE_CUDA=y
</code></pre>
<p>and load the cudatoolkit module</p>
<pre><code>module load cudatoolkit
</code></pre>
<h3 id="compiling-nwchem-aries-eg-xc30xc40">Aries, e.g. XC30/XC40<a class="headerlink" href="#compiling-nwchem-aries-eg-xc30xc40" title="Permanent link">&para;</a></h3>
<h4 id="compiling-nwchem-method-1-armci_networkmpi-pr">Method #1: ARMCI_NETWORK=MPI-PR<a class="headerlink" href="#compiling-nwchem-method-1-armci_networkmpi-pr" title="Permanent link">&para;</a></h4>
<p>This is a <strong><span style="color:#FF0000">new option available in NWChem
6.6.</span></strong></p>
<p>Set the environmental variables for compilation:</p>
<pre><code>% export ARMCI_NETWORK=MPI-PR
</code></pre>
<h4 id="compiling-nwchem-example-hpe-cray">Example: HPE Cray<a class="headerlink" href="#compiling-nwchem-example-hpe-cray" title="Permanent link">&para;</a></h4>
<p>These are variables used for compilation a Cray HPE system
 using Intel compilers (i.e. after issuing the
commands <code>module swap PrgEnv-gnu
PrgEnv-intel</code>).</p>
<pre><code>export NWCHEM_TARGET=LINUX64  
export USE_MPI=y  
export NWCHEM_TARGET=LINUX64  
export ARMCI_NETWORK=MPI-PR  
export USE_MPI=y  
export USE_SCALAPACK=y  
export SCALAPACK=&quot;-L$MKLROOT/lib/intel64 -lmkl_scalapack_ilp64 -lmkl_intel_ilp64 -lmkl_core -lmkl_sequential \  
-lmkl_blacs_intelmpi_ilp64 -lpthread -lm&quot;  
export SCALAPACK_SIZE=8  
export SCALAPACK_LIB=&quot;$SCALAPACK&quot; 
export BLAS_SIZE=8
export BLASOPT=&quot;-L$MKLROOT/lib/intel64 -lmkl_intel_ilp64 -lmkl_core -lmkl_sequential -lmkl_core -liomp5 -lpthread -ldmapp -lm&quot;  
export USE_NOIO=y  
export CRAYPE_LINK_TYPE=dynamic
</code></pre>
<p>To compile</p>
<pre><code>make nwchem_config
make FC=ftn
</code></pre>
<h2 id="compiling-nwchem-how-to-intel-xeon-phi">How-to: Intel Xeon Phi<a class="headerlink" href="#compiling-nwchem-how-to-intel-xeon-phi" title="Permanent link">&para;</a></h2>
<p>This section describes both the newer KNL and older KNC hardware, in
reverse chronological order.</p>
<ul>
<li><strong>Compiling NWChem on self-hosted Intel Xeon Phi Knights Landing
    processors</strong></li>
</ul>
<p>NWChem 6.6 (and later versions) support OpenMP threading, which is
essential to obtaining good performance with NWChem on Intel Xeon Phi
many-core processors.<br />
As of November 2016, the development version of NWChem contains
threading support in the TCE coupled-cluster codes (primarily
non-iterative triples in e.g. CCSD(T)), semi-direct CCSD(T), and
plane-wave DFT (i.e. NWPW).</p>
<p>Required for compilation: Intel compilers, version 16+ (17+ is strongly
recommended).</p>
<p>Environmental variables required for compilation:</p>
<pre><code>% export USE_KNL=1 
% export USE_OPENMP=1  
% export USE_F90_ALLOCATABLE=T  
% export USE_FASTMEM=T
</code></pre>
<p>The latter two options are required to allocate temporaries in MCDRAM
when running in flat mode. Please do not use cache mode for NWChem
CCSD(T) codes. Note that using Fortran heap allocations means the memory
statistics generated by MA are no longer accurate, but we doubt that
anyone has been relying on these anyways.</p>
<p><code>USE_FASTMEM</code> requires the <em>memkind</em> library to be installed.
An open source version of the memkind library can be downloaded from <a href="https://github.com/memkind/memkind">Github</a></p>
<p>Side note: With the exception of <code>USE_FASTMEM</code>, all of the options in
the KNL section apply to Intel Xeon processors as well. OpenMP is
certainly useful on multicore processors as a way to reduce the
communication overhead and memory footprint of NWChem.</p>
<p>When using MKL and Intel 16+, please use the following
settings</p>
<pre><code>% export BLASOPT  =&quot;-mkl -qopenmp&quot; 
% export SCALAPACK=&quot;-mkl -qopenmp -lmkl_scalapack_ilp64 -lmkl_blacs_intelmpi_ilp64&quot;
</code></pre>
<p>The command require for compilation is</p>
<pre><code>$ make FC=ifort CC=icc
</code></pre>
<p>Environmental variables recommended at runtime (assuming Intel OpenMP
and MPI):</p>
<pre><code>% export I_MPI_PIN=1  
% export I_MPI_DEBUG=4  
% export KMP_BLOCKTIME=1 
% export KMP_AFFINITY=scatter,verbose
</code></pre>
<p>Once you are comfortable with the affinity settings, you can use these
instead:</p>
<pre><code>% export I_MPI_PIN=1
% export KMP_BLOCKTIME=1  
% export KMP_AFFINITY=scatter
</code></pre>
<p>Please consult the Intel or similar documentation regarding MPI+OpenMP
affinity on your system. This is a complicated issue that depends on the
software you use; it is impossible to document all the different
combinations of MPI and OpenMP implementations that may be used with
NWChem.</p>
<p>If you encounter segfaults not related to ARMCI, you may need to set the
following or recompile with <code>-heap-arrays</code>. Please create thread in the
Forum if you observe this.</p>
<pre><code>% ulimit -s unlimited  
% export OMP_STACKSIZE=32M
</code></pre>
<ul>
<li><strong>Compiling NWChem on hosts equipped with Intel Xeon Phi Knights
    Corner coprocessors</strong></li>
</ul>
<p>NWChem 6.5 (and later versions) offers the possibility of using Intel
Xeon Phi hardware to perform the most computationally intensive part of
the CCSD(T) calculations (non-iterative triples corrections).</p>
<p>Required for compilation: Intel Composer XE version 14.0.3 (or later
versions)</p>
<p>Environmental variables required for compilation:</p>
<pre><code>% export USE_OPENMP=1 
% export USE_OFFLOAD=1
</code></pre>
<p>When using MKL and Intel Composer XE version 14 (or later versions),
please use the following
settings</p>
<pre><code>% export BLASOPT  =&quot;-mkl -openmp   -lpthread -lm&quot;  
% export SCALAPACK=&quot;-mkl -openmp -lmkl_scalapack_ilp64 -lmkl_blacs_intelmpi_ilp64 -lpthread -lm&quot;
</code></pre>
<p>The command require for compilation is</p>
<pre><code>$ make FC=ifort 
</code></pre>
<ul>
<li><strong>Examples of recommended configurations</strong></li>
</ul>
<p>From our experience using the CCSD(T) TCE module, we have determined
that the optimal configuration is to use a single Global Arrays ranks
for offloading work to each Xeon Phi card.</p>
<p>On the EMSL cascade system, each node is equipped with two coprocessors,
and NWChem can allocate one GA ranks per coprocessor. In the job
scripts, we recommend spawning just 6 GA ranks for each node, instead of
16 (number that would match the number of physical cores). Therefore, 2
out 6 GA ranks assigned to a particular compute node will offload to the
coprocessors, while the remaining 6 cores while be used for traditional
CPU processing duties. Since during offload the host core is idle, we
can double the number of OpenMP threads for the host
(<code>OMP_NUM_THREADS=4</code>) in order to fill the idle core with work from
another GA rank (4 process with 4 threads each will total 16 threads on
each node).</p>
<p>NWChem itself automatically detects the available coprocessors in the
system and properly partitions them for optimal use, therefore no action
is required other than specifying the number of processes on each node
(using the appropriate mpirun/mpiexec options) and setting the value of
<code>OMP_NUM_THREADS</code> as in the example above.</p>
<p>Environmental variables useful at run-time:</p>
<p>OMP_NUM_THREADS is needed for the thread-level parallelization on the
Xeon CPU hosts</p>
<pre><code>% export OMP_NUM_THREADS=4
</code></pre>
<p>MIC_USE_2MB_BUFFER greatly improve communication between host and
Xeon Phi card</p>
<pre><code>% export MIC_USE_2MB_BUFFER=16K
</code></pre>
<p><strong>Very important</strong>: when running on
clusters equipped with Xeon Phi and Infiniband network hardware
(requiring <code>ARMCI_NETWORK=OPENIB</code>), the following env. variable is
required, even in the case when the Xeon Phi hardware is not
utilized.</p>
<pre><code>% export ARMCI_OPENIB_DEVICE=mlx4_0
</code></pre>
<h2 id="compiling-nwchem-how-to-ibm-platforms">How-to: IBM platforms<a class="headerlink" href="#compiling-nwchem-how-to-ibm-platforms" title="Permanent link">&para;</a></h2>
<ul>
<li><strong>Compiling NWChem on BLUEGENE/L</strong></li>
</ul>
<p>The following environment variables need to be
set</p>
<pre><code>% export NWCHEM_TOP=&lt;your path&gt;/nwchem
% export NWCHEM_TARGET=BGL
% export ARMCI_NETWORK=BGMLMPI
% export BGLSYS_DRIVER=/bgl/BlueLight/ppcfloor
% export BGLSYS_ROOT=${BGLSYS_DRIVER}/bglsys
% export BLRTS_GNU_ROOT=${BGLSYS_DRIVER}/blrts-gnu
% export BGDRIVER=${BGLSYS_DRIVER}
% export BGCOMPILERS=${BLRTS_GNU_ROOT}/bin
% export USE_MPI=y
% export LARGE_FILES=TRUE
% export MPI_LIB=${BGLSYS_ROOT}/lib
% export MPI_INCLUDE=${BGLSYS_ROOT}/include
% export LIBMPI=&quot;-lfmpich_.rts -lmpich.rts -lmsglayer.rts -lrts.rts -ldevices.rts&quot;
% export BGMLMPI_INCLUDE=/bgl/BlueLight/ppcfloor/bglsys/include
% export BGMLLIBS=/bgl/BlueLight/ppcfloor/bglsys/lib
</code></pre>
<p>To compile, the following commands should be used:</p>
<pre><code>% cd $NWCHEM_TOP/src
% make nwchem_config
% make FC=blrts_xlf &gt;&amp; make.log
</code></pre>
<ul>
<li><strong>Compiling NWChem on BLUEGENE/P</strong></li>
</ul>
<p>The following environment variables need to be
set</p>
<pre><code>% export NWCHEM_TARGET=BGP
% export ARMCI_NETWORK=DCMFMPI
% export MSG_COMMS=DCMFMPI
% export USE_MPI=y
% export LARGE_FILES=TRUE
% export BGP_INSTALLDIR=/bgsys/drivers/ppcfloor
% export BGCOMPILERS=/bgsys/drivers/ppcfloor/gnu-linux/bin
% export BGP_RUNTIMEPATH=/bgsys/drivers/ppcfloor/runtime
% export ARMCIDRV=${BGP_INSTALLDIR}
% export BGDRIVER=${ARMCIDRV}
% export MPI_LIB=${BGDRIVER}/comm/lib
% export MPI_INCLUDE=${BGDRIVER}/comm/include
% export LIBMPI=&quot;-L${MPI_LIB} -lfmpich_.cnk -lmpich.cnk -ldcmfcoll.cnk -ldcmf.cnk -lpthread -lrt -L${BGP_RUNTIMEPATH}/SPI -lSPI.cna&quot;
% export BGMLMPI_INCLUDE=${MPI_INCLUDE}
</code></pre>
<p>To compile, the following commands should be used:</p>
<pre><code>% cd $NWCHEM_TOP/src
% make nwchem_config
% make FC=bgxlf &gt;&amp; make.log
</code></pre>
<ul>
<li><strong>Compiling NWChem on BLUEGENE/Q</strong></li>
</ul>
<p>The following environment variables need to be
set</p>
<pre><code>% export NWCHEM_TARGET=BGQ
% export USE_MPI=y
% export USE_MPIF=y
% export USE_MPIF4=y
% export MPI_INCLUDE=/bgsys/drivers/ppcfloor/comm/xl/include
% export LIBMPI=&quot; &quot;
% export BLASOPT=&quot;/opt/ibmmath/essl/5.1/lib64/libesslbg.a -llapack -lblas -Wl,-zmuldefs &quot;
% export BLAS_LIB=&quot;/opt/ibmmath/essl/5.1/lib64/libesslbg.a -zmuldefs &quot;
% export BLAS_SIZE=4
% export USE_64TO32=y
% set path=(/bgsys/drivers/ppcfloor/gnu-linux/bin/ $path)
% export ARMCI_NETWORK=MPI-TS
% export DISABLE_GAMIRROR=y
</code></pre>
<p>To compile, the following commands should be used:</p>
<pre><code>% module load bgq-xl
% make nwchem_config
% make 64_to_32 &gt;&amp; make6t3.log
% make &gt;&amp; make.log
</code></pre>
<p><strong>WARNING</strong>: This is just a baseline port that we have tested and
validated against our QA suite. There is large room for improvement both
for serial performance (compiler options) and parallel performance (use
of alternative ARMCI_NETWORKs other than MPI-TS)</p>
<ul>
<li><strong>Compiling NWChem on IBM PowerPC architectures</strong></li>
</ul>
<p>The following environment variables should be set:</p>
<pre><code>% export NWCHEM_TOP=&lt;your path&gt;/nwchem
% export NWCHEM_TARGET=IBM64
% export ARMCI_NETWORK=MPI-MT
% export OBJECT_MODE=64
% export USE_MPI=y
</code></pre>
<p>To compile, the following commands should be used:</p>
<pre><code>% cd $NWCHEM_TOP/src
% make nwchem_config
% make FC=xlf &gt;&amp; make.log
</code></pre>
<h2 id="compiling-nwchem-how-to-commodity-clusters-with-infiniband">How-to: Commodity clusters with Infiniband<a class="headerlink" href="#compiling-nwchem-how-to-commodity-clusters-with-infiniband" title="Permanent link">&para;</a></h2>
<p>Common environmental variables for building and running on most
Infiniband clusters are:</p>
<pre><code>  export NWCHEM_TOP=&lt;your path&gt;/nwchem  
  export NWCHEM_TARGET=LINUX64 
  export NWCHEM_MODULES=&quot;all&quot;  
  export USE_MPI=y 
  export USE_MPIF=y 
  export USE_MPIF4=y  
</code></pre>
<ul>
<li>On Infiniband clusters with the OpenIB software stack, the following
    environment variables should be defined</li>
</ul>
<pre><code>  export ARMCI_NETWORK=OPENIB 
  export IB_INCLUDE=&lt;Location of Infiniband libraries&gt;/include  
</code></pre>
<ul>
<li>Compiling the code on an Infiniband cluster once all variables are
    set</li>
</ul>
<pre><code>  cd $NWCHEM_TOP/src  

  make nwchem_config  

  make &gt;&amp; make.log
</code></pre>
<h2 id="compiling-nwchem-how-to-commodity-clusters-with-intel-omni-path">How-to: Commodity clusters with Intel Omni-Path<a class="headerlink" href="#compiling-nwchem-how-to-commodity-clusters-with-intel-omni-path" title="Permanent link">&para;</a></h2>
<ul>
<li>On  clusters with the Intel Omni-Path network, the following
    environment variables should be defined</li>
</ul>
<pre><code>  export ARMCI_NETWORK=MPI-PR
</code></pre>
<p>The following setting is needed to avoid run-time errors</p>
<pre><code>  export PSM2_MEMORY=large
</code></pre>
<p>More details on this topic discussed a  </p>
<ul>
<li>
<p><a href="https://github.com/nwchemgit/nwchem/issues/284">https://github.com/nwchemgit/nwchem/issues/284</a></p>
</li>
<li>
<p><a href="https://github.com/GlobalArrays/ga/issues/126">https://github.com/GlobalArrays/ga/issues/126</a></p>
</li>
</ul>
<h2 id="compiling-nwchem-how-to-windows-platforms">How-to: Windows Platforms<a class="headerlink" href="#compiling-nwchem-how-to-windows-platforms" title="Permanent link">&para;</a></h2>
<h3 id="compiling-nwchem-mingw">MingW<a class="headerlink" href="#compiling-nwchem-mingw" title="Permanent link">&para;</a></h3>
<p>The current recommended approach for building a NWChem binary for a
Windows platform is to build with the
<a href="https://sourceforge.net/projects/mingw/files/">MinGW/Mingw32</a> environment. MinGW can be
installed using a semi-automatic tool mingw-get-setup.exe
(http://sourceforge.net/projects/mingw/files/Installer/). A basic MinGW
installation is required (Basic Setup), plus pthreads-32,
mingw32-gcc-fortran-dev of &ldquo;All Packages&rdquo; and the MSYS software.<br />
More detailed MinGW/MSYS installation tips can be found in the following
forum discussions</p>
<table>
<thead>
<tr>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://nwchemgit.github.io/Special_AWCforum/sp/id5124.html">https://nwchemgit.github.io/Special_AWCforum/sp/id5124.html</a></td>
</tr>
<tr>
<td><a href="https://nwchemgit.github.io/Special_AWCforum/sp/id6628.html">https://nwchemgit.github.io/Special_AWCforum/sp/id6628.html</a></td>
</tr>
</tbody>
</table>
<p>Another essential prerequisite step is to install Mpich, which can be
found at the following
URL</p>
<p><a href="https://web.archive.org/web/20190105171104/http://www.mpich.org/static/tarballs/1.4.1p1/mpich2-1.4.1p1-win-ia32.msi">https://web.archive.org/web/20190105171104/http://www.mpich.org/static/tarballs/1.4.1p1/mpich2-1.4.1p1-win-ia32.msi</a></p>
<p>Once Mpich is installed, you should copy the installation files to a
different location to avoid the failure of the tools compilation. You
can use the following command</p>
<pre><code>% cp -rp /c/Program\ Files\ \(x86\)/MPICH2/ ~/
</code></pre>
<p>You might want to install Python, too, by using the following
installation file</p>
<p><a href="https://www.python.org/ftp/python/2.7.8/python-2.7.8.msi">https://www.python.org/ftp/python/2.7.8/python-2.7.8.msi</a></p>
<p>Next, you need to set the env.</p>
<pre><code>% export NWCHEM_TOP=~/nwchem-6.8  
% export NWCHEM_TARGET=LINUX
% export USE_MPI=y  
% export MPI_LOC=~/MPICH2  
% export MPI_INCLUDE=$MPI_LOC/include  
% export MPI_LIB=$MPI_LOC/lib  
% export LIBMPI=&quot;-lfmpich2g -lmpi&quot;  
% export PYTHONVERSION=27  
% export DEPEND_CC=gcc
% export USE_INTERNALBLAS=y
% export NWCHEM_MODULES=all
</code></pre>
<p>Then, you can start the compilation by typing</p>
<pre><code>% cd $NWCHEM_TOP/src  
% make nwchem_config  
% make FC=gfortran DEPEND_CC=gcc
</code></pre>
<h4 id="compiling-nwchem-msys2">MSYS2<a class="headerlink" href="#compiling-nwchem-msys2" title="Permanent link">&para;</a></h4>
<p>https://github.com/msys2/msys2/wiki/MSYS2-installation</p>
<pre><code>pacman -Syuu
pacman -S mingw32/mingw-w64-i686-gcc-fortran
pacman -S mingw32/mingw-w64-i686-python3
pacman -S msys/make
</code></pre>
<h3 id="compiling-nwchem-wsl-on-windows-10">WSL on Windows 10<a class="headerlink" href="#compiling-nwchem-wsl-on-windows-10" title="Permanent link">&para;</a></h3>
<p>A good alternative only on Windows 10 is <strong>Windows Subsystem for Linux</strong> (WSL).
This option gives the best performance on Windows when <a href="https://docs.microsoft.com/en-us/windows/wsl/install-win10">WLS 2</a> is used.
<strong>WSL</strong> allows you to obtain a functional command line Linux 64-bit NWChem environment, either by compiling the NWChem code from scratch or by using the Ubuntu precompiled NWChem package. Here is a link to the install guide</p>
<p><a href="https://learn.microsoft.com/en-us/windows/wsl/install">https://learn.microsoft.com/en-us/windows/wsl/install</a></p>
<p>Once Ubuntu is installed, the quickest method to install NWChem is by fetching the Ubuntu NWChem package by typing</p>
<pre><code>sudo apt install nwchem
</code></pre>
<h2 id="compiling-nwchem-general-site-installation">General site installation<a class="headerlink" href="#compiling-nwchem-general-site-installation" title="Permanent link">&para;</a></h2>
<p>The build procedures outlined above will allow use of NWChem within the
NWChem directory structure. The code will look for the basis set library
file in a default place within that directory structure. To install the
code in a general, public place (e.g., /usr/local/NWChem) the following
procedure can be applied:</p>
<ul>
<li>Determine the local storage path for the install files. (e.g.,
    /usr/local/NWChem).</li>
<li>Make directories</li>
</ul>
<pre><code>mkdir /usr/local/NWChem
mkdir /usr/local/NWChem/bin
mkdir /usr/local/NWChem/data
</code></pre>
<ul>
<li>Copy binary</li>
</ul>
<pre><code>cp $NWCHEM_TOP/bin/${NWCHEM_TARGET}/nwchem /usr/local/NWChem/bin
cd /usr/local/NWChem/bin
chmod 755 nwchem
</code></pre>
<ul>
<li>Set links to data files (basis sets, force fields, etc.)</li>
</ul>
<pre><code>cd $NWCHEM_TOP/src/basis
cp -r libraries /usr/local/NWChem/data
cd $NWCHEM_TOP/src/
cp -r data /usr/local/NWChem
cd $NWCHEM_TOP/src/nwpw
cp -r libraryps /usr/local/NWChem/data
</code></pre>
<ul>
<li>Each user will need a .nwchemrc file to point to these default data
    files. A global one could be put in /usr/local/NWChem/data and a
    symbolic link made in each users <code>$HOME</code> directory is probably the
    best plan for new installations. Users would have to issue the following
    command prior to using NWChem:</li>
</ul>
<pre><code>ln -s /usr/local/NWChem/data/default.nwchemrc $HOME/.nwchemrc
</code></pre>
<p>Contents of the <code>default.nwchemrc</code> file based on the above information
should be:</p>
<pre><code>nwchem_basis_library /usr/local/NWChem/data/libraries/
nwchem_nwpw_library /usr/local/NWChem/data/libraryps/
ffield amber
amber_1 /usr/local/NWChem/data/amber_s/
amber_2 /usr/local/NWChem/data/amber_q/
amber_3 /usr/local/NWChem/data/amber_x/
amber_4 /usr/local/NWChem/data/amber_u/
spce    /usr/local/NWChem/data/solvents/spce.rst
charmm_s /usr/local/NWChem/data/charmm_s/
charmm_x /usr/local/NWChem/data/charmm_x/
</code></pre>
<p>Of course users can copy this file instead of making the symbolic link
described above and change these defaults at their discretion.</p>
<p>It is can also be useful to use the <code>NWCHEM_BASIS_LIBRARY</code> environment
variable when testing a new installation when an old one exists. This
will allow you to overwrite the value of <code>nwchem_basis_library</code> in your
<code>.nwchemrc</code> file and point to the new basis library. For example:</p>
<pre><code>% export NWCHEM_BASIS_LIBRARY=&quot;$NWCHEM/data-5.0/libraries/&quot;
</code></pre>
<p>Do not forget the trailing &ldquo;/&rdquo;.</p></section></section>
                    <section class='print-page md-section' id='section-4' heading-number='4'>
                        <h1>Introduction <a class='headerlink' href='#section-4' title='Permanent link'></a>
                        </h1>
                    <section class="print-page" id="getting-started" heading-number="4.1"><h1 id="getting-started-getting-started">Getting Started<a class="headerlink" href="#getting-started-getting-started" title="Permanent link">&para;</a></h1>
<p>This section provides an overview of NWChem input and program
architecture, and the syntax used to describe the input. See <a href="#getting-started-simple-input-file-scf-geometry-optimization">Simple
Input File</a>
and <a href="#getting-started-water-molecule-sample-input-file">Water Molecule Input</a>
for examples of NWChem input files with detailed explanation.</p>
<p>NWChem consists of independent modules that perform the various
functions of the code. Examples of modules include the input parser, SCF
energy, SCF analytic gradient, DFT energy, etc.. Data is passed between
modules and saved for restart using a disk-resident database or dumpfile
(see <a href="#nwchem-architecture">NWChem Architecture</a>).</p>
<p>The input to NWChem is composed of commands, called directives, which
define data (such as basis sets, geometries, and filenames) and the
actions to be performed on that data. Directives are processed in the
order presented in the input file, with the exception of certain
start-up directives (see <a href="#getting-started-input-file-structure">Input File
Structure</a>) which provide critical job
control information, and are processed before all other input. Most
directives are specific to a particular module and define data that is
used by that module only. A few directives (see <a href="#top-level">Top-level
Directives</a>)
potentially affect all modules, for instance by specifying the total
electric charge on the system.</p>
<p>There are two types of directives. Simple directives consist of one line
of input, which may contain multiple fields. Compound directives group
together multiple simple directives that are in some way related and are
terminated with an END directive. See the sample inputs (<a href="#getting-started-simple-input-file-scf-geometry-optimization">Simple Input
File</a> and
<a href="#getting-started-water-molecule-sample-input-file">Water Molecule Input</a>)
and the input syntax specification (<a href="#getting-started-input-format-and-syntax-for-directives">Input Format and Syntax for
Directives</a>).</p>
<p>All input is free format and case is ignored except for actual data
(e.g., names/tags of centers, titles). Directives or blocks of
module-specific directives (i.e., compound directives) can appear in any
order, with the exception of the TASK directive (see <a href="#getting-started-input-file-structure">Input File
Structure</a> and
<a href="#task">Tasks</a>) which is used to invoke an
NWChem module. All input for a given task must precede the TASK
directive. This input specification rule allows the concatenation of
multiple tasks in a single NWChem input file.</p>
<p>To make the input as short and simple as possible, most options have
default values. The user needs to supply input only for those items that
have no defaults, or for items that must be different from the defaults
for the particular application. In the discussion of each directive, the
defaults are noted, where applicable.</p>
<p>The input file structure is described in the following sections, and
illustrated with two examples. The input format and syntax for
directives is also described in detail.</p>
<h2 id="getting-started-nwchemrc-for-environment-variables-and-libraries">.nwchemrc for environment variables and libraries<a class="headerlink" href="#getting-started-nwchemrc-for-environment-variables-and-libraries" title="Permanent link">&para;</a></h2>
<p>Each user should have a <code>.nwchemrc</code> file to point to default data files,
such as basis sets, pseudopotentials, and MD potentials.</p>
<p>Contents of the default.nwchemrc file based on the above information
should
be:</p>
<pre><code> nwchem_basis_library &lt;location of NWChem installation&gt;/src/basis/libraries/  
 nwchem_nwpw_library &lt;location of NWChem installation&gt;/src/nwpw/libraryps/  
 ffield amber  
 amber_1 &lt;location of NWChem installation&gt;/src/data/amber_s/  
 amber_2 &lt;location of NWChem installation&gt;/src/data/amber_q/  
 amber_3 &lt;location of NWChem installation&gt;/src/data/amber_x/  
 amber_4 &lt;location of NWChem installation&gt;/src/data/amber_u/  
 spce   &lt;location of NWChem installation&gt;/src/data/solvents/spce.rst  
 charmm_s &lt;location of NWChem installation&gt;/src/data/charmm_s/  
 charmm_x &lt;location of NWChem installation&gt;/src/data/charmm_x/
</code></pre>
<p>It is can also be useful to use the NWCHEM_BASIS_LIBRARY environment
variable when testing a new libraries in your own directory. This will
allow you to overwrite the value of nwchem_basis_library in your
.nwchemrc file and point to the new basis library. For example:</p>
<pre><code>   % setenv NWCHEM_BASIS_LIBRARY &quot;$NWCHEM/data-5.0/libraries/&quot;
</code></pre>
<p>Do not forget the trailing <code>"/"</code>.</p>
<h2 id="getting-started-input-file-structure">Input File Structure<a class="headerlink" href="#getting-started-input-file-structure" title="Permanent link">&para;</a></h2>
<p>The structure of an input file reflects the internal structure of
NWChem. At the beginning of a calculation, NWChem needs to determine how
much memory to use, the name of the database, whether it is a new or
restarted job, where to put scratch/permanent files, etc.. It is not
necessary to put this information at the top of the input file, however.
NWChem will read through the entire input file looking for the start-up
directives. In this first pass, all other directives are ignored.</p>
<p>The start-up directives are</p>
<pre><code>START  
RESTART  
SCRATCH_DIR  
PERMANENT_DIR  
MEMORY  
ECHO
</code></pre>
<p>After the input file has been scanned for the start-up directives, it is
rewound and read sequentially. Input is processed either by the
top-level parser (for the directives listed in <a href="#top-level">Top-level
Directives</a>, such as TITLE, SET, &hellip;) or
by the parsers for specific computational modules (e.g., SCF, DFT, &hellip;).
Any directives that have already been processed (e.g., MEMORY) are
ignored. Input is read until a TASK directive (see
<a href="#task">Tasks</a>) is encountered. A TASK
directive requests that a calculation be performed and specifies the
level of theory and the operation to be performed. Input processing then
stops and the specified task is executed. The position of the TASK
directive in effect marks the end of the input for that task. Processing
of the input resumes upon the successful completion of the task, and the
results of that task are available to subsequent tasks in the same input
file.</p>
<p>The name of the input file is usually provided as an argument to the
execute command for NWChem. That is, the execute command looks something
like the following</p>
<pre><code> nwchem input_file
</code></pre>
<p>The default name for the input file is <code>nwchem.nw</code>. If an input file name
<code>input_file</code> is specified without an extension, the code assumes .nw as a
default extension, and the input filename becomes <code>input_file.nw</code>. If the
code cannot locate a file named either <code>input_file</code> or <code>input_file.nw</code> (or
nwchem.nw if no file name is provided), an error is reported and
execution terminates. The following section presents two input files to
illustrate the directive syntax and input file format for NWChem
applications.</p>
<h2 id="getting-started-simple-input-file-scf-geometry-optimization">Simple Input File &ndash; SCF geometry optimization<a class="headerlink" href="#getting-started-simple-input-file-scf-geometry-optimization" title="Permanent link">&para;</a></h2>
<p>A simple example of an NWChem input file is an SCF geometry optimization
of the nitrogen molecule, using a Dunning cc-pvdz basis set. This input
file contains the bare minimum of information the user must specify to
run this type of problem &ndash; fewer than ten lines of input, as follows:</p>
<pre><code> title &quot;Nitrogen cc-pvdz SCF geometry optimization&quot;  
 geometry    
   n 0 0 0  
   n 0 0 1.08  
 end  
 basis  
   n library cc-pvdz  
 end  
 task scf optimize
</code></pre>
<p>Examining the input line by line, it can be seen that it contains only
four directives; TITLE, GEOMETRY, BASIS, and TASK. The TITLE directive
is optional, and is provided as a means for the user to more easily
identify outputs from different jobs. An initial geometry is specified
in Cartesian coordinates and Angstrms by means of the GEOMETRY
directive. The Dunning cc-pvdz basis is obtained from the NWChem basis
library, as specified by the BASIS directive input. The TASK directive
requests an SCF geometry optimization.</p>
<p>The <a href="#geometry">GEOMETRY directive</a> defaults to
Cartesian coordinates and Angstrms (options include atomic units and
Z-matrix format). The input blocks for the BASIS and GEOMETRY directives
are structured in similar fashion, i.e., name, keyword, &hellip;, end (In
this simple example, there are no keywords). The BASIS input block must
contain basis set information for every atom type in the geometry with
which it will be used. Refer to <a href="#basis">Basis</a> for a
description of available basis sets and a discussion of how to define
new ones.</p>
<p>The last line of this sample input file (task scf optimize) tells the
program to optimize the molecular geometry by minimizing the SCF energy.
(For a description of possible tasks and the format of the TASK
directive, refer to <a href="#task">Tasks</a>)</p>
<p>If the input is stored in the file <code>n2.nw</code>, the command to run this job on
a typical UNIX workstation is as follows:</p>
<pre><code> nwchem n2
</code></pre>
<p>NWChem output is to UNIX standard output, and error messages are sent to
both standard output and standard error.</p>
<h2 id="getting-started-water-molecule-sample-input-file">Water Molecule Sample Input File<a class="headerlink" href="#getting-started-water-molecule-sample-input-file" title="Permanent link">&para;</a></h2>
<p>A more complex sample problem is the optimization of a positively
charged water molecule using second-order Mller-Plesset perturbation
theory (MP2), followed by a computation of frequencies at the optimized
geometry. A preliminary SCF geometry optimization is performed using a
computationally inexpensive basis set (STO-3G). This yields a good
starting guess for the optimal geometry, and any Hessian information
generated will be used in the next optimization step. Then the
optimization is finished using MP2 and a basis set with polarization
functions. The final task is to calculate the MP2 vibrational
frequencies. The input file to accomplish these three tasks is as
follows:</p>
<pre><code>start h2o_freq  
charge 1  
geometry units angstroms  
 O       0.0  0.0  0.0  
 H       0.0  0.0  1.0  
 H       0.0  1.0  0.0  
end  
basis  
 H library sto-3g  
 O library sto-3g  
end  
scf  
 uhf; doublet  
 print low  
end  
title &quot;H2O+ : STO-3G UHF geometry optimization&quot;  
task scf optimize  
basis  
 H library 6-31g**  
 O library 6-31g**  
end  
title &quot;H2O+ : 6-31g** UMP2 geometry optimization&quot;  
task mp2 optimize  
mp2; print none; end  
scf; print none; end  
title &quot;H2O+ : 6-31g** UMP2 frequencies&quot;  
task mp2 freq
</code></pre>
<p>The START directive
(<a href="#start_restart">START/RESTART</a> tells
NWChem that this run is to be started from the beginning. This directive
need not be at the beginning of the input file, but it is commonly
placed there. Existing database or vector files are to be ignored or
overwritten. The entry h2o_freq on the START line is the prefix to be
used for all files created by the calculation. This convention allows
different jobs to run in the same directory or to share the same scratch
directory
<a href="#scratch_dir">SCRATCH_DIR</a>/<a href="#permanent_dir">PERMANENT_DIR</a>,
as long as they use different prefix names in this field.</p>
<p>As in the first sample problem, the geometry is given in Cartesian
coordinates. In this case, the units are specified as Angstrms. (Since
this is the default, explicit specification of the units is not actually
necessary, however.) The CHARGE directive defines the total charge of
the system. This calculation is to be done on an ion with charge +1.</p>
<p>A small basis set (STO-3G) is specified for the intial geometry
optimization. Next, the multiple lines of the first SCF directive in the
scf &hellip;end block specify details about the SCF calculation to be
performed. Unrestricted Hartree-Fock is chosen here (by specifying the
keyword uhf), rather than the default, restricted open-shell high-spin
Hartree-Fock (ROHF). This is necessary for the subsequent MP2
calculation, because only UMP2 is currently available for open-shell
systems (see Section 4). For open-shell systems, the spin multiplicity
has to be specified (using doublet in this case), or it defaults to
singlet. The print level is set to low to avoid verbose output for the
starting basis calculations.</p>
<p>All input up to this point affects only the settings in the runtime
database. The program takes its information from this database, so the
sequence of directives up to the first TASK directive is irrelevant. An
exchange of order of the different blocks or directives would not affect
the result. The TASK directive, however, must be specified after all
relevant input for a given problem. The TASK directive causes the code
to perform the specified calculation using the parameters set in the
preceding directives. In this case, the first task is an SCF calculation
with geometry optimization, specified with the input scf and optimize.
(See <a href="#task">Tasks</a> for a list of
available tasks and operations.)</p>
<p>After the completion of any task, settings in the database are used in
subsequent tasks without change, unless they are overridden by new input
directives. In this example, before the second task (task mp2 optimize),
a better basis set (6-31G**) is defined and the title is changed. The
second TASK directive invokes an MP2 geometry optimization.</p>
<p>Once the MP2 optimization is completed, the geometry obtained in the
calculation is used to perform a frequency calculation. This task is
invoked by the keyword freq in the final TASK directive, task mp2 freq.
The second derivatives of the energy are calculated as numerical
derivatives of analytical gradients. The intermediate energies and
gradients are not of interest in this case, so output from the SCF and
MP2 modules is disabled with the PRINT directives.</p>
<h2 id="getting-started-input-format-and-syntax-for-directives">Input Format and Syntax for Directives<a class="headerlink" href="#getting-started-input-format-and-syntax-for-directives" title="Permanent link">&para;</a></h2>
<p>This section describes the input format and the syntax used in the rest
of this documentation to describe the format of directives. The input
format for the directives used in NWChem is similar to that of UNIX
shells, which is also used in other chemistry packages, most notably
GAMESS-UK. An input line is parsed into whitespace (blanks or tabs)
separating tokens or fields. Any token that contains whitespace must be
enclosed in double quotes in order to be processed correctly. For
example, the basis set with the descriptive name modified Dunning DZ
must appear in a directive as &ldquo;modified Dunning DZ&rdquo;, since the name
consists of three separate words.</p>
<h3 id="getting-started-input-format">Input Format<a class="headerlink" href="#getting-started-input-format" title="Permanent link">&para;</a></h3>
<p>A (physical) line in the input file is terminated with a newline
character (also known as a `return&rsquo; or `enter&rsquo; character). A semicolon
(;) can be also used to indicate the end of an input line, allowing a
single physical line of input to contain multiple logical lines of
input. For example, five lines of input for the GEOMETRY directive can
be entered as follows;</p>
<pre><code> geometry  
  O 0  0     0  
  H 0  1.430 1.107  
  H 0 -1.430 1.107  
 end
</code></pre>
<p>These same five lines could be entered on a single line, as</p>
<pre><code> geometry; O 0 0 0; H 0 1.430 1.107; H 0 -1.430 1.107; end
</code></pre>
<p>This one physical input line comprises five logical input lines. Each
logical or physical input line must be no longer than 1023 characters.</p>
<p>In the input file:</p>
<ul>
<li>a string, token, or field is a sequence of ASCII characters (NOTE:
    if the string includes blanks or tabs (i.e., white space), the
    entire string must be enclosed in double quotes).</li>
<li><code>\</code> (backslash) at the end of a line concatenates it with the next
    line. Note that a space character is automatically inserted at this
    point so that it is not possible to split tokens across lines. A
    backslash is also used to quote special characters such as
    whitespace, semi-colons, and hash symbols so as to avoid their
    special meaning (NOTE: these special symbols must be quoted with the
    backslash even when enclosed within double quotes).</li>
<li><code>;</code> (semicolon) is used to mark the end of a logical input line within
    a physical line of input.</li>
<li><code>#</code> (the hash or pound symbol) is the comment character. All
    characters following <code>#</code> (up to the end of the physical line) are
    ignored.</li>
<li>If any input line (excluding <a href="#python">Python programs</a> begins
    with the string INCLUDE (ignoring case) and is followed by a valid
    file name, then the data in that file are read as if they were
    included into the current input file at the current line. Up to
    three levels of nested include files are supported. The user should
    note that inputting a basis set from the standard basis library
    (<a href="#basis">Basis Sets</a>) uses one level of include.</li>
<li>Data is read from the input file until an end-of-file is detected,
    or until the string EOF (ignoring case) is encountered at the
    beginning of an input line.</li>
</ul>
<h3 id="getting-started-format-and-syntax-of-directives">Format and syntax of directives<a class="headerlink" href="#getting-started-format-and-syntax-of-directives" title="Permanent link">&para;</a></h3>
<p>Directives consist of a directive name, keywords, and optional input,
and may contain one line or many. Simple directives consist of a single
line of input with one or more fields. Compound directives can have
multiple input lines, and can also include other optional simple and
compound directives. A compound directive is terminated with an END
directive. The directives START (see
<a href="#start_restart">START/RESTART</a>) and
ECHO (see <a href="#echo">ECHO</a>) are examples of
simple directives. The directive GEOMETRY (see
<a href="#geometry">Geometry</a>) is an example of a
compound directive.</p>
<p>Some limited checking of the input for self-consistency is performed by
the input module, but most defaults are imposed by the application
modules at runtime. It is therefore usually impossible to determine
beforehand whether or not all selected options are consistent with each
other.</p>
<p>In the rest of this document, the following notation and syntax
conventions are used in the generic descriptions of the NWChem input.</p>
<ul>
<li>a directive name always appears in all-capitals, and in computer
    typeface (e.g., GEOMETRY, BASIS, SCF). Note that the case of
    directives and keywords is ignored in the actual input.</li>
<li>a keyword always appears in lower case, in computer typeface (e.g.,
    swap, print, units, bqbq).</li>
<li>variable names always appear in lower case, in computer typeface,
    and enclosed in angle brackets to distinguish them from keywords
    (e.g., <code>&lt;input_filename&gt;</code>, <code>&lt;basisname&gt;</code>, <code>&lt;tag&gt;</code>).</li>
<li><code>$variable$</code> is used to indicate the substitution of the value of a
    variable.</li>
<li><code>()</code> is used to group items (the parentheses and other special symbols
    should not appear in the input).</li>
<li><code>||</code> separate exclusive options, parameters, or formats.</li>
<li><code>[  ]</code> enclose optional entries that have a default value.</li>
<li><code>&lt; &gt;</code> enclose a type, a name of a value to be specified, or a
    default value, if any.</li>
<li><code>\</code> is used to concatenate lines in a description.</li>
<li><code>...</code> is used to indicate indefinite continuation of a list.</li>
</ul>
<p>An input parameter is identified in the description of the directive by
prefacing the name of the item with the type of data expected, i.e.,</p>
<ul>
<li>string - an ASCII character string</li>
<li>integer - integer value(s) for a variable or an array</li>
<li>logical - true/false logical variable</li>
<li>real - real floating point value(s) for a variable or an array</li>
<li>double - synonymous with real</li>
</ul>
<p>If an input item is not prefaced by one of these type names, it is
assumed to be of type ``string&rsquo;&lsquo;.</p>
<p>In addition, integer lists may be specified using Fortran triplet
notation, which interprets lo:hi:inc as lo, lo+inc, lo+2*inc, &hellip;, hi.
For example, where a list of integers is expected in the input, the
following two lines are equivalent</p>
<pre><code>  7 10 21:27:2 1:3 99  
  7 10 21 23 25 27 1 2 3 99
</code></pre>
<p>(In Fortran triplet notation, the increment, if unstated, is 1; e.g.,
1:3 = 1:3:1.)</p>
<p>The directive <a href="#hartree-fock-theory-for-molecules-vectors-inputoutput-of-mo-vectors">VECTORS</a> is presented here as an example of
an NWChem input directive. The general form of the directive is as
follows:</p>
<pre><code> VECTORS [input (&lt;string input_movecs default atomic&gt;) || \  
                  (project &lt;string basisname&gt; &lt;string filename&gt;)] \ 
         [swap [(alpha||beta)] &lt;integer vec1 vec2&gt; ...] \ 
         [output &lt;string output_movecs default $file_prefix$.movecs&gt;]
</code></pre>
<p>This directive contains three optional keywords, as indicated by the
three main sets of square brackets enclosing the keywords input, swap,
and output. The keyword input allows the user to specify the source of
the molecular orbital vectors. There are two mutually exclusive options
for specifying the vectors, as indicated by the || symbol separating the
option descriptions;</p>
<pre><code> (&lt;string input_movecs default atomic&gt;) || \ 
                 (project &lt;string basisname&gt; &lt;string filename&gt;) \
</code></pre>
<p>The first option, <code>&lt;string input_movecs default atomic&gt;</code>, can be
used to specify an ASCII character string for the parameter
<code>input_movecs</code>. If no entry is specified, the code uses the default
<code>atomic</code> (i.e., atomic guess). The second option, <code>project
&lt;string basisname&gt; &lt;string filename&gt;</code>, contains the keyword <code>project</code>,
which takes two string arguments. When this keyword is used, the vectors
in file <code>&lt;filename&gt;</code> will be projected from the (smaller) basis
<code>&lt;basisname&gt;</code> into the current atomic orbital (AO) basis.</p>
<p>The second keyword, <code>swap</code>, can be used to re-order the starting
vectors, specifying the pairs of vectors to be swapped. As many pairs as
the user wishes to have swapped can be listed for <integer vec1 vec2
... >. The optional keywords alpha and beta allow the user to swap the
alpha or beta spin orbitals.</p>
<p>The third keyword, <code>output</code>, allows the user to tell the code where to
store the vectors, by specifying an ASCII string for the parameter
<code>output_movecs</code>. If no entry is specified for this parameter, the default
is to write the vectors back into either the user- specified MO vectors
input file or, if this is not available, the file <code>$file_prefix$.movecs</code>.</p>
<p>A particular example of the VECTORS directive is shown below. It
specifies both the input and output keywords, but does not use the swap
option.</p>
<pre><code> vectors input project &quot;small basis&quot; small_basis.movecs \
         output large_basis.movecs
</code></pre>
<p>This directive tells the code to generate input vectors by projecting
from vectors in a smaller basis named &ldquo;small basis&rdquo;, which is stored in
the file <code>small_basis.movecs</code>. The output vectors will be stored in the
file <code>large_basis.movecs</code>.</p>
<p>The order of keyed optional entries within a directive should not
matter, unless noted otherwise in the specific instructions for a
particular directive.</p></section><section class="print-page" id="nwchem-architecture" heading-number="4.2"><h1 id="nwchem-architecture-nwchem-architecture">NWChem Architecture<a class="headerlink" href="#nwchem-architecture-nwchem-architecture" title="Permanent link">&para;</a></h1>
<p>As described in the <a href="#getting-started">Getting Started</a> section, NWChem consists of independent modules that perform the
various functions of the code. Examples include the input parser,
self-consistent field (SCF) energy, SCF analytic gradient, and density
functional theory (DFT) energy modules. The independent NWChem modules
can share data only through a disk-resident database, which is similar
to the GAMESS-UK dumpfile or the Gaussian checkpoint file. This allows
the modules to share data, or to share access to files containing data.</p>
<p>It is not necessary for the user to be intimately familiar with the
contents of the
<a href="#nwchem-architecture-database-structure">database</a> in order
to run NWChem. However, a nodding acquaintance with the design of the
code will help in clarifying the logic behind the input requirements,
especially when restarting jobs or performing multiple tasks within one
job.</p>
<p>As detailed in the section describing the (<a href="#getting-started-input-file-structure">input file
structure</a>), all
start-up directives are processed at the beginning of the job by the
main program, and then the input module is invoked. Each input directive
usually results in one or more entries being made in the database. When
a TASK directive is encountered, control is passed to the appropriate
module, which extracts relevant data from the database and any
associated files. Upon completion of the task, the module will store
significant results in the database, and may also modify other database
entries in order to affect the behavior of subsequent computations.</p>
<h2 id="nwchem-architecture-database-structure">Database Structure<a class="headerlink" href="#nwchem-architecture-database-structure" title="Permanent link">&para;</a></h2>
<p>Data is shared between modules of NWChem by means of the database. Three
main types of information are stored in the data base: (1) arrays of
data, (2) names of files that contain data, and (3) objects. Arrays are
stored directly in the database, and contain the following information:</p>
<ol>
<li>the name of the array, which is a string of ASCII characters (e.g.,
    &ldquo;reference energies&rdquo;)</li>
<li>the type of the data in the array (i.e., real, integer, logical, or
    character)</li>
<li>the number (N) of data items in the array (Note: A scalar is stored
    as an array of unit length.)</li>
<li>the N items of data of the specified type</li>
</ol>
<p>It is possible to enter data directly into the database using the <a href="#set">SET
directive</a>. For example, to store a (64-bit precision)
three-element real array with the name &ldquo;reference energies&rdquo; in the
database, the directive is as follows:</p>
<p><code>set "reference energies" 0.0 1.0 -76.2</code></p>
<p>NWChem determines the data to be real (based on the type of the first
element, 0.0), counts the number of elements in the array, and enters
the array into the database.</p>
<p>Much of the data stored in the database is internally managed by NWChem
and should not be modified by the user. However, other data, including
some NWChem input options, can be freely modified.</p>
<p>Objects are built in the database by storing associated data as multiple
entries, using an internally consistent naming convention. This data is
managed exclusively by the subroutines (or methods) that are associated
with the object. Currently, the code has two main objects: basis sets
and geometries. <a href="#geometry">GEOMETRY</a> and
<a href="#basis">BASIS</a> present a complete discussion of the input to
describe these objects.</p>
<p>As an illustration of what comprises a geometry object, the following
table contains a partial listing of the database contents for a water
molecule geometry named &ldquo;test geom&rdquo;. Each entry contains the field test
geom, which is the unique name of the object.</p>
<pre><code>Contents of RTDB h2o.db  
-----------------------  
Entry                                   Type[nelem]  
---------------------------  ----------------------  
geometry:test geom:efield             double[3]      
geometry:test geom:coords             double[9]      
geometry:test geom:ncenter               int[1]      
geometry:test geom:charges            double[3]      
geometry:test geom:tags                 char[6] 
...
</code></pre>
<p>Using this convention, multiple instances of objects may be stored with
different names in the same database. For example, if a user needed to
do calculations considering alternative geometries for the water
molecule, an input file could be constructed containing all the
geometries of interest by storing them in the database under different
names.</p>
<p>The runtime database contents for the file h2o.db listed above were
generated from the user-specified input directive,</p>
<pre><code> geometry &quot;test geom&quot;  
   O     0.00000000    0.00000000    0.00000000 
   H     0.00000000    1.43042809   -1.10715266 
   H     0.00000000   -1.43042809   -1.10715266  
 end
</code></pre>
<p>The <a href="#geometry">GEOMETRY</a> directive allows the user to specify
the coordinates of the atoms (or centers), and identify the geometry
with a unique name.</p>
<p>Unless a specific name is defined for the geometry, (such as the name
<code>"test geom"</code> shown in the example), the default name of geometry is
assigned. This is the geometry name that computational modules will look
for when executing a calculation. The SET directive can be used in the
input to force NWChem to look for a geometry with a name other than
geometry. For example, to specify use of the geometry with the name
<code>"test geom"</code> in the example above, the SET directive is as follows:</p>
<pre><code>set geometry &quot;test geom&quot;
</code></pre>
<p>NWChem will automatically check for such indirections when loading
geometries. Storage of data associated with basis sets, the other
database resident object, functions in a similar fashion, using the
default name <code>"ao basis"</code>.</p>
<h2 id="nwchem-architecture-persistence-of-data-and-restart">Persistence of data and restart<a class="headerlink" href="#nwchem-architecture-persistence-of-data-and-restart" title="Permanent link">&para;</a></h2>
<p>The database is persistent, meaning that all input data and output data
(calculation results) that are not destroyed in the course of execution
are permanently stored. These data are therefore available to subsequent
tasks or jobs. This makes the input for restart jobs very simple, since
only new or changed data must be provided. It also makes the behavior of
successive restart jobs identical to that of multiple tasks within one
job.</p>
<p>Sometimes, however, this persistence is undesirable, and it is necessary
to return an NWChem module to its default behavior by restoring the
database to its input-free state. In such a case, the
<a href="#unset">UNSET</a> directive can be used to delete all database
entries associated with a given module (including both inputs and
outputs).</p></section></section><section class="print-page" id="capabilities" heading-number="5"><h1 id="capabilities-functionalities">Functionalities<a class="headerlink" href="#capabilities-functionalities" title="Permanent link">&para;</a></h1>
<h2 id="capabilities-overview">Overview<a class="headerlink" href="#capabilities-overview" title="Permanent link">&para;</a></h2>
<p>NWChem provides many methods for computing the properties of molecular
and periodic systems using standard quantum mechanical descriptions of
the electronic wavefunction or density. Its classical molecular dynamics
capabilities provide for the simulation of macromolecules and solutions,
including the computation of free energies using a variety of force
fields. These approaches may be combined to perform mixed
quantum-mechanics and molecular-mechanics simulations.</p>
<p>The specific methods for determining molecular electronic structure,
molecular dynamics, and pseudopotential plane-wave electronic structure
and related attributes are listed in the following sections.</p>
<h2 id="capabilities-molecular-electronic-structure">Molecular Electronic Structure<a class="headerlink" href="#capabilities-molecular-electronic-structure" title="Permanent link">&para;</a></h2>
<p>Methods for determining energies and analytic first derivatives with
respect to atomic coordinates include the following:</p>
<ul>
<li>Hartree-Fock (RHF, UHF, high-spin ROHF)</li>
<li>Gaussian orbital-based density functional theory (DFT) using many
    local and non-local exchange-correlation potentials (LDA, LSDA)</li>
<li>second-order perturbation theory (MP2) with RHF and UHF references</li>
<li>complete active space self-consistent field theory (CASSCF).</li>
</ul>
<p>Analytic second derivatives with respect to atomic coordinates are
available for RHF and UHF, and closed-shell DFT with all functionals.</p>
<p>The following methods are available to compute energies only:</p>
<ul>
<li>iterative CCSD, CCSDT, and CCSDTQ methods and their EOM-CC
    counterparts for RHF, ROHF, and UHF references</li>
<li>active-space CCSDt and EOM-CCSDt approaches</li>
<li>completely renormalized CR-CCSD(T), and CR-EOM-CCSD(T) correction to
    EOM-CCSD excitation energies</li>
<li>locally renormalized CCSD(T) and CCSD(TQ) approaches</li>
<li>non-iterative approaches based on similarity transformed
    Hamiltonian: the CCSD(2)T and CCSD(2) formalisms.</li>
<li>MP2 with RHF reference and resolution of the identity integral
    approximation MP2 (RI-MP2) with RHF and UHF references</li>
<li>selected CI with second-order perturbation correction.</li>
</ul>
<p>For all methods, the following may be performed:</p>
<ul>
<li>single point energy calculations</li>
<li>geometry optimization with constraints (minimization and transition
    state)</li>
<li>molecular dynamics on the fully ab initio potential energy surface</li>
<li>automatic computation of numerical first and second derivatives</li>
<li>normal mode vibrational analysis in Cartesian coordinates</li>
<li>ONIOM hybrid calculations</li>
<li>Conductor-Like Screening Model (COSMO) calculations</li>
<li>electrostatic potential from fit of atomic partial charges</li>
<li>spin-free one-electron Douglas-Kroll calculations</li>
<li>electron transfer (ET)</li>
<li>vibrational SCF and DFT.</li>
</ul>
<p>At the SCF and DFT level of theory various (response) properties are
available, including NMR shielding tensors and indirect spin-spin
coupling.</p>
<h2 id="capabilities-quantum-mechanicsmolecular-mechanics-qmmm">Quantum Mechanics/Molecular Mechanics (QM/MM)<a class="headerlink" href="#capabilities-quantum-mechanicsmolecular-mechanics-qmmm" title="Permanent link">&para;</a></h2>
<p>The QM/MM module in NWChem provides a comprehensive set of capabilities
to study ground and excited state properties of large-molecular systems.
The QM/MM module can be used with practically any quantum mechanical
method available in NWChem. The following tasks are supported</p>
<ul>
<li>single point energy and property calculations</li>
<li>excited states calculation</li>
<li>optimizations and transition state search</li>
<li>dynamics</li>
<li>free energy calculations.</li>
</ul>
<h2 id="capabilities-pseudopotential-plane-wave-electronic-structure">Pseudopotential Plane-Wave Electronic Structure<a class="headerlink" href="#capabilities-pseudopotential-plane-wave-electronic-structure" title="Permanent link">&para;</a></h2>
<p>The NWChem Plane-Wave (NWPW) module uses pseudopotentials and plane-wave
basis sets to perform DFT calculations. This method&rsquo;s efficiency and
accuracy make it a desirable first principles method of simulation in
the study of complex molecular, liquid, and solid-state systems.
Applications for this first principles method include the calculation of
free energies, search for global minima, explicit simulation of solvated
molecules, and simulations of complex vibrational modes that cannot be
described within the harmonic approximation.</p>
<p>The NWPW module is a collection of three modules:</p>
<ul>
<li>PSPW (PSeudopotential Plane-Wave) A gamma point code for calculating
    molecules, liquids, crystals, and surfaces.</li>
<li>Band A band structure code for calculating crystals and surfaces
    with small band gaps (e.g. semi-conductors and metals).</li>
<li>PAW (Projector Augmented Wave) a gamma point projector augmented
    plane-wave code for calculating molecules, crystals, and surfaces.</li>
</ul>
<p>These capabilities are available:</p>
<ul>
<li>constant energy and constant temperature Car-Parrinello molecular
    dynamics (extended Lagrangian dynamics)</li>
<li>LDA, PBE96, and PBE0, exchange-correlation potentials (restricted
    and unrestricted)</li>
<li>SIC, pert-OEP, Hartree-Fock, and hybrid functionals (restricted and
    unrestricted)</li>
<li>Hamann, Troullier-Martins, Hartwigsen-Goedecker-Hutter
    norm-conserving pseudopotentials with semicore corrections</li>
<li>geometry/unit cell optimization, frequency, transition-states</li>
<li>fractional occupation of molecular orbitals for metals</li>
<li>AIMD/MM capability in PSPW</li>
<li>constraints needed for potential of mean force (PMF) calculation</li>
<li>wavefunction, density, electrostatic, Wannier plotting</li>
<li>band structure and density of states generation</li>
</ul>
<h2 id="capabilities-molecular-dynamics">Molecular Dynamics<a class="headerlink" href="#capabilities-molecular-dynamics" title="Permanent link">&para;</a></h2>
<p>The NWChem Molecular Dynamics (MD) module can perform classical
simulations using the AMBER and CHARMM force fields, quantum dynamical
simulations using any of the quantum mechanical methods capable of
returning gradients, and mixed quantum mechanics molecular dynamics
simulation and molecular mechanics energy minimization.</p>
<p>Classical molecular simulation functionality includes the following
methods:</p>
<ul>
<li>single configuration energy evaluation</li>
<li>energy minimization</li>
<li>molecular dynamics simulation</li>
<li>free energy simulation (MCTI and MSTP with single or dual
    topologies, double-wide sampling, and separation-shifted scaling).</li>
</ul>
<p>The classical force field includes the following elements:</p>
<ul>
<li>effective pair potentials</li>
<li>first-order polarization</li>
<li>self-consistent polarization</li>
<li>smooth particle mesh Ewald</li>
<li>twin-range energy and force evaluation</li>
<li>periodic boundary conditions</li>
<li>SHAKE constraints</li>
<li>constant temperature and/or pressure ensembles</li>
<li>dynamic proton hopping using the Q-HOP methodology</li>
<li>advanced system setup capabilities for biomolecular membranes.</li>
</ul></section><section class="print-page" id="top-level" heading-number="6"><h1 id="top-level-top-level-directives">Top-level Directives<a class="headerlink" href="#top-level-top-level-directives" title="Permanent link">&para;</a></h1>
<h2 id="top-level-overview">Overview<a class="headerlink" href="#top-level-overview" title="Permanent link">&para;</a></h2>
<p>Top-level directives are directives that can affect all modules in the code. Some specify molecular properties or other data that should apply to all subsequent calculations with the current database. However, most top-level directives provide the user with the means to manage the resources for a calculation and to start computations. As the first step in the execution of a job, NWChem scans the entire input file looking for start-up directives, which NWChem must process before all other input. The input file is then rewound and processed sequentially, and each directive is processed in the order in which it is encountered. In this second pass, start-up directives are ignored.</p>
<p>The following sections describe each of the top-level directives in detail, noting all keywords, options, required input, and defaults.</p>
<ul>
<li>
<p><a href="#start_restart">START/RESTART</a></p>
</li>
<li>
<p><a href="#permanent_dir">PERMANENT_DIR</a></p>
</li>
<li>
<p><a href="#scratch_dir">SCRATCH_DIR</a></p>
</li>
<li>
<p><a href="#memory">MEMORY</a></p>
</li>
<li>
<p><a href="#echo">ECHO</a></p>
</li>
<li>
<p><a href="#title">TITLE</a></p>
</li>
<li>
<p><a href="#print_noprint">PRINT / NOPRINT</a></p>
</li>
<li>
<p><a href="#set">SET</a></p>
</li>
<li>
<p><a href="#unset">UNSET</a></p>
</li>
<li>
<p><a href="#stop">STOP</a></p>
</li>
<li>
<p><a href="#task">TASK</a></p>
</li>
<li>
<p><a href="#ecce_print">ECCE_PRINT</a></p>
</li>
</ul></section><section class="print-page" id="geometry" heading-number="7"><h1 id="geometry-geometries">Geometries<a class="headerlink" href="#geometry-geometries" title="Permanent link">&para;</a></h1>
<h2 id="geometry-overview">Overview<a class="headerlink" href="#geometry-overview" title="Permanent link">&para;</a></h2>
<p>The GEOMETRY directive is a compound directive that allows the user to
define the geometry to be used for a given calculation. The directive
allows the user to specify the geometry with a relatively small amount
of input, but there are a large number of optional keywords and
additional subordinate directives that the user can specify, if needed.
The directive therefore appears to be rather long and complicated when
presented in its general form, as
follows:</p>
<pre><code> GEOMETRY [&lt;string name default geometry&gt;] \  
          [units &lt;string units default angstroms&gt;] \  
          [(angstrom_to_au || ang2au) \  
                 &lt;real angstrom_to_au default 1.8897265&gt;] \  
          [print [xyz] || noprint] \  
          [center || nocenter] \  
          [bqbq] \  
          [autosym [real tol default 1d-2] || noautosym] \  
          [autoz || noautoz] \  
          [adjust] \  
          [(nuc || nucl || nucleus) &lt;string nucmodel&gt;]  
   [SYMMETRY [group] &lt;string group_name&gt; [print] \  
          [tol &lt;real tol default 1d-2&gt;]]  
   [ [LOAD] [format xyz||pdb]  [frame &lt;int frame&gt;] \  
          [select [not] \  
               [name &lt;string atomname&gt;] \  
               [rname &lt;string residue-name&gt;]  
               [id  &lt;int atom-id&gt;|&lt;int range atom-id1:atom-id2&gt; ... ]  
               [resi &lt;int residue-id&gt;|&lt;int range residue-id1:residue-id2&gt; ... ]  
         ]  
   &lt;string filename&gt; ]  

   &lt;string tag&gt; &lt;real x y z&gt; [vx vy vz] [charge &lt;real charge&gt;] \  
          [mass &lt;real mass&gt;] \  
          [(nuc || nucl || nucleus) &lt;string nucmodel&gt;]  
   ... ]  
   [ZMATRIX || ZMT || ZMAT  
        &lt;string tagn&gt; &lt;list_of_zmatrix_variables&gt;  
        ...   
        [VARIABLES  
             &lt;string symbol&gt; &lt;real value&gt;  
             ... ]  
        [CONSTANTS  
             &lt;string symbol&gt; &lt;real value&gt;  
             ... ]  
    (END || ZEND)]  
    [ZCOORD  
         CVR_SCALING &lt;real value&gt;  
         BOND    &lt;integer i&gt; &lt;integer j&gt; \  
                 [&lt;real value&gt;] [&lt;string name&gt;] [constant]  
         ANGLE   &lt;integer i&gt; &lt;integer j&gt; \  
                     [&lt;real value&gt;] [&lt;string name&gt;] [constant]  
         TORSION &lt;integer i&gt; &lt;integer j&gt; &lt;integer k&gt; &lt;integer l&gt; \  
                 [&lt;real value&gt;] [&lt;string name&gt;] [constant]  
     END]  

     [SYSTEM surface  &lt;molecule polymer surface crystal default molecule&gt;  
          lat_a &lt;real lat_a&gt; lat_b &lt;real lat_b&gt; lat_c &lt;real lat_c&gt;  
          alpha &lt;real alpha&gt; beta &lt;real beta&gt; gamma &lt;real gamma&gt;  
     END]  
  END
</code></pre>
<p>The three main parts of the GEOMETRY directive are:</p>
<ul>
<li>keywords on the first line of the directive (to specify such
    optional input as the geometry name, input units, and print level
    for the output)</li>
<li>symmetry information</li>
<li>Cartesian coordinates or Z-matrix input to specify the locations of
    the atoms and centers</li>
<li>lattice parameters (needed only for periodic systems)</li>
</ul>
<p>The following sections present the input for this compound directive in
detail, describing the options available and the usages of the various
keywords in each of the three main parts.</p>
<ul>
<li>
<p><a href="#keywords-for-the-geometry-directive">Keywords for the GEOMETRY directive</a></p>
</li>
<li>
<p><a href="#symmetry----symmetry-group-input">SYMMETRY: Symmetry Group Input</a></p>
</li>
<li>
<p><a href="#names-of-3-dimensional-space-groups">Names of 3-dimensional space groups</a></p>
</li>
<li>
<p><a href="#cartesian-coordinate-input">Cartesian coordinate input</a></p>
</li>
<li>
<p><a href="#zmatrix">ZMATRIX: Z-matrix input</a></p>
</li>
<li>
<p><a href="#zcoord-forcing-internal-coordinates">ZCOORD: Forcing internal coordinates</a></p>
</li>
<li>
<p><a href="#system----lattice-parameters-for-periodic-systems">SYSTEM: Lattice parameters for periodic systems</a></p>
</li>
<li>
<p><a href="#geometry-load">LOAD: Load geometry from XYZ file</a></p>
</li>
</ul>
<!-- end list --></section><section class="print-page" id="basis" heading-number="8"><h1 id="basis-basis-sets">Basis sets<a class="headerlink" href="#basis-basis-sets" title="Permanent link">&para;</a></h1>
<h2 id="basis-overview">Overview<a class="headerlink" href="#basis-overview" title="Permanent link">&para;</a></h2>
<p>NWChem currently supports basis sets consisting of generally contracted
Cartesian Gaussian functions up to a maximum angular momentum of six (<em>h</em>
functions), and also <em>sp</em> (or L) functions. The BASIS directive is used to
define these, and also to specify use of an <a href="#ecp">Effective Core
Potential</a> that is associated with a
basis set.</p>
<p>The basis functions to be used for a given calculation can be drawn from
a standard set in the <a href="#basis-basis-set-library">Basis set
library</a> that is included in the release
of NWChem. Alternatively, the user can specify particular functions
explicitly in the input, to define a particular basis set.</p>
<h2 id="basis-basis-directive">BASIS directive<a class="headerlink" href="#basis-basis-directive" title="Permanent link">&para;</a></h2>
<p>The general form of the BASIS directive is as follows:</p>
<pre><code> BASIS [&lt;string name default &quot;ao basis&quot;&gt;] \  
       [(spherical || cartesian) default cartesian] \  
       [(print || noprint) default print]  \
       [rel] [bse] 
    &lt;string tag&gt; library [&lt;string tag_in_lib&gt;] \  
                 &lt;string standard_set&gt; [file &lt;filename&gt;] \  
                 [except &lt;string tag list&gt;] [rel]  
       ...  
    &lt;string tag&gt; &lt;string shell_type&gt; [rel]  
       &lt;real exponent&gt; &lt;real list_of_coefficients&gt;  
       ...  
 END
</code></pre>
<p>The following sections examine the keywords on the first line of the <code>BASIS</code> directive:</p>
<h3 id="basis-basis-set-name">Basis set NAME<a class="headerlink" href="#basis-basis-set-name" title="Permanent link">&para;</a></h3>
<p><code>NAME</code> :  <br />
    By default, the basis set is stored in the database with the name
    &ldquo;ao basis&rdquo;. Another name may be specified in the BASIS directive,
    thus, multiple basis sets may be stored simultaneously in the
    database. Also, the
    <a href="#density-functional-theory-for-molecules">DFT</a>
    and <a href="#mp2">RI-MP2</a> modules and the
    <a href="#relativistic-all-electron-approximations-dyalls-modified-dirac-hamitonian-approximation">Dyall-modified-Dirac</a>
    relativistic method require multiple basis sets with specific names.
    The user can associate the &ldquo;ao basis&rdquo; with another named basis using
    the SET directive (see <a href="#set">SET</a>).</p>
<h3 id="basis-spherical-or-cartesian">SPHERICAL or CARTESIAN<a class="headerlink" href="#basis-spherical-or-cartesian" title="Permanent link">&para;</a></h3>
<p><code>SPHERICAL</code> || <code>CARTESIAN</code> : <br />
    The keywords <code>spherical</code> and <code>cartesian</code> offer the option of using
    either spherical-harmonic (5 <em>d</em>, 7 <em>f</em>, 9 <em>g</em>, &hellip;) or Cartesian (6 <em>d</em>, 10
    <em>f</em>, 15 <em>g</em>, &hellip;) angular functions. The default is Cartesian. Note that
    the correlation-consistent basis sets were designed using spherical
    harmonics and to use these, the <code>spherical</code> keyword should be present
    in the <code>BASIS</code> directive. The use of spherical functions also helps
    eliminate problems with linear dependence.</p>
<table>
<thead>
<tr>
<th></th>
<th>Cartesian</th>
<th>Spherical</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>s</td>
<td>s</td>
</tr>
<tr>
<td>1</td>
<td>p<sub>x</sub></td>
<td>p<sub>x</sub></td>
</tr>
<tr>
<td>2</td>
<td>p<sub>y</sub></td>
<td>p<sub>y</sub></td>
</tr>
<tr>
<td>3</td>
<td>p<sub>z</sub></td>
<td>p<sub>z</sub></td>
</tr>
<tr>
<td>1</td>
<td>d<sub>xx</sub></td>
<td>d<sub>xy</sub></td>
</tr>
<tr>
<td>2</td>
<td>d<sub>xy</sub></td>
<td>d<sub>yz</sub></td>
</tr>
<tr>
<td>3</td>
<td>d<sub>xz</sub></td>
<td>d<sub>z<sup>2</sup>-x<sup>2</sup>-y<sup>2</sup></sub></td>
</tr>
<tr>
<td>4</td>
<td>d<sub>yy</sub></td>
<td>d<sub>xz</sub></td>
</tr>
<tr>
<td>5</td>
<td>d<sub>yz</sub></td>
<td>d<sub>x<sup>2</sup>-y<sup>2</sup></sub></td>
</tr>
<tr>
<td>6</td>
<td>d<sub>zz</sub></td>
<td></td>
</tr>
<tr>
<td>1</td>
<td>f<sub>xxx</sub></td>
<td>f<sub>xxy-yyy</sub></td>
</tr>
<tr>
<td>2</td>
<td>f<sub>xxy</sub></td>
<td>f<sub>xyz</sub></td>
</tr>
<tr>
<td>3</td>
<td>f<sub>xxz</sub></td>
<td>f<sub>yzz-xxy-yyy</sub></td>
</tr>
<tr>
<td>4</td>
<td>f<sub>xyy</sub></td>
<td>f<sub>zzz-xxz-yyz</sub></td>
</tr>
<tr>
<td>5</td>
<td>f<sub>xyz</sub></td>
<td>f<sub>-xzz+xxx+xyy</sub></td>
</tr>
<tr>
<td>6</td>
<td>f<sub>xzz</sub></td>
<td>f<sub>xxz-yyz</sub></td>
</tr>
<tr>
<td>7</td>
<td>f<sub>yyy</sub></td>
<td>f<sub>xyy-xxx</sub></td>
</tr>
<tr>
<td>8</td>
<td>f<sub>yyz</sub></td>
<td></td>
</tr>
<tr>
<td>9</td>
<td>f<sub>yzz</sub></td>
<td></td>
</tr>
<tr>
<td>10</td>
<td>f<sub>zzz</sub></td>
<td></td>
</tr>
</tbody>
</table>
<p>Order of functions.<br />
<br></p>
<h3 id="basis-print-keyword">PRINT keyword<a class="headerlink" href="#basis-print-keyword" title="Permanent link">&para;</a></h3>
<p><code>PRINT</code> or <code>NOPRINT</code> :<br />
    The default is for the input module to print all basis sets
    encountered. Specifying the keyword noprint allows the user to
    suppress this output.</p>
<h3 id="basis-rel-keyword">REL keyword<a class="headerlink" href="#basis-rel-keyword" title="Permanent link">&para;</a></h3>
<p><code>REL</code> :<br />
    This keyword marks the entire basis as a relativistic basis for the
    purposes of the Dyall-modified-Dirac relativistic integral code. The
    marking of the basis set is necessary for the code to make the
    proper association between the relativistic shells in the ao basis
    and the shells in the large and/or small component basis. This is
    only necessary for basis sets which are to be used as the ao basis.
    The user is referred to the section on <a href="#relativistic-all-electron-approximations-dyalls-modified-dirac-hamitonian-approximation">Dyall&rsquo;s modified
    Dirac-Hamiltonian
    approximation</a>
    for more details.</p>
<h3 id="basis-bse-keyword">BSE keyword<a class="headerlink" href="#basis-bse-keyword" title="Permanent link">&para;</a></h3>
<p><code>BSE</code> <strong><em>New in NWChem 7.2.0</em></strong>:<br />
    This keyword loads the basis set <a href="#basis-basis-set-library">library</a> using data in <code>$NWCHEM_TOP/src/basis/libraries.bse</code>
    from <a href="#basis-how-to-use-basis-files-from-httpswwwbasissetexchangeorg-new-in-2019">basisetexchanger.org</a>.
    CAVEAT: use of this keyword will also use the
    <a href="#basis-spherical-or-cartesian">spherical/cartesian</a>
    keywords from the basis library files.</p>
<h2 id="basis-basis-sets-tags">Basis sets tags<a class="headerlink" href="#basis-basis-sets-tags" title="Permanent link">&para;</a></h2>
<p>Basis sets are associated with centers by using the tag of a center in a
<a href="#geometry">geometry</a> that has either been input by
the user or is available elsewhere. Each atom or center with the same
tag will have the same basis set. All atoms must have basis functions
assigned to them &ndash; only dummy centers (<code>X</code> or <code>Bq</code>) may have no basis
functions. To facilitate the specification of the geometry and the basis
set for any chemical system, the matching process of a basis set tag to
a geometry tag first looks for an exact match. If no match is found,
NWChem will attempt to match, ignoring case, the name or symbol of the
element. E.g., all hydrogen atoms in a system could be labeled &ldquo;<code>H1</code>&rdquo;,
&ldquo;<code>H2</code>&rdquo;, &hellip;, in the geometry but only one basis set specification for &ldquo;<code>H</code>&rdquo;
or &ldquo;<code>hydrogen</code>&rdquo; is necessary. If desired, a special basis may be added to
one or more centers (e.g., &ldquo;<code>H1</code>&rdquo;) by providing a basis for that tag. If
the matching mechanism fails then NWChem stops with an appropriate error
message.</p>
<p>A special set of tags, &ldquo;<code>*</code>&rdquo; and tags ending with a &ldquo;<code>*</code>&rdquo; (E.g. &ldquo;<code>H*</code>&rdquo;) can
be used in combination with the keyword
<a href="#basis-basis-set-library">library</a>. These tags facilitate the
definition of a certain type of basis set of all atoms, or a group of
atoms, in a geometry using only a single or very few basis set entries.
The &ldquo;<code>*</code>&rdquo; tag will not place basis sets on dummy atoms, <code>Bq*</code> can be used
for that if necessary.</p>
<p>Examined next is how to reference standard basis sets in the basis set
library, and finally, how to define a basis set using exponents and
coefficients.</p>
<h2 id="basis-basis-set-library">Basis set library<a class="headerlink" href="#basis-basis-set-library" title="Permanent link">&para;</a></h2>
<p>The keyword <code>library</code> associated with each specific tag entry specifies
that the calculation will use the standard basis set in NWChem for that
center. The string <code>&lt;standard_set&gt;</code> is the name that identifies the
functions in the library. The names of standard basis sets are not case
sensitive. For a complete list of basis sets and associated ECPs in the
NWChem library see the <a href="#availablebasissets">available basis
sets</a> or the <a href="https://www.basissetexchange.org/">Basis Set
Exchange</a> for naming conventions and
their specifications.</p>
<p>The general form of the input line requesting basis sets from the NWChem
basis set library is:</p>
<pre><code>    &lt;string tag&gt; library [&lt;string tag_in_lib&gt;] \  
                 &lt;string standard set&gt; [file &lt; filename&gt; \  
                 [except &lt;string tag list&gt;] [rel] 
       ...
</code></pre>
<p>For example, the NWChem basis set library contains the Dunning cc-pvdz
basis set. These may be used as follows</p>
<pre><code> basis  
   oxygen library cc-pvdz  
   hydrogen library cc-pvdz  
 end
</code></pre>
<p>A default path of the NWChem basis set libraries is provided on
installation of the code, but a different path can be defined by
specifying the keyword file, and one can explicitly name the file to be
accessed for the basis functions. For example,</p>
<pre><code> basis 
   o  library 3-21g file /usr/d3g681/nwchem/library 
   si library 6-31g file /usr/d3g681/nwchem/libraries/  
 end
</code></pre>
<p>This directive tells the code to use the basis set 3-21g in the file
/usr/d3g681/nwchem/library for atom o and to use the basis set 6-31g in
the directory /usr/d3g681/nwchem/libraries/ for atom si, rather than
look for them in the default libraries. When a directory is defined the
code will search for the basis set in a file with the name 6-31g.</p>
<p>The &ldquo;<code>*</code>&rdquo; tag can be used to efficiently define basis set input
directives for large numbers of atoms. An example is:</p>
<pre><code> basis  
   *  library 3-21g 
 end
</code></pre>
<p>This directive tells the code to assign the basis sets 3-21g to all the
atom tags defined in the geometry. If one wants to place a different
basis set on one of the atoms defined in the geometry, the following
directive can be used:</p>
<pre><code> basis  
   *  library 3-21g except H 
 end
</code></pre>
<p>This directive tells the code to assign the basis sets 3-21g to all the
atoms in the geometry, except the hydrogen atoms. Remember that the user
will have to explicitly define the hydrogen basis set in this
directive! One may also define tags that end with a &ldquo;<code>*</code>&rdquo;:</p>
<pre><code> basis  
   oxy*  library 3-21g   
 end
</code></pre>
<p>This directive tells the code to assign the basis sets 3-21g to all atom
tags in the geometry that start with &ldquo;<code>oxy</code>&rdquo;.</p>
<p>If standard basis sets are to be placed upon a dummy center, the
variable <code>&lt;tag_in_lib&gt;</code> must also be entered on this line, to identify the
correct atom type to use from the basis function library (see the ghost
atom example in <a href="#set">SET</a> and below).
For example: To specify the cc-pvdz basis for a calculation on the water
monomer in the dimer basis, where the dummy oxygen and dummy hydrogen
centers have been identified as <code>bqo</code> and <code>bqh</code> respectively, the <code>BASIS</code>
directive is as follows:</p>
<pre><code> basis  
   o   library cc-pvdz  
   h   library cc-pvdz  
   bqo library o cc-pvdz  
   bqh library h cc-pvdz  
 end
</code></pre>
<p>A special dummy center tag is <code>bq*</code>, which will assign the same basis set
to all bq centers in the geometry. Just as with the <code>*</code> tag, the except
list can be used to assign basis sets to unique dummy centers.</p>
<p>The library basis sets can also be marked as relativistic by adding the
<code>rel</code> keyword to the tag line. See the section on <a href="#relativistic-all-electron-approximations">relativistic
all-electron
approximations</a>
 for more details. The correlation consistent basis sets have been
contracted for relativistic effects and are included in the standard
library.</p>
<p>There are also contractions in the standard library for both a point
nucleus and a finite nucleus of Gaussian shape. These are usually
distinguished by the suffix <code>_pt</code> and <code>_fi</code>. It is the user&rsquo;s
responsibility to ensure that the contraction matches the nuclear type
specified in the geometry object. The specification of a finite nucleus
basis set does NOT automatically set the nuclear type for that atom to
be finite. See <a href="#geometry">Geometries</a> for
information.</p>
<h2 id="basis-how-to-use-basis-files-from-httpswwwbasissetexchangeorg-new-in-2019">How to use basis files from https://www.basissetexchange.org (NEW in 2019)<a class="headerlink" href="#basis-how-to-use-basis-files-from-httpswwwbasissetexchangeorg-new-in-2019" title="Permanent link">&para;</a></h2>
<p>In order to ensure compatibility with the existing basis libraries available in NWChem, we suggest the user  to 
select the &ldquo;Advanced Options&rdquo; menu and tick the boxes &ldquo;Optimize General Contractions&rdquo; and &ldquo;Uncontract General&rdquo;,
as in the image below,
when downloading basis files from <a href="https://www.basissetexchange.org">www.basissetexchange.org</a><br />
<img alt="How to download" src="BSE_NWChem_selection.png" /></p>
<p>As an alternative,  basis set files downloaded from the basissetexchange.org website are available in
the NWChem source code (after release 7.0.0). In order to switch from the default basis libraries to the library formed by files downloaded from <a href="https://www.basissetexchange.org">www.basissetexchange.org</a>,
the following environment variable setting is required</p>
<pre><code>NWCHEM_BASIS_LIBRARY=$NWCHEM_TOP/src/basis/libraries.bse/
</code></pre>
<h2 id="basis-explicit-basis-set-definition">Explicit basis set definition<a class="headerlink" href="#basis-explicit-basis-set-definition" title="Permanent link">&para;</a></h2>
<p>If the basis sets in the library or available in other external files
are not suitable for a given calculation, the basis set may be
explicitly defined. A generally contracted Gaussian basis function is
associated with a center using an input line of the following form:</p>
<pre><code>    &lt;string tag&gt; &lt;string shell_type&gt; [rel]  
       &lt;real exponent&gt; &lt;real list_of_coefficients&gt;  
       ...
</code></pre>
<p>The variable <shell_type> identifies the angular momentum of the shell,
<em>s</em>, <em>p</em>, <em>d</em>, .... NWChem is configured to handle up to <em>h</em> shells. The
keyword <code>rel</code> marks the shell as relativistic &ndash; see the Section  on <a href="#relativistic-all-electron-approximations">relativistic
all-electron
approximations</a> for more
details. Subsequent lines define the primitive function exponents and
contraction coefficients. General contractions are specified by
including multiple columns of coefficients.</p>
<p>The following example defines basis sets for the water molecule:</p>
<pre><code> basis spherical   
   oxygen s  
     11720.0000    0.000710  -0.000160  
      1759.0000    0.005470  -0.001263  
       400.8000    0.027837  -0.006267  
       113.7000    0.104800  -0.025716  
        37.0300    0.283062  -0.070924  
        13.2700    0.448719  -0.165411  
         5.0250    0.270952  -0.116955  
         1.0130    0.015458   0.557368  
         0.3023   -0.002585   0.572759  
   oxygen s                  
         0.3023    1.000000  
   oxygen p                  
        17.7000    0.043018  
         3.8540    0.228913  
         1.0460    0.508728  
         0.2753    0.460531  
   oxygen p                  
         0.2753    1.000000  
   oxygen d  
         1.1850    1.000000  
   hydrogen s  
        13.0100    0.019685  
         1.9620    0.137977  
         0.4446    0.478148  
         0.1220    0.501240  
   hydrogen s    
         0.1220    1.000000  
   hydrogen p    
         0.7270    1.000000  
   oxygen s  
         0.01      1.0  
   hydrogen s  
         0.02974   1.0  
   hydrogen p  
         0.141      1.0  
 end
</code></pre>
<p>Explicit basis set specifications are available from the <a href="https://www.basissetexchange.org/">basis set
exchange</a>. </p>
<h2 id="basis-combinations-of-library-and-explicit-basis-set-input">Combinations of library and explicit basis set input<a class="headerlink" href="#basis-combinations-of-library-and-explicit-basis-set-input" title="Permanent link">&para;</a></h2>
<p>The user can use a mixture of library basis and explicit basis set input
to define the basis sets used on the various atoms.</p>
<p>For example, the following <code>BASIS</code> directive augments the Dunning cc-pvdz
basis set for the water molecule with a diffuse s-shell on oxygen and
adds the aug-cc-pVDZ diffuse functions onto the hydrogen.</p>
<pre><code> basis spherical   
   oxygen library cc-pvdz  
   hydrogen library cc-pvdz  
   oxygen s  
     0.01 1.0  
   hydrogen library &quot;aug-cc-pVDZ Diffuse&quot;  
 end
</code></pre>
<p>The resulting basis set defined is identical to the one defined above in
the explicit basis set input.</p></section><section class="print-page" id="ecp" heading-number="9"><h1 id="ecp-effective-core-potentials">Effective Core Potentials<a class="headerlink" href="#ecp-effective-core-potentials" title="Permanent link">&para;</a></h1>
<h2 id="ecp-overview">Overview<a class="headerlink" href="#ecp-overview" title="Permanent link">&para;</a></h2>
<p>Effective core potentials (ECPs) are a useful means of replacing the
core electrons in a calculation with an effective potential, thereby
eliminating the need for the core basis functions, which usually require
a large set of Gaussians to describe them. In addition to replacing the
core, they may be used to represent relativistic effects, which are
largely confined to the core. In this context, both the scalar
(spin-free) relativistic effects and spin-orbit (spin-dependent)
relativistic effects may be included in effective potentials. NWChem has
the facility to use both, and these are described in the next two
sections.</p>
<p>A brief recapitulation of the development of RECPs is given here,
following L.F. Pacios and P.A. Christiansen, J. Chem. Phys. 82, 2664 (1985). The process can be viewed as starting
from an atomic Dirac-Hartree-Fock calculation, done in jj coupling, and
producing relativistic effective potentials (REPs) for each <span class="arithmatex">\(l\)</span> and <span class="arithmatex">\(j\)</span>
value, <span class="arithmatex">\(U^{\rm REP}_{lj}\)</span>
which for example contains the Coulomb potential of the core electrons
balanced by the part of the nuclear attraction which cancels the core
electron charge. The residue is expressed in a semi-local form,</p>
<div class="arithmatex">\[U^{REP} = U^{REP}_{LJ}(r) + \sum_{l=0}^{L-1}  \sum_{j=|l-1/2|}^{l+1/2} \left[ U^{REP}_{lj}(r) - U^{REP}_{LJ}(r)] \right] \sum_{m=-j}^j \vert lj m \rangle \langle lj m \vert\]</div>
<p>where <span class="arithmatex">\(L\)</span> is one larger than the maximum angular momentum in the atom.
The scalar potential is obtained by averaging the REPs for each <span class="arithmatex">\(j\)</span> for
a given <span class="arithmatex">\(l\)</span> to give an averaged relativistic effective potential, or
AREP</p>
<div class="arithmatex">\[U^{AREP}_l(r) = \frac{1}{2l+1} \left[ lU^{REP}_{l-1/2}(r) + (l+1) U^{\rm REP}_{l+1/2}(r) \right]\]</div>
<p>These are summed into the full potential  </p>
<div class="arithmatex">\[U^{AREP}(r) = U^{AREP}_L(r) + \sum_{l=0}^L \sum_{m=-l}^l \left[ U^{AREP}_{l}(r) - U^{AREP}_{L}(r) \right] \vert l m \rangle \langle l m \vert\]</div>
<p>The spin-orbit potential is obtained from the difference between the
REPs for the two <span class="arithmatex">\(j\)</span> values for a given <span class="arithmatex">\(l\)</span>, and may be represented in
terms of an effective spin-orbit operator,</p>
<div class="arithmatex">\[H^{SO} = \mathbf{\hat{s}} \cdot \sum_{l=1}^{L-1} \frac{2}{2l+1} \Delta U^{REP}_l \sum_{mm'}\vert {lm} \rangle \langle{lm} \vert \mathbf{\hat{l}} \vert {lm'} \rangle \langle{lm'} \vert\]</div>
<p>where</p>
<div class="arithmatex">\[\Delta U^{REP}_{l} = U^{REP}_{l+1/2}(r) - U^{REP}_{l-1/2}(r)\]</div>
<p>The relavistic potential <span class="arithmatex">\(U^{REP}\)</span> is the sum of <span class="arithmatex">\(H^{SO}\)</span> and <span class="arithmatex">\(U^{AREP}\)</span>.  </p>
<p>The spin-orbit integrals generated by NWChem are the integrals over the
sum, including the factor of <span class="arithmatex">\(2/(2l+1)\)</span>
as an effective spin-orbit operator without further factors introduced.</p>
<p>The effective potentials, both scalar and spin-orbit, are fitted to
Gaussians with the form</p>
<div class="arithmatex">\[U_{l}(r) =  r^{-2} \sum_{k} A_{lk} r^{n_{lk}} e^{-B_{lk}r^{2}}\]</div>
<p>where <span class="arithmatex">\(A_{lk}\)</span> is the contraction coefficient, <span class="arithmatex">\(n_{lk}\)</span> is the
exponent of the <span class="arithmatex">\(r\)</span> term (r-exponent), and <span class="arithmatex">\(B_{lk}\)</span> is the Gaussian
exponent. The <span class="arithmatex">\(n_{lk}\)</span> exponent is shifted by 2, in accordance with most of the
ECP literature and implementations, i.e., <span class="arithmatex">\(n_{lk} = 0\)</span> implies
<span class="arithmatex">\(r^{-2}\)</span>. The current implementation allows <span class="arithmatex">\(n_{lk}\)</span>  values of only
0, 1, or 2.</p>
<h2 id="ecp-scalar-ecps">Scalar ECPs<a class="headerlink" href="#ecp-scalar-ecps" title="Permanent link">&para;</a></h2>
<p>The optional directive <code>ECP</code> allows the user to describe an effective core
potential (ECP) in terms of contracted Gaussian functions as given
above. Potentials using these functions must be specified explicitly by
user input in the <code>ECP</code> directive. This directive has essentially the same
form and properties as the standard <code>BASIS</code> directive, except for
essential differences required for ECPs. Because of this, the ECP is
treated internally as a basis set. The form of the input for the <code>ECP</code>
directive is as
follows:</p>
<pre><code> ECP [&lt;string name default &quot;ecp basis&quot;&gt;] \  
       [print || noprint default print]  
    &lt;string tag&gt; library [&lt;string tag_in_lib&gt;] \  
                 &lt;string standard_set&gt; [file &lt;filename&gt;] \  
                 [except&lt;string tag list&gt;]  
    &lt;string tag&gt; [nelec] &lt;integer number_of_electrons_replaced&gt;  
       ...  
    &lt;string tag&gt; &lt;string shell_type&gt;  
    &lt;real r-exponent&gt; &lt;real Gaussian-exponent&gt; &lt;real list_of_coefficients&gt;  
       ...  
 END
</code></pre>
<p>ECPs are automatically segmented, even if general contractions are
input. The projection operators defined in an ECP are spherical by
default, so there is no need to include the <code>CARTESIAN</code> or <code>SPHERICAL</code>
keyword as there is for a standard basis set. ECPs are associated with
centers in geometries through tags or names of centers. These tags must
match in the same manner as for basis sets the tags in a GEOMETRY and
ECP directives, and are limited to sixteen (16) characters. Each center
with the same tag will have the same ECP. By default, the input module
prints each ECP that it encounters. The <code>NOPRINT</code> option can be used to
disable printing. There can be only one active ECP, even though several
may exist in the input deck. The ECP modules load <code>ecp basis</code> inputs
along with any <code>ao basis</code> inputs present. ECPs may be used in both
energy and gradient calculations.</p>
<p>ECPs are named in the same fashion as geometries or regular basis sets,
with the default name being &ldquo;ecp basis&rdquo;. It should be clear from the
above discussion on geometries and database entries how indirection is
supported. All directives that are in common with the standard Gaussian
basis set input have the same function and syntax.</p>
<p>As for regular basis sets, ECPs may be obtained from the standard
library. For a complete list of basis sets and associated ECPs in the
NWChem library see the <a href="#availablebasissets">available basis
sets</a> or the <a href="https://www.basissetexchange.org/">Basis Set
Exchange</a> for naming conventions and
their specifications.</p>
<p>The keyword <code>nelec</code> allows the user to specify the number of core
electrons replaced by the ECP. Additional input lines define the
specific coefficients and exponents. The variable <code>&lt;shell_type&gt;</code> is used
to specify the components of the ECP. The keyword <code>ul</code> entered for
<code>&lt;shell_type&gt;</code> denotes the local part of the ECP. This is equivalent to
the highest angular momentum functions specified in the literature for
most ECPs. The standard entries (<em>s</em>, <em>p</em>, <em>d</em>, etc.) for <code>shell_type</code> specify
the angular momentum projector onto the local function. The shell type
label of <em>s</em> indicates the <em>ul-s</em> projector input, <em>p</em> indicates the <em>ul-p</em>,
etc.</p>
<p>For example, the Christiansen, Ross and Ermler ARECPs are available in
the standard basis set library named <code>crenbl_ecp</code>. To perform a
calculation on uranyl UO<sub>2</sub><sup>2+</sup> with all-electron oxygen
(aug-cc-pvdz basis), and uranium with an ARECP and using the
corresponding basis the following input can be used</p>
<pre><code> geometry
   U 0 0  0
   O 0 0  1.65
   O 0 0 -1.65
 end
 basis 
   U library crenbl_ecp
   O library aug-cc-pvdz
 end
 ecp
   U library crenbl_ecp
 end
</code></pre>
<p>The following is an example of explicit input of an ECP for H<sub>2</sub>CO.<br />
It defines an ECP for the carbon and oxygen atoms in the molecule.</p>
<pre><code> ecp
   C nelec 2 # ecp replaces 2 electrons on C
   C ul      # d
           1       80.0000000       -1.60000000
           1       30.0000000       -0.40000000
           2        0.5498205       -0.03990210
  C s        # s - d 
           0        0.7374760        0.63810832
           0      135.2354832       11.00916230
           2        8.5605569       20.13797020
   C p       # p - d
           2       10.6863587       -3.24684280
           2       23.4979897        0.78505765
   O nelec 2 # ecp replaces 2 electrons on O
   O ul      # d 
           1       80.0000000       -1.60000000
           1       30.0000000       -0.40000000
           2        1.0953760       -0.06623814
   O s       # s - d
           0        0.9212952        0.39552179
           0       28.6481971        2.51654843
           2        9.3033500       17.04478500
   O p       # p - s 
           2       52.3427019       27.97790770
           2       30.7220233      -16.49630500
 end
</code></pre>
<p>Various ECPs without a local function are available, including those of
the Stuttgart group. For those, no <code>ul</code> part needs to be defined. To
define the absence of the local potential, simply specify one
contraction with a zero coefficient:</p>
<pre><code>    &lt;string tag&gt; ul
    2     1.00000     0.00000
</code></pre>
<h2 id="ecp-spin-orbit-ecps">Spin-orbit ECPs<a class="headerlink" href="#ecp-spin-orbit-ecps" title="Permanent link">&para;</a></h2>
<p>The Spin-orbit ECPs can be used with the Density Functional Approach,
but one has to run the calculations without symmetry. Note: when a
Hartree-Fock method is specified the spin-orbit input will be ignored.</p>
<p>Spin-orbit ECPs are fitted in precisely the same functional form as the
scalar RECPs and have the same properties, with the exception that there
is no local potential <em>u<sub>l</sub></em>, no <em>s</em> potential and no effective charge has
to be defined. Spin-orbit potentials are specified in the same way as
ECPs except that the directive SO is used instead of ECP. Note that
there currently are no spin-orbit ECPs defined in the standard NWChem
library. The SO directive is as
follows:</p>
<pre><code> SO [&lt;string name default &quot;so basis&quot;&gt;] \  
       [print || noprint default print]  
    &lt;string tag&gt; library [&lt;string tag_in_lib&gt;] \
                 &lt;string standard_set&gt; [file &lt;filename&gt;]  
                 [except `&lt;string tag list&gt;]  
       ... 
    &lt;string tag&gt; &lt;string shell_type&gt;  
    &lt;real r-exponent&gt; &lt;real Gaussian-exponent&gt; &lt;real list_of_coefficients&gt;  
       ...  
 END
</code></pre>
<p>Note: in the literature the coefficients of the spin-orbit potentials
are NOT always defined in the same manner. The NWChem code assumes that
the spin-orbit potential defined in the input is of the form:
<span class="arithmatex">\(<span class="arithmatex">\(\Delta U^{NWChem}_{l} = \,\! \frac{2}{2l+1} \Delta U_{l}\)</span>\)</span></p>
<p>For example, in the literature (most of) the Stuttgart potentials are
defined as <span class="arithmatex">\(\Delta U_{l}\)</span>  and, hence, have to be multiplied by <span class="arithmatex">\(2/(2l+1)\)</span>
 (Note: On the Stuttgart/Kln web pages<br />
<a href="https://www.tc.uni-koeln.de/PP/clickpse.en.html">https://www.tc.uni-koeln.de/PP/clickpse.en.html</a>,<br />
 spin-orbit potentials have already been corrected by the appropriate scaling factor and can be used as is). On the other hand, the CRENBL
potentials in the published papers are defined as 
<span class="arithmatex">\(\,\!\frac{l}{2l+1} \Delta U_{l}\)</span>
have been corrected with the <span class="arithmatex">\(2/l\)</span> factor, so make sure the
appropriate scaling is applied).</p>
<p>For example, to use the Stuttgart/Kln ECP and SO-ECP for Hg  (ECP60MDF) in NWChem.<br />
The following URL will display bot the the ECP and SO parts.<br />
<a href="http://www.tc.uni-koeln.de/cgi-bin/pp.pl?language=en,format=molpro,element=Hg,job=getecp,ecp=ECP60MDF">http://www.tc.uni-koeln.de/cgi-bin/pp.pl?language=en,format=molpro,element=Hg,job=getecp,ecp=ECP60MDF</a><br />
The highlighted section (last four lines) below is the SO part.<br />
The un-highlighted part (first five lines) is the ECP.    </p>
<pre><code>
!  Q=20., MEFIT, MCDHF+Breit, Ref 37.  
ECP,Hg,60,5,4;  
1; 2,1.000000,0.000000;   
2; 2,12.413071,275.774797; 2,6.897913,49.267898;   
4; 2,11.310320,80.506984; 2,10.210773,161.034824; 2,5.939804,9.083416; 2,5.019755,18.367773;   
4; 2,8.407895,51.137256; 2,8.214086,76.707459; 2,4.012612,6.561821; 2,3.795398,9.818070;  
2; 2,3.273106,9.429001; 2,3.208321,12.494856;   
2; 2,4.485296,-6.338414; 2,4.513200,-8.099863;   
<b>4; 2,11.310320,-161.013967;2,10.210773,161.034824;2,5.939804,-18.166832;2,5.019755,18.367773;  
4; 2,8.407895,-51.137256; 2,8.214086,51.138306; 2,4.012612,-6.561821; 2,3.795398,6.545380;     
2; 2,3.273106,-6.286001; 2,3.208321,6.247428;     
2; 2,4.485296,3.169207; 2,4.513200,-3.239945;</b>
! References:  
! [37] D. Figgen, G. Rauhut, M. Dolg, H. Stoll, Chem. Phys. 311, 227 (2005).  
</code></pre>
<p>The corresponding NWChem input is </p>
<pre><code>ecp  
Hg nelec 60
Hg ul
2      1.0000000              0.0000000
Hg S
2     12.4130710            275.7747970
2      6.8979130             49.2678980
Hg P
2     11.3103200             80.5069840
2     10.2107730            161.0348240
2      5.9398040              9.0834160
2      5.0197550             18.3677730
Hg D
2      8.4078950             51.1372560
2      8.2140860             76.7074590
2      4.0126120              6.5618210
2      3.7953980              9.8180700
Hg F
2      3.2731060              9.4290010
2      3.2083210             12.4948560
Hg G
2      4.4852960             -6.3384140
2      4.5132000             -8.0998630
end

so
Hg P
2  11.310320  161.013967
2  10.210773  161.034824
2   5.939804  -18.166832
2   5.019755   18.367773
Hg D     
2   8.407895  -51.137256
2   8.214086   51.138306
2   4.012612   -6.561821
2   3.795398    6.545380
Hg F      
2   3.273106   -6.286001
2   3.208321    6.247428
Hg G       
2   4.485296    3.169207
2   4.513200   -3.239945
end
</code></pre>
<h3 id="ecp-websites-with-spin-orbits-ecps">Websites with Spin-Orbits ECPs<a class="headerlink" href="#ecp-websites-with-spin-orbits-ecps" title="Permanent link">&para;</a></h3>
<ul>
<li><a href="https://www.tc.uni-koeln.de/PP/clickpse.en.html">https://www.tc.uni-koeln.de/PP/clickpse.en.html</a></li>
<li><a href="https://web.archive.org/web/20241007182207/https://people.clarkson.edu/~pchristi/reps.html">https://people.clarkson.edu/~pchristi/reps.html</a></li>
<li><a href="http://www.qchem.pnpi.spb.ru/recp">http://www.qchem.pnpi.spb.ru/recp</a></li>
<li><a href="https://pseudopotentiallibrary.org">https://pseudopotentiallibrary.org</a></li>
</ul></section><section class="print-page" id="relativistic-all-electron-approximations" heading-number="10"><h1 id="relativistic-all-electron-approximations-relativistic-all-electron-approximations">Relativistic all-electron approximations<a class="headerlink" href="#relativistic-all-electron-approximations-relativistic-all-electron-approximations" title="Permanent link">&para;</a></h1>
<h2 id="relativistic-all-electron-approximations-overview">Overview<a class="headerlink" href="#relativistic-all-electron-approximations-overview" title="Permanent link">&para;</a></h2>
<p>All methods which include treatment of relativistic effects are
ultimately based on the Dirac equation, which has a four component wave
function. The solutions to the Dirac equation describe both positrons
(the &ldquo;negative energy&rdquo; states) and electrons (the &ldquo;positive energy&rdquo;
states), as well as both spin orientations, hence the four components.
The wave function may be broken down into two-component functions
traditionally known as the large and small components; these may further
be broken down into the spin components.</p>
<p>The implementation of approximate all-electron relativistic methods in
quantum chemical codes requires the removal of the negative energy
states and the factoring out of the spin-free terms. Both of these may
be achieved using a transformation of the Dirac Hamiltonian known in
general as a Foldy-Wouthuysen (FW) transformation. Unfortunately this
transformation cannot be represented in closed form for a general
potential, and must be approximated. One popular approach is that
originally formulated by Douglas and Kroll<sup id="relativistic-all-electron-approximations-fnref_footnote_1"><a class="footnote-ref" href="#relativistic-all-electron-approximations-fn_footnote_1">1</a></sup> and developed by
Hess<sup id="relativistic-all-electron-approximations-fnref_footnote_2"><a class="footnote-ref" href="#relativistic-all-electron-approximations-fn_footnote_2">2</a></sup><sup id="fnref_footnote_3"><a class="footnote-ref" href="#relativistic-all-electron-approximations-fn_footnote_3">3</a></sup>. This approach decouples the positive and negative energy
parts to second order in the external potential (and also fourth order
in the fine structure constant, ). Other approaches include the Zeroth
Order Regular Approximation (ZORA)<sup id="relativistic-all-electron-approximations-fnref_footnote_4"><a class="footnote-ref" href="#relativistic-all-electron-approximations-fn_footnote_4">4</a></sup><sup id="fnref_footnote_5"><a class="footnote-ref" href="#relativistic-all-electron-approximations-fn_footnote_5">5</a></sup><sup id="fnref_footnote_6"><a class="footnote-ref" href="#relativistic-all-electron-approximations-fn_footnote_6">6</a></sup><sup id="fnref_footnote_7"><a class="footnote-ref" href="#relativistic-all-electron-approximations-fn_footnote_7">7</a></sup> and modification
of the Dirac equation by Dyall<sup id="relativistic-all-electron-approximations-fnref_footnote_8"><a class="footnote-ref" href="#relativistic-all-electron-approximations-fn_footnote_8">8</a></sup>, and involves an exact FW
transformation on the atomic basis set level<sup id="relativistic-all-electron-approximations-fnref_footnote_9"><a class="footnote-ref" href="#relativistic-all-electron-approximations-fn_footnote_9">9</a></sup><sup id="fnref_footnote_10"><a class="footnote-ref" href="#relativistic-all-electron-approximations-fn_footnote_10">10</a></sup>.</p>
<p>Since these approximations only modify the integrals, they can in
principle be used at all levels of theory. At present the Douglas-Kroll
and ZORA implementations can be used at all levels of theory whereas
Dyall&rsquo;s approach is currently available at the Hartree-Fock level. The
derivatives have been implemented, allowing both methods to be used in
geometry optimizations and frequency calculations.</p>
<h2 id="relativistic-all-electron-approximations-relativistic-directive">RELATIVISTIC directive<a class="headerlink" href="#relativistic-all-electron-approximations-relativistic-directive" title="Permanent link">&para;</a></h2>
<p>The <code>RELATIVISTIC</code> directive provides input for the implemented
relativistic approximations and is a compound directive that encloses
additional directives specific to the
approximations:</p>
<pre><code>RELATIVISTIC
  [DOUGLAS-KROLL [&lt;string (ON||OFF) default ON&gt; \  
                &lt;string (FPP||DKH||DKFULL||DK3||DK3FULL) default DKH&gt;]  || 
   ZORA [ (ON || OFF) default ON ] ||   
   DYALL-MOD-DIRAC [ (ON || OFF) default ON ]  ||
                 [ (NESC1E || NESC2E) default NESC1E ] ]  ||
   X2C [ (ON || OFF) default ON ]
  [CLIGHT &lt;real clight default 137.0359895&gt;]  
 END
</code></pre>
<p>Only one of the methods may be chosen at a time. If both methods are
found to be on in the input block, NWChem will stop and print an error
message. There is one general option for both methods, the definition of
the speed of light in atomic units:</p>
<pre><code> CLIGHT &lt;real clight default 137.0359895&gt;
</code></pre>
<p>The following sections describe the optional sub-directives that can be
specified within the <code>RELATIVISTIC</code> block.</p>
<h3 id="relativistic-all-electron-approximations-douglas-kroll-approximation">Douglas-Kroll approximation<a class="headerlink" href="#relativistic-all-electron-approximations-douglas-kroll-approximation" title="Permanent link">&para;</a></h3>
<p>The spin-free and spin-orbit one-electron Douglas-Kroll approximation
have been implemented. The use of relativistic effects from this
Douglas-Kroll approximation can be invoked by specifying:</p>
<pre><code> DOUGLAS-KROLL [&lt;string (ON||OFF) default ON&gt; \ 
                &lt;string (FPP||DKH||DKFULL|DK3|DK3FULL) default DKH&gt;]
</code></pre>
<p>The ON|OFF string is used to turn on or off the Douglas-Kroll
approximation. By default, if the <code>DOUGLAS-KROLL</code> keyword is found, the
approximation will be used in the calculation. If the user wishes to
calculate a non-relativistic quantity after turning on Douglas-Kroll,
the user will need to define a new <code>RELATIVISTIC</code> block and turn the
approximation <code>OFF</code>. The user could also simply put a blank <code>RELATIVISTIC</code>
block in the input file and all options will be turned off.</p>
<p>The <code>FPP</code> is the approximation based on free-particle projection
operators<sup id="relativistic-all-electron-approximations-fnref_footnote_11"><a class="footnote-ref" href="#relativistic-all-electron-approximations-fn_footnote_11">11</a></sup> whereas the <code>DKH</code> and <code>DKFULL</code> approximations are based on
external-field projection operators<sup id="relativistic-all-electron-approximations-fnref_footnote_12"><a class="footnote-ref" href="#relativistic-all-electron-approximations-fn_footnote_12">12</a></sup>. The latter two are
considerably better approximations than the former. <code>DKH</code> is the
Douglas-Kroll-Hess approach and is the approach that is generally
implemented in quantum chemistry codes. <code>DKFULL</code> includes certain
cross-product integral terms ignored in the <code>DKH</code> approach (see for
example Hberlen and Rsch<sup id="relativistic-all-electron-approximations-fnref_footnote_13"><a class="footnote-ref" href="#relativistic-all-electron-approximations-fn_footnote_13">13</a></sup>). The third-order Douglas-Kroll
approximation has been implemented by T. Nakajima and K.
Hirao<sup id="relativistic-all-electron-approximations-fnref_footnote_14"><a class="footnote-ref" href="#relativistic-all-electron-approximations-fn_footnote_14">14</a></sup><sup id="fnref_footnote_15"><a class="footnote-ref" href="#relativistic-all-electron-approximations-fn_footnote_15">15</a></sup>. This approximation can be called using <code>DK3</code> (DK3
without cross-product integral terms) or <code>DK3FULL</code> (DK3 with cross-product
integral terms).</p>
<p>The contracted basis sets used in the calculations should reflect the
relativistic effects, i.e. one should use contracted basis sets which
were generated using the Douglas-Kroll Hamiltonian. Basis sets that were
contracted using the non-relativistic (Schdinger) Hamiltonian WILL
PRODUCE ERRONEOUS RESULTS for elements beyond the first row. See
appendix A for available basis sets and their naming convention.</p>
<p>NOTE: we suggest that spherical basis sets are used in the calculation.
The use of high quality cartesian basis sets can lead to numerical
inaccuracies.</p>
<p>In order to compute the integrals needed for the Douglas-Kroll
approximation the implementation makes use of a fitting basis set (see
literature given above for details). The current code will create this
fitting basis set based on the given <code>ao basis</code> by simply uncontracting
that basis. This again is what is commonly implemented in quantum
chemistry codes that include the Douglas-Kroll method. Additional
flexibility is available to the user by explicitly specifying a
Douglas-Kroll fitting basis set. This basis set must be named
<code>D-K basis</code> (see <a href="#basis">Basis Sets</a>).</p>
<h3 id="relativistic-all-electron-approximations-zeroth-order-regular-approximation-zora">Zeroth Order regular approximation (ZORA)<a class="headerlink" href="#relativistic-all-electron-approximations-zeroth-order-regular-approximation-zora" title="Permanent link">&para;</a></h3>
<p>The spin-free and spin-orbit one-electron zeroth-order regular
approximation (ZORA) have been implemented. ZORA can be accessed only 
via the <a href="#density-functional-theory-for-molecules">DFT</a> and <a href="#density-functional-theory-for-molecules-spin-orbit-density-functional-theory-sodft">SO-DFT</a> modules. The use of relativistic
effects with ZORA can be invoked by specifying:</p>
<pre><code> ZORA [&lt;string (ON||OFF) &gt;
</code></pre>
<p>The <code>ON</code>|<code>OFF</code> string is used to turn on or off ZORA. No default is present, therefore
<code>ZORA</code> keyword need to be followed by <code>ON</code> in order for the approximation to be used in the
calculation. If the user wishes to calculate a non-relativistic quantity
after turning on ZORA, the user will need to define a new <code>RELATIVISTIC</code>
block and turn the approximation OFF. The user can also simply put a
blank <code>RELATIVISTIC</code> block in the input file and all options will be
turned off.</p>
<p>To increase the accuracy of ZORA calculations, the following settings may be used 
in the relativistic block</p>
<pre><code> relativistic
     zora on
     zora:cutoff 1d-30
 end
</code></pre>
<p>To invoke the relativistic ZORA model potential approach due to van Wullen (references <sup id="fnref_footnote_16"><a class="footnote-ref" href="#relativistic-all-electron-approximations-fn_footnote_16">16</a></sup> and <sup id="fnref_footnote_17"><a class="footnote-ref" href="#relativistic-all-electron-approximations-fn_footnote_17">17</a></sup>).</p>
<p>For model potentials constructed from 4-component densities:  </p>
<pre><code> relativistic
     zora on
     zora:cutoff 1d-30
     modelpotential  1
 end
</code></pre>
<p>For model potentials constructed from 2-component densities:  </p>
<pre><code> relativistic
     zora on
     zora:cutoff 1d-30
     modelpotential  2
 end
</code></pre>
<p>Both approaches are comparable in accuracy and depends on the system.</p>
<h3 id="relativistic-all-electron-approximations-dyalls-modified-dirac-hamitonian-approximation">Dyall&rsquo;s Modified Dirac Hamitonian approximation<a class="headerlink" href="#relativistic-all-electron-approximations-dyalls-modified-dirac-hamitonian-approximation" title="Permanent link">&para;</a></h3>
<p>The approximate methods described in this section are all based on
Dyall&rsquo;s modified Dirac Hamiltonian. This Hamiltonian is entirely
equivalent to the original Dirac Hamiltonian, and its solutions have the
same properties. The modification is achieved by a transformation on the
small component, extracting out &sigma;&sdot;<strong>p</strong>/2mc. This
gives the modified small component the same symmetry as the large
component, and in fact it differs from the large component only at order &alpha;<sup>2</sup>. The advantage of the modification is that the operators
now resemble the operators of the Breit-Pauli Hamiltonian, and can be
classified in a similar fashion into spin-free, spin-orbit and spin-spin
terms. It is the spin-free terms which have been implemented in NWChem,
with a number of further approximations.</p>
<p>The first is that the negative energy states are removed by a normalized
elimination of the small component (NESC), which is equivalent to an
exact Foldy-Wouthuysen (EFW) transformation. The number of components in
the wave function is thereby effectively reduced from 4 to 2. NESC on
its own does not provide any advantages, and in fact complicates things
because the transformation is energy-dependent. The second approximation
therefore performs the elimination on an atom-by-atom basis, which is
equivalent to neglecting blocks which couple different atoms in the EFW
transformation. The advantage of this approximation is that all the
energy dependence can be included in the contraction coefficients of the
basis set. The tests which have been done show that this approximation
gives results well within chemical accuracy. The third approximation
neglects the commutator of the EFW transformation with the two-electron
Coulomb interaction, so that the only corrections that need to be made
are in the one-electron integrals. This is the equivalent of the
Douglas-Kroll(-Hess) approximation as it is usually applied.</p>
<p>The use of these approximations can be invoked with the use of the
<code>DYALL-MOD-DIRAC</code> directive in the <code>RELATIVISTIC</code> directive block. The
syntax is as follows.</p>
<pre><code> DYALL-MOD-DIRAC [ (ON || OFF) default ON ] 
                 [ (NESC1E || NESC2E) default NESC1E ]
</code></pre>
<p>The <code>ON</code>|<code>OFF</code> string is used to turn on or off the Dyall&rsquo;s modified Dirac
approximation. By default, if the <code>DYALL-MOD-DIRAC</code> keyword is found, the
approximation will be used in the calculation. If the user wishes to
calculate a non-relativistic quantity after turning on Dyall&rsquo;s modified
Dirac, the user will need to define a new <code>RELATIVISTIC</code> block and turn
the approximation <code>OFF</code>. The user could also simply put a blank
<code>RELATIVISTIC</code> block in the input file and all options will be turned off.</p>
<p>Both one- and two-electron approximations are available <code>NESC1E</code> ||
<code>NESC2E</code>, and both have analytic gradients. The one-electron approximation
is the default. The two-electron approximation specified by <code>NESC2E</code> has
some sub options which are placed on the same logical line as the
<code>DYALL-MOD-DIRAC</code> directive, with the following
syntax:</p>
<pre><code> NESC2E [ (SS1CENT [ (ON || OFF) default ON ] || SSALL) default SSALL ]
        [ (SSSS [ (ON || OFF) default ON ] || NOSSSS) default SSSS ]
</code></pre>
<p>The first sub-option gives the capability to limit the two-electron
corrections to those in which the small components in any density must
be on the same center. This reduces the <em>(LL|SS)</em> contributions
to at most three-center integrals and the <em>(SS|SS)</em> contributions
to two centers. For a case with only one relativistic atom this option
is redundant. The second controls the inclusion of the <em>(SS|SS)</em>
integrals which are of order &alpha;<sup>4</sup>. For light atoms they may
safely be neglected, but for heavy atoms they should be included.</p>
<p>In addition to the selection of this keyword in the <code>RELATIVISTIC</code>
directive block, it is necessary to supply basis sets in addition to the
<a href="#basis-basis-set-name"><code>ao basis</code></a>. For the one-electron approximation, three basis sets are
needed: the <em>atomic FW</em> basis set, the <em>large component</em> basis set and the
<em>small</em> component basis set. The atomic FW basis set should be included in
the <code>ao basis</code>. The large and small components should similarly be
incorporated in basis sets named <code>large component</code> and <code>small component</code>,
respectively. For the two-electron approximation, only two basis sets
are needed. These are the large component and the small component. The
large component should be included in the <code>ao basis</code> and the small
component is specified separately as <code>small component</code>, as for the
one-electron approximation. This means that the two approximations can
not be run correctly without changing the <code>ao basis</code>, and it is up to the
user to ensure that the basis sets are correctly specified.</p>
<p>There is one further requirement in the specification of the basis sets.
In the ao basis, it is necessary to add the <code>rel</code> keyword either to the
<a href="#basis-rel-keyword">basis</a> directive or the <a href="#basis-basis-set-library">library</a> tag line (See <a href="#relativistic-all-electron-approximations-examples-for-dyall-mod-dirac">below</a> for examples). The
former marks the basis functions specified by the tag as relativistic,
the latter marks the whole basis as relativistic. The marking is
actually done at the unique shell level, so that it is possible not only
to have relativistic and nonrelativistic atoms, it is also possible to
have relativistic and nonrelativistic shells on a given atom. This would
be useful, for example, for diffuse functions or for high angular
momentum correlating functions, where the influence of relativity was
small. The marking of shells as relativistic is necessary to set up a
mapping between the ao basis and the large and/or small component basis
sets. For the one-electron approximation the large and small component
basis sets MUST be of the same size and construction, i.e. differing
only in the contraction coefficients.</p>
<p>It should also be noted that the relativistic code will NOT work with
basis sets that contain sp shells, nor will it work with ECPs. Both of
these are tested and flagged as an error.</p>
<h3 id="relativistic-all-electron-approximations-examples-for-dyall-mod-dirac">Examples for DYALL-MOD-DIRAC<a class="headerlink" href="#relativistic-all-electron-approximations-examples-for-dyall-mod-dirac" title="Permanent link">&para;</a></h3>
<p>Some examples follow. The first example sets up the data for
relativistic calculations on water with the one-electron approximation
and the two-electron approximation, using the library basis sets.</p>
<pre><code> start h2o-dmd
 geometry units bohr
 symmetry c2v
   O       0.000000000    0.000000000   -0.009000000
   H       1.515260000    0.000000000   -1.058900000
   H      -1.515260000    0.000000000   -1.058900000
 end
 basis &quot;fw&quot; rel
   oxygen library cc-pvdz_pt_sf_fw
   hydrogen library cc-pvdz_pt_sf_fw
 end
 basis &quot;large&quot;
   oxygen library cc-pvdz_pt_sf_lc
   hydrogen library cc-pvdz_pt_sf_lc
 end
 basis &quot;large2&quot; rel
   oxygen library cc-pvdz_pt_sf_lc
   hydrogen library cc-pvdz_pt_sf_lc
 end
 basis &quot;small&quot;
   oxygen library cc-pvdz_pt_sf_sc
   hydrogen library cc-pvdz_pt_sf_sc
 end
 set &quot;ao basis&quot; fw
 set &quot;large component&quot; large
 set &quot;small component&quot; small
 relativistic
   dyall-mod-dirac
 end
 task scf
 set &quot;ao basis&quot; large2
 unset &quot;large component&quot;
 set &quot;small component&quot; small
 relativistic
   dyall-mod-dirac nesc2e
 end
 task scf
</code></pre>
<p>The second example has oxygen as a relativistic atom and hydrogen
nonrelativistic.</p>
<pre><code> start h2o-dmd2
 geometry units bohr
 symmetry c2v
   O       0.000000000    0.000000000   -0.009000000
   H       1.515260000    0.000000000   -1.058900000
   H      -1.515260000    0.000000000   -1.058900000
 end
 basis &quot;ao basis&quot;
   oxygen library cc-pvdz_pt_sf_fw rel
   hydrogen library cc-pvdz
 end
 basis &quot;large component&quot;
   oxygen library cc-pvdz_pt_sf_lc
 end
 basis &quot;small component&quot;
   oxygen library cc-pvdz_pt_sf_sc
 end
 relativistic
   dyall-mod-dirac
 end
 task scf
</code></pre>
<h3 id="relativistic-all-electron-approximations-x2c-exact-two-component-relativistic-hamiltonian">X2C: exact two-component relativistic Hamiltonian<a class="headerlink" href="#relativistic-all-electron-approximations-x2c-exact-two-component-relativistic-hamiltonian" title="Permanent link">&para;</a></h3>
<p>The exact two-component Hamiltonian<sup id="fnref_footnote_18"><a class="footnote-ref" href="#relativistic-all-electron-approximations-fn_footnote_18">18</a></sup><sup id="fnref_footnote_19"><a class="footnote-ref" href="#relativistic-all-electron-approximations-fn_footnote_19">19</a></sup> has been implemented in NWChem<sup id="fnref_footnote_20"><a class="footnote-ref" href="#relativistic-all-electron-approximations-fn_footnote_20">20</a></sup><sup id="fnref_footnote_21"><a class="footnote-ref" href="#relativistic-all-electron-approximations-fn_footnote_21">21</a></sup><sup id="fnref_footnote_22"><a class="footnote-ref" href="#relativistic-all-electron-approximations-fn_footnote_22">22</a></sup>.</p>
<pre><code> X2C [&lt;string (ON||OFF) default ON&gt;
</code></pre>
<p>The <code>ON</code>|<code>OFF</code> string is used to turn on or off X2C. By default, if the
<code>X2C</code> keyword is found, the approximation will be used in the
calculation. If the user wishes to calculate a non-relativistic quantity
after turning on X2C, the user will need to define a new <code>RELATIVISTIC</code>
block and turn the approximation OFF. The user can also simply put a
blank <code>RELATIVISTIC</code> block in the input file and all options will be
turned off.</p>
<p>To increase the accuracy of X2C calculations, the following settings may be used 
in the relativistic block</p>
<pre><code> relativistic
     x2c on
     x2c:cutoff 1d-15
 end
</code></pre>
<h2 id="relativistic-all-electron-approximations-references">References<a class="headerlink" href="#relativistic-all-electron-approximations-references" title="Permanent link">&para;</a></h2>
<div class="footnote">
<hr />
<ol>
<li id="relativistic-all-electron-approximations-fn_footnote_1">
<p>Douglas, M.; Kroll, N.M. (1974). &ldquo;Quantum electrodynamical corrections to the fine structure of helium&rdquo;. Annals of Physics 82: 89-155. <a href="https://dx.doi.org/10.1016/0003-4916(74)90333-9">DOI:10.1016/0003-4916(74)90333-9</a>.&#160;<a class="footnote-backref" href="#relativistic-all-electron-approximations-fnref_footnote_1" title="Jump back to footnote 1 in the text">&#8617;</a></p>
</li>
<li id="relativistic-all-electron-approximations-fn_footnote_2">
<p>Hess, B.A. (1985). &ldquo;Applicability of the no-pair equation with free-particle projection operators to atomic and molecular structure calculations&rdquo;. Physical Review A 32: 756-763. <a href="https://dx.doi.org/10.1103/PhysRevA.32.756">DOI:10.1103/PhysRevA.32.756</a>.&#160;<a class="footnote-backref" href="#relativistic-all-electron-approximations-fnref_footnote_2" title="Jump back to footnote 2 in the text">&#8617;</a></p>
</li>
<li id="relativistic-all-electron-approximations-fn_footnote_3">
<p>Hess, B.A. (1986). &ldquo;Relativistic electronic-structure calculations employing a two-component no-pair formalism with external-field projection operators&rdquo;. Physical Review A 33: 3742-3748. <a href="https://dx.doi.org/10.1103/PhysRevA.33.3742">DOI:10.1103/PhysRevA.33.3742</a>.&#160;<a class="footnote-backref" href="#relativistic-all-electron-approximations-fnref_footnote_3" title="Jump back to footnote 3 in the text">&#8617;</a></p>
</li>
<li id="relativistic-all-electron-approximations-fn_footnote_4">
<p>Chang, C; Pelissier, M; Durand, M (1986). &ldquo;Regular Two-Component Pauli-Like Effective Hamiltonians in Dirac Theory&rdquo;. Physica Scripta 34: 394. <a href="https://dx.doi.org/10.1088/0031-8949/34/5/007">DOI:10.1088/0031-8949/34/5/007</a>.&#160;<a class="footnote-backref" href="#relativistic-all-electron-approximations-fnref_footnote_4" title="Jump back to footnote 4 in the text">&#8617;</a></p>
</li>
<li id="relativistic-all-electron-approximations-fn_footnote_5">
<p>van Lenthe, E (1996). &ldquo;The ZORA Equation&rdquo; (in English).&#160;<a class="footnote-backref" href="#relativistic-all-electron-approximations-fnref_footnote_5" title="Jump back to footnote 5 in the text">&#8617;</a></p>
</li>
<li id="relativistic-all-electron-approximations-fn_footnote_6">
<p>Faas, S.; Snijders, J.G.; van Lenthe, J.H.; van Lenthe, E.; Baerends, E.J. (1995). &ldquo;The ZORA formalism applied to the Dirac-Fock equation&rdquo;. Chemical Physics Letters 246: 632-640. <a href="https://dx.doi.org/10.1016/0009-2614(95)01156-0">DOI:10.1016/0009-2614(95)01156-0</a>.&#160;<a class="footnote-backref" href="#relativistic-all-electron-approximations-fnref_footnote_6" title="Jump back to footnote 6 in the text">&#8617;</a></p>
</li>
<li id="relativistic-all-electron-approximations-fn_footnote_7">
<p>Nichols, P.; Govind, N.; Bylaska, E.J.; de Jong, W.A. (2009). &ldquo;Gaussian Basis Set and Planewave Relativistic Spin-Orbit Methods in NWChem&rdquo;. Journal of Chemical Theory and Computation 5: 491-499. <a href="https://dx.doi.org/10.1021/ct8002892">DOI:10.1021/ct8002892</a>.&#160;<a class="footnote-backref" href="#relativistic-all-electron-approximations-fnref_footnote_7" title="Jump back to footnote 7 in the text">&#8617;</a></p>
</li>
<li id="relativistic-all-electron-approximations-fn_footnote_8">
<p>Dyall, K.G. (1994). &ldquo;An exact separation of the spin-free and spin-dependent terms of the Dirac&ndash;Coulomb&ndash;Breit Hamiltonian&rdquo;. The Journal of Chemical Physics 100: 2118-2127. <a href="https://dx.doi.org/10.1063/1.466508">DOI:10.1063/1.466508</a>.&#160;<a class="footnote-backref" href="#relativistic-all-electron-approximations-fnref_footnote_8" title="Jump back to footnote 8 in the text">&#8617;</a></p>
</li>
<li id="relativistic-all-electron-approximations-fn_footnote_9">
<p>Dyall, K.G. (1997). &ldquo;Interfacing relativistic and nonrelativistic methods. I. Normalized elimination of the small component in the modified Dirac equation&rdquo;. The Journal of Chemical Physics 106: 9618-9626. <a href="https://dx.doi.org/10.1063/1.473860">DOI:10.1063/1.473860</a>.&#160;<a class="footnote-backref" href="#relativistic-all-electron-approximations-fnref_footnote_9" title="Jump back to footnote 9 in the text">&#8617;</a></p>
</li>
<li id="relativistic-all-electron-approximations-fn_footnote_10">
<p>Dyall, K.G.; Enevoldsen, T. (1999). &ldquo;Interfacing relativistic and nonrelativistic methods. III. Atomic 4-spinor expansions and integral approximations&rdquo;. The Journal of Chemical Physics 111: 10000-10007. <a href="https://dx.doi.org/10.1063/1.480353">DOI:10.1063/1.480353</a>.&#160;<a class="footnote-backref" href="#relativistic-all-electron-approximations-fnref_footnote_10" title="Jump back to footnote 10 in the text">&#8617;</a></p>
</li>
<li id="relativistic-all-electron-approximations-fn_footnote_11">
<p>Hess, B.A. (1985). &ldquo;Applicability of the no-pair equation with free-particle projection operators to atomic and molecular structure calculations&rdquo;. Physical Review A 32: 756-763. <a href="https://dx.doi.org/10.1103/PhysRevA.32.756">DOI:10.1103/PhysRevA.32.756</a>.&#160;<a class="footnote-backref" href="#relativistic-all-electron-approximations-fnref_footnote_11" title="Jump back to footnote 11 in the text">&#8617;</a></p>
</li>
<li id="relativistic-all-electron-approximations-fn_footnote_12">
<p>Hess, B.A. (1986). &ldquo;Relativistic electronic-structure calculations employing a two-component no-pair formalism with external-field projection operators&rdquo;. Physical Review A 33: 3742-3748. <a href="https://dx.doi.org/10.1103/PhysRevA.33.3742">DOI:10.1103/PhysRevA.33.3742</a>.&#160;<a class="footnote-backref" href="#relativistic-all-electron-approximations-fnref_footnote_12" title="Jump back to footnote 12 in the text">&#8617;</a></p>
</li>
<li id="relativistic-all-electron-approximations-fn_footnote_13">
<p>Haeberlen, O.D.; Roesch, N. (1992). &ldquo;A scalar-relativistic extension of the linear combination of Gaussian-type orbitals local density functional method: application to AuH, AuCl and Au2&rdquo;. Chemical Physics Letters 199: 491-496. <a href="https://dx.doi.org/10.1016/0009-2614(92)87033-L">DOI:10.1016/0009-2614(92)87033-L</a>.&#160;<a class="footnote-backref" href="#relativistic-all-electron-approximations-fnref_footnote_13" title="Jump back to footnote 13 in the text">&#8617;</a></p>
</li>
<li id="relativistic-all-electron-approximations-fn_footnote_14">
<p>Nakajima, T.; Hirao, K. (2000). &ldquo;Numerical illustration of third-order Douglas-Kroll method: atomic and molecular properties of superheavy element 112&rdquo;. Chemical Physics Letters 329: 511-516. <a href="https://dx.doi.org/10.1016/S0009-2614(00)01035-6">DOI:10.1016/S0009-2614(00)01035-6</a>.&#160;<a class="footnote-backref" href="#relativistic-all-electron-approximations-fnref_footnote_14" title="Jump back to footnote 14 in the text">&#8617;</a></p>
</li>
<li id="relativistic-all-electron-approximations-fn_footnote_15">
<p>Nakajima, T.; Hirao, K. (2000). &ldquo;The higher-order Douglas&ndash;Kroll transformation&rdquo;. The Journal of Chemical Physics 113: 7786-7789. <a href="https://dx.doi.org/10.1063/1.1316037">DOI:10.1063/1.1316037</a>.&#160;<a class="footnote-backref" href="#relativistic-all-electron-approximations-fnref_footnote_15" title="Jump back to footnote 15 in the text">&#8617;</a></p>
</li>
<li id="relativistic-all-electron-approximations-fn_footnote_16">
<p>van Wullen, C. (1998). &ldquo;Molecular density functional calculations in the regular relativistic approximation: Method, application to coinage metal diatomics, hydrides, fluorides and chlorides, and comparison with first-order relativistic calculations&rdquo;. The Journal of Chemical Physics 109: 392-399  <a href="https://dx.doi.org/10.1063/1.476576">DOI:10.1063/1.476576</a>&#160;<a class="footnote-backref" href="#relativistic-all-electron-approximations-fnref_footnote_16" title="Jump back to footnote 16 in the text">&#8617;</a></p>
</li>
<li id="relativistic-all-electron-approximations-fn_footnote_17">
<p>van Wullen, C.; Michauk, C. (2005). &ldquo;Accurate and efficient treatment of two-electron contributions in quasirelativistic high-order Douglas-Kroll density-functional calculations&rdquo;. The Journal of Chemical Physics 123, 204113 <a href="https://dx.doi.org/10.1063/1.2133731">DOI:10.1063/1.2133731</a>&#160;<a class="footnote-backref" href="#relativistic-all-electron-approximations-fnref_footnote_17" title="Jump back to footnote 17 in the text">&#8617;</a></p>
</li>
<li id="relativistic-all-electron-approximations-fn_footnote_18">
<p>Liu, W.; Peng, D. (2009). J. Chem. Phys. 2009, 131, 031104 <a href="https://dx.doi.org/10.1063/1.3159445">DOI:10.1063/1.3159445</a>&#160;<a class="footnote-backref" href="#relativistic-all-electron-approximations-fnref_footnote_18" title="Jump back to footnote 18 in the text">&#8617;</a></p>
</li>
<li id="relativistic-all-electron-approximations-fn_footnote_19">
<p>Saue, T. (2011). &ldquo;Relativistic Hamiltonians for Chemistry: A Primer&rdquo;. ChemPhysChem, 12, 30773094 <a href="http://dx.doi.org/10.1002/cphc.201100682">DOI:10.1002/cphc.201100682</a>&#160;<a class="footnote-backref" href="#relativistic-all-electron-approximations-fnref_footnote_19" title="Jump back to footnote 19 in the text">&#8617;</a></p>
</li>
<li id="relativistic-all-electron-approximations-fn_footnote_20">
<p>Autschbach, J.; Peng, D; Reiher, M. (2012). J. Chem. Theory Comput. 2012, 8,  42394248 <a href="https://dx.doi.org/10.1021/ct300623j">DOI:10.1021/ct300623j</a>&#160;<a class="footnote-backref" href="#relativistic-all-electron-approximations-fnref_footnote_20" title="Jump back to footnote 20 in the text">&#8617;</a></p>
</li>
<li id="relativistic-all-electron-approximations-fn_footnote_21">
<p>Peng, D.; Reiher, M. (2012). Theor. Chem. Acc. 131, 1081 <a href="https://dx.doi.org/10.1007/s00214-011-1081-y">DOI:10.1007/s00214-011-1081-y</a>&#160;<a class="footnote-backref" href="#relativistic-all-electron-approximations-fnref_footnote_21" title="Jump back to footnote 21 in the text">&#8617;</a></p>
</li>
<li id="relativistic-all-electron-approximations-fn_footnote_22">
<p>Autschbach, J. (2021). Quantum Theory for Chemical Applications:From Basic Concepts to Advanced Topics, Oxford, University Press, Chapter 24 <a href="https://dx.doi.org/10.1093/oso/9780190920807.001.0001">DOI:10.1093/oso/9780190920807.001.0001</a>&#160;<a class="footnote-backref" href="#relativistic-all-electron-approximations-fnref_footnote_22" title="Jump back to footnote 22 in the text">&#8617;</a></p>
</li>
</ol>
</div></section>
                    <section class='print-page md-section' id='section-11' heading-number='11'>
                        <h1>Quantum-Mechanical-Methods<a class='headerlink' href='#section-11' title='Permanent link'></a>
                        </h1>
                    <section class="print-page" id="hartree-fock-theory-for-molecules" heading-number="11.1"><h1 id="hartree-fock-theory-for-molecules-hartree-fock">Hartree-Fock<a class="headerlink" href="#hartree-fock-theory-for-molecules-hartree-fock" title="Permanent link">&para;</a></h1>
<h2 id="hartree-fock-theory-for-molecules-overview">Overview<a class="headerlink" href="#hartree-fock-theory-for-molecules-overview" title="Permanent link">&para;</a></h2>
<p>The NWChem self-consistent field (SCF) module computes closed-shell
restricted Hartree-Fock (RHF) wavefunctions, restricted high-spin
open-shell Hartree-Fock (ROHF) wavefunctions, and spin-unrestricted
Hartree-Fock (UHF) wavefunctions. The Hartree-Fock equations are solved
using a conjugate-gradient method with an orbital Hessian based
preconditioner<sup id="hartree-fock-theory-for-molecules-fnref_footnote_1"><a class="footnote-ref" href="#hartree-fock-theory-for-molecules-fn_footnote_1">1</a></sup>. The module supports both replicated data and
distributed data Fock builders<sup id="hartree-fock-theory-for-molecules-fnref_footnote_2"><a class="footnote-ref" href="#hartree-fock-theory-for-molecules-fn_footnote_2">2</a></sup>.</p>
<p>The SCF directive provides input to the SCF module and is a compound
directive that encloses additional directives specific to the SCF
module:</p>
<pre><code> SCF
   ...  
 END
</code></pre>
<h2 id="hartree-fock-theory-for-molecules-wavefunction-type">Wavefunction type<a class="headerlink" href="#hartree-fock-theory-for-molecules-wavefunction-type" title="Permanent link">&para;</a></h2>
<p>A spin-restricted, closed shell RHF calculation is performed by default.
An error results if the number of electrons is inconsistent with this
assumption. The number of electrons is inferred from the total charge on
the system and the sum of the effective nuclear charges of all centers
(atoms and dummy atoms, see <a href="#geometry">GEOMETRY</a>).
The total charge on the system is zero by default, unless specified at
some value by input on the CHARGE directive <a href="#charge">Total system
charge</a>.</p>
<p>The options available to define the SCF wavefunction and multiplicity
are as follows:</p>
<pre><code> SINGLET 
 DOUBLET 
 TRIPLET 
 QUARTET 
 QUINTET 
 SEXTET
 SEPTET
 OCTET
 NOPEN &lt;integer nopen default 0&gt;
 RHF
 ROHF
 UHF
</code></pre>
<p>The optional keywords SINGLET, DOUBLET, &hellip;, OCTET and NOPEN allow the
user to specify the number of singly occupied orbitals for a particular
calculation. SINGLET is the default, and specifies a closed shell;
DOUBLET specifies one singly occupied orbital; TRIPLET specifies two
singly occupied orbitals; and so forth. If there are more than seven
singly occupied orbitals, the keyword NOPEN must be used, with the
integer nopen defining the number of singly occupied orbitals (sometimes
referred to as open shells).</p>
<p>If the multiplicity is any value other than SINGLET, the default
calculation will be a spin-restricted, high-spin, open-shell SCF
calculation (keyword ROHF). The open-shell orbitals must be the highest
occupied orbitals. If necessary, any starting vectors may be rearranged
through the use of the SWAP keyword on the
<a href="#hartree-fock-theory-for-molecules-vectors-inputoutput-of-mo-vectors">VECTORS</a> directive
to accomplish this.</p>
<p>A spin-unrestricted solution can also be performed by specifying the
keyword UHF. In UHF calculations, it is assumed that the number of
singly occupied orbitals corresponds to the difference between the
number of alpha-spin and beta-spin orbitals. For example, a UHF
calculation with 2 more alpha-spin orbitals than beta-spin orbitals can
be obtained by specifying</p>
<pre><code> scf  
    triplet ; uhf    # (Note: two logical lines of input)  
    ...  
 end
</code></pre>
<p>The user should be aware that, by default, molecular orbitals are
symmetry adapted in NWChem. This may not be desirable for fully
unrestricted wavefunctions. In such cases, the user has the option of
defeating the defaults by specifying the keywords
<a href="#hartree-fock-theory-for-molecules-adapt-symmetry-adaptation-of-mos">ADAPT</a> OFF and
<a href="#hartree-fock-theory-for-molecules-sym-use-of-symmetry">SYM</a> OFF .</p>
<p>The keywords RHF and ROHF are provided in the code for completeness. It
may be necessary to specify these in order to modify the behavior of a
previous calculation (see <a href="#nwchem-architecture">NWChem
Architecture</a> for restart behavior).</p>
<h2 id="hartree-fock-theory-for-molecules-sym-use-of-symmetry">SYM: use of symmetry<a class="headerlink" href="#hartree-fock-theory-for-molecules-sym-use-of-symmetry" title="Permanent link">&para;</a></h2>
<pre><code>  SYM &lt;string (ON||OFF) default ON&gt;
</code></pre>
<p>This directive enables/disables the use of symmetry to speed up Fock
matrix construction (via the petite-list or skeleton algorithm) in the
SCF, if symmetry was used in the specification of the geometry. Symmetry
adaptation of the molecular orbitals is not affected by this option. The
default is to use symmetry if it is specified in the <a href="#geometry">geometry
directive</a>.</p>
<p>For example, to disable use of symmetry in Fock matrix construction:</p>
<pre><code> sym off
</code></pre>
<h2 id="hartree-fock-theory-for-molecules-adapt-symmetry-adaptation-of-mos">ADAPT: symmetry adaptation of MOs<a class="headerlink" href="#hartree-fock-theory-for-molecules-adapt-symmetry-adaptation-of-mos" title="Permanent link">&para;</a></h2>
<pre><code> ADAPT &lt;string (ON||OFF) default ON&gt;
</code></pre>
<p>The default in the SCF module calculation is to force symmetry adaption
of the molecular orbitals. This does not affect the speed of the
calculation, but without explicit adaption the resulting orbitals may be
symmetry contaminated for some problems. This is especially likely if
the calculation is started using orbitals from a distorted geometry.</p>
<p>The underlying assumption in the use of symmetry in Fock matrix
construction is that the density is totally symmetric. If the orbitals
are symmetry contaminated, this assumption may not be valid &ndash; which
could result in incorrect energies and poor convergence of the
calculation. It is thus advisable when specifying <code>ADAPT OFF</code> to also
specify <code>SYM OFF</code> (<a href="#hartree-fock-theory-for-molecules-sym-use-of-symmetry">Use of Symmetry</a>).</p>
<h2 id="hartree-fock-theory-for-molecules-tol2e-integral-screening-threshold">TOL2E: integral screening threshold<a class="headerlink" href="#hartree-fock-theory-for-molecules-tol2e-integral-screening-threshold" title="Permanent link">&para;</a></h2>
<pre><code> TOL2E &lt;real tol2e default min(10e-7 , 0.01*thresh)&gt;
</code></pre>
<p>The variable tol2e is used in determining the integral screening
threshold for the evaluation of the energy and related Fock-like
matrices. The Schwarz inequality is used to screen the product of
integrals and density matrices in a manner that results in an accuracy
in the energy and Fock matrices that approximates the value specified
for <code>tol2e</code>.</p>
<p>It is generally not necessary to set this parameter directly. Specify
instead the required precision in the wavefunction, using the <code>THRESH</code>
directive (<a href="#hartree-fock-theory-for-molecules-thresh-convergence-threshold">Convergence
threshold</a>). The default
threshold is the minimum of 10<sup>-7</sup> and 0.01 times the requested
convergence threshold for the SCF calculation (<a href="#hartree-fock-theory-for-molecules-thresh-convergence-threshold">Convergence
threshold</a>).</p>
<p>The input to specify the threshold explicitly within the SCF directive
is, for example:</p>
<pre><code> tol2e 1e-9
</code></pre>
<p>For very diffuse basis sets, or for high-accuracy calculations it might
be necessary to set this parameter. A value of 10<sup>-12</sup> is
sufficient for nearly all such
purposes.</p>
<h2 id="hartree-fock-theory-for-molecules-vectors-inputoutput-of-mo-vectors">VECTORS: input/output of MO vectors<a class="headerlink" href="#hartree-fock-theory-for-molecules-vectors-inputoutput-of-mo-vectors" title="Permanent link">&para;</a></h2>
<pre><code> VECTORS [[input] (&lt;string input_movecs default atomic&gt;) || \  
                  (project &lt;string basisname&gt; &lt;string filename&gt;) || \  
                  (fragment &lt;string file1&gt; [&lt;string file2&gt; ...])] \ 
         [swap [alpha||beta] &lt;integer vec1 vec2&gt; ...] \  
         [reorder &lt;integer atom1 atom2&gt; ...] \  
         [output &lt;string output_filename default input_movecs&gt;] \  
         [lock]  
         [rotate  &lt;string input_geometry&gt; &lt;string input_movecs&gt;]
</code></pre>
<p>The VECTORS directive allows the user to specify the source and
destination of the molecular orbital vectors. In a startup calculation
(see <a href="#start_restart">START</a>),
the default source for guess vectors is a diagonalized Fock matrix
constructed from a superposition of the atomic density matrices for the
particular problem. This is usually a very good guess. For a restarted
calculation, the default is to use the previous MO vectors.</p>
<p>The optional keyword INPUT allows the user to specify the source of the
input molecular orbital vectors as any of the following:</p>
<ul>
<li><code>ATOMIC</code> &ndash; eigenvectors of a Fock-like matrix formed from a
    superposition of the atomic densities (the default guess). See
    <a href="#hartree-fock-theory-for-molecules-atomic-guess-orbitals-with-charged-atoms">Atomic
    guess</a> and
    <a href="#hartree-fock-theory-for-molecules-accuracy-of-initial-guess">Accuracy of initial guess</a>.</li>
<li><code>HCORE</code> &ndash; eigenvectors of the bare-nucleus Hamiltonian or the
    one-electron Hamiltonian.</li>
<li>filename &ndash; the name of a file containing the MO vectors from a
    previous calculation. Note that unless the path is fully qualified,
    or begins with a dot (&ldquo;.&rdquo;), then it is assumed to reside in the
    directory for permanent files (see <a href="#permanent_dir">File
    directories</a>).</li>
<li><code>PROJECT basisname filename</code> &ndash; projects the existing MO vectors in
    the file filename from the smaller basis with name basisname into
    the current basis. The definition of the basis basisname must be
    available in the current database, and the basis must be smaller
    than the current basis. In addition, the geometry used for the
    previous calculations must have the atoms in the same order and in
    the same orientation as the current geometry.</li>
<li><code>FRAGMENT file1 ...</code> &ndash; assembles starting MO vectors from previously
    performed calculations on fragments of the system and is described
    in more detail in <a href="#hartree-fock-theory-for-molecules-vectors-fragment-superposition-of-fragment-molecular-orbitals">Superposition of fragment molecular
    orbitals</a>.
    Even though there are some significant restrictions in the use of
    the initial implementation of this method,
    this is the most powerful initial guess option within the code. It
    is very effective for open shell metallic systems.</li>
<li><code>ROTATE input_geometry input_movecs</code> &ndash; rotates MO vectors generated
    at a previous geometry to the current active geometry.</li>
</ul>
<p>The molecular orbitals are saved every iteration if more than 600
seconds have elapsed, and also at the end of the calculation. At
completion (converged or not), the SCF module always canonically
transforms the molecular orbitals by separately diagonalizing the
closed-closed, open-open, and virtual-virtual blocks of the Fock matrix.</p>
<p>The name of the file used to store the MO vectors is determined as
follows:</p>
<ul>
<li>if the <code>OUTPUT</code> keyword was specified on the <code>VECTORS</code> directive, then
    the filename that follows this keyword is used, or</li>
<li>if the input vectors were read from a file, this file is reused for
    the output vectors (overwriting the input vectors); else,</li>
<li>a default file name is generated in the directory for permanent
    files (<a href="#permanent_dir">File
    directories</a>)
    by prepending &ldquo;.movecs&rdquo; with the file prefix, i.e.,
    &ldquo;<file_prefix>.movecs&rdquo;.</li>
</ul>
<p>The name of this file is stored in the database so that a subsequent SCF
calculation will automatically restart from these MO vectors.</p>
<p>Applications of this directive are illustrated in the following
examples.</p>
<p>Example 1:</p>
<pre><code> vectors output h2o.movecs
</code></pre>
<p>Assuming a start-up calculation, this directive will result in use of
the default atomic density guess, and will output the vectors to the
file h2o.movecs.</p>
<p>Example 2:</p>
<pre><code> vectors input initial.movecs output final.movecs
</code></pre>
<p>This directive will result in the initial vectors being read from the
file &ldquo;initial.movecs&rdquo;. The results will be written to the file
final.movecs. The contents of &ldquo;initial.movecs&rdquo; will not be changed.</p>
<p>Example 3:</p>
<pre><code> vectors input project &quot;small basis&quot; small.movecs
</code></pre>
<p>This directive will cause the calculation to start from vectors in the
file &ldquo;small.movecs&rdquo; which are in a basis named &ldquo;small basis&rdquo;. The output
vectors will be written to the default file &ldquo;<file_prefix.movecs>&ldquo;.</p>
<h3 id="hartree-fock-theory-for-molecules-vectors-swap-keyword">VECTORS SWAP keyword<a class="headerlink" href="#hartree-fock-theory-for-molecules-vectors-swap-keyword" title="Permanent link">&para;</a></h3>
<p>Once starting vectors have been obtained using any of the possible
options, they may be reordered through use of the SWAP keyword. This
optional keyword requires a list of orbital pairs that will be swapped.
For UHF calculations, separate SWAP keywords may be provided for the
alpha and beta orbitals, as necessary.</p>
<p>An example of use of the SWAP directive:</p>
<pre><code> vectors input try1.movecs swap 173 175 174 176 output try2.movecs
</code></pre>
<p>This directive will cause the initial orbitals to be read from the file
&ldquo;try1.movecs&rdquo;. The vectors for the orbitals within the pairs 173-175
will be swapped with those within 174-176, so the resulting order is
175, 176, 173, 174. The final orbitals obtained in the calculation will
be written to the file &ldquo;try2.movecs&rdquo;.</p>
<p>The swapping of orbitals occurs as a sequential process in the order
(left to right) input by the user. Thus, regarding each pair as an
elementary transposition it is possible to construct arbitrary
permutations of the orbitals. For instance, to apply the permutation
<code>(6 7 8 9)</code> we note that this permutation is equal to
<code>(6 7)(7 8)(8 9)</code>, and thus may be specified as</p>
<pre><code> vectors swap 8 9  7 8  6 7
</code></pre>
<p>Another example, now illustrating this feature for a UHF calculation, is
the directive</p>
<pre><code> vectors swap beta 4 5 swap alpha 5 6
</code></pre>
<p>This input will result in the swapping of the 5-6 alpha orbital pair and
the 4-5 beta orbital pair. (All other items in the input use the default
values.)</p>
<h3 id="hartree-fock-theory-for-molecules-vectors-lock-keyword">VECTORS LOCK keyword<a class="headerlink" href="#hartree-fock-theory-for-molecules-vectors-lock-keyword" title="Permanent link">&para;</a></h3>
<p>The LOCK keyword allows the user to specify that the ordering of
orbitals will be locked to that of the initial vectors, insofar as
possible. The default is to order by ascending orbital energies within
each orbital space. One application where locking might be desirable is
a calculation where it is necessary to preserve the ordering of a
previous geometry, despite flipping of the orbital energies. For such a
case, the LOCK directive can be used to prevent the SCF calculation from
changing the ordering, even if the orbital energies change.</p>
<h3 id="hartree-fock-theory-for-molecules-vectors-reorder-keyword">VECTORS REORDER keyword<a class="headerlink" href="#hartree-fock-theory-for-molecules-vectors-reorder-keyword" title="Permanent link">&para;</a></h3>
<p>The mapping of the MO&rsquo;s to the nuclei can be changed using the REORDER
keyword. Once starting vectors have been obtained using any of the
possible options, the REORDER keyword moves the MO coefficients between
atoms listed in the integer list. This keyword is particularly useful
for calculating localized electron and hole states.</p>
<p>This optional keyword requires a list containing the new atom ordering.
It is not necessary to provide separate lists for alpha and beta
orbitals.</p>
<p>An example of use of the REORDER keyword:</p>
<pre><code> vectors input try1.movecs reorder 2 1 output try2.movecs
</code></pre>
<p>This directive will cause the initial orbitals to be read from the file
&ldquo;try1.movecs&rdquo;. The MO coefficients for the basis functions on atom 2
will be swapped with those on atom 1. The final orbitals obtained in the
calculation will be written to the file &ldquo;try2.movecs&rdquo;.</p>
<h3 id="hartree-fock-theory-for-molecules-vectors-rotate-keyword">VECTORS ROTATE keyword<a class="headerlink" href="#hartree-fock-theory-for-molecules-vectors-rotate-keyword" title="Permanent link">&para;</a></h3>
<p>The following example shows how the ROTATE keyword can be used to rotate
MO vectors calculated at geometry <code>geom1</code> to geometry <code>geom2</code>, which has a
different rotational orientation:</p>
<pre><code>set geometry geom1
dft
 vectors input atomic output geom1.mo
end
task dft
set geometry geom2
dft
 vectors input rotate geom1 geom1.mo output geom2.mo
end
task dft
</code></pre>
<h3 id="hartree-fock-theory-for-molecules-vectors-fragment-superposition-of-fragment-molecular-orbitals">VECTORS FRAGMENT: Superposition of fragment molecular orbitals<a class="headerlink" href="#hartree-fock-theory-for-molecules-vectors-fragment-superposition-of-fragment-molecular-orbitals" title="Permanent link">&para;</a></h3>
<p>The fragment initial guess is particularly useful in the following
instances:</p>
<ul>
<li>The system naturally decomposes into molecules that can be treated
    individually, e.g., a cluster.</li>
<li>One or more fragments are particularly hard to converge and
    therefore much time can be saved by converging them independently.</li>
<li>A fragment (e.g., a metal atom) must be prepared with a specific
    occupation. This can often be readily accomplished with a
    calculation on the fragment using dummy charges to model a ligand
    field.</li>
<li>The molecular occupation predicted by the atomic initial guess is
    often wrong for systems with heavy metals which may have partially
    occupied orbitals with lower energy than some doubly occupied
    orbitals. The fragment initial guess avoids this problem.</li>
</ul>
<pre><code> VECTORS [input] fragment &lt;string file1&gt; [&lt;string file2&gt; ...]
</code></pre>
<p>The molecular orbitals are formed by superimposing the previously
generated orbitals of fragments of the molecule being studied. These
fragment molecular orbitals must be in the same basis as the current
calculation. The input specifies the files containing the fragment
molecular orbitals. For instance, in a calculation on the water dimer,
one might specify</p>
<pre><code> vectors fragment h2o1.movecs h2o2.movecs
</code></pre>
<p>where h2o1.movecs contains the orbitals for the first fragment, and
h2o2.movecs contains the orbitals for the second fragment.</p>
<p>A complete example of the input for a calculation on the water dimer
using the fragment guess is as follows:</p>
<pre><code>start dimer
title &quot;Water dimer SCF using fragment initial guess&quot;
geometry dimer
  O   -0.595   1.165  -0.048
  H    0.110   1.812  -0.170
  H   -1.452   1.598  -0.154
  O    0.724  -1.284   0.034
  H    0.175  -2.013   0.348
  H    0.177  -0.480   0.010
end
geometry h2o1
  O   -0.595   1.165  -0.048
  H    0.110   1.812  -0.170
  H   -1.452   1.598  -0.154
end
geometry h2o2
  O    0.724  -1.284   0.034
  H    0.175  -2.013   0.348
  H    0.177  -0.480   0.010
end
basis
  o library 3-21g
  h library 3-21g
end
set geometry h2o1
scf; vectors input atomic output h2o1.movecs; end
task scf
set geometry h2o2
scf; vectors input atomic output h2o2.movecs; end
task scf
set geometry dimer
scf
vectors input fragment h2o1.movecs h2o2.movecs \
        output dimer.movecs
end
task scf
</code></pre>
<p>First, the geometry of the dimer and the two monomers are specified and
given names. Then, after the basis specification, calculations are
performed on the fragments by setting the geometry to the appropriate
fragment
(<a href="#set">SET</a>)
and redirecting the output molecular orbitals to an appropriately named
file. Note also that use of the atomic initial guess is forced, since
the default initial guess is to use any existing MOs which would not be
appropriate for the second fragment calculation. Finally, the dimer
calculation is performed by specifying the dimer geometry, indicating
use of the fragment guess, and redirecting the output MOs.</p>
<p>The following points are important in using the fragment initial guess:</p>
<ol>
<li>The fragment calculations must be in the same basis set as the full
    calculation.</li>
<li>The order of atoms in the fragments and the order in which the
    fragment files are specified must be such that when the fragment
    basis sets are concatenated all the basis functions are in the same
    order as in the full system. This is readily accomplished by first
    generating the full geometry with atoms for each fragment
    contiguous, splitting this into numbered fragments and specifying
    the fragment MO files in the correct order on the <code>VECTORS</code> directive.</li>
<li>The occupation of orbitals is preserved when they are merged from
    the fragments to the full molecule and the resulting occupation must
    match the requested occupation for the full molecule. E.g., a
    triplet ROHF calculation must be comprised of fragments that have a
    total of exactly two open-shell orbitals.</li>
<li>Because of these restrictions, it is not possible to introduce
    additional atoms (or basis functions) into fragments for the purpose
    of cleanly breaking real bonds. However, it is possible, and highly
    recommended, to introduce additional point charges to simulate the
    presence of other fragments.</li>
<li>MO vectors of partially occupied or strongly polarized systems are
    very sensitive to orientation. While it is possible to specify the
    same fragment MO vector file multiple times in the <code>VECTORS</code>
    directive, it is usually much better to do a separate calculation
    for each fragment.</li>
<li>Linear dependencies which were present in a fragment calculation may
    be magnified in the full calculation. When this occurs, some of the
    fragment&rsquo;s highest virtual orbitals will not be copied to the full
    system, and a warning will be printed.</li>
</ol>
<p>A more involved example is now presented. We wish to model the sextet
state of Fe(III) complexed with water, imidazole and a heme with a net
unit positive charge. The default atomic guess does not give the correct <em>d<sup>5</sup></em>
 occupation for the metal and also gives an incorrect state for
the double anion of the heme. The following performs calculations on all
of the fragments. Things to note are:</p>
<ol>
<li>The use of a dummy <em>+2</em> charge in the initial guess on the heme
    which in part simulates the presence of the metal ion, and also
    automatically forces an additional two electrons to be added to the
    system (the default net charge being zero).</li>
<li>The iron fragment calculation (charge +3, <em>d<sup>5</sup></em>, sextet) will
    yield the correct open-shell occupation for the full system. If,
    instead, the d-orbitals were partially occupied (e.g., the doublet
    state) it would be useful to introduce dummy charges around the iron
    to model the ligand field and thereby lift the degeneracy to obtain
    the correct occupation.</li>
<li><em>C<sub>s</sub></em> symmetry is used for all of the calculations. It is not
    necessary that the same symmetry be used in all of the calculations,
    provided that the order and orientation of the atoms is preserved.</li>
<li>The unset scf:* directive is used immediately before the
    calculation on the full system so that the default name for the
    output MO vector file can be used, rather than having to specify it
    explicitly.</li>
</ol>
<pre><code>start heme6a1
title  &quot;heme-H2O (6A1) from M.Dupuis&quot;
############################################################
# Define the geometry of the full system and the fragments #
############################################################
geometry full-system
  symmetry cs
  H     0.438   -0.002    4.549
  C     0.443   -0.001    3.457
  C     0.451   -1.251    2.828
  C     0.452    1.250    2.828
  H     0.455    2.652    4.586
  H     0.461   -2.649    4.586
  N1    0.455   -1.461    1.441
  N1    0.458    1.458    1.443
  C     0.460    2.530    3.505
  C     0.462   -2.530    3.506
  C     0.478    2.844    1.249
  C     0.478    3.510    2.534
  C     0.478   -2.848    1.248
  C     0.480   -3.513    2.536
  C     0.484    3.480    0.000
  C     0.485   -3.484    0.000
  H     0.489    4.590    2.664
  H     0.496   -4.592    2.669
  H     0.498    4.573    0.000
  H     0.503   -4.577    0.000
  H    -4.925    1.235    0.000
  H    -4.729   -1.338    0.000
  C    -3.987    0.685    0.000
  N    -3.930   -0.703    0.000
  C    -2.678    1.111    0.000
  C    -2.622   -1.076    0.000
  H    -2.284    2.126    0.000
  H    -2.277   -2.108    0.000
  N    -1.838    0.007    0.000
  Fe    0.307    0.000    0.000
  O     2.673   -0.009    0.000
  H     3.238   -0.804    0.000
  H     3.254    0.777    0.000
end
geometry ring-only
  symmetry cs
  H     0.438   -0.002    4.549
  C     0.443   -0.001    3.457
  C     0.451   -1.251    2.828
  C     0.452    1.250    2.828
  H     0.455    2.652    4.586
  H     0.461   -2.649    4.586
  N1    0.455   -1.461    1.441
  N1    0.458    1.458    1.443
  C     0.460    2.530    3.505
  C     0.462   -2.530    3.506
  C     0.478    2.844    1.249
  C     0.478    3.510    2.534
  C     0.478   -2.848    1.248
  C     0.480   -3.513    2.536
  C     0.484    3.480    0.000
  C     0.485   -3.484    0.000
  H     0.489    4.590    2.664
  H     0.496   -4.592    2.669
  Bq    0.307    0.0      0.0    charge 2  # simulate the iron
end
geometry imid-only
  symmetry cs
  H     0.498    4.573    0.000
  H     0.503   -4.577    0.000
  H    -4.925    1.235    0.000
  H    -4.729   -1.338    0.000
  C    -3.987    0.685    0.000
  N    -3.930   -0.703    0.000
  C    -2.678    1.111    0.000
  C    -2.622   -1.076    0.000
  H    -2.284    2.126    0.000
  H    -2.277   -2.108    0.000
  N    -1.838    0.007    0.000
end
geometry fe-only
  symmetry cs
  Fe    .307    0.000    0.000
end
geometry water-only
  symmetry cs
  O     2.673   -0.009    0.000
  H     3.238   -0.804    0.000
  H     3.254    0.777    0.000
end
############################
# Basis set for everything #
############################
basis nosegment
 O  library 6-31g*
 N  library 6-31g*
 C  library 6-31g*
 H  library 6-31g*
 Fe  library &quot;Ahlrichs pVDZ&quot;
end
##########################################################
# SCF on the fragments for initial guess for full system #
##########################################################
scf; thresh 1e-2; end
set geometry ring-only
scf; vectors atomic swap 80 81 output ring.mo; end
task scf
set geometry water-only
scf; vectors atomic output water.mo; end
task scf
set geometry imid-only
scf; vectors atomic output imid.mo; end
task scf
charge 3
set geometry fe-only
scf; sextet; vectors atomic output fe.mo; end
task scf
##########################
# SCF on the full system #
##########################
unset scf:*     # This restores the defaults
charge 1
set geometry full-system
scf
 sextet
 vectors fragment ring.mo imid.mo fe.mo water.mo
 maxiter 50
end
task scf
</code></pre>
<h3 id="hartree-fock-theory-for-molecules-example-of-projecting-smaller-basis-into-larger-basis">Example of projecting smaller basis into larger basis<a class="headerlink" href="#hartree-fock-theory-for-molecules-example-of-projecting-smaller-basis-into-larger-basis" title="Permanent link">&para;</a></h3>
<p>Key ingredient needed: definition of both the smaller and the larger
basis set, plus mention of the small basis set in the &ldquo;<code>input project</code>&rdquo;
line.</p>
<pre><code>    start he

    geometry
     he 0 0 0
     symmetry oh
    end

    basis  small
     * library sto-3g
    end
    basis  large
     * library 3-21g
    end

    set &quot;ao basis&quot; small
    scf
     vectors input atomic output small.mos
    end
    task scf  

    set &quot;ao basis&quot; large
    scf
     vectors input project small small.mos output large.mos
    end
    task scf
</code></pre>
<h3 id="hartree-fock-theory-for-molecules-atomic-guess-orbitals-with-charged-atoms">Atomic guess orbitals with charged atoms<a class="headerlink" href="#hartree-fock-theory-for-molecules-atomic-guess-orbitals-with-charged-atoms" title="Permanent link">&para;</a></h3>
<p>As noted above, the default guess vectors are based on superimposing the
density matrices of the neutral atoms. If some atoms are significantly
charged, this default guess may be improved upon by modifying the atomic
densities. This is done by setting parameters that add fractional
charges to the occupation of the valence atomic orbitals. Since the
atomic SCF program does not have its own input block, the SET directive
(<a href="#set">SET</a>) must be used to set these
parameters.</p>
<p>The input specifies a list of tags (i.e., names of atoms in a geometry,
see <a href="#geometry">GEOMETRY</a>) and the charges to be
added to those centers. Two parameters must be set as follows:</p>
<pre><code> set atomscf:tags_z &lt;string list_of_tags&gt;  
 set atomscf:z      &lt;real list_of_charges&gt;
</code></pre>
<p>The array of strings atomscf:tags_z should be set to the list of tags,
and the array atomscf:z should be set to the list of charges which must
be real numbers (not integers). All atoms that have a tag specified in
the list of tags will be assigned the corresponding charge from the list
of charges.</p>
<p>For example, the following specifies that all oxygen atoms with tag O be
assigned a charge of -1 and all iron atoms with tag Fe be assigned a
charge of +2</p>
<pre><code> set atomscf:z        -1  2.0  
 set atomscf:tags_z    O  Fe
</code></pre>
<p>There are some limitations to this feature. It is not possible to add
electrons to closed shell atoms, nor is it possible to remove all
electrons from a given atom. Attempts to do so will cause the code to
report an error, and it will not report further errors in the input for
modifying the charge even when they are detected.</p>
<p>Finally, recall that the database is persistent (<a href="#nwchem-architecture-persistence-of-data-and-restart">Data
persistence</a>)
and that the modified settings will be used in subsequent atomic guess
calculations unless the data is deleted from the database with the UNSET
directive
(<a href="#unset">UNSET</a>).</p>
<h2 id="hartree-fock-theory-for-molecules-accuracy-of-initial-guess">Accuracy of initial guess<a class="headerlink" href="#hartree-fock-theory-for-molecules-accuracy-of-initial-guess" title="Permanent link">&para;</a></h2>
<p>For SCF, the initial Fock-matrix construction from the atomic guess is
performed to a default precision of 1e-7. However, other wavefunctions,
notably DFT, use a lower precision. In charged, or diffuse basis sets,
this precision may not be sufficient and could result in incorrect
ordering of the initial orbitals. The accuracy may be increased with the
following directive which should be inserted in the top-level of input
(i.e., outside of the SCF input block) and before the TASK directive.</p>
<pre><code> set tolguess 1e-7
</code></pre>
<h2 id="hartree-fock-theory-for-molecules-thresh-convergence-threshold">THRESH &ndash; convergence threshold<a class="headerlink" href="#hartree-fock-theory-for-molecules-thresh-convergence-threshold" title="Permanent link">&para;</a></h2>
<pre><code> THRESH  &lt;real thresh default 1.0e-4&gt;
</code></pre>
<p>This directive specifies the convergence threshold for the calculation.
The convergence threshold is the norm of the orbital gradient, and has a
default value in the code of 10<sup>-4</sup>.  </p>
<p>The norm of the orbital gradient corresponds roughly to the precision
available in the wavefunction, and the energy should be converged to
approximately the square of this number. It should be noted, however,
that the precision in the energy will not exceed that of the integral
screening tolerance. This tolerance (<a href="#hartree-fock-theory-for-molecules-tol2e-integral-screening-threshold">Integral screening
threshold</a>) is
automatically set from the convergence threshold, so that sufficient
precision is usually available by default.</p>
<p>The default convergence threshold suffices for most SCF energy and
geometry optimization calculations, providing about 6-8 decimal places
in the energy, and about four significant figures in the density and
energy derivative with respect to nuclear coordinates. However, greater
precision may be required for calculations involving weakly interacting
systems, floppy molecules, finite-difference of gradients to compute the
Hessian, and for post-Hartree-Fock calculations. A threshold of 10<sup>-6</sup>
 is adequate for most such purposes, and a threshold of 10<sup>-8</sup>
 might be necessary for very high accuracy or very weak
interactions. A threshold of 10<sup>-8</sup> should be regarded as the best
that can be attained in most circumstances.</p>
<h2 id="hartree-fock-theory-for-molecules-maxiter-iteration-limit">MAXITER &ndash; iteration limit<a class="headerlink" href="#hartree-fock-theory-for-molecules-maxiter-iteration-limit" title="Permanent link">&para;</a></h2>
<pre><code> MAXITER &lt;integer maxiter default 8&gt;
</code></pre>
<p>The maximum number of iterations for the SCF calculation defaults to 20
for both ROHF/RHF and UHF calculations. For most molecules, this number
of iterations is more than sufficient for the quadratically convergent
SCF algorithm to obtain a solution converged to the default threshold
(see <a href="#hartree-fock-theory-for-molecules-thresh-convergence-threshold">Convergence
threshold</a> above). If the
SCF program detects that the quadratically convergent algorithm is not
efficient, then it will resort to a linearly convergent algorithm and
increase the maximum number of iterations by 10.</p>
<p>Convergence may not be reached in the maximum number of iterations for
many reasons, including input error (e.g., an incorrect geometry or a
linearly dependent basis), a very low convergence threshold, a poor
initial guess, or the fact that the system is intrinsically hard to
converge due to the presence of many states with similar energies.</p>
<p>The following sets the maximum number of SCF iterations to 50:</p>
<pre><code> maxiter 50
</code></pre>
<h2 id="hartree-fock-theory-for-molecules-profile-performance-profile">PROFILE &ndash; performance profile<a class="headerlink" href="#hartree-fock-theory-for-molecules-profile-performance-profile" title="Permanent link">&para;</a></h2>
<p>This directive allows the user to obtain timing and parallel execution
information about the SCF module. It is specified by the simple keyword</p>
<pre><code> PROFILE
</code></pre>
<p>This option can be helpful in understanding the computational
performance of an SCF calculation. However, it can introduce a
significant overhead on machines that have expensive timing routines,
such as the SUN.</p>
<h2 id="hartree-fock-theory-for-molecules-diis-diis-convergence">DIIS &ndash; DIIS convergence<a class="headerlink" href="#hartree-fock-theory-for-molecules-diis-diis-convergence" title="Permanent link">&para;</a></h2>
<p>This directive allows the user to specify DIIS convergence rather than
second-order convergence for the SCF calculation. The form of the
directive is as follows:</p>
<pre><code> DIIS
</code></pre>
<p>The implementation of this option is currently fairly rudimentary. It
does not have level-shifting and damping, and does not support open
shells or UHF. It is provided on an &ldquo;as is&rdquo; basis, and should be used
with caution.</p>
<p>When the DIIS directive is specified in the input, the user has the
additional option of specifying the size of the subspace for the DIIS
extrapolation. This is accomplished with the DIISBAS directive, which is
of the form:</p>
<pre><code> DIISBAS &lt;integer diisbas default 5&gt;
</code></pre>
<p>The default of 5 should be adequate for most applications, but may be
increased if convergence is poor. On large systems, it may be necessary
to specify a lower value for diisbas, to conserve memory.</p>
<h2 id="hartree-fock-theory-for-molecules-direct-and-semidirect-recomputation-of-integrals">DIRECT and SEMIDIRECT: recomputation of integrals<a class="headerlink" href="#hartree-fock-theory-for-molecules-direct-and-semidirect-recomputation-of-integrals" title="Permanent link">&para;</a></h2>
<p>In the context of SCF calculations direct means that all integrals are
recomputed as required and none are stored. The other extreme are disk-
or memory-resident (sometimes termed conventional) calculations in which
all integrals are computed once and stored. Semi-direct calculations are
between these two extremes with some integrals being precomputed and
stored, and all other integrals being recomputed as necessary.</p>
<p>The default behavior of the SCF module is</p>
<ul>
<li>If enough memory is available, the integrals are computed once and
    are cached in memory.</li>
<li>If there is not enough memory to store all the integrals at once,
    then 95% of the available disk space in the scratch directory (see
    <a href="#scratch_dir">File
    directories</a>)
    is assumed to be available for this purpose, and as many integrals
    as possible are cached on disk (with no memory being used for
    caching). Some attempt is made to store the most expensive integrals
    in the cache.</li>
<li>If there is not enough room in memory or on disk for all the
    integrals, then the ones that are not cached are recomputed in a
    semidirect fashion.</li>
</ul>
<p>The integral file is deleted at the end of a calculation, so it is not
possible to restart a semidirect calculation when the integrals are
cached in memory or on disk. Many modern computer systems clear the fast
scratch space at the end of each job, adding a further complication to
the problem of restarting a parallel semidirect calculation.</p>
<p>A fully direct calculation (with recomputation of the integrals at each
iteration) is forced by specifying the directive</p>
<pre><code> DIRECT
</code></pre>
<p>Alternatively, the SEMIDIRECT directive can be used to control the
default semidirect calculation by defining the amount of disk space and
the cache memory size. The form of this directive is as follows:</p>
<pre><code>  SEMIDIRECT [filesize &lt;integer filesize default disksize&gt;]   
             [memsize  &lt;integer memsize default available&gt;]  
             [filename &lt;string filename default $file_prefix.aoints$]
</code></pre>
<p>The keyword FILESIZE allows the user to specify the amount of disk space
to be used per process for storing the integrals in 64-bit words.
Similarly, the keyword MEMSIZE allows the user to specify the number of
64-bit words to be used per process for caching integrals in memory.
(Note: If the amount of storage space specified by the entry for memsize
is not available, the code cuts the value in half and checks again for
available space. This process is repeated until the request is
satisfied.)</p>
<p>By default, the integral files are placed into the scratch directory
(see <a href="#scratch_dir">File
directories</a>).
Specifying the keyword FILENAME overrides this default. The
user-specified name entered in the string filename has the process
number appended to it, so that each process has a distinct file but with
a common base-name and directory. Therefore, it is not possible to use
this keyword to specify different disks for different processes. The
<code>SCRATCH_DIR</code> directive (see <a href="#scratch_dir">File
directories</a>)
can be used for this purpose.</p>
<p>For example, to force full recomputation of all integrals:</p>
<pre><code> direct
</code></pre>
<p>Exactly the same result could be obtained by entering the directive:</p>
<pre><code> semidirect filesize 0 memsize 0
</code></pre>
<p>To disable the use of memory for caching integrals and limit disk usage
by each process to 100 megawords (MW):</p>
<pre><code> semidirect memsize 0 filesize 100000000
</code></pre>
<p>The integral records are typically 32769 words long and any non-zero
value for filesize or memsize should be enough to hold at least one
record.</p>
<h3 id="hartree-fock-theory-for-molecules-integral-file-size-and-format-for-the-scf-module">Integral File Size and Format for the SCF Module<a class="headerlink" href="#hartree-fock-theory-for-molecules-integral-file-size-and-format-for-the-scf-module" title="Permanent link">&para;</a></h3>
<p>The file format is rather complex, since it accommodates a variety of
packing and compression options and the distribution of data. This
section presents some information that may help the user understand the
output, and illustrates how to use the output information to estimate
file sizes.</p>
<p>If integrals are stored with a threshold of greater than 10<sup>-10</sup>,
then the integrals are stored in a 32-bit fixed-point format (with
appropriate treatment for large values to retain precision). If
integrals are stored with a threshold less than 10<sup>-10</sup>, however,
the values are stored in 64-bit floating-point format. If a
replicated-data calculation is being run, then 8 bits are used for each
basis function label, unless there are more than 256 functions, in which
case 16 bits are used. If distributed data is being used, then the
labels are always packed to 8 bits (the distributed blocks always being
less than 256; labels are relative to the start of the block).</p>
<p>Thus, the number (<em>W</em>) of 64-bit words required to store <em>N</em> integrals,
may be computed as</p>
<table>
<thead>
<tr>
<th style="text-align: center;">no. 64-bit words</th>
<th style="text-align: center;">labels</th>
<th>values</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: center;">N</td>
<td style="text-align: center;">8-bit</td>
<td>32-bit</td>
</tr>
<tr>
<td style="text-align: center;">1.5N</td>
<td style="text-align: center;">16-bit</td>
<td>32-bit</td>
</tr>
<tr>
<td style="text-align: center;">1.5N</td>
<td style="text-align: center;">8-bit</td>
<td>64-bit</td>
</tr>
<tr>
<td style="text-align: center;">2N</td>
<td style="text-align: center;">16-bit</td>
<td>64-bit</td>
</tr>
</tbody>
</table>
<p>Table 1: number (<em>W</em>) of 64-bit words required to store <em>N</em> integrals  </p>
<p>The actual number of words required can exceed this computed value by up
to one percent, due to bookkeeping overhead, and because the file itself
is organized into fixed-size records.</p>
<p>With at least the default print level, all semidirect (not direct)
calculations will print out information about the integral file and the
number of integrals computed. The form of this output is as
follows:</p>
<pre><code>Integral file          = ./c6h6.aoints.0
Record size in doubles =  32769        No. of integs per rec  =  32768
Max. records in memory =      3        Max. records in file   =      5
No. of bits per label  =      8        No. of bits per value  =     32
#quartets = 2.0D+04  #integrals = 7.9D+05  direct = 63.6%  cached = 36.4%
</code></pre>
<p>The file information above relates only to process 0. The line of
information about the number of quartets, integrals, etc., is a sum over
all processes.</p>
<p>When the integral file is closed, additional information of the
following form is printed:</p>
<pre><code>------------------------------------------------------------
EAF file 0: &quot;./c6h6.aoints.0&quot; size=262152 bytes
------------------------------------------------------------
               write      read    awrite     aread      wait
               -----      ----    ------     -----      ----
     calls:        6        12         0         0         0
   data(b): 1.57e+06  3.15e+06  0.00e+00  0.00e+00
   time(s): 1.09e-01  3.12e-02                      0.00e+00
rate(mb/s): 1.44e+01  1.01e+02
------------------------------------------------------------
 Parallel integral file used       4 records with       0 large values
</code></pre>
<p>Again, the detailed file information relates just to process 0, but the
final line indicates the total number of integral records stored by all
processes.</p>
<p>This information may be used to optimize subsequent calculations, for
instance by assigning more memory or disk space.</p>
<h2 id="hartree-fock-theory-for-molecules-scf-convergence-control-options">SCF Convergence Control Options<a class="headerlink" href="#hartree-fock-theory-for-molecules-scf-convergence-control-options" title="Permanent link">&para;</a></h2>
<p>Note to users: It is desired that the SCF program converge reliably with
the default options for a wide variety of molecules. In addition, it
should be guaranteed to converge for any system, with sufficient
iterations.</p>
<p>The SCF program uses a preconditioned conjugate gradient (PCG) method
that is unconditionally convergent. Basically, a search direction is
generated by multiplying the orbital gradient (the derivative of the
energy with respect to the orbital rotations) by an approximation to the
inverse of the level-shifted orbital Hessian. In the initial iterations
(see <a href="#hartree-fock-theory-for-molecules-nr-controlling-the-newton-raphson">Controlling the Newton-Raphson</a>),
an inexpensive one-electron approximation to the inverse orbital Hessian is
used. Closer to convergence, the full orbital Hessian is used, which
should provide quadratic convergence. For both the full or one-electron
orbital Hessians, the inverse-Hessian matrix-vector product is formed
iteratively. Subsequently, an approximate line search is performed along
the new search direction. If the exact Hessian is being employed, then
the line search should require a single step (of unity). Preconditioning
with approximate Hessians may require additional steps, especially in
the initial iterations. It is the (approximate) line search that
provides the convergence guarantee. The iterations required to solve the
linear equations are referred to as micro-iterations. A macro-iteration
comprises both the iterative solution and a line search.</p>
<p>Level-shifting plays the same role in this algorithm as it does in the
conventional iterative solution of the SCF equations. The approximate
Hessian used for preconditioning should be positive definite. If this is
not the case, then level-shifting by a positive constant () serves to
make the preconditioning matrix positive definite, by adding  to all of
its eigenvalues. The level-shifts employed for the RHF orbital Hessian
should be approximately four times (only twice for UHF) the value that
one would employ in a conventional SCF. Level-shifting is automatically
enabled in the early iterations, and the default options suffice for
most test cases.</p>
<p>So why do things go wrong and what can be done to fix convergence
problems? Most problems encountered so far arise either poor initial
guesses or from small or negative eigenvalues of the orbital Hessian.
The atomic orbital guess is usually very good. However, in calculations
on charged systems, especially with open shells, incorrect initial
occupations may result. The SCF might then converge very slowly since
very large orbital rotations might be required to achieve the correct
occupation or move charge large distances in the molecule. Possible
actions are</p>
<ul>
<li>Modify the atomic guess by assigning charges to the atoms known to
    carry substantial charges (<a href="#hartree-fock-theory-for-molecules-atomic-guess-orbitals-with-charged-atoms">Atomic
    guess</a>)</li>
<li>Examining an analysis of the initial orbitals
    (<a href="#hartree-fock-theory-for-molecules-printing-information-from-the-scf-module">Printing</a>)
    and then swapping them to attain the desired occupation
    (<a href="#hartree-fock-theory-for-molecules-vectors-inputoutput-of-mo-vectors">VECTORS</a>).</li>
<li>Converging the calculation in a minimal basis set, which is usually
    easier, and then projecting into a larger basis set
    (<a href="#hartree-fock-theory-for-molecules-vectors-inputoutput-of-mo-vectors">VECTORS</a>).</li>
<li>Using the fragment orbital initial guess (<a href="#hartree-fock-theory-for-molecules-vectors-fragment-superposition-of-fragment-molecular-orbitals">Fragment molecular
    orbitals</a>).</li>
</ul>
<p>Small or negative Hessian eigenvalues can occur even though the
calculation seem to be close to convergence (as measured by the gradient
norm, or the off-diagonal Fock matrix elements). Small eigenvalues will
cause the iterative linear equation solver to converge slowly, resulting
in an excessive number of micro-iterations. This makes the SCF expensive
in terms of computation time, and it is possible to exceed the maximum
number of iterations without achieving the accuracy required for
quadratic convergence &ndash; which causes more macro-iterations to be
performed.</p>
<p>Two main options are available when a problem will not converge:
Newton-Raphson can be disabled temporarily or permanently (see
<a href="#hartree-fock-theory-for-molecules-nr-controlling-the-newton-raphson">Controlling the
Newton-Raphson</a>), and
level-shifting can be applied to the matrix (see
<a href="#hartree-fock-theory-for-molecules-level-level-shifting-the-orbital-hessian">Level-shifting</a>).
In some cases, both options may be necessary to achieve final
convergence.</p>
<p>If there is reason to suspect a negative eigenvalue, the first course is
to disable the Newton-Raphson iteration until the solution is closer to
convergence. It may be necessary to disable it completely. At some point
close to convergence, the Hessian will be positive definite, so
disabling Newton-Raphson should yield a solution with approximately the
same convergence rate as DIIS.</p>
<p>If temporarily disabling Newton-Raphson is not sufficient to achieve
convergence, it may be necessary to disable it entirely and apply a
small level-shift to the approximate Hessian. This should improve the
convergence rate of the micro-iterations and stabilize the
macro-iterations. The level-shifting will destroy exact quadratic
convergence, but the optimization process is automatically adjusted to
reflect this by enforcing conjugacy and reducing the accuracy to which
the linear equations are solved. The net result of this is that the
solution will do more macro-iterations, but each one should take less
time than it would with the unshifted Hessian.</p>
<p>The following sections describe the directives needed to disable the
Newton-Raphson iteration and specify level-shifting.</p>
<h2 id="hartree-fock-theory-for-molecules-nr-controlling-the-newton-raphson">NR: controlling the Newton-Raphson<a class="headerlink" href="#hartree-fock-theory-for-molecules-nr-controlling-the-newton-raphson" title="Permanent link">&para;</a></h2>
<pre><code>   NR &lt;real nr_switch default 0.1&gt;
</code></pre>
<p>The exact orbital Hessian is adopted as the preconditioner when the
maximum element of the orbital gradient is below the value specified for
nr_switch. The default value is 0.1, which means that Newton-Raphson
will be disabled until the maximum value of the orbital gradient (twice
the largest off-diagonal Fock matrix element) is less than 0.1. To
disable the Newton-Raphson entirely, the value of nr_switch must be set
to zero. The directive to accomplish this is as follows:</p>
<pre><code> nr 0
</code></pre>
<h2 id="hartree-fock-theory-for-molecules-level-level-shifting-the-orbital-hessian">LEVEL: level-shifting the orbital Hessian<a class="headerlink" href="#hartree-fock-theory-for-molecules-level-level-shifting-the-orbital-hessian" title="Permanent link">&para;</a></h2>
<p>This directive allows the user to specify level-shifting to obtain a
positive-definite preconditioning matrix for the SCF solution procedure.
Separate level shifts can be set for the first-order convergent
one-electron approximation to the Hessian used with the preconditioned
conjugate gradient (PCG) method, and for the full Hessian used with the
Newton-Raphson (NR) approach. It is also possible to change the
level-shift automatically as the solution attains some specified
accuracy. The form of the directive is as follows:</p>
<pre><code>  LEVEL [pcg &lt;real initial default 20.0&gt; \
          [&lt;real tol default 0.5&gt; &lt;real final default 0.0&gt;]] \
        [nr &lt;real initial default 0.0&gt; \
          [&lt;real tol default 0.0&gt; &lt;real final default 0.0&gt;]]
</code></pre>
<p>This directive contains only two keywords: one for the PCG method and
the other for the exact Hessian (Newton Raphson, or NR). Use of PCG or
NR is determined by the input specified for nr_switch on the NR
directive, <a href="#hartree-fock-theory-for-molecules-nr-controlling-the-newton-raphson">Controlling the
Newton-Raphson</a> above.</p>
<p>Specifying the keyword pcg on the LEVEL directive allows the user to
define the level shifting for the approximate (i.e., PCG) method.
Specifying the keyword nr allows the user to define the level shifting
for the exact Hessians. In both options, the initial level shift is
defined by the value specified for the variable initial. Optionally, tol
can be specified independently with each keyword to define the level of
accuracy that must be attained in the solution before the level shifting
is changed to the value specified by input in the real variable final.
Level shifts and gradient thresholds are specified in atomic units.</p>
<p>For the PCG method (as specified using the keyword <code>pcg</code>), the defaults
for this input are 20.0 for initial, 0.5 for tol, and 0.0 for final.
This means that the approximate Hessian will be shifted by 20.0 until
the maximum element of the gradient falls below 0.5, at which point the
shift will be set to zero.</p>
<p>For the exact Hessian (as specified using the keyword <code>nr</code>), the defaults
are all zero. The exact Hessian is usually not shifted since this
destroys quadratic convergence. An example of an input directive that
applies a shift of 0.2 to the exact Hessian is as follows:</p>
<pre><code> level nr 0.2
</code></pre>
<p>To apply this shift to the exact Hessian only until the maximum element
of the gradient falls below 0.005, the required input directive is as
follows:</p>
<pre><code> level nr 0.2 0.005 0
</code></pre>
<p>Note that in both of these examples, the parameters for the PCG method
are at the default values. To obtain values different from the defaults,
the keyword pcg must also be specified. For example, to specify the
level shifting in the above example for the exact Hessian and
non-default shifting for the PCG method, the directive would be
something like the following:</p>
<pre><code> level pcg 20 0.3 0.0 nr 0.2 0.005 0.0
</code></pre>
<p>This input will cause the PCG method to be level-shifted by 20.0 until
the maximum element of the gradient falls below 0.3, then the shift will
be zero. For the exact Hessian, the level shifting is initially 0.2,
until the maximum element falls below 0.005, after which the shift is
zero.</p>
<p>The default options correspond to</p>
<pre><code> level pcg 20 0.5 0 nr 0 0 0
</code></pre>
<h2 id="hartree-fock-theory-for-molecules-orbital-localization">Orbital Localization<a class="headerlink" href="#hartree-fock-theory-for-molecules-orbital-localization" title="Permanent link">&para;</a></h2>
<p>The SCF module includes an experimental implementation of orbital
localization, including Foster-Boys and Pipek-Mezey which only works for
closed-shell (RHF) wavefunctions. There is currently no input in the SCF
block to control this so the SET directive
(<a href="#set">SET</a>) must be used.</p>
<p>The directive</p>
<pre><code> set scf:localize t
</code></pre>
<p>will separately localize the core, valence, and virtual orbital spaces
using the Pipek-Mezey algorithm. If the additional directive</p>
<pre><code> set scf:loctype FB
</code></pre>
<p>is included, then the Foster-boys algorithm is used. The partitioning of
core-orbitals is performed using the atomic information described in the section describing how to <a href="#mp2-freeze-freezing-orbitals">freeze the orbitals
</a>.</p>
<p>In the next release, this functionality will be extended to included all
wavefunctions using molecular orbitals.</p>
<h2 id="hartree-fock-theory-for-molecules-printing-information-from-the-scf-module">Printing Information from the SCF Module<a class="headerlink" href="#hartree-fock-theory-for-molecules-printing-information-from-the-scf-module" title="Permanent link">&para;</a></h2>
<p>All output from the SCF module is controlled using the PRINT directive
described in <a href="#print_noprint">Print
control</a>.
The following list describes the items from SCF that are currently under
direct print control, along with the print level for each
one.</p>
<table>
<thead>
<tr>
<th><strong>Name</strong></th>
<th><strong>Print Level</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>&ldquo;atomic guess density&rdquo;</td>
<td>debug</td>
<td>guess density matrix</td>
</tr>
<tr>
<td>&ldquo;atomic scf&rdquo;</td>
<td>debug</td>
<td>details of atomic SCF</td>
</tr>
<tr>
<td>&ldquo;mo guess&rdquo;</td>
<td>default</td>
<td>brief info from mo guess</td>
</tr>
<tr>
<td>&ldquo;information&rdquo;</td>
<td>low</td>
<td>results</td>
</tr>
<tr>
<td>&ldquo;initial vectors&rdquo;</td>
<td>debug</td>
<td></td>
</tr>
<tr>
<td>&ldquo;intermediate vectors&rdquo;</td>
<td>debug</td>
<td></td>
</tr>
<tr>
<td>&ldquo;final vectors&rdquo;</td>
<td>debug</td>
<td></td>
</tr>
<tr>
<td>&ldquo;final vectors analysis&rdquo;</td>
<td>default</td>
<td></td>
</tr>
<tr>
<td>&ldquo;initial vectors analysis&rdquo;</td>
<td>never</td>
<td></td>
</tr>
<tr>
<td>&ldquo;intermediate evals&rdquo;</td>
<td>debug</td>
<td></td>
</tr>
<tr>
<td>&ldquo;final evals&rdquo;</td>
<td>default</td>
<td></td>
</tr>
<tr>
<td>&ldquo;schwarz&rdquo;</td>
<td>high</td>
<td>integral screening info stats at completion</td>
</tr>
<tr>
<td>&ldquo;screening statistics&rdquo;</td>
<td>debug</td>
<td>display stats after every Fock build</td>
</tr>
<tr>
<td>&ldquo;geometry&rdquo;</td>
<td>high</td>
<td></td>
</tr>
<tr>
<td>&ldquo;symmetry&rdquo;</td>
<td>debug</td>
<td>detailed symmetry info</td>
</tr>
<tr>
<td>&ldquo;basis&rdquo;</td>
<td>high</td>
<td></td>
</tr>
<tr>
<td>&ldquo;geombas&rdquo;</td>
<td>debug</td>
<td>detailed basis map info</td>
</tr>
<tr>
<td>&ldquo;vectors i/o&rdquo;</td>
<td>default</td>
<td>report vectors I/O</td>
</tr>
<tr>
<td>&ldquo;parameters&rdquo;</td>
<td>default</td>
<td>convergence parameters</td>
</tr>
<tr>
<td>&ldquo;convergence&rdquo;</td>
<td>default</td>
<td>info each iteration</td>
</tr>
<tr>
<td>&ldquo;mulliken ao&rdquo;</td>
<td>never</td>
<td>Mulliken population of basis functions</td>
</tr>
</tbody>
</table>
<p>Table 2: SCF Print Control Specifications  </p>
<h2 id="hartree-fock-theory-for-molecules-hartree-fock-or-scf-mcscf-and-mp2-gradients">Hartree-Fock or SCF, MCSCF and MP2 Gradients<a class="headerlink" href="#hartree-fock-theory-for-molecules-hartree-fock-or-scf-mcscf-and-mp2-gradients" title="Permanent link">&para;</a></h2>
<p>The input for this directive allows the user to adjust the print control
for the SCF, UHF, ROHF, MCSCF and MP2 gradients. The form of the
directive is as follows:</p>
<pre><code> GRADIENTS   
   [print || noprint] ...  
 END
</code></pre>
<p>The complementary keyword pair print and noprint allows the user some
additional control on the information that can be included in the print
output from the SCF calculation. Currently, only a few items can be
explicitly invoked via print control. These are as follows:</p>
<table>
<thead>
<tr>
<th><strong>Name</strong></th>
<th><strong>Print Level</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>&ldquo;information&rdquo;</td>
<td>low</td>
<td>calculation info</td>
</tr>
<tr>
<td>&ldquo;geometry&rdquo;</td>
<td>high</td>
<td>geometry information</td>
</tr>
<tr>
<td>&ldquo;basis&rdquo;</td>
<td>high</td>
<td>basis set(s) used</td>
</tr>
<tr>
<td>&ldquo;forces&rdquo;</td>
<td>low</td>
<td>details of force components</td>
</tr>
<tr>
<td>&ldquo;timing&rdquo;</td>
<td>default</td>
<td>timing for each phase</td>
</tr>
</tbody>
</table>
<p>Table 3: Gradient Print Control Specifications  </p>
<h2 id="hartree-fock-theory-for-molecules-references">References<a class="headerlink" href="#hartree-fock-theory-for-molecules-references" title="Permanent link">&para;</a></h2>
<div class="footnote">
<hr />
<ol>
<li id="hartree-fock-theory-for-molecules-fn_footnote_1">
<p>Wong, A. T. and Harrison, R. J. (1995) &ldquo;Approaches to large-scale parallel self-consistent field calculation&rdquo;, <em>J. Comp. Chem.</em>  <strong>16</strong>, 1291-1300, DOI: <a href="http://dx.doi.org/10.1002/jcc.540161010">10.1002/jcc.540161010</a>&#160;<a class="footnote-backref" href="#hartree-fock-theory-for-molecules-fnref_footnote_1" title="Jump back to footnote 1 in the text">&#8617;</a></p>
</li>
<li id="hartree-fock-theory-for-molecules-fn_footnote_2">
<p>Foster, I. T.; Tilson, J. L.; Wagner, A. F.; Shepard, R. L.; Harrison, R. J.; Kendall, R. A. and Littlefield, R. J. (1996)   &ldquo;Toward high-performance computational chemistry: I. Scalable Fock matrix construction algorithms&rdquo;, <em>J. Comp. Chem.</em> <strong>17</strong>, 109-123,  DOI: <a href="http://dx.doi.org/10.1002/(SICI)1096-987X(19960115)17:1&lt;109::AID-JCC9&gt;3.0.CO;2-V">10.1002/(SICI)1096-987X(19960115)17:1&lt;109::AID-JCC9&gt;3.0.CO;2-V</a>&#160;<a class="footnote-backref" href="#hartree-fock-theory-for-molecules-fnref_footnote_2" title="Jump back to footnote 2 in the text">&#8617;</a></p>
</li>
</ol>
</div></section><section class="print-page" id="density-functional-theory-for-molecules" heading-number="11.2"><h1 id="density-functional-theory-for-molecules-density-functional-theory-dft">Density Functional Theory (DFT)<a class="headerlink" href="#density-functional-theory-for-molecules-density-functional-theory-dft" title="Permanent link">&para;</a></h1>
<h2 id="density-functional-theory-for-molecules-overview">Overview<a class="headerlink" href="#density-functional-theory-for-molecules-overview" title="Permanent link">&para;</a></h2>
<p>The NWChem density functional theory (DFT) module uses the Gaussian
basis set approach to compute closed shell and open shell densities and
Kohn-Sham orbitals in the:</p>
<ul>
<li>local density approximation (LDA),</li>
<li>non-local density approximation (NLDA),</li>
<li>local spin-density approximation (LSD),</li>
<li>non-local spin-density approximation (NLSD),</li>
<li>non-local meta-GGA approximation (metaGGA),</li>
<li>any empirical mixture of local and non-local approximations
    (including exact exchange), and</li>
<li>asymptotically corrected exchange-correlation potentials.</li>
<li>spin-orbit effects</li>
</ul>
<p>The formal scaling of the DFT computation can be reduced by choosing to
use auxiliary Gaussian basis sets to fit the charge density (CD) and/or
fit the exchange-correlation (XC) potential.</p>
<p>DFT input is provided using the compound DFT directive</p>
<pre><code> DFT  
   ...  
 END
</code></pre>
<p>The actual DFT calculation will be performed when the input module
encounters the <a href="#task">TASK</a> directive.</p>
<pre><code> TASK DFT
</code></pre>
<p>Once a user has specified a geometry and a Kohn-Sham orbital basis set
the DFT module can be invoked with no input directives (defaults invoked
throughout). There are sub-directives which allow for customized
application; those currently provided as options for the DFT module
are:</p>
<pre><code>VECTORS [[input] (&lt;string input_movecs default atomic&gt;) || \  
                  (project &lt;string basisname&gt; &lt;string filename&gt;)] \ 
          [swap [alpha||beta] &lt;integer vec1 vec2&gt; ...] \ 
          [output &lt;string output_filename default input_movecs&gt;] \  
 XC [[acm] [b3lyp] [beckehandh] [pbe0]\  
    [becke97]  [becke97-1] [becke97-2] [becke97-3] [becke97-d] [becke98] \  
     [hcth] [hcth120] [hcth147] [hcth147@tz2p]\
     [hcth407] [becke97gga1]  [hcth407p]\  
     [mpw91] [mpw1k] [xft97] [cft97] [ft97] [op] [bop] [pbeop]\  
     [xpkzb99] [cpkzb99] [xtpss03] [ctpss03] [xctpssh]\ 
     [b1b95] [bb1k] [mpw1b95] [mpwb1k] [pw6b95] [pwb6k] [m05] [m05-2x] [vs98] \ 
     [m06] [m06-hf] [m06-L] [m06-2x] \  
     [HFexch &lt;real prefactor default 1.0&gt;] \  
     [becke88 [nonlocal] &lt;real prefactor default 1.0&gt;] \  
     [xperdew91 [nonlocal] &lt;real prefactor default 1.0&gt;] \  
     [xpbe96 [nonlocal] &lt;real prefactor default 1.0&gt;] \  
     [gill96 [nonlocal] &lt;real prefactor default 1.0&gt;] \  
     [lyp &lt;real prefactor default 1.0&gt;] \  
     [perdew81 &lt;real prefactor default 1.0&gt;] \  
     [perdew86 [nonlocal] &lt;real prefactor default 1.0&gt;] \  
     [perdew91 [nonlocal] &lt;real prefactor default 1.0&gt;] \ 
     [cpbe96 [nonlocal] &lt;real prefactor default 1.0&gt;] \  
     [pw91lda &lt;real prefactor default 1.0&gt;] \ 
     [slater &lt;real prefactor default 1.0&gt;] \  
     [vwn_1 &lt;real prefactor default 1.0&gt;] \  
     [vwn_2 &lt;real prefactor default 1.0&gt;] \  
     [vwn_3 &lt;real prefactor default 1.0&gt;] \ 
     [vwn_4 &lt;real prefactor default 1.0&gt;] \  
     [vwn_5 &lt;real prefactor default 1.0&gt;] \  
     [vwn_1_rpa &lt;real prefactor default 1.0&gt;] \  
     [xtpss03 [nonlocal] &lt;real prefactor default 1.0&gt;] \ 
     [ctpss03 [nonlocal] &lt;real prefactor default 1.0&gt;] \  
     [bc95 [nonlocal] &lt;real prefactor default 1.0&gt;] \  
     [xpw6b95 [nonlocal] &lt;real prefactor default 1.0&gt;] \  
     [xpwb6k [nonlocal] &lt;real prefactor default 1.0&gt;] \  
     [xm05 [nonlocal] &lt;real prefactor default 1.0&gt;] \  
     [xm05-2x [nonlocal] &lt;real prefactor default 1.0&gt;] \  
     [cpw6b95 [nonlocal] &lt;real prefactor default 1.0&gt;] \  
     [cpwb6k [nonlocal] &lt;real prefactor default 1.0&gt;] \  
     [cm05 [nonlocal] &lt;real prefactor default 1.0&gt;] \  
     [cm05-2x [nonlocal] &lt;real prefactor default 1.0&gt;]] \  
     [xvs98 [nonlocal] &lt;real prefactor default 1.0&gt;]] \  
     [cvs98 [nonlocal] &lt;real prefactor default 1.0&gt;]] \  
     [xm06-L [nonlocal] &lt;real prefactor default 1.0&gt;]] \ 
     [xm06-hf [nonlocal] &lt;real prefactor default 1.0&gt;]] \  
     [xm06 [nonlocal] &lt;real prefactor default 1.0&gt;]] \ 
     [xm06-2x [nonlocal] &lt;real prefactor default 1.0&gt;]] \  
     [cm06-L [nonlocal] &lt;real prefactor default 1.0&gt;]] \  
     [cm06-hf [nonlocal] &lt;real prefactor default 1.0&gt;]] \  
     [cm06 [nonlocal] &lt;real prefactor default 1.0&gt;]] \  
     [cm06-2x [nonlocal] &lt;real prefactor default 1.0&gt;]]   
 CONVERGENCE [[energy &lt;real energy default 1e-7&gt;] \  
              [density &lt;real density default 1e-5&gt;] \  
              [gradient &lt;real gradient default 5e-4&gt;] \  
              [dampon &lt;real dampon default 0.0&gt;] \  
              [dampoff &lt;real dampoff default 0.0&gt;] \  
              [diison &lt;real diison default 0.0&gt;] \  
              [diisoff &lt;real diisoff default 0.0&gt;] \  
              [levlon &lt;real levlon default 0.0&gt;] \  
              [levloff &lt;real levloff default 0.0&gt;] \  
              [ncydp &lt;integer ncydp default 2&gt;] \  
              [ncyds &lt;integer ncyds default 30&gt;] \  
              [ncysh &lt;integer ncysh default 30&gt;] \  
              [damp &lt;integer ndamp default 0&gt;] [nodamping] \  
              [diis [nfock &lt;integer nfock default 10&gt;]] \  
              [nodiis] [lshift &lt;real lshift default 0.5&gt;] \  
              [nolevelshifting] \  
              [hl_tol &lt;real hl_tol default 0.1&gt;] \  
              [rabuck [n_rabuck &lt;integer n_rabuck default 25&gt;]\
              [fast] ]  
 GRID [(xcoarse||coarse||medium||fine||xfine||huge) default medium] \  
      [(gausleg||lebedev ) default lebedev ] \  
      [(becke||erf1||erf2||ssf) default erf1] \  
      [(euler||mura||treutler) default mura] \ 
      [rm &lt;real rm default 2.0&gt;] \  
      [nodisk] 
 TOLERANCES [[tight] [tol_rho &lt;real tol_rho default 1e-10&gt;] \  
             [accCoul &lt;integer accCoul default 8&gt;] \  
             [radius &lt;real radius default 25.0&gt;]]  
 [(LB94||CS00 &lt;real shift default none&gt;)]  
 DECOMP 
 ODFT 
 DIRECT  
 SEMIDIRECT [filesize &lt;integer filesize default disksize&gt;] 
            [memsize  &lt;integer memsize default available&gt;]  
            [filename &lt;string filename default $file_prefix.aoints$&gt;]
 INCORE 
 ITERATIONS &lt;integer iterations default 30&gt; 
 MAX_OVL  
 CGMIN  
 RODFT  
 MULLIKEN  
 DISP  
 XDM [ a1 &lt;real a1&gt; ]   [ a2 &lt;real a2&gt; ]  
 MULT &lt;integer mult default 1&gt;  
 NOIO  
 PRINT||NOPRINT
 SYM &lt;string (ON||OFF) default ON&gt;
 ADAPT &lt;string (ON||OFF) default ON&gt;
</code></pre>
<p>The following sections describe these keywords and optional
sub-directives that can be specified for a DFT calculation in NWChem.</p>
<h2 id="density-functional-theory-for-molecules-specification-of-basis-sets-for-the-dft-module">Specification of Basis Sets for the DFT Module<a class="headerlink" href="#density-functional-theory-for-molecules-specification-of-basis-sets-for-the-dft-module" title="Permanent link">&para;</a></h2>
<p>The DFT module requires at a minimum the basis set for the Kohn-Sham
molecular orbitals. This basis set must be in the default basis set
named &ldquo;ao basis&rdquo;, or it must be assigned to this default name using the
<a href="#set">SET</a> directive.</p>
<p>In addition to the basis set for the Kohn-Sham orbitals, the charge
density fitting basis set can also be specified in the input directives
for the DFT module. This basis set is used for the evaluation of the
Coulomb potential in the Dunlap scheme<sup id="density-functional-theory-for-molecules-fnref_footnote_1"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_1">1</a></sup><sup id="fnref_footnote_2"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_2">2</a></sup>. The charge density fitting basis
set must have the name <code>cd basis</code>. This can be the actual name of a
basis set, or a basis set can be assigned this name using the
<a href="#set">SET</a> directive. If this basis set is not
defined by input, the O(N<sup>4</sup>) exact Coulomb contribution is computed.</p>
<p>The user also has the option of specifying a third basis set for the
evaluation of the exchange-correlation potential. This basis set must
have the name <code>xc basis</code>. If this basis set is not specified by input,
the exchange contribution (XC) is evaluated by numerical quadrature. In
most applications, this approach is efficient enough, so the &ldquo;xc basis&rdquo;
basis set is not required.</p>
<p>For the DFT module, the input options for defining the basis sets in a
given calculation can be summarized as follows:</p>
<ul>
<li><code>ao basis</code> - Kohn-Sham molecular orbitals; required for all
    calculations</li>
<li><code>cd basis</code> - charge density fitting basis set; optional, but
    recommended for evaluation of the Coulomb potential</li>
<li>&ldquo;xc basis&rdquo; - exchange-correlation (XC) fitting basis set; optional,
    and not recommended</li>
</ul>
<h2 id="density-functional-theory-for-molecules-adft-new-in-nwchem-720">ADFT <strong><em>New in NWChem 7.2.0</em></strong>:<a class="headerlink" href="#density-functional-theory-for-molecules-adft-new-in-nwchem-720" title="Permanent link">&para;</a></h2>
<p>Use of the auxiliary density functional theory method (ADFT)<sup id="fnref_footnote_3"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_3">3</a></sup> can be triggered by means of the <code>adft</code> keyword. This can result in a large speed-up when using &ldquo;pure&rdquo; GGA functionals (e.g. PBE96) and Laplacian-dependent mGGA functionals (e.g. SCAN-L). The speed-up comes from the use of the fitted density obtained with the charge density fitting technique to approximate both the Coulomb and Exchange-Correlation contributions.</p>
<p>The ADFT method is similar in spirit to the exchange-correlation fitting technique triggered by specifying an <a href="#density-functional-theory-for-molecules-specification-of-basis-sets-for-the-dft-module">xc basis</a> without the <code>adft</code> keyword. It is important to note that, different to straight exchange-correlation fitting, <em>energy derivatives are well-defined</em> within the ADFT framework. As a consequence, geometry optimizations and harmonic vibrational frequencies are well-behaved.</p>
<p>The ADFT method requires a charge density fitting basis set (see <a href="#density-functional-theory-for-molecules-specification-of-basis-sets-for-the-dft-module">DFT basis set section</a>). If not <code>cd basis</code> set is provided, the <code>weigend coulomb fitting</code> basis set will be loaded.</p>
<h2 id="density-functional-theory-for-molecules-vectors-and-max_ovl-ks-mo-vectors">VECTORS and MAX_OVL: KS-MO Vectors<a class="headerlink" href="#density-functional-theory-for-molecules-vectors-and-max_ovl-ks-mo-vectors" title="Permanent link">&para;</a></h2>
<p>The VECTORS directive is the same as that in the <a href="#hartree-fock-theory-for-molecules-vectors-inputoutput-of-mo-vectors">SCF</a> module. Currently, the <a href="#hartree-fock-theory-for-molecules-vectors-lock-keyword">LOCK</a> keyword is not supported by the DFT module, however the directive</p>
<pre><code> MAX_OVL
</code></pre>
<p>has the same effect.</p>
<h2 id="density-functional-theory-for-molecules-xc-and-decomp-exchange-correlation-potentials">XC and DECOMP: Exchange-Correlation Potentials<a class="headerlink" href="#density-functional-theory-for-molecules-xc-and-decomp-exchange-correlation-potentials" title="Permanent link">&para;</a></h2>
<pre><code> XC [[acm] [b3lyp] [beckehandh] [pbe0] [bhlyp]\  
    [becke97]  [becke97-1] [becke97-2] [becke97-3] [becke98] [hcth] [hcth120] [hcth147] [hcth147@tz2p] \
     [hcth407] [becke97gga1] [hcth407p] \  
     [optx] [hcthp14] [mpw91] [mpw1k] [xft97] [cft97] [ft97] [op] [bop] [pbeop]\  
     [m05] [m05-2x] [m06] [m06-l] [m06-2x] [m06-hf] [m08-hx] [m08-so] [m11] [m11-l]\  
     [HFexch &lt;real prefactor default 1.0&gt;] \  
     [becke88 [nonlocal] &lt;real prefactor default 1.0&gt;] \  
     [xperdew91 [nonlocal] &lt;real prefactor default 1.0&gt;] \  
     [xpbe96 [nonlocal] &lt;real prefactor default 1.0&gt;] \  
     [gill96 [nonlocal] &lt;real prefactor default 1.0&gt;] \  
     [lyp &lt;real prefactor default 1.0&gt;] \  
     [perdew81 &lt;real prefactor default 1.0&gt;] \  
     [perdew86 [nonlocal] &lt;real prefactor default 1.0&gt;] \  
     [perdew91 [nonlocal] &lt;real prefactor default 1.0&gt;] \  
     [cpbe96 [nonlocal] &lt;real prefactor default 1.0&gt;] \  
     [pw91lda &lt;real prefactor default 1.0&gt;] \  
     [slater &lt;real prefactor default 1.0&gt;] \  
     [vwn_1 &lt;real prefactor default 1.0&gt;] \  
     [vwn_2 &lt;real prefactor default 1.0&gt;] \  
     [vwn_3 &lt;real prefactor default 1.0&gt;] \  
     [vwn_4 &lt;real prefactor default 1.0&gt;] \  
     [vwn_5 &lt;real prefactor default 1.0&gt;] \  
     [vwn_1_rpa &lt;real prefactor default 1.0&gt;]]
</code></pre>
<p>The user has the option of specifying the exchange-correlation treatment
in the DFT Module (see table below for full list of functionals). The
default exchange-correlation functional is defined as the local density
approximation (LDA) for closed shell systems and its counterpart the
local spin-density (LSD) approximation for open shell systems. Within
this approximation, the exchange functional is the Slater &rho;<sup>1/3</sup>
functional<sup id="density-functional-theory-for-molecules-fnref_footnote_4"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_4">4</a></sup><sup id="fnref_footnote_5"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_5">5</a></sup>, and the correlation functional is the
Vosko-Wilk-Nusair (VWN) functional (functional V)<sup id="density-functional-theory-for-molecules-fnref_footnote_6"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_6">6</a></sup>.
The parameters used in
this formula are obtained by fitting to the Ceperley and Alder Quantum
Monte-Carlo solution of the homogeneous electron gas.</p>
<p>These defaults can be invoked explicitly by specifying the following
keywords within the DFT module input directive, <code>XC slater vwn_5</code>.</p>
<p>That is, this statement in the input file</p>
<pre><code>dft  
 XC slater vwn_5  
end  
task dft
</code></pre>
<p>is equivalent to the simple line</p>
<pre><code>task dft
</code></pre>
<p>The <code>DECOMP</code> directive causes the components of the energy corresponding
to each functional to be printed, rather than just the total
exchange-correlation energy that is the default. You can see an example
of this directive in the <a href="#density-functional-theory-for-molecules-sample-input-file">sample
input</a>.</p>
<p>Many alternative exchange and correlation functionals are available to
the user as listed in the table below. The following sections describe
how to use these options.</p>
<h3 id="density-functional-theory-for-molecules-libxc-interface-new-in-nwchem-720">Libxc interface <strong><em>New in NWChem 7.2.0</em></strong>:<a class="headerlink" href="#density-functional-theory-for-molecules-libxc-interface-new-in-nwchem-720" title="Permanent link">&para;</a></h3>
<p>If NWChem is compiled by linking it with the <a href="https://libxc.gitlab.io/">libxc</a> DFT library
(as described in the <a href="#interfaces-with-external-software-libxc">Interfaces with External Software</a> section),
the user will be able to use most of the XC functionals available in libxc.<br />
The input syntax requires to use the <a href="#density-functional-theory-for-molecules-xc-and-decomp-exchange-correlation-potentials">xc</a> keyword followed by
the functionals name from 
<a href="https://libxc.gitlab.io/functionals/">list available in Libxc</a></p>
<p>For example, the following input for the NWChem libxc interface</p>
<pre><code>dft
 xc gga_x_pbe 1.0 gga_x_pbe 1.0
end
</code></pre>
<p>while trigger use of the same PBE96 functionals as in the NWChem built-in interface</p>
<pre><code>dft
 xc xpbe96 1.0 cpbe96 1.0
end
</code></pre>
<h3 id="density-functional-theory-for-molecules-exchange-correlation-functionals">Exchange-Correlation Functionals<a class="headerlink" href="#density-functional-theory-for-molecules-exchange-correlation-functionals" title="Permanent link">&para;</a></h3>
<p>There are several Exchange and Correlation functionals in addition to
the default <code>slater</code> and <code>vwn_5</code> functionals. These are either local or
gradient-corrected functionals (GCA); a full list can be found in the
table below.</p>
<p>The Hartree-Fock exact exchange functional, (which has O(N<sup>4</sup>)
computation expense), is invoked by specifying</p>
<pre><code>  XC HFexch
</code></pre>
<p>Note that the user also has the ability to include only the local or
nonlocal contributions of a given functional. In addition, the user can
specify a multiplicative prefactor (the variable <prefactor> in the
input) for the local/nonlocal component or total. An example of this
might be,</p>
<pre><code>  XC becke88 nonlocal 0.72
</code></pre>
<p>The user should be aware that the Becke88 local component is simply the
Slater exchange and should be input as such.</p>
<p>Any combination of the supported exchange functional options can be
used. For example, the popular Gaussian B3 exchange could be specified
as:</p>
<pre><code>  XC slater 0.8 becke88 nonlocal 0.72 HFexch 0.2
</code></pre>
<p>Any combination of the supported correlation functional options can be
used. For example, B3LYP could be specified
as:</p>
<pre><code>XC vwn_1_rpa 0.19 lyp 0.81 HFexch 0.20  slater 0.80 becke88 nonlocal 0.72
</code></pre>
<p>and X3LYP as:</p>
<pre><code>xc vwn_1_rpa 0.129 lyp 0.871 hfexch 0.218 slater 0.782 \
becke88 nonlocal 0.542  xperdew91 nonlocal 0.167
</code></pre>
<h3 id="density-functional-theory-for-molecules-setting-up-common-exchange-correlation-functionals">Setting up common exchange-correlation functionals<a class="headerlink" href="#density-functional-theory-for-molecules-setting-up-common-exchange-correlation-functionals" title="Permanent link">&para;</a></h3>
<ul>
<li>B3LYP: <code>xc b3lyp</code></li>
<li>PBE0: <code>xc pbe0</code></li>
<li>PBE96: <code>xc xpbe96 cpbe96</code></li>
<li>PW91: <code>xc xperdew91 perdew91</code></li>
<li>BHLYP: <code>xc bhlyp</code></li>
<li>Becke Half and Half: <code>xc beckehandh</code></li>
<li>BP86: <code>xc becke88 perdew86</code></li>
<li>BP91: <code>xc becke88 perdew91</code></li>
<li>BLYP: <code>xc becke88 lyp</code></li>
</ul>
<p>Minnesota Functionals</p>
<ul>
<li>xc m05</li>
<li>xc m05-2x</li>
<li>xc m06</li>
<li>xc m06-l</li>
<li>xc m06-2x</li>
<li>xc m06-hf</li>
<li>xc m08-hx</li>
<li>xc m08-so</li>
<li>xc m11</li>
<li>xc m11-l</li>
</ul>
<p>Analytic second derivatives are not supported with the Minnesota
functionals yet.</p>
<h3 id="density-functional-theory-for-molecules-combined-exchange-and-correlation-functionals">Combined Exchange and Correlation Functionals<a class="headerlink" href="#density-functional-theory-for-molecules-combined-exchange-and-correlation-functionals" title="Permanent link">&para;</a></h3>
<p>In addition to the options listed above for the exchange and correlation
functionals, the user has the alternative of specifying combined
exchange and correlation functionals.</p>
<p>The available hybrid functionals (where a Hartree-Fock Exchange
component is present) consist of the Becke &ldquo;half and half&rdquo;<sup id="density-functional-theory-for-molecules-fnref_footnote_7"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_7">7</a></sup>, the adiabatic connection method<sup id="fnref_footnote_8"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_8">8</a></sup>,
Becke 1997 (&ldquo;Becke V&rdquo; paper<sup id="density-functional-theory-for-molecules-fnref_footnote_9"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_9">9</a></sup>).</p>
<p>The keyword <code>beckehandh</code> specifies that the exchange-correlation energy
will be computed
as</p>
<p>E<sub>XC</sub> &asymp; &frac12;E<sub>X</sub><sup>HF</sup> + &frac12;E<sub>X</sub><sup>Slater</sup> + &frac12;E<sub>C</sub><sup>PW91LDA</sup>  </p>
<p>We know this is NOT the correct Becke prescribed implementation that
requires the XC potential in the energy expression. But this is what is
currently implemented as an approximation to it.</p>
<p>The keyword <code>acm</code> specifies that the exchange-correlation energy is
computed as  </p>
<p>E<sub>XC</sub> = a<sub>0</sub>E<sub>X</sub><sup>HF</sup> + (1 - a<sub>0</sub>)E<sub>X</sub><sup>Slater</sup> + a<sub>X</sub>&delta;E<sub>X</sub><sup>Becke88</sup> + E<sub>C</sub><sup>VWN</sup> +
 a<sub>C</sub>&delta;E<sub>C</sub><sup>Perdew91</sup>  </p>
<p>where  </p>
<p>a<sub>0</sub> = 0.20, a<sub>X</sub> = 0.72, a<sub>C</sub> = 0.81  </p>
<p>and &delta; stands for a non-local component.</p>
<p>The keyword <code>b3lyp</code> specifies that the exchange-correlation energy is
computed as</p>
<p>E<sub>XC</sub> = a<sub>0</sub>E<sub>X</sub><sup>HF</sup> + (1 - a<sub>0</sub>)E<sub>X</sub><sup>Slater</sup> + a<sub>X</sub>&delta;E<sub>X</sub><sup>Becke88</sup> + (1 - a<sub>C</sub>)E<sub>C</sub><sup>VWN_1_RPA</sup> +
 a<sub>C</sub>&delta;E<sub>C</sub><sup>LYP</sup>  </p>
<p>where  </p>
<p>a<sub>0</sub> = 0.20, a<sub>X</sub> = 0.72, a<sub>C</sub> = 0.81  </p>
<h2 id="density-functional-theory-for-molecules-xc-functionals-summary">XC Functionals Summary<a class="headerlink" href="#density-functional-theory-for-molecules-xc-functionals-summary" title="Permanent link">&para;</a></h2>
<table id="_table-1">
<caption style="caption-side:bottom"><span>Table&nbsp;1:</span> Available Exchange (X) and Correlation (C) functionals. GGA is</caption>
<thead>
<tr>
<th>Keyword</th>
<th>X</th>
<th>C</th>
<th>GGA</th>
<th>Meta</th>
<th>Hybr.</th>
<th>2nd</th>
<th>Ref.</th>
</tr>
</thead>
<tbody>
<tr>
<td>slater</td>
<td>*</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>Y</td>
<td><sup id="fnref2_footnote_4"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_4">4</a></sup><sup id="fnref2_footnote_5"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_5">5</a></sup></td>
</tr>
<tr>
<td>vwn_1</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td></td>
<td>Y</td>
<td><sup id="fnref7_footnote_6"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_6">6</a></sup></td>
</tr>
<tr>
<td>vwn_2</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td></td>
<td>Y</td>
<td><sup id="fnref6_footnote_6"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_6">6</a></sup></td>
</tr>
<tr>
<td>vwn_3</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td></td>
<td>Y</td>
<td><sup id="fnref5_footnote_6"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_6">6</a></sup></td>
</tr>
<tr>
<td>vwn_4</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td></td>
<td>Y</td>
<td><sup id="fnref4_footnote_6"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_6">6</a></sup></td>
</tr>
<tr>
<td>vwn_5</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td></td>
<td>Y</td>
<td><sup id="fnref3_footnote_6"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_6">6</a></sup></td>
</tr>
<tr>
<td>vwn_1_rpa</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td></td>
<td>Y</td>
<td><sup id="fnref2_footnote_6"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_6">6</a></sup></td>
</tr>
<tr>
<td>perdew81</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td></td>
<td>Y</td>
<td><sup id="fnref2_footnote_10"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_10">10</a></sup></td>
</tr>
<tr>
<td>pw91lda</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td></td>
<td>Y</td>
<td><sup id="fnref2_footnote_11"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_11">11</a></sup></td>
</tr>
<tr>
<td>xbecke86b</td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_12"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_12">12</a></sup></td>
</tr>
<tr>
<td>becke88</td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td>Y</td>
<td><sup id="fnref_footnote_13"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_13">13</a></sup></td>
</tr>
<tr>
<td>xperdew86</td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref2_footnote_14"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_14">14</a></sup></td>
</tr>
<tr>
<td>xperdew91</td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td>Y</td>
<td><sup id="fnref_footnote_11"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_11">11</a></sup></td>
</tr>
<tr>
<td>xpbe96</td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td>Y</td>
<td><sup id="fnref2_footnote_15"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_15">15</a></sup><sup id="fnref2_footnote_16"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_16">16</a></sup></td>
</tr>
<tr>
<td>gill96</td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td>Y</td>
<td><sup id="fnref_footnote_17"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_17">17</a></sup></td>
</tr>
<tr>
<td>optx</td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_18"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_18">18</a></sup></td>
</tr>
<tr>
<td>mpw91</td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td>Y</td>
<td><sup id="fnref_footnote_19"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_19">19</a></sup><sup id="fnref_footnote_20"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_20">20</a></sup></td>
</tr>
<tr>
<td>xft97</td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref3_footnote_21"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_21">21</a></sup><sup id="fnref3_footnote_22"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_22">22</a></sup></td>
</tr>
<tr>
<td>rpbe</td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td>Y</td>
<td><sup id="fnref2_footnote_23"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_23">23</a></sup></td>
</tr>
<tr>
<td>revpbe</td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td>Y</td>
<td><sup id="fnref2_footnote_24"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_24">24</a></sup></td>
</tr>
<tr>
<td>xpw6b95</td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref6_footnote_25"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_25">25</a></sup></td>
</tr>
<tr>
<td>xpwb6k</td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref5_footnote_25"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_25">25</a></sup></td>
</tr>
<tr>
<td>perdew86</td>
<td></td>
<td>*</td>
<td>*</td>
<td></td>
<td></td>
<td>Y</td>
<td><sup id="fnref_footnote_14"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_14">14</a></sup></td>
</tr>
<tr>
<td>lyp</td>
<td></td>
<td>*</td>
<td>*</td>
<td></td>
<td></td>
<td>Y</td>
<td><sup id="fnref_footnote_26"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_26">26</a></sup></td>
</tr>
<tr>
<td>perdew91</td>
<td></td>
<td>*</td>
<td>*</td>
<td></td>
<td></td>
<td>Y</td>
<td><sup id="fnref_footnote_27"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_27">27</a></sup><sup id="fnref_footnote_28"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_28">28</a></sup></td>
</tr>
<tr>
<td>cpbe96</td>
<td></td>
<td>*</td>
<td>*</td>
<td></td>
<td></td>
<td>Y</td>
<td><sup id="fnref_footnote_15"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_15">15</a></sup><sup id="fnref_footnote_16"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_16">16</a></sup></td>
</tr>
<tr>
<td>cft97</td>
<td></td>
<td>*</td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref2_footnote_21"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_21">21</a></sup><sup id="fnref2_footnote_22"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_22">22</a></sup></td>
</tr>
<tr>
<td>op</td>
<td></td>
<td>*</td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref2_footnote_29"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_29">29</a></sup></td>
</tr>
<tr>
<td>hcth</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref2_footnote_30"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_30">30</a></sup></td>
</tr>
<tr>
<td>hcth120</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref2_footnote_31"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_31">31</a></sup></td>
</tr>
<tr>
<td>hcth147</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_31"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_31">31</a></sup></td>
</tr>
<tr>
<td>hcth147@tz2p</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_32"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_32">32</a></sup></td>
</tr>
<tr>
<td>hcth407</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_33"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_33">33</a></sup></td>
</tr>
<tr>
<td>becke97gga1</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_34"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_34">34</a></sup></td>
</tr>
<tr>
<td>hcthp14</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_35"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_35">35</a></sup></td>
</tr>
<tr>
<td>ft97</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_21"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_21">21</a></sup><sup id="fnref_footnote_22"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_22">22</a></sup></td>
</tr>
<tr>
<td>htch407p</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_36"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_36">36</a></sup></td>
</tr>
<tr>
<td>bop</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_29"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_29">29</a></sup></td>
</tr>
<tr>
<td>pbeop</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_37"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_37">37</a></sup></td>
</tr>
<tr>
<td>xpkzb99</td>
<td>*</td>
<td></td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref2_footnote_38"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_38">38</a></sup></td>
</tr>
<tr>
<td>cpkzb99</td>
<td></td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_38"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_38">38</a></sup></td>
</tr>
<tr>
<td>xtpss03</td>
<td>*</td>
<td></td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref2_footnote_39"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_39">39</a></sup></td>
</tr>
<tr>
<td>ctpss03</td>
<td></td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_39"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_39">39</a></sup></td>
</tr>
<tr>
<td>bc95</td>
<td></td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_23"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_23">23</a></sup></td>
</tr>
<tr>
<td>cpw6b95</td>
<td></td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref4_footnote_25"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_25">25</a></sup></td>
</tr>
<tr>
<td>cpwb6k</td>
<td></td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref3_footnote_25"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_25">25</a></sup></td>
</tr>
<tr>
<td>xm05</td>
<td>*</td>
<td></td>
<td></td>
<td>*</td>
<td>*</td>
<td>N</td>
<td><sup id="fnref3_footnote_40"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_40">40</a></sup><sup id="fnref2_footnote_41"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_41">41</a></sup></td>
</tr>
<tr>
<td>cm05</td>
<td></td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref2_footnote_40"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_40">40</a></sup><sup id="fnref_footnote_41"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_41">41</a></sup></td>
</tr>
<tr>
<td>m05-2x</td>
<td>*</td>
<td>*</td>
<td></td>
<td>*</td>
<td>*</td>
<td>N</td>
<td><sup id="fnref3_footnote_42"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_42">42</a></sup></td>
</tr>
<tr>
<td>xm05-2x</td>
<td>*</td>
<td></td>
<td></td>
<td>*</td>
<td>*</td>
<td>N</td>
<td><sup id="fnref2_footnote_42"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_42">42</a></sup></td>
</tr>
<tr>
<td>cm05-2x</td>
<td></td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_42"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_42">42</a></sup></td>
</tr>
<tr>
<td>xctpssh</td>
<td></td>
<td></td>
<td></td>
<td>*</td>
<td>*</td>
<td>N</td>
<td><sup id="fnref_footnote_43"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_43">43</a></sup></td>
</tr>
<tr>
<td>bb1k</td>
<td></td>
<td></td>
<td></td>
<td>*</td>
<td>*</td>
<td>N</td>
<td><sup id="fnref_footnote_24"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_24">24</a></sup></td>
</tr>
<tr>
<td>mpw1b95</td>
<td></td>
<td></td>
<td></td>
<td>*</td>
<td>*</td>
<td>N</td>
<td><sup id="fnref2_footnote_44"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_44">44</a></sup></td>
</tr>
<tr>
<td>mpwb1k</td>
<td></td>
<td></td>
<td></td>
<td>*</td>
<td>*</td>
<td>N</td>
<td><sup id="fnref_footnote_44"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_44">44</a></sup></td>
</tr>
<tr>
<td>pw6b95</td>
<td></td>
<td></td>
<td></td>
<td>*</td>
<td>*</td>
<td>N</td>
<td><sup id="fnref2_footnote_25"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_25">25</a></sup></td>
</tr>
<tr>
<td>pwb6k</td>
<td></td>
<td></td>
<td></td>
<td>*</td>
<td>*</td>
<td>N</td>
<td><sup id="fnref_footnote_25"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_25">25</a></sup></td>
</tr>
<tr>
<td>m05</td>
<td></td>
<td></td>
<td></td>
<td>*</td>
<td>*</td>
<td>N</td>
<td><sup id="fnref_footnote_40"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_40">40</a></sup></td>
</tr>
<tr>
<td>vs98</td>
<td></td>
<td></td>
<td></td>
<td>*</td>
<td>*</td>
<td>N</td>
<td><sup id="fnref3_footnote_45"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_45">45</a></sup></td>
</tr>
<tr>
<td>xvs98</td>
<td>*</td>
<td></td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref2_footnote_45"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_45">45</a></sup></td>
</tr>
<tr>
<td>cvs98</td>
<td></td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_45"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_45">45</a></sup></td>
</tr>
<tr>
<td>m06-L</td>
<td>*</td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref6_footnote_46"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_46">46</a></sup></td>
</tr>
<tr>
<td>xm06-L</td>
<td>*</td>
<td></td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref5_footnote_46"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_46">46</a></sup></td>
</tr>
<tr>
<td>cm06-L</td>
<td></td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref4_footnote_46"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_46">46</a></sup></td>
</tr>
<tr>
<td>m06-hf</td>
<td></td>
<td></td>
<td></td>
<td>*</td>
<td>*</td>
<td>N</td>
<td><sup id="fnref3_footnote_47"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_47">47</a></sup></td>
</tr>
<tr>
<td>xm06-hf</td>
<td>*</td>
<td></td>
<td></td>
<td>*</td>
<td>*</td>
<td>N</td>
<td><sup id="fnref2_footnote_47"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_47">47</a></sup></td>
</tr>
<tr>
<td>cm06-hf</td>
<td></td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_47"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_47">47</a></sup></td>
</tr>
<tr>
<td>m06</td>
<td></td>
<td></td>
<td></td>
<td>*</td>
<td>*</td>
<td>N</td>
<td><sup id="fnref3_footnote_48"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_48">48</a></sup></td>
</tr>
<tr>
<td>xm06</td>
<td>*</td>
<td></td>
<td></td>
<td>*</td>
<td>*</td>
<td>N</td>
<td><sup id="fnref2_footnote_48"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_48">48</a></sup></td>
</tr>
<tr>
<td>cm06</td>
<td></td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_48"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_48">48</a></sup></td>
</tr>
<tr>
<td>m06-2x</td>
<td></td>
<td></td>
<td></td>
<td>*</td>
<td>*</td>
<td>N</td>
<td><sup id="fnref3_footnote_46"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_46">46</a></sup></td>
</tr>
<tr>
<td>xm06-2x</td>
<td>*</td>
<td></td>
<td></td>
<td>*</td>
<td>*</td>
<td>N</td>
<td><sup id="fnref2_footnote_46"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_46">46</a></sup></td>
</tr>
<tr>
<td>cm06-2x</td>
<td></td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_46"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_46">46</a></sup></td>
</tr>
<tr>
<td>cm08-hx</td>
<td></td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref9_footnote_49"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_49">49</a></sup></td>
</tr>
<tr>
<td>xm08-hx</td>
<td>*</td>
<td></td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref8_footnote_49"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_49">49</a></sup></td>
</tr>
<tr>
<td>m08-hx</td>
<td>*</td>
<td>*</td>
<td></td>
<td>*</td>
<td>*</td>
<td>N</td>
<td><sup id="fnref7_footnote_49"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_49">49</a></sup></td>
</tr>
<tr>
<td>cm08-so</td>
<td></td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref6_footnote_49"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_49">49</a></sup></td>
</tr>
<tr>
<td>xm08-so</td>
<td>*</td>
<td></td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref5_footnote_49"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_49">49</a></sup></td>
</tr>
<tr>
<td>m08-so</td>
<td>*</td>
<td>*</td>
<td></td>
<td>*</td>
<td>*</td>
<td>N</td>
<td><sup id="fnref4_footnote_49"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_49">49</a></sup></td>
</tr>
<tr>
<td>cm11</td>
<td></td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref3_footnote_50"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_50">50</a></sup></td>
</tr>
<tr>
<td>xm11</td>
<td>*</td>
<td></td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref2_footnote_50"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_50">50</a></sup></td>
</tr>
<tr>
<td>m11</td>
<td>*</td>
<td>*</td>
<td></td>
<td>*</td>
<td>*</td>
<td>N</td>
<td><sup id="fnref_footnote_50"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_50">50</a></sup></td>
</tr>
<tr>
<td>cm11-l</td>
<td></td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref3_footnote_51"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_51">51</a></sup></td>
</tr>
<tr>
<td>xm11-l</td>
<td>*</td>
<td></td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref2_footnote_51"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_51">51</a></sup></td>
</tr>
<tr>
<td>m11-l</td>
<td>*</td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_51"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_51">51</a></sup></td>
</tr>
<tr>
<td>csogga</td>
<td></td>
<td>*</td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref3_footnote_49"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_49">49</a></sup></td>
</tr>
<tr>
<td>xsogga</td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref2_footnote_49"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_49">49</a></sup></td>
</tr>
<tr>
<td>sogga</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_49"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_49">49</a></sup></td>
</tr>
<tr>
<td>csogga11</td>
<td></td>
<td>*</td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref3_footnote_52"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_52">52</a></sup></td>
</tr>
<tr>
<td>xsogga11</td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref2_footnote_52"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_52">52</a></sup></td>
</tr>
<tr>
<td>sogga11</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_52"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_52">52</a></sup></td>
</tr>
<tr>
<td>csogga11-x</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref3_footnote_53"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_53">53</a></sup></td>
</tr>
<tr>
<td>xsogga11-x</td>
<td>*</td>
<td></td>
<td>*</td>
<td></td>
<td></td>
<td>N</td>
<td><sup id="fnref2_footnote_53"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_53">53</a></sup></td>
</tr>
<tr>
<td>sogga11-x</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td>*</td>
<td>N</td>
<td><sup id="fnref_footnote_53"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_53">53</a></sup></td>
</tr>
<tr>
<td>dldf</td>
<td>*</td>
<td>*</td>
<td></td>
<td>*</td>
<td>*</td>
<td>N</td>
<td><sup id="fnref_footnote_54"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_54">54</a></sup></td>
</tr>
<tr>
<td>beckehandh</td>
<td>*</td>
<td>*</td>
<td></td>
<td></td>
<td>*</td>
<td>Y</td>
<td><sup id="fnref2_footnote_7"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_7">7</a></sup></td>
</tr>
<tr>
<td>b3lyp</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td>*</td>
<td>Y</td>
<td><sup id="fnref3_footnote_8"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_8">8</a></sup></td>
</tr>
<tr>
<td>acm</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td>*</td>
<td>Y</td>
<td><sup id="fnref2_footnote_8"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_8">8</a></sup></td>
</tr>
<tr>
<td>becke97</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td>*</td>
<td>N</td>
<td><sup id="fnref2_footnote_9"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_9">9</a></sup></td>
</tr>
<tr>
<td>becke97-1</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td>*</td>
<td>N</td>
<td><sup id="fnref_footnote_30"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_30">30</a></sup></td>
</tr>
<tr>
<td>becke97-2</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td>*</td>
<td>N</td>
<td><sup id="fnref_footnote_55"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_55">55</a></sup></td>
</tr>
<tr>
<td>becke97-3</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td>*</td>
<td>N</td>
<td><sup id="fnref_footnote_56"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_56">56</a></sup></td>
</tr>
<tr>
<td>becke97-d</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td>*</td>
<td>N</td>
<td><sup id="fnref_footnote_57"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_57">57</a></sup></td>
</tr>
<tr>
<td>becke98</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td>*</td>
<td>N</td>
<td><sup id="fnref_footnote_58"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_58">58</a></sup></td>
</tr>
<tr>
<td>pbe0</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td>*</td>
<td>Y</td>
<td><sup id="fnref_footnote_59"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_59">59</a></sup></td>
</tr>
<tr>
<td>mpw1k</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td>*</td>
<td>Y</td>
<td><sup id="fnref_footnote_60"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_60">60</a></sup></td>
</tr>
<tr>
<td>xmvs15</td>
<td>*</td>
<td></td>
<td></td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_61"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_61">61</a></sup></td>
</tr>
<tr>
<td>hle16</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td>*</td>
<td>Y</td>
<td><sup id="fnref_footnote_62"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_62">62</a></sup></td>
</tr>
<tr>
<td>scan</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_63"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_63">63</a></sup></td>
</tr>
<tr>
<td>scanl</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_64"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_64">64</a></sup></td>
</tr>
<tr>
<td>revm06-L</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_65"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_65">65</a></sup></td>
</tr>
<tr>
<td>revm06</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td>N</td>
<td><sup id="fnref_footnote_66"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_66">66</a></sup></td>
</tr>
<tr>
<td>wb97x</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td>*</td>
<td>N</td>
<td><sup id="fnref_footnote_67"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_67">67</a></sup></td>
</tr>
<tr>
<td>wb97x-d3</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td>*</td>
<td>N</td>
<td><sup id="fnref_footnote_68"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_68">68</a></sup></td>
</tr>
<tr>
<td>rscan</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_69"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_69">69</a></sup></td>
</tr>
<tr>
<td>r2scan</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_70"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_70">70</a></sup></td>
</tr>
<tr>
<td>r2scan0</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td>N</td>
<td><sup id="fnref_footnote_71"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_71">71</a></sup></td>
</tr>
<tr>
<td>r2scanl</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td>N</td>
<td><sup id="fnref_footnote_72"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_72">72</a></sup><sup id="fnref_footnote_73"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_73">73</a></sup></td>
</tr>
<tr>
<td>ncap</td>
<td>*</td>
<td>*</td>
<td>*</td>
<td></td>
<td></td>
<td>Y</td>
<td><sup id="fnref_footnote_74"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_74">74</a></sup></td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<h3 id="density-functional-theory-for-molecules-meta-gga-functionals">Meta-GGA Functionals<a class="headerlink" href="#density-functional-theory-for-molecules-meta-gga-functionals" title="Permanent link">&para;</a></h3>
<p>One way to calculate meta-GGA energies is to use orbitals and densities
from fully self-consistent GGA or LDA calculations and run them in one
iteration in the meta-GGA functional. It is expected that meta-GGA
energies obtained this way will be close to fully self consistent
meta-GGA calculations.</p>
<p>It is possible to calculate metaGGA energies both ways in NWChem, that
is, self-consistently or with GGA/LDA orbitals and densities. However,
since second derivatives are not available for metaGGAs, in order to
calculate frequencies, one must use task dft freq numerical. A sample
file with this is shown below, in <a href="#density-functional-theory-for-molecules-sample-input-file">Sample input
file</a>. In this instance, the energy is
calculated self-consistently and geometry is optimized using the
analytical gradients.</p>
<p>(For more information on metaGGAs, see Kurth et al 1999 <sup id="fnref_footnote_75"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_75">75</a></sup>
 for a brief description of meta-GGAs,
and citations 14-27 therein for thorough background)</p>
<p>Note: both TPSS and PKZB correlation require the PBE GGA CORRELATION
(which is itself dependent on an LDA). The decision has been made to use
these functionals with the accompanying local PW91LDA. The user cannot
 set the local part of these metaGGA functionals.</p>
<h3 id="density-functional-theory-for-molecules-range-separated-functionals">Range-Separated Functionals<a class="headerlink" href="#density-functional-theory-for-molecules-range-separated-functionals" title="Permanent link">&para;</a></h3>
<p>Using the Ewald decomposition</p>
<div class="arithmatex">\[\frac{1}{r_{12}} = \frac{\alpha + \beta \texttt{erf}(\mu r_{12})}{r_{12}} + \frac{1 - [\alpha + \beta \texttt{erf}(\mu r_{12})]}{r_{12}}\]</div>
<p>we can split the the Exchange interaction as</p>
<div class="arithmatex">\[E_{X} = E_X^{LR} + E_X^{SR}\]</div>
<p>Therefore, the long-range HF Exchange energy
becomes</p>
<div class="arithmatex">\[E_X^{LR} = \alpha E_X^{HF} - \frac{\beta}{2} \sum_i \sum_j \int \int \phi_i(r_1)\phi_j(r_1)\frac{\texttt{erf}(\mu r_{12})}{r_{12}} \phi_i(r_2)\phi_j(r_2)\]</div>
<h4 id="density-functional-theory-for-molecules-input-parameters-for-range-separated-functionals">Input parameters for Range-Separated functionals<a class="headerlink" href="#density-functional-theory-for-molecules-input-parameters-for-range-separated-functionals" title="Permanent link">&para;</a></h4>
<pre><code> cam &lt;real cam&gt; cam_alpha &lt;real cam_alpha&gt; cam_beta &lt;cam_beta&gt;
</code></pre>
<p><code>cam</code> represents the attenuation parameter &mu;, <code>cam_alpha</code> and
<code>cam_beta</code> are the  &alpha; and &beta; parameters that control the
amount of short-range DFT and long-range HF Exchange according to the
Ewald decomposition. As r<sub>12</sub> &rarr; 0, the HF exchange
fraction is &alpha;,
 while the DFT exchange fraction is 1 - &alpha;.
 As r<sub>12</sub> &rarr; &infin;,
 the HF exchange fraction approaches &alpha; + &beta;  and the DFT exchange fraction
approaches 1 - &alpha; - &beta;. In the HSE functional, the HF part
is short-ranged and DFT is long-ranged.</p>
<p>Range separated functionals (or long-range corrected or LC) can be
specified as follows:</p>
<p>CAM-B3LYP:</p>
<pre><code>xc xcamb88 1.00 lyp 0.81 vwn_5 0.19 hfexch 1.00  
cam 0.33 cam_alpha 0.19 cam_beta 0.46
</code></pre>
<p>LC-BLYP:</p>
<pre><code>xc xcamb88 1.00 lyp 1.0 hfexch 1.00  
cam 0.33 cam_alpha 0.0 cam_beta 1.0
</code></pre>
<p>LC-PBE:</p>
<pre><code>xc xcampbe96 1.0 cpbe96 1.0 HFexch 1.0 
cam 0.30 cam_alpha 0.0 cam_beta 1.0
</code></pre>
<p>LC-PBE0 or CAM-PBE0:</p>
<pre><code>xc xcampbe96 1.0 cpbe96 1.0 HFexch 1.0
cam 0.30 cam_alpha 0.25 cam_beta 0.75
</code></pre>
<p>BNL (Baer, Neuhauser,
Lifshifts):</p>
<pre><code>xc xbnl07 0.90 lyp 1.00 hfexch 1.00  
cam 0.33 cam_alpha 0.0 cam_beta 1.0
</code></pre>
<p>LC-wPBE:</p>
<pre><code>xc xwpbe 1.00 cpbe96 1.0 hfexch 1.00  
cam 0.4 cam_alpha 0.00 cam_beta 1.00
</code></pre>
<p>LRC-wPBEh:</p>
<pre><code>xc xwpbe 0.80 cpbe96 1.0 hfexch 1.00 
cam 0.2 cam_alpha 0.20 cam_beta 0.80
</code></pre>
<p>QTP-00</p>
<pre><code>xc xcamb88 1.00 lyp 0.80 vwn_5 0.2 hfexch 1.00 
cam 0.29 cam_alpha 0.54 cam_beta 0.37
</code></pre>
<p>rCAM-B3LYP</p>
<pre><code>xc xcamb88 1.00 lyp 1.0 vwn_5 0. hfexch 1.00 becke88 nonlocal 0.13590
cam 0.33 cam_alpha 0.18352 cam_beta 0.94979
</code></pre>
<p>HSE03 functional: 0.25*Ex(HF-SR) - 0.25*Ex(PBE-SR) + Ex(PBE) +
Ec(PBE), where gamma(HF-SR) = gamma(PBE-SR)</p>
<pre><code>xc hse03
</code></pre>
<p>or it can be explicitly set as</p>
<pre><code>xc xpbe96 1.0 xcampbe96 -0.25 cpbe96 1.0 srhfexch 0.25
cam 0.33 cam_alpha 0.0 cam_beta 1.0
</code></pre>
<p>HSE06 functional:</p>
<pre><code>xc xpbe96 1.0 xcampbe96 -0.25 cpbe96 1.0 srhfexch 0.25
cam 0.11 cam_alpha 0.0 cam_beta 1.0
</code></pre>
<p>Please see references <sup id="fnref_footnote_76"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_76">76</a></sup><sup id="fnref_footnote_77"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_77">77</a></sup><sup id="fnref_footnote_78"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_78">78</a></sup><sup id="fnref_footnote_79"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_79">79</a></sup><sup id="fnref_footnote_80"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_80">80</a></sup><sup id="fnref_footnote_81"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_81">81</a></sup><sup id="fnref_footnote_82"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_82">82</a></sup><sup id="fnref_footnote_83"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_83">83</a></sup><sup id="fnref_footnote_84"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_84">84</a></sup><sup id="fnref_footnote_85"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_85">85</a></sup><sup id="fnref_footnote_86"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_86">86</a></sup><sup id="fnref_footnote_87"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_87">87</a></sup><sup id="fnref_footnote_88"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_88">88</a></sup>  (not a complete list) for further
details about the theory behind these functionals and applications.</p>
<p>Example illustrating the CAM-B3LYP functional:</p>
<pre><code>start h2o-camb3lyp  
geometry units angstrom  
   O      0.00000000     0.00000000     0.11726921  
   H      0.75698224     0.00000000    -0.46907685  
   H     -0.75698224     0.00000000    -0.46907685  
end  
basis spherical  
 * library aug-cc-pvdz  
end  
dft  
 xc xcamb88 1.00 lyp 0.81 vwn_5 0.19 hfexch 1.00  
 cam 0.33 cam_alpha 0.19 cam_beta 0.46  
 direct  
 iterations 100  
end  
task dft energy
</code></pre>
<p>Example illustrating the HSE03 functional:</p>
<pre><code>echo  
start h2o-hse  
geometry units angstrom  
O      0.00000000     0.00000000     0.11726921  
H      0.75698224     0.00000000    -0.46907685  
H     -0.75698224     0.00000000    -0.46907685  
end  
basis spherical  
 * library aug-cc-pvdz  
end  
dft  
 xc hse03  
 iterations 100  
 direct  
 end  
task dft energy
</code></pre>
<p>or alternatively</p>
<pre><code>dft  
 xc xpbe96 1.0 xcampbe96 -0.25 cpbe96 1.0 srhfexch 0.25  
 cam 0.33 cam_alpha 0.0 cam_beta 1.0  
 iterations 100  
 direct  
end  
task dft energy
</code></pre>
<h3 id="density-functional-theory-for-molecules-ssb-d-functional">SSB-D functional<a class="headerlink" href="#density-functional-theory-for-molecules-ssb-d-functional" title="Permanent link">&para;</a></h3>
<p>The SSB-D<sup id="fnref_footnote_89"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_89">89</a></sup><sup id="fnref_footnote_90"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_90">90</a></sup> functional is a small correction to the non-empirical
PBE functional and includes a portion of Grimme&rsquo;s dispersion correction
(s6=0.847455). It is designed to reproduce the good results of OPBE for
spin-state splittings and reaction barriers, and the good results of PBE
for weak interactions. The SSB-D functional works well for these
systems, including for difficult systems for DFT (dimerization of
anthracene, branching of octane, water-hexamer isomers, C<sub>12</sub>H<sub>12</sub> isomers,
stacked adenine dimers), and for NMR chemical shieldings.</p>
<p>It can be specified as</p>
<pre><code>xc ssb-d
</code></pre>
<h3 id="density-functional-theory-for-molecules-semi-empirical-hybrid-dft-combined-with-perturbative-mp2">Semi-empirical hybrid DFT combined with perturbative MP2<a class="headerlink" href="#density-functional-theory-for-molecules-semi-empirical-hybrid-dft-combined-with-perturbative-mp2" title="Permanent link">&para;</a></h3>
<p>This theory combines hybrid density functional theory with MP2
semi-empirically. The B2PLYP functional, which is an example of this
approximation, can be specified as:</p>
<pre><code>mp2
 freeze atomic
end
dft  
 xc HFexch 0.53 becke88 0.47 lyp 0.73 mp2 0.27  
 dftmp2 
end
</code></pre>
<p>For details of the theory, please see reference<sup id="fnref_footnote_91"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_91">91</a></sup>.</p>
<h3 id="density-functional-theory-for-molecules-lb94-and-cs00-asymptotic-correction">LB94 and CS00: Asymptotic correction<a class="headerlink" href="#density-functional-theory-for-molecules-lb94-and-cs00-asymptotic-correction" title="Permanent link">&para;</a></h3>
<p>The keyword <code>LB94</code> will correct the asymptotic region of the XC definition
of exchange-correlation potential by the van-Leeuwen-Baerends
exchange-correlation potential that has the correct <span class="arithmatex">\(-1/r\)</span> asymptotic
behavior. The total energy will be computed by the XC definition of
exchange-correlation functional. This scheme is known to tend to
overcorrect the deficiency of most uncorrected exchange-correlation
potentials.</p>
<p>The keyword <code>CS00</code>, when supplied with a real value of shift (in atomic
units), will perform Casida-Salahub &lsquo;00 asymptotic correction. This is
primarily intended for use with
<a href="#excited-state-calculations">TDDFT</a>. The shift is normally
positive (which means that the original uncorrected exchange-correlation
potential must be shifted down).</p>
<p>When the keyword <code>CS00</code> is specified without the value of shift, the
program will automatically supply it according to the semi-empirical
formula of Zhan, Nichols, and Dixon (again, see
<a href="#excited-state-calculations">TDDFT</a> for more details and
references). As the Zhan&rsquo;s formula is calibrated against B3LYP results,
it is most meaningful to use this with the B3LYP
functional, although the program does not prohibit (or even warn) the
use of any other functional.</p>
<p>Sample input files of asymptotically corrected TDDFT calculations can be
found in the corresponding
<a href="#excited-state-calculations-sample-input">section</a>.</p>
<h2 id="density-functional-theory-for-molecules-sample-input-file">Sample input file<a class="headerlink" href="#density-functional-theory-for-molecules-sample-input-file" title="Permanent link">&para;</a></h2>
<p>A simple example calculates the geometry of water, using the metaGGA
functionals <code>xtpss03</code> and <code>ctpss03</code>. This also highlights some of the print
features in the DFT module. Note that you must use the line
<code>task dft freq numerical</code>
because analytic hessians are not available for the
metaGGAs:</p>
<pre><code>title &quot;WATER 6-311G* meta-GGA XC geometry&quot;  
echo  
geometry units angstroms  
 O       0.0  0.0  0.0  
 H       0.0  0.0  1.0  
 H       0.0  1.0  0.0  
end  
basis  
 H library 6-311G*  
 O library 6-311G*  
end  
dft  
 iterations 50  
 print  kinetic_energy  
 xc xtpss03 ctpss03  
 decomp  
end  
task dft optimize   
task dft freq numerical
</code></pre>
<h2 id="density-functional-theory-for-molecules-iterations-or-maxiter-number-of-scf-iterations">ITERATIONS or MAXITER: Number of SCF iterations<a class="headerlink" href="#density-functional-theory-for-molecules-iterations-or-maxiter-number-of-scf-iterations" title="Permanent link">&para;</a></h2>
<pre><code> ITERATIONS or MAXITER &lt;integer iterations default 30&gt;
</code></pre>
<p>The default optimization in the DFT module is to iterate on the
Kohn-Sham (SCF) equations for a specified number of iterations (default
30). The keyword that controls this optimization is <code>ITERATIONS</code>, and has
the following general form,</p>
<pre><code>  iterations &lt;integer iterations default 30&gt;
</code></pre>
<p>or</p>
<pre><code>  maxiter &lt;integer iterations default 30&gt;
</code></pre>
<p>The optimization procedure will stop when the specified number of
iterations is reached or convergence is met. See an example that uses
this directive in <a href="#density-functional-theory-for-molecules-sample-input-file">Sample input
file</a>.</p>
<h2 id="density-functional-theory-for-molecules-convergence-scf-convergence-control">CONVERGENCE: SCF Convergence Control<a class="headerlink" href="#density-functional-theory-for-molecules-convergence-scf-convergence-control" title="Permanent link">&para;</a></h2>
<pre><code> CONVERGENCE [energy &lt;real energy default 1e-6&gt;] \  
             [density &lt;real density default 1e-5&gt;] \  
             [gradient &lt;real gradient default 5e-4&gt;] \  
             [hl_tol &lt;real hl_tol default 0.1&gt;]  
             [dampon &lt;real dampon default 0.0&gt;] \  
             [dampoff &lt;real dampoff default 0.0&gt;] \  
             [ncydp &lt;integer ncydp default 2&gt;] \  
             [ncyds &lt;integer ncyds default 30&gt;] \  
             [ncysh &lt;integer ncysh default 30&gt;] \  
             [damp &lt;integer ndamp default 0&gt;] [nodamping] \  
             [diison &lt;real diison default 0.0&gt;] \  
             [diisoff &lt;real diisoff default 0.0&gt;] \  
             [(diis [nfock &lt;integer nfock default 10&gt;]) || nodiis] \  
             [levlon &lt;real levlon default 0.0&gt;] \  
             [levloff &lt;real levloff default 0.0&gt;] \  
             [(lshift &lt;real lshift default 0.5&gt;) || nolevelshifting] \  
             [rabuck [n_rabuck &lt;integer n_rabuck default 25&gt;] \
             [fast] ]
</code></pre>
<p>Convergence is satisfied by meeting any or all of three criteria;</p>
<ul>
<li>convergence of the total energy; this is defined to be when the
    total DFT energy at iteration N and at iteration N-1 differ by a
    value less than a threshold value (the default is 1e-6). This value can be
    modified using the key word,</li>
</ul>
<pre><code>       CONVERGENCE energy &lt;real energy default 1e-6&gt;
</code></pre>
<ul>
<li>convergence of the total density; this is defined to be when the
    total DFT density matrix at iteration N and at iteration N-1 have a
    RMS difference less than some value (the default is 1e-5). This
    value can be modified using the keyword,</li>
</ul>
<pre><code>       CONVERGENCE density &lt;real density default 1e-5&gt;
</code></pre>
<ul>
<li>convergence of the orbital gradient; this is defined to be when the
    DIIS error vector becomes less than some value (the default is
    5e-4). This value can be modified using the keyword,</li>
</ul>
<pre><code>       CONVERGENCE gradient &lt;real gradient default 5e-4&gt;
</code></pre>
<p>The default optimization strategy is to immediately begin direct
inversion of the iterative subspace. In addition, if
the HOMO - LUMO gap is small and the Fock matrix diagonally
dominant, then level-shifting is automatically initiated. There are a
variety of ways to customize this procedure to whatever is desired.</p>
<p>An alternative optimization strategy is to specify, by using the change
in total energy (between iterations <var>N</var> and <var>N-1</var>), when to turn damping,
level-shifting, and/or DIIS on/off. Start and stop keywords for each of
these is available as,</p>
<pre><code> CONVERGENCE  [dampon &lt;real dampon default 0.0&gt;] \  
              [dampoff &lt;real dampoff default 0.0&gt;] \  
              [diison &lt;real diison default 0.0&gt;] \  
              [diisoff &lt;real diisoff default 0.0&gt;] \  
              [levlon &lt;real levlon default 0.0&gt;] \  
              [levloff &lt;real levloff default 0.0&gt;]
</code></pre>
<p>So, for example, damping, DIIS, and/or level-shifting can be turned
on/off as desired.</p>
<p>Another strategy can be to specify how many iterations (cycles)
you wish each type of procedure to be used. The necessary keywords to
control the number of damping cycles (<code>ncydp</code>), the number of DIIS cycles
(<code>ncyds</code>), and the number of level-shifting cycles (<code>ncysh</code>) are input as,</p>
<pre><code> CONVERGENCE  [ncydp &lt;integer ncydp default 2&gt;] \  
              [ncyds &lt;integer ncyds default 30&gt;] \  
              [ncysh &lt;integer ncysh default 0&gt;]
</code></pre>
<p>The amount of damping, level-shifting, time at which level-shifting is
automatically imposed, and Fock matrices used in the DIIS extrapolation
can be modified by the following keywords</p>
<pre><code> CONVERGENCE  [damp &lt;integer ndamp default 0&gt;] \  
              [diis [nfock &lt;integer nfock default 10&gt;]] \  
              [lshift &lt;real lshift default 0.5&gt;] \  
              [hl_tol &lt;real hl_tol default 0.1&gt;]]
</code></pre>
<h3 id="density-functional-theory-for-molecules-convergence-damp-keyword">CONVERGENCE DAMP Keyword<a class="headerlink" href="#density-functional-theory-for-molecules-convergence-damp-keyword" title="Permanent link">&para;</a></h3>
<p>Damping is defined to be the percentage of the previous iterations
density mixed with the current iterations density. So, for example</p>
<pre><code> CONVERGENCE damp 70
</code></pre>
<p>would mix 30% of the current iteration density with 70% of the previous
iteration density.</p>
<h3 id="density-functional-theory-for-molecules-convergence-lshift-keyword">CONVERGENCE LSHIFT Keyword<a class="headerlink" href="#density-functional-theory-for-molecules-convergence-lshift-keyword" title="Permanent link">&para;</a></h3>
<p>Level-Shifting is defined as the amount of shift applied to the diagonal
elements of the unoccupied block of the Fock matrix. The shift is
specified by the keyword <code>lshift</code>. For example the directive,</p>
<pre><code> CONVERGENCE lshift 0.5
</code></pre>
<p>causes the diagonal elements of the Fock matrix corresponding to the
virtual orbitals to be shifted by 0.5 a.u. By default, this
level-shifting procedure is switched on whenever the HOMO-LUMO gap is
small. Small is defined by default to be 0.05 au but can be modified by
the directive hl_tol. An example of changing the HOMO-LUMO gap
tolerance to 0.01 would be,</p>
<pre><code> CONVERGENCE hl_tol 0.01
</code></pre>
<h3 id="density-functional-theory-for-molecules-convergence-diis-keyword">CONVERGENCE DIIS Keyword<a class="headerlink" href="#density-functional-theory-for-molecules-convergence-diis-keyword" title="Permanent link">&para;</a></h3>
<p>Direct inversion of the iterative subspace with extrapolation of up to
10 Fock matrices is a default optimization procedure. For large
molecular systems the amount of available memory may preclude the
ability to store this number of <em>N<sup>2</sup></em> arrays in global memory. The
user may then specify the number of Fock matrices to be used in the
extrapolation (must be greater than three (3) to be effective). To set
the number of Fock matrices stored and used in the extrapolation
procedure to 3 would take the form,</p>
<pre><code> CONVERGENCE diis 3
</code></pre>
<p>The user has the ability to simply turn off any optimization procedures
deemed undesirable with the obvious keywords,</p>
<pre><code> CONVERGENCE [nodamping] [nodiis] [nolevelshifting]
</code></pre>
<p>For systems where the initial guess is very poor, the user can try using
fractional occupation of the orbital levels during the initial cycles of
the SCF convergence<sup id="density-functional-theory-for-molecules-fnref_footnote_92"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_92">92</a></sup>.
The input has the following form</p>
<pre><code> CONVERGENCE rabuck [n_rabuck &lt;integer n_rabuck default 25&gt;]]
</code></pre>
<p>where the optional value <code>n_rabuck</code> determines the number of SCF cycles
during which the method will be active. For example, to set equal to 30
the number of cycles where the Rabuck method is active, you need to use
the following line</p>
<pre><code> CONVERGENCE rabuck 30
</code></pre>
<h3 id="density-functional-theory-for-molecules-convergence-fast-keyword">CONVERGENCE FAST Keyword<a class="headerlink" href="#density-functional-theory-for-molecules-convergence-fast-keyword" title="Permanent link">&para;</a></h3>
<p>The <code>convergence fast</code> option turns on a series of parameters that most often speed-up convergence, but not in 100% of the cases.</p>
<pre><code>  CONVERGENCE fast
</code></pre>
<p>Here is an input snippet that would give you the same result as convergence fast</p>
<pre><code>dft
  convergence lshift 0. ncydp 0 dampon 1d99 dampoff 1d-4 damp 40
end
set quickguess t
task dft  
</code></pre>
<h2 id="density-functional-theory-for-molecules-cdft-constrained-dft">CDFT: Constrained DFT<a class="headerlink" href="#density-functional-theory-for-molecules-cdft-constrained-dft" title="Permanent link">&para;</a></h2>
<p>This option enables the constrained DFT formalism by Wu and Van Voorhis<sup id="fnref_footnote_93"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_93">93</a></sup>: </p>
<pre><code> CDFT &lt;integer fatom1 latom1&gt; [&lt;integer fatom2 latom2&gt;] (charge||spin &lt;real constaint_value&gt;) \ 
      [pop (becke||mulliken||lowdin) default lowdin]
</code></pre>
<p>Variables <code>fatom1</code> and <code>latom1</code> define the first and last atom of the group
of atoms to which the constraint will be applied. Therefore, the atoms in
the same group should be placed continuously in the geometry input. If
<code>fatom2</code> and <code>latom2</code> are specified, the difference between group 1 and 2
(i.e. 1-2) is constrained.</p>
<p>The constraint can be either on the charge or the spin density (number of
alpha - beta electrons) with a user specified <code>constraint_value</code>. Note:
No gradients have been implemented for the spin constraints case.
Geometry optimizations can only be performed using the charge
constraint.</p>
<p>To calculate the charge or spin density, the Becke, Mulliken, and Lowdin
population schemes can be used. The Lowdin scheme is default while the
Mulliken scheme is not recommended. If basis sets with many diffuse
functions are used, the Becke population scheme is recommended.</p>
<p>Multiple constraints can be defined simultaniously by defining multiple
cdft lines in the input. The same population scheme will be used for all
constraints and only needs to be specified once. If multiple population
options are defined, the last one will be used. When there are
convergence problems with multiple constraints, the user is advised to
do one constraint first and to use the resulting orbitals for the next
step of the constrained calculations.</p>
<p>It is best to put <code>convergence nolevelshifting</code> in the dft directive to
avoid issues with gradient calculations and convergence in CDFT. Use
orbital swap to get a broken-symmetry solution.</p>
<p>An input example is given below.</p>
<pre><code>geometry 
symmetry 
 C  0.0  0.0  0.0  
 O  1.2  0.0  0.0  
 C  0.0  0.0  2.0  
 O  1.2  0.0  2.0  
end  
basis  
 * library 6-31G* 
end  
dft  
 xc b3lyp  
 convergence nolevelshifting  
 odft  
 mult 1  
 vectors swap beta 14 15  
 cdft 1 2 charge 1.0  
end  
task dft
</code></pre>
<h2 id="density-functional-theory-for-molecules-smear-fractional-occupation-of-the-molecular-orbitals">SMEAR: Fractional Occupation of the Molecular Orbitals<a class="headerlink" href="#density-functional-theory-for-molecules-smear-fractional-occupation-of-the-molecular-orbitals" title="Permanent link">&para;</a></h2>
<p>The <code>SMEAR</code> keyword is useful in cases with many degenerate states near
the HOMO (eg metallic clusters)</p>
<pre><code> SMEAR &lt;real smear default 0.001&gt;
</code></pre>
<p>This option allows fractional occupation of the molecular orbitals. A
Gaussian broadening function of exponent smear is used as described in
the paper by Warren and Dunlap<sup id="density-functional-theory-for-molecules-fnref_footnote_94"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_94">94</a></sup>.
The user must be aware that an additional energy term is added
to the total energy in order to have energies and gradients consistent.</p>
<h2 id="density-functional-theory-for-molecules-fon-calculations-with-fractional-numbers-of-electrons">FON: Calculations with fractional numbers of electrons<a class="headerlink" href="#density-functional-theory-for-molecules-fon-calculations-with-fractional-numbers-of-electrons" title="Permanent link">&para;</a></h2>
<h3 id="density-functional-theory-for-molecules-restricted">Restricted<a class="headerlink" href="#density-functional-theory-for-molecules-restricted" title="Permanent link">&para;</a></h3>
<pre><code>fon partial 3 electrons 1.8 filled 2
</code></pre>
<p>Here 1.8 electrons will be equally divided over 3 valence orbitals and 2
orbitals are fully filled. The total number of electrons here is 5.8</p>
<p>Example input:</p>
<pre><code>echo 
title &quot;carbon atom&quot; 
start carbon_fon 
geometry  
symmetry c1  
  C   0.0 0.0 0.0  
end  
basis  
 * library 6-31G  
end
dft 
 direct  
 grid xfine  
 convergence energy 1d-8  
 xc pbe0  
 fon partial 3 electrons 1.8 filled 2 
end
task dft energy
</code></pre>
<h3 id="density-functional-theory-for-molecules-unrestricted">Unrestricted<a class="headerlink" href="#density-functional-theory-for-molecules-unrestricted" title="Permanent link">&para;</a></h3>
<pre><code>fon alpha partial 3 electrons 0.9 filled 2 
fon beta partial 3 electrons 0.9 filled 2
</code></pre>
<p>Here 0.9 electrons will be equally divided over 3 alpha valence orbitals
and 2 alpha orbitals are fully filled. Similarly for beta. The total
number of electrons here is 5.8</p>
<p>Example input:</p>
<pre><code>echo  
title &quot;carbon atom&quot;  
start carbon_fon 
geometry  
 C   0.0 0.0 0.0  
end  
basis  
 * library 6-31G  
end  
dft 
 odft  
 fon alpha partial 3 electrons 0.9 filled 2  
 fon beta partial 3 electrons 0.9 filled 2  
end  
task dft energy
</code></pre>
<p>To set fractional numbers in the core orbitals, add the following
directive in the input file:</p>
<pre><code>   set dft:core_fon .true.
</code></pre>
<p>Example input:</p>
<pre><code>dft
 print &quot;final vectors analysis&quot;
 odft
 direct
 fon alpha partial 2 electrons 1.0 filled 2
 fon beta partial 2 electrons 1.0 filled 2
 xc pbe0
 convergence energy 1d-8
end
task dft
</code></pre>
<h2 id="density-functional-theory-for-molecules-occup-controlling-the-occupations-of-molecular-orbitals">OCCUP: Controlling the occupations of molecular orbitals<a class="headerlink" href="#density-functional-theory-for-molecules-occup-controlling-the-occupations-of-molecular-orbitals" title="Permanent link">&para;</a></h2>
<p>Example:</p>
<pre><code>echo  
start h2o_core_hole  
memory 1000 mb  
geometry units au  
 O 0       0        0  
 H 0       1.430   -1.107  
 H 0      -1.430   -1.107  
end  
basis  
 O library 6-31g*  
 H library 6-31g*  
end  
occup     
 6 6      # occupation list for 6 alpha and 6 beta orbitals 
 1.0 0.0  # core-hole in the first beta orbital
 1.0 1.0  
 1.0 1.0  
 1.0 1.0  
 1.0 1.0  
 0.0 0.0  
end  
dft  
 odft  
 mult 1  
 xc beckehandh  
end  
task dft
</code></pre>
<h2 id="density-functional-theory-for-molecules-grid-numerical-integration-of-the-xc-potential">GRID: Numerical Integration of the XC Potential<a class="headerlink" href="#density-functional-theory-for-molecules-grid-numerical-integration-of-the-xc-potential" title="Permanent link">&para;</a></h2>
<pre><code> GRID [(xcoarse||coarse||medium||fine||xfine||huge) default medium] \  
      [(gausleg||lebedev ) default lebedev ] \  
      [(becke||erf1||erf2||ssf) default erf1] \  
      [(euler||mura||treutler) default mura] \  
      [rm &lt;real rm default 2.0&gt;] \  
      [nodisk]
</code></pre>
<p>A numerical integration is necessary for the evaluation of the
exchange-correlation contribution to the density functional. The default
quadrature used for the numerical integration is an Euler-MacLaurin
scheme for the radial components (with a modified Mura-Knowles
transformation) and a Lebedev scheme for the angular components. Within
this numerical integration procedure various levels of accuracy have
been defined and are available to the user. The user can specify the
level of accuracy with the keywords; <code>xcoarse</code>, <code>coarse</code>, <code>medium</code>, <code>fine</code>, <code>xfine</code> and
<code>huge</code>. The default is <code>medium</code>.</p>
<pre><code> GRID [xcoarse||coarse||medium||fine||xfine||huge]
</code></pre>
<p>Our intent is to have a numerical integration scheme which would give us
approximately the accuracy defined below regardless of molecular
composition.</p>
<table>
<thead>
<tr>
<th>Keyword</th>
<th>Total Energy Target Accuracy</th>
</tr>
</thead>
<tbody>
<tr>
<td>xcoarse</td>
<td>1&sdot;10<sup>-4</sup></td>
</tr>
<tr>
<td>coarse</td>
<td>1&sdot;10<sup>-5</sup></td>
</tr>
<tr>
<td>medium</td>
<td>1&sdot;10<sup>-6</sup></td>
</tr>
<tr>
<td>fine</td>
<td>1&sdot;10<sup>-7</sup></td>
</tr>
<tr>
<td>xfine</td>
<td>1&sdot;10<sup>-8</sup></td>
</tr>
<tr>
<td>huge</td>
<td>1&sdot;10<sup>-10</sup></td>
</tr>
</tbody>
</table>
<p>In order to determine the level of radial and angular quadrature needed
to give us the target accuracy, we computed total DFT energies at the LDA
level of theory for many homonuclear atomic, diatomic and triatomic
systems in rows 1-4 of the periodic table. In each case all bond lengths
were set to twice the Bragg-Slater radius. The total DFT energy of the
system was computed using the converged SCF density with atoms having
radial shells ranging from 35-235 (at fixed 48/96 angular quadratures)
and angular quadratures of 12/24-48/96 (at fixed 235 radial shells). The
error of the numerical integration was determined by comparison to a
&ldquo;best&rdquo; or most accurate calculation in which a grid of 235 radial points
48 theta and 96 phi angular points on each atom was used. This
corresponds to approximately 1 million points per atom. The following
tables were empirically determined to give the desired target accuracy
for DFT total energies. These tables below show the number of radial and
angular shells which the DFT module will use for for a given atom
depending on the row it is in (in the periodic table) and the desired
accuracy. Note, differing atom types in a given molecular system will
most likely have differing associated numerical grids. The intent is to
generate the desired energy accuracy (at the expense of speed of the calculation).</p>
<table>
<thead>
<tr>
<th>Keyword</th>
<th>Radial</th>
<th>Angular</th>
</tr>
</thead>
<tbody>
<tr>
<td>xcoarse</td>
<td>21</td>
<td>194</td>
</tr>
<tr>
<td>coarse</td>
<td>35</td>
<td>302</td>
</tr>
<tr>
<td>medium</td>
<td>49</td>
<td>434</td>
</tr>
<tr>
<td>fine</td>
<td>70</td>
<td>590</td>
</tr>
<tr>
<td>xfine</td>
<td>100</td>
<td>1202</td>
</tr>
</tbody>
</table>
<p>Program default number of radial and angular shells empirically
determined for Row 1 atoms (Li &rarr; F) to reach the desired
accuracies.</p>
<table>
<thead>
<tr>
<th>Keyword</th>
<th>Radial</th>
<th>Angular</th>
</tr>
</thead>
<tbody>
<tr>
<td>xcoarse</td>
<td>42</td>
<td>194</td>
</tr>
<tr>
<td>coarse</td>
<td>70</td>
<td>302</td>
</tr>
<tr>
<td>medium</td>
<td>88</td>
<td>434</td>
</tr>
<tr>
<td>fine</td>
<td>123</td>
<td>770</td>
</tr>
<tr>
<td>xfine</td>
<td>125</td>
<td>1454</td>
</tr>
<tr>
<td>huge</td>
<td>300</td>
<td>1454</td>
</tr>
</tbody>
</table>
<p>Program default number of radial and angular shells empirically
determined for Row 2 atoms (Na &rarr; Cl) to reach the desired
accuracies.</p>
<table>
<thead>
<tr>
<th>Keyword</th>
<th>Radial</th>
<th>Angular</th>
</tr>
</thead>
<tbody>
<tr>
<td>xcoarse</td>
<td>75</td>
<td>194</td>
</tr>
<tr>
<td>coarse</td>
<td>95</td>
<td>302</td>
</tr>
<tr>
<td>medium</td>
<td>112</td>
<td>590</td>
</tr>
<tr>
<td>fine</td>
<td>130</td>
<td>974</td>
</tr>
<tr>
<td>xfine</td>
<td>160</td>
<td>1454</td>
</tr>
<tr>
<td>huge</td>
<td>400</td>
<td>1454</td>
</tr>
</tbody>
</table>
<p>Program default number of radial and angular shells empirically
determined for Row 3 atoms (K &rarr; Br) to reach the desired
accuracies.</p>
<table>
<thead>
<tr>
<th>Keyword</th>
<th>Radial</th>
<th>Angular</th>
</tr>
</thead>
<tbody>
<tr>
<td>xcoarse</td>
<td>84</td>
<td>194</td>
</tr>
<tr>
<td>coarse</td>
<td>104</td>
<td>302</td>
</tr>
<tr>
<td>medium</td>
<td>123</td>
<td>590</td>
</tr>
<tr>
<td>fine</td>
<td>141</td>
<td>974</td>
</tr>
<tr>
<td>xfine</td>
<td>205</td>
<td>1454</td>
</tr>
<tr>
<td>huge</td>
<td>400</td>
<td>1454</td>
</tr>
</tbody>
</table>
<p>Program default number of radial and angular shells empirically
determined for Row 4 atoms (Rb &rarr; I) to reach the desired
accuracies.</p>
<h3 id="density-functional-theory-for-molecules-angular-grids">Angular grids<a class="headerlink" href="#density-functional-theory-for-molecules-angular-grids" title="Permanent link">&para;</a></h3>
<p>In addition to the simple keyword specifying the desired accuracy as
described above, the user has the option of specifying a custom
quadrature of this type in which ALL atoms have the same grid
specification. This is accomplished by using the <code>gausleg</code> keyword.</p>
<h4 id="density-functional-theory-for-molecules-gauss-legendre-angular-grid">Gauss-Legendre angular grid<a class="headerlink" href="#density-functional-theory-for-molecules-gauss-legendre-angular-grid" title="Permanent link">&para;</a></h4>
<pre><code> GRID gausleg &lt;integer nradpts default 50&gt; &lt;integer nagrid default 10&gt;
</code></pre>
<p>In this type of grid, the number of phi points is twice the number of
theta points. So, for example, a specification of,</p>
<pre><code> GRID gausleg 80 20
</code></pre>
<p>would be interpreted as 80 radial points, 20 theta points, and 40 phi
points per center (or 64000 points per center before pruning).</p>
<h4 id="density-functional-theory-for-molecules-lebedev-angular-grid">Lebedev angular grid<a class="headerlink" href="#density-functional-theory-for-molecules-lebedev-angular-grid" title="Permanent link">&para;</a></h4>
<p>A second quadrature is the Lebedev scheme for the angular
components. Within this numerical integration procedure various
levels of accuracy have also been defined and are available to the user.
The input for this type of grid takes the form,</p>
<pre><code> GRID lebedev &lt;integer radpts &gt; &lt;integer iangquad &gt;
</code></pre>
<p>In this context the variable <code>iangquad</code> specifies a certain number of
angular points as indicated by the table below:</p>
<table id="_table-2">
<caption style="caption-side:bottom"><span>Table&nbsp;2:</span> List of Lebedev quadratures</caption>
<thead>
<tr>
<th>IANGQUAD</th>
<th>N<sub>angular</sub></th>
<th>l</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>38</td>
<td>9</td>
</tr>
<tr>
<td>2</td>
<td>50</td>
<td>11</td>
</tr>
<tr>
<td>3</td>
<td>74</td>
<td>13</td>
</tr>
<tr>
<td>4</td>
<td>86</td>
<td>15</td>
</tr>
<tr>
<td>5</td>
<td>110</td>
<td>17</td>
</tr>
<tr>
<td>6</td>
<td>146</td>
<td>19</td>
</tr>
<tr>
<td>7</td>
<td>170</td>
<td>21</td>
</tr>
<tr>
<td>8</td>
<td>194</td>
<td>23</td>
</tr>
<tr>
<td>9</td>
<td>230</td>
<td>25</td>
</tr>
<tr>
<td>10</td>
<td>266</td>
<td>27</td>
</tr>
<tr>
<td>11</td>
<td>302</td>
<td>29</td>
</tr>
<tr>
<td>12</td>
<td>350</td>
<td>31</td>
</tr>
<tr>
<td>13</td>
<td>434</td>
<td>35</td>
</tr>
<tr>
<td>14</td>
<td>590</td>
<td>41</td>
</tr>
<tr>
<td>15</td>
<td>770</td>
<td>47</td>
</tr>
<tr>
<td>16</td>
<td>974</td>
<td>53</td>
</tr>
<tr>
<td>17</td>
<td>1202</td>
<td>59</td>
</tr>
<tr>
<td>18</td>
<td>1454</td>
<td>65</td>
</tr>
<tr>
<td>19</td>
<td>1730</td>
<td>71</td>
</tr>
<tr>
<td>20</td>
<td>2030</td>
<td>77</td>
</tr>
<tr>
<td>21</td>
<td>2354</td>
<td>83</td>
</tr>
<tr>
<td>22</td>
<td>2702</td>
<td>89</td>
</tr>
<tr>
<td>23</td>
<td>3074</td>
<td>95</td>
</tr>
<tr>
<td>24</td>
<td>3470</td>
<td>101</td>
</tr>
<tr>
<td>25</td>
<td>3890</td>
<td>107</td>
</tr>
<tr>
<td>26</td>
<td>4334</td>
<td>113</td>
</tr>
<tr>
<td>27</td>
<td>4802</td>
<td>119</td>
</tr>
<tr>
<td>28</td>
<td>5294</td>
<td>125</td>
</tr>
<tr>
<td>29</td>
<td>5810</td>
<td>131</td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<p>Therefore the user can specify any number of radial points along with
the level of angular quadrature (1-29).</p>
<p>The user can also specify grid parameters specific for a given atom
type: parameters that must be supplied are: atom tag and number of
radial points. As an example, here is a grid input line for the water
molecule</p>
<pre><code>grid lebedev 80 11 H 70 8  O 90 11
</code></pre>
<h3 id="density-functional-theory-for-molecules-partitioning-functions">Partitioning functions<a class="headerlink" href="#density-functional-theory-for-molecules-partitioning-functions" title="Permanent link">&para;</a></h3>
<pre><code>GRID [(becke||erf1||erf2||ssf) default erf1]
</code></pre>
<ul>
<li>becke : see paper<sup id="fnref_footnote_95"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_95">95</a></sup></li>
<li>ssf : see paper<sup id="fnref_footnote_96"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_96">96</a></sup></li>
<li>erf1 : modified ssf</li>
<li>erf2 : modified ssf</li>
</ul>
<p>Erf<em>n</em> partitioning functions</p>
<div class="arithmatex">\[\begin{array}{lcl}  w_A(r) &amp; = &amp; \prod_{B\neq A}\frac{1}{2} \left[1 \ - \ erf(\mu^\prime_{AB})\right] \\  \mu^\prime_{AB} &amp; = &amp; \frac{1}{\alpha} \ \frac{\mu_{AB}}{(1-\mu_{AB}^2)^n} \\  \mu_{AB} &amp; = &amp; \frac{{\mathbf r}_A - {\mathbf r}_B} {\left|{\mathbf r}_A - {\mathbf r}_B \right|} \end{array}\]</div>
<h3 id="density-functional-theory-for-molecules-radial-grids">Radial grids<a class="headerlink" href="#density-functional-theory-for-molecules-radial-grids" title="Permanent link">&para;</a></h3>
<pre><code> GRID [[euler||mura||treutler]  default mura]
</code></pre>
<ul>
<li>euler : Euler-McLaurin quadrature with the transformation devised by
    Murray et al<sup id="density-functional-theory-for-molecules-fnref_footnote_97"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_97">97</a></sup>.</li>
<li>mura : Modification of the Murray-Handy-Laming scheme (we are not using the same
    scaling factors proposed in the paper by Mura and Knowles<sup id="density-functional-theory-for-molecules-fnref_footnote_98"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_98">98</a></sup>).</li>
<li>treutler : Gauss-Chebyshev using the transformation suggested by Treutler<sup id="fnref_footnote_99"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_99">99</a></sup>.</li>
</ul>
<h3 id="density-functional-theory-for-molecules-disk-usage-for-grid">Disk usage for Grid<a class="headerlink" href="#density-functional-theory-for-molecules-disk-usage-for-grid" title="Permanent link">&para;</a></h3>
<pre><code> NODISK
</code></pre>
<p>This keyword turns off storage of grid points and weights on disk.</p>
<h2 id="density-functional-theory-for-molecules-tolerances-screening-tolerances">TOLERANCES: Screening tolerances<a class="headerlink" href="#density-functional-theory-for-molecules-tolerances-screening-tolerances" title="Permanent link">&para;</a></h2>
<pre><code> TOLERANCES [[tight] [tol_rho &lt;real tol_rho default 1e-10&gt;] \  
             [accCoul &lt;integer accCoul default 8&gt;] \  
             [radius &lt;real radius default 25.0&gt;]]
</code></pre>
<p>The user has the option of controlling screening for the tolerances in
the integral evaluations for the DFT module. In most applications, the
default values will be adequate for the calculation, but different
values can be specified in the input for the DFT module using the
keywords described below.</p>
<p>The input parameter <code>accCoul</code> is used to define the tolerance in Schwarz
screening for the Coulomb integrals. Only integrals with estimated
values greater than 10<sup>(-accCoul)</sup> are evaluated.</p>
<pre><code> TOLERANCES accCoul &lt;integer accCoul default 8&gt;
</code></pre>
<p>Screening away needless computation of the XC functional (on the grid)
due to negligible density is also possible with the use of,</p>
<pre><code> TOLERANCES tol_rho &lt;real tol_rho default 1e-10&gt;
</code></pre>
<p>XC functional computation is bypassed if the corresponding density
elements are less than <code>tol_rho</code>.</p>
<p>A screening parameter, radius, used in the screening of the Becke or
Delley spatial weights is also available as,</p>
<pre><code> TOLERANCES radius &lt;real radius default 25.0&gt;
</code></pre>
<p>where <code>radius</code> is the cutoff value in bohr.</p>
<p>The tolerances as discussed previously are insured at convergence. More
sleazy tolerances are invoked early in the iterative process which can
speed things up a bit. This can also be problematic at times because it
introduces a discontinuity in the convergence process. To avoid use of
initial sleazy tolerances the user can invoke the tight option:</p>
<pre><code> TOLERANCES tight
</code></pre>
<p>This option sets all tolerances to their default/user specified values
at the very first
iteration.</p>
<h2 id="density-functional-theory-for-molecules-direct-semidirect-and-noio-hardware-resource-control">DIRECT, SEMIDIRECT and NOIO: Hardware Resource Control<a class="headerlink" href="#density-functional-theory-for-molecules-direct-semidirect-and-noio-hardware-resource-control" title="Permanent link">&para;</a></h2>
<pre><code> DIRECT||INCORE  
 SEMIDIRECT [filesize &lt;integer filesize default disksize&gt;]  
            [memsize  &lt;integer memsize default available&gt;]  
            [filename &lt;string filename default $file_prefix.aoints$]  
 NOIO
</code></pre>
<p>The inverted charge-density and exchange-correlation matrices for a DFT
calculation are normally written to disk storage. The user can prevent
this by specifying the keyword <code>noio</code> within the input for the DFT
directive. The input to exercise this option is as follows,</p>
<pre><code>  noio
</code></pre>
<p>If this keyword is encountered, then the two matrices (inverted
charge-density and exchange-correlation) are computed &ldquo;on-the-fly&rdquo;
whenever needed.</p>
<p>The <code>INCORE</code> option is always assumed to be true but can be overridden
with the option <code>DIRECT</code> in which case all integrals are computed
&ldquo;on-the-fly&rdquo;.</p>
<p>The <code>SEMIDIRECT</code> option controls caching of integrals. A full description
of this option is described in the <a href="#hartree-fock-theory-for-molecules-direct-and-semidirect-recomputation-of-integrals">Hartree-Fock</a> section. Some functionality
which is only compatible with the <code>DIRECT</code> option will not, at present,
work when using <code>SEMIDIRECT</code>.</p>
<h2 id="density-functional-theory-for-molecules-odft-and-mult-open-shell-systems">ODFT and MULT: Open shell systems<a class="headerlink" href="#density-functional-theory-for-molecules-odft-and-mult-open-shell-systems" title="Permanent link">&para;</a></h2>
<pre><code> ODFT
 MULT &lt;integer mult default 1&gt;
</code></pre>
<p>Both closed-shell and open-shell systems can be studied using the DFT
module. Specifying the keyword <code>MULT</code> within the <code>DFT</code> directive allows the
user to define the spin multiplicity of the system. The form of the
input line is as follows;</p>
<pre><code>  MULT &lt;integer mult default 1&gt;
</code></pre>
<p>When the keyword <code>MULT</code> is specified, the user can define the integer
variable <code>mult</code>, where mult is equal to the number of alpha electrons
minus beta electrons, plus 1.</p>
<p>When <code>MULT</code> is set to a negative number. For example, if <code>MULT = -3</code>, a
triplet calculation will be performed with the beta electrons
preferentially occupied. For <code>MULT = 3</code>, the alpha electrons will be
preferentially occupied.</p>
<p>The keyword <code>ODFT</code> is unnecessary except in the context of forcing a
singlet system to be computed as an open shell system (i.e., using a
spin-unrestricted wavefunction).</p>
<h2 id="density-functional-theory-for-molecules-cgmin-quadratic-convergence-algorithm">CGMIN: Quadratic convergence algorithm<a class="headerlink" href="#density-functional-theory-for-molecules-cgmin-quadratic-convergence-algorithm" title="Permanent link">&para;</a></h2>
<p>The <code>cgmin</code> keyword will use the quadratic convergence algorithm. It is
possible to turn the use of the quadratic convergence algorithm off with
the <code>nocgmin</code> keyword.</p>
<h2 id="density-functional-theory-for-molecules-rodft-restricted-open-shell-dft">RODFT: Restricted open-shell DFT<a class="headerlink" href="#density-functional-theory-for-molecules-rodft-restricted-open-shell-dft" title="Permanent link">&para;</a></h2>
<p>The <code>rodft</code> keyword will perform restricted open-shell calculations. This
keyword can only be used with the <code>CGMIN</code> keyword.</p>
<h2 id="density-functional-theory-for-molecules-sic-self-interaction-correction">SIC: Self-Interaction Correction<a class="headerlink" href="#density-functional-theory-for-molecules-sic-self-interaction-correction" title="Permanent link">&para;</a></h2>
<pre><code>sic [perturbative || oep || oep-loc ]
&lt;default perturbative&gt;
</code></pre>
<p>The Perdew and Zunger<sup id="fnref_footnote_10"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_10">10</a></sup> method to remove the self-interaction contained in many
exchange-correlation functionals has been implemented with the Optimized
Effective Potential method<sup id="density-functional-theory-for-molecules-fnref_footnote_100"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_100">100</a></sup><sup id="fnref_footnote_101"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_101">101</a></sup> within the Krieger-Li-Iafrate approximation<sup id="fnref_footnote_102"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_102">102</a></sup><sup id="fnref_footnote_103"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_103">103</a></sup><sup id="fnref_footnote_104"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_104">104</a></sup>.
Three variants of these methods are included in NWChem:</p>
<ul>
<li><code>sic perturbative</code> This is the default option for the sic directive.
    After a self-consistent calculation, the Kohn-Sham orbitals are
    localized with the Foster-Boys algorithm (see section on <a href="#hartree-fock-theory-for-molecules-orbital-localization">orbital localization</a>) and the
    self-interaction energy is added to the total energy. All
    exchange-correlation functionals implemented in the NWChem can be
    used with this option.</li>
<li><code>sic oep</code> With this option the optimized effective potential is built
    in each step of the self-consistent process. Because the
    electrostatic potential generated for each orbital involves a
    numerical integration, this method can be expensive.</li>
<li><code>sic oep-loc</code> This option is similar to the oep option with the
    addition of localization of the Kohn-Sham orbitals in each step of
    the self-consistent process.</li>
</ul>
<p>With <code>oep</code> and <code>oep-loc</code> options a <code>xfine grid</code> (see section about <a href="#density-functional-theory-for-molecules-grid-numerical-integration-of-the-xc-potential">numerical integration</a> ) must be
used in order to avoid numerical noise, furthermore the hybrid
functionals can not be used with these options. More details of the
implementation of this method can be found in the paper by Garza<sup id="density-functional-theory-for-molecules-fnref_footnote_105"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_105">105</a></sup>. The components of the
sic energy can be printed out using:</p>
<pre><code>print &quot;SIC information&quot;
</code></pre>
<h2 id="density-functional-theory-for-molecules-mulliken-mulliken-analysis">MULLIKEN: Mulliken analysis<a class="headerlink" href="#density-functional-theory-for-molecules-mulliken-mulliken-analysis" title="Permanent link">&para;</a></h2>
<p>Mulliken analysis of the charge distribution is invoked by the keyword:</p>
<pre><code> MULLIKEN
</code></pre>
<p>When this keyword is encountered, Mulliken analysis of both the input
density as well as the output density will occur. For example, to
perform a mulliken analysis and print the explicit population analysis
of the basis functions, use the following</p>
<pre><code>dft
 mulliken
 print &quot;mulliken ao&quot;
end
task dft
</code></pre>
<h2 id="density-functional-theory-for-molecules-fukui-fukui-indices">FUKUI: Fukui Indices<a class="headerlink" href="#density-functional-theory-for-molecules-fukui-fukui-indices" title="Permanent link">&para;</a></h2>
<p>Fukui indices analysis is invoked by the keyword:</p>
<pre><code> FUKUI
</code></pre>
<p>When this keyword is encounters, the condensed Fukui indices will be
calculated and printed in the output. Detailed information about the
analysis can be obtained using the following</p>
<pre><code> dft  
   fukui 
   print &quot;Fukui information&quot;  
 end  
 task dft
</code></pre>
<p>The implementation of the Fukui analyis in NWChem was based on the papers by Galvan<sup id="fnref_footnote_106"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_106">106</a></sup> and by Chamorro<sup id="fnref_footnote_107"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_107">107</a></sup>.<br />
This implementation makes use of the generalized Fukui indices (<span class="arithmatex">\(f_{SN},  f_{NS}, f_{SS}\)</span>).<br />
The traditional, spin-restricted, Fukui indices are given by <span class="arithmatex">\(f_{NN}^+\)</span>, <span class="arithmatex">\(f_{NN}^-\)</span> and their average:</p>
<div class="arithmatex">\[\begin{array}{lcl} f_A^+ &amp;=&amp; f_{NN}^+  \\ f_A^-  &amp;=&amp; f_{NN}^- \\ f_A^0 &amp;=&amp; \frac{1}{2}\left( f_{NN}^+ + f_{NN}^- \right) \end{array}\]</div>
<h2 id="density-functional-theory-for-molecules-bsse-basis-set-superposition-error">BSSE: Basis Set Superposition Error<a class="headerlink" href="#density-functional-theory-for-molecules-bsse-basis-set-superposition-error" title="Permanent link">&para;</a></h2>
<p>Particular care is required to compute BSSE by the counter-poise method
for the DFT module. In order to include terms deriving from the
numerical grid used in the XC integration, the user must label the ghost
atoms not just bq, but bq followed by the given atomic symbol. For
example, the first component needed to compute the BSSE for the water
dimer, should be written as follows</p>
<pre><code>geometry h2o autosym units au
 O        0.00000000     0.00000000     0.22143139
 H        1.43042868     0.00000000    -0.88572555
 H       -1.43042868     0.00000000    -0.88572555
 bqH      0.71521434     0.00000000    -0.33214708
 bqH     -0.71521434     0.00000000    -0.33214708
 bqO      0.00000000     0.00000000    -0.88572555
end
basis
 H library aug-cc-pvdz
 O library aug-cc-pvdz
 bqH library H aug-cc-pvdz
 bqO library O aug-cc-pvdz
end
</code></pre>
<p>Please note that the <em>ghost</em> oxygen atom has been labeled <code>bqO</code>, and
not just <code>bq</code>.</p>
<h2 id="density-functional-theory-for-molecules-disp-empirical-long-range-contribution-vdw">DISP: Empirical Long-range Contribution (vdW)<a class="headerlink" href="#density-functional-theory-for-molecules-disp-empirical-long-range-contribution-vdw" title="Permanent link">&para;</a></h2>
<pre><code> DISP \
      [ vdw &lt;real vdw integer default 2]] \
      [[s6 &lt;real s6 default depends on XC functional&gt;] \
      [ alpha &lt;real alpha default 20.0d0] \
      [ off ] 
</code></pre>
<p>When systems with high dependence on van der Waals interactions are
computed, the dispersion term may be added empirically through
long-range contribution DFT-D, i.e. <em>E<sub>DFT-D</sub>=E<sub>DFT-KS</sub>+E<sub>disp</sub></em>, 
where:</p>
<div class="arithmatex">\[E_{disp}=-s_6\sum_{i=1}^{N_{atom}-1}\sum_{j=i+1}^{N_{atom}} \frac{C_{6}^{ij}}{R_{ij}^{6}} \frac{1}{ 1+e^{-\alpha \frac{R_{ij}}{R_{vdw}}}}\]</div>
<p>In this equation, the  <em>s<sub>6</sub></em> term depends in the functional and basis
  set used, <em>C<sub>6</sub><sup>ij</sup></em> is the dispersion coefficient between pairs of
  atoms.  <em>R<sub>vdw</sub></em> and  <em>R<sub>ij</sub></em> are related with van der Waals atom
radii and the nucleus distance respectively. The  <em>&alpha;</em> value
contributes to control the corrections at intermediate distances.</p>
<p>There are available three ways to compute   <em>C<sup>6</sup><sub>ij</sub></em>:</p>
<ol>
<li>
<p><code>vdw 0</code> uses the formula <span class="arithmatex">\(C_6^{ij}= \frac{2(C_6^{i}C_6^{j})^{2/3}(N_{eff\ i}N_{eff\ j})^{1/3}} {C_6^{i}(N_{eff\ i}^2)^{1/3}+(C_6^{i}N_{eff\ j}^2)^{1/3}}\)</span>  <br />
     where <em>N<sub>eff</sub></em> and <em>C<sub>6</sub></em> are obtained from references <sup id="fnref_footnote_108"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_108">108</a></sup> and <sup id="fnref_footnote_109"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_109">109</a></sup>  </p>
</li>
<li>
<p><code>vdw 1</code> uses the formula <span class="arithmatex">\(C_6^{ij}=2\ \frac{C_6^{i}C_6^{j}}{C_6^{i}+C_6^{j}}\)</span>.<br />
   See details in reference<sup id="density-functional-theory-for-molecules-fnref_footnote_110"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_110">110</a></sup>.</p>
</li>
<li>
<p><code>vdw 2</code> uses the formula <span class="arithmatex">\(C_6^{ij}=\sqrt{C_6^{i}C_6^{j}}\)</span>.<br />
   See details in reference<sup id="density-functional-theory-for-molecules-fnref2_footnote_91"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_91">91</a></sup>.</p>
</li>
</ol>
<p>Note that in each option there is a certain set of <em>C<sub>6</sub></em> and <em>R<sub>vdw</sub></em>.
Also note that Grimme only defined parameters for elements
up to Z=54 for the dispersion correction above.
<em>C<sub>6</sub></em> values for
elements above Z=54 have been set to zero.</p>
<p>For options <code>vdw 1</code> and <code>vdw 2</code> , there are s<sub>6</sub> values by default for
some functionals and triple-zeta plus double polarization basis set
(TZV2P):</p>
<ul>
<li><code>vdw 1</code> BLYP 1.40, PBE 0.70 and BP86 1.30.</li>
<li><code>vdw 2</code> BLYP 1.20, PBE 0.75, BP86 1.05, B3LYP 1.05, Becke97-D 1.25
    and TPSS 1.00.</li>
</ul>
<p>Grimme&rsquo;s DFT-D3 is also available. Here the dispersion term has the
following form:</p>
<div class="arithmatex">\[ E_{disp} = \sum_{i,j} \sum_{n=6,8} s_n \ \frac{C^{ij}_n}{r_{ij}} \biggl\lbrace 1 + 6 \bigl[\frac{r_{ij}}{s_{r,n} \ R_0^{ij}} \bigr]^{-\alpha n} \biggr\rbrace^{-1}\]</div>
<p>This new dispersion correction covers elements through Z=94.
  <em>C<sup>ij</sup><sub>n</sub> (n=6,8)</em> are coordination and geometry dependent. Details
about the functional form can be found in reference <sup id="density-functional-theory-for-molecules-fnref_footnote_111"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_111">111</a></sup>.</p>
<p>To use the Grimme DFT-D3 dispersion correction, use the option</p>
<ul>
<li>
<p><code>vdw 3</code> (<code>s6</code> and <code>alpha</code> cannot be set manually). Functionals for which
    DFT-D3 is available in NWChem are BLYP, B3LYP, BP86, Becke97-D,
    PBE96, TPSS, PBE0, B2PLYP, BHLYP, TPSSH, PWB6K, B1B95, SSB-D,
    MPW1B95, MPWB1K, M05, M05-2X, M06L, M06, M06-2X, and M06HF</p>
</li>
<li>
<p><code>vdw 4</code> triggers the DFT-D3BJ dispersion model. Currently only BLYP,
    B3LYP, BHLYP, TPSS, TPSSh,  B2-PLYP, B97-D, BP86, PBE96, PW6B95, revPBE, B3PW91, pwb6k, b1b95, CAM-B3LYP, LC-wPBE,  HCTH120, MPW1B95, BOP, OLYP,  BPBE, OPBE and SSB  are supported.</p>
</li>
</ul>
<p>This capability is also supported for energy gradients and Hessian. Is
possible to be deactivated with OFF.</p>
<h2 id="density-functional-theory-for-molecules-noscf-non-self-consistent-calculations">NOSCF: Non Self-Consistent Calculations<a class="headerlink" href="#density-functional-theory-for-molecules-noscf-non-self-consistent-calculations" title="Permanent link">&para;</a></h2>
<p>The <code>noscf</code> keyword can be used to to calculate the non self-consistent energy
for a set of input vectors. For example, the following input shows how a
non self-consistent B3LYP energy can be calculated using a
self-consistent set of vectors calculated at the Hartree-Fock level.</p>
<pre><code>start h2o-noscf

geometry units angstrom
  O      0.00000000     0.00000000     0.11726921
  H      0.75698224     0.00000000    -0.46907685
  H     -0.75698224     0.00000000    -0.46907685
end

basis spherical
  * library aug-cc-pvdz
end
dft
  xc hfexch
  vectors output hf.movecs 
end
task dft energy
dft
  xc b3lyp
  vectors input hf.movecs 
  noscf 
end
task dft energy
</code></pre>
<h2 id="density-functional-theory-for-molecules-xdm-exchange-hole-dipole-moment-dispersion-model">XDM: Exchange-hole dipole moment dispersion model<a class="headerlink" href="#density-functional-theory-for-molecules-xdm-exchange-hole-dipole-moment-dispersion-model" title="Permanent link">&para;</a></h2>
<pre><code>XDM  [ a1 &lt;real a1&gt; ]   [ a2 &lt;real a2&gt; ]
</code></pre>
<p>See details (including list of a1 and a2 parameters) in  paper <sup id="fnref_footnote_112"><a class="footnote-ref" href="#density-functional-theory-for-molecules-fn_footnote_112">112</a></sup>
and the website <a href="https://erin-r-johnson.github.io/software/">https://erin-r-johnson.github.io/software/</a></p>
<pre><code>    geometry 
      O         -0.190010095135     -1.168397415155      0.925531922479
      H         -0.124425719598     -0.832776238160      1.818190662986
      H         -0.063897685990     -0.392575837594      0.364048725248
      O          0.174717244879      1.084630474836     -0.860510672419
      H         -0.566281023931      1.301941006866     -1.427261487135
      H          0.935093179777      1.047335209207     -1.441842151158
    end

    basis spherical
      * library aug-cc-pvdz
    end

    dft
       direct
       xc b3lyp
       xdm a1 0.6224 a2 1.7068
    end

    task dft optimize
</code></pre>
<h2 id="density-functional-theory-for-molecules-print-control">Print Control<a class="headerlink" href="#density-functional-theory-for-molecules-print-control" title="Permanent link">&para;</a></h2>
<pre><code> PRINT||NOPRINT
</code></pre>
<p>The <code>PRINT||NOPRINT</code> options control the level of output in the DFT.
Please see some examples using this directive in <a href="#density-functional-theory-for-molecules-sample-input-file">Sample input
file</a>. Known controllable print options
are:</p>
<table>
<thead>
<tr>
<th>Name</th>
<th>Print Level</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&ldquo;all vector symmetries&rdquo;</td>
<td>high</td>
<td>symmetries of all molecular orbitals</td>
</tr>
<tr>
<td>&ldquo;alpha partner info&rdquo;</td>
<td>high</td>
<td>unpaired alpha orbital analysis</td>
</tr>
<tr>
<td>&ldquo;common&rdquo;</td>
<td>debug</td>
<td>dump of common blocks</td>
</tr>
<tr>
<td>&ldquo;convergence&rdquo;</td>
<td>default</td>
<td>convergence of SCF procedure</td>
</tr>
<tr>
<td>&ldquo;coulomb fit&rdquo;</td>
<td>high</td>
<td>fitting electronic charge density</td>
</tr>
<tr>
<td>&ldquo;dft timings&rdquo;</td>
<td>high</td>
<td></td>
</tr>
<tr>
<td>&ldquo;final vectors&rdquo;</td>
<td>high</td>
<td></td>
</tr>
<tr>
<td>&ldquo;final vectors analysis&rdquo;</td>
<td>high</td>
<td>print all orbital energies and orbitals</td>
</tr>
<tr>
<td>&ldquo;final vector symmetries&rdquo;</td>
<td>default</td>
<td>symmetries of final molecular orbitals</td>
</tr>
<tr>
<td>&ldquo;information&rdquo;</td>
<td>low</td>
<td>general information</td>
</tr>
<tr>
<td>&ldquo;initial vectors&rdquo;</td>
<td>high</td>
<td></td>
</tr>
<tr>
<td>&ldquo;intermediate energy info&rdquo;</td>
<td>high</td>
<td></td>
</tr>
<tr>
<td>&ldquo;intermediate evals&rdquo;</td>
<td>high</td>
<td>intermediate orbital energies</td>
</tr>
<tr>
<td>&ldquo;intermediate fock matrix&rdquo;</td>
<td>high</td>
<td></td>
</tr>
<tr>
<td>&ldquo;intermediate overlap&rdquo;</td>
<td>high</td>
<td>overlaps between the alpha and beta sets</td>
</tr>
<tr>
<td>&ldquo;intermediate S2&rdquo;</td>
<td>high</td>
<td>values of S2</td>
</tr>
<tr>
<td>&ldquo;intermediate vectors&rdquo;</td>
<td>high</td>
<td>intermediate molecular orbitals</td>
</tr>
<tr>
<td>&ldquo;interm vector symm&rdquo;</td>
<td>high</td>
<td>symmetries of intermediate orbitals</td>
</tr>
<tr>
<td>&ldquo;io info&rdquo;</td>
<td>debug</td>
<td>reading from and writing to disk</td>
</tr>
<tr>
<td>&ldquo;kinetic_energy&rdquo;</td>
<td>high</td>
<td>kinetic energy</td>
</tr>
<tr>
<td>&ldquo;mulliken ao&rdquo;</td>
<td>high</td>
<td>mulliken atomic orbital population</td>
</tr>
<tr>
<td>&ldquo;multipole&rdquo;</td>
<td>default</td>
<td>moments of alpha, beta, and nuclear charge densities</td>
</tr>
<tr>
<td>&ldquo;parameters&rdquo;</td>
<td>default</td>
<td>input parameters</td>
</tr>
<tr>
<td>&ldquo;quadrature&rdquo;</td>
<td>high</td>
<td>numerical quadrature</td>
</tr>
<tr>
<td>&ldquo;schwarz&rdquo;</td>
<td>high</td>
<td>integral screening info &amp; stats at completion</td>
</tr>
<tr>
<td>&ldquo;screening parameters&rdquo;</td>
<td>high</td>
<td>integral accuracies</td>
</tr>
<tr>
<td>&ldquo;semi-direct info&rdquo;</td>
<td>default</td>
<td>semi direct algorithm</td>
</tr>
</tbody>
</table>
<p>DFT Print Control Specifications</p>
<h2 id="density-functional-theory-for-molecules-spin-orbit-density-functional-theory-sodft">Spin-Orbit Density Functional Theory (SODFT)<a class="headerlink" href="#density-functional-theory-for-molecules-spin-orbit-density-functional-theory-sodft" title="Permanent link">&para;</a></h2>
<p>The spin-orbit DFT module (SODFT) in the NWChem code allows for the variational
treatment of the one-electron spin-orbit operator within the DFT framework. Calculations
can be performed either with an electron relativistic approach (<a href="#relativistic-all-electron-approximations-zeroth-order-regular-approximation-zora">ZORA</a>) or with an
effective core potential (ECP) and a matching spin-orbit potential (SO). The current
implementation does NOT use symmetry.</p>
<p>The actual SODFT calculation will be performed when the input module
encounters the TASK directive (<a href="#task">TASK</a>).</p>
<pre><code> TASK SODFT
</code></pre>
<p>Input parameters are the same as for the DFT. Some of the DFT options
are not available in the SODFT. These are <code>max_ovl</code> and <code>sic</code>.</p>
<p>Besides using the standard ECP and basis sets, see <a href="#ecp">Effective Core
Potentials</a> for details, one also has to specify a
spin-orbit (SO) potential. The input specification for the SO potential
can be found in <a href="#ecp-spin-orbit-ecps">Effective Core
Potentials</a>. At this time we have not
included any spin-orbit potentials in the basis set library. However,
one can get these from the Stuttgart/Kln web pages
<a href="http://www.tc.uni-koeln.de/PP/clickpse.en.html">http://www.tc.uni-koeln.de/PP/clickpse.en.html</a>.</p>
<p>Note: One should use a combination of ECP and SO potentials that were
designed for the same size core, i.e., don&rsquo;t use a small core ECP
potential with a large core SO potential (it will produce erroneous
results).</p>
<p>The following is an example of a calculation of UO<sub>2</sub>:</p>
<pre><code>start uo2_sodft  
echo  

charge 2  
geometry 
 U     0.00000      0.00000     0.00000  
 O     0.00000      0.00000     1.68000  
 O     0.00000      0.00000    -1.68000  
end  
basis &quot;ao basis&quot;  
 * library &quot;stuttgart rlc ecp&quot;
END
ECP
 * library &quot;stuttgart rlc ecp&quot;
END
SO  
 U p  
 2    3.986181      1.816350  
 2    2.000160     11.543940  
 2    0.960841      0.794644  
 U d  
 2    4.147972      0.353683  
 2    2.234563      3.499282  
 2    0.913695      0.514635  
 U f  
 2    3.998938      4.744214  
 2    1.998840     -5.211731  
 2    0.995641      1.867860  
END  
dft  
 mult 1  
 xc hfexch  
end  
task sodft
</code></pre>
<h2 id="density-functional-theory-for-molecules-sym-and-adapt">SYM and ADAPT<a class="headerlink" href="#density-functional-theory-for-molecules-sym-and-adapt" title="Permanent link">&para;</a></h2>
<p>The options <code>SYM</code> and <code>ADAPT</code> works the same way as the analogous options for the SCF code.
Therefore please use the following links for <a href="#hartree-fock-theory-for-molecules-sym-use-of-symmetry">SYM</a> and
<a href="#hartree-fock-theory-for-molecules-adapt-symmetry-adaptation-of-mos">ADAPT</a>, respectively.</p>
<h2 id="density-functional-theory-for-molecules-references">References<a class="headerlink" href="#density-functional-theory-for-molecules-references" title="Permanent link">&para;</a></h2>
<div class="footnote">
<hr />
<ol>
<li id="density-functional-theory-for-molecules-fn_footnote_1">
<p>Dunlap, B. I.; Connolly, J. W. D.; Sabin, J. R. On First-Row Diatomic Molecules and Local Density Models. <em>The Journal of Chemical Physics</em> <strong>1979</strong>, <em>71</em> (12), 4993. <a href="https://doi.org/10.1063/1.438313">https://doi.org/10.1063/1.438313</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_1" title="Jump back to footnote 1 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_2">
<p>Eichkorn, K.; Treutler, O.; hm, H.; Hser, M.; Ahlrichs, R. Auxiliary Basis Sets to Approximate Coulomb Potentials. <em>Chemical Physics Letters</em> <strong>1995</strong>, <em>240</em> (4), 283290. <a href="https://doi.org/10.1016/0009-2614(95)00621-a">https://doi.org/10.1016/0009-2614(95)00621-a</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_2" title="Jump back to footnote 2 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_3">
<p>Kster, A. M.; Reveles, J. U.; Campo, J. M. del. Calculation of Exchange-Correlation Potentials with Auxiliary Function Densities. <em>The Journal of Chemical Physics</em> <strong>2004</strong>, <em>121</em> (8), 34173424. <a href="https://doi.org/10.1063/1.1771638">https://doi.org/10.1063/1.1771638</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_3" title="Jump back to footnote 3 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_4">
<p>Slater, J. C.; Johnson, K. H. Self-Consistent-Field <em>X</em><em></em> Cluster Method for Polyatomic Molecules and Solids. <em>Phys. Rev. B</em> <strong>1972</strong>, <em>5</em>, 844853. <a href="https://doi.org/10.1103/PhysRevB.5.844">https://doi.org/10.1103/PhysRevB.5.844</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_4" title="Jump back to footnote 4 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_4" title="Jump back to footnote 4 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_5">
<p>Slater, J. C. <em>Quantum Theory of Molecules and Solids: The Self-Consistent Field for Molecules and Solids</em>; McGraw-Hill Education, 1974; p 640.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_5" title="Jump back to footnote 5 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_5" title="Jump back to footnote 5 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_6">
<p>Vosko, S. H.; Wilk, L.; Nusair, M. Accurate Spin-Dependent Electron Liquid Correlation Energies for Local Spin Density Calculations: A Critical Analysis. <em>Canadian Journal of physics</em> <strong>1980</strong>, <em>58</em> (8), 12001211. <a href="https://doi.org/10.1139/p80-159">https://doi.org/10.1139/p80-159</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_6" title="Jump back to footnote 6 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_6" title="Jump back to footnote 6 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref3_footnote_6" title="Jump back to footnote 6 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref4_footnote_6" title="Jump back to footnote 6 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref5_footnote_6" title="Jump back to footnote 6 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref6_footnote_6" title="Jump back to footnote 6 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref7_footnote_6" title="Jump back to footnote 6 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_7">
<p>Becke, A. D. A New Mixing of Hartree-Fock and Local Density-Functional Theories. <em>The Journal of Chemical Physics</em> <strong>1993</strong>, <em>98</em> (2), 13721377. https://doi.org/<a href="https://aip.scitation.org/doi/abs/10.1063/1.464304">https://aip.scitation.org/doi/abs/10.1063/1.464304</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_7" title="Jump back to footnote 7 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_7" title="Jump back to footnote 7 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_8">
<p>Becke, A. D. Density-Functional Thermochemistry. III. The Role of Exact Exchange. <em>The Journal of Chemical Physics</em> <strong>1993</strong>, <em>98</em> (7), 56485652. <a href="https://doi.org/10.1063/1.464913">https://doi.org/10.1063/1.464913</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_8" title="Jump back to footnote 8 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_8" title="Jump back to footnote 8 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref3_footnote_8" title="Jump back to footnote 8 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_9">
<p>Becke, A. D. Density-Functional Thermochemistry. V. Systematic Optimization of Exchange-Correlation Functionals. <em>The Journal of Chemical Physics</em> <strong>1997</strong>, <em>107</em> (20), 85548560. <a href="https://doi.org/10.1063/1.475007">https://doi.org/10.1063/1.475007</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_9" title="Jump back to footnote 9 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_9" title="Jump back to footnote 9 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_10">
<p>Perdew, J. P.; Zunger, A. Self-Interaction Correction to Density-Functional Approximations for Many-Electron Systems. <em>Phys. Rev. B</em> <strong>1981</strong>, <em>23</em>, 50485079. <a href="https://doi.org/10.1103/PhysRevB.23.5048">https://doi.org/10.1103/PhysRevB.23.5048</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_10" title="Jump back to footnote 10 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_10" title="Jump back to footnote 10 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_11">
<p>Perdew, J. P.; Wang, Y. Accurate and Simple Analytic Representation of the Electron-Gas Correlation Energy. <em>Phys. Rev. B</em> <strong>1992</strong>, <em>45</em>, 1324413249. <a href="https://doi.org/10.1103/PhysRevB.45.13244">https://doi.org/10.1103/PhysRevB.45.13244</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_11" title="Jump back to footnote 11 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_11" title="Jump back to footnote 11 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_12">
<p>Becke, A. D. On the Large-Gradient Behavior of the Density Functional Exchange Energy. <em>The Journal of Chemical Physics</em> <strong>1986</strong>, <em>85</em> (12), 71847187. <a href="https://doi.org/10.1063/1.451353">https://doi.org/10.1063/1.451353</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_12" title="Jump back to footnote 12 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_13">
<p>Becke, A. D. Density-Functional Exchange-Energy Approximation with Correct Asymptotic Behavior. <em>Physical Review A</em> <strong>1988</strong>, <em>38</em> (6), 30983100. <a href="https://doi.org/10.1103/PhysRevA.38.3098">https://doi.org/10.1103/PhysRevA.38.3098</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_13" title="Jump back to footnote 13 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_14">
<p>Perdew, J. P. Density-Functional Approximation for the Correlation Energy of the Inhomogeneous Electron Gas. <em>Physical Review B</em> <strong>1986</strong>, <em>33</em> (12), 88228824. <a href="https://doi.org/10.1103/PhysRevB.33.8822">https://doi.org/10.1103/PhysRevB.33.8822</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_14" title="Jump back to footnote 14 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_14" title="Jump back to footnote 14 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_15">
<p>Perdew, J. P.; Burke, K.; Ernzerhof, M. Errata: Generalized Gradient Approximation Made Simple. <em>Phys. Rev. Lett.</em> <strong>1996</strong>, <em>77</em>, 38653868. <a href="https://doi.org/10.1103/PhysRevLett.77.3865">https://doi.org/10.1103/PhysRevLett.77.3865</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_15" title="Jump back to footnote 15 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_15" title="Jump back to footnote 15 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_16">
<p>Perdew, J. P.; Burke, K.; Ernzerhof, M. Generalized Gradient Approximation Made Simple. <em>Phys. Rev. Lett.</em> <strong>1997</strong>, <em>78</em>, 13961396. <a href="https://doi.org/10.1103/PhysRevLett.78.1396">https://doi.org/10.1103/PhysRevLett.78.1396</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_16" title="Jump back to footnote 16 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_16" title="Jump back to footnote 16 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_17">
<p>Gill, P. M. W. A New Gradient-Corrected Exchange Functional. <em>Molecular Physics</em> <strong>1996</strong>, <em>89</em> (2), 433445. <a href="https://doi.org/10.1080/002689796173813">https://doi.org/10.1080/002689796173813</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_17" title="Jump back to footnote 17 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_18">
<p>Handy, N. C.; Cohen, A. J. Left-Right Correlation Energy. <em>Molecular Physics</em> <strong>2001</strong>, <em>99</em> (5), 403412. <a href="https://doi.org/10.1080/00268970010018431">https://doi.org/10.1080/00268970010018431</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_18" title="Jump back to footnote 18 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_19">
<p>Adamo, C.; Barone, V. Exchange Functionals with Improved Long-Range Behavior and Adiabatic Connection Methods Without Adjustable Parameters: The mPW and mPW1PW Models. <em>The Journal of Chemical Physics</em> <strong>1998</strong>, <em>108</em> (2), 664675. <a href="https://doi.org/10.1063/1.475428">https://doi.org/10.1063/1.475428</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_19" title="Jump back to footnote 19 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_20">
<p>Zhao, Y.; Truhlar, D. G. Design of Density Functionals That Are Broadly Accurate for Thermochemistry, Thermochemical Kinetics, and Nonbonded Interactions. <em>The Journal of Physical Chemistry A</em> <strong>2005</strong>, <em>109</em> (25), 56565667. <a href="https://doi.org/10.1021/jp050536c">https://doi.org/10.1021/jp050536c</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_20" title="Jump back to footnote 20 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_21">
<p>Filatov, M.; Thiel, W. A New Gradient-Corrected Exchange-Correlation Density Functional. <em>Molecular Physics</em> <strong>1997</strong>, <em>91</em> (5), 847860. <a href="https://doi.org/10.1080/002689797170950">https://doi.org/10.1080/002689797170950</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_21" title="Jump back to footnote 21 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_21" title="Jump back to footnote 21 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref3_footnote_21" title="Jump back to footnote 21 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_22">
<p>Filatov, M.; Thiel, W. A Nonlocal Correlation Energy Density Functional from a Coulomb Hole Model. <em>International Journal of Quantum Chemistry</em> <strong>1997</strong>, <em>62</em> (6), 603616. <a href="https://doi.org/10.1002/(SICI)1097-461X(1997)62:6&lt;603::AID-QUA4&gt;3.0.CO;2-\%23">https://doi.org/10.1002/(SICI)1097-461X(1997)62:6&lt;603::AID-QUA4&gt;3.0.CO;2-\23</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_22" title="Jump back to footnote 22 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_22" title="Jump back to footnote 22 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref3_footnote_22" title="Jump back to footnote 22 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_23">
<p>Hammer, B.; Hansen, L. B.; Nrskov, J. K. Improved Adsorption Energetics Within Density-Functional Theory Using Revised Perdew-Burke-Ernzerhof Functionals. <em>Physical Review B</em> <strong>1999</strong>, <em>59</em> (11), 74137421. <a href="https://doi.org/10.1103/physrevb.59.7413">https://doi.org/10.1103/physrevb.59.7413</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_23" title="Jump back to footnote 23 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_23" title="Jump back to footnote 23 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_24">
<p>Zhang, Y.; Yang, W. Comment on Generalized Gradient Approximation Made Simple. <em>Physical Review Letters</em> <strong>1998</strong>, <em>80</em> (4), 890890. <a href="https://doi.org/10.1103/physrevlett.80.890">https://doi.org/10.1103/physrevlett.80.890</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_24" title="Jump back to footnote 24 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_24" title="Jump back to footnote 24 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_25">
<p>Zhao, Y.; Lynch, B. J.; Truhlar, D. G. Development and Assessment of a New Hybrid Density Functional Model for Thermochemical Kinetics. <em>The Journal of Physical Chemistry A</em> <strong>2004</strong>, <em>108</em> (14), 27152719. <a href="https://doi.org/10.1021/jp049908s">https://doi.org/10.1021/jp049908s</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_25" title="Jump back to footnote 25 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_25" title="Jump back to footnote 25 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref3_footnote_25" title="Jump back to footnote 25 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref4_footnote_25" title="Jump back to footnote 25 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref5_footnote_25" title="Jump back to footnote 25 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref6_footnote_25" title="Jump back to footnote 25 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_26">
<p>Lee, C.; Yang, W.; Parr, R. G. Development of the Colle-Salvetti Correlation-Energy Formula into a Functional of the Electron Density. <em>Physical Review B</em> <strong>1988</strong>, <em>37</em> (2), 785789. <a href="https://doi.org/10.1103/PhysRevB.37.785">https://doi.org/10.1103/PhysRevB.37.785</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_26" title="Jump back to footnote 26 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_27">
<p>Perdew, J. P.; Chevary, J. A.; Vosko, S. H.; Jackson, K. A.; Pederson, M. R.; Singh, D. J.; Fiolhais, C. Atoms, Molecules, Solids, and Surfaces: Applications of the Generalized Gradient Approximation for Exchange and Correlation. <em>Physical Review B</em> <strong>1992</strong>, <em>46</em> (11), 66716687. <a href="https://doi.org/10.1103/PhysRevB.46.6671">https://doi.org/10.1103/PhysRevB.46.6671</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_27" title="Jump back to footnote 27 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_28">
<p>Perdew, J. P.; Chevary, J. A.; Vosko, S. H.; Jackson, K. A.; Pederson, M. R.; Singh, D. J.; Fiolhais, C. Erratum: Atoms, Molecules, Solids, and Surfaces: Applications of the Generalized Gradient Approximation for Exchange and Correlation. <em>Physical Review B</em> <strong>1993</strong>, <em>48</em> (7), 49784978. <a href="https://doi.org/10.1103/PhysRevB.48.4978.2">https://doi.org/10.1103/PhysRevB.48.4978.2</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_28" title="Jump back to footnote 28 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_29">
<p>Tsuneda, T.; Suzumura, T.; Hirao, K. A New One-Parameter Progressive Collesalvetti-Type Correlation Functional. <em>The Journal of Chemical Physics</em> <strong>1999</strong>, <em>110</em> (22), 1066410678. <a href="https://doi.org/10.1063/1.479012">https://doi.org/10.1063/1.479012</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_29" title="Jump back to footnote 29 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_29" title="Jump back to footnote 29 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_30">
<p>Hamprecht, F. A.; Cohen, A. J.; Tozer, D. J.; Handy, N. C. Development and Assessment of New Exchange-Correlation Functionals. <em>The Journal of Chemical Physics</em> <strong>1998</strong>, <em>109</em> (15), 62646271. <a href="https://doi.org/10.1063/1.477267">https://doi.org/10.1063/1.477267</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_30" title="Jump back to footnote 30 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_30" title="Jump back to footnote 30 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_31">
<p>Boese, A. D.; Doltsinis, N. L.; Handy, N. C.; Sprik, M. New Generalized Gradient Approximation Functionals. <em>The Journal of Chemical Physics</em> <strong>2000</strong>, <em>112</em> (4), 16701678. <a href="https://doi.org/10.1063/1.480732">https://doi.org/10.1063/1.480732</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_31" title="Jump back to footnote 31 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_31" title="Jump back to footnote 31 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_32">
<p>Boese, A. D.; Martin, J. M. L.; Handy, N. C. The Role of the Basis Set: Assessing Density Functional Theory. <em>The Journal of Chemical Physics</em> <strong>2003</strong>, <em>119</em> (6), 30053014. <a href="https://doi.org/10.1063/1.1589004">https://doi.org/10.1063/1.1589004</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_32" title="Jump back to footnote 32 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_33">
<p>Boese, A. D.; Handy, N. C. A New Parametrization of Exchangecorrelation Generalized Gradient Approximation Functionals. <em>The Journal of Chemical Physics</em> <strong>2001</strong>, <em>114</em> (13), 54975503. <a href="https://doi.org/10.1063/1.1347371">https://doi.org/10.1063/1.1347371</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_33" title="Jump back to footnote 33 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_34">
<p>Cohen, A. J.; Handy, N. C. Assessment of Exchange Correlation Functionals. <em>Chemical Physics Letters</em> <strong>2000</strong>, <em>316</em> (1-2), 160166. <a href="https://doi.org/10.1016/s0009-2614(99)01273-7">https://doi.org/10.1016/s0009-2614(99)01273-7</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_34" title="Jump back to footnote 34 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_35">
<p>Menconi, G.; Wilson, P. J.; Tozer, D. J. Emphasizing the Exchange-Correlation Potential in Functional Development. <em>The Journal of Chemical Physics</em> <strong>2001</strong>, <em>114</em> (9), 39583967. <a href="https://doi.org/10.1063/1.1342776">https://doi.org/10.1063/1.1342776</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_35" title="Jump back to footnote 35 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_36">
<p>Boese, A. D.; Chandra, A.; Martin, J. M. L.; Marx, D. From Ab Initio Quantum Chemistry to Molecular Dynamics: The Delicate Case of Hydrogen Bonding in Ammonia. <em>The Journal of Chemical Physics</em> <strong>2003</strong>, <em>119</em> (12), 59655980. <a href="https://doi.org/10.1063/1.1599338">https://doi.org/10.1063/1.1599338</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_36" title="Jump back to footnote 36 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_37">
<p>Tsuneda, T.; Suzumura, T.; Hirao, K. A Reexamination of Exchange Energy Functionals. <em>The Journal of Chemical Physics</em> <strong>1999</strong>, <em>111</em> (13), 56565667. <a href="https://doi.org/10.1063/1.479954">https://doi.org/10.1063/1.479954</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_37" title="Jump back to footnote 37 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_38">
<p>Perdew, J. P.; Kurth, S.; Zupan, A.; Blaha, P. Accurate Density Functional with Correct Formal Properties: A Step Beyond the Generalized Gradient Approximation. <em>Physical Review Letters</em> <strong>1999</strong>, <em>82</em> (12), 25442547. <a href="https://doi.org/10.1103/physrevlett.82.2544">https://doi.org/10.1103/physrevlett.82.2544</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_38" title="Jump back to footnote 38 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_38" title="Jump back to footnote 38 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_39">
<p>Tao, J.; Perdew, J. P.; Staroverov, V. N.; Scuseria, G. E. Climbing the Density Functional Ladder: Nonempirical Meta-Generalized Gradient Approximation Designed for Molecules and Solids. <em>Physical Review Letters</em> <strong>2003</strong>, <em>91</em> (14), 146401. <a href="https://doi.org/10.1103/physrevlett.91.146401">https://doi.org/10.1103/physrevlett.91.146401</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_39" title="Jump back to footnote 39 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_39" title="Jump back to footnote 39 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_40">
<p>Zhao, Y.; Truhlar, D. G. Hybrid Meta Density Functional Theory Methods for Thermochemistry, Thermochemical Kinetics, and Noncovalent Interactions: The MPW1B95 and MPWB1K Models and Comparative Assessments for Hydrogen Bonding and van Der Waals Interactions. <em>The Journal of Physical Chemistry A</em> <strong>2004</strong>, <em>108</em> (33), 69086918. <a href="https://doi.org/10.1021/jp048147q">https://doi.org/10.1021/jp048147q</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_40" title="Jump back to footnote 40 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_40" title="Jump back to footnote 40 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref3_footnote_40" title="Jump back to footnote 40 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_41">
<p>Zhao, Y.; Schultz, N. E.; Truhlar, D. G. Exchange-Correlation Functional with Broad Accuracy for Metallic and Nonmetallic Compounds, Kinetics, and Noncovalent Interactions. <em>The Journal of Chemical Physics</em> <strong>2005</strong>, <em>123</em> (16), 161103. <a href="https://doi.org/10.1063/1.2126975">https://doi.org/10.1063/1.2126975</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_41" title="Jump back to footnote 41 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_41" title="Jump back to footnote 41 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_42">
<p>Zhao, Y.; Schultz, N. E.; Truhlar, D. G. Design of Density Functionals by Combining the Method of Constraint Satisfaction with Parametrization for Thermochemistry, Thermochemical Kinetics, and Noncovalent Interactions. <em>Journal of Chemical Theory and Computation</em> <strong>2006</strong>, <em>2</em> (2), 364382. <a href="https://doi.org/10.1021/ct0502763">https://doi.org/10.1021/ct0502763</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_42" title="Jump back to footnote 42 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_42" title="Jump back to footnote 42 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref3_footnote_42" title="Jump back to footnote 42 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_43">
<p>Staroverov, V. N.; Scuseria, G. E.; Tao, J.; Perdew, J. P. Comparative Assessment of a New Nonempirical Density Functional: Molecules and Hydrogen-Bonded Complexes. <em>The Journal of Chemical Physics</em> <strong>2003</strong>, <em>119</em> (23), 1212912137. <a href="https://doi.org/10.1063/1.1626543">https://doi.org/10.1063/1.1626543</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_43" title="Jump back to footnote 43 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_44">
<p>Becke, A. D. Density-Functional Thermochemistry. IV. A New Dynamical Correlation Functional and Implications for Exact-Exchange Mixing. <em>The Journal of Chemical Physics</em> <strong>1996</strong>, <em>104</em> (3), 10401046. <a href="https://doi.org/10.1063/1.470829">https://doi.org/10.1063/1.470829</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_44" title="Jump back to footnote 44 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_44" title="Jump back to footnote 44 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_45">
<p>Voorhis, T. V.; Scuseria, G. E. A Novel Form for the Exchange-Correlation Energy Functional. <em>The Journal of Chemical Physics</em> <strong>1998</strong>, <em>109</em> (2), 400410. <a href="https://doi.org/10.1063/1.476577">https://doi.org/10.1063/1.476577</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_45" title="Jump back to footnote 45 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_45" title="Jump back to footnote 45 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref3_footnote_45" title="Jump back to footnote 45 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_46">
<p>Zhao, Y.; Truhlar, D. G. A New Local Density Functional for Main-Group Thermochemistry, Transition Metal Bonding, Thermochemical Kinetics, and Noncovalent Interactions. <em>The Journal of Chemical Physics</em> <strong>2006</strong>, <em>125</em> (19), 194101. <a href="https://doi.org/10.1063/1.2370993">https://doi.org/10.1063/1.2370993</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_46" title="Jump back to footnote 46 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_46" title="Jump back to footnote 46 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref3_footnote_46" title="Jump back to footnote 46 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref4_footnote_46" title="Jump back to footnote 46 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref5_footnote_46" title="Jump back to footnote 46 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref6_footnote_46" title="Jump back to footnote 46 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_47">
<p>Zhao, Y.; Truhlar, D. G. Density Functional for Spectroscopy: No Long-Range Self-Interaction Error, Good Performance for Rydberg and Charge-Transfer States, and Better Performance on Average Than B3LYP for Ground States. <em>The Journal of Physical Chemistry A</em> <strong>2006</strong>, <em>110</em> (49), 1312613130. <a href="https://doi.org/10.1021/jp066479k">https://doi.org/10.1021/jp066479k</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_47" title="Jump back to footnote 47 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_47" title="Jump back to footnote 47 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref3_footnote_47" title="Jump back to footnote 47 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_48">
<p>Zhao, Y.; Truhlar, D. G. The M06 Suite of Density Functionals for Main Group Thermochemistry, Thermochemical Kinetics, Noncovalent Interactions, Excited States, and Transition Elements: Two New Functionals and Systematic Testing of Four M06-Class Functionals and 12 Other Functionals. <em>Theoretical Chemistry Accounts</em> <strong>2008</strong>, <em>120</em> (1-3), 215241. <a href="https://doi.org/10.1007/s00214-007-0310-x">https://doi.org/10.1007/s00214-007-0310-x</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_48" title="Jump back to footnote 48 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_48" title="Jump back to footnote 48 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref3_footnote_48" title="Jump back to footnote 48 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_49">
<p>Zhao, Y.; Truhlar, D. G. Construction of a Generalized Gradient Approximation by Restoring the Density-Gradient Expansion and Enforcing a Tight Lieboxford Bound. <em>The Journal of Chemical Physics</em> <strong>2008</strong>, <em>128</em> (18), 184109. <a href="https://doi.org/10.1063/1.2912068">https://doi.org/10.1063/1.2912068</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_49" title="Jump back to footnote 49 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_49" title="Jump back to footnote 49 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref3_footnote_49" title="Jump back to footnote 49 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref4_footnote_49" title="Jump back to footnote 49 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref5_footnote_49" title="Jump back to footnote 49 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref6_footnote_49" title="Jump back to footnote 49 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref7_footnote_49" title="Jump back to footnote 49 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref8_footnote_49" title="Jump back to footnote 49 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref9_footnote_49" title="Jump back to footnote 49 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_50">
<p>Peverati, R.; Truhlar, D. G. Improving the Accuracy of Hybrid Meta-GGA Density Functionals by Range Separation. <em>The Journal of Physical Chemistry Letters</em> <strong>2011</strong>, <em>2</em> (21), 28102817. <a href="https://doi.org/10.1021/jz201170d">https://doi.org/10.1021/jz201170d</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_50" title="Jump back to footnote 50 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_50" title="Jump back to footnote 50 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref3_footnote_50" title="Jump back to footnote 50 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_51">
<p>Peverati, R.; Truhlar, D. G. M11-l: A Local Density Functional That Provides Improved Accuracy for Electronic Structure Calculations in Chemistry and Physics. <em>The Journal of Physical Chemistry Letters</em> <strong>2012</strong>, <em>3</em> (1), 117124. <a href="https://doi.org/10.1021/jz201525m">https://doi.org/10.1021/jz201525m</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_51" title="Jump back to footnote 51 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_51" title="Jump back to footnote 51 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref3_footnote_51" title="Jump back to footnote 51 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_52">
<p>Peverati, R.; Zhao, Y.; Truhlar, D. G. Generalized Gradient Approximation That Recovers the Second-Order Density-Gradient Expansion with Optimized Across-the-Board Performance. <em>The Journal of Physical Chemistry Letters</em> <strong>2011</strong>, <em>2</em> (16), 19911997. <a href="https://doi.org/10.1021/jz200616w">https://doi.org/10.1021/jz200616w</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_52" title="Jump back to footnote 52 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_52" title="Jump back to footnote 52 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref3_footnote_52" title="Jump back to footnote 52 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_53">
<p>Peverati, R.; Truhlar, D. G. Communication: A Global Hybrid Generalized Gradient Approximation to the Exchange-Correlation Functional That Satisfies the Second-Order Density-Gradient Constraint and Has Broad Applicability in Chemistry. <em>The Journal of Chemical Physics</em> <strong>2011</strong>, <em>135</em> (19), 191102. <a href="https://doi.org/10.1063/1.3663871">https://doi.org/10.1063/1.3663871</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_53" title="Jump back to footnote 53 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_53" title="Jump back to footnote 53 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref3_footnote_53" title="Jump back to footnote 53 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_54">
<p>Pernal, K.; Podeszwa, R.; Patkowski, K.; Szalewicz, K. Dispersionless Density Functional Theory. <em>Physical Review Letters</em> <strong>2009</strong>, <em>103</em> (26), 263201. <a href="https://doi.org/10.1103/physrevlett.103.263201">https://doi.org/10.1103/physrevlett.103.263201</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_54" title="Jump back to footnote 54 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_55">
<p>Wilson, P. J.; Bradley, T. J.; Tozer, D. J. Hybrid Exchange-Correlation Functional Determined from Thermochemical Data and Ab Initio Potentials. <em>The Journal of Chemical Physics</em> <strong>2001</strong>, <em>115</em> (20), 92339242. <a href="https://doi.org/10.1063/1.1412605">https://doi.org/10.1063/1.1412605</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_55" title="Jump back to footnote 55 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_56">
<p>Keal, T. W.; Tozer, D. J. Semiempirical Hybrid Functional with Improved Performance in an Extensive Chemical Assessment. <em>The Journal of Chemical Physics</em> <strong>2005</strong>, <em>123</em> (12), 121103. <a href="https://doi.org/10.1063/1.2061227">https://doi.org/10.1063/1.2061227</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_56" title="Jump back to footnote 56 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_57">
<p>Grimme, S. Semiempirical GGA-Type Density Functional Constructed with a Long-Range Dispersion Correction. <em>Journal of Computational Chemistry</em> <strong>2006</strong>, <em>27</em> (15), 17871799. <a href="https://doi.org/10.1002/jcc.20495">https://doi.org/10.1002/jcc.20495</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_57" title="Jump back to footnote 57 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_58">
<p>Schmider, H. L.; Becke, A. D. Optimized Density Functionals from the Extended G2 Test Set. <em>The Journal of Chemical Physics</em> <strong>1998</strong>, <em>108</em> (23), 96249631. <a href="https://doi.org/10.1063/1.476438">https://doi.org/10.1063/1.476438</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_58" title="Jump back to footnote 58 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_59">
<p>Adamo, C.; Barone, V. Toward Reliable Density Functional Methods Without Adjustable Parameters: The PBE0 Model. <em>The Journal of Chemical Physics</em> <strong>1999</strong>, <em>110</em> (13), 61586170. <a href="https://doi.org/10.1063/1.478522">https://doi.org/10.1063/1.478522</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_59" title="Jump back to footnote 59 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_60">
<p>Lynch, B. J.; Fast, P. L.; Harris, M.; Truhlar, D. G. Adiabatic Connection for Kinetics. <em>The Journal of Physical Chemistry A</em> <strong>2000</strong>, <em>104</em> (21), 48114815. <a href="https://doi.org/10.1021/jp000497z">https://doi.org/10.1021/jp000497z</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_60" title="Jump back to footnote 60 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_61">
<p>Sun, J.; Perdew, J. P.; Ruzsinszky, A. Semilocal Density Functional Obeying a Strongly Tightened Bound for Exchange. <em>Proceedings of the National Academy of Sciences</em> <strong>2015</strong>, <em>112</em> (3), 685689. <a href="https://doi.org/10.1073/pnas.1423145112">https://doi.org/10.1073/pnas.1423145112</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_61" title="Jump back to footnote 61 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_62">
<p>Verma, P.; Truhlar, D. G. HLE16: A Local Kohn-Sham Gradient Approximation with Good Performance for Semiconductor Band Gaps and Molecular Excitation Energies. <em>The Journal of Physical Chemistry Letters</em> <strong>2017</strong>, <em>8</em> (2), 380387. <a href="https://doi.org/10.1021/acs.jpclett.6b02757">https://doi.org/10.1021/acs.jpclett.6b02757</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_62" title="Jump back to footnote 62 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_63">
<p>Yang, Z.; Peng, H.; Sun, J.; Perdew, J. P. More Realistic Band Gaps from Meta-Generalized Gradient Approximations: Only in a Generalized Kohn-Sham Scheme. <em>Physical Review B</em> <strong>2016</strong>, <em>93</em> (20), 205205. <a href="https://doi.org/10.1103/physrevb.93.205205">https://doi.org/10.1103/physrevb.93.205205</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_63" title="Jump back to footnote 63 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_64">
<p>Mejia-Rodriguez, D.; Trickey, S. B. Deorbitalization Strategies for Meta-Generalized-Gradient-Approximation Exchange-Correlation Functionals. <em>Physical Review A</em> <strong>2017</strong>, <em>96</em> (5), 052512. <a href="https://doi.org/10.1103/physreva.96.052512">https://doi.org/10.1103/physreva.96.052512</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_64" title="Jump back to footnote 64 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_65">
<p>Wang, Y.; Jin, X.; Yu, H. S.; Truhlar, D. G.; He, X. Revised M06-l Functional for Improved Accuracy on Chemical Reaction Barrier Heights, Noncovalent Interactions, and Solid-State Physics. <em>Proceedings of the National Academy of Sciences</em> <strong>2017</strong>, <em>114</em> (32), 84878492. <a href="https://doi.org/10.1073/pnas.1705670114">https://doi.org/10.1073/pnas.1705670114</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_65" title="Jump back to footnote 65 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_66">
<p>Wang, Y.; Verma, P.; Jin, X.; Truhlar, D. G.; He, X. Revised M06 Density Functional for Main-Group and Transition-Metal Chemistry. <em>Proceedings of the National Academy of Sciences</em> <strong>2018</strong>, <em>115</em> (41), 1025710262. <a href="https://doi.org/10.1073/pnas.1810421115">https://doi.org/10.1073/pnas.1810421115</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_66" title="Jump back to footnote 66 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_67">
<p>Chai, J.-D.; Head-Gordon, M. Systematic Optimization of Long-Range Corrected Hybrid Density Functionals. <em>The Journal of Chemical Physics</em> <strong>2008</strong>, <em>128</em> (8), 084106. <a href="https://doi.org/10.1063/1.2834918">https://doi.org/10.1063/1.2834918</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_67" title="Jump back to footnote 67 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_68">
<p>Lin, Y.-S.; Li, G.-D.; Mao, S.-P.; Chai, J.-D. Long-Range Corrected Hybrid Density Functionals with Improved Dispersion Corrections. <em>Journal of Chemical Theory and Computation</em> <strong>2012</strong>, <em>9</em> (1), 263272. <a href="https://doi.org/10.1021/ct300715s">https://doi.org/10.1021/ct300715s</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_68" title="Jump back to footnote 68 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_69">
<p>Bartk, A. P.; Yates, J. R. Regularized SCAN Functional. <em>The Journal of Chemical Physics</em> <strong>2019</strong>, <em>150</em> (16), 161101. <a href="https://doi.org/10.1063/1.5094646">https://doi.org/10.1063/1.5094646</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_69" title="Jump back to footnote 69 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_70">
<p>Furness, J. W.; Kaplan, A. D.; Ning, J.; Perdew, J. P.; Sun, J. Accurate and Numerically Efficient r2SCAN Meta-Generalized Gradient Approximation. <em>The Journal of Physical Chemistry Letters</em> <strong>2020</strong>, <em>11</em> (19), 82088215. <a href="https://doi.org/10.1021/acs.jpclett.0c02405">https://doi.org/10.1021/acs.jpclett.0c02405</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_70" title="Jump back to footnote 70 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_71">
<p>Bursch, M.; Neugebauer, H.; Ehlert, S.; Grimme, S. Dispersion Corrected r2SCAN Based Global Hybrid Functionals: r2SCANh, r2SCAN0, and r2SCAN50. <em>The Journal of Chemical Physics</em> <strong>2022</strong>, <em>156</em> (13), 134105. <a href="https://doi.org/10.1063/5.0086040">https://doi.org/10.1063/5.0086040</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_71" title="Jump back to footnote 71 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_72">
<p>Meja-Rodrguez, D.; Trickey, S. B. Meta-GGA Performance in Solids at Almost GGA Cost. <em>Physical Review B</em> <strong>2020</strong>, <em>102</em> (12), 121109. <a href="https://doi.org/10.1103/physrevb.102.121109">https://doi.org/10.1103/physrevb.102.121109</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_72" title="Jump back to footnote 72 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_73">
<p>Meja-Rodrguez, D.; Trickey, S. B. Spin-Crossover from a Well-Behaved, Low-Cost Meta-GGA Density Functional. <em>The Journal of Physical Chemistry A</em> <strong>2020</strong>, <em>124</em> (47), 98899894. <a href="https://doi.org/10.1021/acs.jpca.0c08883">https://doi.org/10.1021/acs.jpca.0c08883</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_73" title="Jump back to footnote 73 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_74">
<p>Carmona-Espndola, J.; Gzquez, J. L.; Vela, A.; Trickey, S. B. Generalized Gradient Approximation Exchange Energy Functional with Near-Best Semilocal Performance. <em>Journal of Chemical Theory and Computation</em> <strong>2018</strong>, <em>15</em> (1), 303310. <a href="https://doi.org/10.1021/acs.jctc.8b00998">https://doi.org/10.1021/acs.jctc.8b00998</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_74" title="Jump back to footnote 74 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_75">
<p>Kurth, S.; Perdew, J. P.; Blaha, P. Molecular and Solid-State Tests of Density Functional Approximations: LSD, GGAs, and Meta-GGAs. <em>International Journal of Quantum Chemistry</em> <strong>1999</strong>, <em>75</em> (4-5), 889909. <a href="https://doi.org/10.1002/(sici)1097-461x(1999)75:4/5&lt;889::aid-qua54&gt;3.0.co;2-8">https://doi.org/10.1002/(sici)1097-461x(1999)75:4/5&lt;889::aid-qua54&gt;3.0.co;2-8</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_75" title="Jump back to footnote 75 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_76">
<p>Savin, A. Beyond the Kohn-Sham Determinant. In <em>Recent advances in density functional methods</em>; WORLD SCIENTIFIC, 1995; pp 129153. <a href="https://doi.org/10.1142/9789812830586_0004">https://doi.org/10.1142/9789812830586\0004</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_76" title="Jump back to footnote 76 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_77">
<p>Iikura, H.; Tsuneda, T.; Yanai, T.; Hirao, K. A Long-Range Correction Scheme for Generalized-Gradient-Approximation Exchange Functionals. <em>The Journal of Chemical Physics</em> <strong>2001</strong>, <em>115</em> (8), 35403544. <a href="https://doi.org/10.1063/1.1383587">https://doi.org/10.1063/1.1383587</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_77" title="Jump back to footnote 77 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_78">
<p>Tawada, Y.; Tsuneda, T.; Yanagisawa, S.; Yanai, T.; Hirao, K. A Long-Range-Corrected Time-Dependent Density Functional Theory. <em>The Journal of Chemical Physics</em> <strong>2004</strong>, <em>120</em> (18), 84258433. <a href="https://doi.org/10.1063/1.1688752">https://doi.org/10.1063/1.1688752</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_78" title="Jump back to footnote 78 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_79">
<p>Yanai, T.; Tew, D. P.; Handy, N. C. A New Hybrid Exchange-Correlation Functional Using the Coulomb-Attenuating Method (CAM-B3LYP). <em>Chemical Physics Letters</em> <strong>2004</strong>, <em>393</em> (1-3), 5157. <a href="https://doi.org/10.1016/j.cplett.2004.06.011">https://doi.org/10.1016/j.cplett.2004.06.011</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_79" title="Jump back to footnote 79 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_80">
<p>Peach, M. J. G.; Cohen, A. J.; Tozer, D. J. Influence of Coulomb-Attenuation on Exchange-Correlation Functional Quality. <em>Phys. Chem. Chem. Phys.</em> <strong>2006</strong>, <em>8</em> (39), 45434549. <a href="https://doi.org/10.1039/b608553a">https://doi.org/10.1039/b608553a</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_80" title="Jump back to footnote 80 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_81">
<p>Song, J.-W.; Hirosawa, T.; Tsuneda, T.; Hirao, K. Long-Range Corrected Density Functional Calculations of Chemical Reactions: Redetermination of Parameter. <em>The Journal of Chemical Physics</em> <strong>2007</strong>, <em>126</em> (15), 154105. <a href="https://doi.org/10.1063/1.2721532">https://doi.org/10.1063/1.2721532</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_81" title="Jump back to footnote 81 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_82">
<p>Livshits, E.; Baer, R. A Well-Tempered Density Functional Theory of Electrons in Molecules. <em>Physical Chemistry Chemical Physics</em> <strong>2007</strong>, <em>9</em> (23), 2932. <a href="https://doi.org/10.1039/b617919c">https://doi.org/10.1039/b617919c</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_82" title="Jump back to footnote 82 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_83">
<p>Cohen, A. J.; Mori-Snchez, P.; Yang, W. Development of Exchange-Correlation Functionals with Minimal Many-Electron Self-Interaction Error. <em>The Journal of Chemical Physics</em> <strong>2007</strong>, <em>126</em> (19), 191109. <a href="https://doi.org/10.1063/1.2741248">https://doi.org/10.1063/1.2741248</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_83" title="Jump back to footnote 83 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_84">
<p>Rohrdanz, M. A.; Herbert, J. M. Simultaneous Benchmarking of Ground- and Excited-State Properties with Long-Range-Corrected Density Functional Theory. <em>The Journal of Chemical Physics</em> <strong>2008</strong>, <em>129</em> (3), 034107. <a href="https://doi.org/10.1063/1.2954017">https://doi.org/10.1063/1.2954017</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_84" title="Jump back to footnote 84 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_85">
<p>Govind, N.; Valiev, M.; Jensen, L.; Kowalski, K. Excitation Energies of Zinc Porphyrin in Aqueous Solution Using Long-Range Corrected Time-Dependent Density Functional Theory. <em>The Journal of Physical Chemistry A</em> <strong>2009</strong>, <em>113</em> (21), 60416043. <a href="https://doi.org/10.1021/jp902118k">https://doi.org/10.1021/jp902118k</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_85" title="Jump back to footnote 85 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_86">
<p>Baer, R.; Livshits, E.; Salzner, U. Tuned Range-Separated Hybrids in Density Functional Theory. <em>Annual Review of Physical Chemistry</em> <strong>2010</strong>, <em>61</em> (1), 85109. <a href="https://doi.org/10.1146/annurev.physchem.012809.103321">https://doi.org/10.1146/annurev.physchem.012809.103321</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_86" title="Jump back to footnote 86 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_87">
<p>Autschbach, J.; Srebro, M. Delocalization Error and Functional Tuning in Kohn-Sham Calculations of Molecular Properties. <em>Accounts of Chemical Research</em> <strong>2014</strong>, <em>47</em> (8), 25922602. <a href="https://doi.org/10.1021/ar500171t">https://doi.org/10.1021/ar500171t</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_87" title="Jump back to footnote 87 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_88">
<p>Verma, P.; Bartlett, R. J. Increasing the Applicability of Density Functional Theory. IV. Consequences of Ionization-Potential Improved Exchange-Correlation Potentials. <em>The Journal of Chemical Physics</em> <strong>2014</strong>, <em>140</em> (18), 18A534. <a href="https://doi.org/10.1063/1.4871409">https://doi.org/10.1063/1.4871409</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_88" title="Jump back to footnote 88 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_89">
<p>Swart, M.; Sol, M.; Bickelhaupt, F. M. A New All-Round Density Functional Based on Spin States and SN2 Barriers. <em>The Journal of Chemical Physics</em> <strong>2009</strong>, <em>131</em> (9), 094103. <a href="https://doi.org/10.1063/1.3213193">https://doi.org/10.1063/1.3213193</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_89" title="Jump back to footnote 89 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_90">
<p>Swart, M.; Sol, M.; Bickelhaupt, F. M. Switching Between OPTX and PBE Exchange Functionals. <em>Journal of Computational Methods in Sciences and Engineering</em> <strong>2009</strong>, <em>9</em> (1-2), 6977. <a href="https://doi.org/10.3233/jcm-2009-0230">https://doi.org/10.3233/jcm-2009-0230</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_90" title="Jump back to footnote 90 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_91">
<p>Grimme, S. Semiempirical Hybrid Density Functional with Perturbative Second-Order Correlation. <em>The Journal of Chemical Physics</em> <strong>2006</strong>, <em>124</em> (3), 034108. <a href="https://doi.org/10.1063/1.2148954">https://doi.org/10.1063/1.2148954</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_91" title="Jump back to footnote 91 in the text">&#8617;</a><a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref2_footnote_91" title="Jump back to footnote 91 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_92">
<p>Rabuck, A. D.; Scuseria, G. E. Improving Self-Consistent Field Convergence by Varying Occupation Numbers. <em>The Journal of Chemical Physics</em> <strong>1999</strong>, <em>110</em> (2), 695700. <a href="https://doi.org/10.1063/1.478177">https://doi.org/10.1063/1.478177</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_92" title="Jump back to footnote 92 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_93">
<p>Wu, Q.; Voorhis, T. V. Direct Optimization Method to Study Constrained Systems Within Density-Functional Theory. <em>Physical Review A</em> <strong>2005</strong>, <em>72</em> (2), 024502. <a href="https://doi.org/10.1103/physreva.72.024502">https://doi.org/10.1103/physreva.72.024502</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_93" title="Jump back to footnote 93 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_94">
<p>Warren, R. W.; Dunlap, B. I. Fractional Occupation Numbers and Density Functional Energy Gradients Within the Linear Combination of Gaussian-Type Orbitals Approach. <em>Chemical Physics Letters</em> <strong>1996</strong>, <em>262</em> (3-4), 384392. <a href="https://doi.org/10.1016/0009-2614(96)01107-4">https://doi.org/10.1016/0009-2614(96)01107-4</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_94" title="Jump back to footnote 94 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_95">
<p>Becke, A. D. A Multicenter Numerical Integration Scheme for Polyatomic Molecules. <em>The Journal of Chemical Physics</em> <strong>1988</strong>, <em>88</em> (4), 25472553. <a href="https://doi.org/10.1063/1.454033">https://doi.org/10.1063/1.454033</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_95" title="Jump back to footnote 95 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_96">
<p>Stratmann, R. E.; Scuseria, G. E.; Frisch, M. J. Achieving Linear Scaling in Exchange-Correlation Density Functional Quadratures. <em>Chemical Physics Letters</em> <strong>1996</strong>, <em>257</em> (3-4), 213223. <a href="https://doi.org/10.1016/0009-2614(96)00600-8">https://doi.org/10.1016/0009-2614(96)00600-8</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_96" title="Jump back to footnote 96 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_97">
<p>Murray, C. W.; Handy, N. C.; Laming, G. J. Quadrature Schemes for Integrals of Density Functional Theory. <em>Molecular Physics</em> <strong>1993</strong>, <em>78</em> (4), 9971014. <a href="https://doi.org/10.1080/00268979300100651">https://doi.org/10.1080/00268979300100651</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_97" title="Jump back to footnote 97 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_98">
<p>Mura, M. E.; Knowles, P. J. Improved Radial Grids for Quadrature in Molecular Density-Functional Calculations. <em>The Journal of Chemical Physics</em> <strong>1996</strong>, <em>104</em> (24), 98489858. <a href="https://doi.org/10.1063/1.471749">https://doi.org/10.1063/1.471749</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_98" title="Jump back to footnote 98 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_99">
<p>Treutler, O.; Ahlrichs, R. Efficient Molecular Numerical Integration Schemes. <em>The Journal of Chemical Physics</em> <strong>1995</strong>, <em>102</em> (1), 346354. <a href="https://doi.org/10.1063/1.469408">https://doi.org/10.1063/1.469408</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_99" title="Jump back to footnote 99 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_100">
<p>Sharp, R. T.; Horton, G. K. A Variational Approach to the Unipotential Many-Electron Problem. <em>Physical Review</em> <strong>1953</strong>, <em>90</em> (2), 317317. <a href="https://doi.org/10.1103/physrev.90.317">https://doi.org/10.1103/physrev.90.317</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_100" title="Jump back to footnote 100 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_101">
<p>Talman, J. D.; Shadwick, W. F. Optimized Effective Atomic Central Potential. <em>Physical Review A</em> <strong>1976</strong>, <em>14</em> (1), 3640. <a href="https://doi.org/10.1103/physreva.14.36">https://doi.org/10.1103/physreva.14.36</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_101" title="Jump back to footnote 101 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_102">
<p>Krieger, J. B.; Li, Y.; Iafrate, G. J. Construction and Application of an Accurate Local Spin-Polarized Kohn-Sham Potential with Integer Discontinuity: Exchange-Only Theory. <em>Physical Review A</em> <strong>1992</strong>, <em>45</em> (1), 101126. <a href="https://doi.org/10.1103/physreva.45.101">https://doi.org/10.1103/physreva.45.101</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_102" title="Jump back to footnote 102 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_103">
<p>Krieger, J. B.; Li, Y.; Iafrate, G. J. Systematic Approximations to the Optimized Effective Potential: Application to Orbital-Density-Functional Theory. <em>Physical Review A</em> <strong>1992</strong>, <em>46</em> (9), 54535458. <a href="https://doi.org/10.1103/physreva.46.5453">https://doi.org/10.1103/physreva.46.5453</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_103" title="Jump back to footnote 103 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_104">
<p>Li, Y.; Krieger, J. B.; Iafrate, G. J. Self-Consistent Calculations of Atomic Properties Using Self-Interaction-Free Exchange-Only Kohn-Sham Potentials. <em>Physical Review A</em> <strong>1993</strong>, <em>47</em> (1), 165181. <a href="https://doi.org/10.1103/physreva.47.165">https://doi.org/10.1103/physreva.47.165</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_104" title="Jump back to footnote 104 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_105">
<p>Garza, J.; Nichols, J. A.; Dixon, D. A. The Optimized Effective Potential and the Self-Interaction Correction in Density Functional Theory: Application to Molecules. <em>The Journal of Chemical Physics</em> <strong>2000</strong>, <em>112</em> (18), 78807890. <a href="https://doi.org/10.1063/1.481421">https://doi.org/10.1063/1.481421</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_105" title="Jump back to footnote 105 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_106">
<p>Galvan, M.; Vela, A.; Gazquez, J. L. Chemical Reactivity in Spin-Polarized Density Functional Theory. <em>The Journal of Physical Chemistry</em> <strong>1988</strong>, <em>92</em> (22), 64706474. <a href="https://doi.org/10.1021/j100333a056">https://doi.org/10.1021/j100333a056</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_106" title="Jump back to footnote 106 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_107">
<p>Chamorro, E.; Prez, P. Condensed-to-Atoms Electronic Fukui Functions Within the Framework of Spin-Polarized Density-Functional Theory. <em>The Journal of Chemical Physics</em> <strong>2005</strong>, <em>123</em> (11), 114107. <a href="https://doi.org/10.1063/1.2033689">https://doi.org/10.1063/1.2033689</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_107" title="Jump back to footnote 107 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_108">
<p>Wu, Q.; Yang, W. Empirical Correction to Density Functional Theory for van Der Waals Interactions. <em>The Journal of Chemical Physics</em> <strong>2002</strong>, <em>116</em> (2), 515524. <a href="https://doi.org/10.1063/1.1424928">https://doi.org/10.1063/1.1424928</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_108" title="Jump back to footnote 108 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_109">
<p>Zimmerli, U.; Parrinello, M.; Koumoutsakos, P. Dispersion Corrections to Density Functionals for Water Aromatic Interactions. <em>The Journal of Chemical Physics</em> <strong>2004</strong>, <em>120</em> (6), 26932699. <a href="https://doi.org/10.1063/1.1637034">https://doi.org/10.1063/1.1637034</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_109" title="Jump back to footnote 109 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_110">
<p>Grimme, S. Accurate Description of van Der Waals Complexes by Density Functional Theory Including Empirical Corrections. <em>Journal of Computational Chemistry</em> <strong>2004</strong>, <em>25</em> (12), 14631473. <a href="https://doi.org/10.1002/jcc.20078">https://doi.org/10.1002/jcc.20078</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_110" title="Jump back to footnote 110 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_111">
<p>Grimme, S.; Antony, J.; Ehrlich, S.; Krieg, H. A Consistent and Accurate Ab Initio Parametrization of Density Functional Dispersion Correction (DFT-d) for the 94 Elements h-Pu. <em>The Journal of Chemical Physics</em> <strong>2010</strong>, <em>132</em> (15), 154104. <a href="https://doi.org/10.1063/1.3382344">https://doi.org/10.1063/1.3382344</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_111" title="Jump back to footnote 111 in the text">&#8617;</a></p>
</li>
<li id="density-functional-theory-for-molecules-fn_footnote_112">
<p>Otero-de-la-Roza, A.; Johnson, E. R. Non-Covalent Interactions and Thermochemistry Using XDM-Corrected Hybrid and Range-Separated Hybrid Density Functionals. <em>The Journal of Chemical Physics</em> <strong>2013</strong>, <em>138</em> (20), 204109. <a href="https://doi.org/10.1063/1.4807330">https://doi.org/10.1063/1.4807330</a>.&#160;<a class="footnote-backref" href="#density-functional-theory-for-molecules-fnref_footnote_112" title="Jump back to footnote 112 in the text">&#8617;</a></p>
</li>
</ol>
</div></section><section class="print-page" id="excited-state-calculations" heading-number="11.3"><h1 id="excited-state-calculations-cis-tdhf-tddft">CIS, TDHF, TDDFT<a class="headerlink" href="#excited-state-calculations-cis-tdhf-tddft" title="Permanent link">&para;</a></h1>
<h2 id="excited-state-calculations-overview">Overview<a class="headerlink" href="#excited-state-calculations-overview" title="Permanent link">&para;</a></h2>
<p>NWChem supports a spectrum of single excitation theories for vertical
excitation energy calculations, namely, configuration interaction
singles (CIS)<sup id="excited-state-calculations-fnref_footnote_1"><a class="footnote-ref" href="#excited-state-calculations-fn_footnote_1">1</a></sup>, time-dependent Hartree-Fock (TDHF or also known as
random-phase approximation RPA), time-dependent density functional
theory (TDDFT)<sup id="excited-state-calculations-fnref_footnote_2"><a class="footnote-ref" href="#excited-state-calculations-fn_footnote_2">2</a></sup>, and Tamm-Dancoff approximation<sup id="fnref_footnote_3"><a class="footnote-ref" href="#excited-state-calculations-fn_footnote_3">3</a></sup> to TDDFT. These
methods are implemented in a single framework that invokes Davidson&rsquo;s
trial vector algorithm (or its modification for a non-Hermitian
eigenvalue problem). The capabilities of the module are summarized as
follows:</p>
<ul>
<li>Vertical excitation energies (valence and core),</li>
<li>Spin-restricted singlet and triplet excited states for closed-shell
    systems,</li>
<li>Spin-unrestricted doublet, etc., excited states for open-shell
    systems,</li>
<li>Tamm-Dancoff and full time-dependent linear response theories,</li>
<li>Davidson&rsquo;s trial vector algorithm,</li>
<li>Symmetry (irreducible representation) characterization and
    specification,</li>
<li>Spin multiplicity characterization and specification,</li>
<li>Transition moments and oscillator strengths,</li>
<li>Analytical first derivatives of vertical excitation energies with a
    selected set of exchange-correlation functionals (see TDDFT
    gradients documentation for further information),</li>
<li>Numerical second derivatives of vertical excitation energies,</li>
<li>Disk-based and fully incore algorithms,</li>
<li>Multiple and single trial-vector processing algorithms,</li>
<li>Frozen core and virtual approximation,</li>
<li>Asymptotically correct exchange-correlation potential by van Leeuwen
    and Baerends<sup id="excited-state-calculations-fnref_footnote_4"><a class="footnote-ref" href="#excited-state-calculations-fn_footnote_4">4</a></sup>,</li>
<li>Asymptotic correction by Casida and Salahub<sup id="fnref2_footnote_5"><a class="footnote-ref" href="#excited-state-calculations-fn_footnote_5">5</a></sup>,</li>
<li>Asymptotic correction by Hirata, Zhan, Apr, Windus, and Dixon<sup id="fnref2_footnote_6"><a class="footnote-ref" href="#excited-state-calculations-fn_footnote_6">6</a></sup>.</li>
</ul>
<p>These are very effective way to rectify the shortcomings of TDDFT when
applied to Rydberg excited states (see below).</p>
<h2 id="excited-state-calculations-performance-of-cis-tdhf-and-tddft-methods">Performance of CIS, TDHF, and TDDFT methods<a class="headerlink" href="#excited-state-calculations-performance-of-cis-tdhf-and-tddft-methods" title="Permanent link">&para;</a></h2>
<p>The accuracy of CIS and TDHF for excitation energies of closed-shell
systems are comparable to each other, and are normally considered a
zeroth-order description of the excitation process. These methods are
particularly well balanced in describing Rydberg excited states, in
contrast to TDDFT. However, for open-shell systems, the errors in the
CIS and TDHF excitation energies are often excessive, primarily due to
the multi-determinantal character of the ground and excited state wave
functions of open-shell systems in a HF reference. The scaling of
the computational cost of a CIS or TDHF calculation per state with
respect to the system size is the same as that for a HF calculation for
the ground state, since the critical step of the both methods are the
Fock build, namely, the contraction of two-electron integrals with
density matrices. It is usually necessary to include two sets of diffuse
exponents in the basis set to properly account for the diffuse Rydberg
excited states of neutral species.</p>
<p>The accuracy of TDDFT may vary depending on the exchange-correlation
functional. In general, the exchange-correlation functionals that are
widely used today and are implemented in NWChem work well for low-lying
valence excited states. However, for high-lying diffuse excited states
and Rydberg excited states in particular, TDDFT employing these
conventional functionals breaks down and the excitation energies are
substantially underestimated. This is because of the fact that the
exchange-correlation potentials generated from these functionals decay
too rapidly (exponentially) as opposed to the slow <em>-1/r</em>  asymptotic
decay of the true potential. A rough but useful index is the negative of
the highest occupied KS orbital energy; when the calculated excitation
energies become close to this threshold, these numbers are most likely
underestimated relative to experimental results. It appears that TDDFT
provides a better-balanced description of radical excited states. This
may be traced to the fact that, in DFT, the ground state wave function
is represented well as a single KS determinant, with less
multi-determinantal character and less spin contamination, and hence the
excitation thereof is described well as a simple one electron
transition. The computational cost per state of TDDFT calculations
scales as the same as the ground state DFT calculations, although the
prefactor of the scaling may be much greater in the former.</p>
<p>A very simple and effecive way to rectify the TDDFT&rsquo;s failure for
Rydberg excited states has been proposed by Tozer and Handy<sup id="excited-state-calculations-fnref_footnote_8"><a class="footnote-ref" href="#excited-state-calculations-fn_footnote_8">7</a></sup> and by Casida and
Salahub<sup id="excited-state-calculations-fnref_footnote_5"><a class="footnote-ref" href="#excited-state-calculations-fn_footnote_5">5</a></sup>. They proposed to splice a <em>-1/r</em>
asymptotic tail to an exchange-correlation potential that does not have
the correct asymptotic behavior. Because the approximate
exchange-correlation potentials are too shallow everywhere, a negative
constant must be added to them before they can be spliced to the <em>-1/r</em>
 tail seamlessly in a region that is not sensitive to chemical
effects or to the long-range behavior. The negative constant or the
shift is usually taken to be the difference of the HOMO energy from the
true ionization potential, which can be obtained either from experiment
or from a SCF calculation. Recently, we proposed a new, expedient, and
self-contained asymptotic correction that does not require an ionization
potential (or shift) as an external parameter from a separate
calculation. In this scheme, the shift is computed by a semi-empirical
formula proposed by Zhan, Nichols, and Dixon<sup id="excited-state-calculations-fnref_footnote_6"><a class="footnote-ref" href="#excited-state-calculations-fn_footnote_6">6</a></sup>. Both Casida-Salahub
scheme and this new asymptotic correction scheme give considerably
improved (Koopmans type) ionization potentials and Rydberg excitation
energies. The latter, however, supply the shift by itself unlike to
former.</p>
<h2 id="excited-state-calculations-input-syntax">Input syntax<a class="headerlink" href="#excited-state-calculations-input-syntax" title="Permanent link">&para;</a></h2>
<p>The module is called <code>TDDFT</code> as time-dependent density functional
theory employing a hybrid HF-DFT functional
encompasses all of the above-mentioned methods implemented. To use this
module, one needs to specify <code>TDDFT</code> on the task directive, e.g.,</p>
<pre><code>     TASK TDDFT ENERGY
</code></pre>
<p>for a single-point excitation energy calculation, and</p>
<pre><code>     TASK TDDFT OPTIMIZE
</code></pre>
<p>for an excited-state geometry optimization (and perhaps an adiabatic
excitation energy calculation), and</p>
<pre><code>     TASK TDDFT FREQUENCIES
</code></pre>
<p>for an excited-state vibrational frequency calculation. The TDDFT module
first invokes DFT module for a ground-state calculation (regardless of
whether the calculations uses a HF reference as in CIS or TDHF or a DFT
functional), and hence there is no need to perform a separate
ground-state DFT calculation prior to calling a <code>TDDFT</code> task. When no
second argument of the task directive is given, a single-point
excitation energy calculation will be assumed. For geometry
optimizations, it is usually necessary to specify the target excited
state and its irreducible representation it belongs to. See the
subsections <a href="#excited-state-calculations-target-and-targetsym-the-target-root-and-its-symmetry"><code>TARGET</code> and <code>TARGETSYM</code></a> for more detail.</p>
<p>Individual parameters and keywords may be supplied in the <code>TDDFT</code> input
block. The syntax is:</p>
<pre><code> TDDFT
   [(CIS||RPA) default RPA]  
   [NROOTS &lt;integer nroots default 1&gt;]  
   [MAXVECS &lt;integer maxvecs default 1000&gt;] 
   [(SINGLET||NOSINGLET) default SINGLET]  
   [(TRIPLET||NOTRIPLET) default TRIPLET]  
   [THRESH &lt;double thresh default 1e-4&gt;]  
   [MAXITER &lt;integer maxiter default 100&gt;]  
   [TARGET &lt;integer target default 1&gt;]  
   [TARGETSYM &lt;character targetsym default 'none'&gt;]  
   [SYMMETRY]  
   [ECUT] &lt;-cutoff energy&gt;  
   [EWIN] &lt;-lower cutoff energy&gt;  &lt;-higher cutoff energy&gt;  
   [ALPHA] &lt;integer lower orbital&gt;  &lt;integer upper orbital&gt;  
   [BETA] &lt;integer lower orbital&gt;  &lt;integer upper orbital&gt;  
   [CIVECS]  
   [GRAD, END]  
   [CDSPECTRUM]  
   [GIAO]
   [VELOCITY]
   [SIMPLESO]
   [ALGORITHM &lt;integer algorithm default 0&gt;]  
   [FREEZE [[core] (atomic || &lt;integer nfzc default 0&gt;)] \  
            [virtual &lt;integer nfzv default 0&gt;]]  
   [PRINT (none||low||medium||high||debug)  
     &lt;string list_of_names ...&gt;]
 END
</code></pre>
<p>The user can also specify the reference wave function in the DFT input
block (even when CIS and TDHF calculations are requested). See the
section of Sample input and output for more details.</p>
<p>Since each keyword has a default value, a minimal input file will be</p>
<pre><code> GEOMETRY
  Be 0.0 0.0 0.0  
 END  
 BASIS  
  Be library 6-31G**  
 END  
 TASK TDDFT ENERGY
</code></pre>
<p>Note that the keyword for the asymptotic correction must be given in the
DFT input block, since all the effects of the correction (and also
changes in the computer program) occur in the SCF calculation stage. See
<a href="#density-functional-theory-for-molecules-lb94-and-cs00-asymptotic-correction">DFT</a> (keywords <code>CS00</code> and <code>LB94</code>) for details.</p>
<h2 id="excited-state-calculations-keywords-of-tddft-input-block">Keywords of TDDFT input block<a class="headerlink" href="#excited-state-calculations-keywords-of-tddft-input-block" title="Permanent link">&para;</a></h2>
<h3 id="excited-state-calculations-cis-and-rpa-the-tamm-dancoff-approximation">CIS and RPA: the Tamm-Dancoff approximation<a class="headerlink" href="#excited-state-calculations-cis-and-rpa-the-tamm-dancoff-approximation" title="Permanent link">&para;</a></h3>
<p>These keywords toggle the Tamm-Dancoff approximation. <code>CIS</code> means that the
Tamm-Dancoff approximation is used and the CIS or Tamm-Dancoff TDDFT
calculation is requested. <code>RPA</code>, which is the default, requests TDHF (RPA)
or TDDFT calculation.</p>
<p>The performance of CIS (Tamm-Dancoff TDDFT) and RPA (TDDFT) are
comparable in accuracy. However, the computational cost is slightly
greater in the latter due to the fact that the latter involves a
non-Hermitian eigenvalue problem and requires left and right
eigenvectors while the former needs just one set of eigenvectors of a
Hermitian eigenvalue problem. The latter has much greater chance of
aborting the calculation due to triplet near instability or other
instability problems.</p>
<h3 id="excited-state-calculations-nroots-the-number-of-excited-states">NROOTS: the number of excited states<a class="headerlink" href="#excited-state-calculations-nroots-the-number-of-excited-states" title="Permanent link">&para;</a></h3>
<p>One can specify the number of excited state roots to be determined. The
default value for <code>NROOTS</code> is 1. It is advised that the users request several more
roots than actually needed, since owing to the nature of the trial
vector algorithm, some low-lying roots can be missed when they do not
have sufficient overlap with the initial guess vectors.</p>
<h3 id="excited-state-calculations-maxvecs-the-subspace-size">MAXVECS: the subspace size<a class="headerlink" href="#excited-state-calculations-maxvecs-the-subspace-size" title="Permanent link">&para;</a></h3>
<p>The <code>MAXVECS</code> keyword limits the subspace size of Davidson&rsquo;s algorithm; in other
words, it is the maximum number of trial vectors that the calculation is
allowed to hold. Typically, 10 to 20 trial vectors are needed for each
excited state root to be converged. However, it need not exceed the
product of the number of occupied orbitals and the number of virtual
orbitals. The default value is 1000.</p>
<h3 id="excited-state-calculations-singlet-and-nosinglet-singlet-excited-states">SINGLET and NOSINGLET: singlet excited states<a class="headerlink" href="#excited-state-calculations-singlet-and-nosinglet-singlet-excited-states" title="Permanent link">&para;</a></h3>
<p><code>SINGLET || NOSINGLET</code> requests (suppresses) the calculation of singlet
excited states when the reference wave function is closed shell. The
default is <code>SINGLET</code>.</p>
<h3 id="excited-state-calculations-triplet-and-notriplet-triplet-excited-states">TRIPLET and NOTRIPLET: triplet excited states<a class="headerlink" href="#excited-state-calculations-triplet-and-notriplet-triplet-excited-states" title="Permanent link">&para;</a></h3>
<p><code>TRIPLET || NOTRIPLET</code> requests (suppresses) the calculation of triplet
excited states when the reference wave function is closed shell. The
default <code>is TRIPLET</code>.</p>
<h3 id="excited-state-calculations-thresh-the-convergence-threshold-of-davidson-iteration">THRESH: the convergence threshold of Davidson iteration<a class="headerlink" href="#excited-state-calculations-thresh-the-convergence-threshold-of-davidson-iteration" title="Permanent link">&para;</a></h3>
<p>The <code>THRESH</code> keyword specifies the convergence threshold of Davidson&rsquo;s iterative
algorithm to solve a matrix eigenvalue problem. The threshold refers to
the norm of residual, namely, the difference between the left-hand side
and right-hand side of the matrix eigenvalue equation with the current
solution vector. With the default value of 1e-4, the excitation energies
are usually converged to 1e-5 hartree.</p>
<h3 id="excited-state-calculations-maxiter-the-maximum-number-of-davidson-iteration">MAXITER: the maximum number of Davidson iteration<a class="headerlink" href="#excited-state-calculations-maxiter-the-maximum-number-of-davidson-iteration" title="Permanent link">&para;</a></h3>
<p>It typically takes 10-30 iterations for the Davidson algorithm to get
converged results. The default value for <code>MAXITER</code> is 100.</p>
<h3 id="excited-state-calculations-target-and-targetsym-the-target-root-and-its-symmetry">TARGET and TARGETSYM: the target root and its symmetry<a class="headerlink" href="#excited-state-calculations-target-and-targetsym-the-target-root-and-its-symmetry" title="Permanent link">&para;</a></h3>
<p>At the moment, excited-state first geometry derivatives can be
calculated analytically for a set of functionals, while excited-state
second geometry derivatives are obtained by numerical differentiation.
These keywords may be used to specify which excited state root is being
used for the geometrical derivative calculation. For instance, when
<code>TARGET 3</code> and <code>TARGETSYM a1g</code> are included in the input block, the total
energy (ground state energy plus excitation energy) of the third lowest
excited state root (excluding the ground state) transforming as the
irreducible representation a1g will be passed to the module which
performs the derivative calculations. The default values for <code>TARGET</code> and <code>TARGETSYM</code>
 are <code>1</code> and <code>none</code>, respectively.</p>
<p>The keyword <code>TARGETSYM</code> is essential in excited state geometry
optimization, since it is very common that the order of excited states
changes due to the geometry changes in the course of optimization.
Without specifying the <code>TARGETSYM</code>, the optimizer could (and would likely)
be optimizing the geometry of an excited state that is different from
the one the user had intended to optimize at the starting geometry. On
the other hand, in the frequency calculations, <code>TARGETSYM</code> must be <code>none</code>,
since the finite displacements given in the course of frequency
calculations will lift the spatial symmetry of the equilibrium geometry.
When these finite displacements can alter the order of excited states
including the target state, the frequency calculation is not be
feasible.</p>
<h3 id="excited-state-calculations-symmetry-restricting-the-excited-state-symmetry">SYMMETRY: restricting the excited state symmetry<a class="headerlink" href="#excited-state-calculations-symmetry-restricting-the-excited-state-symmetry" title="Permanent link">&para;</a></h3>
<p>By adding the <code>SYMMETRY</code> keyword to the input block, the user can request the
module to generate the initial guess vectors transforming as the same
irreducible representation as <code>TARGETSYM</code>. This causes the final excited
state roots be (exclusively) dominated by those with the specified
irreducible representation. This may be useful, when the user is
interested in just the optically allowed transitions, or in the geometry
optimization of an excited state root with a particular irreducible
representation. By default, this option is not set. <code>TARGETSYM</code> must be
specified when <code>SYMMETRY</code> is invoked.</p>
<h3 id="excited-state-calculations-ecut-energy-cutoff">ECUT: energy cutoff<a class="headerlink" href="#excited-state-calculations-ecut-energy-cutoff" title="Permanent link">&para;</a></h3>
<p>The <code>ECUT</code> keyword enables restricted excitation window TDDFT (REW-TDDFT)<sup id="fnref_footnote_9"><a class="footnote-ref" href="#excited-state-calculations-fn_footnote_9">8</a></sup>.
This is an approach best suited for core excitations. By specifying this
keyword only excitations from occupied states below the energy cutoff
will be considered.</p>
<h3 id="excited-state-calculations-ewin-energy-window">EWIN: energy window<a class="headerlink" href="#excited-state-calculations-ewin-energy-window" title="Permanent link">&para;</a></h3>
<p>The <code>EWIN</code> keyword enables a restricted energy window between a lower energy
cutoff and a higher energy cutoff. For example, <code>ewin -20.0 -10.0</code> will
only consider excitations from occupied orbitals within the specified
energy window</p>
<h3 id="excited-state-calculations-alpha-beta-alpha-beta-orbital-windows">Alpha, Beta: alpha, beta orbital windows<a class="headerlink" href="#excited-state-calculations-alpha-beta-alpha-beta-orbital-windows" title="Permanent link">&para;</a></h3>
<p>Orbital windows can be specified using the following keywords:</p>
<pre><code>  alpha 1 4
  beta 2 5
</code></pre>
<p>Here alpha excitations will be considered from orbitals 1 through 4
depending on the number of roots requested and beta excitations will be
considered from orbitals 2 through 5 depending on the number of roots
requested.</p>
<h3 id="excited-state-calculations-civecs-ci-vectors">CIVECS: CI vectors<a class="headerlink" href="#excited-state-calculations-civecs-ci-vectors" title="Permanent link">&para;</a></h3>
<p>The <code>CIVECS</code> keyword will result in the CI vectors being written out. By default
this is off. Please note this can be a very large file, so avoid turning
on this keyword if you are calculating a very large number of roots. CI
vectors are needed for excited-state gradient and transition density
calculations.</p>
<h3 id="excited-state-calculations-grad-tddft-gradients">GRAD: TDDFT gradients<a class="headerlink" href="#excited-state-calculations-grad-tddft-gradients" title="Permanent link">&para;</a></h3>
<p>Analytical TDDFT gradients can be calculated by specifying a <code>grad</code> block
within the main <code>TDDFT</code> block</p>
<p>For example, the following will perform a TDDFT optimization on the
first singlet excited state (S1). Note that the <code>civecs</code> keyword must be
specified. To perform a single TDDFT gradient, replace the <code>optimize</code>
keyword with <code>gradient</code> in the task line. A complete TDDFT optimization
input example is given the Sample Inputs section. A TDDFT gradients
calculation can be used to calculate the density of a
specific excited state.
The excited stated density is written to a file with the <code>.dmat</code> suffix.</p>
<pre><code>tddft
 nroots 2
 algorithm 1
 notriplet
 target 1
 targetsym a
 civecs
 grad
   root 1
 end
end
task tddft optimize
</code></pre>
<p>At the moment the following exchange-correlation functionals are
supported with TDDFT
gradients</p>
<pre><code>LDA, BP86, PBE, BLYP, B3LYP, PBE0, BHLYP, CAM-B3LYP, LC-PBE, LC-PBE0, BNL, LC-wPBE, LC-wPBEh, LC-BLYP
</code></pre>
<h3 id="excited-state-calculations-cdspectrum-optical-rotation-calculations">CDSpectrum: optical rotation calculations<a class="headerlink" href="#excited-state-calculations-cdspectrum-optical-rotation-calculations" title="Permanent link">&para;</a></h3>
<p>Perform optical rotation calculations.
We recommend to use the <code>GIAO</code> keyword</p>
<h3 id="excited-state-calculations-velocity-velocity-gauge">VELOCITY: velocity gauge<a class="headerlink" href="#excited-state-calculations-velocity-velocity-gauge" title="Permanent link">&para;</a></h3>
<p>Perform CD spectrum calculations with the velocity gauge.</p>
<h3 id="excited-state-calculations-simpleso-simplified-spin-orbit-coupling">SIMPLESO: simplified Spin-Orbit coupling<a class="headerlink" href="#excited-state-calculations-simpleso-simplified-spin-orbit-coupling" title="Permanent link">&para;</a></h3>
<p>Perform excited states calculations with a simplied Spin-Orbit coupling that uses
eigenvalues from a <a href="#density-functional-theory-for-molecules-spin-orbit-density-functional-theory-sodft">spin-orbit calculation</a>,
instead of a standard dft calculation.<br />
Here is a snippet of an input example (please notice the use of molecular orbitals).</p>
<pre><code>start au2
 geometry
 au 0 0 1
 au 0 0 -1
 symmetry d2h
end
#basis sets, ecp and so-ecp skipped  for simplicity
...
dft
 odft
 vectors output au2_noso.mos
end
task dft
dft
 vectors input au2_noso.mos output au2_so.mos
end
task sodft
dft
 odft
 vectors input u2_noso.mos
end

tddft
  simpleso au2.evals
  nroots 1
  notriplet
end

task tddft
</code></pre>
<h3 id="excited-state-calculations-algorithm-algorithms-for-tensor-contractions">ALGORITHM: algorithms for tensor contractions<a class="headerlink" href="#excited-state-calculations-algorithm-algorithms-for-tensor-contractions" title="Permanent link">&para;</a></h3>
<p>There are four distinct algorithms to choose from, and the default value
of 0 (optimal) means that the program makes an optimal choice from the
four algorithms on the basis of available memory. In the order of
decreasing memory requirement, the four algorithms are:</p>
<ul>
<li>ALGORITHM 1 : Incore, multiple tensor contraction,</li>
<li>ALGORITHM 2 : Incore, single tensor contraction,</li>
<li>ALGORITHM 3 : Disk-based, multiple tensor contraction,</li>
<li>ALGORITHM 4 : Disk-based, single tensor contraction.</li>
</ul>
<p>The incore algorithm stores all the trial and product vectors in memory
across different nodes with the GA, and often decreases the <a href="#excited-state-calculations-maxiter-the-maximum-number-of-davidson-iteration"><code>MAXITER</code></a>
value to accommodate them. The disk-based algorithm stores the vectors
on disks across different nodes with the DRA, and retrieves each vector
one at a time when it is needed. The multiple and single tensor
contraction refers to whether just one or more than one trial vectors
are contracted with integrals. The multiple tensor contraction algorithm
is particularly effective (in terms of speed) for CIS and TDHF, since
the number of the direct evaluations of two-electron integrals is
diminished substantially.</p>
<h3 id="excited-state-calculations-freeze-the-frozen-corevirtual-approximation">FREEZE: the frozen core/virtual approximation<a class="headerlink" href="#excited-state-calculations-freeze-the-frozen-corevirtual-approximation" title="Permanent link">&para;</a></h3>
<p>Some of the lowest-lying core orbitals and/or some of the highest-lying
virtual orbitals may be excluded in the CIS, TDHF, and TDDFT
calculations by the <code>FREEZE</code> keyword (this does not affect the ground state HF
or DFT calculation). No orbitals are frozen by default. To exclude the
atom-like core regions altogether, one may request</p>
<pre><code> FREEZE atomic
</code></pre>
<p>To specify the number of lowest-lying occupied orbitals be excluded, one
may use</p>
<pre><code> FREEZE 10
</code></pre>
<p>which causes 10 lowest-lying occupied orbitals excluded. This is
equivalent to writing</p>
<pre><code> FREEZE core 10
</code></pre>
<p>To freeze the highest virtual orbitals, use the <code>virtual</code> keyword. For
instance, to freeze the top 5 virtuals</p>
<pre><code> FREEZE virtual 5
</code></pre>
<h3 id="excited-state-calculations-trials-restart">TRIALS: restart<a class="headerlink" href="#excited-state-calculations-trials-restart" title="Permanent link">&para;</a></h3>
<p>Setting the keyword <code>trials</code> restart the calculation from the trials vector of a previous run.</p>
<pre><code>trials 
</code></pre>
<h3 id="excited-state-calculations-print-output-verbosity">PRINT: output verbosity<a class="headerlink" href="#excited-state-calculations-print-output-verbosity" title="Permanent link">&para;</a></h3>
<p>The <code>PRINT</code> keyword changes the level of output verbosity. One may also request
some particular items in the table
below.</p>
<table>
<thead>
<tr>
<th>Item</th>
<th>Print Level</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&ldquo;timings&rdquo;</td>
<td>high</td>
<td>CPU and wall times spent in each step</td>
</tr>
<tr>
<td>&ldquo;trial vectors&rdquo;</td>
<td>high</td>
<td>Trial CI vectors</td>
</tr>
<tr>
<td>&ldquo;initial guess&rdquo;</td>
<td>debug</td>
<td>Initial guess CI vectors</td>
</tr>
<tr>
<td>&ldquo;general information&rdquo;</td>
<td>default</td>
<td>General information</td>
</tr>
<tr>
<td>&ldquo;xc information&rdquo;</td>
<td>default</td>
<td>HF/DFT information</td>
</tr>
<tr>
<td>&ldquo;memory information&rdquo;</td>
<td>default</td>
<td>Memory information</td>
</tr>
<tr>
<td>&ldquo;convergence&rdquo;</td>
<td>debug</td>
<td>Convergence</td>
</tr>
<tr>
<td>&ldquo;subspace&rdquo;</td>
<td>debug</td>
<td>Subspace representation of CI matrices</td>
</tr>
<tr>
<td>&ldquo;transform&rdquo;</td>
<td>debug</td>
<td>MO to AO and AO to MO transformation of CI vectors</td>
</tr>
<tr>
<td>&ldquo;diagonalization&rdquo;</td>
<td>debug</td>
<td>Diagonalization of CI matrices</td>
</tr>
<tr>
<td>&ldquo;iteration&rdquo;</td>
<td>default</td>
<td>Davidson iteration update</td>
</tr>
<tr>
<td>&ldquo;contract&rdquo;</td>
<td>debug</td>
<td>Integral transition density contraction</td>
</tr>
<tr>
<td>&ldquo;ground state&rdquo;</td>
<td>default</td>
<td>Final result for ground state</td>
</tr>
<tr>
<td>&ldquo;excited state&rdquo;</td>
<td>low</td>
<td>Final result for target excited state</td>
</tr>
</tbody>
</table>
<p>Printable items in the TDDFT modules and their default print levels.</p>
<h2 id="excited-state-calculations-sample-input">Sample input<a class="headerlink" href="#excited-state-calculations-sample-input" title="Permanent link">&para;</a></h2>
<p>The following is a sample input for a spin-restricted TDDFT calculation
of singlet excitation energies for the water molecule at the
B3LYP/6-31G*.</p>
<pre><code>START h2o  
TITLE &quot;B3LYP/6-31G* H2O&quot;  
GEOMETRY 
 O     0.00000000     0.00000000     0.12982363  
 H     0.75933475     0.00000000    -0.46621158 
 H    -0.75933475     0.00000000    -0.46621158  
END  
BASIS  
 * library 6-31G* 
END  
DFT  
 XC B3LYP  
END  
TDDFT  
 RPA 
 NROOTS 20  
END  
TASK TDDFT ENERGY
</code></pre>
<p>To perform a spin-unrestricted TDHF/aug-cc-pVDZ calculation for the CO+
radical,</p>
<pre><code>START co  
title &quot;TDHF/aug-cc-pVDZ CO+&quot;  
charge 1  
geometry  
 c  0.0  0.0  0.0  
 o  0.0  0.0  1.5
 symmetry c2v # enforcing abelian symmetry
end  
basis  
 * library aug-cc-pvdz  
end  
dft  
 xc hfexch  
 mult 2  
end
task dft optimize
tddft  
 rpa  
 nroots 5  
end  
task tddft energy
</code></pre>
<p>A geometry optimization followed by a frequency calculation for an
excited state is carried out for BF at the CIS/6-31G* level in the
following sample input.</p>
<pre><code>start bf  
title &quot;CIS/6-31G* BF optimization frequencies&quot;  
geometry  
 b 0.0 0.0 0.0  
 f 0.0 0.0 1.2
 symmetry c2v # enforcing abelian symmetry
end  
basis  
 * library 6-31g*  
end  
dft  
 xc hfexch  
end  
tddft  
 cis  
 nroots 3  
 notriplet  
 target 1  
  civecs  
  grad  
    root 1  
  end  
end  
task tddft optimize  
task tddft frequencies
</code></pre>
<p>TDDFT with an asymptotically corrected SVWN exchange-correlation
potential. Casida-Salahub scheme has been used with the shift value of
0.1837 a.u. supplied as an input parameter.</p>
<pre><code>START tddft_ac_co  
GEOMETRY  
 O 0.0 0.0  0.0000  
 C 0.0 0.0  1.1283  
 symmetry c2v # enforcing abelian symmetry
END  
BASIS SPHERICAL  
 C library aug-cc-pVDZ  
 O library aug-cc-pVDZ  
END  
DFT  
 XC Slater VWN_5  
 CS00 0.1837  
END  
TDDFT  
 NROOTS 12  
END  
TASK TDDFT ENERGY
</code></pre>
<p>TDDFT with an asymptotically corrected B3LYP exchange-correlation
potential. Hirata-Zhan-Apra-Windus-Dixon scheme has been used (this is
only meaningful with B3LYP functional).</p>
<pre><code>START tddft_ac_co  
GEOMETRY  
 O 0.0 0.0  0.0000  
 C 0.0 0.0  1.1283  
 symmetry c2v # enforcing abelian symmetry
END  
BASIS SPHERICAL  
 C library aug-cc-pVDZ  
 O library aug-cc-pVDZ  
END  
DFT  
 XC B3LYP  
 CS00  
END  
TDDFT  
 NROOTS 12  
END  
TASK TDDFT ENERGY
</code></pre>
<p>TDDFT for core states. The following example illustrates the usage of an
energy cutoff and energy and orbital windows.<sup id="excited-state-calculations-fnref2_footnote_9"><a class="footnote-ref" href="#excited-state-calculations-fn_footnote_9">8</a></sup></p>
<pre><code>echo  
start h2o_core  
memory 1000 mb  
geometry units au noautosym noautoz  
  O 0.00000000     0.00000000     0.22170860  
  H 0.00000000     1.43758081    -0.88575430  
  H 0.00000000    -1.43758081    -0.88575430  
end  
basis  
 O library 6-31g*  
 H library 6-31g*  
end  
dft  
 xc beckehandh  
 print &quot;final vector analysis&quot;  
end  
task dft  
tddft  
 ecut -10  
 nroots 5  
 notriplet  
 thresh 1d-03  
end  
task tddft  
tddft  
 ewin -20.0 -10.0  
 cis  
 nroots 5  
 notriplet  
 thresh 1d-03  
end  
task tddft  
dft  
 odft  
 mult 1  
 xc beckehandh  
 print &quot;final vector analysis&quot;  
end  
task dft  
tddft  
 alpha 1 1  
 beta 1 1  
 cis  
 nroots 10  
 notriplet  
 thresh 1d-03  
end  
task tddft
</code></pre>
<p>TDDFT optimization with LDA of Pyridine with the 6-31G basis<sup id="fnref_footnote_10"><a class="footnote-ref" href="#excited-state-calculations-fn_footnote_10">9</a></sup></p>
<pre><code>echo  
start tddftgrad_pyridine_opt  
title &quot;TDDFT/LDA geometry optimization of Pyridine with 6-31G&quot;  
geometry nocenter  
 N     0.00000000    0.00000000    1.41599295  
 C     0.00000000   -1.15372936    0.72067272  
 C     0.00000000    1.15372936    0.72067272  
 C     0.00000000   -1.20168790   -0.67391011  
 C     0.00000000    1.20168790   -0.67391011  
 C     0.00000000    0.00000000   -1.38406147  
 H     0.00000000   -2.07614628    1.31521089  
 H     0.00000000    2.07614628    1.31521089  
 H     0.00000000    2.16719803   -1.19243296  
 H     0.00000000   -2.16719803   -1.19243296  
 H     0.00000000    0.00000000   -2.48042299  
 symmetry c1  
end  
basis spherical  
* library &quot;6-31G&quot;  
end  
driver  
  clear  
  maxiter 100  
end  
dft  
  iterations 500  
  grid xfine  
end  
tddft  
  nroots 2  
  algorithm 1  
  notriplet  
  target 1  
  targetsym a  
  civecs  
  grad  
    root 1  
  end  
end  
task tddft optimize
</code></pre>
<p>TDDFT calculation followed by a calculation of the transition density
for a specific excited state using the DPLOT block</p>
<pre><code>echo  
start h2o-td  
title h2o-td  

charge 0  
geometry units au
symmetry group c1  
 O    0.00000000000000      0.00000000000000      0.00000000000000  
 H    0.47043554760291      1.35028113274600      1.06035416576826  
 H   -1.74335410533480     -0.23369304784300      0.27360785442967  
end  
basis &quot;ao basis&quot;
* library &quot;Ahlrichs pVDZ&quot;
end  
dft  
 xc bhlyp  
 grid fine  
 direct  
 convergence energy 1d-5  
end  
tddft  
 rpa  
 nroots 5  
 thresh 1d-5  
 singlet  
 notriplet  
 civecs  
end  
task tddft energy  
dplot  
 civecs h2o-td.civecs_singlet  
 root 2  
 LimitXYZ  
  -3.74335 2.47044 50  
  -2.23369 3.35028 50  
  -2 3.06035 50  
   gaussian  
   output root-2.cube  
end  
task dplot
</code></pre>
<p>TDDFT protocol for calculating the valence-to-core (1s) X-ray emission spectrum <sup id="fnref_footnote_11"><a class="footnote-ref" href="#excited-state-calculations-fn_footnote_11">10</a></sup></p>
<ol>
<li>Calculate the neutral ground state.   </li>
<li>Calculate a full core hole (FCH) ionized state self-consistently, where the<br />
1s core orbital of the absorbing center is swapped with a virtual orbital. Use the <br />
maximum overlap constraint to prevent core hole collapse during the FCH calculation.</li>
<li>Perform a LR-TDDFT calculation within the TDA is performed with the FCH ionized  <br />
state as reference. </li>
<li>Final spectra is produced by taking the absolute value of the negative eigenvalues.</li>
</ol>
<h2 id="excited-state-calculations-spectrum-parser">Spectrum parser<a class="headerlink" href="#excited-state-calculations-spectrum-parser" title="Permanent link">&para;</a></h2>
<p>A Python script is available for parsing NWChem output for TDDFT/vspec excitation energies, and optionally Lorentzian broadenening the spectra . The  nw_spectrum.py file can be found at <a href="https://raw.githubusercontent.com/nwchemgit/nwchem/master/contrib/parsers/nw_spectrum.py">https://raw.githubusercontent.com/nwchemgit/nwchem/master/contrib/parsers/nw_spectrum.py</a></p>
<pre><code>Usage: nw_spectrum.py [options]

Reads NWChem output from stdin, parses for the linear response TDDFT or DFT
vspec excitations, and prints the absorption spectrum to stdout.  It will
optionally broaden peaks using a Lorentzian with FWHM of at least two
energy/wavelength spacings.  By default, it will automatically determine data
format (tddft or vspec) and generate a broadened spectrum in eV.

Example:

    nw_spectrum -b0.3 -p5000 -wnm &lt; water.nwo &gt; spectrum.dat

Create absorption spectrum in nm named &quot;spectrum.dat&quot; from the NWChem output
file &quot;water.nwo&quot; named spectrum.dat with peaks broadened by 0.3 eV and 5000
points in the spectrum.


Options:
  -h, --help            show this help message and exit
  -f FMT, --format=FMT  data file format: auto (default), tddft, vspec, dos
  -b WID, --broad=WID   broaden peaks (FWHM) by WID eV (default 0.1 eV)
  -n NUM, --nbin=NUM    number of eigenvalue bins for DOS calc (default 20)
  -p NUM, --points=NUM  create a spectrum with NUM points (default 2000)
  -w UNT, --units=UNT   units for frequency:  eV (default), au, nm
  -d STR, --delim=STR   use STR as output separator (four spaces default)
  -x, --extract         extract unbroadened roots; do not make spectrum
  -C, --clean           clean output; data only, no header or comments
  -c CHA, --comment=CHA
                        comment character for output ('#' default)
  -v, --verbose         echo warnings and progress to stderr
</code></pre>
<h2 id="excited-state-calculations-references">References<a class="headerlink" href="#excited-state-calculations-references" title="Permanent link">&para;</a></h2>
<div class="footnote">
<hr />
<ol>
<li id="excited-state-calculations-fn_footnote_1">
<p>J. B. Foreman, M. Head-Gordon, J. A. Pople, and M. J. Frisch, <em>J. Phys. Chem.</em> <strong>96</strong>, 135 (1992), DOI:<a href="https://dx.doi.org/10.1021/j100180a030">10.1021/j100180a030</a>&#160;<a class="footnote-backref" href="#excited-state-calculations-fnref_footnote_1" title="Jump back to footnote 1 in the text">&#8617;</a></p>
</li>
<li id="excited-state-calculations-fn_footnote_2">
<p>C. Jamorski, M. E. Casida, and D. R. Salahub, <em>J. Chem. Phys.</em> <strong>104</strong>, 5134 (1996), DOI:<a href="https://dx.doi.org/10.1063/1.471140">10.1063/1.471140</a>; R. Bauernschmitt and R. Ahlrichs, <em>Chem. Phys. Lett.</em> <strong>256</strong>, 454 (1996), DOI:<a href="https://dx.doi.org/10.1016/0009-2614(96)00440-X">10.1016/0009-2614(96)00440-X</a>; R. Bauernschmitt, M. Hser, O. Treutler, and R. Ahlrichs, <em>Chem. Phys. Lett.</em> <strong>264</strong>, 573 (1997), DOI:<a href="https://dx.doi.org/10.1016/S0009-2614(96)01343-7">10.1016/S0009-2614(96)01343-7</a>.&#160;<a class="footnote-backref" href="#excited-state-calculations-fnref_footnote_2" title="Jump back to footnote 2 in the text">&#8617;</a></p>
</li>
<li id="excited-state-calculations-fn_footnote_3">
<p>S. Hirata and M. Head-Gordon, <em>Chem. Phys. Lett.</em> <strong>314</strong>, 291 (1999).  DOI:<a href="https://dx.doi.org/10.1016/S0009-2614(99)01149-5">10.1016/S0009-2614(99)01149-5</a>&#160;<a class="footnote-backref" href="#excited-state-calculations-fnref_footnote_3" title="Jump back to footnote 3 in the text">&#8617;</a></p>
</li>
<li id="excited-state-calculations-fn_footnote_4">
<p>R. van Leeuwen and E. J. Baerends, <em>Phys. Rev. A</em> <strong>49</strong>,  2421 (1994), DOI:<a href="https://dx.doi.org/10.1103/PhysRevA.49.2421">10.1103/PhysRevA.49.2421</a>&#160;<a class="footnote-backref" href="#excited-state-calculations-fnref_footnote_4" title="Jump back to footnote 4 in the text">&#8617;</a></p>
</li>
<li id="excited-state-calculations-fn_footnote_5">
<p>M. E. Casida, C. Jamorski, K. C. Casida, and D. R. Salahub, <em>J. Chem. Phys.</em> <strong>108</strong>, 4439 (1998), DOI:<a href="https://dx.doi.org/10.1063/1.475855">10.1063/1.475855</a>&#160;<a class="footnote-backref" href="#excited-state-calculations-fnref_footnote_5" title="Jump back to footnote 5 in the text">&#8617;</a><a class="footnote-backref" href="#excited-state-calculations-fnref2_footnote_5" title="Jump back to footnote 5 in the text">&#8617;</a></p>
</li>
<li id="excited-state-calculations-fn_footnote_6">
<p>S. Hirata, C.-G. Zhan, E. Apr, T. L. Windus, and D. A. Dixon, <em>J. Phys. Chem. A</em> <strong>107</strong>, 10154 (2003).  DOI:<a href="https://dx.doi.org/10.1021/jp035667x">10.1021/jp035667x</a>&#160;<a class="footnote-backref" href="#excited-state-calculations-fnref_footnote_6" title="Jump back to footnote 6 in the text">&#8617;</a><a class="footnote-backref" href="#excited-state-calculations-fnref2_footnote_6" title="Jump back to footnote 6 in the text">&#8617;</a></p>
</li>
<li id="excited-state-calculations-fn_footnote_8">
<p>D. J. Tozer and N. C. Handy, <em>J. Chem. Phys.</em> <strong>109</strong>, 10180 (1998), DOI:<a href="https://dx.doi.org/10.1063/1.477711">10.1063/1.477711</a>&#160;<a class="footnote-backref" href="#excited-state-calculations-fnref_footnote_8" title="Jump back to footnote 7 in the text">&#8617;</a></p>
</li>
<li id="excited-state-calculations-fn_footnote_9">
<p>K. Lopata, B. E. Van Kuiken, M. Khalil, N. Govind, &ldquo;Linear-Response and Real-Time Time-Dependent Density Functional Theory Studies of Core-Level Near-Edge X-Ray Absorption&rdquo;, <em>J. Chem. Theory Comput.</em>, 2012, <strong>8</strong> (9), pp 32843292, DOI:<a href="https://dx.doi.org/10.1021/ct3005613">10.1021/ct3005613</a>&#160;<a class="footnote-backref" href="#excited-state-calculations-fnref_footnote_9" title="Jump back to footnote 8 in the text">&#8617;</a><a class="footnote-backref" href="#excited-state-calculations-fnref2_footnote_9" title="Jump back to footnote 8 in the text">&#8617;</a></p>
</li>
<li id="excited-state-calculations-fn_footnote_10">
<p>D. W. Silverstein, N. Govind,  H. J. J. van Dam, L. Jensen, &ldquo;Simulating One-Photon Absorption and Resonance Raman Scattering Spectra Using Analytical Excited State Energy Gradients within Time-Dependent Density Functional Theory&rdquo; <em>J. Chem. Theory Comput.</em>, 2013, <strong>9</strong> (12), pp 54905503, DOI:<a href="https://dx.doi.org/10.1021/ct4007772">10.1021/ct4007772</a>&#160;<a class="footnote-backref" href="#excited-state-calculations-fnref_footnote_10" title="Jump back to footnote 9 in the text">&#8617;</a></p>
</li>
<li id="excited-state-calculations-fn_footnote_11">
<p>Y. Zhang, S. Mukamel, M. Khalil, N. Govind, &ldquo;Simulating Valence-to-Core X-ray Emission Spectroscopy of Transition Metal&rdquo;, <em>J. Chem. Theory Comput</em>., 2015, <strong>11</strong> (12), pp 58045809, DOI:<a href="https://dx.doi.org/10.1021/acs.jctc.5b00763">10.1021/acs.jctc.5b00763</a>&#160;<a class="footnote-backref" href="#excited-state-calculations-fnref_footnote_11" title="Jump back to footnote 10 in the text">&#8617;</a></p>
</li>
</ol>
</div></section><section class="print-page" id="rt-tddft" heading-number="11.4"><h1 id="rt-tddft-real-time-tddft">Real-time TDDFT<a class="headerlink" href="#rt-tddft-real-time-tddft" title="Permanent link">&para;</a></h1>
<h2 id="rt-tddft-overview">Overview<a class="headerlink" href="#rt-tddft-overview" title="Permanent link">&para;</a></h2>
<p>Real-time time-dependent density functional theory (RT-TDDFT) is a
DFT-based approach to electronic excited states based on integrating the
time-dependent Kohn-Sham (TDKS) equations in time. The theoretical
underpinnings, strengths, and limitations are similar to traditional
linear-response (LR) <a href="#excited-state-calculations">TDDFT</a>
methods, but instead of a frequency domain solution to the TDKS
equations, RT-TDDFT yields a full time-resolved, potentially non-linear
solution. Real-time simulations can be used to compute not only
spectroscopic properties (e.g., absorption spectra, polarizabilites,
etc), but also the time and space-resolved electronic response to
arbitrary external stimuli (e.g., electron charge dynamics after laser
excitation). For theoretical and computational details, please refer to
the following paper:</p>
<ol>
<li>K. Lopata, N. Govind,&rdquo;Modeling Fast Electron Dynamics with Real-Time
    Time-Dependent Density Functional Theory: Application to Small
    Molecules and Chromophores&rdquo;, J. Chem. Theory Comput., 7, 1344 (2011)
    <a href="https://dx.doi.org/10.1021/ct200137z">DOI:10.1021/ct200137z</a></li>
</ol>
<p>This functionality is built on the <a href="#density-functional-theory-for-molecules">Gaussian basis set
DFT</a> module, and
will work for closed-shell (spin-restricted) and open-shell (spin
unrestricted) calculations with essentially any combination of basis set
and exchange-correlation functional in NWChem. The current
implementation assumes frozen nuclei (Born-Oppenheimer approximation).</p>
<p>In a nutshell, running a RT-TDDFT calculation takes the following form:</p>
<ol>
<li>Compute ground state density matrix with DFT module</li>
<li>Propagate density matrix using RT-TDDFT</li>
<li>Post-process resulting time-dependent observables (e.g., dipole
    moment)</li>
</ol>
<h2 id="rt-tddft-units">Units<a class="headerlink" href="#rt-tddft-units" title="Permanent link">&para;</a></h2>
<p>Unless specified otherwise, all inputs and outputs are in <strong>atomic
units</strong>. Some useful conversions are:</p>
<table>
<thead>
<tr>
<th>Quantity</th>
<th>Conversion</th>
</tr>
</thead>
<tbody>
<tr>
<td>Time</td>
<td>1 au = 0.02419 fs</td>
</tr>
<tr>
<td>Length</td>
<td>1 au = 0.5292 A</td>
</tr>
<tr>
<td>Energy</td>
<td>1 au = 27.2114 eV</td>
</tr>
<tr>
<td>Electric field</td>
<td>1 au = 514.2 V/nm</td>
</tr>
<tr>
<td>Dipole moment</td>
<td>1 au = 2.542 D</td>
</tr>
</tbody>
</table>
<h2 id="rt-tddft-syntax">Syntax<a class="headerlink" href="#rt-tddft-syntax" title="Permanent link">&para;</a></h2>
<p>The <a href="#charge">charge</a>, <a href="#geometry">geometry</a>, <a href="#basis">basis
set</a>, and
<a href="#density-functional-theory-for-molecules">DFT</a> options are
all specified as normal, using their respective syntax. Real-time TDDFT
parameters are supplied in the <code>RT_TDDFT</code> block (note, nothing is
case-sensitive), with all possible options summarized below, and each
discussed in detail
afterwards.</p>
<pre><code>RT_TDDFT 
  [TMAX &lt;double default 1000&gt;]  
  [DT &lt;double default 0.1&gt;]  
  [TAG &lt;string default &quot;&lt;rt_tddft&gt;: &quot;] 
  [LOAD (scf || vectors &lt;string&gt;)]  
  [NCHECKS &lt;integer default 10&gt;] 
  [NPRINTS (* || &lt;integer&gt;)]  
  [NRESTARTS (* || &lt;integer&gt;)]  
  [TOLERANCES (zero &lt;double default 1e-8&gt; || series &lt;double default 1e-10&gt; || interpol &lt;double default 1e-7&gt;)]  
  [PROPAGATOR (euler || rk4 || magnus) default magnus]  
  [EXP (diag || pseries)]  
  [PROF]  
  [NOPROP]  
  [STATIC]  
  [PRINT (*, dipole, quadrupole, field, moocc, field, energy, cputime, charge, convergence, s2)] 
  [EXCITE &lt;string geomname&gt;` with &lt;string fieldname&gt;]  
  [FIELD] 
    ...  
  [END]  
  [VISUALIZATION]  
    ...  
  [END] 
  [LOAD RESTART]
END
</code></pre>
<h3 id="rt-tddft-tmax-simulation-time">TMAX: Simulation time<a class="headerlink" href="#rt-tddft-tmax-simulation-time" title="Permanent link">&para;</a></h3>
<p>This option specifies the maximum time (in au) to run the simulation
before stopping, which must be a positive real number. In practice, you
can just stop the simulation early, so in most cases it is simplest to
just set this to a large value to ensure you capture all the important
dynamics (see the example of <a href="#rt-tddft-resonant-ultraviolet-excitation-of-water">resonant ultraviolet excitation of water</a>). For most
valence excitations, for example, 1000 au is overkill so you might want
to automatically stop at 500 au:</p>
<pre><code>rt_tddft
  ...  
  tmax 500.0 
  ... 
end
</code></pre>
<h3 id="rt-tddft-dt-time-step">DT: Time step<a class="headerlink" href="#rt-tddft-dt-time-step" title="Permanent link">&para;</a></h3>
<p>This specifies the electronic time step for time integration. A larger
time step results in a faster simulation, but there are two issues which
limit the size of the time step. First, the integration algorithm can
become unstable and/or inaccurate for larger time steps, which depends
on the integrator used. Very roughly speaking, the second order Magnus
integrator should be reliable for time steps up to 0.2 au. Second, you
must choose a time step small enough to capture the oscillations of
interest, i.e., to resolve an excitation of energy &omega;, your time
step needs to be smaller than &pi;/&omega;, and typically a tenth of
that for accuracy. For example, to capture high energy oscillations such
as core-level excitations (e.g., &omega; = 50au) you might want a
relative small time step:</p>
<pre><code>rt_tddft
 ...
 dt 0.01
 ...
end
</code></pre>
<p>It is always good practice to check that your results are independent of
time step.</p>
<h3 id="rt-tddft-tag-output-label">TAG: Output label<a class="headerlink" href="#rt-tddft-tag-output-label" title="Permanent link">&para;</a></h3>
<p>This option sets a label for the output for convenient parsing (e.g.,
with &ldquo;grep&rdquo;). Every output line with time-dependent data will begin with
this string (set to <code>&lt;rt_tddft&gt;:</code> by default). For example setting:</p>
<pre><code>rt_tddft
 ...
 tag &quot;nameofrun&quot;
 ...
end
</code></pre>
<p>will result in outputs that look like:</p>
<pre><code>...
nameofrun      2.20000  -7.589146713114E+001     # Etot
...
</code></pre>
<h3 id="rt-tddft-nchecks-number-of-run-time-check-points">NCHECKS: Number of run-time check points<a class="headerlink" href="#rt-tddft-nchecks-number-of-run-time-check-points" title="Permanent link">&para;</a></h3>
<p>This option takes an integer number (default 10), or <code>*</code> which means
every step, which sets the total of number of run-time checkpoints,
where various sanity checks are made such as symmetries, idempotency,
traces, etc. These checks are not terribly efficient (e.g., involve
re-building the TD Fock matrix) so excessive checking will slow down
execution.</p>
<pre><code>rt_tddft
  ...
  nchecks 10
  ...
end
</code></pre>
<h3 id="rt-tddft-nprints-number-of-print-points">NPRINTS: Number of print points<a class="headerlink" href="#rt-tddft-nprints-number-of-print-points" title="Permanent link">&para;</a></h3>
<p>This sets the number of print points, i.e., the total number of
time-dependent observables (e.g., dipole, charge, energy) that are
computed and printed. It either takes an integer number or <code>*</code> which
means every time step (this is the default). Since there is no
appreciable cost to computing and printing these quantities, there is
usually no need to change this from <code>*</code>.</p>
<pre><code>rt_tddft
  ...
  nprints *
  ...
end
</code></pre>
<h3 id="rt-tddft-nrestarts-number-of-restart-checkpoints">NRESTARTS: Number of restart checkpoints<a class="headerlink" href="#rt-tddft-nrestarts-number-of-restart-checkpoints" title="Permanent link">&para;</a></h3>
<p>This sets the number of run-time check points where the time-dependent
complex density matrix is saved to file, allowing the simulation to be
<a href="#rt-tddft-load-restart">restarted</a>) from that point. By
default this is set to 0. There is no significant computational cost to
restart checkpointing, but of course there is some disk I/O cost (which
may become somewhat significant for larger systems). For example, in the
following example there will be 100 restart points, which corresponds to
1 backup every 100 time steps.</p>
<pre><code>rt_tddft
  ...
  tmax 1000.0
  dt 0.1
  nrestarts 100
  ...
end
</code></pre>
<h3 id="rt-tddft-tolerances-controlling-numerical-tolerances">TOLERANCES: Controlling numerical tolerances<a class="headerlink" href="#rt-tddft-tolerances-controlling-numerical-tolerances" title="Permanent link">&para;</a></h3>
<p>This option controls various numerical tolerances:</p>
<ul>
<li><code>zero</code>: threshold for checks that quantities are zero, e.g., in
    symmetry checks (default 1e-8)</li>
<li><code>series</code>: numerical convergence for series, e.g., matrix
    exponentiation (default 1e-10)</li>
<li><code>interpol</code>: numerical convergence for interpolation, e.g., in Magnus
    propagator (default 1e-7)</li>
</ul>
<p>Occasionally it is useful to loosen the interpolation tolerances if the
Magnus interpolator requires an excessive amount of steps; usually this
will not impact accuracy. For example, this sets the tolerances to their
defaults with a looser interpolation tolerance:</p>
<pre><code>rt_tddft
  ...
  tolerances zero 1d-8 series 1d-10 interpol 1e-5
end
</code></pre>
<h3 id="rt-tddft-propagator-selecting-the-integrator-method">PROPAGATOR: Selecting the integrator method<a class="headerlink" href="#rt-tddft-propagator-selecting-the-integrator-method" title="Permanent link">&para;</a></h3>
<p>This selects the propagator (i.e., time integrator) method. Possible
choices include <code>euler</code> for Euler integration (terrible, you should
never use this), <code>rk4</code> for 4th order Runge-Kutta, and <code>magnus</code> for 2nd
order Magnus with self-consistent interpolation. In virtually all cases
Magnus is superior in terms of stability. Euler or rk4 are perhaps only
useful for debugging or simplicity (e.g., for code development).</p>
<pre><code>rt_tddft
  ...
  propagator magnus
  ...
end
</code></pre>
<h3 id="rt-tddft-exp-selecting-the-matrix-exponentiation-method">EXP: Selecting the matrix exponentiation method<a class="headerlink" href="#rt-tddft-exp-selecting-the-matrix-exponentiation-method" title="Permanent link">&para;</a></h3>
<p>This selects the method for exponentiation matrices. For now this can
either be <code>pseries</code> for a contractive power series or <code>diag</code> for
diagonalization. In general the power series (default) is faster.</p>
<pre><code>rt_tddft
  ...
  exp diag
  ...
end
</code></pre>
<h3 id="rt-tddft-prof-run-time-profiling">PROF: Run-time profiling<a class="headerlink" href="#rt-tddft-prof-run-time-profiling" title="Permanent link">&para;</a></h3>
<p>This turns on run-time profiling, which will display time spent in each
component of the code (e.g., building components of the TD Fock matrix,
properites, etc). This slows code down slightly and results in very
verbose output.</p>
<pre><code>rt_tddft
  ...
  prof
  ...
end
</code></pre>
<h3 id="rt-tddft-noprop-skipping-propagation">NOPROP: Skipping propagation<a class="headerlink" href="#rt-tddft-noprop-skipping-propagation" title="Permanent link">&para;</a></h3>
<p>This options causes the RT-TDDFT module skip propagation, i.e., to
initialize and finalize. For now this is largely useful for skipping to
visualization post-processing without having to re-run a simulation.</p>
<pre><code>rt_tddft
  ...
  noprop
  ...
end
</code></pre>
<h3 id="rt-tddft-static-force-static-fock-matrix">STATIC: Force static Fock matrix<a class="headerlink" href="#rt-tddft-static-force-static-fock-matrix" title="Permanent link">&para;</a></h3>
<p>This option sets the static Fock matrix flag, meaning the time-dependent
Fock matrix will not be recalculated at each time, but instead use the
<em>t=0</em> value. This will drastically increase the simulation speed since the
bulk of the work is spent rebuilding the TD Fock matrix, but will give
non-physical results. For example, using <code>static</code> to compute an
absorption spectrum will result in excitations corresponding to the raw
eigenvalue differences without electron-hole relaxation. This option has
few uses besides dry-runs and debugging.</p>
<pre><code>rt_tddft
  ...
  static
  ...
end
</code></pre>
<h3 id="rt-tddft-print-selecting-time-dependent-quantities-to-be-printed">PRINT: Selecting time-dependent quantities to be printed<a class="headerlink" href="#rt-tddft-print-selecting-time-dependent-quantities-to-be-printed" title="Permanent link">&para;</a></h3>
<p>This sets the various time-dependent properties that are to be computed
and printed at each print point. Note that for many of these options,
the values are computed and printed for each geometry specified in the
input deck, not only the active one (i.e., the one set using <code>set
geometry ...</code> in the input deck). Possible choices are:</p>
<ul>
<li><code>dipole</code>: Dipole moment</li>
<li><code>quadrupole</code>: Quadrupole moment</li>
<li><code>field</code>: External (applied) electric field</li>
<li><code>moocc</code>: Molecular orbital occupations</li>
<li><code>energy</code>: Components of system energy (e.g., core, XC, total, etc)</li>
<li><code>cputime</code>: CPU time taken in simulation so far (useful for checking
    scaling)</li>
<li><code>charge</code>: Electronic charge (computed from density matrix, not from
    the XC grid)</li>
<li><code>convergence</code>: Convergence information (e.g., from Magnus)</li>
<li><code>s2</code>: <S2> value (openshell only)</li>
<li><code>*</code>: Print all quantities</li>
</ul>
<p>The defaults correspond to:</p>
<pre><code>rt_tddft
  ...
  print dipole field energy convergence
  ...
end
</code></pre>
<h3 id="rt-tddft-field-sub-block-for-specifying-external-electric-fields">FIELD: Sub-block for specifying external electric fields<a class="headerlink" href="#rt-tddft-field-sub-block-for-specifying-external-electric-fields" title="Permanent link">&para;</a></h3>
<p>This sub-block is used to specify external electric fields, each of
which must be given a unique name. Numerous fields can be specified, but
each will applied to the system only if an appropriate <a href="#rt-tddft-excite-excitation-rules">excitation
rule</a> is set.
There are a few preset field types; others would have to be manually
coded. Note the names are arbitrary, but chosen here to be
descriptive:</p>
<pre><code>field &quot;kick&quot;
  type delta      # E(t=0) = max; E(t&gt;0) = 0
  polarization x  # = x, y, z
  max 0.0001      # maximum value of electric field
  spin total      # = alpha, beta, total (only valid for open-shell)
end
</code></pre>
<pre><code>field &quot;gpulse&quot;
  type gaussian   # Gaussian enveloped quasi-monochromatic pulse: E(t) = max * exp( -(t-t0)^2 / 2s^2) * sin(w0*t + phi0)
  polarization x  # = x, y, z
  frequency 0.12  # frequency of laser pulse in au (e.g., 0.12 au = 3.27 eV)
  phase 0.0       # phase shift of laser pulse (in rad)
  center 200.0    # center of Gaussian envelope (in au time)
  width 50.0      # width of Gaussian pulse (in au time)
  max 0.0001      # maximum value of electric field
  spin total      # = alpha, beta, total (only valid for open-shell)
end
</code></pre>
<pre><code>field &quot;hpulse&quot;
  type hann       # sin^2 (Hann) enveloped quasi-monochromatic pulse
  polarization x  # = x, y, z
  frequency 0.12  # frequency of laser pulse in au (e.g., 0.12 au = 3.27 eV)
  center 200.0    # center of Hann envelope (in au time)
  width 50.0      # width of Hann pulse (in au time)
  max 0.0001      # maximum value of electric field
  spin total      # = alpha, beta, total (only valid for open-shell)
end
</code></pre>
<pre><code>field &quot;resonant&quot;
  type cw         # monochromatic continuous wave
  frequency 0.12  # frequency of laser pulse in au (e.g., 0.12 au = 3.27 eV)
  polarization x  # = x, y, z
  max 0.0001      # maximum value of electric field
  spin total      # = alpha, beta, total (only valid for open-shell)
end
</code></pre>
<h3 id="rt-tddft-excite-excitation-rules">EXCITE: Excitation rules<a class="headerlink" href="#rt-tddft-excite-excitation-rules" title="Permanent link">&para;</a></h3>
<p>This sets the rules for applying external fields to the system. It takes
the form <code>excite &lt;geom&gt; with &lt;field&gt;</code>, where <code>&lt;geom&gt;</code> is the name of a
geometry fragment (defaults to &ldquo;geometry&rdquo; which is the default geometry
name), and <code>&lt;field&gt;</code> is the name of a <a href="#rt-tddft-field-sub-block-for-specifying-external-electric-fields">field
structure</a>.
Assuming, for example, you have defined a field name <code>kick</code> this option
takes the form (note that quotes are optional and shown for clarity):</p>
<pre><code>rt_tddft
  ...
  excite &quot;geometry&quot; with &quot;kick&quot;
  ...
end
</code></pre>
<h3 id="rt-tddft-visualization-sub-block-for-controlling-3d-visualization">VISUALIZATION: Sub-block for controlling 3D visualization<a class="headerlink" href="#rt-tddft-visualization-sub-block-for-controlling-3d-visualization" title="Permanent link">&para;</a></h3>
<p>This block is used to control visualization of the electronic charge
density, which is typically used during a resonant excitation. This is a
two stage process. During the propagation, a series of density matrices
will be dumped to file (see options below). After propagation, if the
<code>dplot</code> option is set, the code will read in options from a separate
<code>DPLOT</code> block and convert the density matrix snapshots to a corresponding
series of real-space charge density <code>cube</code>
files.</p>
<pre><code>visualization
  tstart 0.0        # start visualization at this time
  tend 100.0        # stop visualization at this time
  treference 0.0    # subtract density matrix at this time (useful for difference densities)
  dplot             # post-process density matrices into cube files after propagation
end
</code></pre>
<h3 id="rt-tddft-load-restart">LOAD RESTART<a class="headerlink" href="#rt-tddft-load-restart" title="Permanent link">&para;</a></h3>
<p>This keyword needs to be added to <a href="#start_restart">restart</a> a calculation. In the following example, the calculation will restart from the previous calculation and extend the run to the new <code>tmax</code></p>
<pre><code>rt_tddft
  ...
  tmax 10.0        # new end time
  load restart
end
</code></pre>
<h3 id="rt-tddft-mocap-sub-block-for-molecular-orbital-complex-absorbing-potential">MOCAP: Sub-block for molecular orbital complex absorbing potential<a class="headerlink" href="#rt-tddft-mocap-sub-block-for-molecular-orbital-complex-absorbing-potential" title="Permanent link">&para;</a></h3>
<p>The
K. Lopata and N. Govind,
&ldquo;Near and Above Ionization Electronic Excitations with Non-Hermitian
Real-Time Time-Dependent Density Functional Theory&rdquo;,
Journal of Chemical Theory and Computation 9 (11), 4939-4946 (2013)
    <a href="https://dx.doi.org/10.1021/ct400569s">DOI:10.1021/ct400569s</a></p>
<pre><code>  mocap
    maxval 100.0     # clamp CAP at this value (in Ha)
    emin 0.5         # any MO with eigenvalue &gt;= 0.5 Ha will have CAP applied to it
    prefac 1.0       # prefactor for exponential
    expconst 1.0     # exponential constant for CAP
    on               # turn on/off CAP
    nochecks         # disable checks for speed
    noprint          # don't print CAP value
  end
</code></pre>
<h4 id="rt-tddft-maxval-exponential-maximum-value">MAXVAL: Exponential Maximum Value<a class="headerlink" href="#rt-tddft-maxval-exponential-maximum-value" title="Permanent link">&para;</a></h4>
<h4 id="rt-tddft-emin-vacuum-energy-level">EMIN: Vacuum Energy Level<a class="headerlink" href="#rt-tddft-emin-vacuum-energy-level" title="Permanent link">&para;</a></h4>
<h4 id="rt-tddft-onoff-turn-onoff-cap">ON/OFF: Turn on/off CAP<a class="headerlink" href="#rt-tddft-onoff-turn-onoff-cap" title="Permanent link">&para;</a></h4>
<p>The default value is <code>off</code>, i.e. no complex absorbing potential is applied.</p>
<h2 id="rt-tddft-worked-examples">Worked Examples<a class="headerlink" href="#rt-tddft-worked-examples" title="Permanent link">&para;</a></h2>
<h3 id="rt-tddft-absorption-spectrum-of-water">Absorption spectrum of water<a class="headerlink" href="#rt-tddft-absorption-spectrum-of-water" title="Permanent link">&para;</a></h3>
<p>Here, we compute the 6-31G/TD-PBE0 absorption spectrum of gas-phase
water using RT-TDDFT. In the weak-field limit, these results are
essentially identical to those obtained via traditional linear-response
TDDFT. Although it involves significantly more work do use RT-TDDFT in
this case, for very large systems with many roots a real-time approach
becomes advantageous computationally and also does not suffer from
algorithm stability issues.</p>
<p>To compute the absorption, we find the ground state of the system and
then subject it to three delta-function excitations (x,y,z), which
simultaneously excites all electronic modes of that polarization. The
three resulting dipole moments are then Fourier transformed to give the
frequency-dependent linear polarizability, and thus the absorption
spectrum. The full input deck is
<a href="RT_TDDFT_h2o_abs.nw">RT_TDDFT_h2o_abs.nw</a> and
the corresponding output is
<a href="RT_TDDFT_h2o_abs.nwo.gz">RT_TDDFT_h2o_abs.nwo.gz</a>.</p>
<pre><code>title &quot;Water TD-PBE0 absorption spectrum&quot;
echo


start water

##                                                                                                                                                                           
## aug-cc-pvtz / pbe0 optimized                                                                                                                                              
##                                                                                                                                                                           
## Note: you are required to explicitly name the geometry                                                                                                                    
##                                                                                                                                                                           
geometry &quot;system&quot; units angstroms nocenter noautoz noautosym
  O     0.00000000    -0.00001441    -0.34824012
  H    -0.00000000     0.76001092    -0.93285191
  H     0.00000000    -0.75999650    -0.93290797
end

## Note: We need to explicitly set the &quot;active&quot; geometry even though there is only one geom.                                                                                 
set geometry &quot;system&quot;

## All DFT and basis parameters are inherited by the RT-TDDFT code                                                                                                           
basis
 * library 6-31G
end

dft
  xc pbe0
end

## Compute ground state of the system                                                                                                                                        
task dft energy

##                                                                                                                                                                           
## Now, we compute an x, y, and z kick simulation, which we give separate &quot;tags&quot; for post-processing.                                                                  
##                                                                                                                                                                           

unset rt_tddft:*
rt_tddft
  tmax 200.0
  dt 0.2

  tag &quot;kick_x&quot;

  field &quot;kick&quot;
    type delta
    polarization x
    max 0.0001
  end

  excite &quot;system&quot; with &quot;kick&quot;
 end
task dft rt_tddft

unset rt_tddft:*
rt_tddft
  tmax 200.0
  dt 0.2

  tag &quot;kick_y&quot;

  field &quot;kick&quot;
    type delta
    polarization y
    max 0.0001
  end

  excite &quot;system&quot; with &quot;kick&quot;
 end
task dft rt_tddft

unset rt_tddft:*
rt_tddft
  tmax 200.0
  dt 0.2

  tag &quot;kick_z&quot;

  field &quot;kick&quot;
    type delta
    polarization z
    max 0.0001
  end

  excite &quot;system&quot; with &quot;kick&quot;
 end
task dft rt_tddft
</code></pre>
<p>After running the simulation, we extract the x-dipole moment for the
x-kick and similarly for the y and z-kicks (see <a href="https://github.com/nwchemgit/nwchem/tree/master/contrib/parsers">&ldquo;contrib/parsers&rdquo;
directory</a> for this script or download here:
<a href="RT_TDDFT_scripts.tgz">RT_TDDFT_scripts.tgz</a> ).</p>
<pre><code>nw_rtparse.py -xdipole -px -tkick_x h2o_abs.nwo &gt; x.dat
nw_rtparse.py -xdipole -py -tkick_y h2o_abs.nwo &gt; y.dat
nw_rtparse.py -xdipole -pz -tkick_z h2o_abs.nwo &gt; z.dat
</code></pre>
<p>Note, the syntax for extracting the x polarization for the x-kick, etc.
Alternatively, we could grep and cut, or whatnot. This will give use the
resulting time-dependent dipole moments:</p>
<p><img alt="Time-dependent dipole moments" src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/RT_TDDFT_h2o_td_dipoles.png" align=middle with="730pt" height="200pt"/>  </p>
<p>Now, we need to take the Fourier transforms of these dipole moments to
yield the the x,x element of the 3x3 linear polarizability tensor, and
similarly for the y,y and z,z elements. Here I am using an FFT utility,
although any discrete Fourier transform will do. To accelerate
convergence of the FFT, I have damped the time signals by
<em>exp(-t /&tau;)</em> which results in Lorentzians with FWHM of
<em>2 / &tau;</em> and have also &ldquo;zero padded&rdquo; the data with 50000 points.
This is not critical for extracting frequencies, but creates &ldquo;cleaner&rdquo;
spectra, although care must be taken to damp sufficiently if padding to
avoid artifacts (see small ripples around 23 eV in plot below). After
Fourier transform, I &ldquo;paste&rdquo; the three files together to easily plot the
absorption, which is constructed from the trace of the polarizability
matrix, i.e., the sum of the imaginary parts of the FFTs of the dipole
moments.</p>
<p><em>S (&omega;) = (4&pi;&omega;)/(3c&kappa;)</em>Tr[Im<em> &alpha;(&omega;)</em>]  </p>
<p>where <em>c</em> is the speed of light (137 in atomic units), <em>&kappa;</em>  is
the kick electric field strength, and <em>&alpha;(&omega;)</em> is the linear
polarizabilty tensor computed from the Fourier transforms of the
time-dependent dipole moments. For example,</p>
<pre><code>fft1d -d50 -z -p50000 &lt; x.dat | rotate_fft &gt; xw.dat
fft1d -d50 -z -p50000 &lt; y.dat | rotate_fft &gt; yw.dat
fft1d -d50 -z -p50000 &lt; z.dat | rotate_fft &gt; zw.dat
paste xw.dat yw.dat zw.day &gt; s.dat
</code></pre>
<p>Here, you can just use your favorite Fourier transform utility or
analysis software, but for convenience there is also a simple GNU Octave
fft1d.m utility in the &ldquo;contrib/parsers&rdquo; directory of the trunk or
download here:
<a href="RT_TDDFT_scripts.tgz">RT_TDDFT_scripts.tgz</a>
Note, options are hardcoded at the moment, so the switches above are not
correct instead edit the file and run (also it reads file rather than
redirect from stdin). Assuming the FFT output takes the form (w, Re, Im,
Abs), to plot using gnuplot we would do:</p>
<pre><code>gnuplot&gt; plot &quot;s.dat&quot; u ($1*27.2114):($1*($3+$7+$11))
</code></pre>
<p>where we have scaled by 27.2114 to output in eV instead of atomic units,
and we have not properly scaled to get the absolute oscillator strengths
(thus our magnitudes are in &ldquo;arbitrary units&rdquo;). The real-time spectrum
is shown below, along with the corresponding linear-response TDDFT
excitations are shown in orange for comparison. Since we are in the weak
field regime, the two are identical. Note the oscillator strengths are
arbitrary and scaled, if not scaled the area under each RT-TDDFT curve
should integrate to the linear response oscillator strength.</p>
<p><img alt="Linear absorption spectrum" src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/RT_TDDFT_h2o_abs_spectrum.png" align=middle with="262pt" height="200pt"/>  </p>
<h3 id="rt-tddft-resonant-ultraviolet-excitation-of-water">Resonant ultraviolet excitation of water<a class="headerlink" href="#rt-tddft-resonant-ultraviolet-excitation-of-water" title="Permanent link">&para;</a></h3>
<p>In this example we compute the time-dependent electron response to a
quasi-monochromatic laser pulse tuned to a particular transition. We
will use the results of the previous example (6-31G/PBE0 gas-phase
water). First, we consider the absorption spectrum (computed previously)
but plotted for the three polarizations (x,y,z) rather then as a sum.
Say we are interested in the excitation near 10 eV. We can clearly see
this is a z-polarized transition (green on curve). To selectively excite
this we could use a continuous wave E-field, which has a delta-function,
i.e., single frequency, bandwidth but since we are doing finite
simulations we need a suitable envelope. The broader the envelope in
time the narrower the excitation in frequency domain, but of course long
simulations become costly so we need to put some thought into the choice
of our envelope. In this case the peak of interest is spectrally
isolated from other z-polarized peaks, so this is quite straightforward.
The procedure is outlined below, and the corresponding frequency extent
of the pulse is shown on the absorption figure in orange. Note that it
only covers one excitation, i.e., the field selectively excites one
mode. The full input deck is
<a href="RT_TDDFT_h2o_resonant.nw">RT_TDDFT_h2o_resonant.nw</a>
and the output is
<a href="RT_TDDFT_h2o_resonant.nwo.gz">RT_TDDFT_h2o_resonant.nwo.gz</a>.</p>
<p><img alt="Absorption spectrum and excitation bandwidth" src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/RT_TDDFT_h2o_resonant_spec_field.png" align=middle with="244pt" height="200pt"/>  </p>
<pre><code>title &quot;Water TD-PBE0 resonant excitation&quot; 
echo
scratch_dir ./scratch
permanent_dir ./perm

start water

##
## aug-cc-pvtz / pbe0 optimized
##
## Note: you are required to explicitly name the geometry
##
geometry &quot;system&quot; units angstroms nocenter noautoz noautosym
  O     0.00000000    -0.00001441    -0.34824012
  H    -0.00000000     0.76001092    -0.93285191
  H     0.00000000    -0.75999650    -0.93290797
end

## Note: We need to explicitly set the &quot;active&quot; geometry even though there is only one geom.
set geometry &quot;system&quot; 

## All DFT and basis parameters are inherited by the RT-TDDFT code
basis
 * library 6-31G
end

dft
 xc pbe0
end

## Compute ground state of the system
task dft energy

##
## We excite the system with a quasi-monochromatic
## (Gaussian-enveloped) z-polarized E-field tuned to a transition at
## 10.25 eV.  The envelope takes the form:
##
## G(t) = exp(-(t-t0)^2 / 2s^2)
##
## The target excitation has an energy (frequency) of w = 0.3768 au
## and thus an oscillation period of T = 2 pi / w = 16.68 au
##
## Since we are doing a Gaussian envelope in time, we will get a
## Gaussian envelope in frequency (Gaussians are eigenfunctions of a
## Fourier transform), with width equal to the inverse of the width in
## time.  Say, we want a Gaussian in frequency with FWHM = 1 eV
## (recall FWHM = 2 sqrt (2ln2) s_freq) we want an s_freq = 0.42 eV =
## 0.0154 au, thus in time we need s_time = 1 / s_time = 64.8 au.
##
## Now we want the envelope to be effectively zero at t=0, say 1e-8
## (otherwise we get &quot;windowing&quot; effects).  Reordering G(t):
##
## t0 = t - sqrt(-2 s^2 ln G(t))
##
## That means our Gaussian needs to be centered at t0 = 393.3 au.
##
## The total simulation time will be 1000 au to leave lots of time to
## see oscillations after the field has passed.
##
rt_tddft
  tmax 1000.0
  dt 0.2

  field &quot;driver&quot;
    type gaussian
    polarization z
    frequency 0.3768  # = 10.25 eV
    center 393.3
    width 64.8
    max 0.0001
  end

  excite &quot;system&quot; with &quot;driver&quot;
 end
task dft rt_tddft
</code></pre>
<p><img alt="Time-dependent electric field and dipole moment during resonant excitation" src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/RT_TDDFT_h2o_resonant_td.png" align=middle with="268pt" height="200pt"/>  </p>
<p>From the time-dependent dipole moment you can see the field driving the
system into a superposition of the ground state and the one excited
state, which manifests as monochromatic oscillations. After the field
has passed the dipole oscillations continue forever as there is no
damping in the system.</p>
<h3 id="rt-tddft-charge-transfer-between-a-tcne-dimer">Charge transfer between a TCNE dimer<a class="headerlink" href="#rt-tddft-charge-transfer-between-a-tcne-dimer" title="Permanent link">&para;</a></h3>
<p>Here we compute the time-dependent charge oscillations between a TCNE
(tetracyanoethylene) dimer separated by 3 Angstroms, where the top
molecule starts neutral and the bottom one starts with a -1 charge. This
somewhat non-physical starting point will lead to far-from-equilibrium
dynamics as the charge violently sloshes between the two molecules, with
the oscillation period a function of the molecular separation. The trick
here is to use fragments by have multiple geometries in the input deck,
where each fragment is converged separately, then assembled together
without SCF to use as a starting point. We use a small but diffuse basis
and a range-separated functional (CAM-B3LYP). The input deck is
<a href="RT_TDDFT_tcne_dimer.nw">RT_TDDFT_tcne_dimer.nw</a>
and the full output is
<a href="RT_TDDFT_tcne_dimer.nwo.gz">RT_TDDFT_tcne_dimer.nwo.gz</a>.</p>
<pre><code>title &quot;Tetracyanoethylene dimer charge transfer&quot;

echo
scratch_dir ./scratch
permanent_dir ./perm

start tcne
echo

##
## Each fragment optimized with cc-pvdz/B3LYP
##
geometry &quot;bottom&quot; units angstroms noautosym nocenter noautoz
 C    -1.77576486     0.66496556     0.00004199
 N    -2.94676621     0.71379797     0.00004388
 C    -0.36046718     0.62491168     0.00003506
 C     0.36049301    -0.62492429    -0.00004895
 C     1.77579907    -0.66504145    -0.00006082
 N     2.94680364    -0.71382258    -0.00006592
 C    -0.31262746    -1.87038951    -0.00011201
 N    -0.85519492    -2.90926164    -0.00016331
 C     0.31276207     1.87031662     0.00010870
 N     0.85498782     2.90938919     0.00016857
end

geometry &quot;top&quot; units angstroms noautosym nocenter noautoz
 C    -1.77576486     0.66496556     3.00004199
 N    -2.94676621     0.71379797     3.00004388
 C    -0.36046718     0.62491168     3.00003506
 C     0.36049301    -0.62492429     2.99995105
 C     1.77579907    -0.66504145     2.99993918
 N     2.94680364    -0.71382258     2.99993408
 C    -0.31262746    -1.87038951     2.99988799
 N    -0.85519492    -2.90926164     2.99983669
 C     0.31276207     1.87031662     3.00010870
 N     0.85498782     2.90938919     3.00016857
end


## dimer geometry is the union of bottom and top geometry
geometry &quot;dimer&quot; units angstroms noautosym nocenter noautoz
 C    -1.77576486     0.66496556     0.00004199
 N    -2.94676621     0.71379797     0.00004388
 C    -0.36046718     0.62491168     0.00003506
 C     0.36049301    -0.62492429    -0.00004895
 C     1.77579907    -0.66504145    -0.00006082
 N     2.94680364    -0.71382258    -0.00006592
 C    -0.31262746    -1.87038951    -0.00011201
 N    -0.85519492    -2.90926164    -0.00016331
 C     0.31276207     1.87031662     0.00010870
 N     0.85498782     2.90938919     0.00016857
#---
 C    -1.77576486     0.66496556     3.00004199
 N    -2.94676621     0.71379797     3.00004388
 C    -0.36046718     0.62491168     3.00003506
 C     0.36049301    -0.62492429     2.99995105
 C     1.77579907    -0.66504145     2.99993918
 N     2.94680364    -0.71382258     2.99993408
 C    -0.31262746    -1.87038951     2.99988799
 N    -0.85519492    -2.90926164     2.99983669
 C     0.31276207     1.87031662     3.00010870
 N     0.85498782     2.90938919     3.00016857
end


##
## C, N: 3-21++G
##
basis spherical
C    S
    172.2560000              0.0617669        
     25.9109000              0.3587940        
      5.5333500              0.7007130        
C    SP
      3.6649800             -0.3958970              0.2364600        
      0.7705450              1.2158400              0.8606190        
C    SP
      0.1958570              1.0000000              1.0000000        
C    SP
      0.0438000              1.0000000              1.0000000        
N    S
    242.7660000              0.0598657        
     36.4851000              0.3529550        
      7.8144900              0.7065130        
N    SP
      5.4252200             -0.4133010              0.2379720        
      1.1491500              1.2244200              0.8589530        
N    SP
      0.2832050              1.0000000              1.0000000        
N    SP
      0.0639000              1.0000000              1.0000000        
end


##
## Charge density fitting basis.
##
basis &quot;cd basis&quot;
C    S
      5.91553927E+02         0.31582020
      1.72117940E+02         0.87503863
      5.47992590E+01         2.30760524
C    S
      1.89590940E+01         1.0000000
C    S
      7.05993000E+00         1.0000000
C    S
      2.79484900E+00         1.0000000
C    S
      1.15863400E+00         1.0000000
C    S
      4.94324000E-01         1.0000000
C    S
      2.12969000E-01         1.0000000
C    P
      3.27847358E-01         1.0000000
C    P
      7.86833659E-01         1.0000000
C    P
      1.97101832E+00         1.0000000
C    D
      4.01330100E+00         1.0000000
C    D
      1.24750500E+00         1.0000000
C    D
      4.08148000E-01         1.0000000
C    F
      9.00000000E-01         1.0000000
N    S
      7.91076935E+02         0.41567506
      2.29450184E+02         1.14750694
      7.28869600E+01         3.01935767
N    S
      2.51815960E+01         1.0000000
N    S
      9.37169700E+00         1.0000000
N    S
      3.71065500E+00         1.0000000
N    S
      1.53946300E+00         1.0000000
N    S
      6.57553000E-01         1.0000000
N    S
      2.83654000E-01         1.0000000
N    P
      4.70739194E-01         1.0000000
N    P
      1.12977407E+00         1.0000000
N    P
      2.83008403E+00         1.0000000
N    D
      5.83298650E+00         1.0000000
N    D
      1.73268650E+00         1.0000000
N    D
      5.45242500E-01         1.0000000
N    F
      1.82648000E+00         1.0000000
end


##
## Universal DFT parameters. Note, we are doing open-shell even for
## the neutral fragment so the movecs have the correct size. 
##
## We are using the CAM-B3LYP functional (no need to use &quot;direct&quot;
## since we are doing CD fitting).
##
dft
  xc xcamb88 1.00 lyp 0.81 vwn_5 0.19 hfexch 1.00
  cam 0.33 cam_alpha 0.19 cam_beta 0.46
  odft
  convergence density 1d-9
  grid fine
  maxiter 1000
end


##
## Converge bottom fragment with extra electron and top fragment as
## neutral.
##
charge -1
set geometry &quot;bottom&quot;
dft
  mult 2
  vectors input atomic output &quot;bottom.movecs&quot;
end
task dft energy

charge 0
set geometry &quot;top&quot;
dft
  mult 1
  vectors input atomic output &quot;top.movecs&quot;
end
task dft energy


##
## Assemble the two fragments but don't do SCF--this keeps the system
## in a far-from-equilibrium state from which we will watch the
## dynamics.
##
charge -1
set geometry &quot;dimer&quot;
dft
  mult 2
  vectors input fragment &quot;bottom.movecs&quot; &quot;top.movecs&quot; output &quot;dimer.movecs&quot;
  noscf
end
task dft energy


##
## Now do RT-TDDFT from this crazy state without any electric fields.
##
rt_tddft
  tmax 500.0
  dt 0.2
  load vectors &quot;dimer.movecs&quot;

  print dipole field energy s2 charge
end
task dft rt_tddft
</code></pre>
<p><img alt="Time dependent charge oscillation between a TCNE dimer" src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/RT_TDDFT_tcne_td_charge.png" align=middle with="364pt" height="200pt"/>  </p>
<p>The time-dependent charge shows that the excess electron starts on the
&ldquo;bottom&rdquo; molecule (i.e., a total electronic charge of -65), then swings
to completely occupy the &ldquo;top&rdquo; molecule then oscillates back and forth.
The frequency of this oscillation is dependent on the separation, with
larger separations leading to lower frequencies. It is important to
note, however, this starting point is highly non-physical, specifically
converging the two fragments together and &ldquo;gluing&rdquo; them together
introduces an indeterminate amount of energy to the system, but this
simulation shows how charge dynamics simulations can be done.</p>
<h3 id="rt-tddft-mo-cap-example">MO CAP example<a class="headerlink" href="#rt-tddft-mo-cap-example" title="Permanent link">&para;</a></h3>
<pre><code>## aug-cc-pvtz/PBE0 optimized
geometry &quot;system&quot; units angstroms noautosym noautoz nocenter
  O     0.00000043     0.11188833     0.00000000
  H     0.76000350    -0.47275229     0.00000000
  H    -0.76000393    -0.47275063     0.00000000
end

set geometry &quot;system&quot;

basis spherical
  * library 6-31G*
end

dft
  xc pbe0
  convergence density 1d-9
end
task dft

rt_tddft
  dt 0.2
  tmax 250.0

  print dipole field energy charge

  mocap
    expconst 1.0     # exponential constant for CAP
    emin 0.5         # any MO with eigenvalue &gt;= 0.5 Ha will have CAP applied to it
    prefac 1.0       # prefactor for exponential
    maxval 100.0     # clamp CAP at this value (in Ha)
    on               # turn on CAP
    nochecks         # disable checks for speed
    noprint          # don't print CAP value
  end

  field &quot;kick&quot;
    type delta
    max 0.0001
    polarization z
  end

  excite &quot;system&quot; with &quot;kick&quot;
end
task dft rt_tddft
</code></pre>
<p>After running the parser script on the output files from the input above using either with the <code>on</code> or <code>off</code> keywords, the following plot can be produced from the data file obtained with the command <code>nw_rtparse.py   -xdipole -pz -t"&lt;rt_tddft&gt;"</code>
<img alt="Time dependent dipole with and without CAP" src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/cap_dipole.png" align=middle with="364pt" height="275pt"/></p></section><section class="print-page" id="plane-wave-density-functional-theory" heading-number="11.5"><h1 id="plane-wave-density-functional-theory-pseudopotential-plane-wave-density-functional-theory-nwpw">Pseudopotential plane-wave density functional theory (NWPW)<a class="headerlink" href="#plane-wave-density-functional-theory-pseudopotential-plane-wave-density-functional-theory-nwpw" title="Permanent link">&para;</a></h1>
<div class="toc">
<ul>
<li><a href="#plane-wave-density-functional-theory-pseudopotential-plane-wave-density-functional-theory-nwpw">Pseudopotential plane-wave density functional theory (NWPW)</a><ul>
<li><a href="#plane-wave-density-functional-theory-overview">Overview</a></li>
<li><a href="#plane-wave-density-functional-theory-pspw-tasks-gamma-point-calculations">PSPW Tasks: Gamma Point Calculations</a><ul>
<li><a href="#plane-wave-density-functional-theory-paw-potentials">PAW Potentials</a><ul>
<li><a href="#plane-wave-density-functional-theory-paw-implementation-notes">PAW Implementation Notes</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-exchange-correlation-potentials">Exchange-Correlation Potentials</a><ul>
<li><a href="#plane-wave-density-functional-theory-dft-u-corrections">DFT + U Corrections</a></li>
<li><a href="#plane-wave-density-functional-theory-langreth-style-vdw-and-vdw-van-der-wall-functionals">Langreth style vdw and vdw van der Wall functionals</a></li>
<li><a href="#plane-wave-density-functional-theory-grimme-dispersion-corrections">Grimme Dispersion Corrections</a></li>
<li><a href="#plane-wave-density-functional-theory-using-exchange-correlation-potentials-available-in-the-dft-module">Using Exchange-Correlation Potentials Available in the DFT Module</a></li>
<li><a href="#plane-wave-density-functional-theory-exact-exchange">Exact Exchange</a></li>
<li><a href="#plane-wave-density-functional-theory-self-interaction-corrections">Self-Interaction Corrections</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-wannier">Wannier</a></li>
<li><a href="#plane-wave-density-functional-theory-mulliken-analysis">Mulliken Analysis</a></li>
<li><a href="#plane-wave-density-functional-theory-density-of-states">Density of States</a></li>
<li><a href="#plane-wave-density-functional-theory-projected-density-of-states">Projected Density of States</a></li>
<li><a href="#plane-wave-density-functional-theory-point-charge-analysis">Point Charge Analysis</a></li>
<li><a href="#plane-wave-density-functional-theory-pspw_dplot-generate-gaussian-cube-files">PSPW_DPLOT: Generate Gaussian Cube Files</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-band-tasks-multiple-k-point-calculations">Band Tasks: Multiple k-point Calculations</a><ul>
<li><a href="#plane-wave-density-functional-theory-brillouin-zone">Brillouin Zone</a><ul>
<li><a href="#plane-wave-density-functional-theory-band-structure-paths">Band Structure Paths</a></li>
<li><a href="#plane-wave-density-functional-theory-special-points-of-different-space-groups-conventional-cells">Special Points of Different Space Groups (Conventional Cells)</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-screened-exchange">Screened Exchange</a></li>
<li><a href="#plane-wave-density-functional-theory-density-of-states-and-projected-density-of-states">Density of States and Projected Density of States</a></li>
<li><a href="#plane-wave-density-functional-theory-two-component-wavefunctions-spin-orbit-zora">Two-Component Wavefunctions (Spin-Orbit ZORA)</a></li>
<li><a href="#plane-wave-density-functional-theory-band_dplot-generate-gaussian-cube-files">BAND_DPLOT: Generate Gaussian Cube Files</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-car-parrinello">Car-Parrinello</a><ul>
<li><a href="#plane-wave-density-functional-theory-adding-geometry-constraints-to-a-car-parrinello-simulation">Adding Geometry Constraints to a Car-Parrinello Simulation</a></li>
<li><a href="#plane-wave-density-functional-theory-car-parrinello-output-datafiles">Car-Parrinello Output Datafiles</a><ul>
<li><a href="#plane-wave-density-functional-theory-xyz-motion-file">XYZ motion file</a></li>
<li><a href="#plane-wave-density-functional-theory-ion_motion-motion-file">ION_MOTION motion file</a></li>
<li><a href="#plane-wave-density-functional-theory-emotion-motion-file">EMOTION motion file</a></li>
<li><a href="#plane-wave-density-functional-theory-hmotion-motion-file">HMOTION motion file</a></li>
<li><a href="#plane-wave-density-functional-theory-eigmotion-motion-file">EIGMOTION motion file</a></li>
<li><a href="#plane-wave-density-functional-theory-omotion-motion-file">OMOTION motion file</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-born-oppenheimer-molecular-dynamics">Born-Oppenheimer Molecular Dynamics</a></li>
<li><a href="#plane-wave-density-functional-theory-i-pi-socket-communication">i-PI Socket Communication</a></li>
<li><a href="#plane-wave-density-functional-theory-metropolis-monte-carlo">Metropolis Monte-Carlo</a></li>
<li><a href="#plane-wave-density-functional-theory-free-energy-simulations">Free Energy Simulations</a><ul>
<li><a href="#plane-wave-density-functional-theory-metadynamics">MetaDynamics</a><ul>
<li><a href="#plane-wave-density-functional-theory-input">Input</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-tamd-temperature-accelerated-molecular-dynamics">TAMD - Temperature Accelerated Molecular Dynamics</a><ul>
<li><a href="#plane-wave-density-functional-theory-input_1">Input</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-collective-variables">Collective Variables</a><ul>
<li><a href="#plane-wave-density-functional-theory-bond-distance-collective-variable">Bond Distance Collective Variable</a></li>
<li><a href="#plane-wave-density-functional-theory-angle-collective-variable">Angle Collective Variable</a></li>
<li><a href="#plane-wave-density-functional-theory-coordination-collective-variable">Coordination Collective Variable</a></li>
<li><a href="#plane-wave-density-functional-theory-n-plane-collective-variable">N-Plane Collective Variable</a></li>
<li><a href="#plane-wave-density-functional-theory-user-defined-collective-variable">User defined Collective Variable</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-extended-x-ray-absorption-fine-structure-exafs-integration-with-feff6l">Extended X-Ray Absorption Fine Structure (EXAFS) - Integration with FEFF6L</a></li>
<li><a href="#plane-wave-density-functional-theory-frozen-phonon-calculations">Frozen Phonon Calculations</a></li>
<li><a href="#plane-wave-density-functional-theory-steepest-descent">Steepest Descent</a></li>
<li><a href="#plane-wave-density-functional-theory-simulation-cell">Simulation Cell</a></li>
<li><a href="#plane-wave-density-functional-theory-unit-cell-optimization">Unit Cell Optimization</a></li>
<li><a href="#plane-wave-density-functional-theory-smear-fractional-occupation-of-the-molecular-orbitals">SMEAR - Fractional Occupation of the Molecular Orbitals</a></li>
<li><a href="#plane-wave-density-functional-theory-spin-penalty-functions">Spin Penalty Functions</a></li>
<li><a href="#plane-wave-density-functional-theory-aimdmm-qmmm">AIMD/MM (QM/MM)</a></li>
<li><a href="#plane-wave-density-functional-theory-psp_generator">PSP_GENERATOR</a><ul>
<li><a href="#plane-wave-density-functional-theory-atomic_filling-block">ATOMIC_FILLING Block</a></li>
<li><a href="#plane-wave-density-functional-theory-cutoff">CUTOFF</a></li>
<li><a href="#plane-wave-density-functional-theory-semicore_radius">SEMICORE_RADIUS</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-paw-tasks-legacy-implementation">PAW Tasks: Legacy Implementation</a></li>
<li><a href="#plane-wave-density-functional-theory-pseudopotential-and-paw-basis-libraries">Pseudopotential and PAW basis Libraries</a></li>
<li><a href="#plane-wave-density-functional-theory-nwpw-rtdb-entries-and-miscellaneous-datafiles">NWPW RTDB Entries and Miscellaneous DataFiles</a><ul>
<li><a href="#plane-wave-density-functional-theory-ion-positions">Ion Positions</a></li>
<li><a href="#plane-wave-density-functional-theory-ion-velocities">Ion Velocities</a></li>
<li><a href="#plane-wave-density-functional-theory-wavefunction-datafile">Wavefunction Datafile</a></li>
<li><a href="#plane-wave-density-functional-theory-velocity-wavefunction-datafile">Velocity Wavefunction Datafile</a></li>
<li><a href="#plane-wave-density-functional-theory-formatted-pseudopotential-datafile">Formatted Pseudopotential Datafile</a></li>
<li><a href="#plane-wave-density-functional-theory-one-dimensional-pseudopotential-datafile">One-Dimensional Pseudopotential Datafile</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-car-parrinello-scheme-for-ab-initio-molecular-dynamics">Car-Parrinello Scheme for Ab Initio Molecular Dynamics</a><ul>
<li><a href="#plane-wave-density-functional-theory-verlet-algorithm-for-integration">Verlet Algorithm for Integration</a></li>
<li><a href="#plane-wave-density-functional-theory-constant-temperature-simulations-nose-hoover-thermostats">Constant Temperature Simulations: Nose-Hoover Thermostats</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-nwpw-tutorial-1-s2-dimer-examples-with-pspw">NWPW Tutorial 1: S2 dimer examples with PSPW</a><ul>
<li><a href="#plane-wave-density-functional-theory-total-energy-of-s2-dimer-with-lda-approximation">Total energy of S2 dimer with LDA approximation</a></li>
<li><a href="#plane-wave-density-functional-theory-structural-optimization-of-s2-dimer-with-lda-approximation">Structural optimization of S2 dimer with LDA approximation</a></li>
<li><a href="#plane-wave-density-functional-theory-frequency-calculation-of-s2-dimer-with-lda-approximation">Frequency calculation of S2 dimer with LDA approximation</a></li>
<li><a href="#plane-wave-density-functional-theory-ab-initio-molecular-dynamics-simulation-car-parrinello-of-s2-dimer-using-the-lda-approximation">Ab initio molecular dynamics simulation (Car-Parrinello) of S2 dimer using the LDA approximation</a></li>
<li><a href="#plane-wave-density-functional-theory-ab-initio-molecular-dynamics-simulation-born-oppenheimer-of-ssub2-dimer-using-the-lda-approximation">Ab initio molecular dynamics simulation (Born-Oppenheimer) of S&lt;sub2 dimer using the LDA approximation</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-nwpw-tutorial-2-using-pspw-car-parrinello-simulated-annealing-simulations-to-optimize-structures">NWPW Tutorial 2: Using PSPW Car-Parrinello Simulated Annealing Simulations to Optimize Structures</a><ul>
<li><a href="#plane-wave-density-functional-theory-simulated-annealing-using-constant-energy-simulation">Simulated Annealing Using Constant Energy Simulation</a></li>
<li><a href="#plane-wave-density-functional-theory-simulated-annealing-using-constant-temperature-simulation">Simulated Annealing Using Constant Temperature Simulation</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-nwpw-tutorial-3-using-isodesmic-reaction-energies-to-estimate-gas-phase-thermodynamics">NWPW Tutorial 3: using isodesmic reaction energies to estimate gas-phase thermodynamics</a></li>
<li><a href="#plane-wave-density-functional-theory-nwpw-tutorial-4-aimdmm-simulation-of-ccl_4-64-h_2o">NWPW Tutorial 4: AIMD/MM simulation of CCl\(_4\) + 64 H\(_2\)O</a></li>
<li><a href="#plane-wave-density-functional-theory-nwpw-tutorial-5-optimizing-the-unit-cell-and-geometry-of-diamond">NWPW Tutorial 5: Optimizing the Unit Cell and Geometry of Diamond</a><ul>
<li><a href="#plane-wave-density-functional-theory-optimizing-the-unit-cell-and-geometry-for-an-8-atom-supercell-of-diamond-with-pspw">Optimizing the Unit Cell and Geometry for an 8 Atom Supercell of Diamond with PSPW</a></li>
<li><a href="#plane-wave-density-functional-theory-optimizing-the-unit-cell-for-an-8-atom-supercell-of-diamond-with-band">Optimizing the Unit Cell for an 8 Atom Supercell of Diamond with BAND</a></li>
<li><a href="#plane-wave-density-functional-theory-using-band-to-optimize-the-unit-cell-for-a-2-atom-primitive-cell-of-diamond">Using BAND to Optimize the Unit Cell for a 2 Atom Primitive Cell of Diamond</a></li>
<li><a href="#plane-wave-density-functional-theory-using-band-to-calculate-the-band-structures-of-diamond">Using BAND to Calculate the Band Structures of Diamond</a></li>
<li><a href="#plane-wave-density-functional-theory-using-band-to-calculate-the-density-of-states-of-diamond">Using BAND to Calculate the Density of States of Diamond</a></li>
<li><a href="#plane-wave-density-functional-theory-calculate-the-phonon-spectrum-of-diamond">Calculate the Phonon Spectrum of Diamond</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-nwpw-tutorial-6-optimizing-the-unit-cell-of-nickel-with-fractional-occupation">NWPW Tutorial 6: optimizing the unit cell of nickel with fractional occupation</a></li>
<li><a href="#plane-wave-density-functional-theory-nwpw-tutorial-7-optimizing-the-unit-cells-with-symmetry-diamond-with-fd-3m-symmetry-and-brucite-with-p-3m1-symmetry">NWPW Tutorial 7: Optimizing the unit cells with symmetry: Diamond with Fd-3m symmetry and Brucite with P-3m1 symmetry</a></li>
<li><a href="#plane-wave-density-functional-theory-nwpw-tutorial-8-nvt-metropolis-monte-carlo-simulations">NWPW Tutorial 8: NVT Metropolis Monte-Carlo Simulations</a></li>
<li><a href="#plane-wave-density-functional-theory-nwpw-tutorial-9-npt-metropolis-monte-carlo-simulations">NWPW Tutorial 9: NPT Metropolis Monte-Carlo Simulations</a></li>
<li><a href="#plane-wave-density-functional-theory-nwpw-tutorial-9-free-energy-simulations">NWPW Tutorial 9: Free Energy Simulations</a></li>
<li><a href="#plane-wave-density-functional-theory-paw-tutorial">PAW Tutorial</a><ul>
<li><a href="#plane-wave-density-functional-theory-optimizing-a-water-molecule">Optimizing a water molecule</a></li>
<li><a href="#plane-wave-density-functional-theory-optimizing-a-unit-cell-and-geometry-for-silicon-carbide">Optimizing a unit cell and geometry for Silicon-Carbide</a></li>
<li><a href="#plane-wave-density-functional-theory-running-a-car-parrinello-simulation">Running a Car-Parrinello Simulation</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-nwpw-capabilities-and-limitations">NWPW Capabilities and Limitations</a></li>
<li><a href="#plane-wave-density-functional-theory-questions-and-difficulties">Questions and Difficulties</a></li>
</ul>
</li>
</ul>
</div>
<h2 id="plane-wave-density-functional-theory-overview">Overview<a class="headerlink" href="#plane-wave-density-functional-theory-overview" title="Permanent link">&para;</a></h2>
<p>The NWChem plane-wave (NWPW) module uses pseudopotentials and plane-wave
basis sets to perform Density Functional Theory calculations (simple
introduction <a href="pw-lecture.pdf">pw-lecture.pdf</a>).
This module complements the capabilities of the more traditional
Gaussian function based approaches by having an accuracy at least as
good for many applications, yet is still fast enough to treat systems
containing hundreds of atoms. Another significant advantage is its
ability to simulate dynamics on a ground state potential surface
directly at run-time using the Car-Parrinello algorithm. This method&rsquo;s
efficiency and accuracy make it a desirable first principles method of
simulation in the study of complex molecular, liquid, and solid state
systems. Applications for this first principles method include the
calculation of free energies, search for global minima, explicit
simulation of solvated molecules, and simulations of complex vibrational
modes that cannot be described within the harmonic approximation.</p>
<p>The NWPW module is a collection of three modules.</p>
<ul>
<li>PSPW - (PSeudopotential Plane-Wave) A gamma point code for
    calculating molecules, liquids, crystals, and surfaces.</li>
<li>Band - A band structure code for calculating crystals and surfaces
    with small band gaps (e.g. semi-conductors and metals).</li>
<li>PAW - a (gamma point) projector augmented plane-wave code for
    calculating molecules, crystals, and surfaces
    (<span style="color: red;"> <em>This module will be deprecated in the
    future releases since PAW potentials have been added to PSPW</em>
    </span>)</li>
</ul>
<p>The PSPW, Band, and PAW modules can be used to compute the energy and
optimize the geometry. Both the PSPW and Band modules can also be used
to find saddle points, and compute numerical second derivatives. In
addition the PSPW module can also be used to perform Car-Parrinello
molecular dynamics. Section <a href="#plane-wave-density-functional-theory-pspw-tasks-gamma-point-calculations">PSPW Tasks</a>
describes the tasks contained within the PSPW module, section <a href="#plane-wave-density-functional-theory-band-tasks-multiple-k-point-calculations">Band
Tasks</a> describes the tasks contained within the
Band module, section <a href="#plane-wave-density-functional-theory-paw-tasks-legacy-implementation">PAW Tasks</a> describes the
tasks contained within the PAW module, and section <a href="#plane-wave-density-functional-theory-pseudopotential-and-paw-basis-libraries">Pseudopotential and
PAW basis
Libraries</a>
describes the pseudopotential library included with NWChem. The
datafiles used by the PSPW module are described in section <a href="#plane-wave-density-functional-theory-nwpw-rtdb-entries-and-miscellaneous-datafiles">NWPW RTDB
Entries and DataFiles</a>.
Car-Parrinello output data files are described in section
<a href="#plane-wave-density-functional-theory-car-parrinello-output-datafiles">Car-Parrinello Output
Datafiles</a>, and the
minimization and Car-Parrinello algorithms are described in section
<a href="#plane-wave-density-functional-theory-car-parrinello-scheme-for-ab-initio-molecular-dynamics">Car-Parrinello Scheme for Ab Initio Molecular
Dynamics</a>.
Examples of how to setup and run a PSPW geometry optimization, a
Car-Parrinello simulation, a band structure minimization, and a PAW
geometry optimization are presented at the end. Finally in section <a href="#plane-wave-density-functional-theory-nwpw-capabilities-and-limitations">NWPW
Capabilities and
Limitations</a> the
capabilities and limitations of the NWPW module are discussed.  </p>
<p><span style="color: red;"> <em>As of NWChem 6.6 to use PAW potentials the
user is recommended to use the implementation contained in the PSPW
module (see Sections ). PAW potentials are also being integrated into
the BAND module. Unfortunately, the porting to BAND was not completed
for the NWChem 6.6 release.</em> </span></p>
<p>If you are a first time user of this module it is recommended that you
skip the next five sections and proceed directly to the tutorials.</p>
<h2 id="plane-wave-density-functional-theory-pspw-tasks-gamma-point-calculations">PSPW Tasks: Gamma Point Calculations<a class="headerlink" href="#plane-wave-density-functional-theory-pspw-tasks-gamma-point-calculations" title="Permanent link">&para;</a></h2>
<p>All input to the PSPW Tasks is contained within the compound PSPW block,</p>
<pre><code>PSPW 
  ...
END
</code></pre>
<p>To perform an actual calculation a TASK PSPW directive is used (Section
<a href="#task">Task</a>).</p>
<p><code>TASK PSPW</code></p>
<p>In addition to the directives listed in
<a href="#task">Task</a>, i.e. </p>
<pre><code>TASK PSPW energy 
TASK PSPW gradient 
TASK PSPW optimize 
TASK PSPW saddle 
TASK PSPW freqencies
TASK PSPW vib 
</code></pre>
<p>there are additional directives that are specific to the
PSPW module, which are:</p>
<pre><code>TASK PSPW [Car-Parrinello             ||   
           Born-Oppenheimer           ||
           Metropolis                 ||
           pspw_et                    ||
           noit_energy                ||
           stress                     ||
           pspw_dplot                 ||  
           wannier                    ||
           expand_cell                || 
           exafs                      ||
           ionize                     ||
           lcao                       ||
           rdf                        ||
           aimd_properties            ||
           translate                  ||
           psp_generator              || 
           steepest_descent           ||  
           psp_formatter              || 
           wavefunction_initializer   ||  
           v_wavefunction_initializer ||  
           wavefunction_expander       ]
</code></pre>
<p>Once a user has specified a geometry, the PSPW module can be invoked
with no input directives (defaults invoked throughout). However, the
user will probably always specify the simulation cell used in the
computation, since the default simulation cell is not well suited for
most systems. There are sub-directives which allow for customized
application; those currently provided as options for the PSPW module
are:</p>
<pre><code>NWPW   
  SIMULATION_CELL            ... (see section [Simulation Cell](#simulation-cell)) END
  CELL_NAME &lt;string cell_name default 'cell_default'&gt;
  VECTORS [[input (&lt;string input_wavefunctions default file_prefix.movecs&gt;) || 
           [output(&lt;string output_wavefunctions default file_prefix.movecs&gt;)]] 
  XC (Vosko      || LDA          || PBE96         || revPBE       || PBEsol          ||   
      LDA-SIC    || LDA-SIC/2    || LDA-0.4SIC    || LDA-SIC/4    || LDA-0.2SIC    || 
      PBE96-SIC  || PBE96-SIC/2  || PBE96-0.4SIC  || PBE96-SIC/4  || PBE96-0.2SIC  || 
      revPBE-SIC || revPBE-SIC/2 || revPBE-0.4SIC || revPBE-SIC/4 || revPBE-0.2SIC || 
      PBE96-Grimme2  || PBE96-Grimme3  || PBE96-Grimme4  || BLYP-Grimme2   || BLYP-Grimme3    || BLYP-Grimme4 || 
      revPBE-Grimme2 || revPBE-Grimme3 || revPBE-Grimme4 || PBEsol-Grimme2 || PBEsol-Grimme3 || PBEsol-Grimme4 || 
      PBE0-Grimme2    || PBE0-Grimme3    || PBE0-Grimme4   || B3LYP-Grimme2  || B3LYP-Grimme3  || B3LYP-Grimme4 ||
      revPBE0-Grimme2 || revPBE0-Grimme3 || revPBE0-Grimme4 ||
      PBE0       || revPBE0      || HSE  || HF || default Vosko) 
  XC new ...(see section [Using Exchange-Correlation Potentials Available in the DFT Module](#Using_Exchange-Correlation_Potentials_Available_in_the_DFT_Module))
  DFT||ODFT||RESTRICTED||UNRESTRICTED 
  MULT &lt;integer mult default 1&gt;   
  CG 
  LMBFGS   
  SCF [Anderson|| simple || Broyden]  
      [CG || RMM-DIIS] 
      [density || potential]
      [ALPHA real alpha default 0.25]
      [Kerker real ekerk nodefault] 
      [ITERATIONS integer inner_iterations default 5]  
      [OUTER_ITERATIONS integer outer_iterations default 0]
  LOOP &lt;integer inner_iteration outer_iteration default 10 100&gt;  
  TOLERANCES &lt;real tole tolc default 1.0e-7 1.0e-7&gt; 
  FAKE_MASS &lt;real fake_mass default 400000.0&gt; 
  TIME_STEP &lt;real time_step default 5.8&gt; 
  EWALD_NCUT &lt;integer ncut default 1&gt; 
  EWALD_RCUT &lt;real rcut default (see input description)&gt;  
  CUTOFF  &lt;real cutoff&gt; 
  ENERGY_CUTOFF &lt;real ecut default (see input description)&gt; 
  WAVEFUNCTION_CUTOFF &lt;real wcut default (see input description)&gt;  
  ALLOW_TRANSLATION 
  TRANSLATION (ON || OFF)
  ROTATION (ON || OFF) 
  MULLIKEN [OFF]
  EFIELD   

  BO_STEPS &lt;integer bo_inner_iteration bo_outer_iteration default 10 100&gt; 
  MC_STEPS &lt;integer mc_inner_iteration mc_outer_iteration default 10 100&gt;
  BO_TIME_STEP  &lt;real bo_time_step default 5.0&gt; 
  BO_ALGORITHM [verlet|| velocity-verlet || leap-frog]
  BO_FAKE_MASS &lt;real bo_fake_mass default 500.0&gt; 

  SOCKET (UNIX || IPI_CLIENT) &lt;string socketname default (see input description)&gt;  

  MAPPING &lt;integer mapping default 1&gt;  
  NP_DIMENSIONS &lt;integer npi npj default -1 -1&gt;  
  CAR-PARRINELLO             ... (see section [Car-Parrinello](#car-parrinello-scheme-for-ab-initio-molecular-dynamics)) END 
  STEEPEST_DESCENT           ... (see section [Steepest Descent](#STEEPEST_DESCENT)) END
  DPLOT                      ... (see section [DPLOT](#DPLOT)) END 
  WANNIER                    ... (see section [Wannier](#Wannier)) END 
  PSP_GENERATOR              ... (see section [PSP Generator](#PSP_GENERATOR))) END   

  WAVEFUNCTION_INITIALIZER   ... (see section [Wavefunction Initializer](NWPW_RETIRED.md#WAVEFUNCTION_INITIALIZER) - retired) END  
  V_WAVEFUNCTION_INITIATIZER ... (see section [Wavefunction Velocity Initializer (NWPW_RETIRED#V_WAVEFUNCTION_INITIALIZER) - retired) END  
  WAVEFUNCTION_EXPANDER      ... (see section [Wavefunction Expander](NWPW_RETIRED.md#WAVEFUNCTION_EXPANDER) - retired) END  
  INPUT_WAVEFUNCTION_FILENAME &lt;string input_wavefunctions default file_prefix.movecs&gt;   
 OUTPUT_WAVEFUNCTION_FILENAME &lt;string output_wavefunctions default file_prefix.movecs&gt;   
END
</code></pre>
<p>The following list describes the keywords contained in the PSPW input
block.</p>
<ul>
<li><code>cell_name</code> - name of the simulation_cell named <code>cell_name</code>. See
    section <a href="#plane-wave-density-functional-theory-simulation-cell">Simulation Cell</a>.</li>
<li><code>input_wavefunctions</code> - name of the file containing one-electron
    orbitals</li>
<li><code>output_wavefunctions</code> - name of the file that will contain the
    one-electron orbitals at the end of the run.</li>
<li><code>fake_mass</code> - value for the electronic fake mass <span class="arithmatex">\(\mu\)</span>
    This parameter is not presently used in a conjugate gradient
    simulation.</li>
<li><code>time_step</code> - value for the time step <span class="arithmatex">\(\Delta t\)</span>. This parameter
    is not presently used in a conjugate gradient simulation.</li>
<li><code>inner_iteration</code> - number of iterations between the printing out of
    energies and tolerances</li>
<li><code>outer_iteration</code> - number of outer iterations</li>
<li><code>tole</code> - value for the energy tolerance.</li>
<li><code>tolc</code> - value for the one-electron orbital tolerance.</li>
<li><code>cutoff</code> - value for the cutoff energy used to define the
    wavefunction. In addition using the CUTOFF keyword automatically
    sets the cutoff energy for the density to be twice the wavefunction
    cutoff.</li>
<li><code>ecut</code> - value for the cutoff energy used to define the density.
    Default is set to be the maximum value that will fit within the
    simulation_cell <code>cell_name</code>.</li>
<li><code>wcut</code> - value for the cutoff energy used to define the one-electron
    orbitals. Default is set to be the maximum value that will fit
    within the simulation_cell <code>cell_name</code>.</li>
<li><code>ncut</code> - value for the number of unit cells to sum over (in each
    direction) for the real space part of the Ewald summation. Note
    Ewald summation is only used if the simulation_cell is periodic.</li>
<li><code>rcut</code> - value for the cutoff radius used in the Ewald summation.
    Note Ewald summation is only used if the simulation_cell is
    periodic.</li>
</ul>
<p>Default set to be <span class="arithmatex">\(\frac{MIN(\left| \vec{a_i} \right|)}{\pi}, i=1,2,3\)</span>.</p>
<ul>
<li>(Vosko || PBE96 || revPBE || &hellip;) - Choose between Vosko et al&rsquo;s LDA
    parameterization or the orginal and revised Perdew, Burke, and
    Ernzerhof GGA functional. In addition, several hybrid options.</li>
<li>MULT - optional keyword which if specified allows the user to define
    the spin multiplicity of the system</li>
<li>MULLIKEN - optional keyword which if specified causes a Mulliken
    analysis to be performed at the end of the simulation.</li>
<li>EFIELD - optional keyword which if specified causes an atomic
    electric field analysis to be performed at the end of the
    simulation.</li>
<li>ALLOW_TRANSLATION - By default the the center of mass forces are
    projected out of the computed forces. This optional keyword if
    specified allows the center of mass forces to not be zero.</li>
<li>TRANSLATION - By default the the center of mass forces are projected
    out of the computed forces. TRANSLATION ON allows the center of mass
    forces to not be zero.</li>
<li>ROTATION - By default the overall rotation is not projected out of
    the computed forces. ROTATION OFF projects out the overal rotation
    of the molecule.</li>
<li>CG - optional keyword which sets the minimizer to 1</li>
<li>LMBFGS - optional keyword which sets the minimizer to 2</li>
<li>SCF - optional keyword which sets the minimizer to be a band by band
    minimizer. Several options are available for setting the density or
    potential mixing, and the type of Kohn-Sham minimizer.</li>
<li><code>mapping</code> - for a value of 1 slab FFT is used, for a value of 2 a
    2d-hilbert FFT is used.</li>
</ul>
<p>A variety of prototype minimizers can be used to minimize the energy. To
use these other optimizers the following SET directive needs to be
specified:</p>
<pre><code>set nwpw:mimimizer 1 # Default - Grassman conjugate gradient minimizer is used to minimize the energy. 
set nwpw:mimimizer 2 # Grassman LMBFGS minimimzer is used to minimize the energy.
set nwpw:minimizer 4 # Stiefel conjugate gradient minimizer is used to minimize the energy. 
set nwpw:minimizer 5 # Band-by-band (potential) minimizer is used to minimize the energy.
set nwpw:minimizer 6 # Projected Grassman LMBFGS minimizer is used to minimize the energy.
set nwpw:minimizer 7 # Stiefel LMBFGS minimizer is used to minimize the energy.
set nwpw:minimizer 8 # Band-by-band (density) minimizer is used to minimize the energy.
</code></pre>
<p>Limited testing suggests that the Grassman LMBFGS minimizer is about
twice as fast as the conjugate gradient minimizer. However, there are
several known cases where this optimizer fails, so it is currently not a
default option, and should be used with caution.</p>
<p>In addition the following SET directives can be
specified:</p>
<pre><code>set nwpw:lcao_skip .false. # Initial wavefunctions generated using an LCAO guess.  
set nwpw:lcao_skip .true. # Default - Initial wavefunctions generated using a random plane-wave guess.
set nwpw:lcao_print .false. # Default - Output not produced during the generation of the LCAO guess.   
set nwpw:lcao_print .true. # Output produced during the generation of the LCAO guess.
set nwpw:lcao_iterations 2 #specifies the number of LCAO iterations.
</code></pre>
<h3 id="plane-wave-density-functional-theory-paw-potentials">PAW Potentials<a class="headerlink" href="#plane-wave-density-functional-theory-paw-potentials" title="Permanent link">&para;</a></h3>
<p>The PSPW code can now handle PAW potentials. To use them the
pseudopotentials input block is used to redirect the code to use the paw
potentials located in the default paw potential library
(<code>$NWCHEM_TOP/src/nwpw/libraryp/paw_default</code>). For example, to redirect
the code to use PAW potentials for carbon and hydrogen, the following
input would be used.</p>
<pre><code>nwpw 
    pseudopotentials 
      C library paw_default 
      H library paw_default 
    end  
end
</code></pre>
<p>Most of the capabilities of PSPW will work with PAW potentials including
geometry optimization, Car-Parrinello ab initio molecular dynamics,
Born-Oppenheimer ab initio molecular dynamics, Metropolis Monte-Carlo,
and AIMD/MM. Unfortunately, some of the functionality is missing at this
point in time such as Mulliken analysis, and analytic stresses. However
these small number of missing capabilities should become available over
the next couple of months in the development tree of NWChem.</p>
<p>Even though analytic stresses are not currently available with PAW
potentials unit cell optimization can still be carried out using
numerical stresses. The following SET directives can be used to tell the
code to calculate stresses
numerically.</p>
<pre><code>set includestress  .true.              #this option tells driver to optimize the unit cell 
set includelattice .true.              #this option tells driver to optimize cell using a,b,c,alpha,beta,gamma 
set nwpw:frozen_lattice:thresh 999.0   #large number guarentees the lattice gridding does not adjust during optimization
set nwpw:cif_filename  pspw_corundum
set nwpw:stress_numerical .true. 
set nwpw:lstress_numerical .true.
</code></pre>
<h4 id="plane-wave-density-functional-theory-paw-implementation-notes">PAW Implementation Notes<a class="headerlink" href="#plane-wave-density-functional-theory-paw-implementation-notes" title="Permanent link">&para;</a></h4>
<p>The main idea in the PAW method(Blochl 1994) is to project out the
high-frequency components of the wavefunction in the atomic sphere
region. Effectively this splits the original wavefunction into two
parts:</p>
<div class="arithmatex">\[\psi_n(\mathbf{r}) = \tilde{\psi}_n(\mathbf{r}) + \sum_I \psi_n^I(\mathbf{r})\]</div>
<p>The first part <span class="arithmatex">\(\tilde{\psi}_n(\mathbf{r})\)</span> is smooth and can be
represented using a plane wave basis set of practical size. The second
term is localized with the atomic spheres and is represented on radial
grids centered on the atoms
as</p>
<div class="arithmatex">\[\psi_n^I(\mathbf{r}) = \sum_{\alpha} (\varphi_{\alpha}^I(\mathbf{r})-\tilde{\varphi}_{\alpha}^I(\mathbf{r}))c_{n\alpha}^I\]</div>
<p>where the coefficients <span class="arithmatex">\(c_{n\alpha}^I\)</span> are given by</p>
<div class="arithmatex">\[ c_{n\alpha}^I = &lt;\tilde{p}_{\alpha}^I | \tilde{\psi}_n&gt;\]</div>
<p>This decomposition can be expressed using an invertible linear
transformation, <span class="arithmatex">\(T\)</span>, is defined which relates the stiff one-electron
wavefunctions <span class="arithmatex">\(\psi_n\)</span> to a set of smooth one-electron wavefunctions
<span class="arithmatex">\(\tilde{\psi}_n\)</span>   </p>
<div class="arithmatex">\[\tilde{\psi}_n = T \psi_n\]</div>
<div class="arithmatex">\[\psi_n = T^{-1} \tilde{\psi}_n\]</div>
<p>which can be represented by fairly small plane-wave basis. The
transformation <span class="arithmatex">\(T\)</span> is defined using a local PAW basis, which consists
of atomic orbitals, <span class="arithmatex">\(\varphi{\alpha}^I(\mathbf{r})\)</span>, 
smooth atomic
orbitals, <span class="arithmatex">\(\tilde{\varphi}\)</span><sub>&alpha;</sub><sup>I</sup>(<strong>r</strong>)  which coincide with
the atomic orbitals outside a defined atomic sphere, and projector
functions, <span class="arithmatex">\(p_{\alpha}^I(\mathbf{r})\)</span>. Where I is the atomic index and
 is the orbital index. The projector functions are constructed such
that they are localized within the defined atomic sphere and in addition
are orthonormal to the atomic orbitals. Blchl defined the invertible linear transformations by  </p>
<div class="arithmatex">\[T = 1 + \sum_I \sum_\alpha (|\tilde{\varphi}_\alpha^I&gt; - |\varphi_\alpha^I &gt; ) &lt; p_\alpha^I|\]</div>
<div class="arithmatex">\[\tilde{T} = 1 + \sum_I \sum_{\alpha} (|\varphi_{\alpha}^I&gt; - |\tilde{\varphi}_{\alpha}^I &gt;)&lt;\tilde{\varphi}_{\alpha}^I|\]</div>
<div class="arithmatex">\[|\tilde{p}_{\alpha}^I&gt; = \sum_{\beta} [&lt;\tilde{p}^I | \varphi^I &gt;]_{\alpha\beta}^{-1} | p_{\beta}^I&gt;\]</div>
<p>The main effect of the PAW transformation is that the fast variations of
the valence wave function in the atomic sphere region are projected out
using local basis set, thereby producing a smoothly varying wavefunction
that may be expanded in a plane wave basis set of a manageable size.</p>
<p>The expression for the total energy in PAW method can be separated into
the following 15 terms.</p>
<div class="arithmatex">\[E_{PAW} = \tilde{E}_{kinetic-pw} + \tilde{E}_{vlocal-pw} + \tilde{E}_{Coulomb-pw} + \tilde{E}_{xc-pw} + E_{ion-ion}\]</div>
<div class="arithmatex">\[      + E_{cmp-cmp} + E_{cmp-pw} + E_{valence-core} + E_{kinetic-core} + E_{ion-core}\]</div>
<p>The first five terms are essentially the same as for a standard
pseudopotential plane-wave program, minus the non-local pseudopotential,
where</p>
<div class="arithmatex">\[\tilde{E}_{kinetic-pw} = \sum_i \sum_{\mathbf{G}} \frac{|\mathbf{G}|^2}{2} \tilde{\psi}_i^{*}(\mathbf{G}) \tilde{\psi}_i(\mathbf{G})\]</div>
<div class="arithmatex">\[\tilde{E}_{vlocal-pw} = \sum_{\mathbf{G}} \tilde{\rho}(\mathbf{G}) V_{local}(\mathbf{G})\]</div>
<div class="arithmatex">\[\tilde{E}_{Coulomb-pw} = \frac{\Omega}{2} \sum_{\mathbf{G}} \frac{4\pi}{|\mathbf{G}|} \tilde{\rho}^{*}(\mathbf{G}) \tilde{\rho}(\mathbf{G})\]</div>
<div class="arithmatex">\[\tilde{E}_{xc-pw} = \frac{\Omega}{N_1 N_2 N_3} \sum_{\mathbf{r}} \tilde{\rho}(\mathbf{r}) \epsilon_{xc}(\tilde{\rho}(\mathbf{r}))\]</div>
<div class="arithmatex">\[E_{ion-ion} = \frac{1}{2\Omega} \sum_{\mathbf{G}} \frac{4\pi}{|\mathbf{G}|^2} \exp(\frac{|\mathbf{G}|^2}{4\epsilon}) \sum_{I,J} Z_I \exp (-i \mathbf{G} \cdot \mathbf{R}_I) Z_J \exp ( -i  \mathbf{G} \cdot \mathbf{R}_J) + \]</div>
<div class="arithmatex">\[\frac{1}{2}\sum_{\mathbf{a}} \sum_{I,J \in |\mathbf{R}_I-\mathbf{R}_J+\mathbf{a}|} Z_I Z_J \frac{erf(\epsilon |\mathbf{R}_I-\mathbf{R}_J+\mathbf{a}|)}{|\mathbf{R}_I-\mathbf{R}_J+\mathbf{a}|} - \frac{\epsilon}{\pi}\sum_I Z_I^2 +  \]</div>
<div class="arithmatex">\[ - \frac{\pi}{2\epsilon^2\Omega} \left( \sum_{I} Z_{I} \right)^2 \]</div>
<p>The local potential in the <span class="arithmatex">\(\tilde{E}_{vlocal-pw}\)</span> term is the Fourier
transform of</p>
<div class="arithmatex">\[V_{local}(\mathbf{r}) = -\sum_I Z_I \frac{\frac{|\mathbf{r}-\mathbf{R}_I|}{\sigma_I}}{|\mathbf{r}-\mathbf{R}_I|} + v_{ps}^I(|\mathbf{|\mathbf{r}-\mathbf{R}_I|})\]</div>
<p>It turns out that for many atoms <span class="arithmatex">\(\sigma_I\)</span> needs to be fairly small.
This results in <span class="arithmatex">\(V_{local} (\mathbf{r})\)</span> being stiff. However, since
in the integral above this function is multiplied by a smooth density
<span class="arithmatex">\(\tilde{\rho}(\mathbf{G})\)</span> the expansion of <em>V<sub>local</sub>(<strong>G</strong>)</em>
only needs to be the same as the smooth density. The auxiliary
pseudoptential <span class="arithmatex">\(v_{ps}^I (|\mathbf{r}-\mathbf{R}_I |)\)</span> is defined to
be localized within the atomic sphere and is introduced to remove ghost
states due to local basis set incompleteness.</p>
<p>The next four terms are atomic based and they essentially take into
account the difference between the true valence wavefunctions and the
pseudowavefunctions.</p>
<div class="arithmatex">\[ E_{kinetic-atom}= \sum_I \sum_i \sum_{\alpha\beta} &lt;\tilde{\psi}_i|\tilde{p}_{\alpha}^I &lt; (t_{atom}^I)_{\alpha\beta} &lt;\tilde{p}_{\beta}^I|\tilde{\psi}_i \]</div>
<div class="arithmatex">\[ E_{local-atom}=\sum_I \sum_i \sum_{\alpha\beta} &lt;\tilde{\psi}_i|\tilde{p}_{\alpha}^I&gt; (u_{atom}^I)_{\alpha\beta} &lt;\tilde{p}_{\beta}^I|\tilde{\psi}_i&gt;\]</div>
<div class="arithmatex">\[ E_{xc-atom}=\sum_I \sum_{\theta\phi} w_{\theta\phi} \int_0^{r_{cut}^I} r^2 (\rho^I(r,\theta,\phi) \epsilon_{xc}(\rho^I(r,\theta,\phi)) -\tilde{\rho}^I(r,\theta,\phi) \epsilon_{xc}(\tilde{\rho}^I(r,\theta,\phi))) dr \]</div>
<div class="arithmatex">\[ E_{hartree-atom}= \sum_I W_{atom}^I =\frac{1}{2} \sum_I \sum_i \sum_{\alpha\beta} &lt;\tilde{\psi}_i|\tilde{p}_{\alpha}&gt;  &lt;\tilde{p}_{\beta}^I|\tilde{\psi}_i&gt; \]</div>
<div class="arithmatex">\[ \sum_j \sum_{\mu\nu} &lt;\tilde{\psi}_j|\tilde{p}_{\mu}^I&gt;  &lt;\tilde{p}_{\nu}^I|\tilde{\psi}_j&gt; \sum_{lm} \tau_{l_\alpha m_\alpha, l_\beta m_\beta}^{lm} \tau_{l_\mu m_\mu, l_\nu m_\nu}^{lm} (V_{Heff}^I)_{\alpha\beta\mu\nu}^{l} \]</div>
<p>The next three terms are the terms containing the compensation charge
densities.</p>
<div class="arithmatex">\[E_{cmp-vloc}= \sum_{\mathbf{G}} [\rho_{cmp} (\mathbf{G}) \tilde{V}_{local}(\mathbf{G})+\tilde{\rho}_{cmp}(\mathbf{G})(V_{local}(\mathbf{G})-\tilde{V}_{local}(\mathbf{G}))] + \\
\int (\rho_{cmp}(\mathbf{r})-\tilde{\rho}_{cmp}(\mathbf{r}))(V_{local}(r)-\tilde{V}_{local}(\mathbf{r}))d\mathbf{r}\]</div>
<div class="arithmatex">\[E_{cmp-cmp}= \Omega \sum_{\mathbf{G} \neq 0} \frac{4\pi}{|\mathbf{G}|^2} [ \rho_{cmp}(\mathbf{G}) \tilde{\rho}_{cmp}(\mathbf{G})- \frac{1}{2} \tilde{\rho}_{cmp}(\mathbf{G})\tilde{\rho}_{cmp}(\mathbf{G})] + \\
 \frac{1}{2} \iint \frac{(\rho_{cmp}(\mathbf{r})-\tilde{\rho}_{cmp}(\mathbf{r}))(\rho_{cmp}(\mathbf{r}')-\tilde{\rho}_{cmp}(\mathbf{r}'))} {|\mathbf{r}-\mathbf{r}'|}  d\mathbf{r} d\mathbf{r}'\]</div>
<div class="arithmatex">\[E_{cmp-pw}= \Omega \sum_{\mathbf{G} \neq 0} \frac{4\pi}{|\mathbf{G}|^2} \rho_{cmp}(\mathbf{G}) \tilde{\rho}(\mathbf{G})\]</div>
<p>In the first two formulas the first terms are computed using plane-waves
and the second terms are computed using Gaussian two center integrals.
The smooth local potential in the <span class="arithmatex">\(E_{cmp-vloc}\)</span> term is the Fourier
transform
of</p>
<div class="arithmatex">\[\tilde{V}_{local}(\mathbf{r})=- \sum_I Z_I    \frac{ erf( \frac{ |\mathbf{r}-\mathbf{R}_I | }{\tilde{\sigma}_I })} {|r-R_I |}\]</div>
<p>The stiff and smooth compensation charge densities in the above formula
are</p>
<div class="arithmatex">\[\rho_{cmp}(r)=\sum_I\sum_{lm} Q_{lm}^I g_{lm}^(\sigma_I ) (r-R_I )\]</div>
<div class="arithmatex">\[\tilde{\rho}_{cmp}(r)=\sum_I \sum_{lm} Q_{lm}^I g_{lm}^(\tilde{\sigma}_I ) (r-R_I )\]</div>
<p>where</p>
<div class="arithmatex">\[Q_{lm}^I = \sum_i \sum_{\alpha\beta} &lt;\tilde{\psi}_i|\tilde{p}_\alpha^I &gt; &lt;\tilde{p}_\beta^I|\tilde{\psi}_i &gt;  \tau_{l_\alpha m_\alpha, l_\beta m_\beta}^{lm} (q_{comp}^I )_{\alpha\beta}^l\]</div>
<p>The decay parameter <span class="arithmatex">\(\sigma_I\)</span> is defined the same as above, and
<span class="arithmatex">\(\tilde{\sigma}\)</span><sub>I</sub> is defined to be smooth enough in order that <sub>cmp</sub>(r) and <span class="arithmatex">\(\tilde{V}\)</span><sub>local</sub>(r) can readily be expanded in
terms of plane-waves.</p>
<p>The final three terms are the energies that contain the core
densities</p>
<div class="arithmatex">\[E_{valence-core}=\sum_{i}\sum_{I}\sum_{\alpha \beta}&lt;\tilde{\psi}_{i} | \tilde{p}_{\alpha}^{I} &gt;  (V_{valence-core})_{\alpha \beta}^{I} &lt;\tilde{p}_\beta^{I} | \tilde{\psi}_{i} &gt;\]</div>
<div class="arithmatex">\[E_{kinetic-core}=\sum_c \int_0^\infty \left[ (\varphi_{n_c l_c}^I (r))^{'} (\varphi_{n_c l_c}^{I} (r))^{'} + l_c (l_c+1)  \frac{\varphi_{n_c l_c}^{I} (r) \varphi_{n_c l_c}^{I} (r)}{r^{2}} \right]  dr\]</div>
<div class="arithmatex">\[E_{ion-core}= \sum_I \frac{1}{2} \iint \frac{\rho_c^{I} (r) \rho_c^{I} (r')}{|r-r'|}  drdr^{'} - \int  \frac{\rho_c^{I} (r)}{|r|}  (Z_I+Z_I^{core} ) dr\]</div>
<p>The matrix elements contained in the above formula are</p>
<div class="arithmatex">\[(t_{atom}^{I} )_{\alpha \beta}= {\delta_{m_\alpha m_\beta} \delta_{l_\alpha l_\beta }}2  \int_0^{r_{cut}^I}  \left[
  (\varphi_{n_\alpha l_\alpha}^{I} (r))' (\varphi_{n_\beta l_\beta}^{I} (r))' - (\tilde{\varphi}_{n_\alpha l_\alpha}^{I} (r))' (\tilde{\varphi}_{n_\beta l_\beta}^{I} (r))' + l_\alpha (l_\alpha+1)  \frac{ \varphi_{n_\alpha l_\alpha}^{I}(r) \varphi_{n_\beta l_\beta}^{I}(r) - \tilde{\varphi}_{n_\alpha l_\alpha}^{I}(r) \tilde{\varphi}_{n_\beta l_\beta}^I }{r^2 } \right]  dr\]</div>
<div class="arithmatex">\[(u_{atom}^I )_{\alpha \beta} =\frac{Z_I}{4\pi} (V_{comp}^I )_{\alpha \beta}^{l=0} + \frac{2Z_I}{(\sqrt{2\pi} \sigma_I } (q_comp^I )_{\alpha \beta}^(l=0) +\delta_{m_\alpha m_\beta } \delta_{l_\alpha l_\beta } \int_0^{r_{cut}^I} [ \varphi_{n_\alpha l_\alpha}^I (r) \varphi_(n_\beta l_\beta)^I (r) \frac{-Z_I}{r} + \tilde{\varphi}_{n_\alpha l_\alpha}^I (r) \tilde{\varphi}_{n_\beta l_\beta}^I (r) (-v_{ps}^I (r)) ] dr\]</div>
<div class="arithmatex">\[(V_{Heff}^I )_{\alpha \beta \mu \nu}^l = (V_H^I )_{\alpha\beta\mu\nu}^l - 2(V_{comp}^I )_{\alpha\beta}^l (q_{comp}^I )_{\mu\nu}^l -(v_g^I )^l (q_{comp}^I )_{\alpha\beta}^l (q_{comp}^I )_{\mu\nu}^l\]</div>
<div class="arithmatex">\[(V_H^I )_{\alpha\beta\mu\nu}^l = \frac{4\pi}{2l+1} \int_0^{r_{cut}^I} \int_0^{r_{cut}^I} \left( \frac{r_{&lt;}^l}{r_{&gt;}^{l+1} } \right)( \varphi_{n_\alpha l_\alpha } (r) \varphi_{n_\beta l_\beta } (r) \varphi_{n_\mu l_\mu } (r) \varphi_{n_\nu l_\nu } (r) - \tilde{\varphi}_{n_\alpha l_\alpha } (r) \tilde{\varphi}_(n_\beta l_\beta ) (r) \tilde{\varphi}_{n_\mu l_\mu } (r) \tilde{\varphi}_{n_\nu l_\nu } (r) )  drdr'\]</div>
<div class="arithmatex">\[(V_{comp}^I )_{\alpha\beta}^l = \frac{4\pi}{2l+1} \int_0^{r_{cut}^I} \int_0^{r_{cut}^I} \tilde{\varphi}_{n_\alpha l_\alpha } (r) \tilde{\varphi}_{n_\beta l_\beta } (r) \left( \frac{r_{&lt;}^l}{r_{&gt;}^{l+1} } \right) g_l^I (r')  r'^2  drdr'\]</div>
<div class="arithmatex">\[(q_{comp}^I )_{\alpha\beta}^l = \int_0^\infty r^l \left( \varphi_{n_\alpha l_\alpha } (r) \varphi_{n_\beta l_\beta } (r)- \tilde{\varphi}_{n_\alpha l_\alpha } (r) \tilde{\varphi}_{n_\beta l_\beta } (r) \right) dr\]</div>
<div class="arithmatex">\[(v_g^I )^l =\frac{4\sqrt{2\pi}}{ (2l+1)(2l+1)!! \sigma_I^{2l+1} }\]</div>
<div class="arithmatex">\[\tau_{l_\alpha m_\alpha,l_\beta m_\beta}^{lm}= \int_0^{2\pi} \int_0^{\pi} T_{lm} (\theta,\phi) T_{l_\alpha m_\alpha } (\theta,\phi) T_{l_\beta m_\beta } (\theta,\phi)  \sin(\theta) d\theta d\phi\]</div>
<h3 id="plane-wave-density-functional-theory-exchange-correlation-potentials">Exchange-Correlation Potentials<a class="headerlink" href="#plane-wave-density-functional-theory-exchange-correlation-potentials" title="Permanent link">&para;</a></h3>
<h4 id="plane-wave-density-functional-theory-dft-u-corrections">DFT + U Corrections<a class="headerlink" href="#plane-wave-density-functional-theory-dft-u-corrections" title="Permanent link">&para;</a></h4>
<p>TO DO</p>
<pre><code>nwpw 
  uterm d 0.13634 0.0036749 1 
end
</code></pre>
<h4 id="plane-wave-density-functional-theory-langreth-style-vdw-and-vdw-van-der-wall-functionals">Langreth style vdw and vdw van der Wall functionals<a class="headerlink" href="#plane-wave-density-functional-theory-langreth-style-vdw-and-vdw-van-der-wall-functionals" title="Permanent link">&para;</a></h4>
<p>These potenials that are used to augment standard exchange-correlation potentials area calculated from a double integral over a nonlocal interaction kernel, <span class="arithmatex">\(\phi(\mathbf{r},\mathbf{r}^{'})\)</span></p>
<div class="arithmatex">\[E_{vdw} = \int \rho(\mathbf{r}) \phi(\mathbf{r},\mathbf{r}^{'}) \rho(\mathbf{r}^{'}) d\mathbf{r} d\mathbf{r}^{'} \]</div>
<p>that is evaluated using the fast Fourier transformation method of Roman-Perez and Soler.</p>
<p>G. Roman-Perez and J. M. Soler, Phys. Rev. Lett. 103, 096102 (2009).</p>
<p>Langreth vdw and vdw2 van der Wall functionals are currently available for the BEEF, PBE96,
revPBE, PBEsol, BLYP, PBE0, revPBE0, HSE, and B3LYP exchange-correlation
functionals. To use them the following keywords BEEF-vdw, BEEF-vdw2, PBE96-vdw,
PBE96-vdw2, BLYP-vdw, BLYP-vdw2,
revPBE-vdw, revPBE-vdw, PBEsol-vdw PBEsol-vdw2, PBE0-vdw, PBE0-vdw2,
revPBE0-vdw, revPBE0-vdw2, HSE-vdw, HSE-vdw2, B3LYP-vdw, and B3LYP-vdw2
 can be used in the XC input directive, e.g.</p>
<pre><code>nwpw
   xc beef-vdw  
end
</code></pre>
<pre><code>nwpw
   xc beef-vdw2  
end
</code></pre>
<p>the vdw and vdw2 functionals are defined in  </p>
<p>(vdw) Dion M, Rydberg H, Schrder E, Langreth DC, Lundqvist BI. Van der Waals density functional for general geometries. Physical review letters. 2004 Jun 16;92(24):246401.</p>
<p>(vdw2) K. Lee, E. D. Murray, L. Kong, B. I. Lundqvist, and D. C. Langreth, Phys. Rev. B 82, 081101 (2010).</p>
<h4 id="plane-wave-density-functional-theory-grimme-dispersion-corrections">Grimme Dispersion Corrections<a class="headerlink" href="#plane-wave-density-functional-theory-grimme-dispersion-corrections" title="Permanent link">&para;</a></h4>
<p>Grimme dispersion corrections are currently available for the PBE96,
revPBE, PBEsol, BLYP, PBE0, revPBE0, HSE, and B3LYP exchange-correlation
functionals. To use them the following keywords PBE96-Grimme2,
PBE96-Grimme3, PBE96-Grimme4, BLYP-Grimme2, BLYP-Grimme3, BLYP-Grimme4,
revPBE-Grimme2, revPBE-Grimme3, revPBE-Grimme4, PBEsol-Grimme2,
PBEsol-Grimme3, PBEsol-Grimme4, PBE0-Grimme2, PBE0-Grimme3,
PBE0-Grimme4, revPBE0-Grimme2, revPBE0-Grimme3, revPBE0-Grimme4,
HSE-Grimme2, HSE-Grimme3, HSE-Grimme4, B3LYP-Grimme2, B3LYP-Grimme3, and
B3LYP-Grimme4 can be used in the XC input directive, e.g.</p>
<pre><code>nwpw
   xc pbe96-grimme2  
end
</code></pre>
<p>In these functionals Grimme2, Grimme3 and Grimme4 are defined in the
following papers by S. Grimme.</p>
<p>Grimme2 - Commonly known as DFT-D2, S. Grimme, J. Comput. Chem., 27
(2006), 1787-1799.</p>
<p>Grimme3 - Commonly known as DFT-D3, S. Grimme, J. Antony, S. Ehrlich and
H. Krieg A consistent and accurate ab initio parameterization of density
functional dispersion correction (DFT-D) for the 94 elements H-Pu, J.
Chem. Phys, 132 (2010), 154104</p>
<p>Grimme4 - Commonly known as DFT-D3 with BJ damping. This correction
augments the Grimme3 correction by including BJ-damping, S. Grimme, J.
Antony, S. Ehrlich and H. Krieg A consistent and accurate ab initio
parameterization of density functional dispersion correction (DFT-D) for
the 94 elements H-Pu, J. Chem. Phys, 132 (2010), 154104. S. Grimme, S.
Ehrlich and L. Goerigk, J. Comput. Chem, 32 (2011), 1456-1465. This
correction augments the Grimme3 correction by including BJ-damping.</p>
<p>If these functionals are used in a publication please include in your
citations the references to Grimme&rsquo;s work.</p>
<h4 id="plane-wave-density-functional-theory-using-exchange-correlation-potentials-available-in-the-dft-module">Using Exchange-Correlation Potentials Available in the DFT Module<a class="headerlink" href="#plane-wave-density-functional-theory-using-exchange-correlation-potentials-available-in-the-dft-module" title="Permanent link">&para;</a></h4>
<p>(<span style="color: red;"><em>Warning - To use this capability in NWChem
6.6 the user must explicitly include the nwxc module in the
NWCHEM_MODULES list when compiling. Unfortunately, there was too much
uncertainty in how the nwxc computed higher-order derivatives used by
some of the functionality in nwdft module to include it in a release for
all the functionality in NWChem. We are planning to have a debug release
in winter 2016 to take fix this problem. This capability is still
included by default in NWChem 6.5</em></span>)</p>
<p>The user has the option of using many of the exchange-correlation
potentials available in DFT Module (see Section <a href="#density-functional-theory-for-molecules-xc-and-decomp-exchange-correlation-potentials">XC and DECOMP &ndash;
Exchange-Correlation
Potentials</a>).</p>
<pre><code>  XC [[acm] [b3lyp] [beckehandh] [pbe0] [bhlyp]\
      [becke97]  [becke97-1] [becke97-2] [becke97-3] [becke98] [hcth] [hcth120] [hcth147] \ 
      [hcth407] [becke97gga1] [hcth407p] \
      [optx] [hcthp14] [mpw91] [mpw1k] [xft97] [cft97] [ft97] [op] [bop] [pbeop]\
      [HFexch &lt;real prefactor default 1.0&gt;] \
      [becke88 [nonlocal] &lt;real prefactor default 1.0&gt;] \
      [xperdew91 [nonlocal] &lt;real prefactor default 1.0&gt;] \
      [xpbe96 [nonlocal] &lt;real prefactor default 1.0&gt;] \
      [gill96 [nonlocal] &lt;real prefactor default 1.0&gt;] \
      [lyp &lt;real prefactor default 1.0&gt;] \
      [perdew81 &lt;real prefactor default 1.0&gt;] \
      [perdew86 [nonlocal] &lt;real prefactor default 1.0&gt;] \
      [perdew91 [nonlocal] &lt;real prefactor default 1.0&gt;] \
      [cpbe96 [nonlocal] &lt;real prefactor default 1.0&gt;] \
      [pw91lda &lt;real prefactor default 1.0&gt;] \
      [slater &lt;real prefactor default 1.0&gt;] \
      [vwn_1 &lt;real prefactor default 1.0&gt;] \
      [vwn_2 &lt;real prefactor default 1.0&gt;] \
      [vwn_3 &lt;real prefactor default 1.0&gt;] \
      [vwn_4 &lt;real prefactor default 1.0&gt;] \
      [vwn_5 &lt;real prefactor default 1.0&gt;] \
      [vwn_1_rpa &lt;real prefactor default 1.0&gt;]]
</code></pre>
<p>These functional can be invoked by prepending the &ldquo;new&rdquo; directive before
the exchange correlation potetntials in the input directive, XC new
slater vwn_5.</p>
<p>That is, this statement in the input file</p>
<pre><code>nwpw 
 XC new slater vwn_5  
end  
task pspw energy
</code></pre>
<p>Using this input the user has ability to include only the local or
nonlocal contributions of a given functional. The user can also specify
a multiplicative prefactor (the variable <code>prefactor</code> in the input) for
the local/nonlocal component or total (for more details see Section <a href="#density-functional-theory-for-molecules-xc-and-decomp-exchange-correlation-potentials">XC
and DECOMP &ndash; Exchange-Correlation
Potentials</a>).
An example of this might be,</p>
<p><code>XC new becke88 nonlocal 0.72</code></p>
<p>The user should be aware that the Becke88 local component is simply the
Slater exchange and should be input as such.</p>
<p>Any combination of the supported exchange functional options can be
used. For example the popular Gaussian B3 exchange could be specified
as:</p>
<p><code>XC new slater 0.8 becke88 nonlocal 0.72 HFexch 0.2</code></p>
<p>Any combination of the supported correlation functional options can be
used. For example B3LYP could be specified
as:</p>
<p><code>XC new vwn_1_rpa 0.19 lyp 0.81 HFexch 0.20  slater 0.80 becke88 nonlocal 0.72</code></p>
<p>and X3LYP as:</p>
<pre><code>xc new vwn_1_rpa 0.129 lyp 0.871 hfexch 0.218 slater 0.782 \ 
becke88 nonlocal 0.542  xperdew91 nonlocal 0.167
</code></pre>
<h4 id="plane-wave-density-functional-theory-exact-exchange">Exact Exchange<a class="headerlink" href="#plane-wave-density-functional-theory-exact-exchange" title="Permanent link">&para;</a></h4>
<h4 id="plane-wave-density-functional-theory-self-interaction-corrections">Self-Interaction Corrections<a class="headerlink" href="#plane-wave-density-functional-theory-self-interaction-corrections" title="Permanent link">&para;</a></h4>
<p>The SET directive is used to specify the molecular orbitals contribute
to the self-interaction-correction (SIC) term.</p>
<pre><code>set pspw:SIC_orbitals &lt;integer list_of_molecular_orbital_numbers&gt;
</code></pre>
<p>This defines only the molecular orbitals in the list as SIC active. All
other molecular orbitals will not contribute to the SIC term. For
example the following directive specifies that the molecular orbitals
numbered 1,5,6,7,8, and 15 are SIC active.</p>
<p><code>set pspw:SIC_orbitals 1 5:8 15</code></p>
<p>or equivalently</p>
<p><code>set pspw:SIC_orbitals 1 5 6 7 8 15</code></p>
<p>The following directive turns on self-consistent SIC.</p>
<pre><code>set pspw:SIC_relax .false. # Default - Perturbative SIC calculation  
set pspw:SIC_relax .true. # Self-consistent SIC calculation
</code></pre>
<p>Two types of solvers can be used and they are specified using the
following SET directive</p>
<pre><code>set pspw:SIC_solver_type 1 # Default - cutoff coulomb kernel 
set pspw:SIC_solver_type 2 # Free-space boundary condition kernel
</code></pre>
<p>The parameters for the cutoff coulomb kernel are defined by the
following SET directives:</p>
<pre><code>set pspw:SIC_screening_radius &lt;real rcut&gt;  
set pspw:SIC_screening_power &lt;real rpower&gt;
</code></pre>
<h3 id="plane-wave-density-functional-theory-wannier">Wannier<a class="headerlink" href="#plane-wave-density-functional-theory-wannier" title="Permanent link">&para;</a></h3>
<p>The pspw wannier task is generate maximally localized (Wannier)
molecular orbitals. The algorithm proposed by Silvestrelli et al is use
to generate the Wannier orbitals.</p>
<p>Input to the Wannier task is contained within the Wannier sub-block.</p>
<pre><code>NWPW  
...  
 Wannier  
   ...   
 END 
... 
END
</code></pre>
<p>To run a Wannier calculation the following directive is used:</p>
<p><code>TASK PSPW Wannier</code></p>
<p>Listed below is the format of a Wannier
sub-block.</p>
<pre><code>NWPW  
...   
 Wannier   
  OLD_WAVEFUNCTION_FILENAME &lt;string input_wavefunctions default input_movecs&gt;  
  NEW_WAVEFUNCTION_FILENAME &lt;string output_wavefunctions default input_movecs&gt; 
 END
...  
END
</code></pre>
<p>The following list describes the input for the Wannier sub-block.</p>
<ul>
<li><code>input_wavefunctions</code> - name of pspw wavefunction file.</li>
<li><code>output_wavefunctions</code> - name of pspw wavefunction file that will
    contain the Wannier orbitals.</li>
</ul>
<h3 id="plane-wave-density-functional-theory-mulliken-analysis">Mulliken Analysis<a class="headerlink" href="#plane-wave-density-functional-theory-mulliken-analysis" title="Permanent link">&para;</a></h3>
<h3 id="plane-wave-density-functional-theory-density-of-states">Density of States<a class="headerlink" href="#plane-wave-density-functional-theory-density-of-states" title="Permanent link">&para;</a></h3>
<p>The &ldquo;dos&rdquo; option is used to turn on a density of states analysis. This
option can be specified without additional parameters, i.e.</p>
<pre><code>nwpw 
  dos  
end
</code></pre>
<p>in which case default values are used, or it can be specified with
additional parameters, e.g.</p>
<pre><code>nwpw
   dos 0.002 700 -0.80000 0.8000 
end
</code></pre>
<p>The parameters are</p>
<pre><code>nwpw  
 dos  [&lt;alpha&gt; &lt;npoints&gt; &lt;emin&gt; &lt;emax&gt;] 
end
</code></pre>
<p>where</p>
<ul>
<li><code>alpha</code>  value for the broadening the eigenvalues, default
    0.05/27.2116 au</li>
<li><code>npoints</code>  number of plotting points in dos files, default 500</li>
<li><code>emin</code>  minimum energy in dos plots, default min(eigenvalues)-0.1
    au</li>
<li><code>emax</code>  maximimum energy in dos plots, default max(eigenvalues)+0.1
    au</li>
</ul>
<p>The units for dos parameters are in atomic units. Note that if virtual
states are specified in the pspw calculation then the virtual density of
states will also be generated in addition to the filled density of
states.</p>
<p>The following files are generated and written to the permanent_dir for
restricted calculations</p>
<ul>
<li>file_prefix.smear_dos_both - total density of states</li>
<li>file_prefix.smear_fdos_both - density of states of filled states</li>
<li>file_prefix.smear_vdos_both - density of states of virtual states</li>
</ul>
<p>For unrestricted calculations</p>
<ul>
<li>file_prefix.smear_dos_alpha - total density of states for up
    electrons</li>
<li>file_prefix.smear_dos_beta - total density of states for down
    electrons</li>
<li>file_prefix.smear_fdos_alpha - density of states for filled up
    electrons</li>
<li>file_prefix.smear_fdos_beta - density of states for filled down
    electrons</li>
<li>file_prefix.smear_vdos_alpha - density of states for virtual up
    electrons</li>
<li>file_prefix.smear_vdos_beta - density of states for virtual down
    electrons</li>
</ul>
<p>The nwpw:dos:actlist variable is used to specify the atoms used to
determine weights for dos generation. If the variable is not set then
all the atoms are used, e.g.</p>
<p><code>set nwpw:dos:actlist 1 2 3 4</code></p>
<h3 id="plane-wave-density-functional-theory-projected-density-of-states">Projected Density of States<a class="headerlink" href="#plane-wave-density-functional-theory-projected-density-of-states" title="Permanent link">&para;</a></h3>
<p>For projected density of states the &ldquo;Mulliken&rdquo; keyword needs to be set,
e.g.</p>
<pre><code>nwpw 
  Mulliken  
  dos
end
</code></pre>
<p>The following additional files are generated and written to the
permanent_dir for restricted calculations</p>
<ul>
<li>file_prefix.mulliken_dos_both_s - total s projected density of
    restricted states</li>
<li>file_prefix.mulliken_fdos_both_s - s projected density of states
    of filled restricted states</li>
<li>file_prefix.mulliken_vdos_both_s - s projected density of states
    of virtual restricted states</li>
<li>file_prefix.mulliken_dos_both_p - total p projected density of
    states</li>
<li>file_prefix.mulliken_fdos_both_p - p projected density of states
    of filled states</li>
<li>file_prefix.mulliken_vdos_both_p - p projected density of states
    of virtual states</li>
</ul>
<p>&hellip;</p>
<ul>
<li>file_prefix.mulliken_dos_both_all - total of projected density
    of filled and virtual restricted states</li>
<li>file_prefix.mulliken_fdos_both_all - total of projected density
    of filled restricted states</li>
<li>file_prefix.mulliken_vdos_both_all - total of projected density
    of states of virtual restricted states</li>
</ul>
<p>Similarly for unrestricted calculations</p>
<ul>
<li>file_prefix.mulliken_dos_alpha_s - total s projected density of
    up states</li>
<li>file_prefix.mulliken_fdos_alpha_s - s projected density of
    states of filled up states</li>
<li>file_prefix.mulliken_vdos_alpha_s - s projected density of
    states of virtual up states</li>
<li>file_prefix.mulliken_dos_alpha_p - total p projected density of
    up states</li>
<li>file_prefix.mulliken_fdos_alpha_p - p projected density of
    states of filled up states</li>
<li>file_prefix.mulliken_vdos_alpha_p - p projected density of
    states of virtual up states</li>
</ul>
<p>&hellip;</p>
<ul>
<li>file_prefix.mulliken_dos_alpha_all - total of projected density
    of filled up states</li>
<li>file_prefix.mulliken_fdos_alpha_all - total of projected density
    of filled up states</li>
<li>file_prefix.mulliken_vdos_alpha_all - total of projected density
    of states of virtual up states</li>
</ul>
<p>&hellip;</p>
<ul>
<li>file_prefix.mulliken_dos_beta_s - total s projected density of
    down states</li>
<li>file_prefix.mulliken_fdos_beta_s - s projected density of states
    of filled down states</li>
<li>file_prefix.mulliken_vdos_beta_s - s projected density of states
    of virtual down states</li>
<li>file_prefix.mulliken_dos_beta_p - total p projected density of
    down states</li>
<li>file_prefix.mulliken_fdos_beta_p - p projected density of states
    of filled down states</li>
<li>file_prefix.mulliken_vdos_beta_p - p projected density of states
    of virtual down states</li>
</ul>
<p>&hellip;</p>
<ul>
<li>file_prefix.mulliken_dos_beta_all - total of projected density
    of filled down states</li>
<li>file_prefix.mulliken_fdos_beta_all - total of projected density
    of filled down states</li>
<li>file_prefix.mulliken_vdos_beta_all - total of projected density
    of states of virtual down states</li>
</ul>
<h3 id="plane-wave-density-functional-theory-point-charge-analysis">Point Charge Analysis<a class="headerlink" href="#plane-wave-density-functional-theory-point-charge-analysis" title="Permanent link">&para;</a></h3>
<p>The MULLIKEN option can be used to generate derived atomic point charges
from a plane-wave density. This analysis is based on a strategy
suggested in the work of P.E. Blochl, J. Chem. Phys. vol. 103, page 7422
(1995). In this strategy the low-frequency components a plane-wave
density are fit to a linear combination of atom centered Gaussian
functions.</p>
<p>The following SET directives are used to define the
fitting.</p>
<pre><code>set nwpw_APC:Gc &lt;real Gc_cutoff&gt; # specifies the maximum frequency component of the density to be used in the fitting in units of au.  
set nwpw_APC:nga &lt;integer number_gauss&gt; # specifies the the number of Gaussian functions per atom.
set nwpw_APC:gamma &lt;real gamma_list&gt; # specifies the decay lengths of each atom centered Gaussian. 
</code></pre>
<p>We suggest using the following parameters.</p>
<pre><code>set nwpw_APC:Gc 2.5
set nwpw_APC:nga 3 
set nwpw_APC:gamma 0.6 0.9 1.35  
</code></pre>
<h3 id="plane-wave-density-functional-theory-pspw_dplot-generate-gaussian-cube-files">PSPW_DPLOT: Generate Gaussian Cube Files<a class="headerlink" href="#plane-wave-density-functional-theory-pspw_dplot-generate-gaussian-cube-files" title="Permanent link">&para;</a></h3>
<p>The pspw dplot task is used to generate plots of various types of
electron densities (or orbitals) of a molecule. The electron density is
calculated on the specified set of grid points from a PSPW calculation.
The output file generated is in the Gaussian Cube format. Input to the
DPLOT task is contained within the DPLOT sub-block.</p>
<pre><code>NWPW 
...  
 DPLOT   
   ...   
 END  
... 
END
</code></pre>
<p>To run a DPLOT calculation the following directive is used:</p>
<p><code>TASK PSPW PSPW_DPLOT</code></p>
<p>Listed below is the format of a DPLOT
sub-block.</p>
<pre><code>NWPW  
...   
 DPLOT  
   VECTORS &lt;string input_wavefunctions default input_movecs&gt;  
   DENSITY [total||diff||alpha||beta||laplacian||potential default total]   
           &lt;string density_name no default&gt;  
   ELF [restricted|alpha|beta] &lt;string elf_name no default&gt;
   ORBITAL &lt;integer orbital_number no default&gt; &lt;string orbital_name no default&gt;  
   [LIMITXYZ [units &lt;string Units default au&gt;]   
   &lt;real X_From&gt; &lt;real X_To&gt; &lt;integer No_Of_Spacings_X&gt;   
   &lt;real Y_From&gt; &lt;real Y_To&gt; &lt;integer No_Of_Spacings_Y&gt;  
   &lt;real Z_From&gt; &lt;real Z_To&gt; &lt;integer No_Of_Spacings_Z&gt;]  
   NCELL &lt;integer nx default 0&gt; &lt;integer ny default 0&gt; &lt;integer nz default 0&gt;
   POSITION_TOLERANCE &lt;real rtol default 0.001&gt;
 END  
...  
END
</code></pre>
<p>The following list describes the input for the DPLOT sub-block.</p>
<pre><code>VECTORS &lt;string input_wavefunctions default input_movecs&gt;
</code></pre>
<p>This sub-directive specifies the name of the molecular orbital file. If
the second file is optionally given the density is computed as the
difference between the corresponding electron densities. The vector
files have to
match.</p>
<pre><code>DENSITY [total||difference||alpha||beta||laplacian||potential default total] 
        &lt;string density_name no default&gt;
</code></pre>
<p>This sub-directive specifies, what kind of density is to be plotted. The
known names for total, difference, alpha, beta, laplacian, and
potential.</p>
<pre><code>ELF [restricted|alpha|beta] &lt;string elf_name no default&gt;
</code></pre>
<p>This sub-directive specifies that an electron localization function
(ELF) is to be
plotted.</p>
<pre><code>ORBITAL &lt;integer orbital_number no default&gt; &lt;string orbital_name no default&gt;
</code></pre>
<p>This sub-directive specifies the molecular orbital number that is to be
plotted.</p>
<pre><code>LIMITXYZ [units &lt;string Units default angstroms&gt;]  
&lt;real X_From&gt; &lt;real X_To&gt; &lt;integer No_Of_Spacings_X&gt;  
&lt;real Y_From&gt; &lt;real Y_To&gt; &lt;integer No_Of_Spacings_Y&gt;  
&lt;real Z_From&gt; &lt;real Z_To&gt; &lt;integer No_Of_Spacings_Z&gt;
</code></pre>
<p>By default the grid spacing and the limits of the cell to be plotted are
defined by the input wavefunctions. Alternatively the user can use the
LIMITXYZ sub-directive to specify other limits. The grid is generated
using No_Of_Spacings + 1 points along each direction. The known names
for Units are angstroms, au and bohr.</p>
<h2 id="plane-wave-density-functional-theory-band-tasks-multiple-k-point-calculations">Band Tasks: Multiple k-point Calculations<a class="headerlink" href="#plane-wave-density-functional-theory-band-tasks-multiple-k-point-calculations" title="Permanent link">&para;</a></h2>
<p>All input to the Band Tasks is contained within the compound NWPW block,</p>
<pre><code>NWPW   
 ...  
END
</code></pre>
<p>To perform an actual calculation a Task Band directive is used (Section
<a href="#task">Task</a>).</p>
<p><code>Task Band</code></p>
<p>Once a user has specified a geometry, the Band module can be invoked
with no input directives (defaults invoked throughout). There are
sub-directives which allow for customized application; those currently
provided as options for the Band module
are:</p>
<pre><code>NWPW   
 CELL_NAME &lt;string cell_name default cell_default&gt; 
 ZONE_NAME &lt;string zone_name default zone_default&gt;  
 INPUT_WAVEFUNCTION_FILENAME &lt;string input_wavefunctions default input_movecs&gt;  
 OUTPUT_WAVEFUNCTION_FILENAME &lt;string output_wavefunctions default input_movecs&gt;  
 FAKE_MASS &lt;real fake_mass default 400000.0&gt;   
 TIME_STEP &lt;real time_step default 5.8&gt;   
 LOOP &lt;integer inner_iteration outer_iteration default 10 100&gt;   
 TOLERANCES &lt;real tole tolc default 1.0e-7 1.0e-7&gt; 
 CUTOFF &lt;real cutoff&gt;  
 ENERGY_CUTOFF &lt;real ecut default (see input description)&gt; 
 WAVEFUNCTION_CUTOFF &lt;real wcut default (see input description)&gt;  
 EWALD_NCUT &lt;integer ncut default 1&gt;] 
 EWALD_RCUT &lt;real rcut default (see input description)&gt; 

 XC (Vosko      || LDA          || PBE96         || revPBE       || PBEsol          ||  `  
                || HSE  || default Vosko)                                                                `  
  #Note that HSE is the only hybrid functional implemented in BAND

 DFT||ODFT||RESTRICTED||UNRESTRICTED   
 MULT &lt;integer mult default 1&gt;   
 CG  
 LMBFGS  
 SCF [Anderson|| simple || Broyden]  
 [CG || RMM-DIIS] [density || potential]  
 [ALPHA real alpha default 0.25]  
 [ITERATIONS integer inner_iterations default 5]  
 [OUTER_ITERATIONS integer outer_iterations default 0]

 SIMULATION_CELL  [units &lt;string units default bohrs&gt;]
   ... (see input description)   
 END 
 BRILLOUIN_ZONE 
   ... (see input description) 
 END 
 MONKHORST-PACK &lt;real n1 n2 n3 default 1 1 1&gt;
 BAND_DPLOT 
   ... (see input description)  
 END  
 MAPPING &lt;integer mapping default 1&gt;  
 SMEAR &lt;sigma default 0.001&gt; 
 [TEMPERATURE &lt;temperature&gt;] 
 [FERMI || GAUSSIAN || MARZARI-VANDERBILT default FERMI] 
 [ORBITALS &lt;integer orbitals default 4&gt;] 
END 
</code></pre>
<p>The following list describes these keywords.</p>
<ul>
<li><code>cell_name</code> - name of the simulation_cell named <code>cell_name</code>. See
    <a href="#plane-wave-density-functional-theory-simulation-cell">Simulation Cell</a>.</li>
<li><code>input_wavefunctions</code> - name of the file containing one-electron
    orbitals</li>
<li><code>output_wavefunctions</code> - name that will point to file containing the
    one-electron orbitals at the end of the run.</li>
<li><code>fake_mass</code> - value for the electronic fake mass <span class="arithmatex">\(\mu\)</span>. This
    parameter is not presently used in a conjugate gradient simulation</li>
<li><code>time_step</code> - value for the time step <span class="arithmatex">\(\Delta t\)</span>. This parameter
    is not presently used in a conjugate gradient simulation.</li>
<li><code>inner_iteration</code> - number of iterations between the printing out of
    energies and tolerances</li>
<li><code>outer_iteration</code> - number of outer iterations</li>
<li><code>tole</code> - value for the energy tolerance.</li>
<li><code>tolc</code> - value for the one-electron orbital tolerance.</li>
<li><code>cutoff</code> - value for the cutoff energy used to define the
    wavefunction. In addition using the CUTOFF keyword automatically
    sets the cutoff energy for the density to be twice the wavefunction
    cutoff.</li>
<li><code>ecut</code> - value for the cutoff energy used to define the density.
    Default is set to be the maximum value that will fit within the
    simulation_cell <code>cell_name</code>.</li>
<li><code>wcut</code> - value for the cutoff energy used to define the one-electron
    orbitals. Default is set to be the maximum value that will fix
    within the simulation_cell <code>cell_name</code>.</li>
<li><code>ncut</code> - value for the number of unit cells to sum over (in each
    direction) for the real space part of the Ewald summation. Note
    Ewald summation is only used if the simulation_cell is periodic.</li>
<li><code>rcut</code> - value for the cutoff radius used in the Ewald summation.
    Note Ewald summation is only used if the simulation_cell is
    periodic.</li>
</ul>
<p>Default set to be <span class="arithmatex">\(\frac{MIN(\left| \vec{a_i} \right|)}{\pi}, i=1,2,3\)</span>.</p>
<ul>
<li>(Vosko || PBE96 || revPBE) - Choose between Vosko et al&rsquo;s LDA
    parameterization or the orginal and revised Perdew, Burke, and
    Ernzerhof GGA functional.</li>
<li>SIMULATION_CELL (see section -sec:pspw_cell-)</li>
<li>BRILLOUIN_ZONE (see section -sec:band_brillouin_zone-)</li>
<li>MONKHORST-PACK - Alternatively, the MONKHORST-PACK keyword can be
    used to enter a MONKHORST-PACK sampling of the Brillouin zone.</li>
<li><code>smear</code> - value for smearing broadending</li>
<li><code>temperature</code> - same as smear but in units of K.</li>
<li>CG - optional keyword which sets the minimizer to 1</li>
<li>LMBFGS - optional keyword which sets the minimizer to 2</li>
<li>SCF - optional keyword which sets the minimizer to be a band by band
    minimizer. Several options are available for setting the density or
    potential mixing, and the type of Kohn-Sham minimizer.</li>
</ul>
<h3 id="plane-wave-density-functional-theory-brillouin-zone">Brillouin Zone<a class="headerlink" href="#plane-wave-density-functional-theory-brillouin-zone" title="Permanent link">&para;</a></h3>
<p>To supply the special points of the Brillouin zone, the user defines a
brillouin_zone sub-block within the NWPW block. Listed below is the
format of a brillouin_zone
sub-block.</p>
<pre><code>NWPW  
...   
 BRILLOUIN_ZONE 
   ZONE_NAME &lt;string name default zone_default&gt; 
   (KVECTOR &lt;real k1 k2 k3 no default&gt; &lt;real weight default (see input description)&gt;   
    ...)   
 END  
...  
END
</code></pre>
<p>The user enters the special points and weights of the Brillouin zone.
The following list describes the input in detail.</p>
<ul>
<li><code>name</code> - user-supplied name for the simulation block.</li>
<li><code>k1 k2 k3</code> - user-supplied values for a special point in the
    Brillouin zone.</li>
<li><code>weight</code> - user-supplied weight. Default is to set the weight so
    that the sum of all the weights for the entered special points adds
    up to unity.</li>
</ul>
<h4 id="plane-wave-density-functional-theory-band-structure-paths">Band Structure Paths<a class="headerlink" href="#plane-wave-density-functional-theory-band-structure-paths" title="Permanent link">&para;</a></h4>
<p>SC: gamma, m, r, x</p>
<p>FCC: gamma, k, l, u, w, x</p>
<p>BCC: gamma, h, n, p</p>
<p>Rhombohedral: not currently implemented</p>
<p>Hexagonal: gamma, a, h, k, l, m</p>
<p>Simple Tetragonal: gamma, a, m, r, x, z</p>
<p>Simple Orthorhombic: gamma, r, s, t, u, x, y, z</p>
<p>Body-Centered Tetragonal: gamma, m, n, p, x</p>
<h4 id="plane-wave-density-functional-theory-special-points-of-different-space-groups-conventional-cells">Special Points of Different Space Groups (Conventional Cells)<a class="headerlink" href="#plane-wave-density-functional-theory-special-points-of-different-space-groups-conventional-cells" title="Permanent link">&para;</a></h4>
<p>(1) P1</p>
<p>(2) P-1</p>
<p>(3)</p>
<h3 id="plane-wave-density-functional-theory-screened-exchange">Screened Exchange<a class="headerlink" href="#plane-wave-density-functional-theory-screened-exchange" title="Permanent link">&para;</a></h3>
<h3 id="plane-wave-density-functional-theory-density-of-states-and-projected-density-of-states">Density of States and Projected Density of States<a class="headerlink" href="#plane-wave-density-functional-theory-density-of-states-and-projected-density-of-states" title="Permanent link">&para;</a></h3>
<p>The &ldquo;dos&rdquo; option is used to calculate density of states using broadening
of the eigenvalues . This option can be specified without additional
parameters, i.e.</p>
<pre><code>nwpw  
  dos
end
</code></pre>
<p>in which case default values are used, or it can be specified with
additional parameters, e.g.</p>
<pre><code>nwpw  
   dos 0.002 700 -0.80000 0.8000
end
</code></pre>
<p>The parameters are</p>
<pre><code>nwpw 
 dos  [&lt;alpha&gt; &lt;npoints&gt; &lt;emin&gt; &lt;emax&gt;]
end
</code></pre>
<p>where</p>
<ul>
<li><code>alpha</code> - value for the broadening the eigenvalues, default
    0.05/27.2116 au</li>
<li><code>npoints</code> - number of plotting points in dos files, default 500</li>
<li><code>emin</code> - minimum energy in dos plots, default min(eigenvalues)-0.1
    au</li>
<li><code>emax</code> - maximimum energy in dos plots, default max(eigenvalues)+0.1
    au</li>
</ul>
<p>The units for dos parameters are in atomic units. Note that if virtual
states are specified in the pspw calculation then the virtual density of
states will also be generated in addition to the filled density of
states.</p>
<p>The following files are generated and written to the permanent_dir for
restricted calculations</p>
<ul>
<li>file_prefix.smear_dos_both - total density of states</li>
<li>file_prefix.smear_fdos_both - density of states of filled states</li>
<li>file_prefix.smear_vdos_both - density of states of virtual states</li>
</ul>
<p>For unrestricted calculations</p>
<ul>
<li>file_prefix.smear_dos_alpha - total density of states for up
    electrons</li>
<li>file_prefix.smear_dos_beta - total density of states for down
    electrons</li>
<li>file_prefix.smear_fdos_alpha - density of states for filled up
    electrons</li>
<li>file_prefix.smear_fdos_beta - density of states for filled down
    electrons</li>
<li>file_prefix.smear_vdos_alpha - density of states for virtual up
    electrons</li>
<li>file_prefix.smear_vdos_beta - density of states for virtual down
    electrons</li>
</ul>
<p>The nwpw:dos:actlist variable is used to specify the atoms used to
determine weights for dos generation. If the variable is not set then
all the atoms are used, e.g.</p>
<p><code>set nwpw:dos:actlist 1 2 3 4</code></p>
<p>For projected density of states the &ldquo;Mulliken&rdquo; keyword needs to be set,
e.g.</p>
<pre><code>nwpw
  Mulliken
  dos 
end
</code></pre>
<p>The following additional files are generated and written to the
permanent_dir for restricted calculations</p>
<ul>
<li>file_prefix.mulliken_dos_both_s - total s projected density of
    restricted states</li>
<li>file_prefix.mulliken_fdos_both_s - s projected density of states
    of filled restricted states</li>
<li>file_prefix.mulliken_vdos_both_s - s projected density of states
    of virtual restricted states</li>
<li>file_prefix.mulliken_dos_both_p - total p projected density of
    states</li>
<li>file_prefix.mulliken_fdos_both_p - p projected density of states
    of filled states</li>
<li>file_prefix.mulliken_vdos_both_p - p projected density of states
    of virtual states</li>
</ul>
<p>&hellip;</p>
<ul>
<li>file_prefix.mulliken_dos_both_all - total of projected density
    of filled and virtual restricted states</li>
<li>file_prefix.mulliken_fdos_both_all - total of projected density
    of filled restricted states</li>
<li>file_prefix.mulliken_vdos_both_all - total of projected density
    of states of virtual restricted states</li>
</ul>
<p>Similarly for unrestricted calculations</p>
<ul>
<li>file_prefix.mulliken_dos_alpha_s - total s projected density of
    up states</li>
<li>file_prefix.mulliken_fdos_alpha_s - s projected density of
    states of filled up states</li>
<li>file_prefix.mulliken_vdos_alpha_s - s projected density of
    states of virtual up states</li>
<li>file_prefix.mulliken_dos_alpha_p - total p projected density of
    up states</li>
<li>file_prefix.mulliken_fdos_alpha_p - p projected density of
    states of filled up states</li>
<li>file_prefix.mulliken_vdos_alpha_p - p projected density of
    states of virtual up states</li>
</ul>
<p>&hellip;</p>
<ul>
<li>file_prefix.mulliken_dos_alpha_all - total of projected density
    of filled up states</li>
<li>file_prefix.mulliken_fdos_alpha_all - total of projected density
    of filled up states</li>
<li>file_prefix.mulliken_vdos_alpha_all - total of projected density
    of states of virtual up states</li>
</ul>
<p>&hellip;</p>
<ul>
<li>file_prefix.mulliken_dos_beta_s - total s projected density of
    down states</li>
<li>file_prefix.mulliken_fdos_beta_s - s projected density of states
    of filled down states</li>
<li>file_prefix.mulliken_vdos_beta_s - s projected density of states
    of virtual down states</li>
<li>file_prefix.mulliken_dos_beta_p - total p projected density of
    down states</li>
<li>file_prefix.mulliken_fdos_beta_p - p projected density of states
    of filled down states</li>
<li>file_prefix.mulliken_vdos_beta_p - p projected density of states
    of virtual down states</li>
</ul>
<p>&hellip;</p>
<ul>
<li>file_prefix.mulliken_dos_beta_all - total of projected density
    of filled down states</li>
<li>file_prefix.mulliken_fdos_beta_all - total of projected density
    of filled down states</li>
<li>file_prefix.mulliken_vdos_beta_all - total of projected density
    of states of virtual down states</li>
</ul>
<h3 id="plane-wave-density-functional-theory-two-component-wavefunctions-spin-orbit-zora">Two-Component Wavefunctions (Spin-Orbit ZORA)<a class="headerlink" href="#plane-wave-density-functional-theory-two-component-wavefunctions-spin-orbit-zora" title="Permanent link">&para;</a></h3>
<h3 id="plane-wave-density-functional-theory-band_dplot-generate-gaussian-cube-files">BAND_DPLOT: Generate Gaussian Cube Files<a class="headerlink" href="#plane-wave-density-functional-theory-band_dplot-generate-gaussian-cube-files" title="Permanent link">&para;</a></h3>
<p>The BAND BAND_DPLOT task is used to generate plots of various types of
electron densities (or orbitals) of a crystal. The electron density is
calculated on the specified set of grid points from a Band calculation.
The output file generated is in the Gaussian Cube format. Input to the
BAND_DPLOT task is contained within the BAND_DPLOT sub-block.</p>
<pre><code>NWPW   
...  
 BAND_DPLOT 
   ... 
 END 
...
END
</code></pre>
<p>To run a BAND_DPLOT calculation the following directive is used:</p>
<p><code>TASK BAND BAND_DPLOT</code></p>
<p>Listed below is the format of a BAND_DPLOT
sub-block.</p>
<pre><code>NWPW
... 
 BAND_DPLOT 
   VECTORS &lt;string input_wavefunctions default input_movecs&gt;
   DENSITY [total||difference||alpha||beta||laplacian||potential default total] &lt;string density_name no default&gt;
   ELF [restricted|alpha|beta] &lt;string elf_name no default&gt;  
   ORBITAL (density || real || complex default density)  
           &lt;integer orbital_number no default&gt; 
           &lt;integer brillion_number default 1&gt; 
           &lt;string orbital_name no default&gt;  
   [LIMITXYZ [units &lt;string Units default angstroms&gt;]   
    &lt;real X_From&gt; &lt;real X_To&gt; &lt;integer No_Of_Spacings_X&gt;  
    &lt;real Y_From&gt; &lt;real Y_To&gt; &lt;integer No_Of_Spacings_Y&gt;  
    &lt;real Z_From&gt; &lt;real Z_To&gt; &lt;integer No_Of_Spacings_Z&gt;] 
 END
...
END
</code></pre>
<p>The following list describes the input for the BAND_DPLOT sub-block.</p>
<p><code>VECTORS &lt;string input_wavefunctions default input_movecs&gt;</code></p>
<p>This sub-directive specifies the name of the molecular orbital file. If
the second file is optionally given the density is computed as the
difference between the corresponding electron densities. The vector
files have to
match.</p>
<pre><code>DENSITY [total||difference||alpha||beta||laplacian||potential default total] &lt;string density_name no default&gt;
</code></pre>
<p>This sub-directive specifies, what kind of density is to be plotted. The
known names for total, difference, alpha, beta, laplacian, and
potential.</p>
<pre><code>ELF [restricted|alpha|beta] &lt;string elf_name no default&gt;
</code></pre>
<p>This sub-directive specifies that an electron localization function
(ELF) is to be
plotted.</p>
<pre><code>ORBITAL (density || real || complex default density) &lt;integer orbital_number no default&gt;&lt;integer brillion_number default 1&gt; &lt;string orbital_name no default&gt;
</code></pre>
<p>This sub-directive specifies the molecular orbital number that is to be
plotted.</p>
<pre><code>LIMITXYZ [units &lt;string Units default angstroms&gt;] 
&lt;real X_From&gt; &lt;real X_To&gt; &lt;integer No_Of_Spacings_X&gt;  
&lt;real Y_From&gt; &lt;real Y_To&gt; &lt;integer No_Of_Spacings_Y&gt;  
&lt;real Z_From&gt; &lt;real Z_To&gt; &lt;integer No_Of_Spacings_Z&gt;
</code></pre>
<p>By default the grid spacing and the limits of the cell to be plotted are
defined by the input wavefunctions. Alternatively the user can use the
LIMITXYZ sub-directive to specify other limits. The grid is generated
using No_Of_Spacings + 1 points along each direction. The known names
for Units are angstroms, au and bohr.</p>
<h2 id="plane-wave-density-functional-theory-car-parrinello">Car-Parrinello<a class="headerlink" href="#plane-wave-density-functional-theory-car-parrinello" title="Permanent link">&para;</a></h2>
<p>The Car-Parrinello task is used to perform ab initio molecular dynamics
using the scheme developed by Car and Parrinello. In this unified ab
initio molecular dynamics scheme the motion of the ion cores is coupled
to a fictitious motion for the Kohn-Sham orbitals of density functional
theory. Constant energy or constant temperature simulations can be
performed. A detailed description of this method is described in section
<a href="#plane-wave-density-functional-theory-car-parrinello-scheme-for-ab-initio-molecular-dynamics">Car-Parrinello Scheme for Ab Initio Molecular
Dynamics</a>.</p>
<p>Input to the Car-Parrinello simulation is contained within the
Car-Parrinello sub-block.</p>
<pre><code>NWPW 
... 
 Car-Parrinello  
   ...  
 END  
...
END
</code></pre>
<p>To run a Car-Parrinello calculation the following directives are used:</p>
<pre><code>TASK PSPW Car-Parrinello   
 TASK BAND Car-Parrinello
 TASK PAW Car-Parrinello
</code></pre>
<p>The Car-Parrinello sub-block contains a great deal of input, including
pointers to data, as well as parameter input. Listed below is the format
of a Car-Parrinello
sub-block.</p>
<pre><code>NWPW  
...   
 Car-Parrinello   
   CELL_NAME &lt;string cell_name default 'cell_default'&gt;
   INPUT_WAVEFUNCTION_FILENAME &lt;string input_wavefunctions default file_prefix.movecs&gt;  
   OUTPUT_WAVEFUNCTION_FILENAME &lt;string output_wavefunctions default file_prefix.movecs&gt;   
   INPUT_V_WAVEFUNCTION_FILENAME &lt;string input_v_wavefunctions default file_prefix.vmovecs&gt; 
   OUTPUT_V_WAVEFUNCTION_FILENAME &lt;string output_v_wavefunctions default file_prefix.vmovecs&gt;  
   FAKE_MASS &lt;real fake_mass default default 1000.0&gt;
   TIME_STEP &lt;real time_step default 5.0&gt;  
   LOOP &lt;integer inner_iteration outer_iteration default 10 1&gt; 
   SCALING &lt;real scale_c scale_r default 1.0 1.0&gt;   
   ENERGY_CUTOFF &lt;real ecut default (see input description)&gt;   
   WAVEFUNCTION_CUTOFF &lt;real wcut default (see input description)&gt;   
   EWALD_NCUT &lt;integer ncut default 1&gt;   
   EWALD_RCUT &lt;real rcut default (see input description)&gt;   
    XC (Vosko      || LDA          || PBE96         || revPBE       || HF            ||    
        LDA-SIC    || LDA-SIC/2    || LDA-0.4SIC    || LDA-SIC/4    || LDA-0.2SIC    ||   
        PBE96-SIC  || PBE96-SIC/2  || PBE96-0.4SIC  || PBE96-SIC/4  || PBE96-0.2SIC  ||   
        revPBE-SIC || revPBE-SIC/2 || revPBE-0.4SIC || revPBE-SIC/4 || revPBE-0.2SIC ||   
        PBE0       || revPBE0      || default Vosko) 
   [Nose-Hoover &lt;real Period_electron real Temperature_electron 
                 real Period_ion      real Temperature_ion     
                 integer Chainlength_electron integer Chainlength_ion default 100.0 298.15 100.0 298.15 1 1&gt;]   
   [TEMPERATURE &lt;real Temperature_ion      real Period_ion 
                 real Temperature_electron real Period_electron  
                 integer Chainlength_ion integer Chainlength_electron default 298.15 1200 298.15 1200.0 1 1&gt;]   
   [SA_decay &lt;real sa_scale_c sa_scale_r default 1.0 1.0&gt;] 
   XYZ_FILENAME &lt;string xyz_filename default file_prefix.xyz&gt; 
   ION_MOTION_FILENAME &lt;string ion_motion_filename default file_prefix.ion_motion
   EMOTION_FILENAME &lt;string emotion_filename default file_prefix.emotion&gt; 
   HMOTION_FILENAME &lt;string hmotion_filename nodefault&gt;
   OMOTION_FILENAME &lt;string omotion_filename nodefault&gt;
   EIGMOTION_FILENAME &lt;string eigmotion_filename nodefault&gt;  
 END  
...  
END
</code></pre>
<p>The following list describes the input for the Car-Parrinello sub-block.</p>
<ul>
<li><code>cell_name</code> - name of the the simulation_cell named <code>cell_name</code>.
    See section <a href="#plane-wave-density-functional-theory-simulation-cell">Simulation Cell</a>.</li>
<li><code>input_wavefunctions</code> - name of the file containing one-electron
    orbitals</li>
<li><code>output_wavefunctions</code> - name of the file that will contain the
    one-electron orbitals at the end of the run.</li>
<li><code>input_v_wavefunctions</code> - name of the file containing one-electron
    orbital velocities.</li>
<li><code>output_v_wavefunctions</code> - name of the file that will contain the
    one-electron orbital velocities at the end of the run.</li>
<li><code>fake_mass</code> - value for the electronic fake mass (<span class="arithmatex">\(\mu\)</span> ).</li>
<li><code>time_step</code> - value for the Verlet integration time step
    (<span class="arithmatex">\(Delta t\)</span>).</li>
<li><code>inner_iteration</code> - number of iterations between the printing out of
    energies.</li>
<li><code>outer_iteration</code> - number of outer iterations</li>
<li><code>scale_c</code> - value for the initial velocity scaling of the
    one-electron orbital velocities.</li>
<li><code>scale_r</code> - value for the initial velocity scaling of the ion
    velocities.</li>
<li><code>ecut</code> - value for the cutoff energy used to define the density.
    Default is set to be the maximum value that will fit within the
    simulation_cell <code>cell_name</code>.</li>
<li><code>wcut</code> - value for the cutoff energy used to define the one-electron
    orbitals. Default is set to be the maximum value that will fit
    within the simulation_cell <code>cell_name</code>.</li>
<li><code>ncut</code> - value for the number of unit cells to sum over (in each
    direction) for the real space part of the Ewald summation. Note
    Ewald summation is only used if the simulation_cell is periodic.</li>
<li><code>rcut</code> - value for the cutoff radius used in the Ewald summation.
    Note Ewald summation is only used if the simulation_cell is
    periodic.</li>
</ul>
<p>Default set to be <span class="arithmatex">\(\frac{MIN(\left| \vec{a_i} \right|)}{\pi}, i=1,2,3\)</span>.</p>
<ul>
<li>(Vosko || PBE96 || revPBE || &hellip;) - Choose between Vosko et al&rsquo;s LDA
    parameterization or the orginal and revised Perdew, Burke, and
    Ernzerhof GGA functional. In addition, several hybrid options.</li>
<li>Nose-Hoover or Temperature - optional subblock which if specified
    causes the simulation to perform Nose-Hoover dynamics. If this
    subblock is not specified the simulation performs constant energy
    dynamics. See section -sec:pspw_nose- for a description of the
    parameters. Note that the Temperature subblock is just a reordering
    of the Nose-Hoover subblock.<ul>
<li><code>Period_electron</code> <span class="arithmatex">\(\equiv P_{electron}\)</span> - estimated period for
    fictitious electron thermostat.</li>
<li><code>Temperature_electron</code> <span class="arithmatex">\(\equiv T_{electron}\)</span> - temperature for
    fictitious electron motion</li>
<li><code>Period_ion</code> <span class="arithmatex">\(\equiv P_{ion}\)</span> - estimated period for ionic
    thermostat</li>
<li><code>Temperature_ion</code> <span class="arithmatex">\(\equiv T_{ion}\)</span> - temperature for ion
    motion</li>
<li><code>Chainlength_electron</code> - number of electron thermostat chains</li>
<li><code>Chainlength_ion</code> - number of ion thermostat chains</li>
</ul>
</li>
<li>SA_decay - optional subblock which if specified causes the
    simulation to run a simulated annealing simulation. For simulated
    annealing to work the Nose-Hoover subblock needs to be specified.
    The initial temperature are taken from the Nose-Hoover subblock. See
    section -sec:pspw_nose- for a description of the parameters.<ul>
<li><code>sa_scale_c</code> <span class="arithmatex">\(\equiv \tau_{electron}\)</span> - decay rate in atomic
    units for electronic temperature.</li>
<li><code>sa_scale_r</code> <span class="arithmatex">\(\equiv \tau_{ionic}\)</span> - decay rate in atomic
    units for the ionic temperature.  </li>
</ul>
</li>
<li><code>xyz_filename</code> - name of the XYZ motion file generated</li>
<li><code>emotion_filename</code> - name of the emotion motion file. See section
    <a href="#plane-wave-density-functional-theory-emotion-motion-file">EMOTION motion file</a> for a
    description of the datafile.</li>
<li><code>hmotion_filenameh</code> - name of the hmotion motion file. See section
    <a href="#plane-wave-density-functional-theory-hmotion-motion-file">HMOTION motion file</a> for a
    description of the datafile.</li>
<li><code>eigmotion_filename</code> - name of the eigmotion motion file. See
    section <a href="#plane-wave-density-functional-theory-eigmotion-motion-file">EIGMOTION motion file</a>
    for a description of the datafile.</li>
<li><code>ion_motion_filename</code> - name of the ion_motion motion file. See
    section <a href="#plane-wave-density-functional-theory-ion_motion-motion-file">ION_MOTION motion file</a>- for a description of the
    datafile.</li>
<li>MULLIKEN - optional keyword which if specified causes an omotion
    motion file to be created.</li>
<li><code>omotion_filename</code> - name of the omotion motion file. See section
    <a href="#plane-wave-density-functional-theory-omotion-motion-file">OMOTION motion file</a> for a description of
    the datafile.</li>
</ul>
<p>When a DPLOT sub-block is specified the following SET directive can be
used to output dplot data during a PSPW Car-Parrinello simulation:</p>
<pre><code>set pspw_dplot:iteration_list &lt;integer list_of_iteration_numbers&gt;
</code></pre>
<p>The Gaussian cube files specified in the DPLOT sub-block are appended
with the specified iteration number.</p>
<p>For example, the following directive specifies that at the
3,10,11,12,13,14,15, and 50 iterations Gaussian cube files are to be
produced.</p>
<p><code>set pspw_dplot:iteration_list 3,10:15,50</code></p>
<h3 id="plane-wave-density-functional-theory-adding-geometry-constraints-to-a-car-parrinello-simulation">Adding Geometry Constraints to a Car-Parrinello Simulation<a class="headerlink" href="#plane-wave-density-functional-theory-adding-geometry-constraints-to-a-car-parrinello-simulation" title="Permanent link">&para;</a></h3>
<p>The Car-Parrinello module allows users to freeze the cartesian
coordinates in a simulation (Note - the Car-Parrinello code recognizes
Cartesian constraints, but it does not recognize internal coordinate
constraints). The +SET+ directive (Section <a href="#zcoord-forcing-internal-coordinates-applying-constraints-in-geometry-optimizations">Applying constraints in
geometry
optimizations</a>) 
is used to freeze atoms, by specifying a directive of the form:</p>
<pre><code>set geometry:actlist &lt;integer list_of_center_numbers&gt;
</code></pre>
<p>This defines only the centers in the list as active. All other centers
will have zero force assigned to them, and will remain frozen at their
starting coordinates during a Car-Parrinello simulation.</p>
<p>For example, the following directive specifies that atoms numbered 1, 5,
6, 7, 8, and 15 are active and all other atoms are frozen:</p>
<p><code>set geometry:actlist 1 5:8 15</code></p>
<p>or equivalently,</p>
<p><code>set geometry:actlist 1 5 6 7 8 15</code></p>
<p>If this option is not specified by entering a +SET+ directive, the
default behavior in the code is to treat all atoms as active. To revert
to this default behavior after the option to define frozen atoms has
been invoked, the +UNSET+ directive must be used (since the database is
persistent, see Section <a href="#nwchem-architecture">NWChem Architecture</a>).
The form of the +UNSET+ directive is as follows:</p>
<p><code>unset geometry:actlist</code></p>
<p>In addition, the Car-Parrinello module allows users to freeze bond
lengths via a Shake algorithm. The following +SET+ directive shows how
to do this.</p>
<p><code>set nwpw:shake_constraint "2 6 L 6.9334"</code></p>
<p>This input fixes the bond length between atoms 2 and 6 to be 6.9334
bohrs. Note that this input only recognizes bohrs.</p>
<p>When using constraints it is usually necessary to turn off center of
mass shifting. This can be done by the following +SET+ directive.</p>
<p><code>set nwpw:com_shift .false.</code></p>
<h3 id="plane-wave-density-functional-theory-car-parrinello-output-datafiles">Car-Parrinello Output Datafiles<a class="headerlink" href="#plane-wave-density-functional-theory-car-parrinello-output-datafiles" title="Permanent link">&para;</a></h3>
<h4 id="plane-wave-density-functional-theory-xyz-motion-file">XYZ motion file<a class="headerlink" href="#plane-wave-density-functional-theory-xyz-motion-file" title="Permanent link">&para;</a></h4>
<p>Data file that stores ion positions and velocities as a function of time
in XYZ
format.</p>
<pre><code>[line 1: ] n_ion
[line 2: ] do ii=1,n_ion
[line 2+ii: ] atom_name(ii), x(ii),y(ii),z(ii),vx(ii),vy(ii),vz(ii)
end do 
[line n_ion+3 ] n_nion 
 do ii=1,n_ion
[line n_ion+3+ii: ] atom_name(ii), x(ii),y(ii),z(ii), vx(ii),vy(ii),vz(ii) 
end do
[line 2*n_ion+4: ] ....
</code></pre>
<h4 id="plane-wave-density-functional-theory-ion_motion-motion-file">ION_MOTION motion file<a class="headerlink" href="#plane-wave-density-functional-theory-ion_motion-motion-file" title="Permanent link">&para;</a></h4>
<p>Datafile that stores ion positions and velocities as a function of
time</p>
<pre><code>[line 1: ] it_out, n_ion, omega, a1.x,a1.y,a1.z, a2.x,a2,y,a2.z, a3.x,a3.y,a3.z 
[line 2: ] time 
do ii=1,n_ion
[line 2+ii: ] ii, atom_symbol(ii),atom_name(ii), x(ii),y(ii),z(ii), vx(ii),vy(ii),vz(ii)  
end do
[line n_ion+3 ] time 
do do ii=1,n_ion  
[line n_ion+3+ii: ] ii, atom_symbol(ii),atom_name(ii), x(ii),y(ii),z(ii), vx(ii),vy(ii),vz(ii) 
end do 
[line 2*n_ion+4: ] ....
</code></pre>
<h4 id="plane-wave-density-functional-theory-emotion-motion-file">EMOTION motion file<a class="headerlink" href="#plane-wave-density-functional-theory-emotion-motion-file" title="Permanent link">&para;</a></h4>
<p>Datafile that store energies as a function of
time.</p>
<pre><code>[line 1: ] time, E1,E2,E3,E4,E5,E6,E7,E8,(E9,E10, if Nose-Hoover),eave,evar,have,hvar,ion_Temp 
[line 2: ] ...
</code></pre>
<p>where</p>
<pre><code>E1 = total energy
E2 = potential energy
E3 = ficticious kinetic energy
E4 = ionic kinetic energy
E5 = orbital energy
E6 = hartree energy
E7 = exchange-correlation energy 
E8 = ionic energy
eave = average potential energy 
evar = variance of potential energy
have = average total energy
hvar = variance of total energy
ion_Temp = temperature
</code></pre>
<h4 id="plane-wave-density-functional-theory-hmotion-motion-file">HMOTION motion file<a class="headerlink" href="#plane-wave-density-functional-theory-hmotion-motion-file" title="Permanent link">&para;</a></h4>
<p>Datafile that stores the rotation matrix as a function of time.</p>
<pre><code>[line 1: ] time
[line 2: ] ms,ne(ms),ne(ms)
do i=1,ne(ms)
[line 2+i: ] (hml(i,j), j=1,ne(ms)
end do
[line 3+ne(ms): ] time
[line 4+ne(ms): ] ....
</code></pre>
<h4 id="plane-wave-density-functional-theory-eigmotion-motion-file">EIGMOTION motion file<a class="headerlink" href="#plane-wave-density-functional-theory-eigmotion-motion-file" title="Permanent link">&para;</a></h4>
<p>Datafile that stores the eigenvalues for the one-electron orbitals as a
function of time.</p>
<pre><code>[line 1: ] time, (eig(i), i=1,number_orbitals) 
[line 2: ] ...
</code></pre>
<h4 id="plane-wave-density-functional-theory-omotion-motion-file">OMOTION motion file<a class="headerlink" href="#plane-wave-density-functional-theory-omotion-motion-file" title="Permanent link">&para;</a></h4>
<p>Datafile that stores a reduced representation of the one-electron
orbitals. To be used with a molecular orbital viewer that will be ported
to NWChem in the near future.</p>
<h2 id="plane-wave-density-functional-theory-born-oppenheimer-molecular-dynamics">Born-Oppenheimer Molecular Dynamics<a class="headerlink" href="#plane-wave-density-functional-theory-born-oppenheimer-molecular-dynamics" title="Permanent link">&para;</a></h2>
<pre><code>NWPW
...
  BO_STEPS &lt;integer bo_inner_iteration bo_outer_iteration default 10 100&gt; 
  BO_TIME_STEP  &lt;real bo_time_step default 5.0&gt; 
  BO_ALGORITHM [verlet|| velocity-verlet || leap-frog]
  BO_FAKE_MASS &lt;real bo_fake_mass default 500.0&gt; 
END
</code></pre>
<h2 id="plane-wave-density-functional-theory-i-pi-socket-communication">i-PI Socket Communication<a class="headerlink" href="#plane-wave-density-functional-theory-i-pi-socket-communication" title="Permanent link">&para;</a></h2>
<pre><code>NWPW
   SOCKET (UNIX || IPI_CLIENT) &lt;string socketname default (see input description)&gt;
END
</code></pre>
<p>The <code>NWPW</code> module provides native communication via the i-PI socket protocol.
The behavior is identical to the
<a href="#geometry-optimization-i-pi-socket-communication">i-PI socket communication</a>
provided by the <code>DRIVER</code> module.
The <code>NWPW</code> implementation of the <code>SOCKET</code> directive is better optimized for
plane-wave calculations.</p>
<p>For proper behavior, the <code>TASK</code> directive should be set to <code>GRADIENT</code>,
e.g. <code>TASK PSPW GRADIENT</code> or <code>TASK BAND GRADIENT</code>.</p>
<h2 id="plane-wave-density-functional-theory-metropolis-monte-carlo">Metropolis Monte-Carlo<a class="headerlink" href="#plane-wave-density-functional-theory-metropolis-monte-carlo" title="Permanent link">&para;</a></h2>
<pre><code>NWPW
...
  MC_STEPS &lt;integer mc_inner_iteration mc_outer_iteration default 10 100&gt; 
END
</code></pre>
<h2 id="plane-wave-density-functional-theory-free-energy-simulations">Free Energy Simulations<a class="headerlink" href="#plane-wave-density-functional-theory-free-energy-simulations" title="Permanent link">&para;</a></h2>
<h3 id="plane-wave-density-functional-theory-metadynamics">MetaDynamics<a class="headerlink" href="#plane-wave-density-functional-theory-metadynamics" title="Permanent link">&para;</a></h3>
<p><img alt="Metadynamics bias potential. Courtesy of Raymond Atta-Fynn" src="ray2.gif" title="Metadynamics bias potential. Courtesy of Raymond Atta-Fynn&quot;.md) ![Metadynamics simulation of the first hydrolysis of U(IV) from 1.  Courtesy of Raymond Atta-Fynn](ray.gif &quot;Metadynamics simulation of the first hydrolysis of U(IV.md) from 1. Courtesy of Raymond Atta-Fynn" />
Metadynamics<a href="http://en.wikipedia.org/wiki/Metadynamics">2</a><a href="http://iopscience.iop.org/0034-4885/71/12/126601">3</a><a href="http://people.sissa.it/~laio/Research/Res_metadynamics.php">4</a>
is a powerful, non-equilibrium molecular dynamics method which
accelerates the sampling of the multidimensional free energy surfaces of
chemical reactions. This is achieved by adding an external
time-dependent bias potential that is a function of user defined
collective variables, <span class="arithmatex">\(\mathbf{s}\)</span>. The bias potential discourages the
system from sampling previously visited values of <span class="arithmatex">\(\mathbf{s}\)</span> (i.e.,
encourages the system to explore new values of <span class="arithmatex">\(\mathbf{s}\)</span>. During the
simulation the bias potential accumulates in low energy wells which then
allows the system to cross energy barriers much more quickly than would
occur in standard dynamics. The collective variable <span class="arithmatex">\(\mathbf{s}\)</span> is a
generic function of the system coordinates, <span class="arithmatex">\(\mathbf{R}\)</span> (e.g. bond
distance, bond angle, coordination numbers, etc.) that is capable of
describing the chemical reaction of interest.
<span class="arithmatex">\(\mathbf{s}\left(\mathbf{R}\right)\)</span> can be regarded as a reaction
coordinate if it can distinguish between the reactant, transition, and
products states, and also capture the kinetics of the reaction.</p>
<p>The biasing is achieved by flooding the energy landscape with
repulsive Gaussian hills centered on the current location of
<span class="arithmatex">\(\mathbf{s}\left(\mathbf{R}\right)\)</span> at a constant time interval
<span class="arithmatex">\(\Delta t\)</span>. If the height of the Gaussians is constant in time then we
have standard metadynamics; if the heights vary (slowly decreased) over
time then we have well-tempered metadynamics. In between the addition of
Gaussians, the system is propagated by normal (but out of equilibrium)
dynamics. Suppose that the dimension of the collective space is <span class="arithmatex">\(d\)</span>,
i.e.
<span class="arithmatex">\(\mathbf{s}\left(\mathbf{R}\right)=\bigl[s_1\left(\mathbf{R}\right),s_2\left(\mathbf{R}\right),\ldots,s_d\left(\mathbf{R}\right)\bigr]\)</span>
and that prior to any time <span class="arithmatex">\(t\)</span> during the simulation, <span class="arithmatex">\(N +1\)</span>
Gaussians centered on 
<span class="arithmatex">\(\mathbf{S}^{t_g}\)</span>
are deposited along the
trajectory of <span class="arithmatex">\(\mathbf{s}\left(\mathbf{R}\right)\)</span> at times
<span class="arithmatex">\(t_g = 0, \Delta t, 2\Delta t, \ldots ,N\Delta t\)</span>. Then, the value of
the bias potential, <span class="arithmatex">\(V\)</span>, at an arbitrary point,
<span class="arithmatex">\(\mathbf{s}\left(\mathbf{R}\right)=\bigl[s_1\left(\mathbf{R}\right),s_2\left(\mathbf{R}\right),\ldots,s_d\left(\mathbf{R}\right)\bigr]\)</span>,
along the trajectory of <span class="arithmatex">\(\mathbf{s}\left(\mathbf{R}\right)\)</span> at time <span class="arithmatex">\(t\)</span>
is given
by</p>
<div class="arithmatex">\[V_{meta}\left(\pmb{s},t\right) = \sum_{t_g=0}^{t_g&lt;} W(t) \exp\left(-\sum_{i=1}^{d}\frac{({s_i}-{s_i}^{t_g})^2}{2\sigma_i^2}\right)\]</div>
<p>where
<span class="arithmatex">\(W(t)=W_0 \exp\left(-\frac{V_{meta}\left(\mathbf{s},t-\Delta t\right)}{k_B T_{tempered}}\right)\)</span>
is the time-dependent Gaussian height. <span class="arithmatex">\(\sigma_i \,(i=1,2,\ldots,d)\)</span>
and <span class="arithmatex">\(W_0\)</span> are width and initial height respectively of Gaussians, and
<span class="arithmatex">\(T_{tempered}\)</span>
is the tempered metadynamics temperature.
<span class="arithmatex">\(T_{tempered}=0\)</span>
corresponds to standard molecular dynamics because
<span class="arithmatex">\(W(t)=0\)</span>
and therfore there is no bias. <span class="arithmatex">\(T_{tempered}=\infty\)</span>
corresponds to standard metadynamics since in this case
<span class="arithmatex">\(W(t)=W_0\)</span>=constant. A positive, finite value of <span class="arithmatex">\(T_{tempered}\)</span> (eg.
<span class="arithmatex">\(T_{tempered}\)</span> &gt;=1500 K) corresponds to <em>well-tempered</em> metadynamics in which  <span class="arithmatex">\(0 &lt; W(t)&lt;= W_0\)</span>.</p>
<p>For sufficiently large <span class="arithmatex">\(t\)</span>, the history potential
<span class="arithmatex">\(V_{meta}\left(\mathbf{s},t\right)\)</span> will nearly flatten the free energy
surface, <span class="arithmatex">\(F(\mathbf{s})\)</span>, along <span class="arithmatex">\(\mathbf{S}\)</span> and an unbiased estimator
of F(s) is given by</p>
<div class="arithmatex">\[F(\mathbf{s}) = -\left(1+\frac{T}{T_{tempered}}\right)\lim_{t \to \infty} V_{meta}(\mathbf{s},t)\]</div>
<h4 id="plane-wave-density-functional-theory-input">Input<a class="headerlink" href="#plane-wave-density-functional-theory-input" title="Permanent link">&para;</a></h4>
<p>Input to a metadynamics simulation is contained within the METADYNAMICS
sub-block. Listed below is the the format of a METADYNAMICS
sub-block,</p>
<pre><code>NWPW  
 METADYNAMICS
    [
    BOND &lt;integer atom1_index no default&gt; &lt;integer atom2_index no default&gt; 
         [W &lt;real w default 0.00005&gt;]  
         [SIGMA &lt;real sigma default 0.1&gt;] 
         [RANGE &lt;real a b default (see input description)&gt;]  
         [NRANGE &lt;integer nrange default 501&gt;] 
    ...]  
    [
    ANGLE &lt;integer atom1_index no default&gt; &lt;integer atom2_index no default&gt; &lt;integer atom3_index no default&gt;  
         [W &lt;real w default 0.00005&gt;] 
         [SIGMA &lt;real sigma default 0.1&gt;]
         [RANGE &lt;real a b default 0]
         [NRANGE &lt;integer nrange default 501&gt;]  
    ...]
    [
    COORD_NUMBER [INDEX1 &lt;integer_list atom1_indexes no default&gt;][INDEX2 &lt;integer_list atom2_indexes no default&gt;]  
         [SPRIK] 
         [N &lt;real n default 6.0&gt;]
         [M &lt;real m default 12.0&gt;]
         [R0 &lt;real r0 default 3.0&gt;]  

         [W &lt;real w default 0.00005&gt;] 
         [SIGMA &lt;real sigma default 0.1&gt;] 
         [RANGE &lt;real a b no default&gt;]  
         [NRANGE &lt;integer nrange default 501&gt;] 
    ...] 
    [  
    N-PLANE  &lt;integer atom1_index no default&gt; &lt;integer_list atom_indexes no default&gt;  
         [W &lt;real w default 0.00005&gt;]   
         [SIGMA &lt;real sigma default 0.1&gt;]  
         [RANGE &lt;real a b no default&gt;] 
         [NRANGE &lt;integer nrange default 501&gt;]  
         [NVECTOR &lt;real(3) nx ny nz&gt;] 
    ...] 
    [UPDATE &lt;integer meta_update default 1&gt;]  
    [PRINT_SHIFT &lt;integer print_shift default 0&gt;]
    [TEMPERED &lt;real tempered_temperature no default&gt;]  
    [BOUNDARY &lt;real w_boundary sigma_boundary no default&gt;]
  END
END
</code></pre>
<p>Multiple collective variables can be defined at the same time, e.g.</p>
<pre><code>NWPW  
 METADYNAMICS 
    BOND 1 8  W 0.0005 SIGMA 0.1 
    BOND 1 15 W 0.0005 SIGMA 0.1 
  END
END
</code></pre>
<p>will produce a two-dimensional potential energy surface.</p>
<h3 id="plane-wave-density-functional-theory-tamd-temperature-accelerated-molecular-dynamics">TAMD - Temperature Accelerated Molecular Dynamics<a class="headerlink" href="#plane-wave-density-functional-theory-tamd-temperature-accelerated-molecular-dynamics" title="Permanent link">&para;</a></h3>
<h4 id="plane-wave-density-functional-theory-input_1">Input<a class="headerlink" href="#plane-wave-density-functional-theory-input_1" title="Permanent link">&para;</a></h4>
<h3 id="plane-wave-density-functional-theory-collective-variables">Collective Variables<a class="headerlink" href="#plane-wave-density-functional-theory-collective-variables" title="Permanent link">&para;</a></h3>
<h4 id="plane-wave-density-functional-theory-bond-distance-collective-variable">Bond Distance Collective Variable<a class="headerlink" href="#plane-wave-density-functional-theory-bond-distance-collective-variable" title="Permanent link">&para;</a></h4>
<p>This describes the bond distance between any pair of atoms <span class="arithmatex">\(i\)</span> and
<span class="arithmatex">\(j\)</span>:</p>
<div class="arithmatex">\[s\left(r_{ij}\right) = \left \vert \mathbf{r}_{i}-\mathbf{r}_j\right\vert = r_{ij}\]</div>
<h4 id="plane-wave-density-functional-theory-angle-collective-variable">Angle Collective Variable<a class="headerlink" href="#plane-wave-density-functional-theory-angle-collective-variable" title="Permanent link">&para;</a></h4>
<p>This describes the bond angle formed at <span class="arithmatex">\(i\)</span> by the triplet <span class="arithmatex">\(&lt;ijk&gt;\)</span></p>
<div class="arithmatex">\[s\left(r_{ij},r_{ik}\right) = \frac{\pmb{r}_{ij}\cdot\pmb{r}_{ik}}{r_{ij}r_{ik}}\]</div>
<h4 id="plane-wave-density-functional-theory-coordination-collective-variable">Coordination Collective Variable<a class="headerlink" href="#plane-wave-density-functional-theory-coordination-collective-variable" title="Permanent link">&para;</a></h4>
<p>The coordination number collective variable is defined as</p>
<div class="arithmatex">\[s\left(r_{ij},r_{0}\right) =\sum_{i,j}\xi_{ij}\]</div>
<p>where the summation over <span class="arithmatex">\(i\)</span> and <span class="arithmatex">\(j\)</span> runs over two types of atoms,
<span class="arithmatex">\(\xi_{ij}\)</span> is the <em>weighting function</em>, and <span class="arithmatex">\(r_{0}\)</span> is the cut-off
distance.
In the standard procedure for computing the coordination
number, <span class="arithmatex">\(\xi_{ij}\)</span> =1 if <span class="arithmatex">\(r_{ij} &lt; r_0\)</span>, otherwise 
<span class="arithmatex">\(\xi_{ij}\)</span> =0,
implying that <span class="arithmatex">\(\xi_{ij}\)</span> is not continuous when 
<span class="arithmatex">\(r_{ij}=r_{0}\)</span>. To
ensure a smooth and continuous definition of the coordination number, we
adopt two variants of the weighting function. The first variant
is</p>
<div class="arithmatex">\[\xi_{ij} = \frac{1-\left(r_{ij}/r_{0}\right)^n}{1-\left(r_{ij}/r_{0}\right)^m}\]</div>
<p>where <span class="arithmatex">\(n\)</span> and <span class="arithmatex">\(m\)</span> are integers (m &gt; n) chosen such that
<span class="arithmatex">\(\xi_{ij}\approx 1\)</span> when <span class="arithmatex">\(r_{ij} &lt; r_0\)</span>  and
<span class="arithmatex">\(\xi_{ij}\rightarrow 0\)</span> when <span class="arithmatex">\(r_{ij}\)</span> is much larger than <span class="arithmatex">\(r_{0}\)</span>.
For example, the parameters of the O-H coordination in water is well
described by <span class="arithmatex">\(r_{0}\)</span> =1.6 , 
<span class="arithmatex">\(n=6\)</span>
and <span class="arithmatex">\(m=18\)</span>. 
In practice, <span class="arithmatex">\(n\)</span>
and <span class="arithmatex">\(m\)</span> must tuned for a given <span class="arithmatex">\(r_{0}\)</span> to ensure that <span class="arithmatex">\(\xi_{ij}\)</span>
is smooth and satisfies the above mentioned properties, particularly the
large <span class="arithmatex">\(r_{ij}\)</span></p>
<p>The second form of the weighting function, which is due to Sprik, is</p>
<div class="arithmatex">\[\xi_{ij}=\frac{1}{1 + \exp\left[n\left(r_{ij}-r_{0}\right)\right]}\]</div>
<p>In this definition <span class="arithmatex">\(\xi_{ij}\)</span> is analogous to the Fermi function and
its width is controlled by the parameter <span class="arithmatex">\(\frac{1}{n}\)</span>. Large and
small values of <span class="arithmatex">\(n\)</span> respectively correspond to sharp and soft
transitions at <span class="arithmatex">\(r_{ij} = r_{0}\)</span>. 
Furthermore <span class="arithmatex">\(\xi_{ij}\)</span> should
approach 1 and 0 when <span class="arithmatex">\(r_{ij} &lt; 0\)</span>
and <script type="math/tex">r_{ij} < r_0 </script>
respectively. In practice <span class="arithmatex">\(n\)</span> =6-10  <span class="arithmatex">\(^{-1}\)</span>. 
For example, a good set
of parameters of the O-H coordination in water is <span class="arithmatex">\(r_{0}\)</span> =1.4  and
<span class="arithmatex">\(n\)</span> =10  <span class="arithmatex">\(^{-1}\)</span>.</p>
<h4 id="plane-wave-density-functional-theory-n-plane-collective-variable">N-Plane Collective Variable<a class="headerlink" href="#plane-wave-density-functional-theory-n-plane-collective-variable" title="Permanent link">&para;</a></h4>
<p>The N-Plane collective variable is useful for probing the adsorption of
adatom/admolecules to a surface. It is defined as the <em>average distance</em>
of the adatom/admolecule from a given layer in the slab along the
surface normal:</p>
<div class="arithmatex">\[s = Z_{ads}-\frac{1}{N_{plane}}\sum_{i=1}^{N_{plane}}Z_i\]</div>
<p>where <span class="arithmatex">\(Z_{ads}\)</span> denotes the position of the adatom/admolecule/impurity
along the surface normal (here, we assume the surface normal to be the
z-axis) and the summation over 
<span class="arithmatex">\(i\)</span> 
runs over 
<span class="arithmatex">\(N_{plane}\)</span> 
atoms at
<span class="arithmatex">\(Z_i\)</span> which form the layer. The layer could be on the face or in the
interior of the
slab.</p>
<h4 id="plane-wave-density-functional-theory-user-defined-collective-variable">User defined Collective Variable<a class="headerlink" href="#plane-wave-density-functional-theory-user-defined-collective-variable" title="Permanent link">&para;</a></h4>
<h2 id="plane-wave-density-functional-theory-extended-x-ray-absorption-fine-structure-exafs-integration-with-feff6l">Extended X-Ray Absorption Fine Structure (EXAFS) - Integration with FEFF6L<a class="headerlink" href="#plane-wave-density-functional-theory-extended-x-ray-absorption-fine-structure-exafs-integration-with-feff6l" title="Permanent link">&para;</a></h2>
<h2 id="plane-wave-density-functional-theory-frozen-phonon-calculations">Frozen Phonon Calculations<a class="headerlink" href="#plane-wave-density-functional-theory-frozen-phonon-calculations" title="Permanent link">&para;</a></h2>
<h2 id="plane-wave-density-functional-theory-steepest-descent">Steepest Descent<a class="headerlink" href="#plane-wave-density-functional-theory-steepest-descent" title="Permanent link">&para;</a></h2>
<p>The functionality of this task is now performed automatically by the
PSPW and BAND. For backward compatibility, we provide a description of
the input to this task.</p>
<p>The steepest_descent task is used to optimize the one-electron orbitals
with respect to the total energy. In addition it can also be used to
optimize geometries. This method is meant to be used for coarse
optimization of the one-electron orbitals.</p>
<p>Input to the steepest_descent simulation is contained within the
steepest_descent sub-block.</p>
<pre><code>NWPW   
...   
 STEEPEST_DESCENT   
   ...   
 END   
...  
END
</code></pre>
<p>To run a steepest_descent calculation the following directive is used:</p>
<pre><code>TASK PSPW steepest_descent  
TASK BAND steepest_descent 
</code></pre>
<p>The steepest_descent sub-block contains a great deal of input,
including pointers to data, as well as parameter input. Listed below is
the format of a STEEPEST_DESCENT
sub-block.</p>
<pre><code>NWPW  
...   
 STEEPEST_DESCENT   
   CELL_NAME &lt;string cell_name&gt;   
   [GEOMETRY_OPTIMIZE]   
   INPUT_WAVEFUNCTION_FILENAME &lt;string input_wavefunctions default file_prefix.movecs&gt;   
   OUTPUT_WAVEFUNCTION_FILENAME &lt;string output_wavefunctions default file_prefix.movecs&gt;   
   FAKE_MASS &lt;real fake_mass default 400000.0&gt;   
   TIME_STEP &lt;real time_step default 5.8&gt;   
   LOOP &lt;integer inner_iteration outer_iteration default 10 1&gt;   
   TOLERANCES &lt;real tole tolc tolr default 1.0d-9 1.0d-9 1.0d-4&gt;   
   ENERGY_CUTOFF &lt;real ecut default (see input desciption)&gt;   
   WAVEFUNCTION_CUTOFF &lt;real wcut default (see input description)&gt;   
   EWALD_NCUT &lt;integer ncut default 1&gt;   
   EWALD_RCUT &lt;real rcut default (see input description)&gt;   
   XC (Vosko      || LDA          || PBE96         || revPBE       || HF            ||    
       LDA-SIC    || LDA-SIC/2    || LDA-0.4SIC    || LDA-SIC/4    || LDA-0.2SIC    ||   
       PBE96-SIC  || PBE96-SIC/2  || PBE96-0.4SIC  || PBE96-SIC/4  || PBE96-0.2SIC  ||   
       revPBE-SIC || revPBE-SIC/2 || revPBE-0.4SIC || revPBE-SIC/4 || revPBE-0.2SIC ||   
       PBE0       || revPBE0      || default Vosko)   
   [MULLIKEN]  
 END  
...  
END
</code></pre>
<p>The following list describes the input for the STEEPEST_DESCENT
sub-block.</p>
<ul>
<li><code>cell_name</code> - name of the simulation_cell named <code>cell_name</code>. See
    <a href="#plane-wave-density-functional-theory-simulation-cell">Simulation Cell</a>.</li>
<li>GEOMETRY_OPTIMIZE - optional keyword which if specified turns on
    geometry optimization.</li>
<li><code>input_wavefunctions</code> - name of the file containing one-electron
    orbitals</li>
<li><code>output_wavefunctions</code> - name of the file tha will contain the
    one-electron orbitals at the end of the run.</li>
<li><code>fake_mass</code> - value for the electronic fake mass <span class="arithmatex">\(\mu\)</span></li>
<li><code>time_step</code> - value for the time step <span class="arithmatex">\(Delta t\)</span>.</li>
<li><code>inner_iteration</code> - number of iterations between the printing out of
    energies and tolerances</li>
<li><code>outer_iteration</code> - number of outer iterations</li>
<li><code>tole</code> - value for the energy tolerance.</li>
<li><code>tolc</code> - value for the one-electron orbital tolerance.</li>
<li><code>tolr</code> - value for the ion position tolerance.</li>
<li><code>ecut</code> - value for the cutoff energy used to define the density.
    Default is set to be the maximum value that will fit within the
    simulation_cell <code>cell_name</code>.</li>
<li><code>wcut</code> - value for the cutoff energy used to define the one-electron
    orbitals. Default is set to be the maximum value that will fit
    within the simulation_cell <code>cell_name</code>.</li>
<li><code>ncut</code> - value for the number of unit cells to sum over (in each
    direction) for the real space part of the Ewald summation. Note
    Ewald summation is only used if the simulation_cell is periodic.</li>
<li><code>rcut</code> - value for the cutoff radius used in the Ewald summation.
    Note Ewald summation is only used if the simulation_cell is
    periodic.</li>
</ul>
<p>Default set to be <span class="arithmatex">\(\frac{MIN(\left| \vec{a_i} \right|)}{\pi}, i=1,2,3\)</span>.</p>
<ul>
<li>(Vosko || PBE96 || revPBE || &hellip;) - Choose between Vosko et al&rsquo;s LDA
    parameterization or the orginal and revised Perdew, Burke, and
    Ernzerhof GGA functional. In addition, several hybrid options
    (hybrid options are not available in BAND).</li>
<li>MULLIKEN - optional keyword which if specified causes a Mulliken
    analysis to be performed at the end of the simulation.</li>
</ul>
<h2 id="plane-wave-density-functional-theory-simulation-cell">Simulation Cell<a class="headerlink" href="#plane-wave-density-functional-theory-simulation-cell" title="Permanent link">&para;</a></h2>
<p>The simulation cell parameters are entered by defining a
simulation_cell sub-block within the PSPW block. Listed below is the
format of a simulation_cell sub-block.</p>
<pre><code>NWPW 
...   
 SIMULATION_CELL  [units &lt;string units default bohrs&gt;]
    CELL_NAME &lt;string name default 'cell_default'&gt;  
    BOUNDARY_CONDITIONS (periodic || aperiodic default periodic) 
    LATTICE_VECTORS   
      &lt;real a1.x a1.y a1.z default 20.0 0.0 0.0&gt; 
      &lt;real a2.x a2.y a2.z default 0.0 20.0 0.0&gt;   
      &lt;real a3.x a3.y a3.z default 0.0 0.0 20.0&gt; 
    NGRID &lt;integer na1 na2 na3 default 32 32 32&gt; 
  END 
...  
END
</code></pre>
<p>Basically, the user needs to enter the dimensions, gridding and boundary
conditions of the simulation cell. The following list describes the
input in detail.</p>
<ul>
<li><code>name</code> - user-supplied name for the simulation block.</li>
<li>periodic - keyword specifying that the simulation cell has periodic
    boundary conditions.</li>
<li>aperiodic - keyword specifying that the simulation cell has
    free-space boundary conditions.</li>
<li><code>a1.x a1.y a1.z</code> - user-supplied values for the first lattice
    vector</li>
<li><code>a2.x a2.y a2.z</code> - user-supplied values for the second lattice
    vector</li>
<li><code>a3.x a3.y a3.z</code> - user-supplied values for the third lattice
    vector</li>
<li><code>na1 na2 na3</code> - user-supplied values for discretization along
    lattice vector directions.</li>
</ul>
<p>Alternatively, instead of explicitly entering lattice vectors, users can
enter the unit cell using the standard cell parameters, a, b, c,
<span class="arithmatex">\(alpha\)</span>,
<span class="arithmatex">\(\beta\)</span>, 
and <span class="arithmatex">\(\gamma\)</span>, by using the LATTICE block. The
format for input is as follows:</p>
<pre><code>NWPW 
... 
 SIMULATION_CELL [units &lt;string units default bohrs&gt;]  
   ...   
   LATTICE   
     [lat_a &lt;real a default 20.0&gt;]  
     [lat_b &lt;real b default 20.0&gt;]  
     [lat_c &lt;real c default 20.0&gt;] 
     [alpha &lt;real alpha default 90.0&gt;]  
     [beta &lt;real beta default 90.0&gt;] 
     [gamma &lt;real gamma default 90.0&gt;]  
   END 
   ...   
 END
...
END
</code></pre>
<p>The user can also enter the lattice vectors of standard unit cells using
the keywords SC, FCC, BCC, for simple cubic, face-centered cubic, and
body-centered cubic respectively. Listed below is an example of the
format of this type of input.</p>
<pre><code>NWPW 
... 
 SIMULATION_CELL [units &lt;string units default bohrs&gt;]
   SC 20.0 
   .... 
 END
...
END
</code></pre>
<p>Finally, the lattice vectors from the unit cell can also be defined
using the fractional coordinate input in the GEOMETRY input (see section
<a href="#system----lattice-parameters-for-periodic-systems">Geometry Lattice
Parameters</a>).
Listed below is an example of the format of this type of input for an 8
atom silicon carbide unit cell.</p>
<pre><code>geometry units au  
 system crystal 
   lat_a 8.277
   lat_b 8.277   
   lat_c 8.277   
   alpha 90.0  
   beta 90.0   
   gamma 90.0   
 end
 Si -0.50000 -0.50000 -0.50000
 Si  0.00000  0.00000 -0.50000 
 Si  0.00000 -0.50000  0.00000
 Si -0.50000  0.00000  0.00000  
 C  -0.25000 -0.25000 -0.25000  
 C   0.25000  0.25000 -0.25000 
 C   0.25000 -0.25000  0.25000 
 C  -0.25000  0.25000  0.25000  
end
</code></pre>
<p>Warning - Currently only the &ldquo;system crystal&rdquo; option is recognized by
NWPW. The &ldquo;system slab&rdquo; and &ldquo;system polymer&rdquo; options will be supported
in the future.</p>
<h2 id="plane-wave-density-functional-theory-unit-cell-optimization">Unit Cell Optimization<a class="headerlink" href="#plane-wave-density-functional-theory-unit-cell-optimization" title="Permanent link">&para;</a></h2>
<p>The PSPW module using the DRIVER geometry optimizer can optimize a
crystal unit cell. Currently this type of optimization works only if the
geometry is specified in fractional coordinates. The following SET
directive is used to tell the DRIVER geometry optimizer to optimize the
crystal unit cell in addition to the geometry.</p>
<p><code>set includestress .true.</code></p>
<h2 id="plane-wave-density-functional-theory-smear-fractional-occupation-of-the-molecular-orbitals">SMEAR - Fractional Occupation of the Molecular Orbitals<a class="headerlink" href="#plane-wave-density-functional-theory-smear-fractional-occupation-of-the-molecular-orbitals" title="Permanent link">&para;</a></h2>
<p>The smear keyword to turn on fractional occupation of the molecular
orbitals in PSPW and BAND
calculations</p>
<pre><code>SMEAR &lt;sigma default 0.001&gt; [TEMPERATURE &lt;temperature&gt;]
                            [FERMI || GAUSSIAN || MARZARI-VANDERBILT default FERMI]
                            [ORBITALS &lt;integer orbitals default 4&gt;]
</code></pre>
<p>Fermi-Dirac (FERMI), Gaussian, and Marzari-Vanderbilt broadening functions are
available. The ORBITALS keyword is used to change the number of virtual
orbitals to be used in the calculation. Note to use this option the user
must currently use the SCF minimizer. The following SCF options are
recommended for running fractional occupation</p>
<p><code>SCF Anderson outer_iterations 0 Kerker 2.0</code></p>
<h2 id="plane-wave-density-functional-theory-spin-penalty-functions">Spin Penalty Functions<a class="headerlink" href="#plane-wave-density-functional-theory-spin-penalty-functions" title="Permanent link">&para;</a></h2>
<p>Spin-penalty functions makes it easier to define antiferromagnetic
structures. These functions are implemented by adding a scaling factor
to the non-local psp for up/down spins on atoms and angular momentum
that you specify.</p>
<p>Basically, the pseudopotential
energy</p>
<div class="arithmatex">\[E_{psp}= \sum_{\sigma=\uparrow,\downarrow} \sum_{i=1}^{n_{elc}^\sigma} \sum_{I=1}^{n_{ions}} \left( &lt;\psi_i^\sigma|V_{local}^{I}|\psi_i^\sigma&gt;  + \sum_{l=0}^{l_{max}^I} \sum_{m=-l}^{l} \sum_{n=1}^{n_{max}^I} \sum_{n'=1}^{n_{max}^I} &lt;\psi_i^\sigma|P_{nlm}^I&gt; h_{l,n,n'}^I &lt;P_{n'lm}^{I}|\psi_i^\sigma&gt; \right)\]</div>
<p>was modified
to</p>
<div class="arithmatex">\[E_{psp}= \sum_{\sigma=\uparrow,\downarrow} \sum_{i=1}^{n_{elc}^\sigma} \sum_{I=1}^{n_{ions}} \left( &lt;\psi_i^\sigma|V_{local}^{I}|\psi_i^\sigma&gt;  + \sum_{l=0}^{l_{max}^I} \sum_{m=-l}^{l} \sum_{n=1}^{n_{max}^I} \sum_{n'=1}^{n_{max}^I}  \left(1-\delta_{l,l^\sigma} \delta_{I,ionlist^\sigma}(\xi^\sigma-1)\right) &lt;\psi_i^\sigma|P_{nlm}^I&gt; h_{l,n,n'}^I &lt;P_{n'lm}^{I}|\psi_i^\sigma&gt; \right)\]</div>
<p>An example input is as follows:</p>
<pre><code>title &quot;hematite 10 atoms&quot;

start hema10

memory 1900 mb

permanent_dir ./perm
scratch_dir   ./perm

geometry center noautosym noautoz print
 system crystal
   lat_a 5.42 
   lat_b 5.42 
   lat_c 5.42 
   alpha 55.36  
   beta  55.36  
   gamma 55.36  
 end
Fe                  0.355000   0.355000   0.355000
Fe                  0.145000   0.145000   0.145000  
Fe                 -0.355000  -0.355000  -0.355000  
Fe                  0.855000   0.855000   0.855000  
O                   0.550000  -0.050000   0.250000  
O                   0.250000   0.550000  -0.050000  
O                  -0.050000   0.250000   0.550000  
O                  -0.550000   0.050000  -0.250000  
O                  -0.250000  -0.550000   0.050000  
O                   0.050000  -0.250000  -0.550000\  
end  

nwpw
  virtual 8
  odft
  ewald_rcut 3.0
  ewald_ncut 8 
  xc pbe96
  lmbfgs 
  mult 1
  dplot
     density diff diff1.cube
  end

  #spin penalty functions  
  pspspin up   d -1.0 1:2  
  pspspin down d -1.0 3:4  
end
task pspw energy 
task pspw pspw_dplot  

nwpw
  pspspin off
  dplot
     density diff diff2.cube
  end  
end 
task pspw energy
task pspw pspw_dplot
</code></pre>
<h2 id="plane-wave-density-functional-theory-aimdmm-qmmm">AIMD/MM (QM/MM)<a class="headerlink" href="#plane-wave-density-functional-theory-aimdmm-qmmm" title="Permanent link">&para;</a></h2>
<p>A QM/MM capability is available that is integrated with PSPW module and
can be used with Car-Parrinello simulations. Currently, the input is not
very robust but it is straightforward. The first step to run a QM/MM
simulations is to define the MM atoms in the geometry block. The MM
atoms must be at the end of the geometry and a carat, &rdquo; ^ &ldquo;, must be
appended to the end of the atom name, e.g.</p>
<pre><code>geometry units angstrom nocenter noautosym noautoz print xyz  
 C -0.000283 0.000106 0.000047  
 Cl -0.868403 1.549888 0.254229  
 Cl 0.834043 -0.474413 1.517103  
 Cl -1.175480 -1.275747 -0.460606  
 Cl 1.209940 0.200235 -1.310743  
 O^ 0.3226E+01 -0.4419E+01 -0.5952E+01   
 H^ 0.3193E+01 -0.4836E+01 -0.5043E+01   
 H^ 0.4167E+01 -0.4428E+01 -0.6289E+01  
 O^ 0.5318E+01 -0.3334E+01 -0.1220E+01  
 H^ 0.4978E+01 -0.3040E+01 -0.2113E+01  
 H^ 0.5654E+01 -0.2540E+01 -0.7127E+00  
end
</code></pre>
<p>Another way to specify the MM atoms is to use the mm_tags option which
appends the atoms with a &rdquo; ^ &ldquo;.</p>
<pre><code>geometry units angstrom nocenter noautosym noautoz print xyz  
 C -0.000283 0.000106 0.000047  
 Cl -0.868403 1.549888 0.254229  
 Cl 0.834043 -0.474413 1.517103  
 Cl -1.175480 -1.275747 -0.460606  
 Cl 1.209940 0.200235 -1.310743  
 O 0.3226E+01 -0.4419E+01 -0.5952E+01   
 H 0.3193E+01 -0.4836E+01 -0.5043E+01   
 H 0.4167E+01 -0.4428E+01 -0.6289E+01  
 O 0.5318E+01 -0.3334E+01 -0.1220E+01  
 H 0.4978E+01 -0.3040E+01 -0.2113E+01  
 H 0.5654E+01 -0.2540E+01 -0.7127E+00  
end  
NWPW  
  QMMM  
    mm_tags 6:11  
  END  
END
</code></pre>
<p>The option &ldquo;mm_tags off&rdquo; can be used to remove the &rdquo; ^ &rdquo; from the
atoms, i.e.</p>
<pre><code>NWPW  
  QMMM  
     mm_tags 6:11 off  
  END  
END 
</code></pre>
<p>Next the pseudopotentials have be defined for the every type of MM atom
contained in the geometry blocks. The following local pseudopotential
suggested by Laio, VandeVondele and Rothlisberger can be automatically
generated.</p>
<div class="arithmatex">\[\begin{matrix}V(\vec{r}) = -Z_{ion}\frac{{r_c}^{n_{\sigma}} - r^{n_{\sigma}}}{-sign(Z_{ion})*{r_c}^{n_{\sigma}+1}-r^{n_{\sigma}+1}}\end{matrix}\]</div>
<p>The following input To define this pseudopo the O^ MM atom using the
following input</p>
<pre><code>NWPW   
 QMMM   
   mm_psp O^ -0.8476 4 0.70   
 END  
END
</code></pre>
<p>defines the local pseudopotential for the O^ MM atom , where
<span class="arithmatex">\(Z_{ion}=-0.8476\)</span>, 
<span class="arithmatex">\(n_{\sigma}=4\)</span>, and <span class="arithmatex">\(r_c=0.7\)</span>. The following
input can be used to define the local pseudopotentials for all the MM
atoms in the geometry block defined above</p>
<pre><code>NWPW   
 QMMM   
   mm_psp O^ -0.8476 4 0.70   
   mm_psp H^ 0.4238 4 0.40   
 END  
END
</code></pre>
<p>Next the Lenard-Jones potentials for the QM and MM atoms need to be
defined. This is done as as follows</p>
<pre><code>NWPW   
 QMMM   
   lj_ion_parameters C 3.41000000d0 0.10d0   
   lj_ion_parameters Cl 3.45000000d0 0.16d0   
   lj_ion_parameters O^ 3.16555789d0 0.15539425d0   
 END  
END
</code></pre>
<p>Note that the Lenard-Jones potential is not defined for the MM H atoms
in this example. The final step is to define the MM fragments in the
simulation. MM fragments are a set of atoms in which bonds and angle
harmonic potentials are defined, or alternatively shake constraints are
defined. The following input defines the fragments for the two water
molecules in the above
geometry,</p>
<pre><code>NWPW   
 QMMM   
   fragment spc   
     size 3                  #size of fragment   
     index_start 6:9:3       #atom index list that defines the start of   
                             # the fragments (start:final:stride)  
     bond_spring 1 2 0.467307856 1.889726878   #bond i j Kspring r0   
     bond_spring 1 3 0.467307856 1.889726878   #bond i j Kspring r0   
     angle_spring 2 1 3 0.07293966 1.910611932 #angle i j k Kspring theta0   
   end   
 END  
END
</code></pre>
<p>The fragments can be defined using shake constraints
as</p>
<pre><code>NWPW   
 QMMM   
   fragment spc   
     size 3                  #size of fragment   
     index_start 6:9:3       #atom index list that defines the start of   
                             # the fragments (start:final:stride)  
     shake units angstroms 1 2 3 cyclic 1.0 1.632993125 1.0   
   end   
 END  
END
</code></pre>
<p>Alternatively, each water could be defined independently as
follows.</p>
<pre><code>NWPW   
 QMMM   
   fragment spc1   
     size 3                  #size of fragment   
     index_start 6           #atom index list that defines the start of   
                             #the fragments   
     bond_spring 1 2 0.467307856 1.889726878 #bond i j Kspring r0   
     bond_spring 1 3 0.467307856 1.889726878 #bond i j Kspring r0   
     angle_spring 2 1 3 0.07293966 1.910611932 #angle i j k Kspring theta0   
   end   
   fragment spc2   
     size 3                  #size of fragment   
     index_start 9           #atom index list that defines the start of   
                             #the fragments   
     shake units angstroms 1 2 3 cyclic 1.0 1.632993125 1.0   
   end   
 END  
END
</code></pre>
<h2 id="plane-wave-density-functional-theory-psp_generator">PSP_GENERATOR<a class="headerlink" href="#plane-wave-density-functional-theory-psp_generator" title="Permanent link">&para;</a></h2>
<p>A one-dimensional pseudopotential code has been integrated into NWChem.
This code allows the user to modify and develop pseudopotentials.
Currently, only the Hamann and Troullier-Martins norm-conserving
pseudopotentials can be generated. In future releases, the
pseudopotential library (section <a href="#plane-wave-density-functional-theory-pseudopotential-and-paw-basis-libraries">Pseudopotential and PAW basis
Libraries</a>) will be
more complete, so that the user will not have explicitly generate
pseudopotentials using this module.</p>
<p>Input to the PSP_GENERATOR task is contained within the PSP_GENERATOR
sub-block.</p>
<pre><code>NWPW   
...   
 PSP_GENERATOR   
   ...   
 END   
...  
END
</code></pre>
<p>To run a PSP_GENERATOR calculation the following directive is used:</p>
<p><code>TASK PSPW PSP_GENERATOR</code></p>
<p>Listed below is the format of a PSP_GENERATOR
sub-block.</p>
<pre><code>NWPW  
...   
 PSP_GENERATOR   
   PSEUDOPOTENTIAL_FILENAME: &lt;string psp_name&gt;   
   ELEMENT: &lt;string element&gt;   
   CHARGE: &lt;real charge&gt;   
   MASS_NUMBER: &lt;real mass_number&gt;   
   ATOMIC_FILLING: &lt;integer ncore nvalence&gt; ( (1||2||...) (s||p||d||f||...) &lt;real filling&gt;  ...)

   [CUTOFF: &lt;integer lmax&gt; ( (s||p||d||f||g) &lt;real rcut&gt; ...) ] 

   PSEUDOPOTENTIAL_TYPE: (TROULLIER-MARTINS || HAMANN default HAMANN)   
   SOLVER_TYPE: (PAULI || SCHRODINGER default PAULI)   
   EXCHANGE_TYPE: (dirac || PBE96 default DIRAC)   
   CORRELATION_TYPE: (VOSKO || PBE96 default VOSKO)   
   [SEMICORE_RADIUS: &lt;real rcore&gt;]

 END  
...   
END
</code></pre>
<p>The following list describes the input for the PSP_GENERATOR sub-block.</p>
<ul>
<li><code>psp_name</code> - name that points to a.</li>
<li><code>element</code> - Atomic symbol.</li>
<li><code>charge</code> - charge of the atom</li>
<li><code>mass</code> - mass number for the atom</li>
<li><code>ncore</code> - number of core states</li>
<li><code>nvalence</code> - number of valence states.</li>
<li>ATOMIC_FILLING:.....(see below)</li>
<li><code>filling</code> - occupation of atomic state</li>
<li>CUTOFF:....(see below)</li>
<li><code>rcore</code> - value for the semicore radius (see below)</li>
</ul>
<h3 id="plane-wave-density-functional-theory-atomic_filling-block">ATOMIC_FILLING Block<a class="headerlink" href="#plane-wave-density-functional-theory-atomic_filling-block" title="Permanent link">&para;</a></h3>
<p>This required block is used to define the reference atom which is used
to define the pseudopotential. After the ATOMIC_FILLING: <ncore>
<nvalence> line, the core states are listed (one per line), and then the
valence states are listed (one per line). Each state contains two
integer and a value. The first integer specifies the radial quantum
number, <span class="arithmatex">\(n\)</span>, the second integer specifies the angular momentum quantum
number, <span class="arithmatex">\(l\)</span>, and the third value specifies the occupation of the
state.</p>
<p>For example to define a pseudopotential for the Neon atom in the
<span class="arithmatex">\(1s^2 2s^2 2p^6\)</span> state could have the block</p>
<pre><code>ATOMIC_FILLING: 1 2   
 1 s 2.0 #core state - 1s^2   
 2 s 2.0 #valence state - 2s^2   
 2 p 6.0 #valence state - 2p^6
</code></pre>
<p>for a pseudopotential with a <span class="arithmatex">\(2s\)</span> and <span class="arithmatex">\(2p\)</span> valence electrons or the
block</p>
<pre><code>ATOMIC_FILLING: 3 0   
 1 s 2.0 #core state   
 2 s 2.0 #core state   
 2 p 6.0 #core state
</code></pre>
<p>could be used for a pseudopotential with no valence electrons.</p>
<h3 id="plane-wave-density-functional-theory-cutoff">CUTOFF<a class="headerlink" href="#plane-wave-density-functional-theory-cutoff" title="Permanent link">&para;</a></h3>
<p>This optional block specifies the cutoff distances used to match the
all-electron atom to the pseudopotential atom. For Hamann
pseudopotentials <span class="arithmatex">\(r_{cut}(l)\)</span> defines the distance where the
all-electron potential is matched to the pseudopotential, and for
Troullier-Martins pseudopotentials <span class="arithmatex">\(r_{cut}(l)\)</span> defines the distance
where the all-electron orbital is matched to the pseudowavefunctions.
Thus the definition of the radii depends on the type of pseudopotential.
The cutoff radii used in Hamann pseudopotentials will be smaller than
the cutoff radii used in Troullier-Martins pseudopotentials.</p>
<p>For example to define a softened Hamann pseudopotential for Carbon would
be</p>
<pre><code>ATOMIC_FILLING: 1 2   
 1 s 2.0   
 2 s 2.0   
 2 p 2.0  
CUTOFF: 2  
 s 0.8   
 p 0.85   
 d 0.85
</code></pre>
<p>while a similarly softened Troullier-Marting pseudopotential for Carbon
would be</p>
<pre><code>ATOMIC_FILLING: 1 2   
 1 s 2.0   
 2 s 2.0   
 2 p 2.0  
CUTOFF: 2   
 s 1.200   
 p 1.275   
 d 1.275
</code></pre>
<h3 id="plane-wave-density-functional-theory-semicore_radius">SEMICORE_RADIUS<a class="headerlink" href="#plane-wave-density-functional-theory-semicore_radius" title="Permanent link">&para;</a></h3>
<p>Specifying the SEMICORE_RADIUS option turns on the semicore correction
approximation proposed by Louie et al (S.G. Louie, S. Froyen, and M.L.
Cohen, Phys. Rev. B, <strong>26</strong>(, 1738, (1982)). This approximation is known
to dramatically improve results for systems containing alkali and
transition metal atoms.</p>
<p>The implementation in the PSPW module defines the semi-core density,
<span class="arithmatex">\(\rho_{semicore}\)</span>, by
using the sixth-order polynomial</p>
<div class="arithmatex">\[\rho_{semicore}(r) = \begin{cases} \rho_{core}  \mbox{if } r \ge r_{semicore} \\ c_0 + c_3 r^3 + c_4 r^4 + c_5 r^5 + c_6 r^6  \mbox{if } r &lt; r_{semicore} \end{cases}\]</div>
<p>This expansion was suggested by Fuchs and Scheffler (M. Fuchs, and M.
Scheffler, Comp. Phys. Comm.,<strong>119</strong>,67 (1999)), and is better behaved
for taking derivatives (i.e. calculating ionic forces) than the
expansion suggested by Louie et al.</p>
<h2 id="plane-wave-density-functional-theory-paw-tasks-legacy-implementation">PAW Tasks: Legacy Implementation<a class="headerlink" href="#plane-wave-density-functional-theory-paw-tasks-legacy-implementation" title="Permanent link">&para;</a></h2>
<p>(<span style="color: red;"><em>This capability is now available in PSPW. It
is recommended that this module only be used for testing purposes.</em>
</span>)</p>
<p>All input to the PAW Tasks is contained within the compound NWPW block,</p>
<pre><code>NWPW   
 ...  
END
</code></pre>
<p>To perform an actual calculation a TASK PAW directive is used
(<a href="#task">Task</a>).</p>
<p><code>TASK PAW</code></p>
<p>In addition to the directives listed in <a href="#task">Task</a>, i.e.</p>
<pre><code>TASK paw energy   
TASK paw gradient   
TASK paw optimize   
TASK paw saddle   
TASK paw freqencies   
TASK paw vib
</code></pre>
<p>there are additional directives that are specific to the PSPW module,
which are:</p>
<p><code>TASK PAW [Car-Parrinello || steepest_descent ]</code></p>
<p>Once a user has specified a geometry, the PAW module can be invoked with
no input directives (defaults invoked throughout). There are
sub-directives which allow for customized application; those currently
provided as options for the PAW module
are:</p>
<pre><code>NWPW   
 CELL_NAME &lt;string cell_name default 'cell_default'&gt;   
 [GEOMETRY_OPTIMIZE]   
 INPUT_WAVEFUNCTION_FILENAME &lt;string input_wavefunctions default input_movecs&gt;   
 OUTPUT_WAVEFUNCTION_FILENAME &lt;string output_wavefunctions default input_movecs&gt;   
 FAKE_MASS &lt;real fake_mass default 400000.0&gt;   
 TIME_STEP &lt;real time_step default 5.8&gt;   
 LOOP &lt;integer inner_iteration outer_iteration default 10 100&gt;   
 TOLERANCES &lt;real tole tolc default 1.0e-7 1.0e-7&gt;   
 CUTOFF &lt;real cutoff&gt;   
 ENERGY_CUTOFF &lt;real ecut default (see input description)&gt;   
 WAVEFUNCTION_CUTOFF &lt;real wcut default (see input description)&gt;   
 EWALD_NCUT &lt;integer ncut default 1&gt;]   
 EWALD_RCUT &lt;real rcut default (see input description)&gt;   
 XC (Vosko || PBE96 || revPBE default Vosko)   
 DFT||ODFT||RESTRICTED||UNRESTRICTED   
 MULT &lt;integer mult default 1&gt;   
 INTEGRATE_MULT_L &lt;integer imult default 1&gt;  
 SIMULATION_CELL   
   ... (see input description)   
 END   
 CAR-PARRINELLO   
   ... (see input description)   
 END  
 MAPPING &lt;integer mapping default 1&gt;  
END 
</code></pre>
<p>The following list describes these keywords.</p>
<ul>
<li><code>cell_name</code> - name of the the simulation_cell named <code>cell_name</code>.
    The current version of PAW only accepts periodic unit cells. See
    <a href="#plane-wave-density-functional-theory-simulation-cell">Simulation Cell</a>.</li>
<li>GEOMETRY_OPTIMIZE - optional keyword which if specified turns on
    geometry optimization.</li>
<li><code>input_wavefunctions</code> - name of the file containing one-electron
    orbitals</li>
<li><code>output_wavefunctions</code> - name of the file that will contain the
    one-electron orbitals at the end of the run.</li>
<li><code>fake_mass</code> - value for the electronic fake mass <span class="arithmatex">\(\mu\)</span>. This
    parameter is not presently used in a conjugate gradient simulation</li>
<li><code>time_step</code> - value for the time step (<span class="arithmatex">\(\Delta t\)</span>). This parameter
    is not presently used in a conjugate gradient simulation.</li>
<li><code>inner_iteration</code> - number of iterations between the printing out of
    energies and tolerances</li>
<li><code>outer_iteration</code> - number of outer iterations</li>
<li><code>tole</code> - value for the energy tolerance.</li>
<li><code>tolc</code> - value for the one-electron orbital tolerance.</li>
<li><code>cutoff</code> - value for the cutoff energy used to define the
    wavefunction. In addition using the CUTOFF keyword automatically
    sets the cutoff energy for the density to be twice the wavefunction
    cutoff.</li>
<li><code>ecut</code> - value for the cutoff energy used to define the density.
    Default is set to be the maximum value that will fit within the
    simulation_cell <code>cell_name</code>.</li>
<li><code>wcut</code> - value for the cutoff energy used to define the one-electron
    orbitals. Default is set to be the maximum value that will fix
    within the simulation_cell <code>cell_name</code>.</li>
<li><code>ncuth</code> - value for the number of unit cells to sum over (in each
    direction) for the real space part of the smooth compensation
    summation.</li>
<li><code>rcut</code> - value for the cutoff radius used in the smooth compensation
    summation.</li>
</ul>
<p>Default set to be <span class="arithmatex">\(\frac{MIN(\left| \vec{a_i} \right|)}{\pi}, i=1,2,3\)</span>.</p>
<ul>
<li>(Vosko || PBE96 || revPBE) - Choose between Vosko et al&rsquo;s LDA
    parametrization or the original and revised Perdew, Burke, and
    Ernzerhof GGA functional.</li>
<li>MULT - optional keyword which if specified allows the user to define
    the spin multiplicity of the system</li>
<li>INTEGRATE_MULT_L - optional keyword which if specified allows the
    user to define the angular XC integration of the augmented region</li>
<li>SIMULATION_CELL (see <a href="#plane-wave-density-functional-theory-simulation-cell">Simulation Cell</a> )</li>
<li>CAR-PARRINELLO (see <a href="#plane-wave-density-functional-theory-car-parrinello-scheme-for-ab-initio-molecular-dynamics">Car-Parrinello</a>)</li>
<li><code>mapping</code> - for a value of 1 slab FFT is used, for a value of 2 a
    2d-Hilbert FFT is used.</li>
</ul>
<h2 id="plane-wave-density-functional-theory-pseudopotential-and-paw-basis-libraries">Pseudopotential and PAW basis Libraries<a class="headerlink" href="#plane-wave-density-functional-theory-pseudopotential-and-paw-basis-libraries" title="Permanent link">&para;</a></h2>
<p>A library of pseudopotentials used by PSPW and BAND is currently
available in the directory <code>$NWCHEM_TOP/src/nwpw/libraryp/pspw_default</code></p>
<p>The elements listed in the following table are
present:</p>
<pre><code> H                                                  He  
-------                              ------------------  
 Li Be                               B  C  N  O  F  Ne  
-------                             -------------------  
 Na Mg                               Al Si P  S  Cl Ar  
-------------------------------------------------------  
 K  Ca Sc Ti V  Cr Mn Fe Co Ni Cu Zn Ga Ge As Se Br Kr           
-------------------------------------------------------  
 Rb Sr Y  Zr Nb Mo Tc Ru Rh Pd Ag Cd In Sn Sb Te I  Xe  
-------------------------------------------------------  
 Cs Ba La Hf Ta W  Re Os Ir Pt Au Hg Tl Pb Bi Po At Rn  
-------------------------------------------------------  
 Fr Ra .   
-----------------  
         ------------------------------------------  
          .  .  .  .  .  .  Gd .  .  .  .  .  .  .                        
         ------------------------------------------  
          .  .  U  .  Pu .  .  .  .  .  .  .  .  .        
         ------------------------------------------
</code></pre>
<p>The pseudopotential libraries are continually being tested and added.
Also, the PSPW program can read in pseudopotentials in CPI and TETER
format generated with pseudopotential generation programs such as the
OPIUM package of Rappe et al. The user can request additional
pseudopotentials from Eric J. Bylaska at (Eric.Bylaska@pnl.gov).</p>
<p>Similarly, a library of PAW basis used by PAW is currently available in
the directory
<code>$NWCHEM_TOP/src/nwpw/libraryp/paw_default</code></p>
<pre><code> H                                                  He  
-------                              -----------------  
 Li Be                               B  C  N  O  F  Ne  
-------                             ------------------  
 Na Mg                               Al Si P  S  Cl Ar  
------------------------------------------------------  
 K  Ca Sc Ti V  Cr Mn Fe Co Ni Cu Zn Ga Ge As Se Br Kr           
------------------------------------------------------  
 .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  
------------------------------------------------------  
 .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  
------------------------------------------------------  
 .  .  .   
-----------------           
         ------------------------------------------  
          .  .  .  .  .  .  .  .  .  .  .  .  .  .                        
         ------------------------------------------  
          .  .  .  .  .  .  .  .  .  .  .  .  .  .        
         ------------------------------------------
</code></pre>
<p>Currently there are not very many elements available for PAW. However,
the user can request additional basis sets from Eric J. Bylaska at
(Eric.Bylaska@pnl.gov).</p>
<p>A preliminary implementation of the HGH pseudopotentials (Hartwigsen,
Goedecker, and Hutter) has been implemented into the PSPW module. To
access the pseudopotentials the pseudopotentials input block is used.
For example, to redirect the code to use HGH pseudopotentials for carbon
and hydrogen, the following input would be used.</p>
<pre><code>nwpw   
...   
 pseudopotentials   
  C library HGH_LDA   
  H library HGH_LDA   
 end   
...  
end
</code></pre>
<p>The implementation of HGH pseudopotentials is rather limited in this
release. HGH pseudopotentials cannot be used to optimize unit cells, and
they do not work with the MULLIKEN option. They also have not yet been
implemented into the BAND structure code.<br />
To read in pseudopotentials in CPI format the following input would be
used.</p>
<pre><code>nwpw   
...   
 pseudopotentials   
  C CPI c.cpi   
  H CPI h.cpi   
 end   
...  
end
</code></pre>
<p>In order for the program to recognize the CPI format the CPI files, e.g.
c.cpi have to be prepended with the &ldquo;<CPI>&rdquo; keyword.</p>
<p>To read in pseudopotentials in TETER format the following input would be
used.</p>
<pre><code>nwpw   
...   
 pseudopotentials   
  C TETER c.teter   
  H TETER h.teter   
 end   
...  
end
</code></pre>
<p>In order for the program to recognize the TETER format the TETER files,
e.g. c.teter have to be prepended with the &ldquo;<TETER>&rdquo; keyword.</p>
<p>If you wish to redirect the code to a different directory other than the
default one, you need to set the environmental variable
NWCHEM_NWPW_LIBRARY to the new location of the libraryps directory.</p>
<h2 id="plane-wave-density-functional-theory-nwpw-rtdb-entries-and-miscellaneous-datafiles">NWPW RTDB Entries and Miscellaneous DataFiles<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-rtdb-entries-and-miscellaneous-datafiles" title="Permanent link">&para;</a></h2>
<p>Input to the PSPW and Band modules are contained in both the RTDB and
datafiles. The RTDB is used to store input that the user will need to
directly specify. Input of this kind includes ion positions, ion
velocities, and simulation cell parameters. The datafiles are used to
store input, such the one-electron orbitals, one-electron orbital
velocities, formatted pseudopotentials, and one-dimensional
pseudopotentials, that the user will in most cases run a program to
generate.</p>
<h3 id="plane-wave-density-functional-theory-ion-positions">Ion Positions<a class="headerlink" href="#plane-wave-density-functional-theory-ion-positions" title="Permanent link">&para;</a></h3>
<p>The positions of the ions are stored in the default geometry structure
in the RTDB and must be specified using the GEOMETRY directive.</p>
<h3 id="plane-wave-density-functional-theory-ion-velocities">Ion Velocities<a class="headerlink" href="#plane-wave-density-functional-theory-ion-velocities" title="Permanent link">&para;</a></h3>
<p>The velocities of the ions are stored in the default geometry structure
in the RTDB, and must be specified using the GEOMETRY directive.</p>
<h3 id="plane-wave-density-functional-theory-wavefunction-datafile">Wavefunction Datafile<a class="headerlink" href="#plane-wave-density-functional-theory-wavefunction-datafile" title="Permanent link">&para;</a></h3>
<p>The one-electron orbitals are stored in a wavefunction datafile. This is
a binary file and cannot be directly edited. This datafile is used by
steepest_descent and Car-Parrinello tasks and can be generated using
the wavefunction_initializer or wavefunction_expander tasks.</p>
<h3 id="plane-wave-density-functional-theory-velocity-wavefunction-datafile">Velocity Wavefunction Datafile<a class="headerlink" href="#plane-wave-density-functional-theory-velocity-wavefunction-datafile" title="Permanent link">&para;</a></h3>
<p>The one-electron orbital velocities are stored in a velocity
wavefunction datafile. This is a binary file and cannot be directly
edited. This datafile is automatically generated the first time a
Car-Parrinello task is run.</p>
<h3 id="plane-wave-density-functional-theory-formatted-pseudopotential-datafile">Formatted Pseudopotential Datafile<a class="headerlink" href="#plane-wave-density-functional-theory-formatted-pseudopotential-datafile" title="Permanent link">&para;</a></h3>
<p>The pseudopotentials in Kleinman-Bylander form expanded on a simulation
cell (3d grid) are stored in a formatted pseudopotential datafile
(PSPW-&ldquo;atomname.vpp&rdquo;, BAND-&ldquo;atomname.cpp&rdquo;, PAW-&ldquo;atomname.jpp&rdquo;). These
are binary files and cannot be directly edited. These datafiles are
automatically generated.</p>
<h3 id="plane-wave-density-functional-theory-one-dimensional-pseudopotential-datafile">One-Dimensional Pseudopotential Datafile<a class="headerlink" href="#plane-wave-density-functional-theory-one-dimensional-pseudopotential-datafile" title="Permanent link">&para;</a></h3>
<p>The one-dimensional pseudopotentials are stored in a one-dimensional
pseudopotential file (&ldquo;atomname.psp&rdquo;). This is an ASCII file and can be
directly edited with a text editor or can be generated using the
pspw_generator task. However, these datafiles are usually atomatically
generated.</p>
<p>The data stored in the one-dimensional pseudopotential file is</p>
<pre><code>character*2 element :: element name   
integer charge :: valence charge of ion   
real mass :: mass of ion   
integer lmax :: maximum angular component   
real rcut(lmax) :: cutoff radii used to define pseudopotentials   
integer nr :: number of points in the radial grid   
real dr :: linear spacing of the radial grid    
real r(nr):: one-dimensional radial grid   
real Vpsp(nr,lmax) :: one-dimensional pseudopotentials   
real psi(nr,lmax) :: one-dimensional pseudowavefunctions   
real r_semicore :: semicore radius   
real rho_semicore(nr) :: semicore density  
</code></pre>
<p>and the format of it is:</p>
<pre><code>[line 1: ] element [line 2: ] charge mass lmax  
[line 3: ] (rcut(l), l=1,lmax)  
[line 4: ] nr dr  
[line 5: ] r(1) (Vpsp(1,l), l=1,lmax)  
[line 6: ] ....  
[line nr+4: ] r(nr) (Vpsp(nr,l), l=1,lmax)  
[line nr+5: ] r(1) (psi(1,l), l=1,lmax) [line nr+6: ] ....  
[line 2*nr+4:] r(nr) (psi(nr,l), l=1,lmax)  
[line 2*nr+5:] r_semicore  
if (r_semicore read) then  
 [line 2*nr+6:] r(1) rho_semicore(1)  
 [line 2*nr+7:] ....  
 [line 3*nr+5:] r(nr) rho_semicore(nr)  
end if
</code></pre>
<h2 id="plane-wave-density-functional-theory-car-parrinello-scheme-for-ab-initio-molecular-dynamics">Car-Parrinello Scheme for Ab Initio Molecular Dynamics<a class="headerlink" href="#plane-wave-density-functional-theory-car-parrinello-scheme-for-ab-initio-molecular-dynamics" title="Permanent link">&para;</a></h2>
<p>Car and Parrinello developed a unified scheme for doing <em>ab initio</em>
molecular dynamics by combining the motion of the ion cores and a
fictitious motion for the Kohn-Sham orbitals of density-functional
theory (R. Car and M. Parrinello, Phys. Rev. Lett. <strong>55</strong>, 2471, (1985)
- simple introduction
<a href="cpmd-lecture.pdf">cpmd-lecture.pdf</a>). At the
heart of this method they introduced a fictitious kinetic energy
functional for the Kohn-Sham
orbitals.</p>
<div class="arithmatex">\[\begin{align}KE \left ( \bigl[ \psi_{i,\sigma}(\vec{r}) \bigr] \right ) = \sum_{i,\sigma}^{occ} \int d\vec{r}\ \mu \left | \dot{\psi}_{i,\sigma}(\vec{r}) \right | ^2\end{align}\]</div>
<p>Given this kinetic energy the constrained equations of motion are found
by taking the first variation of the auxiliary Lagrangian.</p>
<div class="arithmatex">\[ L = \sum_{i,\sigma}^{occ} \int d\vec{r}\ \mu \left | \dot{\psi}_{i,\sigma}(\vec{r}) \right | ^2;     + \frac{1}{2} \sum_{I} M_I \left | \dot{\vec{R}}_{I} \right | ^2 - E \left [  \bigl[ \psi_{i,\sigma}(\vec{r})  \bigr] , \bigl[ \vec{R}_I  \bigr] \right ] \\ + \sum_{ij,\sigma} \Lambda_{ij,\sigma} \left ( \int d\vec{r}\ \psi_{i,\sigma}^{*}(\vec{r}) \psi_{j,\sigma}(\vec{r}) - \delta_{ij\sigma} \right )\]</div>
<p>Which generates a dynamics for the wavefunctions
<span class="arithmatex">\(\psi_{i,\sigma}(\vec{r})\)</span> and atoms positions <span class="arithmatex">\(\vec{R}_I\)</span> through
the constrained equations of motion:</p>
<div class="arithmatex">\[\begin{matrix}\mu \ddot{\psi}_{i,\sigma}(\vec{r},t) = -\frac{\delta E}{\delta \psi_{i,\sigma }^{*} \left( \vec{r},t \right) } + \sum\limits_j \Lambda_{ij,\sigma} \psi_{j,\sigma} \left( \vec{r},t \right) \end{matrix}\]</div>
<div class="arithmatex">\[\begin{matrix}M_I \ddot{\vec{R}}_I = -\frac{\partial E}{\partial \vec{R}_I} \end{matrix}\]</div>
<p>where <span class="arithmatex">\(\mu\)</span> is the fictitious mass for the electronic degrees of
freedom and <span class="arithmatex">\(M_I\)</span> are the ionic masses. The adjustable parameter
<span class="arithmatex">\(\mu\)</span> is used to describe the relative rate at which the wavefunctions
change with time. <span class="arithmatex">\(\Lambda_{ij,\sigma}\)</span> are the Lagrangian multipliers
for the orthonormalization of the single-particle orbitals
<span class="arithmatex">\(\psi_{i,\sigma}(\vec{r})\)</span>. They are defined by the orthonormalization
constraint conditions and can be rigorously found. However, the
equations of motion for the Lagrange multipliers depend on the specific
algorithm used to integrate the Eqns. above.</p>
<p>For this method to give ionic motions that are physically meaningful the
kinetic energy of the Kohn-Sham orbitals must be relatively small when
compared to the kinetic energy of the ions. There are two ways where
this criterion can fail. First, the numerical integrations for the
Car-Parrinello equations of motion can often lead to large relative
values of the kinetic energy of the Kohn-Sham orbitals relative to the
kinetic energy of the ions. This kind of failure is easily fixed by
requiring a more accurate numerical integration, i.e. use a smaller time
step for the numerical integration. Second, during the motion of the
system a the ions can be in locations where there is an Kohn-Sham
orbital level crossing, i.e. the density-functional energy can have two
states that are nearly degenerate. This kind of failure often occurs in
the study of chemical reactions. This kind of failure is not easily
fixed and requires the use of a more sophisticated density-functional
energy that accounts for low-lying excited electronic states.</p>
<h3 id="plane-wave-density-functional-theory-verlet-algorithm-for-integration">Verlet Algorithm for Integration<a class="headerlink" href="#plane-wave-density-functional-theory-verlet-algorithm-for-integration" title="Permanent link">&para;</a></h3>
<p>Integrating the Eqns. above using the Verlet algorithm results
in</p>
<div class="arithmatex">\[\begin{matrix}\psi_{i,\sigma}^{t+ \Delta t} \leftarrow 2 \psi_{i,\sigma}^{t} - \psi_{i,\sigma}^{t-\Delta t} + \frac{(\Delta t)^2}{\mu} \left[ \frac{\delta E}{\delta \psi_{i,\sigma}^{*}} + \sum_{j} \psi_{j,\sigma} \Lambda_{ji,\sigma} \right]_{t} \end{matrix}\]</div>
<div class="arithmatex">\[\begin{matrix}\vec{R}_I^{t+\Delta t} \leftarrow 2 \vec{R}_I^{t} - \vec{R}_I^{t-\Delta t} + \frac{(\Delta t)^2}{M_I} \frac{\partial E}{\partial \vec{R}_I}\end{matrix}\]</div>
<p>In this molecular dynamic procedure we have to know variational
derivative  </p>
<div class="arithmatex">\[\frac{\delta E}{\delta \psi_{i,\sigma}^{*}}\]</div>
<p>and the matrix <span class="arithmatex">\(\Lambda_{ij,\sigma}\)</span>.<br />
The variational derivative</p>
<div class="arithmatex">\[\frac{\delta E}{\delta \psi_{i,\sigma}^{*}}\]</div>
<p>can be analytically found and is  </p>
<div class="arithmatex">\[\begin{align}\frac{\delta E}{\delta \psi_{i,\sigma}^{*}}  = -\frac{1}{2} \nabla^2 \psi_{i,\sigma}(\vec{r}) \\  + \int d\vec{r^{\prime}} W_{ext}(\vec{r},\vec{r^{\prime}}) \psi_{i,\sigma}(\vec{r^{\prime}}) \\  + \int d\vec{r^{\prime}} \frac{n(\vec{r^{\prime}})}{|\vec{r}-\vec{r^{\prime}}|} \psi_{i,\sigma}(\vec{r})\\  + \mu_{xc}^{\sigma}(\vec{r}) \psi_{i,\sigma}(\vec{r}) \\  \equiv \hat{H} \psi_{i,\sigma} \end{align}\]</div>
<p>To find the matrix <span class="arithmatex">\(\Lambda_{ij,\sigma}\)</span> impose the orthonormality
constraint on <span class="arithmatex">\(\psi_{i,\sigma}^{t+\Delta t}\)</span> obtain a matrix
Riccatti equation, and then Riccatti equation is solved by an iterative
solution.</p>
<h3 id="plane-wave-density-functional-theory-constant-temperature-simulations-nose-hoover-thermostats">Constant Temperature Simulations: Nose-Hoover Thermostats<a class="headerlink" href="#plane-wave-density-functional-theory-constant-temperature-simulations-nose-hoover-thermostats" title="Permanent link">&para;</a></h3>
<p>Nose-Hoover Thermostats for the electrons and ions can also be added to
the Car-Parrinello simulation. In this type of simulation thermostats
variables <span class="arithmatex">\(x_e\)</span> and <span class="arithmatex">\(x_R\)</span> are added to the simulation by adding the
auxiliary energy functionals to the total
energy.</p>
<div class="arithmatex">\[ ION_{THERMOSTAT}(x_R)  = \frac{1}{2} Q_R \dot{x_R} + E_{R0}x_R\]</div>
<div class="arithmatex">\[ELECTRON_{THERMOSTAT}(x_e)  = \frac{1}{2} Q_e \dot{x_e} + E_{e0}x_e \]</div>
<p>In these equations, the average kinetic energy for the ions is</p>
<div class="arithmatex">\[\begin{matrix}E_{R0} = \frac{1}{2} f k_B T \end{matrix}\]</div>
<p>where <span class="arithmatex">\(f\)</span> is the number of atomic degrees of freedom, <span class="arithmatex">\(k_B\)</span>  is
Boltzmann&rsquo;s constant, and T is the desired temperature. Defining the
average fictitious kinetic energy of the electrons is not as
straightforward. Blchl and Parrinello (P.E. Blchl and M. Parrinello,
Phys. Rev. B, <strong>45</strong>, 9413, (1992)) have suggested the following formula
for determining the average fictitious kinetic
energy</p>
<div class="arithmatex">\[\begin{matrix}E_{e0} = 4 k_B T \frac{\mu}{M} \sum_{i} &lt;\psi_i|-\frac{1}{2} \nabla^2 |\psi_{i}&gt; \end{matrix}\]</div>
<p>where <span class="arithmatex">\(\mu\)</span> is the fictitious electronic mass, <span class="arithmatex">\(M\)</span> is average mass
of one atom, and
<span class="arithmatex">\(\sum_{i} &lt;\psi_i|-\frac{1}{2} \nabla^2 |\psi_{i}&gt;\)</span>
is the kinetic energy of the electrons.</p>
<p>Blchl and Parrinello suggested that the choice of mass parameters,
<span class="arithmatex">\(Q_e\)</span>, and <span class="arithmatex">\(Q_R\)</span> should be made such that the period of oscillating
thermostats should be chosen larger than the typical time scale for the
dynamical events of interest but shorter than the simulation time.</p>
<div class="arithmatex">\[P_{ion}  = 2\pi \sqrt{\frac{Q_R}{4E_{R0}}}\]</div>
<div class="arithmatex">\[P_{electron}  = 2\pi \sqrt{\frac{Q_e}{4E_{e0}}}\]</div>
<p>where <span class="arithmatex">\(P_{ion}\)</span> and <span class="arithmatex">\(P_{electron}\)</span> are the periods of oscillation
for the ionic and fictitious electronic thermostats.</p>
<p>In simulated annealing simulations the electronic and ionic Temperatures
are scaled according to an exponential cooling schedule,</p>
<div class="arithmatex">\[T_e(t)  = T_e^0 \exp^{-\frac{t}{\tau_e}}\]</div>
<div class="arithmatex">\[T_{ionic}(t)  = T_{ionic}^0 \exp^{-\frac{t}{\tau_{ionic}}}\]</div>
<p>where <span class="arithmatex">\(T_e^0\)</span> and <span class="arithmatex">\(T_{ionic}^0\)</span> are the initial temperatures, and
<span class="arithmatex">\(\tau_e\)</span> and <span class="arithmatex">\(\tau_{ionic}\)</span> are the cooling rates in atomic units.</p>
<h2 id="plane-wave-density-functional-theory-nwpw-tutorial-1-s2-dimer-examples-with-pspw">NWPW Tutorial 1: S<sub>2</sub> dimer examples with PSPW<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-tutorial-1-s2-dimer-examples-with-pspw" title="Permanent link">&para;</a></h2>
<p>A description of all the examples in NWPW Tutorial 1 can be found in the
attached pdf <a href="nwpwexample1.pdf">nwpwexample1.pdf</a></p>
<h3 id="plane-wave-density-functional-theory-total-energy-of-s2-dimer-with-lda-approximation">Total energy of S<sub>2</sub> dimer with LDA approximation<a class="headerlink" href="#plane-wave-density-functional-theory-total-energy-of-s2-dimer-with-lda-approximation" title="Permanent link">&para;</a></h3>
<p>(input:<a href="s2-example1.nw">Media:s2-example1.nw</a>,
output:<a href="s2-example1.nwout">Media:s2-example1.nwout</a>)</p>
<p>In this example, the total energy of the S<sub>2</sub> dimer using LDA
approximation for the exchange-correlation functional is calculated.</p>
<pre><code> echo  
 title &quot;total energy of s2-dimer LDA/30Ry with PSPW method&quot;  
 scratch_dir   ./scratch  
 permanent_dir ./perm  
 start s2-pspw-energy  
 geometry  
 S 0.0 0.0 0.0  
 S 0.0 0.0 1.88  
 end  
 nwpw  
   simulation_cell  
     SC 20.0  
   end  
   cutoff 15.0  
   mult 3  
   xc lda  
   lmbfgs  
 end  
 task pspw energy
</code></pre>
<p>The energies from the simulation will
be</p>
<pre><code> ...  
 ==  Summary Of Results  ==  

  number of electrons: spin up=    7.00000  down=    5.00000 (real space)  

  total     energy    :  -0.2041363137E+02 (   -0.10207E+02/ion)  
  total orbital energy:  -0.4944372503E+01 (   -0.41203E+00/electron)  
  hartree   energy    :   0.1680529987E+02 (    0.14004E+01/electron)  
  exc-corr  energy    :  -0.4320620600E+01 (   -0.36005E+00/electron)  
  ion-ion   energy    :   0.8455644190E-02 (    0.42278E-02/ion)  

  kinetic (planewave) :   0.7529965882E+01 (    0.62750E+00/electron)  
  V_local (planewave) :  -0.4506036741E+02 (   -0.37550E+01/electron)  
  V_nl    (planewave) :   0.4623635248E+01 (    0.38530E+00/electron)  
  V_Coul  (planewave) :   0.3361059973E+02 (    0.28009E+01/electron)  
  V_xc.   (planewave) :  -0.5648205953E+01 (   -0.47068E+00/electron)  
  Virial Coefficient  :  -0.1656626150E+01  

  orbital energies:  
     -0.2001309E+00 (  -5.446eV)  
     -0.2001309E+00 (  -5.446eV)  
     -0.3294434E+00 (  -8.965eV)    -0.2991148E+00 (  -8.139eV)  
     -0.3294435E+00 (  -8.965eV)    -0.2991151E+00 (  -8.139eV)  
     -0.3582269E+00 (  -9.748eV)    -0.3352434E+00 (  -9.123eV)  
     -0.5632339E+00 ( -15.326eV)    -0.5246249E+00 ( -14.276eV)  
     -0.7642738E+00 ( -20.797eV)    -0.7413909E+00 ( -20.174eV)  

  Total PSPW energy   :  -0.2041363137E+02  
 ...
</code></pre>
<h3 id="plane-wave-density-functional-theory-structural-optimization-of-s2-dimer-with-lda-approximation">Structural optimization of S<sub>2</sub> dimer with LDA approximation<a class="headerlink" href="#plane-wave-density-functional-theory-structural-optimization-of-s2-dimer-with-lda-approximation" title="Permanent link">&para;</a></h3>
<p>(input:<a href="s2-example2.nw">Media:s2-example2.nw</a>,
output:<a href="s2-example2.nwout">Media:s2-example2.nwout</a>)</p>
<p>In this example, the structure of the S<sub>2</sub> dimer using results
generated from prior energy calculation is calculated. Since most of the
parameters are already stored in the run-time database the input is very
simple.</p>
<pre><code> echo  
 title &quot;optimization of s2-dimer LDA/30Ry with PSPW method&quot;  
 scratch_dir   ./scratch  
 permanent_dir ./perm  
 restart s2-pspw-energy  
 driver  
    maxiter 20  
    xyz s2  
 end  
 task pspw optimize
</code></pre>
<p>As the optimization process consists of series of total energy
evaluations the contents of the output file are very much similar to
that in Example I. At each step the total energy and force information
will be outputed as
follows</p>
<pre><code>   Step       Energy      Delta E   Gmax     Grms     Xrms     Xmax   Walltime  
   ---- ---------------- -------- -------- -------- -------- -------- --------  
 @    1     -20.41364254 -7.1D-05  0.00004  0.00004  0.00605  0.01048      7.8
</code></pre>
<p>The best way to keep track of the optimization calculation is to run the
following grep command on the output
file.</p>
<pre><code> grep @ outputfile  

 @ Step       Energy      Delta E   Gmax     Grms     Xrms     Xmax   Walltime  
 @ ---- ---------------- -------- -------- -------- -------- -------- --------  
 @    0     -20.41357202  0.0D+00  0.00672  0.00672  0.00000  0.00000      1.5  
 @    1     -20.41364254 -7.1D-05  0.00004  0.00004  0.00605  0.01048      7.8  
 @    2     -20.41364256 -2.3D-08  0.00020  0.00020  0.00003  0.00005      9.7  
 @    2     -20.41364256 -2.3D-08  0.00020  0.00020  0.00003  0.00005      9.7
</code></pre>
<p>The optimized energy and geometry will
be</p>
<pre><code> ...  
       ----------------------  
       Optimization converged  
       ----------------------  


   Step       Energy      Delta E   Gmax     Grms     Xrms     Xmax   Walltime  
   ---- ---------------- -------- -------- -------- -------- -------- --------  
 @    2     -20.41364256 -2.3D-08  0.00020  0.00020  0.00003  0.00005      9.7  
                                      ok       ok       ok       ok          



                                 Z-matrix (autoz)  
                                 --------   

  Units are Angstrom for bonds and degrees for angles  

       Type          Name      I     J     K     L     M      Value     Gradient  
       ----------- --------  ----- ----- ----- ----- ----- ---------- ----------  
     1 Stretch                  1     2                       1.89115    0.00020  



                          Geometry &quot;geometry&quot; -&gt; &quot;geometry&quot;  
                          ---------------------------------  

  Output coordinates in angstroms (scale by  1.889725989 to convert to a.u.)  

   No.       Tag          Charge          X              Y              Z  
  ---- ---------------- ---------- -------------- -------------- --------------  
     1 S                   16.0000     0.00000000     0.00000000    -0.94557591  
     2 S                   16.0000     0.00000000     0.00000000     0.94557591  

 ...
</code></pre>
<h3 id="plane-wave-density-functional-theory-frequency-calculation-of-s2-dimer-with-lda-approximation">Frequency calculation of S<sub>2</sub> dimer with LDA approximation<a class="headerlink" href="#plane-wave-density-functional-theory-frequency-calculation-of-s2-dimer-with-lda-approximation" title="Permanent link">&para;</a></h3>
<p>(input:<a href="s2-example3.nw">Media:s2-example3.nw</a>,
output:<a href="s2-example3.nwout">Media:s2-example3.nwout</a>)</p>
<p>In this example, the vibrational frequency of the S<sub>2</sub> dimer using
results generated from prior geometry optimization is calculated. Since
most of the parameters are already stored in the run-time database the
input is very simple.</p>
<pre><code> echo  
 title &quot;frequency calculation of s2-dimer LDA/30Ry with PSPW method&quot;  
 scratch_dir   ./scratch  
 permanent_dir ./perm  
 restart s2-pspw-energy  
 freq  
    animate  
 end  
 task pspw freq
</code></pre>
<p>The frequency and thermodynamic analysis
generated</p>
<pre><code> ...  
  Temperature                      =   298.15K  
  frequency scaling parameter      =   1.0000  


  Linear Molecule  

  Zero-Point correction to Energy  =    1.034 kcal/mol  (  0.001647 au)  
  Thermal correction to Energy     =    2.579 kcal/mol  (  0.004110 au)  
  Thermal correction to Enthalpy   =    3.171 kcal/mol  (  0.005054 au)  

  Total Entropy                    =   52.277 cal/mol-K  
    - Translational                =   38.368 cal/mol-K (mol. weight =  63.9441)  
    - Rotational                   =   13.630 cal/mol-K (symmetry #  =        2)  
    - Vibrational                  =    0.279 cal/mol-K  

  Cv (constant volume heat capacity) =    5.750 cal/mol-K  
    - Translational                  =    2.979 cal/mol-K  
    - Rotational                     =    1.986 cal/mol-K  
    - Vibrational                    =    0.785 cal/mol-K  
 ...  
  ----------------------------------------------------------------------------  
  Normal Eigenvalue ||           Projected Infra Red Intensities  
   Mode   [cm**-1]  || [atomic units] [(debye/angs)**2] [(KM/mol)] [arbitrary]  
  ------ ---------- || -------------- ----------------- ---------- -----------  
     1        0.000 ||    0.000030           0.001         0.029       0.000  
     2        0.000 ||    2.466908          56.914      2404.864      15.000  
     3        0.000 ||    2.466908          56.914      2404.864      15.000  
     4        0.000 ||    2.466908          56.914      2404.864      15.000  
     5        0.000 ||    2.466908          56.914      2404.864      15.000  
     6      723.419 ||    0.000000           0.000         0.000       0.000  
  ----------------------------------------------------------------------------  
 ...
</code></pre>
<h3 id="plane-wave-density-functional-theory-ab-initio-molecular-dynamics-simulation-car-parrinello-of-s2-dimer-using-the-lda-approximation">Ab initio molecular dynamics simulation (Car-Parrinello) of S<sub>2</sub> dimer using the LDA approximation<a class="headerlink" href="#plane-wave-density-functional-theory-ab-initio-molecular-dynamics-simulation-car-parrinello-of-s2-dimer-using-the-lda-approximation" title="Permanent link">&para;</a></h3>
<p>(input:<a href="s2-example4.nw">Media:s2-example4.nw</a>,
output:<a href="s2-example4.nwout">Media:s2-example4.nwout</a>
<a href="s2-md.xyz">Media:s2-md.xyz</a>
<a href="s2-md.emotion.dat">Media:s2-md.emotion.dat</a> )</p>
<p>In this example, a constant energy Car-Parrinello simulation of S<sub>2</sub>
dimer using LDA approximation is calculated. A brief introduction to the
Car-Parrinello method can be found in
<a href="cpmd-lecture.pdf">cpmd-lecture.pdf</a></p>
<pre><code> echo  
 title &quot;AIMD simulation of s2-dimer&quot;  
 scratch_dir   ./scratch  
 permanent_dir ./perm  
 start s2-md  
 geometry  
 S 0.0 0.0 0.0  
 S 0.0 0.0 1.95  
 end  
 nwpw  
   simulation_cell  
     SC 20.0  
   end  
   cutoff 15.0  
   mult 3  
   xc lda  
   lmbfgs  
   car-parrinello  
     time_step 5.0  
     fake_mass 600.0  
     loop 1 1000  
     xyz_filename s2-md.xyz  
   end  
 end  
 task pspw energy  
 task pspw car-parrinello
</code></pre>
<p>A plotting program (e.g. gnuplot, xmgrace) can be used to look at the
total, potential, kinetic energies, contained in the s2-md.emotion file
(see section <a href="#plane-wave-density-functional-theory-emotion-motion-file">EMOTION motion file</a> for
datafile format)
i.e.,</p>
<pre><code> seattle-1604% gnuplot  

       G N U P L O T  
       Version 4.0 patchlevel 0  
       last modified Thu Apr 15 14:44:22 CEST 2004  
       System: Linux 2.6.18-194.8.1.el5  

       Copyright (C) 1986 - 1993, 1998, 2004  
       Thomas Williams, Colin Kelley and many others  

       This is gnuplot version 4.0.  Please refer to the documentation  
       for command syntax changes.  The old syntax will be accepted  
       throughout the 4.0 series, but all save files use the new syntax.  

       Type help to access the on-line reference manual.  
       The gnuplot FAQ is available from  
               &lt;http://www.gnuplot.info/faq/&gt;  

       Send comments and requests for help to  
               &lt;gnuplot-info@lists.sourceforge.net&gt;  
       Send bugs, suggestions and mods to  
               &lt;gnuplot-bugs@lists.sourceforge.net&gt;  


 Terminal type set to 'x11'  
 gnuplot&gt; plot &quot;s2-md.emotion&quot;,&quot;s2-md.emotion&quot; using 1:3  
 gnuplot&gt; 
</code></pre>
<p>The following plot shows the Car-Parrinello <span class="arithmatex">\(^3\Sigma_g^-\)</span> S<span class="arithmatex">\(_2\)</span>
energy surface generated from the
simulation.</p>
<p><img alt="s2-md" src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/s2-md.png" align=middle with="600pt" height="450pt"/>  </p>
<h3 id="plane-wave-density-functional-theory-ab-initio-molecular-dynamics-simulation-born-oppenheimer-of-ssub2-dimer-using-the-lda-approximation">Ab initio molecular dynamics simulation (Born-Oppenheimer) of S&lt;sub2</sub> dimer using the LDA approximation<a class="headerlink" href="#plane-wave-density-functional-theory-ab-initio-molecular-dynamics-simulation-born-oppenheimer-of-ssub2-dimer-using-the-lda-approximation" title="Permanent link">&para;</a></h3>
<p>(input:<a href="s2-example5.nw">Media:s2-example5.nw</a>,
output:<a href="s2-example5.nwout">Media:s2-example5.nwout</a>
<a href="s2-bomd.xyz">Media:s2-bomd.xyz</a>
<a href="s2-bomd.emotion.dat">Media:s2-bomd.emotion.dat</a> ) In
this example, a constant energy Born-Oppenheimer simulation of S2 dimer
using LDA approximation is calculated.</p>
<pre><code>title &quot;AIMD simulation of s2-dimer&quot; 
echo

scratch_dir   ./scratch
permanent_dir ./perm

start s2-bomd  

geometry
S 0.0 0.0 0.0
S 0.0 0.0 1.95
end

nwpw 
  simulation_cell
    SC 20.0 
  end
  cutoff 15.0  
  mult 3 
  xc lda 
  lmbfgs 
end
task pspw energy 

nwpw 
   bo_steps 1 500 
   bo_time_step 10.0  
end 
task pspw born-oppenheimer
</code></pre>
<p>The following plot shows the <span class="arithmatex">\(^3\Sigma_g^{-}_{2}\)</span> energy surface
generated from the
simulation.</p>
<p><img alt="s2-bomd" src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/s2-bomd.png" align=middle with="600pt" height="450pt"/>  </p>
<h2 id="plane-wave-density-functional-theory-nwpw-tutorial-2-using-pspw-car-parrinello-simulated-annealing-simulations-to-optimize-structures">NWPW Tutorial 2: Using PSPW Car-Parrinello Simulated Annealing Simulations to Optimize Structures<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-tutorial-2-using-pspw-car-parrinello-simulated-annealing-simulations-to-optimize-structures" title="Permanent link">&para;</a></h2>
<p>In principle quantum mechanical calculations can be used to determine
the structure of any chemical system. One chooses a structure,
calculates the total energy of the system, and repeats the calculation
for all possible geometries. Of course the major limitation of this
approach is that the number of local minima structures increases
dramatically with system size and the cost of quantum mechanical
calculations also increases dramatically with system size. Not
surprisingly most quantum mechanical calculations limit the number of
structures to be calculated by using experimental results or chemical
intuition. One could speed up the calculations by using simplified
inter-atomic force fields instead of quantum mechanical calculations.
However, inter-atomic forces fields have many simplifying assumptions
that can severely limit their predictability. Another approach is to use
ab initio molecular dynamics methods combined with simulated annealing.
These methods are quite robust and allow strongly interacting many body
systems to be studied by direct dynamics simulation without the
introduction of empirical interactions. In these methods, the atomic
forces are calculated from ab initio calculations that are performed
on-the-fly as the molecular dynamics trajectory is generated.</p>
<p>The following examples demonstrate how to use the ab initio molecular
dynamics methods and simulated annealing strategies of NWChem to
determine the lowest energy structures of the B<sub>12</sub> cluster. This
example is based on a study performed by Kiran Boggavarapu et al.. One
might expect from chemical intuition that lowest energy structure of
B<sub>12</sub> will be an icosahedron, since B<sub>12</sub> icosahedra are a
common structural unit found in many boron rich materials. Despite this
prevalence, ab initio calculations performed by several researchers have
   suggested that B<sub>12</sub>, as well as B<sub>12</sub><sup>+</sup> and B<sub>12</sub><sup>-</sup>,
will have a more open geometry.</p>
<p><img alt="boron" src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/boron.png" align=middle with="600pt" height="246pt"/>  </p>
<h3 id="plane-wave-density-functional-theory-simulated-annealing-using-constant-energy-simulation">Simulated Annealing Using Constant Energy Simulation<a class="headerlink" href="#plane-wave-density-functional-theory-simulated-annealing-using-constant-energy-simulation" title="Permanent link">&para;</a></h3>
<p>(input:<a href="b12-example2a.nw">Media:b12-example2a.nw</a>,
output:<a href="b12-example2a.nwout">Media:b12-example2a.nwout</a>
<a href="b12.00.xyz">Media:b12.00.xyz</a>
<a href="b12.00.emotion.dat">Media:b12.00.emotion.dat</a>
<a href="b12.01.xyz">Media:b12.01.xyz</a>
<a href="b12.01.emotion.dat">Media:b12.01.emotion.dat</a>)</p>
<p>This example uses a series of constant energy Car-Parrinello simulations
with velocity scaling to do simulated annealing. The initial four
Car-Parrinello simulations are used to heat up the system to several
thousand Kelvin. Then the system is cooled down thru a series of
constant energy simulations in which the electronic and ionic velocities
are scaled by 0.99 at the start of each Car-Parrinello simulation.
Energy minimization calculations are used periodically in this
simulation to bring the system back down to Born-Oppenheimer surface.
This is necessary because of electronic heating.</p>
<p>The Car-Parrinello keyword scaling scales the wavefunction and ionic
velocities at the start of the simulation. The following input is used
to increase the ionic velocities by a factor of two at the start of the
Car-Parrinello simulation.</p>
<p>Key Input</p>
<pre><code>...
Car-Parrinello  
fake_mass 500.0
time_step 5.0 
loop 10 100  
** scaling 1.0 2.0**  
emotion_filename b12.00.emotion
xyz_filename     b12.00.xyz
end 
...
</code></pre>
<p>Output</p>
<pre><code>...        
     wavefnc cutoff= 10.000  fft= 42x 42x 42(     6027 waves     1004 per task)  

technical parameters:  
     translation contrained  
     time step=      5.00     ficticious mass=     500.0  
     **cooling/heatting rates: 0.10000E+01 (psi)
0.20000E+01
(ion)**  
     maximum iterations =    1000 (   10 inner  100 outer )  
     initial kinetic energy:  0.99360E-05 (psi)   0.27956E-03 (ion)  
                                                  0.20205E-28 (c.o.m.)  
     **after scaling: 0.99360E-05 (psi) 0.11182E-02
(ion)**  
     **increased energy: 0.00000E+00 (psi)
0.83868E-03 (ion)**  

Constant Energy Simulation           
...
</code></pre>
<p>The program checks to see if the initial input ionic velocities have a
non-zero center of mass velocity. If there is a non-zero center of mass
velocity in the system then by default the program removes it. To turn
off this feature set the following</p>
<pre><code> nwpw 
    translation on  
 end
</code></pre>
<p>or</p>
<pre><code> set nwpw:com_shift .false.
</code></pre>
<h3 id="plane-wave-density-functional-theory-simulated-annealing-using-constant-temperature-simulation">Simulated Annealing Using Constant Temperature Simulation<a class="headerlink" href="#plane-wave-density-functional-theory-simulated-annealing-using-constant-temperature-simulation" title="Permanent link">&para;</a></h3>
<p>(input:<a href="b12-example2b.nw">Media:b12-example2b.nw</a>,
output:<a href="b12-example2b.nwout">Media:b12-example2b.nwout</a>
<a href="b12.10.xyz">Media:b12.10.xyz</a>
<a href="b12.10.emotion.dat">Media:b12.10.emotion.dat</a>
<a href="b12.11.xyz.gz">Media:b12.11.xyz.gz</a>
<a href="b12.11.emotion.dat">Media:b12.11.emotion.dat</a>)</p>
<p>(mpeg movie of simulation:
<a href="boron.mpg">Media:boron.mpg</a>)</p>
<p>The simulated annealing calculation in this example uses a constant
temperature Car-Parrinello simulation with an exponential cooling
schedule,</p>
<div class="arithmatex">\[T(t)=T_0e^{-t/\tau}\]</div>
<p>where T0 and  are an initial temperature and a time scale of cooling,
respectively. In the present calculations T0=3500K and =4.134e+4 au
(1.0 ps) were used and the thermostat masses were kept fixed to the
initial values determined by T=Te=3500K and (2/)=250 a.u. (6 fs).
Annealing proceeded for 50000 steps, until a temperature of 10K was
reached. After which, the metastable structure is optimized using the
driver optimizer. The keyword SA_decay is used to enter the decay
rates, electron and ion, used in the simulated annealing algorithm in
the constant temperature car-parrinello simulation. The decay rates are
in units of au (conversion 1 au = 2.41889e-17 seconds).</p>
<p>Key
Input</p>
<pre><code> .
 Car-Parrinello  
 SA_decay 4.134d4 4.134d4 #decay rate in units of au (1au=2.41889e-17seconds)  
 .
</code></pre>
<h2 id="plane-wave-density-functional-theory-nwpw-tutorial-3-using-isodesmic-reaction-energies-to-estimate-gas-phase-thermodynamics">NWPW Tutorial 3: using isodesmic reaction energies to estimate gas-phase thermodynamics<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-tutorial-3-using-isodesmic-reaction-energies-to-estimate-gas-phase-thermodynamics" title="Permanent link">&para;</a></h2>
<p>(<a href="isodesmic.pdf">isodesmic.pdf</a>
<a href="isodesmic.tgz">isodesmic.tgz</a>)</p>
<p>The development of a computational scheme that can accurately predict
reaction energies requires some care. As shown in Table 1 energy errors
associated with ab initio calculations can be quite high. Even though ab
initio electronic structure methods are constantly being developed and
improved upon, these methods are rarely able to give heat of formations
of a broad class of molecules with error limits of less than a few
kcal/mol. Only when very large basis sets such as the
correlation-consistent basis sets, high level treatments of correlation
energy such as coupled cluster methods (CCSD(T)), and small correction
factors such as core-valence correlation energies and relativistic
effects, are included will the heat of formation from ab initio
electronic structure methods be accurate to within one kcal/mol.
Although one can now accurately calculate the heats of formation of
molecules with up to 6 first row atoms, such high-level calculations are
extremely demanding and scale computationally as <span class="arithmatex">\(N^7\)</span> for <span class="arithmatex">\(N\)</span> basis
functions.</p>
<p>Examples of these types of large errors are shown in the following
Table, where the enthalpies of formation of CCl<span class="arithmatex">\(_3\)</span>SH are calculated
by using atomization energies from different levels of ab initio
theory.</p>
<table>
<thead>
<tr>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td></td>
<td>MP2/cc-pVDZ</td>
<td>LDA/DZVP2</td>
<td>BP91/DZVP2</td>
<td>B3LYP/DZVP2</td>
<td>G2 Theory</td>
</tr>
<tr>
<td>H <span class="arithmatex">\(_f^o\)</span></td>
<td>+4.9</td>
<td>-80.0</td>
<td>-2.6</td>
<td>+26.5</td>
<td>-13.0</td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<p>Table 1: Standard enthalpy of formation (H<span class="arithmatex">\(_f^o\)</span>(298K) for
CCl<span class="arithmatex">\(_3\)</span>SH in kcal/mol from atomization energies with various
electronic structure methods. Results taken from reference [2].</p>
<p>Differences of up to 106.5 kcal/mol are found between different levels
of theory. This example demonstrates that care must be taken in choosing
the appropriate method for calculating the heats of formation from total
atomization energies.</p>
<p>The difficulties associated with calculating absolute heats of formation
from atomization energies can be avoided by using a set of isodesmic
reactions[1]. The defining property of an isodesmic reaction - that
there are an equal number of like bonds on the left-hand and right-hand
sides of the reaction - helps to minimize the error in the reaction
energy. These reactions are designed to separate out the interactions
between molecular subsistents and non-bonding electrons from the direct
bonding interactions by having the direct bonding interactions largely
canceling one another. This separation is quite attractive. Most ab
initio methods give substantial errors when estimating direct bonding
interactions due to the computational difficulties associated with
electron pair correlation, whereas ab initio methods are expected to be
more accurate for estimating neighboring interactions and long-range
through-bond effects.</p>
<p>The following isodesmic reaction can be used determine the enthalpy of
formation for CCl<span class="arithmatex">\(_3\)</span>SH that is significantly more accurate than the
estimates based on atomization energies.</p>
<p>CCl<span class="arithmatex">\(_3\)</span>SH + CH<span class="arithmatex">\(_4\)</span> <span class="arithmatex">\(\rightarrow\)</span> CH<span class="arithmatex">\(_3\)</span>SH + CCl<span class="arithmatex">\(_3\)</span>H,
H<span class="arithmatex">\(_r\)</span>(calc).</p>
<p>The first step is to calculate the reaction enthalpy of this reaction
from electronic, thermal and vibrational energy differences at 298.15K
at a consistent level of theory. The defining property of an isodesmic
reaction that there are an equal number of like bonds on the left-hand
and right-hand sides of the reaction helps to minimize the error in the
calculation of the reaction energy. The enthalpy of formation of
CCl<span class="arithmatex">\(_3\)</span>SH can then be calculated by using Hesss law with the
calculated enthalpy change and the experimentally known heats of
formation of the other 3 species (see Table 3).</p>
<p>H<span class="arithmatex">\(_f\)</span>(CCl<span class="arithmatex">\(_3\)</span>SH) = H<span class="arithmatex">\(_f\)</span>(CH<span class="arithmatex">\(_3\)</span>SH)(exp) +
H<span class="arithmatex">\(_f\)</span>(CCl<span class="arithmatex">\(_3\)</span>H)(exp) - H<span class="arithmatex">\(_f\)</span>(CH<span class="arithmatex">\(_4\)</span>)(exp)- H<span class="arithmatex">\(_r\)</span>(calc).</p>
<p>In this example, try to design and run NWPW simulations that can be used
to estimate the enthalpy of formation for CCl<span class="arithmatex">\(_3\)</span>SH using its
atomization energy and using the reaction enthalpy of the isodesmic
reaction and compare your results to Table 2. Be careful to make sure
that you use the same cutoff energy for all the simulations (.e.g.
cutoff 35.0). You might also try to estimate enthalpies of formation for
CHCl<span class="arithmatex">\(_2\)</span>SH and CH<span class="arithmatex">\(_2\)</span>ClSH. Also try designing simulations that use
the SCF, DFT, MP2, and TCE modules.</p>
<p>CCl<span class="arithmatex">\(_3\)</span>SH + CH<span class="arithmatex">\(_4\)</span> <span class="arithmatex">\(\rightarrow\)</span> CH<span class="arithmatex">\(_3\)</span>SH + CCl<span class="arithmatex">\(_3\)</span>H</p>
<p>Un-optimized geometries for CCl<span class="arithmatex">\(_3\)</span>SH, CH<span class="arithmatex">\(_3\)</span>SH, CCl<span class="arithmatex">\(_3\)</span>H and
CH<span class="arithmatex">\(_4\)</span> which are needed to design your simulations are contained in
the file
<a href="thermodynamics.xyz">Media:thermodynamics.xyz</a>. You
will also need to calculate the energies for the H, C, S, and Cl atoms
to calculate the atomization energies. The multiplicities for these
atoms are 2, 3, 3 and 2 respectively. You will also need to calculate
the enthalpy of a molecule. The enthalpy of a molecule at 298.15K is sum
of the total energy and a thermal correction to the enthalpy. A good
estimate for the thermal correction to the enthalpy can be obtained from
a frequency calculation, i.e.</p>
<p>H = E + H<span class="arithmatex">\(_{correction}\)</span></p>
<p>Thermodynamic output from a frequency
calculation:</p>
<pre><code>Temperature                      =   298.15K  
frequency scaling parameter      =   1.0000  

Zero-Point correction to Energy  =   27.528 kcal/mol  (  0.043869 au)  
Thermal correction to Energy     =   29.329 kcal/mol  (  0.046739 au)
</code></pre>
<p>The following line contains the value for
H<span class="arithmatex">\(_{correction}\)</span></p>
<pre><code>Thermal correction to Enthalpy   =   29.922 kcal/mol  (  0.047683 au)

Total Entropy                    =   44.401 cal/mol-K  
  - Translational                =   34.246 cal/mol-K (mol. weight =  16.0313)  
  - Rotational                   =   10.060 cal/mol-K (symmetry #  =       12)  
  - Vibrational                  =    0.095 cal/mol-K  

Cv (constant volume heat capacity) =    6.503 cal/mol-K  
  - Translational                  =    2.979 cal/mol-K  
  - Rotational                     =    2.979 cal/mol-K  
  - Vibrational                    =    0.544 cal/mol-K
</code></pre>
<table>
<thead>
<tr>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td>Compounds</td>
<td>MP2/cc-pVDZ</td>
<td>LDA/DZVP2</td>
<td>BP91/DZVP2</td>
<td>B3LYP/DZVP2</td>
<td>G2</td>
<td>Experiment</td>
</tr>
<tr>
<td></td>
<td>(isodesmic)</td>
<td>(isodesmic)</td>
<td>(isodesmic)</td>
<td>(isodesmic)</td>
<td>(atomization)</td>
<td></td>
</tr>
<tr>
<td>CCl$_3SH</td>
<td>-13.40</td>
<td>-11.86</td>
<td>-8.68</td>
<td>-7.64</td>
<td>-12.95</td>
<td></td>
</tr>
<tr>
<td>CHCl<span class="arithmatex">\(_2\)</span>SH</td>
<td>-11.48</td>
<td>-11.07</td>
<td>-8.66</td>
<td>-7.92</td>
<td>-11.52</td>
<td></td>
</tr>
<tr>
<td>CH<span class="arithmatex">\(_2\)</span>ClSH</td>
<td>-7.01</td>
<td>-6.66</td>
<td>-5.44</td>
<td>-5.20</td>
<td>-6.98</td>
<td></td>
</tr>
<tr>
<td>CH<span class="arithmatex">\(_3\)</span>SH</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>-4.76</td>
<td>-5.34</td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<table id="_table-1">
<caption style="caption-side:bottom"><span>Table&nbsp;1:</span> Gas-phase standard enthalpies of formation ( H</caption>
<thead>
<tr>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td>Compounds</td>
<td>H<span class="arithmatex">\(_f^o\)</span>(298K)</td>
</tr>
<tr>
<td>H</td>
<td>52.095</td>
</tr>
<tr>
<td>C</td>
<td>171.291</td>
</tr>
<tr>
<td>S</td>
<td>66.636</td>
</tr>
<tr>
<td>Cl</td>
<td>29.082</td>
</tr>
<tr>
<td>CCl<span class="arithmatex">\(_4\)</span></td>
<td>-24.59</td>
</tr>
<tr>
<td>CCl<span class="arithmatex">\(_3\)</span>H</td>
<td>-24.65</td>
</tr>
<tr>
<td>CCl<span class="arithmatex">\(_2\)</span>H<span class="arithmatex">\(_2\)</span></td>
<td>-22.10</td>
</tr>
<tr>
<td>CClH<span class="arithmatex">\(_3\)</span></td>
<td>-19.32</td>
</tr>
<tr>
<td>CH<span class="arithmatex">\(_4\)</span></td>
<td>-17.88</td>
</tr>
<tr>
<td>CH<span class="arithmatex">\(_3\)</span>SH</td>
<td>-5.34</td>
</tr>
<tr>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<p>Table 3: Miscellaneous experimental gas-phase enthalpies of formation
(kcal/mol) taken from [3].</p>
<ol>
<li>Hehre, W. J., L. Radom, P.v.R. Schleyer, and J.A. Pople Ab Initio
    Molecular Orbital Theory; John Wiley &amp; Sons: New York, 1986).</li>
<li>E.J. Bylaska, D.A. Dixon, and A.R. Felmy(2000), The Free Energies
    of Reactions of Chlorinated Methanes with Aqueous Monovalent Anions:
    Application of ab initio Electronic Structure Theory, J. Phys.
    Chem. A, 104(3), 610-617.</li>
<li>Chase, M. W., Jr. Phys. Chem. Ref. Data, Monograph No. 9 1998, 9,
    1-1951.</li>
</ol>
<h2 id="plane-wave-density-functional-theory-nwpw-tutorial-4-aimdmm-simulation-of-ccl_4-64-h_2o">NWPW Tutorial 4: AIMD/MM simulation of CCl<span class="arithmatex">\(_4\)</span> + 64 H<span class="arithmatex">\(_2\)</span>O<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-tutorial-4-aimdmm-simulation-of-ccl_4-64-h_2o" title="Permanent link">&para;</a></h2>
<p>(input:<a href="ccl4-64water.nw">Media:ccl4-64water.nw</a>,
output:<a href="ccl4-64water.nwout">Media:ccl4-64water.nwout</a>)</p>
<p>In this section we show how use the PSPW module to perform a
Car-Parrinello AIMD/MM simulation for a CCl<span class="arithmatex">\(_4\)</span> molecule in a box of
64 H<span class="arithmatex">\(_2\)</span>O. Before running a PSPW Car-Parrinello simulation the system
should be on the Born-Oppenheimer surface, i.e. the one-electron
orbitals should be minimized with respect to the total energy (i.e. task
pspw energy). In this example, default pseudopotentials from the
pseudopotential library are used for C, Cl, O^ and H^, exchange
correlation functional is PBE96, The boundary condition is periodic, and
with a side length of 23.577 Bohrs and has a cutoff energy is 50 Ry).
The time step and fake mass for the Car-Parrinello run are specified to
be 5.0 au and 600.0 au, respectively.</p>
<h2 id="plane-wave-density-functional-theory-nwpw-tutorial-5-optimizing-the-unit-cell-and-geometry-of-diamond">NWPW Tutorial 5: Optimizing the Unit Cell and Geometry of Diamond<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-tutorial-5-optimizing-the-unit-cell-and-geometry-of-diamond" title="Permanent link">&para;</a></h2>
<p><img alt="diamong" src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/diamond.png" align=left with="200pt" height="190pt"/>   The PSPW and BAND
codes can be used to determine structures and energies for a wide range
of crystalline systems. It can also be used to generate band structure
and density of state
plots.</p>
<h3 id="plane-wave-density-functional-theory-optimizing-the-unit-cell-and-geometry-for-an-8-atom-supercell-of-diamond-with-pspw">Optimizing the Unit Cell and Geometry for an 8 Atom Supercell of Diamond with PSPW<a class="headerlink" href="#plane-wave-density-functional-theory-optimizing-the-unit-cell-and-geometry-for-an-8-atom-supercell-of-diamond-with-pspw" title="Permanent link">&para;</a></h3>
<p>(input:<a href="diamond-pspw.nw">Media:diamond-pspw.nw</a>,
output:<a href="diamond-pspw.nwout">Media:diamond-pspw.nwout</a>,
<a href="diamond.opt.cif">Media:diamond.opt.cif</a>)</p>
<p>(input:<a href="catom-pspw.nw">Media:catom-pspw.nw</a>,
output:<a href="catom-pspw.nwout">Media:catom-pspw.nwout</a>)</p>
<p>The following example uses the PSPW module to optimize the unit cell and
geometry for a diamond crystal. The fractional coordinates and the unit
cell are defined in the geometry block. The simulation_cell block is
not needed since NWPW automatically uses the unit cell defined in the
geometry
block.</p>
<pre><code>title &quot;Diamond 8 atom cubic cell - geometry and unit cell optimization&quot;  
echo  

permanent_dir ./perm  
scratch_dir   ./scratch  

start diamond  

memory 950 mb  

#**** Enter the geometry using fractional coordinates ****  
geometry center noautosym noautoz print   
  system crystal   
    lat_a 3.56d0   
    lat_b 3.56d0   
    lat_c 3.56d0   
    alpha 90.0d0   
    beta  90.0d0   
    gamma 90.0d0   
  end  
  C -0.50000d0 -0.50000d0 -0.50000d0  
  C  0.00000d0  0.00000d0 -0.50000d0  
  C  0.00000d0 -0.50000d0  0.00000d0  
  C -0.50000d0  0.00000d0  0.00000d0  
  C -0.25000d0 -0.25000d0 -0.25000d0  
  C  0.25000d0  0.25000d0 -0.25000d0  
  C  0.25000d0 -0.25000d0  0.25000d0  
  C -0.25000d0  0.25000d0  0.25000d0  
end  

nwpw   
  ewald_rcut 3.0  
  ewald_ncut 8  #The default value of 1 needs to be increased for small cells  
  lmbfgs  
  xc pbe96  
end  

driver   
  clear   
  maxiter 40  
end  

set nwpw:cif_filename diamond.opt  # create a CIF file containing optimization history   
set includestress .true.           # this option tells driver to optimize the unit cell  
task pspw optimize ignore
</code></pre>
<p>The optimized energy and geometry will
be</p>
<pre><code> ...  
     ----------------------  
     Optimization converged  
     ----------------------  

  Step       Energy      Delta E   Gmax     Grms     Xrms     Xmax   Walltime  
  ---- ---------------- -------- -------- -------- -------- -------- --------  
@    6     -45.07688304 -1.1D-07  0.00037  0.00021  0.00002  0.00003    174.5  
                                    ok       ok       ok       ok    



                        Geometry &quot;geometry&quot; -&gt; &quot;geometry&quot;  
                        ---------------------------------  

Output coordinates in angstroms (scale by  1.889725989 to convert to a.u.)  

 No.       Tag          Charge          X              Y              Z  
---- ---------------- ---------- -------------- -------------- --------------  
   1 C                    6.0000     1.82723789     1.82729813     1.82705440  
   2 C                    6.0000     0.00000857    -0.00006053     1.82730027  
   3 C                    6.0000    -0.00000584     1.82706061     0.00002852  
   4 C                    6.0000     1.82712018     0.00006354    -0.00002544  
   5 C                    6.0000     2.74074195     2.74072805     2.74088522  
   6 C                    6.0000     0.91366407     0.91370055     2.74064976  
   7 C                    6.0000     0.91351181     2.74080771     0.91352917  
   8 C                    6.0000     2.74078843     0.91348115     0.91365446  

     Lattice Parameters   
     ------------------   

     lattice vectors in angstroms (scale by  1.889725989 to convert to a.u.)  

     a1=&lt;   3.654   0.000   0.000 &gt;  
     a2=&lt;   0.000   3.654   0.000 &gt;  
     a3=&lt;   0.000   0.000   3.654 &gt;  
     a=       3.654 b=      3.654 c=       3.654  
     alpha=  90.000 beta=  90.000 gamma=  90.000  
     omega=    48.8  

     reciprocal lattice vectors in a.u.  

     b1=&lt;   0.910   0.000   0.000 &gt;  
     b2=&lt;   0.000   0.910   0.000 &gt;  
     b3=&lt;   0.000   0.000   0.910 &gt;  

     Atomic Mass   
     -----------   

     C                 12.000000  


==============================================================================  
                               internuclear distances  
------------------------------------------------------------------------------  
      center one      |      center two      | atomic units |  angstroms  
------------------------------------------------------------------------------  
   5 C                |   1 C                |     2.99027  |     1.58238  
   6 C                |   1 C                |     2.99027  |     1.58238  
   6 C                |   2 C                |     2.99027  |     1.58238  
   7 C                |   1 C                |     2.99026  |     1.58238  
   7 C                |   3 C                |     2.99027  |     1.58238  
   8 C                |   1 C                |     2.99027  |     1.58238  
   8 C                |   4 C                |     2.99027  |     1.58238  
------------------------------------------------------------------------------  
                        number of included internuclear distances:          7  
==============================================================================  

==============================================================================  
                                internuclear angles  
------------------------------------------------------------------------------  
       center 1       |       center 2       |       center 3       |  degrees  
------------------------------------------------------------------------------  
   5 C                |   1 C                |   6 C                |   109.46  
   5 C                |   1 C                |   7 C                |   109.48  
   5 C                |   1 C                |   8 C                |   109.48  
   6 C                |   1 C                |   7 C                |   109.47  
   6 C                |   1 C                |   8 C                |   109.46  
   7 C                |   1 C                |   8 C                |   109.48  
   1 C                |   6 C                |   2 C                |   109.48  
   1 C                |   7 C                |   3 C                |   109.47  
   1 C                |   8 C                |   4 C                |   109.47  
------------------------------------------------------------------------------  
                           number of included internuclear angles:          9  
==============================================================================  ...
</code></pre>
<p>The C-C bond distance after the geometry optimization is 1.58 Angs. and
agrees very well with the experimental value of 1.54 Angs.. Another
quantity that can be calculated from this simulation is the cohesive
energy.The cohesive energy of a crystal is the energy needed to separate
the atoms of the solid into isolated atoms,
i.e.</p>
<div class="arithmatex">\[ E_{coh}= -\left( E_{solid} - \sum_{a} E_{atom}^a  \right)  \]</div>
<p>where <span class="arithmatex">\(E_{solid}\)</span> is the energy of the solid and <span class="arithmatex">\(E_{atom}^a\)</span> are
the energies of the isolated atoms. In order to calculate the cohesive
energy the energy of an isolated carbon atom at the same level of theory
and cutoff energy will need to be calculated. The following input can be
used to the energy of an isolated carbon atom.</p>
<p>(input:<a href="catom-pspw.nw">file:catom-pspw.nw</a>,
output:<a href="catom-pspw.nwout">file:catom-pspw.nwout</a>)</p>
<pre><code>title &quot;triplet carbon atom at pbe96 level using a large unit cell&quot;  
start c1-pspw  
memory 1400 mb  

permanent_dir ./perm  
scratch_dir   ./scratch  

geometry  
C 0 0 0  
end  

nwpw  
   simulation_cell  
      FCC 38.0     #large unit cell   
      boundary_conditions   periodic  # periodic boundary conditions are used by default.  
      #boundary_conditions aperiodic  # free-space (or aperiodic) boundary conditions could also be used.  
   end  
   xc pbe96  
   mult 3  
  lmbfgs  
end  
task pspw energy
</code></pre>
<p>The total energy from the simulation will be</p>
<p><code>Total PSPW energy   :  -0.5421213534E+01</code></p>
<p>Using this energy and energy of diamond the cohesive energy per atom is
calculated to be</p>
<div class="arithmatex">\[\begin{align} E_{coh}= -\left( -45.07688304au/8 - (-5.421213534au)  \right)= 0.2133968 au = 5.8 eV \end{align}\]</div>
<p>This value is substantially lower than the experimental value of
<span class="arithmatex">\(7.37eV\)</span>! It turns out this error is a result of the unit cell being
too small for the diamond calculation (or too small of a Brillioun zone
sampling). In the next section, we show how increasing the Brillouin
zone sampling reduces the error in the calculated cohesive
energy.</p>
<h3 id="plane-wave-density-functional-theory-optimizing-the-unit-cell-for-an-8-atom-supercell-of-diamond-with-band">Optimizing the Unit Cell for an 8 Atom Supercell of Diamond with BAND<a class="headerlink" href="#plane-wave-density-functional-theory-optimizing-the-unit-cell-for-an-8-atom-supercell-of-diamond-with-band" title="Permanent link">&para;</a></h3>
<p>(input:<a href="diamond-band.nw">Media:diamond-band.nw</a>,
output:<a href="diamond-band.nwout">Media:diamond-band.nwout</a>)</p>
<p>In this example the BAND module is used to optimize the unit cell and
geometry for a diamond crystal at different Brillouin zone
samplings.</p>
<pre><code>title &quot;Diamond 8 atom cubic cell - geometry and unit cell optimization&quot;   
echo  

permanent_dir ./perm  
scratch_dir   ./scratch  

start diamond-band  

memory 1950 mb  

#**** Enter the geometry using fractional coordinates ****  
geometry center noautosym noautoz print   
  system crystal   
    lat_a 3.58d0   
    lat_b 3.58d0   
    lat_c 3.58d0   
    alpha 90.0d0   
    beta  90.0d0   
    gamma 90.0d0   
  end  
  C -0.50000d0 -0.50000d0 -0.50000d0  
  C  0.00000d0  0.00000d0 -0.50000d0  
  C  0.00000d0 -0.50000d0  0.00000d0  
  C -0.50000d0  0.00000d0  0.00000d0  
  C -0.25000d0 -0.25000d0 -0.25000d0  
  C  0.25000d0  0.25000d0 -0.25000d0  
  C  0.25000d0 -0.25000d0  0.25000d0  
  C -0.25000d0  0.25000d0  0.25000d0  
end  
set includestress    .true.   # option tells driver to optimize the unit cell  
set nwpw:zero_forces .true.   # option zeros the forces on the atoms--&gt; only lattice parameters optimized  

nwpw   
  ewald_rcut 3.0  
  ewald_ncut 8    #The default value of 1 needs to be increased  
  lmbfgs  
  xc pbe96  
end  

#1x1x1 k-point mesh  
nwpw   
  monkhorst-pack 1 1 1  
end  
set nwpw:cif_filename diamond111.opt  
driver; clear; maxiter 40; end; task band optimize ignore

#2x2x2 k-point mesh  
nwpw   
  monkhorst-pack 2 2 2  
end  
set nwpw:cif_filename diamond222.opt  
driver; clear; maxiter 40; end; task band optimize ignore

#3x3x3 k-point mesh  
nwpw   
  monkhorst-pack 3 3 3  
end  
set nwpw:cif_filename diamond333.opt  
driver; clear; maxiter 40; end; task band optimize ignore

#4x4x4 k-point mesh  
nwpw   
  monkhorst-pack 4 4 4   
end  
set nwpw:cif_filename diamond444.opt  
driver; clear; maxiter 40; end; task band optimize ignore

#5x5x5 k-point mesh  
nwpw   
  monkhorst-pack 5 5 5  
end  
set nwpw:cif_filename diamond555.opt  
driver; clear; maxiter 40; end; task band optimize ignore
</code></pre>
<p>The following figure shows a plot of the cohesive energy and C-C bond
distance versus the Brillouin zone sampling. As can be seen in this
figure the cohesive energy (w/o zero-point correction) and C-C bond
distance agree very well with the experimental values of 7.37 eV
(including zero-point correction) and 1.54
Angs.</p>
<p><img alt=" " src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/ecohdist.png" align=middle with="792pt" height="612pt"/>  </p>
<h3 id="plane-wave-density-functional-theory-using-band-to-optimize-the-unit-cell-for-a-2-atom-primitive-cell-of-diamond">Using BAND to Optimize the Unit Cell for a 2 Atom Primitive Cell of Diamond<a class="headerlink" href="#plane-wave-density-functional-theory-using-band-to-optimize-the-unit-cell-for-a-2-atom-primitive-cell-of-diamond" title="Permanent link">&para;</a></h3>
<p>(input:<a href="diamond-fcc.nw">Media:diamond-fcc.nw</a>,
output:<a href="diamond-fcc.nwout.gz">Media:diamond-fcc.nwout.gz</a>)</p>
<p>In this example the BAND module is used to optimize a 2 atom unit cell
for a diamond crystal at different Brillouin zone samplings. The
optimized energy and geometry will be (Monkhorst-Pack sampling of
11x11x11)</p>
<pre><code>     ----------------------  
     Optimization converged  
     ----------------------  

  Step       Energy      Delta E   Gmax     Grms     Xrms     Xmax   Walltime  
  ---- ---------------- -------- -------- -------- -------- -------- --------  
@    1     -11.40586236  5.2D-07  0.00039  0.00018  0.00002  0.00003    662.0  
                                     ok       ok       ok       ok    



                         Geometry &quot;geometry&quot; -&gt; &quot;geometry&quot;  
                         ---------------------------------  

 Output coordinates in angstroms (scale by  1.889725989 to convert to a.u.)  

  No.       Tag          Charge          X              Y              Z  
 ---- ---------------- ---------- -------------- -------------- --------------  
    1 C                    6.0000     0.00000000     0.00000000     0.00000000  
    2 C                    6.0000     0.72201500     1.25056532     0.51054180  

      Lattice Parameters   
      ------------------   

      lattice vectors in angstroms (scale by  1.889725989 to convert to a.u.)  

      a1=&lt;   2.165   1.251   0.001 &gt;  
      a2=&lt;   0.001   2.500   0.001 &gt;  
      a3=&lt;   0.722   1.251   2.041 &gt;  
      a=       2.500 b=      2.500 c=       2.500  
      alpha=  59.966 beta=  59.966 gamma=  59.966  
      omega=    11.0  

      reciprocal lattice vectors in a.u.  

      b1=&lt;   1.536  -0.768   0.000 &gt;  
      b2=&lt;   0.000   1.330   0.000 &gt;  
      b3=&lt;  -0.543  -0.543   1.629 &gt;  

      Atomic Mass   
      -----------   

      C                 12.000000  


 ==============================================================================  
                                internuclear distances  
 ------------------------------------------------------------------------------  
       center one      |      center two      | atomic units |  angstroms  
 ------------------------------------------------------------------------------  
    2 C                |   1 C                |     2.89435  |     1.53162  
 ------------------------------------------------------------------------------  
                         number of included internuclear distances:          1  
 ==============================================================================
</code></pre>
<p>The following figure shows a plot of the cohesive energy and C-C bond
distance versus the Brillouin zone sampling for the 8 atom SC unit cell
and the 2 atom FCC unit
cell.</p>
<p><img alt=" " src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/band-fcc.png" align=middle with="792pt" height="612pt"/>  </p>
<h3 id="plane-wave-density-functional-theory-using-band-to-calculate-the-band-structures-of-diamond">Using BAND to Calculate the Band Structures of Diamond<a class="headerlink" href="#plane-wave-density-functional-theory-using-band-to-calculate-the-band-structures-of-diamond" title="Permanent link">&para;</a></h3>
<p>(input:<a href="diamond-structure.nw">Media:diamond-structure.nw</a>,
output:<a href="diamond-structure.nwout">Media:diamond-structure.nwout</a>,
<a href="diamondfcc.restricted_band.dat">file:diamondfcc.restricted_band.dat</a>)</p>
<p>The following example uses the BAND module to calculate the band
structure for the FCC cell of the a diamond crystal. The fractional
coordinates and the unit cell are defined in the geometry block. The
simulation_cell block is not needed since NWPW automatically uses the
unit cell defined in the geometry
block.</p>
<pre><code>title &quot;Diamond 2 atom fcc cell Brillouin sampling=9x9x9 M-P - Band structure plot&quot;  
echo  

permanent_dir ./perm  
scratch_dir   ./scratch  

start diamondfcc  

memory 1950 mb  

#**** Enter the geometry using fractional coordinates ****  
geometry center noautosym noautoz print   
  system crystal   
    lat_a 2.500d0   
    lat_b 2.500d0   
    lat_c 2.500d0   
    alpha 60.0d0   
    beta  60.0d0   
    gamma 60.0d0   
  end  
 C  0.00000d0  0.00000d0  0.00000d0  
 C  0.25000d0  0.25000d0  0.25000d0  
end  

nwpw  
  ewald_rcut 3.0  
  ewald_ncut 8    #The default value of 1 needs to be increased  
  lmbfgs  
  xc pbe96  

  monkhorst-pack 9 9 9  
end  

#need to run &quot;task band energy&quot; before &quot;task band structure&quot; can be run  
task band energy  

nwpw  
   virtual 16  
   brillouin_zone  
     zone_name fccpath  
     path fcc l gamma x w k gamma  
   end  
   zone_structure_name fccpath  
end  
task band structure
</code></pre>
<p>This calculation outputs the
<a href="diamondfcc.restricted_band.dat">file:diamondfcc.restricted_band.dat</a>)
data file in the permanent_directory. A plotting (e.g. gnuplot or
xmgrace) can be used to display the band structure.</p>
<p><img alt=" " src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/band-structure.png" align=middle with="792pt" height="612pt"/>  </p>
<h3 id="plane-wave-density-functional-theory-using-band-to-calculate-the-density-of-states-of-diamond">Using BAND to Calculate the Density of States of Diamond<a class="headerlink" href="#plane-wave-density-functional-theory-using-band-to-calculate-the-density-of-states-of-diamond" title="Permanent link">&para;</a></h3>
<p>(2 atom cell -
input:<a href="diamond-dos.nw">diamond-dos.nw</a>
output:<a href="diamond-dos.nwout">diamond-dos.nwout</a>,
<a href="diamond-dos.dos.dat">diamond-dos.dos.dat</a> (8
atom cell -
input:<a href="diamond-dos8.nw">diamond-dos8.nw</a> output:
<a href="diamond-dos8.nwout.gz">diamond-dos8.nwout.gz</a>,
<a href="diamond-dos8.dos.dat">diamond-dos8.dos.dat</a></p>
<p>There are two possible ways to use the BAND module to calculate the density and projected density of states.  The first approach just uses the eigenvalues generated from an energy calculation to generate a density of states.  The following example uses this strategy to calculate the density of states and projected density of states of diamond.</p>
<pre><code>title &quot;Diamond 2 atom fcc cell Brillouin sampling=9x9x9 M-P - density of states plot&quot;  
echo  

permanent_dir ./perm  
scratch_dir   ./scratch  

start diamond-dos  

memory 1950 mb  

#**** Enter the geometry using fractional coordinates ****  
geometry center noautosym noautoz print   
  system crystal   
    lat_a 2.500d0   
    lat_b 2.500d0   
    lat_c 2.500d0   
    alpha 60.0d0   
    beta  60.0d0   
    gamma 60.0d0   
  end  
 C  0.00000d0  0.00000d0  0.00000d0  
 C  0.25000d0  0.25000d0  0.25000d0  
end  

nwpw  
  ewald_rcut 3.0  
  ewald_ncut 8    #The default value of 1 needs to be increased  
  lmbfgs  
  xc pbe96  

  monkhorst-pack 9 9 9  
  dos                   # dos keyword tells the code to calculate dos at the end of an energy calculation
  mulliken              # turn on projected density of states
  virtual 8             # include 8 virtual states
end  

task band energy  
</code></pre>
<p>The other approach uses the band structure code to calculate the eigenvalues given a precomputed density.  The approach is slower than the first approach, however, it can be used to substantially increase the number of k-points and virtual orbitals used to generate the density of states.  The following example demonstrates this capability to calculate the density of states and projected density of states of the diamond crystal.</p>
<pre><code>title &quot;Diamond 2 atom fcc cell Brillouin sampling=9x9x9 M-P - density of states plot&quot;  
echo  

permanent_dir ./perm  
scratch_dir   ./scratch  

start diamond-dos  

memory 1950 mb  

#**** Enter the geometry using fractional coordinates ****  
geometry center noautosym noautoz print   
  system crystal   
    lat_a 2.500d0   
    lat_b 2.500d0   
    lat_c 2.500d0   
    alpha 60.0d0   
    beta  60.0d0   
    gamma 60.0d0   
  end  
 C  0.00000d0  0.00000d0  0.00000d0  
 C  0.25000d0  0.25000d0  0.25000d0  
end  

nwpw  
  ewald_rcut 3.0  
  ewald_ncut 8    #The default value of 1 needs to be increased  
  lmbfgs  
  xc pbe96  

  monkhorst-pack 9 9 9  
end  

#need to run &quot;task band energy&quot; before &quot;task band dos&quot; can be run  
task band energy  

nwpw  
   virtual 26                     #26 virtual orbitals included in the DOS calculation  
   dos 0.002 700 -1.00000 2.0000  #alpha npoints emin emax,....,change default energy range and gridding. note alpha not used in task band dos calculations
   dos-grid 11 11 11  
   mulliken                       # mulliken keyword used to turn on projected density of states
end  
task band dos
</code></pre>
<p>This calculation outputs the <img alt="diamond-dos.dos.dat" src="diamond-dos.dos.dat" /> data file in the permanent_directory. A plotting
(e.g. gnuplot or xmgrace) can be used to display the density of
states.</p>
<p><img alt=" " src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/band-dos.png" align=middle with="792pt" height="612pt"/>  </p>
<h3 id="plane-wave-density-functional-theory-calculate-the-phonon-spectrum-of-diamond">Calculate the Phonon Spectrum of Diamond<a class="headerlink" href="#plane-wave-density-functional-theory-calculate-the-phonon-spectrum-of-diamond" title="Permanent link">&para;</a></h3>
<pre><code>title &quot;Diamond 2 atom fcc cell Brillouin sampling=9x9x9 M-P - Phonon spectra&quot;  
echo  

permanent_dir ./perm  
scratch_dir   ./scratch  

start diamond-dos  

memory 1950 mb  

#**** Enter the geometry using fractional coordinates ****  
geometry center noautosym noautoz print   
  system crystal   
    lat_a 2.500d0   
    lat_b 2.500d0   
    lat_c 2.500d0   
    alpha 60.0d0   
    beta  60.0d0   
    gamma 60.0d0   
  end  
 C  0.00000d0  0.00000d0  0.00000d0  
 C  0.25000d0  0.25000d0  0.25000d0  
end  

nwpw  
  ewald_rcut 3.0  
  ewald_ncut 8    #The default value of 1 needs to be increased  
  lmbfgs  
  xc pbe96  

  monkhorst-pack 9 9 9  
end  

task band energy  
task band freq

</code></pre>
<h2 id="plane-wave-density-functional-theory-nwpw-tutorial-6-optimizing-the-unit-cell-of-nickel-with-fractional-occupation">NWPW Tutorial 6: optimizing the unit cell of nickel with fractional occupation<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-tutorial-6-optimizing-the-unit-cell-of-nickel-with-fractional-occupation" title="Permanent link">&para;</a></h2>
<p>(input:<a href="Ni-band.nw">Media:Ni-band.nw</a>
output:<a href="Ni-band.nwout">Media:Ni-band.nwout</a>) The
following example demonstrates how to uses the BAND module to optimize
the unit cell and geometry for FCC cell of Nickel
metal</p>
<pre><code>title &quot;Ni FCC metal, monkhorst-pack=3x3x3, 5x5x5, and 7x7x7, fermi smearing, xc=pbe96&quot;  
echo  

start Ni-band  

memory 1900 mb  

permanent_dir ./perm  
scratch_dir   ./scratch  

geometry units angstroms center noautosym noautoz print  
 system crystal  
    lat_a 3.5451d0  
    lat_b 3.5451d0  
    lat_c 3.5454d0  
    alpha 90.0d0  
    beta  90.0d0  
    gamma 90.0d0  
  end  

Ni 0.000000   0.000000   0.000000  
Ni 0.000000   0.500000   0.500000   
Ni 0.500000   0.000000   0.500000   
Ni 0.500000   0.500000   0.000000  
end  
set nwpw:cif_filename Ni-band  
set nwpw:zero_forces .true.  
set includestress    .true.  

#turn on pseudopotential filtering   
set nwpw:kbpp_ray .true.  
set nwpw:kbpp_filter .true.  

nwpw  
   #fractional occupation  
   smear fermi   

   #scf option used with smear  
   scf anderson outer_iterations 0 kerker 2.0    

   ewald_ncut 8  
   ewald_rcut 3.0  
   xc pbe96  
   monkhorst-pack 3 3 3  
   np_dimensions -1 -1 4   
end  

#generate initial wavefunctions w/ low cutoff energy  
nwpw  
   loop 10 10  
   cutoff 10.0  
end  
task band energy  

#increase cutoff energy and number of iterations  
nwpw  
   cutoff 50.0  
   loop 10 100  
end

#3x3x3 k-point mesh  
nwpw   
  monkhorst-pack 3 3 3  
end  
set nwpw:cif_filename nickel333.opt  
driver; clear; maxiter 40; end; task band optimize ignore

#5x5x5 k-point mesh  
nwpw   
  monkhorst-pack 5 5 5  
end  
set nwpw:cif_filename nickel555.opt  
driver; clear; maxiter 40; end; task band optimize ignore

#7x7x7 k-point mesh  
nwpw   
  monkhorst-pack 7 7 7  
end  
set nwpw:cif_filename nickel777.opt  
driver; clear; maxiter 40; end; task band optimize ignore
</code></pre>
<p>The following figure shows a plot of the cohesive energy and Ni-Ni bond
distance versus the Brillouin zone sampling. As can be seen in this
figure the cohesive energy (w/o zero-point correction) and Ni-Ni bond
distance agree very well with the experimental values of 4.44 eV
(including zero-point correction) and 2.49
Angs.</p>
<p><img alt=" " src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/niecohdist.png" align=middle with="792pt" height="612pt"/>  </p>
<h2 id="plane-wave-density-functional-theory-nwpw-tutorial-7-optimizing-the-unit-cells-with-symmetry-diamond-with-fd-3m-symmetry-and-brucite-with-p-3m1-symmetry">NWPW Tutorial 7: Optimizing the unit cells with symmetry: Diamond with Fd-3m symmetry and Brucite with P-3m1 symmetry<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-tutorial-7-optimizing-the-unit-cells-with-symmetry-diamond-with-fd-3m-symmetry-and-brucite-with-p-3m1-symmetry" title="Permanent link">&para;</a></h2>
<p>(Diamond example,
input:<a href="diamond-symmetry.nw">Media:diamond-symmetry.nw</a>,
output:<a href="diamond-symmetry.nwout">Media:diamond-symmetry.nwout</a>)</p>
<p>(Brucite example,
input:<a href="brucite-symmetry.nw">Media:brucite-symmetry.nw</a>,
output:<a href="brucite-symmetry.nwout">Media:brucite-symmetry.nwout</a>)</p>
<p>The following example uses the BAND module to optimize the unit cell and
geometry for a Diamond crystal with Fd-3m symmetry. The fractional
coordinates, unit cell, and symmetry are defined in the geometry
block.</p>
<pre><code>title &quot;Diamond 8 atom cubic cell generated using Fd-3m symmetry - geometry and unit cell optimization&quot; 
echo 

memory 1500 mb

permanent_dir ./perm
scratch_dir   ./scratch

start diamond-symmetry  


geometry nocenter noautosym noautoz print 
 system crystal 
   lat_a 3.58 
   lat_b 3.58 
   lat_c 3.58  
   alpha 90.0  
   beta  90.0  
   gamma 90.0  
 end
symmetry Fd-3m
C 0.0 0.0 0.0
end 
set nwpw:cif_filename diamond-symmetry

#turn on pseudopotential filtering  
set nwpw:kbpp_ray    .true.
set nwpw:kbpp_filter .true.

#***** setup the nwpw Band code - 3x3x3 k-point mesh ****  
nwpw
  ewald_rcut 3.0
  ewald_ncut 8
  xc pbe96  
  lmbfgs
  monkhorst-pack 3 3 3
  np_dimensions -1 -1 4
end  

set includestress  .true.   # tell driver to optimize unit cell
set includelattice .true.   # tell driver to optimize with a,b,c,alpha,beta,gamma
task band optimize ignore
</code></pre>
<p>The optimized geometry will also contain the information about the
symmetry being
used</p>
<pre><code>....  
     ----------------------  
     Optimization converged  
     ----------------------  


 Step       Energy      Delta E   Gmax     Grms     Xrms     Xmax   Walltime  
 ---- ---------------- -------- -------- -------- -------- -------- --------  
@    7     -45.62102901 -4.1D-07  0.00010  0.00003  0.00019  0.00060    287.1  
                                     ok       ok       ok       ok    



                        Geometry &quot;geometry&quot; -&gt; &quot;geometry&quot;  
                        ---------------------------------  

Output coordinates in angstroms (scale by  1.889725989 to convert to a.u.)  

 No.       Tag          Charge          X              Y              Z  
---- ---------------- ---------- -------------- -------------- --------------  
   1 C                    6.0000     0.00000000     0.00000000     0.00000000  
   2 C                    6.0000     0.00000000     1.76715074     1.76715074  
   3 C                    6.0000     1.76715074     1.76715074     0.00000000  
   4 C                    6.0000     1.76715074     0.00000000     1.76715074  
   5 C                    6.0000     2.65072611     0.88357537     2.65072611  
   6 C                    6.0000     0.88357537     0.88357537     0.88357537  
   7 C                    6.0000     0.88357537     2.65072611     2.65072611  
   8 C                    6.0000     2.65072611     2.65072611     0.88357537  

     Lattice Parameters   
     ------------------   

     lattice vectors in angstroms (scale by  1.889725989 to convert to a.u.)  

     a1=&lt;   3.534   0.000   0.000 &gt;  
     a2=&lt;   0.000   3.534   0.000 &gt;  
     a3=&lt;   0.000   0.000   3.534 &gt;  
     a=       3.534 b=      3.534 c=       3.534  
     alpha=  90.000 beta=  90.000 gamma=  90.000  
     omega=    44.1  

     reciprocal lattice vectors in a.u.  

     b1=&lt;   0.941   0.000   0.000 &gt;  
     b2=&lt;   0.000   0.941   0.000 &gt;  
     b3=&lt;   0.000   0.000   0.941 &gt;  

     Atomic Mass   
     -----------   

     C                 12.000000  


     Symmetry information  
     --------------------  

Group name                  Fd-3m  
Group number            227  
Group order             192  
No. of unique centers     1  
Setting number            1  

     Symmetry unique atoms  

    1  

==============================================================================  
                               internuclear distances  
------------------------------------------------------------------------------  
      center one      |      center two      | atomic units |  angstroms  
------------------------------------------------------------------------------  
   5 C                |   4 C                |     2.89203  |     1.53040  
   6 C                |   1 C                |     2.89203  |     1.53040  
   6 C                |   2 C                |     2.89203  |     1.53040  
   6 C                |   3 C                |     2.89203  |     1.53040  
   6 C                |   4 C                |     2.89203  |     1.53040  
   7 C                |   2 C                |     2.89203  |     1.53040  
   8 C                |   3 C                |     2.89203  |     1.53040  
------------------------------------------------------------------------------  
                        number of included internuclear distances:          7  
==============================================================================  



==============================================================================  
                                internuclear angles  
------------------------------------------------------------------------------  
       center 1       |       center 2       |       center 3       |  degrees  
------------------------------------------------------------------------------  
   6 C                |   2 C                |   7 C                |   109.47  
   6 C                |   3 C                |   8 C                |   109.47  
   5 C                |   4 C                |   6 C                |   109.47  
   1 C                |   6 C                |   2 C                |   109.47  
   1 C                |   6 C                |   3 C                |   109.47  
   1 C                |   6 C                |   4 C                |   109.47  
   2 C                |   6 C                |   3 C                |   109.47  
   2 C                |   6 C                |   4 C                |   109.47  
   3 C                |   6 C                |   4 C                |   109.47  
------------------------------------------------------------------------------  
                           number of included internuclear angles:          9  
==============================================================================
</code></pre>
<p>The following example uses the BAND module to optimize the unit cell and
geometry for a <a href="http://en.wikipedia.org/wiki/Brucite">Brucite</a> crystal
(Mg(OH)<sub>2</sub> with P-3m1
symmetry.</p>
<p><img alt="400px" src="Brucite-small.jpg" /></p>
<pre><code>title &quot;brucite testing - using P-3m1 symmetry&quot;  
echo  

memory 1500 mb  

permanent_dir ./perm  
scratch_dir      ./scratch  

geometry nocenter noautosym noautoz print   
 system crystal   
   lat_a 3.14979  
   lat_b 3.14979  
   lat_c 4.7702  
   alpha 90.0  
   beta  90.0  
   gamma 120.0  
 end  
symmetry P-3m1  
Mg   0.00000   0.00000  0.00000  
O   0.33333   0.66667   0.22030  
H   0.33333   0.66667   0.41300  
end   
set nwpw:cif_filename brucite  

#turn on pseudopotential filtering   
set nwpw:kbpp_ray .true.  
set nwpw:kbpp_filter .true.  

#***** setup the nwpw gamma point code ****  
nwpw  
  ewald_rcut 3.0  
  ewald_ncut 8  
  xc pbe96  
  lmbfgs  
  monkhorst-pack 3 3 2  
  #np_dimensions -1 -1 4  
end  

driver  
  clear  
  maxiter 31  
end  

set includestress  .true.          # tell driver to optimize unit cell  
set includelattice .true.  

task band optimize ignore
</code></pre>
<p>Optimizing Brucite, which is a soft layered material (2.5-3 Mohs scale),
is more difficult to optimize than a hard material such as Diamond. For
these types of materials using symmetry can often result in a faster
optimization. For example, with symmetry the optimization converges
within 20 to 30 geometry optimization
steps,</p>
<pre><code>@ Step       Energy      Delta E   Gmax     Grms     Xrms     Xmax   Walltime  
@ ---- ---------------- -------- -------- -------- -------- -------- --------  
@    0     -34.39207476  0.0D+00  0.24673  0.10223  0.00000  0.00000    172.7  
@    1     -34.39340208 -1.3D-03  0.00872  0.00302  0.00198  0.00485    328.5  
....  
@   20     -34.39042736 -1.2D-05  0.00195  0.00083  0.00440  0.01964   3019.2  
@   21     -34.39043463 -7.3D-06  0.00028  0.00011  0.00493  0.02042   3150.6  
@   22     -34.39043484 -2.1D-07  0.00043  0.00014  0.00002  0.00008   3278.5  
@   22     -34.39043484 -2.1D-07  0.00043  0.00014  0.00002  0.00008   3278.5
</code></pre>
<p>whereas, without symmetry the optimization may not be converged even at
100 geometry steps
(input:<a href="brucite-nosymmetry.nw">Media:brucite-nosymmetry.nw</a>,
output:<a href="brucite-nosymmetry.nwout">Media:brucite-nosymmetry.nwout</a>).</p>
<pre><code>@ Step       Energy      Delta E   Gmax     Grms     Xrms     Xmax   Walltime  
@ ---- ---------------- -------- -------- -------- -------- -------- --------  
@    0     -34.39207476  0.0D+00  0.24673  0.10250  0.00000  0.00000     18.4  
@    1     -34.39340765 -1.3D-03  0.02963  0.00715  0.00202  0.00500     30.7  
...  
@   49     -34.39027641 -2.1D-06  0.01870  0.00646  0.00074  0.00202    595.7  
@   50     -34.39027503  1.4D-06  0.01962  0.00669  0.00069  0.00197    608.4  
...  
@  100     -34.39034236 -3.8D-07  0.00380  0.00150  0.00036  0.00132   1155.3  
@  101     -34.39034431 -1.9D-06  0.00305  0.00118  0.00012  0.00045   1166.8  
@  102     -34.39034449 -1.8D-07  0.00370  0.00144  0.00006  0.00020   1177.9  
...
</code></pre>
<h2 id="plane-wave-density-functional-theory-nwpw-tutorial-8-nvt-metropolis-monte-carlo-simulations">NWPW Tutorial 8: NVT Metropolis Monte-Carlo Simulations<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-tutorial-8-nvt-metropolis-monte-carlo-simulations" title="Permanent link">&para;</a></h2>
<p>In this example the PSPW module is used to run an NVT simulation for a
diamond crystal using the a Metropolis Monte-Carlo algorithm.</p>
<!-- (input:[Media:diamond-nvt.nw](diamond-nvt.nw.md),
output:[Media:diamond-nvt.nwout.gz](diamond-nvt.nwout.gz.md),
datafiles:[Media:diamond-nvt.emotion.gz](diamond-nvt.emotion.gz.md),
[Media:diamond-nvt.ion_motion.gz](diamond-nvt.ion_motion.gz.md),
[Media:diamond-nvt.xyz.gz](diamond-nvt.xyz.gz.md),
[Media:diamond_nvt_234.cif.gz](diamond_nvt_234.cif.gz.md)) -->

<p><img alt="NVT Metropolis Monte-Carlo Trajectory (Markov Chain) for diamond,
T=300K." src="diamond-nvt.gif" title="NVT Metropolis Monte-Carlo Trajectory (Markov Chain.md) for diamond, T=300K." /></p>
<pre><code>title &quot;Metropolis NVT simulation of diamond - this input is used to put the system in equilibrium&quot;  
echo  

start diamond-nvt  

#permanent_dir ./perm  
#scratch_dir   ./perm  

#**** Enter the geometry using fractional coordinates ****  
geometry center noautosym noautoz print   
system crystal   
   lat_a 3.56d0   
   lat_b 3.56d0   
   lat_c 3.56d0   
   alpha 90.0d0   
   beta  90.0d0   
   gamma 90.0d0   
end  
C -0.50000d0 -0.50000d0 -0.50000d0  
C  0.00000d0  0.00000d0 -0.50000d0  
C  0.00000d0 -0.50000d0  0.00000d0  
C -0.50000d0  0.00000d0  0.00000d0  
C -0.25000d0 -0.25000d0 -0.25000d0  
C  0.25000d0  0.25000d0 -0.25000d0  
C  0.25000d0 -0.25000d0  0.25000d0  
C -0.25000d0  0.25000d0  0.25000d0  
end  
set nwpw:cif_filename diamond_nvt_234  

###### setup the nwpw gamma point code ######  
set nwpw:kbpp_ray .true.  
set nwpw:kbpp_filter .true.  
set nwpw:frozen_lattice:thresh 999.0  
nwpw  
   lmbfgs  
   ewald_rcut 3.0  
   ewald_ncut 8  
   xc pbe  
end  
task pspw energy  

##### optimize the unit cell #####  
set includestress  .true.  #this option tells driver to optimize the unit cell  
set includelattice .true.  #this option tells driver to optimize cell using a,b,c,alpha,beta,gamma   
driver  
   clear  
   maxiter 51  
end  
task pspw optimize ignore  

####################################################################################  
###### setup Metropolis NVT code - input will change in a forthcoming release ######  
####################################################################################  
set nwpw:mc_seed 234              # Seed for random number generator  
set nwpw:mc_algorithm 1            # 1-NVT; 2-NPT  
set nwpw:mc_aratio 0.234           # targeted acceptance ratio  
set nwpw:mc_ddx 0.1                # parameter used to adjust geometry dispacement to have sampling with targeted acceptance  
set nwpw:mc_temperature 300.0      # Temperature in K  
set nwpw:mc_step_size   0.250      # initial geometry displacement step size  

nwpw  
   mc_steps 10 100  #total number of iterations = 10*100, number of iterations between step size adjustments = 10  
   cpmd_properties on  
end  
task pspw Metropolis
</code></pre>
<h2 id="plane-wave-density-functional-theory-nwpw-tutorial-9-npt-metropolis-monte-carlo-simulations">NWPW Tutorial 9: NPT Metropolis Monte-Carlo Simulations<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-tutorial-9-npt-metropolis-monte-carlo-simulations" title="Permanent link">&para;</a></h2>
<p>In this example the PSPW module is used to run an NPT simulation for a
diamond crystal using the a Metropolis Monte-Carlo
algorithm.</p>
<p>(input:<a href="diamond-metropolis.nw">Media:diamond-metropolis.nw</a>,
output:<a href="diamond-metropolis.nwout.gz">Media:diamond-metropolis.nwout.gz</a>,
datafiles:<a href="diamond-metropolis.emotion.gz">Media:diamond-metropolis.emotion.gz</a>,
<a href="diamond-metropolis.ion_motion.gz">Media:diamond-metropolis.ion_motion.gz</a>,
<a href="diamond-metropolis.xyz.gz">Media:diamond-metropolis.xyz.gz</a>,
<a href="diamond_metropolis_1234.cif.gz">Media:diamond_metropolis_1234.cif.gz</a>)</p>
<pre><code>title &quot;Metropolis NPT simulation of diamond - this input is used to put the system in equilibrium&quot;  
echo  

start diamond-metropolis  

#permanent_dir ./perm  
#scratch_dir   ./perm  

#**** Enter the geometry using fractional coordinates ****  
geometry center noautosym noautoz print   
  system crystal   
    lat_a 3.56d0   
    lat_b 3.56d0   
    lat_c 3.56d0   
    alpha 90.0d0   
    beta  90.0d0   
    gamma 90.0d0    
  end  
 C -0.50000d0 -0.50000d0 -0.50000d0  
 C  0.00000d0  0.00000d0 -0.50000d0  
 C  0.00000d0 -0.50000d0  0.00000d0  
 C -0.50000d0  0.00000d0  0.00000d0  
 C -0.25000d0 -0.25000d0 -0.25000d0  
 C  0.25000d0  0.25000d0 -0.25000d0  
 C  0.25000d0 -0.25000d0  0.25000d0  
 C -0.25000d0  0.25000d0  0.25000d0  
end  
set nwpw:cif_filename pspw_metropolis  

###### setup the nwpw gamma point code ######  
set nwpw:kbpp_ray .true.  
set nwpw:kbpp_filter .true.  
set nwpw:frozen_lattice:thresh 999.0  
nwpw  
   lmbfgs  
   ewald_rcut 3.0  
   ewald_ncut 8  
   xc pbe  
end  
task pspw energy  


####################################################################################  
###### setup Metropolis NPT code - input will change in a forthcoming release ######  
####################################################################################  
set nwpw:mc_seed 1234              # Seed for random number generator  
set nwpw:mc_algorithm 2            # 1-NVT; 2-NPT  
set nwpw:mc_aratio 0.234           # targeted acceptance ratio  
set nwpw:mc_ddx 0.1                # parameter used to adjust geometry dispacement to have sampling with targeted acceptance  
set nwpw:mc_ddv 0.1                # parameter used to adjust volume change to have sampling with targeted acceptance  
set nwpw:mc_temperature 300.0      # Temperature in K  
set nwpw:mc_step_size   0.250      # geometry displacement step size  
set nwpw:mc_volume_step 0.130      # volume displacement step size  

nwpw  
   bo_steps 10 100  #total number of iterations = 10*100, number of iterations between step size adjustments = 10  
end  
task pspw Metropolis
</code></pre>
<p><img alt=" " src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/diamond-metropolis.png" align=middle with="640pt" height="480pt"/>  </p>
<p>(inputs:<a href="diamond-metropolis-sampling.nw.tgz">Media:diamond-metropolis-sampling.nw.tgz</a>)</p>
<p>(python analysis
program:<a href="makehistogram.gz">Media:makehistogram.gz</a>)</p>
<pre><code>[WE27972:~/Projects/NWChem/Metropolis] bylaska% makehistogram -t 300 -c 2 1235/diamond-metropolis-1235.emotion 1236/diamond-metropolis-1236.emotion   
1237/diamond-metropolis-1237.emotion 1238/diamond-metropolis-1238.emotion 1239/diamond-metropolis-1239.emotion 1240/diamond-metropolis-1240.emotion   
1241/diamond-metropolis-1241.emotion 1242/diamond-metropolis-1242.emotion 1243/diamond-metropolis-1243.emotion 1244/diamond-metropolis-1244.emotion   
1245/diamond-metropolis-1245.emotion 1246/diamond-metropolis-1246.emotion 1248/diamond-metropolis-1248.emotion 1249/diamond-metropolis-1249.emotion   
makehistogram Program  
len(args)= 14  

unitconversion  = 1.0  
temperature (K) = 300  
RT (au)         = 0.000949482834326  ( 0.5958  kcal/mol)  

data columns -1 = [1]  
data rows (n)   = 52000  

delta (au)       = 0.01  
xmin-delta (au)  = -45.08080365  
xmax+delta (au)  = -45.05079515  

data averaging:  
- xbar       (au)           = -45.0668093497  
- S2_{n-1}   (au)           = 1.08378343101e-05  
- &lt;exp((x-xmin)/RT)&gt; (au)   = 5293374903.39  
- &lt;exp((x-xbar)/RT)&gt; (au)   = 2102.44405413  
- Free energy               = -45.0595449934  
- Free energy1              = -45.0595449934  

histogram distribution parameters:  
- number of bins (Rice k) = 75  
- bin width               = 0.00040552027027  
- norm                    = 1.0  
- xbar (au)               = -45.0668107987  (error= -1.44908364064e-06 )  
- S2_{n-1} (au)           = 1.0858459744e-05  (error= 2.06254339582e-08 )  
- &lt;exp((x-xmin)/RT)&gt; (au) = 5184600342.01  (error= -108774561.378 )  
- &lt;exp((x-xbar)/RT)&gt; (au) = 2062.38570923  (error= -40.0583449011 )  
- Free energy               = -45.0595647078  (error= -1.9714360235e-05 )  
- Free energy1              = -45.0595647078  (error= -1.9714360235e-05 )  
- histogram plot file     = histogram.dat  

normal distribution parameters:  
- average x (input xbar)                   = -45.0668093497  
- unbiased sample variance (input S2_(n-1))= 1.08378343101e-05  
- xbar-xmin                                = 0.0139943003357  
- norm                    = 0.99998877936  
- xbar (au)               = -45.0663035243  (error= 0.000505825397738 )  
- S2_{n-1} (au)           = 1.1091077321e-05  (error= 2.53243010936e-07 )  
- &lt;exp((x-xmin)/RT)&gt; (au) = 943482808.939  (error= -4349892094.45 )  
- &lt;exp((x-xbar)/RT)&gt; (au) = 219.968603653  (error= -1882.47545048 )  
- Free energy               = -45.061182503  (error= -0.00163750957643 )  
- Free energy1              = -45.061182503  (error= -0.00163750957643 )  
- normal distribution plot file = normdist.dat  
- number data points = 1500  

gamma distribution parameters:  
- alpha0= 18.0700715921  
- beta0 = 1291.24508969  
- xmin + alpha0/beta0 = -45.0668093497  
- alpha = 18.5003178357  
- beta  = 1321.98948086  
- xmin + alpha/beta   = -45.0668093497  
- norm                    = 0.999923464137 0.99993569948  
- xbar (au)               = -45.0633614482 -45.0639126423  (error= 0.00344790150088 0.00289670733491 )  
- S2_{n-1} (au)           = 2.27110055327e-05 1.89632753897e-05  (error= 1.18731712226e-05 8.12544107961e-06 )  
- &lt;exp((x-xmin)/RT)&gt; (au) = 7932775654.26 7060892836.07  (error= 2639400750.87 1767517932.68 )  
- &lt;exp((x-xbar)/RT)&gt; (au) = 83.43400035 132.707151194  (error= -2019.01005378 -1969.73690294 )  
- Free energy               = -45.059160883 -45.0592714327  (error= 0.000384110406969 0.000273560709338 )  
- Free energy1              = -45.059160883 -45.0592714327  (error= 0.000384110406969 0.000273560709338 )  
- gamma distribution plot file = gammadist.dat  
- number data points = 1500  

Hausdorff distribution parameters:  
- xmin = -45.08080365  
- xmax = -45.05079515  
- number moments     = 15  
  -- &lt; x^0 &gt;    =    1.000000000000000  
  -- &lt; x^1 &gt;    =    0.466344546904007  
  -- &lt; x^2 &gt;    =    0.229512222180349  
  -- &lt; x^3 &gt;    =    0.119040323347820  
  -- &lt; x^4 &gt;    =    0.064946164109284  
  -- &lt; x^5 &gt;    =    0.037186896798964  
  -- &lt; x^6 &gt;    =    0.022287980659815  
  -- &lt; x^7 &gt;    =    0.013942929105868  
  -- &lt; x^8 &gt;    =    0.009076370636747  
  -- &lt; x^9 &gt;    =    0.006128509645342  
  -- &lt; x^10 &gt;   =    0.004278147917961  
  -- &lt; x^11 &gt;   =    0.003077410986590  
  -- &lt; x^12 &gt;   =    0.002273768533280  
  -- &lt; x^13 &gt;   =    0.001720304299285  
  -- &lt; x^14 &gt;   =    0.001328990330385  
- norm                    = 1.0000000003  
- xbar (au)               = -45.066809363  (error= -1.33426993898e-08 )  
- S2_{n-1} (au)           = 1.08376258908e-05  (error= -2.08419282206e-10 )  
- &lt;exp((x-xmin)/RT)&gt; (au) = 5423305875.35  (error= 129930971.958 )  
- &lt;exp((x-xbar)/RT)&gt; (au) = 2154.08083332  (error= 51.6367791881 )  
- Free energy               = -45.0595219689  (error= 2.30245307122e-05 )  
- Free energy1              = -45.0595219689  (error= 2.30245307122e-05 )  
- Hausdorff moment history file = moment_hist.dat  
- Hausdorff distribution plot file = hausdorff.dat  
- number data points = 1500
</code></pre>
<p><img alt=" " src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/diamond-e-histogram.png" align=middle with="640pt" height="480pt"/><br />
<img alt=" " src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/diamond-v-histogram.png" align=middle with="640pt" height="480pt"/>  </p>
<h2 id="plane-wave-density-functional-theory-nwpw-tutorial-9-free-energy-simulations">NWPW Tutorial 9: Free Energy Simulations<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-tutorial-9-free-energy-simulations" title="Permanent link">&para;</a></h2>
<p>A description of using the WHAM method for generating free energy of the
gas-phase dissociation reaction CH<span class="arithmatex">\(_3\)</span>Cl <span class="arithmatex">\(\rightarrow\)</span> CH<span class="arithmatex">\(_3\)</span>+Cl
can be found in the attached pdf
(<a href="nwchem-new-pmf.pdf">nwchem-new-pmf.pdf</a>)</p>
<h2 id="plane-wave-density-functional-theory-paw-tutorial">PAW Tutorial<a class="headerlink" href="#plane-wave-density-functional-theory-paw-tutorial" title="Permanent link">&para;</a></h2>
<h3 id="plane-wave-density-functional-theory-optimizing-a-water-molecule">Optimizing a water molecule<a class="headerlink" href="#plane-wave-density-functional-theory-optimizing-a-water-molecule" title="Permanent link">&para;</a></h3>
<p>The following input deck performs for a water molecule a PSPW energy
calculation followed by a PAW energy calculation and a PAW geometry
optimization calculation. The default unit cell parameters are used
(SC=20.0, ngrid 32 32 32). In this simulation, the first PAW run
optimizes the wavefunction and the second PAW run optimizes the
wavefunction and geometry in tandem.</p>
<pre><code>title &quot;paw steepest descent test&quot;  
start paw_test 
charge 0 
geometry units au nocenter noautoz noautosym  
O      0.00000    0.00000    0.01390 
H     -1.49490    0.00000   -1.18710  
H      1.49490    0.00000   -1.18710  
end 
nwpw  
  time_step 15.8  
  ewald_rcut 1.50  
  tolerances 1.0d-8 1.0d-8  
end  
set nwpw:lcao_iterations 1  
set nwpw:minimizer 2  
task pspw energy  
task paw energy  
nwpw  
  time_step 5.8  
  geometry_optimize  
  ewald_rcut 1.50  
  tolerances 1.0d-7 1.0d-7 1.0d-4  
end
task paw steepest_descent 
task paw optimize
</code></pre>
<h3 id="plane-wave-density-functional-theory-optimizing-a-unit-cell-and-geometry-for-silicon-carbide">Optimizing a unit cell and geometry for Silicon-Carbide<a class="headerlink" href="#plane-wave-density-functional-theory-optimizing-a-unit-cell-and-geometry-for-silicon-carbide" title="Permanent link">&para;</a></h3>
<p>The following example demonstrates how to uses the PAW module to
optimize the unit cell and geometry for a silicon-carbide
crystal.</p>
<pre><code>title &quot;SiC 8 atom cubic cell - geometry and unit cell optimization&quot; 
start SiC
#**** Enter the geometry using fractional coordinates ****  
geometry units au center noautosym noautoz print
 system crystal
   lat_a 8.277d0
   lat_b 8.277d0 
   lat_c 8.277d0  
   alpha 90.0d0  
   beta  90.0d0
   gamma 90.0d0
 end 
Si    -0.50000d0  -0.50000d0  -0.50000d0
Si     0.00000d0   0.00000d0  -0.50000d0  
Si     0.00000d0  -0.50000d0   0.00000d0 
Si    -0.50000d0   0.00000d0   0.00000d0
C     -0.25000d0  -0.25000d0  -0.25000d0
C      0.25000d0   0.25000d0  -0.25000d0 
C      0.25000d0  -0.25000d0   0.25000d0
C     -0.25000d0   0.25000d0   0.25000d0  
end  
#***** setup the nwpw gamma point code ****  
nwpw
  simulation_cell 
    ngrid 16 16 16
  end
  ewald_ncut 8 
end  
set nwpw:minimizer 2
set nwpw:psi_nolattice .true.  # turns of unit cell checking for wavefunctions 
driver
  clear 
  maxiter 40  
end
set includestress .true.         # this option tells driver to optimize the unit cell
set nwpw:stress_numerical .true. #currently only numerical stresses implemented in paw
task paw optimize
</code></pre>
<h3 id="plane-wave-density-functional-theory-running-a-car-parrinello-simulation">Running a Car-Parrinello Simulation<a class="headerlink" href="#plane-wave-density-functional-theory-running-a-car-parrinello-simulation" title="Permanent link">&para;</a></h3>
<p>In this section we show how use the PAW module to perform a
   Car-Parrinello molecular dynamic simulation for a C<sub>2</sub> molecule at
the LDA level. Before running a PAW Car-Parrinello simulation the system
should be on the Born-Oppenheimer surface, i.e. the one-electron
orbitals should be minimized with respect to the total energy (i.e. task
pspw energy). The input needed is basically the same as for optimizing
   the geometry of a C<sub>2</sub> molecule at the LDA level,except that and
additional Car-Parrinello sub-block is added.</p>
<p>In the following example we show the input needed to run a
   Car-Parrinello simulation for a C<sub>2</sub> molecule at the LDA level. In
this example, default pseudopotentials from the pseudopotential library
are used for C, the boundary condition is free-space, the exchange
correlation functional is LDA, The boundary condition is free-space, and
the simulation cell cell is aperiodic and cubic with a side length of
10.0 Angstroms and has 40 grid points in each direction (cutoff energy
is 44 Ry). The time step and fake mass for the Car-Parrinello run are
specified to be 5.0 au and 600.0 au,
respectively.</p>
<pre><code>start c2_paw_lda_md  
title &quot;C2 restricted singlet dimer, LDA/44Ry - constant energy Car-Parrinello simulation&quot;
geometry  
 C    -0.62 0.0 0.0 
 C     0.62 0.0 0.0 
end
pspw 
  simulation_cell units angstroms  
     boundary_conditions aperiodic 
     lattice 
       lat_a 10.00d0  
       lat_b 10.00d0
       lat_c 10.00d0
     end  
     ngrid 40 40 40
  end  
  Car-Parrinello 
    fake_mass 600.0
    time_step 5.0 
    loop 10 10  
  end 
end 
set nwpw:minimizer 2
task paw energy
task paw Car-Parrinello
</code></pre>
<h2 id="plane-wave-density-functional-theory-nwpw-capabilities-and-limitations">NWPW Capabilities and Limitations<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-capabilities-and-limitations" title="Permanent link">&para;</a></h2>
<ul>
<li>Hybrid Functionals (e.g. PBE0, LDA-SIC) only work in PSPW.</li>
<li>Wannier orbital task only works in PSPW.</li>
<li>AIMD/MM simulation only works with PSPW.</li>
</ul>
<h2 id="plane-wave-density-functional-theory-questions-and-difficulties">Questions and Difficulties<a class="headerlink" href="#plane-wave-density-functional-theory-questions-and-difficulties" title="Permanent link">&para;</a></h2>
<p>Questions and encountered problems should be reported to the <a href="#forum">NWChem
Community Forum</a> or to Eric J. Bylaska,
Eric.Bylaska@pnl.gov</p></section><section class="print-page" id="tce" heading-number="11.6"><h1 id="tce-tensor-contraction-engine-module-ci-mbpt-and-cc">Tensor Contraction Engine Module: CI, MBPT, and CC<a class="headerlink" href="#tce-tensor-contraction-engine-module-ci-mbpt-and-cc" title="Permanent link">&para;</a></h1>
<h2 id="tce-overview">Overview<a class="headerlink" href="#tce-overview" title="Permanent link">&para;</a></h2>
<p>The Tensor Contraction Engine (TCE) Module of NWChem implements a
variety of approximations that converge at the exact solutions of
Schrdinger equation. They include configuration interaction theory
through singles, doubles, triples, and quadruples substitutions,
coupled-cluster theory through connected singles, doubles, triples, and
quadruples substitutions, and many-body perturbation theory through
fourth order in its tensor formulation. Not only optimized parallel
programs of some of these high-end correlation theories are new, but
also the way in which they have been developed is unique. The working
equations of all of these methods have been derived completely
automatically by a symbolic manipulation program called a Tensor
Contraction Engine (TCE), and the optimized parallel programs have also
been computer-generated by the same program, which were interfaced to
NWChem. The development of the TCE program and this portion of the
NWChem program has been financially supported by the United States
Department of Energy, Office of Science, Office of Basic Energy Science,
through the SciDAC program.</p>
<p>The capabilities of the module include:</p>
<ul>
<li>Restricted Hartree-Fock, unrestricted Hartree-Fock, and restricted
    open-shell Hartree-Fock references,</li>
<li>Restricted KS DFT and unrestricted KS DFT references,</li>
<li>Unrestricted configuration interaction theory (CISD, CISDT, and
    CISDTQ),</li>
<li>Unrestricted coupled-cluster theory (LCCD, CCD, LCCSD, CCSD, QCISD,
    CCSDT, CCSDTQ),</li>
<li>Unrestricted iterative many-body perturbation theory [MBPT(2),
    MBPT(3), MBPT(4)] in its tensor formulation,</li>
<li>Unrestricted coupled-cluster singles and doubles with perturbative
    connected triples {CCSD(T), CCSD[T]},</li>
<li>Unrestricted equation-of-motion coupled-cluster theory (EOM-CCSD,
    EOM-CCSDT, EOM-CCSDTQ) for excitation energies, transition moments
    and oscillator strengths, and excited-state dipole moments,</li>
<li>Unrestricted coupled-cluster theory (CCSD, CCSDT, CCSDTQ) for dipole
    moments.</li>
<li>Several variants of active-space CCSDt and EOMCCSDt methods that
    employ limited set of triply excited cluster amplitudes defined by
    active orbitals.</li>
<li>Ground-state non-iterative CC approaches that account for the effect
    of triply and/or quadruply excited connected clusters: the
    perturbative approaches based on the similarity transformed
    Hamiltonian: CCSD(2), CCSD(2)<sub>T</sub>, CCSDT(2)<sub>Q</sub>, the
    completely and locally renormalized methods: CR-CCSD(T), LR-CCSD(T),
    LR-CCSD(TQ)-1.</li>
<li>Excited-state non-iterative corrections due to triples to the
    EOMCCSD excitation energies: the completely renormalized EOMCCSD(T)
    method (CR-EOMCCSD(T)).</li>
<li>Dynamic dipole polarizabilities at the CCSD and CCSDT levels using
    the linear response formalism.</li>
<li>Ground- and excited- states the iterative second-order model CC2.</li>
<li>Dynamic dipole polarizabilities at the CCSDTQ level using the linear
    response formalism.</li>
<li>State-specific Multireference Coupled Cluster methods (MRCC)
    (Brillouin-Wigner (BW-MRCC) and Mukherjee (Mk-MRCC) approaches).</li>
<li>Universally State Selective corrections to the BW-MRCC and Mk-MRCC methods 
    (diagonal USS(2) and perturbative USS(pt) methods).</li>
<li>Electron affinity/Ionization potential EOMCCSD formulations
    (EA/IP-EOMCC; available for RHF reference only).</li>
</ul>
<p>The distributed binary executables do not contain CCSDTQ and its
derivative methods, owing to their large volume. The source code
includes them, so a user can reinstate them by setenv CCSDTQ yes and
recompile TCE module. The following optimizations have been used in the
module:</p>
<ul>
<li>Spin symmetry (spin integration is performed wherever possible
    within the unrestricted framework, making the present unrestricted
    program optimal for an open-shell system. The spin adaption was not
    performed, although in a restricted calculation for a closed-shell
    system, certain spin blocks of integrals and amplitudes are further
    omitted by symmetry, and consequently, the present unrestricted CCSD
    requires only twice as many operations as a spin-adapted restricted
    CCSD for a closed-shell system),</li>
<li>Point-group symmetry,</li>
<li>Index permutation symmetry,</li>
<li>Runtime orbital range tiling for memory management,</li>
<li>Dynamic load balancing (local index sort and matrix multiplications)
    parallelism,</li>
<li>Multiple parallel I/O schemes including fully in-core algorithm
    using Global Arrays,</li>
<li>Frozen core and virtual approximation.</li>
<li>DIIS extrapolation and Jacobi update of excitation amplitudes</li>
<li>Additional algorithms for the 2-e integral transformation, including
    efficient and scalable spin-free out-of-core N<sup>5</sup> algorithms.</li>
<li>Hybrid I/O schemes for both spin-orbital and spin-free calculations
    which eliminate the memory bottleneck of the 2-e integrals in favor
    of disk storage. Calculations with nearly 400 basis functions at the
    CCSD(T) can be performed on workstation using this method.</li>
<li>Parallel check-pointing and restart for ground-state (including
    property) calculations at the CCSD, CCSDT and CCSDTQ levels of
    theory.</li>
</ul>
<h2 id="tce-performance-of-ci-mbpt-and-cc-methods">Performance of CI, MBPT, and CC methods<a class="headerlink" href="#tce-performance-of-ci-mbpt-and-cc-methods" title="Permanent link">&para;</a></h2>
<p>For reviews or tutorials of these highly-accurate correlation methods,
the user is referred to:</p>
<ul>
<li>Trygve Helgaker, Poul Jorgensen and Jeppe Olsen, Molecular
    Electronic-Structure Theory.</li>
<li>A. Szabo and N. S. Ostlund, Modern Quantum Chemistry: Introduction
    to Advanced Electronic Structure Theory.</li>
<li>B. O. Roos (editor), Lecture Notes in Quantum Chemistry.</li>
</ul>
<p>For background on development of the symbolic algebra tools which help
create the code used in this model see:</p>
<ul>
<li>S. Hirata, J. Phys. Chem. A 107, 9887 (2003).</li>
<li>S. Hirata, J. Chem. Phys. 121, 51 (2004).</li>
<li>S. Hirata, Theor. Chem. Acc. 116, 2 (2006).</li>
</ul>
<p>For details of particular CC implementations, see:</p>
<ul>
<li>S. Hirata, P.-D. Fan, A.A. Auer, M. Nooijen, P. Piecuch, J. Chem.
    Phys. 121, 12197 (2004).</li>
<li>K. Kowalski, S. Hirata, M. Wloch, P. Piecuch, T.L. Windus, J. Chem.
    Phys. 123, 074319 (2005).</li>
<li>K. Kowalski, P. Piecuch, J. Chem. Phys. 115, 643 (2001).</li>
<li>K. Kowalski, P. Piecuch, Chem. Phys. Lett. 347, 237 (2001).</li>
<li>P. Piecuch, S.A. Kucharski, and R.J. Bartlett, J. Chem. Phys. 110,
    6103 (1999).</li>
<li>P. Piecuch, N. Oliphant, and L. Adamowicz, J. Chem. Phys. 99, 1875
    (1993).</li>
<li>N. Oliphant and L. Adamowicz, Int. Rev. Phys. Chem. 12, 339 (1993).</li>
<li>P. Piecuch, N. Oliphant, and L. Adamowicz, J. Chem. Phys. 99, 1875
    (1993).</li>
<li>K. Kowalski, P. Piecuch, J. Chem. Phys. 120, 1715 (2004).</li>
<li>K. Kowalski, J. Chem. Phys. 123, 014102 (2005).</li>
<li>P. Piecuch, K. Kowalski, I.S.O. Pimienta, M.J. McGuire, Int. Rev.
    Phys. Chem. 21, 527 (2002).</li>
<li>K. Kowalski, P. Piecuch, J. Chem. Phys. 122, 074107 (2005)</li>
<li>K. Kowalski, W. de Jong, J. Mol. Struct.: THEOCHEM, 768, 45 (2006).</li>
<li>O. Christiansen, H. Koch, P. Jrgensen, Chem. Phys. Lett. 243, 409
    (1995).</li>
<li>K. Kowalski, J. Chem. Phys. 125, 124101 (2006).</li>
<li>K. Kowalski, S. Krishnamoorthy, O. Villa, J.R. Hammond, N. Govind,
    J. Chem. Phys. 132, 154103 (2010).</li>
<li>J.R. Hammond, K. Kowalski, W.A. de Jong, J. Chem. Phys. 127, 144105
    (2007).</li>
<li>J.R. Hammond, W.A. de Jong, K. Kowalski, J. Chem. Phys. 128, 224102
    (2008).</li>
<li>J.R. Hammond, K. Kowalski, J. Chem. Phys. 130, 194108 (2009).</li>
<li>W. Ma. S. Krishnamoorthy, O. Villa, J. Chem. Theory Comput. 7, 1316
    (2011).</li>
<li>J.Brabec, J. Pittner, H.J.J. van Dam, E. Apra, K. Kowalski, J. Chem.
    Theory Comput. 8, 487 (2012).</li>
<li>K. Bhaskaran-Nair, J. Brabec, E. Apra, H.J.J. van Dam. J. Pittner,
    K. Kowalski, J. Chem. Phys. 137, 094112 (2012).</li>
<li>K. Bhaskaran-Nair, K. Kowalski, J. Moreno, M. Jarrell, W.A. Shelton,
    J. Chem. Phys. 141, 074304 (2014).</li>
<li>J. Brabec, S. Banik, K. Kowalski, J.  Pittner, J. Chem. Phys. 145, 164106
    (2016).</li>
<li>S. Rajbhandari, F. Rastello, K. Kowalski, S. Krishnamoorthy, P.  Sadayappan, 
    Proceedings of the 22nd ACM SIGPLAN Symposium on Principles and Practice of Parallel Programming
    327 (2017). </li>
</ul>
<h2 id="tce-algorithms-of-ci-mbpt-and-cc-methods">Algorithms of CI, MBPT, and CC methods<a class="headerlink" href="#tce-algorithms-of-ci-mbpt-and-cc-methods" title="Permanent link">&para;</a></h2>
<h3 id="tce-spin-spatial-and-index-permutation-symmetry">Spin, spatial, and index permutation symmetry<a class="headerlink" href="#tce-spin-spatial-and-index-permutation-symmetry" title="Permanent link">&para;</a></h3>
<p>The TCE thoroughly analyzes the working equation of many-electron theory
models and automatically generates a program that takes full advantage
of these symmetries at the same time. To do so, the TCE first recognizes
the index permutation symmetries among the working equations, and
perform strength reduction and factorization by carefully monitoring the
index permutation symmetries of intermediate tensors. Accordingly, every
input and output tensor (such as integrals, excitation amplitudes,
residuals) has just two independent but strictly ordered index strings,
and each intermediate tensor has just four independent but strictly
ordered index strings. The operation cost and storage size of tensor
contraction is minimized by using the index range restriction arising
from these index permutation symmetries and also spin and spatial
symmetry integration.</p>
<h3 id="tce-runtime-orbital-range-tiling">Runtime orbital range tiling<a class="headerlink" href="#tce-runtime-orbital-range-tiling" title="Permanent link">&para;</a></h3>
<p>To maintain the peak local memory usage at a manageable level, in the
beginning of the calculation, the orbitals are rearranged into tiles
(blocks) that contains orbitals with the same spin and spatial
symmetries. So the tensor contractions in these methods are carried out
at the tile level; the spin, spatial, and index permutation symmetry is
employed to reduce the operation and storage cost at the tile level
also. The so-called tile-structure of all tensors used in CC equations
is also the key-factor determining the parallel structure of the TCE CC
codes . The tiling scheme corresponds to partitioning of the
spin-orbital domain into smaller subsets containing the spin-orbitals of
the same spin and spatial symmetries (the so-called tiles). This
partitioning of the spin-orbital domain entails the blocking of all
tensors corresponding to one- and two-electron integrals, cluster
amplitudes, and all recursive intermediates, into smaller blocks of the
size defined by the size of the tile (or tilesize for short). Since the
parallel scheme used in all TCE generated codes is deeply rooted in
dynamic load balancing techniques, the tile-structure defines the
granularity of the work to be distributed. The size of tiles (tilesize)
defines also the local memory requirements in all TCE derived CC
implementations. For CI/CC/EOMCC/LR-CC models based on the sinlges and
doubles models (CISD,CCSD,EOMCCSD,LR-CCSD) the peak local memory
requirement is proportional to the <code>tilesize</code><sup>4</sup>. In approaches
accounting for triples, either in iterative or non-iterative fashion,
the local memory usage is proportional to <code>tilesize</code><sup>6</sup>. This means
that in the CCSD(T), CCSDt, CCSDT, CR-EOMCCSD(T), EOMCCSDt, EOMCCSDT,
LR-CCSDT caluclations the tilesize cannot be defined too large.</p>
<h3 id="tce-dynamic-load-balancing-parallelism">Dynamic load balancing parallelism<a class="headerlink" href="#tce-dynamic-load-balancing-parallelism" title="Permanent link">&para;</a></h3>
<p>In a parallel execution, dynamic load balancing of tile-level local
tensor index sorting and local tensor contraction (matrix
multiplication) will be invoked.</p>
<h3 id="tce-parallel-io-schemes">Parallel I/O schemes<a class="headerlink" href="#tce-parallel-io-schemes" title="Permanent link">&para;</a></h3>
<p>Each process is assigned a local tensor index sorting and tensor
contraction dynamically. It must first retrieve the tiles of input
tensors, and perform these local operations, and accumulate the output
tensors to the storage. We have developed a uniform interface for these
I/O operations to either (1) a global file on a global file system, (2)
a global memory on a global or distributed memory system, and (3)
semi-replicated files on a distributed file systems. Some of these
operations depend on the ParSoft library.</p>
<h2 id="tce-input-syntax">Input syntax<a class="headerlink" href="#tce-input-syntax" title="Permanent link">&para;</a></h2>
<p>The keyword to invoke the many-electron theories in the module is TCE.
To perform a single-point energy calculation, include</p>
<pre><code> TASK TCE ENERGY
</code></pre>
<p>in the input file, which may be preceded by the TCE input block that
details the calculations:</p>
<pre><code> TCE
   [(DFT||HF||SCF) default HF=SCF]
   [FREEZE [[core] (atomic || &lt;integer nfzc default 0&gt;)] \
            [virtual &lt;integer nfzv default 0&gt;]]
   [(LCCD||CCD||CCSD||CC2||LR-CCSD||LCCSD||CCSDT||CCSDTA||CCSDTQ|| \
     CCSD(T)||CCSD[T]||CCSD(2)_T||CCSD(2)||CCSDT(2)_Q|| \
     CR-CCSD[T]||CR-CCSD(T)|| \
     LR-CCSD(T)||LR-CCSD(TQ)-1||CREOMSD(T)|| \
     QCISD||CISD||CISDT||CISDTQ|| \
     MBPT2||MBPT3||MBPT4||MP2||MP3||MP4) default CCSD]
   [THRESH &lt;double thresh default 1e-6&gt;]
   [MAXITER &lt;integer maxiter default 100&gt;]
   [PRINT (none||low||medium||high||debug)
     &lt;string list_of_names ...&gt;]
   [IO (fortran||eaf||ga||sf||replicated||dra||ga_eaf) default ga]
   [DIIS &lt;integer diis default 5&gt;]
   [LSHIFT &lt;double lshift default is 0.0d0&gt;]
   [NROOTS &lt;integer nroots default 0&gt;]
   [TARGET &lt;integer target default 1&gt;]
   [TARGETSYM &lt;character targetsym default 'none'&gt;]
   [SYMMETRY]
   [2EORB]
   [2EMET &lt;integer fast2e default 1&gt;]
   [T3A_LVL] 
   [ACTIVE_OA]
   [ACTIVE_OB]
   [ACTIVE_VA]
   [ACTIVE_VB]
   [DIPOLE]
   [TILESIZE &lt;no default (automatically adjusted)&gt;]
   [(NO)FOCK &lt;logical recompf default .true.&gt;]
   [FRAGMENT &lt;default -1 (off)&gt;]
 END
</code></pre>
<p>Also supported are energy gradient calculation, geometry optimization,
and vibrational frequency (or hessian) calculation, on the basis of
numerical differentiation. To perform these calculations, use</p>
<pre><code>     TASK TCE GRADIENT
</code></pre>
<p>or</p>
<pre><code>     TASK TCE OPTIMIZE
</code></pre>
<p>or</p>
<pre><code>     TASK TCE FREQUENCIES
</code></pre>
<p>The user may also specify the parameters of reference wave function
calculation in a separate block for either HF (SCF) or DFT, depending on
the first keyword in the above syntax.</p>
<p>Since every keyword except the model has a default value, a minimal
input file will be</p>
<pre><code> GEOMETRY
   Be 0.0 0.0 0.0
 END
 BASIS
   Be library cc-pVDZ
 END
 TCE
   ccsd
 END
 TASK TCE ENERGY
</code></pre>
<p>which performs a CCSD/cc-pVDZ calculation of the Be atom in its singlet
ground state with a spin-restricted HF reference.</p>
<p>New implementations of the iterative CCSD and EOMCCSD methods based on the
improved task scheduling can be enable by the <code>set tce:nts T</code> command as
in the following example:</p>
<pre><code>geometry/basis set specifications 
tce
freeze atomic
creomccsd(t)
tilesize 20
2eorb
2emet 13
eomsol 2
end 

set tce:nts T

task tce energy
</code></pre>
<p>New task scheduling should reduce time to solutions and provide better
parallel performance especially in large CCSD/EOMCCSD runs.</p>
<h2 id="tce-keywords-of-tce-input-block">Keywords of TCE input block<a class="headerlink" href="#tce-keywords-of-tce-input-block" title="Permanent link">&para;</a></h2>
<h3 id="tce-hf-scf-or-dft-the-reference-wave-function">HF, SCF, or DFT: the reference wave function<a class="headerlink" href="#tce-hf-scf-or-dft-the-reference-wave-function" title="Permanent link">&para;</a></h3>
<p>This keyword tells the module which of the HF (SCF) or DFT module is
going to be used for the calculation of a reference wave function. The
keyword HF and SCF are one and the same keyword internally, and are
default. When these are used, the details of the HF (SCF) calculation
can be specified in the SCF input block, whereas if DFT is chosen, DFT
input block may be provided.</p>
<p>For instance, RHF-RCCSDT calculation (R standing for spin-restricted)
can be performed with the following input blocks:</p>
<pre><code> SCF
   SINGLET
   RHF
 END
 TCE
   SCF
   CCSDT
 END
 TASK TCE ENERGY
</code></pre>
<p>This calculation (and any correlation calculation in the TCE module
using a RHF or RDFT reference for a closed-shell system) skips the
storage and computation of all &beta; spin blocks of integrals and excitation
amplitudes. ROHF-UCCSDT (U standing for spin-unrestricted) for an
open-shell doublet system can be requested by</p>
<pre><code> SCF
   DOUBLET
   ROHF
 END
 TCE
   SCF
   CCSDT
 END
 TASK TCE ENERGY
</code></pre>
<p>and likewise, UHF-UCCSDT for an open-shell doublet system can be
specified with</p>
<pre><code> SCF
   DOUBLET
   UHF
 END
 TCE
   SCF
   CCSDT
 END
 TASK TCE ENERGY
</code></pre>
<p>The operation and storage costs of the last two calculations are
identical. To use the KS DFT reference wave function for a UCCSD
calculation of an open-shell doublet system,</p>
<pre><code> DFT
   ODFT
   MULT 2
 END
 TCE
   DFT
   CCSD
 END
 TASK TCE ENERGY
</code></pre>
<p>Note that the default model of the DFT module is
LDA.</p>
<h3 id="tce-ccsdccsdtccsdtqcisdcisdtcisdtq-mbpt2mbpt3mbpt4-etc-the-correlation-models">CCSD,CCSDT,CCSDTQ,CISD,CISDT,CISDTQ, MBPT2,MBPT3,MBPT4, etc.: the correlation models<a class="headerlink" href="#tce-ccsdccsdtccsdtqcisdcisdtcisdtq-mbpt2mbpt3mbpt4-etc-the-correlation-models" title="Permanent link">&para;</a></h3>
<p>These keywords stand for the following models:</p>
<ul>
<li>LCCD: linearized coupled-cluster doubles,</li>
<li>CCD: coupled-cluster doubles,</li>
<li>LCCSD: linearized coupled-cluster singles &amp; doubles,</li>
<li>CCSD: coupled-cluster singles &amp; doubles (also EOM-CCSD),</li>
<li>CCSD_ACT: coupled-cluster singles &amp; active doubles (also
    active-space EOMCCSD),</li>
<li>LR-CCSD: locally renormalized EOMCCSD method.</li>
<li>EACCSD: Electron affinity EOMCCSD method.</li>
<li>IPCCSD: Ionization potential EOMCCSD method.</li>
<li>CC2: second-order approximate coupled cluster with singles and
    doubles model</li>
<li>CCSDT: coupled-cluster singles, doubles, &amp; triples (also EOM-CCSDT),</li>
<li>CCSDTA: coupled-cluster singles, doubles, &amp; active triples (also
    EOM-CCSDT). Three variants of the active-space CCSDt and EOMCCSDt
    approaches can be selected based on various definitions of triply
    excited clusters: (1) version I (keyword T3A_LVL 1) uses the
    largest set of triply excited amplitudes defined by at least one
    occupied and one unoccupied active spinorbital labels. (2) Version
    II (keyword T3A_LVL 2) uses triply excited amplitudes that carry at
    least two occupied and unoccupied active spinorbital labels. (3)
    Version III (keyword T3A_LVL 3) uses triply excited amplitudes that
    are defined by active indices only. Each version requires defining
    relevant set of occupied active &alpha; and &beta; spinorbitals (ACTIVE_OA and
    <code>ACTIVE_OB</code>) as well as active unoccupied &alpha; and &beta; spinorbitals
    (<code>ACTIVE_VA</code> and <code>ACTIVE_VB</code>).</li>
<li>CCSDTQ: coupled-cluster singles, doubles, triples, &amp; quadruples
    (also EOM-CCSDTQ),</li>
<li>CCSD(T): CCSD and perturbative connected triples,</li>
<li>CCSD[T]: CCSD and perturbative connected triples,</li>
<li>CR-CCSD[T]: completely renormalized CCSD[T] method,</li>
<li>CR-CCSD(T): completely renormalized CCSD(T) method,</li>
<li>CCSD(2)_T: CCSD and perturbative CCSD(T)_T correction,</li>
<li>CCSD(2)_TQ: CCSD and perturbative CCSD(2) correction,</li>
<li>CCSDT(2)_Q: CCSDT and perturbative CCSDT(2)_Q correction.</li>
<li>LR-CCSD(T): CCSD and perturbative locally renormalized CCSD(T)
    correction,</li>
<li>LR-CCSD(TQ)-1: CCSD and perturbative locally renormalized CCSD(TQ)
    (LR-CCSD(TQ)-1) correction,</li>
<li>CREOMSD(T): EOMCCSD energies and completely renormalized
    EOMCCSD(T)(IA) correction. In this option NWCHEM prints two
    components: (1) total energy of the K-th state
    E<sub>K</sub>=E<sub>K</sub> </sup>EOMCCSD</sup>+&delta;<sub>K</sub><sup>CR-EOMCCSD(T),IA</sup>(T) and (2)
    the so-called -corrected EOMCCSD excitation energy
    &omega;<sub>K</sub><sup>CR-EOMCCSD(T),IA</sup>=&omega;<sub>K</sub><sup>EOMCCSD</sup>+&delta;<sub>K</sub><sup> CR-EOMCCSD(T),IA</sup>(T).</li>
<li>CREOM(T)AC: active-space CR-EOMCCSD(T) approach,</li>
<li>QCISD: quadratic configuration interaction singles &amp; doubles,</li>
<li>CISD: configuration interaction singles &amp; doubles,</li>
<li>CISDT: configuration interaction singles, doubles, &amp; triples,</li>
<li>CISDTQ: configuration interaction singles, doubles, triples, &amp;
    quadruples,</li>
<li>MBPT2=MP2: iterative tensor second-order many-body or Mller-Plesset
    perturbation theory,</li>
<li>MBPT3=MP3: iterative tensor third-order many-body or Mller-Plesset
    perturbation theory,</li>
<li>MBPT4=MP4: iterative tensor fourth-order many-body or Mller-Plesset
    perturbation theory,</li>
</ul>
<p>All of these models are based on spin-orbital expressions of the
amplitude and energy equations, and designed primarily for
spin-unrestricted reference wave functions. However, for a restricted
reference wave function of a closed-shell system, some further reduction
of operation and storage cost will be made. Within the unrestricted
framework, all these methods take full advantage of spin, spatial, and
index permutation symmetries to save operation and storage costs at
every stage of the calculation. Consequently, these computer-generated
programs will perform significantly faster than, for instance, a
hand-written spin-adapted CCSD program in NWChem, although the nominal
operation cost for a spin-adapted CCSD is just one half of that for
spin-unrestricted CCSD (in spin-unrestricted CCSD there are three
independent sets of excitation amplitudes, whereas in spin-adapted CCSD
there is only one set, so the nominal operation cost for the latter is
one third of that of the former. For a restricted reference wave
function of a closed-shell system, all  spin block of the excitation
amplitudes and integrals can be trivially mapped to the all  spin
block, reducing the ratio to one half).</p>
<p>While the MBPT (MP) models implemented in the TCE module give identical
correlation energies as conventional implementation for a canonical HF
reference of a closed-shell system, the former are intrinsically more
general and theoretically robust for other less standard reference wave
functions and open-shell systems. This is because the zeroth order of
Hamiltonian is chosen to be the full Fock operator (not just the
diagonal part), and no further approximation was invoked. So unlike the
conventional implementation where the Fock matrix is assumed to be
diagonal and a correlation energy is evaluated in a single analytical
formula that involves orbital energies (or diagonal Fock matrix
elements), the present tensor MBPT requires the iterative solution of
amplitude equations and subsequent energy evaluation and is generally
more expensive than the former. For example, the operation cost of many
conventional implementation of MBPT(2) scales as the fourth power of the
system size, but the cost of the present tensor MBPT(2) scales as the
fifth power of the system size, as the latter permits non-canonical HF
reference and the former does not (to reinstate the non-canonical HF
reference in the former makes it also scale as the fifth power of the
system size).</p>
<h3 id="tce-state-specific-multireference-coupled-cluster-methods-mrcc">State-Specific Multireference Coupled Cluster methods (MRCC)<a class="headerlink" href="#tce-state-specific-multireference-coupled-cluster-methods-mrcc" title="Permanent link">&para;</a></h3>
<p>Several State-Specific MRCC methods have been implemented in 6.3 release
of nwchem. These include:</p>
<ul>
<li>Iterative Brillouin-Wigner MRCC method employing single and double
    excitations (BW-MRCCSD)</li>
<li>Iterative Mukherjee MRCC method employing single and double
    excitations (Mk-MRCCSD)</li>
<li>Non-iterative methods accounting for triple excitations in MRCC
    formalisms: BW-MRCCSD(T) and Mk-MRCCSD(T)</li>
</ul>
<p>The current implementation can be used in studies of systems composed of
an even number of correlated electrons (this limitation will be removed
in the next release). This includes typical examples of diradical,
open-shell singlets, and bond-forming/breaking processes where the
corresponding wavefunctions have strong quasidegenerate character.</p>
<p>To enable the compilation of the MRCC codes one has to set the following
variable before the compilation of NWChem</p>
<pre><code> export MRCC_METHODS=y
</code></pre>
<p>To run MRCC calculations the user has to define two groups in the input
file. First, the TCE group and secondly the MRCCDATA group. In the TCE
group the iterative level of theory is defined, e.g. BWCCSD or MKCCSD.
This implementation was designed for complete model spaces (CMS) which
means that the modelspace contains all Slater determinants of all
possible (in the context of the spatial and spin symmetry, M<sub>s</sub>)
distributions of active electrons among active spin orbitals. The user
can define the modelspace in two ways. As a first approach the model
space can be defined by hand, as shown in the two examples below. The
input of the model space starts with the <code>NREF</code> keyword followed by the
number of reference configurations that will be used, which should equal
the number of strings for references below. In the input <code>2</code> refers to
doubly occupied orbitals, <code>a</code> to alpha electrons, <code>b</code> to beta electrons
and <code>0</code> identifies an unoccupied orbital. When the model space is
defined by hand the occupation strings have to include the frozen
orbitals as well. In the second way the CMS can be generated using the
keyword <code>CAS</code> followed by the number of active electrons and the number
of active orbitals. When using the <code>CAS</code> keyword we strongly recommend
that users check the references that are generated.</p>
<p>As the model space typically includes multiple configurations it is
possible to use the MRCC method to calculate excited states instead of
the ground state. For this reason it is required to specify the root of
interest. The <code>ROOT</code> keyword followed by the root number specifies the
state the code calculates. The lowest root, the ground state, is
identified as <code>root 1</code>. If one wants to calculate the third root the
keyword <code>ROOT 3</code> should be used. An example is given below.</p>
<pre><code> echo
 start tce_mrcc_bwcc 
 memory stack 1000 mb heap 100 mb global 500 mb verify
 geometry units au
   H                     0.00000000     -2.27289450     -1.58834700
   O                     0.00000000      0.00000000     -0.01350000
   H                     0.00000000      2.27289450     -1.58834700
 end
 basis spherical
   O library cc-pvdz
   H library cc-pvdz
 end
 charge 0
 scf
   rohf
   singlet
   thresh 1.0e-10
   tol2e 1.0e-10
 end
 tce
   bwccsd
   thresh 1.0e-7
   targetsym a1
   io ga
   tilesize 18
 end
 mrccdata
   root 1
   nref 4
   222220
   222202
   2222ab
   2222ba
 end
 task tce energy
</code></pre>
<pre><code> echo
 start tce_mrcc_mkcc 
 memory stack 1000 mb heap 100 mb global 500 mb verify
 geometry units au
   H                   0.00000000     -2.27289450     -1.58834700
   O                   0.00000000      0.00000000     -0.01350000
   H                   0.00000000      2.27289450     -1.58834700
 end
 basis spherical
   O library cc-pvdz
   H library cc-pvdz
 end
 charge 0
 scf
   rohf
   singlet
   thresh 1.0e-10
   tol2e 1.0e-10
 end
 tce
   mkccsd
   thresh 1.0e-5
   targetsym a1
   maxiter 100
   io ga
   tilesize 18
 end
 mrccdata
   root 1
   cas 2 2 # Please make sure the references generated are correct.
 end
 task tce energy
</code></pre>
<p>This version of MRCC works only with GA as specified by the <code>IO GA</code>
option. In addition this code works only with the spin-orbit 4-index
transformation, however in all circumstances an RHF Hartree-Fock initial
calculation has to be used. In this implementation the effective
Hamiltonian operator contains only scalar, one- and two-body many body
components. Finally, in our implementation the BWCCSD methods use the
energy threshold for the convergence, whereas the MKCCSD method uses the
norm of the residual.</p>
<p>In addition to the iterative single-double calculations the code can
calculate non-iterative triples corrections. To request these triples
corrections the keyword <code>SE4T</code> should be added to the MRCCDATA block.
The implementation details and the from of the triples correction are
given in equation 20 [ <a href="http://dx.doi.org/10.1063/1.4747698">J. Chem. Phys. 137, 094112
(2012)</a>].</p>
<pre><code> echo
 start tce_mrcc_bwcc 
 memory stack 1000 mb heap 100 mb global 500 mb verify
 geometry units au
   H                     0.00000000     -2.27289450     -1.58834700
   O                     0.00000000      0.00000000     -0.01350000
   H                     0.00000000      2.27289450     -1.58834700
 end
 basis spherical
   O library cc-pvdz
   H library cc-pvdz
 end
 charge 0
 scf
   rohf
   singlet
   thresh 1.0e-10
   tol2e 1.0e-10
 end
 tce
   bwccsd
   thresh 1.0e-7
   targetsym a1
   io ga
   tilesize 18
 end
 mrccdata
   se4t
   no_aposteriori
   root 1
   nref 4
   222220
   222202
   2222ab
   2222ba
 end
 task tce ener
</code></pre>
<pre><code> echo
 start tce_mrcc_mkcc 
 memory stack 1000 mb heap 100 mb global 500 mb verify
 geometry units au
   H                     0.00000000     -2.27289450     -1.58834700
   O                     0.00000000      0.00000000     -0.01350000
   H                     0.00000000      2.27289450     -1.58834700
 end
 basis spherical
   O library cc-pvdz
   H library cc-pvdz
 end
 charge 0
 scf
   rohf
   singlet
   thresh 1.0e-10
   tol2e 1.0e-10
 end
 tce
   mkccsd
   thresh 1.0e-5
   targetsym a1
   io ga
   tilesize 18
   maxiter 100
 end
 mrccdata
   se4t
   root 1
   nref 4
   222220
   222202
   2222ab
   2222ba
 end
 task tce ener
</code></pre>
<h3 id="tce-implementation-notes-for-reference-level-parallelism-in-mrcc">Implementation notes for reference-level-parallelism in MRCC<a class="headerlink" href="#tce-implementation-notes-for-reference-level-parallelism-in-mrcc" title="Permanent link">&para;</a></h3>
<p>The current version of the MRCC codes contains also a pilot
implementation of the reference-level-parallelism based on the use of
processor groups for BWCCSD and BWCCSD(T) approaches. The main ideas of
this approach have been described in</p>
<ul>
<li>J.Brabec, J. Pittner, H.J.J. van Dam, E. Apra, K. Kowalski, <a href="http://dx.doi.org/10.1021/ct200809m">J.
    Chem. Theory Comput. 8, 487
    (2012)</a>.</li>
<li>K. Bhaskaran-Nair, J. Brabec, E. Apra, H.J.J. van Dam. J. Pittner,
    K. Kowalski, <a href="http://dx.doi.org/10.1063/1.4747698">J. Chem. Phys. 137, 094112
    (2012)</a>.</li>
</ul>
<p>Two essential keywords have to be added to the <code>mrccdata</code> block of the
input:</p>
<pre><code>subgroupsize n
improvetiling
</code></pre>
<p>and</p>
<pre><code>diis 0
</code></pre>
<p>in tce block. The line <code>subgroupsize n</code> defines the size of the subgroup and
<code>improvetiling</code> refers to the data representation in the MRCC subgroup
algorithm. For example, if user has 4 references and total 32 cores/CPU
then n should be defined as 32/4=8. If user has 10 references and 1200
cores/CPU available then the size of the subgroupsize (n) is 120.</p>
<pre><code> echo
 start tce_mrcc_bwcc_subgroups 
 memory stack 1000 mb heap 100 mb global 500 mb verify
 geometry units au
   H                   0.00000000     -2.27289450     -1.58834700
   O                   0.00000000      0.00000000     -0.01350000
   H                   0.00000000      2.27289450     -1.58834700
 end
 basis spherical
   O library cc-pvdz
   H library cc-pvdz
 end
 charge 0
 scf
   rohf
   singlet
   thresh 1e-12
   tol2e 1e-12
 end
 tce
   bwccsd
   targetsym a1
   io ga
   diis 0
   thresh 1e-7
   tilesize 18
 end
 mrccdata
   subgroupsize 2 # Please read the documentation below.
   improvetiling
   root 1
   cas 2 2
 end
 task tce ener
</code></pre>
<p>CAUTION: Before using the subgroup-based algorithm the users should
perform the GA subgroup test in
<code>$NWCHEM_TOP/src/tools/ga-5-6-3/global/testing/pgtest.x</code> and <code>pg2test.x</code> in the
same location. Additionally it is strongly encouraged to run the NWChem
QA tests from the $NWCHEM_TOP/QA/tests/tce_mrcc_bwcc_subgroups directory
with various combinations of subgroup size and total number of CPU.</p>
<p>The USS corrections can be enabled by using <code>usspt</code> directive
keyword in the mrccdata</p>
<pre><code>tce
  mkccsd
  thresh 1.0e-10
  targetsym a1
  maxiter 600
  io ga
end

mrccdata
  usspt
  root 1
  cas 2 2
end
</code></pre>
<p>In effect both diagonal and perturbative USS corrections will be calculated after
the completion of iterative Mk-MRCCSD or BW-MRCCSD calculations.</p>
<h3 id="tce-electron-affinity-ionization-potential-eomccsd-methods">Electron affinity, ionization potential EOMCCSD methods<a class="headerlink" href="#tce-electron-affinity-ionization-potential-eomccsd-methods" title="Permanent link">&para;</a></h3>
<p>The EA/IP-EOMCCSD methodologies are available in the 6.5 NWChem release.
These implementation are available for the RHF type of the reference
function. To enable the compilation of the EA/IP-EOMCCSD codes one has
to set the following variable before the compilation of NWChem</p>
<pre><code>export EACCSD=y
export IPCCSD=y
</code></pre>
<p>Two input examples for the EA/IP-EOMCCSD calculations are shown below.</p>
<ul>
<li>EA-EOMCCSD calculations for the ozone molecule</li>
</ul>
<pre><code>start tce_eaccsd_ozone
title &quot;tce_eaccsd_ozone&quot;
echo

memory stack 1000 mb heap 200 mb global 500 mb

geometry units bohr
symmetry c1 
O   0.0000000000        0.0000000000        0.0000000000
O   0.0000000000       -2.0473224350       -1.2595211660
O   0.0000000000        2.0473224350       -1.2595211660
end

basis spherical
 * library cc-pvdz
end

scf
thresh 1.0e-10
tol2e 1.0e-10
singlet
rhf
end

tce
eaccsd
nroots 2
freeze atomic
tilesize 20
thresh 1.0d-6
end

task tce energy
</code></pre>
<ul>
<li>IP-EOMCCSD calculations for the F2 molecule</li>
</ul>
<pre><code>start  tce_ipccsd_f2
title &quot;tce_ipccsd_f2&quot;
echo

memory stack 1000 mb heap 200 mb global 500 mb

geometry units angstroms
symmetry c1
 F    0.0000000000        0.0000000000   0.7059650
 F    0.0000000000        0.0000000000  -0.7059650
end

basis spherical
 * library cc-pvdz
end

scf
thresh 1.0e-10
tol2e 1.0e-10
singlet
rhf
end

tce
ipccsd
nroots 1
freeze atomic
thresh 1.0e-7
end

task tce energy
</code></pre>
<p>As in the EOMCCSD input we can request any number of roots.</p>
<p>In analogy to the EOMCC calculations we can customize the number of
initial guesses by using <code>set tce:maxeorb</code> directive. For example for
system with the symmetry with the orbital energy structure shown
below  <em>ea ip</em>
one can use the energy window (in the sense of the absolute value of the
HF orbital energies) to pinpoint the initial guesses. If one is
interested in calculating one EA-EOMCCSD root of the a1 symmetry the</p>
<pre><code> set tce:maxeorb 0.1
</code></pre>
<p>should be used. This means that the number of starting vectors will be
equal to the number of the unoccupied a1 symmetry orbitals with the
corresponding orbital energies less than 0.1 (in our example there will
be only one such a vector corresponding to the unoccupied orbital energy
0.072). If one looks for two roots</p>
<pre><code> set tce:maxeorb 0.16
</code></pre>
<p>option should be used(there are two a1 unoccupied orbitals with energies
less than 0.16).</p>
<p>For the IP-EOMCCSD case the <code>set tce:maxeorb</code> option works in a similar
way. For example if one is looks for 1 IP-EOMCCSD root of a1 symmetry ,</p>
<pre><code>set tce:maxeorb 0.24 
</code></pre>
<p>directive should be used (there is only one occupied orbital of a1
symmetry with the absolute value of orbital energy less than 0.24),
etc.</p>
<h3 id="tce-thresh-the-convergence-threshold-of-iterative-solutions-of-amplitude-equations">THRESH: the convergence threshold of iterative solutions of amplitude equations<a class="headerlink" href="#tce-thresh-the-convergence-threshold-of-iterative-solutions-of-amplitude-equations" title="Permanent link">&para;</a></h3>
<p>This keyword specifies the convergence threshold of iterative solutions
of amplitude equations, and applies to all of the CI, CC, and MBPT
models. The threshold refers to the norm of residual, namely, the
deviation from the amplitude equations. The default value is 1e-6.</p>
<h3 id="tce-maxiter-the-maximum-number-of-iterations">MAXITER: the maximum number of iterations<a class="headerlink" href="#tce-maxiter-the-maximum-number-of-iterations" title="Permanent link">&para;</a></h3>
<p>It sets the maximum allowed number iterations for the iterative
solutions of amplitude equations. The default value is 100.</p>
<h3 id="tce-io-parallel-io-scheme">IO: parallel I/O scheme<a class="headerlink" href="#tce-io-parallel-io-scheme" title="Permanent link">&para;</a></h3>
<p>There are five parallel I/O schemes implemented for all the models,
which need to be wisely chosen for a particular problem and computer
architecture.</p>
<ul>
<li>fortran : Fortran77 direct access,</li>
<li>eaf : Exclusive Access File library,</li>
<li>ga : Fully incore, Global Array virtual file,</li>
<li>sf : Shared File library,</li>
<li>replicated : Semi-replicated file on distributed file system with
    EAF library.</li>
<li>dra : Distributed file on distributed file system with DRA library.</li>
<li>ga_eaf : Semi-replicated file on distributed file system with EAF
    library. GA is used to speedup the file reconciliation.</li>
</ul>
<p>The GA algorithm, which is default, stores all input (integrals and
excitation amplitudes), output (residuals), and intermediate tensors in
the shared memory area across all nodes by virtue of GA library. This
fully incore algorithm replaces disk I/O by inter-process
communications. This is a recommended algorithm whenever feasible. Note
that the memory management through runtime orbital range tiling
described above applies to local (unshared) memory of each node, which
may be separately allocated from the shared memory space for GA. So when
there is not enough shared memory space (either physically or due to
software limitations, in particular, shmmax setting), the GA algorithm
can crash due to an out-of-memory error. The replicated scheme is the
currently the only disk-based algorithm for a genuinely distributed file
system. This means that each node keeps an identical copy of input
tensors and it holds non-identical overlapping segments of intermediate
and output tensors in its local disk. Whenever data coherency is
required, a file reconcilation process will take place to make the
intermediate and output data identical throughout the nodes. This
algorithm, while requiring redundant data space on local disk, performs
reasonably efficiently in parallel. For sequential execution, this
reduces to the EAF scheme. For a global file system, the SF scheme is
recommended. This together with the Fortran77 direct access scheme does
not usually exhibit scalability unless shared files on the global file
system also share the same I/O buffer. For sequential executions, the
SF, EAF, and replicated schemes are interchangeable, while the Fortran77
scheme is appreciably slower.</p>
<p>Two new I/O algorithms dra and ga_eaf combines GA and DRA or EAF based
replicated algorithm. In the former, arrays that are not active (e.g.,
prior T amplitudes used in DIIS or EOM-CC trial vectors) in GA algorithm
will be moved to DRA. In the latter, the intermediates that are formed
by tensor contractions are initially stored in GA, thereby avoiding the
need to accumulate the fragments of the intermediate scattered in EAFs
in the original EAF algorithm. Once the intermediate is formed
completely, then it will be replicated as EAFs.</p>
<p>The spin-free 4-index transformation algorithms are exclusively
compatible with the GA I/O scheme, although out-of-core algorithms for
the 4-index transformation are accessible using the 2emet options. See
<a href="#tce-2emet-alternative-storage-of-two-electron-integrals">Alternative storage of two-electron
integrals</a> 
for details.</p>
<h3 id="tce-diis-the-convergence-acceleration">DIIS: the convergence acceleration<a class="headerlink" href="#tce-diis-the-convergence-acceleration" title="Permanent link">&para;</a></h3>
<p>It sets the number iterations in which a DIIS extrapolation is performed
to accelerate the convergence of excitation amplitudes. The default
value is 5, which means in every five iteration, one DIIS extrapolation
is performed (and in the rest of the iterations, Jacobi rotation is
used). When zero or negative value is specified, the DIIS is turned off.
It is not recommended to perform DIIS every iteration, whereas setting a
large value for this parameter necessitates a large memory (disk) space
to keep the excitation amplitudes of previous iterations. <br />
Another tool for convergence acceleration is the level shift option (<code>lshift</code> keyword) that allows 
to increase small orbital energy differences used in calculating the
up-dates for cluster amplitudes. Typical values for <code>lshift</code> oscillates
between 0.3 and 0.5 for CC calculations for ground states of
multi-configurational character. Otherwise, the value of <code>lshift</code> is by
default set equal to 0.</p>
<h3 id="tce-freeze-the-frozen-corevirtual-approximation">FREEZE: the frozen core/virtual approximation<a class="headerlink" href="#tce-freeze-the-frozen-corevirtual-approximation" title="Permanent link">&para;</a></h3>
<p>Some of the lowest-lying core orbitals and/or some of the highest-lying
virtual orbitals may be excluded in the calculations by this keyword
(this does not affect the ground state HF or DFT calculation). No
orbitals are frozen by default. To exclude the atom-like core regions
altogether, one may request</p>
<pre><code> FREEZE atomic
</code></pre>
<p>To specify the number of lowest-lying occupied orbitals be excluded, one
may use</p>
<pre><code> FREEZE 10
</code></pre>
<p>which causes 10 lowest-lying occupied orbitals excluded. This is
equivalent to writing</p>
<pre><code> FREEZE core 10
</code></pre>
<p>To freeze the highest virtual orbitals, use the virtual keyword. For
instance, to freeze the top 5 virtuals</p>
<pre><code> FREEZE virtual 5
</code></pre>
<h3 id="tce-nroots-the-number-of-excited-states">NROOTS: the number of excited states<a class="headerlink" href="#tce-nroots-the-number-of-excited-states" title="Permanent link">&para;</a></h3>
<p>One can specify the number of excited state roots to be determined. The
default value is 1. It is advised that the users request several more
roots than actually needed, since owing to the nature of the trial
vector algorithm, some low-lying roots can be missed when they do not
have sufficient overlap with the initial guess vectors.</p>
<h3 id="tce-target-and-targetsym-the-target-root-and-its-symmetry">TARGET and TARGETSYM: the target root and its symmetry<a class="headerlink" href="#tce-target-and-targetsym-the-target-root-and-its-symmetry" title="Permanent link">&para;</a></h3>
<p>At the moment, the first and second geometrical derivatives of
excitation energies that are needed in force, geometry, and frequency
calculations are obtained by numerical differentiation. These keywords
may be used to specify which excited state root is being used for the
geometrical derivative calculation. For instance, when <code>TARGET 3</code> and
<code>TARGETSYM a1g</code> are included in the input block, the total energy (ground
state energy plus excitation energy) of the third lowest excited state
root (excluding the ground state) transforming as the irreducible
representation a1g will be passed to the module which performs the
derivative calculations. The default values of these keywords are 1 and
none, respectively.</p>
<p>The keyword <code>TARGETSYM</code> is essential in excited state geometry
optimization, since it is very common that the order of excited states
changes due to the geometry changes in the course of optimization.
Without specifying the <code>TARGETSYM</code>, the optimizer could (and would likely)
be optimizing the geometry of an excited state that is different from
the one the user had intended to optimize at the starting geometry. On
the other hand, in the frequency calculations, <code>TARGETSYM</code> must be <code>none</code>,
since the finite displacements given in the course of frequency
calculations will lift the spatial symmetry of the equilibrium geometry.
When these finite displacements can alter the order of excited states
including the target state, the frequency calculation is not be
feasible.</p>
<h3 id="tce-symmetry-restricting-the-excited-state-symmetry">SYMMETRY: restricting the excited state symmetry<a class="headerlink" href="#tce-symmetry-restricting-the-excited-state-symmetry" title="Permanent link">&para;</a></h3>
<p>By adding this keyword to the input block, the user can request the
module to seek just the roots of the specified irreducible
representation as <code>TARGETSYM</code>. By default, this option is not set.
<code>TARGETSYM</code> must be specified when <code>SYMMETRY</code> is
invoked.</p>
<h3 id="tce-eomsol-alternative-solver-for-the-right-eomccsd-eigenvalue-problem">EOMSOL: alternative solver for the right EOMCCSD eigenvalue problem<a class="headerlink" href="#tce-eomsol-alternative-solver-for-the-right-eomccsd-eigenvalue-problem" title="Permanent link">&para;</a></h3>
<p>The EOMSOL enables the user to switch between two algorithms for solving
EOMCCSD eigenproblem. When EOMSOL is set equal to 1 (<code>eomsol 1</code>
directive in the tce group) the old solver is invoked. The advantage of
this solver is a possibility of finding states of complicated
configurational structure, for example doubly excited states. However,
the dimension of the iterative space increases with each iteration and
in effect this algorithm requires large memory allocations especially
for large systems. In order to address this bottleneck, new algorithm
(<code>eomsol 2</code> directive in the tce group) was designed. In EOMSOL 2
algorithm all iterations are split into microcycles corresponding to
diis microiterations (the use of <code>diis</code> parameter is discussed earlier).
This algorithm enables the user to precisely estimate the memory usage
in the EOMCCSD calculations, which is equal to
diis*nroots*(size_x1+size_x2), where diis is the length of the DIIS
cycle, nroots is the number of sought roots, size_x1 corresponds to the
size of GA storing singly excited EOMCC almplitudes, and size_x2 is the
size of GA with doubly excited EOMCC amplitudes. Generally, larger
values of diis parameter lead to a faster convergence, however, this
happens at the expense of larger memory requirements. It is recommended
not to use in the EOMCCSD calculations with <code>eomsol 2</code> diis parameter
smaller than 5, which is its default value. The EOMSOL 2 algorithm uses
the CIS vectors as initial guesses, and for this reason is suited mainly
to track singly excited states. By default, the EOMSOL 1 option is
called in the EOMCCSD calculations. It should be also stressed that all
iterative EOMCC methods with higher than double excitations use EOMSOL 1
approach.</p>
<p>In some situations it is convenient to use separate convergence
threshold for the CCSD and EOMCCSD solvers. This can be achieved by
setting proper environmetal variables. In the following example</p>
<pre><code>geometry/basis set specifications
tce 
 thresh 1.0d-6
 ccsd
 nroots 2
end
set tce:thresheom 1.0d-4
task tce energy
</code></pre>
<p>the CCSD equations will be converged to the 1.0d-6 threshold while the
EOMCCSD ones to 1.0d-4. This option shoul dbe used with the <code>eomsol 2</code>
option. In some situations finding several (n) roots to the EOMCCSD
equations can be quite challenging. To by-pass this problem one can use
the &ldquo;n+1&rdquo; model, i.e., we request another root to be converged. Usually,
the presence the &ldquo;buffer&rdquo; root can imporve the iterative process for n
roots of interest. However, the buffer root does not have to be
converged to the same accuracy as n roots of interest. The follwing
example, shows how to handle this process (we chose n=2, n+1=3):</p>
<pre><code>geometry/basis set specifications
tce 
 freeze core
 ccsd
 nroots 3
 thresh 1.0d-6
end
set tce:thresheom 1.0d-4
set tce:threshl 1.0d-3
task tce energy
</code></pre>
<p>In this example the CCSD equations are solved with the 1.0d-6 threshold,
the first n (2) EOMCCSD roots are determined with the 10d-4 accuracy,
while the buffer root is determined with relax conv. criterion 1.0d-3.</p>
<h3 id="tce-2eorb-alternative-storage-of-two-electron-integrals">2EORB: alternative storage of two-electron integrals<a class="headerlink" href="#tce-2eorb-alternative-storage-of-two-electron-integrals" title="Permanent link">&para;</a></h3>
<p>In the 5.0 version a new option has been added in order to provide more
economical way of storing two-electron integrals used in CC calculations
based on the RHF and ROHF references. The <code>2EORB</code> keyword can be used for
all CC methods except for those using an active-space (CCSDt) up to NWChem version 
6.3. After that, further optimization restricted the use of <code>2EORB</code> to 
CCSD-based methods. Note that the four-index transformation is usually
an insignificant amount of the wall time for methods involving iterative
triples anyway. 
With 2EORB, all two-electron integrals are transformed and subsequently stored in a way
which is compatible with assumed tiling scheme. The transformation from
orbital to spinorbital form of the two-electron integrals is performed
on-the-fly during execution of the CC module. This option, although
slower, allows to significantly reduce the memory requirements needed by
the first half of 4-index transformation and final file with fully
transformed two-electron integrals. Savings in the memory requirements
on the order of magnitude (and more) have been observed for large-scale
open-shell calculations.</p>
<h3 id="tce-2emet-alternative-storage-of-two-electron-integrals">2EMET: alternative storage of two-electron integrals<a class="headerlink" href="#tce-2emet-alternative-storage-of-two-electron-integrals" title="Permanent link">&para;</a></h3>
<p>Several new computation-intensive algorithms has been added with the
purpose of improving scalability and overcoming local memory bottleneck
of the 5.0 <code>2EORB</code> 4-index transformation. In order to give the user a
full control over this part of the TCE code several keywords were
designed to define the most vital parameters that determine the
perfromance of 4-index transformation. All new keywords must be used
with the <code>2EORB</code> keyword, and thus will not work beyond CCSD methods after 
NWChem 6.3 (see explanation for <code>2EORB</code> above). The 2emet keyword (default 
value 1 or <code>2emet 1</code>, refers to the older 4-index transformation), defines the algorithm to be
used. By putting <code>2emet 2</code> the TCE code will execute the algoritm based on
the two step procedure with two intermediate files. In some instances
this algorithm is characterized by better timings compared to algorithms
3 and 4, although it is more memory demanding. In contrast to algorithms
nr 1,3, and 4 this approach can make use of a disk to store intermediate
files. For this purpose one should use the keyword <code>idiskx</code> (<code>idiskx 0</code>
causes that all intermediate files are stored on global arrays, while
<code>idiskx 1</code> tells the code to use a disk to store intermediates; default
value of <code>idiskx</code> is equal 0). Algorithm nr 3 (<code>2emet 3</code>) uses only one
intermediate file whereas algorithm nr 4 (<code>2emet 4</code>) is a version of
algorithm 3 with the option of reducing the memory requirements. For
example, by using the new keyword <code>split 4</code> we will reduce the size of only
intermediate file by factor of 4 (the <code>split</code> keyword can be only used in
the context of algorithm nr 4). All new algorithms (i.e. 2emet 2+) use
the attilesize setting to define the size of the atomic tile. By default
<code>attilesize</code> is set equal 30. For larger systems the use of larger values
of <code>attilesize</code> is recommended (typically between 40-60).</p>
<p>Additional algorithms are numbered 5, 6 and 9. Other values of <code>2emet</code> are
not supported and refer to methods which do not function properly.
Algorithms 5 and 6 were written as out-of-core N<sup>5</sup> methods (<code>idiskx1</code>) and are the most efficient algorithms at the present time. The
corresponding in-core variants (<code>idiskx 0</code>) are available but require
excessive memory with respect to the methods discussed above, although
they may be faster if sufficient memory is available (to get enough
memory often requires excessive nodes, which decreases performance in
the later stages of the calculation). The difference between 5 and 6 is
that 5 writes to a single file (SF or GA) while 6 uses multiple files.
For smaller calculations, particularly single-node jobs, 5 is faster
than 6, but for more than a handful of processors, algorithm 6 should be
used. The perforance discrepancy depends on the hardware used but in
algorithm eliminates simulataneous disk access on parallel file systems
or memory mutexes for the in-core case. For NFS filesystems attached to
parallel clusters, no performance differences have been observed, but
for Lustre and PVFS they are signficant. Using algorithm 5 for large
parallel file systems will make the file system inaccessible to other
users, invoking the wrath of system administrators.</p>
<p>Algorithm 9 is an out-of-core solution to the memory bottleneck of the
2-e integrals. In this approach, the intermediates of the 4-index
transformation as well as the MO integrals are stored in an SF file. As
before, this requires a shared file system. Because algorithm 9 is based
upon algorithm 5, described above, it is not expected to scale. The
primary purpose of algorithm 9 is to make the performance of the NWChem
coupled-cluster codes competive with fast serial codes on workstations.
It succeeds in this purpose when corresponding functionality is
compared. A more scalable version of this algorithm is possible, but the
utility is limited since large parallel computers do not permit the wall
times necessary to use an out-of-core method, which is necessarily
slower than the in-core variant. An extensible solution to these issues
using complex heterogeneous I/O is in development. Restarting with
algorithm 9 is not supported and attempting to use this feature with the
present version may produce meaningless results.</p>
<p>New is the inclusion of multiple <code>2emet</code> options for the spin-orbital
transformations, which are the default when 2eorb is not set and are
mandatory for UHF and KS references. The are currently three algorithms
1, 2 and 3 available. The numbering scheme does not correspond in any
way to the numbering scheme for the 2eorb case, except that <code>2emet 1</code>
corresponds to the default algorithm present in previous releases, which
uses the user-defined I/O scheme. Algorithm 2 (<code>2emet 2</code>) writes an SF
file for the half-transformed integrals, which is at least an
order-of-magnitude larger than the fully-transformed integrals, but
stores the fully-transformed integrals in core. Thus, once the 4-index
transformation is complete, this algorithm will perform exactly as when
algorithm 1 is used. Unfortuntely, the spin-orbital 2-e
fully-transformed integrals are still quite large and an algorithm
corresponding to 2eorb/2emet=9 is available with <code>2emet 3</code>. Algorithm 3 is
also limited in its scalability, but it permits relatively large
UHF-based calculations using single workstations for patient users.</p>
<p>In cases where the user has access to both shared and local filesystems
for parallel calculations, the <code>permanent_dir</code> setting refers to the
location of SF files. The file system for <code>scratch_dir</code> will not be used
for any of the 4-index transformation algorithms which are compatible
with <code>io=ga</code>.</p>
<p>Algorithms 13 and 14 are the N<sup>5</sup> variants of algorithms 3 and 4.
They are the most efficient in core GA-based algorithms for the RHF and
ROHF reference functions. Again, two parameters are needed to define the
perfromance of these algorithms: tilesize and attilesize. By default
attilesize is set equal to 40. In all runs tilesize is required to be
less than attilesize (tilesize &lt; attilesize).</p>
<p>New 4-index transformation for RHF/ROHF references (<code>2emet 15</code>) is
available in NWChem 6.5. In contrast to algorithms 13 and 14
inter-processor communication is significantly reduced resulting in much
better performance.</p>
<p>In the later part of this manual several examples illustrate the use of
the newly introduced keywords.</p>
<p>An efficient loop-fused four-index transfromations for RHF and ROHF references can be
enabled by the sequence <code>2eorb/2emet 16</code>.</p>
<h3 id="tce-ccsdtcr-eomccsdt-calculations-with-large-tiles">CCSD(T)/CR-EOMCCSD(T) calculations with large tiles<a class="headerlink" href="#tce-ccsdtcr-eomccsdt-calculations-with-large-tiles" title="Permanent link">&para;</a></h3>
<p>In 6.5 version of NWChem we have enabled versions of the CCSD(T) and
CR-EOMCCSD(T) codes, which by-pass the local memory limitation of
previous implementations. For this purpose a sliced versions of the
CCSD(T)/CR-EOMCCSD(T) codes have been developed (see K. Kowalski, S.
Krishnamoorthy, R. Olson, V. Tipparaju, E. Apra, Supercomputing 2011,
Seattle). In order to enable these versions it is enough to add</p>
<pre><code>set tce:xmem 100
</code></pre>
<p>which defines maximum memory size (in MB) for the slice of 6-dimensional
tensors (in the current example 100 MB; for more details see QA tests
tce_ccsd_t_xmem and tce_cr_eomccsd_t_xmem).</p>
<h3 id="tce-dipole-the-ground-and-excited-state-dipole-moments">DIPOLE: the ground- and excited-state dipole moments<a class="headerlink" href="#tce-dipole-the-ground-and-excited-state-dipole-moments" title="Permanent link">&para;</a></h3>
<p>When this is set, the ground-state CC calculation will enter another
round of iterative step for the so-called &Lambda; equation to obtain
the one-particle density matrix and dipole moments. Likewise, for
excited-states (EOM-CC), the transition moments and dipole moments will
be computed when (and only when) this option is set. In the latter case,
EOM-CC left hand side solutions will be sought incurring approximately
three times the computational cost of excitation energies alone (note
that the EOM-CC effective Hamiltonian is not Hermitian and has distinct
left and right eigenvectors).</p>
<h3 id="tce-nofock-not-recompute-fock-matrix">(NO)FOCK: (not) recompute Fock matrix<a class="headerlink" href="#tce-nofock-not-recompute-fock-matrix" title="Permanent link">&para;</a></h3>
<p>The default is <code>FOCK</code> meaning that the Fock matrix will be reconstructed
(as opposed to using the orbital energies as the diagonal part of Fock).
This is essential in getting correct correlation energies with ROHF or
DFT reference wave functions. However, currently, this module cannot
reconstruct the Fock matrix when one-component relativistic effects are
operative. So when a user wishes to run TCE&rsquo;s correlation methods with
DK or other relativistic reference, <code>NOFOCK</code> must be set and orbital
energies must be used for the Fock matrix.</p>
<h3 id="tce-densmat">DENSMAT<a class="headerlink" href="#tce-densmat" title="Permanent link">&para;</a></h3>
<p>Generate Density Matrix that can be used in the DPLOT module as described in
the <a href="#dplot-example-of-ccsd-charge-density-plot-with-gaussian-cube-output">Example section</a>. </p>
<h3 id="tce-print-the-verbosity">PRINT: the verbosity<a class="headerlink" href="#tce-print-the-verbosity" title="Permanent link">&para;</a></h3>
<p>This keyword changes the level of output verbosity. One may also request
some particular items in the table
below.</p>
<center>

| Item                      | Print Level | Description                          |
| ------------------------- | ----------- | ------------------------------------ |
| "time"                    | vary        | CPU and wall times                   |
| "tile"                    | vary        | Orbital range tiling information     |
| "t1"                      | debug       | T<sub>1</sub> excitation amplitude dumping |
| "t2"                      | debug       | T<sub>2</sub> excitation amplitude dumping |
| "t3"                      | debug       | T<sub>3</sub> excitation amplitude dumping |
| "t4"                      | debug       | T<sub>4</sub>  excitation amplitude dumping |
| "general information"     | default     | General information                  |
| "correlation information" | default     | TCE information                      |
| "mbpt2"                   | debug       | Canonical HF MBPT2 test               |
| "get_block"              | debug       | I/O information                      |
| "put_block"              | debug       | I/O information                      |
| "add_block"              | debug       | I/O information                      |
| "files"                   | debug       | File information                     |
| "offset"                  | debug       | File offset information              |
| "ao1e"                    | debug       | AO one-electron integral evaluation  |
| "ao2e"                    | debug       | AO two-electron integral evaluation  |
| "mo1e"                    | debug       | One-electron integral transformation |
| "mo2e"                    | debug       | Two-electron integral transformation |

Printable items in the TCE modules and their default print levels

</center>

<h2 id="tce-sample-input">Sample input<a class="headerlink" href="#tce-sample-input" title="Permanent link">&para;</a></h2>
<p>The following is a sample input for a ROHF-UCCSD energy calculation of a
water radical cation.</p>
<pre><code>START h2o
TITLE &quot;ROHF-UCCSD/cc-pVTZ H2O&quot;
CHARGE 1
GEOMETRY
 O     0.00000000     0.00000000     0.12982363
 H     0.75933475     0.00000000    -0.46621158
 H    -0.75933475     0.00000000    -0.46621158
END
BASIS
 * library cc-pVTZ
END
SCF
 ROHF
 DOUBLET
 THRESH 1.0e-10
 TOL2E  1.0e-10
END
TCE
 CCSD
END
TASK TCE ENERGY
</code></pre>
<p>The same result can be obtained by the following input:</p>
<pre><code>START h2o
TITLE &quot;ROHF-UCCSD/cc-pVTZ H2O&quot;
CHARGE 1
GEOMETRY
 O     0.00000000     0.00000000     0.12982363
 H     0.75933475     0.00000000    -0.46621158
 H    -0.75933475     0.00000000    -0.46621158
END
BASIS
 * library cc-pVTZ
END
SCF
 ROHF
 DOUBLET
 THRESH 1.0e-10
 TOL2E  1.0e-10
END
TASK UCCSD ENERGY
</code></pre>
<p>EOMCCSD calculations with EOMSOL 2 algorithm. In these claculations the
diis value of 8 will be used both in the CCSD and EOMCCSD iterations.</p>
<pre><code>TITLE &quot;tce_eomccsd_eomsol2&quot;
ECHO
START tce_eomccsd_eomsol2
GEOMETRY UNITS ANGSTROM
N            .034130    -.986909     .000000
N          -1.173397     .981920     .000000
C          -1.218805    -.408164     .000000
C           -.007302    1.702153     .000000
C           1.196200    1.107045     .000000
C           1.289085    -.345905     .000000
O           2.310232    -.996874     .000000
O          -2.257041   -1.026495     .000000
H            .049329   -1.997961     .000000
H          -2.070598    1.437050     .000000
H           -.125651    2.776484     .000000
H           2.111671    1.674079     .000000
END
BASIS
 * library 6-31G
END
SCF
 THRESH 1.0e-10
 TOL2E 1.0e-10
 SINGLET
 RHF
END
TCE
 FREEZE ATOMIC
 CREOMSD(T)
 EOMSOL 2
 DIIS 8
 TILESIZE 15
 THRESH 1.0d-5
 2EORB
 2EMET 13
 NROOTS 1
END
TASK TCE ENERGY
</code></pre>
<p>EOM-CCSDT calculation for excitation energies, excited-state dipole, and
transition moments.</p>
<pre><code>START tce_h2o_eomcc
GEOMETRY UNITS BOHR
 H    1.474611052297904   0.000000000000000   0.863401706825835
 O    0.000000000000000   0.000000000000000  -0.215850436155089
 H   -1.474611052297904   0.000000000000000   0.863401706825835
END
BASIS
 * library sto-3g
END
SCF
 SINGLET
 RHF
END
TCE
 CCSDT
 DIPOLE
 FREEZE CORE ATOMIC
 NROOTS 1
END
TASK TCE ENERGY
</code></pre>
<p>Active-space CCSDt/EOMCCSDt calculations (version I) of several excited
states of the Be<sub>3</sub> molecule. Three highest-lying occupied &alpha; and &beta;
orbitals (active_oa and active_ob) and nine lowest-lying unoccupied &alpha; and &beta;
orbitals (active_va and active_vb) define the active space.</p>
<pre><code>START TCE_ACTIVE_CCSDT
ECHO
GEOMETRY UNITS ANGSTROM
SYMMETRY C2V
 BE  0.00  0.00   0.00
 BE  0.00  1.137090 -1.96949
end
BASIS spherical
 # --- DEFINE YOUR BASIS SET ---
END
SCF
 THRESH 1.0e-10
 TOL2E 1.0e-10
 SINGLET
 RHF
END
TCE
 FREEZE ATOMIC
 CCSDTA
 TILESIZE 15
 THRESH 1.0d-5
 ACTIVE_OA 3
 ACTIVE_OB 3
 ACTIVE_VA 9
 ACTIVE_VB 9
 T3A_LVL 1
 NROOTS  2
END 
TASK TCE ENERGY
</code></pre>
<p>Completely renormalized EOMCCSD(T) (CR-EOMCCSD(T)) calculations for the
ozone molecule as described by the POL1 basis set. The CREOMSD(T)
directive automatically initialize three-step procedure: (1) CCSD
calculations; (2) EOMCCSD calculations; (3) non-iterative CR-EOMCCSD(T)
corrections.</p>
<pre><code>START TCE_CR_EOM_T_OZONE
ECHO
GEOMETRY UNITS BOHR
SYMMETRY C2V
 O   0.0000000000        0.0000000000        0.0000000000
 O   0.0000000000       -2.0473224350       -1.2595211660
 O   0.0000000000        2.0473224350       -1.2595211660
END
BASIS SPHERICAL
O    S
    10662.285000000      0.00079900
     1599.709700000      0.00615300
      364.725260000      0.03115700
      103.651790000      0.11559600
       33.905805000      0.30155200
O    S
       12.287469000      0.44487000
        4.756805000      0.24317200
O    S
        1.004271000      1.00000000
O    S
        0.300686000      1.00000000
O    S
        0.090030000      1.00000000
O    P
       34.856463000      0.01564800
        7.843131000      0.09819700
        2.306249000      0.30776800
        0.723164000      0.49247000
O    P
        0.214882000      1.00000000
O    P
        0.063850000      1.00000000
O    D
        2.306200000      0.20270000
        0.723200000      0.57910000
O    D
        0.214900000      0.78545000
        0.063900000      0.53387000
END
SCF
 THRESH 1.0e-10
 TOL2E 1.0e-10
 SINGLET
 RHF
END
TCE
 FREEZE ATOMIC
 CREOMSD(T)
 TILESIZE 20
 THRESH 1.0d-6
 NROOTS 2
END
TASK TCE ENERGY
</code></pre>
<p>The input for the active-space CR-EOMCCSD(T) calculations (the uracil
molecule in the 6-31G* basis set). In this example, the model space is
specified by defining the number of highest occupied orbitals (noact)
and the number of lowest unoccupied orbitals (nuact) that will be
considered as the active orbitals. In any type of the active-space
CR-EOMCCSD(T) calculatoins based on the RHF and ROHF references more
efficient versions of the orbital 4-index transformation can be invoked
(i.e., <code>2emet 13</code> or <code>2emet 14</code>).</p>
<pre><code>title &quot;uracil-6-31-Gs-act&quot;
echo
start uracil-6-31-Gs-act 
memory stack 1000 mb heap 100 mb global 1000 mb noverify
geometry units angstrom
  N             .034130    -.986909     .000000
  N          -1.173397     .981920     .000000
  C          -1.218805    -.408164     .000000
  C           -.007302    1.702153     .000000
  C           1.196200    1.107045     .000000
  C           1.289085    -.345905     .000000
  O           2.310232    -.996874     .000000
  O          -2.257041   -1.026495     .000000
  H            .049329   -1.997961     .000000
  H          -2.070598    1.437050     .000000
  H           -.125651    2.776484     .000000
  H           2.111671    1.674079     .000000
end
basis cartesian
  * library 6-31G*
end
scf
  thresh 1.0e-10
  tol2e 1.0e-10
  singlet
  rhf
end
tce
  freeze atomic
  creom(t)ac
  oact 21
  uact 99
  tilesize 15
  thresh 1.0d-5
  2eorb
  2emet 13
  nroots 1
  symmetry
  targetsym a'
end
task tce energy
</code></pre>
<p>The active-space in the active-space CR-EOMCCSD(T) calculations can be
alternatively specified by defining the energy <code>window</code>
[emin_act,emax_act]. All orbitals with orbital energies falling into
this widnow will considered as active (the active space in the following
example is different from the one used in the previous example).</p>
<pre><code>title &quot;uracil-6-31-Gs-act&quot;
echo
start uracil-6-31-Gs-act 
memory stack 1000 mb heap 100 mb global 1000 mb noverify
geometry units angstrom
  N             .034130    -.986909     .000000
  N          -1.173397     .981920     .000000
  C          -1.218805    -.408164     .000000
  C           -.007302    1.702153     .000000
  C           1.196200    1.107045     .000000
  C           1.289085    -.345905     .000000
  O           2.310232    -.996874     .000000
  O          -2.257041   -1.026495     .000000
  H            .049329   -1.997961     .000000
  H          -2.070598    1.437050     .000000
  H           -.125651    2.776484     .000000
  H           2.111671    1.674079     .000000
end
basis cartesian
  * library 6-31G*
end
scf
  thresh 1.0e-10
  tol2e 1.0e-10
  singlet
  rhf
end
tce
  freeze atomic
  creom(t)ac
  emin_act -0.5
  emax_act 1.0
  tilesize 15
  thresh 1.0d-5
  2eorbe
  2emet 13
  nroots 1
  symmetry
  targetsym a'
end
task tce energy 
</code></pre>
<p>The LR-CCSD(T) calculations for the glycine molecule in the aug-cc-pVTZ
basis set. Option 2EORB is used in order to minimize memory requirements
associated with the storage of two-electron integrals.</p>
<pre><code>START TCE_LR_CCSD_T
ECHO
GEOMETRY UNITS BOHR
 O      -2.8770919486        1.5073755650        0.3989960497
 C      -0.9993929716        0.2223265108       -0.0939400216
 C       1.6330980507        1.1263991128       -0.7236778647
 O      -1.3167079358       -2.3304840070       -0.1955378962
 N       3.5887721300       -0.1900460352        0.6355723246
 H       1.7384347574        3.1922914768       -0.2011420479
 H       1.8051078216        0.9725472539       -2.8503867814
 H       3.3674278149       -2.0653924379        0.5211399625
 H       5.2887327108        0.3011058554       -0.0285088728
 H      -3.0501350657       -2.7557071585        0.2342441831
END
BASIS
 * library aug-cc-pVTZ
END
SCF
 THRESH 1.0e-10
 TOL2E 1.0e-10
 SINGLET
 RHF
END
TCE
 FREEZE ATOMIC
 2EORB
 TILESIZE 15
 LR-CCSD(T)
 THRESH 1.0d-7
END
TASK TCE ENERGY
</code></pre>
<p>The CCSD calculations for the triplet state of the C<sub>20</sub> molecule.
New algorithms for 4-index tranformation are used.</p>
<pre><code>title &quot;c20_cage&quot;
echo
start c20_cage
memory stack 2320 mb heap 180 mb global 2000 mb noverify
geometry print xyz units bohr
  symmetry c2
  C      -0.761732  -1.112760   3.451966
  C       0.761732   1.112760   3.451966
  C       0.543308  -3.054565   2.168328
  C      -0.543308   3.054565   2.168328
  C       3.190553   0.632819   2.242986
  C      -3.190553  -0.632819   2.242986
  C       2.896910  -1.982251   1.260270
  C      -2.896910   1.982251   1.260270
  C      -0.951060  -3.770169   0.026589
  C       0.951060   3.770169   0.026589
  C       3.113776   2.128908   0.076756
  C      -3.113776  -2.128908   0.076756
  C       3.012003  -2.087494  -1.347695
  C      -3.012003   2.087494  -1.347695
  C       0.535910  -2.990532  -2.103427
  C      -0.535910   2.990532  -2.103427
  C       3.334106   0.574125  -2.322563
  C      -3.334106  -0.574125  -2.322563
  C      -0.764522  -1.081362  -3.453211
  C       0.764522   1.081362  -3.453211
end
basis spherical
 * library cc-pvtz
end
scf
  triplet
  rohf
  thresh 1.e-8
  maxiter 200
end
tce
  ccsd
  maxiter 60
  diis 5
  thresh 1.e-6
  2eorb
  2emet 3
  attilesize 40
  tilesize 30
  freeze atomic
end
task tce energy
</code></pre>
<h2 id="tce-tce-response-properties">TCE Response Properties<a class="headerlink" href="#tce-tce-response-properties" title="Permanent link">&para;</a></h2>
<h3 id="tce-introduction">Introduction<a class="headerlink" href="#tce-introduction" title="Permanent link">&para;</a></h3>
<p>Response properties can be calculated within the TCE. Ground-state
dipole polarizabilities can be performed at the CCSD, CCSDT and CCSDTQ
levels of theory. Neither CCSDT-LR nor CCSDTQ-LR are compiled by
default. Like the rest of the TCE, properties can be calculated with
RHF, UHF, ROHF and DFT reference wavefunctions.</p>
<p>Specific details for the implementation of CCSD-LR and CCSDT-LR can be
found in the following papers:</p>
<ul>
<li>J. R. Hammond, M. Valiev, W. A. deJong and K. Kowalski, J. Phys.
    Chem. A, 111, 5492 (2007).</li>
<li>J. R. Hammond, K. Kowalski and W. A. de Jong, J. Chem. Phys., 127,
    144105 (2007).</li>
<li>J. R. Hammond, W. A. de Jong and K. Kowalski , J. Chem. Phys., 128,
    224102 (2008).</li>
</ul>
<p>An appropriate background on coupled-cluster linear response (CC-LR) can
be found in the references of those papers.</p>
<h3 id="tce-performance">Performance<a class="headerlink" href="#tce-performance" title="Permanent link">&para;</a></h3>
<p>The coupled-cluster response codes were generated in the same manner as
the rest of the TCE, thus all previous comments on performance apply
here as well. The improved offsets available in the CCSD and EOM-CCSD
codes is now also available in the CCSD-&Lambda; and CCSD-LR codes. The
bottleneck for CCSD-LR is the same as EOM-CCSD, likewise for CCSDT-LR
and EOM-CCSDT. The CCSD-LR code has been tested on as many as 1024
processors for systems with more than 2000 spin-orbitals, while the
CCSDT-LR code has been run on as many as 1024 processors. The CCSDTQ-LR
code is not particularly useful due to the extreme memory requirements
of quadruples amplitudes, limited scalability and poor convergence in
the CCSDTQ equations in general.</p>
<h3 id="tce-input">Input<a class="headerlink" href="#tce-input" title="Permanent link">&para;</a></h3>
<p>The input commands for TCE response properties exclusively use set
directives (see <a href="#set">SET</a>) instead of TCE input
block keywords. There are currently only three commands available:</p>
<pre><code>set tce:lineresp &lt;logical lineresp default: F&gt;
set tce:afreq &lt;double precision afreq(9) default: 0.0&gt; 
set tce:respaxis &lt;logical respaxis(3) default: T T T&gt;
</code></pre>
<p>The boolean variable lineresp invokes the linear response equations for
the corresponding coupled-cluster method (only CCSD and CCSDT possess
this feature) and evaluates the dipole polarizability. When lineresp is
true, the &Lambda;-equations will also be solved, so the dipole moment is also
calculated. If no other options are set, the complete dipole
polarizability tensor will be calculated at zero frequency (static). Up
to nine real frequencies can be set; adding more should not crash the
code but it will calculate meaningless quantities. If one desires to
calculate more frequencies at one time, merely change the line <code>double
precision afreq(9)</code> in <code>$NWCHEM_TOP/src/tce/include/tce.fh</code>
appropriately and recompile.</p>
<p>The user can choose to calculate response amplitudes only for certain
axis, either because of redundancy due to symmetry or because of memory
limitations. The boolean vector of length three respaxis is used to
determine whether or not a given set of response amplitudes are
allocated, solved for, and used in the polarizability tensor evaluation.
The logical variables represent the X, Y, Z axes, respectively. If
respaxis is set to T F T, for example, the responses with respect to the
X and Z dipoles will be calculated, and the four (three unique) tensor
components will be evaluated. This feature is also useful for conserving
memory. By calculating only one axis at a time, memory requirements can
be reduced by 25% or more, depending on the number of DIIS vectors
used. Reducing the number of DIIS vectors also reduces the memory
requirements.</p>
<p>It is strongly advised that when calculating polarizabilities at
high-frequencies, that user set the frequencies in increasing order,
preferably starting with zero or other small value. This approach is
computationally efficient (the initial guess for subsequent responses is
the previously converged value) and mitigates starting from a zero
vector for the response amplitudes.</p>
<h3 id="tce-examples">Examples<a class="headerlink" href="#tce-examples" title="Permanent link">&para;</a></h3>
<p>This example runs in-core on a large workstation.</p>
<pre><code>geometry units angstrom
 symmetry d2h
 C               0.000    1.390    0.000
 H               0.000    2.470    0.000
 C               1.204    0.695    0.000
 H               2.139    1.235    0.000
 C               0.000   -1.390    0.000
 H               0.000   -2.470    0.000
 C              -1.204   -0.695    0.000
 H              -2.139   -1.235    0.000
 C               1.204   -0.695    0.000
 H               2.139   -1.235    0.000
 C              -1.204    0.695    0.000
 H              -2.139    1.235    0.000
end
basis spherical
 * library aug-cc-pvdz
end
tce
 freeze atomic
 ccsd
 io ga
 2eorb
 tilesize 16
end
set tce:lineresp T
set tce:afreq 0.000 0.072
set tce:respaxis T T T
task tce energy
</code></pre>
<p>This is a relatively simple example for CCSDT-LR.</p>
<pre><code>geometry units au
 symmetry c2v
 H 0       0        0
 F 0       0        1.7328795
end
basis spherical
 * library aug-cc-pvdz
end
tce
 ccsdt
 io ga
 2eorb
end
set tce:lineresp T
set tce:afreq 0.0 0.1 0.2 0.3 0.4
set tce:respaxis T F T
task tce energy
</code></pre>
<h2 id="tce-tce-restart-capability">TCE Restart Capability<a class="headerlink" href="#tce-tce-restart-capability" title="Permanent link">&para;</a></h2>
<h3 id="tce-overview_1">Overview<a class="headerlink" href="#tce-overview_1" title="Permanent link">&para;</a></h3>
<p>Check-pointing and restart are critical for computational chemistry
applications of any scale, but particularly those done on
supercomputers, or run for an extended period on workstations and
clusters. The TCE supports parallel check-pointing and restart using the
Shared Files (SF) library in the Global Arrays Tools. The SF library
requires that the file system be accessible by every node, so reading
and writing restart files can only be performed on a shared file system.
For workstations, this condition is trivially met. Restart files must be
persistent to be useful, so volatile file systems or those which are
periodicly erased should not be used for check-pointing.</p>
<p>Restart is possible for all ground-state amplitudes (<em>T</em>, &Lambda; and
T<sup>(1)</sup> but not for excited-state amplitudes, as in an EOM-CC
calculation. The latter functionality is under development.</p>
<p>Restart capability is useful in the following situations:</p>
<ul>
<li>The total run time is limited, as is the case for most
    supercomputing facilities.</li>
<li>The system is volatile and jobs die randomly.</li>
<li>When doing property calculations which require a large number of
    responses which cannot all be stored in-core simultaneously.</li>
</ul>
<p>At the present time, restarting the amplitudes during a potential energy
surface scan or numerical geometry optmization/frequency calculation is
not advised due to the phase issue in the molecular orbital
coefficients. If the phase changes, the amplitudes will no longer be a
useful guess and may lead to nonsense results. Expert users may be able
to use restart when the geometry varies using careful choices in the SCF
input by using the <code>rotate</code> and <code>lock</code> options but this use of restart
is not supported.</p>
<p>If SF encounters a failure during restart I/O, the job will fail. The
capability to ignore a subset of failures, such as when saving the
amplitudes prior to convergence, will be available in the future. This
is useful on some large machines when the filesystem is being taxed by
another job and may be appear unavailable at the moment a check-point
write is attempted.</p>
<p>The performance of SF I/O for restart is excellent and the wall time for
reading and writing integrals and amplitudes is negligible, even on a
supercomputer (such systems have very fast parallel file systems in most
cases). The only platform for which restart may cause I/O problems is
BlueGene, due to ratio of compute to I/O nodes (64 on BlueGene/P).</p>
<h3 id="tce-input_1">Input<a class="headerlink" href="#tce-input_1" title="Permanent link">&para;</a></h3>
<pre><code>set tce:read_integrals &lt;logical read_integrals default: F F F F F&gt;
set tce:read_t &lt;logical read_t default: F F F F&gt;
set tce:read_l &lt;logical read_l default: F F F F&gt;
set tce:read_tr &lt;logical read_tr default: F F F F&gt;
set tce:save_integrals &lt;logical save_integrals default: F F F F F&gt;
set tce:save_t &lt;logical save_t default: F F F F&gt;
set tce:save_l &lt;logical save_l default: F F F F&gt;
set tce:save_tr &lt;logical save_tr default: F F F F&gt;
set tce:save_interval &lt;integer save_interval default: 100000&gt;
</code></pre>
<p>The boolean variables read_integrals and save_integrals control which
integrals are read/saved. The first location is the 1-e integrals, the
second is for the 2-e integrals, and the third is for dipole integrals.
The fourth and fifth positions are reserved for quadrupole and octupole
integrals but this functionality is not available. The <code>read_t</code>, <code>read_l</code>,
<code>read_tr</code>, <code>save_t</code>, <code>save_l</code> and <code>save_tr</code> variables control the
reading/saving of the T, &Lambda; and T<sup>(1)</sup> (response) amplitudes.
Restart control on the response amplitudes is implicitly controlled by
the value of respaxis (see above). Requesting amplitudes that are beyond
the scope of a given calculation, such as T<sub>3</sub> in a CCSD calculation,
does not produce an error as these commands will never be processed.</p>
<p>Attempting to restart with a set of amplitudes without the corresponding
integrals is ill-advised, due to the phase issue discussed above. For
the same reason, one cannot save a subset of the integrals, so if it is
even remotely possible that the dipole moment or polarizabilities will
be desired for a given molecule, the dipole integrals should be saved as
well. It is possible to save the dipole integrals without setting dipole
in the TCE input block; setting save_integrals(3) true is sufficient
for this to occur.</p>
<p>The save_interval variable controls the frequency with which amplitudes
are saved. By default, the amplitudes are saved only when the iterative
process has converged, meaning that if the iterations do not converge in
less than the maximum, one must start the calculation again from
scratch. The solution is to set save_interval to a smaller value, such
as the number of DIIS cycles being used.</p>
<p>The user shall not change the tilesize when reading in saved amplitudes.
The results of this are catastrophic and under no circumstance will this
lead to physically meaningful results. Restart does not work for 2eorb
and <code>2emet 9</code>; no error will be produced but the results may be
meaningless.</p>
<h3 id="tce-examples_1">Examples<a class="headerlink" href="#tce-examples_1" title="Permanent link">&para;</a></h3>
<pre><code>geometry units au
 symmetry c2v
 H 0       0        0
 F 0       0        1.7328795
end
basis spherical
 * library aug-cc-pvdz
end
tce
 ccsdt
 io ga
end
set tce:lineresp T
set tce:afreq 0.0 0.1 0.2 0.3 0.4
set tce:respaxis T F T
task tce energy
</code></pre>
<h2 id="tce-maximizing-performance">Maximizing performance<a class="headerlink" href="#tce-maximizing-performance" title="Permanent link">&para;</a></h2>
<p>The following are recommended parameters for getting the best
performance and efficiency for common methods on various hardware
configurations. The optimal settings are far from extensible and it is
extremely important that users take care in how they apply these
recommendations. Testing a variety of settings on a simple example is
recommended when optimal settings are desired. Nonetheless, a few
guiding principles will improve the performance of TCE jobs markedly,
including making otherwise impossible jobs possible.</p>
<h3 id="tce-memory-considerations">Memory considerations<a class="headerlink" href="#tce-memory-considerations" title="Permanent link">&para;</a></h3>
<p>The default memory settings for NWChem are not optimal for TCE
calculations. When 2 GB of memory is available per process, the
following settings are close to optimal for CCSD jobs</p>
<pre><code>memory stack 800 mb heap 100 mb global 1000 mb
</code></pre>
<p>for property jobs, which require more amplitudes to be stored, it is
wise to favor the global allocation</p>
<pre><code>memory stack 500 mb heap 100 mb global 1300 mb
</code></pre>
<p>If you get an error for ga_create during the iterative steps, reduce
the number of DIIS vectors. If this error occurs during the four-index
transformation (after d_v2 filesize appears) you need more GA space, a
different 2emet, or more nodes.</p>
<p>The memory requirements for CCSD(T) are quite different because the
triples are generated in local memory. The value of tilesize should not
be larger than 30 in most cases and one should set something similar to
the following</p>
<pre><code>memory stack 1200 mb heap 100 mb global 600 mb
</code></pre>
<p>The local memory requires will be tilesize<sup>N</sup> where N=4 for
CCSD, N=6 for CCSD(T) and CCSDT, and N=8 for CCSDTQ. One should set
tilesize to 16 or less for CCSDT and CCSDTQ, although symmetry will
affect the local memory use significantly. The local memory usage of the
CR-EOMCCSD(T) approach has recently been significantly reduced to the
level of the CCSD(T) approach (2*tilesize<sup>6</sup>).</p>
<h3 id="tce-using-openmp-in-tce">Using OpenMP in TCE<a class="headerlink" href="#tce-using-openmp-in-tce" title="Permanent link">&para;</a></h3>
<p>TCE compute kernels are both floating-point and bandwidth intensive,
hence are amenable to multithreading. However, not all TCE kernels
support OpenMP, and therefore performance may be limited by Amdahl&rsquo;s
law. Furthermore, Global Arrays communication is not yet thread-safe and
must be called outside of threaded regions, potentially limiting
performance. However, even partial OpenMP is likely to improve
performance relative to idle cores, in the case where memory limitations
or other considerations (see below for the case of Xeon Phi
coprocessors) force the user to run NWChem on a subset of the available
CPU cores.</p>
<p>Currently, OpenMP threading is available in the following kernels:</p>
<ul>
<li>BLAS (assuming the chosen library supports this).</li>
<li>CCSD(T) and MRCCSD(T) triples kernels.</li>
</ul>
<p>The development version of NWChem (post-6.6) supports OpenMP more
kernels, including:</p>
<ul>
<li>4- and 6-dimensional tensor transposes.</li>
<li>Loop-driven CCSD tensor contractions.</li>
</ul>
<p>In most cases, NWChem runs best on CPU-only systems without OpenMP
threads. However, modest OpenMP has been found to improve performance of
CCSD(T) jobs. We expect the benefits of OpenMP to be more visible with
time, as NWChem achieves more complete coverage with OpenMP and as
platforms evolve to have more and more cores per node.</p>
<h3 id="tce-how-to-run-large-ccsdeomccsd-calculations">How to run large CCSD/EOMCCSD calculations<a class="headerlink" href="#tce-how-to-run-large-ccsdeomccsd-calculations" title="Permanent link">&para;</a></h3>
<p>When running large CCSD or EOMCCSD calculations for large systems
(number of orbitals larger than 400) and using large number of cores it
is recommended to switch to workflow based implementation of
CCSD/EOMCCSD methods.</p>
<p>The original CCSD/EOMCCSD TCE implementations are aggregates of a large
number of subroutines, which calculate either recursive intermediates or
contributions to residual vector. The dimensionalities of the tensors
involved in a given subroutine greatly impact the memory, computation,
and communication characteristics of each subroutine, which can lead to
pronounced problems with load balancing. For example, for the most
computationally intensive part of the CCSD/EOMCCSD approaches associated
with the inclusion of 4-particle integrals, the corresponding task pool
(the number of tasks in a subroutine) can easily be 2 orders of
magnitude larger than the task pool for subroutines calculating one-body
intermediates. To address this problem and improve the scalability of
the CCSD/EOMCCSD implementations, we exploited the dependencies exposed
between the task pools into classes (C) characterized by a collective
task pool. This was done in such a way as to ensure sufficient
parallelism in each class while minimizing the total number of such
classes. This procedure enabled us to reduce the number of
synchronization steps from nearly 80, in the EOMCCSD case, down to 4.
Optimized versions of the CCSD/EOMCCSD codes are enabled once the</p>
<pre><code>set tce:nts T
</code></pre>
<p>directive is used in the input file. Compared to the original
CCSD/EOMCCSD implementations the new approaches requires more global
memory. The new CCSD/EOMCCSD implementations provides significant
improvements in the parallel performance and average time per iteration.</p>
<p>References:</p>
<p>H.S. Hu, K. Bhaskaran-Nair, E. Apra, N. Govind, K. Kowalski, J. Phys.
Chem. A 118, 9087 (2014).</p>
<p>K. Kowalski, S. Krishnamoorthy, R.M. Olson, V. Tipparaju, E. Apra, K.
Kowalski, High Performance Computing, Networking, Storage and Analysis
(SC), 2011 International Conference 1 (2011).</p>
<h3 id="tce-scf-options">SCF options<a class="headerlink" href="#tce-scf-options" title="Permanent link">&para;</a></h3>
<p>For parallel jobs on clusters with poor disk performance on the
filesystem used for scratch_dir, it is a good idea to disable disk IO
during the SCF stage of the calculation. This is done by adding
semidirect memsize N filesize 0, where N is 80% of the stack memory
divided by 8, as the value in this directive is the number of dwords,
rather than bytes. With these settings, if the aggregate memory is
sufficient to store the integrals, the SCF performance will be
excellent, and it will be better than if direct is set in the SCF input
block. If scratch_dir is set to a local disk, then one should use as
much disk as is permissible, controlled by the value of filesize. On
many high-performance computers, filling up the local scratch disk will
crash the node, so one cannot be careless with these settings. In
addition, on many such machines, the shared file system performance is
better than that of the local disk (this is true for many NERSC
systems).</p>
<h3 id="tce-convergence-criteria">Convergence criteria<a class="headerlink" href="#tce-convergence-criteria" title="Permanent link">&para;</a></h3>
<p>It makes no sense to converge a calculation to a precision not relevant
to experiment. However, the relationship between convergence criteria
and calculated quantities is not fully known for some properties. For
example, the effect of the convergence criteria on the polarizability is
significant in some cases. In the case of CN, convergence of
10<sup>-11</sup> is necessary to resolve the polarizability tensor
components to 10<sup>-2</sup>. However, for many systems 10<sup>-7</sup>
convergence is sufficient to get accurate results for all properties. It
is important to calibrate the effect of convergence on property
calculations, particularly for open-shell and post-CCSD methods, on a
modest basis set before relaxing the convergence criteria too much.</p>
<h3 id="tce-io-schemes-and-integral-transformation-algorithms">IO schemes and integral transformation algorithms<a class="headerlink" href="#tce-io-schemes-and-integral-transformation-algorithms" title="Permanent link">&para;</a></h3>
<p>The effect on memory use of using the 2eorb keyword is huge. However,
this option can only be used with IO=GA and an RHF/ROHF reference. There
are a number of choices for the integral transformation algorithm when
using spin-free integrals. The fastest algorithm is 2EMET=5, but
significant disk IO is required for this algorithm. One must set
permanent_dir to a fast, shared file system for this algorithm to work.
If disk performance is not good, one should use either 2EMET=3 or
2EMET=4 depending on how much memory is available. If one sees a
ga_create error with 2EMET=3, then switch to algorithm 4 and add split
8 to the TCE input block.</p>
<h2 id="tce-using-coprocessor-architectures">Using coprocessor architectures<a class="headerlink" href="#tce-using-coprocessor-architectures" title="Permanent link">&para;</a></h2>
<h3 id="tce-ccsdt-and-mrccsdt-implementations-for-intel-mic-architectures">CCSD(T) and MRCCSD(T) implementations for Intel MIC architectures<a class="headerlink" href="#tce-ccsdt-and-mrccsdt-implementations-for-intel-mic-architectures" title="Permanent link">&para;</a></h3>
<p><span style="color:red;">This option is no longer available from version 7.0.0</span></p>
<p>NWChem 6.5 and 6.6 offer the possibility of using Intel Xeon Phi
hardware to perform the most computationally intensive part of the
CCSD(T) and MRCCSD(T) (only in NWChem 6.6) calculations (non-iterative
triples corrections). The form of input is the same as used in standard
TCE CCSD(T) and MRCCSD(T) runs. To enable new implementations please
follow compilation directives described below.</p>
<p>Required for compilation: Intel Composer XE version 14.0.3 (or later
versions)</p>
<p>Environmental variables required for compilation:</p>
<pre><code> % setenv USE_OPENMP 1

 % setenv USE_OFFLOAD 1
</code></pre>
<p>When using MKL and Intel Composer XE version 14 (or later versions),
please use the following
settings</p>
<pre><code> % setenv BLASOPT   &quot;-mkl -openmp   -lpthread -lm&quot;
 % setenv SCALAPACK &quot;-mkl -openmp -lmkl_scalapack_ilp64 -lmkl_blacs_intelmpi_ilp64 -lpthread -lm&quot;
</code></pre>
<p>The command require for compilation is</p>
<pre><code> make FC=ifort 
</code></pre>
<p>From our experience using the CCSD(T) and MRCCSD(T) TCE modules, we have
determined that the optimal configuration is to use a single Global
Arrays ranks for offloading work to each Xeon Phi card.</p>
<p>On the EMSL cascade system, each node is equipped with two coprocessors,
and NWChem can allocate one GA ranks per coprocessor. In the job
scripts, we recommend spawning just 6 GA ranks for each node, instead of
16 (number that would match the number of physical cores). Therefore, 2
out 6 GA ranks assigned to a particular compute node will offload to the
coprocessors, while the remaining 6 cores while be used for traditional
CPU processing duties. Since during offload the host core is idle, we
can double the number of OpenMP threads for the host
(<code>OMP_NUM_THREADS=4</code>) in order to fill the idle core with work from
another GA rank (4 process with 4 threads each will total 16 threads on
each node).</p>
<p>NWChem itself automatically detects the available coprocessors in the
system and properly partitions them for optimal use, therefore no action
is required other than specifying the number of processes on each node
(using the appropriate mpirun/mpiexec options) and setting the value of
<code>OMP_NUM_THREADS</code> as in the example above.</p>
<p>Environmental variables useful at run-time:</p>
<p><code>OMP_NUM_THREADS</code> is needed for the thread-level parallelization on the
Xeon CPU hosts</p>
<pre><code> % setenv OMP_NUM_THREADS 4
</code></pre>
<p>MIC_USE_2MB_BUFFER greatly improve communication between host and
Xeon Phi card</p>
<pre><code> % setenv MIC_USE_2MB_BUFFER 16K
</code></pre>
<h3 id="tce-ccsdt-method-with-cuda">CCSD(T) method with CUDA<a class="headerlink" href="#tce-ccsdt-method-with-cuda" title="Permanent link">&para;</a></h3>
<p>NWChem 6.3 offers a possibility of using GPU accelerators to perform the
most computationally intensive part of the CCSD(T) calculations
(non-iterative triples corrections). To enable this option one has to
enable compilation options described below and add the <code>cuda n</code>
directive to the tce block of input, where <code>n</code> refers to number of CUDA
devices per node.</p>
<p>geometry/basis set specifications</p>
<pre><code>tce 
 io ga
 freeze atomic
 thresh 1.0d-6
 tilesize 15
 ccsd(t)
 cuda 1
end
</code></pre>
<p>In the example above the number of CUDA devises is set equal to 1, which
means that user will use 1 GPU per node.</p>
<p>To enable the compilation of CUDA code one has to set the follwoing
variables before the compilation of
NWChem.</p>
<pre><code>export TCE_CUDA=Y 
export CUDA_LIBS=&quot;-L&lt;Your Path to cuda&gt;/lib64 -L&lt;Your Path to cuda&gt;/lib -lcudart&quot;
export CUDA_FLAGS=&quot;-arch  &lt;Your Cuda architecture&gt;&quot;  
export CUDA_INCLUDE=&quot;-I. -I&lt;Your Path to cuda&gt;/include&quot;
</code></pre>
<p>For
example:</p>
<pre><code>export TCE_CUDA=Y
export CUDA_LIBS=&quot;-L/usr/local/cuda-5.0/lib64 -L/usr/local/cuda-5.0/lib -lcudart&quot;
export CUDA_FLAGS=&quot;-arch sm_20 &quot;
export CUDA_INCLUDE=&quot;-I. -I/usr/local/cuda-5.0/include&quot;
</code></pre>
<p>In addition the code needs to be compiled with the following make
command</p>
<pre><code> make FC=&lt;fortran compiler&gt; CUDA=nvcc
</code></pre>
<p>Before running production style calculations we strongly suggest the
users to perform QA test from the /nwchem/QA/tests/tce_cuda directory.
A full example of a TCE CUDA input file is given below:</p>
<pre><code>start tce_cuda
echo
memory stack 1000 mb heap 100 mb global 500 mb verify
geometry units bohr
  O     0.00000000     0.00000000     0.22138519
  H     0.00000000    -1.43013023    -0.88554075
  H     0.00000000     1.43013023    -0.88554075
end
basis spherical
  H library cc-pVDZ
  O library cc-pVDZ
end
charge 0
scf
  thresh 1.0e-10
  tol2e 1.0e-10
  singlet
  rhf
end
tce
  ccsd(t)
  io ga
  cuda 1
  tilesize 18
end
task tce energy
</code></pre></section><section class="print-page" id="mp2" heading-number="11.7"><h1 id="mp2-mp2">MP2<a class="headerlink" href="#mp2-mp2" title="Permanent link">&para;</a></h1>
<h2 id="mp2-overview">Overview<a class="headerlink" href="#mp2-overview" title="Permanent link">&para;</a></h2>
<p>There are (at least) three algorithms within NWChem that compute the
Mller-Plesset (or many-body) perturbation theory second-order
correction<sup id="mp2-fnref_footnote_1"><a class="footnote-ref" href="#mp2-fn_footnote_1">1</a></sup> to the Hartree-Fock energy (MP2). They vary in
capability, the size of system that can be treated and use of other
approximations</p>
<ul>
<li>Semi-direct &ndash; this is recommended for most large applications (up
    to about 2800 basis functions), especially on 
    machines with significant disk I/O capability. Partially transformed
    integrals are stored on disk, multi-passing as necessary. RHF and
    UHF references may be treated including computation of analytic
    derivatives. This is selected by specifying mp2 on the task
    directive, e.g.</li>
</ul>
<pre><code>TASKMP2
</code></pre>
<ul>
<li>Fully-direct<sup id="fnref_footnote_2"><a class="footnote-ref" href="#mp2-fn_footnote_2">2</a></sup> &ndash; this is of utility if only limited I/O
    resources are available (up to about 2800 functions). Only RHF
    references and energies are available. This is selected by
    specifying <code>direct_mp2</code> on the task directive, e.g.</li>
</ul>
<pre><code>TASKDIRECT_MP2
</code></pre>
<ul>
<li>Resolution of the identity (RI) approximation MP2 (RI-MP2)<sup id="fnref_footnote_3"><a class="footnote-ref" href="#mp2-fn_footnote_3">3</a></sup> &ndash;
    this uses the RI approximation and is therefore only exact in the
    limit of a complete fitting basis. However, with some care, high
    accuracy may be obtained with relatively modest fitting basis sets.
    An RI-MP2 calculation can cost over 40 times less than the
    corresponding exact MP2 calculation. RHF and UHF references with
    only energies are available. This is selected by specifying rimp2 on
    the task directive, e.g.,</li>
</ul>
<pre><code>TASKRIMP2
</code></pre>
<p>All three MP2 tasks share the same input block.</p>
<pre><code> MP2  
   [FREEZE [[core] (atomic || &lt;integer nfzc default 0&gt;)] \  
           [virtual &lt;integer nfzv default 0&gt;]]  
   [TIGHT]  
   [PRINT]  
   [NOPRINT]  
   [VECTORS &lt;string filename default scf-output-vectors&gt; \  
     [swap [(alpha||beta)] &lt;integer pair-list&gt;] ]  
   [RIAPPROX &lt;string riapprox default V&gt;]  
   [FILE3C &lt;string filename default $file_prefix$.mo3cint&gt;  
   [SCRATCHDISK &lt;integer&gt;] 
 END
</code></pre>
<h2 id="mp2-freeze-freezing-orbitals">FREEZE: Freezing orbitals<a class="headerlink" href="#mp2-freeze-freezing-orbitals" title="Permanent link">&para;</a></h2>
<p>All MP2 modules support frozen core orbitals, however, only the direct
MP2 and RI-MP2 modules support frozen virtual orbitals.</p>
<p>By default, no orbitals are frozen. The atomic keyword causes orbitals
to be frozen according to the rules in the table below. Note that no
orbitals are frozen on atoms on which the nuclear charge has been
modified either by the user or due to the presence of an ECP. The actual
input would be</p>
<pre><code> freeze atomic
</code></pre>
<p>For example, in a calculation on Si(OH)<sub>2</sub>, by default the lowest
seven orbitals would be frozen (the oxygen 1<em>s</em>, and the silicon
1<em>s</em>, 2<em>s</em> and 2<em>p</em>).</p>
<h3 id="mp2-number-of-orbitals-considered-core-in-the-freeze-by-atoms-algorithm">Number of orbitals considered &ldquo;core&rdquo; in the &ldquo;freeze by atoms&rdquo; algorithm<a class="headerlink" href="#mp2-number-of-orbitals-considered-core-in-the-freeze-by-atoms-algorithm" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Period</th>
<th>Elements</th>
<th>Core Orbitals</th>
<th>Number of Core</th>
</tr>
</thead>
<tbody>
<tr>
<td>0</td>
<td>H - He</td>
<td>-</td>
<td>0</td>
</tr>
<tr>
<td>1</td>
<td>Li - Ne</td>
<td><em>1s</em></td>
<td>1</td>
</tr>
<tr>
<td>2</td>
<td>Na - Ar</td>
<td><em>1s2s2p</em></td>
<td>5</td>
</tr>
<tr>
<td>3</td>
<td>K - Kr</td>
<td><em>1s2s2p3s3p</em></td>
<td>9</td>
</tr>
<tr>
<td>4</td>
<td>Rb - Xe</td>
<td><em>1s2s2p3s3p4s3d4p</em></td>
<td>18</td>
</tr>
<tr>
<td>5</td>
<td>Cs - Rn</td>
<td><em>1s2s2p3s3p4s3d4p5s4d5p</em></td>
<td>27</td>
</tr>
<tr>
<td>6</td>
<td>Fr - Lr</td>
<td><em>1s2s2p3s3p4s3d4p5s4d5p6s4f5d6p</em></td>
<td>43</td>
</tr>
</tbody>
</table>
<p><strong>Caution:</strong> The rule for freezing orbitals &ldquo;by atoms&rdquo; are rather
unsophisticated since the number of orbitals to be frozen is computed from
the table above by summing the number of core orbitals in each atom
present. Therefore, the corresponding number of lowest-energy orbitals are frozen.
If for some reason the actual core orbitals are not the lowest lying,
then correct results will not be obtained. It is likely that
special attention should be paid to systems including third-
and higher- period atoms.</p>
<p>The user may also specify the number of orbitals to be frozen by atom.
Following the  Si(OH)<sub>2</sub> example, the user could specify</p>
<pre><code>freeze atomic O 1 Si 3
</code></pre>
<p>In this case only the lowest four orbitals would be frozen. If the user
does not specify the orbitals by atom, the rules default  to values reported in the <a href="#mp2-number-of-orbitals-considered-core-in-the-freeze-by-atoms-algorithm">Table above</a>.</p>
<p>Caution: The system does not check for a valid number of orbitals per
atom. If the user specifies to freeze more orbitals then are available
for the atom, the system will not catch the error. The user must specify
a logical number of orbitals to be frozen for the atom.</p>
<p>The FREEZE directive may also be used to specify the number of core
orbitals to freeze. For instance, to freeze the first 10 orbitals</p>
<pre><code> freeze 10
</code></pre>
<p>or equivalently, using the optional keyword core</p>
<pre><code> freeze core 10
</code></pre>
<p>Again, note that if the 10 orbitals to be frozen do not correspond to
the first 10 orbitals, then the swap keyword of the VECTORS directive
must be used to order the input orbitals correctly (<a href="#mp2-vectors-mo-vectors">MO
vectors</a>).</p>
<p>To freeze the highest virtual orbitals, use the virtual keyword. For
instance, to freeze the top 5 virtuals</p>
<pre><code> freeze virtual 5
</code></pre>
<p>Again, note that this only works for the direct-MP2 and RI-MP2 energy
codes.</p>
<h2 id="mp2-tight-increased-precision">TIGHT: Increased precision<a class="headerlink" href="#mp2-tight-increased-precision" title="Permanent link">&para;</a></h2>
<p>The TIGHT directive can be used to increase the precision in the MP2
energy and gradients.</p>
<p>By default the MP2 gradient package should compute energies accurate to
better than a micro-Hartree, and gradients accurate to about five
decimal places (atomic units). However, if there is significant linear
dependence in the basis set the precision might not be this good. Also,
for computing very accurate geometries or numerical frequencies, greater
precision may be desirable.</p>
<p>This option increases the precision to which both the SCF (from
10<sup>-6</sup> to 10<sup>-8</sup> and CPHF 10<sup>-4</sup> to $10<sup>-6</sup>
are solved, and also tightens thresholds for computation of the AO and
MO integrals (from 10<sup>-9</sup> to 10<sup>-11</sup> within the MP2 code.</p>
<h2 id="mp2-scratchdisk-limiting-io-usage">SCRATCHDISK: Limiting I/O usage<a class="headerlink" href="#mp2-scratchdisk-limiting-io-usage" title="Permanent link">&para;</a></h2>
<p>This directive - used only in the semi-direct algorithm - allows to
limit the per process disk usage. Mandatory argument for this keyword is
the maximum number of MBytes. For example, the following input line</p>
<pre><code> scratchdisk 512
</code></pre>
<p>puts an upper limit of 512 MBytes to the semi-direct MP2 usage of disk
(again, on a per process base).</p>
<h2 id="mp2-print-and-noprint">PRINT and NOPRINT<a class="headerlink" href="#mp2-print-and-noprint" title="Permanent link">&para;</a></h2>
<p>The standard print control options are recognized. The list of
recognized names are given in the table below.  </p>
<table>
<thead>
<tr>
<th>Item</th>
<th>Print Level</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>RI-MP2</strong></td>
<td></td>
<td></td>
</tr>
<tr>
<td>&ldquo;2/3 ints&rdquo;</td>
<td>debug</td>
<td>Partial 3-center integrals</td>
</tr>
<tr>
<td>&ldquo;3c ints&rdquo;</td>
<td>debug</td>
<td>MO 3-center integrals</td>
</tr>
<tr>
<td>&ldquo;4c ints b&rdquo;</td>
<td>debug</td>
<td>&ldquo;B&rdquo; matrix with approx. 4c integrals</td>
</tr>
<tr>
<td>&ldquo;4c ints&rdquo;</td>
<td>debug</td>
<td>Approximate 4-center integrals</td>
</tr>
<tr>
<td>&ldquo;amplitudes&rdquo;</td>
<td>debug</td>
<td>&ldquo;B&rdquo; matrix with denominators</td>
</tr>
<tr>
<td>&ldquo;basis&rdquo;</td>
<td>high</td>
<td></td>
</tr>
<tr>
<td>&ldquo;fit xf&rdquo;</td>
<td>debug</td>
<td>Transformation for fitting basis</td>
</tr>
<tr>
<td>&ldquo;geombas&rdquo;</td>
<td>debug</td>
<td>Detailed basis map info</td>
</tr>
<tr>
<td>&ldquo;geometry&rdquo;</td>
<td>high</td>
<td></td>
</tr>
<tr>
<td>&ldquo;information&rdquo;</td>
<td>low</td>
<td>General information about calc.</td>
</tr>
<tr>
<td>&ldquo;integral i/o&rdquo;</td>
<td>high</td>
<td>File size information</td>
</tr>
<tr>
<td>&ldquo;mo ints&rdquo;</td>
<td>debug</td>
<td></td>
</tr>
<tr>
<td>&ldquo;pair energies&rdquo;</td>
<td>debug</td>
<td>(working only in <code>direct_mp2</code>)</td>
</tr>
<tr>
<td>&ldquo;partial pair energies&rdquo;</td>
<td>debug</td>
<td>Pair energy matrix each time it is updated</td>
</tr>
<tr>
<td>&ldquo;progress reports&rdquo;</td>
<td>default</td>
<td>Report completion of time-consuming steps</td>
</tr>
<tr>
<td>&ldquo;reference&rdquo;</td>
<td>high</td>
<td>Details about reference wavefunction</td>
</tr>
<tr>
<td>&ldquo;warnings&rdquo;</td>
<td>low</td>
<td>Non-fatal warnings</td>
</tr>
</tbody>
</table>
<h2 id="mp2-vectors-mo-vectors">VECTORS: MO vectors<a class="headerlink" href="#mp2-vectors-mo-vectors" title="Permanent link">&para;</a></h2>
<p>All of the (supported) MP2 modules require use of converged canonical
SCF (RHF or UHF) orbitals for correct results. The vectors are by
default obtained from the preceding SCF calculation, but it is possible
to specify a different source using the VECTORS directive. For instance,
to obtain vectors from the file /tmp/h2o.movecs, use the directive</p>
<pre><code> vectors /tmp/h2o.movecs
</code></pre>
<p>As noted above (<a href="#mp2-freeze-freezing-orbitals">FREEZE</a>) if
the SCF orbitals are not in the correct order, it is necessary to
permute the input orbitals using the swap keyword of the VECTORS
directive. For instance, if it is desired to freeze a total six orbitals
corresponding to the SCF orbitals 1-5, and 7, it is necessary to swap
orbital 7 into the 6th position. This is accomplished by</p>
<pre><code> vectors swap 6 7
</code></pre>
<p>The swap capability is examined in more detail in <a href="#hartree-fock-theory-for-molecules-vectors-inputoutput-of-mo-vectors">Input/output of MO
vectors</a>.</p>
<h2 id="mp2-ri-mp2-fitting-basis">RI-MP2 fitting basis<a class="headerlink" href="#mp2-ri-mp2-fitting-basis" title="Permanent link">&para;</a></h2>
<p>The RI-MP2 method requires a fitting basis, which must be specified with
the name &ldquo;ri-mp2 basis&rdquo; (see <a href="#basis">Basis</a>). For instance,</p>
<pre><code> basis &quot;ri-mp2 basis&quot;
   O s; 10000.0 1
   O s;  1000.0 1
   O s;   100.0 1
   ...
 end
</code></pre>
<p>Alternatively, using a standard capability of basis sets
(<a href="#basis">Basis</a>) another named basis may be associated with
the fitting basis. For instance, the following input specifies a basis
with the name &ldquo;small fitting basis&rdquo; and then defines this to be the
&ldquo;ri-mp2 basis&rdquo;.</p>
<pre><code> basis &quot;small fitting basis&quot;
   H s; 10    1
   H s;  3    1
   H s;  1    1
   H s;  0.1  1
   H s;  0.01 1
 end
</code></pre>
<pre><code> set &quot;ri-mp2 basis&quot; &quot;small fitting basis&quot;
</code></pre>
<h2 id="mp2-file3c-ri-mp2-3-center-integral-filename">FILE3C: RI-MP2 3-center integral filename<a class="headerlink" href="#mp2-file3c-ri-mp2-3-center-integral-filename" title="Permanent link">&para;</a></h2>
<p>The default name for the file used to store the transformed 3-center
integrals is &ldquo;file_prefix.mo3cint&rdquo; in the scratch directory. This may
be overridden using the FILE3C directive. For instance, to specify the
file /scratch/h2o.3c, use this directive</p>
<pre><code> file3c /scratch/h2o.3c
</code></pre>
<h2 id="mp2-riapprox-ri-mp2-approximation">RIAPPROX: RI-MP2 Approximation<a class="headerlink" href="#mp2-riapprox-ri-mp2-approximation" title="Permanent link">&para;</a></h2>
<p>The type of RI approximation used in the RI-MP2 calculation is
controlled by means of the RIAPPROX directive. The two possible values
are V and SVS (case sensitive), which correspond to the approximations
with the same names described by Vahtras et al.<sup id="mp2-fnref_footnote_4"><a class="footnote-ref" href="#mp2-fn_footnote_4">4</a></sup>. The default is V.</p>
<h2 id="mp2-advanced-options-for-ri-mp2">Advanced options for RI-MP2<a class="headerlink" href="#mp2-advanced-options-for-ri-mp2" title="Permanent link">&para;</a></h2>
<p>These options, which functioned at the time of writing, are not
currently supported.</p>
<h3 id="mp2-control-of-linear-dependence">Control of linear dependence<a class="headerlink" href="#mp2-control-of-linear-dependence" title="Permanent link">&para;</a></h3>
<p>Construction of the RI fit requires the inversion of a matrix of fitting
basis integrals which is carried out via diagonalization. If the fitting
basis includes near linear dependencies, there will be small eigenvalues
which can ultimately lead to non-physical RI-MP2 correlation energies.
Eigenvectors of the fitting matrix are discarded if the corresponding
eigenvalue is less than <code>min eval</code> which defaults to 10<sup>-8</sup>. This
parameter may be changed by setting the a parameter in the database. For
instance, to set it to 10<sup>-10</sup></p>
<pre><code> set &quot;mp2:fit min eval&quot; 1e-10
</code></pre>
<h3 id="mp2-reference-spin-mapping-for-ri-mp2-calculations">Reference Spin Mapping for RI-MP2 Calculations<a class="headerlink" href="#mp2-reference-spin-mapping-for-ri-mp2-calculations" title="Permanent link">&para;</a></h3>
<p>The user has the option of specifying that the RI-MP2 calculations are
to be done with variations of the SCF reference wavefunction. This is
accomplished with a SET directive of the form,</p>
<pre><code> set &quot;mp2:reference spin mapping&quot; &lt;integer array default 0&gt;
</code></pre>
<p>Each element specified for array is the SCF spin case to be used for the
corresponding spin case of the correlated calculation. The number of
elements set determines the overall type of correlated calculation to be
performed. The default is to use the unadulterated SCF reference
wavefunction.</p>
<p>For example, to perform a spin-unrestricted calculation (two elements)
using the alpha spin orbitals (spin case 1) from the reference for both
of the correlated reference spin cases, the SET directive would be as
follows,</p>
<pre><code> set &quot;mp2:reference spin mapping&quot; 1 1
</code></pre>
<p>The SCF calculation to produce the reference wavefunction could be
either RHF or UHF in this case.</p>
<p>The SET directive for a similar case, but this time using the beta-spin
SCF orbitals for both correlated spin cases, is as follows,</p>
<pre><code> set &quot;mp2:reference spin mapping&quot; 2 2
</code></pre>
<p>The SCF reference calculation must be UHF in this case.</p>
<p>The SET directive for a spin-restricted calculation (one element) from
the beta-spin SCF orbitals using this option is as follows,</p>
<pre><code> set &quot;mp2:reference spin mapping&quot; 2
</code></pre>
<p>The SET directive for a spin-unrestricted calculation with the spins
flipped from the original SCF reference wavefunction is as follows,</p>
<pre><code> set &quot;mp2:reference spin mapping&quot; 2 1
</code></pre>
<h3 id="mp2-batch-sizes-for-the-ri-mp2-calculation">Batch Sizes for the RI-MP2 Calculation<a class="headerlink" href="#mp2-batch-sizes-for-the-ri-mp2-calculation" title="Permanent link">&para;</a></h3>
<p>The user can control the size of each batch in the transformation and
energy evaluation in the MP2 calculation, and consequently the memory
requirements and number of passes required. This is done using two SET
directives of the following form,</p>
<pre><code> set &quot;mp2:transformation batch size&quot; &lt;integer size default -1&gt;
 set &quot;mp2:energy batch size&quot; &lt;integer isize jsize default -1 -1&gt;
</code></pre>
<p>The default is for the code to determine the batch size based on the
available memory. Should there be problems with the program-determined
batch sizes, these variables allow the user to override them. The
program will always use the smaller of the user&rsquo;s value of these entries
and the internally computed batch size.</p>
<p>The transformation batch size computed in the code is the number of
occupied orbitals in the <em>(occ vir|fit)</em> three-center integrals to be
produced at a time. If this entry is less than the number of occupied
orbitals in the system, the transformation will require multiple passes
through the two-electron integrals. The memory requirements of this
stage are two global arrays of dimension <em><code>batch size</code> x vir x fit</em> with
the &ldquo;fit&rdquo; dimension distributed across all processors (on shell-block
boundaries). The compromise here is memory space versus multiple
integral evaluations.</p>
<p>The energy evaluation batch sizes are computed in the code from the
number of occupied orbitals in the two sets of three-center integrals to
be multiplied together to produce a matrix of approximate four-center
integrals. Two blocks of integrals of dimension <em>(<code>batch isize</code> x vir)</em>
and <em>(<code>batch jsize</code> x vir)</em> by fit are read in from disk and multiplied
together to produce <em><code>batch isize</code> <code>batch jsize</code> vir^2</em> approximate
integrals. The compromise here is performance of the distributed matrix
multiplication (which requires large matrices) versus memory space.</p>
<h3 id="mp2-energy-memory-allocation-mode-ri-mp2-calculation">Energy Memory Allocation Mode: RI-MP2 Calculation<a class="headerlink" href="#mp2-energy-memory-allocation-mode-ri-mp2-calculation" title="Permanent link">&para;</a></h3>
<p>The user must choose a strategy for the memory allocation in the energy
evaluation phase of the RI-MP2 calculation, either by minimizing the
amount of I/O, or minimizing the amount of computation. This can be
accomplished using a SET directive of the form,</p>
<pre><code> set &quot;mp2:energy mem minimize&quot; &lt;string mem_opt default I&gt;
</code></pre>
<p>A value of I entered for the string <code>mem_opt</code> means that a strategy to
minimize I/O will be employed. A value of C tells the code to use a
strategy that minimizes computation.</p>
<p>When the option to minimize I/O is selected, the block sizes are made as
large as possible so that the total number of passes through the
integral files is as small as possible. When the option to minimize
computation is selected, the blocks are chosen as close to square as
possible so that permutational symmetry in the energy evaluation can be
used most effectively.</p>
<h3 id="mp2-local-memory-usage-in-three-center-transformation">Local Memory Usage in Three-Center Transformation<a class="headerlink" href="#mp2-local-memory-usage-in-three-center-transformation" title="Permanent link">&para;</a></h3>
<p>For most applications, the code will be able to size the blocks without
help from the user. Therefore, it is unlikely that users will have any
reason to specify values for these entries except when doing very
particular performance measurements.</p>
<p>The size of <code>xf3ci:AO 1 batch size</code> is the most important of the three, in
terms of the effect on performance.</p>
<p>Local memory usage in the first two steps of the transformation is
controlled in the RI-MP2 calculation using the following SET directives,</p>
<pre><code> set &quot;xf3ci:AO 1 batch size&quot; &lt;integer max&gt;
 set &quot;xf3ci:AO 2 batch size&quot; &lt;integer max&gt;
 set &quot;xf3ci:fit batch size&quot; &lt;integer max&gt;
</code></pre>
<p>The size of the local arrays determines the sizes of the two matrix
multiplications. These entries set limits on the size of blocks to be
used in each index. The listing above is in order of importance of the
parameters to performance, with xf3ci:AO 1 batch size being most
important.</p>
<p>Note that these entries are only upper bounds and that the program will
size the blocks according to what it determines as the best usage of the
available local memory. The absolute maximum for a block size is the
number of functions in the AO basis, or the number of fitting basis
functions on a node. The absolute minimum value for block size is the
size of the largest shell in the appropriate basis. Batch size entries
specified for max that are larger than these limits are automatically
reset to an appropriate value.</p>
<h2 id="mp2-one-electron-properties-and-natural-orbitals">One-electron properties and natural orbitals<a class="headerlink" href="#mp2-one-electron-properties-and-natural-orbitals" title="Permanent link">&para;</a></h2>
<p>If an MP2 energy gradient is computed, all contributions are available
to form the MP2 linear-response density. This is the density that when
contracted with any spin-free, one-electron operator yields the
associated property defined as the derivative of the energy. Thus, the
reported MP2 dipole moment is the derivative of the energy w.r.t. an
external electric field and is not the expectation value of the operator
over the wavefunction. It has been shown that evaluating the MP2 density
through a derivative provides more accurate results, presumably because
this matches the way experiments probe the electron density more
closely[raghavachari1981]<sup id="mp2-fnref_footnote_5"><a class="footnote-ref" href="#mp2-fn_footnote_5">5</a></sup><sup id="fnref_footnote_6"><a class="footnote-ref" href="#mp2-fn_footnote_6">6</a></sup><sup id="fnref_footnote_7"><a class="footnote-ref" href="#mp2-fn_footnote_7">7</a></sup>.</p>
<p>Only dipole moments are printed by the MP2 gradient code, but natural
orbitals are produced and stored in the permanent directory with a file
extension of &ldquo;.mp2nos&rdquo;. These may be fed into the <a href="#properties">property
package</a> to compute more general properties as in the following example.</p>
<pre><code>start h2o
geometry
   O        2.15950        0.88132        0.00000
   H        3.12950        0.88132        0.00000
   H        1.83617        0.89369       -0.91444
end

basis spherical
  * library aug-cc-pVDZ
end

mp2
  freeze atomic
end

task mp2 gradient

property
  vectors  h2o.mp2nos
  mulliken
end

task mp2 property
</code></pre>
<p>Note that the MP2 linear response density matrix is not necessarily positive
definite so it is not unusual to see a few small negative natural
orbital occupation numbers. Significant negative occupation numbers have
been argued to be a sign that the system might be near degenerate<sup id="mp2-fnref_footnote_8"><a class="footnote-ref" href="#mp2-fn_footnote_8">8</a></sup>.</p>
<h2 id="mp2-scs-mp2-spin-component-scaled-mp2">SCS-MP2: Spin-Component Scaled MP2<a class="headerlink" href="#mp2-scs-mp2-spin-component-scaled-mp2" title="Permanent link">&para;</a></h2>
<p>Each MP2 output contains the calculation of the SCS-MP2 correlation
energies as suggested by S.Grimme<sup id="mp2-fnref_footnote_9"><a class="footnote-ref" href="#mp2-fn_footnote_9">9</a></sup></p>
<p>The SCS keyword is only required for gradients calculations:</p>
<pre><code> MP2  
   [SCS]  
 END
</code></pre>
<p>Scaling factors for the two components (parallel and opposite spin) can
be defined by using the keywords FSS (same spin factor) and FOS
(opposite spin factor):</p>
<pre><code>mp2  
   scs  
   fss   1.13  
   fos   0.56  
end
</code></pre>
<p>Default values are FSS=0.333333333, FOS=1.2 for MP2, and FSS=1.13, FOS=1.27 for
CCSD.</p>
<h2 id="mp2-references">References<a class="headerlink" href="#mp2-references" title="Permanent link">&para;</a></h2>
<div class="footnote">
<hr />
<ol>
<li id="mp2-fn_footnote_1">
<p>Mller, Chr.; Plesset, M. S. Note on an Approximation Treatment for Many-Electron Systems. <em>Physical Review</em> <strong>1934</strong>, <em>46</em> (7), 618622. <a href="https://doi.org/10.1103/PhysRev.46.618">https://doi.org/10.1103/PhysRev.46.618</a>.&#160;<a class="footnote-backref" href="#mp2-fnref_footnote_1" title="Jump back to footnote 1 in the text">&#8617;</a></p>
</li>
<li id="mp2-fn_footnote_2">
<p>Wong, A. T.; Harrison, R. J.; Rendell, A. P. Parallel Direct Four-Index Transformations. <em>Theoretica Chimica Acta</em> <strong>1996</strong>, <em>93</em> (6), 317331. <a href="https://doi.org/10.1007/BF01129213">https://doi.org/10.1007/BF01129213</a>.&#160;<a class="footnote-backref" href="#mp2-fnref_footnote_2" title="Jump back to footnote 2 in the text">&#8617;</a></p>
</li>
<li id="mp2-fn_footnote_3">
<p>Bernholdt, D. E.; Harrison, R. J. Large-Scale Correlated Electronic Structure Calculations: The RI-MP2 Method on Parallel Computers. <em>Chemical Physics Letters</em> <strong>1996</strong>, <em>250</em> (5-6), 477484. <a href="https://doi.org/10.1016/0009-2614(96)00054-1">https://doi.org/10.1016/0009-2614(96)00054-1</a>.&#160;<a class="footnote-backref" href="#mp2-fnref_footnote_3" title="Jump back to footnote 3 in the text">&#8617;</a></p>
</li>
<li id="mp2-fn_footnote_4">
<p>Vahtras, O.; Almlf, J.; Feyereisen, M. W. Integral Approximations for LCAO-SCF Calculations. <em>Chemical Physics Letters</em> <strong>1993</strong>, <em>213</em> (5-6), 514518. <a href="https://doi.org/10.1016/0009-2614(93)89151-7">https://doi.org/10.1016/0009-2614(93)89151-7</a>.&#160;<a class="footnote-backref" href="#mp2-fnref_footnote_4" title="Jump back to footnote 4 in the text">&#8617;</a></p>
</li>
<li id="mp2-fn_footnote_5">
<p>Diercksen, G. H. F.; Roos, B. O.; Sadlej, A. J. Legitimate Calculation of First-Order Molecular Properties in the Case of Limited CI Functions. Dipole Moments. <em>Chemical Physics</em> <strong>1981</strong>, <em>59</em> (1-2), 2939. <a href="https://doi.org/10.1016/0301-0104(81)80082-1">https://doi.org/10.1016/0301-0104(81)80082-1</a>.&#160;<a class="footnote-backref" href="#mp2-fnref_footnote_5" title="Jump back to footnote 5 in the text">&#8617;</a></p>
</li>
<li id="mp2-fn_footnote_6">
<p>Rice, J. E.; Amos, R. D. On the Efficient Evaluation of Analytic Energy Gradients. <em>Chemical Physics Letters</em> <strong>1985</strong>, <em>122</em> (6), 585590. <a href="https://doi.org/10.1016/0009-2614(85)87275-4">https://doi.org/10.1016/0009-2614(85)87275-4</a>.&#160;<a class="footnote-backref" href="#mp2-fnref_footnote_6" title="Jump back to footnote 6 in the text">&#8617;</a></p>
</li>
<li id="mp2-fn_footnote_7">
<p>Wiberg, K. B.; Hadad, C. M.; LePage, T. J.; Breneman, C. M.; Frisch, M. J. Analysis of the Effect of Electron Correlation on Charge Density Distributions. <em>The Journal of Physical Chemistry</em> <strong>1992</strong>, <em>96</em> (2), 671679. <a href="https://doi.org/10.1021/j100181a030">https://doi.org/10.1021/j100181a030</a>.&#160;<a class="footnote-backref" href="#mp2-fnref_footnote_7" title="Jump back to footnote 7 in the text">&#8617;</a></p>
</li>
<li id="mp2-fn_footnote_8">
<p>Gordon, M. S.; Schmidt, M. W.; Chaban, G. M.; Glaesemann, K. R.; Stevens, W. J.; Gonzalez, C. A Natural Orbital Diagnostic for Multiconfigurational Character in Correlated Wave Functions. <em>The Journal of Chemical Physics</em> <strong>1999</strong>, <em>110</em> (9), 41994207. <a href="https://doi.org/10.1063/1.478301">https://doi.org/10.1063/1.478301</a>.&#160;<a class="footnote-backref" href="#mp2-fnref_footnote_8" title="Jump back to footnote 8 in the text">&#8617;</a></p>
</li>
<li id="mp2-fn_footnote_9">
<p>Grimme, S. Improved Second-Order Mller-Plesset Perturbation Theory by Separate Scaling of Parallel- and Antiparallel-Spin Pair Correlation Energies. <em>The Journal of Chemical Physics</em> <strong>2003</strong>, <em>118</em> (20), 90959102. <a href="https://doi.org/10.1063/1.1569242">https://doi.org/10.1063/1.1569242</a>.&#160;<a class="footnote-backref" href="#mp2-fnref_footnote_9" title="Jump back to footnote 9 in the text">&#8617;</a></p>
</li>
</ol>
</div></section><section class="print-page" id="ccsd" heading-number="11.8"><h1 id="ccsd-coupled-cluster-calculations">Coupled Cluster Calculations<a class="headerlink" href="#ccsd-coupled-cluster-calculations" title="Permanent link">&para;</a></h1>
<h2 id="ccsd-overview">Overview<a class="headerlink" href="#ccsd-overview" title="Permanent link">&para;</a></h2>
<p>The NWChem coupled cluster energy module is primarily the work of
Alistair Rendell and Rika Kobayashi<sup id="ccsd-fnref_footnote_1"><a class="footnote-ref" href="#ccsd-fn_footnote_1">1</a></sup><sup id="fnref_footnote_2"><a class="footnote-ref" href="#ccsd-fn_footnote_2">2</a></sup>, with contributions from
Bert de Jong, David Bernholdt and Edoardo Apr<sup id="ccsd-fnref_footnote_3"><a class="footnote-ref" href="#ccsd-fn_footnote_3">3</a></sup>.</p>
<p>The coupled cluster code can perform calculations with full iterative
treatment of single and double excitations and non-iterative inclusion
of triple excitation effects. It is presently limited to closed-shell
(RHF) references.</p>
<p>Note that symmetry is not used within most of the CCSD(T) code. This can
have a profound impact on performance since the speed-up from symmetry
is roughly the square of the number of irreducible representations. In
the absence of symmetry, the performance of this code is competitive
with other programs.</p>
<p>The operation of the coupled cluster code is controlled by the input
block</p>
<pre><code>CCSD
  [MAXITER &lt;integer maxiter default 20&gt;]
  [THRESH  &lt;real thresh default 1e-6&gt;]
  [TOL2E &lt;real tol2e default min(1e-12 , 0.01**`thresh`*`)&gt;]
  [DIISBAS  &lt;integer diisbas default 5&gt;]
  [FREEZE [[core] (atomic || &lt;integer nfzc default 0&gt;)] \
          [virtual &lt;integer nfzv default 0&gt;]]
  [NODISK]
  [IPRT  &lt;integer IPRT default 0&gt;]
  [PRINT ...]
  [NOPRINT ...]
END
</code></pre>
<p>Note that the keyword CCSD is used for the input block regardless of the
actual level of theory desired (specified with the TASK directive). The
following directives are recognized within the CCSD group.</p>
<h2 id="ccsd-maxiter-maximum-number-of-iterations">MAXITER &ndash; Maximum number of iterations<a class="headerlink" href="#ccsd-maxiter-maximum-number-of-iterations" title="Permanent link">&para;</a></h2>
<p>The maximum number of iterations is set to 20 by default. This should be
quite enough for most calculations, although particularly troublesome
cases may require more.</p>
<pre><code> MAXITER  &lt;integer maxiter default 20&gt;
</code></pre>
<h2 id="ccsd-thresh-convergence-threshold">THRESH &ndash; Convergence threshold<a class="headerlink" href="#ccsd-thresh-convergence-threshold" title="Permanent link">&para;</a></h2>
<p>Controls the convergence threshold for the iterative part of the
calculation. Both the RMS error in the amplitudes and the change in
energy must be less than thresh.</p>
<pre><code> THRESH  &lt;real thresh default 1e-6&gt;
</code></pre>
<h2 id="ccsd-tol2e-integral-screening-threshold">TOL2E &ndash; integral screening threshold<a class="headerlink" href="#ccsd-tol2e-integral-screening-threshold" title="Permanent link">&para;</a></h2>
<pre><code> TOL2E &lt;real tol2e default min(1e-12, 0.01*thresh)&gt;
</code></pre>
<p>The variable tol2e is used in determining the integral screening
threshold for the evaluation of the energy and related quantities.</p>
<p>CAUTION! At the present time, the tol2e parameter only affects the
three- and four-virtual contributions, and the triples, all of which are
done &ldquo;on the fly&rdquo;. The transformations used for the other parts of the
code currently have a hard-wired threshold of 10<sup>-12</sup>. The default
for tol2e is set to match this, and since user input can only make the
threshold smaller, setting this parameter can only make calculations
take longer.</p>
<h2 id="ccsd-diisbas-diis-subspace-dimension">DIISBAS &ndash; DIIS subspace dimension<a class="headerlink" href="#ccsd-diisbas-diis-subspace-dimension" title="Permanent link">&para;</a></h2>
<p>Specifies the maximum size of the subspace used in DIIS convergence
acceleration. Note that DIIS requires the amplitudes and errors be
stored for each iteration in the subspace. Obviously this can
significantly increase memory requirements, and could force the user to
reduce DIISBAS for large calculations.</p>
<p>Measures to alleviate this problem, including more compact storage of
the quantities involved, and the possibility of disk storage are being
considered, but have not yet been implemented.</p>
<pre><code> DIISBAS  &lt;integer diisbas default 5&gt;
</code></pre>
<h2 id="ccsd-freeze-freezing-orbitals">FREEZE &ndash; Freezing orbitals<a class="headerlink" href="#ccsd-freeze-freezing-orbitals" title="Permanent link">&para;</a></h2>
<pre><code>  [FREEZE [[core] (atomic || &lt;integer nfzc default 0&gt;)] \
          [virtual &lt;integer nfzv default 0&gt;]]
</code></pre>
<p>This directive is identical to that used in the <a href="#mp2">MP2</a>
module.</p>
<h2 id="ccsd-nodisk-on-the-fly-computation-of-integrals">NODISK &ndash; On-the-fly computation of integrals<a class="headerlink" href="#ccsd-nodisk-on-the-fly-computation-of-integrals" title="Permanent link">&para;</a></h2>
<p>The CCSD modules by default computes once and stores on disk the
integrals. To avoid this kind of I/O operations, specify the keyword
<code>NODISK</code></p>
<h2 id="ccsd-iprt-debug-printing">IPRT &ndash; Debug printing<a class="headerlink" href="#ccsd-iprt-debug-printing" title="Permanent link">&para;</a></h2>
<p>This directive controls the level of output from the code, mostly to
facilitate debugging and the like. The larger the value, the more output
printed. From looking at the source code, the interesting values seem to
be IPRT &gt; 5, 10, and 50.</p>
<pre><code> IPRT  &lt;integer IPRT default 0&gt;
</code></pre>
<h2 id="ccsd-print-and-noprint">PRINT and NOPRINT<a class="headerlink" href="#ccsd-print-and-noprint" title="Permanent link">&para;</a></h2>
<p>The coupled cluster module supports the standard NWChem print control
keywords, although very little in the code is actually hooked into this
mechanism yet.  </p>
<table>
<thead>
<tr>
<th>Item</th>
<th>Print Level</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&ldquo;reference&rdquo;</td>
<td>high</td>
<td>Wavefunction information</td>
</tr>
<tr>
<td>&ldquo;guess pair energies&rdquo;</td>
<td>debug</td>
<td>MP2 pair energies</td>
</tr>
<tr>
<td>&ldquo;byproduct energies&rdquo;</td>
<td>default</td>
<td>Intermediate energies</td>
</tr>
<tr>
<td>&ldquo;term debugging switches&rdquo;</td>
<td>debug</td>
<td>Switches for individual terms</td>
</tr>
</tbody>
</table>
<h2 id="ccsd-methods-tasks-recognized">Methods (Tasks) Recognized<a class="headerlink" href="#ccsd-methods-tasks-recognized" title="Permanent link">&para;</a></h2>
<p>Currently available methods are</p>
<ul>
<li>CCSD - Full iterative inclusion of single and double excitations</li>
<li>CCSD+T(CCSD) - The fourth order triples contribution computed with converged
   singles and doubles amplitudes</li>
<li>CCSD(T) - The linearized triples approximation due to Raghavachari.</li>
</ul>
<p>The calculation is invoked using the TASK directive, so to perform a
CCSD+T(CCSD) calculation, for example, the input file should include the directive</p>
<pre><code> TASK CCSD+T(CCSD)
</code></pre>
<p>Lower-level results which come as by-products (such as MP3/MP4) of the
requested calculation are generally also printed in the output file and
stored on the run-time database, but the method specified in the TASK
directive is considered the primary result.</p>
<h2 id="ccsd-debugging-and-development-aids">Debugging and Development Aids<a class="headerlink" href="#ccsd-debugging-and-development-aids" title="Permanent link">&para;</a></h2>
<p>The information in this section is intended for use by experts (both
with the methodology and with the code), primarily for debugging and
development work. Messing with stuff in listed in this section will
probably make your calculation quantitatively wrong! 
Consider yourself warned!</p>
<h3 id="ccsd-switching-on-and-off-terms">Switching On and Off Terms<a class="headerlink" href="#ccsd-switching-on-and-off-terms" title="Permanent link">&para;</a></h3>
<p>The /DEBUG/ common block contains a number of arrays which control the
calculation of particular terms in the program. These are 15-element
integer arrays (although from the code only a few elements actually
effect anything) which can be set from the input deck. See the code for
details of how the arrays are interpreted.</p>
<p>Printing of this data at run-time is controlled by the &ldquo;term debugging
switches&rdquo; print option. The values are checked against the defaults at
run-time and a warning is printed to draw attention to the fact that the
calculation does not correspond precisely to the requested method.</p>
<pre><code>DOA  &lt;integer array default 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2&gt;
DOB  &lt;integer array default 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2&gt;
DOG  &lt;integer array default 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1&gt;
DOH  &lt;integer array default 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1&gt;
DOJK &lt;integer array default 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2&gt;
DOS  &lt;integer array default 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1&gt;
DOD  &lt;integer array default 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1&gt;
</code></pre>
<h2 id="ccsd-alternative-implementations-of-triples">Alternative Implementations of Triples<a class="headerlink" href="#ccsd-alternative-implementations-of-triples" title="Permanent link">&para;</a></h2>
<p>There are four customized versions of the CCSD(T) triples driver that
may improve performance on some architectures. These are not the default
implementation and are not tested regularly. The burden is on the user
to evaluate their correctness in comparison to the default triples
driver. The triples driver only affects how the (T) energy contribution
is evaluated; the CCSD code is the same in all cases.</p>
<p>All of the non-standard triples drivers are activated using RTDB set
directives, which are specified outside of the CCSD input block.</p>
<h3 id="ccsd-nonblocking">Nonblocking<a class="headerlink" href="#ccsd-nonblocking" title="Permanent link">&para;</a></h3>
<p>The nonblocking variant of the triples driver uses nonblocking Global
Arrays get operations. It may improve communication overlap at large
node code, provided that nonblocking communication makes asynchronous
progress.</p>
<pre><code> set ccsd:use_trpdrv_nb T
</code></pre>
<h3 id="ccsd-openmp">OpenMP<a class="headerlink" href="#ccsd-openmp" title="Permanent link">&para;</a></h3>
<p>As of November 2016, the development version of semidirect CCSD(T) uses
OpenMP extensively. The OpenMP variant of the triples driver includes
OpenMP threaded kernels and attempts to run multiple DGEMM calls
simultaneously. The CCSD iteration uses OpenMP threading in kernels with
a relatively small number of parallel regions. It also uses nonblocking
Global Arrays get operations.</p>
<pre><code>set ccsd:use_ccsd_omp T
set ccsd:use_trpdrv_omp T
</code></pre>
<p>If one runs with only the (T) portion of the code using threads, the
CCSD code will run slower when using fewer cores. Thus, it may be
prudent to run the CCSD portion with a larger number of processes and
then run a second job for (T) that restarts the computation on a smaller
number of processes and a larger number of threads.</p>
<p>Preliminary evaluation of this implementation indicates that a small
number of threads (2 to 4) is optimal, with the assumption that
single-threaded execution can utilize all of the cores. It is expected
that nodes with a large number of cores may not be able to support
process-only parallelism due to memory-capacity constraints, in which
case the OpenMP implementation allows the user to make use of more cores
than otherwise possible.</p>
<p>Because of the extensive refactoring of the code to maximize OpenMP
performance and the intrinsic non-associativity of floating-point
arithmetic, the OpenMP variant may not produce the exact same answer as
the default one. If there is concern about the numerical fidelity of
results, a more stringent numerical threshold for the CCSD equations may
be required.</p>
<h3 id="ccsd-offload">Offload<a class="headerlink" href="#ccsd-offload" title="Permanent link">&para;</a></h3>
<p>The offload variant of the triples driver supports Intel Xeon Phi
coprocessors (Knights Corner family), in addition to the aforementioned
OpenMP and nonblocking features. This implementation has not been tested
extensively and a recommendation concerning the right number of
processes and threads is not available.</p>
<pre><code> set ccsd:use_trpdrv_offload T
</code></pre>
<h2 id="ccsd-references">References<a class="headerlink" href="#ccsd-references" title="Permanent link">&para;</a></h2>
<div class="footnote">
<hr />
<ol>
<li id="ccsd-fn_footnote_1">
<p>Rendell, A.P., Lee, T.J., Komornicki, A., and Wilson, S. (1992)
&ldquo;Evaluation of the contribution from triply excited intermediates to
the fourth-order perturbation theory energy on Intel distributed
memory supercomputers&rdquo;, <em>Theor. Chem. Acc.</em>, <strong>84</strong>, 271-287, doi:
<a href="http://dx.doi.org/10.1007/BF01113267">10.1007/BF01113267</a>&#160;<a class="footnote-backref" href="#ccsd-fnref_footnote_1" title="Jump back to footnote 1 in the text">&#8617;</a></p>
</li>
<li id="ccsd-fn_footnote_2">
<p>Kobayashi, R. and Rendell, A.P. (1997) &ldquo;A direct coupled cluster
algorithm for massively parallel computers&rdquo;, <em>Chem. Phys. Lett.</em>,
<strong>265</strong>, 1-11, doi:
<a href="http://dx.doi.org/10.1016/S0009-2614\(96\)01387-5">10.1016/S0009-2614(96)01387-5</a>&#160;<a class="footnote-backref" href="#ccsd-fnref_footnote_2" title="Jump back to footnote 2 in the text">&#8617;</a></p>
</li>
<li id="ccsd-fn_footnote_3">
<p>Apr, E., Harrison, R.J., de Jong, W.A., Rendell, A.P., Tipparaju,
V. and Xantheas, S.S. (2009) &ldquo;Liquid Water: Obtaining the Right
Answer for the Right Reasons&rdquo;, <em>Proc. SC&lsquo;09</em>, doi:
<a href="http://dx.doi.org/10.1145/1654059.1654127">10.1145/1654059.1654127</a>&#160;<a class="footnote-backref" href="#ccsd-fnref_footnote_3" title="Jump back to footnote 3 in the text">&#8617;</a></p>
</li>
</ol>
</div></section><section class="print-page" id="multiconfiguration_scf" heading-number="11.9"><h1 id="multiconfiguration_scf-mcscf">MCSCF<a class="headerlink" href="#multiconfiguration_scf-mcscf" title="Permanent link">&para;</a></h1>
<h2 id="multiconfiguration_scf-overview">Overview<a class="headerlink" href="#multiconfiguration_scf-overview" title="Permanent link">&para;</a></h2>
<p>The NWChem multiconfiguration SCF (MCSCF) module can currently perform
complete active space SCF (CASSCF) calculations with at most 20 active
orbitals and about 500 basis
functions.</p>
<pre><code> MCSCF  
   STATE &lt;string state&gt;  
   ACTIVE &lt;integer nactive&gt;  
   ACTELEC &lt;integer nactelec&gt;  
   MULTIPLICITY &lt;integer multiplicity&gt;  
   [SYMMETRY &lt;integer symmetry default 1&gt;]  
   [VECTORS [[input] &lt;string input_file default file_prefix.movecs&gt;]   
          [swap &lt;integer vec1 vec2&gt; ...] \  
          [output &lt;string output_file default input_file&gt;] \  
          [lock]  
   [HESSIAN (exact||onel)]  
   [MAXITER &lt;integer maxiter default 20&gt;]  
   [THRESH  &lt;real thresh default 1.0e-4&gt;]  
   [TOL2E &lt;real tol2e default 1.0e-9&gt;] 
   [LEVEL &lt;real shift default 0.1d0&gt;]  
 END
</code></pre>
<p>Note that the <code>ACTIVE</code>, <code>ACTELEC</code>, and <code>MULTIPLICITY</code> directives are required.
The symmetry and multiplicity may alternatively be entered using the
STATE directive.</p>
<h2 id="multiconfiguration_scf-active-number-of-active-orbitals">ACTIVE: Number of active orbitals<a class="headerlink" href="#multiconfiguration_scf-active-number-of-active-orbitals" title="Permanent link">&para;</a></h2>
<p>The number of orbitals in the CASSCF active space must be specified
using the ACTIVE directive.</p>
<p>E.g.,</p>
<pre><code> active 10
</code></pre>
<p>The input molecular orbitals (see the vectors directive in <a href="#multiconfiguration_scf-vectors-inputoutput-of-mo-vectors">MCSCF
Vectors</a> and <a href="#hartree-fock-theory-for-molecules-vectors-inputoutput-of-mo-vectors">SCF
Vectors</a>)
must be arranged in order</p>
<ol>
<li>doubly occupied orbitals,</li>
<li>active orbitals, and</li>
<li>unoccupied orbitals.</li>
</ol>
<h2 id="multiconfiguration_scf-actelec-number-of-active-electrons">ACTELEC: Number of active electrons<a class="headerlink" href="#multiconfiguration_scf-actelec-number-of-active-electrons" title="Permanent link">&para;</a></h2>
<p>The number of electrons in the CASSCF active space must be specified
using the <code>ACTELEC</code> directive. An error is reported if the number of
active electrons and the multiplicity are inconsistent.</p>
<p>The number of closed shells is determined by subtracting the number of
active electrons from the total number of electrons (which in turn is
derived from the sum of the nuclear charges minus the total system
charge).</p>
<h2 id="multiconfiguration_scf-multiplicity">MULTIPLICITY<a class="headerlink" href="#multiconfiguration_scf-multiplicity" title="Permanent link">&para;</a></h2>
<p>The spin multiplicity must be specified and is enforced by projection of
the determinant wavefunction.</p>
<p>E.g., to obtain a triplet state</p>
<pre><code> multiplicity 3
</code></pre>
<h2 id="multiconfiguration_scf-symmetry-spatial-symmetry-of-the-wavefunction">SYMMETRY: Spatial symmetry of the wavefunction<a class="headerlink" href="#multiconfiguration_scf-symmetry-spatial-symmetry-of-the-wavefunction" title="Permanent link">&para;</a></h2>
<p>This species the irreducible representation of the wavefunction as an
integer in the range 1&ndash;8 using the same numbering of representations as
output by the SCF program. Note that only Abelian point groups are
supported.</p>
<p>E.g., to specify a B<sub>1</sub> state when using the C<sub>2v</sub> group</p>
<pre><code> symmetry 3
</code></pre>
<h2 id="multiconfiguration_scf-state-symmetry-and-multiplicity">STATE: Symmetry and multiplicity<a class="headerlink" href="#multiconfiguration_scf-state-symmetry-and-multiplicity" title="Permanent link">&para;</a></h2>
<p>The electronic state (spatial symmetry and multiplicity) may
alternatively be specified using the conventional notation for an
electronic state, such as <sup>3</sup>B<sub>2</sub> for a triplet state of B<sub>2</sub>
symmetry. This would be accomplished with the input</p>
<pre><code> state 3b2
</code></pre>
<p>which is equivalent to</p>
<pre><code> symmetry 4 
 multiplicity 3
</code></pre>
<h2 id="multiconfiguration_scf-vectors-inputoutput-of-mo-vectors">VECTORS: Input/output of MO vectors<a class="headerlink" href="#multiconfiguration_scf-vectors-inputoutput-of-mo-vectors" title="Permanent link">&para;</a></h2>
<p>Calculations are best started from RHF/ROHF molecular orbitals (see
<a href="#hartree-fock-theory-for-molecules">SCF</a>), and by
default vectors are taken from the previous MCSCF or SCF calculation. To
specify another input file use the <code>VECTORS</code> directive. Vectors are by
default output to the input file, and may be redirected using the output
keyword. The swap keyword of the
<a href="#hartree-fock-theory-for-molecules-vectors-inputoutput-of-mo-vectors">VECTORS</a>
directive may be used to reorder orbitals to obtain the correct active
space.</p>
<p>The
<a href="#hartree-fock-theory-for-molecules-vectors-inputoutput-of-mo-vectors">LOCK</a>
keyword allows the user to specify that the ordering of orbitals will be
locked to that of the initial vectors, insofar as possible. The default
is to order by ascending orbital energies within each orbital space. One
application where locking might be desirable is a calculation where it
is necessary to preserve the ordering of a previous geometry, despite
flipping of the orbital energies. For such a case, the <code>LOCK</code> directive
can be used to prevent the SCF calculation from changing the ordering,
even if the orbital energies change.</p>
<p>Output orbitals of a converged MCSCF calculation are canonicalized as
follows:</p>
<ul>
<li>Doubly occupied and unoccupied orbitals diagonalize the
    corresponding blocks of an effective Fock operator. Note that in the
    case of degenerate orbital energies this does not fully determine
    the orbtials.</li>
<li>Active-space orbitals are chosen as natural orbitals by
    diagonalization of the active space 1-particle density matrix. Note
    that in the case of degenerate occupations that this does not fully
    determine the orbitals.</li>
</ul>
<h2 id="multiconfiguration_scf-hessian-select-preconditioner">HESSIAN: Select preconditioner<a class="headerlink" href="#multiconfiguration_scf-hessian-select-preconditioner" title="Permanent link">&para;</a></h2>
<p>The MCSCF will use a one-electron approximation to the orbital-orbital
Hessian until some degree of convergence is obtained, whereupon it will
attempt to use the exact orbital-orbital Hessian which makes the micro
iterations more expensive but potentially reduces the total number of
macro iterations. Either choice may be forced throughout the calculation
by specifying the appropriate keyword on the <code>HESSIAN</code> directive.</p>
<p>E.g., to specify the one-electron approximation throughout</p>
<pre><code> hessian onel
</code></pre>
<h2 id="multiconfiguration_scf-level-level-shift-for-convergence">LEVEL: Level shift for convergence<a class="headerlink" href="#multiconfiguration_scf-level-level-shift-for-convergence" title="Permanent link">&para;</a></h2>
<p>The <a href="#hessians-and-vibrational-frequencies">Hessian</a>
used in the MCSCF optimization is by default level shifted by 0.1 until
the orbital gradient norm falls below 0.01, at which point the level
shift is reduced to zero. The initial value of 0.1 may be changed using
the LEVEL directive. Increasing the level shift may make convergence
more stable in some instances.</p>
<p>E.g., to set the initial level shift to 0.5</p>
<pre><code> level 0.5
</code></pre>
<h2 id="multiconfiguration_scf-print-and-noprint">PRINT and NOPRINT<a class="headerlink" href="#multiconfiguration_scf-print-and-noprint" title="Permanent link">&para;</a></h2>
<p>Specific output items can be selectively enabled or disabled using the
<a href="#print_noprint">print control mechanism</a> with the
available print options listed in the table below.  </p>
<table>
<thead>
<tr>
<th>Option</th>
<th>Class</th>
<th>Synopsis</th>
</tr>
</thead>
<tbody>
<tr>
<td>ci energy</td>
<td>default</td>
<td>CI energy eigenvalue</td>
</tr>
<tr>
<td>fock energy</td>
<td>default</td>
<td>Energy derived from Fock matrices</td>
</tr>
<tr>
<td>gradient norm</td>
<td>default</td>
<td>Gradient norm</td>
</tr>
<tr>
<td>movecs</td>
<td>default</td>
<td>Converged occupied MO vectors</td>
</tr>
<tr>
<td>trace energy</td>
<td>high</td>
<td>Trace Energy</td>
</tr>
<tr>
<td>converge info</td>
<td>high</td>
<td>Convergence data and monitoring</td>
</tr>
<tr>
<td>precondition</td>
<td>high</td>
<td>Orbital preconditioner iterations</td>
</tr>
<tr>
<td>microci</td>
<td>high</td>
<td>CI iterations in line search</td>
</tr>
<tr>
<td>canonical</td>
<td>high</td>
<td>Canonicalization information</td>
</tr>
<tr>
<td>new movecs</td>
<td>debug</td>
<td>MO vectors at each macro-iteration</td>
</tr>
<tr>
<td>ci guess</td>
<td>debug</td>
<td>Initial guess CI vector</td>
</tr>
<tr>
<td>density matrix</td>
<td>debug</td>
<td>One- and Two-particle density matrices</td>
</tr>
</tbody>
</table></section><section class="print-page" id="gw" heading-number="11.10"><h1 id="gw-gw">GW<a class="headerlink" href="#gw-gw" title="Permanent link">&para;</a></h1>
<h2 id="gw-overview">Overview<a class="headerlink" href="#gw-overview" title="Permanent link">&para;</a></h2>
<p>Electron attachment and detachment energies can be accurately described
by many-body perturbation theory (MBPT) methods. In particular, the 
<em>GW</em> approximation (GWA) to the self-energy is a MBPT method that has 
seen recent interest in its application to molecules due to a
promising cost/accuracy ratio.</p>
<p>The <em>GW</em> module implemented in NWChem takes a <a href="#density-functional-theory-for-molecules">DFT</a>
mean-field approximation to the Green&rsquo;s function, <em>G<sub>0</sub></em>, in order to
solve the quasiparticle equation at the one-shot <em>G<sub>0</sub>W<sub>0</sub></em> or
at various <em>levels</em> of the eigenvalue self-consistent <em>GW</em> approach (ev<em>GW</em>).
Since the mean-field orbitals are kept fixed in all these
approaches, the results depend on the actual starting
point <em>G<sub>0</sub></em> (hence, they depend on the exchange-correlation
functional chosen for the underlying DFT calculation).
For example, it has been known that a large fraction of exact exchange
is needed for the accurate prediction of core-level binding energies at the
one-shot <em>G<sub>0</sub>W<sub>0</sub></em> level.</p>
<p>For further theoretical insights and details about the actual implementation in
NWChem, please refer to the paper by Mejia-Rodriguez et al<sup id="gw-fnref_footnote_1"><a class="footnote-ref" href="#gw-fn_footnote_1">1</a></sup>.  </p>
<p><em>GW</em> input is provided using the compound directive</p>
<pre><code>GW
  ...
END
</code></pre>
<p>The actual <em>GW</em> calculation will be performed when the 
input module encounters the <a href="#task">TASK</a> directive.</p>
<pre><code>TASK DFT GW
</code></pre>
<p>Note that <code>DFT</code> must be specified as the underlying <em>QM</em> theory before 
<code>GW</code>. The <a href="#charge">charge</a>, <a href="#geometry">geometry</a>, and 
<a href="#density-functional-theory-for-molecules">DFT</a> options are all specified as
normal. </p>
<p>In addition to an atomic orbital <a href="#basis">basis set</a>, the <em>GW</em> module <strong>requires</strong> 
an <em>auxiliary basis set</em> to be provided in order to fit the four-center 
electron repulsion integrals. The auxliary basis set can have either the
 <code>cd basis</code> or <code>ri basis</code> names (see also <a href="#density-functional-theory-for-molecules-specification-of-basis-sets-for-the-dft-module">DFT</a>). Three combinations can be obtained:</p>
<ul>
<li>If a <code>ri basis</code> is given without a <code>cd basis</code>,
    the ground-state DFT will be performed without density fitting,
    and the <em>GW</em> task will use the <code>ri basis</code> to fit the integrals.</li>
<li>If a <code>cd basis</code> is given without a <code>ri basis</code>,
    <strong>both</strong> DFT and <em>GW</em> tasks will be performed using the 
    <code>cd basis</code> to fit the integrals.</li>
<li>If both <code>cd basis</code> and <code>ri basis</code> are present, the <code>cd basis</code> will be used
    for the DFT task, while the <code>ri basis</code> will be used
    for the <em>GW</em> task.</li>
</ul>
<h2 id="gw-gw-input-directive">GW Input directive<a class="headerlink" href="#gw-gw-input-directive" title="Permanent link">&para;</a></h2>
<p>There are sub-directives which allow for customized <em>GW</em> calculations.
The most general <em>GW</em> input block directive will look like:</p>
<pre><code>GW
  RPA 
  CORE
  EVGW [&lt;integer eviter default 4&gt;]
  EVGW0 [&lt;integer eviter default 4&gt;]
  FIRST &lt;integer first_orbital default 1&gt;
  METHOD [ [analytic] || [cdgw &lt;integer grid_points default 200&gt;] ]
  ETA &lt;real infinitesimal default 0.001&gt; 
  SOLVER [ [newton &lt;integer maxiter default 10&gt; ] || [graph] ]
  STATES [ [ alpha || beta ] [occ &lt;integer number default 1&gt;] [vir &lt;integer default 0&gt;] ]
  CONVERGENCE &lt;real threshold default 0.005&gt; [&lt;string units default ev&gt;]
END
</code></pre>
<p>The following sections describe these keywords.</p>
<h3 id="gw-rpa">RPA<a class="headerlink" href="#gw-rpa" title="Permanent link">&para;</a></h3>
<p>The keyword <code>RPA</code> triggers the computation of the RPA correlation energy.
This adds a little overhead to the <a href="#gw-method"><em>CD-GW</em></a> approach.</p>
<h3 id="gw-core-and-first">CORE and FIRST<a class="headerlink" href="#gw-core-and-first" title="Permanent link">&para;</a></h3>
<p>The <code>CORE</code> keyword forces to start counting the <a href="#gw-states"><code>STATES</code></a>
from the <code>FIRST</code> molecular orbital <strong>upwards</strong>. </p>
<p>The <code>FIRST</code> keyword has no meaning without <code>CORE</code> specified.</p>
<h3 id="gw-evgw-and-evgw0">EVGW and EVGW0<a class="headerlink" href="#gw-evgw-and-evgw0" title="Permanent link">&para;</a></h3>
<p>The <code>EVGW</code> keyword trigger the partial self-consistnet ev<em>GW</em> approach,
where both the Green&rsquo;s function <em>G</em> and the screened Coulomb <em>W</em> are updated
by using the quasiparticle energies from the previous step in their 
construction. </p>
<p>Similarly, the <code>EVGW0</code> triggers the ev<em>GW<sub>0</sub></em> approach, where
only the Green&rsquo;s function <em>G</em> is updated with the quasiparticle energies
of the previous iterations. <em>W<sub>0</sub></em> is kept fixed.</p>
<p>Both partial self-consistent cycles run for <code>eviter</code> number of cycles.</p>
<p>The use of <code>EVGW</code> or <code>EVGW0</code> will trigger the use of a scissor-shift
operator for all states not updated in the <em>evGW</em> cycle.</p>
<h3 id="gw-method">METHOD<a class="headerlink" href="#gw-method" title="Permanent link">&para;</a></h3>
<p>Two different techniques to obtain the diagonal self-energy
matrix elements are implemented in NWChem. </p>
<p>The <code>analytic</code> method builds and diagonalizes the full Casida RPA matrix
in order to obtain the screened Coulomb matrix elements. The Casida RPA
matrix grows very rapidly in size (<em>N<sub>occ</sub></em> &times; <em>N<sub>vir</sub></em>)
and ultimately yields a <em>N<sup>6</sup></em> scaling due to the diagonalization step.
It is therefore recommended to link the <a href="https://elpa.mpcdf.mpg.de/">ELPA</a>
and turn on its use by setting</p>
<pre><code>SET dft:scaleig e
</code></pre>
<p>The <code>cdgw</code> method uses the Contour-Deformation technique in order to
avoid the <em>N<sup>6</sup></em> diagonalization step. The diagonal self-energy
matrix elements &#x01A9;<sub><em>nn</em></sub> are obtained via a numerical
integration on the imaginary axis and the integrals over closed contours
on the first and third quadrants of the complex plane. The <code>grid_points</code>
value controls  the density of the modified Gauss-Legendre grid used
in the numerical integration over the imaginary axis.</p>
<p>Both <code>analytic</code> and <code>cdgw</code> methods are suitable for <em>core</em> <strong>and</strong>
<em>valence</em> calculations.</p>
<h3 id="gw-eta">ETA<a class="headerlink" href="#gw-eta" title="Permanent link">&para;</a></h3>
<p>The magnitude of the imaginary infinitesimal can be controlled using the
keyword <code>ETA</code>. The default value of <code>0.001</code> should work rather well
for <em>valence</em> calculations, but <code>CORE</code> calculations might need a
larger value, sometimes between <code>0.005</code> or even <code>0.01</code>.</p>
<h3 id="gw-solver">SOLVER<a class="headerlink" href="#gw-solver" title="Permanent link">&para;</a></h3>
<p>Two methods to solver the quasiparticle equations are implemented
in NWChem. </p>
<p>The <code>newton</code> method uses a modified Newton approach to find the fixed-point
of the quasiparticle equations. The Newton method tries to bracket the
solution and switches to a golden section method whenever the Newton step goes
beyond the bracketing values.</p>
<p>The <code>graph</code> method uses a frequency grid in order to bracket the solution
between two consecutive grid points. The number of grid points is controlled 
heuristically depending on the <a href="#gw-method"><code>METHOD</code></a>
and on the presence, or not, of nearby states in a cluster
of energy (see below).</p>
<p>Regardless of the solver, the energies of the states are always
classified in clusters with a maximum extension of <code>1.5 eV</code>. For
a given cluster of energies, the
<code>newton</code> method will start with the state closer to the Fermi level
and use its solution as guess for the rest of the states in the cluster.
The <code>graph</code> method will look for the solution of all the states
in a given cluster at once with a frequency grid with range large 
enough to encompass all the cluster &plusmn; <code>0.2 eV</code>.</p>
<h3 id="gw-states">STATES<a class="headerlink" href="#gw-states" title="Permanent link">&para;</a></h3>
<p>The keyword <code>STATES</code> controls for which particular state
the <em>GW</em> quasiparticle equations are to be solved. The
keyword might appear twice, one for the <em>alpha</em> spin channel
and one for the <em>beta</em> channel. The <em>beta</em> channel keyword is
meaningless for restricted closed-shell DFT calculations 
(<code>MULT 1</code> without <code>ODFT</code> in the <code>DFT</code> input block).</p>
<p>The number of occupied states will be counted starting from the 
state closest to the Fermi level (HOMO) unless the keyword
<a href="#gw-core-and-first"><code>CORE</code></a> is present. The virtual states will <strong>always</strong>
be counted from the state closest to the Fermi level upwards.</p>
<p>A <strong>-1</strong> following either <code>occ</code> or <code>vir</code> stands for all
states in the respective space.</p>
<h3 id="gw-convergence">CONVERGENCE<a class="headerlink" href="#gw-convergence" title="Permanent link">&para;</a></h3>
<p>The converegnce threshold of the quasiparticle equations can
be controlled with the keyword <code>CONVERGENCE</code> and might be
given either in <code>eV</code> or Hartree <code>au</code>.</p>
<h2 id="gw-sample-input-file">Sample Input File<a class="headerlink" href="#gw-sample-input-file" title="Permanent link">&para;</a></h2>
<ul>
<li>A <em>GW</em> calculation requesting the core-level binding 
energies of all 1<em>s</em> states (6 Fluorines and 6 Carbons)
using the <a href="#gw-method">CD-GW</a> method.</li>
</ul>
<pre><code>title &quot;CDGW C6F6 core&quot;
start
echo

memory 2000 mb

geometry
 C     -0.21589696     1.38358991     0.00000000
 C     -1.30618181     0.50480033     0.00000000
 C     -1.09023026    -0.87871037     0.00000000
 C      0.21590562    -1.38360671     0.00000000
 C      1.30610372    -0.50476737     0.00000000
 C      1.09020243     0.87883094     0.00000000
 F     -0.42025331     2.69273557     0.00000000
 F     -2.54211642     0.98238922     0.00000000
 F     -2.12174279    -1.71033945     0.00000000
 F      0.42026196    -2.69275237     0.00000000
 F      2.54203111    -0.98237286     0.00000000
 F      2.12188428     1.71024875     0.00000000
end

basis &quot;ao basis&quot; spherical
 * library cc-pvdz
end

basis &quot;cd basis&quot; spherical
 * library cc-pvdz-ri
end

dft
 xc xpbe96 0.55 hfexch 0.45 cpbe96 1.0
 direct
end

gw
 core
 eta 0.01
 method cdgw
 solver newton 15
 states alpha occ 12
end

task dft gw
</code></pre>
<ul>
<li>A <em>valence</em> <em>GW</em> calculation to obtain the vertical ionization
potential and the vertical electron affinity of the water
molecule using the <a href="#gw-method">analytic</a> method.</li>
</ul>
<pre><code>start

geometry
O     -0.000545       1.517541       0.000000
H      0.094538       0.553640       0.000000
H      0.901237       1.847958       0.000000
end

basis &quot;ao basis&quot; spherical
 h library def2-svp
 o library def2-svp
end

basis &quot;cd basis&quot; spherical
 h library def2-universal-jkfit
 o library def2-universal-jkfit
end

dft
 mult 1
 xc pbe96
 grid fine
 direct
end

gw
 states alpha occ 1 vir 1
end

task dft gw
</code></pre>
<ul>
<li>An ev<em>GW</em><sub>0</sub> calculation with 10 iterations
using the <a href="#gw-method">analytic</a> method.
All occupied energies, but only 10 virtual ones, are
updated using <em>GW</em>. The rest of the virtual states
are shifted using the so-called scissor operator.</li>
</ul>
<pre><code>start

geometry
O     -0.000545       1.517541       0.000000
H      0.094538       0.553640       0.000000
H      0.901237       1.847958       0.000000
end

basis &quot;ao basis&quot; spherical
 h library def2-svp
 o library def2-svp
end

basis &quot;cd basis&quot; spherical
 h library def2-universal-jkfit
 o library def2-universal-jkfit
end

dft
 mult 1
 xc pbe96
 grid fine
 direct
end

gw
 evgw0 10
 states alpha occ -1 vir 10
end

task dft gw
</code></pre>
<h2 id="gw-references">References<a class="headerlink" href="#gw-references" title="Permanent link">&para;</a></h2>
<div class="footnote">
<hr />
<ol>
<li id="gw-fn_footnote_1">
<p>Mejia-Rodriguez, D.; Kunitsa, A.; Apr, E.; Govind, N. Scalable Molecular GW Calculations: Valence and Core Spectra. <em>Journal of Chemical Theory and Computation</em> <strong>2021</strong>, <em>17</em> (12), 75047517. <a href="https://doi.org/10.1021/acs.jctc.1c00738">https://doi.org/10.1021/acs.jctc.1c00738</a>.&#160;<a class="footnote-backref" href="#gw-fnref_footnote_1" title="Jump back to footnote 1 in the text">&#8617;</a></p>
</li>
</ol>
</div></section><section class="print-page" id="xtb" heading-number="11.11"><h1 id="xtb-xtb">XTB<a class="headerlink" href="#xtb-xtb" title="Permanent link">&para;</a></h1>
<h2 id="xtb-overview">Overview<a class="headerlink" href="#xtb-overview" title="Permanent link">&para;</a></h2>
<p>XTB method <sup id="fnref_footnote_1"><a class="footnote-ref" href="#xtb-fn_footnote_1">1</a></sup>. Full documentation available at the <a href="https://xtb-docs.readthedocs.io">XTB website</a>.<br />
The NWChem implementation makes use of the <a href="https://tblite.readthedocs.io/">TBlite library</a>.  </p>
<pre><code> XTB
   [ACC &lt;real acc default 1.0&gt;] 
   [UHF &lt;integer uhf default 0&gt;]
   [METHOD gfn1 || gfn2 default gfn2]
   [GUESS]
   [VERBOSITY]
   [PRINT]  
   [NOPRINT]
   [NSPIN &lt;integer nspin default 1&gt;]
   [BROYDEN &lt;real broyden default 0.4&gt;]
 END
</code></pre>
<h2 id="xtb-acc-accuracy">ACC: Accuracy<a class="headerlink" href="#xtb-acc-accuracy" title="Permanent link">&para;</a></h2>
<p>For example, if the user wants to increase the accuracy:</p>
<pre><code> acc 1d-4
</code></pre>
<h2 id="xtb-uhf-number-of-unpaired-electrons">UHF: number of unpaired electrons<a class="headerlink" href="#xtb-uhf-number-of-unpaired-electrons" title="Permanent link">&para;</a></h2>
<pre><code> uhf 3
</code></pre>
<h2 id="xtb-nspin">NSPIN:<a class="headerlink" href="#xtb-nspin" title="Permanent link">&para;</a></h2>
<p>A value of <code>nspin</code> equal to two triggers an open-shell calculation (while the default value of 1 triggers a closed-shell calculation).</p>
<pre><code> nspin 2
</code></pre>
<h2 id="xtb-references">References<a class="headerlink" href="#xtb-references" title="Permanent link">&para;</a></h2>
<div class="footnote">
<hr />
<ol>
<li id="xtb-fn_footnote_1">
<p>Bannwarth, C.; Caldeweyher, E.; Ehlert, S.; Hansen, A.; Pracht, P.; Seibert, J.; Spicher, S.; Grimme, S. Extended Tight-Binding Quantum Chemistry Methods. <em>WIREs Computational Molecular Science</em> <strong>2020</strong>, <em>11</em> (2). <a href="https://doi.org/10.1002/wcms.1493">https://doi.org/10.1002/wcms.1493</a>.&#160;<a class="footnote-backref" href="#xtb-fnref_footnote_1" title="Jump back to footnote 1 in the text">&#8617;</a></p>
</li>
</ol>
</div></section></section>
                    <section class='print-page md-section' id='section-12' heading-number='12'>
                        <h1>Quantum Molecular Dynamics<a class='headerlink' href='#section-12' title='Permanent link'></a>
                        </h1>
                    <section class="print-page" id="plane-wave-density-functional-theory" heading-number="12.1"><h1 id="plane-wave-density-functional-theory-pseudopotential-plane-wave-density-functional-theory-nwpw">Pseudopotential plane-wave density functional theory (NWPW)<a class="headerlink" href="#plane-wave-density-functional-theory-pseudopotential-plane-wave-density-functional-theory-nwpw" title="Permanent link">&para;</a></h1>
<div class="toc">
<ul>
<li><a href="#plane-wave-density-functional-theory-pseudopotential-plane-wave-density-functional-theory-nwpw">Pseudopotential plane-wave density functional theory (NWPW)</a><ul>
<li><a href="#plane-wave-density-functional-theory-overview">Overview</a></li>
<li><a href="#plane-wave-density-functional-theory-pspw-tasks-gamma-point-calculations">PSPW Tasks: Gamma Point Calculations</a><ul>
<li><a href="#plane-wave-density-functional-theory-paw-potentials">PAW Potentials</a><ul>
<li><a href="#plane-wave-density-functional-theory-paw-implementation-notes">PAW Implementation Notes</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-exchange-correlation-potentials">Exchange-Correlation Potentials</a><ul>
<li><a href="#plane-wave-density-functional-theory-dft-u-corrections">DFT + U Corrections</a></li>
<li><a href="#plane-wave-density-functional-theory-langreth-style-vdw-and-vdw-van-der-wall-functionals">Langreth style vdw and vdw van der Wall functionals</a></li>
<li><a href="#plane-wave-density-functional-theory-grimme-dispersion-corrections">Grimme Dispersion Corrections</a></li>
<li><a href="#plane-wave-density-functional-theory-using-exchange-correlation-potentials-available-in-the-dft-module">Using Exchange-Correlation Potentials Available in the DFT Module</a></li>
<li><a href="#plane-wave-density-functional-theory-exact-exchange">Exact Exchange</a></li>
<li><a href="#plane-wave-density-functional-theory-self-interaction-corrections">Self-Interaction Corrections</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-wannier">Wannier</a></li>
<li><a href="#plane-wave-density-functional-theory-mulliken-analysis">Mulliken Analysis</a></li>
<li><a href="#plane-wave-density-functional-theory-density-of-states">Density of States</a></li>
<li><a href="#plane-wave-density-functional-theory-projected-density-of-states">Projected Density of States</a></li>
<li><a href="#plane-wave-density-functional-theory-point-charge-analysis">Point Charge Analysis</a></li>
<li><a href="#plane-wave-density-functional-theory-pspw_dplot-generate-gaussian-cube-files">PSPW_DPLOT: Generate Gaussian Cube Files</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-band-tasks-multiple-k-point-calculations">Band Tasks: Multiple k-point Calculations</a><ul>
<li><a href="#plane-wave-density-functional-theory-brillouin-zone">Brillouin Zone</a><ul>
<li><a href="#plane-wave-density-functional-theory-band-structure-paths">Band Structure Paths</a></li>
<li><a href="#plane-wave-density-functional-theory-special-points-of-different-space-groups-conventional-cells">Special Points of Different Space Groups (Conventional Cells)</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-screened-exchange">Screened Exchange</a></li>
<li><a href="#plane-wave-density-functional-theory-density-of-states-and-projected-density-of-states">Density of States and Projected Density of States</a></li>
<li><a href="#plane-wave-density-functional-theory-two-component-wavefunctions-spin-orbit-zora">Two-Component Wavefunctions (Spin-Orbit ZORA)</a></li>
<li><a href="#plane-wave-density-functional-theory-band_dplot-generate-gaussian-cube-files">BAND_DPLOT: Generate Gaussian Cube Files</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-car-parrinello">Car-Parrinello</a><ul>
<li><a href="#plane-wave-density-functional-theory-adding-geometry-constraints-to-a-car-parrinello-simulation">Adding Geometry Constraints to a Car-Parrinello Simulation</a></li>
<li><a href="#plane-wave-density-functional-theory-car-parrinello-output-datafiles">Car-Parrinello Output Datafiles</a><ul>
<li><a href="#plane-wave-density-functional-theory-xyz-motion-file">XYZ motion file</a></li>
<li><a href="#plane-wave-density-functional-theory-ion_motion-motion-file">ION_MOTION motion file</a></li>
<li><a href="#plane-wave-density-functional-theory-emotion-motion-file">EMOTION motion file</a></li>
<li><a href="#plane-wave-density-functional-theory-hmotion-motion-file">HMOTION motion file</a></li>
<li><a href="#plane-wave-density-functional-theory-eigmotion-motion-file">EIGMOTION motion file</a></li>
<li><a href="#plane-wave-density-functional-theory-omotion-motion-file">OMOTION motion file</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-born-oppenheimer-molecular-dynamics">Born-Oppenheimer Molecular Dynamics</a></li>
<li><a href="#plane-wave-density-functional-theory-i-pi-socket-communication">i-PI Socket Communication</a></li>
<li><a href="#plane-wave-density-functional-theory-metropolis-monte-carlo">Metropolis Monte-Carlo</a></li>
<li><a href="#plane-wave-density-functional-theory-free-energy-simulations">Free Energy Simulations</a><ul>
<li><a href="#plane-wave-density-functional-theory-metadynamics">MetaDynamics</a><ul>
<li><a href="#plane-wave-density-functional-theory-input">Input</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-tamd-temperature-accelerated-molecular-dynamics">TAMD - Temperature Accelerated Molecular Dynamics</a><ul>
<li><a href="#plane-wave-density-functional-theory-input_1">Input</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-collective-variables">Collective Variables</a><ul>
<li><a href="#plane-wave-density-functional-theory-bond-distance-collective-variable">Bond Distance Collective Variable</a></li>
<li><a href="#plane-wave-density-functional-theory-angle-collective-variable">Angle Collective Variable</a></li>
<li><a href="#plane-wave-density-functional-theory-coordination-collective-variable">Coordination Collective Variable</a></li>
<li><a href="#plane-wave-density-functional-theory-n-plane-collective-variable">N-Plane Collective Variable</a></li>
<li><a href="#plane-wave-density-functional-theory-user-defined-collective-variable">User defined Collective Variable</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-extended-x-ray-absorption-fine-structure-exafs-integration-with-feff6l">Extended X-Ray Absorption Fine Structure (EXAFS) - Integration with FEFF6L</a></li>
<li><a href="#plane-wave-density-functional-theory-frozen-phonon-calculations">Frozen Phonon Calculations</a></li>
<li><a href="#plane-wave-density-functional-theory-steepest-descent">Steepest Descent</a></li>
<li><a href="#plane-wave-density-functional-theory-simulation-cell">Simulation Cell</a></li>
<li><a href="#plane-wave-density-functional-theory-unit-cell-optimization">Unit Cell Optimization</a></li>
<li><a href="#plane-wave-density-functional-theory-smear-fractional-occupation-of-the-molecular-orbitals">SMEAR - Fractional Occupation of the Molecular Orbitals</a></li>
<li><a href="#plane-wave-density-functional-theory-spin-penalty-functions">Spin Penalty Functions</a></li>
<li><a href="#plane-wave-density-functional-theory-aimdmm-qmmm">AIMD/MM (QM/MM)</a></li>
<li><a href="#plane-wave-density-functional-theory-psp_generator">PSP_GENERATOR</a><ul>
<li><a href="#plane-wave-density-functional-theory-atomic_filling-block">ATOMIC_FILLING Block</a></li>
<li><a href="#plane-wave-density-functional-theory-cutoff">CUTOFF</a></li>
<li><a href="#plane-wave-density-functional-theory-semicore_radius">SEMICORE_RADIUS</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-paw-tasks-legacy-implementation">PAW Tasks: Legacy Implementation</a></li>
<li><a href="#plane-wave-density-functional-theory-pseudopotential-and-paw-basis-libraries">Pseudopotential and PAW basis Libraries</a></li>
<li><a href="#plane-wave-density-functional-theory-nwpw-rtdb-entries-and-miscellaneous-datafiles">NWPW RTDB Entries and Miscellaneous DataFiles</a><ul>
<li><a href="#plane-wave-density-functional-theory-ion-positions">Ion Positions</a></li>
<li><a href="#plane-wave-density-functional-theory-ion-velocities">Ion Velocities</a></li>
<li><a href="#plane-wave-density-functional-theory-wavefunction-datafile">Wavefunction Datafile</a></li>
<li><a href="#plane-wave-density-functional-theory-velocity-wavefunction-datafile">Velocity Wavefunction Datafile</a></li>
<li><a href="#plane-wave-density-functional-theory-formatted-pseudopotential-datafile">Formatted Pseudopotential Datafile</a></li>
<li><a href="#plane-wave-density-functional-theory-one-dimensional-pseudopotential-datafile">One-Dimensional Pseudopotential Datafile</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-car-parrinello-scheme-for-ab-initio-molecular-dynamics">Car-Parrinello Scheme for Ab Initio Molecular Dynamics</a><ul>
<li><a href="#plane-wave-density-functional-theory-verlet-algorithm-for-integration">Verlet Algorithm for Integration</a></li>
<li><a href="#plane-wave-density-functional-theory-constant-temperature-simulations-nose-hoover-thermostats">Constant Temperature Simulations: Nose-Hoover Thermostats</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-nwpw-tutorial-1-s2-dimer-examples-with-pspw">NWPW Tutorial 1: S2 dimer examples with PSPW</a><ul>
<li><a href="#plane-wave-density-functional-theory-total-energy-of-s2-dimer-with-lda-approximation">Total energy of S2 dimer with LDA approximation</a></li>
<li><a href="#plane-wave-density-functional-theory-structural-optimization-of-s2-dimer-with-lda-approximation">Structural optimization of S2 dimer with LDA approximation</a></li>
<li><a href="#plane-wave-density-functional-theory-frequency-calculation-of-s2-dimer-with-lda-approximation">Frequency calculation of S2 dimer with LDA approximation</a></li>
<li><a href="#plane-wave-density-functional-theory-ab-initio-molecular-dynamics-simulation-car-parrinello-of-s2-dimer-using-the-lda-approximation">Ab initio molecular dynamics simulation (Car-Parrinello) of S2 dimer using the LDA approximation</a></li>
<li><a href="#plane-wave-density-functional-theory-ab-initio-molecular-dynamics-simulation-born-oppenheimer-of-ssub2-dimer-using-the-lda-approximation">Ab initio molecular dynamics simulation (Born-Oppenheimer) of S&lt;sub2 dimer using the LDA approximation</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-nwpw-tutorial-2-using-pspw-car-parrinello-simulated-annealing-simulations-to-optimize-structures">NWPW Tutorial 2: Using PSPW Car-Parrinello Simulated Annealing Simulations to Optimize Structures</a><ul>
<li><a href="#plane-wave-density-functional-theory-simulated-annealing-using-constant-energy-simulation">Simulated Annealing Using Constant Energy Simulation</a></li>
<li><a href="#plane-wave-density-functional-theory-simulated-annealing-using-constant-temperature-simulation">Simulated Annealing Using Constant Temperature Simulation</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-nwpw-tutorial-3-using-isodesmic-reaction-energies-to-estimate-gas-phase-thermodynamics">NWPW Tutorial 3: using isodesmic reaction energies to estimate gas-phase thermodynamics</a></li>
<li><a href="#plane-wave-density-functional-theory-nwpw-tutorial-4-aimdmm-simulation-of-ccl_4-64-h_2o">NWPW Tutorial 4: AIMD/MM simulation of CCl\(_4\) + 64 H\(_2\)O</a></li>
<li><a href="#plane-wave-density-functional-theory-nwpw-tutorial-5-optimizing-the-unit-cell-and-geometry-of-diamond">NWPW Tutorial 5: Optimizing the Unit Cell and Geometry of Diamond</a><ul>
<li><a href="#plane-wave-density-functional-theory-optimizing-the-unit-cell-and-geometry-for-an-8-atom-supercell-of-diamond-with-pspw">Optimizing the Unit Cell and Geometry for an 8 Atom Supercell of Diamond with PSPW</a></li>
<li><a href="#plane-wave-density-functional-theory-optimizing-the-unit-cell-for-an-8-atom-supercell-of-diamond-with-band">Optimizing the Unit Cell for an 8 Atom Supercell of Diamond with BAND</a></li>
<li><a href="#plane-wave-density-functional-theory-using-band-to-optimize-the-unit-cell-for-a-2-atom-primitive-cell-of-diamond">Using BAND to Optimize the Unit Cell for a 2 Atom Primitive Cell of Diamond</a></li>
<li><a href="#plane-wave-density-functional-theory-using-band-to-calculate-the-band-structures-of-diamond">Using BAND to Calculate the Band Structures of Diamond</a></li>
<li><a href="#plane-wave-density-functional-theory-using-band-to-calculate-the-density-of-states-of-diamond">Using BAND to Calculate the Density of States of Diamond</a></li>
<li><a href="#plane-wave-density-functional-theory-calculate-the-phonon-spectrum-of-diamond">Calculate the Phonon Spectrum of Diamond</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-nwpw-tutorial-6-optimizing-the-unit-cell-of-nickel-with-fractional-occupation">NWPW Tutorial 6: optimizing the unit cell of nickel with fractional occupation</a></li>
<li><a href="#plane-wave-density-functional-theory-nwpw-tutorial-7-optimizing-the-unit-cells-with-symmetry-diamond-with-fd-3m-symmetry-and-brucite-with-p-3m1-symmetry">NWPW Tutorial 7: Optimizing the unit cells with symmetry: Diamond with Fd-3m symmetry and Brucite with P-3m1 symmetry</a></li>
<li><a href="#plane-wave-density-functional-theory-nwpw-tutorial-8-nvt-metropolis-monte-carlo-simulations">NWPW Tutorial 8: NVT Metropolis Monte-Carlo Simulations</a></li>
<li><a href="#plane-wave-density-functional-theory-nwpw-tutorial-9-npt-metropolis-monte-carlo-simulations">NWPW Tutorial 9: NPT Metropolis Monte-Carlo Simulations</a></li>
<li><a href="#plane-wave-density-functional-theory-nwpw-tutorial-9-free-energy-simulations">NWPW Tutorial 9: Free Energy Simulations</a></li>
<li><a href="#plane-wave-density-functional-theory-paw-tutorial">PAW Tutorial</a><ul>
<li><a href="#plane-wave-density-functional-theory-optimizing-a-water-molecule">Optimizing a water molecule</a></li>
<li><a href="#plane-wave-density-functional-theory-optimizing-a-unit-cell-and-geometry-for-silicon-carbide">Optimizing a unit cell and geometry for Silicon-Carbide</a></li>
<li><a href="#plane-wave-density-functional-theory-running-a-car-parrinello-simulation">Running a Car-Parrinello Simulation</a></li>
</ul>
</li>
<li><a href="#plane-wave-density-functional-theory-nwpw-capabilities-and-limitations">NWPW Capabilities and Limitations</a></li>
<li><a href="#plane-wave-density-functional-theory-questions-and-difficulties">Questions and Difficulties</a></li>
</ul>
</li>
</ul>
</div>
<h2 id="plane-wave-density-functional-theory-overview">Overview<a class="headerlink" href="#plane-wave-density-functional-theory-overview" title="Permanent link">&para;</a></h2>
<p>The NWChem plane-wave (NWPW) module uses pseudopotentials and plane-wave
basis sets to perform Density Functional Theory calculations (simple
introduction <a href="pw-lecture.pdf">pw-lecture.pdf</a>).
This module complements the capabilities of the more traditional
Gaussian function based approaches by having an accuracy at least as
good for many applications, yet is still fast enough to treat systems
containing hundreds of atoms. Another significant advantage is its
ability to simulate dynamics on a ground state potential surface
directly at run-time using the Car-Parrinello algorithm. This method&rsquo;s
efficiency and accuracy make it a desirable first principles method of
simulation in the study of complex molecular, liquid, and solid state
systems. Applications for this first principles method include the
calculation of free energies, search for global minima, explicit
simulation of solvated molecules, and simulations of complex vibrational
modes that cannot be described within the harmonic approximation.</p>
<p>The NWPW module is a collection of three modules.</p>
<ul>
<li>PSPW - (PSeudopotential Plane-Wave) A gamma point code for
    calculating molecules, liquids, crystals, and surfaces.</li>
<li>Band - A band structure code for calculating crystals and surfaces
    with small band gaps (e.g. semi-conductors and metals).</li>
<li>PAW - a (gamma point) projector augmented plane-wave code for
    calculating molecules, crystals, and surfaces
    (<span style="color: red;"> <em>This module will be deprecated in the
    future releases since PAW potentials have been added to PSPW</em>
    </span>)</li>
</ul>
<p>The PSPW, Band, and PAW modules can be used to compute the energy and
optimize the geometry. Both the PSPW and Band modules can also be used
to find saddle points, and compute numerical second derivatives. In
addition the PSPW module can also be used to perform Car-Parrinello
molecular dynamics. Section <a href="#plane-wave-density-functional-theory-pspw-tasks-gamma-point-calculations">PSPW Tasks</a>
describes the tasks contained within the PSPW module, section <a href="#plane-wave-density-functional-theory-band-tasks-multiple-k-point-calculations">Band
Tasks</a> describes the tasks contained within the
Band module, section <a href="#plane-wave-density-functional-theory-paw-tasks-legacy-implementation">PAW Tasks</a> describes the
tasks contained within the PAW module, and section <a href="#plane-wave-density-functional-theory-pseudopotential-and-paw-basis-libraries">Pseudopotential and
PAW basis
Libraries</a>
describes the pseudopotential library included with NWChem. The
datafiles used by the PSPW module are described in section <a href="#plane-wave-density-functional-theory-nwpw-rtdb-entries-and-miscellaneous-datafiles">NWPW RTDB
Entries and DataFiles</a>.
Car-Parrinello output data files are described in section
<a href="#plane-wave-density-functional-theory-car-parrinello-output-datafiles">Car-Parrinello Output
Datafiles</a>, and the
minimization and Car-Parrinello algorithms are described in section
<a href="#plane-wave-density-functional-theory-car-parrinello-scheme-for-ab-initio-molecular-dynamics">Car-Parrinello Scheme for Ab Initio Molecular
Dynamics</a>.
Examples of how to setup and run a PSPW geometry optimization, a
Car-Parrinello simulation, a band structure minimization, and a PAW
geometry optimization are presented at the end. Finally in section <a href="#plane-wave-density-functional-theory-nwpw-capabilities-and-limitations">NWPW
Capabilities and
Limitations</a> the
capabilities and limitations of the NWPW module are discussed.  </p>
<p><span style="color: red;"> <em>As of NWChem 6.6 to use PAW potentials the
user is recommended to use the implementation contained in the PSPW
module (see Sections ). PAW potentials are also being integrated into
the BAND module. Unfortunately, the porting to BAND was not completed
for the NWChem 6.6 release.</em> </span></p>
<p>If you are a first time user of this module it is recommended that you
skip the next five sections and proceed directly to the tutorials.</p>
<h2 id="plane-wave-density-functional-theory-pspw-tasks-gamma-point-calculations">PSPW Tasks: Gamma Point Calculations<a class="headerlink" href="#plane-wave-density-functional-theory-pspw-tasks-gamma-point-calculations" title="Permanent link">&para;</a></h2>
<p>All input to the PSPW Tasks is contained within the compound PSPW block,</p>
<pre><code>PSPW 
  ...
END
</code></pre>
<p>To perform an actual calculation a TASK PSPW directive is used (Section
<a href="#task">Task</a>).</p>
<p><code>TASK PSPW</code></p>
<p>In addition to the directives listed in
<a href="#task">Task</a>, i.e. </p>
<pre><code>TASK PSPW energy 
TASK PSPW gradient 
TASK PSPW optimize 
TASK PSPW saddle 
TASK PSPW freqencies
TASK PSPW vib 
</code></pre>
<p>there are additional directives that are specific to the
PSPW module, which are:</p>
<pre><code>TASK PSPW [Car-Parrinello             ||   
           Born-Oppenheimer           ||
           Metropolis                 ||
           pspw_et                    ||
           noit_energy                ||
           stress                     ||
           pspw_dplot                 ||  
           wannier                    ||
           expand_cell                || 
           exafs                      ||
           ionize                     ||
           lcao                       ||
           rdf                        ||
           aimd_properties            ||
           translate                  ||
           psp_generator              || 
           steepest_descent           ||  
           psp_formatter              || 
           wavefunction_initializer   ||  
           v_wavefunction_initializer ||  
           wavefunction_expander       ]
</code></pre>
<p>Once a user has specified a geometry, the PSPW module can be invoked
with no input directives (defaults invoked throughout). However, the
user will probably always specify the simulation cell used in the
computation, since the default simulation cell is not well suited for
most systems. There are sub-directives which allow for customized
application; those currently provided as options for the PSPW module
are:</p>
<pre><code>NWPW   
  SIMULATION_CELL            ... (see section [Simulation Cell](#simulation-cell)) END
  CELL_NAME &lt;string cell_name default 'cell_default'&gt;
  VECTORS [[input (&lt;string input_wavefunctions default file_prefix.movecs&gt;) || 
           [output(&lt;string output_wavefunctions default file_prefix.movecs&gt;)]] 
  XC (Vosko      || LDA          || PBE96         || revPBE       || PBEsol          ||   
      LDA-SIC    || LDA-SIC/2    || LDA-0.4SIC    || LDA-SIC/4    || LDA-0.2SIC    || 
      PBE96-SIC  || PBE96-SIC/2  || PBE96-0.4SIC  || PBE96-SIC/4  || PBE96-0.2SIC  || 
      revPBE-SIC || revPBE-SIC/2 || revPBE-0.4SIC || revPBE-SIC/4 || revPBE-0.2SIC || 
      PBE96-Grimme2  || PBE96-Grimme3  || PBE96-Grimme4  || BLYP-Grimme2   || BLYP-Grimme3    || BLYP-Grimme4 || 
      revPBE-Grimme2 || revPBE-Grimme3 || revPBE-Grimme4 || PBEsol-Grimme2 || PBEsol-Grimme3 || PBEsol-Grimme4 || 
      PBE0-Grimme2    || PBE0-Grimme3    || PBE0-Grimme4   || B3LYP-Grimme2  || B3LYP-Grimme3  || B3LYP-Grimme4 ||
      revPBE0-Grimme2 || revPBE0-Grimme3 || revPBE0-Grimme4 ||
      PBE0       || revPBE0      || HSE  || HF || default Vosko) 
  XC new ...(see section [Using Exchange-Correlation Potentials Available in the DFT Module](#Using_Exchange-Correlation_Potentials_Available_in_the_DFT_Module))
  DFT||ODFT||RESTRICTED||UNRESTRICTED 
  MULT &lt;integer mult default 1&gt;   
  CG 
  LMBFGS   
  SCF [Anderson|| simple || Broyden]  
      [CG || RMM-DIIS] 
      [density || potential]
      [ALPHA real alpha default 0.25]
      [Kerker real ekerk nodefault] 
      [ITERATIONS integer inner_iterations default 5]  
      [OUTER_ITERATIONS integer outer_iterations default 0]
  LOOP &lt;integer inner_iteration outer_iteration default 10 100&gt;  
  TOLERANCES &lt;real tole tolc default 1.0e-7 1.0e-7&gt; 
  FAKE_MASS &lt;real fake_mass default 400000.0&gt; 
  TIME_STEP &lt;real time_step default 5.8&gt; 
  EWALD_NCUT &lt;integer ncut default 1&gt; 
  EWALD_RCUT &lt;real rcut default (see input description)&gt;  
  CUTOFF  &lt;real cutoff&gt; 
  ENERGY_CUTOFF &lt;real ecut default (see input description)&gt; 
  WAVEFUNCTION_CUTOFF &lt;real wcut default (see input description)&gt;  
  ALLOW_TRANSLATION 
  TRANSLATION (ON || OFF)
  ROTATION (ON || OFF) 
  MULLIKEN [OFF]
  EFIELD   

  BO_STEPS &lt;integer bo_inner_iteration bo_outer_iteration default 10 100&gt; 
  MC_STEPS &lt;integer mc_inner_iteration mc_outer_iteration default 10 100&gt;
  BO_TIME_STEP  &lt;real bo_time_step default 5.0&gt; 
  BO_ALGORITHM [verlet|| velocity-verlet || leap-frog]
  BO_FAKE_MASS &lt;real bo_fake_mass default 500.0&gt; 

  SOCKET (UNIX || IPI_CLIENT) &lt;string socketname default (see input description)&gt;  

  MAPPING &lt;integer mapping default 1&gt;  
  NP_DIMENSIONS &lt;integer npi npj default -1 -1&gt;  
  CAR-PARRINELLO             ... (see section [Car-Parrinello](#car-parrinello-scheme-for-ab-initio-molecular-dynamics)) END 
  STEEPEST_DESCENT           ... (see section [Steepest Descent](#STEEPEST_DESCENT)) END
  DPLOT                      ... (see section [DPLOT](#DPLOT)) END 
  WANNIER                    ... (see section [Wannier](#Wannier)) END 
  PSP_GENERATOR              ... (see section [PSP Generator](#PSP_GENERATOR))) END   

  WAVEFUNCTION_INITIALIZER   ... (see section [Wavefunction Initializer](NWPW_RETIRED.md#WAVEFUNCTION_INITIALIZER) - retired) END  
  V_WAVEFUNCTION_INITIATIZER ... (see section [Wavefunction Velocity Initializer (NWPW_RETIRED#V_WAVEFUNCTION_INITIALIZER) - retired) END  
  WAVEFUNCTION_EXPANDER      ... (see section [Wavefunction Expander](NWPW_RETIRED.md#WAVEFUNCTION_EXPANDER) - retired) END  
  INPUT_WAVEFUNCTION_FILENAME &lt;string input_wavefunctions default file_prefix.movecs&gt;   
 OUTPUT_WAVEFUNCTION_FILENAME &lt;string output_wavefunctions default file_prefix.movecs&gt;   
END
</code></pre>
<p>The following list describes the keywords contained in the PSPW input
block.</p>
<ul>
<li><code>cell_name</code> - name of the simulation_cell named <code>cell_name</code>. See
    section <a href="#plane-wave-density-functional-theory-simulation-cell">Simulation Cell</a>.</li>
<li><code>input_wavefunctions</code> - name of the file containing one-electron
    orbitals</li>
<li><code>output_wavefunctions</code> - name of the file that will contain the
    one-electron orbitals at the end of the run.</li>
<li><code>fake_mass</code> - value for the electronic fake mass <span class="arithmatex">\(\mu\)</span>
    This parameter is not presently used in a conjugate gradient
    simulation.</li>
<li><code>time_step</code> - value for the time step <span class="arithmatex">\(\Delta t\)</span>. This parameter
    is not presently used in a conjugate gradient simulation.</li>
<li><code>inner_iteration</code> - number of iterations between the printing out of
    energies and tolerances</li>
<li><code>outer_iteration</code> - number of outer iterations</li>
<li><code>tole</code> - value for the energy tolerance.</li>
<li><code>tolc</code> - value for the one-electron orbital tolerance.</li>
<li><code>cutoff</code> - value for the cutoff energy used to define the
    wavefunction. In addition using the CUTOFF keyword automatically
    sets the cutoff energy for the density to be twice the wavefunction
    cutoff.</li>
<li><code>ecut</code> - value for the cutoff energy used to define the density.
    Default is set to be the maximum value that will fit within the
    simulation_cell <code>cell_name</code>.</li>
<li><code>wcut</code> - value for the cutoff energy used to define the one-electron
    orbitals. Default is set to be the maximum value that will fit
    within the simulation_cell <code>cell_name</code>.</li>
<li><code>ncut</code> - value for the number of unit cells to sum over (in each
    direction) for the real space part of the Ewald summation. Note
    Ewald summation is only used if the simulation_cell is periodic.</li>
<li><code>rcut</code> - value for the cutoff radius used in the Ewald summation.
    Note Ewald summation is only used if the simulation_cell is
    periodic.</li>
</ul>
<p>Default set to be <span class="arithmatex">\(\frac{MIN(\left| \vec{a_i} \right|)}{\pi}, i=1,2,3\)</span>.</p>
<ul>
<li>(Vosko || PBE96 || revPBE || &hellip;) - Choose between Vosko et al&rsquo;s LDA
    parameterization or the orginal and revised Perdew, Burke, and
    Ernzerhof GGA functional. In addition, several hybrid options.</li>
<li>MULT - optional keyword which if specified allows the user to define
    the spin multiplicity of the system</li>
<li>MULLIKEN - optional keyword which if specified causes a Mulliken
    analysis to be performed at the end of the simulation.</li>
<li>EFIELD - optional keyword which if specified causes an atomic
    electric field analysis to be performed at the end of the
    simulation.</li>
<li>ALLOW_TRANSLATION - By default the the center of mass forces are
    projected out of the computed forces. This optional keyword if
    specified allows the center of mass forces to not be zero.</li>
<li>TRANSLATION - By default the the center of mass forces are projected
    out of the computed forces. TRANSLATION ON allows the center of mass
    forces to not be zero.</li>
<li>ROTATION - By default the overall rotation is not projected out of
    the computed forces. ROTATION OFF projects out the overal rotation
    of the molecule.</li>
<li>CG - optional keyword which sets the minimizer to 1</li>
<li>LMBFGS - optional keyword which sets the minimizer to 2</li>
<li>SCF - optional keyword which sets the minimizer to be a band by band
    minimizer. Several options are available for setting the density or
    potential mixing, and the type of Kohn-Sham minimizer.</li>
<li><code>mapping</code> - for a value of 1 slab FFT is used, for a value of 2 a
    2d-hilbert FFT is used.</li>
</ul>
<p>A variety of prototype minimizers can be used to minimize the energy. To
use these other optimizers the following SET directive needs to be
specified:</p>
<pre><code>set nwpw:mimimizer 1 # Default - Grassman conjugate gradient minimizer is used to minimize the energy. 
set nwpw:mimimizer 2 # Grassman LMBFGS minimimzer is used to minimize the energy.
set nwpw:minimizer 4 # Stiefel conjugate gradient minimizer is used to minimize the energy. 
set nwpw:minimizer 5 # Band-by-band (potential) minimizer is used to minimize the energy.
set nwpw:minimizer 6 # Projected Grassman LMBFGS minimizer is used to minimize the energy.
set nwpw:minimizer 7 # Stiefel LMBFGS minimizer is used to minimize the energy.
set nwpw:minimizer 8 # Band-by-band (density) minimizer is used to minimize the energy.
</code></pre>
<p>Limited testing suggests that the Grassman LMBFGS minimizer is about
twice as fast as the conjugate gradient minimizer. However, there are
several known cases where this optimizer fails, so it is currently not a
default option, and should be used with caution.</p>
<p>In addition the following SET directives can be
specified:</p>
<pre><code>set nwpw:lcao_skip .false. # Initial wavefunctions generated using an LCAO guess.  
set nwpw:lcao_skip .true. # Default - Initial wavefunctions generated using a random plane-wave guess.
set nwpw:lcao_print .false. # Default - Output not produced during the generation of the LCAO guess.   
set nwpw:lcao_print .true. # Output produced during the generation of the LCAO guess.
set nwpw:lcao_iterations 2 #specifies the number of LCAO iterations.
</code></pre>
<h3 id="plane-wave-density-functional-theory-paw-potentials">PAW Potentials<a class="headerlink" href="#plane-wave-density-functional-theory-paw-potentials" title="Permanent link">&para;</a></h3>
<p>The PSPW code can now handle PAW potentials. To use them the
pseudopotentials input block is used to redirect the code to use the paw
potentials located in the default paw potential library
(<code>$NWCHEM_TOP/src/nwpw/libraryp/paw_default</code>). For example, to redirect
the code to use PAW potentials for carbon and hydrogen, the following
input would be used.</p>
<pre><code>nwpw 
    pseudopotentials 
      C library paw_default 
      H library paw_default 
    end  
end
</code></pre>
<p>Most of the capabilities of PSPW will work with PAW potentials including
geometry optimization, Car-Parrinello ab initio molecular dynamics,
Born-Oppenheimer ab initio molecular dynamics, Metropolis Monte-Carlo,
and AIMD/MM. Unfortunately, some of the functionality is missing at this
point in time such as Mulliken analysis, and analytic stresses. However
these small number of missing capabilities should become available over
the next couple of months in the development tree of NWChem.</p>
<p>Even though analytic stresses are not currently available with PAW
potentials unit cell optimization can still be carried out using
numerical stresses. The following SET directives can be used to tell the
code to calculate stresses
numerically.</p>
<pre><code>set includestress  .true.              #this option tells driver to optimize the unit cell 
set includelattice .true.              #this option tells driver to optimize cell using a,b,c,alpha,beta,gamma 
set nwpw:frozen_lattice:thresh 999.0   #large number guarentees the lattice gridding does not adjust during optimization
set nwpw:cif_filename  pspw_corundum
set nwpw:stress_numerical .true. 
set nwpw:lstress_numerical .true.
</code></pre>
<h4 id="plane-wave-density-functional-theory-paw-implementation-notes">PAW Implementation Notes<a class="headerlink" href="#plane-wave-density-functional-theory-paw-implementation-notes" title="Permanent link">&para;</a></h4>
<p>The main idea in the PAW method(Blochl 1994) is to project out the
high-frequency components of the wavefunction in the atomic sphere
region. Effectively this splits the original wavefunction into two
parts:</p>
<div class="arithmatex">\[\psi_n(\mathbf{r}) = \tilde{\psi}_n(\mathbf{r}) + \sum_I \psi_n^I(\mathbf{r})\]</div>
<p>The first part <span class="arithmatex">\(\tilde{\psi}_n(\mathbf{r})\)</span> is smooth and can be
represented using a plane wave basis set of practical size. The second
term is localized with the atomic spheres and is represented on radial
grids centered on the atoms
as</p>
<div class="arithmatex">\[\psi_n^I(\mathbf{r}) = \sum_{\alpha} (\varphi_{\alpha}^I(\mathbf{r})-\tilde{\varphi}_{\alpha}^I(\mathbf{r}))c_{n\alpha}^I\]</div>
<p>where the coefficients <span class="arithmatex">\(c_{n\alpha}^I\)</span> are given by</p>
<div class="arithmatex">\[ c_{n\alpha}^I = &lt;\tilde{p}_{\alpha}^I | \tilde{\psi}_n&gt;\]</div>
<p>This decomposition can be expressed using an invertible linear
transformation, <span class="arithmatex">\(T\)</span>, is defined which relates the stiff one-electron
wavefunctions <span class="arithmatex">\(\psi_n\)</span> to a set of smooth one-electron wavefunctions
<span class="arithmatex">\(\tilde{\psi}_n\)</span>   </p>
<div class="arithmatex">\[\tilde{\psi}_n = T \psi_n\]</div>
<div class="arithmatex">\[\psi_n = T^{-1} \tilde{\psi}_n\]</div>
<p>which can be represented by fairly small plane-wave basis. The
transformation <span class="arithmatex">\(T\)</span> is defined using a local PAW basis, which consists
of atomic orbitals, <span class="arithmatex">\(\varphi{\alpha}^I(\mathbf{r})\)</span>, 
smooth atomic
orbitals, <span class="arithmatex">\(\tilde{\varphi}\)</span><sub>&alpha;</sub><sup>I</sup>(<strong>r</strong>)  which coincide with
the atomic orbitals outside a defined atomic sphere, and projector
functions, <span class="arithmatex">\(p_{\alpha}^I(\mathbf{r})\)</span>. Where I is the atomic index and
 is the orbital index. The projector functions are constructed such
that they are localized within the defined atomic sphere and in addition
are orthonormal to the atomic orbitals. Blchl defined the invertible linear transformations by  </p>
<div class="arithmatex">\[T = 1 + \sum_I \sum_\alpha (|\tilde{\varphi}_\alpha^I&gt; - |\varphi_\alpha^I &gt; ) &lt; p_\alpha^I|\]</div>
<div class="arithmatex">\[\tilde{T} = 1 + \sum_I \sum_{\alpha} (|\varphi_{\alpha}^I&gt; - |\tilde{\varphi}_{\alpha}^I &gt;)&lt;\tilde{\varphi}_{\alpha}^I|\]</div>
<div class="arithmatex">\[|\tilde{p}_{\alpha}^I&gt; = \sum_{\beta} [&lt;\tilde{p}^I | \varphi^I &gt;]_{\alpha\beta}^{-1} | p_{\beta}^I&gt;\]</div>
<p>The main effect of the PAW transformation is that the fast variations of
the valence wave function in the atomic sphere region are projected out
using local basis set, thereby producing a smoothly varying wavefunction
that may be expanded in a plane wave basis set of a manageable size.</p>
<p>The expression for the total energy in PAW method can be separated into
the following 15 terms.</p>
<div class="arithmatex">\[E_{PAW} = \tilde{E}_{kinetic-pw} + \tilde{E}_{vlocal-pw} + \tilde{E}_{Coulomb-pw} + \tilde{E}_{xc-pw} + E_{ion-ion}\]</div>
<div class="arithmatex">\[      + E_{cmp-cmp} + E_{cmp-pw} + E_{valence-core} + E_{kinetic-core} + E_{ion-core}\]</div>
<p>The first five terms are essentially the same as for a standard
pseudopotential plane-wave program, minus the non-local pseudopotential,
where</p>
<div class="arithmatex">\[\tilde{E}_{kinetic-pw} = \sum_i \sum_{\mathbf{G}} \frac{|\mathbf{G}|^2}{2} \tilde{\psi}_i^{*}(\mathbf{G}) \tilde{\psi}_i(\mathbf{G})\]</div>
<div class="arithmatex">\[\tilde{E}_{vlocal-pw} = \sum_{\mathbf{G}} \tilde{\rho}(\mathbf{G}) V_{local}(\mathbf{G})\]</div>
<div class="arithmatex">\[\tilde{E}_{Coulomb-pw} = \frac{\Omega}{2} \sum_{\mathbf{G}} \frac{4\pi}{|\mathbf{G}|} \tilde{\rho}^{*}(\mathbf{G}) \tilde{\rho}(\mathbf{G})\]</div>
<div class="arithmatex">\[\tilde{E}_{xc-pw} = \frac{\Omega}{N_1 N_2 N_3} \sum_{\mathbf{r}} \tilde{\rho}(\mathbf{r}) \epsilon_{xc}(\tilde{\rho}(\mathbf{r}))\]</div>
<div class="arithmatex">\[E_{ion-ion} = \frac{1}{2\Omega} \sum_{\mathbf{G}} \frac{4\pi}{|\mathbf{G}|^2} \exp(\frac{|\mathbf{G}|^2}{4\epsilon}) \sum_{I,J} Z_I \exp (-i \mathbf{G} \cdot \mathbf{R}_I) Z_J \exp ( -i  \mathbf{G} \cdot \mathbf{R}_J) + \]</div>
<div class="arithmatex">\[\frac{1}{2}\sum_{\mathbf{a}} \sum_{I,J \in |\mathbf{R}_I-\mathbf{R}_J+\mathbf{a}|} Z_I Z_J \frac{erf(\epsilon |\mathbf{R}_I-\mathbf{R}_J+\mathbf{a}|)}{|\mathbf{R}_I-\mathbf{R}_J+\mathbf{a}|} - \frac{\epsilon}{\pi}\sum_I Z_I^2 +  \]</div>
<div class="arithmatex">\[ - \frac{\pi}{2\epsilon^2\Omega} \left( \sum_{I} Z_{I} \right)^2 \]</div>
<p>The local potential in the <span class="arithmatex">\(\tilde{E}_{vlocal-pw}\)</span> term is the Fourier
transform of</p>
<div class="arithmatex">\[V_{local}(\mathbf{r}) = -\sum_I Z_I \frac{\frac{|\mathbf{r}-\mathbf{R}_I|}{\sigma_I}}{|\mathbf{r}-\mathbf{R}_I|} + v_{ps}^I(|\mathbf{|\mathbf{r}-\mathbf{R}_I|})\]</div>
<p>It turns out that for many atoms <span class="arithmatex">\(\sigma_I\)</span> needs to be fairly small.
This results in <span class="arithmatex">\(V_{local} (\mathbf{r})\)</span> being stiff. However, since
in the integral above this function is multiplied by a smooth density
<span class="arithmatex">\(\tilde{\rho}(\mathbf{G})\)</span> the expansion of <em>V<sub>local</sub>(<strong>G</strong>)</em>
only needs to be the same as the smooth density. The auxiliary
pseudoptential <span class="arithmatex">\(v_{ps}^I (|\mathbf{r}-\mathbf{R}_I |)\)</span> is defined to
be localized within the atomic sphere and is introduced to remove ghost
states due to local basis set incompleteness.</p>
<p>The next four terms are atomic based and they essentially take into
account the difference between the true valence wavefunctions and the
pseudowavefunctions.</p>
<div class="arithmatex">\[ E_{kinetic-atom}= \sum_I \sum_i \sum_{\alpha\beta} &lt;\tilde{\psi}_i|\tilde{p}_{\alpha}^I &lt; (t_{atom}^I)_{\alpha\beta} &lt;\tilde{p}_{\beta}^I|\tilde{\psi}_i \]</div>
<div class="arithmatex">\[ E_{local-atom}=\sum_I \sum_i \sum_{\alpha\beta} &lt;\tilde{\psi}_i|\tilde{p}_{\alpha}^I&gt; (u_{atom}^I)_{\alpha\beta} &lt;\tilde{p}_{\beta}^I|\tilde{\psi}_i&gt;\]</div>
<div class="arithmatex">\[ E_{xc-atom}=\sum_I \sum_{\theta\phi} w_{\theta\phi} \int_0^{r_{cut}^I} r^2 (\rho^I(r,\theta,\phi) \epsilon_{xc}(\rho^I(r,\theta,\phi)) -\tilde{\rho}^I(r,\theta,\phi) \epsilon_{xc}(\tilde{\rho}^I(r,\theta,\phi))) dr \]</div>
<div class="arithmatex">\[ E_{hartree-atom}= \sum_I W_{atom}^I =\frac{1}{2} \sum_I \sum_i \sum_{\alpha\beta} &lt;\tilde{\psi}_i|\tilde{p}_{\alpha}&gt;  &lt;\tilde{p}_{\beta}^I|\tilde{\psi}_i&gt; \]</div>
<div class="arithmatex">\[ \sum_j \sum_{\mu\nu} &lt;\tilde{\psi}_j|\tilde{p}_{\mu}^I&gt;  &lt;\tilde{p}_{\nu}^I|\tilde{\psi}_j&gt; \sum_{lm} \tau_{l_\alpha m_\alpha, l_\beta m_\beta}^{lm} \tau_{l_\mu m_\mu, l_\nu m_\nu}^{lm} (V_{Heff}^I)_{\alpha\beta\mu\nu}^{l} \]</div>
<p>The next three terms are the terms containing the compensation charge
densities.</p>
<div class="arithmatex">\[E_{cmp-vloc}= \sum_{\mathbf{G}} [\rho_{cmp} (\mathbf{G}) \tilde{V}_{local}(\mathbf{G})+\tilde{\rho}_{cmp}(\mathbf{G})(V_{local}(\mathbf{G})-\tilde{V}_{local}(\mathbf{G}))] + \\
\int (\rho_{cmp}(\mathbf{r})-\tilde{\rho}_{cmp}(\mathbf{r}))(V_{local}(r)-\tilde{V}_{local}(\mathbf{r}))d\mathbf{r}\]</div>
<div class="arithmatex">\[E_{cmp-cmp}= \Omega \sum_{\mathbf{G} \neq 0} \frac{4\pi}{|\mathbf{G}|^2} [ \rho_{cmp}(\mathbf{G}) \tilde{\rho}_{cmp}(\mathbf{G})- \frac{1}{2} \tilde{\rho}_{cmp}(\mathbf{G})\tilde{\rho}_{cmp}(\mathbf{G})] + \\
 \frac{1}{2} \iint \frac{(\rho_{cmp}(\mathbf{r})-\tilde{\rho}_{cmp}(\mathbf{r}))(\rho_{cmp}(\mathbf{r}')-\tilde{\rho}_{cmp}(\mathbf{r}'))} {|\mathbf{r}-\mathbf{r}'|}  d\mathbf{r} d\mathbf{r}'\]</div>
<div class="arithmatex">\[E_{cmp-pw}= \Omega \sum_{\mathbf{G} \neq 0} \frac{4\pi}{|\mathbf{G}|^2} \rho_{cmp}(\mathbf{G}) \tilde{\rho}(\mathbf{G})\]</div>
<p>In the first two formulas the first terms are computed using plane-waves
and the second terms are computed using Gaussian two center integrals.
The smooth local potential in the <span class="arithmatex">\(E_{cmp-vloc}\)</span> term is the Fourier
transform
of</p>
<div class="arithmatex">\[\tilde{V}_{local}(\mathbf{r})=- \sum_I Z_I    \frac{ erf( \frac{ |\mathbf{r}-\mathbf{R}_I | }{\tilde{\sigma}_I })} {|r-R_I |}\]</div>
<p>The stiff and smooth compensation charge densities in the above formula
are</p>
<div class="arithmatex">\[\rho_{cmp}(r)=\sum_I\sum_{lm} Q_{lm}^I g_{lm}^(\sigma_I ) (r-R_I )\]</div>
<div class="arithmatex">\[\tilde{\rho}_{cmp}(r)=\sum_I \sum_{lm} Q_{lm}^I g_{lm}^(\tilde{\sigma}_I ) (r-R_I )\]</div>
<p>where</p>
<div class="arithmatex">\[Q_{lm}^I = \sum_i \sum_{\alpha\beta} &lt;\tilde{\psi}_i|\tilde{p}_\alpha^I &gt; &lt;\tilde{p}_\beta^I|\tilde{\psi}_i &gt;  \tau_{l_\alpha m_\alpha, l_\beta m_\beta}^{lm} (q_{comp}^I )_{\alpha\beta}^l\]</div>
<p>The decay parameter <span class="arithmatex">\(\sigma_I\)</span> is defined the same as above, and
<span class="arithmatex">\(\tilde{\sigma}\)</span><sub>I</sub> is defined to be smooth enough in order that <sub>cmp</sub>(r) and <span class="arithmatex">\(\tilde{V}\)</span><sub>local</sub>(r) can readily be expanded in
terms of plane-waves.</p>
<p>The final three terms are the energies that contain the core
densities</p>
<div class="arithmatex">\[E_{valence-core}=\sum_{i}\sum_{I}\sum_{\alpha \beta}&lt;\tilde{\psi}_{i} | \tilde{p}_{\alpha}^{I} &gt;  (V_{valence-core})_{\alpha \beta}^{I} &lt;\tilde{p}_\beta^{I} | \tilde{\psi}_{i} &gt;\]</div>
<div class="arithmatex">\[E_{kinetic-core}=\sum_c \int_0^\infty \left[ (\varphi_{n_c l_c}^I (r))^{'} (\varphi_{n_c l_c}^{I} (r))^{'} + l_c (l_c+1)  \frac{\varphi_{n_c l_c}^{I} (r) \varphi_{n_c l_c}^{I} (r)}{r^{2}} \right]  dr\]</div>
<div class="arithmatex">\[E_{ion-core}= \sum_I \frac{1}{2} \iint \frac{\rho_c^{I} (r) \rho_c^{I} (r')}{|r-r'|}  drdr^{'} - \int  \frac{\rho_c^{I} (r)}{|r|}  (Z_I+Z_I^{core} ) dr\]</div>
<p>The matrix elements contained in the above formula are</p>
<div class="arithmatex">\[(t_{atom}^{I} )_{\alpha \beta}= {\delta_{m_\alpha m_\beta} \delta_{l_\alpha l_\beta }}2  \int_0^{r_{cut}^I}  \left[
  (\varphi_{n_\alpha l_\alpha}^{I} (r))' (\varphi_{n_\beta l_\beta}^{I} (r))' - (\tilde{\varphi}_{n_\alpha l_\alpha}^{I} (r))' (\tilde{\varphi}_{n_\beta l_\beta}^{I} (r))' + l_\alpha (l_\alpha+1)  \frac{ \varphi_{n_\alpha l_\alpha}^{I}(r) \varphi_{n_\beta l_\beta}^{I}(r) - \tilde{\varphi}_{n_\alpha l_\alpha}^{I}(r) \tilde{\varphi}_{n_\beta l_\beta}^I }{r^2 } \right]  dr\]</div>
<div class="arithmatex">\[(u_{atom}^I )_{\alpha \beta} =\frac{Z_I}{4\pi} (V_{comp}^I )_{\alpha \beta}^{l=0} + \frac{2Z_I}{(\sqrt{2\pi} \sigma_I } (q_comp^I )_{\alpha \beta}^(l=0) +\delta_{m_\alpha m_\beta } \delta_{l_\alpha l_\beta } \int_0^{r_{cut}^I} [ \varphi_{n_\alpha l_\alpha}^I (r) \varphi_(n_\beta l_\beta)^I (r) \frac{-Z_I}{r} + \tilde{\varphi}_{n_\alpha l_\alpha}^I (r) \tilde{\varphi}_{n_\beta l_\beta}^I (r) (-v_{ps}^I (r)) ] dr\]</div>
<div class="arithmatex">\[(V_{Heff}^I )_{\alpha \beta \mu \nu}^l = (V_H^I )_{\alpha\beta\mu\nu}^l - 2(V_{comp}^I )_{\alpha\beta}^l (q_{comp}^I )_{\mu\nu}^l -(v_g^I )^l (q_{comp}^I )_{\alpha\beta}^l (q_{comp}^I )_{\mu\nu}^l\]</div>
<div class="arithmatex">\[(V_H^I )_{\alpha\beta\mu\nu}^l = \frac{4\pi}{2l+1} \int_0^{r_{cut}^I} \int_0^{r_{cut}^I} \left( \frac{r_{&lt;}^l}{r_{&gt;}^{l+1} } \right)( \varphi_{n_\alpha l_\alpha } (r) \varphi_{n_\beta l_\beta } (r) \varphi_{n_\mu l_\mu } (r) \varphi_{n_\nu l_\nu } (r) - \tilde{\varphi}_{n_\alpha l_\alpha } (r) \tilde{\varphi}_(n_\beta l_\beta ) (r) \tilde{\varphi}_{n_\mu l_\mu } (r) \tilde{\varphi}_{n_\nu l_\nu } (r) )  drdr'\]</div>
<div class="arithmatex">\[(V_{comp}^I )_{\alpha\beta}^l = \frac{4\pi}{2l+1} \int_0^{r_{cut}^I} \int_0^{r_{cut}^I} \tilde{\varphi}_{n_\alpha l_\alpha } (r) \tilde{\varphi}_{n_\beta l_\beta } (r) \left( \frac{r_{&lt;}^l}{r_{&gt;}^{l+1} } \right) g_l^I (r')  r'^2  drdr'\]</div>
<div class="arithmatex">\[(q_{comp}^I )_{\alpha\beta}^l = \int_0^\infty r^l \left( \varphi_{n_\alpha l_\alpha } (r) \varphi_{n_\beta l_\beta } (r)- \tilde{\varphi}_{n_\alpha l_\alpha } (r) \tilde{\varphi}_{n_\beta l_\beta } (r) \right) dr\]</div>
<div class="arithmatex">\[(v_g^I )^l =\frac{4\sqrt{2\pi}}{ (2l+1)(2l+1)!! \sigma_I^{2l+1} }\]</div>
<div class="arithmatex">\[\tau_{l_\alpha m_\alpha,l_\beta m_\beta}^{lm}= \int_0^{2\pi} \int_0^{\pi} T_{lm} (\theta,\phi) T_{l_\alpha m_\alpha } (\theta,\phi) T_{l_\beta m_\beta } (\theta,\phi)  \sin(\theta) d\theta d\phi\]</div>
<h3 id="plane-wave-density-functional-theory-exchange-correlation-potentials">Exchange-Correlation Potentials<a class="headerlink" href="#plane-wave-density-functional-theory-exchange-correlation-potentials" title="Permanent link">&para;</a></h3>
<h4 id="plane-wave-density-functional-theory-dft-u-corrections">DFT + U Corrections<a class="headerlink" href="#plane-wave-density-functional-theory-dft-u-corrections" title="Permanent link">&para;</a></h4>
<p>TO DO</p>
<pre><code>nwpw 
  uterm d 0.13634 0.0036749 1 
end
</code></pre>
<h4 id="plane-wave-density-functional-theory-langreth-style-vdw-and-vdw-van-der-wall-functionals">Langreth style vdw and vdw van der Wall functionals<a class="headerlink" href="#plane-wave-density-functional-theory-langreth-style-vdw-and-vdw-van-der-wall-functionals" title="Permanent link">&para;</a></h4>
<p>These potenials that are used to augment standard exchange-correlation potentials area calculated from a double integral over a nonlocal interaction kernel, <span class="arithmatex">\(\phi(\mathbf{r},\mathbf{r}^{'})\)</span></p>
<div class="arithmatex">\[E_{vdw} = \int \rho(\mathbf{r}) \phi(\mathbf{r},\mathbf{r}^{'}) \rho(\mathbf{r}^{'}) d\mathbf{r} d\mathbf{r}^{'} \]</div>
<p>that is evaluated using the fast Fourier transformation method of Roman-Perez and Soler.</p>
<p>G. Roman-Perez and J. M. Soler, Phys. Rev. Lett. 103, 096102 (2009).</p>
<p>Langreth vdw and vdw2 van der Wall functionals are currently available for the BEEF, PBE96,
revPBE, PBEsol, BLYP, PBE0, revPBE0, HSE, and B3LYP exchange-correlation
functionals. To use them the following keywords BEEF-vdw, BEEF-vdw2, PBE96-vdw,
PBE96-vdw2, BLYP-vdw, BLYP-vdw2,
revPBE-vdw, revPBE-vdw, PBEsol-vdw PBEsol-vdw2, PBE0-vdw, PBE0-vdw2,
revPBE0-vdw, revPBE0-vdw2, HSE-vdw, HSE-vdw2, B3LYP-vdw, and B3LYP-vdw2
 can be used in the XC input directive, e.g.</p>
<pre><code>nwpw
   xc beef-vdw  
end
</code></pre>
<pre><code>nwpw
   xc beef-vdw2  
end
</code></pre>
<p>the vdw and vdw2 functionals are defined in  </p>
<p>(vdw) Dion M, Rydberg H, Schrder E, Langreth DC, Lundqvist BI. Van der Waals density functional for general geometries. Physical review letters. 2004 Jun 16;92(24):246401.</p>
<p>(vdw2) K. Lee, E. D. Murray, L. Kong, B. I. Lundqvist, and D. C. Langreth, Phys. Rev. B 82, 081101 (2010).</p>
<h4 id="plane-wave-density-functional-theory-grimme-dispersion-corrections">Grimme Dispersion Corrections<a class="headerlink" href="#plane-wave-density-functional-theory-grimme-dispersion-corrections" title="Permanent link">&para;</a></h4>
<p>Grimme dispersion corrections are currently available for the PBE96,
revPBE, PBEsol, BLYP, PBE0, revPBE0, HSE, and B3LYP exchange-correlation
functionals. To use them the following keywords PBE96-Grimme2,
PBE96-Grimme3, PBE96-Grimme4, BLYP-Grimme2, BLYP-Grimme3, BLYP-Grimme4,
revPBE-Grimme2, revPBE-Grimme3, revPBE-Grimme4, PBEsol-Grimme2,
PBEsol-Grimme3, PBEsol-Grimme4, PBE0-Grimme2, PBE0-Grimme3,
PBE0-Grimme4, revPBE0-Grimme2, revPBE0-Grimme3, revPBE0-Grimme4,
HSE-Grimme2, HSE-Grimme3, HSE-Grimme4, B3LYP-Grimme2, B3LYP-Grimme3, and
B3LYP-Grimme4 can be used in the XC input directive, e.g.</p>
<pre><code>nwpw
   xc pbe96-grimme2  
end
</code></pre>
<p>In these functionals Grimme2, Grimme3 and Grimme4 are defined in the
following papers by S. Grimme.</p>
<p>Grimme2 - Commonly known as DFT-D2, S. Grimme, J. Comput. Chem., 27
(2006), 1787-1799.</p>
<p>Grimme3 - Commonly known as DFT-D3, S. Grimme, J. Antony, S. Ehrlich and
H. Krieg A consistent and accurate ab initio parameterization of density
functional dispersion correction (DFT-D) for the 94 elements H-Pu, J.
Chem. Phys, 132 (2010), 154104</p>
<p>Grimme4 - Commonly known as DFT-D3 with BJ damping. This correction
augments the Grimme3 correction by including BJ-damping, S. Grimme, J.
Antony, S. Ehrlich and H. Krieg A consistent and accurate ab initio
parameterization of density functional dispersion correction (DFT-D) for
the 94 elements H-Pu, J. Chem. Phys, 132 (2010), 154104. S. Grimme, S.
Ehrlich and L. Goerigk, J. Comput. Chem, 32 (2011), 1456-1465. This
correction augments the Grimme3 correction by including BJ-damping.</p>
<p>If these functionals are used in a publication please include in your
citations the references to Grimme&rsquo;s work.</p>
<h4 id="plane-wave-density-functional-theory-using-exchange-correlation-potentials-available-in-the-dft-module">Using Exchange-Correlation Potentials Available in the DFT Module<a class="headerlink" href="#plane-wave-density-functional-theory-using-exchange-correlation-potentials-available-in-the-dft-module" title="Permanent link">&para;</a></h4>
<p>(<span style="color: red;"><em>Warning - To use this capability in NWChem
6.6 the user must explicitly include the nwxc module in the
NWCHEM_MODULES list when compiling. Unfortunately, there was too much
uncertainty in how the nwxc computed higher-order derivatives used by
some of the functionality in nwdft module to include it in a release for
all the functionality in NWChem. We are planning to have a debug release
in winter 2016 to take fix this problem. This capability is still
included by default in NWChem 6.5</em></span>)</p>
<p>The user has the option of using many of the exchange-correlation
potentials available in DFT Module (see Section <a href="#density-functional-theory-for-molecules-xc-and-decomp-exchange-correlation-potentials">XC and DECOMP &ndash;
Exchange-Correlation
Potentials</a>).</p>
<pre><code>  XC [[acm] [b3lyp] [beckehandh] [pbe0] [bhlyp]\
      [becke97]  [becke97-1] [becke97-2] [becke97-3] [becke98] [hcth] [hcth120] [hcth147] \ 
      [hcth407] [becke97gga1] [hcth407p] \
      [optx] [hcthp14] [mpw91] [mpw1k] [xft97] [cft97] [ft97] [op] [bop] [pbeop]\
      [HFexch &lt;real prefactor default 1.0&gt;] \
      [becke88 [nonlocal] &lt;real prefactor default 1.0&gt;] \
      [xperdew91 [nonlocal] &lt;real prefactor default 1.0&gt;] \
      [xpbe96 [nonlocal] &lt;real prefactor default 1.0&gt;] \
      [gill96 [nonlocal] &lt;real prefactor default 1.0&gt;] \
      [lyp &lt;real prefactor default 1.0&gt;] \
      [perdew81 &lt;real prefactor default 1.0&gt;] \
      [perdew86 [nonlocal] &lt;real prefactor default 1.0&gt;] \
      [perdew91 [nonlocal] &lt;real prefactor default 1.0&gt;] \
      [cpbe96 [nonlocal] &lt;real prefactor default 1.0&gt;] \
      [pw91lda &lt;real prefactor default 1.0&gt;] \
      [slater &lt;real prefactor default 1.0&gt;] \
      [vwn_1 &lt;real prefactor default 1.0&gt;] \
      [vwn_2 &lt;real prefactor default 1.0&gt;] \
      [vwn_3 &lt;real prefactor default 1.0&gt;] \
      [vwn_4 &lt;real prefactor default 1.0&gt;] \
      [vwn_5 &lt;real prefactor default 1.0&gt;] \
      [vwn_1_rpa &lt;real prefactor default 1.0&gt;]]
</code></pre>
<p>These functional can be invoked by prepending the &ldquo;new&rdquo; directive before
the exchange correlation potetntials in the input directive, XC new
slater vwn_5.</p>
<p>That is, this statement in the input file</p>
<pre><code>nwpw 
 XC new slater vwn_5  
end  
task pspw energy
</code></pre>
<p>Using this input the user has ability to include only the local or
nonlocal contributions of a given functional. The user can also specify
a multiplicative prefactor (the variable <code>prefactor</code> in the input) for
the local/nonlocal component or total (for more details see Section <a href="#density-functional-theory-for-molecules-xc-and-decomp-exchange-correlation-potentials">XC
and DECOMP &ndash; Exchange-Correlation
Potentials</a>).
An example of this might be,</p>
<p><code>XC new becke88 nonlocal 0.72</code></p>
<p>The user should be aware that the Becke88 local component is simply the
Slater exchange and should be input as such.</p>
<p>Any combination of the supported exchange functional options can be
used. For example the popular Gaussian B3 exchange could be specified
as:</p>
<p><code>XC new slater 0.8 becke88 nonlocal 0.72 HFexch 0.2</code></p>
<p>Any combination of the supported correlation functional options can be
used. For example B3LYP could be specified
as:</p>
<p><code>XC new vwn_1_rpa 0.19 lyp 0.81 HFexch 0.20  slater 0.80 becke88 nonlocal 0.72</code></p>
<p>and X3LYP as:</p>
<pre><code>xc new vwn_1_rpa 0.129 lyp 0.871 hfexch 0.218 slater 0.782 \ 
becke88 nonlocal 0.542  xperdew91 nonlocal 0.167
</code></pre>
<h4 id="plane-wave-density-functional-theory-exact-exchange">Exact Exchange<a class="headerlink" href="#plane-wave-density-functional-theory-exact-exchange" title="Permanent link">&para;</a></h4>
<h4 id="plane-wave-density-functional-theory-self-interaction-corrections">Self-Interaction Corrections<a class="headerlink" href="#plane-wave-density-functional-theory-self-interaction-corrections" title="Permanent link">&para;</a></h4>
<p>The SET directive is used to specify the molecular orbitals contribute
to the self-interaction-correction (SIC) term.</p>
<pre><code>set pspw:SIC_orbitals &lt;integer list_of_molecular_orbital_numbers&gt;
</code></pre>
<p>This defines only the molecular orbitals in the list as SIC active. All
other molecular orbitals will not contribute to the SIC term. For
example the following directive specifies that the molecular orbitals
numbered 1,5,6,7,8, and 15 are SIC active.</p>
<p><code>set pspw:SIC_orbitals 1 5:8 15</code></p>
<p>or equivalently</p>
<p><code>set pspw:SIC_orbitals 1 5 6 7 8 15</code></p>
<p>The following directive turns on self-consistent SIC.</p>
<pre><code>set pspw:SIC_relax .false. # Default - Perturbative SIC calculation  
set pspw:SIC_relax .true. # Self-consistent SIC calculation
</code></pre>
<p>Two types of solvers can be used and they are specified using the
following SET directive</p>
<pre><code>set pspw:SIC_solver_type 1 # Default - cutoff coulomb kernel 
set pspw:SIC_solver_type 2 # Free-space boundary condition kernel
</code></pre>
<p>The parameters for the cutoff coulomb kernel are defined by the
following SET directives:</p>
<pre><code>set pspw:SIC_screening_radius &lt;real rcut&gt;  
set pspw:SIC_screening_power &lt;real rpower&gt;
</code></pre>
<h3 id="plane-wave-density-functional-theory-wannier">Wannier<a class="headerlink" href="#plane-wave-density-functional-theory-wannier" title="Permanent link">&para;</a></h3>
<p>The pspw wannier task is generate maximally localized (Wannier)
molecular orbitals. The algorithm proposed by Silvestrelli et al is use
to generate the Wannier orbitals.</p>
<p>Input to the Wannier task is contained within the Wannier sub-block.</p>
<pre><code>NWPW  
...  
 Wannier  
   ...   
 END 
... 
END
</code></pre>
<p>To run a Wannier calculation the following directive is used:</p>
<p><code>TASK PSPW Wannier</code></p>
<p>Listed below is the format of a Wannier
sub-block.</p>
<pre><code>NWPW  
...   
 Wannier   
  OLD_WAVEFUNCTION_FILENAME &lt;string input_wavefunctions default input_movecs&gt;  
  NEW_WAVEFUNCTION_FILENAME &lt;string output_wavefunctions default input_movecs&gt; 
 END
...  
END
</code></pre>
<p>The following list describes the input for the Wannier sub-block.</p>
<ul>
<li><code>input_wavefunctions</code> - name of pspw wavefunction file.</li>
<li><code>output_wavefunctions</code> - name of pspw wavefunction file that will
    contain the Wannier orbitals.</li>
</ul>
<h3 id="plane-wave-density-functional-theory-mulliken-analysis">Mulliken Analysis<a class="headerlink" href="#plane-wave-density-functional-theory-mulliken-analysis" title="Permanent link">&para;</a></h3>
<h3 id="plane-wave-density-functional-theory-density-of-states">Density of States<a class="headerlink" href="#plane-wave-density-functional-theory-density-of-states" title="Permanent link">&para;</a></h3>
<p>The &ldquo;dos&rdquo; option is used to turn on a density of states analysis. This
option can be specified without additional parameters, i.e.</p>
<pre><code>nwpw 
  dos  
end
</code></pre>
<p>in which case default values are used, or it can be specified with
additional parameters, e.g.</p>
<pre><code>nwpw
   dos 0.002 700 -0.80000 0.8000 
end
</code></pre>
<p>The parameters are</p>
<pre><code>nwpw  
 dos  [&lt;alpha&gt; &lt;npoints&gt; &lt;emin&gt; &lt;emax&gt;] 
end
</code></pre>
<p>where</p>
<ul>
<li><code>alpha</code>  value for the broadening the eigenvalues, default
    0.05/27.2116 au</li>
<li><code>npoints</code>  number of plotting points in dos files, default 500</li>
<li><code>emin</code>  minimum energy in dos plots, default min(eigenvalues)-0.1
    au</li>
<li><code>emax</code>  maximimum energy in dos plots, default max(eigenvalues)+0.1
    au</li>
</ul>
<p>The units for dos parameters are in atomic units. Note that if virtual
states are specified in the pspw calculation then the virtual density of
states will also be generated in addition to the filled density of
states.</p>
<p>The following files are generated and written to the permanent_dir for
restricted calculations</p>
<ul>
<li>file_prefix.smear_dos_both - total density of states</li>
<li>file_prefix.smear_fdos_both - density of states of filled states</li>
<li>file_prefix.smear_vdos_both - density of states of virtual states</li>
</ul>
<p>For unrestricted calculations</p>
<ul>
<li>file_prefix.smear_dos_alpha - total density of states for up
    electrons</li>
<li>file_prefix.smear_dos_beta - total density of states for down
    electrons</li>
<li>file_prefix.smear_fdos_alpha - density of states for filled up
    electrons</li>
<li>file_prefix.smear_fdos_beta - density of states for filled down
    electrons</li>
<li>file_prefix.smear_vdos_alpha - density of states for virtual up
    electrons</li>
<li>file_prefix.smear_vdos_beta - density of states for virtual down
    electrons</li>
</ul>
<p>The nwpw:dos:actlist variable is used to specify the atoms used to
determine weights for dos generation. If the variable is not set then
all the atoms are used, e.g.</p>
<p><code>set nwpw:dos:actlist 1 2 3 4</code></p>
<h3 id="plane-wave-density-functional-theory-projected-density-of-states">Projected Density of States<a class="headerlink" href="#plane-wave-density-functional-theory-projected-density-of-states" title="Permanent link">&para;</a></h3>
<p>For projected density of states the &ldquo;Mulliken&rdquo; keyword needs to be set,
e.g.</p>
<pre><code>nwpw 
  Mulliken  
  dos
end
</code></pre>
<p>The following additional files are generated and written to the
permanent_dir for restricted calculations</p>
<ul>
<li>file_prefix.mulliken_dos_both_s - total s projected density of
    restricted states</li>
<li>file_prefix.mulliken_fdos_both_s - s projected density of states
    of filled restricted states</li>
<li>file_prefix.mulliken_vdos_both_s - s projected density of states
    of virtual restricted states</li>
<li>file_prefix.mulliken_dos_both_p - total p projected density of
    states</li>
<li>file_prefix.mulliken_fdos_both_p - p projected density of states
    of filled states</li>
<li>file_prefix.mulliken_vdos_both_p - p projected density of states
    of virtual states</li>
</ul>
<p>&hellip;</p>
<ul>
<li>file_prefix.mulliken_dos_both_all - total of projected density
    of filled and virtual restricted states</li>
<li>file_prefix.mulliken_fdos_both_all - total of projected density
    of filled restricted states</li>
<li>file_prefix.mulliken_vdos_both_all - total of projected density
    of states of virtual restricted states</li>
</ul>
<p>Similarly for unrestricted calculations</p>
<ul>
<li>file_prefix.mulliken_dos_alpha_s - total s projected density of
    up states</li>
<li>file_prefix.mulliken_fdos_alpha_s - s projected density of
    states of filled up states</li>
<li>file_prefix.mulliken_vdos_alpha_s - s projected density of
    states of virtual up states</li>
<li>file_prefix.mulliken_dos_alpha_p - total p projected density of
    up states</li>
<li>file_prefix.mulliken_fdos_alpha_p - p projected density of
    states of filled up states</li>
<li>file_prefix.mulliken_vdos_alpha_p - p projected density of
    states of virtual up states</li>
</ul>
<p>&hellip;</p>
<ul>
<li>file_prefix.mulliken_dos_alpha_all - total of projected density
    of filled up states</li>
<li>file_prefix.mulliken_fdos_alpha_all - total of projected density
    of filled up states</li>
<li>file_prefix.mulliken_vdos_alpha_all - total of projected density
    of states of virtual up states</li>
</ul>
<p>&hellip;</p>
<ul>
<li>file_prefix.mulliken_dos_beta_s - total s projected density of
    down states</li>
<li>file_prefix.mulliken_fdos_beta_s - s projected density of states
    of filled down states</li>
<li>file_prefix.mulliken_vdos_beta_s - s projected density of states
    of virtual down states</li>
<li>file_prefix.mulliken_dos_beta_p - total p projected density of
    down states</li>
<li>file_prefix.mulliken_fdos_beta_p - p projected density of states
    of filled down states</li>
<li>file_prefix.mulliken_vdos_beta_p - p projected density of states
    of virtual down states</li>
</ul>
<p>&hellip;</p>
<ul>
<li>file_prefix.mulliken_dos_beta_all - total of projected density
    of filled down states</li>
<li>file_prefix.mulliken_fdos_beta_all - total of projected density
    of filled down states</li>
<li>file_prefix.mulliken_vdos_beta_all - total of projected density
    of states of virtual down states</li>
</ul>
<h3 id="plane-wave-density-functional-theory-point-charge-analysis">Point Charge Analysis<a class="headerlink" href="#plane-wave-density-functional-theory-point-charge-analysis" title="Permanent link">&para;</a></h3>
<p>The MULLIKEN option can be used to generate derived atomic point charges
from a plane-wave density. This analysis is based on a strategy
suggested in the work of P.E. Blochl, J. Chem. Phys. vol. 103, page 7422
(1995). In this strategy the low-frequency components a plane-wave
density are fit to a linear combination of atom centered Gaussian
functions.</p>
<p>The following SET directives are used to define the
fitting.</p>
<pre><code>set nwpw_APC:Gc &lt;real Gc_cutoff&gt; # specifies the maximum frequency component of the density to be used in the fitting in units of au.  
set nwpw_APC:nga &lt;integer number_gauss&gt; # specifies the the number of Gaussian functions per atom.
set nwpw_APC:gamma &lt;real gamma_list&gt; # specifies the decay lengths of each atom centered Gaussian. 
</code></pre>
<p>We suggest using the following parameters.</p>
<pre><code>set nwpw_APC:Gc 2.5
set nwpw_APC:nga 3 
set nwpw_APC:gamma 0.6 0.9 1.35  
</code></pre>
<h3 id="plane-wave-density-functional-theory-pspw_dplot-generate-gaussian-cube-files">PSPW_DPLOT: Generate Gaussian Cube Files<a class="headerlink" href="#plane-wave-density-functional-theory-pspw_dplot-generate-gaussian-cube-files" title="Permanent link">&para;</a></h3>
<p>The pspw dplot task is used to generate plots of various types of
electron densities (or orbitals) of a molecule. The electron density is
calculated on the specified set of grid points from a PSPW calculation.
The output file generated is in the Gaussian Cube format. Input to the
DPLOT task is contained within the DPLOT sub-block.</p>
<pre><code>NWPW 
...  
 DPLOT   
   ...   
 END  
... 
END
</code></pre>
<p>To run a DPLOT calculation the following directive is used:</p>
<p><code>TASK PSPW PSPW_DPLOT</code></p>
<p>Listed below is the format of a DPLOT
sub-block.</p>
<pre><code>NWPW  
...   
 DPLOT  
   VECTORS &lt;string input_wavefunctions default input_movecs&gt;  
   DENSITY [total||diff||alpha||beta||laplacian||potential default total]   
           &lt;string density_name no default&gt;  
   ELF [restricted|alpha|beta] &lt;string elf_name no default&gt;
   ORBITAL &lt;integer orbital_number no default&gt; &lt;string orbital_name no default&gt;  
   [LIMITXYZ [units &lt;string Units default au&gt;]   
   &lt;real X_From&gt; &lt;real X_To&gt; &lt;integer No_Of_Spacings_X&gt;   
   &lt;real Y_From&gt; &lt;real Y_To&gt; &lt;integer No_Of_Spacings_Y&gt;  
   &lt;real Z_From&gt; &lt;real Z_To&gt; &lt;integer No_Of_Spacings_Z&gt;]  
   NCELL &lt;integer nx default 0&gt; &lt;integer ny default 0&gt; &lt;integer nz default 0&gt;
   POSITION_TOLERANCE &lt;real rtol default 0.001&gt;
 END  
...  
END
</code></pre>
<p>The following list describes the input for the DPLOT sub-block.</p>
<pre><code>VECTORS &lt;string input_wavefunctions default input_movecs&gt;
</code></pre>
<p>This sub-directive specifies the name of the molecular orbital file. If
the second file is optionally given the density is computed as the
difference between the corresponding electron densities. The vector
files have to
match.</p>
<pre><code>DENSITY [total||difference||alpha||beta||laplacian||potential default total] 
        &lt;string density_name no default&gt;
</code></pre>
<p>This sub-directive specifies, what kind of density is to be plotted. The
known names for total, difference, alpha, beta, laplacian, and
potential.</p>
<pre><code>ELF [restricted|alpha|beta] &lt;string elf_name no default&gt;
</code></pre>
<p>This sub-directive specifies that an electron localization function
(ELF) is to be
plotted.</p>
<pre><code>ORBITAL &lt;integer orbital_number no default&gt; &lt;string orbital_name no default&gt;
</code></pre>
<p>This sub-directive specifies the molecular orbital number that is to be
plotted.</p>
<pre><code>LIMITXYZ [units &lt;string Units default angstroms&gt;]  
&lt;real X_From&gt; &lt;real X_To&gt; &lt;integer No_Of_Spacings_X&gt;  
&lt;real Y_From&gt; &lt;real Y_To&gt; &lt;integer No_Of_Spacings_Y&gt;  
&lt;real Z_From&gt; &lt;real Z_To&gt; &lt;integer No_Of_Spacings_Z&gt;
</code></pre>
<p>By default the grid spacing and the limits of the cell to be plotted are
defined by the input wavefunctions. Alternatively the user can use the
LIMITXYZ sub-directive to specify other limits. The grid is generated
using No_Of_Spacings + 1 points along each direction. The known names
for Units are angstroms, au and bohr.</p>
<h2 id="plane-wave-density-functional-theory-band-tasks-multiple-k-point-calculations">Band Tasks: Multiple k-point Calculations<a class="headerlink" href="#plane-wave-density-functional-theory-band-tasks-multiple-k-point-calculations" title="Permanent link">&para;</a></h2>
<p>All input to the Band Tasks is contained within the compound NWPW block,</p>
<pre><code>NWPW   
 ...  
END
</code></pre>
<p>To perform an actual calculation a Task Band directive is used (Section
<a href="#task">Task</a>).</p>
<p><code>Task Band</code></p>
<p>Once a user has specified a geometry, the Band module can be invoked
with no input directives (defaults invoked throughout). There are
sub-directives which allow for customized application; those currently
provided as options for the Band module
are:</p>
<pre><code>NWPW   
 CELL_NAME &lt;string cell_name default cell_default&gt; 
 ZONE_NAME &lt;string zone_name default zone_default&gt;  
 INPUT_WAVEFUNCTION_FILENAME &lt;string input_wavefunctions default input_movecs&gt;  
 OUTPUT_WAVEFUNCTION_FILENAME &lt;string output_wavefunctions default input_movecs&gt;  
 FAKE_MASS &lt;real fake_mass default 400000.0&gt;   
 TIME_STEP &lt;real time_step default 5.8&gt;   
 LOOP &lt;integer inner_iteration outer_iteration default 10 100&gt;   
 TOLERANCES &lt;real tole tolc default 1.0e-7 1.0e-7&gt; 
 CUTOFF &lt;real cutoff&gt;  
 ENERGY_CUTOFF &lt;real ecut default (see input description)&gt; 
 WAVEFUNCTION_CUTOFF &lt;real wcut default (see input description)&gt;  
 EWALD_NCUT &lt;integer ncut default 1&gt;] 
 EWALD_RCUT &lt;real rcut default (see input description)&gt; 

 XC (Vosko      || LDA          || PBE96         || revPBE       || PBEsol          ||  `  
                || HSE  || default Vosko)                                                                `  
  #Note that HSE is the only hybrid functional implemented in BAND

 DFT||ODFT||RESTRICTED||UNRESTRICTED   
 MULT &lt;integer mult default 1&gt;   
 CG  
 LMBFGS  
 SCF [Anderson|| simple || Broyden]  
 [CG || RMM-DIIS] [density || potential]  
 [ALPHA real alpha default 0.25]  
 [ITERATIONS integer inner_iterations default 5]  
 [OUTER_ITERATIONS integer outer_iterations default 0]

 SIMULATION_CELL  [units &lt;string units default bohrs&gt;]
   ... (see input description)   
 END 
 BRILLOUIN_ZONE 
   ... (see input description) 
 END 
 MONKHORST-PACK &lt;real n1 n2 n3 default 1 1 1&gt;
 BAND_DPLOT 
   ... (see input description)  
 END  
 MAPPING &lt;integer mapping default 1&gt;  
 SMEAR &lt;sigma default 0.001&gt; 
 [TEMPERATURE &lt;temperature&gt;] 
 [FERMI || GAUSSIAN || MARZARI-VANDERBILT default FERMI] 
 [ORBITALS &lt;integer orbitals default 4&gt;] 
END 
</code></pre>
<p>The following list describes these keywords.</p>
<ul>
<li><code>cell_name</code> - name of the simulation_cell named <code>cell_name</code>. See
    <a href="#plane-wave-density-functional-theory-simulation-cell">Simulation Cell</a>.</li>
<li><code>input_wavefunctions</code> - name of the file containing one-electron
    orbitals</li>
<li><code>output_wavefunctions</code> - name that will point to file containing the
    one-electron orbitals at the end of the run.</li>
<li><code>fake_mass</code> - value for the electronic fake mass <span class="arithmatex">\(\mu\)</span>. This
    parameter is not presently used in a conjugate gradient simulation</li>
<li><code>time_step</code> - value for the time step <span class="arithmatex">\(\Delta t\)</span>. This parameter
    is not presently used in a conjugate gradient simulation.</li>
<li><code>inner_iteration</code> - number of iterations between the printing out of
    energies and tolerances</li>
<li><code>outer_iteration</code> - number of outer iterations</li>
<li><code>tole</code> - value for the energy tolerance.</li>
<li><code>tolc</code> - value for the one-electron orbital tolerance.</li>
<li><code>cutoff</code> - value for the cutoff energy used to define the
    wavefunction. In addition using the CUTOFF keyword automatically
    sets the cutoff energy for the density to be twice the wavefunction
    cutoff.</li>
<li><code>ecut</code> - value for the cutoff energy used to define the density.
    Default is set to be the maximum value that will fit within the
    simulation_cell <code>cell_name</code>.</li>
<li><code>wcut</code> - value for the cutoff energy used to define the one-electron
    orbitals. Default is set to be the maximum value that will fix
    within the simulation_cell <code>cell_name</code>.</li>
<li><code>ncut</code> - value for the number of unit cells to sum over (in each
    direction) for the real space part of the Ewald summation. Note
    Ewald summation is only used if the simulation_cell is periodic.</li>
<li><code>rcut</code> - value for the cutoff radius used in the Ewald summation.
    Note Ewald summation is only used if the simulation_cell is
    periodic.</li>
</ul>
<p>Default set to be <span class="arithmatex">\(\frac{MIN(\left| \vec{a_i} \right|)}{\pi}, i=1,2,3\)</span>.</p>
<ul>
<li>(Vosko || PBE96 || revPBE) - Choose between Vosko et al&rsquo;s LDA
    parameterization or the orginal and revised Perdew, Burke, and
    Ernzerhof GGA functional.</li>
<li>SIMULATION_CELL (see section -sec:pspw_cell-)</li>
<li>BRILLOUIN_ZONE (see section -sec:band_brillouin_zone-)</li>
<li>MONKHORST-PACK - Alternatively, the MONKHORST-PACK keyword can be
    used to enter a MONKHORST-PACK sampling of the Brillouin zone.</li>
<li><code>smear</code> - value for smearing broadending</li>
<li><code>temperature</code> - same as smear but in units of K.</li>
<li>CG - optional keyword which sets the minimizer to 1</li>
<li>LMBFGS - optional keyword which sets the minimizer to 2</li>
<li>SCF - optional keyword which sets the minimizer to be a band by band
    minimizer. Several options are available for setting the density or
    potential mixing, and the type of Kohn-Sham minimizer.</li>
</ul>
<h3 id="plane-wave-density-functional-theory-brillouin-zone">Brillouin Zone<a class="headerlink" href="#plane-wave-density-functional-theory-brillouin-zone" title="Permanent link">&para;</a></h3>
<p>To supply the special points of the Brillouin zone, the user defines a
brillouin_zone sub-block within the NWPW block. Listed below is the
format of a brillouin_zone
sub-block.</p>
<pre><code>NWPW  
...   
 BRILLOUIN_ZONE 
   ZONE_NAME &lt;string name default zone_default&gt; 
   (KVECTOR &lt;real k1 k2 k3 no default&gt; &lt;real weight default (see input description)&gt;   
    ...)   
 END  
...  
END
</code></pre>
<p>The user enters the special points and weights of the Brillouin zone.
The following list describes the input in detail.</p>
<ul>
<li><code>name</code> - user-supplied name for the simulation block.</li>
<li><code>k1 k2 k3</code> - user-supplied values for a special point in the
    Brillouin zone.</li>
<li><code>weight</code> - user-supplied weight. Default is to set the weight so
    that the sum of all the weights for the entered special points adds
    up to unity.</li>
</ul>
<h4 id="plane-wave-density-functional-theory-band-structure-paths">Band Structure Paths<a class="headerlink" href="#plane-wave-density-functional-theory-band-structure-paths" title="Permanent link">&para;</a></h4>
<p>SC: gamma, m, r, x</p>
<p>FCC: gamma, k, l, u, w, x</p>
<p>BCC: gamma, h, n, p</p>
<p>Rhombohedral: not currently implemented</p>
<p>Hexagonal: gamma, a, h, k, l, m</p>
<p>Simple Tetragonal: gamma, a, m, r, x, z</p>
<p>Simple Orthorhombic: gamma, r, s, t, u, x, y, z</p>
<p>Body-Centered Tetragonal: gamma, m, n, p, x</p>
<h4 id="plane-wave-density-functional-theory-special-points-of-different-space-groups-conventional-cells">Special Points of Different Space Groups (Conventional Cells)<a class="headerlink" href="#plane-wave-density-functional-theory-special-points-of-different-space-groups-conventional-cells" title="Permanent link">&para;</a></h4>
<p>(1) P1</p>
<p>(2) P-1</p>
<p>(3)</p>
<h3 id="plane-wave-density-functional-theory-screened-exchange">Screened Exchange<a class="headerlink" href="#plane-wave-density-functional-theory-screened-exchange" title="Permanent link">&para;</a></h3>
<h3 id="plane-wave-density-functional-theory-density-of-states-and-projected-density-of-states">Density of States and Projected Density of States<a class="headerlink" href="#plane-wave-density-functional-theory-density-of-states-and-projected-density-of-states" title="Permanent link">&para;</a></h3>
<p>The &ldquo;dos&rdquo; option is used to calculate density of states using broadening
of the eigenvalues . This option can be specified without additional
parameters, i.e.</p>
<pre><code>nwpw  
  dos
end
</code></pre>
<p>in which case default values are used, or it can be specified with
additional parameters, e.g.</p>
<pre><code>nwpw  
   dos 0.002 700 -0.80000 0.8000
end
</code></pre>
<p>The parameters are</p>
<pre><code>nwpw 
 dos  [&lt;alpha&gt; &lt;npoints&gt; &lt;emin&gt; &lt;emax&gt;]
end
</code></pre>
<p>where</p>
<ul>
<li><code>alpha</code> - value for the broadening the eigenvalues, default
    0.05/27.2116 au</li>
<li><code>npoints</code> - number of plotting points in dos files, default 500</li>
<li><code>emin</code> - minimum energy in dos plots, default min(eigenvalues)-0.1
    au</li>
<li><code>emax</code> - maximimum energy in dos plots, default max(eigenvalues)+0.1
    au</li>
</ul>
<p>The units for dos parameters are in atomic units. Note that if virtual
states are specified in the pspw calculation then the virtual density of
states will also be generated in addition to the filled density of
states.</p>
<p>The following files are generated and written to the permanent_dir for
restricted calculations</p>
<ul>
<li>file_prefix.smear_dos_both - total density of states</li>
<li>file_prefix.smear_fdos_both - density of states of filled states</li>
<li>file_prefix.smear_vdos_both - density of states of virtual states</li>
</ul>
<p>For unrestricted calculations</p>
<ul>
<li>file_prefix.smear_dos_alpha - total density of states for up
    electrons</li>
<li>file_prefix.smear_dos_beta - total density of states for down
    electrons</li>
<li>file_prefix.smear_fdos_alpha - density of states for filled up
    electrons</li>
<li>file_prefix.smear_fdos_beta - density of states for filled down
    electrons</li>
<li>file_prefix.smear_vdos_alpha - density of states for virtual up
    electrons</li>
<li>file_prefix.smear_vdos_beta - density of states for virtual down
    electrons</li>
</ul>
<p>The nwpw:dos:actlist variable is used to specify the atoms used to
determine weights for dos generation. If the variable is not set then
all the atoms are used, e.g.</p>
<p><code>set nwpw:dos:actlist 1 2 3 4</code></p>
<p>For projected density of states the &ldquo;Mulliken&rdquo; keyword needs to be set,
e.g.</p>
<pre><code>nwpw
  Mulliken
  dos 
end
</code></pre>
<p>The following additional files are generated and written to the
permanent_dir for restricted calculations</p>
<ul>
<li>file_prefix.mulliken_dos_both_s - total s projected density of
    restricted states</li>
<li>file_prefix.mulliken_fdos_both_s - s projected density of states
    of filled restricted states</li>
<li>file_prefix.mulliken_vdos_both_s - s projected density of states
    of virtual restricted states</li>
<li>file_prefix.mulliken_dos_both_p - total p projected density of
    states</li>
<li>file_prefix.mulliken_fdos_both_p - p projected density of states
    of filled states</li>
<li>file_prefix.mulliken_vdos_both_p - p projected density of states
    of virtual states</li>
</ul>
<p>&hellip;</p>
<ul>
<li>file_prefix.mulliken_dos_both_all - total of projected density
    of filled and virtual restricted states</li>
<li>file_prefix.mulliken_fdos_both_all - total of projected density
    of filled restricted states</li>
<li>file_prefix.mulliken_vdos_both_all - total of projected density
    of states of virtual restricted states</li>
</ul>
<p>Similarly for unrestricted calculations</p>
<ul>
<li>file_prefix.mulliken_dos_alpha_s - total s projected density of
    up states</li>
<li>file_prefix.mulliken_fdos_alpha_s - s projected density of
    states of filled up states</li>
<li>file_prefix.mulliken_vdos_alpha_s - s projected density of
    states of virtual up states</li>
<li>file_prefix.mulliken_dos_alpha_p - total p projected density of
    up states</li>
<li>file_prefix.mulliken_fdos_alpha_p - p projected density of
    states of filled up states</li>
<li>file_prefix.mulliken_vdos_alpha_p - p projected density of
    states of virtual up states</li>
</ul>
<p>&hellip;</p>
<ul>
<li>file_prefix.mulliken_dos_alpha_all - total of projected density
    of filled up states</li>
<li>file_prefix.mulliken_fdos_alpha_all - total of projected density
    of filled up states</li>
<li>file_prefix.mulliken_vdos_alpha_all - total of projected density
    of states of virtual up states</li>
</ul>
<p>&hellip;</p>
<ul>
<li>file_prefix.mulliken_dos_beta_s - total s projected density of
    down states</li>
<li>file_prefix.mulliken_fdos_beta_s - s projected density of states
    of filled down states</li>
<li>file_prefix.mulliken_vdos_beta_s - s projected density of states
    of virtual down states</li>
<li>file_prefix.mulliken_dos_beta_p - total p projected density of
    down states</li>
<li>file_prefix.mulliken_fdos_beta_p - p projected density of states
    of filled down states</li>
<li>file_prefix.mulliken_vdos_beta_p - p projected density of states
    of virtual down states</li>
</ul>
<p>&hellip;</p>
<ul>
<li>file_prefix.mulliken_dos_beta_all - total of projected density
    of filled down states</li>
<li>file_prefix.mulliken_fdos_beta_all - total of projected density
    of filled down states</li>
<li>file_prefix.mulliken_vdos_beta_all - total of projected density
    of states of virtual down states</li>
</ul>
<h3 id="plane-wave-density-functional-theory-two-component-wavefunctions-spin-orbit-zora">Two-Component Wavefunctions (Spin-Orbit ZORA)<a class="headerlink" href="#plane-wave-density-functional-theory-two-component-wavefunctions-spin-orbit-zora" title="Permanent link">&para;</a></h3>
<h3 id="plane-wave-density-functional-theory-band_dplot-generate-gaussian-cube-files">BAND_DPLOT: Generate Gaussian Cube Files<a class="headerlink" href="#plane-wave-density-functional-theory-band_dplot-generate-gaussian-cube-files" title="Permanent link">&para;</a></h3>
<p>The BAND BAND_DPLOT task is used to generate plots of various types of
electron densities (or orbitals) of a crystal. The electron density is
calculated on the specified set of grid points from a Band calculation.
The output file generated is in the Gaussian Cube format. Input to the
BAND_DPLOT task is contained within the BAND_DPLOT sub-block.</p>
<pre><code>NWPW   
...  
 BAND_DPLOT 
   ... 
 END 
...
END
</code></pre>
<p>To run a BAND_DPLOT calculation the following directive is used:</p>
<p><code>TASK BAND BAND_DPLOT</code></p>
<p>Listed below is the format of a BAND_DPLOT
sub-block.</p>
<pre><code>NWPW
... 
 BAND_DPLOT 
   VECTORS &lt;string input_wavefunctions default input_movecs&gt;
   DENSITY [total||difference||alpha||beta||laplacian||potential default total] &lt;string density_name no default&gt;
   ELF [restricted|alpha|beta] &lt;string elf_name no default&gt;  
   ORBITAL (density || real || complex default density)  
           &lt;integer orbital_number no default&gt; 
           &lt;integer brillion_number default 1&gt; 
           &lt;string orbital_name no default&gt;  
   [LIMITXYZ [units &lt;string Units default angstroms&gt;]   
    &lt;real X_From&gt; &lt;real X_To&gt; &lt;integer No_Of_Spacings_X&gt;  
    &lt;real Y_From&gt; &lt;real Y_To&gt; &lt;integer No_Of_Spacings_Y&gt;  
    &lt;real Z_From&gt; &lt;real Z_To&gt; &lt;integer No_Of_Spacings_Z&gt;] 
 END
...
END
</code></pre>
<p>The following list describes the input for the BAND_DPLOT sub-block.</p>
<p><code>VECTORS &lt;string input_wavefunctions default input_movecs&gt;</code></p>
<p>This sub-directive specifies the name of the molecular orbital file. If
the second file is optionally given the density is computed as the
difference between the corresponding electron densities. The vector
files have to
match.</p>
<pre><code>DENSITY [total||difference||alpha||beta||laplacian||potential default total] &lt;string density_name no default&gt;
</code></pre>
<p>This sub-directive specifies, what kind of density is to be plotted. The
known names for total, difference, alpha, beta, laplacian, and
potential.</p>
<pre><code>ELF [restricted|alpha|beta] &lt;string elf_name no default&gt;
</code></pre>
<p>This sub-directive specifies that an electron localization function
(ELF) is to be
plotted.</p>
<pre><code>ORBITAL (density || real || complex default density) &lt;integer orbital_number no default&gt;&lt;integer brillion_number default 1&gt; &lt;string orbital_name no default&gt;
</code></pre>
<p>This sub-directive specifies the molecular orbital number that is to be
plotted.</p>
<pre><code>LIMITXYZ [units &lt;string Units default angstroms&gt;] 
&lt;real X_From&gt; &lt;real X_To&gt; &lt;integer No_Of_Spacings_X&gt;  
&lt;real Y_From&gt; &lt;real Y_To&gt; &lt;integer No_Of_Spacings_Y&gt;  
&lt;real Z_From&gt; &lt;real Z_To&gt; &lt;integer No_Of_Spacings_Z&gt;
</code></pre>
<p>By default the grid spacing and the limits of the cell to be plotted are
defined by the input wavefunctions. Alternatively the user can use the
LIMITXYZ sub-directive to specify other limits. The grid is generated
using No_Of_Spacings + 1 points along each direction. The known names
for Units are angstroms, au and bohr.</p>
<h2 id="plane-wave-density-functional-theory-car-parrinello">Car-Parrinello<a class="headerlink" href="#plane-wave-density-functional-theory-car-parrinello" title="Permanent link">&para;</a></h2>
<p>The Car-Parrinello task is used to perform ab initio molecular dynamics
using the scheme developed by Car and Parrinello. In this unified ab
initio molecular dynamics scheme the motion of the ion cores is coupled
to a fictitious motion for the Kohn-Sham orbitals of density functional
theory. Constant energy or constant temperature simulations can be
performed. A detailed description of this method is described in section
<a href="#plane-wave-density-functional-theory-car-parrinello-scheme-for-ab-initio-molecular-dynamics">Car-Parrinello Scheme for Ab Initio Molecular
Dynamics</a>.</p>
<p>Input to the Car-Parrinello simulation is contained within the
Car-Parrinello sub-block.</p>
<pre><code>NWPW 
... 
 Car-Parrinello  
   ...  
 END  
...
END
</code></pre>
<p>To run a Car-Parrinello calculation the following directives are used:</p>
<pre><code>TASK PSPW Car-Parrinello   
 TASK BAND Car-Parrinello
 TASK PAW Car-Parrinello
</code></pre>
<p>The Car-Parrinello sub-block contains a great deal of input, including
pointers to data, as well as parameter input. Listed below is the format
of a Car-Parrinello
sub-block.</p>
<pre><code>NWPW  
...   
 Car-Parrinello   
   CELL_NAME &lt;string cell_name default 'cell_default'&gt;
   INPUT_WAVEFUNCTION_FILENAME &lt;string input_wavefunctions default file_prefix.movecs&gt;  
   OUTPUT_WAVEFUNCTION_FILENAME &lt;string output_wavefunctions default file_prefix.movecs&gt;   
   INPUT_V_WAVEFUNCTION_FILENAME &lt;string input_v_wavefunctions default file_prefix.vmovecs&gt; 
   OUTPUT_V_WAVEFUNCTION_FILENAME &lt;string output_v_wavefunctions default file_prefix.vmovecs&gt;  
   FAKE_MASS &lt;real fake_mass default default 1000.0&gt;
   TIME_STEP &lt;real time_step default 5.0&gt;  
   LOOP &lt;integer inner_iteration outer_iteration default 10 1&gt; 
   SCALING &lt;real scale_c scale_r default 1.0 1.0&gt;   
   ENERGY_CUTOFF &lt;real ecut default (see input description)&gt;   
   WAVEFUNCTION_CUTOFF &lt;real wcut default (see input description)&gt;   
   EWALD_NCUT &lt;integer ncut default 1&gt;   
   EWALD_RCUT &lt;real rcut default (see input description)&gt;   
    XC (Vosko      || LDA          || PBE96         || revPBE       || HF            ||    
        LDA-SIC    || LDA-SIC/2    || LDA-0.4SIC    || LDA-SIC/4    || LDA-0.2SIC    ||   
        PBE96-SIC  || PBE96-SIC/2  || PBE96-0.4SIC  || PBE96-SIC/4  || PBE96-0.2SIC  ||   
        revPBE-SIC || revPBE-SIC/2 || revPBE-0.4SIC || revPBE-SIC/4 || revPBE-0.2SIC ||   
        PBE0       || revPBE0      || default Vosko) 
   [Nose-Hoover &lt;real Period_electron real Temperature_electron 
                 real Period_ion      real Temperature_ion     
                 integer Chainlength_electron integer Chainlength_ion default 100.0 298.15 100.0 298.15 1 1&gt;]   
   [TEMPERATURE &lt;real Temperature_ion      real Period_ion 
                 real Temperature_electron real Period_electron  
                 integer Chainlength_ion integer Chainlength_electron default 298.15 1200 298.15 1200.0 1 1&gt;]   
   [SA_decay &lt;real sa_scale_c sa_scale_r default 1.0 1.0&gt;] 
   XYZ_FILENAME &lt;string xyz_filename default file_prefix.xyz&gt; 
   ION_MOTION_FILENAME &lt;string ion_motion_filename default file_prefix.ion_motion
   EMOTION_FILENAME &lt;string emotion_filename default file_prefix.emotion&gt; 
   HMOTION_FILENAME &lt;string hmotion_filename nodefault&gt;
   OMOTION_FILENAME &lt;string omotion_filename nodefault&gt;
   EIGMOTION_FILENAME &lt;string eigmotion_filename nodefault&gt;  
 END  
...  
END
</code></pre>
<p>The following list describes the input for the Car-Parrinello sub-block.</p>
<ul>
<li><code>cell_name</code> - name of the the simulation_cell named <code>cell_name</code>.
    See section <a href="#plane-wave-density-functional-theory-simulation-cell">Simulation Cell</a>.</li>
<li><code>input_wavefunctions</code> - name of the file containing one-electron
    orbitals</li>
<li><code>output_wavefunctions</code> - name of the file that will contain the
    one-electron orbitals at the end of the run.</li>
<li><code>input_v_wavefunctions</code> - name of the file containing one-electron
    orbital velocities.</li>
<li><code>output_v_wavefunctions</code> - name of the file that will contain the
    one-electron orbital velocities at the end of the run.</li>
<li><code>fake_mass</code> - value for the electronic fake mass (<span class="arithmatex">\(\mu\)</span> ).</li>
<li><code>time_step</code> - value for the Verlet integration time step
    (<span class="arithmatex">\(Delta t\)</span>).</li>
<li><code>inner_iteration</code> - number of iterations between the printing out of
    energies.</li>
<li><code>outer_iteration</code> - number of outer iterations</li>
<li><code>scale_c</code> - value for the initial velocity scaling of the
    one-electron orbital velocities.</li>
<li><code>scale_r</code> - value for the initial velocity scaling of the ion
    velocities.</li>
<li><code>ecut</code> - value for the cutoff energy used to define the density.
    Default is set to be the maximum value that will fit within the
    simulation_cell <code>cell_name</code>.</li>
<li><code>wcut</code> - value for the cutoff energy used to define the one-electron
    orbitals. Default is set to be the maximum value that will fit
    within the simulation_cell <code>cell_name</code>.</li>
<li><code>ncut</code> - value for the number of unit cells to sum over (in each
    direction) for the real space part of the Ewald summation. Note
    Ewald summation is only used if the simulation_cell is periodic.</li>
<li><code>rcut</code> - value for the cutoff radius used in the Ewald summation.
    Note Ewald summation is only used if the simulation_cell is
    periodic.</li>
</ul>
<p>Default set to be <span class="arithmatex">\(\frac{MIN(\left| \vec{a_i} \right|)}{\pi}, i=1,2,3\)</span>.</p>
<ul>
<li>(Vosko || PBE96 || revPBE || &hellip;) - Choose between Vosko et al&rsquo;s LDA
    parameterization or the orginal and revised Perdew, Burke, and
    Ernzerhof GGA functional. In addition, several hybrid options.</li>
<li>Nose-Hoover or Temperature - optional subblock which if specified
    causes the simulation to perform Nose-Hoover dynamics. If this
    subblock is not specified the simulation performs constant energy
    dynamics. See section -sec:pspw_nose- for a description of the
    parameters. Note that the Temperature subblock is just a reordering
    of the Nose-Hoover subblock.<ul>
<li><code>Period_electron</code> <span class="arithmatex">\(\equiv P_{electron}\)</span> - estimated period for
    fictitious electron thermostat.</li>
<li><code>Temperature_electron</code> <span class="arithmatex">\(\equiv T_{electron}\)</span> - temperature for
    fictitious electron motion</li>
<li><code>Period_ion</code> <span class="arithmatex">\(\equiv P_{ion}\)</span> - estimated period for ionic
    thermostat</li>
<li><code>Temperature_ion</code> <span class="arithmatex">\(\equiv T_{ion}\)</span> - temperature for ion
    motion</li>
<li><code>Chainlength_electron</code> - number of electron thermostat chains</li>
<li><code>Chainlength_ion</code> - number of ion thermostat chains</li>
</ul>
</li>
<li>SA_decay - optional subblock which if specified causes the
    simulation to run a simulated annealing simulation. For simulated
    annealing to work the Nose-Hoover subblock needs to be specified.
    The initial temperature are taken from the Nose-Hoover subblock. See
    section -sec:pspw_nose- for a description of the parameters.<ul>
<li><code>sa_scale_c</code> <span class="arithmatex">\(\equiv \tau_{electron}\)</span> - decay rate in atomic
    units for electronic temperature.</li>
<li><code>sa_scale_r</code> <span class="arithmatex">\(\equiv \tau_{ionic}\)</span> - decay rate in atomic
    units for the ionic temperature.  </li>
</ul>
</li>
<li><code>xyz_filename</code> - name of the XYZ motion file generated</li>
<li><code>emotion_filename</code> - name of the emotion motion file. See section
    <a href="#plane-wave-density-functional-theory-emotion-motion-file">EMOTION motion file</a> for a
    description of the datafile.</li>
<li><code>hmotion_filenameh</code> - name of the hmotion motion file. See section
    <a href="#plane-wave-density-functional-theory-hmotion-motion-file">HMOTION motion file</a> for a
    description of the datafile.</li>
<li><code>eigmotion_filename</code> - name of the eigmotion motion file. See
    section <a href="#plane-wave-density-functional-theory-eigmotion-motion-file">EIGMOTION motion file</a>
    for a description of the datafile.</li>
<li><code>ion_motion_filename</code> - name of the ion_motion motion file. See
    section <a href="#plane-wave-density-functional-theory-ion_motion-motion-file">ION_MOTION motion file</a>- for a description of the
    datafile.</li>
<li>MULLIKEN - optional keyword which if specified causes an omotion
    motion file to be created.</li>
<li><code>omotion_filename</code> - name of the omotion motion file. See section
    <a href="#plane-wave-density-functional-theory-omotion-motion-file">OMOTION motion file</a> for a description of
    the datafile.</li>
</ul>
<p>When a DPLOT sub-block is specified the following SET directive can be
used to output dplot data during a PSPW Car-Parrinello simulation:</p>
<pre><code>set pspw_dplot:iteration_list &lt;integer list_of_iteration_numbers&gt;
</code></pre>
<p>The Gaussian cube files specified in the DPLOT sub-block are appended
with the specified iteration number.</p>
<p>For example, the following directive specifies that at the
3,10,11,12,13,14,15, and 50 iterations Gaussian cube files are to be
produced.</p>
<p><code>set pspw_dplot:iteration_list 3,10:15,50</code></p>
<h3 id="plane-wave-density-functional-theory-adding-geometry-constraints-to-a-car-parrinello-simulation">Adding Geometry Constraints to a Car-Parrinello Simulation<a class="headerlink" href="#plane-wave-density-functional-theory-adding-geometry-constraints-to-a-car-parrinello-simulation" title="Permanent link">&para;</a></h3>
<p>The Car-Parrinello module allows users to freeze the cartesian
coordinates in a simulation (Note - the Car-Parrinello code recognizes
Cartesian constraints, but it does not recognize internal coordinate
constraints). The +SET+ directive (Section <a href="#zcoord-forcing-internal-coordinates-applying-constraints-in-geometry-optimizations">Applying constraints in
geometry
optimizations</a>) 
is used to freeze atoms, by specifying a directive of the form:</p>
<pre><code>set geometry:actlist &lt;integer list_of_center_numbers&gt;
</code></pre>
<p>This defines only the centers in the list as active. All other centers
will have zero force assigned to them, and will remain frozen at their
starting coordinates during a Car-Parrinello simulation.</p>
<p>For example, the following directive specifies that atoms numbered 1, 5,
6, 7, 8, and 15 are active and all other atoms are frozen:</p>
<p><code>set geometry:actlist 1 5:8 15</code></p>
<p>or equivalently,</p>
<p><code>set geometry:actlist 1 5 6 7 8 15</code></p>
<p>If this option is not specified by entering a +SET+ directive, the
default behavior in the code is to treat all atoms as active. To revert
to this default behavior after the option to define frozen atoms has
been invoked, the +UNSET+ directive must be used (since the database is
persistent, see Section <a href="#nwchem-architecture">NWChem Architecture</a>).
The form of the +UNSET+ directive is as follows:</p>
<p><code>unset geometry:actlist</code></p>
<p>In addition, the Car-Parrinello module allows users to freeze bond
lengths via a Shake algorithm. The following +SET+ directive shows how
to do this.</p>
<p><code>set nwpw:shake_constraint "2 6 L 6.9334"</code></p>
<p>This input fixes the bond length between atoms 2 and 6 to be 6.9334
bohrs. Note that this input only recognizes bohrs.</p>
<p>When using constraints it is usually necessary to turn off center of
mass shifting. This can be done by the following +SET+ directive.</p>
<p><code>set nwpw:com_shift .false.</code></p>
<h3 id="plane-wave-density-functional-theory-car-parrinello-output-datafiles">Car-Parrinello Output Datafiles<a class="headerlink" href="#plane-wave-density-functional-theory-car-parrinello-output-datafiles" title="Permanent link">&para;</a></h3>
<h4 id="plane-wave-density-functional-theory-xyz-motion-file">XYZ motion file<a class="headerlink" href="#plane-wave-density-functional-theory-xyz-motion-file" title="Permanent link">&para;</a></h4>
<p>Data file that stores ion positions and velocities as a function of time
in XYZ
format.</p>
<pre><code>[line 1: ] n_ion
[line 2: ] do ii=1,n_ion
[line 2+ii: ] atom_name(ii), x(ii),y(ii),z(ii),vx(ii),vy(ii),vz(ii)
end do 
[line n_ion+3 ] n_nion 
 do ii=1,n_ion
[line n_ion+3+ii: ] atom_name(ii), x(ii),y(ii),z(ii), vx(ii),vy(ii),vz(ii) 
end do
[line 2*n_ion+4: ] ....
</code></pre>
<h4 id="plane-wave-density-functional-theory-ion_motion-motion-file">ION_MOTION motion file<a class="headerlink" href="#plane-wave-density-functional-theory-ion_motion-motion-file" title="Permanent link">&para;</a></h4>
<p>Datafile that stores ion positions and velocities as a function of
time</p>
<pre><code>[line 1: ] it_out, n_ion, omega, a1.x,a1.y,a1.z, a2.x,a2,y,a2.z, a3.x,a3.y,a3.z 
[line 2: ] time 
do ii=1,n_ion
[line 2+ii: ] ii, atom_symbol(ii),atom_name(ii), x(ii),y(ii),z(ii), vx(ii),vy(ii),vz(ii)  
end do
[line n_ion+3 ] time 
do do ii=1,n_ion  
[line n_ion+3+ii: ] ii, atom_symbol(ii),atom_name(ii), x(ii),y(ii),z(ii), vx(ii),vy(ii),vz(ii) 
end do 
[line 2*n_ion+4: ] ....
</code></pre>
<h4 id="plane-wave-density-functional-theory-emotion-motion-file">EMOTION motion file<a class="headerlink" href="#plane-wave-density-functional-theory-emotion-motion-file" title="Permanent link">&para;</a></h4>
<p>Datafile that store energies as a function of
time.</p>
<pre><code>[line 1: ] time, E1,E2,E3,E4,E5,E6,E7,E8,(E9,E10, if Nose-Hoover),eave,evar,have,hvar,ion_Temp 
[line 2: ] ...
</code></pre>
<p>where</p>
<pre><code>E1 = total energy
E2 = potential energy
E3 = ficticious kinetic energy
E4 = ionic kinetic energy
E5 = orbital energy
E6 = hartree energy
E7 = exchange-correlation energy 
E8 = ionic energy
eave = average potential energy 
evar = variance of potential energy
have = average total energy
hvar = variance of total energy
ion_Temp = temperature
</code></pre>
<h4 id="plane-wave-density-functional-theory-hmotion-motion-file">HMOTION motion file<a class="headerlink" href="#plane-wave-density-functional-theory-hmotion-motion-file" title="Permanent link">&para;</a></h4>
<p>Datafile that stores the rotation matrix as a function of time.</p>
<pre><code>[line 1: ] time
[line 2: ] ms,ne(ms),ne(ms)
do i=1,ne(ms)
[line 2+i: ] (hml(i,j), j=1,ne(ms)
end do
[line 3+ne(ms): ] time
[line 4+ne(ms): ] ....
</code></pre>
<h4 id="plane-wave-density-functional-theory-eigmotion-motion-file">EIGMOTION motion file<a class="headerlink" href="#plane-wave-density-functional-theory-eigmotion-motion-file" title="Permanent link">&para;</a></h4>
<p>Datafile that stores the eigenvalues for the one-electron orbitals as a
function of time.</p>
<pre><code>[line 1: ] time, (eig(i), i=1,number_orbitals) 
[line 2: ] ...
</code></pre>
<h4 id="plane-wave-density-functional-theory-omotion-motion-file">OMOTION motion file<a class="headerlink" href="#plane-wave-density-functional-theory-omotion-motion-file" title="Permanent link">&para;</a></h4>
<p>Datafile that stores a reduced representation of the one-electron
orbitals. To be used with a molecular orbital viewer that will be ported
to NWChem in the near future.</p>
<h2 id="plane-wave-density-functional-theory-born-oppenheimer-molecular-dynamics">Born-Oppenheimer Molecular Dynamics<a class="headerlink" href="#plane-wave-density-functional-theory-born-oppenheimer-molecular-dynamics" title="Permanent link">&para;</a></h2>
<pre><code>NWPW
...
  BO_STEPS &lt;integer bo_inner_iteration bo_outer_iteration default 10 100&gt; 
  BO_TIME_STEP  &lt;real bo_time_step default 5.0&gt; 
  BO_ALGORITHM [verlet|| velocity-verlet || leap-frog]
  BO_FAKE_MASS &lt;real bo_fake_mass default 500.0&gt; 
END
</code></pre>
<h2 id="plane-wave-density-functional-theory-i-pi-socket-communication">i-PI Socket Communication<a class="headerlink" href="#plane-wave-density-functional-theory-i-pi-socket-communication" title="Permanent link">&para;</a></h2>
<pre><code>NWPW
   SOCKET (UNIX || IPI_CLIENT) &lt;string socketname default (see input description)&gt;
END
</code></pre>
<p>The <code>NWPW</code> module provides native communication via the i-PI socket protocol.
The behavior is identical to the
<a href="#geometry-optimization-i-pi-socket-communication">i-PI socket communication</a>
provided by the <code>DRIVER</code> module.
The <code>NWPW</code> implementation of the <code>SOCKET</code> directive is better optimized for
plane-wave calculations.</p>
<p>For proper behavior, the <code>TASK</code> directive should be set to <code>GRADIENT</code>,
e.g. <code>TASK PSPW GRADIENT</code> or <code>TASK BAND GRADIENT</code>.</p>
<h2 id="plane-wave-density-functional-theory-metropolis-monte-carlo">Metropolis Monte-Carlo<a class="headerlink" href="#plane-wave-density-functional-theory-metropolis-monte-carlo" title="Permanent link">&para;</a></h2>
<pre><code>NWPW
...
  MC_STEPS &lt;integer mc_inner_iteration mc_outer_iteration default 10 100&gt; 
END
</code></pre>
<h2 id="plane-wave-density-functional-theory-free-energy-simulations">Free Energy Simulations<a class="headerlink" href="#plane-wave-density-functional-theory-free-energy-simulations" title="Permanent link">&para;</a></h2>
<h3 id="plane-wave-density-functional-theory-metadynamics">MetaDynamics<a class="headerlink" href="#plane-wave-density-functional-theory-metadynamics" title="Permanent link">&para;</a></h3>
<p><img alt="Metadynamics bias potential. Courtesy of Raymond Atta-Fynn" src="ray2.gif" title="Metadynamics bias potential. Courtesy of Raymond Atta-Fynn&quot;.md) ![Metadynamics simulation of the first hydrolysis of U(IV) from 1.  Courtesy of Raymond Atta-Fynn](ray.gif &quot;Metadynamics simulation of the first hydrolysis of U(IV.md) from 1. Courtesy of Raymond Atta-Fynn" />
Metadynamics<a href="http://en.wikipedia.org/wiki/Metadynamics">2</a><a href="http://iopscience.iop.org/0034-4885/71/12/126601">3</a><a href="http://people.sissa.it/~laio/Research/Res_metadynamics.php">4</a>
is a powerful, non-equilibrium molecular dynamics method which
accelerates the sampling of the multidimensional free energy surfaces of
chemical reactions. This is achieved by adding an external
time-dependent bias potential that is a function of user defined
collective variables, <span class="arithmatex">\(\mathbf{s}\)</span>. The bias potential discourages the
system from sampling previously visited values of <span class="arithmatex">\(\mathbf{s}\)</span> (i.e.,
encourages the system to explore new values of <span class="arithmatex">\(\mathbf{s}\)</span>. During the
simulation the bias potential accumulates in low energy wells which then
allows the system to cross energy barriers much more quickly than would
occur in standard dynamics. The collective variable <span class="arithmatex">\(\mathbf{s}\)</span> is a
generic function of the system coordinates, <span class="arithmatex">\(\mathbf{R}\)</span> (e.g. bond
distance, bond angle, coordination numbers, etc.) that is capable of
describing the chemical reaction of interest.
<span class="arithmatex">\(\mathbf{s}\left(\mathbf{R}\right)\)</span> can be regarded as a reaction
coordinate if it can distinguish between the reactant, transition, and
products states, and also capture the kinetics of the reaction.</p>
<p>The biasing is achieved by flooding the energy landscape with
repulsive Gaussian hills centered on the current location of
<span class="arithmatex">\(\mathbf{s}\left(\mathbf{R}\right)\)</span> at a constant time interval
<span class="arithmatex">\(\Delta t\)</span>. If the height of the Gaussians is constant in time then we
have standard metadynamics; if the heights vary (slowly decreased) over
time then we have well-tempered metadynamics. In between the addition of
Gaussians, the system is propagated by normal (but out of equilibrium)
dynamics. Suppose that the dimension of the collective space is <span class="arithmatex">\(d\)</span>,
i.e.
<span class="arithmatex">\(\mathbf{s}\left(\mathbf{R}\right)=\bigl[s_1\left(\mathbf{R}\right),s_2\left(\mathbf{R}\right),\ldots,s_d\left(\mathbf{R}\right)\bigr]\)</span>
and that prior to any time <span class="arithmatex">\(t\)</span> during the simulation, <span class="arithmatex">\(N +1\)</span>
Gaussians centered on 
<span class="arithmatex">\(\mathbf{S}^{t_g}\)</span>
are deposited along the
trajectory of <span class="arithmatex">\(\mathbf{s}\left(\mathbf{R}\right)\)</span> at times
<span class="arithmatex">\(t_g = 0, \Delta t, 2\Delta t, \ldots ,N\Delta t\)</span>. Then, the value of
the bias potential, <span class="arithmatex">\(V\)</span>, at an arbitrary point,
<span class="arithmatex">\(\mathbf{s}\left(\mathbf{R}\right)=\bigl[s_1\left(\mathbf{R}\right),s_2\left(\mathbf{R}\right),\ldots,s_d\left(\mathbf{R}\right)\bigr]\)</span>,
along the trajectory of <span class="arithmatex">\(\mathbf{s}\left(\mathbf{R}\right)\)</span> at time <span class="arithmatex">\(t\)</span>
is given
by</p>
<div class="arithmatex">\[V_{meta}\left(\pmb{s},t\right) = \sum_{t_g=0}^{t_g&lt;} W(t) \exp\left(-\sum_{i=1}^{d}\frac{({s_i}-{s_i}^{t_g})^2}{2\sigma_i^2}\right)\]</div>
<p>where
<span class="arithmatex">\(W(t)=W_0 \exp\left(-\frac{V_{meta}\left(\mathbf{s},t-\Delta t\right)}{k_B T_{tempered}}\right)\)</span>
is the time-dependent Gaussian height. <span class="arithmatex">\(\sigma_i \,(i=1,2,\ldots,d)\)</span>
and <span class="arithmatex">\(W_0\)</span> are width and initial height respectively of Gaussians, and
<span class="arithmatex">\(T_{tempered}\)</span>
is the tempered metadynamics temperature.
<span class="arithmatex">\(T_{tempered}=0\)</span>
corresponds to standard molecular dynamics because
<span class="arithmatex">\(W(t)=0\)</span>
and therfore there is no bias. <span class="arithmatex">\(T_{tempered}=\infty\)</span>
corresponds to standard metadynamics since in this case
<span class="arithmatex">\(W(t)=W_0\)</span>=constant. A positive, finite value of <span class="arithmatex">\(T_{tempered}\)</span> (eg.
<span class="arithmatex">\(T_{tempered}\)</span> &gt;=1500 K) corresponds to <em>well-tempered</em> metadynamics in which  <span class="arithmatex">\(0 &lt; W(t)&lt;= W_0\)</span>.</p>
<p>For sufficiently large <span class="arithmatex">\(t\)</span>, the history potential
<span class="arithmatex">\(V_{meta}\left(\mathbf{s},t\right)\)</span> will nearly flatten the free energy
surface, <span class="arithmatex">\(F(\mathbf{s})\)</span>, along <span class="arithmatex">\(\mathbf{S}\)</span> and an unbiased estimator
of F(s) is given by</p>
<div class="arithmatex">\[F(\mathbf{s}) = -\left(1+\frac{T}{T_{tempered}}\right)\lim_{t \to \infty} V_{meta}(\mathbf{s},t)\]</div>
<h4 id="plane-wave-density-functional-theory-input">Input<a class="headerlink" href="#plane-wave-density-functional-theory-input" title="Permanent link">&para;</a></h4>
<p>Input to a metadynamics simulation is contained within the METADYNAMICS
sub-block. Listed below is the the format of a METADYNAMICS
sub-block,</p>
<pre><code>NWPW  
 METADYNAMICS
    [
    BOND &lt;integer atom1_index no default&gt; &lt;integer atom2_index no default&gt; 
         [W &lt;real w default 0.00005&gt;]  
         [SIGMA &lt;real sigma default 0.1&gt;] 
         [RANGE &lt;real a b default (see input description)&gt;]  
         [NRANGE &lt;integer nrange default 501&gt;] 
    ...]  
    [
    ANGLE &lt;integer atom1_index no default&gt; &lt;integer atom2_index no default&gt; &lt;integer atom3_index no default&gt;  
         [W &lt;real w default 0.00005&gt;] 
         [SIGMA &lt;real sigma default 0.1&gt;]
         [RANGE &lt;real a b default 0]
         [NRANGE &lt;integer nrange default 501&gt;]  
    ...]
    [
    COORD_NUMBER [INDEX1 &lt;integer_list atom1_indexes no default&gt;][INDEX2 &lt;integer_list atom2_indexes no default&gt;]  
         [SPRIK] 
         [N &lt;real n default 6.0&gt;]
         [M &lt;real m default 12.0&gt;]
         [R0 &lt;real r0 default 3.0&gt;]  

         [W &lt;real w default 0.00005&gt;] 
         [SIGMA &lt;real sigma default 0.1&gt;] 
         [RANGE &lt;real a b no default&gt;]  
         [NRANGE &lt;integer nrange default 501&gt;] 
    ...] 
    [  
    N-PLANE  &lt;integer atom1_index no default&gt; &lt;integer_list atom_indexes no default&gt;  
         [W &lt;real w default 0.00005&gt;]   
         [SIGMA &lt;real sigma default 0.1&gt;]  
         [RANGE &lt;real a b no default&gt;] 
         [NRANGE &lt;integer nrange default 501&gt;]  
         [NVECTOR &lt;real(3) nx ny nz&gt;] 
    ...] 
    [UPDATE &lt;integer meta_update default 1&gt;]  
    [PRINT_SHIFT &lt;integer print_shift default 0&gt;]
    [TEMPERED &lt;real tempered_temperature no default&gt;]  
    [BOUNDARY &lt;real w_boundary sigma_boundary no default&gt;]
  END
END
</code></pre>
<p>Multiple collective variables can be defined at the same time, e.g.</p>
<pre><code>NWPW  
 METADYNAMICS 
    BOND 1 8  W 0.0005 SIGMA 0.1 
    BOND 1 15 W 0.0005 SIGMA 0.1 
  END
END
</code></pre>
<p>will produce a two-dimensional potential energy surface.</p>
<h3 id="plane-wave-density-functional-theory-tamd-temperature-accelerated-molecular-dynamics">TAMD - Temperature Accelerated Molecular Dynamics<a class="headerlink" href="#plane-wave-density-functional-theory-tamd-temperature-accelerated-molecular-dynamics" title="Permanent link">&para;</a></h3>
<h4 id="plane-wave-density-functional-theory-input_1">Input<a class="headerlink" href="#plane-wave-density-functional-theory-input_1" title="Permanent link">&para;</a></h4>
<h3 id="plane-wave-density-functional-theory-collective-variables">Collective Variables<a class="headerlink" href="#plane-wave-density-functional-theory-collective-variables" title="Permanent link">&para;</a></h3>
<h4 id="plane-wave-density-functional-theory-bond-distance-collective-variable">Bond Distance Collective Variable<a class="headerlink" href="#plane-wave-density-functional-theory-bond-distance-collective-variable" title="Permanent link">&para;</a></h4>
<p>This describes the bond distance between any pair of atoms <span class="arithmatex">\(i\)</span> and
<span class="arithmatex">\(j\)</span>:</p>
<div class="arithmatex">\[s\left(r_{ij}\right) = \left \vert \mathbf{r}_{i}-\mathbf{r}_j\right\vert = r_{ij}\]</div>
<h4 id="plane-wave-density-functional-theory-angle-collective-variable">Angle Collective Variable<a class="headerlink" href="#plane-wave-density-functional-theory-angle-collective-variable" title="Permanent link">&para;</a></h4>
<p>This describes the bond angle formed at <span class="arithmatex">\(i\)</span> by the triplet <span class="arithmatex">\(&lt;ijk&gt;\)</span></p>
<div class="arithmatex">\[s\left(r_{ij},r_{ik}\right) = \frac{\pmb{r}_{ij}\cdot\pmb{r}_{ik}}{r_{ij}r_{ik}}\]</div>
<h4 id="plane-wave-density-functional-theory-coordination-collective-variable">Coordination Collective Variable<a class="headerlink" href="#plane-wave-density-functional-theory-coordination-collective-variable" title="Permanent link">&para;</a></h4>
<p>The coordination number collective variable is defined as</p>
<div class="arithmatex">\[s\left(r_{ij},r_{0}\right) =\sum_{i,j}\xi_{ij}\]</div>
<p>where the summation over <span class="arithmatex">\(i\)</span> and <span class="arithmatex">\(j\)</span> runs over two types of atoms,
<span class="arithmatex">\(\xi_{ij}\)</span> is the <em>weighting function</em>, and <span class="arithmatex">\(r_{0}\)</span> is the cut-off
distance.
In the standard procedure for computing the coordination
number, <span class="arithmatex">\(\xi_{ij}\)</span> =1 if <span class="arithmatex">\(r_{ij} &lt; r_0\)</span>, otherwise 
<span class="arithmatex">\(\xi_{ij}\)</span> =0,
implying that <span class="arithmatex">\(\xi_{ij}\)</span> is not continuous when 
<span class="arithmatex">\(r_{ij}=r_{0}\)</span>. To
ensure a smooth and continuous definition of the coordination number, we
adopt two variants of the weighting function. The first variant
is</p>
<div class="arithmatex">\[\xi_{ij} = \frac{1-\left(r_{ij}/r_{0}\right)^n}{1-\left(r_{ij}/r_{0}\right)^m}\]</div>
<p>where <span class="arithmatex">\(n\)</span> and <span class="arithmatex">\(m\)</span> are integers (m &gt; n) chosen such that
<span class="arithmatex">\(\xi_{ij}\approx 1\)</span> when <span class="arithmatex">\(r_{ij} &lt; r_0\)</span>  and
<span class="arithmatex">\(\xi_{ij}\rightarrow 0\)</span> when <span class="arithmatex">\(r_{ij}\)</span> is much larger than <span class="arithmatex">\(r_{0}\)</span>.
For example, the parameters of the O-H coordination in water is well
described by <span class="arithmatex">\(r_{0}\)</span> =1.6 , 
<span class="arithmatex">\(n=6\)</span>
and <span class="arithmatex">\(m=18\)</span>. 
In practice, <span class="arithmatex">\(n\)</span>
and <span class="arithmatex">\(m\)</span> must tuned for a given <span class="arithmatex">\(r_{0}\)</span> to ensure that <span class="arithmatex">\(\xi_{ij}\)</span>
is smooth and satisfies the above mentioned properties, particularly the
large <span class="arithmatex">\(r_{ij}\)</span></p>
<p>The second form of the weighting function, which is due to Sprik, is</p>
<div class="arithmatex">\[\xi_{ij}=\frac{1}{1 + \exp\left[n\left(r_{ij}-r_{0}\right)\right]}\]</div>
<p>In this definition <span class="arithmatex">\(\xi_{ij}\)</span> is analogous to the Fermi function and
its width is controlled by the parameter <span class="arithmatex">\(\frac{1}{n}\)</span>. Large and
small values of <span class="arithmatex">\(n\)</span> respectively correspond to sharp and soft
transitions at <span class="arithmatex">\(r_{ij} = r_{0}\)</span>. 
Furthermore <span class="arithmatex">\(\xi_{ij}\)</span> should
approach 1 and 0 when <span class="arithmatex">\(r_{ij} &lt; 0\)</span>
and <script type="math/tex">r_{ij} < r_0 </script>
respectively. In practice <span class="arithmatex">\(n\)</span> =6-10  <span class="arithmatex">\(^{-1}\)</span>. 
For example, a good set
of parameters of the O-H coordination in water is <span class="arithmatex">\(r_{0}\)</span> =1.4  and
<span class="arithmatex">\(n\)</span> =10  <span class="arithmatex">\(^{-1}\)</span>.</p>
<h4 id="plane-wave-density-functional-theory-n-plane-collective-variable">N-Plane Collective Variable<a class="headerlink" href="#plane-wave-density-functional-theory-n-plane-collective-variable" title="Permanent link">&para;</a></h4>
<p>The N-Plane collective variable is useful for probing the adsorption of
adatom/admolecules to a surface. It is defined as the <em>average distance</em>
of the adatom/admolecule from a given layer in the slab along the
surface normal:</p>
<div class="arithmatex">\[s = Z_{ads}-\frac{1}{N_{plane}}\sum_{i=1}^{N_{plane}}Z_i\]</div>
<p>where <span class="arithmatex">\(Z_{ads}\)</span> denotes the position of the adatom/admolecule/impurity
along the surface normal (here, we assume the surface normal to be the
z-axis) and the summation over 
<span class="arithmatex">\(i\)</span> 
runs over 
<span class="arithmatex">\(N_{plane}\)</span> 
atoms at
<span class="arithmatex">\(Z_i\)</span> which form the layer. The layer could be on the face or in the
interior of the
slab.</p>
<h4 id="plane-wave-density-functional-theory-user-defined-collective-variable">User defined Collective Variable<a class="headerlink" href="#plane-wave-density-functional-theory-user-defined-collective-variable" title="Permanent link">&para;</a></h4>
<h2 id="plane-wave-density-functional-theory-extended-x-ray-absorption-fine-structure-exafs-integration-with-feff6l">Extended X-Ray Absorption Fine Structure (EXAFS) - Integration with FEFF6L<a class="headerlink" href="#plane-wave-density-functional-theory-extended-x-ray-absorption-fine-structure-exafs-integration-with-feff6l" title="Permanent link">&para;</a></h2>
<h2 id="plane-wave-density-functional-theory-frozen-phonon-calculations">Frozen Phonon Calculations<a class="headerlink" href="#plane-wave-density-functional-theory-frozen-phonon-calculations" title="Permanent link">&para;</a></h2>
<h2 id="plane-wave-density-functional-theory-steepest-descent">Steepest Descent<a class="headerlink" href="#plane-wave-density-functional-theory-steepest-descent" title="Permanent link">&para;</a></h2>
<p>The functionality of this task is now performed automatically by the
PSPW and BAND. For backward compatibility, we provide a description of
the input to this task.</p>
<p>The steepest_descent task is used to optimize the one-electron orbitals
with respect to the total energy. In addition it can also be used to
optimize geometries. This method is meant to be used for coarse
optimization of the one-electron orbitals.</p>
<p>Input to the steepest_descent simulation is contained within the
steepest_descent sub-block.</p>
<pre><code>NWPW   
...   
 STEEPEST_DESCENT   
   ...   
 END   
...  
END
</code></pre>
<p>To run a steepest_descent calculation the following directive is used:</p>
<pre><code>TASK PSPW steepest_descent  
TASK BAND steepest_descent 
</code></pre>
<p>The steepest_descent sub-block contains a great deal of input,
including pointers to data, as well as parameter input. Listed below is
the format of a STEEPEST_DESCENT
sub-block.</p>
<pre><code>NWPW  
...   
 STEEPEST_DESCENT   
   CELL_NAME &lt;string cell_name&gt;   
   [GEOMETRY_OPTIMIZE]   
   INPUT_WAVEFUNCTION_FILENAME &lt;string input_wavefunctions default file_prefix.movecs&gt;   
   OUTPUT_WAVEFUNCTION_FILENAME &lt;string output_wavefunctions default file_prefix.movecs&gt;   
   FAKE_MASS &lt;real fake_mass default 400000.0&gt;   
   TIME_STEP &lt;real time_step default 5.8&gt;   
   LOOP &lt;integer inner_iteration outer_iteration default 10 1&gt;   
   TOLERANCES &lt;real tole tolc tolr default 1.0d-9 1.0d-9 1.0d-4&gt;   
   ENERGY_CUTOFF &lt;real ecut default (see input desciption)&gt;   
   WAVEFUNCTION_CUTOFF &lt;real wcut default (see input description)&gt;   
   EWALD_NCUT &lt;integer ncut default 1&gt;   
   EWALD_RCUT &lt;real rcut default (see input description)&gt;   
   XC (Vosko      || LDA          || PBE96         || revPBE       || HF            ||    
       LDA-SIC    || LDA-SIC/2    || LDA-0.4SIC    || LDA-SIC/4    || LDA-0.2SIC    ||   
       PBE96-SIC  || PBE96-SIC/2  || PBE96-0.4SIC  || PBE96-SIC/4  || PBE96-0.2SIC  ||   
       revPBE-SIC || revPBE-SIC/2 || revPBE-0.4SIC || revPBE-SIC/4 || revPBE-0.2SIC ||   
       PBE0       || revPBE0      || default Vosko)   
   [MULLIKEN]  
 END  
...  
END
</code></pre>
<p>The following list describes the input for the STEEPEST_DESCENT
sub-block.</p>
<ul>
<li><code>cell_name</code> - name of the simulation_cell named <code>cell_name</code>. See
    <a href="#plane-wave-density-functional-theory-simulation-cell">Simulation Cell</a>.</li>
<li>GEOMETRY_OPTIMIZE - optional keyword which if specified turns on
    geometry optimization.</li>
<li><code>input_wavefunctions</code> - name of the file containing one-electron
    orbitals</li>
<li><code>output_wavefunctions</code> - name of the file tha will contain the
    one-electron orbitals at the end of the run.</li>
<li><code>fake_mass</code> - value for the electronic fake mass <span class="arithmatex">\(\mu\)</span></li>
<li><code>time_step</code> - value for the time step <span class="arithmatex">\(Delta t\)</span>.</li>
<li><code>inner_iteration</code> - number of iterations between the printing out of
    energies and tolerances</li>
<li><code>outer_iteration</code> - number of outer iterations</li>
<li><code>tole</code> - value for the energy tolerance.</li>
<li><code>tolc</code> - value for the one-electron orbital tolerance.</li>
<li><code>tolr</code> - value for the ion position tolerance.</li>
<li><code>ecut</code> - value for the cutoff energy used to define the density.
    Default is set to be the maximum value that will fit within the
    simulation_cell <code>cell_name</code>.</li>
<li><code>wcut</code> - value for the cutoff energy used to define the one-electron
    orbitals. Default is set to be the maximum value that will fit
    within the simulation_cell <code>cell_name</code>.</li>
<li><code>ncut</code> - value for the number of unit cells to sum over (in each
    direction) for the real space part of the Ewald summation. Note
    Ewald summation is only used if the simulation_cell is periodic.</li>
<li><code>rcut</code> - value for the cutoff radius used in the Ewald summation.
    Note Ewald summation is only used if the simulation_cell is
    periodic.</li>
</ul>
<p>Default set to be <span class="arithmatex">\(\frac{MIN(\left| \vec{a_i} \right|)}{\pi}, i=1,2,3\)</span>.</p>
<ul>
<li>(Vosko || PBE96 || revPBE || &hellip;) - Choose between Vosko et al&rsquo;s LDA
    parameterization or the orginal and revised Perdew, Burke, and
    Ernzerhof GGA functional. In addition, several hybrid options
    (hybrid options are not available in BAND).</li>
<li>MULLIKEN - optional keyword which if specified causes a Mulliken
    analysis to be performed at the end of the simulation.</li>
</ul>
<h2 id="plane-wave-density-functional-theory-simulation-cell">Simulation Cell<a class="headerlink" href="#plane-wave-density-functional-theory-simulation-cell" title="Permanent link">&para;</a></h2>
<p>The simulation cell parameters are entered by defining a
simulation_cell sub-block within the PSPW block. Listed below is the
format of a simulation_cell sub-block.</p>
<pre><code>NWPW 
...   
 SIMULATION_CELL  [units &lt;string units default bohrs&gt;]
    CELL_NAME &lt;string name default 'cell_default'&gt;  
    BOUNDARY_CONDITIONS (periodic || aperiodic default periodic) 
    LATTICE_VECTORS   
      &lt;real a1.x a1.y a1.z default 20.0 0.0 0.0&gt; 
      &lt;real a2.x a2.y a2.z default 0.0 20.0 0.0&gt;   
      &lt;real a3.x a3.y a3.z default 0.0 0.0 20.0&gt; 
    NGRID &lt;integer na1 na2 na3 default 32 32 32&gt; 
  END 
...  
END
</code></pre>
<p>Basically, the user needs to enter the dimensions, gridding and boundary
conditions of the simulation cell. The following list describes the
input in detail.</p>
<ul>
<li><code>name</code> - user-supplied name for the simulation block.</li>
<li>periodic - keyword specifying that the simulation cell has periodic
    boundary conditions.</li>
<li>aperiodic - keyword specifying that the simulation cell has
    free-space boundary conditions.</li>
<li><code>a1.x a1.y a1.z</code> - user-supplied values for the first lattice
    vector</li>
<li><code>a2.x a2.y a2.z</code> - user-supplied values for the second lattice
    vector</li>
<li><code>a3.x a3.y a3.z</code> - user-supplied values for the third lattice
    vector</li>
<li><code>na1 na2 na3</code> - user-supplied values for discretization along
    lattice vector directions.</li>
</ul>
<p>Alternatively, instead of explicitly entering lattice vectors, users can
enter the unit cell using the standard cell parameters, a, b, c,
<span class="arithmatex">\(alpha\)</span>,
<span class="arithmatex">\(\beta\)</span>, 
and <span class="arithmatex">\(\gamma\)</span>, by using the LATTICE block. The
format for input is as follows:</p>
<pre><code>NWPW 
... 
 SIMULATION_CELL [units &lt;string units default bohrs&gt;]  
   ...   
   LATTICE   
     [lat_a &lt;real a default 20.0&gt;]  
     [lat_b &lt;real b default 20.0&gt;]  
     [lat_c &lt;real c default 20.0&gt;] 
     [alpha &lt;real alpha default 90.0&gt;]  
     [beta &lt;real beta default 90.0&gt;] 
     [gamma &lt;real gamma default 90.0&gt;]  
   END 
   ...   
 END
...
END
</code></pre>
<p>The user can also enter the lattice vectors of standard unit cells using
the keywords SC, FCC, BCC, for simple cubic, face-centered cubic, and
body-centered cubic respectively. Listed below is an example of the
format of this type of input.</p>
<pre><code>NWPW 
... 
 SIMULATION_CELL [units &lt;string units default bohrs&gt;]
   SC 20.0 
   .... 
 END
...
END
</code></pre>
<p>Finally, the lattice vectors from the unit cell can also be defined
using the fractional coordinate input in the GEOMETRY input (see section
<a href="#system----lattice-parameters-for-periodic-systems">Geometry Lattice
Parameters</a>).
Listed below is an example of the format of this type of input for an 8
atom silicon carbide unit cell.</p>
<pre><code>geometry units au  
 system crystal 
   lat_a 8.277
   lat_b 8.277   
   lat_c 8.277   
   alpha 90.0  
   beta 90.0   
   gamma 90.0   
 end
 Si -0.50000 -0.50000 -0.50000
 Si  0.00000  0.00000 -0.50000 
 Si  0.00000 -0.50000  0.00000
 Si -0.50000  0.00000  0.00000  
 C  -0.25000 -0.25000 -0.25000  
 C   0.25000  0.25000 -0.25000 
 C   0.25000 -0.25000  0.25000 
 C  -0.25000  0.25000  0.25000  
end
</code></pre>
<p>Warning - Currently only the &ldquo;system crystal&rdquo; option is recognized by
NWPW. The &ldquo;system slab&rdquo; and &ldquo;system polymer&rdquo; options will be supported
in the future.</p>
<h2 id="plane-wave-density-functional-theory-unit-cell-optimization">Unit Cell Optimization<a class="headerlink" href="#plane-wave-density-functional-theory-unit-cell-optimization" title="Permanent link">&para;</a></h2>
<p>The PSPW module using the DRIVER geometry optimizer can optimize a
crystal unit cell. Currently this type of optimization works only if the
geometry is specified in fractional coordinates. The following SET
directive is used to tell the DRIVER geometry optimizer to optimize the
crystal unit cell in addition to the geometry.</p>
<p><code>set includestress .true.</code></p>
<h2 id="plane-wave-density-functional-theory-smear-fractional-occupation-of-the-molecular-orbitals">SMEAR - Fractional Occupation of the Molecular Orbitals<a class="headerlink" href="#plane-wave-density-functional-theory-smear-fractional-occupation-of-the-molecular-orbitals" title="Permanent link">&para;</a></h2>
<p>The smear keyword to turn on fractional occupation of the molecular
orbitals in PSPW and BAND
calculations</p>
<pre><code>SMEAR &lt;sigma default 0.001&gt; [TEMPERATURE &lt;temperature&gt;]
                            [FERMI || GAUSSIAN || MARZARI-VANDERBILT default FERMI]
                            [ORBITALS &lt;integer orbitals default 4&gt;]
</code></pre>
<p>Fermi-Dirac (FERMI), Gaussian, and Marzari-Vanderbilt broadening functions are
available. The ORBITALS keyword is used to change the number of virtual
orbitals to be used in the calculation. Note to use this option the user
must currently use the SCF minimizer. The following SCF options are
recommended for running fractional occupation</p>
<p><code>SCF Anderson outer_iterations 0 Kerker 2.0</code></p>
<h2 id="plane-wave-density-functional-theory-spin-penalty-functions">Spin Penalty Functions<a class="headerlink" href="#plane-wave-density-functional-theory-spin-penalty-functions" title="Permanent link">&para;</a></h2>
<p>Spin-penalty functions makes it easier to define antiferromagnetic
structures. These functions are implemented by adding a scaling factor
to the non-local psp for up/down spins on atoms and angular momentum
that you specify.</p>
<p>Basically, the pseudopotential
energy</p>
<div class="arithmatex">\[E_{psp}= \sum_{\sigma=\uparrow,\downarrow} \sum_{i=1}^{n_{elc}^\sigma} \sum_{I=1}^{n_{ions}} \left( &lt;\psi_i^\sigma|V_{local}^{I}|\psi_i^\sigma&gt;  + \sum_{l=0}^{l_{max}^I} \sum_{m=-l}^{l} \sum_{n=1}^{n_{max}^I} \sum_{n'=1}^{n_{max}^I} &lt;\psi_i^\sigma|P_{nlm}^I&gt; h_{l,n,n'}^I &lt;P_{n'lm}^{I}|\psi_i^\sigma&gt; \right)\]</div>
<p>was modified
to</p>
<div class="arithmatex">\[E_{psp}= \sum_{\sigma=\uparrow,\downarrow} \sum_{i=1}^{n_{elc}^\sigma} \sum_{I=1}^{n_{ions}} \left( &lt;\psi_i^\sigma|V_{local}^{I}|\psi_i^\sigma&gt;  + \sum_{l=0}^{l_{max}^I} \sum_{m=-l}^{l} \sum_{n=1}^{n_{max}^I} \sum_{n'=1}^{n_{max}^I}  \left(1-\delta_{l,l^\sigma} \delta_{I,ionlist^\sigma}(\xi^\sigma-1)\right) &lt;\psi_i^\sigma|P_{nlm}^I&gt; h_{l,n,n'}^I &lt;P_{n'lm}^{I}|\psi_i^\sigma&gt; \right)\]</div>
<p>An example input is as follows:</p>
<pre><code>title &quot;hematite 10 atoms&quot;

start hema10

memory 1900 mb

permanent_dir ./perm
scratch_dir   ./perm

geometry center noautosym noautoz print
 system crystal
   lat_a 5.42 
   lat_b 5.42 
   lat_c 5.42 
   alpha 55.36  
   beta  55.36  
   gamma 55.36  
 end
Fe                  0.355000   0.355000   0.355000
Fe                  0.145000   0.145000   0.145000  
Fe                 -0.355000  -0.355000  -0.355000  
Fe                  0.855000   0.855000   0.855000  
O                   0.550000  -0.050000   0.250000  
O                   0.250000   0.550000  -0.050000  
O                  -0.050000   0.250000   0.550000  
O                  -0.550000   0.050000  -0.250000  
O                  -0.250000  -0.550000   0.050000  
O                   0.050000  -0.250000  -0.550000\  
end  

nwpw
  virtual 8
  odft
  ewald_rcut 3.0
  ewald_ncut 8 
  xc pbe96
  lmbfgs 
  mult 1
  dplot
     density diff diff1.cube
  end

  #spin penalty functions  
  pspspin up   d -1.0 1:2  
  pspspin down d -1.0 3:4  
end
task pspw energy 
task pspw pspw_dplot  

nwpw
  pspspin off
  dplot
     density diff diff2.cube
  end  
end 
task pspw energy
task pspw pspw_dplot
</code></pre>
<h2 id="plane-wave-density-functional-theory-aimdmm-qmmm">AIMD/MM (QM/MM)<a class="headerlink" href="#plane-wave-density-functional-theory-aimdmm-qmmm" title="Permanent link">&para;</a></h2>
<p>A QM/MM capability is available that is integrated with PSPW module and
can be used with Car-Parrinello simulations. Currently, the input is not
very robust but it is straightforward. The first step to run a QM/MM
simulations is to define the MM atoms in the geometry block. The MM
atoms must be at the end of the geometry and a carat, &rdquo; ^ &ldquo;, must be
appended to the end of the atom name, e.g.</p>
<pre><code>geometry units angstrom nocenter noautosym noautoz print xyz  
 C -0.000283 0.000106 0.000047  
 Cl -0.868403 1.549888 0.254229  
 Cl 0.834043 -0.474413 1.517103  
 Cl -1.175480 -1.275747 -0.460606  
 Cl 1.209940 0.200235 -1.310743  
 O^ 0.3226E+01 -0.4419E+01 -0.5952E+01   
 H^ 0.3193E+01 -0.4836E+01 -0.5043E+01   
 H^ 0.4167E+01 -0.4428E+01 -0.6289E+01  
 O^ 0.5318E+01 -0.3334E+01 -0.1220E+01  
 H^ 0.4978E+01 -0.3040E+01 -0.2113E+01  
 H^ 0.5654E+01 -0.2540E+01 -0.7127E+00  
end
</code></pre>
<p>Another way to specify the MM atoms is to use the mm_tags option which
appends the atoms with a &rdquo; ^ &ldquo;.</p>
<pre><code>geometry units angstrom nocenter noautosym noautoz print xyz  
 C -0.000283 0.000106 0.000047  
 Cl -0.868403 1.549888 0.254229  
 Cl 0.834043 -0.474413 1.517103  
 Cl -1.175480 -1.275747 -0.460606  
 Cl 1.209940 0.200235 -1.310743  
 O 0.3226E+01 -0.4419E+01 -0.5952E+01   
 H 0.3193E+01 -0.4836E+01 -0.5043E+01   
 H 0.4167E+01 -0.4428E+01 -0.6289E+01  
 O 0.5318E+01 -0.3334E+01 -0.1220E+01  
 H 0.4978E+01 -0.3040E+01 -0.2113E+01  
 H 0.5654E+01 -0.2540E+01 -0.7127E+00  
end  
NWPW  
  QMMM  
    mm_tags 6:11  
  END  
END
</code></pre>
<p>The option &ldquo;mm_tags off&rdquo; can be used to remove the &rdquo; ^ &rdquo; from the
atoms, i.e.</p>
<pre><code>NWPW  
  QMMM  
     mm_tags 6:11 off  
  END  
END 
</code></pre>
<p>Next the pseudopotentials have be defined for the every type of MM atom
contained in the geometry blocks. The following local pseudopotential
suggested by Laio, VandeVondele and Rothlisberger can be automatically
generated.</p>
<div class="arithmatex">\[\begin{matrix}V(\vec{r}) = -Z_{ion}\frac{{r_c}^{n_{\sigma}} - r^{n_{\sigma}}}{-sign(Z_{ion})*{r_c}^{n_{\sigma}+1}-r^{n_{\sigma}+1}}\end{matrix}\]</div>
<p>The following input To define this pseudopo the O^ MM atom using the
following input</p>
<pre><code>NWPW   
 QMMM   
   mm_psp O^ -0.8476 4 0.70   
 END  
END
</code></pre>
<p>defines the local pseudopotential for the O^ MM atom , where
<span class="arithmatex">\(Z_{ion}=-0.8476\)</span>, 
<span class="arithmatex">\(n_{\sigma}=4\)</span>, and <span class="arithmatex">\(r_c=0.7\)</span>. The following
input can be used to define the local pseudopotentials for all the MM
atoms in the geometry block defined above</p>
<pre><code>NWPW   
 QMMM   
   mm_psp O^ -0.8476 4 0.70   
   mm_psp H^ 0.4238 4 0.40   
 END  
END
</code></pre>
<p>Next the Lenard-Jones potentials for the QM and MM atoms need to be
defined. This is done as as follows</p>
<pre><code>NWPW   
 QMMM   
   lj_ion_parameters C 3.41000000d0 0.10d0   
   lj_ion_parameters Cl 3.45000000d0 0.16d0   
   lj_ion_parameters O^ 3.16555789d0 0.15539425d0   
 END  
END
</code></pre>
<p>Note that the Lenard-Jones potential is not defined for the MM H atoms
in this example. The final step is to define the MM fragments in the
simulation. MM fragments are a set of atoms in which bonds and angle
harmonic potentials are defined, or alternatively shake constraints are
defined. The following input defines the fragments for the two water
molecules in the above
geometry,</p>
<pre><code>NWPW   
 QMMM   
   fragment spc   
     size 3                  #size of fragment   
     index_start 6:9:3       #atom index list that defines the start of   
                             # the fragments (start:final:stride)  
     bond_spring 1 2 0.467307856 1.889726878   #bond i j Kspring r0   
     bond_spring 1 3 0.467307856 1.889726878   #bond i j Kspring r0   
     angle_spring 2 1 3 0.07293966 1.910611932 #angle i j k Kspring theta0   
   end   
 END  
END
</code></pre>
<p>The fragments can be defined using shake constraints
as</p>
<pre><code>NWPW   
 QMMM   
   fragment spc   
     size 3                  #size of fragment   
     index_start 6:9:3       #atom index list that defines the start of   
                             # the fragments (start:final:stride)  
     shake units angstroms 1 2 3 cyclic 1.0 1.632993125 1.0   
   end   
 END  
END
</code></pre>
<p>Alternatively, each water could be defined independently as
follows.</p>
<pre><code>NWPW   
 QMMM   
   fragment spc1   
     size 3                  #size of fragment   
     index_start 6           #atom index list that defines the start of   
                             #the fragments   
     bond_spring 1 2 0.467307856 1.889726878 #bond i j Kspring r0   
     bond_spring 1 3 0.467307856 1.889726878 #bond i j Kspring r0   
     angle_spring 2 1 3 0.07293966 1.910611932 #angle i j k Kspring theta0   
   end   
   fragment spc2   
     size 3                  #size of fragment   
     index_start 9           #atom index list that defines the start of   
                             #the fragments   
     shake units angstroms 1 2 3 cyclic 1.0 1.632993125 1.0   
   end   
 END  
END
</code></pre>
<h2 id="plane-wave-density-functional-theory-psp_generator">PSP_GENERATOR<a class="headerlink" href="#plane-wave-density-functional-theory-psp_generator" title="Permanent link">&para;</a></h2>
<p>A one-dimensional pseudopotential code has been integrated into NWChem.
This code allows the user to modify and develop pseudopotentials.
Currently, only the Hamann and Troullier-Martins norm-conserving
pseudopotentials can be generated. In future releases, the
pseudopotential library (section <a href="#plane-wave-density-functional-theory-pseudopotential-and-paw-basis-libraries">Pseudopotential and PAW basis
Libraries</a>) will be
more complete, so that the user will not have explicitly generate
pseudopotentials using this module.</p>
<p>Input to the PSP_GENERATOR task is contained within the PSP_GENERATOR
sub-block.</p>
<pre><code>NWPW   
...   
 PSP_GENERATOR   
   ...   
 END   
...  
END
</code></pre>
<p>To run a PSP_GENERATOR calculation the following directive is used:</p>
<p><code>TASK PSPW PSP_GENERATOR</code></p>
<p>Listed below is the format of a PSP_GENERATOR
sub-block.</p>
<pre><code>NWPW  
...   
 PSP_GENERATOR   
   PSEUDOPOTENTIAL_FILENAME: &lt;string psp_name&gt;   
   ELEMENT: &lt;string element&gt;   
   CHARGE: &lt;real charge&gt;   
   MASS_NUMBER: &lt;real mass_number&gt;   
   ATOMIC_FILLING: &lt;integer ncore nvalence&gt; ( (1||2||...) (s||p||d||f||...) &lt;real filling&gt;  ...)

   [CUTOFF: &lt;integer lmax&gt; ( (s||p||d||f||g) &lt;real rcut&gt; ...) ] 

   PSEUDOPOTENTIAL_TYPE: (TROULLIER-MARTINS || HAMANN default HAMANN)   
   SOLVER_TYPE: (PAULI || SCHRODINGER default PAULI)   
   EXCHANGE_TYPE: (dirac || PBE96 default DIRAC)   
   CORRELATION_TYPE: (VOSKO || PBE96 default VOSKO)   
   [SEMICORE_RADIUS: &lt;real rcore&gt;]

 END  
...   
END
</code></pre>
<p>The following list describes the input for the PSP_GENERATOR sub-block.</p>
<ul>
<li><code>psp_name</code> - name that points to a.</li>
<li><code>element</code> - Atomic symbol.</li>
<li><code>charge</code> - charge of the atom</li>
<li><code>mass</code> - mass number for the atom</li>
<li><code>ncore</code> - number of core states</li>
<li><code>nvalence</code> - number of valence states.</li>
<li>ATOMIC_FILLING:.....(see below)</li>
<li><code>filling</code> - occupation of atomic state</li>
<li>CUTOFF:....(see below)</li>
<li><code>rcore</code> - value for the semicore radius (see below)</li>
</ul>
<h3 id="plane-wave-density-functional-theory-atomic_filling-block">ATOMIC_FILLING Block<a class="headerlink" href="#plane-wave-density-functional-theory-atomic_filling-block" title="Permanent link">&para;</a></h3>
<p>This required block is used to define the reference atom which is used
to define the pseudopotential. After the ATOMIC_FILLING: <ncore>
<nvalence> line, the core states are listed (one per line), and then the
valence states are listed (one per line). Each state contains two
integer and a value. The first integer specifies the radial quantum
number, <span class="arithmatex">\(n\)</span>, the second integer specifies the angular momentum quantum
number, <span class="arithmatex">\(l\)</span>, and the third value specifies the occupation of the
state.</p>
<p>For example to define a pseudopotential for the Neon atom in the
<span class="arithmatex">\(1s^2 2s^2 2p^6\)</span> state could have the block</p>
<pre><code>ATOMIC_FILLING: 1 2   
 1 s 2.0 #core state - 1s^2   
 2 s 2.0 #valence state - 2s^2   
 2 p 6.0 #valence state - 2p^6
</code></pre>
<p>for a pseudopotential with a <span class="arithmatex">\(2s\)</span> and <span class="arithmatex">\(2p\)</span> valence electrons or the
block</p>
<pre><code>ATOMIC_FILLING: 3 0   
 1 s 2.0 #core state   
 2 s 2.0 #core state   
 2 p 6.0 #core state
</code></pre>
<p>could be used for a pseudopotential with no valence electrons.</p>
<h3 id="plane-wave-density-functional-theory-cutoff">CUTOFF<a class="headerlink" href="#plane-wave-density-functional-theory-cutoff" title="Permanent link">&para;</a></h3>
<p>This optional block specifies the cutoff distances used to match the
all-electron atom to the pseudopotential atom. For Hamann
pseudopotentials <span class="arithmatex">\(r_{cut}(l)\)</span> defines the distance where the
all-electron potential is matched to the pseudopotential, and for
Troullier-Martins pseudopotentials <span class="arithmatex">\(r_{cut}(l)\)</span> defines the distance
where the all-electron orbital is matched to the pseudowavefunctions.
Thus the definition of the radii depends on the type of pseudopotential.
The cutoff radii used in Hamann pseudopotentials will be smaller than
the cutoff radii used in Troullier-Martins pseudopotentials.</p>
<p>For example to define a softened Hamann pseudopotential for Carbon would
be</p>
<pre><code>ATOMIC_FILLING: 1 2   
 1 s 2.0   
 2 s 2.0   
 2 p 2.0  
CUTOFF: 2  
 s 0.8   
 p 0.85   
 d 0.85
</code></pre>
<p>while a similarly softened Troullier-Marting pseudopotential for Carbon
would be</p>
<pre><code>ATOMIC_FILLING: 1 2   
 1 s 2.0   
 2 s 2.0   
 2 p 2.0  
CUTOFF: 2   
 s 1.200   
 p 1.275   
 d 1.275
</code></pre>
<h3 id="plane-wave-density-functional-theory-semicore_radius">SEMICORE_RADIUS<a class="headerlink" href="#plane-wave-density-functional-theory-semicore_radius" title="Permanent link">&para;</a></h3>
<p>Specifying the SEMICORE_RADIUS option turns on the semicore correction
approximation proposed by Louie et al (S.G. Louie, S. Froyen, and M.L.
Cohen, Phys. Rev. B, <strong>26</strong>(, 1738, (1982)). This approximation is known
to dramatically improve results for systems containing alkali and
transition metal atoms.</p>
<p>The implementation in the PSPW module defines the semi-core density,
<span class="arithmatex">\(\rho_{semicore}\)</span>, by
using the sixth-order polynomial</p>
<div class="arithmatex">\[\rho_{semicore}(r) = \begin{cases} \rho_{core}  \mbox{if } r \ge r_{semicore} \\ c_0 + c_3 r^3 + c_4 r^4 + c_5 r^5 + c_6 r^6  \mbox{if } r &lt; r_{semicore} \end{cases}\]</div>
<p>This expansion was suggested by Fuchs and Scheffler (M. Fuchs, and M.
Scheffler, Comp. Phys. Comm.,<strong>119</strong>,67 (1999)), and is better behaved
for taking derivatives (i.e. calculating ionic forces) than the
expansion suggested by Louie et al.</p>
<h2 id="plane-wave-density-functional-theory-paw-tasks-legacy-implementation">PAW Tasks: Legacy Implementation<a class="headerlink" href="#plane-wave-density-functional-theory-paw-tasks-legacy-implementation" title="Permanent link">&para;</a></h2>
<p>(<span style="color: red;"><em>This capability is now available in PSPW. It
is recommended that this module only be used for testing purposes.</em>
</span>)</p>
<p>All input to the PAW Tasks is contained within the compound NWPW block,</p>
<pre><code>NWPW   
 ...  
END
</code></pre>
<p>To perform an actual calculation a TASK PAW directive is used
(<a href="#task">Task</a>).</p>
<p><code>TASK PAW</code></p>
<p>In addition to the directives listed in <a href="#task">Task</a>, i.e.</p>
<pre><code>TASK paw energy   
TASK paw gradient   
TASK paw optimize   
TASK paw saddle   
TASK paw freqencies   
TASK paw vib
</code></pre>
<p>there are additional directives that are specific to the PSPW module,
which are:</p>
<p><code>TASK PAW [Car-Parrinello || steepest_descent ]</code></p>
<p>Once a user has specified a geometry, the PAW module can be invoked with
no input directives (defaults invoked throughout). There are
sub-directives which allow for customized application; those currently
provided as options for the PAW module
are:</p>
<pre><code>NWPW   
 CELL_NAME &lt;string cell_name default 'cell_default'&gt;   
 [GEOMETRY_OPTIMIZE]   
 INPUT_WAVEFUNCTION_FILENAME &lt;string input_wavefunctions default input_movecs&gt;   
 OUTPUT_WAVEFUNCTION_FILENAME &lt;string output_wavefunctions default input_movecs&gt;   
 FAKE_MASS &lt;real fake_mass default 400000.0&gt;   
 TIME_STEP &lt;real time_step default 5.8&gt;   
 LOOP &lt;integer inner_iteration outer_iteration default 10 100&gt;   
 TOLERANCES &lt;real tole tolc default 1.0e-7 1.0e-7&gt;   
 CUTOFF &lt;real cutoff&gt;   
 ENERGY_CUTOFF &lt;real ecut default (see input description)&gt;   
 WAVEFUNCTION_CUTOFF &lt;real wcut default (see input description)&gt;   
 EWALD_NCUT &lt;integer ncut default 1&gt;]   
 EWALD_RCUT &lt;real rcut default (see input description)&gt;   
 XC (Vosko || PBE96 || revPBE default Vosko)   
 DFT||ODFT||RESTRICTED||UNRESTRICTED   
 MULT &lt;integer mult default 1&gt;   
 INTEGRATE_MULT_L &lt;integer imult default 1&gt;  
 SIMULATION_CELL   
   ... (see input description)   
 END   
 CAR-PARRINELLO   
   ... (see input description)   
 END  
 MAPPING &lt;integer mapping default 1&gt;  
END 
</code></pre>
<p>The following list describes these keywords.</p>
<ul>
<li><code>cell_name</code> - name of the the simulation_cell named <code>cell_name</code>.
    The current version of PAW only accepts periodic unit cells. See
    <a href="#plane-wave-density-functional-theory-simulation-cell">Simulation Cell</a>.</li>
<li>GEOMETRY_OPTIMIZE - optional keyword which if specified turns on
    geometry optimization.</li>
<li><code>input_wavefunctions</code> - name of the file containing one-electron
    orbitals</li>
<li><code>output_wavefunctions</code> - name of the file that will contain the
    one-electron orbitals at the end of the run.</li>
<li><code>fake_mass</code> - value for the electronic fake mass <span class="arithmatex">\(\mu\)</span>. This
    parameter is not presently used in a conjugate gradient simulation</li>
<li><code>time_step</code> - value for the time step (<span class="arithmatex">\(\Delta t\)</span>). This parameter
    is not presently used in a conjugate gradient simulation.</li>
<li><code>inner_iteration</code> - number of iterations between the printing out of
    energies and tolerances</li>
<li><code>outer_iteration</code> - number of outer iterations</li>
<li><code>tole</code> - value for the energy tolerance.</li>
<li><code>tolc</code> - value for the one-electron orbital tolerance.</li>
<li><code>cutoff</code> - value for the cutoff energy used to define the
    wavefunction. In addition using the CUTOFF keyword automatically
    sets the cutoff energy for the density to be twice the wavefunction
    cutoff.</li>
<li><code>ecut</code> - value for the cutoff energy used to define the density.
    Default is set to be the maximum value that will fit within the
    simulation_cell <code>cell_name</code>.</li>
<li><code>wcut</code> - value for the cutoff energy used to define the one-electron
    orbitals. Default is set to be the maximum value that will fix
    within the simulation_cell <code>cell_name</code>.</li>
<li><code>ncuth</code> - value for the number of unit cells to sum over (in each
    direction) for the real space part of the smooth compensation
    summation.</li>
<li><code>rcut</code> - value for the cutoff radius used in the smooth compensation
    summation.</li>
</ul>
<p>Default set to be <span class="arithmatex">\(\frac{MIN(\left| \vec{a_i} \right|)}{\pi}, i=1,2,3\)</span>.</p>
<ul>
<li>(Vosko || PBE96 || revPBE) - Choose between Vosko et al&rsquo;s LDA
    parametrization or the original and revised Perdew, Burke, and
    Ernzerhof GGA functional.</li>
<li>MULT - optional keyword which if specified allows the user to define
    the spin multiplicity of the system</li>
<li>INTEGRATE_MULT_L - optional keyword which if specified allows the
    user to define the angular XC integration of the augmented region</li>
<li>SIMULATION_CELL (see <a href="#plane-wave-density-functional-theory-simulation-cell">Simulation Cell</a> )</li>
<li>CAR-PARRINELLO (see <a href="#plane-wave-density-functional-theory-car-parrinello-scheme-for-ab-initio-molecular-dynamics">Car-Parrinello</a>)</li>
<li><code>mapping</code> - for a value of 1 slab FFT is used, for a value of 2 a
    2d-Hilbert FFT is used.</li>
</ul>
<h2 id="plane-wave-density-functional-theory-pseudopotential-and-paw-basis-libraries">Pseudopotential and PAW basis Libraries<a class="headerlink" href="#plane-wave-density-functional-theory-pseudopotential-and-paw-basis-libraries" title="Permanent link">&para;</a></h2>
<p>A library of pseudopotentials used by PSPW and BAND is currently
available in the directory <code>$NWCHEM_TOP/src/nwpw/libraryp/pspw_default</code></p>
<p>The elements listed in the following table are
present:</p>
<pre><code> H                                                  He  
-------                              ------------------  
 Li Be                               B  C  N  O  F  Ne  
-------                             -------------------  
 Na Mg                               Al Si P  S  Cl Ar  
-------------------------------------------------------  
 K  Ca Sc Ti V  Cr Mn Fe Co Ni Cu Zn Ga Ge As Se Br Kr           
-------------------------------------------------------  
 Rb Sr Y  Zr Nb Mo Tc Ru Rh Pd Ag Cd In Sn Sb Te I  Xe  
-------------------------------------------------------  
 Cs Ba La Hf Ta W  Re Os Ir Pt Au Hg Tl Pb Bi Po At Rn  
-------------------------------------------------------  
 Fr Ra .   
-----------------  
         ------------------------------------------  
          .  .  .  .  .  .  Gd .  .  .  .  .  .  .                        
         ------------------------------------------  
          .  .  U  .  Pu .  .  .  .  .  .  .  .  .        
         ------------------------------------------
</code></pre>
<p>The pseudopotential libraries are continually being tested and added.
Also, the PSPW program can read in pseudopotentials in CPI and TETER
format generated with pseudopotential generation programs such as the
OPIUM package of Rappe et al. The user can request additional
pseudopotentials from Eric J. Bylaska at (Eric.Bylaska@pnl.gov).</p>
<p>Similarly, a library of PAW basis used by PAW is currently available in
the directory
<code>$NWCHEM_TOP/src/nwpw/libraryp/paw_default</code></p>
<pre><code> H                                                  He  
-------                              -----------------  
 Li Be                               B  C  N  O  F  Ne  
-------                             ------------------  
 Na Mg                               Al Si P  S  Cl Ar  
------------------------------------------------------  
 K  Ca Sc Ti V  Cr Mn Fe Co Ni Cu Zn Ga Ge As Se Br Kr           
------------------------------------------------------  
 .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  
------------------------------------------------------  
 .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  
------------------------------------------------------  
 .  .  .   
-----------------           
         ------------------------------------------  
          .  .  .  .  .  .  .  .  .  .  .  .  .  .                        
         ------------------------------------------  
          .  .  .  .  .  .  .  .  .  .  .  .  .  .        
         ------------------------------------------
</code></pre>
<p>Currently there are not very many elements available for PAW. However,
the user can request additional basis sets from Eric J. Bylaska at
(Eric.Bylaska@pnl.gov).</p>
<p>A preliminary implementation of the HGH pseudopotentials (Hartwigsen,
Goedecker, and Hutter) has been implemented into the PSPW module. To
access the pseudopotentials the pseudopotentials input block is used.
For example, to redirect the code to use HGH pseudopotentials for carbon
and hydrogen, the following input would be used.</p>
<pre><code>nwpw   
...   
 pseudopotentials   
  C library HGH_LDA   
  H library HGH_LDA   
 end   
...  
end
</code></pre>
<p>The implementation of HGH pseudopotentials is rather limited in this
release. HGH pseudopotentials cannot be used to optimize unit cells, and
they do not work with the MULLIKEN option. They also have not yet been
implemented into the BAND structure code.<br />
To read in pseudopotentials in CPI format the following input would be
used.</p>
<pre><code>nwpw   
...   
 pseudopotentials   
  C CPI c.cpi   
  H CPI h.cpi   
 end   
...  
end
</code></pre>
<p>In order for the program to recognize the CPI format the CPI files, e.g.
c.cpi have to be prepended with the &ldquo;<CPI>&rdquo; keyword.</p>
<p>To read in pseudopotentials in TETER format the following input would be
used.</p>
<pre><code>nwpw   
...   
 pseudopotentials   
  C TETER c.teter   
  H TETER h.teter   
 end   
...  
end
</code></pre>
<p>In order for the program to recognize the TETER format the TETER files,
e.g. c.teter have to be prepended with the &ldquo;<TETER>&rdquo; keyword.</p>
<p>If you wish to redirect the code to a different directory other than the
default one, you need to set the environmental variable
NWCHEM_NWPW_LIBRARY to the new location of the libraryps directory.</p>
<h2 id="plane-wave-density-functional-theory-nwpw-rtdb-entries-and-miscellaneous-datafiles">NWPW RTDB Entries and Miscellaneous DataFiles<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-rtdb-entries-and-miscellaneous-datafiles" title="Permanent link">&para;</a></h2>
<p>Input to the PSPW and Band modules are contained in both the RTDB and
datafiles. The RTDB is used to store input that the user will need to
directly specify. Input of this kind includes ion positions, ion
velocities, and simulation cell parameters. The datafiles are used to
store input, such the one-electron orbitals, one-electron orbital
velocities, formatted pseudopotentials, and one-dimensional
pseudopotentials, that the user will in most cases run a program to
generate.</p>
<h3 id="plane-wave-density-functional-theory-ion-positions">Ion Positions<a class="headerlink" href="#plane-wave-density-functional-theory-ion-positions" title="Permanent link">&para;</a></h3>
<p>The positions of the ions are stored in the default geometry structure
in the RTDB and must be specified using the GEOMETRY directive.</p>
<h3 id="plane-wave-density-functional-theory-ion-velocities">Ion Velocities<a class="headerlink" href="#plane-wave-density-functional-theory-ion-velocities" title="Permanent link">&para;</a></h3>
<p>The velocities of the ions are stored in the default geometry structure
in the RTDB, and must be specified using the GEOMETRY directive.</p>
<h3 id="plane-wave-density-functional-theory-wavefunction-datafile">Wavefunction Datafile<a class="headerlink" href="#plane-wave-density-functional-theory-wavefunction-datafile" title="Permanent link">&para;</a></h3>
<p>The one-electron orbitals are stored in a wavefunction datafile. This is
a binary file and cannot be directly edited. This datafile is used by
steepest_descent and Car-Parrinello tasks and can be generated using
the wavefunction_initializer or wavefunction_expander tasks.</p>
<h3 id="plane-wave-density-functional-theory-velocity-wavefunction-datafile">Velocity Wavefunction Datafile<a class="headerlink" href="#plane-wave-density-functional-theory-velocity-wavefunction-datafile" title="Permanent link">&para;</a></h3>
<p>The one-electron orbital velocities are stored in a velocity
wavefunction datafile. This is a binary file and cannot be directly
edited. This datafile is automatically generated the first time a
Car-Parrinello task is run.</p>
<h3 id="plane-wave-density-functional-theory-formatted-pseudopotential-datafile">Formatted Pseudopotential Datafile<a class="headerlink" href="#plane-wave-density-functional-theory-formatted-pseudopotential-datafile" title="Permanent link">&para;</a></h3>
<p>The pseudopotentials in Kleinman-Bylander form expanded on a simulation
cell (3d grid) are stored in a formatted pseudopotential datafile
(PSPW-&ldquo;atomname.vpp&rdquo;, BAND-&ldquo;atomname.cpp&rdquo;, PAW-&ldquo;atomname.jpp&rdquo;). These
are binary files and cannot be directly edited. These datafiles are
automatically generated.</p>
<h3 id="plane-wave-density-functional-theory-one-dimensional-pseudopotential-datafile">One-Dimensional Pseudopotential Datafile<a class="headerlink" href="#plane-wave-density-functional-theory-one-dimensional-pseudopotential-datafile" title="Permanent link">&para;</a></h3>
<p>The one-dimensional pseudopotentials are stored in a one-dimensional
pseudopotential file (&ldquo;atomname.psp&rdquo;). This is an ASCII file and can be
directly edited with a text editor or can be generated using the
pspw_generator task. However, these datafiles are usually atomatically
generated.</p>
<p>The data stored in the one-dimensional pseudopotential file is</p>
<pre><code>character*2 element :: element name   
integer charge :: valence charge of ion   
real mass :: mass of ion   
integer lmax :: maximum angular component   
real rcut(lmax) :: cutoff radii used to define pseudopotentials   
integer nr :: number of points in the radial grid   
real dr :: linear spacing of the radial grid    
real r(nr):: one-dimensional radial grid   
real Vpsp(nr,lmax) :: one-dimensional pseudopotentials   
real psi(nr,lmax) :: one-dimensional pseudowavefunctions   
real r_semicore :: semicore radius   
real rho_semicore(nr) :: semicore density  
</code></pre>
<p>and the format of it is:</p>
<pre><code>[line 1: ] element [line 2: ] charge mass lmax  
[line 3: ] (rcut(l), l=1,lmax)  
[line 4: ] nr dr  
[line 5: ] r(1) (Vpsp(1,l), l=1,lmax)  
[line 6: ] ....  
[line nr+4: ] r(nr) (Vpsp(nr,l), l=1,lmax)  
[line nr+5: ] r(1) (psi(1,l), l=1,lmax) [line nr+6: ] ....  
[line 2*nr+4:] r(nr) (psi(nr,l), l=1,lmax)  
[line 2*nr+5:] r_semicore  
if (r_semicore read) then  
 [line 2*nr+6:] r(1) rho_semicore(1)  
 [line 2*nr+7:] ....  
 [line 3*nr+5:] r(nr) rho_semicore(nr)  
end if
</code></pre>
<h2 id="plane-wave-density-functional-theory-car-parrinello-scheme-for-ab-initio-molecular-dynamics">Car-Parrinello Scheme for Ab Initio Molecular Dynamics<a class="headerlink" href="#plane-wave-density-functional-theory-car-parrinello-scheme-for-ab-initio-molecular-dynamics" title="Permanent link">&para;</a></h2>
<p>Car and Parrinello developed a unified scheme for doing <em>ab initio</em>
molecular dynamics by combining the motion of the ion cores and a
fictitious motion for the Kohn-Sham orbitals of density-functional
theory (R. Car and M. Parrinello, Phys. Rev. Lett. <strong>55</strong>, 2471, (1985)
- simple introduction
<a href="cpmd-lecture.pdf">cpmd-lecture.pdf</a>). At the
heart of this method they introduced a fictitious kinetic energy
functional for the Kohn-Sham
orbitals.</p>
<div class="arithmatex">\[\begin{align}KE \left ( \bigl[ \psi_{i,\sigma}(\vec{r}) \bigr] \right ) = \sum_{i,\sigma}^{occ} \int d\vec{r}\ \mu \left | \dot{\psi}_{i,\sigma}(\vec{r}) \right | ^2\end{align}\]</div>
<p>Given this kinetic energy the constrained equations of motion are found
by taking the first variation of the auxiliary Lagrangian.</p>
<div class="arithmatex">\[ L = \sum_{i,\sigma}^{occ} \int d\vec{r}\ \mu \left | \dot{\psi}_{i,\sigma}(\vec{r}) \right | ^2;     + \frac{1}{2} \sum_{I} M_I \left | \dot{\vec{R}}_{I} \right | ^2 - E \left [  \bigl[ \psi_{i,\sigma}(\vec{r})  \bigr] , \bigl[ \vec{R}_I  \bigr] \right ] \\ + \sum_{ij,\sigma} \Lambda_{ij,\sigma} \left ( \int d\vec{r}\ \psi_{i,\sigma}^{*}(\vec{r}) \psi_{j,\sigma}(\vec{r}) - \delta_{ij\sigma} \right )\]</div>
<p>Which generates a dynamics for the wavefunctions
<span class="arithmatex">\(\psi_{i,\sigma}(\vec{r})\)</span> and atoms positions <span class="arithmatex">\(\vec{R}_I\)</span> through
the constrained equations of motion:</p>
<div class="arithmatex">\[\begin{matrix}\mu \ddot{\psi}_{i,\sigma}(\vec{r},t) = -\frac{\delta E}{\delta \psi_{i,\sigma }^{*} \left( \vec{r},t \right) } + \sum\limits_j \Lambda_{ij,\sigma} \psi_{j,\sigma} \left( \vec{r},t \right) \end{matrix}\]</div>
<div class="arithmatex">\[\begin{matrix}M_I \ddot{\vec{R}}_I = -\frac{\partial E}{\partial \vec{R}_I} \end{matrix}\]</div>
<p>where <span class="arithmatex">\(\mu\)</span> is the fictitious mass for the electronic degrees of
freedom and <span class="arithmatex">\(M_I\)</span> are the ionic masses. The adjustable parameter
<span class="arithmatex">\(\mu\)</span> is used to describe the relative rate at which the wavefunctions
change with time. <span class="arithmatex">\(\Lambda_{ij,\sigma}\)</span> are the Lagrangian multipliers
for the orthonormalization of the single-particle orbitals
<span class="arithmatex">\(\psi_{i,\sigma}(\vec{r})\)</span>. They are defined by the orthonormalization
constraint conditions and can be rigorously found. However, the
equations of motion for the Lagrange multipliers depend on the specific
algorithm used to integrate the Eqns. above.</p>
<p>For this method to give ionic motions that are physically meaningful the
kinetic energy of the Kohn-Sham orbitals must be relatively small when
compared to the kinetic energy of the ions. There are two ways where
this criterion can fail. First, the numerical integrations for the
Car-Parrinello equations of motion can often lead to large relative
values of the kinetic energy of the Kohn-Sham orbitals relative to the
kinetic energy of the ions. This kind of failure is easily fixed by
requiring a more accurate numerical integration, i.e. use a smaller time
step for the numerical integration. Second, during the motion of the
system a the ions can be in locations where there is an Kohn-Sham
orbital level crossing, i.e. the density-functional energy can have two
states that are nearly degenerate. This kind of failure often occurs in
the study of chemical reactions. This kind of failure is not easily
fixed and requires the use of a more sophisticated density-functional
energy that accounts for low-lying excited electronic states.</p>
<h3 id="plane-wave-density-functional-theory-verlet-algorithm-for-integration">Verlet Algorithm for Integration<a class="headerlink" href="#plane-wave-density-functional-theory-verlet-algorithm-for-integration" title="Permanent link">&para;</a></h3>
<p>Integrating the Eqns. above using the Verlet algorithm results
in</p>
<div class="arithmatex">\[\begin{matrix}\psi_{i,\sigma}^{t+ \Delta t} \leftarrow 2 \psi_{i,\sigma}^{t} - \psi_{i,\sigma}^{t-\Delta t} + \frac{(\Delta t)^2}{\mu} \left[ \frac{\delta E}{\delta \psi_{i,\sigma}^{*}} + \sum_{j} \psi_{j,\sigma} \Lambda_{ji,\sigma} \right]_{t} \end{matrix}\]</div>
<div class="arithmatex">\[\begin{matrix}\vec{R}_I^{t+\Delta t} \leftarrow 2 \vec{R}_I^{t} - \vec{R}_I^{t-\Delta t} + \frac{(\Delta t)^2}{M_I} \frac{\partial E}{\partial \vec{R}_I}\end{matrix}\]</div>
<p>In this molecular dynamic procedure we have to know variational
derivative  </p>
<div class="arithmatex">\[\frac{\delta E}{\delta \psi_{i,\sigma}^{*}}\]</div>
<p>and the matrix <span class="arithmatex">\(\Lambda_{ij,\sigma}\)</span>.<br />
The variational derivative</p>
<div class="arithmatex">\[\frac{\delta E}{\delta \psi_{i,\sigma}^{*}}\]</div>
<p>can be analytically found and is  </p>
<div class="arithmatex">\[\begin{align}\frac{\delta E}{\delta \psi_{i,\sigma}^{*}}  = -\frac{1}{2} \nabla^2 \psi_{i,\sigma}(\vec{r}) \\  + \int d\vec{r^{\prime}} W_{ext}(\vec{r},\vec{r^{\prime}}) \psi_{i,\sigma}(\vec{r^{\prime}}) \\  + \int d\vec{r^{\prime}} \frac{n(\vec{r^{\prime}})}{|\vec{r}-\vec{r^{\prime}}|} \psi_{i,\sigma}(\vec{r})\\  + \mu_{xc}^{\sigma}(\vec{r}) \psi_{i,\sigma}(\vec{r}) \\  \equiv \hat{H} \psi_{i,\sigma} \end{align}\]</div>
<p>To find the matrix <span class="arithmatex">\(\Lambda_{ij,\sigma}\)</span> impose the orthonormality
constraint on <span class="arithmatex">\(\psi_{i,\sigma}^{t+\Delta t}\)</span> obtain a matrix
Riccatti equation, and then Riccatti equation is solved by an iterative
solution.</p>
<h3 id="plane-wave-density-functional-theory-constant-temperature-simulations-nose-hoover-thermostats">Constant Temperature Simulations: Nose-Hoover Thermostats<a class="headerlink" href="#plane-wave-density-functional-theory-constant-temperature-simulations-nose-hoover-thermostats" title="Permanent link">&para;</a></h3>
<p>Nose-Hoover Thermostats for the electrons and ions can also be added to
the Car-Parrinello simulation. In this type of simulation thermostats
variables <span class="arithmatex">\(x_e\)</span> and <span class="arithmatex">\(x_R\)</span> are added to the simulation by adding the
auxiliary energy functionals to the total
energy.</p>
<div class="arithmatex">\[ ION_{THERMOSTAT}(x_R)  = \frac{1}{2} Q_R \dot{x_R} + E_{R0}x_R\]</div>
<div class="arithmatex">\[ELECTRON_{THERMOSTAT}(x_e)  = \frac{1}{2} Q_e \dot{x_e} + E_{e0}x_e \]</div>
<p>In these equations, the average kinetic energy for the ions is</p>
<div class="arithmatex">\[\begin{matrix}E_{R0} = \frac{1}{2} f k_B T \end{matrix}\]</div>
<p>where <span class="arithmatex">\(f\)</span> is the number of atomic degrees of freedom, <span class="arithmatex">\(k_B\)</span>  is
Boltzmann&rsquo;s constant, and T is the desired temperature. Defining the
average fictitious kinetic energy of the electrons is not as
straightforward. Blchl and Parrinello (P.E. Blchl and M. Parrinello,
Phys. Rev. B, <strong>45</strong>, 9413, (1992)) have suggested the following formula
for determining the average fictitious kinetic
energy</p>
<div class="arithmatex">\[\begin{matrix}E_{e0} = 4 k_B T \frac{\mu}{M} \sum_{i} &lt;\psi_i|-\frac{1}{2} \nabla^2 |\psi_{i}&gt; \end{matrix}\]</div>
<p>where <span class="arithmatex">\(\mu\)</span> is the fictitious electronic mass, <span class="arithmatex">\(M\)</span> is average mass
of one atom, and
<span class="arithmatex">\(\sum_{i} &lt;\psi_i|-\frac{1}{2} \nabla^2 |\psi_{i}&gt;\)</span>
is the kinetic energy of the electrons.</p>
<p>Blchl and Parrinello suggested that the choice of mass parameters,
<span class="arithmatex">\(Q_e\)</span>, and <span class="arithmatex">\(Q_R\)</span> should be made such that the period of oscillating
thermostats should be chosen larger than the typical time scale for the
dynamical events of interest but shorter than the simulation time.</p>
<div class="arithmatex">\[P_{ion}  = 2\pi \sqrt{\frac{Q_R}{4E_{R0}}}\]</div>
<div class="arithmatex">\[P_{electron}  = 2\pi \sqrt{\frac{Q_e}{4E_{e0}}}\]</div>
<p>where <span class="arithmatex">\(P_{ion}\)</span> and <span class="arithmatex">\(P_{electron}\)</span> are the periods of oscillation
for the ionic and fictitious electronic thermostats.</p>
<p>In simulated annealing simulations the electronic and ionic Temperatures
are scaled according to an exponential cooling schedule,</p>
<div class="arithmatex">\[T_e(t)  = T_e^0 \exp^{-\frac{t}{\tau_e}}\]</div>
<div class="arithmatex">\[T_{ionic}(t)  = T_{ionic}^0 \exp^{-\frac{t}{\tau_{ionic}}}\]</div>
<p>where <span class="arithmatex">\(T_e^0\)</span> and <span class="arithmatex">\(T_{ionic}^0\)</span> are the initial temperatures, and
<span class="arithmatex">\(\tau_e\)</span> and <span class="arithmatex">\(\tau_{ionic}\)</span> are the cooling rates in atomic units.</p>
<h2 id="plane-wave-density-functional-theory-nwpw-tutorial-1-s2-dimer-examples-with-pspw">NWPW Tutorial 1: S<sub>2</sub> dimer examples with PSPW<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-tutorial-1-s2-dimer-examples-with-pspw" title="Permanent link">&para;</a></h2>
<p>A description of all the examples in NWPW Tutorial 1 can be found in the
attached pdf <a href="nwpwexample1.pdf">nwpwexample1.pdf</a></p>
<h3 id="plane-wave-density-functional-theory-total-energy-of-s2-dimer-with-lda-approximation">Total energy of S<sub>2</sub> dimer with LDA approximation<a class="headerlink" href="#plane-wave-density-functional-theory-total-energy-of-s2-dimer-with-lda-approximation" title="Permanent link">&para;</a></h3>
<p>(input:<a href="s2-example1.nw">Media:s2-example1.nw</a>,
output:<a href="s2-example1.nwout">Media:s2-example1.nwout</a>)</p>
<p>In this example, the total energy of the S<sub>2</sub> dimer using LDA
approximation for the exchange-correlation functional is calculated.</p>
<pre><code> echo  
 title &quot;total energy of s2-dimer LDA/30Ry with PSPW method&quot;  
 scratch_dir   ./scratch  
 permanent_dir ./perm  
 start s2-pspw-energy  
 geometry  
 S 0.0 0.0 0.0  
 S 0.0 0.0 1.88  
 end  
 nwpw  
   simulation_cell  
     SC 20.0  
   end  
   cutoff 15.0  
   mult 3  
   xc lda  
   lmbfgs  
 end  
 task pspw energy
</code></pre>
<p>The energies from the simulation will
be</p>
<pre><code> ...  
 ==  Summary Of Results  ==  

  number of electrons: spin up=    7.00000  down=    5.00000 (real space)  

  total     energy    :  -0.2041363137E+02 (   -0.10207E+02/ion)  
  total orbital energy:  -0.4944372503E+01 (   -0.41203E+00/electron)  
  hartree   energy    :   0.1680529987E+02 (    0.14004E+01/electron)  
  exc-corr  energy    :  -0.4320620600E+01 (   -0.36005E+00/electron)  
  ion-ion   energy    :   0.8455644190E-02 (    0.42278E-02/ion)  

  kinetic (planewave) :   0.7529965882E+01 (    0.62750E+00/electron)  
  V_local (planewave) :  -0.4506036741E+02 (   -0.37550E+01/electron)  
  V_nl    (planewave) :   0.4623635248E+01 (    0.38530E+00/electron)  
  V_Coul  (planewave) :   0.3361059973E+02 (    0.28009E+01/electron)  
  V_xc.   (planewave) :  -0.5648205953E+01 (   -0.47068E+00/electron)  
  Virial Coefficient  :  -0.1656626150E+01  

  orbital energies:  
     -0.2001309E+00 (  -5.446eV)  
     -0.2001309E+00 (  -5.446eV)  
     -0.3294434E+00 (  -8.965eV)    -0.2991148E+00 (  -8.139eV)  
     -0.3294435E+00 (  -8.965eV)    -0.2991151E+00 (  -8.139eV)  
     -0.3582269E+00 (  -9.748eV)    -0.3352434E+00 (  -9.123eV)  
     -0.5632339E+00 ( -15.326eV)    -0.5246249E+00 ( -14.276eV)  
     -0.7642738E+00 ( -20.797eV)    -0.7413909E+00 ( -20.174eV)  

  Total PSPW energy   :  -0.2041363137E+02  
 ...
</code></pre>
<h3 id="plane-wave-density-functional-theory-structural-optimization-of-s2-dimer-with-lda-approximation">Structural optimization of S<sub>2</sub> dimer with LDA approximation<a class="headerlink" href="#plane-wave-density-functional-theory-structural-optimization-of-s2-dimer-with-lda-approximation" title="Permanent link">&para;</a></h3>
<p>(input:<a href="s2-example2.nw">Media:s2-example2.nw</a>,
output:<a href="s2-example2.nwout">Media:s2-example2.nwout</a>)</p>
<p>In this example, the structure of the S<sub>2</sub> dimer using results
generated from prior energy calculation is calculated. Since most of the
parameters are already stored in the run-time database the input is very
simple.</p>
<pre><code> echo  
 title &quot;optimization of s2-dimer LDA/30Ry with PSPW method&quot;  
 scratch_dir   ./scratch  
 permanent_dir ./perm  
 restart s2-pspw-energy  
 driver  
    maxiter 20  
    xyz s2  
 end  
 task pspw optimize
</code></pre>
<p>As the optimization process consists of series of total energy
evaluations the contents of the output file are very much similar to
that in Example I. At each step the total energy and force information
will be outputed as
follows</p>
<pre><code>   Step       Energy      Delta E   Gmax     Grms     Xrms     Xmax   Walltime  
   ---- ---------------- -------- -------- -------- -------- -------- --------  
 @    1     -20.41364254 -7.1D-05  0.00004  0.00004  0.00605  0.01048      7.8
</code></pre>
<p>The best way to keep track of the optimization calculation is to run the
following grep command on the output
file.</p>
<pre><code> grep @ outputfile  

 @ Step       Energy      Delta E   Gmax     Grms     Xrms     Xmax   Walltime  
 @ ---- ---------------- -------- -------- -------- -------- -------- --------  
 @    0     -20.41357202  0.0D+00  0.00672  0.00672  0.00000  0.00000      1.5  
 @    1     -20.41364254 -7.1D-05  0.00004  0.00004  0.00605  0.01048      7.8  
 @    2     -20.41364256 -2.3D-08  0.00020  0.00020  0.00003  0.00005      9.7  
 @    2     -20.41364256 -2.3D-08  0.00020  0.00020  0.00003  0.00005      9.7
</code></pre>
<p>The optimized energy and geometry will
be</p>
<pre><code> ...  
       ----------------------  
       Optimization converged  
       ----------------------  


   Step       Energy      Delta E   Gmax     Grms     Xrms     Xmax   Walltime  
   ---- ---------------- -------- -------- -------- -------- -------- --------  
 @    2     -20.41364256 -2.3D-08  0.00020  0.00020  0.00003  0.00005      9.7  
                                      ok       ok       ok       ok          



                                 Z-matrix (autoz)  
                                 --------   

  Units are Angstrom for bonds and degrees for angles  

       Type          Name      I     J     K     L     M      Value     Gradient  
       ----------- --------  ----- ----- ----- ----- ----- ---------- ----------  
     1 Stretch                  1     2                       1.89115    0.00020  



                          Geometry &quot;geometry&quot; -&gt; &quot;geometry&quot;  
                          ---------------------------------  

  Output coordinates in angstroms (scale by  1.889725989 to convert to a.u.)  

   No.       Tag          Charge          X              Y              Z  
  ---- ---------------- ---------- -------------- -------------- --------------  
     1 S                   16.0000     0.00000000     0.00000000    -0.94557591  
     2 S                   16.0000     0.00000000     0.00000000     0.94557591  

 ...
</code></pre>
<h3 id="plane-wave-density-functional-theory-frequency-calculation-of-s2-dimer-with-lda-approximation">Frequency calculation of S<sub>2</sub> dimer with LDA approximation<a class="headerlink" href="#plane-wave-density-functional-theory-frequency-calculation-of-s2-dimer-with-lda-approximation" title="Permanent link">&para;</a></h3>
<p>(input:<a href="s2-example3.nw">Media:s2-example3.nw</a>,
output:<a href="s2-example3.nwout">Media:s2-example3.nwout</a>)</p>
<p>In this example, the vibrational frequency of the S<sub>2</sub> dimer using
results generated from prior geometry optimization is calculated. Since
most of the parameters are already stored in the run-time database the
input is very simple.</p>
<pre><code> echo  
 title &quot;frequency calculation of s2-dimer LDA/30Ry with PSPW method&quot;  
 scratch_dir   ./scratch  
 permanent_dir ./perm  
 restart s2-pspw-energy  
 freq  
    animate  
 end  
 task pspw freq
</code></pre>
<p>The frequency and thermodynamic analysis
generated</p>
<pre><code> ...  
  Temperature                      =   298.15K  
  frequency scaling parameter      =   1.0000  


  Linear Molecule  

  Zero-Point correction to Energy  =    1.034 kcal/mol  (  0.001647 au)  
  Thermal correction to Energy     =    2.579 kcal/mol  (  0.004110 au)  
  Thermal correction to Enthalpy   =    3.171 kcal/mol  (  0.005054 au)  

  Total Entropy                    =   52.277 cal/mol-K  
    - Translational                =   38.368 cal/mol-K (mol. weight =  63.9441)  
    - Rotational                   =   13.630 cal/mol-K (symmetry #  =        2)  
    - Vibrational                  =    0.279 cal/mol-K  

  Cv (constant volume heat capacity) =    5.750 cal/mol-K  
    - Translational                  =    2.979 cal/mol-K  
    - Rotational                     =    1.986 cal/mol-K  
    - Vibrational                    =    0.785 cal/mol-K  
 ...  
  ----------------------------------------------------------------------------  
  Normal Eigenvalue ||           Projected Infra Red Intensities  
   Mode   [cm**-1]  || [atomic units] [(debye/angs)**2] [(KM/mol)] [arbitrary]  
  ------ ---------- || -------------- ----------------- ---------- -----------  
     1        0.000 ||    0.000030           0.001         0.029       0.000  
     2        0.000 ||    2.466908          56.914      2404.864      15.000  
     3        0.000 ||    2.466908          56.914      2404.864      15.000  
     4        0.000 ||    2.466908          56.914      2404.864      15.000  
     5        0.000 ||    2.466908          56.914      2404.864      15.000  
     6      723.419 ||    0.000000           0.000         0.000       0.000  
  ----------------------------------------------------------------------------  
 ...
</code></pre>
<h3 id="plane-wave-density-functional-theory-ab-initio-molecular-dynamics-simulation-car-parrinello-of-s2-dimer-using-the-lda-approximation">Ab initio molecular dynamics simulation (Car-Parrinello) of S<sub>2</sub> dimer using the LDA approximation<a class="headerlink" href="#plane-wave-density-functional-theory-ab-initio-molecular-dynamics-simulation-car-parrinello-of-s2-dimer-using-the-lda-approximation" title="Permanent link">&para;</a></h3>
<p>(input:<a href="s2-example4.nw">Media:s2-example4.nw</a>,
output:<a href="s2-example4.nwout">Media:s2-example4.nwout</a>
<a href="s2-md.xyz">Media:s2-md.xyz</a>
<a href="s2-md.emotion.dat">Media:s2-md.emotion.dat</a> )</p>
<p>In this example, a constant energy Car-Parrinello simulation of S<sub>2</sub>
dimer using LDA approximation is calculated. A brief introduction to the
Car-Parrinello method can be found in
<a href="cpmd-lecture.pdf">cpmd-lecture.pdf</a></p>
<pre><code> echo  
 title &quot;AIMD simulation of s2-dimer&quot;  
 scratch_dir   ./scratch  
 permanent_dir ./perm  
 start s2-md  
 geometry  
 S 0.0 0.0 0.0  
 S 0.0 0.0 1.95  
 end  
 nwpw  
   simulation_cell  
     SC 20.0  
   end  
   cutoff 15.0  
   mult 3  
   xc lda  
   lmbfgs  
   car-parrinello  
     time_step 5.0  
     fake_mass 600.0  
     loop 1 1000  
     xyz_filename s2-md.xyz  
   end  
 end  
 task pspw energy  
 task pspw car-parrinello
</code></pre>
<p>A plotting program (e.g. gnuplot, xmgrace) can be used to look at the
total, potential, kinetic energies, contained in the s2-md.emotion file
(see section <a href="#plane-wave-density-functional-theory-emotion-motion-file">EMOTION motion file</a> for
datafile format)
i.e.,</p>
<pre><code> seattle-1604% gnuplot  

       G N U P L O T  
       Version 4.0 patchlevel 0  
       last modified Thu Apr 15 14:44:22 CEST 2004  
       System: Linux 2.6.18-194.8.1.el5  

       Copyright (C) 1986 - 1993, 1998, 2004  
       Thomas Williams, Colin Kelley and many others  

       This is gnuplot version 4.0.  Please refer to the documentation  
       for command syntax changes.  The old syntax will be accepted  
       throughout the 4.0 series, but all save files use the new syntax.  

       Type help to access the on-line reference manual.  
       The gnuplot FAQ is available from  
               &lt;http://www.gnuplot.info/faq/&gt;  

       Send comments and requests for help to  
               &lt;gnuplot-info@lists.sourceforge.net&gt;  
       Send bugs, suggestions and mods to  
               &lt;gnuplot-bugs@lists.sourceforge.net&gt;  


 Terminal type set to 'x11'  
 gnuplot&gt; plot &quot;s2-md.emotion&quot;,&quot;s2-md.emotion&quot; using 1:3  
 gnuplot&gt; 
</code></pre>
<p>The following plot shows the Car-Parrinello <span class="arithmatex">\(^3\Sigma_g^-\)</span> S<span class="arithmatex">\(_2\)</span>
energy surface generated from the
simulation.</p>
<p><img alt="s2-md" src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/s2-md.png" align=middle with="600pt" height="450pt"/>  </p>
<h3 id="plane-wave-density-functional-theory-ab-initio-molecular-dynamics-simulation-born-oppenheimer-of-ssub2-dimer-using-the-lda-approximation">Ab initio molecular dynamics simulation (Born-Oppenheimer) of S&lt;sub2</sub> dimer using the LDA approximation<a class="headerlink" href="#plane-wave-density-functional-theory-ab-initio-molecular-dynamics-simulation-born-oppenheimer-of-ssub2-dimer-using-the-lda-approximation" title="Permanent link">&para;</a></h3>
<p>(input:<a href="s2-example5.nw">Media:s2-example5.nw</a>,
output:<a href="s2-example5.nwout">Media:s2-example5.nwout</a>
<a href="s2-bomd.xyz">Media:s2-bomd.xyz</a>
<a href="s2-bomd.emotion.dat">Media:s2-bomd.emotion.dat</a> ) In
this example, a constant energy Born-Oppenheimer simulation of S2 dimer
using LDA approximation is calculated.</p>
<pre><code>title &quot;AIMD simulation of s2-dimer&quot; 
echo

scratch_dir   ./scratch
permanent_dir ./perm

start s2-bomd  

geometry
S 0.0 0.0 0.0
S 0.0 0.0 1.95
end

nwpw 
  simulation_cell
    SC 20.0 
  end
  cutoff 15.0  
  mult 3 
  xc lda 
  lmbfgs 
end
task pspw energy 

nwpw 
   bo_steps 1 500 
   bo_time_step 10.0  
end 
task pspw born-oppenheimer
</code></pre>
<p>The following plot shows the <span class="arithmatex">\(^3\Sigma_g^{-}_{2}\)</span> energy surface
generated from the
simulation.</p>
<p><img alt="s2-bomd" src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/s2-bomd.png" align=middle with="600pt" height="450pt"/>  </p>
<h2 id="plane-wave-density-functional-theory-nwpw-tutorial-2-using-pspw-car-parrinello-simulated-annealing-simulations-to-optimize-structures">NWPW Tutorial 2: Using PSPW Car-Parrinello Simulated Annealing Simulations to Optimize Structures<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-tutorial-2-using-pspw-car-parrinello-simulated-annealing-simulations-to-optimize-structures" title="Permanent link">&para;</a></h2>
<p>In principle quantum mechanical calculations can be used to determine
the structure of any chemical system. One chooses a structure,
calculates the total energy of the system, and repeats the calculation
for all possible geometries. Of course the major limitation of this
approach is that the number of local minima structures increases
dramatically with system size and the cost of quantum mechanical
calculations also increases dramatically with system size. Not
surprisingly most quantum mechanical calculations limit the number of
structures to be calculated by using experimental results or chemical
intuition. One could speed up the calculations by using simplified
inter-atomic force fields instead of quantum mechanical calculations.
However, inter-atomic forces fields have many simplifying assumptions
that can severely limit their predictability. Another approach is to use
ab initio molecular dynamics methods combined with simulated annealing.
These methods are quite robust and allow strongly interacting many body
systems to be studied by direct dynamics simulation without the
introduction of empirical interactions. In these methods, the atomic
forces are calculated from ab initio calculations that are performed
on-the-fly as the molecular dynamics trajectory is generated.</p>
<p>The following examples demonstrate how to use the ab initio molecular
dynamics methods and simulated annealing strategies of NWChem to
determine the lowest energy structures of the B<sub>12</sub> cluster. This
example is based on a study performed by Kiran Boggavarapu et al.. One
might expect from chemical intuition that lowest energy structure of
B<sub>12</sub> will be an icosahedron, since B<sub>12</sub> icosahedra are a
common structural unit found in many boron rich materials. Despite this
prevalence, ab initio calculations performed by several researchers have
   suggested that B<sub>12</sub>, as well as B<sub>12</sub><sup>+</sup> and B<sub>12</sub><sup>-</sup>,
will have a more open geometry.</p>
<p><img alt="boron" src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/boron.png" align=middle with="600pt" height="246pt"/>  </p>
<h3 id="plane-wave-density-functional-theory-simulated-annealing-using-constant-energy-simulation">Simulated Annealing Using Constant Energy Simulation<a class="headerlink" href="#plane-wave-density-functional-theory-simulated-annealing-using-constant-energy-simulation" title="Permanent link">&para;</a></h3>
<p>(input:<a href="b12-example2a.nw">Media:b12-example2a.nw</a>,
output:<a href="b12-example2a.nwout">Media:b12-example2a.nwout</a>
<a href="b12.00.xyz">Media:b12.00.xyz</a>
<a href="b12.00.emotion.dat">Media:b12.00.emotion.dat</a>
<a href="b12.01.xyz">Media:b12.01.xyz</a>
<a href="b12.01.emotion.dat">Media:b12.01.emotion.dat</a>)</p>
<p>This example uses a series of constant energy Car-Parrinello simulations
with velocity scaling to do simulated annealing. The initial four
Car-Parrinello simulations are used to heat up the system to several
thousand Kelvin. Then the system is cooled down thru a series of
constant energy simulations in which the electronic and ionic velocities
are scaled by 0.99 at the start of each Car-Parrinello simulation.
Energy minimization calculations are used periodically in this
simulation to bring the system back down to Born-Oppenheimer surface.
This is necessary because of electronic heating.</p>
<p>The Car-Parrinello keyword scaling scales the wavefunction and ionic
velocities at the start of the simulation. The following input is used
to increase the ionic velocities by a factor of two at the start of the
Car-Parrinello simulation.</p>
<p>Key Input</p>
<pre><code>...
Car-Parrinello  
fake_mass 500.0
time_step 5.0 
loop 10 100  
** scaling 1.0 2.0**  
emotion_filename b12.00.emotion
xyz_filename     b12.00.xyz
end 
...
</code></pre>
<p>Output</p>
<pre><code>...        
     wavefnc cutoff= 10.000  fft= 42x 42x 42(     6027 waves     1004 per task)  

technical parameters:  
     translation contrained  
     time step=      5.00     ficticious mass=     500.0  
     **cooling/heatting rates: 0.10000E+01 (psi)
0.20000E+01
(ion)**  
     maximum iterations =    1000 (   10 inner  100 outer )  
     initial kinetic energy:  0.99360E-05 (psi)   0.27956E-03 (ion)  
                                                  0.20205E-28 (c.o.m.)  
     **after scaling: 0.99360E-05 (psi) 0.11182E-02
(ion)**  
     **increased energy: 0.00000E+00 (psi)
0.83868E-03 (ion)**  

Constant Energy Simulation           
...
</code></pre>
<p>The program checks to see if the initial input ionic velocities have a
non-zero center of mass velocity. If there is a non-zero center of mass
velocity in the system then by default the program removes it. To turn
off this feature set the following</p>
<pre><code> nwpw 
    translation on  
 end
</code></pre>
<p>or</p>
<pre><code> set nwpw:com_shift .false.
</code></pre>
<h3 id="plane-wave-density-functional-theory-simulated-annealing-using-constant-temperature-simulation">Simulated Annealing Using Constant Temperature Simulation<a class="headerlink" href="#plane-wave-density-functional-theory-simulated-annealing-using-constant-temperature-simulation" title="Permanent link">&para;</a></h3>
<p>(input:<a href="b12-example2b.nw">Media:b12-example2b.nw</a>,
output:<a href="b12-example2b.nwout">Media:b12-example2b.nwout</a>
<a href="b12.10.xyz">Media:b12.10.xyz</a>
<a href="b12.10.emotion.dat">Media:b12.10.emotion.dat</a>
<a href="b12.11.xyz.gz">Media:b12.11.xyz.gz</a>
<a href="b12.11.emotion.dat">Media:b12.11.emotion.dat</a>)</p>
<p>(mpeg movie of simulation:
<a href="boron.mpg">Media:boron.mpg</a>)</p>
<p>The simulated annealing calculation in this example uses a constant
temperature Car-Parrinello simulation with an exponential cooling
schedule,</p>
<div class="arithmatex">\[T(t)=T_0e^{-t/\tau}\]</div>
<p>where T0 and  are an initial temperature and a time scale of cooling,
respectively. In the present calculations T0=3500K and =4.134e+4 au
(1.0 ps) were used and the thermostat masses were kept fixed to the
initial values determined by T=Te=3500K and (2/)=250 a.u. (6 fs).
Annealing proceeded for 50000 steps, until a temperature of 10K was
reached. After which, the metastable structure is optimized using the
driver optimizer. The keyword SA_decay is used to enter the decay
rates, electron and ion, used in the simulated annealing algorithm in
the constant temperature car-parrinello simulation. The decay rates are
in units of au (conversion 1 au = 2.41889e-17 seconds).</p>
<p>Key
Input</p>
<pre><code> .
 Car-Parrinello  
 SA_decay 4.134d4 4.134d4 #decay rate in units of au (1au=2.41889e-17seconds)  
 .
</code></pre>
<h2 id="plane-wave-density-functional-theory-nwpw-tutorial-3-using-isodesmic-reaction-energies-to-estimate-gas-phase-thermodynamics">NWPW Tutorial 3: using isodesmic reaction energies to estimate gas-phase thermodynamics<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-tutorial-3-using-isodesmic-reaction-energies-to-estimate-gas-phase-thermodynamics" title="Permanent link">&para;</a></h2>
<p>(<a href="isodesmic.pdf">isodesmic.pdf</a>
<a href="isodesmic.tgz">isodesmic.tgz</a>)</p>
<p>The development of a computational scheme that can accurately predict
reaction energies requires some care. As shown in Table 1 energy errors
associated with ab initio calculations can be quite high. Even though ab
initio electronic structure methods are constantly being developed and
improved upon, these methods are rarely able to give heat of formations
of a broad class of molecules with error limits of less than a few
kcal/mol. Only when very large basis sets such as the
correlation-consistent basis sets, high level treatments of correlation
energy such as coupled cluster methods (CCSD(T)), and small correction
factors such as core-valence correlation energies and relativistic
effects, are included will the heat of formation from ab initio
electronic structure methods be accurate to within one kcal/mol.
Although one can now accurately calculate the heats of formation of
molecules with up to 6 first row atoms, such high-level calculations are
extremely demanding and scale computationally as <span class="arithmatex">\(N^7\)</span> for <span class="arithmatex">\(N\)</span> basis
functions.</p>
<p>Examples of these types of large errors are shown in the following
Table, where the enthalpies of formation of CCl<span class="arithmatex">\(_3\)</span>SH are calculated
by using atomization energies from different levels of ab initio
theory.</p>
<table>
<thead>
<tr>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td></td>
<td>MP2/cc-pVDZ</td>
<td>LDA/DZVP2</td>
<td>BP91/DZVP2</td>
<td>B3LYP/DZVP2</td>
<td>G2 Theory</td>
</tr>
<tr>
<td>H <span class="arithmatex">\(_f^o\)</span></td>
<td>+4.9</td>
<td>-80.0</td>
<td>-2.6</td>
<td>+26.5</td>
<td>-13.0</td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<p>Table 1: Standard enthalpy of formation (H<span class="arithmatex">\(_f^o\)</span>(298K) for
CCl<span class="arithmatex">\(_3\)</span>SH in kcal/mol from atomization energies with various
electronic structure methods. Results taken from reference [2].</p>
<p>Differences of up to 106.5 kcal/mol are found between different levels
of theory. This example demonstrates that care must be taken in choosing
the appropriate method for calculating the heats of formation from total
atomization energies.</p>
<p>The difficulties associated with calculating absolute heats of formation
from atomization energies can be avoided by using a set of isodesmic
reactions[1]. The defining property of an isodesmic reaction - that
there are an equal number of like bonds on the left-hand and right-hand
sides of the reaction - helps to minimize the error in the reaction
energy. These reactions are designed to separate out the interactions
between molecular subsistents and non-bonding electrons from the direct
bonding interactions by having the direct bonding interactions largely
canceling one another. This separation is quite attractive. Most ab
initio methods give substantial errors when estimating direct bonding
interactions due to the computational difficulties associated with
electron pair correlation, whereas ab initio methods are expected to be
more accurate for estimating neighboring interactions and long-range
through-bond effects.</p>
<p>The following isodesmic reaction can be used determine the enthalpy of
formation for CCl<span class="arithmatex">\(_3\)</span>SH that is significantly more accurate than the
estimates based on atomization energies.</p>
<p>CCl<span class="arithmatex">\(_3\)</span>SH + CH<span class="arithmatex">\(_4\)</span> <span class="arithmatex">\(\rightarrow\)</span> CH<span class="arithmatex">\(_3\)</span>SH + CCl<span class="arithmatex">\(_3\)</span>H,
H<span class="arithmatex">\(_r\)</span>(calc).</p>
<p>The first step is to calculate the reaction enthalpy of this reaction
from electronic, thermal and vibrational energy differences at 298.15K
at a consistent level of theory. The defining property of an isodesmic
reaction that there are an equal number of like bonds on the left-hand
and right-hand sides of the reaction helps to minimize the error in the
calculation of the reaction energy. The enthalpy of formation of
CCl<span class="arithmatex">\(_3\)</span>SH can then be calculated by using Hesss law with the
calculated enthalpy change and the experimentally known heats of
formation of the other 3 species (see Table 3).</p>
<p>H<span class="arithmatex">\(_f\)</span>(CCl<span class="arithmatex">\(_3\)</span>SH) = H<span class="arithmatex">\(_f\)</span>(CH<span class="arithmatex">\(_3\)</span>SH)(exp) +
H<span class="arithmatex">\(_f\)</span>(CCl<span class="arithmatex">\(_3\)</span>H)(exp) - H<span class="arithmatex">\(_f\)</span>(CH<span class="arithmatex">\(_4\)</span>)(exp)- H<span class="arithmatex">\(_r\)</span>(calc).</p>
<p>In this example, try to design and run NWPW simulations that can be used
to estimate the enthalpy of formation for CCl<span class="arithmatex">\(_3\)</span>SH using its
atomization energy and using the reaction enthalpy of the isodesmic
reaction and compare your results to Table 2. Be careful to make sure
that you use the same cutoff energy for all the simulations (.e.g.
cutoff 35.0). You might also try to estimate enthalpies of formation for
CHCl<span class="arithmatex">\(_2\)</span>SH and CH<span class="arithmatex">\(_2\)</span>ClSH. Also try designing simulations that use
the SCF, DFT, MP2, and TCE modules.</p>
<p>CCl<span class="arithmatex">\(_3\)</span>SH + CH<span class="arithmatex">\(_4\)</span> <span class="arithmatex">\(\rightarrow\)</span> CH<span class="arithmatex">\(_3\)</span>SH + CCl<span class="arithmatex">\(_3\)</span>H</p>
<p>Un-optimized geometries for CCl<span class="arithmatex">\(_3\)</span>SH, CH<span class="arithmatex">\(_3\)</span>SH, CCl<span class="arithmatex">\(_3\)</span>H and
CH<span class="arithmatex">\(_4\)</span> which are needed to design your simulations are contained in
the file
<a href="thermodynamics.xyz">Media:thermodynamics.xyz</a>. You
will also need to calculate the energies for the H, C, S, and Cl atoms
to calculate the atomization energies. The multiplicities for these
atoms are 2, 3, 3 and 2 respectively. You will also need to calculate
the enthalpy of a molecule. The enthalpy of a molecule at 298.15K is sum
of the total energy and a thermal correction to the enthalpy. A good
estimate for the thermal correction to the enthalpy can be obtained from
a frequency calculation, i.e.</p>
<p>H = E + H<span class="arithmatex">\(_{correction}\)</span></p>
<p>Thermodynamic output from a frequency
calculation:</p>
<pre><code>Temperature                      =   298.15K  
frequency scaling parameter      =   1.0000  

Zero-Point correction to Energy  =   27.528 kcal/mol  (  0.043869 au)  
Thermal correction to Energy     =   29.329 kcal/mol  (  0.046739 au)
</code></pre>
<p>The following line contains the value for
H<span class="arithmatex">\(_{correction}\)</span></p>
<pre><code>Thermal correction to Enthalpy   =   29.922 kcal/mol  (  0.047683 au)

Total Entropy                    =   44.401 cal/mol-K  
  - Translational                =   34.246 cal/mol-K (mol. weight =  16.0313)  
  - Rotational                   =   10.060 cal/mol-K (symmetry #  =       12)  
  - Vibrational                  =    0.095 cal/mol-K  

Cv (constant volume heat capacity) =    6.503 cal/mol-K  
  - Translational                  =    2.979 cal/mol-K  
  - Rotational                     =    2.979 cal/mol-K  
  - Vibrational                    =    0.544 cal/mol-K
</code></pre>
<table>
<thead>
<tr>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td>Compounds</td>
<td>MP2/cc-pVDZ</td>
<td>LDA/DZVP2</td>
<td>BP91/DZVP2</td>
<td>B3LYP/DZVP2</td>
<td>G2</td>
<td>Experiment</td>
</tr>
<tr>
<td></td>
<td>(isodesmic)</td>
<td>(isodesmic)</td>
<td>(isodesmic)</td>
<td>(isodesmic)</td>
<td>(atomization)</td>
<td></td>
</tr>
<tr>
<td>CCl$_3SH</td>
<td>-13.40</td>
<td>-11.86</td>
<td>-8.68</td>
<td>-7.64</td>
<td>-12.95</td>
<td></td>
</tr>
<tr>
<td>CHCl<span class="arithmatex">\(_2\)</span>SH</td>
<td>-11.48</td>
<td>-11.07</td>
<td>-8.66</td>
<td>-7.92</td>
<td>-11.52</td>
<td></td>
</tr>
<tr>
<td>CH<span class="arithmatex">\(_2\)</span>ClSH</td>
<td>-7.01</td>
<td>-6.66</td>
<td>-5.44</td>
<td>-5.20</td>
<td>-6.98</td>
<td></td>
</tr>
<tr>
<td>CH<span class="arithmatex">\(_3\)</span>SH</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>-4.76</td>
<td>-5.34</td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<table id="_table-1">
<caption style="caption-side:bottom"><span>Table&nbsp;1:</span> Gas-phase standard enthalpies of formation ( H</caption>
<thead>
<tr>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td>Compounds</td>
<td>H<span class="arithmatex">\(_f^o\)</span>(298K)</td>
</tr>
<tr>
<td>H</td>
<td>52.095</td>
</tr>
<tr>
<td>C</td>
<td>171.291</td>
</tr>
<tr>
<td>S</td>
<td>66.636</td>
</tr>
<tr>
<td>Cl</td>
<td>29.082</td>
</tr>
<tr>
<td>CCl<span class="arithmatex">\(_4\)</span></td>
<td>-24.59</td>
</tr>
<tr>
<td>CCl<span class="arithmatex">\(_3\)</span>H</td>
<td>-24.65</td>
</tr>
<tr>
<td>CCl<span class="arithmatex">\(_2\)</span>H<span class="arithmatex">\(_2\)</span></td>
<td>-22.10</td>
</tr>
<tr>
<td>CClH<span class="arithmatex">\(_3\)</span></td>
<td>-19.32</td>
</tr>
<tr>
<td>CH<span class="arithmatex">\(_4\)</span></td>
<td>-17.88</td>
</tr>
<tr>
<td>CH<span class="arithmatex">\(_3\)</span>SH</td>
<td>-5.34</td>
</tr>
<tr>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<p>Table 3: Miscellaneous experimental gas-phase enthalpies of formation
(kcal/mol) taken from [3].</p>
<ol>
<li>Hehre, W. J., L. Radom, P.v.R. Schleyer, and J.A. Pople Ab Initio
    Molecular Orbital Theory; John Wiley &amp; Sons: New York, 1986).</li>
<li>E.J. Bylaska, D.A. Dixon, and A.R. Felmy(2000), The Free Energies
    of Reactions of Chlorinated Methanes with Aqueous Monovalent Anions:
    Application of ab initio Electronic Structure Theory, J. Phys.
    Chem. A, 104(3), 610-617.</li>
<li>Chase, M. W., Jr. Phys. Chem. Ref. Data, Monograph No. 9 1998, 9,
    1-1951.</li>
</ol>
<h2 id="plane-wave-density-functional-theory-nwpw-tutorial-4-aimdmm-simulation-of-ccl_4-64-h_2o">NWPW Tutorial 4: AIMD/MM simulation of CCl<span class="arithmatex">\(_4\)</span> + 64 H<span class="arithmatex">\(_2\)</span>O<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-tutorial-4-aimdmm-simulation-of-ccl_4-64-h_2o" title="Permanent link">&para;</a></h2>
<p>(input:<a href="ccl4-64water.nw">Media:ccl4-64water.nw</a>,
output:<a href="ccl4-64water.nwout">Media:ccl4-64water.nwout</a>)</p>
<p>In this section we show how use the PSPW module to perform a
Car-Parrinello AIMD/MM simulation for a CCl<span class="arithmatex">\(_4\)</span> molecule in a box of
64 H<span class="arithmatex">\(_2\)</span>O. Before running a PSPW Car-Parrinello simulation the system
should be on the Born-Oppenheimer surface, i.e. the one-electron
orbitals should be minimized with respect to the total energy (i.e. task
pspw energy). In this example, default pseudopotentials from the
pseudopotential library are used for C, Cl, O^ and H^, exchange
correlation functional is PBE96, The boundary condition is periodic, and
with a side length of 23.577 Bohrs and has a cutoff energy is 50 Ry).
The time step and fake mass for the Car-Parrinello run are specified to
be 5.0 au and 600.0 au, respectively.</p>
<h2 id="plane-wave-density-functional-theory-nwpw-tutorial-5-optimizing-the-unit-cell-and-geometry-of-diamond">NWPW Tutorial 5: Optimizing the Unit Cell and Geometry of Diamond<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-tutorial-5-optimizing-the-unit-cell-and-geometry-of-diamond" title="Permanent link">&para;</a></h2>
<p><img alt="diamong" src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/diamond.png" align=left with="200pt" height="190pt"/>   The PSPW and BAND
codes can be used to determine structures and energies for a wide range
of crystalline systems. It can also be used to generate band structure
and density of state
plots.</p>
<h3 id="plane-wave-density-functional-theory-optimizing-the-unit-cell-and-geometry-for-an-8-atom-supercell-of-diamond-with-pspw">Optimizing the Unit Cell and Geometry for an 8 Atom Supercell of Diamond with PSPW<a class="headerlink" href="#plane-wave-density-functional-theory-optimizing-the-unit-cell-and-geometry-for-an-8-atom-supercell-of-diamond-with-pspw" title="Permanent link">&para;</a></h3>
<p>(input:<a href="diamond-pspw.nw">Media:diamond-pspw.nw</a>,
output:<a href="diamond-pspw.nwout">Media:diamond-pspw.nwout</a>,
<a href="diamond.opt.cif">Media:diamond.opt.cif</a>)</p>
<p>(input:<a href="catom-pspw.nw">Media:catom-pspw.nw</a>,
output:<a href="catom-pspw.nwout">Media:catom-pspw.nwout</a>)</p>
<p>The following example uses the PSPW module to optimize the unit cell and
geometry for a diamond crystal. The fractional coordinates and the unit
cell are defined in the geometry block. The simulation_cell block is
not needed since NWPW automatically uses the unit cell defined in the
geometry
block.</p>
<pre><code>title &quot;Diamond 8 atom cubic cell - geometry and unit cell optimization&quot;  
echo  

permanent_dir ./perm  
scratch_dir   ./scratch  

start diamond  

memory 950 mb  

#**** Enter the geometry using fractional coordinates ****  
geometry center noautosym noautoz print   
  system crystal   
    lat_a 3.56d0   
    lat_b 3.56d0   
    lat_c 3.56d0   
    alpha 90.0d0   
    beta  90.0d0   
    gamma 90.0d0   
  end  
  C -0.50000d0 -0.50000d0 -0.50000d0  
  C  0.00000d0  0.00000d0 -0.50000d0  
  C  0.00000d0 -0.50000d0  0.00000d0  
  C -0.50000d0  0.00000d0  0.00000d0  
  C -0.25000d0 -0.25000d0 -0.25000d0  
  C  0.25000d0  0.25000d0 -0.25000d0  
  C  0.25000d0 -0.25000d0  0.25000d0  
  C -0.25000d0  0.25000d0  0.25000d0  
end  

nwpw   
  ewald_rcut 3.0  
  ewald_ncut 8  #The default value of 1 needs to be increased for small cells  
  lmbfgs  
  xc pbe96  
end  

driver   
  clear   
  maxiter 40  
end  

set nwpw:cif_filename diamond.opt  # create a CIF file containing optimization history   
set includestress .true.           # this option tells driver to optimize the unit cell  
task pspw optimize ignore
</code></pre>
<p>The optimized energy and geometry will
be</p>
<pre><code> ...  
     ----------------------  
     Optimization converged  
     ----------------------  

  Step       Energy      Delta E   Gmax     Grms     Xrms     Xmax   Walltime  
  ---- ---------------- -------- -------- -------- -------- -------- --------  
@    6     -45.07688304 -1.1D-07  0.00037  0.00021  0.00002  0.00003    174.5  
                                    ok       ok       ok       ok    



                        Geometry &quot;geometry&quot; -&gt; &quot;geometry&quot;  
                        ---------------------------------  

Output coordinates in angstroms (scale by  1.889725989 to convert to a.u.)  

 No.       Tag          Charge          X              Y              Z  
---- ---------------- ---------- -------------- -------------- --------------  
   1 C                    6.0000     1.82723789     1.82729813     1.82705440  
   2 C                    6.0000     0.00000857    -0.00006053     1.82730027  
   3 C                    6.0000    -0.00000584     1.82706061     0.00002852  
   4 C                    6.0000     1.82712018     0.00006354    -0.00002544  
   5 C                    6.0000     2.74074195     2.74072805     2.74088522  
   6 C                    6.0000     0.91366407     0.91370055     2.74064976  
   7 C                    6.0000     0.91351181     2.74080771     0.91352917  
   8 C                    6.0000     2.74078843     0.91348115     0.91365446  

     Lattice Parameters   
     ------------------   

     lattice vectors in angstroms (scale by  1.889725989 to convert to a.u.)  

     a1=&lt;   3.654   0.000   0.000 &gt;  
     a2=&lt;   0.000   3.654   0.000 &gt;  
     a3=&lt;   0.000   0.000   3.654 &gt;  
     a=       3.654 b=      3.654 c=       3.654  
     alpha=  90.000 beta=  90.000 gamma=  90.000  
     omega=    48.8  

     reciprocal lattice vectors in a.u.  

     b1=&lt;   0.910   0.000   0.000 &gt;  
     b2=&lt;   0.000   0.910   0.000 &gt;  
     b3=&lt;   0.000   0.000   0.910 &gt;  

     Atomic Mass   
     -----------   

     C                 12.000000  


==============================================================================  
                               internuclear distances  
------------------------------------------------------------------------------  
      center one      |      center two      | atomic units |  angstroms  
------------------------------------------------------------------------------  
   5 C                |   1 C                |     2.99027  |     1.58238  
   6 C                |   1 C                |     2.99027  |     1.58238  
   6 C                |   2 C                |     2.99027  |     1.58238  
   7 C                |   1 C                |     2.99026  |     1.58238  
   7 C                |   3 C                |     2.99027  |     1.58238  
   8 C                |   1 C                |     2.99027  |     1.58238  
   8 C                |   4 C                |     2.99027  |     1.58238  
------------------------------------------------------------------------------  
                        number of included internuclear distances:          7  
==============================================================================  

==============================================================================  
                                internuclear angles  
------------------------------------------------------------------------------  
       center 1       |       center 2       |       center 3       |  degrees  
------------------------------------------------------------------------------  
   5 C                |   1 C                |   6 C                |   109.46  
   5 C                |   1 C                |   7 C                |   109.48  
   5 C                |   1 C                |   8 C                |   109.48  
   6 C                |   1 C                |   7 C                |   109.47  
   6 C                |   1 C                |   8 C                |   109.46  
   7 C                |   1 C                |   8 C                |   109.48  
   1 C                |   6 C                |   2 C                |   109.48  
   1 C                |   7 C                |   3 C                |   109.47  
   1 C                |   8 C                |   4 C                |   109.47  
------------------------------------------------------------------------------  
                           number of included internuclear angles:          9  
==============================================================================  ...
</code></pre>
<p>The C-C bond distance after the geometry optimization is 1.58 Angs. and
agrees very well with the experimental value of 1.54 Angs.. Another
quantity that can be calculated from this simulation is the cohesive
energy.The cohesive energy of a crystal is the energy needed to separate
the atoms of the solid into isolated atoms,
i.e.</p>
<div class="arithmatex">\[ E_{coh}= -\left( E_{solid} - \sum_{a} E_{atom}^a  \right)  \]</div>
<p>where <span class="arithmatex">\(E_{solid}\)</span> is the energy of the solid and <span class="arithmatex">\(E_{atom}^a\)</span> are
the energies of the isolated atoms. In order to calculate the cohesive
energy the energy of an isolated carbon atom at the same level of theory
and cutoff energy will need to be calculated. The following input can be
used to the energy of an isolated carbon atom.</p>
<p>(input:<a href="catom-pspw.nw">file:catom-pspw.nw</a>,
output:<a href="catom-pspw.nwout">file:catom-pspw.nwout</a>)</p>
<pre><code>title &quot;triplet carbon atom at pbe96 level using a large unit cell&quot;  
start c1-pspw  
memory 1400 mb  

permanent_dir ./perm  
scratch_dir   ./scratch  

geometry  
C 0 0 0  
end  

nwpw  
   simulation_cell  
      FCC 38.0     #large unit cell   
      boundary_conditions   periodic  # periodic boundary conditions are used by default.  
      #boundary_conditions aperiodic  # free-space (or aperiodic) boundary conditions could also be used.  
   end  
   xc pbe96  
   mult 3  
  lmbfgs  
end  
task pspw energy
</code></pre>
<p>The total energy from the simulation will be</p>
<p><code>Total PSPW energy   :  -0.5421213534E+01</code></p>
<p>Using this energy and energy of diamond the cohesive energy per atom is
calculated to be</p>
<div class="arithmatex">\[\begin{align} E_{coh}= -\left( -45.07688304au/8 - (-5.421213534au)  \right)= 0.2133968 au = 5.8 eV \end{align}\]</div>
<p>This value is substantially lower than the experimental value of
<span class="arithmatex">\(7.37eV\)</span>! It turns out this error is a result of the unit cell being
too small for the diamond calculation (or too small of a Brillioun zone
sampling). In the next section, we show how increasing the Brillouin
zone sampling reduces the error in the calculated cohesive
energy.</p>
<h3 id="plane-wave-density-functional-theory-optimizing-the-unit-cell-for-an-8-atom-supercell-of-diamond-with-band">Optimizing the Unit Cell for an 8 Atom Supercell of Diamond with BAND<a class="headerlink" href="#plane-wave-density-functional-theory-optimizing-the-unit-cell-for-an-8-atom-supercell-of-diamond-with-band" title="Permanent link">&para;</a></h3>
<p>(input:<a href="diamond-band.nw">Media:diamond-band.nw</a>,
output:<a href="diamond-band.nwout">Media:diamond-band.nwout</a>)</p>
<p>In this example the BAND module is used to optimize the unit cell and
geometry for a diamond crystal at different Brillouin zone
samplings.</p>
<pre><code>title &quot;Diamond 8 atom cubic cell - geometry and unit cell optimization&quot;   
echo  

permanent_dir ./perm  
scratch_dir   ./scratch  

start diamond-band  

memory 1950 mb  

#**** Enter the geometry using fractional coordinates ****  
geometry center noautosym noautoz print   
  system crystal   
    lat_a 3.58d0   
    lat_b 3.58d0   
    lat_c 3.58d0   
    alpha 90.0d0   
    beta  90.0d0   
    gamma 90.0d0   
  end  
  C -0.50000d0 -0.50000d0 -0.50000d0  
  C  0.00000d0  0.00000d0 -0.50000d0  
  C  0.00000d0 -0.50000d0  0.00000d0  
  C -0.50000d0  0.00000d0  0.00000d0  
  C -0.25000d0 -0.25000d0 -0.25000d0  
  C  0.25000d0  0.25000d0 -0.25000d0  
  C  0.25000d0 -0.25000d0  0.25000d0  
  C -0.25000d0  0.25000d0  0.25000d0  
end  
set includestress    .true.   # option tells driver to optimize the unit cell  
set nwpw:zero_forces .true.   # option zeros the forces on the atoms--&gt; only lattice parameters optimized  

nwpw   
  ewald_rcut 3.0  
  ewald_ncut 8    #The default value of 1 needs to be increased  
  lmbfgs  
  xc pbe96  
end  

#1x1x1 k-point mesh  
nwpw   
  monkhorst-pack 1 1 1  
end  
set nwpw:cif_filename diamond111.opt  
driver; clear; maxiter 40; end; task band optimize ignore

#2x2x2 k-point mesh  
nwpw   
  monkhorst-pack 2 2 2  
end  
set nwpw:cif_filename diamond222.opt  
driver; clear; maxiter 40; end; task band optimize ignore

#3x3x3 k-point mesh  
nwpw   
  monkhorst-pack 3 3 3  
end  
set nwpw:cif_filename diamond333.opt  
driver; clear; maxiter 40; end; task band optimize ignore

#4x4x4 k-point mesh  
nwpw   
  monkhorst-pack 4 4 4   
end  
set nwpw:cif_filename diamond444.opt  
driver; clear; maxiter 40; end; task band optimize ignore

#5x5x5 k-point mesh  
nwpw   
  monkhorst-pack 5 5 5  
end  
set nwpw:cif_filename diamond555.opt  
driver; clear; maxiter 40; end; task band optimize ignore
</code></pre>
<p>The following figure shows a plot of the cohesive energy and C-C bond
distance versus the Brillouin zone sampling. As can be seen in this
figure the cohesive energy (w/o zero-point correction) and C-C bond
distance agree very well with the experimental values of 7.37 eV
(including zero-point correction) and 1.54
Angs.</p>
<p><img alt=" " src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/ecohdist.png" align=middle with="792pt" height="612pt"/>  </p>
<h3 id="plane-wave-density-functional-theory-using-band-to-optimize-the-unit-cell-for-a-2-atom-primitive-cell-of-diamond">Using BAND to Optimize the Unit Cell for a 2 Atom Primitive Cell of Diamond<a class="headerlink" href="#plane-wave-density-functional-theory-using-band-to-optimize-the-unit-cell-for-a-2-atom-primitive-cell-of-diamond" title="Permanent link">&para;</a></h3>
<p>(input:<a href="diamond-fcc.nw">Media:diamond-fcc.nw</a>,
output:<a href="diamond-fcc.nwout.gz">Media:diamond-fcc.nwout.gz</a>)</p>
<p>In this example the BAND module is used to optimize a 2 atom unit cell
for a diamond crystal at different Brillouin zone samplings. The
optimized energy and geometry will be (Monkhorst-Pack sampling of
11x11x11)</p>
<pre><code>     ----------------------  
     Optimization converged  
     ----------------------  

  Step       Energy      Delta E   Gmax     Grms     Xrms     Xmax   Walltime  
  ---- ---------------- -------- -------- -------- -------- -------- --------  
@    1     -11.40586236  5.2D-07  0.00039  0.00018  0.00002  0.00003    662.0  
                                     ok       ok       ok       ok    



                         Geometry &quot;geometry&quot; -&gt; &quot;geometry&quot;  
                         ---------------------------------  

 Output coordinates in angstroms (scale by  1.889725989 to convert to a.u.)  

  No.       Tag          Charge          X              Y              Z  
 ---- ---------------- ---------- -------------- -------------- --------------  
    1 C                    6.0000     0.00000000     0.00000000     0.00000000  
    2 C                    6.0000     0.72201500     1.25056532     0.51054180  

      Lattice Parameters   
      ------------------   

      lattice vectors in angstroms (scale by  1.889725989 to convert to a.u.)  

      a1=&lt;   2.165   1.251   0.001 &gt;  
      a2=&lt;   0.001   2.500   0.001 &gt;  
      a3=&lt;   0.722   1.251   2.041 &gt;  
      a=       2.500 b=      2.500 c=       2.500  
      alpha=  59.966 beta=  59.966 gamma=  59.966  
      omega=    11.0  

      reciprocal lattice vectors in a.u.  

      b1=&lt;   1.536  -0.768   0.000 &gt;  
      b2=&lt;   0.000   1.330   0.000 &gt;  
      b3=&lt;  -0.543  -0.543   1.629 &gt;  

      Atomic Mass   
      -----------   

      C                 12.000000  


 ==============================================================================  
                                internuclear distances  
 ------------------------------------------------------------------------------  
       center one      |      center two      | atomic units |  angstroms  
 ------------------------------------------------------------------------------  
    2 C                |   1 C                |     2.89435  |     1.53162  
 ------------------------------------------------------------------------------  
                         number of included internuclear distances:          1  
 ==============================================================================
</code></pre>
<p>The following figure shows a plot of the cohesive energy and C-C bond
distance versus the Brillouin zone sampling for the 8 atom SC unit cell
and the 2 atom FCC unit
cell.</p>
<p><img alt=" " src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/band-fcc.png" align=middle with="792pt" height="612pt"/>  </p>
<h3 id="plane-wave-density-functional-theory-using-band-to-calculate-the-band-structures-of-diamond">Using BAND to Calculate the Band Structures of Diamond<a class="headerlink" href="#plane-wave-density-functional-theory-using-band-to-calculate-the-band-structures-of-diamond" title="Permanent link">&para;</a></h3>
<p>(input:<a href="diamond-structure.nw">Media:diamond-structure.nw</a>,
output:<a href="diamond-structure.nwout">Media:diamond-structure.nwout</a>,
<a href="diamondfcc.restricted_band.dat">file:diamondfcc.restricted_band.dat</a>)</p>
<p>The following example uses the BAND module to calculate the band
structure for the FCC cell of the a diamond crystal. The fractional
coordinates and the unit cell are defined in the geometry block. The
simulation_cell block is not needed since NWPW automatically uses the
unit cell defined in the geometry
block.</p>
<pre><code>title &quot;Diamond 2 atom fcc cell Brillouin sampling=9x9x9 M-P - Band structure plot&quot;  
echo  

permanent_dir ./perm  
scratch_dir   ./scratch  

start diamondfcc  

memory 1950 mb  

#**** Enter the geometry using fractional coordinates ****  
geometry center noautosym noautoz print   
  system crystal   
    lat_a 2.500d0   
    lat_b 2.500d0   
    lat_c 2.500d0   
    alpha 60.0d0   
    beta  60.0d0   
    gamma 60.0d0   
  end  
 C  0.00000d0  0.00000d0  0.00000d0  
 C  0.25000d0  0.25000d0  0.25000d0  
end  

nwpw  
  ewald_rcut 3.0  
  ewald_ncut 8    #The default value of 1 needs to be increased  
  lmbfgs  
  xc pbe96  

  monkhorst-pack 9 9 9  
end  

#need to run &quot;task band energy&quot; before &quot;task band structure&quot; can be run  
task band energy  

nwpw  
   virtual 16  
   brillouin_zone  
     zone_name fccpath  
     path fcc l gamma x w k gamma  
   end  
   zone_structure_name fccpath  
end  
task band structure
</code></pre>
<p>This calculation outputs the
<a href="diamondfcc.restricted_band.dat">file:diamondfcc.restricted_band.dat</a>)
data file in the permanent_directory. A plotting (e.g. gnuplot or
xmgrace) can be used to display the band structure.</p>
<p><img alt=" " src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/band-structure.png" align=middle with="792pt" height="612pt"/>  </p>
<h3 id="plane-wave-density-functional-theory-using-band-to-calculate-the-density-of-states-of-diamond">Using BAND to Calculate the Density of States of Diamond<a class="headerlink" href="#plane-wave-density-functional-theory-using-band-to-calculate-the-density-of-states-of-diamond" title="Permanent link">&para;</a></h3>
<p>(2 atom cell -
input:<a href="diamond-dos.nw">diamond-dos.nw</a>
output:<a href="diamond-dos.nwout">diamond-dos.nwout</a>,
<a href="diamond-dos.dos.dat">diamond-dos.dos.dat</a> (8
atom cell -
input:<a href="diamond-dos8.nw">diamond-dos8.nw</a> output:
<a href="diamond-dos8.nwout.gz">diamond-dos8.nwout.gz</a>,
<a href="diamond-dos8.dos.dat">diamond-dos8.dos.dat</a></p>
<p>There are two possible ways to use the BAND module to calculate the density and projected density of states.  The first approach just uses the eigenvalues generated from an energy calculation to generate a density of states.  The following example uses this strategy to calculate the density of states and projected density of states of diamond.</p>
<pre><code>title &quot;Diamond 2 atom fcc cell Brillouin sampling=9x9x9 M-P - density of states plot&quot;  
echo  

permanent_dir ./perm  
scratch_dir   ./scratch  

start diamond-dos  

memory 1950 mb  

#**** Enter the geometry using fractional coordinates ****  
geometry center noautosym noautoz print   
  system crystal   
    lat_a 2.500d0   
    lat_b 2.500d0   
    lat_c 2.500d0   
    alpha 60.0d0   
    beta  60.0d0   
    gamma 60.0d0   
  end  
 C  0.00000d0  0.00000d0  0.00000d0  
 C  0.25000d0  0.25000d0  0.25000d0  
end  

nwpw  
  ewald_rcut 3.0  
  ewald_ncut 8    #The default value of 1 needs to be increased  
  lmbfgs  
  xc pbe96  

  monkhorst-pack 9 9 9  
  dos                   # dos keyword tells the code to calculate dos at the end of an energy calculation
  mulliken              # turn on projected density of states
  virtual 8             # include 8 virtual states
end  

task band energy  
</code></pre>
<p>The other approach uses the band structure code to calculate the eigenvalues given a precomputed density.  The approach is slower than the first approach, however, it can be used to substantially increase the number of k-points and virtual orbitals used to generate the density of states.  The following example demonstrates this capability to calculate the density of states and projected density of states of the diamond crystal.</p>
<pre><code>title &quot;Diamond 2 atom fcc cell Brillouin sampling=9x9x9 M-P - density of states plot&quot;  
echo  

permanent_dir ./perm  
scratch_dir   ./scratch  

start diamond-dos  

memory 1950 mb  

#**** Enter the geometry using fractional coordinates ****  
geometry center noautosym noautoz print   
  system crystal   
    lat_a 2.500d0   
    lat_b 2.500d0   
    lat_c 2.500d0   
    alpha 60.0d0   
    beta  60.0d0   
    gamma 60.0d0   
  end  
 C  0.00000d0  0.00000d0  0.00000d0  
 C  0.25000d0  0.25000d0  0.25000d0  
end  

nwpw  
  ewald_rcut 3.0  
  ewald_ncut 8    #The default value of 1 needs to be increased  
  lmbfgs  
  xc pbe96  

  monkhorst-pack 9 9 9  
end  

#need to run &quot;task band energy&quot; before &quot;task band dos&quot; can be run  
task band energy  

nwpw  
   virtual 26                     #26 virtual orbitals included in the DOS calculation  
   dos 0.002 700 -1.00000 2.0000  #alpha npoints emin emax,....,change default energy range and gridding. note alpha not used in task band dos calculations
   dos-grid 11 11 11  
   mulliken                       # mulliken keyword used to turn on projected density of states
end  
task band dos
</code></pre>
<p>This calculation outputs the <img alt="diamond-dos.dos.dat" src="diamond-dos.dos.dat" /> data file in the permanent_directory. A plotting
(e.g. gnuplot or xmgrace) can be used to display the density of
states.</p>
<p><img alt=" " src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/band-dos.png" align=middle with="792pt" height="612pt"/>  </p>
<h3 id="plane-wave-density-functional-theory-calculate-the-phonon-spectrum-of-diamond">Calculate the Phonon Spectrum of Diamond<a class="headerlink" href="#plane-wave-density-functional-theory-calculate-the-phonon-spectrum-of-diamond" title="Permanent link">&para;</a></h3>
<pre><code>title &quot;Diamond 2 atom fcc cell Brillouin sampling=9x9x9 M-P - Phonon spectra&quot;  
echo  

permanent_dir ./perm  
scratch_dir   ./scratch  

start diamond-dos  

memory 1950 mb  

#**** Enter the geometry using fractional coordinates ****  
geometry center noautosym noautoz print   
  system crystal   
    lat_a 2.500d0   
    lat_b 2.500d0   
    lat_c 2.500d0   
    alpha 60.0d0   
    beta  60.0d0   
    gamma 60.0d0   
  end  
 C  0.00000d0  0.00000d0  0.00000d0  
 C  0.25000d0  0.25000d0  0.25000d0  
end  

nwpw  
  ewald_rcut 3.0  
  ewald_ncut 8    #The default value of 1 needs to be increased  
  lmbfgs  
  xc pbe96  

  monkhorst-pack 9 9 9  
end  

task band energy  
task band freq

</code></pre>
<h2 id="plane-wave-density-functional-theory-nwpw-tutorial-6-optimizing-the-unit-cell-of-nickel-with-fractional-occupation">NWPW Tutorial 6: optimizing the unit cell of nickel with fractional occupation<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-tutorial-6-optimizing-the-unit-cell-of-nickel-with-fractional-occupation" title="Permanent link">&para;</a></h2>
<p>(input:<a href="Ni-band.nw">Media:Ni-band.nw</a>
output:<a href="Ni-band.nwout">Media:Ni-band.nwout</a>) The
following example demonstrates how to uses the BAND module to optimize
the unit cell and geometry for FCC cell of Nickel
metal</p>
<pre><code>title &quot;Ni FCC metal, monkhorst-pack=3x3x3, 5x5x5, and 7x7x7, fermi smearing, xc=pbe96&quot;  
echo  

start Ni-band  

memory 1900 mb  

permanent_dir ./perm  
scratch_dir   ./scratch  

geometry units angstroms center noautosym noautoz print  
 system crystal  
    lat_a 3.5451d0  
    lat_b 3.5451d0  
    lat_c 3.5454d0  
    alpha 90.0d0  
    beta  90.0d0  
    gamma 90.0d0  
  end  

Ni 0.000000   0.000000   0.000000  
Ni 0.000000   0.500000   0.500000   
Ni 0.500000   0.000000   0.500000   
Ni 0.500000   0.500000   0.000000  
end  
set nwpw:cif_filename Ni-band  
set nwpw:zero_forces .true.  
set includestress    .true.  

#turn on pseudopotential filtering   
set nwpw:kbpp_ray .true.  
set nwpw:kbpp_filter .true.  

nwpw  
   #fractional occupation  
   smear fermi   

   #scf option used with smear  
   scf anderson outer_iterations 0 kerker 2.0    

   ewald_ncut 8  
   ewald_rcut 3.0  
   xc pbe96  
   monkhorst-pack 3 3 3  
   np_dimensions -1 -1 4   
end  

#generate initial wavefunctions w/ low cutoff energy  
nwpw  
   loop 10 10  
   cutoff 10.0  
end  
task band energy  

#increase cutoff energy and number of iterations  
nwpw  
   cutoff 50.0  
   loop 10 100  
end

#3x3x3 k-point mesh  
nwpw   
  monkhorst-pack 3 3 3  
end  
set nwpw:cif_filename nickel333.opt  
driver; clear; maxiter 40; end; task band optimize ignore

#5x5x5 k-point mesh  
nwpw   
  monkhorst-pack 5 5 5  
end  
set nwpw:cif_filename nickel555.opt  
driver; clear; maxiter 40; end; task band optimize ignore

#7x7x7 k-point mesh  
nwpw   
  monkhorst-pack 7 7 7  
end  
set nwpw:cif_filename nickel777.opt  
driver; clear; maxiter 40; end; task band optimize ignore
</code></pre>
<p>The following figure shows a plot of the cohesive energy and Ni-Ni bond
distance versus the Brillouin zone sampling. As can be seen in this
figure the cohesive energy (w/o zero-point correction) and Ni-Ni bond
distance agree very well with the experimental values of 4.44 eV
(including zero-point correction) and 2.49
Angs.</p>
<p><img alt=" " src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/niecohdist.png" align=middle with="792pt" height="612pt"/>  </p>
<h2 id="plane-wave-density-functional-theory-nwpw-tutorial-7-optimizing-the-unit-cells-with-symmetry-diamond-with-fd-3m-symmetry-and-brucite-with-p-3m1-symmetry">NWPW Tutorial 7: Optimizing the unit cells with symmetry: Diamond with Fd-3m symmetry and Brucite with P-3m1 symmetry<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-tutorial-7-optimizing-the-unit-cells-with-symmetry-diamond-with-fd-3m-symmetry-and-brucite-with-p-3m1-symmetry" title="Permanent link">&para;</a></h2>
<p>(Diamond example,
input:<a href="diamond-symmetry.nw">Media:diamond-symmetry.nw</a>,
output:<a href="diamond-symmetry.nwout">Media:diamond-symmetry.nwout</a>)</p>
<p>(Brucite example,
input:<a href="brucite-symmetry.nw">Media:brucite-symmetry.nw</a>,
output:<a href="brucite-symmetry.nwout">Media:brucite-symmetry.nwout</a>)</p>
<p>The following example uses the BAND module to optimize the unit cell and
geometry for a Diamond crystal with Fd-3m symmetry. The fractional
coordinates, unit cell, and symmetry are defined in the geometry
block.</p>
<pre><code>title &quot;Diamond 8 atom cubic cell generated using Fd-3m symmetry - geometry and unit cell optimization&quot; 
echo 

memory 1500 mb

permanent_dir ./perm
scratch_dir   ./scratch

start diamond-symmetry  


geometry nocenter noautosym noautoz print 
 system crystal 
   lat_a 3.58 
   lat_b 3.58 
   lat_c 3.58  
   alpha 90.0  
   beta  90.0  
   gamma 90.0  
 end
symmetry Fd-3m
C 0.0 0.0 0.0
end 
set nwpw:cif_filename diamond-symmetry

#turn on pseudopotential filtering  
set nwpw:kbpp_ray    .true.
set nwpw:kbpp_filter .true.

#***** setup the nwpw Band code - 3x3x3 k-point mesh ****  
nwpw
  ewald_rcut 3.0
  ewald_ncut 8
  xc pbe96  
  lmbfgs
  monkhorst-pack 3 3 3
  np_dimensions -1 -1 4
end  

set includestress  .true.   # tell driver to optimize unit cell
set includelattice .true.   # tell driver to optimize with a,b,c,alpha,beta,gamma
task band optimize ignore
</code></pre>
<p>The optimized geometry will also contain the information about the
symmetry being
used</p>
<pre><code>....  
     ----------------------  
     Optimization converged  
     ----------------------  


 Step       Energy      Delta E   Gmax     Grms     Xrms     Xmax   Walltime  
 ---- ---------------- -------- -------- -------- -------- -------- --------  
@    7     -45.62102901 -4.1D-07  0.00010  0.00003  0.00019  0.00060    287.1  
                                     ok       ok       ok       ok    



                        Geometry &quot;geometry&quot; -&gt; &quot;geometry&quot;  
                        ---------------------------------  

Output coordinates in angstroms (scale by  1.889725989 to convert to a.u.)  

 No.       Tag          Charge          X              Y              Z  
---- ---------------- ---------- -------------- -------------- --------------  
   1 C                    6.0000     0.00000000     0.00000000     0.00000000  
   2 C                    6.0000     0.00000000     1.76715074     1.76715074  
   3 C                    6.0000     1.76715074     1.76715074     0.00000000  
   4 C                    6.0000     1.76715074     0.00000000     1.76715074  
   5 C                    6.0000     2.65072611     0.88357537     2.65072611  
   6 C                    6.0000     0.88357537     0.88357537     0.88357537  
   7 C                    6.0000     0.88357537     2.65072611     2.65072611  
   8 C                    6.0000     2.65072611     2.65072611     0.88357537  

     Lattice Parameters   
     ------------------   

     lattice vectors in angstroms (scale by  1.889725989 to convert to a.u.)  

     a1=&lt;   3.534   0.000   0.000 &gt;  
     a2=&lt;   0.000   3.534   0.000 &gt;  
     a3=&lt;   0.000   0.000   3.534 &gt;  
     a=       3.534 b=      3.534 c=       3.534  
     alpha=  90.000 beta=  90.000 gamma=  90.000  
     omega=    44.1  

     reciprocal lattice vectors in a.u.  

     b1=&lt;   0.941   0.000   0.000 &gt;  
     b2=&lt;   0.000   0.941   0.000 &gt;  
     b3=&lt;   0.000   0.000   0.941 &gt;  

     Atomic Mass   
     -----------   

     C                 12.000000  


     Symmetry information  
     --------------------  

Group name                  Fd-3m  
Group number            227  
Group order             192  
No. of unique centers     1  
Setting number            1  

     Symmetry unique atoms  

    1  

==============================================================================  
                               internuclear distances  
------------------------------------------------------------------------------  
      center one      |      center two      | atomic units |  angstroms  
------------------------------------------------------------------------------  
   5 C                |   4 C                |     2.89203  |     1.53040  
   6 C                |   1 C                |     2.89203  |     1.53040  
   6 C                |   2 C                |     2.89203  |     1.53040  
   6 C                |   3 C                |     2.89203  |     1.53040  
   6 C                |   4 C                |     2.89203  |     1.53040  
   7 C                |   2 C                |     2.89203  |     1.53040  
   8 C                |   3 C                |     2.89203  |     1.53040  
------------------------------------------------------------------------------  
                        number of included internuclear distances:          7  
==============================================================================  



==============================================================================  
                                internuclear angles  
------------------------------------------------------------------------------  
       center 1       |       center 2       |       center 3       |  degrees  
------------------------------------------------------------------------------  
   6 C                |   2 C                |   7 C                |   109.47  
   6 C                |   3 C                |   8 C                |   109.47  
   5 C                |   4 C                |   6 C                |   109.47  
   1 C                |   6 C                |   2 C                |   109.47  
   1 C                |   6 C                |   3 C                |   109.47  
   1 C                |   6 C                |   4 C                |   109.47  
   2 C                |   6 C                |   3 C                |   109.47  
   2 C                |   6 C                |   4 C                |   109.47  
   3 C                |   6 C                |   4 C                |   109.47  
------------------------------------------------------------------------------  
                           number of included internuclear angles:          9  
==============================================================================
</code></pre>
<p>The following example uses the BAND module to optimize the unit cell and
geometry for a <a href="http://en.wikipedia.org/wiki/Brucite">Brucite</a> crystal
(Mg(OH)<sub>2</sub> with P-3m1
symmetry.</p>
<p><img alt="400px" src="Brucite-small.jpg" /></p>
<pre><code>title &quot;brucite testing - using P-3m1 symmetry&quot;  
echo  

memory 1500 mb  

permanent_dir ./perm  
scratch_dir      ./scratch  

geometry nocenter noautosym noautoz print   
 system crystal   
   lat_a 3.14979  
   lat_b 3.14979  
   lat_c 4.7702  
   alpha 90.0  
   beta  90.0  
   gamma 120.0  
 end  
symmetry P-3m1  
Mg   0.00000   0.00000  0.00000  
O   0.33333   0.66667   0.22030  
H   0.33333   0.66667   0.41300  
end   
set nwpw:cif_filename brucite  

#turn on pseudopotential filtering   
set nwpw:kbpp_ray .true.  
set nwpw:kbpp_filter .true.  

#***** setup the nwpw gamma point code ****  
nwpw  
  ewald_rcut 3.0  
  ewald_ncut 8  
  xc pbe96  
  lmbfgs  
  monkhorst-pack 3 3 2  
  #np_dimensions -1 -1 4  
end  

driver  
  clear  
  maxiter 31  
end  

set includestress  .true.          # tell driver to optimize unit cell  
set includelattice .true.  

task band optimize ignore
</code></pre>
<p>Optimizing Brucite, which is a soft layered material (2.5-3 Mohs scale),
is more difficult to optimize than a hard material such as Diamond. For
these types of materials using symmetry can often result in a faster
optimization. For example, with symmetry the optimization converges
within 20 to 30 geometry optimization
steps,</p>
<pre><code>@ Step       Energy      Delta E   Gmax     Grms     Xrms     Xmax   Walltime  
@ ---- ---------------- -------- -------- -------- -------- -------- --------  
@    0     -34.39207476  0.0D+00  0.24673  0.10223  0.00000  0.00000    172.7  
@    1     -34.39340208 -1.3D-03  0.00872  0.00302  0.00198  0.00485    328.5  
....  
@   20     -34.39042736 -1.2D-05  0.00195  0.00083  0.00440  0.01964   3019.2  
@   21     -34.39043463 -7.3D-06  0.00028  0.00011  0.00493  0.02042   3150.6  
@   22     -34.39043484 -2.1D-07  0.00043  0.00014  0.00002  0.00008   3278.5  
@   22     -34.39043484 -2.1D-07  0.00043  0.00014  0.00002  0.00008   3278.5
</code></pre>
<p>whereas, without symmetry the optimization may not be converged even at
100 geometry steps
(input:<a href="brucite-nosymmetry.nw">Media:brucite-nosymmetry.nw</a>,
output:<a href="brucite-nosymmetry.nwout">Media:brucite-nosymmetry.nwout</a>).</p>
<pre><code>@ Step       Energy      Delta E   Gmax     Grms     Xrms     Xmax   Walltime  
@ ---- ---------------- -------- -------- -------- -------- -------- --------  
@    0     -34.39207476  0.0D+00  0.24673  0.10250  0.00000  0.00000     18.4  
@    1     -34.39340765 -1.3D-03  0.02963  0.00715  0.00202  0.00500     30.7  
...  
@   49     -34.39027641 -2.1D-06  0.01870  0.00646  0.00074  0.00202    595.7  
@   50     -34.39027503  1.4D-06  0.01962  0.00669  0.00069  0.00197    608.4  
...  
@  100     -34.39034236 -3.8D-07  0.00380  0.00150  0.00036  0.00132   1155.3  
@  101     -34.39034431 -1.9D-06  0.00305  0.00118  0.00012  0.00045   1166.8  
@  102     -34.39034449 -1.8D-07  0.00370  0.00144  0.00006  0.00020   1177.9  
...
</code></pre>
<h2 id="plane-wave-density-functional-theory-nwpw-tutorial-8-nvt-metropolis-monte-carlo-simulations">NWPW Tutorial 8: NVT Metropolis Monte-Carlo Simulations<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-tutorial-8-nvt-metropolis-monte-carlo-simulations" title="Permanent link">&para;</a></h2>
<p>In this example the PSPW module is used to run an NVT simulation for a
diamond crystal using the a Metropolis Monte-Carlo algorithm.</p>
<!-- (input:[Media:diamond-nvt.nw](diamond-nvt.nw.md),
output:[Media:diamond-nvt.nwout.gz](diamond-nvt.nwout.gz.md),
datafiles:[Media:diamond-nvt.emotion.gz](diamond-nvt.emotion.gz.md),
[Media:diamond-nvt.ion_motion.gz](diamond-nvt.ion_motion.gz.md),
[Media:diamond-nvt.xyz.gz](diamond-nvt.xyz.gz.md),
[Media:diamond_nvt_234.cif.gz](diamond_nvt_234.cif.gz.md)) -->

<p><img alt="NVT Metropolis Monte-Carlo Trajectory (Markov Chain) for diamond,
T=300K." src="diamond-nvt.gif" title="NVT Metropolis Monte-Carlo Trajectory (Markov Chain.md) for diamond, T=300K." /></p>
<pre><code>title &quot;Metropolis NVT simulation of diamond - this input is used to put the system in equilibrium&quot;  
echo  

start diamond-nvt  

#permanent_dir ./perm  
#scratch_dir   ./perm  

#**** Enter the geometry using fractional coordinates ****  
geometry center noautosym noautoz print   
system crystal   
   lat_a 3.56d0   
   lat_b 3.56d0   
   lat_c 3.56d0   
   alpha 90.0d0   
   beta  90.0d0   
   gamma 90.0d0   
end  
C -0.50000d0 -0.50000d0 -0.50000d0  
C  0.00000d0  0.00000d0 -0.50000d0  
C  0.00000d0 -0.50000d0  0.00000d0  
C -0.50000d0  0.00000d0  0.00000d0  
C -0.25000d0 -0.25000d0 -0.25000d0  
C  0.25000d0  0.25000d0 -0.25000d0  
C  0.25000d0 -0.25000d0  0.25000d0  
C -0.25000d0  0.25000d0  0.25000d0  
end  
set nwpw:cif_filename diamond_nvt_234  

###### setup the nwpw gamma point code ######  
set nwpw:kbpp_ray .true.  
set nwpw:kbpp_filter .true.  
set nwpw:frozen_lattice:thresh 999.0  
nwpw  
   lmbfgs  
   ewald_rcut 3.0  
   ewald_ncut 8  
   xc pbe  
end  
task pspw energy  

##### optimize the unit cell #####  
set includestress  .true.  #this option tells driver to optimize the unit cell  
set includelattice .true.  #this option tells driver to optimize cell using a,b,c,alpha,beta,gamma   
driver  
   clear  
   maxiter 51  
end  
task pspw optimize ignore  

####################################################################################  
###### setup Metropolis NVT code - input will change in a forthcoming release ######  
####################################################################################  
set nwpw:mc_seed 234              # Seed for random number generator  
set nwpw:mc_algorithm 1            # 1-NVT; 2-NPT  
set nwpw:mc_aratio 0.234           # targeted acceptance ratio  
set nwpw:mc_ddx 0.1                # parameter used to adjust geometry dispacement to have sampling with targeted acceptance  
set nwpw:mc_temperature 300.0      # Temperature in K  
set nwpw:mc_step_size   0.250      # initial geometry displacement step size  

nwpw  
   mc_steps 10 100  #total number of iterations = 10*100, number of iterations between step size adjustments = 10  
   cpmd_properties on  
end  
task pspw Metropolis
</code></pre>
<h2 id="plane-wave-density-functional-theory-nwpw-tutorial-9-npt-metropolis-monte-carlo-simulations">NWPW Tutorial 9: NPT Metropolis Monte-Carlo Simulations<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-tutorial-9-npt-metropolis-monte-carlo-simulations" title="Permanent link">&para;</a></h2>
<p>In this example the PSPW module is used to run an NPT simulation for a
diamond crystal using the a Metropolis Monte-Carlo
algorithm.</p>
<p>(input:<a href="diamond-metropolis.nw">Media:diamond-metropolis.nw</a>,
output:<a href="diamond-metropolis.nwout.gz">Media:diamond-metropolis.nwout.gz</a>,
datafiles:<a href="diamond-metropolis.emotion.gz">Media:diamond-metropolis.emotion.gz</a>,
<a href="diamond-metropolis.ion_motion.gz">Media:diamond-metropolis.ion_motion.gz</a>,
<a href="diamond-metropolis.xyz.gz">Media:diamond-metropolis.xyz.gz</a>,
<a href="diamond_metropolis_1234.cif.gz">Media:diamond_metropolis_1234.cif.gz</a>)</p>
<pre><code>title &quot;Metropolis NPT simulation of diamond - this input is used to put the system in equilibrium&quot;  
echo  

start diamond-metropolis  

#permanent_dir ./perm  
#scratch_dir   ./perm  

#**** Enter the geometry using fractional coordinates ****  
geometry center noautosym noautoz print   
  system crystal   
    lat_a 3.56d0   
    lat_b 3.56d0   
    lat_c 3.56d0   
    alpha 90.0d0   
    beta  90.0d0   
    gamma 90.0d0    
  end  
 C -0.50000d0 -0.50000d0 -0.50000d0  
 C  0.00000d0  0.00000d0 -0.50000d0  
 C  0.00000d0 -0.50000d0  0.00000d0  
 C -0.50000d0  0.00000d0  0.00000d0  
 C -0.25000d0 -0.25000d0 -0.25000d0  
 C  0.25000d0  0.25000d0 -0.25000d0  
 C  0.25000d0 -0.25000d0  0.25000d0  
 C -0.25000d0  0.25000d0  0.25000d0  
end  
set nwpw:cif_filename pspw_metropolis  

###### setup the nwpw gamma point code ######  
set nwpw:kbpp_ray .true.  
set nwpw:kbpp_filter .true.  
set nwpw:frozen_lattice:thresh 999.0  
nwpw  
   lmbfgs  
   ewald_rcut 3.0  
   ewald_ncut 8  
   xc pbe  
end  
task pspw energy  


####################################################################################  
###### setup Metropolis NPT code - input will change in a forthcoming release ######  
####################################################################################  
set nwpw:mc_seed 1234              # Seed for random number generator  
set nwpw:mc_algorithm 2            # 1-NVT; 2-NPT  
set nwpw:mc_aratio 0.234           # targeted acceptance ratio  
set nwpw:mc_ddx 0.1                # parameter used to adjust geometry dispacement to have sampling with targeted acceptance  
set nwpw:mc_ddv 0.1                # parameter used to adjust volume change to have sampling with targeted acceptance  
set nwpw:mc_temperature 300.0      # Temperature in K  
set nwpw:mc_step_size   0.250      # geometry displacement step size  
set nwpw:mc_volume_step 0.130      # volume displacement step size  

nwpw  
   bo_steps 10 100  #total number of iterations = 10*100, number of iterations between step size adjustments = 10  
end  
task pspw Metropolis
</code></pre>
<p><img alt=" " src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/diamond-metropolis.png" align=middle with="640pt" height="480pt"/>  </p>
<p>(inputs:<a href="diamond-metropolis-sampling.nw.tgz">Media:diamond-metropolis-sampling.nw.tgz</a>)</p>
<p>(python analysis
program:<a href="makehistogram.gz">Media:makehistogram.gz</a>)</p>
<pre><code>[WE27972:~/Projects/NWChem/Metropolis] bylaska% makehistogram -t 300 -c 2 1235/diamond-metropolis-1235.emotion 1236/diamond-metropolis-1236.emotion   
1237/diamond-metropolis-1237.emotion 1238/diamond-metropolis-1238.emotion 1239/diamond-metropolis-1239.emotion 1240/diamond-metropolis-1240.emotion   
1241/diamond-metropolis-1241.emotion 1242/diamond-metropolis-1242.emotion 1243/diamond-metropolis-1243.emotion 1244/diamond-metropolis-1244.emotion   
1245/diamond-metropolis-1245.emotion 1246/diamond-metropolis-1246.emotion 1248/diamond-metropolis-1248.emotion 1249/diamond-metropolis-1249.emotion   
makehistogram Program  
len(args)= 14  

unitconversion  = 1.0  
temperature (K) = 300  
RT (au)         = 0.000949482834326  ( 0.5958  kcal/mol)  

data columns -1 = [1]  
data rows (n)   = 52000  

delta (au)       = 0.01  
xmin-delta (au)  = -45.08080365  
xmax+delta (au)  = -45.05079515  

data averaging:  
- xbar       (au)           = -45.0668093497  
- S2_{n-1}   (au)           = 1.08378343101e-05  
- &lt;exp((x-xmin)/RT)&gt; (au)   = 5293374903.39  
- &lt;exp((x-xbar)/RT)&gt; (au)   = 2102.44405413  
- Free energy               = -45.0595449934  
- Free energy1              = -45.0595449934  

histogram distribution parameters:  
- number of bins (Rice k) = 75  
- bin width               = 0.00040552027027  
- norm                    = 1.0  
- xbar (au)               = -45.0668107987  (error= -1.44908364064e-06 )  
- S2_{n-1} (au)           = 1.0858459744e-05  (error= 2.06254339582e-08 )  
- &lt;exp((x-xmin)/RT)&gt; (au) = 5184600342.01  (error= -108774561.378 )  
- &lt;exp((x-xbar)/RT)&gt; (au) = 2062.38570923  (error= -40.0583449011 )  
- Free energy               = -45.0595647078  (error= -1.9714360235e-05 )  
- Free energy1              = -45.0595647078  (error= -1.9714360235e-05 )  
- histogram plot file     = histogram.dat  

normal distribution parameters:  
- average x (input xbar)                   = -45.0668093497  
- unbiased sample variance (input S2_(n-1))= 1.08378343101e-05  
- xbar-xmin                                = 0.0139943003357  
- norm                    = 0.99998877936  
- xbar (au)               = -45.0663035243  (error= 0.000505825397738 )  
- S2_{n-1} (au)           = 1.1091077321e-05  (error= 2.53243010936e-07 )  
- &lt;exp((x-xmin)/RT)&gt; (au) = 943482808.939  (error= -4349892094.45 )  
- &lt;exp((x-xbar)/RT)&gt; (au) = 219.968603653  (error= -1882.47545048 )  
- Free energy               = -45.061182503  (error= -0.00163750957643 )  
- Free energy1              = -45.061182503  (error= -0.00163750957643 )  
- normal distribution plot file = normdist.dat  
- number data points = 1500  

gamma distribution parameters:  
- alpha0= 18.0700715921  
- beta0 = 1291.24508969  
- xmin + alpha0/beta0 = -45.0668093497  
- alpha = 18.5003178357  
- beta  = 1321.98948086  
- xmin + alpha/beta   = -45.0668093497  
- norm                    = 0.999923464137 0.99993569948  
- xbar (au)               = -45.0633614482 -45.0639126423  (error= 0.00344790150088 0.00289670733491 )  
- S2_{n-1} (au)           = 2.27110055327e-05 1.89632753897e-05  (error= 1.18731712226e-05 8.12544107961e-06 )  
- &lt;exp((x-xmin)/RT)&gt; (au) = 7932775654.26 7060892836.07  (error= 2639400750.87 1767517932.68 )  
- &lt;exp((x-xbar)/RT)&gt; (au) = 83.43400035 132.707151194  (error= -2019.01005378 -1969.73690294 )  
- Free energy               = -45.059160883 -45.0592714327  (error= 0.000384110406969 0.000273560709338 )  
- Free energy1              = -45.059160883 -45.0592714327  (error= 0.000384110406969 0.000273560709338 )  
- gamma distribution plot file = gammadist.dat  
- number data points = 1500  

Hausdorff distribution parameters:  
- xmin = -45.08080365  
- xmax = -45.05079515  
- number moments     = 15  
  -- &lt; x^0 &gt;    =    1.000000000000000  
  -- &lt; x^1 &gt;    =    0.466344546904007  
  -- &lt; x^2 &gt;    =    0.229512222180349  
  -- &lt; x^3 &gt;    =    0.119040323347820  
  -- &lt; x^4 &gt;    =    0.064946164109284  
  -- &lt; x^5 &gt;    =    0.037186896798964  
  -- &lt; x^6 &gt;    =    0.022287980659815  
  -- &lt; x^7 &gt;    =    0.013942929105868  
  -- &lt; x^8 &gt;    =    0.009076370636747  
  -- &lt; x^9 &gt;    =    0.006128509645342  
  -- &lt; x^10 &gt;   =    0.004278147917961  
  -- &lt; x^11 &gt;   =    0.003077410986590  
  -- &lt; x^12 &gt;   =    0.002273768533280  
  -- &lt; x^13 &gt;   =    0.001720304299285  
  -- &lt; x^14 &gt;   =    0.001328990330385  
- norm                    = 1.0000000003  
- xbar (au)               = -45.066809363  (error= -1.33426993898e-08 )  
- S2_{n-1} (au)           = 1.08376258908e-05  (error= -2.08419282206e-10 )  
- &lt;exp((x-xmin)/RT)&gt; (au) = 5423305875.35  (error= 129930971.958 )  
- &lt;exp((x-xbar)/RT)&gt; (au) = 2154.08083332  (error= 51.6367791881 )  
- Free energy               = -45.0595219689  (error= 2.30245307122e-05 )  
- Free energy1              = -45.0595219689  (error= 2.30245307122e-05 )  
- Hausdorff moment history file = moment_hist.dat  
- Hausdorff distribution plot file = hausdorff.dat  
- number data points = 1500
</code></pre>
<p><img alt=" " src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/diamond-e-histogram.png" align=middle with="640pt" height="480pt"/><br />
<img alt=" " src="https://raw.githubusercontent.com/wiki/nwchemgit/nwchem/diamond-v-histogram.png" align=middle with="640pt" height="480pt"/>  </p>
<h2 id="plane-wave-density-functional-theory-nwpw-tutorial-9-free-energy-simulations">NWPW Tutorial 9: Free Energy Simulations<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-tutorial-9-free-energy-simulations" title="Permanent link">&para;</a></h2>
<p>A description of using the WHAM method for generating free energy of the
gas-phase dissociation reaction CH<span class="arithmatex">\(_3\)</span>Cl <span class="arithmatex">\(\rightarrow\)</span> CH<span class="arithmatex">\(_3\)</span>+Cl
can be found in the attached pdf
(<a href="nwchem-new-pmf.pdf">nwchem-new-pmf.pdf</a>)</p>
<h2 id="plane-wave-density-functional-theory-paw-tutorial">PAW Tutorial<a class="headerlink" href="#plane-wave-density-functional-theory-paw-tutorial" title="Permanent link">&para;</a></h2>
<h3 id="plane-wave-density-functional-theory-optimizing-a-water-molecule">Optimizing a water molecule<a class="headerlink" href="#plane-wave-density-functional-theory-optimizing-a-water-molecule" title="Permanent link">&para;</a></h3>
<p>The following input deck performs for a water molecule a PSPW energy
calculation followed by a PAW energy calculation and a PAW geometry
optimization calculation. The default unit cell parameters are used
(SC=20.0, ngrid 32 32 32). In this simulation, the first PAW run
optimizes the wavefunction and the second PAW run optimizes the
wavefunction and geometry in tandem.</p>
<pre><code>title &quot;paw steepest descent test&quot;  
start paw_test 
charge 0 
geometry units au nocenter noautoz noautosym  
O      0.00000    0.00000    0.01390 
H     -1.49490    0.00000   -1.18710  
H      1.49490    0.00000   -1.18710  
end 
nwpw  
  time_step 15.8  
  ewald_rcut 1.50  
  tolerances 1.0d-8 1.0d-8  
end  
set nwpw:lcao_iterations 1  
set nwpw:minimizer 2  
task pspw energy  
task paw energy  
nwpw  
  time_step 5.8  
  geometry_optimize  
  ewald_rcut 1.50  
  tolerances 1.0d-7 1.0d-7 1.0d-4  
end
task paw steepest_descent 
task paw optimize
</code></pre>
<h3 id="plane-wave-density-functional-theory-optimizing-a-unit-cell-and-geometry-for-silicon-carbide">Optimizing a unit cell and geometry for Silicon-Carbide<a class="headerlink" href="#plane-wave-density-functional-theory-optimizing-a-unit-cell-and-geometry-for-silicon-carbide" title="Permanent link">&para;</a></h3>
<p>The following example demonstrates how to uses the PAW module to
optimize the unit cell and geometry for a silicon-carbide
crystal.</p>
<pre><code>title &quot;SiC 8 atom cubic cell - geometry and unit cell optimization&quot; 
start SiC
#**** Enter the geometry using fractional coordinates ****  
geometry units au center noautosym noautoz print
 system crystal
   lat_a 8.277d0
   lat_b 8.277d0 
   lat_c 8.277d0  
   alpha 90.0d0  
   beta  90.0d0
   gamma 90.0d0
 end 
Si    -0.50000d0  -0.50000d0  -0.50000d0
Si     0.00000d0   0.00000d0  -0.50000d0  
Si     0.00000d0  -0.50000d0   0.00000d0 
Si    -0.50000d0   0.00000d0   0.00000d0
C     -0.25000d0  -0.25000d0  -0.25000d0
C      0.25000d0   0.25000d0  -0.25000d0 
C      0.25000d0  -0.25000d0   0.25000d0
C     -0.25000d0   0.25000d0   0.25000d0  
end  
#***** setup the nwpw gamma point code ****  
nwpw
  simulation_cell 
    ngrid 16 16 16
  end
  ewald_ncut 8 
end  
set nwpw:minimizer 2
set nwpw:psi_nolattice .true.  # turns of unit cell checking for wavefunctions 
driver
  clear 
  maxiter 40  
end
set includestress .true.         # this option tells driver to optimize the unit cell
set nwpw:stress_numerical .true. #currently only numerical stresses implemented in paw
task paw optimize
</code></pre>
<h3 id="plane-wave-density-functional-theory-running-a-car-parrinello-simulation">Running a Car-Parrinello Simulation<a class="headerlink" href="#plane-wave-density-functional-theory-running-a-car-parrinello-simulation" title="Permanent link">&para;</a></h3>
<p>In this section we show how use the PAW module to perform a
   Car-Parrinello molecular dynamic simulation for a C<sub>2</sub> molecule at
the LDA level. Before running a PAW Car-Parrinello simulation the system
should be on the Born-Oppenheimer surface, i.e. the one-electron
orbitals should be minimized with respect to the total energy (i.e. task
pspw energy). The input needed is basically the same as for optimizing
   the geometry of a C<sub>2</sub> molecule at the LDA level,except that and
additional Car-Parrinello sub-block is added.</p>
<p>In the following example we show the input needed to run a
   Car-Parrinello simulation for a C<sub>2</sub> molecule at the LDA level. In
this example, default pseudopotentials from the pseudopotential library
are used for C, the boundary condition is free-space, the exchange
correlation functional is LDA, The boundary condition is free-space, and
the simulation cell cell is aperiodic and cubic with a side length of
10.0 Angstroms and has 40 grid points in each direction (cutoff energy
is 44 Ry). The time step and fake mass for the Car-Parrinello run are
specified to be 5.0 au and 600.0 au,
respectively.</p>
<pre><code>start c2_paw_lda_md  
title &quot;C2 restricted singlet dimer, LDA/44Ry - constant energy Car-Parrinello simulation&quot;
geometry  
 C    -0.62 0.0 0.0 
 C     0.62 0.0 0.0 
end
pspw 
  simulation_cell units angstroms  
     boundary_conditions aperiodic 
     lattice 
       lat_a 10.00d0  
       lat_b 10.00d0
       lat_c 10.00d0
     end  
     ngrid 40 40 40
  end  
  Car-Parrinello 
    fake_mass 600.0
    time_step 5.0 
    loop 10 10  
  end 
end 
set nwpw:minimizer 2
task paw energy
task paw Car-Parrinello
</code></pre>
<h2 id="plane-wave-density-functional-theory-nwpw-capabilities-and-limitations">NWPW Capabilities and Limitations<a class="headerlink" href="#plane-wave-density-functional-theory-nwpw-capabilities-and-limitations" title="Permanent link">&para;</a></h2>
<ul>
<li>Hybrid Functionals (e.g. PBE0, LDA-SIC) only work in PSPW.</li>
<li>Wannier orbital task only works in PSPW.</li>
<li>AIMD/MM simulation only works with PSPW.</li>
</ul>
<h2 id="plane-wave-density-functional-theory-questions-and-difficulties">Questions and Difficulties<a class="headerlink" href="#plane-wave-density-functional-theory-questions-and-difficulties" title="Permanent link">&para;</a></h2>
<p>Questions and encountered problems should be reported to the <a href="#forum">NWChem
Community Forum</a> or to Eric J. Bylaska,
Eric.Bylaska@pnl.gov</p></section><section class="print-page" id="gaussian-basis-aimd" heading-number="12.2"><h1 id="gaussian-basis-aimd-gaussian-basis-aimd">Gaussian Basis AIMD<a class="headerlink" href="#gaussian-basis-aimd-gaussian-basis-aimd" title="Permanent link">&para;</a></h1>
<h2 id="gaussian-basis-aimd-overview">Overview<a class="headerlink" href="#gaussian-basis-aimd-overview" title="Permanent link">&para;</a></h2>
<p>This module performs adiabatic ab initio molecular dynamics on finite
systems. The nuclei are integrated using the velocity-Verlet algorithm,
and the electronic potential can be provided by any of the Gaussian
basis set based methods in NWChem, e.g. DFT, TDDFT, TCE, MP2, SCF,
MCSCF, etc. If analytic gradients are not available for the selected
level of theory, numerical gradients will automatically be used. Initial
velocities are randomly selected from the Maxwell-Boltzmann distribution
at the specified temperature, unless a restart file (<code>.qmdrst</code>) is
present. If a restart file is present, the trajectory information will
be read from that file and the trajectory will resume from that point.</p>
<p>For computational details and a case study using the module, please
refer to the 2016 paper by Fischer<sup id="gaussian-basis-aimd-fnref_footnote_1"><a class="footnote-ref" href="#gaussian-basis-aimd-fn_footnote_1">1</a></sup>.  </p>
<pre><code>QMD
  [dt_nucl &lt;double default 10.0&gt;]  
  [nstep_nucl &lt;integer default 1000&gt;]  
  [targ_temp &lt;double default 298.15&gt;]  
  [thermostat &lt;string default none&gt; &lt;thermostat parameters&gt;]  
  [rand_seed &lt;integer default new one generated for each run&gt;]  
  [com_step &lt;integer default 100&gt;]  
  [print_xyz &lt;integer default 1&gt;]  
  [linear]   
  [property &lt;integer default 1&gt;] 
  [tddft &lt;integer default 1&gt;]
  [namd ]
END
</code></pre>
<p>The module is called as:</p>
<pre><code>task &lt;level of theory&gt; qmd
</code></pre>
<p>where <level of theory> is any Gaussian basis set method in NWChem</p>
<h2 id="gaussian-basis-aimd-qmd-keywords">QMD Keywords<a class="headerlink" href="#gaussian-basis-aimd-qmd-keywords" title="Permanent link">&para;</a></h2>
<h3 id="gaussian-basis-aimd-dt_nucl-nuclear-time-step">DT_NUCL: Nuclear time step<a class="headerlink" href="#gaussian-basis-aimd-dt_nucl-nuclear-time-step" title="Permanent link">&para;</a></h3>
<p>This specifies the nuclear time step in atomic units (1 a.u. = 0.02419
fs).
Default: 10.0 a.u.</p>
<h3 id="gaussian-basis-aimd-nstep_nucl-simulation-steps">NSTEP_NUCL: Simulation steps<a class="headerlink" href="#gaussian-basis-aimd-nstep_nucl-simulation-steps" title="Permanent link">&para;</a></h3>
<p>This specifies the number of steps to take in the nuclear dynamics.<br />
Default: 1000</p>
<h3 id="gaussian-basis-aimd-targ_temp-temperature-of-the-system">TARG_TEMP: Temperature of the system<a class="headerlink" href="#gaussian-basis-aimd-targ_temp-temperature-of-the-system" title="Permanent link">&para;</a></h3>
<p>This specifies the temperature to use with the thermostat. Also it is
used in generating initial velocities from the Maxwell-Boltzmann
distribution.<br />
Default: 298.15 K</p>
<h3 id="gaussian-basis-aimd-thermostat-thermostat-for-controling-temperature-of-the-simulation">THERMOSTAT: Thermostat for controling temperature of the simulation<a class="headerlink" href="#gaussian-basis-aimd-thermostat-thermostat-for-controling-temperature-of-the-simulation" title="Permanent link">&para;</a></h3>
<p>This specifies the thermostat to use for regulating the temperature of
the nuclei. Possible options are:</p>
<ul>
<li><strong><code>none</code></strong></li>
</ul>
<p>No thermostat is used, i.e. an NVE ensemble is simulated. Default  </p>
<ul>
<li><strong><code>svr</code></strong> <code>&lt;double default 1000.0&gt;</code></li>
</ul>
<p>Stochastic velocity rescaling thermostat of Bussi, Donadio, and Parrinello<sup id="fnref_footnote_2"><a class="footnote-ref" href="#gaussian-basis-aimd-fn_footnote_2">2</a></sup>.<br />
   Number sets the relaxation parameter of the thermostat  </p>
<ul>
<li><strong><code>langevin</code></strong> <code>&lt;double default 0.1&gt;</code></li>
</ul>
<p>Langevin dynamics, implementation according to Bussi and Parrinello<sup id="fnref_footnote_3"><a class="footnote-ref" href="#gaussian-basis-aimd-fn_footnote_3">3</a></sup>.<br />
   The optional input parameter sets the value of the friction  </p>
<ul>
<li><strong><code>berendsen</code></strong> <code>&lt;double default 1000.0&gt;</code>  </li>
</ul>
<p>Berendsen thermostat, the optional input parameter sets the relaxation parameter of the thermostat  </p>
<ul>
<li><strong><code>rescale</code></strong></li>
</ul>
<p>Velocity rescaling, i.e. isokinetic ensemble  </p>
<ul>
<li><strong><code>nose-hoover</code></strong> <code>&lt;integer default 3&gt;</code></li>
</ul>
<p>NosHoover thermostat (only available in release 7.2.0 and later). 
  The optional input parameter defines the number of oscillators.</p>
<h3 id="gaussian-basis-aimd-rand_seed-seed-for-the-random-number-generator">RAND_SEED: Seed for the random number generator<a class="headerlink" href="#gaussian-basis-aimd-rand_seed-seed-for-the-random-number-generator" title="Permanent link">&para;</a></h3>
<p><code>rand_seed</code> specifies the seed for initializing the random number generator. If
not given, a unique random seed will be generated. Even without a
thermostat, this will influence the initial
velocities.</p>
<h3 id="gaussian-basis-aimd-com_step-how-often-center-of-mass-translations-and-rotations-are-removed">COM_STEP: How often center-of-mass translations and rotations are removed<a class="headerlink" href="#gaussian-basis-aimd-com_step-how-often-center-of-mass-translations-and-rotations-are-removed" title="Permanent link">&para;</a></h3>
<p><code>com_step</code> specifies that center-of-mass translations and rotations will be
removed every <code>com_step</code> steps. Default 10 COM translations and rotations
are removed on startup (either randomized initial velocities or those
read from the restart file).</p>
<h3 id="gaussian-basis-aimd-print_xyz-how-often-to-print-trajectory-information-to-xyz-file">PRINT_XYZ: How often to print trajectory information to xyz file<a class="headerlink" href="#gaussian-basis-aimd-print_xyz-how-often-to-print-trajectory-information-to-xyz-file" title="Permanent link">&para;</a></h3>
<p><code>print_xyz</code> specifies how often the trajectory information (coordinates, velocities,
total energy, step number, dipole (if available)) is written to the xyz file. The
units for the coordinates and velocities in the xyz file are Angstrom
and Angstrom/fs, respectively.<br />
For example, <code>print_xyz 5</code> will write the xyz trajectory file every 5 steps.<br />
Default: 1 </p>
<h3 id="gaussian-basis-aimd-linear-flag-for-linear-molecules">LINEAR: Flag for linear molecules<a class="headerlink" href="#gaussian-basis-aimd-linear-flag-for-linear-molecules" title="Permanent link">&para;</a></h3>
<p>If the <code>linear</code> keyword is present, the code assumes the molecule is linear.</p>
<h3 id="gaussian-basis-aimd-property-how-often-to-calculate-molecular-properties-as-part-of-the-md-simulation">PROPERTY: How often to calculate molecular properties as part of the MD simulation<a class="headerlink" href="#gaussian-basis-aimd-property-how-often-to-calculate-molecular-properties-as-part-of-the-md-simulation" title="Permanent link">&para;</a></h3>
<p>If the <code>property</code> keyword present, the code will look for the property block and calculate the requested properties.<br />
For example, <code>property 5</code> will calculate properties on the current geometry every 5 steps.<br />
Default: 0 (e.g properties are not computed)  </p>
<h3 id="gaussian-basis-aimd-tddft-how-often-to-perform-tddft-calculation-as-part-of-the-md-simulation">TDDFT: How often to perform TDDFT calculation as part of the MD simulation<a class="headerlink" href="#gaussian-basis-aimd-tddft-how-often-to-perform-tddft-calculation-as-part-of-the-md-simulation" title="Permanent link">&para;</a></h3>
<p>If the <code>tddft</code> keyword is present, the code will look for the tddft block and calculate the absorption spectrum.<br />
For example, <code>tddft 5</code> will perform tddft calculations on the current geometry every 5 steps.<br />
Default: 0 (e.g tddft is not run)</p>
<h2 id="gaussian-basis-aimd-sample-input-files">Sample input files<a class="headerlink" href="#gaussian-basis-aimd-sample-input-files" title="Permanent link">&para;</a></h2>
<h3 id="gaussian-basis-aimd-ground-state-molecular-dynamics">Ground state Molecular Dynamics<a class="headerlink" href="#gaussian-basis-aimd-ground-state-molecular-dynamics" title="Permanent link">&para;</a></h3>
<p>The following is a sample input for a ground state MD simulation. The
simulation is 200 steps long with a 10 a.u. time step, using the
stochastic velocity rescaling thermostat with a relaxation parameter of
100 a.u. and a target temperature of 200 K. Center-of-mass rotations and
translations will be removed every 10 steps and trajectory information
will be output to the xyz file every 5 steps.</p>
<pre><code>start qmd_dft_h2o_svr  
echo  
print low  
geometry noautosym noautoz  
  O   0.00000000    -0.01681748     0.11334792  
  H   0.00000000     0.81325914    -0.34310308  
  H   0.00000000    -0.67863597    -0.56441201  
end  
basis  
  * library 6-31G*  
end  
dft  
  xc pbe0  
end  
qmd  
  nstep_nucl  200  
  dt_nucl     10.0  
  targ_temp   200.0  
  com_step    10  
  thermostat  svr 100.0  
  print_xyz   5  
end  
task dft qmd
</code></pre>
<h3 id="gaussian-basis-aimd-excited-state-molecular-dynamics">Excited state Molecular Dynamics<a class="headerlink" href="#gaussian-basis-aimd-excited-state-molecular-dynamics" title="Permanent link">&para;</a></h3>
<p>The following is a sample input for an excited state MD simulation on
the first excited state. The simulation is 200 steps long with a 10 a.u.
time step, run in the microcanonical ensemble. Center-of-mass rotations
and translations will be removed every 10 steps and trajectory
information will be output to the xyz file every 5 steps.</p>
<pre><code>start qmd_tddft_h2o_svr  
echo  
print low  
geometry noautosym noautoz  
  O   0.00000000    -0.01681748     0.11334792  
  H   0.00000000     0.81325914    -0.34310308  
  H   0.00000000    -0.67863597    -0.56441201  
end  
basis  
  * library 6-31G*  
end  
dft  
  xc pbe0  
end  
tddft  
  nroots 5  
  notriplet  
  target 1  
  civecs  
  grad  
    root 1  
  end  
end  
qmd  
  nstep_nucl  200  
  dt_nucl     10.0  
  com_step    10  
  thermostat  none  
  print_xyz   5  
end  
task tddft qmd
</code></pre>
<h3 id="gaussian-basis-aimd-property-calculation-in-a-molecular-dynamics-simulation">Property calculation in a Molecular Dynamics simulation<a class="headerlink" href="#gaussian-basis-aimd-property-calculation-in-a-molecular-dynamics-simulation" title="Permanent link">&para;</a></h3>
<p>Thefollowing is a sample input for an MD simulation that compute
polarizability by means of the
<a href="#properties-polarizability-computed-with-the-sum-over-orbitals-method">SOS method</a>
at each time step.</p>
<pre><code>start qmd_prop_h2o_svr  
echo  
print low  
geometry noautosym noautoz  
  O   0.00000000    -0.01681748     0.11334792  
  H   0.00000000     0.81325914    -0.34310308  
  H   0.00000000    -0.67863597    -0.56441201  
end  
basis  
  * library 6-31G*  
end  
dft  
  xc pbe0  
end  

qmd  
  nstep_nucl  200  
  dt_nucl     10.0  
  com_step    10  
  thermostat  none  
  print_xyz   5
  property 1
end  

property
 polfromsos
end
task tddft qmd
</code></pre>
<p>Additional sample inputs can be found in $NWCHEM_TOP/QA/tests/qmd_*
(e.g. <a href="https://github.com/nwchemgit/nwchem/tree/master/QA/tests/qmd_dft_h2o_berendsen_props">https://github.com/nwchemgit/nwchem/tree/master/QA/tests/qmd_dft_h2o_berendsen_props</a>)</p>
<h2 id="gaussian-basis-aimd-processing-the-output-of-a-qmd-run">Processing the output of a QMD run<a class="headerlink" href="#gaussian-basis-aimd-processing-the-output-of-a-qmd-run" title="Permanent link">&para;</a></h2>
<p>The xyz file produced by the QMD module contains the velocities (given
in Angstrom/fs), in addition to the coordinates (given in Angstrom). The
comment lines also contain the time step, total energy (atomic units),
and dipole moment (atomic units). In the directory <a href="https://github.com/nwchemgit/nwchem/tree/master/contrib/qmd_tools">$NWCHEM_TOP/contrib/qmd_tools</a>, the code <a href="https://github.com/nwchemgit/nwchem/blob/master/contrib/qmd_tools/qmd_analysis.f90">qmd_analysis.f90</a> will used the xyz trajectory as input to calculate the IR spectrum and
vibrational density of states from Fourier transforms of the dipole and
atomic momenta autocorrelation functions, respectively. The code needs
to be linked to a LAPACK library when compiled; the Makefile in the
directory will compile the code with the LAPACK routines included with
the NWChem source.</p>
<p>Here we compute the IR spectrum and the element-wise breakdown of the
  vibrational density of states for silicon tetrachloride (SiCl<sub>4</sub>). The
following input file was used.</p>
<pre><code>start SiCl4  
echo  
print low  
geometry noautosym noautoz  
  Si              -0.00007905     0.00044148     0.00000001  
  Cl               0.71289590     1.00767685     1.74385011  
  Cl              -2.13658008    -0.00149375    -0.00000001  
  Cl               0.71086735    -2.01430142    -0.00000001  
  Cl               0.71289588     1.00767684    -1.74385011  
end  
basis  
  * library 6-31G  
end  
dft  
  xc hfexch 1.0  
end  
qmd  
  nstep_nucl  20000  
  dt_nucl     10.0  
  targ_temp   20.0  
  com_step    10  
  rand_seed   12345  
  thermostat  none  
end  
task dft qmd
</code></pre>
<p>The IR spectrum and vibrational density of states were generated from
the <code>qmd_analysis</code> code  with the following
command.</p>
<pre><code>./qmd_analysis -xyz SiCl4.xyz -steps 15000 -skip 5000 -ts 10.0 -temp 20.0 -smax 800 -width 10.0
</code></pre>
<p>where we have skipped the first 5000 steps from the simulation and only
used the data from the last 15000 steps to compute the spectra. The time
step is given as 10 a.u. since that was the time step in the simulation
and we output the trajectory information every step. The temperature was
set to 20 K (for analysis, this is only used in the calculation of the
quantum correction factor for the autocorrelation function of the dipole
moment). The option smax sets the maximum of the spectral window that is
output to 800 wave numbers. The width option sets the full-width at
half-maximum of the peaks in the resulting spectra.</p>
<p>The computed IR spectrum and vibrational density of states are shown
here.</p>
<p><img alt="pic1" src="SiCl4IR.png" />
<img alt="pic2" src="SiCl4VDOS.png" /></p>
<h2 id="gaussian-basis-aimd-namd-non-adiabatic-excited-stated-molecular-dynamics">NAMD: Non-adiabatic Excited Stated Molecular Dynamics<a class="headerlink" href="#gaussian-basis-aimd-namd-non-adiabatic-excited-stated-molecular-dynamics" title="Permanent link">&para;</a></h2>
<p>For  details of the NAMD implementation, please
refer to the 2020 paper by Song<sup id="gaussian-basis-aimd-fnref_footnote_4"><a class="footnote-ref" href="#gaussian-basis-aimd-fn_footnote_4">4</a></sup>.    </p>
<pre><code>[namd]
  [init_state &lt;integer default 2&gt;]
  [nstates &lt;integer default 2&gt;]
  [dt_elec &lt;double default 0.01&gt;]
  [deco &lt;logical default .false.]
  [tdks &lt;integer default 1&gt;]
[end]  
</code></pre>
<p>In the <code>namd</code> sub-block within the <code>qmd</code> block, please note:</p>
<ul>
<li>The number of roots requested in the tddft block must be at least nstates-1.</li>
<li>The nuclear time step (dt_nucl) must be an integer multiple of the electronic time step (<code>mod(dt_nucl,dt_elec)=0</code>).</li>
</ul>
<h3 id="gaussian-basis-aimd-deco-decoherence-flag">DECO: Decoherence flag<a class="headerlink" href="#gaussian-basis-aimd-deco-decoherence-flag" title="Permanent link">&para;</a></h3>
<p>The <code>deco</code> flag applies the EDC electronic decoherence correction described in the papey bt Granucci and Persico<sup id="fnref_footnote_5"><a class="footnote-ref" href="#gaussian-basis-aimd-fn_footnote_5">5</a></sup>.<br />
The default value is <code>.false.</code>, i.e. no decoherence correction is applied.</p>
<h3 id="gaussian-basis-aimd-dt_elec-electronic-dynamics-time-step">DT_ELEC: Electronic dynamics time step<a class="headerlink" href="#gaussian-basis-aimd-dt_elec-electronic-dynamics-time-step" title="Permanent link">&para;</a></h3>
<p>The keyword <code>dt_elec</code> sets the electronic time step in atomic units.</p>
<h3 id="gaussian-basis-aimd-n_states-number-of-states">N_STATES: Number of states<a class="headerlink" href="#gaussian-basis-aimd-n_states-number-of-states" title="Permanent link">&para;</a></h3>
<p>The keyword <code>nstates</code> sets the number of electronic states to include in the calculation, i.e.
the number of states for use with Eq. 5 of the 2020 Song paper.</p>
<h3 id="gaussian-basis-aimd-init_state-initial-state">INIT_STATE: Initial state<a class="headerlink" href="#gaussian-basis-aimd-init_state-initial-state" title="Permanent link">&para;</a></h3>
<p>The keyword <code>init_state</code> sets the initial electronic state to be occupied;
the numbering for this keyword and the output that reports the currently
occupied state runs from 0 (ground state) to <code>nstates-1</code>.
So if you want to start a calculation in the first excited state, you would set
<code>init_state</code> to 1.</p>
<h3 id="gaussian-basis-aimd-tdks-time-dependent-kohn-sham">TDKS: Time-Dependent Kohn-Sham<a class="headerlink" href="#gaussian-basis-aimd-tdks-time-dependent-kohn-sham" title="Permanent link">&para;</a></h3>
<p>The keyword <code>tdks</code> will use Time-Dependent Kohn-Sham instead of the default <a href="#excited-state-calculations-cis-and-rpa-the-tamm-dancoff-approximation">Tamm-Dancoff</a> approximation.<br />
The keyword requires the keyword <a href="#density-functional-theory-for-molecules-odft-and-mult-open-shell-systems"><code>odft</code></a>
in the <code>dft</code> input block to work.<br />
It can have two values:</p>
<ul>
<li><code>1</code> (default) selects the alpha spin channel</li>
<li><code>2</code>  selects the beta spin channel</li>
</ul>
<h3 id="gaussian-basis-aimd-namd-input-example">NAMD Input Example<a class="headerlink" href="#gaussian-basis-aimd-namd-input-example" title="Permanent link">&para;</a></h3>
<p>Example input for  fewest-switches surface-hopping (FSSH) approach.</p>
<pre><code>geometry noautosym nocenter
O 0.0000  0.0000  0.1197
H 0.0000  0.7615 -0.4790
H 0.0000 -0.7615 -0.4790
end
basis
* library 6-31G*
end

dft
  xc b3lyp
end

tddft
  nroots 10
  notriplet
  cis
  civecs
  grad
    root 1
  end
end

qmd
  nstep_nucl 50
  dt_nucl 0.5
  targ_temp 300.0
  thermostat svr 500
  namd 
    nstates 5
    init_state 3
    dt_elec 0.1
    deco .true.
  end
end
task tddft qmd
</code></pre>
<h2 id="gaussian-basis-aimd-references">References<a class="headerlink" href="#gaussian-basis-aimd-references" title="Permanent link">&para;</a></h2>
<div class="footnote">
<hr />
<ol>
<li id="gaussian-basis-aimd-fn_footnote_1">
<p>Fischer, S. A.; Ueltschi, T. W.; El-Khoury, P. Z.; Mifflin, A. L.; Hess, W. P.; Wang, H.-F.; Cramer, C. J.; Govind, N. Infrared and Raman Spectroscopy from Ab Initio Molecular Dynamics and Static Normal Mode Analysis: The C-H Region of DMSO as a Case Study. <em>The Journal of Physical Chemistry B</em> <strong>2015</strong>, <em>120</em> (8), 14291436. <a href="https://doi.org/10.1021/acs.jpcb.5b03323">https://doi.org/10.1021/acs.jpcb.5b03323</a>.&#160;<a class="footnote-backref" href="#gaussian-basis-aimd-fnref_footnote_1" title="Jump back to footnote 1 in the text">&#8617;</a></p>
</li>
<li id="gaussian-basis-aimd-fn_footnote_2">
<p>Bussi, G.; Donadio, D.; Parrinello, M. Canonical Sampling Through Velocity Rescaling. <em>The Journal of Chemical Physics</em> <strong>2007</strong>, <em>126</em> (1), 014101. <a href="https://doi.org/10.1063/1.2408420">https://doi.org/10.1063/1.2408420</a>.&#160;<a class="footnote-backref" href="#gaussian-basis-aimd-fnref_footnote_2" title="Jump back to footnote 2 in the text">&#8617;</a></p>
</li>
<li id="gaussian-basis-aimd-fn_footnote_3">
<p>Bussi, G.; Parrinello, M. Accurate Sampling Using Langevin Dynamics. <em>Physical Review E</em> <strong>2007</strong>, <em>75</em> (5), 056707. <a href="https://doi.org/10.1103/PhysRevE.75.056707">https://doi.org/10.1103/PhysRevE.75.056707</a>.&#160;<a class="footnote-backref" href="#gaussian-basis-aimd-fnref_footnote_3" title="Jump back to footnote 3 in the text">&#8617;</a></p>
</li>
<li id="gaussian-basis-aimd-fn_footnote_4">
<p>Song, H.; Fischer, S. A.; Zhang, Y.; Cramer, C. J.; Mukamel, S.; Govind, N.; Tretiak, S. First Principles Nonadiabatic Excited-State Molecular Dynamics in NWChem. <em>Journal of Chemical Theory and Computation</em> <strong>2020</strong>, <em>16</em> (10), 64186427. <a href="https://doi.org/10.1021/acs.jctc.0c00295">https://doi.org/10.1021/acs.jctc.0c00295</a>.&#160;<a class="footnote-backref" href="#gaussian-basis-aimd-fnref_footnote_4" title="Jump back to footnote 4 in the text">&#8617;</a></p>
</li>
<li id="gaussian-basis-aimd-fn_footnote_5">
<p>Granucci, G.; Persico, M. Critical Appraisal of the Fewest Switches Algorithm for Surface Hopping. <em>The Journal of Chemical Physics</em> <strong>2007</strong>, <em>126</em> (13), 134114. <a href="https://doi.org/10.1063/1.2715585">https://doi.org/10.1063/1.2715585</a>.&#160;<a class="footnote-backref" href="#gaussian-basis-aimd-fnref_footnote_5" title="Jump back to footnote 5 in the text">&#8617;</a></p>
</li>
</ol>
</div></section></section>
                    <section class='print-page md-section' id='section-13' heading-number='13'>
                        <h1>Hybrid Approaches<a class='headerlink' href='#section-13' title='Permanent link'></a>
                        </h1>
                    <section class="print-page" id="solvation-models" heading-number="13.1"><h1 id="solvation-models-solvation-models">Solvation Models<a class="headerlink" href="#solvation-models-solvation-models" title="Permanent link">&para;</a></h1>
<h2 id="solvation-models-overview">Overview<a class="headerlink" href="#solvation-models-overview" title="Permanent link">&para;</a></h2>
<p>Two solvation models are available in NWChem: <a href="#solvation-models-cosmo">COSMO</a> and <a href="#solvation-models-smd">SMD</a>.
Since some of the COSMO parameters are used for SMD, we suggest to read the COSMO section before the SMD one.</p>
<h2 id="solvation-models-cosmo">COSMO<a class="headerlink" href="#solvation-models-cosmo" title="Permanent link">&para;</a></h2>
<h3 id="solvation-models-overview_1">Overview<a class="headerlink" href="#solvation-models-overview_1" title="Permanent link">&para;</a></h3>
<p>COSMO is the continuum solvation &lsquo;COnductor-like Screening MOdel&rsquo; of A.
Klamt and G. Schrmann to describe dielectric screening effects in
solvents<sup id="solvation-models-fnref_footnote_1"><a class="footnote-ref" href="#solvation-models-fn_footnote_1">1</a></sup>. This model has been enhanced by D.M. York and M.
Karplus<sup id="solvation-models-fnref_footnote_2"><a class="footnote-ref" href="#solvation-models-fn_footnote_2">2</a></sup> to create a smooth potential energy surface. The latter
facilitates geometry optimization and dynamics and the implementation
has been adapted to take advantage of those ideas.</p>
<p>The NWChem COSMO module implements algorithm for calculation of the
energy for the following methods:</p>
<ol>
<li>Restricted Hartree-Fock (RHF),</li>
<li>Restricted open-shell Hartree-Fock (ROHF),</li>
<li>Restricted Kohn-Sham DFT (DFT),</li>
<li>Unrestricted Kohn-Sham DFT (ODFT),</li>
</ol>
<p>by determining the solvent reaction field self-consistently with the
solute charge distribution from the respective methods. Note that COSMO
for unrestricted Hartree-Fock (UHF) method can also be performed by
invoking the DFT module with appropriate keywords.</p>
<p>Correlation energy of solvent molecules may also be evaluated at</p>
<ol>
<li>MP2,</li>
<li>CCSD,</li>
<li>CCSD+T(CCSD),</li>
<li>CCSD(T),</li>
</ol>
<p>levels of theory. It is cautioned, however, that these correlated COSMO
calculations determine the solvent reaction field using the HF charge
distribution of the solute rather than the charge distribution of the
correlation theory and are not entirely self consistent in that respect.
In other words, these calculations assume that the correlation effect
and solvation effect are largely additive, and the combination effect
thereof is neglected. COSMO for MCSCF has not been implemented yet.</p>
<p>In the current implementation the code calculates the gas-phase energy
of the system followed by the solution-phase energy, and returns the
electrostatic contribution to the solvation free energy. At the present
gradients are calculated analytically, but frequencies are calculated by
finite difference of the gradients.   <br />
The non-electrostatic
contributions can be calculated by turning on the <a href="#solvation-models-smd">SMD model</a>. It should
be noted that one must in general take into account the standard state
correction besides the electrostatic and cavitation/dispersion
contribution to the solvation free energy, when a comparison to
experimental data is made.</p>
<h3 id="solvation-models-cosmo-input-parameters">COSMO Input Parameters<a class="headerlink" href="#solvation-models-cosmo-input-parameters" title="Permanent link">&para;</a></h3>
<p>Invoking the COSMO solvation model is done by specifying the input COSMO
input block with the input options as:</p>
<pre><code>cosmo  
  [off]  
  [dielec  &lt;real dielec default 78.4&gt;]  
  [parameters &lt;filename&gt;]  
  [radius  &lt;real atom1&gt;  
           &lt;real atom2&gt;  
      . . .  
           &lt;real atomN&gt;]
  [rsolv &lt;real rsolv default 0.5&gt;]  
  [screen  &lt;ideal||ks||st&gt;]
  [iscren  &lt;integer iscren default 0, deprecated&gt;]  
  [minbem  &lt;integer minbem default 2&gt;]
  [maxbem  &lt;integer maxbem default 4, deprecated&gt;]
  [ificos  &lt;integer ificos default 0&gt;]  
  [lineq   &lt;integer lineq default 1&gt;]  
  [zeta &lt;real zeta default 0.98&gt;]  
  [gamma_s &lt;real gammas default 1.0&gt;]  
  [sw_tol &lt;real swtol default 1.0e-4&gt;]
  [charge_correction &lt;no||scale||lagrangian default lagrangian&gt;]  
  [do_gasphase  &lt;logical do_gasphase default True&gt;] 
  [do_cosmo_ks]
  [do_cosmo_yk]
  [do_cosmo_smd]
end
</code></pre>
<p>followed by the task directive specifying the wavefunction and type of
calculation, e.g., <code>task scf energy</code>, <code>task mp2 energy</code>, <code>task dft
optimize</code>, etc.</p>
<h4 id="solvation-models-cosmo-off-keyword">COSMO: OFF keyword<a class="headerlink" href="#solvation-models-cosmo-off-keyword" title="Permanent link">&para;</a></h4>
<p><code>off</code> can be used to turn off COSMO in a compound (multiple task) run.
By default, once the COSMO solvation model has been defined it will be
used in subsequent calculations. Add the keyword <code>off</code> if COSMO is not
needed in subsequent calculations.</p>
<h4 id="solvation-models-cosmo-dielec-keyword">COSMO: DIELEC keyword<a class="headerlink" href="#solvation-models-cosmo-dielec-keyword" title="Permanent link">&para;</a></h4>
<p><code>dielec</code> is the value of the dielectric constant <span class="arithmatex">\( \epsilon \)</span> of the medium, with a
default value of 78.4 (the dielectric constant for water).
This value is used to scale the dielectric screening energy, as detailed in the <a href="#solvation-models-cosmo-screen-keyword">SCREEN section</a>.</p>
<h4 id="solvation-models-cosmo-parameters-keyword">COSMO: PARAMETERS keyword<a class="headerlink" href="#solvation-models-cosmo-parameters-keyword" title="Permanent link">&para;</a></h4>
<p><code>parameters</code> specifies COSMO radii parameters file that stores custom
setting for COSMO parameters. The format for such file consists of the
atom or element name followed by the radii. The program will first
attempt to match based on atom name and only then the element name.
Otherwise radius will be set based on default parameters. The file has
to present in one of the three location ( in the order of preference) -
directory specified by the environmental variable
<code>NWCHEM_COSMO_LIBRARY</code>, permanent directory, and run directory. </p>
<h4 id="solvation-models-cosmo-radius-keyword">COSMO: RADIUS keyword<a class="headerlink" href="#solvation-models-cosmo-radius-keyword" title="Permanent link">&para;</a></h4>
<p><code>radius</code> is an array that specifies the radius of the spheres associated
with each atom and that make up the molecule-shaped cavity. These values
will override default radii setting including those specified in the
COSMO parameter file (if any) Default values are Van der Waals radii.
Values are in units of angstroms. The codes uses the following Van der
Waals radii by default:</p>
<p>Default radii provided by Andreas Klamt (Cosmologic)</p>
<p>vdw radii: 1.17 ( 0.02) * Bondi radius<sup id="fnref_footnote_3"><a class="footnote-ref" href="#solvation-models-fn_footnote_3">3</a></sup></p>
<p>optimal vdw radii for H, C, N, O, F, S, Cl, Br, I<sup id="fnref_footnote_4"><a class="footnote-ref" href="#solvation-models-fn_footnote_4">4</a></sup></p>
<p>for heavy elements: 1.17*1.9</p>
<pre><code>     data (vander(i),i=1,102)  
    1 / 1.300,1.638,1.404,1.053,2.0475,2.00,  
    2   1.830,1.720,1.720,1.8018,1.755,1.638,  
    3   1.404,2.457,2.106,2.160,2.05,2.223,  
    4   2.223,2.223,2.223,2.223,2.223,2.223,  
    5   2.223,2.223,2.223,2.223,2.223,2.223,  
    6   2.223,2.223,2.223,2.223,2.160,2.223,  
    7   2.223,2.223,2.223,2.223,2.223,2.223,  
    8   2.223,2.223,2.223,2.223,2.223,2.223,  
    9   2.223,2.223,2.223,2.223,2.320,2.223,  
    1   2.223,2.223,2.223,2.223,2.223,2.223,  
    2   2.223,2.223,2.223,2.223,2.223,2.223,  
    3   2.223,2.223,2.223,2.223,2.223,2.223,  
    4   2.223,2.223,2.223,2.223,2.223,2.223,  
    5   2.223,2.223,2.223,2.223,2.223,2.223,  
    6   2.223,2.223,2.223,2.223,2.223,2.223,  
    7   2.223,2.223,2.223,2.223,2.223,2.223,  
    7   2.223,2.223,2.223,2.223,2.223,2.223/
</code></pre>
<p>For examples see Stefanovich et al.<sup id="fnref_footnote_5"><a class="footnote-ref" href="#solvation-models-fn_footnote_5">5</a></sup> and Barone et al.<sup id="fnref_footnote_6"><a class="footnote-ref" href="#solvation-models-fn_footnote_6">6</a></sup></p>
<h4 id="solvation-models-cosmo-rsolv-keyword">COSMO: RSOLV keyword<a class="headerlink" href="#solvation-models-cosmo-rsolv-keyword" title="Permanent link">&para;</a></h4>
<p><code>rsolv</code> is a parameter used to define the solvent accessible surface. See the original reference of Klamt and Schrmann <sup id="fnref2_footnote_1"><a class="footnote-ref" href="#solvation-models-fn_footnote_1">1</a></sup> for a description. 
The default value is 0.5 (in angstroms). It is not used in the York and Karplus model (used by default in NWChem).</p>
<h4 id="solvation-models-cosmo-screen-keyword">COSMO: SCREEN keyword<a class="headerlink" href="#solvation-models-cosmo-screen-keyword" title="Permanent link">&para;</a></h4>
<p>The <code>screen</code> parameter defines the dielectric screening scaling.</p>
<ul>
<li><code>screen ideal</code> applies no scaling, meaning <span class="arithmatex">\( f(\epsilon) = 1 \)</span>. In this case,
the <a href="#solvation-models-cosmo-dielec-keyword">dielectric constant</a> <span class="arithmatex">\( \epsilon \)</span> is irrelevant, as the solvent is treated as a perfect conductor with <span class="arithmatex">\( \epsilon = \infty \)</span>.</li>
<li><code>screen ks</code> uses the original scaling by Klamt and Schrmann<sup id="fnref3_footnote_1"><a class="footnote-ref" href="#solvation-models-fn_footnote_1">1</a></sup>, given by <span class="arithmatex">\( f(\epsilon) = \frac{\epsilon - 1}{\epsilon + 1/2} \)</span>.</li>
<li><code>screen st</code> applies the modified scaling proposed by Stefanovich and Truong<sup id="fnref3_footnote_5"><a class="footnote-ref" href="#solvation-models-fn_footnote_5">5</a></sup>, defined as <span class="arithmatex">\( f(\epsilon) = \frac{\epsilon - 1}{\epsilon} \)</span>.</li>
</ul>
<p>By default, the modified scaling is used. For high dielectric values, the differences between these scaling methods become negligible.</p>
<h4 id="solvation-models-cosmo-iscreen-keyword-deprecated-see-screen">COSMO: ISCREEN keyword (deprecated, see <a href="#solvation-models-cosmo-screen-keyword">SCREEN</a>)<a class="headerlink" href="#solvation-models-cosmo-iscreen-keyword-deprecated-see-screen" title="Permanent link">&para;</a></h4>
<p><code>iscren</code> is a flag to define the dielectric charge scaling option.
<code>iscren 1</code> implies the original scaling from Klamt and Schrmann,
mainly &ldquo;(&epsilon;-1)/(&epsilon;+1/2)&rdquo;, where &epsilon; is the
dielectric constant. <code>iscren 0</code> implies the modified scaling suggested
by Stefanovich and Truong<sup id="solvation-models-fnref2_footnote_5"><a class="footnote-ref" href="#solvation-models-fn_footnote_5">5</a></sup>, mainly &ldquo;(&epsilon;-1)/&epsilon;&ldquo;.
Default is to use the modified scaling. For high dielectric the
difference between the scaling is not significant.</p>
<h4 id="solvation-models-cosmo-minbem-and-ificos-keywords">COSMO: MINBEM and IFICOS keywords<a class="headerlink" href="#solvation-models-cosmo-minbem-and-ificos-keywords" title="Permanent link">&para;</a></h4>
<p>The <code>minbem</code> and <code>ificos</code> parameters define the tesselation of the unit sphere. The
approach still follows the original proposal by Klamt and Schrmann to
some degree. Basically a tesselation is generated from <code>minbem</code> refining
passes starting from either an octahedron or an icosahedron. Each level
of refinement partitions the triangles of the current tesselation into
four triangles. This procedure is repeated recursively until the desired
granularity of the tesselation is reached. The induced point charges
from the polarization of the medium are assigned to the centers of the
tesselation. The default value is <code>minbem 2</code>. The flag <code>ificos</code> serves
to select the original tesselation, <code>ificos 0</code> for an octahedron
(default) and <code>ificos 1</code> for an icosahedron. Starting from an icosahedron
yields a somewhat finer tesselation that converges somewhat faster.
Solvation energies are not really sensitive to this choice for
sufficiently fine tesselations. The old <code>maxbem</code> directive is no longer
used for the York-Karplus model, while it was used in the <a href="#solvation-models-cosmo-do_cosmo_ks-keyword">Klamt-Schrmann model</a> .</p>
<h4 id="solvation-models-cosmo-lineq-keyword">COSMO: LINEQ keyword<a class="headerlink" href="#solvation-models-cosmo-lineq-keyword" title="Permanent link">&para;</a></h4>
<p>The <code>lineq</code> parameter serves to select the numerical algorithm to solve
the linear equations yielding the effective charges that represent the
polarization of the medium. <code>lineq 0</code> selects a dense matrix linear equation solver
(default), <code>lineq 1</code> selects an iterative method. For
large molecules where the number of effective charges is large, the
code selects the iterative method.</p>
<h4 id="solvation-models-cosmo-zeta-keyword">COSMO: ZETA keyword<a class="headerlink" href="#solvation-models-cosmo-zeta-keyword" title="Permanent link">&para;</a></h4>
<p><code>zeta</code> sets the width of the Gaussian charge distributions that were
suggested by York and Karplus to avoid singularities when two surface
charges coincide. The default value is <code>zeta 0.98</code> this value was chosen
to ensure that the results of the current implementation are as close as
possible to those of the original Klamt and Schrmann based
implementation.</p>
<h4 id="solvation-models-cosmo-gamma_s-keyword">COSMO: GAMMA_S keyword<a class="headerlink" href="#solvation-models-cosmo-gamma_s-keyword" title="Permanent link">&para;</a></h4>
<p><code>gamma_s</code> modifies the width of the smooth switching function that
eliminates surface charges when their positions move into the sphere of
a neighboring atom. <code>gamma_s 0.0</code> leads to a heavyside or abrupt
switching function, whereas <code>gamma_s 1.0</code> maximizes the width of the
switching function. The default value is <code>gamma_s 1.0</code>.</p>
<h4 id="solvation-models-cosmo-sw_tol-keyword">COSMO: SW_TOL keyword<a class="headerlink" href="#solvation-models-cosmo-sw_tol-keyword" title="Permanent link">&para;</a></h4>
<p><code>sw_tol</code> specifies the cutoff of the switching function below which a
surface charge at a particular point is eliminated. The values of the
switching function lie in the domain from 0 to 1. This value should not
be set too small as that leads to instabilities in the linear system
solvers. The default value is <code>sw_tol 1.0e-4</code>.</p>
<h4 id="solvation-models-cosmo-charge_correction-keyword">COSMO: CHARGE_CORRECTION keyword<a class="headerlink" href="#solvation-models-cosmo-charge_correction-keyword" title="Permanent link">&para;</a></h4>
<p>The <code>charge_correction</code> input option defines the surface charge correction to use.<br />
With the input line <code>charge_correction no</code>, no surface charge corrections are used.<br />
With the input line <code>charge_correction scale</code>, both the the electron surface charges and the nuclear surface charges are scaled.<br />
With the default input line <code>charge_correction lagrangian</code>, a lagrangian constraint is used to variationally correct the charges.<br />
The default is <code>charge_correction lagrangian</code>  unless the Klamt-Schuurmann COSMO model is used when the default becomes <code>charge_correction scale</code>.           </p>
<h4 id="solvation-models-cosmo-do_gasphase-keyword">COSMO: DO_GASPHASE keyword<a class="headerlink" href="#solvation-models-cosmo-do_gasphase-keyword" title="Permanent link">&para;</a></h4>
<p><code>do_gasphase</code> is a flag to control whether the calculation of the
solvation energy is preceded by a gas phase calculation. The default is
to always perform a gas phase calculation first and then calculate the
solvation starting from the converged gas phase electron density.
However, in geometry optimizations this approach can double the cost. In
such a case setting <code>do_gasphase false</code> suppresses the gas phase
calculations and only the solvated system calculations are performed.
This option needs to be used with care as in some cases starting the
COSMO solvation from an unconverged electron density can generate
unphysical charges that lock the calculation into strange electron
distributions.</p>
<h4 id="solvation-models-cosmo-do_cosmo_ks-keyword">COSMO: DO_COSMO_KS keyword<a class="headerlink" href="#solvation-models-cosmo-do_cosmo_ks-keyword" title="Permanent link">&para;</a></h4>
<p><code>do_cosmo_ks</code> is a flag to turn on the Klamt-Schrmann model  </p>
<h4 id="solvation-models-cosmo-do_cosmo_yk-keyword">COSMO: DO_COSMO_YK keyword<a class="headerlink" href="#solvation-models-cosmo-do_cosmo_yk-keyword" title="Permanent link">&para;</a></h4>
<p><code>do_cosmo_yk</code> is a flag to turn on the York-Karplus model (default)  </p>
<h4 id="solvation-models-cosmo-do_cosmo_smd-keyword">COSMO: DO_COSMO_SMD keyword<a class="headerlink" href="#solvation-models-cosmo-do_cosmo_smd-keyword" title="Permanent link">&para;</a></h4>
<p><code>do_cosmo_smd</code> is a flag to turn on the SMD model. More details can be found
at the <a href="#solvation-models-smd">SMD Model documentation</a>  </p>
<p>The following example is for a water molecule in &lsquo;water&rsquo;, using the
HF/6-31G** level of
theory:</p>
<pre><code>start  

geometry  
 o                  .0000000000         .0000000000        -.0486020332  
 h                  .7545655371         .0000000000         .5243010666  
 h                 -.7545655371         .0000000000         .5243010666  
end  
basis
 o library 6-31g**  
 h library 6-31g**  
end  
cosmo  
 dielec 78.0  
 radius 1.40  
        1.16  
        1.16  
 lineq  0  
end  
task scf energy
</code></pre>
<p>Alternatively, instead of listing COSMO radii parameters in the input,
the former can be loaded using an external file through the <code>parameters</code>
directive</p>
<pre><code>start  

geometry  
 ow                  .0000000000         .0000000000        -.0486020332  
 hw                  .7545655371         .0000000000         .5243010666  
 h                  -.7545655371         .0000000000         .5243010666  
end  
basis
 * library 6-31g**  
end

cosmo  
 dielec 78.0  
 lineq  0  
 parameters water.par  
end

task scf energy
</code></pre>
<p>where the <code>water.par</code> file has the following form:</p>
<pre><code>O 1.40
H 1.16
</code></pre>
<p>This will set radii of all oxygen atoms to 1.4 and all hydrogen atoms to
1.16. More fine grained control may be achieved using specific atom
names. For example, the following parameter file</p>
<pre><code>O    1.40
H    1.16
HW  1.06
</code></pre>
<p>will set a different radii of 1.06 to hydrogen atoms named HW. Note
that, as per general rule in NWChem, all names are case insensitive.</p>
<p>and placed in one of the these locations - directory specified by the
environmental variable <code>NWCHEM_COSMO_LIBRARY</code>, permanent directory, or
run directory.  </p>
<h2 id="solvation-models-smd">SMD<a class="headerlink" href="#solvation-models-smd" title="Permanent link">&para;</a></h2>
<h3 id="solvation-models-overview_2">Overview<a class="headerlink" href="#solvation-models-overview_2" title="Permanent link">&para;</a></h3>
<p>SMD denotes solvation model based on density and it is described in
detail in the 2009 paper by Marenich, Cramer and Truhlar<sup id="solvation-models-fnref_footnote_7"><a class="footnote-ref" href="#solvation-models-fn_footnote_7">7</a></sup>.  </p>
<p>The SMD model is a universal continuum solvation model where universal
denotes its applicability to any charged or uncharged solute in any
solvent or liquid medium for which a few key descriptors are known. The
word continuum denotes that the solvent is not represented explicitly
as a collection of discrete solvent molecules but rather as a dielectric
medium with surface tensions at the solute-solvent interface.</p>
<p>SMD directly calculates the free energy of solvation of an ideal
solvation process that occurs at fixed concentration (for example, from
an ideal gas at a concentration of 1 mol/L to an ideal solution at a
liquid-phase concentration of 1 mol/L) at 298 K, but this may converted
by standard thermodynamic formulas to a standard-state free energy of
solvation, which is defined as the transfer of molecules from an ideal
gas at 1 bar to an ideal 1 molar solution.</p>
<p>The SMD model separates the fixed-concentration free energy of solvation
into two components. The first component is the bulk-electrostatic
contribution arising from a self-consistent reaction field (SCRF)
treatment. The SCRF treatment involves an integration of the
nonhomogeneous-dielectric Poisson equation for bulk electrostatics in
terms of the <a href="#solvation-models-cosmo">COSMO model</a>
of Klamt and Schrmann with the modified <a href="#solvation-models-cosmo-input-parameters">COSMO scaling factor</a> 
suggested by Stefanovich and Truong and by using the SMD
intrinsic atomic Coulomb radii. These radii have been optimized for H,
C, N, O, F, Si, P, S, Cl, and Br. For any other atom the current
implementation of the SMD model uses scaled values of the van der Waals
radii of Mantina et al<sup id="solvation-models-fnref_footnote_8"><a class="footnote-ref" href="#solvation-models-fn_footnote_8">8</a></sup>.</p>
<p>The scaling factor equals 1.52 for group 17 elements heavier than Br
(i.e., for I and At) and 1.18 for all other elements for which there are
no optimized SMD radii.</p>
<p>The second contribution to the fixed-concentration free energy of
solvation is the contribution arising from short-range interactions
between the solute and solvent molecules in the first solvation shell.
This contribution is called the cavitydispersionsolvent-structure
(CDS) term, and it is a sum of terms that are proportional (with
geometry-dependent proportionality constants called atomic surface
tensions) to the solvent-accessible surface areas (SASAs) of the
individual atoms of the solute.</p>
<h3 id="solvation-models-smd-input-parameters">SMD Input Parameters<a class="headerlink" href="#solvation-models-smd-input-parameters" title="Permanent link">&para;</a></h3>
<p>The SMD model requires additional parameters in the <a href="#solvation-models-cosmo-input-parameters">COSMO input block</a></p>
<pre><code>cosmo  
  [do_cosmo_smd &lt;logical&gt;]
  [solvent (keyword)]
  [icds &lt;integer&gt;]
  [sola &lt;real&gt;]
  [solb &lt;real&gt;]
  [solc &lt;real&gt;]
  [solg &lt;real&gt;]
  [solh &lt;real&gt;]
  [soln &lt;real&gt;]
end  
</code></pre>
<p>At the moment the SMD model is available in NWChem only with the DFT
block</p>
<p>The SMD input options are as follows:</p>
<pre><code>do_cosmo_smd &lt;logical&gt;
</code></pre>
<p>The <code>do_cosmo_smd</code> keyword instructs NWChem to perform a ground-state SMD calculation
when set to a <code>true</code> value.</p>
<h4 id="solvation-models-smd-solvent-keyword">SMD: SOLVENT keyword<a class="headerlink" href="#solvation-models-smd-solvent-keyword" title="Permanent link">&para;</a></h4>
<pre><code>solvent (keyword)
</code></pre>
<p>a <code>solvent</code> keyword from the short name entry in the
<a href="#solvation-models-solvents-list-solvent-keyword">list of available SMD solvent names</a>.</p>
<p>When a solvent is specified by name, the descriptors for the solvent are
based on the Minnesota Solvent Descriptor Database<sup id="solvation-models-fnref_footnote_9"><a class="footnote-ref" href="#solvation-models-fn_footnote_9">9</a></sup>.  </p>
<p>The user can specify a solvent (by using a string using up to eight characters)
that is not on the list by using a new
solvent keyword and introducing user-provided values for the
following solvent descriptors:</p>
<h4 id="solvation-models-smd-dielec-keyword">SMD: DIELEC keyword<a class="headerlink" href="#solvation-models-smd-dielec-keyword" title="Permanent link">&para;</a></h4>
<pre><code>dielec (real input)
</code></pre>
<p>dielectric constant at 298 K  </p>
<h4 id="solvation-models-smd-sola-keyword">SMD: SOLA keyword<a class="headerlink" href="#solvation-models-smd-sola-keyword" title="Permanent link">&para;</a></h4>
<pre><code>sola (real input) 
</code></pre>
<p>Abrahams hydrogen bond acidity   </p>
<h4 id="solvation-models-smd-solb-keyword">SMD: SOLB keyword<a class="headerlink" href="#solvation-models-smd-solb-keyword" title="Permanent link">&para;</a></h4>
<pre><code>solb (real input) 
</code></pre>
<p>Abrahams hydrogen bond basicity  </p>
<h4 id="solvation-models-smd-solc-keyword">SMD: SOLC keyword<a class="headerlink" href="#solvation-models-smd-solc-keyword" title="Permanent link">&para;</a></h4>
<pre><code>solc (real input)
</code></pre>
<p>aromaticity as a fraction of non-hydrogenic solvent atoms that are aromatic carbon atoms  </p>
<h4 id="solvation-models-smd-solg-keyword">SMD: SOLG keyword<a class="headerlink" href="#solvation-models-smd-solg-keyword" title="Permanent link">&para;</a></h4>
<pre><code>solg (real input)
</code></pre>
<p>macroscopic surface tension of the solvent at an air/solvent interface at 298 K in units of
cal mol<sup>1</sup> <sup>2</sup><br />
(note that 1 dyne/cm = 1.43932 cal mol<sup>1</sup> <sup>2</sup>)  </p>
<h4 id="solvation-models-smd-solh-keyword">SMD: SOLH keyword<a class="headerlink" href="#solvation-models-smd-solh-keyword" title="Permanent link">&para;</a></h4>
<pre><code>solh (real input)
</code></pre>
<p>electronegative halogenicity as the fraction of non-hydrogenic solvent atoms that are F, Cl, or Br  </p>
<h4 id="solvation-models-smd-soln-keyword">SMD: SOLN keyword<a class="headerlink" href="#solvation-models-smd-soln-keyword" title="Permanent link">&para;</a></h4>
<pre><code>soln (real input)
</code></pre>
<p>index of refraction at optical frequencies at 293 K</p>
<h4 id="solvation-models-smd-icds-keyword">SMD: ICDS keyword<a class="headerlink" href="#solvation-models-smd-icds-keyword" title="Permanent link">&para;</a></h4>
<pre><code>icds (integer input)
</code></pre>
<p><code>icds</code> should have a value of 1 for water.
<code>icds</code> should have a value of 2  for any nonaqueous solvent.<br />
If <code>icds</code> is set equal to 2, then you need to provide the following solvent descriptors
(see the <a href="https://comp.chem.umn.edu/solvation/mnsddb.pdf">MN solvent descriptor database</a> ):</p>
<ul>
<li><a href="#solvation-models-smd-sola-keyword">sola</a> </li>
<li><a href="#solvation-models-smd-solb-keyword">solb</a></li>
<li><a href="#solvation-models-smd-solc-keyword">solc</a></li>
<li><a href="#solvation-models-smd-solg-keyword">solg</a> </li>
<li><a href="#solvation-models-smd-solh-keyword">solh</a> </li>
<li><a href="#solvation-models-smd-soln-keyword">soln</a> </li>
</ul>
<h3 id="solvation-models-smd-examples">SMD Examples<a class="headerlink" href="#solvation-models-smd-examples" title="Permanent link">&para;</a></h3>
<h3 id="solvation-models-smd-example-water-solvent">SMD Example: water solvent<a class="headerlink" href="#solvation-models-smd-example-water-solvent" title="Permanent link">&para;</a></h3>
<pre><code>echo 
title &quot;SMD/M06-2X/6-31G(d) solvation free energy for CF3COO- in water&quot;
start
charge -1
geometry nocenter
C    0.512211   0.000000  -0.012117
C   -1.061796   0.000000  -0.036672
O   -1.547400   1.150225  -0.006609
O   -1.547182  -1.150320  -0.006608
F    1.061911   1.087605  -0.610341
F    1.061963  -1.086426  -0.612313
F    0.993255  -0.001122   1.266928
symmetry c1
end
basis
* library 6-31G*
end
dft
 XC m06-2x
end
cosmo
 do_cosmo_smd true
 solvent water
end
task dft energy
</code></pre>
<h3 id="solvation-models-smd-example-new-solvent">SMD Example: new solvent<a class="headerlink" href="#solvation-models-smd-example-new-solvent" title="Permanent link">&para;</a></h3>
<p>Example using a user defined solvent, not present in the SMD list of solvents</p>
<pre><code>echo 
title &quot;SMD/M06-2X/6-31G(d) solvation free energy for CF3COO- in my solvent&quot;
start
charge -1
geometry nocenter
C    0.512211   0.000000  -0.012117
C   -1.061796   0.000000  -0.036672
O   -1.547400   1.150225  -0.006609
O   -1.547182  -1.150320  -0.006608
F    1.061911   1.087605  -0.610341
F    1.061963  -1.086426  -0.612313
F    0.993255  -0.001122   1.266928
symmetry c1
end
basis
* library 6-31G*
end
dft
 XC m06-2x
end
cosmo
 do_cosmo_smd true
 solvent mysolv 
 dielec 11.4
 sola 1.887
 solb 0.0
 soln 0.98
 icds 2
end
task dft energy
</code></pre>
<h2 id="solvation-models-solvents-list-solvent-keyword">Solvents List - Solvent keyword<a class="headerlink" href="#solvation-models-solvents-list-solvent-keyword" title="Permanent link">&para;</a></h2>
<p>The short name for the solvent from the table can be used  with the <code>solvent</code> keyword to define the solvent.<br />
Example with acetonitrile.</p>
<pre><code>cosmo
  solvent acetntrl
end
</code></pre>
<table>
<thead>
<tr>
<th>Long name</th>
<th>short name</th>
<th style="text-align: right;">dielec</th>
</tr>
</thead>
<tbody>
<tr>
<td>acetic acid</td>
<td>acetacid</td>
<td style="text-align: right;">6.2528</td>
</tr>
<tr>
<td>acetone</td>
<td>acetone</td>
<td style="text-align: right;">20.493</td>
</tr>
<tr>
<td>acetonitrile</td>
<td>acetntrl</td>
<td style="text-align: right;">35.688</td>
</tr>
<tr>
<td>acetophenone</td>
<td>acetphen</td>
<td style="text-align: right;">17.440</td>
</tr>
<tr>
<td>aniline</td>
<td>aniline</td>
<td style="text-align: right;">6.8882</td>
</tr>
<tr>
<td>anisole</td>
<td>anisole</td>
<td style="text-align: right;">4.2247</td>
</tr>
<tr>
<td>benzaldehyde</td>
<td>benzaldh</td>
<td style="text-align: right;">18.220</td>
</tr>
<tr>
<td>benzene</td>
<td>benzene</td>
<td style="text-align: right;">2.2706</td>
</tr>
<tr>
<td>benzonitrile</td>
<td>benzntrl</td>
<td style="text-align: right;">25.592</td>
</tr>
<tr>
<td>benzyl chloride</td>
<td>benzylcl</td>
<td style="text-align: right;">6.7175</td>
</tr>
<tr>
<td>1-bromo-2-methylpropane</td>
<td>brisobut</td>
<td style="text-align: right;">7.7792</td>
</tr>
<tr>
<td>bromobenzene</td>
<td>brbenzen</td>
<td style="text-align: right;">5.3954</td>
</tr>
<tr>
<td>bromoethane</td>
<td>brethane</td>
<td style="text-align: right;">9.01</td>
</tr>
<tr>
<td>bromoform</td>
<td>bromform</td>
<td style="text-align: right;">4.2488</td>
</tr>
<tr>
<td>1-bromooctane</td>
<td>broctane</td>
<td style="text-align: right;">5.0244</td>
</tr>
<tr>
<td>1-bromopentane</td>
<td>brpentan</td>
<td style="text-align: right;">6.269</td>
</tr>
<tr>
<td>2-bromopropane</td>
<td>brpropa2</td>
<td style="text-align: right;">9.3610</td>
</tr>
<tr>
<td>1-bromopropane</td>
<td>brpropan</td>
<td style="text-align: right;">8.0496</td>
</tr>
<tr>
<td>butanal</td>
<td>butanal</td>
<td style="text-align: right;">13.450</td>
</tr>
<tr>
<td>butanoic acid</td>
<td>butacid</td>
<td style="text-align: right;">2.9931</td>
</tr>
<tr>
<td>1-butanol</td>
<td>butanol</td>
<td style="text-align: right;">17.332</td>
</tr>
<tr>
<td>2-butanol</td>
<td>butanol2</td>
<td style="text-align: right;">15.944</td>
</tr>
<tr>
<td>butanone</td>
<td>butanone</td>
<td style="text-align: right;">18.246</td>
</tr>
<tr>
<td>butanonitrile</td>
<td>butantrl</td>
<td style="text-align: right;">24.291</td>
</tr>
<tr>
<td>butyl acetate</td>
<td>butile</td>
<td style="text-align: right;">4.9941</td>
</tr>
<tr>
<td>butylamine</td>
<td>nba</td>
<td style="text-align: right;">4.6178</td>
</tr>
<tr>
<td>n-butylbenzene</td>
<td>nbutbenz</td>
<td style="text-align: right;">2.360</td>
</tr>
<tr>
<td>sec-butylbenzene</td>
<td>sbutbenz</td>
<td style="text-align: right;">2.3446</td>
</tr>
<tr>
<td>tert-butylbenzene</td>
<td>tbutbenz</td>
<td style="text-align: right;">2.3447</td>
</tr>
<tr>
<td>carbon disulfide</td>
<td>cs2</td>
<td style="text-align: right;">2.6105</td>
</tr>
<tr>
<td>carbon tetrachloride</td>
<td>carbntet</td>
<td style="text-align: right;">2.2280</td>
</tr>
<tr>
<td>chlorobenzene</td>
<td>clbenzen</td>
<td style="text-align: right;">5.6968</td>
</tr>
<tr>
<td>sec-butyl chloride</td>
<td>secbutcl</td>
<td style="text-align: right;">8.3930</td>
</tr>
<tr>
<td>chloroform</td>
<td>chcl3</td>
<td style="text-align: right;">4.7113</td>
</tr>
<tr>
<td>1-chlorohexane</td>
<td>clhexane</td>
<td style="text-align: right;">5.9491</td>
</tr>
<tr>
<td>1-chloropentane</td>
<td>clpentan</td>
<td style="text-align: right;">6.5022</td>
</tr>
<tr>
<td>1-chloropropane</td>
<td>clpropan</td>
<td style="text-align: right;">8.3548</td>
</tr>
<tr>
<td>o-chlorotoluene</td>
<td>ocltolue</td>
<td style="text-align: right;">4.6331</td>
</tr>
<tr>
<td>m-cresol</td>
<td>m-cresol</td>
<td style="text-align: right;">12.440</td>
</tr>
<tr>
<td>o-cresol</td>
<td>o-cresol</td>
<td style="text-align: right;">6.760</td>
</tr>
<tr>
<td>cyclohexane</td>
<td>cychexan</td>
<td style="text-align: right;">2.0165</td>
</tr>
<tr>
<td>cyclohexanone</td>
<td>cychexon</td>
<td style="text-align: right;">15.619</td>
</tr>
<tr>
<td>cyclopentane</td>
<td>cycpentn</td>
<td style="text-align: right;">1.9608</td>
</tr>
<tr>
<td>cyclopentanol</td>
<td>cycpntol</td>
<td style="text-align: right;">16.989</td>
</tr>
<tr>
<td>cyclopentanone</td>
<td>cycpnton</td>
<td style="text-align: right;">13.58</td>
</tr>
<tr>
<td>cis-decalin</td>
<td>declncis</td>
<td style="text-align: right;">2.2139</td>
</tr>
<tr>
<td>trans-decalin</td>
<td>declntra</td>
<td style="text-align: right;">2.1781</td>
</tr>
<tr>
<td>decalin (cis/trans mixture)</td>
<td>declnmix</td>
<td style="text-align: right;">2.196</td>
</tr>
<tr>
<td>n-decane</td>
<td>decane</td>
<td style="text-align: right;">1.9846</td>
</tr>
<tr>
<td>1-decanol</td>
<td>decanol</td>
<td style="text-align: right;">7.5305</td>
</tr>
<tr>
<td>1,2-dibromoethane</td>
<td>edb12</td>
<td style="text-align: right;">4.9313</td>
</tr>
<tr>
<td>dibromomethane</td>
<td>dibrmetn</td>
<td style="text-align: right;">7.2273</td>
</tr>
<tr>
<td>dibutyl ether</td>
<td>butyleth</td>
<td style="text-align: right;">3.0473</td>
</tr>
<tr>
<td>o-dichlorobenzene</td>
<td>odiclbnz</td>
<td style="text-align: right;">9.9949</td>
</tr>
<tr>
<td>1,2-dichloroethane</td>
<td>edc12</td>
<td style="text-align: right;">10.125</td>
</tr>
<tr>
<td>cis-dichloroethylene</td>
<td>c12dce</td>
<td style="text-align: right;">9.200</td>
</tr>
<tr>
<td>trans-dichloroethylene</td>
<td>t12dce</td>
<td style="text-align: right;">2.140</td>
</tr>
<tr>
<td>dichloromethane</td>
<td>dcm</td>
<td style="text-align: right;">8.930</td>
</tr>
<tr>
<td>diethyl ether</td>
<td>ether</td>
<td style="text-align: right;">4.2400</td>
</tr>
<tr>
<td>diethyl sulfide</td>
<td>et2s</td>
<td style="text-align: right;">5.723</td>
</tr>
<tr>
<td>diethylamine</td>
<td>dietamin</td>
<td style="text-align: right;">3.5766</td>
</tr>
<tr>
<td>diiodomethane</td>
<td>mi</td>
<td style="text-align: right;">5.320</td>
</tr>
<tr>
<td>diisopropyl ether</td>
<td>dipe</td>
<td style="text-align: right;">3.380</td>
</tr>
<tr>
<td>dimethyl disulfide</td>
<td>dmds</td>
<td style="text-align: right;">9.600</td>
</tr>
<tr>
<td>dimethylsulfoxide</td>
<td>dmso</td>
<td style="text-align: right;">46.826</td>
</tr>
<tr>
<td>N,N-dimethylacetamide</td>
<td>dma</td>
<td style="text-align: right;">37.781</td>
</tr>
<tr>
<td>cis-1,2-dimethylcyclohexane</td>
<td>cisdmchx</td>
<td style="text-align: right;">2.060</td>
</tr>
<tr>
<td>N,N-dimethylformamide</td>
<td>dmf</td>
<td style="text-align: right;">37.219</td>
</tr>
<tr>
<td>2,4-dimethylpentane</td>
<td>dmepen24</td>
<td style="text-align: right;">1.8939</td>
</tr>
<tr>
<td>2,4-dimethylpyridine</td>
<td>dmepyr24</td>
<td style="text-align: right;">9.4176</td>
</tr>
<tr>
<td>2,6-dimethylpyridine</td>
<td>dmepyr26</td>
<td style="text-align: right;">7.1735</td>
</tr>
<tr>
<td>1,4-dioxane</td>
<td>dioxane</td>
<td style="text-align: right;">2.2099</td>
</tr>
<tr>
<td>diphenyl ether</td>
<td>phoph</td>
<td style="text-align: right;">3.730</td>
</tr>
<tr>
<td>dipropylamine</td>
<td>dproamin</td>
<td style="text-align: right;">2.9112</td>
</tr>
<tr>
<td>n-dodecane</td>
<td>dodecan</td>
<td style="text-align: right;">2.0060</td>
</tr>
<tr>
<td>1,2-ethanediol</td>
<td>meg</td>
<td style="text-align: right;">40.245</td>
</tr>
<tr>
<td>ethanethiol</td>
<td>etsh</td>
<td style="text-align: right;">6.667</td>
</tr>
<tr>
<td>ethanol</td>
<td>ethanol</td>
<td style="text-align: right;">24.852</td>
</tr>
<tr>
<td>ethyl acetate</td>
<td>etoac</td>
<td style="text-align: right;">5.9867</td>
</tr>
<tr>
<td>ethyl formate</td>
<td>etome</td>
<td style="text-align: right;">8.3310</td>
</tr>
<tr>
<td>ethylbenzene</td>
<td>eb</td>
<td style="text-align: right;">2.4339</td>
</tr>
<tr>
<td>ethylphenyl ether</td>
<td>phenetol</td>
<td style="text-align: right;">4.1797</td>
</tr>
<tr>
<td>fluorobenzene</td>
<td>c6h5f</td>
<td style="text-align: right;">5.420</td>
</tr>
<tr>
<td>1-fluorooctane</td>
<td>foctane</td>
<td style="text-align: right;">3.890</td>
</tr>
<tr>
<td>formamide</td>
<td>formamid</td>
<td style="text-align: right;">108.94</td>
</tr>
<tr>
<td>formic acid</td>
<td>formacid</td>
<td style="text-align: right;">51.100</td>
</tr>
<tr>
<td>n-heptane</td>
<td>heptane</td>
<td style="text-align: right;">1.9113</td>
</tr>
<tr>
<td>1-heptanol</td>
<td>heptanol</td>
<td style="text-align: right;">11.321</td>
</tr>
<tr>
<td>2-heptanone</td>
<td>heptnon2</td>
<td style="text-align: right;">11.658</td>
</tr>
<tr>
<td>4-heptanone</td>
<td>heptnon4</td>
<td style="text-align: right;">12.257</td>
</tr>
<tr>
<td>n-hexadecane</td>
<td>hexadecn</td>
<td style="text-align: right;">2.0402</td>
</tr>
<tr>
<td>n-hexane</td>
<td>hexane</td>
<td style="text-align: right;">1.8819</td>
</tr>
<tr>
<td>hexanoic acid</td>
<td>hexnacid</td>
<td style="text-align: right;">2.600</td>
</tr>
<tr>
<td>1-hexanol</td>
<td>hexanol</td>
<td style="text-align: right;">12.51</td>
</tr>
<tr>
<td>2-hexanone</td>
<td>hexanon2</td>
<td style="text-align: right;">14.136</td>
</tr>
<tr>
<td>1-hexene</td>
<td>hexene</td>
<td style="text-align: right;">2.0717</td>
</tr>
<tr>
<td>1-hexyne</td>
<td>hexyne</td>
<td style="text-align: right;">2.615</td>
</tr>
<tr>
<td>iodobenzene</td>
<td>c6h5i</td>
<td style="text-align: right;">4.5470</td>
</tr>
<tr>
<td>1-iodobutane</td>
<td>iobutane</td>
<td style="text-align: right;">6.173</td>
</tr>
<tr>
<td>iodoethane</td>
<td>c2h5i</td>
<td style="text-align: right;">7.6177</td>
</tr>
<tr>
<td>1-iodohexadecane</td>
<td>iohexdec</td>
<td style="text-align: right;">3.5338</td>
</tr>
<tr>
<td>iodomethane</td>
<td>ch3i</td>
<td style="text-align: right;">6.8650</td>
</tr>
<tr>
<td>1-iodopentane</td>
<td>iopentan</td>
<td style="text-align: right;">5.6973</td>
</tr>
<tr>
<td>1-iodopropane</td>
<td>iopropan</td>
<td style="text-align: right;">6.9626</td>
</tr>
<tr>
<td>isopropylbenzene</td>
<td>cumene</td>
<td style="text-align: right;">2.3712</td>
</tr>
<tr>
<td>p-isopropyltoluene</td>
<td>p-cymene</td>
<td style="text-align: right;">2.2322</td>
</tr>
<tr>
<td>mesitylene</td>
<td>mesityln</td>
<td style="text-align: right;">2.2650</td>
</tr>
<tr>
<td>methanol</td>
<td>methanol</td>
<td style="text-align: right;">32.613</td>
</tr>
<tr>
<td>2-methoxyethanol</td>
<td>egme</td>
<td style="text-align: right;">17.200</td>
</tr>
<tr>
<td>methyl acetate</td>
<td>meacetat</td>
<td style="text-align: right;">6.8615</td>
</tr>
<tr>
<td>methyl benzoate</td>
<td>mebnzate</td>
<td style="text-align: right;">6.7367</td>
</tr>
<tr>
<td>methyl butanoate</td>
<td>mebutate</td>
<td style="text-align: right;">5.5607</td>
</tr>
<tr>
<td>methyl formate</td>
<td>meformat</td>
<td style="text-align: right;">8.8377</td>
</tr>
<tr>
<td>4-methyl-2-pentanone</td>
<td>mibk</td>
<td style="text-align: right;">12.887</td>
</tr>
<tr>
<td>methyl propanoate</td>
<td>mepropyl</td>
<td style="text-align: right;">6.0777</td>
</tr>
<tr>
<td>2-methyl-1-propanol</td>
<td>isobutol</td>
<td style="text-align: right;">16.777</td>
</tr>
<tr>
<td>2-methyl-2-propanol</td>
<td>terbutol</td>
<td style="text-align: right;">12.470</td>
</tr>
<tr>
<td>N-methylaniline</td>
<td>nmeaniln</td>
<td style="text-align: right;">5.9600</td>
</tr>
<tr>
<td>methylcyclohexane</td>
<td>mecychex</td>
<td style="text-align: right;">2.024</td>
</tr>
<tr>
<td>N-methylformamide (E/Z mixture)</td>
<td>nmfmixtr</td>
<td style="text-align: right;">181.56</td>
</tr>
<tr>
<td>2-methylpentane</td>
<td>isohexan</td>
<td style="text-align: right;">1.890</td>
</tr>
<tr>
<td>2-methylpyridine</td>
<td>mepyrid2</td>
<td style="text-align: right;">9.9533</td>
</tr>
<tr>
<td>3-methylpyridine</td>
<td>mepyrid3</td>
<td style="text-align: right;">11.645</td>
</tr>
<tr>
<td>4-methylpyridine</td>
<td>mepyrid4</td>
<td style="text-align: right;">11.957</td>
</tr>
<tr>
<td>nitrobenzene</td>
<td>c6h5no2</td>
<td style="text-align: right;">34.809</td>
</tr>
<tr>
<td>nitroethane</td>
<td>c2h5no2</td>
<td style="text-align: right;">28.290</td>
</tr>
<tr>
<td>nitromethane</td>
<td>ch3no2</td>
<td style="text-align: right;">36.562</td>
</tr>
<tr>
<td>1-nitropropane</td>
<td>ntrprop1</td>
<td style="text-align: right;">23.730</td>
</tr>
<tr>
<td>2-nitropropane</td>
<td>ntrprop2</td>
<td style="text-align: right;">25.654</td>
</tr>
<tr>
<td>o-nitrotoluene</td>
<td>ontrtolu</td>
<td style="text-align: right;">25.669</td>
</tr>
<tr>
<td>n-nonane</td>
<td>nonane</td>
<td style="text-align: right;">1.9605</td>
</tr>
<tr>
<td>1-nonanol</td>
<td>nonanol</td>
<td style="text-align: right;">8.5991</td>
</tr>
<tr>
<td>5-nonanone</td>
<td>nonanone</td>
<td style="text-align: right;">10.600</td>
</tr>
<tr>
<td>n-octane</td>
<td>octane</td>
<td style="text-align: right;">1.9406</td>
</tr>
<tr>
<td>1-octanol</td>
<td>octanol</td>
<td style="text-align: right;">9.8629</td>
</tr>
<tr>
<td>2-octanone</td>
<td>octanon2</td>
<td style="text-align: right;">9.4678</td>
</tr>
<tr>
<td>n-pentadecane</td>
<td>pentdecn</td>
<td style="text-align: right;">2.0333</td>
</tr>
<tr>
<td>pentanal</td>
<td>pentanal</td>
<td style="text-align: right;">10.000</td>
</tr>
<tr>
<td>n-pentane</td>
<td>npentane</td>
<td style="text-align: right;">1.8371</td>
</tr>
<tr>
<td>pentanoic acid</td>
<td>pentacid</td>
<td style="text-align: right;">2.6924</td>
</tr>
<tr>
<td>1-pentanol</td>
<td>pentanol</td>
<td style="text-align: right;">15.130</td>
</tr>
<tr>
<td>2-pentanone</td>
<td>pentnon2</td>
<td style="text-align: right;">15.200</td>
</tr>
<tr>
<td>3-pentanone</td>
<td>pentnon3</td>
<td style="text-align: right;">16.780</td>
</tr>
<tr>
<td>1-pentene</td>
<td>pentene</td>
<td style="text-align: right;">1.9905</td>
</tr>
<tr>
<td>E-2-pentene</td>
<td>e2penten</td>
<td style="text-align: right;">2.051</td>
</tr>
<tr>
<td>pentyl acetate</td>
<td>pentacet</td>
<td style="text-align: right;">4.7297</td>
</tr>
<tr>
<td>pentylamine</td>
<td>pentamin</td>
<td style="text-align: right;">4.2010</td>
</tr>
<tr>
<td>perfluorobenzene</td>
<td>pfb</td>
<td style="text-align: right;">2.029</td>
</tr>
<tr>
<td>phenylmethanol</td>
<td>benzalcl</td>
<td style="text-align: right;">12.457</td>
</tr>
<tr>
<td>propanal</td>
<td>propanal</td>
<td style="text-align: right;">18.500</td>
</tr>
<tr>
<td>propanoic acid</td>
<td>propacid</td>
<td style="text-align: right;">3.440</td>
</tr>
<tr>
<td>1-propanol</td>
<td>propanol</td>
<td style="text-align: right;">20.524</td>
</tr>
<tr>
<td>2-propanol</td>
<td>propnol2</td>
<td style="text-align: right;">19.264</td>
</tr>
<tr>
<td>propanonitrile</td>
<td>propntrl</td>
<td style="text-align: right;">29.324</td>
</tr>
<tr>
<td>2-propen-1-ol</td>
<td>propenol</td>
<td style="text-align: right;">19.011</td>
</tr>
<tr>
<td>propyl acetate</td>
<td>propacet</td>
<td style="text-align: right;">5.5205</td>
</tr>
<tr>
<td>propylamine</td>
<td>propamin</td>
<td style="text-align: right;">4.9912</td>
</tr>
<tr>
<td>pyridine</td>
<td>pyridine</td>
<td style="text-align: right;">12.978</td>
</tr>
<tr>
<td>tetrachloroethene</td>
<td>c2cl4</td>
<td style="text-align: right;">2.268</td>
</tr>
<tr>
<td>tetrahydrofuran</td>
<td>thf</td>
<td style="text-align: right;">7.4257</td>
</tr>
<tr>
<td>tetrahydrothiophene-S,S-dioxide</td>
<td>sulfolan</td>
<td style="text-align: right;">43.962</td>
</tr>
<tr>
<td>tetralin</td>
<td>tetralin</td>
<td style="text-align: right;">2.771</td>
</tr>
<tr>
<td>thiophene</td>
<td>thiophen</td>
<td style="text-align: right;">2.7270</td>
</tr>
<tr>
<td>thiophenol</td>
<td>phsh</td>
<td style="text-align: right;">4.2728</td>
</tr>
<tr>
<td>toluene</td>
<td>toluene</td>
<td style="text-align: right;">2.3741</td>
</tr>
<tr>
<td>tributyl phosphate</td>
<td>tbp</td>
<td style="text-align: right;">8.1781</td>
</tr>
<tr>
<td>1,1,1-trichloroethane</td>
<td>tca111</td>
<td style="text-align: right;">7.0826</td>
</tr>
<tr>
<td>1,1,2-trichloroethane</td>
<td>tca112</td>
<td style="text-align: right;">7.1937</td>
</tr>
<tr>
<td>trichloroethene</td>
<td>tce</td>
<td style="text-align: right;">3.422</td>
</tr>
<tr>
<td>triethylamine</td>
<td>et3n</td>
<td style="text-align: right;">2.3832</td>
</tr>
<tr>
<td>2,2,2-trifluoroethanol</td>
<td>tfe222</td>
<td style="text-align: right;">26.726</td>
</tr>
<tr>
<td>1,2,4-trimethylbenzene</td>
<td>tmben124</td>
<td style="text-align: right;">2.3653</td>
</tr>
<tr>
<td>2,2,4-trimethylpentane</td>
<td>isoctane</td>
<td style="text-align: right;">1.9358</td>
</tr>
<tr>
<td>n-undecane</td>
<td>undecane</td>
<td style="text-align: right;">1.9910</td>
</tr>
<tr>
<td>m-xylene</td>
<td>m-xylene</td>
<td style="text-align: right;">2.3478</td>
</tr>
<tr>
<td>o-xylene</td>
<td>o-xylene</td>
<td style="text-align: right;">2.5454</td>
</tr>
<tr>
<td>p-xylene</td>
<td>p-xylene</td>
<td style="text-align: right;">2.2705</td>
</tr>
<tr>
<td>xylene (mixture)</td>
<td>xylenemx</td>
<td style="text-align: right;">2.3879</td>
</tr>
<tr>
<td>water</td>
<td>h2o</td>
<td style="text-align: right;">78.400</td>
</tr>
</tbody>
</table>
<h2 id="solvation-models-usage-tips">Usage Tips<a class="headerlink" href="#solvation-models-usage-tips" title="Permanent link">&para;</a></h2>
<p>Authors of paper <sup id="fnref2_footnote_7"><a class="footnote-ref" href="#solvation-models-fn_footnote_7">7</a></sup> report that<br />
&rdquo; &hellip; <em>the SMD/COSMO/NWChem calculations we employed finer grids (options minbem=3, maxbem=4, ificos=1) because the default NWChem tessellation parameters (options: minbem=2, maxbem=3, ificos=0) produced very large errors in solvation free energies</em>.&rdquo;<br />
Since the <code>maxbem</code> keyword is no longer in use, this paper&rsquo;s recommended input translate into</p>
<pre><code>cosmo
 minbem 3
 ificos 1
end
</code></pre>
<h2 id="solvation-models-references">References<a class="headerlink" href="#solvation-models-references" title="Permanent link">&para;</a></h2>
<div class="footnote">
<hr />
<ol>
<li id="solvation-models-fn_footnote_1">
<p>Klamt, A.; Schrmann, G. COSMO: A New Approach to Dielectric Screening in Solvents with Explicit Expressions for the Screening Energy and Its Gradient. <em>J. Chem. Soc., Perkin Trans. 2</em> <strong>1993</strong>, No. 5, 799805. <a href="https://doi.org/10.1039/p29930000799">https://doi.org/10.1039/p29930000799</a>.&#160;<a class="footnote-backref" href="#solvation-models-fnref_footnote_1" title="Jump back to footnote 1 in the text">&#8617;</a><a class="footnote-backref" href="#solvation-models-fnref2_footnote_1" title="Jump back to footnote 1 in the text">&#8617;</a><a class="footnote-backref" href="#solvation-models-fnref3_footnote_1" title="Jump back to footnote 1 in the text">&#8617;</a></p>
</li>
<li id="solvation-models-fn_footnote_2">
<p>York, D. M.; Karplus, M. A Smooth Solvation Potential Based on the Conductor-Like Screening Model. <em>The Journal of Physical Chemistry A</em> <strong>1999</strong>, <em>103</em> (50), 1106011079. <a href="https://doi.org/10.1021/jp992097l">https://doi.org/10.1021/jp992097l</a>.&#160;<a class="footnote-backref" href="#solvation-models-fnref_footnote_2" title="Jump back to footnote 2 in the text">&#8617;</a></p>
</li>
<li id="solvation-models-fn_footnote_3">
<p>Bondi, A. Van Der Waals Volumes and Radii. <em>The Journal of Physical Chemistry</em> <strong>1964</strong>, <em>68</em> (3), 441451. <a href="https://doi.org/10.1021/j100785a001">https://doi.org/10.1021/j100785a001</a>.&#160;<a class="footnote-backref" href="#solvation-models-fnref_footnote_3" title="Jump back to footnote 3 in the text">&#8617;</a></p>
</li>
<li id="solvation-models-fn_footnote_4">
<p>Klamt, A.; Jonas, V.; Brger, T.; Lohrenz, J. C. W. Refinement and Parametrization of COSMO-RS. <em>The Journal of Physical Chemistry A</em> <strong>1998</strong>, <em>102</em> (26), 50745085. <a href="https://doi.org/10.1021/jp980017s">https://doi.org/10.1021/jp980017s</a>.&#160;<a class="footnote-backref" href="#solvation-models-fnref_footnote_4" title="Jump back to footnote 4 in the text">&#8617;</a></p>
</li>
<li id="solvation-models-fn_footnote_5">
<p>Stefanovich, E. V.; Truong, T. N. Optimized Atomic Radii for Quantum Dielectric Continuum Solvation Models. <em>Chemical Physics Letters</em> <strong>1995</strong>, <em>244</em> (1-2), 6574. <a href="https://doi.org/10.1016/0009-2614(95)00898-e">https://doi.org/10.1016/0009-2614(95)00898-e</a>.&#160;<a class="footnote-backref" href="#solvation-models-fnref_footnote_5" title="Jump back to footnote 5 in the text">&#8617;</a><a class="footnote-backref" href="#solvation-models-fnref2_footnote_5" title="Jump back to footnote 5 in the text">&#8617;</a><a class="footnote-backref" href="#solvation-models-fnref3_footnote_5" title="Jump back to footnote 5 in the text">&#8617;</a></p>
</li>
<li id="solvation-models-fn_footnote_6">
<p>Barone, V.; Cossi, M.; Tomasi, J. A New Definition of Cavities for the Computation of Solvation Free Energies by the Polarizable Continuum Model. <em>The Journal of Chemical Physics</em> <strong>1997</strong>, <em>107</em> (8), 32103221. <a href="https://doi.org/10.1063/1.474671">https://doi.org/10.1063/1.474671</a>.&#160;<a class="footnote-backref" href="#solvation-models-fnref_footnote_6" title="Jump back to footnote 6 in the text">&#8617;</a></p>
</li>
<li id="solvation-models-fn_footnote_7">
<p>Marenich, A. V.; Cramer, C. J.; Truhlar, D. G. Universal Solvation Model Based on Solute Electron Density and on a Continuum Model of the Solvent Defined by the Bulk Dielectric Constant and Atomic Surface Tensions. <em>The Journal of Physical Chemistry B</em> <strong>2009</strong>, <em>113</em> (18), 63786396. <a href="https://doi.org/10.1021/jp810292n">https://doi.org/10.1021/jp810292n</a>.&#160;<a class="footnote-backref" href="#solvation-models-fnref_footnote_7" title="Jump back to footnote 7 in the text">&#8617;</a><a class="footnote-backref" href="#solvation-models-fnref2_footnote_7" title="Jump back to footnote 7 in the text">&#8617;</a></p>
</li>
<li id="solvation-models-fn_footnote_8">
<p>Haynes, W. M. CRC Handbook of Chemistry and Physics; Mantina, M., Valero, R., Cramer, C. J., Truhlar, D. G., Eds.; Taylor &amp; Francis Group, 2013; pp 949.&#160;<a class="footnote-backref" href="#solvation-models-fnref_footnote_8" title="Jump back to footnote 8 in the text">&#8617;</a></p>
</li>
<li id="solvation-models-fn_footnote_9">
<p>Winget, P.; Dolney, D. M.; Giesen, D. J.; Cramer, C. J.; Truhlar, D. G. Minnesota Solvent Descriptor Database. <em>Minneapolis, MN: Department of Chemistry and Supercomputer Institute</em> <strong>1999</strong>.&#160;<a class="footnote-backref" href="#solvation-models-fnref_footnote_9" title="Jump back to footnote 9 in the text">&#8617;</a></p>
</li>
</ol>
</div></section><section class="print-page" id="vem-model" heading-number="13.2"><h1 id="vem-model-vem-vertical-excitation-or-emission-model">VEM (Vertical Excitation or Emission) Model<a class="headerlink" href="#vem-model-vem-vertical-excitation-or-emission-model" title="Permanent link">&para;</a></h1>
<h2 id="vem-model-overview">Overview<a class="headerlink" href="#vem-model-overview" title="Permanent link">&para;</a></h2>
<p>The VEM is a model for calculating the vertical excitation (absorption)
or vertical emission (fluorescence) energy in solution according to a
two-time-scale model of solvent polarization. The model is described in reference<sup id="vem-model-fnref_footnote_1"><a class="footnote-ref" href="#vem-model-fn_footnote_1">1</a></sup>.  </p>
<p>The current implementation is based on the VEM(d,RD) algorithm as
described in the above paper. The method is available only at the <a href="#excited-state-calculations">TDDFT</a>
level of theory, including both full-linear response TDDFT (sometimes
called LR-TDDFT or regular TDDFT) and the TammDancoff approximation,
<a href="#excited-state-calculations-cis-and-rpa-the-tamm-dancoff-approximation">TDDFT-TDA</a>
(sometimes just called TDA). The configuration interaction
singles (CIS) wave function method can also be used along with VEM by
considering CIS to be a special case of TDDFT-TDA.</p>
<p>The abbreviation VEM originally referred to the vertical excitation
model of reference<sup id="vem-model-fnref_footnote_2"><a class="footnote-ref" href="#vem-model-fn_footnote_2">2</a></sup>,
but the current implementation of VEM extends to both excitation and
emission calculations in solution, and the E in VEM now stands for
excitation/emission. Furthermore, the current version of VEM (based on
the Marenich et al. paper<sup id="vem-model-fnref2_footnote_1"><a class="footnote-ref" href="#vem-model-fn_footnote_1">1</a></sup>) does
not reduce to the original VEM of Li et al., but is improved as
described in reference<sup id="vem-model-fnref3_footnote_1"><a class="footnote-ref" href="#vem-model-fn_footnote_1">1</a></sup>.</p>
<p>The VEM model is based on a nonequilibrium dielectric-continuum approach
in terms of two-time-scale solvent response. The solvent&rsquo;s
bulk-electrostatic polarization is described by using the reaction field
partitioned into slow and fast components, and only the fast component
is self-consistently (iteratively) equilibrated with the charge density
of the solute molecule in its final state. During the VEM calculation,
the slow component is kept in equilibrium with the initial state&rsquo;s
solute charge density but not with the final state&rsquo;s one. In the case of
vertical absorption the initial state is the ground electronic state of
the solute molecule in solution and the final state is an excited
electronic state in solution (and vice versa in the case of an emission
spectrum). Both the ground- and excited-state calculations involve an
integration of the nonhomogeneous-dielectric Poisson equation for bulk
electrostatics in terms of the COSMO model as implemented in NWChem with
the modified COSMO scaling factor (<code>iscren 0</code>) and by using the SMD
intrinsic atomic Coulomb radii (by default; see the section of the
manual describing SMD). The excited-state electron density is calculated
using the Z-Vector &ldquo;relaxed density&rdquo; approach.</p>
<p>The VEM excitation or emission energy includes only a bulk-electrostatic
contribution without any cavitydispersionsolvent-structure (CDS)
contributions (such contributions are used in SMD ground-state
calculations as described in the SMD section of this manual, but are not
used in VEM calculations). When one considers solvatochromic shifts, the
main contributions beyond bulk electrostatics are solutesolvent
dispersion interactions, hydrogen bonding (the latter is most important
in protic solvents), and perhaps charge transfer between the solute and
the solvent. To explicitly account for solutesolvent charge transfer
and hydrogen bonding, the user can run a VEM calculation on a
supersolute that involves a solutesolvent molecular cluster with one or
a few solvent molecules added explicitly to a bare solute. The
solutesolvent dispersion contribution to the solvatochromic shift, if
desired, can be estimated by the solvation model with state-specific
polarizability (SMSSP) described in reference<sup id="vem-model-fnref_footnote_3"><a class="footnote-ref" href="#vem-model-fn_footnote_3">3</a></sup>.  </p>
<p>In this case, the user needs to provide values of ground- and
excited-state spherically averaged molecular polarizabilities of the
solvent.</p>
<h2 id="vem-model-syntax">Syntax<a class="headerlink" href="#vem-model-syntax" title="Permanent link">&para;</a></h2>
<p>The VEM-specific input options are as
follows:</p>
<h3 id="vem-model-do_cosmo_vem">DO_COSMO_VEM:<a class="headerlink" href="#vem-model-do_cosmo_vem" title="Permanent link">&para;</a></h3>
<pre><code> do_cosmo_vem &lt;integer do_cosmo_vem default 0&gt;
</code></pre>
<p>The <code>do_cosmo_vem</code> can be set to the following values:<br />
- <code>0</code> (do not do any VEM calculation even if the task tddft gradient line is present; default).<br />
- <code>1</code> (do a nonequilibrium VEM excitation energy calculation;<br />
in this case the <code>task tddft gradient</code> line should be present, too)  <br />
- <code>2</code> (do an equilibrium VEM excitation energy calculation followed by<br />
a nonequilibrium emission energy calculation;<br />
<code>task tddft gradient</code> line should be present)    </p>
<h3 id="vem-model-vem-solvent">VEM Solvent<a class="headerlink" href="#vem-model-vem-solvent" title="Permanent link">&para;</a></h3>
<p>The VEM solvent (which is water by default) can be specified by using
the solvent keyword described in the <a href="#smd-model-smd-solvent-keyword">SMD section of this manual</a> or by
specifying the VEM solvent descriptors such as</p>
<p><code>dielec (real input)</code><br />
<code>static dielectric constant</code>  </p>
<p><code>dielecinf (real input)</code>  <br />
optical dielectric constant which is set (by default) to the squared value of the solvent&rsquo;s index of refraction <br />
(see the keyword <code>soln</code>, but note 
that if the solvent is specified with the solvent keyword,<br />
 the refractive index is set by the program without needing the user to supply it.)   </p>
<p>Solvent descriptors set by the program are based on the Minnesota
Solvent Descriptor Database<sup id="vem-model-fnref_footnote_4"><a class="footnote-ref" href="#vem-model-fn_footnote_4">4</a></sup>:</p>
<p>If the option <code>do_cosmo_vem 1</code> or <code>do_cosmo_vem 2</code> is specified the
program will run VEM ground- and excited-state bulk-electrostatic
calculations using the COSMO algorithm with the SMD Coulomb radii by
default. If the user wants to use the default COSMO radii in such
calculations (this is not recommended) the option <code>do_cosmo_smd .false.</code> should be specified.</p>
<h2 id="vem-model-smssp-estimate-of-the-solutesolvent-dispersion-contribution">SMSSP estimate of the solutesolvent dispersion contribution<a class="headerlink" href="#vem-model-smssp-estimate-of-the-solutesolvent-dispersion-contribution" title="Permanent link">&para;</a></h2>
<p>If the SMSSP estimate of a solutesolvent dispersion contribution to the
solvatochromic shift is desired, the following options should be
used:</p>
<p><code>polgs_cosmo_vem (real input)</code><br />
user-provided value of the spherically-averaged molecular polarizability of the solute in the ground state (in <sup>3</sup>)</p>
<p><code>poles_cosmo_vem (real input)</code><br />
user-provided value of the spherically-averaged molecular polarizability of the solute in an exited state of interest (in <sup>3</sup>)</p>
<h2 id="vem-model-examples">Examples<a class="headerlink" href="#vem-model-examples" title="Permanent link">&para;</a></h2>
<p>An example of the VEM input file is provided
below.</p>
<pre><code>echo  
title  'VEM/TDDFT-B3LYP/6-311+G(d) vertical excitation energy + SMSSP for formaldehyde in methanol'  
start  
geometry nocenter  
 O    0.0000000000    0.0000000000    0.6743110000  
 C    0.0000000000    0.0000000000   -0.5278530000  
 H    0.0000000000    0.9370330000   -1.1136860000  
 H    0.0000000000   -0.9370330000   -1.1136860000  
symmetry c1  
end  
basis  
* library 6-311+G*  
end  
dft  
 XC b3lyp  
end  
cosmo  
 do_cosmo_smd true  
 do_cosmo_vem 1  
 solvent methanol  
 polgs_cosmo_vem 2.429  
 poles_cosmo_vem 3.208  
end  
tddft  
 nroots 10  
 target 1  
 singlet  
 notriplet  
 algorithm 1  
 civecs  
end  
grad  
  root 1  
  solve_thresh 1d-05  
end  
task tddft gradient
</code></pre>
<h2 id="vem-model-references">References<a class="headerlink" href="#vem-model-references" title="Permanent link">&para;</a></h2>
<div class="footnote">
<hr />
<ol>
<li id="vem-model-fn_footnote_1">
<p>Marenich, A. V.; Cramer, C. J.; Truhlar, D. G.; Guido, C. A.; Mennucci, B.; Scalmani, G.; Frisch, M. J. Practical Computation of Electronic Excitation in Solution: Vertical Excitation Model. <em>Chemical Science</em> <strong>2011</strong>, <em>2</em> (11), 2143. <a href="https://doi.org/10.1039/c1sc00313e">https://doi.org/10.1039/c1sc00313e</a>.&#160;<a class="footnote-backref" href="#vem-model-fnref_footnote_1" title="Jump back to footnote 1 in the text">&#8617;</a><a class="footnote-backref" href="#vem-model-fnref2_footnote_1" title="Jump back to footnote 1 in the text">&#8617;</a><a class="footnote-backref" href="#vem-model-fnref3_footnote_1" title="Jump back to footnote 1 in the text">&#8617;</a></p>
</li>
<li id="vem-model-fn_footnote_2">
<p>Li, J.; Cramer, C. J.; Truhlar, D. G. Two-Response-Time Model Based on CM2/INDO/S2 Electrostatic Potentials for the Dielectric Polarization Component of Solvatochromic Shifts on Vertical Excitation Energies. <em>International Journal of Quantum Chemistry</em> <strong>2000</strong>, <em>77</em> (1), 264280. <a href="https://doi.org/10.1002/(sici)1097-461x(2000)77:1&lt;264::aid-qua24&gt;3.0.co;2-j">https://doi.org/10.1002/(sici)1097-461x(2000)77:1&lt;264::aid-qua24&gt;3.0.co;2-j</a>.&#160;<a class="footnote-backref" href="#vem-model-fnref_footnote_2" title="Jump back to footnote 2 in the text">&#8617;</a></p>
</li>
<li id="vem-model-fn_footnote_3">
<p>Marenich, A. V.; Cramer, C. J.; Truhlar, D. G. Uniform Treatment of Solute-Solvent Dispersion in the Ground and Excited Electronic States of the Solute Based on a Solvation Model with State-Specific Polarizability. <em>Journal of Chemical Theory and Computation</em> <strong>2013</strong>, <em>9</em> (8), 36493659. <a href="https://doi.org/10.1021/ct400329u">https://doi.org/10.1021/ct400329u</a>.&#160;<a class="footnote-backref" href="#vem-model-fnref_footnote_3" title="Jump back to footnote 3 in the text">&#8617;</a></p>
</li>
<li id="vem-model-fn_footnote_4">
<p>Winget, P.; Dolney, D. M.; Giesen, D. J.; Cramer, C. J.; Truhlar, D. G. Minnesota Solvent Descriptor Database. <em>Minneapolis, MN: Department of Chemistry and Supercomputer Institute</em> <strong>1999</strong>.&#160;<a class="footnote-backref" href="#vem-model-fnref_footnote_4" title="Jump back to footnote 4 in the text">&#8617;</a></p>
</li>
</ol>
</div></section><section class="print-page" id="oniom" heading-number="13.3"><h1 id="oniom-hybrid-calculations-with-oniom">Hybrid Calculations with ONIOM<a class="headerlink" href="#oniom-hybrid-calculations-with-oniom" title="Permanent link">&para;</a></h1>
<h2 id="oniom-overview">Overview<a class="headerlink" href="#oniom-overview" title="Permanent link">&para;</a></h2>
<p>ONIOM is the hybrid method of Morokuma and co-workers that enables
different levels of theory to be applied to different parts of a
molecule/system and combined to produce a consistent energy expression.
The objective is to perform a high-level calculation on just a small
part of the system and to include the effects of the remainder at lower
levels of theory, with the end result being of similar accuracy to a
high-level calculation on the full system.</p>
<ol>
<li>M. Svensson, S. Humbel, R.D.J. Froese, T. Mastubara, S. Sieber, and
    K. Morokuma, J. Phys. Chem., 100, 19357 (1996).</li>
<li>S. Dapprich, I. Komaromi, K.S. Byun, K. Morokuma, and M.J. Frisch,
    J. Mol. Struct. (Theochem), 461-462, 1 (1999).</li>
<li>R.D.J. Froese and K. Morokuma in &ldquo;Encylopedia of Computational
    Chemistry&rdquo;, volume 2, pp.1244-1257, (ed. P. von Rague Schleyer,
    John Wiley and Sons, Chichester, Sussex, 1998).</li>
</ol>
<p>The NWChem ONIOM module implements two- and three-layer ONIOM models for
use in energy, gradient, geometry optimization, and vibrational
frequency calculations with any of the pure quantum mechanical methods
within NWChem. At the present time, it is not possible to perform ONIOM
calculations with either solvation models or classical force fields. Nor
is it yet possible to compute properties except as derivatives of the
total energy.</p>
<p>Using the terminology of Morokuma et al., the full molecular geometry
including all atoms is referred to as the &ldquo;real&rdquo; geometry and it is
treated using a &ldquo;low&rdquo;-level of theory. A subset of these atoms (referred
to as the &ldquo;model&rdquo; geometry) are treated using both the &ldquo;low&rdquo;-level and a
&ldquo;high&rdquo;-level of theory. A three-layer model also introduces an
&ldquo;intermediate&rdquo; model geometry and a &ldquo;medium&rdquo; level of theory.</p>
<p>The two-layer model requires a high and low level of theory and a real
and model molecular geometry. The energy at the high-level of theory for
the real geometry is estimated as</p>
<pre><code>     E(High,Real) = E(Low,Real) + [E(High,Model) - E(Low,Model)].
</code></pre>
<p>The three-layer model requires high, medium and low levels of theory,
and real, intermediate and model geometries and the corresponding energy
estimate is</p>
<pre><code>     E(High,Real) = E(Low,Real) + [E(High,Model) - E(Medium,Model)]
                    +  [E(Medium,Inter) - E(Low,Inter)].
</code></pre>
<p>When does ONIOM work well? The approximation for a two-layer model will
be good if</p>
<ul>
<li>the model system includes the interactions that dominate the energy
    difference being computed and the high-level of theory describes
    these to the required precision, and</li>
<li>the interactions between the model and the rest of the real system
    (substitution effects) are described to sufficient accuracy at the
    lower level of theory.</li>
</ul>
<p>ONIOM is used to compute energy differences and the absolute energies
are not all that meaningful even though they are well defined. Due to
cancellation of errors, ONIOM actually works better than you might
expect, but a poorly designed calculation can yield very bad results.
Please read and heed the caution at the end of the article by Dapprich
et al.</p>
<p>The input options are as
follows</p>
<pre><code>ONIOM  
  HIGH   &lt;string theory&gt; [basis &lt;string basis default &quot;ao basis&quot;&gt;] \  
                         [ecp &lt;string ecp&gt;] [input &lt;string input&gt;]  
 [MEDIUM &lt;string theory&gt; [basis &lt;string basis default &quot;ao basis&quot;&gt;] \  
                         [ecp &lt;string ecp&gt;] [input &lt;string input&gt;]]  
  LOW    &lt;string theory&gt; [basis &lt;string basis default &quot;ao basis&quot;&gt;] \  
                         [ecp &lt;string ecp&gt;] [input &lt;string input&gt;]  
  MODEL &lt;integer natoms&gt; [charge &lt;double charge&gt;] \  
                         [&lt;integer i1 j1&gt; &lt;real g1&gt; [&lt;string tag1&gt;] ...]  
 [INTER &lt;integer natoms&gt; [charge &lt;double charge&gt;] \  
                         [&lt;integer i1 j1&gt; &lt;real g1&gt; [&lt;string tag1&gt;] ...]]  
 [VECTORS [low-real &lt;string mofile&gt;] [low-model &lt;string mofile&gt;] \  
          [high-model &lt;string mofile&gt;] [medium-model &lt;string mofile]\  
          [medium-inter &lt;string mofile&gt;] [low-inter &lt;string mofile&gt;]]  
 [PRINT ...]  
 [NOPRINT ...]  
END
</code></pre>
<p>which are described in detail below.</p>
<p>For better validation of user input, the <code>HIGH</code>, <code>LOW</code> and <code>MODEL</code> directives
must always be specified. If the one of the <code>MEDIUM</code> or <code>INTER</code> directives
are specified, then so must the other.</p>
<h2 id="oniom-real-model-and-intermediate-geometries">Real, model and intermediate geometries<a class="headerlink" href="#oniom-real-model-and-intermediate-geometries" title="Permanent link">&para;</a></h2>
<p>The geometry and total charge of the full or real system should be
specified as normal using the <a href="#geometry">geometry</a> directive.
If <span class="arithmatex">\(N_{model}\)</span> of the atoms are to be included in the model system,
then these should be specified first in the geometry. Similarly, in a
three-layer calculation, if there are <span class="arithmatex">\(N_{inter}\)</span> atoms to be included
in the intermediate system, then these should also be arranged together
at the beginning of the geometry. The implict assumption is that the
model system is a subset of the intermediate system which is a subset of
the real system. The number of atoms to be included in the model and
intemediate systems are specified using the MODEL and INTER directives.
Optionally, the total charge of the model and intermediate systems may
be adjusted. The default is that all three systems have the same total
charge.</p>
<p>Example 1. A two-layer calculation on <span class="arithmatex">\(K^{+}(H_{2}O)\)</span> taking the
potassium ion as the model system. Note that no bonds are broken so no
link atoms are introduced. The real geometry would be specified with
potassium (the model) first.</p>
<pre><code>   geometry autosym  
     K  0    0.00    1.37  
     O  0    0.00   -1.07  
     H  0   -0.76   -1.68  
     H  0    0.76   -1.68  
   end
</code></pre>
<p>and the following directive in the ONIOM input block indicates that one
atom (implicitly the first in the geometry) is in the model system</p>
<pre><code>   model 1
</code></pre>
<h3 id="oniom-link-atoms">Link atoms<a class="headerlink" href="#oniom-link-atoms" title="Permanent link">&para;</a></h3>
<p>Link atoms for bonds spanning two regions are automatically generated
from the bond information. The additional parameters on the <code>MODEL</code> and
<code>INTER</code> directives describe the broken bonds including scale factors for
placement of the link atom and, optionally, the type of link atom. The
type of link atom defaults to hydrogen, but any type may be specified
(actually here you are specifying a geometry tag which is used to
associate a <a href="#geometry">geometrical center</a> with an atom type
and basis sets, etc. For each broken bond specify the numbers of the two
atoms (i and j), the scale factor (g) and optionally the tag of the link
atom. Link atoms are placed along the vector connecting the the first to
the second atom of the bond according to the equation</p>
<div class="arithmatex">\[\underline{R}_{link} = (1-g)\underline{R}_{1} + g*\underline{R}_{2}\]</div>
<p>where <em>g</em> is the scale factor. If the scale factor is one, then the link
atom is placed where the second atom was. More usually, the scale factor
is less than one, in which case the link atom is placed between the
original two atoms. The scale factor should be chosen so that the link
atom (usually hydrogen) is placed near its equilibrium bond length from
the model atom. E.g., when breaking a single carbon-carbon bond (typical
length 1.528 Angstrms) using a hydrogen link atom we will want a
carbon-hydrogen bond length of about 1.084 Angstrms, so the scale
factor should be chosen as 1.084/1.528 ~ 0.709.</p>
<p>Example 2. A calculation on acetaldehyde (H<sub>3</sub>C-CHO) using aldehyde
(H-CHO) as the model system. The covalent bond between the two
carbon atoms is broken and a link atom must be introduced to replace the
methyl group. The link atom is automatically generated &ndash; all you need
to do is specify the atoms in the model system that are also in the real
system (here CHO) and the broken bonds. Here is the geometry of
acetaldehyde with the CHO of aldehyde first</p>
<pre><code>   geometry  
     C    -0.383    0.288    0.021  
     H    -1.425    0.381    0.376  
     O     0.259    1.263   -0.321  

     H     0.115   -1.570    1.007  
     H    -0.465   -1.768   -0.642  
     H     1.176   -1.171   -0.352  
     C     0.152   -1.150    0.005  
   end
</code></pre>
<p>There are three atoms (the first three) of the real geometry included in
the model geometry, and we are breaking the bond between atoms 1 and 7,
replacing atom 7 with a hydrogen link atom. This is all accomplished by
the directive</p>
<pre><code>   model 3   1 7 0.709 H
</code></pre>
<p>Since the default link atom is hydrogen there is actually no need to
specify the &ldquo;H&rdquo;.</p>
<p>See also the
<a href="#oniom-a-three-layer-example">ONIOM three layer example</a>
for a more complex example.</p>
<h3 id="oniom-numbering-of-the-link-atoms">Numbering of the link atoms<a class="headerlink" href="#oniom-numbering-of-the-link-atoms" title="Permanent link">&para;</a></h3>
<p>The link atoms are appended to the atoms of the model or intermediate
systems in the order that the broken bonds are specified in the input.
This is of importance only if manually constructing an initial guess.</p>
<h2 id="oniom-high-medium-and-low-theories">High, medium and low theories<a class="headerlink" href="#oniom-high-medium-and-low-theories" title="Permanent link">&para;</a></h2>
<p>The two-layer model requires both the high-level and low-level theories
be specified. The three-layer model also requires the medium-level
theory. Each of these includes a theory (such as SCF, MP2, DFT, CCSD,
CCSD(T), etc.), an optional basis set, an optional ECP, and an optional
string containing general NWChem input.</p>
<h3 id="oniom-basis-specification">Basis specification<a class="headerlink" href="#oniom-basis-specification" title="Permanent link">&para;</a></h3>
<p>The basis name on the theory directive (high, medium, or low) is that
specified on a basis set directive (see Section 7) and not the name of a
standard basis in the library. If not specified, the basis set for the
high-level theory defaults to the standard &ldquo;ao basis&rdquo;. That for the
medium level defaults to the high-level basis, and the low-level basis
defaults to the medium-level basis. Other wavefunction parameters are
obtained from the standard wavefunction input blocks. See <a href="#oniom-effective-core-potentials">Effective
core potentials</a> for an example.</p>
<h3 id="oniom-effective-core-potentials">Effective core potentials<a class="headerlink" href="#oniom-effective-core-potentials" title="Permanent link">&para;</a></h3>
<p>If an <a href="#ecp">effective core potential</a> is specified in the
usual fashion outside of the ONIOM input then this will be used in all
calculations. If an alternative ECP name (the name specified on the ECP
directive in the same manner as done for basis sets) is specified on one
of the theory directives, then this ECP will be used in preference for
that level of theory. See the
<a href="#oniom-a-three-layer-example">ONIOM three layer example</a>
for sample input.</p>
<h3 id="oniom-general-input-strings">General input strings<a class="headerlink" href="#oniom-general-input-strings" title="Permanent link">&para;</a></h3>
<p>For many purposes, the ability to specify the theory, basis and
effective core potential is adequate. All of the options for each theory
are determined from their independent input blocks. However, if the same
theory (e.g., DFT) is to be used with different options for the ONIOM
theoretical models, then the general input strings must be used. These
strings are processed as NWChem input each time the theoretical model is
invoked. The strings may contain any NWChem input, except for options
pertaining to ONIOM and the <code>task</code> directive. The intent that the strings
be used just to control the options pertaining to the theory being used.</p>
<p>A word of caution. Be sure to check that the options are producing the
desired results. Since the NWChem database is persistent and the ONIOM
calculations happen in an undefined order, the input strings should
fully define the calculation you wish to have happen.</p>
<p>For instance, if the high model is DFT/B3LYP/6-311g** and the low
model is DFT/LDA/3-21g, the ONIOM input might look like this</p>
<pre><code>   oniom  
     model 3  
     low  dft basis 3-21g    input &quot;dft; xc; end&quot;  
     high dft basis 6-311g** input &quot;dft; xc b3lyp; end&quot;  
   end
</code></pre>
<p>The empty XC directive restores the default LDA exchange-correlation
option (see Section 11.3). Note that semi-colons and other quotation
marks inside the input string must be preceded by a backslash to avoid
special interpretation.</p>
<p>See <a href="#oniom-dft-with-and-without-charge-fitting">|DFT with and without charge
fitting</a> for another
example.</p>
<h2 id="oniom-use-of-symmetry">Use of symmetry<a class="headerlink" href="#oniom-use-of-symmetry" title="Permanent link">&para;</a></h2>
<p>Symmetry should work just fine as long as the model and intermediate
regions respect the symmetry &ndash; i.e., symmetry equivalent atoms need to
be treated equivalently. If symmetry equivalent atoms must be treated in
separate regions then the symmetry must be lowered (or completely
switched off).</p>
<h2 id="oniom-molecular-orbital-files">Molecular orbital files<a class="headerlink" href="#oniom-molecular-orbital-files" title="Permanent link">&para;</a></h2>
<p>The VECTORS directive in the ONIOM block is different to that elsewhere
in NWChem. For each of the necessary combinations of theory and geometry
you can specify a different file for the molecular orbitals. By default
each combination will store the MO vectors in the permanent directory
using a file name created by appending to the name of the calculation
the following string</p>
<ul>
<li>low-real &ndash; &ldquo;.lrmos&rdquo;</li>
<li>low-inter &ndash; &ldquo;.limos&rdquo;</li>
<li>low-model &ndash; &ldquo;.lmmos&rdquo;</li>
<li>medium-inter &ndash; &ldquo;.mimos&rdquo;</li>
<li>medium-model &ndash; &ldquo;.mmmos&rdquo;</li>
<li>high-model &ndash; &ldquo;.hmmos&rdquo;</li>
</ul>
<p>Each calculation will utilize the appropriate vectors which is more
efficient during geometry optimizations and frequency calculations, and
is also useful for the initial calculation. In the absence of existing
MO vectors files, the default atomic guess is used (see
<a href="#hartree-fock-theory-for-molecules-vectors-inputoutput-of-mo-vectors">|Input/output of MO vectors</a>).</p>
<p>If special measures must be taken to converge the initial SCF, DFT or
MCSCF calculation for one or more of the systems, then initial vectors
may be saved in a file with the default name, or another name may be
specified using the VECTORS directive. Note that subsequent vectors
(e.g., from a geometry optimization) will be written back to this file,
so take a copy if you wish to preserve it. To generate the initial guess
for the model or intermediate systems it is necessary to generate the
geometries which is most readily done, if there are link atoms, by just
running NWChem on the input for the ONIOM calculation on your
workstation. It will print these geometries before starting any
calculations which you can then terminate.</p>
<p>E.g., in a calculation on Fe(III) surrounded by some ligands, it is hard
to converge the full (real) system from the atomic guess so as to obtain
a <span class="arithmatex">\(d^5\)</span> configuration for the iron atom since the <em>d</em> orbitals are
often nominally lower in energy than some of the ligand orbitals. The
most effective mechanism is to converge the isolated Fe(III) and then to
use the <a href="#hartree-fock-theory-for-molecules-vectors-fragment-superposition-of-fragment-molecular-orbitals">fragment
guess</a>
as a starting guess for the real system. The resulting converged
molecular orbitals can be saved either with the default name (as
described above in this section), in which case no additional input is
necessary. If an alternative name is desired, then the <code>VECTORS</code> directive
may be used as follows</p>
<pre><code>   vectors low-real /u/rjh/jobs/fe_ether_water.mos
</code></pre>
<h2 id="oniom-restarting">Restarting<a class="headerlink" href="#oniom-restarting" title="Permanent link">&para;</a></h2>
<p>Restart of ONIOM calculations does not currently work as smoothly as we
would like. For geometry optimizations that terminated gracefully by
running out of iterations, the restart will work as normal. Otherwise,
specify in the input of the restart job the last geometry of the
optimization. The Hessian information will be reused and the calculation
should proceed losing at most the cost of one ONIOM gradient evaluation.
For energy or frequency calculations, restart may not currently be
possible.</p>
<h2 id="oniom-examples">Examples<a class="headerlink" href="#oniom-examples" title="Permanent link">&para;</a></h2>
<h3 id="oniom-hydrocarbon-bond-energy">Hydrocarbon bond energy<a class="headerlink" href="#oniom-hydrocarbon-bond-energy" title="Permanent link">&para;</a></h3>
<p>A simple two-layer model changing just the wavefunction with one link
atom.</p>
<p>This reproduces the two-layer ONIOM (MP2:HF) result from Dapprich et al.
for the reaction <span class="arithmatex">\(R-CH_{3} = R-CH_{2} + H\)</span> with <span class="arithmatex">\(R=CH_3\)</span> using
<span class="arithmatex">\(CH_4\)</span> as the model. The geometries of <span class="arithmatex">\(R-CH_3\)</span> and <span class="arithmatex">\(R-CH_2\)</span> are
optimized at the DFT-B3LYP/6-311++G** level of theory, and then ONIOM
is used to compute the binding energy using UMP2 for the model system
and HF for the real system. The results, including MP2 calculations on
the full system for comparison, are as given in the table
below.  </p>
<table>
<thead>
<tr>
<th>Theory</th>
<th>Me-CH2</th>
<th>Me-Me</th>
<th>H</th>
<th>De(Hartree)</th>
<th>De(kcal/mol)</th>
</tr>
</thead>
<tbody>
<tr>
<td>B3LYP</td>
<td>-79.185062</td>
<td>-79.856575</td>
<td>-0.502256</td>
<td>0.169257</td>
<td>106.2</td>
</tr>
<tr>
<td>HF</td>
<td>-78.620141</td>
<td>-79.251701</td>
<td>-0.499817</td>
<td>0.131741</td>
<td>82.7</td>
</tr>
<tr>
<td>MP2</td>
<td>-78.904716</td>
<td>-79.571654</td>
<td>-0.499817</td>
<td>0.167120</td>
<td>104.9</td>
</tr>
<tr>
<td>MP2:HF</td>
<td>-78.755223</td>
<td>-79.422559</td>
<td>-0.499817</td>
<td>0.167518</td>
<td>105.1</td>
</tr>
</tbody>
</table>
<p>Energies for ONIOM example 1, hydrocarbon bond energy using MP2:HF
two-layer model.  </p>
<p>The following input first performs a calculation on <span class="arithmatex">\(CH_{3}-CH_2\)</span>, and
then on <span class="arithmatex">\(CH_{3}-CH_3\)</span>. Note that in the second calculation we cannot
use the full symmetry since we are breaking the C-C bond in forming the
model system (the non-equivalence of the methyl groups is perhaps more
apparent if we write <span class="arithmatex">\(R-CH_3\)</span>).</p>
<pre><code>   start  

   basis spherical  
     H library 6-311++G**; C library 6-311++G**  
   end  

   title &quot;ONIOM Me-CH2&quot;  

   geometry autosym  
     H    -0.23429328     1.32498565     0.92634814  
     H    -0.23429328     1.32498565    -0.92634814  
     C    -0.13064265     0.77330370     0.00000000  
     H    -1.01618703    -1.19260361     0.00000000  
     H     0.49856072    -1.08196901    -0.88665533  
     H     0.49856072    -1.08196901     0.88665533  
     C    -0.02434414    -0.71063687     0.00000000  
   end  

   scf; uhf; doublet; thresh 1e-6; end  
   mp2; freeze atomic; end  

   oniom  
     high mp2  
     low  scf  
     model 3   3 7 0.724  
   end  

   task oniom  

   title &quot;ONIOM Me-Me&quot;  

   geometry   # Note cannot use full D3D symmetry here, either specify noautosym, or change an atom tag (here C -&gt; C1)  
     H   -0.72023641     0.72023641    -1.16373235  
     H    0.98386124     0.26362482    -1.16373235  
     H   -0.26362482    -0.98386124    -1.16373235  
     C    0.00000000     0.00000000    -0.76537515  
     H    0.72023641    -0.72023641     1.16373235  
     H   -0.98386124    -0.26362482     1.16373235  
     H    0.26362482     0.98386124     1.16373235  
     C1   0.00000000     0.00000000     0.76537515  
   end  

   scf; rhf; singlet; end  

   oniom  
     high mp2  
     low  scf  
     model 4   4 8 0.724  
   end  

   task oniom
</code></pre>
<h3 id="oniom-optimization-and-frequencies">Optimization and frequencies<a class="headerlink" href="#oniom-optimization-and-frequencies" title="Permanent link">&para;</a></h3>
<p>A two-layer model including modification of theory, basis, ECP and total
charge and no link atoms.</p>
<p>This input reproduces the ONIOM optimization and vibrational frequency
calculation of Rh(CO)<sub>2</sub>Cp of Dapprich et al. The model system is
Rh(CO)<sub>2</sub><sup>+</sup>. The low theory is the Gaussian LANL2MB model
(Hay-Wadt n+1 ECP with minimal basis on Rh, STO-3G on others) with SCF.
The high theory is the Gaussian LANL2DZ model (another Hay-Wadt ECP with
a DZ basis set on Rh, Dunning split valence on the other atoms) with
DFT/B3LYP. Note that different names should be used for the basis set
and ECP since the same mechanism is used to store them in the
database.</p>
<pre><code>   start  

   ecp LANL2DZ_ECP  
     rh library LANL2DZ_ECP  
   end  

   basis LANL2DZ spherical  
     rh library LANL2DZ_ECP  
     o library SV_(Dunning-Hay); c library SV_(Dunning-Hay); h library SV_(Dunning-Hay)  
   end  

   ecp Hay-Wadt_MB_(n+1)_ECP  
     rh library Hay-Wadt_MB_(n+1)_ECP  
   end  

   # This is the minimal basis used by Gaussian.  It is not the same   
   # as the one in the EMSL basis set library for this ECP.  
   basis Hay-Wadt_MB_(n+1) spherical   
     Rh s; .264600D+01 -.135541D+01; .175100D+01  .161122D+01; .571300D+00  .589381D+00  
     Rh s; .264600D+01  .456934D+00; .175100D+01 -.595199D+00; .571300D+00 -.342127D+00  
           .143800D+00  .410138D+00; .428000D-01  .780486D+00  
     Rh p; .544000D+01 -.987699D-01; .132900D+01  .743359D+00; .484500D+00  .366846D+00  
     Rh p; .659500D+00 -.370046D-01; .869000D-01  .452364D+00; .257000D-01  .653822D+00  
     Rh d; .366900D+01  .670480D-01; .142300D+01  .455084D+00; .509100D+00  .479584D+00  
           .161000D+00  .233826D+00  
     o  library sto-3g; c  library sto-3g; h  library sto-3g  
   end  

   charge 0  
   geometry autosym  
     rh       0.00445705    -0.15119674     0.00000000  
     c       -0.01380554    -1.45254070     1.35171818  
     c       -0.01380554    -1.45254070    -1.35171818  
     o       -0.01805883    -2.26420212     2.20818932  
     o       -0.01805883    -2.26420212    -2.20818932  
     c        1.23209566     1.89314720     0.00000000  
     c        0.37739392     1.84262319    -1.15286640  
     c       -1.01479160     1.93086461    -0.70666350  
     c       -1.01479160     1.93086461     0.70666350  
     c        0.37739392     1.84262319     1.15286640  
     h        2.31251453     1.89903673     0.00000000  
     h        0.70378132     1.86131979    -2.18414218  
     h       -1.88154273     1.96919306    -1.35203550  
     h       -1.88154273     1.96919306     1.35203550  
     h        0.70378132     1.86131979     2.18414218  
   end  

   dft; grid fine; convergence gradient 1e-6 density 1e-6; xc b3lyp; end  
   scf; thresh 1e-6; end  

   oniom  
     low scf basis Hay-Wadt_MB_(n+1) ecp  Hay-Wadt_MB_(n+1)_ECP  
     high dft basis LANL2DZ ecp LANL2DZ_ECP  
     model 5 charge 1  
     print low  
   end  

   task oniom optimize  
   task oniom freq
</code></pre>
<h3 id="oniom-a-three-layer-example">A three-layer example<a class="headerlink" href="#oniom-a-three-layer-example" title="Permanent link">&para;</a></h3>
<p>A three layer example combining CCSD(T), and MP2 with two different
quality basis sets, and using multiple link atoms.</p>
<p>The full system is tetra-dimethyl-amino-ethylene (TAME) or
(N(Me)2)2-C=C-(N(Me)2)2. The intermediate system is (NH2)2-C=C-(NH2)2
and H2C=CH2 is the model system. CCSD(T)+aug-cc-pvtz is used for the
model region, MP2+aug-cc-pvtz for the intermediate region, and
MP2+aug-cc-pvdz for everything.</p>
<p>In the real geometry the first two atoms (C, C) are the model system
(link atoms will be added automatically). The first six atoms (C, C, N,
N, N, N) describe the intermediate system (again with link atoms to be
added automatically). The atoms have been numbered using comments to
make the bonding input easier to generate.</p>
<p>To make the model system, four C-N bonds are broken between the ethylene
fragment and the dimethyl-amino groups and replaced with C-H bonds. To
make the intermediate system, eight C-N bonds are broken between the
nitrogens and the methyl groups and replaced with N-H bonds. The scaling
factor could be chosen differently for each of the bonds.</p>
<pre><code>   start   

   geometry   
     C  0.40337795 -0.17516305 -0.51505208  # 1  
     C -0.40328664  0.17555927  0.51466084  # 2  
     N  1.87154979 -0.17516305 -0.51505208  # 3  
     N -0.18694782 -0.60488524 -1.79258692  # 4  
     N  0.18692927  0.60488318  1.79247594  # 5  
     N -1.87148219  0.17564718  0.51496494  # 6  
     C  2.46636552  1.18039452 -0.51505208  # 7  
     C  2.48067731 -1.10425355  0.46161675  # 8  
     C -2.46642715 -1.17982091  0.51473105  # 9  
     C -2.48054940  1.10495864 -0.46156202  # 10  
     C  0.30027136  0.14582197 -2.97072148  # 11  
     C -0.14245927 -2.07576980 -1.96730852  # 12  
     C -0.29948109 -0.14689874  2.97021079  # 13  
     C  0.14140463  2.07558249  1.96815181  # 14  
     H  0.78955302  2.52533887  1.19760764  
     H -0.86543435  2.50958894  1.88075113  
     ... and 22 other hydrogen atoms on the methyl groups  
   end  

   basis aug-cc-pvtz spherical  
     C library aug-cc-pvtz; H library aug-cc-pvtz  
   end  

   basis aug-cc-pvdz spherical  
     C library aug-cc-pvtz; H library aug-cc-pvtz  
   end  

   oniom        
     high ccsd(t) basis aug-cc-pvtz  
     medium mp2 basis aug-cc-pvtz  
     low mp2 basis aug-cc-pvdz  
     model 2   1 3  0.87   1 4  0.87   2 5  0.87   2 6  0.87  

     inter 6   3 7  0.69   3 8  0.69   4 11 0.69   4 12 0.69 \  
               5 13 0.69   5 14 0.69   6 9  0.69   6 10 0.69  
   end  

   task oniom
</code></pre>
<h3 id="oniom-dft-with-and-without-charge-fitting">DFT with and without charge fitting<a class="headerlink" href="#oniom-dft-with-and-without-charge-fitting" title="Permanent link">&para;</a></h3>
<p>Demonstrates use of general input strings.</p>
<p>A two-layer model for anthracene (a linear chain of three fused benzene
rings) using benzene as the model system. The high-level theory is
DFT/B3LYP/TZVP with exact Coulomb. The low level is DFT/LDA/DZVP2 with
charge fitting.</p>
<p>Note the following.</p>
<ol>
<li>The semi-colons and quotation marks inside the input string must be
    quoted with backslash.</li>
<li>The low level of theory sets the fitting basis set and the high
    level of theory unsets it.  </li>
</ol>
<pre><code>   start  
   geometry   
     symmetry d2h  
     C    0.71237329    -1.21458940     0.0  
     C   -0.71237329    -1.21458940     0.0  
     C    0.71237329     1.21458940     0.0  
     C   -0.71237329     1.21458940     0.0  
     C   -1.39414269     0.00000000     0.0  
     C    1.39414269     0.00000000     0.0  
     H   -2.47680865     0.00000000     0.0  
     H    2.47680865     0.00000000     0.0  
     C    1.40340535    -2.48997027     0.0  
     C   -1.40340535    -2.48997027     0.0  
     C    1.40340535     2.48997027     0.0  
     C   -1.40340535     2.48997027     0.0  
     C    0.72211503     3.64518615     0.0  
     C   -0.72211503     3.64518615     0.0  
     C    0.72211503    -3.64518615     0.0  
     C   -0.72211503    -3.64518615     0.0  
     H    2.48612947     2.48094825     0.0  
     H    1.24157357     4.59507342     0.0  
     H   -1.24157357     4.59507342     0.0  
     H   -2.48612947     2.48094825     0.0  
     H    2.48612947    -2.48094825     0.0  
     H    1.24157357    -4.59507342     0.0  
     H   -1.24157357    -4.59507342     0.0  
     H   -2.48612947    -2.48094825     0.0  
   end  

   basis small  
     h library DZVP_(DFT_Orbital)  
     c library DZVP_(DFT_Orbital)  
   end  

   basis fitting  
     h library DGauss_A1_DFT_Coulomb_Fitting  
     c library DGauss_A1_DFT_Coulomb_Fitting  
   end  

   basis big  
     h library TZVP_(DFT_Orbital)  
     c library TZVP_(DFT_Orbital)  
   end  

   oniom  
     model 8   1 9 0.75   2 10 0.75   3 11 0.75   4 12 0.75  
     high dft basis big   input &quot;unset &quot;cd basis&quot;; dft; xc b3lyp; end&quot;  
     low  dft basis small input &quot;set &quot;cd basis&quot; fitting; dft; xc; end&quot;  
   end  

   task oniom
</code></pre></section>
                    <section class='print-page md-section' id='section-13-4' heading-number='13.4'>
                        <h1>QMMM<a class='headerlink' href='#section-13-4' title='Permanent link'></a>
                        </h1>
                    <section class="print-page" id="qmmm_introduction" heading-number="13.4.1"><h1 id="qmmm_introduction-qmmm-introduction">QMMM Introduction<a class="headerlink" href="#qmmm_introduction-qmmm-introduction" title="Permanent link">&para;</a></h1>
<p>The combined quantum mechanical molecular mechanics (QM/MM) approach
provides a simple and effective tool to study localized molecular
transformations in large scale systems such as those encountered in
solution chemistry or enzyme catalysis. In this method an accurate but
computationally intensive quantum mechanical (QM) description is only
used for the regions where electronic structure transformations are
occurring (e.g. bond making and breaking). The rest of the system, whose
chemical identity remains essentially the same, is treated at the
approximate classical molecular mechanics (MM)
level.</p>
<p>The QM/MM module in NWChem is built as a top level interface between the classical MD module and various QM modules,managing initialization, data transfer, and various high level operations. The size of the system (10^3 - 10^5 atoms)
and the need for classical force field parameters precludes description of the system through just the geometry input
block as would be done in pure QM simulations.<br />
Instead a separate <a href="#qmmm_restart_and_topology_files">preparation
stage</a> is required. 
In a typical setting this <a href="#qmmm_restart_and_topology_files">preparation
run</a> will be done separately from the main QM/MM simulations resulting in the generation of topology and restart files. The topology file contains a list of all relevant force field interactions encountered in the system but has no information about the actual atom positions. Typically the topology file will be generated once and reused throughout the entire simulation. The actual structural information about the system is contained in the restart file, which will be changing as the system coordinates are updated during the course of the simulation.</p>
<p>Once restart and topology files are generated, the QM/MM simulation can
be initiated by defining the specifics of the
<a href="#qm_parameters">QM</a> and <a href="#mm_parameters">MM</a>
descriptions, and if necessary <a href="#qmmm_parameters">QM/MM</a>
interface parameters.</p>
<p>The actual QM/MM calculation is invoked with the following task
directive.</p>
<pre><code> task qmmm &lt;string qmtheory&gt; &lt;string operation&gt; [numerical] [ignore]
</code></pre>
<p>where qmtheory specifies quantum method for the calculation of the
quantum region. It is expected that most of QM/MM simulations will be
performed with with HF, DFT. or <a href="#tce">CC</a> theories, but any
other QM theory supported by NWChem should also work. NWChem supports
wide range of QM/MM tasks including</p>
<ul>
<li><a href="#qmmm_sp_energy">single point energy/gradient
    calculations</a></li>
<li><a href="#qmmm_sp_property">properties</a></li>
<li><a href="#qmmm_esp">ESP charge analysis</a></li>
<li><a href="#qmmm_optimization">optimization and transition states</a></li>
<li><a href="#qmmm_freq">hessians and frequency</a></li>
<li><a href="#qmmm_neb_calculations">reaction pathway calculations</a></li>
<li><a href="#qmmm_dynamics">dynamics</a></li>
<li><a href="#qmmm_free_energy">free energy calculations</a></li>
</ul></section><section class="print-page" id="qmmm" heading-number="13.4.2"><h1 id="qmmm-qmmm-simulations">QMMM Simulations<a class="headerlink" href="#qmmm-qmmm-simulations" title="Permanent link">&para;</a></h1>
<ul>
<li><a href="#qmmm_introduction">Introduction</a></li>
<li><a href="#qmmm_restart_and_topology_files">Topology and Restart
    Files</a><ul>
<li><a href="#qmmm_preparation_prerequisites">Prerequisites</a></li>
<li><a href="#qmmm_preparation_basic">QM region definition</a></li>
<li><a href="#qmmm_preparation_solvation">Solvation</a></li>
<li><a href="#qmmm_preparation_constraints">Permanent Constraints</a></li>
</ul>
</li>
<li><a href="#qmmm_input_file">Input File</a><ul>
<li><a href="#qm_parameters">QM Parameters</a></li>
<li><a href="#mm_parameters">MM Parameters</a></li>
<li><a href="#qmmm_parameters">QM/MM Parameters</a></li>
</ul>
</li>
<li>Single Point Calculations<ul>
<li><a href="#qmmm_sp_energy">Ground State Energy and Gradient</a></li>
<li><a href="#qmmm_excited_states">Excited State Energy</a></li>
<li><a href="#qmmm_sp_property">Properties</a></li>
<li><a href="#qmmm_esp">ESP Charge Analysis</a></li>
</ul>
</li>
<li>Potential Energy Surface Analysis<ul>
<li><a href="#qmmm_optimization">Optimization</a></li>
<li><a href="#qmmm_transition_states"> Transition States</a></li>
<li><a href="#qmmm_freq">Hessians and Frequency</a></li>
<li><a href="#qmmm_neb_calculations">Reaction Pathway Calculations with
    NEB</a></li>
</ul>
</li>
<li><a href="#qmmm_dynamics">Dynamics</a></li>
<li><a href="#qmmm_free_energy">Free Energy Calculations</a></li>
<li><a href="#qmmm_appendix">Appendix</a><ul>
<li><a href="#qmmm_appendix-format-of-nwchem-parameter-file">Format of NWChem parameter
    file</a></li>
<li><a href="#qmmm_appendix-conversion-of-standard-amber-program-parameter-files">Conversion from AMBER program parameter files to
    NWChem</a></li>
</ul>
</li>
<li><a href="#qmmm_references">References</a></li>
</ul></section>
                    <section class='print-page md-section' id='section-13-4-3' heading-number='13.4.3'>
                        <h1>QMMM Restart and Topology Files<a class='headerlink' href='#section-13-4-3' title='Permanent link'></a>
                        </h1>
                    <section class="print-page" id="qmmm_restart_and_topology_files" heading-number="13.4.3.1"><p>The structure of the system in QM/MM calculations is provided by the
restart file (extension <code>.rst</code>) and not by the geometry block as it would
be for pure QM calculations. The parameters of classical interaction are
given by the topology file (extension <code>.top</code>). These two files are
<strong>REQUIRED</strong> for QM/MM calculations and can be generated by the <a href="#prepare">prepare
module</a>. In a typical setting this &ldquo;preparation
stage&rdquo; is performed separately from the main QM/MM simulation.</p>
<ul>
<li><a href="#qmmm_preparation_prerequisites">Prerequisites</a>  </li>
<li><a href="#qmmm_preparation_basic">QM region definition</a>  </li>
<li><a href="#qmmm_preparation_solvation">Solvation</a>  </li>
<li><a href="#qmmm_preparation_constraints">Permanent Constraints</a>  </li>
</ul></section><section class="print-page" id="qmmm_preparation_prerequisites" heading-number="13.4.3.2"><p>The necessary prerequisites for
the preparation of topology and restart files for QM/MM simulations are:
* Properly formatted <a href="http://www.wwpdb.org/docs.html">PDB</a>  file for the system.
* Fragment Files (extension .frg)
* Parameter Files (extension .par)</p>
<p>A number of fragment files as well as standard Amber type force parameter files are provided with the NWChem distribution. However, user should be prepared
to generate additional fragment and <a href="Parameter-file.pdf">parameter files</a> for nonstandard cases</p></section><section class="print-page" id="qmmm_preparation_basic" heading-number="13.4.3.3"><p>One of major required pieces of information that has to be provided in
the prepare block for QM/MM simulations is the definition of the QM
region. This can be accomplished using <em>modify</em> directive used either
per atom</p>
<pre><code>modify atom &lt;integer isgm&gt;:&lt;string atomname&gt; quantum
</code></pre>
<p>or per segment/residue basis</p>
<pre><code>modify segment &lt;integer isgm&gt; quantum
</code></pre>
<p>Here <em>isgm</em> and <em>atomname</em> refer to the residue number and atom name
record as given in the PDB file. It is important to note that that the
leading blanks in atom name record should be indicated with underscores.
Per PDB format guidelines the atom name record starts at column 13. If,
for example, the atom name record &ldquo;OW&rdquo; starts in the 14th column in PDB
file, it will appear as &ldquo;_OW&rdquo; in the modify atom directive in the
prepare block.</p>
<p>In the current implementation only solute atoms can be declared as
quantum. If part of the solvent has to be treated quantum mechanically
then it has to redeclared to be solute. The definition of QM region
should be accompanied by <em>update lists</em> and <em>ignore</em> directives.
<span id="example"></span> Here is an example input file that will
generate QM/MM restart and topology files for the ethanol molecule:</p>
<pre><code>title &quot;Prepare QM/MM calculation of ethanol&quot;  
start etl  

[prepare](Prepare)  
#--*name of the pdbfile*  
   source [etl0.pdb](etl0.pdb)                      
#--*generate new topology and sequence file*  
   new_top new_seq                      
#--*generate new restart file*  
   new_rst                              
#--*define quantum region (note the use of underscore)*  
   modify atom 1:_C1  quantum           
   modify atom 1:2H1  quantum           
   modify atom 1:3H1  quantum            
   modify atom 1:4H1  quantum           
   update lists  
   ignore  
#--*save restart file*     
   write etl_ref.rst  
#--*generate pdb file*  
   write [etl_ref.pdb](etl_ref.pdb)  
end   
task prepare
</code></pre>
<p>Running the input shown above will produce (among other things) the
topology file (etl.top), the restart file (etl_ref.rst), and the pdb
file <a href="etl_ref.pdb">etl_ref.pdb</a>. The prefix for the topology file follows after the
rtdb name specified in the start directive in the input (i.e. &ldquo;start
etl&rdquo;), while the names for the restart and pdb files were specified
explicitly in the input file. In the absence of the explicit write
statement for the restart file, it would be generated under the name
&ldquo;etl_md.rst&rdquo;. The pdb file would only be written in the presence of the
explicit write statement.</p>
<p><span id="tip"></span><strong>Tip</strong>: <em>It is strongly recommended to check the
correctness of the generated pdb file versus the original &ldquo;source&rdquo; pdb
file to catch possible errors in the formatting of the pdb and fragment
files.</em></p></section><section class="print-page" id="qmmm_preparation_solvation" heading-number="13.4.3.4"><p>During the preparation stage of QM/MM calculations the system may also
be solvated using <code>solvate</code> directive of the prepare module. It is
recommended that solvation is performed in conjunction with <code>center</code> and
<code>orient</code> directives.</p>
<p>Here is an example where the ethanol molecule is declared quantum and solvated in a box of spce waters:</p>
<pre><code>title &quot;Prepare QM/MM calculation of solvated ethanol&quot;
start etl
prepare
source etl0.pdb
new_top new_seq
new_rst
#center and orient prior to solvation
center
orient
#solvation in 1 nm by 2 nm by 3 nm box
solvate box 1.0 2.0 3.0
#the whole ethanol is declared quantum now
modify segment 1 quantum
update lists
ignore
write etl_ref.rst
write etl_ref.pdb
end
task prepare
</code></pre></section><section class="print-page" id="qmmm_preparation_constraints" heading-number="13.4.3.5"><p>Fixing atoms outside a certain distance from the QM region can also be
accomplished using prepare module. These constraints will then be
permanently embedded in the resulting restart file, which may be
advantageous for certain types of QM/MM simulations. The actual format
for the constraint directive to fix whole residues
is</p>
<pre><code>fix segments beyond  &lt;real radius&gt;  &lt;integer residue number&gt;:&lt;string atom name&gt;
</code></pre>
<p>or to fix on atom
basis</p>
<pre><code>fix atoms beyond  &lt;real radius&gt;  &lt;integer residue number&gt;:&lt;string atom name&gt;
</code></pre>
<p>This example illustrates the use of permanent fix directives during preparation stage  </p>
<pre><code>start etl
prepare
source etl0.pdb
new_top new_seq
new_rst
center
orient
#solvation in 40 A cubic box
solvate cube 4.0
modify segment 1 quantum
#fix residues more than 20 A away from ethanol oxygen atom
fix segments beyond 2.0 1:_O
update lists
ignore
write etl_ref.rst
write etl_ref.pdb
end
task prepare
</code></pre></section></section>
                    <section class='print-page md-section' id='section-13-4-4' heading-number='13.4.4'>
                        <h1>QMMM Input File<a class='headerlink' href='#section-13-4-4' title='Permanent link'></a>
                        </h1>
                    <section class="print-page" id="qmmm_input_file" heading-number="13.4.4.1"><h1 id="qmmm_input_file-qmmm-input-file">QM/MM Input file<a class="headerlink" href="#qmmm_input_file-qmmm-input-file" title="Permanent link">&para;</a></h1>
<p>The input file for QM/MM
calculations contains definition of molecular mechanics parameters,
quantum mechanical parameters, and QM/MM interface parameters.</p>
<ul>
<li><a href="#qmmm_qm_parameters">QM/MM QM Parameters</a>  </li>
<li><a href="#qmmm_mm_parameters">QM/MM MM Parameters</a>  </li>
<li><a href="#qmmm_qmmm_parameters">QM/MM QM/MM Parameters</a>  </li>
</ul></section><section class="print-page" id="qmmm_qm_parameters" heading-number="13.4.4.2"><h1 id="qmmm_qm_parameters-qm_parameters">QM_Parameters<a class="headerlink" href="#qmmm_qm_parameters-qm_parameters" title="Permanent link">&para;</a></h1>
<p>The parameters defining calculation of the QM region (including basis sets) must
   be present in the traditional NWChem input format except for the geometry block.</p>
<p>The geometrical information will be constructed automatically using information
   contained in the restart file.</p></section><section class="print-page" id="qmmm_mm_parameters" heading-number="13.4.4.3"><p>The molecular mechanics parameters are given in the form of standard MD
input block as used by the <a href="#md">MD module</a>. At
the basic level the molecular mechanics input block specifies the
restart and topology file that were generated during QM/MM <a href="#qmmm_restart_and_topology_files">preparation
stage</a>. It also
contains information relevant to the calculation of the classical region
(e.g. cutoff distances, constraints, optimization and dynamics
parameters, etc) in the system. In this input block one can also set
fixed atom constraints on classical atoms. Continuing with our <a href="#qmmm_preparation_basic-example">prepare
example for ethanol
molecule</a> here is
a simple input block that may be used for this system.</p>
<pre><code>md  
#thisspecifiesthatetl_md.rstwillbeusedasarestartfile  
#andetl.topwillbeatopologyfile  
systemetl_md  
#ifweeverwantedtofixC1atom  
fixsolute1_C1  
noshakesolute  
end
</code></pre>
<p>The <em>noshake solute</em>, shown in the above example is a recommended
directive for QM/MM simulations that involve
<a href="#qmmm_optimization">optimizations</a>. Otherwise
user has to ensure that the <a href="#qmmm_method">optimization
method</a> for classical solute atoms
is a steepest descent</p></section><section class="print-page" id="qmmm_qmmm_parameters" heading-number="13.4.4.4"><h1 id="qmmm_qmmm_parameters-qmmm-parameters">QM/MM parameters<a class="headerlink" href="#qmmm_qmmm_parameters-qmmm-parameters" title="Permanent link">&para;</a></h1>
<p>The QM/MM interface parameters define the interaction between classical
and quantum regions. </p>
<pre><code> qmmm  
 [ [eref] &lt;double precision default 0.0d0&gt;]  
 [ [bqzone] &lt;double precision default 9.0d0&gt;]  
 [ [mm_charges] [exclude &lt;(none||all||linkbond||linkbond_H) default none&gt;]  
          [ expand  &lt;none||all||solute||solvent&gt; default none]  
          [ update  &lt;integer default 0&gt;]   
 [ [link_atoms] &lt;(hydrogen||halogen) default hydrogen&gt;]  
 [ [link_ecp]  &lt;(auto||user) default auto&gt;]  
 [ [region]   &lt; [region1]  [region2]  [region3] &gt; ]  
 [ [method]   [method1]  [method2]  [method3]  ]  
 [ [maxiter]  [maxiter1] [maxiter2] [maxiter3] ]  
 [ [ncycles]  &lt; [number] default 1 &gt; ]  
 [ [density]  [espfit] [static] [dynamical] ]  
 [ [xyz]  ]  
 [ [convergence] &lt;double precision default 1.0d-4&gt;] ]  
 [ [load] ]  
 [ [nsamples] ]  
 end
</code></pre>
<p>Detailed explanation of the subdirectives in the QM/MM input block is given below:  </p>
<h3 id="qmmm_qmmm_parameters-qmmm-eref">QM/MM eref<a class="headerlink" href="#qmmm_qmmm_parameters-qmmm-eref" title="Permanent link">&para;</a></h3>
<pre><code>eref&lt;doubleprecisiondefault0.0d0&gt;
</code></pre>
<p>This directive sets the relative zero of energy for the QM component of
the system. The need for this directive arises from different
definitions of zero energy for QM and MM methods. In QM methods the zero
of energy for the system is typically vacuum. The zero of energy for the
MM system is by definition of most parameterized force fields the
separated atom energy. Therefore in many cases the energetics of the QM
system will likely overshadow the MM component of the system. This
imbalance can be corrected by suitably chosen value of <code>eref</code>. In most
cases <em>IT IS OK</em> to leave <code>eref</code> at its default value of zero.</p>
<h3 id="qmmm_qmmm_parameters-qmmm-bqzone">QM/MM bqzone<a class="headerlink" href="#qmmm_qmmm_parameters-qmmm-bqzone" title="Permanent link">&para;</a></h3>
<pre><code>bqzone &lt;double precision default 9.0d0&gt;
</code></pre>
<p>This directive defines the radius of the zone (in angstroms) around the
quantum region where classical residues/segments will be allowed to
interact with quantum region both electrostatically and through Van der
Waals interactions. It should be noted that classical atoms interacting
with quantum region via bonded interactions are always included (this is
true even if bqzone is set to 0). In addition, even if one atom of a
given charged group is in the bqzone (residues are typically treated as
one charged group) then the whole group will be included.</p>
<h3 id="qmmm_qmmm_parameters-qmmm-mm_charges">QM/MM mm_charges<a class="headerlink" href="#qmmm_qmmm_parameters-qmmm-mm_charges" title="Permanent link">&para;</a></h3>
<pre><code>mm_charges [exclude &lt;(none||all||linkbond||linkbond_H) default none&gt;]
           [expand  &lt;none||all||solute||solvent&gt; default none]
           [update  &lt;integer default 0&gt;]
</code></pre>
<p>This directive controls treatment of classical point (MM) charges that
are interacting with QM region. For most QM/MM applications the use of
directive will be not be necessary. Its absence would be simply mean
that all MM charges within the cuttof distance ( as specified by cutoff
) as well those belonging to the charges groups directly bonded to QM
region will be allowed to interact with QM region.</p>
<p>Keyword <code>exclude</code> specifies the subset MM charges that will be
specifically excluded from interacting with QM region.</p>
<ul>
<li><code>none</code> default value reverts to the original set of MM charges as
    described above.</li>
<li><code>all</code> excludes all MM charges from interacting with QM region (&ldquo;gas
    phase&rdquo; calculation).</li>
<li><code>linkbond</code> excludes MM charges that are connected to a quantum region
    by at most two bonds,</li>
<li><code>linkbond_H</code> similar to <code>linkbond</code> but excludes only hydrogen atoms.</li>
</ul>
<p>Keyword <code>expand</code> expands the set MM charges interacting with QM region
beyond the limits imposed by cutoff value.</p>
<ul>
<li><code>none</code> default value reverts to the original set of MM charges</li>
<li><code>solute</code> expands electrostatic interaction to all solute MM charges</li>
<li><code>solvent</code> expands electrostatic interaction to all solvent MM charges</li>
<li><code>all</code> expands electrostatic interaction to all MM charges</li>
</ul>
<p>Keyword <code>update</code> specifies how often list of MM charges will be updated in
the course of the calculation. Default behavior is not to update.</p>
<h3 id="qmmm_qmmm_parameters-qmmm-link_atoms">QM/MM link_atoms<a class="headerlink" href="#qmmm_qmmm_parameters-qmmm-link_atoms" title="Permanent link">&para;</a></h3>
<pre><code>link_atoms &lt;(hydrogen||halogen) default halogen&gt;
</code></pre>
<p>This directive controls the treatment of bonds crossing the boundary
between quantum and classical regions. The use of hydrogen keyword will
trigger truncation of such bonds with hydrogen link atoms. The position
of the hydrogen atom will be calculated from the coordinates of the
quantum and classical atom of the truncated bond using the following
expression</p>
<div class="arithmatex">\[R_{hlink} = (1-g)R_{quant} + g*R_{class}\]</div>
<p>where <em>g</em> is the scale factor set at 0.709</p>
<p>Setting <code>link_atoms</code> to <code>halogen</code> will result in the modification of the
quantum atom of the truncated bond to the fluoride atom. This
fluoride atom will typically carry an effective core potential (ECP)
basis set as specified in <a href="#qmmm_qmmm_parameters-qmmm-link_ecp"><code>link_ecp</code></a> directive.</p>
<h3 id="qmmm_qmmm_parameters-qmmm-link_ecp">QM/MM link_ecp<a class="headerlink" href="#qmmm_qmmm_parameters-qmmm-link_ecp" title="Permanent link">&para;</a></h3>
<pre><code>link_ecp &lt;(auto||user)default auto&gt; 
</code></pre>
<p>This directive specifies ECP basis set on fluoride link atoms. If set to
<code>auto</code> the ECP basis set given by Zhang, Lee, Yang<sup id="fnref_footnote_1"><a class="footnote-ref" href="#qmmm_qmmm_parameters-fn_footnote_1">1</a></sup> for 6-31G<em> basis
will be used. Strictly speaking, this implies the use of 6-31G</em>
spherical basis as the main basis set. If other choices are desired then
keyword user should be used and ECP basis set should be entered
separatelly following the format given in section dealing with <a href="#ecp">ECPs</a> .
The name tag for fluoride link atoms is <code>F_L</code>.</p>
<h3 id="qmmm_qmmm_parameters-qmmm-region">QM/MM region<a class="headerlink" href="#qmmm_qmmm_parameters-qmmm-region" title="Permanent link">&para;</a></h3>
<pre><code>region&lt;[region1][region2][region3]&gt;
</code></pre>
<p>This directive specifies active region(s) for optimization, dynamics,
frequency, and free energy calculations. Up to three regions can be
specified, those are limited to</p>
<ul>
<li><code>qm</code> - all quantum atoms </li>
<li><code>qmlink</code> - quantum and link atoms <span id="qmlink"></span></li>
<li><code>mm_solute</code> - all classical solute atoms excluding link atoms</li>
<li><code>solute</code> - all solute atoms including quantum</li>
<li><code>solvent</code> all solvent atoms</li>
<li><code>mm</code> all classical solute and solvent atoms, excluding link atoms</li>
<li><code>all</code> all atoms</li>
</ul>
<p>Only the first region will be used in dynamics, frequency, and free
energy calculations. In the geometry optimizations, all three regions
will be optimized using the following optimization methods</p>
<pre><code>if(region.eq.&quot;qm&quot;)then  
method=&quot;bfgs&quot;  
elseif(region.eq.&quot;qmlink&quot;)then  
method=&quot;bfgs&quot;  
elseif(region.eq.&quot;mm_solute&quot;)then  
method=&quot;lbfgs&quot;  
elseif(region.eq.&quot;mm&quot;)then  
method=&quot;sd&quot;  
elseif(region.eq.&quot;solute&quot;)then  
method=&quot;sd&quot;  
elseif(region.eq.&quot;solvent&quot;)then  
method=&quot;sd&quot;  
elseif(region.eq.&quot;all&quot;)then  
method=&quot;sd&quot;  
endif
</code></pre>
<p>where &ldquo;bfgs&rdquo; stands for BroydenFletcherGoldfarbShanno (BFGS)
optimization method, &ldquo;lbfgs&rdquo; limited memory version of quasi-newton, and
&ldquo;sd&rdquo; simple steepest descent algorithm. These assignments can be
potentially altered using <a href="#qmmm_method">method</a> directive.</p>
<h3 id="qmmm_qmmm_parameters-qmmm-method">QM/MM method<a class="headerlink" href="#qmmm_qmmm_parameters-qmmm-method" title="Permanent link">&para;</a></h3>
<pre><code>  method   [method1]  [method2]  [method3]
</code></pre>
<p>This directive controls which optimization algorithm will be used for the regions as defined by [[qmmm_region|Qmmm_region]] directive.
The allowed values are <code>bfgs</code> aka <a href="#geometry-optimization-geometry-optimization-with-driver">DRIVER</a>,
<code>lbfgs</code> limited memory version of quasi-newton,
and <code>sd</code> simple steepest descent algorithm.
The use of this directive is not recommended in all but special cases. In particular, 
<code>bfgs</code> should be used for QM region if there are any constraints,
<code>sd</code> method should always be used for classical solute and solvent atoms with shake constraints.</p>
<h3 id="qmmm_qmmm_parameters-qmmm-maxiter">QM/MM maxiter<a class="headerlink" href="#qmmm_qmmm_parameters-qmmm-maxiter" title="Permanent link">&para;</a></h3>
<pre><code> maxiter  [maxiter1] [maxiter2] [maxiter3]
</code></pre>
<p>This directive controls maximum number of iterations for the optimizations of regions as defined by by regions directive. User is strongly encouraged to set this directive explicitly as the default value shown below may not be appropriate in all the cases:</p>
<pre><code>      if(region.eq.&quot;qm&quot;) then
        maxiter = 20
      else if (region.eq.&quot;qmlink&quot;) then
        maxiter = 20
      else if (region.eq.&quot;mm&quot;) then
        maxiter = 100
      else if (region.eq.&quot;solvent&quot;) then
        maxiter = 100
      else
        maxiter = 50
      end if
</code></pre>
<h3 id="qmmm_qmmm_parameters-qmmm-ncycles">QM/MM ncycles<a class="headerlink" href="#qmmm_qmmm_parameters-qmmm-ncycles" title="Permanent link">&para;</a></h3>
<pre><code> ncycles  &lt; [number] default 1 &gt;
</code></pre>
<p>This directive controls the number of optimization cycles where the defined regions will be optimized in succession, or number of sampling cycles in free energy calculations.</p>
<h3 id="qmmm_qmmm_parameters-qmmm-density">QM/MM density<a class="headerlink" href="#qmmm_qmmm_parameters-qmmm-density" title="Permanent link">&para;</a></h3>
<pre><code>density[espfit][static][dynamical]defaultdynamical
</code></pre>
<p>This directive controls the electrostatic representation of <em>fixed</em> QM
region during optimization/dynamics of classical regions. It has no
effect when position of QM atoms are changing.</p>
<ul>
<li>
<p><strong>dynamical</strong> is an option where QM region is treated the standard
    way, through the recalculation of the wavefunction calculated and
    the resulting electron density is used to compute electrostatic
    interactions with classical atoms. This option is the most expensive
    one and becomes impractical for large scale optimizations.</p>
</li>
<li>
<p><strong>static</strong> option will not recalculate QM wavefunction but will
    still use full electron density in the computations of electrostatic
    interactions.</p>
</li>
<li>
<p><strong>espfit</strong> option will not recalculate QM wavefunction nor it will
    use full electron density. Instead, a set of ESP charges for QM
    region will be calculated and used to compute electrostatic
    interactions with the MM regions. This option is the most efficient
    and is strongly recommended for large systems.</p>
</li>
</ul>
<p>Note that both &ldquo;static&rdquo; and &ldquo;espfit&rdquo; options do require the presence of
the movecs file. It is typically available as part as part of
calculation process. Otherwise it can be generated by running qmmm
energy calculation.</p>
<p>In most calculations default value for <strong>density</strong> would <strong>dynamical</strong>,
with the exception of free energy calculations where default setting
<strong>espfit</strong> will be used.</p>
<h3 id="qmmm_qmmm_parameters-qmmm-load">QM/MM load<a class="headerlink" href="#qmmm_qmmm_parameters-qmmm-load" title="Permanent link">&para;</a></h3>
<pre><code>load &lt; esp &gt; [&lt;filename&gt;]
</code></pre>
<p>This directive instructs to load external file (located in permanent
directory) containing esp charges for QM region. If filename is not
provided it will be constructed from the name of the restart file by
replacing &ldquo;.rst&rdquo; suffix with &ldquo;.esp&rdquo;. Note that file containing esp
charges is always generated whenever esp charge calculation is performed</p>
<h3 id="qmmm_qmmm_parameters-qmmm-convergence">QM/MM convergence<a class="headerlink" href="#qmmm_qmmm_parameters-qmmm-convergence" title="Permanent link">&para;</a></h3>
<pre><code>convergence&lt;doubleprecisionetoldefault1.0d-4&gt;
</code></pre>
<p>This directive controls convergence of geometry optimization. The
optimization is deemed converged if absolute difference in total
energies between consecutive optimization cycles becomes less than
<em>etol</em>.</p>
<h3 id="qmmm_qmmm_parameters-qmmm-nsamples">QM/MM nsamples<a class="headerlink" href="#qmmm_qmmm_parameters-qmmm-nsamples" title="Permanent link">&para;</a></h3>
<pre><code>nsamples
</code></pre>
<p>This directive is required for <a href="#qmmm_free_energy-nsamples">free energy
calculations</a> and defines number
of samples for averaging during single cycle.</p>
<h2 id="qmmm_qmmm_parameters-references">References<a class="headerlink" href="#qmmm_qmmm_parameters-references" title="Permanent link">&para;</a></h2>
<div class="footnote">
<hr />
<ol>
<li id="qmmm_qmmm_parameters-fn_footnote_1">
<p>Zhang, Y.; Lee, T.-S.; Yang, W. A Pseudobond Approach to Combining Quantum Mechanical and Molecular Mechanical Methods. <em>The Journal of Chemical Physics</em> <strong>1999</strong>, <em>110</em> (1), 4654. <a href="https://doi.org/10.1063/1.478083">https://doi.org/10.1063/1.478083</a>.&#160;<a class="footnote-backref" href="#qmmm_qmmm_parameters-fnref_footnote_1" title="Jump back to footnote 1 in the text">&#8617;</a></p>
</li>
</ol>
</div></section></section>
                    <section class='print-page md-section' id='section-13-4-5' heading-number='13.4.5'>
                        <h1>QMMMM Single Point Calculations<a class='headerlink' href='#section-13-4-5' title='Permanent link'></a>
                        </h1>
                    <section class="print-page" id="qmmm_sp_energy" heading-number="13.4.5.1"><h1 id="qmmm_sp_energy-qmmm-single-point-calculations">QMMM Single Point Calculations<a class="headerlink" href="#qmmm_sp_energy-qmmm-single-point-calculations" title="Permanent link">&para;</a></h1>
<p>The task directive for QM/MM
single point energy and gradient calculations is given by</p>
<pre><code>task qmmm &lt;qmtheory&gt; energy
</code></pre>
<p>or</p>
<pre><code>task qmmm &lt;qmtheory&gt; gradient [numerical]
</code></pre>
<p>where <code>qmtheory</code> refers to the level of QM theory (e.g. dft, tce, mp2,
&hellip;).</p>
<p>The ground state QM/MM energy calculations should be possible with all
QM descriptions available in NWChem, however most of testing was
performed using core QM methods (scf,dft,mp2,tce). The ground state
QM/MM gradient calculations can be performed analytically with
scf,dft,mp2 levels of theory and numerically for all the others.</p>
<p>The relevant settings for QM/MM interface block for energy and gradient
calculations include</p>
<ul>
<li><a href="#qmmm_bq_zone">bqzone</a></li>
<li><a href="#qmmm_mm_charges">mm_charges</a></li>
<li><a href="#qmmm_link_atoms">link_atoms</a></li>
<li><a href="#qmmm_link_ecp">link_ecp</a>.</li>
</ul>
<p><a href="#qmmm_example3">qmmm_example3</a></p></section><section class="print-page" id="qmmm_excited_states" heading-number="13.4.5.2"><p>The excited
state QM/MM energy calculations can be performed with
<a href="#tce">TCE</a></p>
<p><code>taskqmmmtceenergy</code></p>
<p>or <a href="#excited-state-calculations">TDDFT</a></p>
<p><code>taskqmmmtddftenergy</code></p>
<p>levels of theory. The excited state QM/MM gradient energy calculations
can be performed only numerically.</p></section><section class="print-page" id="qmmm_sp_property" heading-number="13.4.5.3"><h1 id="qmmm_sp_property-qmmm-properties">QM/MM properties<a class="headerlink" href="#qmmm_sp_property-qmmm-properties" title="Permanent link">&para;</a></h1>
<p>A number of electronic structure properties can be calculated with QM/MM
using capabilities provided by <a href="#properties">property</a>,
<a href="#esp">esp</a>, and <a href="#dplot">dplot modules</a>.</p>
<p>The example below illustrates dipole property QM/MM DFT/B3LYP
calculation for quantum water molecule embedded into 20 angstrom box of
classical SPCE/E water molecules.</p>
<p>The preparation stage that involves definition of the QM region and
solvation is performed as part of the calculation. Note that water
fragment file <a href="WTR.frg">wtr.frg</a> is required in this
calculation. Prepare run will generate restart file
(<a href="wtr_ref.rst">wtr_ref.rst</a>) and topology file
(<a href="wtr.top">wtr.top</a>)</p>
<p>In the QM/MM interface block the use of <code>bq_zone</code> value of 3.0 Angstrom
is specified.</p>
<pre><code>  start wtr

  permanent_dir ./perm  
  scratch_dir ./data

  prepare  
  source wtr0.pdb  
  new_top new_seq  
  new_rst  
  modify segment 1  quantum  
  center  
  orient  
  solvate box 3.0  
  update lists  
  ignore  
  write wtr_ref.rst  
  write wtr_ref.pdb  
  end

  task prepare

  md  
  system wtr_ref  
  end

  basis  
  * library &quot;6-31G&quot;  
  end

  dft  
  xc b3lyp  
  end

  qmmm  
  bq_zone 3.0  
  end

  property  
   dipole  
  end

  task qmmm dft property
</code></pre></section><section class="print-page" id="qmmm_esp" heading-number="13.4.5.4"><ul>
<li>Example QM/MM ESP Calculation:</li>
</ul>
<p>The example below illustrates dipole property QM/MM DFT/B3LYP
calculation for quantum water molecule embedded into 20 angstrom box of
classical SPCE/E water molecules.</p>
<p>The preparation stage that involves definition of the QM region and
solvation is performed as part of the calculation. Note that water
fragment file <a href="WTR.frg">wtr.frg</a> is required in this
calculation. Prepare run will generate restart file
(<a href="wtr_ref.rst">wtr_ref.rst</a>) and topology file
(<a href="wtr.top">wtr.top</a>)</p>
<p>In the QM/MM interface block the use of bq_zone value of 3.0 Angstrom
is specified.</p>
<pre><code>  start wtr

  permanent_dir ./perm  
  scratch_dir ./data

  prepare  
  source wtr0.pdb  
  new_top new_seq  
  new_rst  
  modify segment 1  quantum  
  center  
  orient  
  solvate box 3.0  
  update lists  
  ignore  
  write wtr_ref.rst  
  write wtr_ref.pdb  
  end

  task prepare

  md  
  system wtr_ref  
  end

  basis  
  * library &quot;6-31G&quot;  
  end

  dft  
  xc b3lyp  
  end

  qmmm  
  bq_zone 3.0  
  end

  property  
   dipole  
  end

  task qmmm dft property
</code></pre></section></section>
                    <section class='print-page md-section' id='section-13-4-6' heading-number='13.4.6'>
                        <h1>QMMM Potential Energy Surface Analysis<a class='headerlink' href='#section-13-4-6' title='Permanent link'></a>
                        </h1>
                    <section class="print-page" id="qmmm_optimization" heading-number="13.4.6.1"><p>QM/MM optimization is based on multi-region optimization
methodology and is invoked by</p>
<pre><code>taskqmmm&lt;qmtheory&gt;optimize
</code></pre>
<p>The overall algorithm involves alternating optimizations of QM and MM
regions until convergence is achieved. This type of approach offers
substantial savings compared to direct optimization of the entire system
as a whole. In the simplest case of two regions (QM and MM) the
algorithm is comprised of the following steps:</p>
<ol>
<li>Optimization of the QM region keeping MM region fixed</li>
<li>Calculation of reduced electrostatic representation for the QM
    region (e.g. ESP charges)</li>
<li>Optimization of MM region keeping QM region fixed</li>
<li>Repeat from Step 1 until converged</li>
</ol>
<p>The optimization process is controlled by the following keywords:</p>
<ul>
<li><a href="#qmmm_region"><strong>region</strong></a> - <strong>required</strong>
    keyword which specifies which regions will optimized and in which
    order.</li>
<li><a href="#qmmm_maxiter"><strong>maxiter</strong></a> - number of
    optimizations steps for each region within single optimization pass</li>
<li><a href="#qmmm_ncycles"><strong>ncycles</strong></a> - number of
    optimization cycles</li>
<li><a href="#qmmm_density"><strong>density</strong></a> - electrostatic
    representation of the QM region during MM optimization</li>
<li><a href="#qmmm_xyz"><strong>xyz</strong></a> - output of xyz structure
    files</li>
<li><a href="#qmmm_convergence"><strong>convergence</strong></a> -
    convergence criteria</li>
</ul>
<p>Here is an example QM/MM block that provides practical illustration of
all these keywords for a generic optimization case where QM molecule(s)
are embedded in the solvent</p>
<pre><code>qmmm
[region]qmsolvent
[maxiter]103000
[ncycles]5
[density]espfit
[xyz]foo
end
</code></pre>
<p>We have two regions in the system &ldquo;qm&rdquo; and &ldquo;solvent&rdquo; and we would like
to optimize them both, thus the line</p>
<pre><code>[region]qmsolvent
</code></pre>
<p>Our QM region is presumably small and the maximum number of iterations
(within a single optimization pass) is set to 10. The solvent region is
typically much larger (thousands of atoms) and the maximum number of
iterations is set to a much large number 3000:</p>
<pre><code>[maxiter]103000
</code></pre>
<p>We would like to perform a total of 5 optimization passes, giving us a
total of 5*10=50 optimization steps for QM region and 5*3000=15000
optimization steps for solvent region:</p>
<pre><code>[ncycles]5
</code></pre>
<p>We are requesting QM region to be represented by point ESP charges
during the solvent optimization:</p>
<pre><code>[density]espfit
</code></pre>
<p>Finally we are requesting that the coordinates of the first region to be
saved in the form of numbered xyz files:</p>
<pre><code>[xyz]foo
</code></pre></section><section class="print-page" id="qmmm_transition_states" heading-number="13.4.6.2"><p>QM/MM transition states calculations for qm or qmlink regions can be
performed using</p>
<p><code>taskqmmm</code><qmtheory><code>saddle</code></p>
<p>The overall algorithm is very similar to QM/MM optimization
calculations, but instead of optimization, transition state search will
be performed for <a href="#qmmm_region-qm">qm</a> or
<a href="#qmmm_region-qmlink">qmlink</a> region for specified number of
steps ( as defined by <a href="#qmmm_maxiter">maxiter</a> keyword). The
remaining classical regions (if any) will be optimized following the
standard optimization protocol, which may involve, if specified, ESP
charge representation of the QM atoms (a recommended option).</p>
<p>The success transition state calculations is strongly dependent on the
initial guess. User may consider generation of the latter using QM/MM
reaction pathway calculation. Another useful strategy involves
precalculation of the Hessian. Following the example presented above one
could have precalculated numerical Hessian for the qm region</p>
<p><code>...</code><br />
<code>qmmm</code><br />
<code>regionqm</code><br />
<code>end</code>  </p>
<p><code>freq</code><br />
<code>animate</code><br />
<code>end</code>  </p>
<p><code>taskqmmmdftfreq</code></p>
<p>and then used this information in the TS calculation</p>
<p><code>...</code><br />
<code>driver</code><br />
<code>clear</code><br />
<code>inhess2#readinhessianfrompermdirectory</code><br />
<code>moddir1#follow1stmode</code><br />
<code>end</code>  </p>
<p><code>qmmm</code><br />
<code>bqzone15.0</code><br />
<code>regionqmsolvent</code><br />
<code>xyzts</code><br />
<code>maxiter101000</code><br />
<code>ncycles2</code><br />
<code>densityespfit</code><br />
<code>end</code>  </p>
<p><code>taskqmmmdftsaddle</code></p></section><section class="print-page" id="qmmm_freq" heading-number="13.4.6.3"><h1 id="qmmm_freq-qmmm-hessian-and-frequency-calculations">QM/MM hessian and frequency calculations<a class="headerlink" href="#qmmm_freq-qmmm-hessian-and-frequency-calculations" title="Permanent link">&para;</a></h1>
<h3 id="qmmm_freq-setup">Setup<a class="headerlink" href="#qmmm_freq-setup" title="Permanent link">&para;</a></h3>
<p>QM/MM hessian and frequency calculations are invoked though the
following task directives</p>
<pre><code>task qmmm `&lt;qmtheory&gt;` hessian
</code></pre>
<p>or</p>
<pre><code>task qmmm `&lt;qmtheory&gt;` freq
</code></pre>
<p>Only numerical implementation are supported at this point and will be
used even in the absence of &ldquo;numerical&rdquo; keyword. Other than standard
QM/MM directives no additional QM/MM input is required for default
hessian/frequency for all the QM atoms. Using
<a href="#qmmm_region">region</a> keyword(first region only),
hessian/frequency calculations can also be performed for classical
solute atoms. If only classical atoms are involved
<a href="#qmmm_density">density</a> keyword can be utilized to enable
frozen density or ESP charge representation for fixed QM region.
Hessian/frequency calculations for solvent are not possible.</p>
<h3 id="qmmm_freq-examples">Examples<a class="headerlink" href="#qmmm_freq-examples" title="Permanent link">&para;</a></h3>
<h4 id="qmmm_freq-example-of-qmmm-frequency-calculation-for-classical-region">Example of QM/MM frequency calculation for classical region<a class="headerlink" href="#qmmm_freq-example-of-qmmm-frequency-calculation-for-classical-region" title="Permanent link">&para;</a></h4>
<p>This example illustrates QM/MM frequency calculation for Ala-Ser-Ala
system. In this case instead of default QM region (see prepare block),
the calculation is performed on classical solute part of the system as
defined by region directive in QM/MM block. The electrostatic field from
fixed QM region is represented by point ESP charges (see density
directive). These ESP charges are calculated from wavefunction generated
as a result of energy calculation.</p>
<pre><code> memory total 800 Mb  

 start asa  

 permanent_dir ./perm  
 scratch_dir ./data  

 #this will generate topology file ([asa.top](asa.top)), restart ([asa_ref.rst](asa_ref.rst)), and pdb ([asa_ref.pdb](asa_ref.pdb)) files.  
 prepare  
   source [asa.pdb](asa.pdb)  
   new_top new_seq  
   new_rst  
   modify atom 2:_CB quantum  
   modify atom 2:2HB quantum  
   modify atom 2:3HB quantum  
   modify atom 2:_OG quantum  
   modify atom 2:_HG quantum  
   center  
   orient  
   solvate  
   update lists  
   ignore  
   write [asa_ref.rst](asa_ref.rst)  
   write [asa_ref.pdb](asa_ref.pdb)   # Write out PDB file to check structure  
 end  
 task prepare

 md  
   system asa_ref  
 end  

 basis &quot;ao basis&quot;  
   * library &quot;6-31G*&quot;  
 end  

 dft  
  print low  
  iterations 500  
 end  

 qmmm  
 region mm_solute  
 density espfit  
 end  

 # run energy calculation to generate wavefunction file for subsequent ESP charge generation  
 task qmmm dft energy  
 task qmmm dft freq
</code></pre></section><section class="print-page" id="qmmm_neb_calculations" heading-number="13.4.6.4"><p>Experimental implementation of Nudged Elastic Band (NEB) method is
available for reaction pathway calculations with QM/MM. The actual
pathway/beads construction involves (by default) only the region
containing QM and link atoms (referred to as qmlink). The rest of the
system plays a passive role and is quenched/optimized each time a
gradient on a bead is calculated.</p>
<p>The initial guess for NEB pathway can be generated using geometries of
the starting and ending point provided by the .rst files. These are set
in the input using the following directive</p>
<pre><code>setqmmm:neb_path_limitsxxx_start.rstxxx_end.rst
</code></pre>
<p>where xxx_start.rst xxx_end.rst refers to starting and ending point of
the NEB pathway. Both rst files have to be present at the <strong>top level</strong>
directory. It should be noted that only coordinates of qmlink region
will be used from these two files. The initial coordinates for the rest
of the system come from reference rst file provided in the MD block</p>
<pre><code>md
systemxxx_ref
...
end
</code></pre>
<p>Typically this reference restart file ( xxx_ref.rst ) would be a copy
of a restart file for starting or ending point.</p>
<p>The number of beads in the NEB pathway, initial optimization step size,
and number of optimization steps are set using the following directives</p>
<pre><code>setneb:nbeads10
setneb:stepsize10
setneb:steps20
</code></pre>
<p>The calculation starts by constructing initial guess for the pathway
(consisting of a sequence of numbered rst files) by combining linearly
interpolated coordinates of the qmlink regions from starting and ending
rst files and classical coordinates from the reference file. Next phase
involves calculation of the gradients on qmlink region atoms for each of
the beads. This involves two steps. First classical region around the
qmlink region is relaxed following standard QM/MM optimization protocol.
Aside the fact that optimization region cannot be qmlink, all other
optimization directives apply and should be set in the QM/MM block
following standard convention, e.g.</p>
<pre><code>qmmm
regionsolvent
maxiter1000
ncycles1
densityespfit
end
</code></pre>
<p>or</p>
<pre><code>qmmm
regionmm_solutesolvent
maxiter3001000
ncycles3
densityespfit
end
</code></pre>
<p>In both examples presented above we utilized espfit option for density
to speed up calculations. <strong>Note that optimization region cannot be
qmlink!</strong></p>
<p>After the optimization has been performed the gradient on qmlink region
is calculated. The procedure is repeated for all the beads. After that
the bead coordinates will be advanced following NEB protocol and the
entire cycle will be repeated again.</p>
<p>In addition to interpolated initial guess, one can also specify custom
initial path represented by numbered sequence of restart files stored in
the <strong>perm</strong> directory. This behavior will be triggered automatically in
the absence of <em>qmmm:neb_path_limits</em> directive. The default naming of
the custom initial path is of the form <system>XXX.rst, where <system>
is the prefix of reference restart file as defined in MD block and XXX
is the 3-digit integer counter with zero blanks (001,002, &hellip;, 010, 011,
..). If needed the prefix for the custom initial path can be adjusted
using</p>
<p><code>setqmmm:neb_path</code><string prefix></p>
<p>The progress of NEB calculation can be monitored by</p>
<pre><code>grepgnorm&lt;output file&gt;
</code></pre>
<p>Experience shows that the value of gnorm less or around O(10^-2)
indicates converged pathway. The current pathway in the XYZ format can
be found in the output file (look for XYZ FILE string) and viewed as
animation in some of the molecular viewers (e.g.
<a href="https://jmol.sourceforge.net/">JMOL</a>)</p>
<p>The entire example directory including the output file can be downloaded from <a href="QMMM_NEB_example1.tar.gz">here</a>. </p></section></section><section class="print-page" id="qmmm_dynamics" heading-number="13.4.7"><p>Dynamical simulations within QM/MM framework can be
initiated using</p>
<pre><code>taskqmmm &lt;qmtheory&gt; dynamics 
</code></pre>
<p>directive. User has to specify the
<a href="#qmmm_region">region</a> for which simulation will
performed. If dynamics is performed only for the classical parts of the
system (QM region is fixed) then ESP point charge representation
(<a href="#qmmm_density">density espfit</a>) is recommended to
speed up simulations. If this option is utilized then wavefunction file
(<system>.movecs) has to be available and present in the permanent
directory (this can be most easily achieved by running energy
calculation prior to dynamics).</p></section><section class="print-page" id="qmmm_free_energy" heading-number="13.4.8"><h1 id="qmmm_free_energy-qmmm-free-energy">QMMM Free Energy<a class="headerlink" href="#qmmm_free_energy-qmmm-free-energy" title="Permanent link">&para;</a></h1>
<h2 id="qmmm_free_energy-overview">Overview<a class="headerlink" href="#qmmm_free_energy-overview" title="Permanent link">&para;</a></h2>
<p>Free energy capabilities of QM/MM module are at this point restricted to
calculations of free energy differences between two fixed configurations
of the QM region.</p>
<p><strong>Users must be warned that this the least automated QM/MM functionality
containing several calculation stages. Solid understanding of free
energy calculations is required to achieve a meaningful calculation.</strong></p>
<p>Description of the implemented methodology can be found in <a href="https://dx.doi.org/10.1063/1.2768343">the
following paper</a>. In this
approach the free energy difference between the two configurations of
the QM region (e.g. A and B):  </p>
<p><span class="arithmatex">\(\Delta W_{A\to B}=-1/\beta \ln \langle (e^{-\beta (E_B-E_A)})  \rangle_{A}\)</span></p>
<p>is approximated as a sum of internal QM contribution and solvation:  </p>
<p><span class="arithmatex">\(\Delta W_{A\to B}\approx\Delta W_{A\to B}^{int}+\Delta W_{A\to B}^{solv}\)</span></p>
<p>It is presumed that structures of A and B configurations are available
as restart files sharing <strong>common</strong> topology file.  </p>
<h2 id="qmmm_free_energy-internal-contribution">Internal contribution<a class="headerlink" href="#qmmm_free_energy-internal-contribution" title="Permanent link">&para;</a></h2>
<p>The internal QM contribution is given by the differences in the internal
QM energies evaluated at the <strong>optimized</strong> MM environment:  </p>
<p><span class="arithmatex">\(\Delta W_{A\to B}^{int}=E_{B}^{int}-E_{A}^{int}\)</span></p>
<p>The internal QM energy is nothing more but a gas phase expression total
energy but evaluated with the wavefunction obtained in the presence of
the environment. To calculate internal QM contribution to free energy
difference one has to</p>
<ol>
<li>Optimize MM environment for each configuration</li>
<li>Perform total energy calculation for each configuration</li>
<li>Calculate internal energy difference</li>
</ol>
<p>Note that internal QM energy can be found in the QM/MM output file under
&ldquo;quantum energy internal&rdquo; name.</p>
<h2 id="qmmm_free_energy-solvation-contribution">Solvation contribution<a class="headerlink" href="#qmmm_free_energy-solvation-contribution" title="Permanent link">&para;</a></h2>
<p>The solvation contribution is evaluated by averaging energy difference
between A and B configurations of the QM system represented by a set of
ESP charges.  </p>
<p><span class="arithmatex">\(\Delta W_{A\to B}^{solv}=-1/\beta \ln \langle (e^{-\beta (E_B^{ESP}-E_A^{ESP})})  \rangle_{A}\)</span></p>
<p>where <span class="arithmatex">\(E_A^{ESP}\)</span> is the total energy of the system where QM region is
replaced by a set of fixed point ESP charges.  </p>
<p>In majority of cases the A and B configuration are &ldquo;too far apart&rdquo; and
one step free energy calculation as shown above will not lead to
meaningful results. One solution is to introduce intermediate points
that bridge A and B configurations by linear interpolation  </p>
<p><span class="arithmatex">\(R_{\lambda_i} = (1-\lambda_i)R_A + \lambda_i R_B\)</span></p>
<p><span class="arithmatex">\(Q_{\lambda_i} = (1-\lambda_i)Q_A + \lambda_i Q_B\)</span></p>
<p>where  </p>
<p><span class="arithmatex">\(\lambda_i = \frac {i}{n}, \quad i=0,..,n\,\!\)</span></p>
<p>The solvation free energy difference can be then written as sum of
differences for the subintervals <span class="arithmatex">\(\,\! [\lambda_i\to\lambda_{i+1}]\)</span>  :  </p>
<p><span class="arithmatex">\( \Delta W_{A\to B}^{\rm esp} = \sum_{i=0}^{n}\Delta W_{\lambda_i\to\lambda_{i+1}}^{\rm esp} \)</span></p>
<p>To expedite the calculation it is convenient to use a double wide
sampling strategy where the free energy differences for the intervals
<span class="arithmatex">\(\,\! [\lambda_{i-1}\to\lambda_{i}]\)</span>  and
<span class="arithmatex">\(\,\! [\lambda_i\to\lambda_{i+1}]\)</span>  are calculated simultaneously by
sampling around <span class="arithmatex">\(\,\!\lambda_{i}\)</span>   point. In the simplest case where we
use two subintervals (n=2)  </p>
<p><span class="arithmatex">\(\Delta W_{A\to B}^{solv} \equiv \Delta W_{0\to 1}= \Delta W_{0\to 0.5}^{solv}+\Delta W_{0.5\to 1}^{solv}\)</span></p>
<p>or  </p>
<p><span class="arithmatex">\(\Delta W_{A\to B}^{\rm solv} = -\Delta W_{0.5\to 0}^{\rm solv}+\Delta W_{0.5 \to 1}^{\rm solv}\)</span></p>
<p>The following items are necessary:</p>
<ol>
<li>Restart file corresponding to either A or B configuration of the QM
    region (sharing the same topology file)</li>
<li>ESP charges for QM region in .esp format for both configurations</li>
<li>Coordinates for QM regions in .xyzi format for both configurations</li>
</ol>
<p>Both .esp and .xyzi files would be typically obtained during the
calculation of internal free energy (see above). ESP charges would be
generated in the perm directory during optimization of the MM region.
The xyzi is basically xyz structure file with an extra column that
allows to map coordinates of QM atoms to the overall system. The xyzi
file can also be obtained as part of calculation of internal free energy
by inserting</p>
<pre><code>set qmmm:region_print .true.
</code></pre>
<p>anywhere in the input file during energy calculation. <strong>Both xyzi and
esp files should be placed into the perm directory!!!</strong></p>
<p>In the input file the restart file is specified in the MD block
following the standard notation</p>
<pre><code>md
 system &lt; name of rst file without extension&gt;
 ...
end
</code></pre>
<p>while coordinates of QM region (xyzi files) and ESP charges (esp files)
are set using the following directives (at the top level outside of any
input blocks)</p>
<pre><code>set qmmm:fep_geom xxx_A.xyzi xxx_B.xyzi
set qmmm:fep_esp  xxx_A.esp xxx_B.esp
</code></pre>
<p>The current interpolation interval <span class="arithmatex">\(\,\! [\lambda_i\to\lambda_{i+1}]\)</span><br />
for which free energy difference is calculated is defined as</p>
<pre><code>set qmmm:fep_lambda lambda_i lambda_i+1
</code></pre>
<p>To enable double wide sampling use the following directive</p>
<pre><code>set qmmm:fep_deriv .true.
</code></pre>
<p>If set, the above directive will perform both
<span class="arithmatex">\(\,\! [\lambda_i\to\lambda_{i+1}]\)</span>  and
<span class="arithmatex">\(\,\! [\lambda_i\to\lambda_{i-1}]\)</span>  calculations, where  </p>
<p><span class="arithmatex">\(\,\! \lambda_{i-1}=\lambda_i - (\lambda_{i+1}-\lambda_i)\)</span></p>
<p>The calculation proceeds in cycles, each cycle consisting of two phases.
First phase is generation of classical MD trajectory around
<span class="arithmatex">\(\,\! \lambda_i\)</span> point. Second phase is processing of the generated
trajectory to calculate averages of relevant energy differences. The
number of MD steps in the first phase is controlled by the QM/MM
directive <span id="nsamples"></span></p>
<ul>
<li><strong><a href="#qmmm_nsamples">nsamples</a></strong>
    <integer number of MD steps for sampling></li>
</ul>
<p>This is a <strong>required</strong> directive for QM/MM free energy calculations.</p>
<p>Number of overall cycles is defined by the QM/MM directive</p>
<ul>
<li><strong><a href="#qmmm_ncycles">ncycles</a></strong> <integer number
    of cycles default 1></li>
</ul>
<p>In most cases explicit definition of QM/MM
<strong><a href="#qmmm_density">density</a></strong> and
<strong><a href="#qmmm_region">region</a></strong> should not be required.
The QM/MM <strong><a href="#qmmm_density">density</a></strong> will
automatically default to <strong>espfit</strong> and
<strong><a href="#qmmm_region">region</a></strong> to <strong>mm</strong>.</p>
<p>Prior to data collection for free energy calculations user may want to
prequilibrate the system, which can be achieved by <strong>equil</strong> keyword in
the MD block:</p>
<pre><code> md
  ... 
  equil &lt;number of equilibration steps&gt;
 end
</code></pre>
<p>Other parameters (e.g. temperature and pressure can be also set in the
MD block.</p>
<p>The actual QM/MM solvation free energy calculation is invoked through
the following task directive</p>
<pre><code>task qmmm fep
</code></pre>
<p>The current value of solvation free energy differences may be tracked
though</p>
<pre><code>grep free &lt;name of the output file&gt;
</code></pre>
<p>The first number is a forward (<span class="arithmatex">\(\,\! [\lambda_i\to\lambda_{i+1}]\)</span>)
free energy difference and second number is backward
(<span class="arithmatex">\(\,\! [\lambda_i\to\lambda_{i-1}]\)</span>) free energy difference, both in
kcal/mol. The same numbers can also be found in the 4th and 6th columns
of <system>.thm file but this time in <strong>atomic units</strong>.</p>
<p>The same <system>.thm file can also be used to continue from the prior
calculation. This will require the presence of</p>
<pre><code> set qmmm:extend .true.
</code></pre>
<p>directive, the <system>.thm file, and the appropriate rst file.</p>
<p>Here is an <a href="#qmmm_fep_example">example of the input file for QM/MM solvation free energy
calculation</a>.</p></section><section class="print-page" id="qmmm_appendix" heading-number="13.4.9"><h1 id="qmmm_appendix-qmmm-apppendix">QMMM Apppendix<a class="headerlink" href="#qmmm_appendix-qmmm-apppendix" title="Permanent link">&para;</a></h1>
<h2 id="qmmm_appendix-preparing-qmmm-calculations-from-scratch">Preparing QM/MM calculations from scratch<a class="headerlink" href="#qmmm_appendix-preparing-qmmm-calculations-from-scratch" title="Permanent link">&para;</a></h2>
<p>Setting up QM/MM calculations for a new system for which classical force
analog is not readily available would typically involve the following
steps</p>
<ol>
<li><a href="#qmmm_appendix-generation-of-the-proper-pdb-file">Generation of the proper PDB file</a></li>
<li><a href="#qmmm_appendix-generation-of-new-fragment-files">Construction the fragment file(s)</a></li>
<li><a href="#qmmm_appendix-generation-of-new-parameter-files">Creation of parameter file if new atom types were defined</a></li>
</ol>
<h2 id="qmmm_appendix-generation-of-the-proper-pdb-file">Generation of the proper PDB file<a class="headerlink" href="#qmmm_appendix-generation-of-the-proper-pdb-file" title="Permanent link">&para;</a></h2>
<p>It is often the case that the input structure for the system comes in
the form of the xyz file. Let us take a concrete example of N3O3-
molecule, which we would like to embed in classical solvent and perform
QM/MM calculations. Here is the structure of just N3O3- in xyz format
(generated in course of gas phase optimizations)</p>
<pre><code>   6      
  geometry   
  N                     1.31562667     0.93574165    -0.42424728  
  O                     1.56161766     0.18015298    -1.36827899  
  N                     2.36373381     1.02559495     0.48834956  
  O                     3.47240000     0.42852552     0.42137570  
  N                     1.95804013     1.90608355     1.48799418  
  O                     2.81393172     2.06788134     2.36142683
</code></pre>
<p>We cannot use this file as is in the QM/MM simulations, and it has to be
converted into PDB format. This is needed even if we plan to treat this
molecule quantum mechanically. There is more than way to do it. For
example, we could use Babel <a href="https://openbabel.org">https://openbabel.org</a>,
which will generate the PDB file as  </p>
<pre><code>COMPND    geometry   
AUTHOR    GENERATED BY OPEN BABEL 2.3.0  
HETATM    1  O   LIG     1       1.562   0.180  -1.368  1.00  0.00           O    
HETATM    2  N   LIG     1       1.316   0.936  -0.424  1.00  0.00           N    
HETATM    3  N   LIG     1       2.364   1.026   0.488  1.00  0.00           N    
HETATM    4  O   LIG     1       3.472   0.429   0.421  1.00  0.00           O    
HETATM    5  N   LIG     1       1.958   1.906   1.488  1.00  0.00           N    
HETATM    6  O   LIG     1       2.814   2.068   2.361  1.00  0.00           O    
CONECT    1    2                                                        
CONECT    2    1    3                                                   
CONECT    3    2    4    5                                              
CONECT    4    3                                                        
CONECT    5    3    6                                                   
CONECT    6    5                                                        
MASTER        0    0    0    0    0    0    0    0    6    0    6    0  
END
</code></pre>
<p>which after stripping nonessential information becomes  </p>
<pre><code> HETATM    1  O   LIG     1       1.562   0.180  -1.368  1.00  0.00           O    
 HETATM    2  N   LIG     1       1.316   0.936  -0.424  1.00  0.00           N    
 HETATM    3  N   LIG     1       2.364   1.026   0.488  1.00  0.00           N    
 HETATM    4  O   LIG     1       3.472   0.429   0.421  1.00  0.00           O    
 HETATM    5  N   LIG     1       1.958   1.906   1.488  1.00  0.00           N    
 HETATM    6  O   LIG     1       2.814   2.068   2.361  1.00  0.00           O    
 END
</code></pre>
<p>This is not yet the format we want. So what we need to do is</p>
<ol>
<li><em>replace HETATM field by ATOM preserving the format (column
locations) of the rest of the file</em>:</li>
</ol>
<p>I would typically use sed for this purpose</p>
<pre><code> sed 's/HETATM/ATOM  /' n3o3-bad.pdb &gt; n3o3-step1.pdb
</code></pre>
<p>where <code>n3o3-bad.pdb</code> is the original pdb file from Babel and
<code>n3o3-step1.pdb</code> is the converted one as shown below</p>
<pre><code>ATOM      1  O   LIG     1       1.562   0.180  -1.368  1.00  0.00           O  
ATOM      2  N   LIG     1       1.316   0.936  -0.424  1.00  0.00           N  
ATOM      3  N   LIG     1       2.364   1.026   0.488  1.00  0.00           N  
ATOM      4  O   LIG     1       3.472   0.429   0.421  1.00  0.00           O  
ATOM      5  N   LIG     1       1.958   1.906   1.488  1.00  0.00           N  
ATOM      6  O   LIG     1       2.814   2.068   2.361  1.00  0.00           O  
END
</code></pre>
<ol>
<li><em>define residues (aka molecules):</em></li>
</ol>
<p>In our case Babel did this for us and the entire system is defined as
one residue with the name LIG (see columns 4 and 5). We can leave it as,
but I will redefine residue name to NN3 (keep it to 3 characters !).
Again running <code>sed 's/LIG/NN3/' n3o3-step1.pdb &gt; n3o3-step2.pdb</code></p>
<pre><code>ATOM      1  O   NN3     1       1.562   0.180  -1.368  1.00  0.00           O  
ATOM      2  N   NN3     1       1.316   0.936  -0.424  1.00  0.00           N  
ATOM      3  N   NN3     1       2.364   1.026   0.488  1.00  0.00           N  
ATOM      4  O   NN3     1       3.472   0.429   0.421  1.00  0.00           O  
ATOM      5  N   NN3     1       1.958   1.906   1.488  1.00  0.00           N  
ATOM      6  O   NN3     1       2.814   2.068   2.361  1.00  0.00           O  
END
</code></pre>
<p>You could have also broken it up into several residues as</p>
<pre><code>ATOM      1  O   NN2     1       1.562   0.180  -1.368  1.00  0.00           O  
ATOM      2  N   NN2     1       1.316   0.936  -0.424  1.00  0.00           N  
ATOM      3  N   NN2     1       2.364   1.026   0.488  1.00  0.00           N  
ATOM      4  O   NN2     1       3.472   0.429   0.421  1.00  0.00           O  
ATOM      5  N   NN1     2       1.958   1.906   1.488  1.00  0.00           N  
ATOM      6  O   NN1     2       2.814   2.068   2.361  1.00  0.00           O  
END
</code></pre>
<p>where I defined two residues NN2 and NN1 (note changes in columns 4 and
5). To keep things simple I will stay with one residue version for now</p>
<ol>
<li><em>make unique atom names</em>:</li>
</ol>
<p>This has to do with the requirement that all atoms names have to be
unique within a given residue. So if we take our one residue version it
could be modified as (notice column 3)</p>
<pre><code>ATOM      1  O1  NN3     1       1.562   0.180  -1.368  1.00  0.00           O  
ATOM      2  N1  NN3     1       1.316   0.936  -0.424  1.00  0.00           N  
ATOM      3  N2  NN3     1       2.364   1.026   0.488  1.00  0.00           N  
ATOM      4  O2  NN3     1       3.472   0.429   0.421  1.00  0.00           O  
ATOM      5  N3  NN3     2       1.958   1.906   1.488  1.00  0.00           N  
ATOM      6  O3  NN3     2       2.814   2.068   2.361  1.00  0.00           O  
END
</code></pre>
<p>Now we have a PDB file for our system in &ldquo;proper&rdquo; PDB format. Before we
move to next step, I should mention that the last 3 columns are not
necessary and could have been removed at any point leading to
<a href="n3o3.pdb">n3o3.pdb</a></p>
<pre><code>ATOM      1  O1  NN3     1       1.562   0.180  -1.368      
ATOM      2  N1  NN3     1       1.316   0.936  -0.424      
ATOM      3  N2  NN3     1       2.364   1.026   0.488      
ATOM      4  O2  NN3     1       3.472   0.429   0.421      
ATOM      5  N3  NN3     2       1.958   1.906   1.488      
ATOM      6  O3  NN3     2       2.814   2.068   2.361      
END
</code></pre>
<h2 id="qmmm_appendix-generation-of-new-fragment-files">Generation of new fragment files<a class="headerlink" href="#qmmm_appendix-generation-of-new-fragment-files" title="Permanent link">&para;</a></h2>
<p>While setting up QM/MM calculations is often necessary to generate new
fragment files for the molecules that are not available as part of
standard set. The IMPORTANT assumption here is that these new
molecules/residues will be part of OQM region, and as a result only
minimum information needs to be provided to include them in QM/MM
calculations.</p>
<p>First we must ensure that we have a proper PDB format for our as
discussed in the <a href="#qmmm_appendix-generation-of-the-proper-pdb-file">Generation of the proper PDB file
</a> section. As a concrete
example we will start with N3O3 example that was discussed there as well</p>
<pre><code>ATOM      1  O1  NN3     1       1.562   0.180  -1.368      
ATOM      2  N1  NN3     1       1.316   0.936  -0.424      
ATOM      3  N2  NN3     1       2.364   1.026   0.488      
ATOM      4  O2  NN3     1       3.472   0.429   0.421      
ATOM      5  N3  NN3     1       1.958   1.906   1.488      
ATOM      6  O3  NN3     1       2.814   2.068   2.361      
END
</code></pre>
<p>We have residue named NN3 and therefore looking to construct fragment
file <code>NN3.frg</code>. The best way to do it is to run a simple prepare job</p>
<pre><code>start n3o3  

prepare  
source n3o3.pdb  
new_top new_seq  
new_rst  
modify segment 1 quantum   
update lists  
ignore  
write n3o3_ref.pdb  
write n3o3_ref.rst  
end  

task prepare
</code></pre>
<p>This prepare job will necessarily fail because NN3.frg is not yet
available:</p>
<pre><code>Created fragment                      ./NN3.frg_TMP  

Unresolved atom types in fragment NN3  


 **********  
 *   0: pre_mkfrg failed 9999  
 **********
</code></pre>
<p>As part of this process skeleton fragment file (<code>NN3.frg_TMP</code>) will be
generated that can be modified into the final correct form. Let us take
a look at <code>NN3.frg_TMP</code></p>
<pre><code># This is an automatically generated fragment file  
# Atom types and connectivity were derived from coordinates  
# Atomic partial charges are crude estimates  
# 00/00/00  00:00:00    
#  
$NN3  
    6    1    1    0  
NN3  
    1 O1             0    0    0    1    1    0.000000    0.000000  
    2 N1             0    0    0    1    1    0.000000    0.000000  
    3 N2             0    0    0    1    1    0.000000    0.000000  
    4 O2             0    0    0    1    1    0.000000    0.000000  
    5 N3             0    0    0    1    1    0.000000    0.000000  
    6 O3             0    0    0    1    1    0.000000    0.000000  
    1    2  
    2    3  
    3    4  
    3    5  
    5    6
</code></pre>
<p>The main problem with this fragment file is that there are no atom types
to be found in column 12. What atom type does is to identify what
classical parameters should be assigned to it. Since, as mentioned in
the beginning, we are planing to treat this residue/molecule as part of
QM region, the only classical information needed is VDW parameters. We
will assume that all nitrogens atoms in our molecule can use the same
set of parameters, and the same for oxygens. Therefore we will define
two atom types NX and OX  </p>
<pre><code>$NN3  
    6    1    1    0  
NN3  
    1 O1   OX        0    0    0    1    1    0.000000    0.000000  
    2 N1   NX        0    0    0    1    1    0.000000    0.000000  
    3 N2   NX        0    0    0    1    1    0.000000    0.000000  
    4 O2   OX        0    0    0    1    1    0.000000    0.000000  
    5 N3   NX        0    0    0    1    1    0.000000    0.000000  
    6 O3   OX        0    0    0    1    1    0.000000    0.000000  
    1    2  
    2    3  
    3    4  
    3    5  
    5    6
</code></pre>
<p>and rename resulting file as <a href="NN3.frg">NN3.frg</a>. Please
note that the overall format of the fragment file was preserved and atom
types were entered starting at column 12. Rerunning the same prepare job
moves as a bit further this time</p>
<pre><code> modify segment     1 set 0 quantum  
 Parameter file                        /Users/marat/opt/codes/nwchem-new/src/data/amber_s/amber.par  
 Parameter file                        /Users/marat/opt/codes/nwchem-new/src/data/amber_q/amber.par  
 Parameter file                        /Users/marat/opt/codes/nwchem-new/src/data/amber_x/amber.par  
 Parameter file                        /Users/marat/opt/codes/nwchem-new/src/data/amber_u/amber.par

 Undetermined force field parameters

 Parameters could not be found for atom type     OX   Q  
 Parameters could not be found for atom type     NX   Q

 **********  
 *   0: pre_check failed 9999  
 **********
</code></pre>
<p>complaining now that atom types OX and NX are not defined. This brings
us to the next step of defining new parameter file for our calculation. </p>
<h2 id="qmmm_appendix-generation-of-new-parameter-files">Generation of new parameter files<a class="headerlink" href="#qmmm_appendix-generation-of-new-parameter-files" title="Permanent link">&para;</a></h2>
<p>Continuing with our fragment construction in <a href="#qmmm_appendix-generation-of-new-fragment-files">Generation of new fragment
files</a> section, we now
need to define VDW parameters for our new atom types NX and OX. The best
way to do it is to create <code>amber.par</code> file in the directory where you plan
to rerun final prepare  </p>
<pre><code>Electrostatic 1-4 scaling factor     0.833333  
Relative dielectric constant     1.000000  
Parameters epsilon R*  
#  
Atoms  
NX     14.01000 7.11280E-01 1.82400E-01                            1 1111111111  
      Q       7 3.55640E-01 1.82400E-01  
OX     16.00000 6.35968E-01 1.76830E-01                            1 1111111111  
      Q       8 3.17984E-01 1.76830E-01  
End
</code></pre>
<p>The format of this file is documented in <a href="#qmmm_appendix-format-of-nwchem-parameter-file">Format of NWChem parameter
file</a>. How to actually
choose the appropriate values for VDW parameters is a whole new subject,
which I do not think anybody yet fully addressed. The practical strategy
is to copy from known atom types, which are chemically similar to the
ones in your system. In the case above I copied parameters from standard
AMBER atom types N and OW.</p>
<h2 id="qmmm_appendix-format-of-nwchem-parameter-file">Format of NWChem parameter file<a class="headerlink" href="#qmmm_appendix-format-of-nwchem-parameter-file" title="Permanent link">&para;</a></h2>
<p>The format of NWChem parameter is illustrated on the figure below and
also available as <a href="Parameter-file.pdf">pdf file</a>.</p>
<p><img alt="Format of NWChem parameter file" src="Nwchem-parameter-file.png" /></p>
<h2 id="qmmm_appendix-conversion-of-standard-amber-program-parameter-files">Conversion of standard AMBER program parameter files<a class="headerlink" href="#qmmm_appendix-conversion-of-standard-amber-program-parameter-files" title="Permanent link">&para;</a></h2>
<p>Fortran code that performs conversion from AMBER program parameter file
format to NWChem can be found <a href="amber-helper.tgz">here</a>.
It works by parsing out free format AMBER style parameter file contained
in <code>amber.in</code></p>
<pre><code>MASS  
C  12.01  
CA    12.01  
BOND  
#this is a comment  
C -CA  469.0    1.409           this is also a comment  
C  -   CB     447.0      1.419    
ANGLE  
C -CA-CA    63.0      120.00 another comment  
C -CB-NB    70.0      130.00  
DIHEDRAL  
X -C -CA-X    4   14.50        180.0             2.         intrpol.bsd.on C6H6  
 X - C -  CB -X    4   12.00        180.0             2.         intrpol.bsd.on C6H6  
IMPROPER  
X -CT-N -CT         1.0          180.          2.           JCC,7,(1986),230  
CT -O  - C -OH         10.5         180.           2.  
NONBOND  
 CA 1.9080  0.0860  
C 1.9080  0.0860
</code></pre>
<p>to <strong>fixed format</strong> NWChem style amber.par file  </p>
<pre><code>#Generated amber.par file  
Electrostatic 1-4 scaling factor     0.833333  
Relative dielectric constant     1.000000  
Parameters epsilon R*  
#  
Atoms  
CA     12.01000 3.59824E-01 1.90800E-01                            1 1111111111  
              6 1.79912E-01 1.90800E-01  
C      12.01000 3.59824E-01 1.90800E-01                            1 1111111111  
              6 1.79912E-01 1.90800E-01  
Bonds  
C    -CA      0.14090 3.92459E+05  
C    -CB      0.14190 3.74050E+05  
Angles  
C    -CA   -CA      2.09440 5.27184E+02  
C    -CB   -NB      2.26893 5.85760E+02  
Proper dihedrals  
     -C    -CA   -        3.14159 1.51670E+01    2  
     -C    -CB   -        3.14159 1.25520E+01    2  
Improper dihedrals  
     -CT   -N    -CT      3.14159 4.18400E+00    2  
CT   -O    -C    -OH      3.14159 4.39320E+01    2  
End
</code></pre></section></section><section class="print-page" id="bq" heading-number="13.5"><h1 id="bq-point-charges">Point charges<a class="headerlink" href="#bq-point-charges" title="Permanent link">&para;</a></h1>
<h2 id="bq-overview">Overview<a class="headerlink" href="#bq-overview" title="Permanent link">&para;</a></h2>
<p>The Bq module provides a way to perform QM calculations in the presence of point charges or Bq&rsquo;s,  (as typically referred to in quantum chemistry community). 
Using Bq module versus  geometry block is a recommended way to include point charges in your calculations, in particular if number of charges are big.</p>
<p>The format for including external point charges using the Bq module is shown below, supporting both explicit charge definition in the body of the block and/or loading from external files.</p>
<pre><code> bq [units au|nm|pm|ang...] [namespace]  
   [clear] 
   [force|noforce]  
   [load &lt;file&gt; [charges &lt;chargefile&gt;] [format ix iy iz iq] [units au|nm|pm|ang|...] [ scale &lt;factor&gt; ]]  
   x  y  z  q
   ...  
end  
</code></pre>
<ul>
<li><strong>units</strong> - specify the global units for the coordinates of point charges. Allowed values are au, bohr, nanometers, nm, pm, picometers, angstrom (note that only first 3 characters are important). The default units are angstrom</li>
<li><strong>namespace</strong> - an optional name that can be used to distinguish between potentially several point charge sets. The default value for namespace is &ldquo;default&rdquo;, and only the set that has this namespace value will be actually used in the calculation. If for example different namespace is used, this point charge set will be processed into the run time database but not actually used unless the following set directive is encountered</li>
</ul>
<pre><code>set bq &lt;namespace&gt;
</code></pre>
<p>Here is an example that illustrates this </p>
<pre><code> ...  
 #store point charge in namespace &quot;foo&quot;  
 bq &quot;foo&quot;
  ...  
 end  

 #perform calculation without actually loading charges in foo  
 task dft energy  

 #activate charges in foo  
 set bq foo  
 # now DFT calculation will performed in the presence of charges in foo  
 task dft energy
</code></pre>
<ul>
<li><strong>clear</strong> - this directive erases all the previously specified point charges in a given namespace, prior to new setup (if any).</li>
<li><strong>force</strong> <output_file> | <strong>noforce</strong> this directive triggers|disables calculation of forces on Bq charges. Default value is <strong>noforce</strong>, which disables force calculation. The forces will be written to <output_file> if provided or to <prefix>.bqforce.dat file. The format of the file is</li>
</ul>
<pre><code> #comment line
 fx fy fz
 ...
</code></pre>
<ul>
<li><strong>load</strong> <file> [charges <chargefile> ] [format ix iy iz iq] [units au|nm|pm|<strong>ang</strong>|&hellip;] [ scale <factor> ] - this directive allows to load point charges from external file(s). You can load charges and their coordinates from a single file, or from separate files. The files do not have to follow any specific format, but blank lines and comments (starting with #) will be ignored. The actual specification of how the coordinates/charges are laid out in in the file is given by format keyword. Multiple load directives are a supported.<br />
<strong> <file> - the name of the file where Bq coordinates and charges are stored. <br />
</strong> charges <chargefile> - this optional keyword allows to load charges (NOT the coordinates) from a separate <chargefile>. In this case, only coordinates would be loaded from <file>  </li>
</ul>
<p>** format ix iy iz iq - this optional keyword allows to set the fields (separated by blanks) where x,y,z coordinates and respective charge values are to be found. If a specified field doe nor exist or contains no numerical value, the processing will skip to the next line. The default value for format is &lsquo;&rsquo;&lsquo;2 3 4 5&rsquo;&lsquo;&rsquo;, which will work for the following example (note that the second line will not be processed here)  </p>
<pre><code> #this is a comment
 coordinates are in fields 2,3,4 and charge is field 5
 O  2.384   1.738   1.380  -0.9
 H  2.448   1.608   0.416   0.45
 H  1.560   1.268   1.608   0.45
</code></pre>
<p>** units au|nm|pm|ang|&hellip; - this optional keyword sets the local coordinate units valid only for this particular load directive. Otherwise global unit definition will apply (see above)</p>
<p>** scale <factor> - this optional keyword allows to scale loaded charge values by some factor</p>
<ul>
<li>x y z q  - explicit definition of coordinates and charge values of point charges. These can be mixed in with load directive at will. </li>
</ul>
<pre><code>  start w1  

  BASIS &quot;ao basis&quot; PRINT  
  * library &quot;3-21G&quot;  
  END  

  dft  
   mult 1  
   XC b3lyp  
   iterations 5000  
  end  

  geometry nocenter noautosym units angstrom noautoz print  
  O          2.045   1.011  -1.505  
  H1         1.912   0.062  -1.314  
  H2         1.119   1.318  -1.544  
  end  

  #example of explicit Bq input
  bq  
  2.384   1.738   1.380  -0.9  
  2.448   1.608   0.416   0.45  
  1.560   1.268   1.608   0.45  
  end  

  task dft energy  

  #example of implicit Bq input using load directive  
  bq  
  load bq.xyz format 1 2 3 4  
  end  
  task dft energy  

  #example of loading coordinates and charges separately  
  bq  
  load   bq.xyz charges bq.xyz format 1 2 3 4  
  end  
  task dft energy  

  #example of loading Bq's with default format (1 2 3 4) and scaling charges (to zero)  
  bq  
  load bq.xyz scale 0.0  
  end  
  task dft energy  

  #example of mixed Bq input  
  bq  
  load bqO.xyz format 2 3 4 6  
  2.448   1.608   0.416   0.45  
  1.560   1.268   1.608   0.45  
  end  
  task dft energy  

  #example of erasing Bq's  
  bq  
  clear  
  end  
  task dft energy  

  #example of storing Bq's in custom namespace (not activated)  
  bq marat  
  load bq.xyz format 1 2 3 4  
  end  
  task dft energy  

  #example of activating Bq's stored in custom namespace  
  set bq marat  

  task dft energy  
</code></pre>
<ul>
<li><strong>Increasing the limit on the number of bq  charges</strong>
  There is an internal limit of 25000 bq charges. To increase this use the following set directive in the input file
  set bq:max_nbq 30000</li>
</ul></section><section class="print-page" id="1d-rism" heading-number="13.6"><h1 id="1d-rism-1-d-rism">1-D RISM<a class="headerlink" href="#1d-rism-1-d-rism" title="Permanent link">&para;</a></h1>
<h2 id="1d-rism-overview">Overview<a class="headerlink" href="#1d-rism-overview" title="Permanent link">&para;</a></h2>
<p>The 1D-RISM module in NWChem provides description of solvated systems
following one-dimensional reference interaction site of model of
Chandler and Anderson. Similar to ab-initio density-functional theory,
1D-RISM can be thought of as an approach where discrete particle
representation of solvent degrees of freedom is replaced by average
density field. Unlike traditional continuum solvation model, this
density based representation is inherently inhomogenous and incorporates
specific molecular features of the solvent. In the current
implementation, 1D-RISM is not directly coupled to QM calculations but
presumed to be used as a post processing step after QM calculations
which provide ESP point charges for a given solute geometry.</p>
<p>Then parameters for 1D-RISM calculations are defined in the rism input
block</p>
<pre><code>rism
soluteconfiguration&lt;filename&gt;
vdw[rule&lt;arithmetic|geometric&gt;]parameters&lt;filename&gt;
[temp&lt;floatdefault298.15&gt;]
[closure&lt;hnc|kh&gt;]
end
</code></pre>
<p>At this point energy task is supported, which is invoked using standard
directive</p>
<pre><code>taskrismenergy
</code></pre>
<ul>
<li><strong>solute configuration</strong> - points to the file that contains
    information about the solute geometry. charges, and atom type
    mapping. The format is similar to xyz style with additional fields
    that specify charge and atom type. The atom type maps back to the
    vdw parameter file (see below). The example file is shown below</li>
</ul>
<pre><code>7

O1-1.0921110.7334611.237573-1.104415O
O20.758765-0.2016870.473908-1.043019O
C1-0.2129541.568653-0.833617-0.474263C1
C2-0.1742050.6304320.3571351.276672C2
H10.3606361.160405-1.6688590.102898H
H20.2424192.521128-0.5319520.118979H
H3-1.2439671.772778-1.1395470.123148H
</code></pre>
<ul>
<li><strong>vdw</strong> - defines van der waals parameters<ul>
<li><strong>rule</strong> - optional setting that specifies combination rule,
    defaults to &ldquo;arithmetic&rdquo;</li>
<li><strong>parameters</strong> - points to the file that contains vdw parameters
    for the system. Example file is shown below (note comments in
    the file)</li>
</ul>
</li>
</ul>
<pre><code>#VanderWaalsparametersfileforRISM
#typesigma(Angstrom)epsilon(kj/mol)
C0.3400E+010.3601E+00
H0.2600E+010.0628E-00
</code></pre>
<ul>
<li><strong>temp</strong> - defines temperature of the system with default value of
    298.15</li>
<li><strong>closure</strong> - specifies choice of closure</li>
</ul>
<p>Upon completion of the run, the resulting radial distribution functions
are saved into rdf_out.data file.</p>
<p>The computed chemical potentials in both HNC and gaussian approximations
are written in the output file.</p>
<p>Here is the complete example input file for solvated calculation of
acetic acid.</p>
<pre><code>echo
startrism

memoryglobal40mbstack23mbheap5mb

rism
closurekh
temp298
vdwrulearithmeticparametersvdw.par
soluteconfigurationsolute2.data
end

taskenergyrism
</code></pre>
<p>solute2.data
file</p>
<pre><code>8

O10.15566663-0.860695081.9256322-0.7323104568123959O1
O22.31302544-0.615505201.32869265-0.7721248369124809O2
C10.69252260-1.26942616-0.34814880-0.4444201659837397C1
C21.15967680-0.885318051.026428401.004561861052242C2
H10.22862001-2.26160688-0.310111320.1303963270585546H
H2-0.08170478-0.56654621-0.678346920.1513209389506027H
H31.53138139-1.28351200-1.046441340.1454517042730594H
H40.48680931-0.663628202.835512880.5171246283741536H4
</code></pre>
<p>vdw.par file</p>
<pre><code>O13.06600.8809
O22.96000.8792
C13.40000.4580
C23.40000.3601
H2.11500.0657
H40.80000.1926
</code></pre></section></section>
                    <section class='print-page md-section' id='section-14' heading-number='14'>
                        <h1>Potential Energy Surface Analysis<a class='headerlink' href='#section-14' title='Permanent link'></a>
                        </h1>
                    <section class="print-page" id="geometry-optimization" heading-number="14.1"><h1 id="geometry-optimization-geometry-optimization">Geometry Optimization<a class="headerlink" href="#geometry-optimization-geometry-optimization" title="Permanent link">&para;</a></h1>
<h2 id="geometry-optimization-geometry-optimization-with-driver">Geometry Optimization with DRIVER<a class="headerlink" href="#geometry-optimization-geometry-optimization-with-driver" title="Permanent link">&para;</a></h2>
<p>The DRIVER module is one of two drivers (also see documentation on
<a href="#geometry-optimization-geometry-optimization-with-stepper">STEPPER</a>)
to perform a geometry optimization function on the molecule defined by
input using the <a href="#geometry">GEOMETRY</a> directive.
Geometry optimization is either an energy minimization or a transition
state optimization. The algorithm programmed in DRIVER is a quasi-newton
optimization with line searches and approximate energy Hessian updates.</p>
<p>DRIVER is selected by default out of the two available modules to
perform geometry optimization. In order to force use of DRIVER (e.g.,
because a previous optimization used STEPPER) provide a DRIVER input
block (below) &ndash; even an empty block will force use of DRIVER.</p>
<p>Optional input for this module is specified within the compound
directive,</p>
<pre><code> DRIVER   
   (LOOSE || DEFAULT || TIGHT)  
   GMAX &lt;real value&gt;  
   GRMS &lt;real value&gt;  
   XMAX &lt;real value&gt;  
   XRMS &lt;real value&gt;  
   EPREC &lt;real eprec default 1e-7&gt;  
   TRUST &lt;real trust default 0.3&gt;  
   SADSTP &lt;real sadstp default 0.1&gt;  
   CLEAR  
   REDOAUTOZ  
   INHESS &lt;integer inhess default 0&gt;  
   (MODDIR || VARDIR) &lt;integer dir default 0&gt; 
   (FIRSTNEG || NOFIRSTNEG)  
   MAXITER &lt;integer maxiter default 20&gt;  
   BSCALE &lt;real BSCALE default 1.0&gt;` 
   ASCALE &lt;real ASCALE default 0.25&gt;  
   TSCALE &lt;real TSCALE default 0.1&gt; 
   HSCALE &lt;real HSCALE default 1.0&gt; 
      PRINT ... 
   XYZ &lt;string xyz default *file_prefix*&gt;]  
   NOXYZ  
   SOCKET (UNIX || IPI_CLIENT) &lt;string socketname default (see input description)&gt;  
 END
</code></pre>
<p>On each optimization step a line search is performed. To speed up
calculations (up to two times), it may be beneficial to turn off the
line search using following directive:</p>
<pre><code> set driver:linopt 0
</code></pre>
<h3 id="geometry-optimization-convergence-criteria">Convergence criteria<a class="headerlink" href="#geometry-optimization-convergence-criteria" title="Permanent link">&para;</a></h3>
<pre><code>   (LOOSE || DEFAULT || TIGHT)  
   GMAX &lt;real value&gt;  
   GRMS &lt;real value&gt;  
   XMAX &lt;real value&gt;  
   XRMS &lt;real value&gt;
</code></pre>
<p>The defaults may be used, or the directives LOOSE, DEFAULT, or TIGHT
specified to use standard sets of values, or the individual criteria
adjusted. All criteria are in atomic units. GMAX and GRMS control the
maximum and root mean square gradient in the coordinates being used
(Z-matrix, redundant internals, or Cartesian). XMAX and XRMS control the
maximum and root mean square of the Cartesian step.</p>
<table>
<thead>
<tr>
<th></th>
<th>LOOSE</th>
<th>DEFAULT</th>
<th>TIGHT</th>
</tr>
</thead>
<tbody>
<tr>
<td>GMAX</td>
<td>0.00450</td>
<td>0.00045</td>
<td>0.000015</td>
</tr>
<tr>
<td>GRMS</td>
<td>0.00300</td>
<td>0.00030</td>
<td>0.00001</td>
</tr>
<tr>
<td>XMAX</td>
<td>0.01800</td>
<td>0.00180</td>
<td>0.00006</td>
</tr>
<tr>
<td>XRMS</td>
<td>0.01200</td>
<td>0.00120</td>
<td>0.00004</td>
</tr>
</tbody>
</table>
<p>Note that GMAX and GRMS used for convergence of geometry may
significantly vary in different coordinate systems such as Z-matrix,
redundant internals, or Cartesian. The coordinate system is defined in
the input file (default is Z-matrix). Therefore the choice of coordinate
system may slightly affect converged energy. Although in most cases XMAX
and XRMS are last to converge which are always done in Cartesian
coordinates, which insures convergence to the same geometry in different
coordinate systems.</p>
<p>The old criterion may be recovered with the input</p>
<pre><code>  gmax 0.0008; grms 1; xrms 1; xmax 1
</code></pre>
<h3 id="geometry-optimization-available-precision">Available precision<a class="headerlink" href="#geometry-optimization-available-precision" title="Permanent link">&para;</a></h3>
<pre><code>   EPREC &lt;real eprec default 1e-7&gt;
</code></pre>
<p>In performing a line search the optimizer must know the precision of the
energy (this has nothing to do with convergence criteria). The default
value of 1e-7 should be adjusted if less, or more, precision is
available. Note that the default EPREC for DFT calculations is 5e-6
instead of 1e-7.</p>
<h3 id="geometry-optimization-controlling-the-step-length">Controlling the step length<a class="headerlink" href="#geometry-optimization-controlling-the-step-length" title="Permanent link">&para;</a></h3>
<pre><code>   TRUST &lt;real trust default 0.3&gt;
   SADSTP &lt;real sadstp default 0.1&gt;
</code></pre>
<p>A fixed trust radius (trust) is used to control the step during
minimizations, and is also used for modes being minimized during
saddle-point searches. It defaults to 0.3 for minimizations and 0.1 for
saddle-point searches. The parameter sadstp is the trust radius used for
the mode being maximized during a saddle-point search and defaults to
0.1.</p>
<h3 id="geometry-optimization-maximum-number-of-steps">Maximum number of steps<a class="headerlink" href="#geometry-optimization-maximum-number-of-steps" title="Permanent link">&para;</a></h3>
<pre><code>   MAXITER &lt;integer maxiter default 40&gt;
</code></pre>
<p>By default at most 40 geometry optimization steps will be taken, but
this may be modified with this directive.</p>
<h3 id="geometry-optimization-discard-restart-information">Discard restart information<a class="headerlink" href="#geometry-optimization-discard-restart-information" title="Permanent link">&para;</a></h3>
<pre><code>   CLEAR
</code></pre>
<p>By default Driver reuses Hessian information from a previous
optimization, and, to facilitate a restart also stores which mode is
being followed for a saddle-point search. This option deletes all
restart data.</p>
<h3 id="geometry-optimization-regenerate-internal-coordinates">Regenerate internal coordinates<a class="headerlink" href="#geometry-optimization-regenerate-internal-coordinates" title="Permanent link">&para;</a></h3>
<pre><code>   REDOAUTOZ
</code></pre>
<p>Deletes Hessian data and regenerates internal coordinates at the current
geometry. Useful if there has been a large change in the geometry that
has rendered the current set of coordinates invalid or non-optimal.</p>
<h3 id="geometry-optimization-initial-hessian">Initial Hessian<a class="headerlink" href="#geometry-optimization-initial-hessian" title="Permanent link">&para;</a></h3>
<pre><code>   INHESS &lt;integer inhess default 0&gt;
</code></pre>
<ul>
<li><code>0</code> = Default &hellip; use restart data if available, otherwise use
    diagonal guess.</li>
<li><code>1</code> = Use diagonal initial guess.</li>
<li><code>2</code> = Use restart data if available, otherwise transform Cartesian
    Hessian from previous frequency calculation.</li>
</ul>
<p>In addition, the diagonal elements of the initial Hessian for internal
coordinates may be scaled using separate factors for bonds, angles and
torsions with the following</p>
<pre><code>   BSCALE &lt;real bscale default 1.0&gt;  
   ASCALE &lt;real ascale default 0.25&gt;  
   TSCALE &lt;real tscale default 0.1&gt;
</code></pre>
<p>These values typically give a two-fold speedup over unit values, based
on about 100 test cases up to 15 atoms using 3-21g and 6-31g* SCF.
However, if doing many optimizations on physically similar systems it
may be worth fine tuning these parameters.</p>
<p>Finally, the entire Hessian from any source may be scaled by a factor
using the directive</p>
<pre><code>   HSCALE &lt;real hscale default 1.0&gt;
</code></pre>
<p>It might be of utility, for instance, when computing an initial Hessian
using SCF to start a large MP2 optimization. The SCF vibrational modes
are expected to be stiffer than the MP2, so scaling the initial Hessian
by a number less than one might be beneficial.</p>
<h3 id="geometry-optimization-mode-or-variable-to-follow-to-saddle-point">Mode or variable to follow to saddle point<a class="headerlink" href="#geometry-optimization-mode-or-variable-to-follow-to-saddle-point" title="Permanent link">&para;</a></h3>
<pre><code>   (MODDIR || VARDIR) &lt;integer dir default 0&gt;  
   (FIRSTNEG || NOFIRSTNEG)
</code></pre>
<p>When searching for a transition state the program, by default, will take
an initial step uphill and then do mode following using a fuzzy maximum
overlap (the lowest eigen-mode with an overlap with the previous search
direction of 0.7 times the maximum overlap is selected). Once a negative
eigen-value is found, that mode is followed regardless of overlap.</p>
<p>The initial uphill step is appropriate if the gradient points roughly in
the direction of the saddle point, such as might be the case if a
constrained optimization was performed at the starting geometry.
Alternatively, the initial search direction may be chosen to be along a
specific internal variable (using the directive <code>VARDIR</code>) or along a
specific eigen-mode (using <code>MODDIR</code>). Following a variable might be
valuable if the initial gradient is either very small or very large.
Note that the eigen-modes in the optimizer have next-to-nothing to do
with the output from a frequency calculation. You can examine the
eigen-modes used by the optimizer with</p>
<pre><code>  driver; print hvecs; end
</code></pre>
<p>The selection of the first negative mode is usually a good choice if the
search is started in the vicinity of the transition state and the
initial search direction is satisfactory. However, sometimes the first
negative mode might not be the one of interest (e.g., transverse to the
reaction direction). If <code>NOFIRSTNEG</code> is specified, the code will not take
the first negative direction and will continue doing mode-following
until that mode goes negative.</p>
<h3 id="geometry-optimization-optimization-history-as-xyz-files">Optimization history as XYZ files<a class="headerlink" href="#geometry-optimization-optimization-history-as-xyz-files" title="Permanent link">&para;</a></h3>
<pre><code>   XYZ [&lt;string xyz default $fileprefix&gt;] 
   NOXYZ
</code></pre>
<p>The XYZ directive causes the geometry at each step (but not intermediate
points of a line search) to be output into separate files in the
permanent directory in XYZ format. The optional string will prefix the
filename. The NOXYZ directive turns this off.</p>
<p>For example, the input</p>
<pre><code>   driver; xyz test; end
</code></pre>
<p>will cause files test-000.xyz, test-001.xyz, &hellip; to be created in the
permanent directory.</p>
<p>The script rasmolmovie in the NWChem contrib directory can be used to
turn these into an animated GIF movie.</p>
<h3 id="geometry-optimization-i-pi-socket-communication">i-PI Socket communication<a class="headerlink" href="#geometry-optimization-i-pi-socket-communication" title="Permanent link">&para;</a></h3>
<pre><code>   SOCKET (UNIX | IPI_CLIENT) &lt;socketname&gt; [RETRIES &lt;integer&gt;] [DELAY &lt;float&gt;]
</code></pre>
<p>The <code>SOCKET</code> directive enables NWChem to communicate with other software
packages &ndash; such as <a href="http://ipi-code.org/">i-PI</a> or
<a href="https://wiki.fysik.dtu.dk/ase/ase/calculators/socketio/socketio.html">ASE</a> or <a href="https://github.com/TheochemUI/eOn/pull/244">EON</a> &ndash; via the i-PI socket protocol.</p>
<p>Communication is done either over Unix sockets (<code>SOCKET UNIX</code>) or IP
sockets (<code>SOCKET IPI_CLIENT</code>):</p>
<ul>
<li>Unix sockets - NWChem will create and bind to a UNIX socket
    file located at <code>/tmp/ipi_&lt;socketname&gt;</code>. If not specified,
    <code>&lt;socketname&gt;</code> will default to <code>nwchem</code>.</li>
<li>IP sockets - NWChem will bind to the IP address and port
    specified by <code>&lt;socketname&gt;</code>. If not specified, <code>&lt;socketname&gt;</code>
    will default to <code>127.0.0.1:31415</code>.</li>
</ul>
<p>The behavior of the client when first attempting to connect to the server can be
controlled with two optional keywords:</p>
<ul>
<li><code>RETRIES &lt;integer&gt;</code>: Specifies the maximum number of times the client will try
  to connect to the server socket before exiting with an error. If not provided,
  this defaults to 30 tries.</li>
<li><code>DELAY &lt;integer&gt;</code>: Specifies the time in seconds to wait between failed
  connection attempts. If not provided, this defaults to 2 seconds.</li>
</ul>
<p>The <code>SOCKET</code> directive is only useful when used in conjunction with other
software packages that support communication via the i-PI socket
protocol. For more information, see the
<a href="https://docs.ipi-code.org/distributed.html#communication-protocol">i-PI documentation</a>.</p>
<h3 id="geometry-optimization-print-options">Print options<a class="headerlink" href="#geometry-optimization-print-options" title="Permanent link">&para;</a></h3>
<p>The UNIX command   </p>
<pre><code>grep '^@' \&lt; output
</code></pre>
<p>will extract a pretty table summarizing the optimization.  </p>
<p>If you specify the NWChem input</p>
<pre><code>     scf; print none; end  
     driver; print low; end  
     task scf optimize
</code></pre>
<p>you&rsquo;ll obtain a pleasantly terse output.</p>
<p>For more control, these options for the standard print directive are
recognized</p>
<ul>
<li><code>debug</code> - prints a large amount of data. Don&rsquo;t use in parallel.</li>
<li><code>high</code> - print the search direction in internals</li>
<li><code>default</code> - prints geometry for each major step (not during the line
    search), gradient in internals (before and after application of
    constraints)</li>
<li><code>low</code> - prints convergence and energy information. At convergence
    prints final geometry, change in internals from initial geometry</li>
</ul>
<p>and these specific print options</p>
<ul>
<li><code>finish</code> (low) - print geometry data at end of calculation</li>
<li><code>bonds</code> (default) - print bonds at end of calculation</li>
<li><code>angles</code> (default) - print angles at end of calculation</li>
<li><code>hvecs</code> (never) - print eigen-values/vectors of the Hessian</li>
<li><code>searchdir</code> (high) - print the search direction in internals</li>
<li><code>"internal gradient"</code> (default) - print the gradient in internals</li>
<li><code>sadmode</code> (default) - print the mode being followed to the saddle point</li>
</ul>
<h2 id="geometry-optimization-geometry-optimization-with-stepper">Geometry Optimization with STEPPER<a class="headerlink" href="#geometry-optimization-geometry-optimization-with-stepper" title="Permanent link">&para;</a></h2>
<p>The STEPPER module performs a search for critical points on the
potential energy surface of the molecule defined by input using the
<a href="#geometry">GEOMETRY</a> directive. Since STEPPER is
not the primary geometry optimization module in NWChem the compound
directive is required; the
<a href="#geometry-optimization-geometry-optimization-with-driver">DRIVER</a> module is the
default. Input for this module is specified within the compound
directive,</p>
<pre><code> STEPPER  
   ...  
 END
</code></pre>
<p>The presence of the <code>STEPPER</code> compound directive automatically turns off
the default geometry optimization tool <code>DRIVER</code>. Input specified for the
<code>STEPPER</code> module must appear in the input file after the <code>GEOMETRY</code>
directive, since it must know the number of atoms that are to be used in
the geometry optimization. In the current version of NWChem, STEPPER can
be used only with geometries that are defined in Cartesian coordinates.
<code>STEPPER</code> removes translational and rotational components before
determining the step direction (5 components for linear systems and 6
for others) using a standard Eckart algorithm. The default initial guess
nuclear Hessian is the identity matrix.</p>
<p>The default in <code>STEPPER</code> is to minimize the energy as a function of the
geometry with a maximum of 20 geometry optimization iterations. When
this is the desired calculation, no input is required other than the
<code>STEPPER</code> compound directive. However, the user also has the option of
defining different tasks for the <code>STEPPER</code> module, and can vary the number
of iterations and the convergence criteria from the default values. The
input for these options is described in the following sections.</p>
<h3 id="geometry-optimization-min-and-ts-minimum-or-transition-state-search">MIN and TS: Minimum or transition state search<a class="headerlink" href="#geometry-optimization-min-and-ts-minimum-or-transition-state-search" title="Permanent link">&para;</a></h3>
<p>The default is for STEPPER to minimize the energy with respect to the
geometry of the system. This default behavior may be forced with the
directive</p>
<pre><code> MIN
</code></pre>
<p>STEPPER can also be used to find the transition state by following the
lowest eigenvector of the nuclear Hessian. This is usually invoked by
using the saddle keyword on the <a href="#task">TASK
directive</a>, but it may also be
selected by specifying the directive</p>
<pre><code> TS
</code></pre>
<p>in the STEPPER input.</p>
<h3 id="geometry-optimization-track-mode-selection">TRACK: Mode selection<a class="headerlink" href="#geometry-optimization-track-mode-selection" title="Permanent link">&para;</a></h3>
<p><code>STEPPER</code> has the ability to <em>track</em> a specific mode during an
optimization for a transition state search, the user can also have the
module track the eigenvector corresponding to a specific mode. This is
done by specifying the directive</p>
<pre><code> TRACK [nmode &lt;integer nmode default 1&gt;]
</code></pre>
<p>The keyword <code>TRACK</code> tells <code>STEPPER</code> to track the eigenvector corresponding
to the integer value of <nmode> during a transition state walk. (Note:
this input is invalid for a minimization walk since following a specific
eigenvector will not necessarily give the desired local minimum.) The
step is constructed to go up in energy along the nmode eigenvector and
down in all other degrees of freedom.</p>
<h3 id="geometry-optimization-maxiter-maximum-number-of-steps">MAXITER: Maximum number of steps<a class="headerlink" href="#geometry-optimization-maxiter-maximum-number-of-steps" title="Permanent link">&para;</a></h3>
<p>In most applications, 20 stepper iterations will be sufficient to obtain
the energy minimization. However, the user has the option of specifying
the maximum number of iterations allowed, using the input line,</p>
<pre><code> MAXITER &lt;integer maxiter default 20&gt;
</code></pre>
<p>The value specified for the integer <maxiter> defines the maximum number
of geometry optimization steps. The geometry optimization will restart
automatically.</p>
<h3 id="geometry-optimization-trust-trust-radius">TRUST: Trust radius<a class="headerlink" href="#geometry-optimization-trust-trust-radius" title="Permanent link">&para;</a></h3>
<p>The size of steps that can be taken in STEPPER is controlled by the
trust radius which has a default value of 0.1. Steps are constrained to
be no larger than the trust radius. The user has the option of
overriding this default using the keyword TRUST, with the following
input line,</p>
<pre><code> TRUST &lt;real radius default 0.1&gt;
</code></pre>
<p>The larger the value specified for the variable radius, the larger the
steps that can be taken by <code>STEPPER</code>. Experience has shown that for larger
systems (i.e., those with 20 or more atoms), a value of 0.5, or greater,
usually should be entered for <radius>.</p>
<h3 id="geometry-optimization-convggm-convgg-and-convge-convergence-criteria">CONVGGM, CONVGG and CONVGE: Convergence criteria<a class="headerlink" href="#geometry-optimization-convggm-convgg-and-convge-convergence-criteria" title="Permanent link">&para;</a></h3>
<p>Three convergence criteria can be specified explicitly for the STEPPER
calculations. The keyword <code>CONVGGM</code> allows the user to specify the
convergence tolerance for the largest component of the gradient. This is
the primary convergence criterion, as per the default settings, although
all three criteria are in effect. this default setting is consistent
with the other optimizer module <code>DRIVER</code>. The input line for <code>CONVGGM</code> has
the following form,</p>
<pre><code>  CONVGGM &lt;real convggm default 8.0d-04&gt;
</code></pre>
<p>The keyword <code>CONVGG</code> allows the user to specify the convergence tolerance
for the gradient norm for all degrees of freedom. The input line is of
the following form,</p>
<pre><code>  CONVGG &lt;real convgg default 1.0d-02&gt;
</code></pre>
<p>The entry for the real variable <code>&lt;convgg&gt;</code> should be approximately equal
to the square root of the energy convergence tolerance.</p>
<p>The energy convergence tolerance is the convergence criterion for the
energy difference in the geometry optimization in <code>STEPPER</code>. It can be
specified by input using a line of the following form,</p>
<pre><code>  CONVGE &lt;real convge default 1.0d-04&gt;
</code></pre>
<h3 id="geometry-optimization-backstepping-in-stepper">Backstepping in STEPPER<a class="headerlink" href="#geometry-optimization-backstepping-in-stepper" title="Permanent link">&para;</a></h3>
<p>If a step taken during the optimization is too large (e.g., the step
causes the energy to go up for a minimization or down for a transition
state search), the <code>STEPPER</code> optimizer will automatically &ldquo;backstep&rdquo;
and correct the step based on information prior to the faulty step. If
you have an optimization that &ldquo;backsteps&rdquo; frequently then the initial
trust radius should most likely be decreased.</p>
<h3 id="geometry-optimization-initial-nuclear-hessian-options">Initial Nuclear Hessian Options<a class="headerlink" href="#geometry-optimization-initial-nuclear-hessian-options" title="Permanent link">&para;</a></h3>
<p>Stepper uses a modified Fletcher-Powell algorithm to find the transition
state or energy minimum on the potential energy hypersurface.<br />
There are
two files left in the user&rsquo;s permanent directory that are used to
provide an initial hessian to the critical point search algorithm. If
these files do not exist then the default is to use a unit matrix as the
initial hessian.<br />
Once Stepper executes it generates a binary dump file
by the name of <code>name.stpr41</code> which will be used on all subsequent stepper
runs and modified with the current updated hessian. The default file
prefix is the &ldquo;name&rdquo; that is used (see
<a href="#start_restart">START</a>).
It also stores the information for the last valid step in case the
algorithm must take a <a href="#geometry-optimization-backstepping-in-stepper">&ldquo;backstep&rdquo;</a>.
This file is the working data store for all stepper-based optimizations.
This file is never deleted by default and is the first source of an
initial hessian.<br />
The second source of an initial hessian is an ASCII file
that contains the lower triangular values of the initial hessian. This
is stored in file <code>name.hess</code>, where &ldquo;name&rdquo; is again the default file
prefix. This is the second source of an initial hessian and is the
method used to incorporate an initial hessian from any other source
(e.g., another ab initio code, a molecular mechanics code, etc.,). To
get a decent starting hessian at a given point you can use the task
specification task scf hessian, with a smaller basis set, which will by
default generate the <code>name.hess</code> file. Then you may define your basis set
of choice and proceed with the optimization you desire.</p></section><section class="print-page" id="hessians-and-vibrational-frequencies" heading-number="14.2"><h1 id="hessians-and-vibrational-frequencies-hessians">Hessians<a class="headerlink" href="#hessians-and-vibrational-frequencies-hessians" title="Permanent link">&para;</a></h1>
<h2 id="hessians-and-vibrational-frequencies-overview">Overview<a class="headerlink" href="#hessians-and-vibrational-frequencies-overview" title="Permanent link">&para;</a></h2>
<p>This section relates to the computation of analytic hessians which are
available for open and closed shell SCF, except ROHF and for closed
shell and unrestricted open shell DFT <sup id="hessians-and-vibrational-frequencies-fnref_footnote_1"><a class="footnote-ref" href="#hessians-and-vibrational-frequencies-fn_footnote_1">1</a></sup>. Analytic hessians are not
currently available for SCF or DFT calculations relativistic
all-electron methodologies or for charge fitting with DFT. The current
algorithm is fully in-core and does not use symmetry.</p>
<p>There is no required input for the Hessian module. This module only
impacts the hessian calculation. For options for calculating the
frequencies, please see the <a href="#vibration">Vibrational</a> module.</p>
<h2 id="hessians-and-vibrational-frequencies-hessian-module-input">Hessian Module Input<a class="headerlink" href="#hessians-and-vibrational-frequencies-hessian-module-input" title="Permanent link">&para;</a></h2>
<p>All input for the Hessian Module is optional since the default
definitions are usually correct for most purposes. The generic module
input begins with hessian and has the form:</p>
<pre><code> hessian  
   thresh &lt;real tol default 1d-6&gt;
   print ... 
   profile  
 end
</code></pre>
<h3 id="hessians-and-vibrational-frequencies-defining-the-wavefunction-threshold">Defining the wavefunction threshold<a class="headerlink" href="#hessians-and-vibrational-frequencies-defining-the-wavefunction-threshold" title="Permanent link">&para;</a></h3>
<p>You may modify the default threshold for the wavefunction. This keyword
is identical to
<a href="#hartree-fock-theory-for-molecules-thresh-convergence-threshold">THRESH</a>
in the SCF, and the
<a href="#density-functional-theory-for-molecules-convergence-scf-convergence-control">CONVERGENCE</a>
gradient in the DFT. The usual defaults for the convergence of the
wavefunction for single point and gradient calculations is generally not
tight enough for analytic hessians. Therefore, the hessian, by default,
tightens these up to 1d-6 and runs an additional energy point if needed.
If, during an analytic hessian calculation, you encounter an error:</p>
<pre><code> cphf_solve:the available MOs do not satisfy the SCF equations
</code></pre>
<p>the convergence criteria of the wavefunction generally needs to be
tightened.</p>
<h3 id="hessians-and-vibrational-frequencies-profile">Profile<a class="headerlink" href="#hessians-and-vibrational-frequencies-profile" title="Permanent link">&para;</a></h3>
<p>The PROFILE keyword provides additional information concerning the
computation times of different sections of the hessian code. Summary
information is given about the maximum, minimum and average times that a
particular section of the code took to complete. This is normally only
useful for developers.</p>
<h3 id="hessians-and-vibrational-frequencies-print-control">Print Control<a class="headerlink" href="#hessians-and-vibrational-frequencies-print-control" title="Permanent link">&para;</a></h3>
<p>Known controllable print options are shown in the table
below:  </p>
<table>
<thead>
<tr>
<th>Name</th>
<th>Print Level</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>hess_follow</code></td>
<td>high</td>
<td>more information about where the calculation is</td>
</tr>
<tr>
<td><code>cphf_cont</code></td>
<td>debug</td>
<td>detailed CPHF information</td>
</tr>
<tr>
<td><code>nucdd_cont</code></td>
<td>debug</td>
<td>detailed nuclear contribution information</td>
</tr>
<tr>
<td><code>onedd_cont</code></td>
<td>debug</td>
<td>detailed one electron contribution information</td>
</tr>
<tr>
<td><code>twodd_cont</code></td>
<td>debug</td>
<td>detailed two electron contribution information</td>
</tr>
<tr>
<td><code>fock_xc</code></td>
<td>debug</td>
<td>detailed XC information during the fock builds</td>
</tr>
</tbody>
</table>
<h1 id="hessians-and-vibrational-frequencies-vibrational-frequencies">Vibrational frequencies<a class="headerlink" href="#hessians-and-vibrational-frequencies-vibrational-frequencies" title="Permanent link">&para;</a></h1>
<p>The nuclear hessian which is used to compute the vibrational frequencies
can be computed by finite difference for any ab initio wave-function
that has analytic gradients or by analytic methods for SCF and DFT (see
<a href="#hessians-and-vibrational-frequencies-overview">Hessians</a> for details). The appropriate nuclear
hessian generation algorithm is chosen based on the user input when TASK
<theory> frequencies is the task directive.</p>
<p>The vibrational package was integrated from the Utah Messkit and can use
any nuclear hessian generated from the driver routines, finite
difference routines or any analytic hessian modules. There is no
required input for the &ldquo;VIB&rdquo; package. VIB computes the Infra Red
frequencies and intensities for the computed nuclear hessian and the
&ldquo;projected&rdquo; nuclear hessian. The VIB module projects out the
translations and rotations of the nuclear hessian using the standard
Eckart projection algorithm. It also computes the zero point energy for
the molecular system based on the frequencies obtained from the
projected hessian.</p>
<p>The default mass of each atom is used unless an alternative mass is
provided via the <a href="#geometry">geometry</a> input or
redefined using the vibrational module input. The default mass is the
mass of the most abundant isotope of each element. If the abundance was
roughly equal, the mass of the isotope with the longest half life was
used.</p>
<p>In addition, the vibrational analysis is given at the default standard
temperature of 298.15 degrees.</p>
<h2 id="hessians-and-vibrational-frequencies-vibrational-module-input">Vibrational Module Input<a class="headerlink" href="#hessians-and-vibrational-frequencies-vibrational-module-input" title="Permanent link">&para;</a></h2>
<p>All input for the Vibrational Module is optional since the default
definitions will compute the frequencies and IR intensities. The generic
module input can begin with vib, freq, frequency and has the form:</p>
<pre><code> {freq || vib || frequency}` 
   [reuse [&lt;string hessian_filename&gt;]]  
   [mass &lt;integer lexical_index&gt; &lt;real new_mass&gt;]  
   [mass &lt;string tag_identifier&gt; &lt;real new_mass&gt;]  
   [{temp || temperature} &lt;integer number_of_temperatures&gt;\ 
         &lt;real temperature1 temperature2 ...&gt;] 
   [animate [&lt;real step_size_for_animation&gt;]]  
   [fd_delta [&lt;real step_size_for_fd_hessian&gt;]]  
   [filename &lt;string file_set_name&gt; [overwrite]]  
 end
</code></pre>
<h3 id="hessians-and-vibrational-frequencies-hessian-file-reuse">Hessian File Reuse<a class="headerlink" href="#hessians-and-vibrational-frequencies-hessian-file-reuse" title="Permanent link">&para;</a></h3>
<p>By default the task <theory> frequencies directive will recompute the
hessian. To reuse the previously computed hessian you need only specify
reuse in the module input block. If you have stored the hessian in an
alternate place you may redirect the reuse directive to that file by
specifying the path to that file.</p>
<pre><code> reuse /path_to_hessian_file
</code></pre>
<p>This will reuse your saved Hessian data but one caveat is that the
geometry specification at the point where the hessian is computed must
be the default &ldquo;geometry&rdquo; on the current run-time-data-base for the
projection to work properly.</p>
<h3 id="hessians-and-vibrational-frequencies-redefining-masses-of-elements">Redefining Masses of Elements<a class="headerlink" href="#hessians-and-vibrational-frequencies-redefining-masses-of-elements" title="Permanent link">&para;</a></h3>
<p>You may also modify the mass of a specific center or a group of centers
via the input.</p>
<p>To modify the mass of a specific center you can simply use:</p>
<pre><code> mass 3 4.00260324
</code></pre>
<p>which will set the mass of center 3 to 4.00260324 AMUs. The lexical
index of centers is determined by the geometry object.</p>
<p>To modify all Hydrogen atoms in a molecule you may use the tag based
mechanism:</p>
<pre><code> mass hydrogen 2.014101779
</code></pre>
<p>The mass redefinitions always start with the default masses and change
the masses in the order given in the input. Care must be taken to change
the masses properly. For example, if you want all hydrogens to have the
mass of Deuterium and the third hydrogen (which is the 6th atomic
center) to have the mass of Tritium you must set the Deuterium masses
first with the tag based mechanism and then set the 6th center&rsquo;s mass to
that of Tritium using the lexical center index mechanism.</p>
<p>The mass redefinitions are not fully persistent on the
run-time-data-base. Each input block that redefines masses will
invalidate the mass definitions of the previous input block. For
example,</p>
<pre><code>freq
  reuse
  mass hydrogen 2.014101779
end
task scf frequencies
freq
  reuse
  mass oxygen 17.9991603
end
task scf frequencies
</code></pre>
<p>will use the new mass for all hydrogens in the first frequency analysis.
The mass of the oxygen atoms will be redefined in the second frequency
analysis but the hydrogen atoms will use the default mass. To get a
modified oxygen and hydrogen analysis you would have to use:</p>
<pre><code>freq
  reuse
  mass hydrogen 2.014101779
end
task scf frequencies
freq
  reuse
  mass hydrogen 2.014101779
  mass oxygen 17.9991603
end
task scf frequencies
</code></pre>
<h3 id="hessians-and-vibrational-frequencies-temp-or-temperature">Temp or Temperature<a class="headerlink" href="#hessians-and-vibrational-frequencies-temp-or-temperature" title="Permanent link">&para;</a></h3>
<p>The &ldquo;VIB&rdquo; module can generate the vibrational analysis at various
temperatures other than at standard room temperature. Either temp or
temperature can be used to initiate this command.</p>
<p>To modify the temperature of the computation you can simply use:</p>
<pre><code> temp 4 298.15 300.0 350.0 400.0
</code></pre>
<p>At this point, the temperatures are persistant and so the user must
&ldquo;reset&rdquo; the temperature if the standard behavior is required after
setting the temperatures in a previous &ldquo;VIB&rdquo; command, i.e.</p>
<pre><code> temp 1 298.15
</code></pre>
<h3 id="hessians-and-vibrational-frequencies-animation">Animation<a class="headerlink" href="#hessians-and-vibrational-frequencies-animation" title="Permanent link">&para;</a></h3>
<p>The &ldquo;VIB&rdquo; module also can generate mode animation input files in the
standard xyz file format for graphics packages like RasMol or XMol There
are scripts to automate this for RasMol in
<code>$NWCHEM_TOP/contrib/rasmolmovie</code>. Each mode will have 20 xyz files
generated that cycle from the equilibrium geometry to 5 steps in the
positive direction of the mode vector, back to 5 steps in the negative
direction of the mode vector, and finally back to the equilibrium
geometry. By default these files are not generated. To activate this
mechanism simply use the following input directive</p>
<pre><code> animate
</code></pre>
<p>anywhere in the frequency/vib input block.</p>
<p>Given an ordered list of files containing molecular coordinates in XYZ
format, the rasmolmovie shell script generates an animated gif for each
of the six possible views down a Cartesian axis.</p>
<p>It uses the free utilities</p>
<ul>
<li>rasmol 
   <a href="http://www.umass.edu/microbio/rasmol/">https://www.umass.edu/microbio/rasmol/</a>
   to manipulate the molecule and generate the individual frames</li>
<li>convert from ImageMagick 
   <a href="https://www.imagemagick.org">https://www.imagemagick.org</a>
   to combine the frames into an animated gif</li>
</ul>
<p>It should be easy to modify the script to other file formats or
animation tools.</p>
<h4 id="hessians-and-vibrational-frequencies-controlling-the-step-size-along-the-mode-vector">Controlling the Step Size Along the Mode Vector<a class="headerlink" href="#hessians-and-vibrational-frequencies-controlling-the-step-size-along-the-mode-vector" title="Permanent link">&para;</a></h4>
<p>By default, the step size used is 0.15 a.u. which will give reliable
animations for most systems. This can be changed via the animate input
directive, e.g.</p>
<pre><code>vib
   animate 0.20
end
</code></pre>
<p>where <step_size> is the real number that is the magnitude of each step
along the eigenvector of each nuclear hessian mode in atomic units.</p>
<h3 id="hessians-and-vibrational-frequencies-specifying-filenames-for-animated-normal-modes">Specifying filenames for animated normal modes<a class="headerlink" href="#hessians-and-vibrational-frequencies-specifying-filenames-for-animated-normal-modes" title="Permanent link">&para;</a></h3>
<p>By default, normal modes will be stored in files that start with
&ldquo;freq.m-<mode number>&ldquo;. This is inconvenient if more than vibrational
analysis is run in a single input file. To specify different filename
for a particular vibrational analysis use the directive</p>
<pre><code>filename &lt;file_set_name&gt; [overwrite]
</code></pre>
<p>where <file_set_name> is the name that will be prepended to the usual
filenames. In addition the code by default requires all files to be new
files. When the option &ldquo;overwrite&rdquo; is provided any pre-existing files
will simply be overwritten.</p>
<h3 id="hessians-and-vibrational-frequencies-controlling-the-step-size-of-the-finite-difference-hessian">Controlling the Step Size of the Finite difference Hessian<a class="headerlink" href="#hessians-and-vibrational-frequencies-controlling-the-step-size-of-the-finite-difference-hessian" title="Permanent link">&para;</a></h3>
<p>By default, the step size used for calculating the finite difference
Hessian is 0.010 a.u. for DFT and NWPW modules, and 0.001 a.u. otherwise
This can be changed via the fd_delta input directive, e.g.</p>
<pre><code>vib
   fd_delta 0.005
end
</code></pre>
<p>where <fd_hessian_step_size> is the real number that is the magnitude of
each displacement in atomic units for the calculation of the finite
difference Hessian. For older versions of NWChem without the fd_delta
option just set the &ldquo;stpr_gen:delta&rdquo; value on the runtime database,
e.g.</p>
<pre><code>set stpr_gen:delta 0.005
</code></pre>
<h3 id="hessians-and-vibrational-frequencies-an-example-input-deck">An Example Input Deck<a class="headerlink" href="#hessians-and-vibrational-frequencies-an-example-input-deck" title="Permanent link">&para;</a></h3>
<p>This example input deck will optimize the geometry for the given basis
          set, compute the frequencies for H<sub>2</sub>O, H<sub>2</sub>O at different
temperatures, D<sub>2</sub>O, HDO, and TDO.</p>
<pre><code>start  h2o
title Water 
geometry units au autosym
  O      0.00000000    0.00000000    0.00000000
  H      0.00000000    1.93042809   -1.10715266
  H      0.00000000   -1.93042809   -1.10715266
end
basis noprint
  H library sto-3g 
  O library sto-3g
end
scf; thresh 1e-6; end
driver; tight; end
task scf optimize

scf; thresh 1e-8; print none; end
task scf freq 

freq
 reuse; temp 4 298.15 300.0 350.0 400.0
end
task scf freq

freq 
 reuse; mass H 2.014101779
 temp 1 298.15
end
task scf freq

freq
 reuse; mass 2 2.014101779
end
task scf freq

freq
 reuse; mass 2 2.014101779 ; mass 3 3.01604927
end
task scf freq
</code></pre>
<h2 id="hessians-and-vibrational-frequencies-references">References<a class="headerlink" href="#hessians-and-vibrational-frequencies-references" title="Permanent link">&para;</a></h2>
<div class="footnote">
<hr />
<ol>
<li id="hessians-and-vibrational-frequencies-fn_footnote_1">
<p>Johnson, B. G.; Frisch, M. J. An Implementation of Analytic Second Derivatives of the Gradient-Corrected Density Functional Energy. <em>The Journal of Chemical Physics</em> <strong>1994</strong>, <em>100</em> (10), 74297442. <a href="https://doi.org/10.1063/1.466887">https://doi.org/10.1063/1.466887</a>.&#160;<a class="footnote-backref" href="#hessians-and-vibrational-frequencies-fnref_footnote_1" title="Jump back to footnote 1 in the text">&#8617;</a></p>
</li>
</ol>
</div></section><section class="print-page" id="constraints" heading-number="14.3"><h1 id="constraints-constraints">Constraints<a class="headerlink" href="#constraints-constraints" title="Permanent link">&para;</a></h1>
<p>The constraints directive allows the user to specify which constraints
should be imposed on the system during the analysis of potential energy
surface. Currently such constraints are limited to fixed atom positions
and harmonic restraints (springs) on the distance between the two atoms.
The general form of constraints block is presented
below:</p>
<pre><code>CONSTRAINTS [string name ] \  
          [clear] \  
          [enable||disable] \  
          [fix atom &lt;integer list&gt;] \  
          [spring bond &lt;integer atom1&gt; &lt;integer atom2&gt; &lt;real k&gt; &lt;real r0&gt; ]  
          [spring dihedral &lt;integer atom1&gt; &lt;integer atom2&gt; &lt;integer atom3&gt; &lt;integer atom4&gt; &lt;real k&gt; &lt;real phi0 in degrees&gt; ] 
          [spring bondings &lt;real K0&gt; &lt;real gamma0&gt; [&lt;real ca&gt; &lt;integer atom1a&gt; &lt;integer atom2a&gt; 
                                             &lt;real cb&gt; &lt;integer atom1b&gt; &lt;integer atom2b&gt;
                                             ...]]
          [penalty pbondings &lt;real K0&gt; &lt;real gcut0&gt; &lt;real gamma0&gt; [&lt;real ca&gt; &lt;integer atom1a&gt; &lt;integer atom2a&gt; 
                                                           &lt;real cb&gt; &lt;integer atom1b&gt; &lt;integer atom2b&gt;
                                                           ...]]
 END
</code></pre>
<p>The keywords are described below</p>
<ul>
<li><code>name</code> - optional keyword that associates a name with a given set of
    constraints. Any unnamed set of constraints will be given a name
    <em>default</em> and will be automatically loaded prior to a calculation.
    Any constraints with the name other than <em>default</em> will have to be
    loaded manually using the <a href="#set">SET</a> directive. For example,</li>
</ul>
<pre><code> CONSTRAINTS one  
   spring bond 1 3 5.0 1.3
   fix atom 1  
 END 
</code></pre>
<p>the above constraints can be loaded using the following set directive that assigns the name <code>one</code> as the current constraint  </p>
<pre><code> set constraints one  
 ....  
 task ....
</code></pre>
<ul>
<li><code>clear</code> - destroys any prior constraint information. This may be
    useful when the same constraints have to be redefined or completely
    removed from the runtime database.</li>
</ul>
<!-- end list -->

<ul>
<li><code>enable||disable</code> - enables or disables 
    without actually removing the information from the runtime database.</li>
</ul>
<!-- end list -->

<ul>
<li><code>fix atom</code> - fixes atom positions during geometry optimization. This
    directive requires an integer list that specifies which atoms are to
    be fixed. This directive can be repeated within a given constraints
    block. To illustrate the use <code>fix atom</code> directive let us consider a
    situation where we would like to fix atoms 1, 3, 4, 5, 6 while
    performing an optimization on some hypothetical system. There are
    actually several ways to enter this particular constraint.</li>
</ul>
<!-- end list -->

<ul>
<li>There is a straightforward option which requires the most
        typing:</li>
</ul>
<pre><code>       constraints  
         fix atom 1 3 4 5 6 
       end
</code></pre>
<ul>
<li>Second method uses list input:</li>
</ul>
<pre><code>       constraints  
         fix atom 1 3:6  
       end
</code></pre>
<ul>
<li>Third approach illustrates the use of multiple <code>fix atom</code> directives:</li>
</ul>
<pre><code>       constraints 
         fix atom 1 
         fix atom 3:6 
       end
</code></pre>
<ul>
<li>the <code>spring bond</code> <em>&lt; i j k r<sub>0</sub> &gt;</em> - places a spring with a spring constant
    <em>k</em> and equilibrium length <em>r<sub>0</sub></em> between atoms <em>i</em> and <em>j</em> (all in
    atomic units). Please note that this type of constraint adds an
    additional term to the total energy expression</li>
</ul>
<p><em>E = E<sub>total</sub> + &frac12; k (r<sub>ij</sub> - r<sub>0</sub>)<sup>2</sup></em>  </p>
<p>This additional term forces the distance between atoms <em>i</em> and <em>j</em> to be
in the vicinity of <em>r<sub>0</sub></em> but never exactly that. In general the spring
energy term will always have some nonzero residual value, and this has
to be accounted for when comparing total energies. The <code>spring bond</code>
directive can be repeated within a given constraints block. If the
spring between the same pair of atoms is defined more than once, it will
be replaced by the latest specification in the order it appears in the
input block.</p></section><section class="print-page" id="nudged-elastic-band-and-zero-temperature-string-methods" heading-number="14.4"><h1 id="nudged-elastic-band-and-zero-temperature-string-methods-nudged-elastic-band-neb-method">Nudged Elastic Band (NEB) method<a class="headerlink" href="#nudged-elastic-band-and-zero-temperature-string-methods-nudged-elastic-band-neb-method" title="Permanent link">&para;</a></h1>
<h2 id="nudged-elastic-band-and-zero-temperature-string-methods-overview">Overview<a class="headerlink" href="#nudged-elastic-band-and-zero-temperature-string-methods-overview" title="Permanent link">&para;</a></h2>
<p>The NEB module is an implementation of the nudged elastic band (NEB)
method of Jonsson et al., and it is one of two drivers in NWChem that
can be used to perform minimum energy path optimizations. NEB can be
used at all levels of theory, including SCF, HF, DFT, PSPW, BAND, MP2,
RIMP2, CCSD, TCE.</p>
<p>Input to the NEB modules is contained with the NEB block</p>
<pre><code> NEB  
  ...  
 END
</code></pre>
<p>To run a NEB calculation the following the following task directives is
used</p>
<pre><code>TASK &lt;theory&gt; NEB  
TASK &lt;theory&gt; NEB ignore
</code></pre>
<p>where <code>&lt;theory&gt;</code> is SCF, HF, DFT, PSPW, BAND, MP2, CCSD, TCE, etc.. The
Task directive with the ignore option is recommended, otherwise NWChem
will crash if the path is not optimized in the allowed maximum number of
iterations.</p>
<p>Optional input for this module is specified within the compound
directive,</p>
<pre><code> NEB 
    NBEADS &lt;integer nbeads default 5&gt;  
    KBEADS &lt;float kbeads default 0.1&gt; 
    MAXITER &lt;integer maxiter default 5&gt;  

    STEPSIZE &lt;integer stepsize default 1.0&gt;  
    NHIST &lt;integer nhist default 5&gt; 
    ALGORITHM &lt;integer algorithm default 0&gt;  

    [loose | default | tight]  
    GMAX &lt;float gmax default 0.00045&gt;  
    GRMS &lt;float grms default 0.00030&gt;  
    XMAX &lt;float xmax default 0.00018&gt;  
    XMRS &lt;float xmrs default 0.00012&gt;  

    [IMPOSE]  
    [HASMIDDLE]  
    [XYZ_PATH &lt;string xyzfilename&gt;]  
    [RESET]  
    [PRINT_SHIFT &lt;integer print_shift default 0&gt;]  
 END
</code></pre>
<p>The following list describes the input for the NEB block</p>
<ul>
<li><strong>nbeads</strong> - number of beads (or images) used to represent the path</li>
<li><strong>kbeads</strong> - value for the NEB spring constant</li>
<li><strong>maxiter</strong> - maximum number of NEB path optimizations to be performed</li>
<li><strong>stepsize</strong> - value for the stepsize used in the optimization.
    Typically less than 1.</li>
<li><strong>nhist</strong> - number of histories to use for quasi-Newton optimization
    (algorithm =0)</li>
<li><strong>LOOSE|DEFAULT|TIGHT</strong> - options specifying thresholds for convergence</li>
<li><strong>gmax</strong> - value for the maximum gradient used to determine
    convergence</li>
<li><strong>grms</strong> - value for the root mean square gradient used to determine
    convergence</li>
<li><strong>xmax</strong> - value for the maximum cartesian step used to determine
    convergence</li>
<li><strong>xrmx</strong> - value for the root mean square cartesian step used to
    determine convergence</li>
<li><strong>algorithm</strong> - 0: quasi-Newton Fixed Point optimization, 1: dampled
    Verlet optimization, 2: refining conjugate gradient optimization</li>
<li><strong>IMPOSE</strong> - if specified causes the initial geometries used to specify
    the path to be aligned with one another</li>
<li><strong>HASMIDDLE</strong> - if specified causes the initial path to use the the
    &ldquo;midgeom&rdquo; geometry to be used as the midpoint, i.e. the initial path
    is defined as a linear morphing from &ldquo;geometry&rdquo; &ndash;&gt; &ldquo;midgeom&rdquo; &ndash;&gt;
    &ldquo;endgeom&rdquo;</li>
<li><strong>XYZ_PATH</strong> - if specified the initial path is defined from the
    sequence of geometries contained in <strong>xyzfilename</strong></li>
<li><strong>RESET</strong> - if specified causes the NEB optimization and path to be
    started from scratch</li>
<li><strong>print_shift</strong> - setting the <code>PRINT_SHIFT</code> directive causes the path
    energies and geometries to be outputed every <code>&lt;print_shift&gt;</code> steps.
    The current path energies are appended to the file
    <code>jobname.neb_epath</code> and the current geometries are appended to the
    file <code>jobname.nebpath_"current iteration".xyz</code>.</li>
</ul>
<h2 id="nudged-elastic-band-and-zero-temperature-string-methods-setting-up-initial-path">Setting up initial path<a class="headerlink" href="#nudged-elastic-band-and-zero-temperature-string-methods-setting-up-initial-path" title="Permanent link">&para;</a></h2>
<p>There are three different ways to define the initial path for NEB
optimization.</p>
<ul>
<li>Linear interpolation between two geometries</li>
</ul>
<p>The geometries in the path are defined
by</p>
<div class="arithmatex">\[\vec{R}^i_{xyz} = \vec{R}^1_{xyz} + \frac{i-1}{nbeads-1} (\vec{R}^{nbeads}_{xyz}-\vec{R}^{1}_{xyz}), \ for \ i=1,nbeads\]</div>
<p>where the starting geometry <span class="arithmatex">\(\vec{R}^1_{xyz}\)</span> is entered in the
geometry block labeled <code>geometry</code>, e.g.</p>
<pre><code>geometry nocenter noautosym noautoz  
O  0.00000000    -0.02293938     0.00000000  
H  0.00000000     0.55046969     0.75406534  
H  0.00000000     0.55046969    -0.75406534  
end
</code></pre>
<p>and the last geometry in the path  <span class="arithmatex">\(\vec{R}^{nbeads}_{xyz}\)</span> is
entered in the geometry block label <code>endgeom</code>, e.g.</p>
<pre><code>geometry endgeom nocenter noautosym noautoz  
O  0.00000000     0.02293938     0.00000000  
H  0.00000000    -0.55046969     0.75406534  
H  0.00000000    -0.55046969    -0.75406534  
end
</code></pre>
<ul>
<li>Linear interpolation between three geometries</li>
</ul>
<p>The geometries for this path are defined
by</p>
<div class="arithmatex">\[\vec{R}^i_{xyz} = \vec{R}^1_{xyz} + \frac{i-1}{nbeads/2-1} (\vec{R}^{nbeads/2}_{xyz}-\vec{R}^{1}_{xyz}), \ for \   i=1,nbeads/2\]</div>
<p>and</p>
<div class="arithmatex">\[\vec{R}^i_{xyz} = \vec{R}^{nbeads/2}_{xyz} + \frac{i-nbeads/2}{nbeads/2-1} (\vec{R}^{nbeads}_{xyz}-\vec{R}^{nbeads/2}_{xyz}), \ for \  i=nbeads/2+1,nbeads\]</div>
<p>where the starting <span class="arithmatex">\(\vec{R}^1_{xyz}\)</span> , middle <span class="arithmatex">\(\vec{R}_{xyz}^{nbeads/2}\)</span>  and last <span class="arithmatex">\(\vec{R}^{nbeads}_{xyz}\)</span>
geometries are entered in the geometry blocks <code>geometry</code>, <code>midgeom</code> and
<code>endgeom</code> respectively, e.g.</p>
<pre><code>geometry nocenter noautosym noautoz  
O  0.00000000    -0.02293938     0.00000000  
H  0.00000000     0.55046969     0.75406534  
H  0.00000000     0.55046969    -0.75406534  
end

geometry midgeom nocenter noautosym noautoz  
O  0.00000000     0.00000000     0.00000000  
H  0.00000000     0.00000000     1.00000000  
H  0.00000000     0.00000000    -1.00000000  
end

geometry endgeom nocenter noautosym noautoz  
O  0.00000000     0.02293938     0.00000000  
H  0.00000000    -0.55046969     0.75406534  
H  0.00000000    -0.55046969    -0.75406534  
end
</code></pre>
<ul>
<li>Using <code>xyz_path</code> to explicitly input a path of geometries</li>
</ul>
<p>The <code>xyz_path</code> option can also be used to define the initial path.</p>
<pre><code>...  
NEB  
   ...  
  XYZ_PATH path.xyz
END  
...
</code></pre>
<p>where path.xyz contains a list of geometries in xyz format, e.g.</p>
<pre><code>--------------- path.xyz ------------------  
                   3  
energy=  -17.107207699285738       
O           0.000000   -0.022939    0.000000  
H           0.000000    0.550469    0.754065  
H           0.000000    0.550469   -0.754065  
                   3  
energy=  -17.094903833074170       
O          -0.000003   -0.110080   -0.000000  
H          -0.000000    0.273180    0.847029  
H          -0.000000    0.273180   -0.847029  
                   3  
energy=  -17.063823686395292       
O          -0.000000   -0.000080   -0.000000  
H           0.000000   -0.000002    0.941236  
H           0.000000   -0.000002   -0.941236  
                   3  
energy=  -17.094944036147005       
O          -0.000000    0.110472   -0.000000  
H          -0.000000   -0.273172    0.846957  
H          -0.000000   -0.273172   -0.846957  
                   3  
energy=  -17.107208157343706       
O           0.000000    0.022939    0.000000  
H           0.000000   -0.550469    0.754065  
H           0.000000   -0.550469   -0.754065  
--------------- path.xyz ------------------
</code></pre>
<h2 id="nudged-elastic-band-and-zero-temperature-string-methods-convergence-criteria">Convergence criteria<a class="headerlink" href="#nudged-elastic-band-and-zero-temperature-string-methods-convergence-criteria" title="Permanent link">&para;</a></h2>
<p>The defaults may be used, or the directives LOOSE, DEFAULT, or TIGHT
specified to use standard sets of values, or the individual criteria
adjusted. All criteria are in atomic units. GMAX and GRMS control the
maximum and root mean square gradient in the coordinates. XMAX and XRMS
control the maximum and root mean square of the Cartesian step.</p>
<table>
<thead>
<tr>
<th></th>
<th>LOOSE</th>
<th>DEFAULT</th>
<th>TIGHT</th>
</tr>
</thead>
<tbody>
<tr>
<td>GMAX</td>
<td>0.0045d0</td>
<td>0.00045</td>
<td>0.000015</td>
</tr>
<tr>
<td>GRMS</td>
<td>0.0030d0</td>
<td>0.00030</td>
<td>0.00001</td>
</tr>
<tr>
<td>XMAX</td>
<td>0.0054d0</td>
<td>0.00180</td>
<td>0.00006</td>
</tr>
<tr>
<td>XRMS</td>
<td>0.0036d0</td>
<td>0.00120</td>
<td>0.00004</td>
</tr>
</tbody>
</table>
<h2 id="nudged-elastic-band-and-zero-temperature-string-methods-neb-tutorial-1-h2o-inversion">NEB Tutorial 1: H2O Inversion<a class="headerlink" href="#nudged-elastic-band-and-zero-temperature-string-methods-neb-tutorial-1-h2o-inversion" title="Permanent link">&para;</a></h2>
<p>(input:<a href="h2o-neb.nw">h2o-neb.nw</a>,
output:<a href="h2o-neb.nwout">h2o-neb.nwout</a>, datafiles:
<a href="h2o-neb.neb_epath.dat">h2o-neb.neb_epath.dat</a>
<a href="h2o-neb.neb_final_epath.dat">h2o-neb.neb_final_epath.dat</a>
)</p>
<p>(xyzfiles: <a href="h2o-neb.nebpath_000001.xyz">h2o-neb.nebpath_000001.xyz</a>
<a href="h2o-neb.nebpath_000005.xyz">h2o-neb.nebpath_000005.xyz</a>
<a href="h2o-neb.nebpath_000010.xyz">h2o-neb.nebpath_000010.xyz</a>
<a href="h2o-neb.nebpath_000020.xyz">h2o-neb.nebpath_000020.xyz</a>
<a href="h2o-neb.nebpath_final.xyz">h2o-neb.nebpath_final.xyz</a>
)</p>
<p><img alt="H2O Inversion Pathway" src="h2o.gif" /></p>
<pre><code>Title &quot;H2O inversion calculation&quot;
echo
start h2o-neb

geometry nocenter noautosym noautoz  
O  0.00000000    -0.02293938     0.00000000  
H  0.00000000     0.55046969     0.75406534  
H  0.00000000     0.55046969    -0.75406534  
end  

geometry endgeom nocenter noautosym noautoz  
O  0.00000000     0.02293938     0.00000000  
H  0.00000000    -0.55046969     0.75406534  
H  0.00000000    -0.55046969    -0.75406534  
end  
### Gaussian DFT ###
basis  
* library 3-21G  
end  

dft  
  xc b3lyp  
  maxiter 5001  
  cgmin  
end  

neb  
  nbeads 10  
  kbeads 1.0  
  maxiter 10  
  stepsize 0.10  
  print_shift 1  
end  
task dft neb ignore
neb  
  # increase the number of images  
  nbeads 20  
  kbeads 1.0  
  stepsize 1.0  
  maxiter 30  
  loose  
end 
task dft neb ignore
</code></pre>
<p>After each optimization step the path energies are outputed as follows</p>
<pre><code>neb: Path Energy #                    9  
neb:                     1  -75.970000166349976       
neb:                     2  -75.973958450556779       
neb:                     3  -75.973964391052448       
neb:                     4  -75.973965560274110       
neb:                     5  -75.973961077512683       
neb:                     6  -75.973087554095144       
neb:                     7  -75.965847261117744       
neb:                     8  -75.950292780255126       
neb:                     9  -75.932932759963109       
neb:                    10  -75.921912278179292       
neb:                    11  -75.921834552460439       
neb:                    12  -75.932680002200939       
neb:                    13  -75.949868818688529       
neb:                    14  -75.965372754426866       
neb:                    15  -75.972788885848303       
neb:                    16  -75.973958649400714       
neb:                    17  -75.973965255113598       
neb:                    18  -75.973964962774133       
neb:                    19  -75.973959526041568       
neb:                    20  -75.970000163960066     
</code></pre>
<p>Another way to keep track of the optimization process is to run the
following grep command on the output
file.</p>
<pre><code>[WE24397:NWChem/NEB/Example2] bylaska% grep @ h2o-neb.nwout  
@neb    
@neb NEB Method  
@neb algorithm      =         0  
@neb maxiter        =        10  
@neb nbeads         =        10  
@neb nhist          =         5  
@neb natoms         =         3  
@neb stepsize       = 0.100E+01  
@neb trust          = 0.100E+00  
@neb kbeads         = 0.100E+00  
@neb Gmax tolerance = 0.450E-03  
@neb Grms tolerance = 0.300E-03  
@neb Xmax tolerance = 0.180E-03  
@neb Xrms tolerance = 0.120E-03  
@neb    
@neb Step    Intrinsic E    Mid-Point E      Minimum E      Maximum E   Gmax     Grms     Xrms     Xmax   Walltime  
@neb ---- -------------- -------------- -------------- -------------- -------- -------- -------- -------- --------  
@neb    1     -75.951572     -75.921109     -75.970632     -75.921109  0.55875  0.01606  0.14221  1.54029    454.9  
@neb    2     -75.953755     -75.923180     -75.972590     -75.923177  0.38930  0.01116  0.01588  0.45644    624.4  
@neb    3     -75.956726     -75.924391     -75.972861     -75.924387  0.25587  0.00961  0.03673  0.83118    805.2  
@neb    4     -75.957861     -75.924279     -75.973059     -75.924275  0.23572  0.00894  0.01793  0.24399    971.8  
@neb    5     -75.959613     -75.925045     -75.973869     -75.925036  0.10257  0.00464  0.03197  0.20350   1152.8  
@neb    6     -75.959964     -75.925503     -75.973957     -75.925486  0.04762  0.00196  0.00905  0.10433   1316.4  
@neb    7     -75.960068     -75.925822     -75.973956     -75.925791  0.03897  0.00141  0.00308  0.04432   1519.9  
@neb    8     -75.960091     -75.925914     -75.973959     -75.925877  0.03707  0.00127  0.00070  0.01691   2055.8  
@neb    9     -75.960129     -75.926078     -75.973962     -75.926028  0.03353  0.00108  0.00127  0.03707   2297.2  
@neb   10     -75.960142     -75.926142     -75.973963     -75.926085  0.03199  0.00101  0.00054  0.00420   2756.6  
@neb   NEB calculation not converged  
@neb    
@neb NEB Method  
@neb algorithm      =         0  
@neb maxiter        =        30  
@neb nbeads         =        20  
@neb nhist          =         5  
@neb natoms         =         3  
@neb stepsize       = 0.100E+01  
@neb trust          = 0.100E+00  
@neb kbeads         = 0.100E+01  
@neb Gmax tolerance = 0.450E-02  
@neb Grms tolerance = 0.300E-02  
@neb Xmax tolerance = 0.540E-02  
@neb Xrms tolerance = 0.360E-02  
@neb    
@neb Step    Intrinsic E    Mid-Point E      Minimum E      Maximum E   Gmax     Grms     Xrms     Xmax   Walltime  
@neb ---- -------------- -------------- -------------- -------------- -------- -------- -------- -------- --------  
@neb    1     -75.960225     -75.921704     -75.973965     -75.921669  0.24799  0.00398  0.00272  0.08741   3966.5  
@neb    2     -75.960339     -75.921782     -75.973965     -75.921745  0.24794  0.00328  0.00199  0.12148   5023.2  
@neb    3     -75.960424     -75.921742     -75.973965     -75.921701  0.19390  0.00286  0.00164  0.08342   5741.4  
@neb    4     -75.960494     -75.921849     -75.973965     -75.921804  0.19681  0.00266  0.00143  0.09030   6079.7  
@neb    5     -75.960646     -75.921874     -75.973965     -75.921820  0.17459  0.00240  0.00241  0.22047   6751.5  
@neb    6     -75.960674     -75.921856     -75.973965     -75.921797  0.14246  0.00165  0.00060  0.00256   7572.3  
@neb    7     -75.960724     -75.921884     -75.973966     -75.921817  0.13004  0.00153  0.00082  0.05401   7893.3  
@neb    8     -75.960747     -75.921892     -75.973966     -75.921822  0.12809  0.00149  0.00038  0.00237   8631.2  
@neb    9     -75.960792     -75.921912     -75.973966     -75.921835  0.12267  0.00142  0.00075  0.05081   9222.0  
@neb   10     -75.960813     -75.921923     -75.973966     -75.921841  0.11902  0.00138  0.00035  0.00212  10163.2  
@neb   11     -75.960834     -75.921934     -75.973966     -75.921846  0.11569  0.00135  0.00035  0.00203  10478.3  
@neb   12     -75.961060     -75.922060     -75.973966     -75.921889  0.07709  0.00104  0.00365  0.30944  10863.8  
@neb   13     -75.961255     -75.922186     -75.973966     -75.921919  0.04600  0.00087  0.00309  0.19999  11357.0  
@neb   14     -75.961405     -75.922286     -75.973966     -75.921927  0.03549  0.00079  0.00244  0.03857  11860.0  
@neb  NEB calculation converged
</code></pre>
<h1 id="nudged-elastic-band-and-zero-temperature-string-methods-zero-temperature-string-method">Zero Temperature String Method<a class="headerlink" href="#nudged-elastic-band-and-zero-temperature-string-methods-zero-temperature-string-method" title="Permanent link">&para;</a></h1>
<h2 id="nudged-elastic-band-and-zero-temperature-string-methods-overview_1">Overview<a class="headerlink" href="#nudged-elastic-band-and-zero-temperature-string-methods-overview_1" title="Permanent link">&para;</a></h2>
<p>The STRING module is an implementation of the zero temperature string
method of vanden Eijden et al., and it is one of two drivers in NWChem
that can be used to perform minimum energy path optimizations. STRING
can be used at all levels of theory, including SCF, HF, DFT, PSPW, BAND,
MP2, RIMP2, CCSD, TCE.</p>
<p>Input to the STRING module is contained with the STRING block</p>
<pre><code> STRING
  ...
 END
</code></pre>
<p>To run a STRING calculation the following the following task directives
is used</p>
<pre><code>TASK &lt;theory&gt; STRING
TASK &lt;theory&gt; STRING ignore
</code></pre>
<p>where <code>&lt;theory&gt;</code> is SCF, HF, DFT, PSPW, BAND, MP2, CCSD, TCE, etc.. The
Task directive with the ignore option is recommended, otherwise NWChem
will crash if the path is not optimized in the allowed maximum number of
iterations.</p>
<p>Optional input for this module is specified within the compound
directive,</p>
<pre><code> STRING 
    NBEADS &lt;integer nbeads default 5&gt; 
    MAXITER &lt;integer maxiter default 5&gt; 

    STEPSIZE &lt;integer stepsize default 1.0&gt; 
    NHIST &lt;integer nhist default 5&gt;  
    INTERPOL &lt;integer algorithm default 1&gt;  

    FREEZE1 &lt;logical freeze1 default .false.&gt;  
    FREEZEN &lt;logical freezen default .false.&gt;  

    TOL &lt;float tol default 0.00045&gt;  

    [IMPOSE]  
    [HASMIDDLE]  
    [XYZ_PATH &lt;string xyzfilename&gt;]  
    [RESET]  
    PRINT_SHIFT &lt;integer print_shift default 0&gt;  
 END
</code></pre>
<p>The following list describes the input for the STRING block</p>
<ul>
<li><strong>nbeads</strong> - number of beads (or images) used to represent the path</li>
<li><strong>maxiter</strong> - maximum number of NEB path optimizations to be performed</li>
<li><strong>stepsize</strong> - value for the stepsize used in the optimization.
    Typically less than 1.</li>
<li><strong>nhist</strong> - number of histories to use for quasi-Newton optimization
    (algorithm =0)</li>
<li><strong>tol</strong> - value for the maximum gradient used to determine convergence</li>
<li><strong>freeze1</strong> - <code>.true.</code>: first bead of simulation frozen, <code>.false.</code>:first
    bead of simulation not frozen.</li>
<li><strong>freezeN</strong> - <code>.true.</code>:last bead of simulation frozen, <code>.false.</code>:last bead
    of simulation not frozen</li>
<li><strong>interpol</strong> - 1: linear, 2: spline, 3: Akima spline</li>
<li>IMPOSE - if specified causes the initial geometries used to specify
    the path to be aligned with one another</li>
<li>HASMIDDLE - if specified causes the initial path to use the the
    &ldquo;midgeom&rdquo; geometry to be used as the midpoint, i.e. the initial path
    is defined as a linear morphing from &ldquo;geometry&rdquo; &ndash;&gt; &ldquo;midgeom&rdquo; &ndash;&gt;
    &ldquo;endgeom&rdquo;</li>
<li>XYZ_PATH - if specified the initial path is defined from the
    sequence of geometries contained in <strong>xyzfilename</strong></li>
<li>RESET - if specified causes the NEB optimization and path to be
    started from scratch</li>
<li><strong>print_shift</strong> - setting the PRINT_SHIFT directive causes the path
    energies and geometries to be outputed every <code>&lt;print_shift&gt;</code> steps.
    The current path energies are appended to the file
    <code>jobname.neb_epath</code> and the current geometries are appended to the
    file <code>jobname.nebpath _"current iteration".xyz</code></li>
</ul>
<h2 id="nudged-elastic-band-and-zero-temperature-string-methods-setting-up-the-initial-path">Setting up the initial path<a class="headerlink" href="#nudged-elastic-band-and-zero-temperature-string-methods-setting-up-the-initial-path" title="Permanent link">&para;</a></h2>
<p>There are three different ways to define the initial path for NEB
optimization.</p>
<ul>
<li>Linear interpolation between two geometries</li>
</ul>
<p>The geometries in the path are defined
by</p>
<div class="arithmatex">\[\vec{R}^i_{xyz} = \vec{R}^1_{xyz} + \frac{i-1}{nbeads-1} (\vec{R}^{nbeads}_{xyz}-\vec{R}^{1}_{xyz}), \  for \   i=1,nbeads\]</div>
<p>where the starting geometry  <span class="arithmatex">\(\vec{R}^1_{xyz}\)</span>  is entered in the
geometry block labeled <code>geometry</code>, e.g.</p>
<pre><code>geometry nocenter noautosym noautoz  
O  0.00000000    -0.02293938     0.00000000  
H  0.00000000     0.55046969     0.75406534  
H  0.00000000     0.55046969    -0.75406534  
end
</code></pre>
<p>and the last geometry in the path  <span class="arithmatex">\(\vec{R}^{nbeads}_{xyz}\)</span>  is
entered in the geometry block label <code>endgeom</code>, e.g.</p>
<pre><code>geometry endgeom nocenter noautosym noautoz  
O  0.00000000     0.02293938     0.00000000  
H  0.00000000    -0.55046969     0.75406534  
H  0.00000000    -0.55046969    -0.75406534  
end
</code></pre>
<ul>
<li>Linear interpolation between three geometries</li>
</ul>
<p>The geometries for this path are defined
by</p>
<div class="arithmatex">\[\vec{R}^i_{xyz} = \vec{R}^1_{xyz} + \frac{i-1}{nbeads/2-1} (\vec{R}^{nbeads/2}_{xyz}-\vec{R}^{1}_{xyz}), \  for \   i=1,nbeads/2\]</div>
<p>and</p>
<div class="arithmatex">\[\vec{R}^i_{xyz} = \vec{R}^{nbeads/2}_{xyz} + \frac{i-nbeads/2}{nbeads/2-1} (\vec{R}^{nbeads}_{xyz}-\vec{R}^{nbeads/2}_{xyz}), \  for \   i=nbeads/2+1,nbeads\]</div>
<p>where the starting  <span class="arithmatex">\(\vec{R}^{1}_{xyz}\)</span> , middle
<span class="arithmatex">\(\vec{R}^{nbeads/2}_{xyz}\)</span>  and last  <span class="arithmatex">\(\vec{R}^{nbeads}_{xyz}\)</span>&rdquo; 
geometries are entered in the geometry blocks <code>geometry</code>, <code>midgeom</code> and
<code>endgeom</code> respectively, e.g.</p>
<pre><code>geometry nocenter noautosym noautoz  
O  0.00000000    -0.02293938     0.00000000  
H  0.00000000     0.55046969     0.75406534  
H  0.00000000     0.55046969    -0.75406534  
end

geometry midgeom nocenter noautosym noautoz  
O  0.00000000     0.00000000     0.00000000  
H  0.00000000     0.00000000     1.00000000  
H  0.00000000     0.00000000    -1.00000000  
end

geometry endgeom nocenter noautosym noautoz  
O  0.00000000     0.02293938     0.00000000  
H  0.00000000    -0.55046969     0.75406534  
H  0.00000000    -0.55046969    -0.75406534  
end
</code></pre>
<ul>
<li>Using <code>xyz_path</code> to explicitly input a path of geometries</li>
</ul>
<p>The <code>xyz_path</code> option can also be used to define the initial path, e.g.</p>
<pre><code>...
STRING
   ...
  XYZ_PATH path.xyz
END
...
</code></pre>
<h2 id="nudged-elastic-band-and-zero-temperature-string-methods-string-tutorial-1hcn-hnc-path-optimization">String Tutorial 1:HCN &ndash;&gt; HNC path optimization<a class="headerlink" href="#nudged-elastic-band-and-zero-temperature-string-methods-string-tutorial-1hcn-hnc-path-optimization" title="Permanent link">&para;</a></h2>
<p>(input:<a href="HCN-string1.nw">HCN-string1.nw</a>,
output:<a href="HCN-string1.nwout">HCN-string1.nwout</a>,
datafiles:
<a href="HCN-string1.string_epath.dat">HCN-string1.string_epath.dat</a>
<a href="HCN-string1.string_final_epath.dat">HCN-string1.string_final_epath.dat</a>
)</p>
<p>(xyzfiles: <a href="HCN-string1.stringpath_000001.xyz">
HCN-string1.stringpath_000001.xyz</a>
<a href="HCN-string1.stringpath_000005.xyz">
HCN-string1.stringpath_000005.xyz</a>
<a href="HCN-string1.stringpath_000010.xyz">
HCN-string1.stringpath_000010.xyz</a>
<a href="HCN-string1.stringpath_000020.xyz">
HCN-string1.stringpath_000020.xyz</a>
<a href="HCN-string1.stringpath_000030.xyz">
HCN-string1.stringpath_000030.xyz</a>
<a href="HCN-string1.stringpath_final.xyz">
HCN-string1.stringpath_final.xyz</a>
)</p>
<center>

![HCN--\>HNC Reaction Pathway](hcn-hnc-dft.gif)

</center>

<center>

![400px](hcn-hnc-dft.png)

</center>

<p>In this example, the path energy for the reaction HCN &ndash;&gt; HNC is
calculated.</p>
<pre><code>#    
# The initial path has the Carbon moving through the Nitrogen.  
# So for this simulation to work that atom avoidance code needs to work.  
# Because the initial path is so stiff the wavefunction optimizer needs to requires  
# lots of iterations during the early stages of the path optimization.  
#  
#  
Title &quot;HCN --&gt; HNC Zero-Temperature String Simulation&quot;  
echo  
start hcn-hnc-dft  

geometry noautoz noautosym  
C         0.00000000     0.00000000    -0.49484657  
N         0.00000000     0.00000000     0.64616359  
H         0.00000000     0.00000000    -1.56151539  
end  

geometry endgeom  noautoz noautosym  
C         0.00000000     0.00000000     0.73225318   
N         0.00000000     0.00000000    -0.42552059  
H         0.00000000     0.00000000    -1.42351006  
end  

### Gaussian DFT ###
basis  
* library 3-21G  
end  

dft  
  xc b3lyp  
  maxiter 501  
end  

string  
  nhist 10  
  nbeads 10  
  maxiter 10  
  stepsize 0.10  
  print_shift 1  

  # don't allow the end points of the path to move  
  freeze1 .true.  
  freezeN .true.  
end  
task dft string ignore

string  
  # increase the number of images  
  nbeads 20  
  maxiter 20  

  # allow the end points of the path to move  
  freeze1 .false.  
  freezeN .false.  
end  
task dft string ignore
</code></pre>
<p>After each optimization step the path energies are outputed as follows</p>
<pre><code> string: Path Energy #                    2  
 string:                     1  -92.906682492969779  
 string:                     2  -92.743446565848473  
 string:                     3  -92.751945829987775  
 string:                     4  -92.756507971834026  
 string:                     5  -92.726984154346979  
 string:                     6  -92.701651474021503  
 string:                     7  -92.672613497521183  
 string:                     8  -92.825096796032099  
 string:                     9  -92.716422030970662  
 string:                    10  -92.881713271394148
</code></pre>
<p>Another way to keep track of the optimization process is to run the
following grep command on the output
file.</p>
<pre><code>[WE24397:NWChem/NEB/Example2] bylaska% grep @ HCN-dft.out  
@zts  
@zts String method.  
@zts Temperature          =   0.00000  
@zts Covergence Tolerance =   0.00010  
@zts Step Size            =   0.10000  
@zts Maximum Time Steps   =        10  
@zts Number of replicas   =        10  
@zts Number of histories  =        10  
@zts String Interpolator  =         1  
@zts First Replica        = frozen  
@zts Last Replica         = frozen  
@zts  
@zts  Step     xrms     xmax        E start       E middle          E end          E max      E average  
@zts     1 0.460700 2.602234    -92.9066825    -83.4767173    -92.8817133    -83.4767173    -91.6169775  
@zts     2 0.862226 5.405612    -92.9066825    -92.3028437    -92.8817133    -92.3028437    -92.6631831  
@zts     3 0.105285 0.530157    -92.9066825    -92.3289676    -92.8817133    -92.3289676    -92.6702949  
@zts     4 0.134687 0.740991    -92.9066825    -92.3512584    -92.8817133    -92.3512584    -92.6821949  
@zts     5 0.117113 0.916210    -92.9066825    -92.3767826    -92.8817133    -92.3767826    -92.6899234  
@zts     6 0.124464 0.844439    -92.9066825    -92.4195957    -92.8817133    -92.4195957    -92.7045117  
@zts     7 0.092105 0.731434    -92.9066825    -92.4510785    -92.8817133    -92.4510785    -92.7156403  
@zts     8 0.049227 0.330651    -92.9066825    -92.4690983    -92.8817133    -92.4690983    -92.7288274  
@zts     9 0.032819 0.177356    -92.9066825    -92.4827444    -92.8817133    -92.4827444    -92.7344806  
@zts    10 0.076249 0.444246    -92.9066825    -92.4930430    -92.8817133    -92.4930430    -92.7381477  
@zts The string calculation failed to converge  
@zts Bead number    1  Potential Energy =     -92.906682487840  
@zts Bead number    2  Potential Energy =     -92.850640135623  
@zts Bead number    3  Potential Energy =     -92.819370566454  
@zts Bead number    4  Potential Energy =     -92.680821335407  
@zts Bead number    5  Potential Energy =     -92.505231918657  
@zts Bead number    6  Potential Energy =     -92.493042984646  
@zts Bead number    7  Potential Energy =     -92.637367419044  
@zts Bead number    8  Potential Energy =     -92.775376312982  
@zts Bead number    9  Potential Energy =     -92.831230727986  
@zts Bead number   10  Potential Energy =     -92.881713271394  
@zts  
@zts String method.  
@zts Temperature          =   0.00000  
@zts Covergence Tolerance =   0.00010  
@zts Step Size            =   0.10000  
@zts Maximum Time Steps   =        20  
@zts Number of replicas   =        20  
@zts Number of histories  =        10  
@zts String Interpolator  =         1  
@zts First Replica        = moves  
@zts Last Replica         = moves  
@zts  
@zts  Step     xrms     xmax        E start       E middle          E end          E max      E average  
@zts     1 1.039809 5.039486    -92.9071472    -92.4998400    -92.8820628    -92.4998400    -92.7500136  
@zts     2 0.192562 0.999019    -92.9073958    -92.5259828    -92.8821500    -92.5259828    -92.7624061  
@zts     3 0.244943 1.236459    -92.9075306    -92.5735140    -92.8821223    -92.5735140    -92.7816692  
@zts     4 0.207031 1.093667    -92.9075888    -92.6229190    -92.8821177    -92.6154678    -92.7979112  
@zts     5 0.056648 0.293829    -92.9075975    -92.6672565    -92.8821033    -92.6507897    -92.8101666  
@zts     6 0.078950 0.555245    -92.9076044    -92.7245122    -92.8822536    -92.7014407    -92.8241914  
@zts     7 0.065564 0.521110    -92.9076101    -92.7539982    -92.8822915    -92.7376310    -92.8326007  
@zts     8 0.050188 0.319477    -92.9076113    -92.7695725    -92.8824219    -92.7612604    -92.8378464  
@zts     9 0.055301 0.322130    -92.9076168    -92.7754581    -92.8825732    -92.7740099    -92.8408900  
@zts    10 0.038769 0.195102    -92.9076177    -92.7775695    -92.8826652    -92.7775695    -92.8425440  
@zts    11 0.064900 0.273480    -92.9076215    -92.7800330    -92.8827175    -92.7800330    -92.8443574  
@zts    12 0.062593 0.266337    -92.9076224    -92.7823972    -92.8826993    -92.7823972    -92.8458976  
@zts    13 0.205437 0.948190    -92.9076243    -92.7842034    -92.8826408    -92.7842034    -92.8469810  
@zts    14 0.015025 0.068924    -92.9076247    -92.7844362    -92.8826536    -92.7844362    -92.8472227  
@zts    15 0.129208 0.602636    -92.9076254    -92.7849856    -92.8826676    -92.7849856    -92.8477169  
@zts    16 0.013479 0.056561    -92.9076260    -92.7855201    -92.8826783    -92.7855201    -92.8481626  
@zts    17 0.472858 2.220715    -92.9076271    -92.7878088    -92.8826913    -92.7878088    -92.8497919  
@zts    18 0.162617 0.766201    -92.9076273    -92.7879912    -92.8826934    -92.7879912    -92.8499197  
@zts    19 0.013204 0.060562    -92.9076276    -92.7885097    -92.8826994    -92.7885097    -92.8502675  
@zts    20 0.718205 3.423813    -92.9076278    -92.7905066    -92.8827009    -92.7895258    -92.8514863  
@zts The string calculation failed to converge  
@zts Bead number    1  Potential Energy =     -92.907627751439  
@zts Bead number    2  Potential Energy =     -92.905047596626  
@zts Bead number    3  Potential Energy =     -92.897944354806  
@zts Bead number    4  Potential Energy =     -92.887494117302  
@zts Bead number    5  Potential Energy =     -92.874059841858  
@zts Bead number    6  Potential Energy =     -92.857382758537  
@zts Bead number    7  Potential Energy =     -92.837207959079  
@zts Bead number    8  Potential Energy =     -92.815902497386  
@zts Bead number    9  Potential Energy =     -92.798474907121  
@zts Bead number   10  Potential Energy =     -92.789525765222  
@zts Bead number   11  Potential Energy =     -92.790506632257  
@zts Bead number   12  Potential Energy =     -92.799861168980  
@zts Bead number   13  Potential Energy =     -92.814252430183  
@zts Bead number   14  Potential Energy =     -92.830704548760  
@zts Bead number   15  Potential Energy =     -92.847248091296  
@zts Bead number   16  Potential Energy =     -92.861557132126  
@zts Bead number   17  Potential Energy =     -92.871838446832  
@zts Bead number   18  Potential Energy =     -92.878543965696  
@zts Bead number   19  Potential Energy =     -92.881844751735  
@zts Bead number   20  Potential Energy =     -92.882700859222
</code></pre>
<p>A plotting program (e.g. gnuplot, xmgrace) can be used to look at final
path as well as the the convergence of the path
i.e.,</p>
<pre><code>[WE24397:NEB/Example2/perm] bylaska% gnuplot  

   G N U P L O T  
   Version 4.6 patchlevel 0    last modified 2012-03-04   
   Build System: Darwin x86_64  

   Copyright (C) 1986-1993, 1998, 2004, 2007-2012  
   Thomas Williams, Colin Kelley and many others  

   gnuplot home:     &lt;http://www.gnuplot.info&gt;  
   faq, bugs, etc:   type &quot;help FAQ&quot;  
   immediate help:   type &quot;help&quot;  (plot window: hit 'h')  

Terminal type set to 'aqua'  
gnuplot&gt; set xlabel &quot;Reaction Coordinate&quot;  
gnuplot&gt; set ylabel &quot;Energy (kcal/mol)&quot;  
gnuplot&gt; set yrange [0:100]  
gnuplot&gt; set grid  
gnuplot&gt; set style data linespoints  
gnuplot&gt; plot &quot;hcn-hnc-dft.string_epath&quot; using 1:($2+92.908)*27.2116*23.06,&quot;hcn-hnc-dft.string_final_epath&quot; using 1:($2+92.908)*27.2116*23.06  
gnuplot&gt; 
</code></pre>
<p><a href="hcn-hnc-dft0.png">400px</a></p>
<h2 id="nudged-elastic-band-and-zero-temperature-string-methods-string-tutorial-2">String Tutorial 2:<a class="headerlink" href="#nudged-elastic-band-and-zero-temperature-string-methods-string-tutorial-2" title="Permanent link">&para;</a></h2>
<pre><code>Title &quot;2SiO4H4 --&gt; H3O3Si-O-SiO3H3 + H2O&quot;  
echo  
start sio4h4-dimer  

geometry noautoz noautosym  
Si        -3.90592       -0.11789        0.03791   
O         -2.32450       -0.24327       -0.05259  
O         -4.45956       -1.13247        1.13159  
O         -4.53584       -0.45118       -1.38472  
O         -4.28179        1.37363        0.44838  
Si         1.27960        0.06912        0.14555  
O          2.85122        0.23514        0.32761  
O          0.54278        0.38513        1.52092  
O          0.94484       -1.42248       -0.29913  
O          0.75605        1.07390       -0.97272  
H         -1.66762       -0.74425       -0.29362  
H         -4.05734        2.06481        0.90983  
H         -4.30983       -1.85807        1.57116  
H         -4.43621       -0.88060       -2.12508  
H          3.59374       -0.16315        0.50572  
H          0.36896        0.10990        2.31839  
H          0.53993       -2.15495       -0.09488  
H          0.43207        1.85525       -1.13531  
end  

geometry endgeom  noautoz noautosym  
Si        -3.07373        0.18232       -0.24945  
O         -1.50797        0.23823       -0.53062  
O         -3.36758       -0.93058        0.85023  
O         -3.83958       -0.20093       -1.59101  
O         -3.57993        1.59735        0.27471  
Si        -0.05186        0.25441        0.11277  
O          0.94679       -0.58168       -0.80206  
O         -0.10091       -0.40972        1.55838  
O          1.41035       -3.75872        1.22931  
O          0.47135        1.75206        0.24209  
H          1.03624       -4.62405        0.92620  
H         -3.81554        2.06192        0.96069  
H         -3.97094       -1.38510        1.26383  
H         -4.39754       -0.73964       -1.96563  
H          1.45990       -0.57144       -1.49361  
H         -0.44444       -0.37536        2.34765  
H          2.15751       -4.00850        1.82933  
H          0.77180        2.44229       -0.17616  
end  

nwpw  
  simulation_cell  
     SC 18.0  
  end  
  cutoff 30.0  
  lmbfgs  
end  

string  
  nhist 10  
  nbeads 10  
  maxiter 10  
  stepsize 0.10  
  print_shift 1  

  # don't allow the end points of the path to move  
  freeze1 .true.  
  freezeN .true.  
end  
task pspw string ignore

string  
  # increase the number of images  
  nbeads 20  
  maxiter 20  

  # allow the end points of the path to move  
  freeze1 .false.  
  freezeN .false.  
end  
task pspw string ignore
</code></pre>
<h2 id="nudged-elastic-band-and-zero-temperature-string-methods-string-tutorial-3-combining-neb-and-string-path-optimizations">String Tutorial 3: Combining NEB and String path optimizations<a class="headerlink" href="#nudged-elastic-band-and-zero-temperature-string-methods-string-tutorial-3-combining-neb-and-string-path-optimizations" title="Permanent link">&para;</a></h2></section></section>
                    <section class='print-page md-section' id='section-15' heading-number='15'>
                        <h1>Electronic Structure Analysis<a class='headerlink' href='#section-15' title='Permanent link'></a>
                        </h1>
                    <section class="print-page" id="properties" heading-number="15.1"><h1 id="properties-properties">Properties<a class="headerlink" href="#properties-properties" title="Permanent link">&para;</a></h1>
<h2 id="properties-overview">Overview<a class="headerlink" href="#properties-overview" title="Permanent link">&para;</a></h2>
<p>Properties can be calculated for both the Hartree-Fock and DFT wave
functions. The properties that are available are:</p>
<ul>
<li>Natural bond analysis</li>
<li>Dipole, quadrupole, and octupole moment</li>
<li>Mulliken population analysis and bond order analysis</li>
<li>Electrostatic potential (diamagnetic shielding) at nuclei</li>
<li>Electric field and field gradient at nuclei</li>
<li>Electric field gradients with relativistic effects</li>
<li>Electron and spin density at nuclei</li>
<li>NMR shielding (GIAO method)</li>
<li>NMR hyperfine coupling (Fermi-Contact and Spin-Dipole expectation
    values)</li>
<li>NMR indirect spin-spin coupling</li>
<li>Gshift</li>
<li>Response to electric and magnetic fields (static and dynamic)</li>
<li>Raman</li>
<li>Localization of molecular orbitals</li>
</ul>
<p>The properties module is started when the task directive TASK <theory>
property is defined in the user input file. The input format has the
form:</p>
<pre><code> PROPERTY 
   [property keyword]  
   [CENTER ((com || coc || origin || arb &lt;real x y z&gt;) default coc)] 
 END
</code></pre>
<p>Most of the properties can be computed for Hartree-Fock (closed-shell
RHF, open-shell ROHF, and open-shell UHF), and DFT (closed-shell and
open-shell spin unrestricted) wavefunctions. The NMR hyperfine and
indirect spin-spin coupling require a UHF or ODFT wave function.</p>
<h2 id="properties-vectors-keyword">Vectors keyword<a class="headerlink" href="#properties-vectors-keyword" title="Permanent link">&para;</a></h2>
<pre><code> VECTORS [ (&lt;string input_movecs &gt;)]
</code></pre>
<p>The VECTORS directive allows the user to specify the input molecular
orbital vectors for the property calculation</p>
<h2 id="properties-property-keywords">Property keywords<a class="headerlink" href="#properties-property-keywords" title="Permanent link">&para;</a></h2>
<p>Each property can be requested by defining one of the following
keywords:</p>
<pre><code> NBOFILE  
 DIPOLE  
 QUADRUPOLE  
 OCTUPOLE  
 MULLIKEN  
 ESP  
 EFIELD  
 EFIELDGRAD  
 EFIELDGRADZ4  
 GSHIFT  
 ELECTRONDENSITY  
 HYPERFINE [&lt;integer&gt; number_of_atoms &lt;integer&gt; atom_list]  
 SHIELDING [&lt;integer&gt; number_of_atoms &lt;integer&gt; atom_list]  
 SPINSPIN [&lt;integer&gt; number_of_pairs &lt;integer&gt; pair_list]  
 RESPONSE [&lt;integer&gt; response_order &lt;real&gt; frequency]  
 AIMFILE  
 MOLDENFILE
 LOCALIZATION  
 ALL
</code></pre>
<p>The <code>ALL</code> keyword generates all currently available properties.</p>
<h3 id="properties-nmr-and-epr">NMR and EPR<a class="headerlink" href="#properties-nmr-and-epr" title="Permanent link">&para;</a></h3>
<p>Both the NMR shielding and spin-spin coupling have additional optional
parameters that can be defined in the input. For the shielding the user
can define the number of atoms for which the shielding tensor should be
calculated, followed by the list of specific atom centers. In the case
of spin-spin coupling the number of atom pairs, followed by the atom
pairs, can be defined (i.e., <code>spinspin 1 1 2</code> will calculate the coupling
for one pair, and the coupling will be between atoms 1 and 2).</p>
<p>For both the NMR spin-spin and hyperfine coupling the isotope that has
the highest abundance and has spin, will be chosen for each atom under
consideration.</p>
<h4 id="properties-calculating-epr-and-paramagnetic-nmr-parameters">Calculating EPR and paramagnetic NMR parameters<a class="headerlink" href="#properties-calculating-epr-and-paramagnetic-nmr-parameters" title="Permanent link">&para;</a></h4>
<p>The following
<a href="#epr-pnmr">tutorial</a> illustrates how to combine the
hyperfine, gshift and shielding to calculate the EPR and paramagnetic
NMR parameters of an open-shell system. All calculations are compatible 
with the ZORA model potential approach.</p>
<p>For theoretical and computational details, please refer to 
references<sup id="properties-fnref_footnote_1"><a class="footnote-ref" href="#properties-fn_footnote_1">1</a></sup><sup id="fnref_footnote_2"><a class="footnote-ref" href="#properties-fn_footnote_2">2</a></sup><sup id="fnref_footnote_3"><a class="footnote-ref" href="#properties-fn_footnote_3">3</a></sup>.</p>
<h4 id="properties-nmr-input-example">NMR: Input Example<a class="headerlink" href="#properties-nmr-input-example" title="Permanent link">&para;</a></h4>
<pre><code>geometry  nocenter
  C    0.00000000    0.00000000    0.00000000
  O    1.18337200    0.00000000    0.00000000
  H    -.63151821    0.94387462    0.00000000
end

basis
 &quot;*&quot; library 6-311G**
end
property
 efieldgradz4 1 3
 shielding 2 1 2
 hyperfine 2 1 3
 gshift
end

relativistic
  zora on
  zora:cutoff_NMR 1d-8
  zora:cutoff 1d-30
end

dft
mult 2
xc becke88 perdew86
end

task dft property
</code></pre>
<h3 id="properties-center-center-of-expansion-for-multipole-calculations">CENTER: Center of expansion for multipole calculations<a class="headerlink" href="#properties-center-center-of-expansion-for-multipole-calculations" title="Permanent link">&para;</a></h3>
<p>The user also has the option to choose the center of expansion for the
dipole, quadrupole, and octupole calculations.</p>
<pre><code>   [CENTER ((com || coc || origin || arb &lt;real x y z&gt;) default coc)]
</code></pre>
<p>com is the center of mass, coc is the center of charge, origin is (0.0,
0.0, 0.0) and arb is any arbitrary point which must be accompanied by
the coordinated to be used. Currently the x, y, and z coordinates must
be given in the same units as UNITS in <a href="#geometry">GEOMETRY</a>.</p>
<h3 id="properties-response-calculations">Response Calculations<a class="headerlink" href="#properties-response-calculations" title="Permanent link">&para;</a></h3>
<p>Response calculations can be calculated as
follows:</p>
<pre><code>property
 response  1 7.73178E-2   # response order and frequency in Hartree energy units  
 velocity                 # use modified velocity gauge for electric dipole  
 orbeta                   # calculate optical rotation 'beta' directly [^4]  
 giao                     # GIAO optical rotation [^5][^6][^7], forces orbeta  
 bdtensor                 # calculates B-tilde of Refs. [^5][^7] 
 analysis                 # analyze response in terms of MOs [^7]  
 damping 0.007            # complex response functions with damping, Ref [^8] 
 convergence 1e-4         # set CPKS convergence criterion (default 1e-4)  
end
</code></pre>
<p>Response calculations are currently supported only for   </p>
<ul>
<li>order 1 (linear response),   </li>
<li>single frequency,   </li>
<li>electric field,</li>
<li>mixed electric-magnetic field perturbations.    </li>
</ul>
<p>The output consists of the electric polarizability
and optical rotation tensors (alpha, beta for optical rotation) in
atomic units.<br />
The <code>response</code> keyword requires two arguments: response order and frequency in Hartree energy units
(the <code>aoresponse</code> keyword can be used with same effect as the <code>response</code> keyword).<br />
If the <code>velocity</code> or <code>giao</code> keywords are absent, the
dipole-length form will be used for the dipole integrals. This is a bit
faster. <br />
The isotropic optical rotation is origin independent when using
the velocity gauge (by means of <code>velocity</code> keyword) or with GIAOs <sup id="fnref_footnote_5"><a class="footnote-ref" href="#properties-fn_footnote_5">5</a></sup> (by means of the <code>giao</code> keyword). <br />
With the keyword <code>bdtensor</code>, a
fully origin-invariant optical rotation tensor is calculated <sup id="properties-fnref2_footnote_5"><a class="footnote-ref" href="#properties-fn_footnote_5">5</a></sup><sup id="fnref_footnote_7"><a class="footnote-ref" href="#properties-fn_footnote_7">7</a></sup>.<br />
Note that <code>velocity</code> and <code>orbeta</code> are incompatible. <br />
The input line</p>
<pre><code>set prop:newaoresp 0
</code></pre>
<p>outside of the <code>properties</code> block forces the use of an
older version of the response code, which has fewer features (in
particular, no working GIAO optical rotation) but which has been tested
more thoroughly. In the default newer version you may encounter
undocumented features (bugs).<br />
The keyword <code>analysis</code> triggers an analysis of the
response tensors in terms of molecular orbitals.<br />
If the property input block also contains the keyword <code>pmlocalization</code>, then the analysis is
performed in terms of Pipek-Mezey localized MOs, otherwise the canonical
set is used (this feature may currently not work, please check the sum
of the analysis carefully). See Ref. [6] for an example. Works with HF
and density functionals for which linear response kernels are
implemented in NWChem.</p>
<p>Please refer to papers<sup id="fnref3_footnote_5"><a class="footnote-ref" href="#properties-fn_footnote_5">5</a></sup><sup id="fnref_footnote_4"><a class="footnote-ref" href="#properties-fn_footnote_4">4</a></sup><sup id="fnref_footnote_6"><a class="footnote-ref" href="#properties-fn_footnote_6">6</a></sup><sup id="fnref_footnote_9"><a class="footnote-ref" href="#properties-fn_footnote_9">9</a></sup><sup id="fnref_footnote_8"><a class="footnote-ref" href="#properties-fn_footnote_8">8</a></sup><sup id="fnref2_footnote_7"><a class="footnote-ref" href="#properties-fn_footnote_7">7</a></sup> for further details:</p>
<h3 id="properties-raman">Raman<a class="headerlink" href="#properties-raman" title="Permanent link">&para;</a></h3>
<p>Raman calculations can be performed by specifying the Raman block. These
calculations are performed in conjunction with polarizability
calculations. Detailed description of input parameters at
<a href="https://pubs.acs.org/doi/10.1021/jp411039m#notes-1">https://pubs.acs.org/doi/10.1021/jp411039m#notes-1</a></p>
<pre><code>RAMAN 
 [ (NORMAL | | RESONANCE) default NORMAL ]  
 [ (LORENTZIAN | | GAUSSIAN) default LORENTZIAN ]  
 [ LOW &lt;double low default 0.0&gt; ]  
 [ HIGH &lt;double high default highest normal mode&gt; ]  
 [ FIRST &lt;integer first default 7&gt; ]  
 [ LAST &lt; integer last default number of normal modes &gt; ]  
 [ WIDTH &lt;double width default 20.0&gt; ]  
 [ DQ &lt;double dq default 0.01&gt; ]  
END  
task dft raman
</code></pre>
<p>or</p>
<pre><code>task dft raman numerical
</code></pre>
<p>Sample input block:</p>
<pre><code>property
 response 1 8.8559E-2  
 damping 0.007  
end  
raman  
 normal  
 lorentzian  
end
</code></pre>
<h4 id="properties-raman-keywords">Raman Keywords<a class="headerlink" href="#properties-raman-keywords" title="Permanent link">&para;</a></h4>
<ul>
<li><code>NORMAL</code> and <code>RESONANCE</code>: Type of Raman plot to make.</li>
<li><code>LORENTZIAN</code> and <code>GAUSSIAN</code>: Generation of smoothed spectra (rather than
    sticks) using either a Lorentzian function or a Gaussian function.
    The default is <code>LORENTZIAN</code>.</li>
<li><code>LOW</code> and <code>HIGH</code>: The default range in which to generate the Raman
    spectrum plot is (0.0, highest wavenumber normal mode) cm-1. The <code>LOW</code>
    and <code>HIGH</code> keywords modify the frequency range.</li>
<li><code>FIRST</code> and <code>LAST</code>: The default range of indices of normal modes used in
    the plot is (7, number of normal modes). The <code>FIRST</code> and <code>LAST</code> keywords
    modify the range of indices.</li>
<li><code>WIDTH</code>: Controls the width in the smoothed peaks, using Lorentzians or
    Gaussians, in the plot. The default value for <code>WIDTH</code> is 20.0.</li>
<li><code>DQ</code>: Size of the steps along the normal modes. The default value for
    <code>DQ</code> is 0.01. It is related to the step size dR used in numerical
    evaluation of polarizability derivative</li>
</ul>
<h4 id="properties-raman-output">Raman Output<a class="headerlink" href="#properties-raman-output" title="Permanent link">&para;</a></h4>
<p>Raman spectrum in stick format and smoothed using Lorentzians or
Gaussians stored in a filename with format <code>[fname].normal</code>.<br />
The number of points is 1000 by default. This value can be changed by adding the following <a href="#set">SET</a> directive to the input file</p>
<pre><code>set raman:numpts &lt;integer&gt;
</code></pre>
<h4 id="properties-raman-references">Raman References<a class="headerlink" href="#properties-raman-references" title="Permanent link">&para;</a></h4>
<p>Please refer to papers<sup id="fnref_footnote_10"><a class="footnote-ref" href="#properties-fn_footnote_10">10</a></sup><sup id="fnref_footnote_11"><a class="footnote-ref" href="#properties-fn_footnote_11">11</a></sup> for further details:</p>
<h3 id="properties-polarizability-computed-with-the-sum-over-orbitals-method">Polarizability computed with the Sum over Orbitals method<a class="headerlink" href="#properties-polarizability-computed-with-the-sum-over-orbitals-method" title="Permanent link">&para;</a></h3>
<p>As an alternative to the <a href="#properties-response-calculations">linear response
method</a>, the Sum over Orbitals (SOO) method is
available to compute polarizabilities. Results of these method are
much less accurate than linear response calculations, with values off
by a factor of 2-4x. However, the qualitative nature of this results
can be used to compute Raman frequencies when coupled with
<a href="#gaussian-basis-aimd-property-calculation-in-a-molecular-dynamics-simulation">QMD</a>,
as described in references <sup id="properties-fnref_footnote_12"><a class="footnote-ref" href="#properties-fn_footnote_12">12</a></sup><sup id="fnref_footnote_13"><a class="footnote-ref" href="#properties-fn_footnote_13">13</a></sup>.</p>
<p>Sample input computing polarizability both with the SOO method and the linear response method:</p>
<pre><code>property
 polfromsos
end

task dft property

property
 response 1 0
end
task dft property
</code></pre>
<h3 id="properties-nbofile">Nbofile<a class="headerlink" href="#properties-nbofile" title="Permanent link">&para;</a></h3>
<p>The keyword <code>NBOFILE</code> does not execute the Natural Bond Analysis code, but
simply creates an input file to be used as input to the stand-alone NBO
code. All other properties are calculated upon request.</p>
<p>Following the successful completion of an electronic structure
calculation, a Natural Bond Orbital (NBO) analysis may be carried out by
providing the keyword <code>NBOFILE</code> in the <code>PROPERTY</code> directive. NWChem will
query the rtdb and construct an ASCII file, <em>file_prefix</em><code>.gen</code>, that may
be used as input to the stand alone version of the NBO program, GenNBO.
<em>file_prefix</em> is equal to string following the <code>START</code> directive. The
input deck may be edited to provide additional options to the NBO
calculation, (see the NBO user&rsquo;s manual for details.)</p>
<p>Users that have their own NBO version can compile and link the code into
the NWChem software. See the INSTALL file in the source for details.</p>
<h2 id="properties-gaussian-cube-files">Gaussian Cube Files<a class="headerlink" href="#properties-gaussian-cube-files" title="Permanent link">&para;</a></h2>
<p>Electrostatic potential (keyword <code>esp</code>) and the magnitude of the
electric field (keyword <code>efield</code>) on the grid can be generated in the
form of the Gaussian Cube File. This behavior is triggered by the
inclusion of grid keyword as shown
below</p>
<pre><code> grid [pad dx [dy dz]] [rmax x y z] [rmin x y z] [ngrid nx [ny nz]] [output filename]
</code></pre>
<p>where</p>
<ul>
<li>
<p><code>pad dx [dy dz]</code> - specifies amount of padding (in angstroms) in
    x,y, and z dimensions that will be applied in the automatic
    construction of the rectangular grid volume based on the geometry of
    the system. If only one number is provided then the same amount of
    padding will be applied in all dimensions. The default setting is 4
    angstrom padding in all dimensions.</p>
</li>
<li>
<p><code>rmin x y z</code> - specifies the coordinates (in angstroms) of the minimum
    corner of the rectangular grid volume. This will override any
    padding in this direction.</p>
</li>
<li>
<p><code>rmax x y z</code> - specifies the coordinates (in angstroms) of the maximum
    corner of the rectangular grid volume. This will override any
    padding in this direction.</p>
</li>
<li>
<p><code>ngrid nx [ny nz]</code> - specifies number of grid points along each
    dimension. If only one number is provided then the same number of
    grid points are assumed all dimensions. In the absence of this
    directive the number of grid points would be computed such that grid
    spacing will be close to 0.2 angstrom, but not exceeding 50 grid
    points in either dimension.</p>
</li>
<li>
<p><code>output filename</code> - specifies name of the output cube file. The
    default behavior is to use <em>prefix</em><code>-elp.cube</code> or <em>prefix</em><code>-elf.cube</code>
    file names for electrostatic potential or electric field
    respectively. Here <em>prefix</em> denotes the system name as specified in
    start directive. Note that Gaussian cube files will be written in
    the run directory (where the input file resides).</p>
</li>
</ul>
<p>Example input file</p>
<pre><code>  echo  
  start nacl  


  geometry nocenter noautoz noautosym  
   Na                   -0.00000000     0.00000000    -0.70428494  
   Cl                    0.00000000    -0.00000000     1.70428494  
  end  


  basis  
    * library 6-31g*  
  end  

  #electric field would be written out to nacl.elf.cube file  
  #with  
  #ngrid     : 20 20 20  
  #rmax      : 4.000     4.000     5.704  
  #rmin      :-4.000    -4.000    -4.704  

  property  
  efield  
  grid pad 4.0 ngrid 20  
  end  

  task dft property  

  #electrostatic potential would be written to esp-pad.cube file  
  # with the same parameters as above  

  property  
  esp  
  grid pad 4.0 ngrid 20 output esp-pad.cube  
  end  

  task dft property  

  #illustrating explicit specification of minumum box coordinates  

  property  
  esp  
  grid pad 4.0 rmax 4.000 4.000 5.704 ngrid 20  
  end  

  task dft property
</code></pre>
<h2 id="properties-aimfile">Aimfile<a class="headerlink" href="#properties-aimfile" title="Permanent link">&para;</a></h2>
<p>This keyword generates AIM Wavefunction files. The resulting AIM
wavefunction file (.wfn/.wfx) can be post-processed with a variety of
codes, e.g.</p>
<ul>
<li><a href="http://www.quimica.urv.es/XAIM/">XAIM</a></li>
<li><a href="https://github.com/aoterodelaroza/nciplot">NCIPLOT</a></li>
<li><a href="http://sobereva.com/multiwfn/">Multiwfn</a></li>
<li><a href="https://github.com/aoterodelaroza/postg">Postg</a></li>
</ul>
<p><strong>WARNING:</strong> Since we have discovered issues in generating .WFN files with this module (e.g. systems with ECPs), the recommended method for generating .WFN file is to first generate a Molden file with the <a href="#properties-moldenfile">Moldenfile</a> option, then convert the Molden file into a WFN file by using the <a href="https://github.com/zorkzou/Molden2AIM">Molden2AIM</a> program.</p>
<h2 id="properties-moldenfile">Moldenfile<a class="headerlink" href="#properties-moldenfile" title="Permanent link">&para;</a></h2>
<pre><code>MOLDENFILE
MOLDEN_NORM (JANPA | | NWCHEM || NONE)
</code></pre>
<p>This keyword generates files using the <a href="https://www.theochem.ru.nl/molden/molden_format.html">Molden
format</a>. The resulting
Molden file (.molden) should compatible with a variety of codes that can
input Molden files, e.g.</p>
<ul>
<li><a href="https://www.theochem.ru.nl/molden/">Molden</a></li>
<li><a href="https://janpa.sf.net">JANPA</a> (the nwchem2molden step is no longer
    required when using .molden files and the <code>MOLDEN_NORM JANPA</code>
    keyword)</li>
<li><a href="http://orbkit.github.io/">orbkit</a></li>
<li><a href="http://github.com/Konjkov/molden2qmc">Molden2qmc</a></li>
<li><a href="https://github.com/zorkzou/Molden2AIM">Molden2AIM</a></li>
<li><a href="http://sobereva.com/multiwfn/">Multiwfn</a></li>
</ul>
<p>the <code>MOLDEN_NORM</code> option allows the renormalization of the basis set
coefficients. By default, the coefficient values from input are not
modified. Using the <code>JANPA</code> value coefficients are normalized following
<a href="https://janpa.sourceforge.net/">JANPA</a>&rsquo;s
convention (where basis coefficients are normalized to unity), while the <code>NWCHEM</code> will produce coefficients normalized
according to NWChem&rsquo;s convention. Using <code>MOLDEN_NORM</code> equal <code>NONE</code> will
leave the input coefficients unmodified.<br />
It is strongly recommended to use <strong>spherical</strong> <a href="#basis">basis set</a> when using the NWChem Molden output for JANPA analysis</p>
<p>Example input file for a scf calculation. The resulting Molden file will
be named <code>h2o.molden</code></p>
<pre><code> start heat

 geometry; he 0. 0. 0.; end  

 basis spherical; * library 6-31g ; end  

 task scf  

 property 
  vectors heat.movecs  
  moldenfile   
  molden_norm janpa 
 end

 task scf property
</code></pre>
<p>Then, the resulting <code>h2o.molden</code> file can be post processed by <a href="https://janpa.sourceforge.net/">Janpa</a> with the following command</p>
<pre><code>java -jar janpa.jar h2o.molden &gt; h2o.janpa.txt
</code></pre>
<h2 id="properties-localization">Localization<a class="headerlink" href="#properties-localization" title="Permanent link">&para;</a></h2>
<p>Localized molecular orbitals (LMOs) can be computed with the <code>localization</code> keyword.  </p>
<pre><code> property  
  localization (( pm || boys || ibo) default pm) ((0 || 1 || 2) default 0)
 end  
</code></pre>
<p>The following methods are available:</p>
<ul>
<li>Pipek-Mezey<sup id="fnref_footnote_14"><a class="footnote-ref" href="#properties-fn_footnote_14">14</a></sup>, <code>pm</code> keyword (default) </li>
<li>Foster-Boys<sup id="fnref_footnote_15"><a class="footnote-ref" href="#properties-fn_footnote_15">15</a></sup>, <code>boys</code> keyword </li>
<li>IAO/IBO<sup id="fnref2_footnote_16"><a class="footnote-ref" href="#properties-fn_footnote_16">16</a></sup><sup id="fnref_footnote_17"><a class="footnote-ref" href="#properties-fn_footnote_17">17</a></sup>, <code>ibo</code> keyword </li>
</ul>
<p>An optional integer switch may be supplied after the localization method specification. The default <code>0</code> means only occupied orbitals are localized.  Use <code>1</code> to localize virtual orbitals instead, or <code>2</code> to separately localize occupied and virtual orbitals in the same run. Localization of virtual orbitals is not implemented for the Boys localization. If only the keyword <code>localization</code> is given, this defaults to a Pipek-Mezey localization of the occupied molecular orbitals.</p>
<p>In spin-unrestricted calculations, alpha and beta spin orbitals are localized separately. Fractional-occupation DFT and restricted open-shell Hartree-Fock or Kohn-Sham are not yet supported; attempting to use localization with fractional orbital occupations may run but will likely produce unwanted and wrong results.</p>
<p>A localization run prints a characterizations of the LMOs. Specifically, the PM and IBO localizations print an energy-sorted list of LMOs indicating the density weight of each LMO on the various atoms for weights exceeding a 1% threshold; the atoms are listed in terms of their input number. A Boys localization prints the LMO centroid and the expectation value of <em>r</em>^2 per LMO.</p>
<p>Typically, however, the user will want to visualize the LMOs. The localization run saves the LMO coefficients in a file <code>locorb.movecs</code>, which can be used as the MO vector file in an NWChem <a href="https://nwchemgit.github.io/DPLOT.html">DPLOT task</a> to generate corresponding volume data maps, for example, in the `cube&rsquo; format commonly used in computational chemistry. The latter can be visualized with a considerable number of open source, free, or proprietary software packages.</p>
<p>NWChem test jobs that use the orbital localization functionality (under <code>$NWCHEM_TOP/QA/tests/</code>) are <code>localize-ibo-allyl</code>, <code>localize-pm-allyl</code>, and <code>localize-ibo-aa</code>. The latter is based on the acrylic acid example from the original IBO publication.<sup id="fnref_footnote_16"><a class="footnote-ref" href="#properties-fn_footnote_16">16</a></sup> IBO number 19 represents the delocalized C-C pi bond LMO of the system. After running the test job, the corresponding cube file for LMO 19 can be generated, for example, with</p>
<pre><code> dplot
 vectors locorb.movecs
 limitxyz
 -4.3334997 4.9318898 50
 -3.3065879000000002 3.718637 50
 -2.0 2.0 50
 gaussian
 spin alpha
 orbitals view;1;19
 output localize-ibo-aa-00019.cube
 end
 task dplot
</code></pre>
<p>The resulting plot with &#177;0.03 isosurfaces may look like the following</p>
<p><img alt="Acrylic Acid IBO number 19" src="acrylic-acid-ibo.png" /></p>
<p>A <a href="https://github.com/jautschbach/nwchem-dplot-utility">Python script</a> is available to facilitate the creation of volume data files for visualizing multiple MOs or LMOs of a given molecule. For the acrylic acid example, after running NWChem with the <code>localize-ibo-aa.nw</code> input, the following command will create an input file with <code>dplot</code> tasks to generate cubes for the 19 occupied and 10 virtual IBOs of the molecule on a 50x50x50 grid:</p>
<pre><code> dplot.py -i localize-ibo-aa.nw -m locorb.movecs -g 50 -l 1-29
</code></pre>
<p>The <code>dplot</code> task shown above is included in the resulting NWChem input file <code>dplot.nw</code>.</p>
<h2 id="properties-references">References<a class="headerlink" href="#properties-references" title="Permanent link">&para;</a></h2>
<div class="footnote">
<hr />
<ol>
<li id="properties-fn_footnote_1">
<p>Autschbach, J.; Patchkovskii, S.; Pritchard, B. Calculation of Hyperfine Tensors and Paramagnetic NMR Shifts Using the Relativistic Zeroth-Order Regular Approximation and Density Functional Theory. <em>Journal of Chemical Theory and Computation</em> <strong>2011</strong>, <em>7</em> (7), 21752188. <a href="https://doi.org/10.1021/ct200143w">https://doi.org/10.1021/ct200143w</a>.&#160;<a class="footnote-backref" href="#properties-fnref_footnote_1" title="Jump back to footnote 1 in the text">&#8617;</a></p>
</li>
<li id="properties-fn_footnote_2">
<p>Aquino, F.; Pritchard, B.; Autschbach, J. Scalar Relativistic Computations and Localized Orbital Analyses of Nuclear Hyperfine Coupling and Paramagnetic NMR Chemical Shifts. <em>Journal of Chemical Theory and Computation</em> <strong>2012</strong>, <em>8</em> (2), 598609. <a href="https://doi.org/10.1021/ct2008507">https://doi.org/10.1021/ct2008507</a>.&#160;<a class="footnote-backref" href="#properties-fnref_footnote_2" title="Jump back to footnote 2 in the text">&#8617;</a></p>
</li>
<li id="properties-fn_footnote_3">
<p>Aquino, F.; Govind, N.; Autschbach, J. Scalar Relativistic Computations of Nuclear Magnetic Shielding and &lt;i&gt;g&lt;/i&gt;-Shifts with the Zeroth-Order Regular Approximation and Range-Separated Hybrid Density Functionals. <em>Journal of Chemical Theory and Computation</em> <strong>2011</strong>, <em>7</em> (10), 32783292. <a href="https://doi.org/10.1021/ct200408j">https://doi.org/10.1021/ct200408j</a>.&#160;<a class="footnote-backref" href="#properties-fnref_footnote_3" title="Jump back to footnote 3 in the text">&#8617;</a></p>
</li>
<li id="properties-fn_footnote_4">
<p>Autschbach. Computation of Optical Rotation Using Timedependent Density Functional Theory. <em>Computing Letters</em> <strong>2007</strong>, <em>3</em> (2), 131150. <a href="https://doi.org/10.1163/157404007782913327">https://doi.org/10.1163/157404007782913327</a>.&#160;<a class="footnote-backref" href="#properties-fnref_footnote_4" title="Jump back to footnote 4 in the text">&#8617;</a></p>
</li>
<li id="properties-fn_footnote_5">
<p>Autschbach, J. Time-Dependent Density Functional Theory for Calculating Origin-Independent Optical Rotation and Rotatory Strength Tensors. <em>ChemPhysChem</em> <strong>2011</strong>, <em>12</em> (17), 32243235. <a href="https://doi.org/10.1002/cphc.201100225">https://doi.org/10.1002/cphc.201100225</a>.&#160;<a class="footnote-backref" href="#properties-fnref_footnote_5" title="Jump back to footnote 5 in the text">&#8617;</a><a class="footnote-backref" href="#properties-fnref2_footnote_5" title="Jump back to footnote 5 in the text">&#8617;</a><a class="footnote-backref" href="#properties-fnref3_footnote_5" title="Jump back to footnote 5 in the text">&#8617;</a></p>
</li>
<li id="properties-fn_footnote_6">
<p>Krykunov, M.; Autschbach, J. Calculation of Optical Rotation with Time-Periodic Magnetic-Field-Dependent Basis Functions in Approximate Time-Dependent Density-Functional Theory. <em>The Journal of Chemical Physics</em> <strong>2005</strong>, <em>123</em> (11), 114103. <a href="https://doi.org/10.1063/1.2032428">https://doi.org/10.1063/1.2032428</a>.&#160;<a class="footnote-backref" href="#properties-fnref_footnote_6" title="Jump back to footnote 6 in the text">&#8617;</a></p>
</li>
<li id="properties-fn_footnote_7">
<p>Moore, B.; Srebro, M.; Autschbach, J. Analysis of Optical Activity in Terms of Bonds and Lone-Pairs: The Exceptionally Large Optical Rotation of Norbornenone. <em>Journal of Chemical Theory and Computation</em> <strong>2012</strong>, <em>8</em> (11), 43364346. <a href="https://doi.org/10.1021/ct300839y">https://doi.org/10.1021/ct300839y</a>.&#160;<a class="footnote-backref" href="#properties-fnref_footnote_7" title="Jump back to footnote 7 in the text">&#8617;</a><a class="footnote-backref" href="#properties-fnref2_footnote_7" title="Jump back to footnote 7 in the text">&#8617;</a></p>
</li>
<li id="properties-fn_footnote_8">
<p>Krykunov, M.; Kundrat, M. D.; Autschbach, J. Calculation of Circular Dichroism Spectra from Optical Rotatory Dispersion, and Vice Versa, as Complementary Tools for Theoretical Studies of Optical Activity Using Time-Dependent Density Functional Theory. <em>The Journal of Chemical Physics</em> <strong>2006</strong>, <em>125</em> (19), 194110. <a href="https://doi.org/10.1063/1.2363372">https://doi.org/10.1063/1.2363372</a>.&#160;<a class="footnote-backref" href="#properties-fnref_footnote_8" title="Jump back to footnote 8 in the text">&#8617;</a></p>
</li>
<li id="properties-fn_footnote_9">
<p>Hammond, J. R.; Govind, N.; Kowalski, K.; Autschbach, J.; Xantheas, S. S. Accurate Dipole Polarizabilities for Water Clusters n=212 at the Coupled-Cluster Level of Theory and Benchmarking of Various Density Functionals. <em>The Journal of Chemical Physics</em> <strong>2009</strong>, <em>131</em> (21), 214103. <a href="https://doi.org/10.1063/1.3263604">https://doi.org/10.1063/1.3263604</a>.&#160;<a class="footnote-backref" href="#properties-fnref_footnote_9" title="Jump back to footnote 9 in the text">&#8617;</a></p>
</li>
<li id="properties-fn_footnote_10">
<p>Mullin, J. M.; Autschbach, J.; Schatz, G. C. Time-Dependent Density Functional Methods for Surface Enhanced Raman Scattering (SERS) Studies. <em>Computational and Theoretical Chemistry</em> <strong>2012</strong>, <em>987</em>, 3241. <a href="https://doi.org/10.1016/j.comptc.2011.08.027">https://doi.org/10.1016/j.comptc.2011.08.027</a>.&#160;<a class="footnote-backref" href="#properties-fnref_footnote_10" title="Jump back to footnote 10 in the text">&#8617;</a></p>
</li>
<li id="properties-fn_footnote_11">
<p>Aquino, F. W.; Schatz, G. C. Time-Dependent Density Functional Methods for Raman Spectra in Open-Shell Systems. <em>The Journal of Physical Chemistry A</em> <strong>2014</strong>, <em>118</em> (2), 517525. <a href="https://doi.org/10.1021/jp411039m">https://doi.org/10.1021/jp411039m</a>.&#160;<a class="footnote-backref" href="#properties-fnref_footnote_11" title="Jump back to footnote 11 in the text">&#8617;</a></p>
</li>
<li id="properties-fn_footnote_12">
<p>Fischer, S. A.; Ueltschi, T. W.; El-Khoury, P. Z.; Mifflin, A. L.; Hess, W. P.; Wang, H.-F.; Cramer, C. J.; Govind, N. Infrared and Raman Spectroscopy from Ab Initio Molecular Dynamics and Static Normal Mode Analysis: The C-H Region of DMSO as a Case Study. <em>The Journal of Physical Chemistry B</em> <strong>2015</strong>, <em>120</em> (8), 14291436. <a href="https://doi.org/10.1021/acs.jpcb.5b03323">https://doi.org/10.1021/acs.jpcb.5b03323</a>.&#160;<a class="footnote-backref" href="#properties-fnref_footnote_12" title="Jump back to footnote 12 in the text">&#8617;</a></p>
</li>
<li id="properties-fn_footnote_13">
<p>Apr, E.; Bhattarai, A.; Baxter, E.; Wang, S.; Johnson, G. E.; Govind, N.; El-Khoury, P. Z. Simplified Ab Initio Molecular Dynamics-Based Raman Spectral Simulations. <em>Applied Spectroscopy</em> <strong>2020</strong>, <em>74</em> (11), 13501357. <a href="https://doi.org/10.1177/0003702820923392">https://doi.org/10.1177/0003702820923392</a>.&#160;<a class="footnote-backref" href="#properties-fnref_footnote_13" title="Jump back to footnote 13 in the text">&#8617;</a></p>
</li>
<li id="properties-fn_footnote_14">
<p>Pipek, J.; Mezey, P. G. A Fast Intrinsic Localization Procedure Applicable for Ab Initio and Semiempirical Linear Combination of Atomic Orbital Wave Functions. <em>The Journal of Chemical Physics</em> <strong>1989</strong>, <em>90</em> (9), 49164926. <a href="https://doi.org/10.1063/1.456588">https://doi.org/10.1063/1.456588</a>.&#160;<a class="footnote-backref" href="#properties-fnref_footnote_14" title="Jump back to footnote 14 in the text">&#8617;</a></p>
</li>
<li id="properties-fn_footnote_15">
<p>Foster, J. M.; Boys, S. F. Canonical Configurational Interaction Procedure. <em>Reviews of Modern Physics</em> <strong>1960</strong>, <em>32</em> (2), 300302. <a href="https://doi.org/10.1103/revmodphys.32.300">https://doi.org/10.1103/revmodphys.32.300</a>.&#160;<a class="footnote-backref" href="#properties-fnref_footnote_15" title="Jump back to footnote 15 in the text">&#8617;</a></p>
</li>
<li id="properties-fn_footnote_16">
<p>Knizia, G. Intrinsic Atomic Orbitals: An Unbiased Bridge Between Quantum Theory and Chemical Concepts. <em>Journal of Chemical Theory and Computation</em> <strong>2013</strong>, <em>9</em> (11), 48344843. <a href="https://doi.org/10.1021/ct400687b">https://doi.org/10.1021/ct400687b</a>.&#160;<a class="footnote-backref" href="#properties-fnref_footnote_16" title="Jump back to footnote 16 in the text">&#8617;</a><a class="footnote-backref" href="#properties-fnref2_footnote_16" title="Jump back to footnote 16 in the text">&#8617;</a></p>
</li>
<li id="properties-fn_footnote_17">
<p>Knizia, G.; Klein, J. E. M. N. Electron Flow in Reaction Mechanisms-Revealed from First Principles. <em>Angewandte Chemie International Edition</em> <strong>2015</strong>, <em>54</em> (18), 55185522. <a href="https://doi.org/10.1002/anie.201410637">https://doi.org/10.1002/anie.201410637</a>.&#160;<a class="footnote-backref" href="#properties-fnref_footnote_17" title="Jump back to footnote 17 in the text">&#8617;</a></p>
</li>
</ol>
</div></section><section class="print-page" id="esp" heading-number="15.2"><h1 id="esp-electrostatic-potentials">Electrostatic potentials<a class="headerlink" href="#esp-electrostatic-potentials" title="Permanent link">&para;</a></h1>
<h2 id="esp-overview">Overview<a class="headerlink" href="#esp-overview" title="Permanent link">&para;</a></h2>
<p>The NWChem Electrostatic Potential (ESP) module derives partial atomic
charges that fit the quantum mechanical electrostatic potential on
selected grid points.</p>
<p>The ESP module is specified by the NWChem task directive</p>
<pre><code>task esp
</code></pre>
<p>The input for the module is taken from the ESP input block</p>
<pre><code>ESP  
  ... 
END
</code></pre>
<h2 id="esp-grid-specification">Grid specification<a class="headerlink" href="#esp-grid-specification" title="Permanent link">&para;</a></h2>
<p>The grid points for which the quantum mechanical electrostatic potential
is evaluated and used in the fitting procedure of the partial atomic
charges all lie outside the van der Waals radius of the atoms and within
a cutoff distance from the atomic centers. The following input
parameters determine the selection of grid points.</p>
<ul>
<li>If a grid file is found, the grid will be read from that file. If no
    grid file is found, or the keyword</li>
</ul>
<pre><code>       recalculate  
</code></pre>
<p>is given, the grid and the electrostatic potential is recalculated.</p>
<ul>
<li>The extent of the grid is determined by the <code>range</code> keyword</li>
</ul>
<pre><code>       range &lt;real rcut&gt; 
</code></pre>
<p>where <code>rcut</code> is the maximum distance in nm between a grid point and any of the atomic centers. 
When omitted, a default value for rcut of 0.3 nm is used.</p>
<ul>
<li>The grid spacing is specified by the <code>spacing</code> keyword</li>
</ul>
<pre><code>       spacing &lt;real spac&gt;  
</code></pre>
<p>where <code>spac</code> is the grid spacing in <em>nm</em> for the regularly spaced grid points.
If not specified, a default spacing of 0.05 nm is used.</p>
<ul>
<li>The van der Waals <code>radius</code> of an element can be specified by</li>
</ul>
<pre><code>       radius &lt;integer iatnum&gt; &lt;real atrad&gt;  
</code></pre>
<p>where iatnum is the atomic number for which a van der Waals radius of atrad in nm will be used in the grid point determination.<br />
Default values will be used for atoms not specified.</p>
<ul>
<li>The <code>probe</code> radius in nm determining the envelope around the molecule is specified by</li>
</ul>
<pre><code>       probe &lt;real probe default 0.07&gt;
</code></pre>
<ul>
<li>The distance between atomic center and probe center can be
    multiplied by a constant <code>factor</code> specified by</li>
</ul>
<pre><code>       factor &lt;real factor default 1.0&gt;
</code></pre>
<p>All grid points are discarded that lie within a distance <code>factor*(radius(i)+probe)</code> from any atom <strong>i</strong>.</p>
<ul>
<li>Schwarz screening is applied using</li>
</ul>
<pre><code>      screen [&lt;real scrtol default 1.0D-5&gt;]
</code></pre>
<h2 id="esp-constraints">Constraints<a class="headerlink" href="#esp-constraints" title="Permanent link">&para;</a></h2>
<p>Additional constraints to the partial atomic charges can be imposed
during the fitting procedure.</p>
<ul>
<li>The net charge of a subset of atoms can be constrained
using</li>
</ul>
<pre><code>       constrain &lt;real charge {&lt;integer iatom&gt;}
</code></pre>
<p>where <code>charge</code> is the net charge of the set of atoms <code>{iatom}</code>. A negative atom number iatom can be<br />
used to specify that the partial charge of that atom is substracted in the sum for the set.</p>
<ul>
<li>The net charge of a sequence of atoms can be constrained using</li>
</ul>
<pre><code>       constrain &lt;real charge&gt; &lt;integer iatom&gt; through &lt;integer jatom&gt; 
</code></pre>
<p>where charge is the net charge of the set of atoms <code>{[iatom:jatom]}</code>.</p>
<ul>
<li>A group of atoms can be constrained to have the same charge with</li>
</ul>
<pre><code>       constrain equal {&lt;integer iatom&gt;}
</code></pre>
<ul>
<li>The individual charge of a group of atoms can be constrained to be equal to those of a second group of atoms
with</li>
</ul>
<pre><code>      constrain group  &lt;integer iatom&gt; &lt;integer jatom&gt;  to  &lt;integer katom&gt; &lt;integer latom&gt;
</code></pre>
<p>resulting in the same charge for atoms iatom and katom, for atoms iatom+1 and k atom+1, &hellip; for atoms jatom and latom.</p>
<ul>
<li>A special constraint</li>
</ul>
<pre><code>       constrain xhn  &lt;integer iatom&gt; {&lt;integer jatom&gt;}
</code></pre>
<p>can be used to constrain the set <code>{iatom,{jatom}}</code> to zero charge, and constrain all atoms  in <code>{jatom}</code>
to have the same charge. This can be used, for example, to restrain a methyl    group to zero charge,
and have all hydrogen carrying identical charges.</p>
<h2 id="esp-restraints">Restraints<a class="headerlink" href="#esp-restraints" title="Permanent link">&para;</a></h2>
<p>Restraints can be applied to each partial charge using the RESP charge
fitting procedure.</p>
<ul>
<li>The directive for charge restraining is</li>
</ul>
<pre><code>      restrain [hfree] (harmonic [&lt;real scale&gt;] | \ 
      hyperbolic [&lt;real scale&gt; [&lt;real tight&gt;]]  \  
       [maxiter &lt;integer maxit&gt;]  [tolerance &lt;real toler&gt;])
</code></pre>
<p>Here <code>hfree</code> can be specified to exclude hydrogen atoms from the
restraining procedure. Variable <code>scale</code> is the strength of the restraint
potential, with a default of 0.005 au for the <code>harmonic</code> restraint and a
default value of 0.001 au for the <code>hyperbolic</code> restraint. For the
hyperbolic restraints the tightness <code>tight</code> can be specified to change the
default value of 0.1 e. The iteration count that needs to be carried out
for the hyperbolic restraint is determined by the maximum number of
allowed iterations <code>maxiter</code>, with a default value of 25, and the
<code>tolerance</code> in the convergence of the partial charges <code>toler</code>, with a
default of 0.001 e.</p></section><section class="print-page" id="dplot" heading-number="15.3"><h1 id="dplot-dplot">DPLOT<a class="headerlink" href="#dplot-dplot" title="Permanent link">&para;</a></h1>
<h2 id="dplot-overview">Overview<a class="headerlink" href="#dplot-overview" title="Permanent link">&para;</a></h2>
<pre><code> DPLOT  
   ...  
 END
</code></pre>
<p>This directive is used to obtain the plots of various types of electron
densities (or orbitals) of the molecule. The electron density is
calculated on a specified set of grid points using the molecular
orbitals from SCF or DFT calculation. The output file is either in MSI
Insight II contour format (default) or in the <a href="http://paulbourke.net/dataformats/cube/">Gaussian Cube format</a>.
DPLOT is not executed until the <code>task dplot</code> directive is given.
Different sub-directives are described below.</p>
<p>The implementation of the dplot functionality uses mostly local memory.
The quantities stored in local memory are:</p>
<ul>
<li>the positions of the grid points</li>
<li>the property of interest expressed at the grid points</li>
</ul>
<p>The stack memory setting in the input file must be sufficient to hold
these quantities in local memory on a processor.</p>
<h2 id="dplot-gaussian-gaussian-cube-format">GAUSSIAN: Gaussian Cube format<a class="headerlink" href="#dplot-gaussian-gaussian-cube-format" title="Permanent link">&para;</a></h2>
<pre><code> GAUSSIAN
</code></pre>
<p>A outputfile is generate in Gaussian Cube format. You can visualize this
file using gOpenMol (after converting the Gaussian Cube file with
gcube2plt), Molden or Molekel.</p>
<h2 id="dplot-title-title-directive">TITLE: Title directive<a class="headerlink" href="#dplot-title-title-directive" title="Permanent link">&para;</a></h2>
<pre><code> TITLE &lt;string Title default Unknown Title&gt;
</code></pre>
<p>This sub-directive specifies a title line for the generated input to the
Insight program or for the Gaussian cube file. Only one line is allowed.</p>
<h2 id="dplot-limitxyz-plot-limits">LIMITXYZ: Plot limits<a class="headerlink" href="#dplot-limitxyz-plot-limits" title="Permanent link">&para;</a></h2>
<pre><code> LIMITXYZ [units &lt;string Units default angstroms&gt;]  
  &lt;real X_From&gt; &lt;real X_To&gt; &lt;integer No_Of_Spacings_X&gt;  
   &lt;real Y_From&gt; &lt;real Y_To&gt; &lt;integer No_Of_Spacings_Y&gt;  
   &lt;real Z_From&gt; &lt;real Z_To&gt; &lt;integer No_Of_Spacings_Z&gt;
</code></pre>
<p>This sub-directive specifies the limits of the cell to be plotted. The
grid is generated using No_Of_Spacings + 1 points along each
direction. The known names for Units are angstroms, au and bohr.</p>
<h2 id="dplot-spin-density-to-be-plotted">SPIN: Density to be plotted<a class="headerlink" href="#dplot-spin-density-to-be-plotted" title="Permanent link">&para;</a></h2>
<pre><code> SPIN &lt;string Spin default total&gt;
</code></pre>
<p>This sub-directive specifies what kind of density is to be computed.
The known names for Spin are <code>total</code>, <code>alpha</code>, <code>beta</code> and <code>spindens</code>, the last
being computed as the difference between  and  electron densities.</p>
<h2 id="dplot-output-filename">OUTPUT: Filename<a class="headerlink" href="#dplot-output-filename" title="Permanent link">&para;</a></h2>
<pre><code> OUTPUT &lt;string File_Name default dplot&gt;
</code></pre>
<p>This sub-directive specifies the name of the generated input to the
Insight program or the generated Gaussian cube file. The name <code>OUTPUT</code> is
reserved for the standard NWChem output.</p>
<h2 id="dplot-vectors-mo-vector-file-name">VECTORS: MO vector file name<a class="headerlink" href="#dplot-vectors-mo-vector-file-name" title="Permanent link">&para;</a></h2>
<pre><code> VECTORS &lt;string File_Name default movecs&gt; [&lt;string File_Name2&gt;]
</code></pre>
<p>This sub-directive specifies the name of the molecular orbital file. If
the second file is optionally given the density is computed as the
difference between the corresponding electron densities. The vector
files have to match.</p>
<h2 id="dplot-where-density-evaluation">WHERE: Density evaluation<a class="headerlink" href="#dplot-where-density-evaluation" title="Permanent link">&para;</a></h2>
<pre><code> WHERE &lt;string Where default grid&gt;
</code></pre>
<p>This sub-directive specifies where the density is to be computed. The
known names for <code>where</code> are <code>grid</code> (the calculation of the density is
performed on the set of a grid points specified by the sub-directive
<code>LimitXYZ</code> and the file specified by the sub-directive Output is
generated), <code>nuclei</code> (the density is computed at the position of the
nuclei and written to the NWChem output) and <code>g+n</code> (both).</p>
<h2 id="dplot-orbital-orbital-sub-space">ORBITAL: Orbital sub-space<a class="headerlink" href="#dplot-orbital-orbital-sub-space" title="Permanent link">&para;</a></h2>
<pre><code> ORBITALS [&lt;string Option default density&gt;]
 &lt;integer No_Of_Orbitals&gt;  
 &lt;integer Orb_No_1 Orb_No_2 ...&gt;
</code></pre>
<p>This sub-directive specifies the subset of the orbital space for the
calculation of the electron density. The density is computed using the
occupation numbers from the orbital file modified according to the <code>spin</code>
directive. If the contours of the orbitals are to be plotted Option
should be set to <code>view</code> (instead of the implicit default value
<code>density</code>). Note, that in this case No_Of_Orbitals should be set to 1
and sub-directive <code>where</code> is automatically set to <code>grid</code>. Also specification
of two orbital files conflicts with the view option. alpha orbitals are
always plotted unless <code>spin</code> is set to <code>beta</code>.</p>
<h2 id="dplot-civecs-ci-vectors">CIVECS: CI vectors<a class="headerlink" href="#dplot-civecs-ci-vectors" title="Permanent link">&para;</a></h2>
<pre><code>CIVECS [&lt;string Name of civecs file&gt;]
</code></pre>
<h2 id="dplot-densmat-density-matrix">DENSMAT: Density matrix<a class="headerlink" href="#dplot-densmat-density-matrix" title="Permanent link">&para;</a></h2>
<pre><code>DENSMAT [&lt;string Name of density matrix file&gt;]
</code></pre>
<h2 id="dplot-examples">Examples<a class="headerlink" href="#dplot-examples" title="Permanent link">&para;</a></h2>
<h3 id="dplot-charge-density">Charge Density<a class="headerlink" href="#dplot-charge-density" title="Permanent link">&para;</a></h3>
<h3 id="dplot-example-of-hf-charge-density-plot-with-gaussian-cube-output">Example of HF charge density plot (with Gaussian Cube output)<a class="headerlink" href="#dplot-example-of-hf-charge-density-plot-with-gaussian-cube-output" title="Permanent link">&para;</a></h3>
<pre><code>start n2  
geometry  
  n  0 0   0.53879155 
  n  0 0  -0.53879155  
end  
basis;  n library cc-pvdz;end 
scf  
vectors  output n2.movecs  
end  
dplot  
  TITLE HOMO  
  vectors n2.movecs  
   LimitXYZ  
 -3.0 3.0 10    
 -3.0 3.0 10   
 -3.0  3.0  10  
  spin total 
  gaussian  
  output chargedensity.cube  
end  
task scf       
task dplot
</code></pre>
<h3 id="dplot-example-of-ccsd-charge-density-plot-with-gaussian-cube-output">Example of CCSD charge density plot (with Gaussian Cube output)<a class="headerlink" href="#dplot-example-of-ccsd-charge-density-plot-with-gaussian-cube-output" title="Permanent link">&para;</a></h3>
<pre><code>start n2
geometry
 n 0 0  0.53879155
 n 0 0 -0.53879155
 symmetry c2v
end
basis; n library cc-pvdz;end
tce
 ccsd
 densmat n2.densmat
end
task tce energy
dplot
 TITLE HOMO
 LimitXYZ
 -3.0 3.0 10
 -3.0 3.0 10
 -3.0 3.0 10
 spin total
 gaussian
 densmat n2.densmat
 output ccsddensity.cube
end
task dplot
</code></pre>
<h3 id="dplot-molecular-orbital">Molecular Orbital<a class="headerlink" href="#dplot-molecular-orbital" title="Permanent link">&para;</a></h3>
<p>Example of orbital plot (with Insight II contour output):</p>
<pre><code>start n2  
geometry 
  n  0 0   0.53879155  
  n  0 0  -0.53879155  
end  
basis;  n library cc-pvdz;end 
scf 
vectors  output n2.movecs  
end  
dplot  
  TITLE HOMO  
  vectors n2.movecs  
   LimitXYZ  
 -3.0 3.0 10    
 -3.0 3.0 10   
 -3.0  3.0  10  
  spin total  
  orbitals view; 1; 7  
  output homo.grd  
end  
task scf       
task dplot
</code></pre>
<h3 id="dplot-transition-density">Transition Density<a class="headerlink" href="#dplot-transition-density" title="Permanent link">&para;</a></h3>
<p>TDDFT calculation followed by a calculation of the transition density
for a specific excited state using the DPLOT block</p>
<pre><code>echo  
start h2o-td  
title h2o-td  
memory stack 400 mb heap 50 mb global 350 mb  
charge 0  
geometry units au noautoz nocenter  
symmetry group c1  
 O    0.00000000000000      0.00000000000000      0.00000000000000  
 H    0.47043554760291      1.35028113274600      1.06035416576826  
 H   -1.74335410533480     -0.23369304784300      0.27360785442967  
end  
basis &quot;ao basis&quot; print  
 H    S  
    13.0107010              0.19682158E-01  
     1.9622572              0.13796524  
     0.44453796             0.47831935  
 H    S  
     0.12194962             1.0000000  
 H    P  
     0.8000000              1.0000000  
 O    S  
  2266.1767785             -0.53431809926E-02  
   340.87010191            -0.39890039230E-01  
    77.363135167           -0.17853911985  
    21.479644940           -0.46427684959  
     6.6589433124          -0.44309745172  
 O    S  
     0.80975975668          1.0000000  
 O    S  
     0.25530772234          1.0000000  
 O    P  
    17.721504317            0.43394573193E-01  
     3.8635505440           0.23094120765  
     1.0480920883           0.51375311064  
 O    P  
     0.27641544411          1.0000000  
 O    D  
     1.2000000              1.0000000  
end  
dft  
 xc bhlyp  
 grid fine  
 direct  
 convergence energy 1d-5  
end  
tddft  
 rpa  
 nroots 5  
 thresh 1d-5  
 singlet  
 notriplet  
 civecs  
end  
task tddft energy  
dplot  
 civecs h2o-td.civecs_singlet  
 root 2  
 LimitXYZ  
  -3.74335 2.47044 50  
  -2.23369 3.35028 50  
  -2 3.06035 50  
   gaussian  
   output root-2.cube  
end  
task dplot
</code></pre>
<h3 id="dplot-plot-the-excited-state-density">Plot the excited state density<a class="headerlink" href="#dplot-plot-the-excited-state-density" title="Permanent link">&para;</a></h3>
<pre><code>echo  
start tddftgrad_co_exden  
geometry  
 C     0.00000000     0.00000000    -0.64628342  
 O     0.00000000     0.00000000     0.48264375  
 symmetry c1  
end  
basis spherical  
 * library &quot;3-21G&quot;  
end  
dft  
 xc pbe0  
 direct  
end  
tddft  
 nroots 3  
 notriplet  
 target 1  
 civecs  
 grad  
  root 1  
 end  
end  
task tddft gradient  
dplot  
 densmat tddftgrad_co_exden.dmat  
 LimitXYZ  
-4.0 4.0 50  
-4.0 4.0 50  
-4.0 4.0 50  
 gaussian  
 output co_exden.cube  
end  
task dplot
</code></pre></section></section>
                    <section class='print-page md-section' id='section-16' heading-number='16'>
                        <h1>Other-Capabilities<a class='headerlink' href='#section-16' title='Permanent link'></a>
                        </h1>
                    <section class="print-page" id="electron-transfer" heading-number="16.1"><h1 id="electron-transfer-electron-transfer">Electron Transfer<a class="headerlink" href="#electron-transfer-electron-transfer" title="Permanent link">&para;</a></h1>
<p>The NWChem electron transfer (ET) module calculates the electronic
coupling energy (also called the electron transfer matrix element)
between ET reactant and product states. The electronic coupling
(<i>V<sub>RP</sub></i>), and nuclear
reorganization energy () are all components of the electron transfer
rate defined by Marcus&rsquo; theory, which also depends on the temperature
(see Reference 1
below):</p>
<div class="arithmatex">\[{k_{ET}}= \frac{2\pi}{\hbar} V_{RP}^{2} \frac{1}{\sqrt{4\pi \lambda k_{B}T}} \exp \left( \frac{- \Delta G^{*}}{k_{B} T} \right)\]</div>
<p>The ET module utilizes the method of <em>Corresponding Orbital
Transformation</em> to calculate <i>V<sub>RP</sub></i>. The only input required are the
names of the files containing the open-shell (UHF) MO vectors for the ET
reactant and product states (<em>R</em> and <em>P</em>).</p>
<p>The basis set used in the calculation of <i>V<sub>RP</sub></i> must be the same as
the basis set used to calculate the MO vectors of <em>R</em> and <em>P</em>. The
magnitude of <i>V<sub>RP</sub></i> depends on the amount of overlap between <em>R</em> and
<em>P</em>, which is important to consider when choosing the basis set. Diffuse
functions may be necessary to fill in the overlap, particularly when the
ET distance is long.</p>
<p>The MO&rsquo;s of <em>R</em> and <em>P</em> must correspond to localized states. for
instance, in the reaction A<sup>-</sup> + B &rarr; A + B<sup>-</sup>  the
transferring electron is localized on A in the reactant state and is
localized on B in the product state. To verify the localization of the
electron in the calculation of the vectors, carefully examine the
Mulliken population analysis. In order to determine which orbitals are
involved in the electron transfer, use the print keyword &ldquo;mulliken ao&rdquo;
which prints the Mulliken population of each basis function.</p>
<p>An effective core potential (ECP) basis can be used to replace core
electrons. However, there is one caveat: the orbitals involved in
electron transfer must not be replaced with ECP&rsquo;s. Since the ET orbitals
are valence orbitals, this is not usually a problem, but the user should
use ECP&rsquo;s with care.</p>
<p>Suggested references are listed below. The first two references gives a
good description of Marcus&rsquo; two-state ET model, and the appendix of the
third reference details the method used in the ET module.</p>
<ol>
<li>R.A. Marcus, N. Sutin, Biochimica Biophysica Acta 35, 437, (1985).</li>
<li>J.R. Bolton, N. Mataga, and G. McLendon in &ldquo;Electron Transfer in
    Inorganic, Organic and Biological Systems&rdquo; (American Chemical
    Society, Washington, D.C., 1991)</li>
<li>A. Farazdel, M. Dupuis, E. Clementi, and A. Aviram, J. Am. Chem.
    Soc., 112, 4206 (1990).</li>
</ol>
<h2 id="electron-transfer-vectors-input-of-mo-vectors-for-et-reactant-and-product-states">VECTORS: input of MO vectors for ET reactant and product states<a class="headerlink" href="#electron-transfer-vectors-input-of-mo-vectors-for-et-reactant-and-product-states" title="Permanent link">&para;</a></h2>
<pre><code> VECTORS [reactants] &lt;string reactants_filename&gt;  
 VECTORS [products ] &lt;string products_filename&gt;
</code></pre>
<p>In the VECTORS directive the user specifies the source of the molecular
orbital vectors for the ET reactant and product states. This is required
input, as no default filename will be set by the program. In fact, this
is the only required input in the ET module, although there are other
optional keywords described
below.</p>
<h2 id="electron-transfer-focknofock-method-for-calculating-the-two-electron-contribution-to-vrp">FOCK/NOFOCK: method for calculating the two-electron contribution to  <i>V<sub>RP</sub></i><a class="headerlink" href="#electron-transfer-focknofock-method-for-calculating-the-two-electron-contribution-to-vrp" title="Permanent link">&para;</a></h2>
<pre><code>  &lt;string (FOCK||NOFOCK) default FOCK&gt;
</code></pre>
<p>This directive enables/disables the use of the NWChem&rsquo;s Fock matrix
routine in the calculation of the two-electron portion of the ET
Hamiltonian. Since the Fock matrix routine has been optimized for speed,
accuracy and parallel performance, it is the most efficient choice.</p>
<p>Alternatively, the user can calculate the two-electron contribution to
the ET Hamiltonian with another subroutine which may be more accurate
for systems with a small number of basis functions, although it is
slower.</p>
<h2 id="electron-transfer-tol2e-integral-screening-threshold">TOL2E: integral screening threshold<a class="headerlink" href="#electron-transfer-tol2e-integral-screening-threshold" title="Permanent link">&para;</a></h2>
<pre><code> TOL2E &lt;real tol2e default max(10e-12,min(10e-7, S(RP)*10e-7 )&gt;
</code></pre>
<p>The variable tol2e is used in determining the integral screening
threshold for the evaluation of the two-electron contribution to the
Hamiltonian between the electron transfer reactant and product states.
As a default, tol2e is set depending on the magnitude of the overlap
between the ET reactant and product states (<i>S<sub>RP</sub></i>), and is not less
than 1.0d-12 or greater than 1.0d-7.</p>
<p>The input to specify the threshold explicitly within the ET directive
is, for example:</p>
<pre><code> tol2e 1e-9
</code></pre>
<!---  ##  FMO: Fragment Orbital Approach 

The keyword <tt>FMO</tt> can be used to compute electronic couplings by means of the fragment orbital approach
(EF Valeev, V Coropceanu, DA da Silva Filho, S Salman, JL Bredas,. J. Am. Chem. Soc., 128, 9882 (2006))


<pre><code>fmo
</code></pre>

An example input file can be found in  the [etrans-fmo QA directory](https://github.com/nwchemgit/nwchem/tree/master/QA/tests/etrans-fmo)
-->
<h2 id="electron-transfer-example">Example<a class="headerlink" href="#electron-transfer-example" title="Permanent link">&para;</a></h2>
<p>The following example is for a simple electron transfer reaction,
He &rarr; He<sup>+</sup>. The ET calculation is easy to execute,
but it is crucial that ET reactant and product wavefunctions reflect
localized states. This can be accomplished using either a <a href="#hartree-fock-theory-for-molecules-vectors-fragment-superposition-of-fragment-molecular-orbitals">fragment
guess</a>, or
a <a href="#hartree-fock-theory-for-molecules-atomic-guess-orbitals-with-charged-atoms">charged atomic density
guess</a>. For
self-exchange ET reactions such as this one, you can use the
<a href="#hartree-fock-theory-for-molecules-vectors-inputoutput-of-mo-vectors">REORDER</a> keyword
to move the electron from the first helium to the second.</p>
<p>Example input :</p>
<pre><code>basis &quot;ao basis&quot; 
 * library aug-cc-pvtz
end

geometry
 He 0 0 0
end

charge 1

scf
 tol2e 1d-9
 uhf
 doublet
 vectors output HeP.movecs
end
task scf

charge 0

scf
 uhf
 singlet
 vectors output He.movecs
end
task scf

geometry noautosym noautoz
  He 0.0   0.0   0.0
  He 5.0   0.0   0.0
end

charge 1
#ET reactants:
scf
  doublet; uhf; vectors input fragment HeP.movecs He.movecs output HeA.movecs
end
task scf

#ET products:
scf
  doublet; uhf; vectors input HeA.movecs reorder 2 1 output HeB.movecs
end
task scf

et
 vectors reactants HeA.movecs
 vectors products HeB.movecs
end
task scf et
</code></pre>
<p>Here is what the output looks like for this example:</p>
<pre><code>                           Electron Transfer Calculation
                           -----------------------------

 MO vectors for reactants: HeA.movecs
 MO vectors for products : HeB.movecs

 Electronic energy of reactants     H(RR)      -5.2836825646
 Electronic energy of products      H(PP)      -5.2836825646

 Reactants/Products overlap S(RP) : -4.20D-04

 Reactants/Products interaction energy:    
 -------------------------------------           
 One-electron contribution         H1(RP)       0.0027017960

 Beginning calculation of 2e contribution
 Two-electron integral screening (tol2e) : 4.20D-11

 Two-electron contribution         H2(RP)      -0.0004625156
 Total interaction energy           H(RP)       0.0022392804

 Electron Transfer Coupling Energy |V(RP)|      0.0000220152
                                                       4.832 cm-1
                                                    0.000599 eV
                                                       0.014 kcal/mol
</code></pre>
<p>The overlap between the ET reactant and product states (<i>S<sub>RP</sub></i>) is
small, so the magnitude of the coupling between the states is also
small. If the fragment guess or charged atomic density guess were not
used, the Mulliken spin population would be 0.5 on both He atoms, the
overlap between the ET reactant and product states would be 100% and an
infinite <i>V<sub>RP</sub></i> would result.</p></section><section class="print-page" id="python" heading-number="16.2"><h1 id="python-controlling-nwchem-with-python">Controlling NWChem with Python<a class="headerlink" href="#python-controlling-nwchem-with-python" title="Permanent link">&para;</a></h1>
<h2 id="python-overview">Overview<a class="headerlink" href="#python-overview" title="Permanent link">&para;</a></h2>
<p>Python programs may be embedded into the NWChem input
and used to control the execution of NWChem. Python is a very powerful
and widely used scripting language that provides useful things such as
variables, conditional branches and loops, and is also readily extended.
Example applications include scanning potential energy surfaces,
computing properties in a variety of basis sets, optimizing the energy
w.r.t. parameters in the basis set, computing polarizabilities with
finite field, and simple molecular dynamics.</p>
<p>Look in the NWChem contrib directory for useful scripts and examples.
Visit the Python web-site <a href="https://www.python.org">https://www.python.org</a> for a full manual and
lots of useful code and resources.</p>
<h2 id="python-how-to-input-and-run-a-python-program-inside-nwchem">How to input and run a Python program inside NWChem<a class="headerlink" href="#python-how-to-input-and-run-a-python-program-inside-nwchem" title="Permanent link">&para;</a></h2>
<p>A Python program is input into NWChem inside a Python compound
directive.</p>
<pre><code> python [print|noprint]
   ...
 end
</code></pre>
<p>The END directive must be flush against the left margin (see the
Troubleshooting section for the reason why).</p>
<p>The program is by default printed to standard output when read, but this
may be disabled with the noprint keyword. Python uses indentation to
indicate scope (and the initial level of indentation must be zero),
whereas NWChem uses optional indentation only to make the input more
readable. For example, in Python, the contents of a loop, or
conditionally-executed block of code must be indented further than the
surrounding code. Also, Python attaches special meaning to several
symbols also used by NWChem. For these reasons, the input inside a
PYTHON compound directive is read verbatim except that if the first line
of the Python program is indented, the same amount of indentation is
removed from all subsequent lines. This is so that a program may be
indented inside the PYTHON input block for improved readability of the
NWChem input, while satisfying the constraint that when given to Python
the first line has zero indentation.</p>
<p>E.g., the following two sets of input specify the same Python program.</p>
<pre><code>python  
  print (&quot;Hello&quot;)
  print (&quot;Goodbye&quot;)
end

python  
print (&quot;Hello&quot;)
print (&quot;Goodbye&quot;)
end
</code></pre>
<p>whereas this program is in error since the indentation of the second
line is less than that of the first.</p>
<pre><code>python  
  print (&quot;Hello&quot;)
print (&quot;Goodbye&quot;)
end
</code></pre>
<p>The Python program is not executed until the following directive is
encountered</p>
<pre><code> task python
</code></pre>
<p>which is to maintain consistency with the behavior of NWChem in general.
The program is executed by all nodes. This enables the full
functionality and speed of NWChem to be accessible from Python, but
there are some gotchas</p>
<ul>
<li>Print statements and other output will be executed by all nodes so
    you will get a lot more output than probably desired unless the
    output is restricted to just one node (by convention node zero).</li>
<li>The calls to NWChem functions are all collective (i.e., all nodes
    must execute them). If these calls are not made collectively your
    program may deadlock (i.e., cease to make progress).</li>
<li>When writing to the database (<code>rtdb_put()</code>) it is the data from node
    zero that is written.</li>
<li>NWChem overrides certain default signal handlers so care must be
    taken when creating processes (see Section describing <a href="#python-scaning-a-basis-exponent-yet-again-plotting-and-handling-child-processes">a scanning example</a>).</li>
</ul>
<h2 id="python-nwchem-extensions">NWChem extensions<a class="headerlink" href="#python-nwchem-extensions" title="Permanent link">&para;</a></h2>
<p>Since we have little experience using Python, the NWChem-Python
interface might change in a non-backwardly compatible fashion as we
discover better ways of providing useful functionality. We would
appreciate suggestions about useful things that can be added to the
NWChem-Python interface. In principle, nearly any Fortran or C routine
within NWChem can be extended to Python, but we are also interested in
ideas that will enable users to build completely new things. For
instance, how about being able to define your own energy functions that
can be used with the existing optimizers or dynamics package?</p>
<p>Python has been extended with a module named &ldquo;nwchem&rdquo; which is
automatically imported and contains the following NWChem-specific
commands. They all handle NWChem-related errors by raising the exception
&ldquo;NWChemError&rdquo;, which may be handled in the standard Python manner (see
Section describing <a href="#python-handling-exceptions-from-nwchem">handling exception</a>).</p>
<ul>
<li><code>input_parse(string)</code> &ndash; invokes the standard NWChem input parser
    with the data in string as input. Note that the usual behavior of
    NWChem will apply &ndash; the parser only reads input up to either end of
    input or until a <code>TASK</code> directive is encountered (the task directive
    is not executed by the parser).</li>
<li><code>task_energy(theory)</code> &ndash; returns the energy as if computed with the
    NWChem directive <code>TASK ENERGY &lt;THEORY&gt;</code>.</li>
<li><code>task_gradient(theory)</code> &ndash; returns a tuple (energy,gradient) as if
    computed with the NWChem directive <code>TASK GRADIENT &lt;THEORY&gt;</code>.</li>
<li><code>task_optimize(theory)</code> &ndash; returns a tuple (energy,gradient) as if
    computed with the NWChem directive <code>TASK OPTIMIZE &lt;THEORY&gt;</code>. The
    energy and gradient will be those at the last point in the
    optimization and consistent with the current geometry in the
    database.</li>
<li><code>ga_nodeid()</code> &ndash; returns the number of the parallel process.</li>
<li><code>rtdb_print(print_values)</code> &ndash; prints the contents of the RTDB. If
    <code>print_values</code> is 0, only the keys are printed, if it is 1 then the
    values are also printed.</li>
<li><code>rtdb_put(name, values)</code> or <code>rtdb_put(name, values, type)</code> &ndash; puts the
    values into the database with the given name. In the first form, the
    type is inferred from the first value, and in the second form the
    type is specified using the last argument as one of INT, DBL,
    LOGICAL, or CHAR.</li>
<li><code>rtdb_get(name)</code> &ndash; returns the data from the database associated
    with the given name.</li>
</ul>
<p>An
<a href="#python-accessing-geometry-information-a-temporary-hack">example</a>
below explains, in lieu of a Python wrapper for the geometry object, how
to obtain the Cartesian molecular coordinates directly from the
database.</p>
<h2 id="python-examples">Examples<a class="headerlink" href="#python-examples" title="Permanent link">&para;</a></h2>
<p>Several examples will provide the best explanation of how the extensions
are used, and how Python might prove useful.</p>
<h3 id="python-hello-world">Hello world<a class="headerlink" href="#python-hello-world" title="Permanent link">&para;</a></h3>
<pre><code>python
  print ('Hello world from process %i'  % ga_nodeid())
end

task python
</code></pre>
<p>This input prints the traditional greeting from each parallel
process.</p>
<h3 id="python-scanning-a-basis-exponent">Scanning a basis exponent<a class="headerlink" href="#python-scanning-a-basis-exponent" title="Permanent link">&para;</a></h3>
<pre><code>geometry units au
 O 0 0 0; H 0 1.430 -1.107; H 0 -1.430 -1.107
end

python
exponent = 0.1
while (exponent &lt;= 2.01):
    input_parse('''                                                                                                
    basis noprint                                                                                                  
     H library 3-21g; O library 3-21g; O d; %f 1.0                                                                 
    end                                                                                                            
    ''' % (exponent))
    print (' exponent = %5.4f,' % exponent, ', energy = %16.10f' % task_energy('scf'))
    exponent = exponent + 0.1
end

print none

task python
</code></pre>
<p>This program augments a 3-21g basis for water with a d-function on
oxygen and varies the exponent from 0.1 to 2.0 in steps of 0.1, printing
the exponent and energy at each step.</p>
<p>The geometry is input as usual, but the basis set input is embedded
inside a call to <code>input_parse()</code> in the Python program. The standard
Python string substitution is used to put the current value of the
exponent into the basis set (replacing the %f) before being parsed by
NWChem. The energy is returned by <code>task_energy('scf')</code> and printed out.
The print none in the NWChem input switches off all NWChem output so all
you will see is the output from your Python program.</p>
<p>Note that execution in parallel may produce unwanted output since all
process execute the print statement inside the Python program.</p>
<p>Look in the NWChem contrib directory for a routine that makes the above
task easier.</p>
<h3 id="python-scanning-a-basis-exponent-revisited">Scanning a basis exponent revisited<a class="headerlink" href="#python-scanning-a-basis-exponent-revisited" title="Permanent link">&para;</a></h3>
<pre><code>geometry units au
O 0 0 0; H 0 1.430 -1.107; H 0 -1.430 -1.107
end


python
  if (ga_nodeid() == 0): plotdata = open(&quot;plotdata&quot;,'w')

  def energy_at_exponent(exponent):
    input_parse('''                                                                                                
    basis noprint                                                                                                  
      H library 3-21g; O library 3-21g; O d; %f 1.0                                                                
    end                                                                                                            
    ''' % (exponent))
    return task_energy('scf')

  exponent = 0.1
  while exponent &lt;= 2.01:
    energy = energy_at_exponent(exponent)
    if (ga_nodeid() == 0):
        print (' exponent = %5.4f,' % exponent, ', energy = %16.10f' % energy)
        plotdata.write('%f %f\n' % (exponent , energy))
    exponent = exponent + 0.1

  if (ga_nodeid() == 0): plotdata.close()
end

print none
task python
</code></pre>
<p>This input performs exactly the same calculation as the previous one,
but uses a slightly more sophisticated Python program, also writes the
data out to a file for easy visualization with a package such as
gnuplot, and protects write statements to prevent duplicate output in a
parallel job. The only significant differences are in the Python
program. A file called &ldquo;plotdata&rdquo; is opened, and then a procedure is
defined which given an exponent returns the energy. Next comes the main
loop that scans the exponent through the desired range and prints the
results to standard output and to the file. When the loop is finished
the additional output file is closed.</p>
<h3 id="python-scanning-a-geometric-variable">Scanning a geometric variable<a class="headerlink" href="#python-scanning-a-geometric-variable" title="Permanent link">&para;</a></h3>
<pre><code>python
   geometry = '''
     geometry noprint; symmetry d2h
        C 0 0 %f; H 0  0.916 1.224
     end
   '''
   x = 0.6
   while (x &lt; 0.721):
     input_parse(geometry % x)
     energy = task_energy('scf')
     print (' x = %5.2f   energy = %10.6f' % (x, energy))
     x = x + 0.01
end

basis; C library 6-31g*; H library 6-31g*; end

print none

task python
</code></pre>
<p>This scans the bond length in ethene from 1.2 to 1.44 in steps of 0.2
computing the energy at each geometry. Since it is using D<sub>2h</sub>
symmetry the program actually uses a variable (x) that is half the bond
length.</p>
<p>Look in the NWChem contrib directory for a routine that makes the above
task
easier.</p>
<h3 id="python-scan-using-the-bsse-counterpoise-corrected-energy">Scan using the BSSE counterpoise corrected energy<a class="headerlink" href="#python-scan-using-the-bsse-counterpoise-corrected-energy" title="Permanent link">&para;</a></h3>
<pre><code>basis spherical
  Ne library cc-pvdz; BqNe library Ne cc-pvdz
  He library cc-pvdz; BqHe library He cc-pvdz
end

mp2; tight; freeze core atomic; end

print none

python
  supermolecule = 'geometry noprint;   Ne 0 0 0;   He 0 0 %f; end\n'
  fragment1     = 'geometry noprint;   Ne 0 0 0; BqHe 0 0 %f; end\n'
  fragment2     = 'geometry noprint; BqNe 0 0 0;   He 0 0 %f; end\n'

  def energy(geometry):
    input_parse(geometry + 'scf; vectors atomic; end\n')
    return task_energy('mp2')

  def bsse_energy(z):
    return energy(supermolecule % z) - \
        energy(fragment1 % z) - \
        energy(fragment2 % z)
  z = 3.3
  while (z &lt; 4.301):
    e = bsse_energy(z)
    if (ga_nodeid() == 0): print (' z = %5.2f   energy = %10.7f ' % (z, e))
    z = z + 0.1
end

task python
</code></pre>
<p>This example scans the He&ndash;Ne bond-length from 3.3 to 4.3 and prints out
the BSSE counterpoise corrected MP2 energy.</p>
<p>The basis set is specified as usual, noting that we will need functions
on ghost centers to do the counterpoise correction. The Python program
commences by defining strings containing the geometry of the
super-molecule and two fragments, each having one variable to be
substituted. Next, a function is defined to compute the energy given a
geometry, and then a function is defined to compute the counterpoise
corrected energy at a given bond length. Finally, the bond length is
scanned and the energy printed. When computing the energy, the atomic
guess has to be forced in the SCF since by default it will attempt to
use orbitals from the previous calculation which is not appropriate
here.</p>
<p>Since the counterpoise corrected energy is a linear combination of other
standard energies, it is possible to compute the analytic derivatives
term by term. Thus, combining this example and the next could yield the
foundation of a BSSE corrected geometry optimization
package.</p>
<h3 id="python-scan-the-geometry-and-compute-the-energy-and-gradient">Scan the geometry and compute the energy and gradient<a class="headerlink" href="#python-scan-the-geometry-and-compute-the-energy-and-gradient" title="Permanent link">&para;</a></h3>
<pre><code>basis noprint; H library sto-3g; O library sto-3g; end

python
 print ('   y     z     energy                gradient')
 print (' ----- ----- ---------- ------------------------------------')
 y = 1.2
 while y &lt;= 1.61:
     z = 1.0
     while z &lt;= 1.21:
         input_parse('''
           geometry noprint units atomic
             O 0   0   0
             H 0  %f -%f
             H 0 -%f -%f
           end
         ''' % (y, z, y, z))

         (energy,gradient) = task_gradient('scf')

         print (' %5.2f %5.2f %9.6f' % (y, z, energy)),
         i = 0
         while (i &lt; len(gradient)):
             print ('%5.2f' % gradient[i]),
             i = i + 1
         print ('')
         z = z + 0.1
     y = y + 0.1
end

print none

task python
</code></pre>
<p>This program illustrates evaluating the energy and gradient by calling
task_gradient(). A water molecule is scanned through several C<sub>2v</sub>
geometries by varying the y and z coordinates of the two hydrogen atoms.
At each geometry the coordinates, energy and gradient are printed.</p>
<p>The basis set (sto-3g) is input as usual. The two while loops vary the y
and z coordinates. These are then substituted into a geometry which is
parsed by NWChem using <code>input_parse()</code>. The energy and gradient are then
evaluated by calling <code>task_gradient()</code> which returns a tuple containing
the energy (a scalar) and the gradient (a vector or list). These are
printed out exploiting the Python convention that a print statement
ending in a comma does not print end-of-line.</p>
<h3 id="python-reaction-energies-varying-the-basis-set">Reaction energies varying the basis set<a class="headerlink" href="#python-reaction-energies-varying-the-basis-set" title="Permanent link">&para;</a></h3>
<pre><code>mp2; freeze atomic; end

print none

python
  energies = {}
  c2h4 = 'geometry noprint; symmetry d2h; \
          C 0 0 0.672; H 0 0.935 1.238; end\n'
  ch4  = 'geometry noprint; symmetry td; \
          C 0 0 0; H 0.634 0.634 0.634; end\n'
  h2   = 'geometry noprint; H 0 0 0.378; H 0 0 -0.378; end\n'

  def energy(basis, geometry):
    input_parse('''
      basis spherical noprint
        c library %s ; h library %s 
      end
    ''' % (basis, basis))
    input_parse(geometry)
    return task_energy('mp2')

  for basis in ('sto-3g', '6-31g', '6-31g*', 'cc-pvdz', 'cc-pvtz'):
     energies[basis] =   2*energy(basis, ch4) \
                       - 2*energy(basis, h2) - energy(basis, c2h4)
     if (ga_nodeid() == 0): print (basis, ' %8.6f' % energies[basis])
end 

task python
</code></pre>
<p>In this example the reaction energy for  2H<sub>2</sub> + C<sub>2</sub>H<sub>4</sub> &rarr;
2CH<sub>4</sub> is evaluated using MP2 in several basis sets. The geometries are
fixed, but could be re-optimized in each basis. To illustrate the useful
associative arrays in Python, the reaction energies are put into the
associative array energies &ndash; note its declaration at the top of the
program.</p>
<h3 id="python-using-the-database">Using the database<a class="headerlink" href="#python-using-the-database" title="Permanent link">&para;</a></h3>
<pre><code> python  
   rtdb_put(&quot;test_int2&quot;, 22)  
   rtdb_put(&quot;test_int&quot;, [22, 10, 3],    INT)  
   rtdb_put(&quot;test_dbl&quot;, [22.9, 12.4, 23.908],  DBL)  
   rtdb_put(&quot;test_str&quot;, &quot;hello&quot;, CHAR)  
   rtdb_put(&quot;test_logic&quot;, [0,1,0,1,0,1], LOGICAL)  
   rtdb_put(&quot;test_logic2&quot;, 0, LOGICAL)  

   rtdb_print(1)  

   print &quot;test_str    = &quot;, rtdb_get(&quot;test_str&quot;)  
   print &quot;test_int    = &quot;, rtdb_get(&quot;test_int&quot;)  
   print &quot;test_in2    = &quot;, rtdb_get(&quot;test_int2&quot;)  
   print &quot;test_dbl    = &quot;, rtdb_get(&quot;test_dbl&quot;)  
   print &quot;test_logic  = &quot;, rtdb_get(&quot;test_logic&quot;)  
   print &quot;test_logic2 = &quot;, rtdb_get(&quot;test_logic2&quot;)  
 end  

 task python
</code></pre>
<p>This example illustrates how to access the database from Python.</p>
<h3 id="python-handling-exceptions-from-nwchem">Handling exceptions from NWChem<a class="headerlink" href="#python-handling-exceptions-from-nwchem" title="Permanent link">&para;</a></h3>
<pre><code> geometry; he 0 0 0; he 0 0 2; end  
 basis; he library 3-21g; end  
 scf; maxiter 1; end  

 python  
   try:  
     task_energy('scf')  
   except NWChemError, message:  
     print 'Error from NWChem ... ', message  
 end  

 task python
</code></pre>
<p>The above test program shows how to handle exceptions generated by
NWChem by forcing an SCF calculation on He<sub>2&lt;\sub&gt; to fail due to
insufficient iterations.</p>
<p>If an NWChem command fails it will raise the exception &ldquo;NWChemError&rdquo;
(case sensitive) unless the error was fatal. If the exception is not
caught, then it will cause the entire Python program to terminate with
an error. This Python program catches the exception, prints out the
message, and then continues as if all was well since the exception has
been handled.</p>
<p>If your Python program detects an error, raise an unhandled exception.
Do not call exit(1) since this may circumvent necessary clean-up of the
NWChem execution environment.</p>
<h3 id="python-accessing-geometry-information-a-temporary-hack">Accessing geometry information: a temporary hack<a class="headerlink" href="#python-accessing-geometry-information-a-temporary-hack" title="Permanent link">&para;</a></h3>
<p>In an ideal world the geometry and basis set objects would have full
Python wrappers, but until then a back-door solution will have to
suffice. We&rsquo;ve already seen how to use <code>input_parse()</code> to put geometry
(and basis) data into NWChem, so it only remains to get the geometry
data back after it has been updated by a geometry optimzation or some
other operation.</p>
<p>The following Python procedure retrieves the coordinates in the same
units as initially input for a geometry of a given name. Its full source
is included in the NWChem contrib directory.</p>
<pre><code> def geom_get_coords(name):  
   try:  
     actualname = rtdb_get(name)  
   except NWChemError:  
     actualname = name  
   coords = rtdb_get('geometry:' + actualname + ':coords')  
   units  = rtdb_get('geometry:' + actualname + ':user units')  
   if (units == 'a.u.'):  
     factor = 1.0  
   elif (units == 'angstroms'):  
     factor = rtdb_get('geometry:'+actualname+':angstrom_to_au')  
   else:  
     raise NWChemError,'unknown units'  
   i = 0  
   while (i &lt; len(coords)):  
     coords[i] = coords[i] / factor  
     i = i + 1  
   return coords
</code></pre>
<p>A <a href="#geometry">geometry</a> with name NAME has its coordinates (in
atomic units) stored in the database entry <code>geometry:NAME:coords</code>. A minor
wrinkle here is that indirection is possible (and used by the
optimizers) so that we must first check if NAME actually points to
another name. In the program this is done in the first try&hellip;except
sequence. With the actual name of the geometry, we can get the
coordinates. Any exceptions are passed up to the caller. The rest of the
code is just to convert back into the initial input units &ndash; only atomic
units or Angstrms are handled in this simple example. Returned is a
list of the atomic coordinates in the same units as your initial input.</p>
<p>The routine is used as follows</p>
<pre><code>   coords = geom_get_coords('geometry')
</code></pre>
<p>or, if you want better error handling</p>
<pre><code>   try:  
     coords = geom_get_coords('geometry')  
   except NWChemError,message:  
     print 'Coordinates for geometry not found ', message  
   else:  
     print coords
</code></pre>
<p>This is very dirty and definitely not supported from one release to
another, but, browsing the output of <code>rtdb_print()</code> at the end of a
calculation is a good way to find stuff. To be on safer ground, look in
the programmers manual since some of the high-level routines do pass
data via the database in a well-defined and supported manner. Be warned
&ndash; you must be very careful if you try to modify data in the database.
The input parser does many important things that are not immediately
apparent (e.g., ensure the geometry is consistent with the point group,
mark the SCF as not converged if the SCF options are changed, &hellip;).
Where at all possible your Python program should generate standard
NWChem input and pass it to <code>input_parse()</code> rather than setting
parameters directly in the
database.</p>
<h3 id="python-scaning-a-basis-exponent-yet-again-plotting-and-handling-child-processes">Scaning a basis exponent yet again: plotting and handling child processes<a class="headerlink" href="#python-scaning-a-basis-exponent-yet-again-plotting-and-handling-child-processes" title="Permanent link">&para;</a></h3>
<pre><code> geometry units au  
   O 0 0 0; H 0 1.430 -1.107; H 0 -1.430 -1.107  
 end  

 print none  

 python  
   import Gnuplot, time, signal  

   def energy_at_exponent(exponent):  
      input_parse('''  
         basis noprint  
            H library 3-21g; O library 3-21g; O d; %f 1.0  
         end  
      ''' % (exponent))  
      return task_energy('scf')  

   data = []  
   exponent = 0.5  
   while exponent &lt;= 0.6:  
      energy = energy_at_exponent(exponent)  
      print ' exponent = ', exponent, ' energy = ', energy  
      data = data + [exponent,energy](exponent,energy.md)  
      exponent = exponent + 0.02  

   if (ga_nodeid() == 0):  
      signal.signal(signal.SIGCHLD, signal.SIG_DFL)  
      g = Gnuplot.Gnuplot()  
      g('set data style linespoints')  
      g.plot(data)  
      time.sleep(30)  # 30s to look at the plot  

 end  

 task python
</code></pre>
<p>This illustrates how to handle signals from terminating child processes
and how to generate simple plots on UNIX systems. The <a href="#python-scanning-a-basis-exponent-revisited">scanning
example</a> is modified so
that instead of writing the data to a file for subsequent visualization,
it is saved for subsequent visualization with Gnuplot (you&rsquo;ll need both
Gnuplot and the corresponding package for Python in your PYTHONPATH.
Look at <a href="https://web.archive.org/web/20010717060625/http://monsoon.harvard.edu/~mhagger/download/">https://web.archive.org/web/20010717060625/http://monsoon.harvard.edu/~mhagger/download/</a>).</p>
<p>The issue is that NWChem traps various signals from the O/S that usually
indicate bad news in order to provide better error handling and reliable
clean-up of shared, parallel resources. One of these signals is SIGCHLD
which is generated whenever a child process terminates. If you want to
create child processes within Python, then the NWChem handler for
SIGCHLD must be replaced with the default handler. There seems to be no
easy way to restore the NWChem handler after the child has completed,
but this should have no serious side effect.</p>
<h2 id="python-troubleshooting">Troubleshooting<a class="headerlink" href="#python-troubleshooting" title="Permanent link">&para;</a></h2>
<p>Common problems with Python programs inside NWChem.</p>
<ol>
<li>
<p>You get the message<br />
<code>0:python_input: indentation must be &gt;= that of first line: 4</code>    <br />
   This indicates that NWChem thinks that a line is less indented than
   the first line. If this is not the case then perhaps there is a tab
    in your input which NWChem treats as a single space character but
    appears to you as more spaces. Try running untabify in Emacs. The
    problem could also be the <code>END</code> directive that terminates the <code>PYTHON</code>
    compound directive, since Python also has an end statement. To
    avoid confusion the <code>END</code> directive for NWChem must be at the start of
    the line.  </p>
</li>
<li>
<p>Your program hangs or deadlocks &ndash; most likely you have a piece of
    code that is restricted to executing on a subset of the processors
    (perhaps just node 0) but is calling (perhaps indirectly) a function
    that must execute on all nodes.</p>
</li>
</ol></section><section class="print-page" id="quantum" heading-number="16.3"><h1 id="quantum-quantum">Quantum<a class="headerlink" href="#quantum-quantum" title="Permanent link">&para;</a></h1>
<p>The quantum features are routines for printing out one- and
two-electron integrals that can be utilized in quantum algorithms. The
outputs can be interpreted and formatted in
<a href="https://en.wikipedia.org/wiki/YAML">YAML</a> files that include the
integrals and basic information including initial wavefunction guesses
for quantum algorithms. There are currently two options for integrals:</p>
<h2 id="quantum-bare-hamiltonian">Bare Hamiltonian<a class="headerlink" href="#quantum-bare-hamiltonian" title="Permanent link">&para;</a></h2>
<p>To print the bare Hamiltonian, you must set up a CCSD calculation with the following parameters: <br />
<em> The symmetry must be declared as &lsquo;C1&rsquo; <br />
</em> In the TCE block, you must include the following:   </p>
<pre><code>        2eorb
        2emet 13
        nroots ### (Optional, but will include leading excitations for excited-state quantum calculations)
</code></pre>
<p>You must set the printing parameters at the end of the input file:  </p>
<pre><code>        set tce:print_integrals T
        set tce:qorb ### (Total number of orbitals you wish to be printed)
        set tce:qela ### (Number of alpha electrons)
        set tce:qelb ### (number of beta electrons)
</code></pre>
<p><em>Notes:</em>  </p>
<ul>
<li>The two-electron integrals are printed out in compact Mulliken
notation (each printed element corresponds to (ij|kl), (ij|lk),
(ji|kl), (ji|lk), (kl|ij), (kl|ji), (lk|ij), and (lk|ji) )  </li>
<li>The number
of alpha and beta electrons that are set in the printing does not need
to correspond to the total number of alpha and beta electrons. Rather
it is the number of electrons included in the printing, starting from
the HOMO and counting down. This allows you to &lsquo;freeze&rsquo; core
orbitals. However, this is not recommended, because unlike freezing
Fock matrix elements and two-electron integrals in correlated
calculations, this procedure is dropping core orbitals of the standard
one- and two-electron integrals and can lead to an imbalance of the
correlation energy.  </li>
<li>If the number of electrons and orbitals do not
correspond to the total numbers of electrons and orbitals, then the
calculation will perform a corresponding frozen core/frozen virtual
CC/EOMCC calculation. This is to ensure that leading CC/EOMCC
excitations do not correspond to orbitals outside of the printed
integrals.  </li>
</ul>
<h2 id="quantum-ducc-hamiltonian">DUCC Hamiltonian<a class="headerlink" href="#quantum-ducc-hamiltonian" title="Permanent link">&para;</a></h2>
<p>The double unitary coupled-cluster (DUCC)
Hamiltonian is a way of incorporating correlation effects (mainly
dynamical) into a reduced dimensionality Hamiltonian based on a
defined active space. The procedure currently only reduces the virtual
space and all occupied orbitals are considered active. Only the
ground-state implementation is available in the current <a href="https://github.com/nwchemgit/nwchem/releases/tag/v7.0.0-release">NWChem
release</a>.  </p>
<p>To print the DUCC Hamiltonian, you must set up a CCSD calculation with the following parameters:  </p>
<ul>
<li>The symmetry must be declared as &lsquo;C1&rsquo;  </li>
<li>In the TCE block, you must include the following:  </li>
</ul>
<pre><code>        2eorb
        2emet 13
</code></pre>
<p>You must set the printing parameters at the end of the input file:  </p>
<pre><code>        set tce:qducc T
        set tce:nactv ### (The number of active virtual orbitals. 
                           Remember that all occupied orbitals are active as well)
        set tce:nonhf F/T (If a non-RHF reference is used, set to T. Otherwise, keep as F.)
        set tce:ducc_model ### (Determines how the similarity transformed Hamiltonian is truncated. See note below.)
</code></pre>
<p><em>Notes:</em></p>
<ul>
<li>Six models for the truncation of the similarity transformed are implemented and numbered 1-6. They correspond to approximations A(2)-A(7), respectively, that are detailed in <a href="https://arxiv.org/abs/2110.12077">arXiv:2110.12077</a>. Model 3 corresponds to the earliest implementation described in <a href="https://doi.org/10.1063/1.5094643">J. Chem. Phys. 151, 014107 (2019)</a> and <a href="https://doi.org/10.1063/1.5128103">J. Chem. Phys. 151, 234114 (2019)</a>. In contrast to these early publications, the full form of the two-electron integral is printed. Model 2 is an aggressive truncation and is not recommended for use. </li>
<li>The entire two-electron integral is printed out since the DUCC Hamiltonian does not have the same symmetry as the bare Hamiltonian. Instead of the 8-fold symmetry, it has dual 4-fold symmetries    </li>
</ul>
<pre><code>        (IJ|KL) = (JI|LK) = (KL|IJ) = (LK|JI)  
        Separately, (IJ|LK) = (JI|KL) = (KL|JI) = (LK|IJ)  
        But (IJ|KL) /= (IJ|LK)    
</code></pre>
<ul>
<li>When utilizing the <a href="#quantum-yaml-interpreter">YAML interpreter</a>, the full two-electron integral will be printed in the YAML file, despite it being labeled as &lsquo;sparse&rsquo; in the YAML file.   </li>
<li>Frozen core calculations should not be performed. There is nothing preventing such calculation from being executed, but the integrals will be incorrect.   </li>
<li>Since the all-orbital coupled-cluster calculation must be performed, then there is a chance that a leading excitation outside of the active-space will be printed and therefore included in the initial wave function description in the YAML file. The user ought to be aware of this risk and make changes to the YAML file to exclude these excitations, which would include renormalizing the remaining amplitudes of the initial guess.  </li>
</ul>
<h2 id="quantum-yaml-interpreter">YAML Interpreter<a class="headerlink" href="#quantum-yaml-interpreter" title="Permanent link">&para;</a></h2>
<p>Once the calculation is completed, the integrals and basic information including initial wavefunction guesses for quantum algorithms can be extracted into a YAML file. To generate the YAML file, executed the following command:</p>
<pre><code>python $NWCHEM_TOP/contrib/quasar/export_chem_library_yaml.py &lt; {NWChem output file} &gt; {YAML file name}
</code></pre>
<p><em>Note:</em>
* the &lsquo;fci_energy&rsquo; in the YAML file is taken to be the energy of the correlated method, which, in either case, is the CCSD energy.</p>
<h2 id="quantum-example-input-file-for-bare-hamiltonian">Example input file for Bare Hamiltonian<a class="headerlink" href="#quantum-example-input-file-for-bare-hamiltonian" title="Permanent link">&para;</a></h2>
<pre><code>echo

start Lih

geometry units angstroms
symmetry C1
 Li 0 0   0.000
 H  0 0   1.600
end

basis spherical
 * library sto-3g
end

scf
  singlet
  rhf
  thresh 1e-10
end

tce
  2eorb
  2emet 13
  ccsd
  thresh 1.0d-8
  maxiter 150
  nroots 2
end

set tce:print_integrals T
set tce:qorb 6
set tce:qela 2
set tce:qelb 2

task tce energy
</code></pre>
<h2 id="quantum-example-input-file-for-ducc">Example input file for DUCC<a class="headerlink" href="#quantum-example-input-file-for-ducc" title="Permanent link">&para;</a></h2>
<pre><code>echo

start Lih

geometry units angstroms
symmetry C1
 Li 0 0   0.000
 H  0 0   1.600
end

basis spherical
 * library sto-3g
end

scf
  singlet
  rhf
  thresh 1e-10
end

tce
  2eorb
  2emet 13
  ccsd
  thresh 1.0d-8
  maxiter 150
end

set tce:qducc T
set tce:nactv 4
set tce:nonhf F
set tce:ducc_model 3

task tce energy
</code></pre></section><section class="print-page" id="vscf" heading-number="16.4"><h1 id="vscf-vibrational-scf-vscf">Vibrational SCF (VSCF)<a class="headerlink" href="#vscf-vibrational-scf-vscf" title="Permanent link">&para;</a></h1>
<h2 id="vscf-overview">Overview<a class="headerlink" href="#vscf-overview" title="Permanent link">&para;</a></h2>
<p>The VSCF module<sup id="fnref_footnote_1"><a class="footnote-ref" href="#vscf-fn_footnote_1">1</a></sup> can be used to calculate the anharmonic contributions to
the vibrational modes of the molecule of interest. Energies are
calculated on a one-dimensional grid along each normal mode, on a
two-dimensional grid along each pair of normal modes, and optionally on
a three-dimensional grid along each triplet of normal modes. These
energies are then used to calculate the vibrational nuclear wavefunction
at an SCF- (VSCF) and MP2-like (cc-VSCF) level of theory.</p>
<p>VSCF can be used at all levels of theory, SCF and correlated methods,
and DFT. For correlated methods, only the SCF level dipole is evaluated
and used to calculate the IR intensity values.</p>
<p>The VSCF module is started when the task directive <code>TASK &lt;theory&gt; vscf</code> is
defined in the user input file. The input format has the form:</p>
<pre><code>VSCF
[coupling&lt;stringcoupleleveldefault&quot;pair&quot;&gt;]  
[ngrid&lt;integerdefault16&gt;]  
[iexcite&lt;integerdefault1&gt;]  
[vcfct&lt;realdefault1.0&gt;]  
END
</code></pre>
<h2 id="vscf-coupling-keyword">Coupling keyword<a class="headerlink" href="#vscf-coupling-keyword" title="Permanent link">&para;</a></h2>
<p>The order of coupling of the harmonic normal modes included in the
calculation is controlled by the specifying:</p>
<pre><code>coupling&lt;stringcoupleleveldefault&quot;pair&quot;&gt;
</code></pre>
<p>For <code>coupling=diagonal</code> a one-dimensional grid along each normal mode is
computed. <br />
For <code>coupling=pair</code> a two-dimensional grid along each pair of normal modes is computed. <br />
For <code>coupling=triplet</code> a three-dimensional grid along each triplet of normal modes is computed. </p>
<h2 id="vscf-ngrid-keyword">Ngrid keyword<a class="headerlink" href="#vscf-ngrid-keyword" title="Permanent link">&para;</a></h2>
<p>The number of grid points along each normal mode, or pair of modes can
be defined by specifying:</p>
<pre><code>ngrid&lt;integerdefault16&gt;
</code></pre>
<h2 id="vscf-iexcite-keyword">Iexcite keyword<a class="headerlink" href="#vscf-iexcite-keyword" title="Permanent link">&para;</a></h2>
<p>This VSCF module by default calculates the ground state (&nu;=0), but can
also calculate excited states (such as &nu;=1). The number of excited
states calculated is defined by specifying:</p>
<pre><code>iexcite&lt;integerdefault1&gt;
</code></pre>
<p>With <code>iexcite=1</code> the fundamental frequencies are calculated.<br />
With <code>iexcite=2</code> the first overtones are calculated.<br />
With <code>iexcite=3</code> the second overtones are calculated.</p>
<h2 id="vscf-vcfct-keyword">Vcfct keyword<a class="headerlink" href="#vscf-vcfct-keyword" title="Permanent link">&para;</a></h2>
<p>In certain cases the pair coupling potentials can become larger than
those for a single normal mode. In this case the pair potentials need to
be scaled down. The scaling factor used can be defined by specifying:</p>
<pre><code>vcfct&lt;realdefault1.0&gt;
</code></pre>
<h2 id="vscf-references">References<a class="headerlink" href="#vscf-references" title="Permanent link">&para;</a></h2>
<div class="footnote">
<hr />
<ol>
<li id="vscf-fn_footnote_1">
<p>Chaban, G. M.; Jung, J. O.; Gerber, R. B. Ab Initio Calculation of Anharmonic Vibrational States of Polyatomic Systems: Electronic Structure Combined with Vibrational Self-Consistent Field. <em>The Journal of Chemical Physics</em> <strong>1999</strong>, <em>111</em> (5), 18231829. <a href="https://doi.org/10.1063/1.479452">https://doi.org/10.1063/1.479452</a>.&#160;<a class="footnote-backref" href="#vscf-fnref_footnote_1" title="Jump back to footnote 1 in the text">&#8617;</a></p>
</li>
</ol>
</div></section><section class="print-page" id="ccca_method" heading-number="16.5"><h1 id="ccca_method-correlation-consistent-composite-approach-ccca">Correlation consistent Composite Approach (ccCA)<a class="headerlink" href="#ccca_method-correlation-consistent-composite-approach-ccca" title="Permanent link">&para;</a></h1>
<p>The CCCA module calculates the total energy using the correlation
consistent Composite Approach (ccCA). At present the ccCA module is
designed for the study of main group species
only.</p>
<div class="arithmatex">\[E_{ccCA} = \Delta E_{MP2/CBS} \  \   + \Delta E_{CC} + \Delta E_{CV} + \Delta E_{SR} + \Delta E_{ZPE}\]</div>
<p>where EMP2/CBS is the complete basis set extrapolation of MP2 energies
with the aug-cc-pVnZ (n=T,D,Q) series of basis sets, <span class="arithmatex">\(\Delta E_{CC}\)</span> is the correlation correction,</p>
<div class="arithmatex">\[\Delta E_{CC} = E_{CCSD(T)/cc-pVTZ} \  \ \  -  E_{MP2/cc-pVTZ}\]</div>
<p><span class="arithmatex">\(\Delta E_{CV}\)</span> is the core-valence correction,</p>
<div class="arithmatex">\[\Delta E_{CV} = E_{MP2(FC1)/aug-cc-pCVTZ} \  \  \  \  \   -  E_{MP2/aug-cc-pVTZ}\]</div>
<p>where <span class="arithmatex">\(FC1\)</span> symbolizes that all electrons of first-row atoms are correlated, all electrons of second-row atoms are correlated except the 1s MOs, and all electrons of atoms KKr are correlated except the 1s2s2p MOs.<br />
<span class="arithmatex">\(\Delta E_{SR}\)</span> is the scalar-relativistic correction,</p>
<div class="arithmatex">\[\Delta E_{SR} = E_{MP2/cc-pVTZ-DK} \  \  \  -  E_{MP2/cc-pVTZ}\]</div>
<p>and <span class="arithmatex">\(\Delta E_{ZPE}\)</span> is the zero-point energy correction or thermal
correction. Geometry optimization and subsequent frequency analysis are
performed with B3LYP/cc-pVTZ.</p>
<p>Suggested reference: N.J. DeYonker, B. R. Wilson, A.W. Pierpont, T.R.
Cundari, A.K. Wilson, Mol. Phys. 107, 1107 (2009). Earlier variants for
ccCA algorithms can also be found in: N. J. DeYonker, T. R. Cundari, A.
K. Wilson, J. Chem. Phys. 124, 114104 (2006).</p>
<p>The ccCA module can be used to calculate the total single point energy
for a fixed geometry and the zero-point energy correction is not
available in this calculation. Alternatively the geometry optimization
by B3LYP/cc-pVTZ is performed before the single point energy evaluation.
For open shell molecules, the number of unpaired electrons must be given
explicitly.</p>
<pre><code>CCCA

[(ENERGY||OPTIMIZE)defaultENERGY] 
[(DFT||DIRECT)defaultDFT]  
[(MP2||MBPT2)defaultMP2]  
[(RHF||ROHF||UHF)defaultRHF]  
[(CCSD(T)||TCE)defaultCCSD(T)]  
[NOPEN&lt;integernumberofunpairedelectronsdefault0&gt;]  
[(THERM||NOTHERM)defaultTHERM]  
[(PRINT||NOPRINT)defaultNOPRINT]  
[BASIS&lt;basis name for orbital projection guess&gt;]  
[MOVEC&lt;file name for orbital projection guess&gt;]  
END
</code></pre>
<p>One example of input file for single point energy evaluation is given
here:</p>
<pre><code>start h2o_ccca

title &quot;H2O, ccCA test&quot;

geometry units au

H0.00000000001.4140780900-1.1031626600  
H0.0000000000-1.4140780900-1.1031626600  
 O0.00000000000.0000000000-0.0080100000
end

task ccca
</code></pre>
<p>An input file for the ground state of O2 with geometry optimization is
given below:</p>
<pre><code>start o2_ccca

title &quot;O2, ccCA test&quot;

geometry units au

O0.00000000000.0000000000-2.0000  
O0.00000000000.00000000000.0000

end

ccca
optimize
dft  
nopen2
end

task ccca
</code></pre></section><section class="print-page" id="interface" heading-number="16.6"><h1 id="interface-interfaces-to-other-programs">Interfaces to Other Programs<a class="headerlink" href="#interface-interfaces-to-other-programs" title="Permanent link">&para;</a></h1>
<p>NWChem has interfaces to several different packages which are listed
below. In general, the NWChem authors work with the authors of the other
packages to make sure that the interface works. However, any problems
with the interface should be reported through the github issue page https://github.com/nwchemgit/nwchem/issues</p>
<h2 id="interface-dirdyvtst-direct-dynamics-for-variational-transition-state-theory">DIRDYVTST &ndash; DIRect Dynamics for Variational Transition State Theory<a class="headerlink" href="#interface-dirdyvtst-direct-dynamics-for-variational-transition-state-theory" title="Permanent link">&para;</a></h2>
<p>by Bruce C. Garrett, Environmental Molecular Sciences Laboratory,
Pacific Northwest Laboratory, Richland, Washington</p>
<p>Yao-Yuan Chuang and Donald G. Truhlar, Department of Chemistry and Super
Computer Institute, University of Minnesota, MN 55455-0431</p>
<p>and interfaced to NWChem by</p>
<p>Ricky A. Kendall, Scalable Computing Laboratory, Ames Laboratory and
Iowa State University, Ames, IA 50011</p>
<p>Theresa L. Windus, Environmental Molecular Sciences Laboratory, Pacific
Northwest Laboratory, Richland, Washington</p>
<p>If you use the DIRDYVTST portion of NWChem, please use following
citation in addition to the usual NWChem
citation:</p>
<pre><code>   DIRDYVTST, Yao-Yuan Chuang and Donald G. Truhlar, Department of Chemistry and Super Computer Institute,   
   University of Minnesota; Ricky A. Kendall,Scalable Computing Laboratory, Ames Laboratory and Iowa State   
   University; Bruce C. Garrett and Theresa L. Windus, Environmental Molecular Sciences Laboratory,  
   Pacific Northwest Laboratory. `
</code></pre>
<h3 id="interface-introduction">Introduction<a class="headerlink" href="#interface-introduction" title="Permanent link">&para;</a></h3>
<p>By using DIRDYVTST, a user can carry out electronic structure
calculations with NWChem and use the resulting energies, gradients, and
Hessians for direct dynamics calculations with <a href="https://comp.chem.umn.edu/polyrate/">POLYRATE</a>. 
This program
prepares the file30 input for POLYRATE from NWChem electronic structure
calculations of energies, gradients and Hessians at the reactant,
product, and saddle point geometries and along the minimum energy path.
Cartesian geometries for the reactants, products, and saddle points need
to be input to this program; optimization of geometries is not performed
in this program. Note that DIRDYVTST is based on the DIRDYGAUSS program
and is similar to two other programs: DDUTILITIES and GAUSSRATE. Users
of this module are encouraged to read the <a href="https://comp.chem.umn.edu/polyrate/190727_Polyrate_17-C_manuall.pdf">POLYRATE manual</a> since they
will need to create the file fu5 input to run calculations with
POLYRATE.</p>
<p>Notes about the code:</p>
<p>Input. The code has been written to parallel, as much as possible, the
POLYRATE code.</p>
<p>Output. There is one default output file for each DIRDYVTST run -
.file30.</p>
<p>Integrators for following the reaction path. Currently the Euler and
three Page-McIver (PM) methods are implemented. The PM methods are the
local quadratic approximation (LQA), the corrected LQA (CLQA), and the
cubic (CUBE) algorithm. The PM methods are implemented so that the
Hessian can be reused at intermediate steps at which only the gradient
is updated.</p>
<h3 id="interface-files">Files<a class="headerlink" href="#interface-files" title="Permanent link">&para;</a></h3>
<p>Test runs are located in directories in $NWCHEM_TOP/QA/tests. Test runs
are available for two systems: H + H<sub>2</sub> and OH + H<sub>2</sub>.</p>
<p>The H + H<sub>2</sub> test uses the Euler integration method at the SCF/3-21G
level of theory to calculate points along the reaction path. This test
is located in the $NWCHEM_TOP/QA/tests/h3tr1 directory.</p>
<p>The OH + H<sub>2</sub> test uses the Page-McIver CUBE algorithm to calculate
points on the SCF/3-21G surface and does additional single point
calculations at the SCF/6-31G* level of theory. This test is located in
the $NWCHEM_TOP/QA/tests/oh3tr3 directory.</p>
<p>Note: These tests are set up with SCF, however, other levels of theory
can be used. The initial hessian calculations at the reactants, products
and saddle point can cause some problems when numerical hessians are
required (especially when there is symmetry breaking in the
wavefunction).</p>
<h3 id="interface-detailed-description-of-the-input">Detailed description of the input<a class="headerlink" href="#interface-detailed-description-of-the-input" title="Permanent link">&para;</a></h3>
<p>The input consists of keywords for NWChem and keywords related to
POLYRATE input. The first set of inputs are for NWChem with the general
input block of the
form:</p>
<pre><code> DIRDYVTST [autosym [real tol default 1d-2] | noautosym]  
   [THEORY &lt;string theory&gt; [basis &lt;string basis default &quot;ao basis&quot;&gt;] \  
                        [ecp &lt;string ecp&gt;] [input &lt;string input&gt;]]  
   [SPTHEORY &lt;string theory&gt; [basis &lt;string basis default &quot;ao basis&quot;&gt;]   
                        [ecp &lt;string ecp&gt;] [input &lt;string input&gt;`]]  
   ...
 END
</code></pre>
<h4 id="interface-use-of-symmetry">Use of symmetry<a class="headerlink" href="#interface-use-of-symmetry" title="Permanent link">&para;</a></h4>
<p>The use of symmetry in the calculation is controlled by the keyword
<code>autosym | noautosym</code> which is used as described in the
<a href="#geometry">geometry</a> directive. <code>Autosym</code> is on by
default. A couple words of warning here. The tolerance related to
<code>autosym</code> can cause problems when taking the initial step off of the
transition state. If the tolerance is too large and the initial step
relatively small, the resulting geometry will be close to a higher
symmetry than is really wanted and the molecule will be symmetrized into
the higher symmetry. To check this, the code prints out the symmetry at
each geometry along the path. It is up to the user to check the symmetry
and make sure that it is the required one. In perverse cases, the user
may need to turn autosym off (noautosym) if changing the tolerance
doesn&rsquo;t produce the desired results. In the case that autosym is used,
the user does not need to worry about the different alignment of the
molecule between NWChem and POLYRATE, this is taken care of internally
in the DIRDYVTST module.</p>
<h4 id="interface-basis-specification">Basis specification<a class="headerlink" href="#interface-basis-specification" title="Permanent link">&para;</a></h4>
<p>The basis name on the theory or sptheory directive is that specified on
a <a href="#basis">basis set directive</a> and not the name of
a standard basis in the library. If not specified, the basis set for the
sptheory defaults to the theory basis which defaults to &ldquo;ao basis&rdquo;.</p>
<h4 id="interface-effective-core-potentials">Effective core potentials<a class="headerlink" href="#interface-effective-core-potentials" title="Permanent link">&para;</a></h4>
<p>If an <a href="#ecp">effective core potential</a> is specified
in the usual fashion outside of the DIRDYVTST input then this will be
used in all calculations. If an alternative ECP name (the name specified
on the ECP directive in the same manner as done for basis sets) is
specified on one of the theory directives, then this ECP will be used in
preference for that level of theory.</p>
<h4 id="interface-general-input-strings">General input strings<a class="headerlink" href="#interface-general-input-strings" title="Permanent link">&para;</a></h4>
<p>For many purposes, the ability to specify the theory, basis and
effective core potential is adequate. All of the options for each theory
are determined from their independent input blocks. However, if the same
theory (e.g., DFT) is to be used with different options for theory and
sptheory, then the general input strings must be used. These strings are
processed as NWChem input each time the theoretical calculation is
invoked. The strings may contain any NWChem input, except for options
pertaining to DIRDYVTST and the task directive. The intent is that the
strings be used just to control the options pertaining to the theory
being used.</p>
<p>A word of caution. Be sure to check that the options are producing the
desired results. Since the NWChem database is persistent, the input
strings should fully define the calculation you wish to have happen.</p>
<p>For instance, if the theory model is DFT/LDA/3-21g and the sptheory
model is DFT/B3LYP/6-311g**, the DIRDYVTST input might look like this</p>
<pre><code>   dirdyvtst  
     theory  dft basis 3-21g    input &quot;dft\; xc\; end&quot;  
     sptheory dft basis 6-311g** input &quot;dft\; xc b3lyp\; end&quot;  
     ....  
   end
</code></pre>
<p>The empty XC directive restores the default <a href="#density-functional-theory-for-molecules-xc-and-decomp-exchange-correlation-potentials">LDA exchange-correlation
functional</a>.
Note that semi-colons and other quotation marks inside the input string
must be preceded by a backslash to avoid special interpretation.</p>
<h4 id="interface-polyrate-related-options">POLYRATE related options<a class="headerlink" href="#interface-polyrate-related-options" title="Permanent link">&para;</a></h4>
<p>These keyword options are simlar to the POLYRATE input format, except
there are no ENERGETICS, OPTIMIZATION, SECOND, TUNNELING, and RATE
sections.</p>
<h5 id="interface-general-section">GENERAL section<a class="headerlink" href="#interface-general-section" title="Permanent link">&para;</a></h5>
<p>The GENERAL section has the following format:</p>
<pre><code>  * GENERAL

 [TITLE &lt;string title&gt;]  
 ATOMS  
   &lt;integer num&gt; &lt;string tag&gt; [&lt;real mass&gt;]  
   ...  
 END  
 [SINGLEPOINT]  
 [SAVEFILE (vecs || hess || spc)
</code></pre>
<p>Descriptions</p>
<p>TITLE is a keyword that allows the user to input a description of the
calculation. In this version, the user can only have a single-line
description.</p>
<p>For example:</p>
<p><code>TITLE Calculation of D + HCl reaction</code></p>
<p>ATOMS is a list keyword that is used to input a list of the atoms. It is
similar to POLYRATE in that the order of the atom and the atomic symbol
are required in a single line. If isotope of the element is considered
then the atomic mass is required in units of amu.</p>
<p>For example:</p>
<pre><code>                         ATOMS  
                           1     H     2.014  
                           2     H  
                           3     Cl  
                         END`
</code></pre>
<p>SINGLEPOINT is a keyword that specifies that a single point calculation
is to be performed at the reactants, products and saddle point
geometries. The type of single point calculation is specified in the
sptheory line.</p>
<p>SAVEFILE is a keyword that specifies that NWChem files are to be saved.
Allowed values of variable input to SAVEFILE are vecs, hess, and spc for
saving the files base theory movecs, base theory hessian and singlepoint
calculation movecs.</p>
<h5 id="interface-react1-react2-prod1-prod2-and-start-sections">REACT1, REACT2, PROD1, PROD2, and START sections<a class="headerlink" href="#interface-react1-react2-prod1-prod2-and-start-sections" title="Permanent link">&para;</a></h5>
<p>These sections have the following
format:</p>
<pre><code>*(REACT1 || REACT2 || PROD1 || PROD2 || START)  
  GEOM  
    &lt;integer num&gt; &lt;real x y z&gt;  
    ...  
  END  
  SPECIES (ATOMIC || LINRP || NONLINRP || LINTS || NONLINTS default NONLINRP)
</code></pre>
<p>REACT1 and REACT2 are input for each of the reactants and PROD1 and
PROD2 are input for each of the products. REACT1 and PROD1 are required.
START is the input for the transition state if one exists, or starting
point to follow downhill the MEP.</p>
<p>Descriptions</p>
<p>GEOM is a list keyword that indicates the geometry of the molecule in
Cartesian coordinates with atomic unit.</p>
<p>For example:</p>
<pre><code>                         GEOM  
                            1      0.0     0.0     0.0  
                            2      0.0     0.0     1.5  
                         END
</code></pre>
<p>SPECIES is a variable keyword that indicates the type of the molecule.
Options are: ATOMIC (atomic reactant or product), LINRP (linear reactant
or product), NONLINRP (nonlinear reactant or product), LINTS (linear
transition state), and NONLINTS (nonlinear transition state).</p>
<p>For example:</p>
<p><code>SPECIES atomic</code></p>
<h5 id="interface-path-section">PATH section<a class="headerlink" href="#interface-path-section" title="Permanent link">&para;</a></h5>
<p>The Path section has the format:</p>
<pre><code>*PATH  
 [SCALEMASS &lt;real scalemass default 1.0&gt;]  
 [SSTEP &lt;real sstep default 0.01&gt;]  
 [SSAVE &lt;real ssave default 0.1&gt;]  
 [SHESS &lt;real shess default SSAVE&gt;]  
 [SLP &lt;real slp default 1.0&gt;]  
 [SLM &lt;real slm default -1.0&gt;]  
 [SIGN (REACTANT || PRODUCT default REACTANT)]  
 [INTEGRA (EULER || LQA || CLQA || CUBE default EULER)] 
 [PRINTFREQ (on || off default off)]
</code></pre>
<p>Descriptions</p>
<p>SCALEMASS is a variable keyword that indicates the arbitrary mass (in
amu) used for mass-scaled Cartesian coordinates. This is the variable
called mu in published papers. Normally, this is taken as either 1.0 amu
or, for bimolecular reactions, as the reduced mass of relative
translation of the reactants.</p>
<p>SSTEP is a variable keyword that indicates the numerical step size (in
bohrs) for the gradient grid. This is the step size for following the
minimum energy path.</p>
<p>SSAVE is a variable keyword that indicates the numerical step size (in
bohrs) for saving the Hessian grid. At each save point the potential and
its first and second derivatives are recalculated and written to the
.file30 file. For example, if SSTEP=0.01 and SSAVE=0.1, then the
potential information is written to .file30 every 10 steps along the
gradient grid.</p>
<p>SHESS is a variable keyword that indicates the numerical step size (in
bohrs) for recomputing the Hessian when using a Page-McIver integrator
(e.g., LQA, CLQA, or CUBE). For Euler integration SHESS = SSAVE. For
intermediate points along the gradient grid, the Hessian matrix from the
last Hessian calculation is reused. For example, if SSTEP=0.01 and
SHESS=0.05, then the Hessian matrix is recomputed every 5 steps along
the gradient grid.</p>
<p>SLP is a variable keyword that indicates the positive limit of the
reaction coordinate (in bohrs).</p>
<p>SLM is a variable keyword that indicates the negative limit of the
reaction coordinate (in bohrs).</p>
<p>SIGN is a variable keyword used to ensure the conventional definition of
the sign of s, s \&lt; 0 for the reactant side and s &gt; 0 for the product
side, is followed. PRODUCT should be used if the eigenvector at the
saddle point points toward the product side and REACTANT if the
eigenvector points toward the reactant side.</p>
<p>INTEGRA is a variable keyword that indicates the integration method used
to follow the reaction path. Options are: EULER, LQA, CLQA, and CUBE.</p>
<p>PRINTFREQ is a variable keyword that indicates that projected
frequencies and eigenvectors will be printed along the MEP.</p>
<h4 id="interface-restart">Restart<a class="headerlink" href="#interface-restart" title="Permanent link">&para;</a></h4>
<p>DIRDYVTST calculations should be
<a href="#start_restart">restarted</a>
through the normal NWChem mechanism. The user needs to change the start
directive to a restart directive and get rid of any information that
will overwrite important information in the RTDB. The file.db and
file.file30 need to be available for the calculation to restart
properly.</p>
<h4 id="interface-example">Example<a class="headerlink" href="#interface-example" title="Permanent link">&para;</a></h4>
<p>This is an example that creates the file30 file for POLYRATE for
H + H<sub>2</sub>. Note that the multiplicity is that of the entire
supermolecule, a doublet. In this example, the initial energies,
gradients, and Hessians are calculated at the UHF/3-21G level of theory
and the singlepoint calculations are calculated at the MP2/cc-pVDZ level
of theory with a tighter convergence threshold than the first SCF.</p>
<pre><code>start h3test  

basis  
 h library 3-21G  
end  

basis singlepoint  
 h library cc-pVDZ  
end  

scf  
  uhf  
  doublet  
  thresh 1.0e-6  
end  

dirdyvtst autosym 0.001  
  theory scf input &quot;scf\; uhf\; doublet\; thresh 1.0e-06\; end&quot;  
  sptheory mp2 basis singlepoint input \  
    &quot;scf\; uhf\; doublet\; thresh 1.0e-07\; end&quot;  
*GENERAL  
  TITLE  
    Test run: H+H2 reaction, Page-McIver CLQA algorithm, no restart  

  ATOMS 
     1    H 
     2    H  
     3    H  
  END 

  SINGLEPOINT  

*REACT1  
   GEOM  
     1  0.0   0.0   0.0  
     2  0.0   0.0   1.3886144 
   END  

   SPECIES LINRP  

*REACT2  
  GEOM  
    3    0.0   0.0 190.3612132  
  END  

  SPECIES  ATOMIC  

*PROD2  
  GEOM  
    1   0.0   0.0 190.3612132  
  END  

  SPECIES   ATOMIC  

*PROD1  

  GEOM  
    2  0.0   0.0   1.3886144 
    3   0.0   0.0   0.0  
  END  

  SPECIES  LINRP  

*START  

  GEOM  
    1  0.0   0.0  -1.76531973  
    2  0.0   0.0   0.0  
    3  0.0   0.0   1.76531973  
  END  

  SPECIES  LINTS  

*PATH  
  SSTEP  0.05  
  SSAVE  0.05  
  SLP    0.50  
  SLM   -0.50  
  SCALEMASS 0.6718993  
  INTEGRA   CLQA  
end  

task dirdyvtst
</code></pre></section></section>
                    <section class='print-page md-section' id='section-17' heading-number='17'>
                        <h1>Molecular Mechanics<a class='headerlink' href='#section-17' title='Permanent link'></a>
                        </h1>
                    <section class="print-page" id="prepare" heading-number="17.1"><h1 id="prepare-prepare">Prepare<a class="headerlink" href="#prepare-prepare" title="Permanent link">&para;</a></h1>
<p>The prepare module is used to set up the necessary files for a molecular dynamics simulation with NWChem. User supplied coordinates can be used to generate topology and restart files. The topology file contains all static information about a molecular system, such as lists of atoms, bonded interactions and force field parameters. The restart file contains all dynamic information about a molecular system, such as coordinates, velocities and properties.</p>
<p>Without any input, the prepare module checks the existence of a topology and restart file for the molecular systems. If these files exist, the module returns to the main task level without action. The module will generate these files when they do not exist. Without any input to the module, the generated system will be for a non-solvated isolated solute system.</p>
<p>To update existing files, including solvation, the module requires input directives read from an input deck,</p>
<pre><code>prepare
  ...
end
</code></pre>
<p>The prepare module performs three sub-tasks:</p>
<ul>
<li>sequence generation :
    This sub-task analyzes the supplied coordinates from a PDB-formatted file or from the input geometry, and generates a sequence file, containing the description of the system in terms of basic building blocks found as fragment or segment files in the database directories for the force field used. If these files do not exist, they are generated based on the supplied coordinates. This process constists of generating a fragment file with the list of atoms with their force field dependent atom types, partial atomic charges calculated from a Hartree Fock calculation for the fragment, followed by a restrained electrostatic potential fit, and a connectivity list. From the information on this fragment file the lists of all bonded interactions are generated, and the complete lists are written to a segment file.</li>
<li>topology generation :
    Based on the generated or user-supplied e file and the force field specific segment database files, this sub-task compiles the lists of atoms, bonded interactions, excluded pairs, and substitutes the force field parameters. Special commands may be given to specify interaction parameters that will be changing in a free energy evaluation.</li>
<li>restart generation :
    Using the user supplied coordinates and the topology file for the chemical system, this sub-task generates a restart file for the system with coordinates, velocities and other dynamic information. This step may include solvation of the chemical system and specifying periodic boundary conditions.</li>
</ul>
<p>Files involved in the preparation phase exist in the following hierarchy:</p>
<ul>
<li>standards :
    The standard database files contain the original force field information. These files are to reside in a directory that is specified in the file <code>$HOME/.nwchemrc</code>. There will be such a directory for each supported force field. These directories contain fragment files (with extension frg), segment files (with extension sgm) and a parameter file (with the name of the force field and with extension par).</li>
<li>extensions : These database files contain generally accepted extensions to the original force field and are to reside in a separate directory that is specified in the file <code>$HOME/.nwchemrc</code>. There will be such a directory for each supported force field. These directories contain fragment files (with extension frg), segment files (with extension sgm) and a parameter file (with the name of the force field and with extension par).</li>
<li>contributed :
    These database files contain contributed definitions, also required for the quality assurance tests and are to reside in a separate directory that is specified in the file <code>$HOME/.nwchemrc</code>. There will be such a directory for each supported force field. These directories contain fragment files (with extension frg), segment files (with extension sgm) and a parameter file (with the name of the force field and with extension par).</li>
<li>user preferences :
    These database files contain user preferred extensions to the original force field and are to reside in a separate directory that is specified in the file <code>$HOME/.nwchemrc</code>. Separate directories of this type should be defined for each supported force field. This directory may contain fragment files (with extension frg), segment files (with extension sgm) and a parameter file (with the name of the force field and with extension par).</li>
<li>temporary files :
    Temporary database files contain user preferred extensions to the original force field and are to reside in a separate directory that is specified in the file <code>$HOME/.nwchemrc</code>. There may be such a directory for each supported force field. This directory may contain fragment files (with extension frg), segment files (with extension sgm) and a parameter file (with the name of the force field and with extension par).</li>
<li>current files :
    Database files that contain user preferred extensions to the original force field and are to reside in a separate directory that is specified in the file <code>$HOME/.nwchemrc</code>. Typically this will be the current working directory, although it may be defined as a specific directory. This directory may contain fragment files (with extension frg), segment files (with extension sgm) and a parameter file (with the name of the force field and with extension par). If not specified, files will be taken from the current directory.</li>
</ul>
<p>Data is taken from the database files searched in the above order. If data is specified more than once, the last found values are used. For example, if some standard segment is redefined in a temporary file, the latter one will be used. This allows the user to redefine standards or extensions without having to modify those database files, which may reside in a generally available, non-modifyable directory. If a filename is specified rather than a directory, the filename indicates the parameter file definition. All other files (frg and sgm files) will be take from the specified directory.</p>
<p>The most common problems with the prepare module are</p>
<ol>
<li>The format of the pdb file does not conform to the pdb standard. In particular, atom names need to correspond with definitions in the fragment and segment database files, and should adhere to IUPAC recommendations as adopted by the pdb standard. If this problem occurs, the pdb file will need to be corrected.</li>
</ol>
<!-- -->

<ol>
<li>Non-standard segments may contain atoms that could not be atom typed with the existing typing rules in the force field parameter files. When this happens, additional typing rules can be included in the parameter file, or the fragment file may be manually typed.</li>
</ol>
<!-- -->

<ol>
<li>Parameters for atom types or bonded interactions do not exist in the force field. When this happens, additional parameters may be defined in the parameter files, or the segment file may be edited to include explicit parameters.</li>
</ol>
<h2 id="prepare-default-database-directories">Default database directories<a class="headerlink" href="#prepare-default-database-directories" title="Permanent link">&para;</a></h2>
<p>The file <code>$HOME/.nwchemrc</code> may contain the following entries that determine which files are used by the prepare module.</p>
<pre><code>ffield &lt;string ffname&gt;
</code></pre>
<p>This entry specifies the default force field. Database files supplied with NWChem currently support values for ffname of amber, referring to AMBER95, and charmm, referring to the academic CHARMM22 force field.</p>
<pre><code>&lt;string ffname&gt;_(1-9) &lt;string ffdir&gt;[{&lt;string parfile&gt;}]
</code></pre>
<p>Entries of this type specify the directory ffdir in which force field database files can be found. Optionally the parameterfile in this directory may be specified as parfile. The prepare module will only use files in directories specified here. One exception is that files in the current work directory will be used if no directory with current files is specified. The directories are read in the order 1-9 with duplicate parameters taken from the last occurrence found. Note that multiple parameter files may be specified that will be read in the order in which they are specified.</p>
<pre><code>&lt;string solvnam&gt; &lt;string solvfil&gt;
</code></pre>
<p>This entry may be used to identify a pure solvent restart file solvfil by a name solvnam</p>
<p>An example file <code>$HOME/.nwchemrc</code> is:</p>
<pre><code>ffield amber

amber_1 /soft/nwchem/share/amber/amber_s/amber99.par,spce.par

amber_2 /soft/nwchem/share/amber/amber_x/

amber_3 /usr/people/username/data/amber/amber_u/

spce /soft/nwchem/share/solvents/spce.rst

charmm_1 /soft/nwchem/share/charmm/charmm_s/

charmm_2 /soft/nwchem/share/charmm/charmm_x/
</code></pre>
<h2 id="prepare-system-name-and-coordinate-source">System name and coordinate source<a class="headerlink" href="#prepare-system-name-and-coordinate-source" title="Permanent link">&para;</a></h2>
<pre><code>system &lt;string sys_calc&gt;
</code></pre>
<p>The system name can be explicitly specified for the prepare module. If not specified, the system name will be taken from a specification in a previous md input block, or derived from the run time database name.</p>
<pre><code>source ( pdb | rtdb )
</code></pre>
<p>The source of the coordinates can be explicitly specified to be from a PDB formatted file sys.pdb, or from a geometry object in the run time database. If not specified, a pdb file will be used when it exists in the current directory or the rtdb geometry otherwise.</p>
<pre><code>model &lt;integer modpdb default 0&gt;
</code></pre>
<p>If a PDB formatted source file contains different MODELs, the model keyword can be used to specify which MODEL will be used to generate the topology and restart file. If not specified, the first MODEL found on the PDB file will be read.</p>
<pre><code>altloc &lt;character locpdb default ' '&gt;
</code></pre>
<p>The altloc keyword may be used to specify the use of alternate location coordinates on a PDB file.</p>
<pre><code>chain &lt;character chnpdb default ' '&gt;
</code></pre>
<p>The chain keyword may be used to specify the chain identifier for coordinates on a PDB file.</p>
<pre><code>histidine ( hid | hie | hip )
</code></pre>
<p>specifies the default protonation state of histidine.</p>
<pre><code>sscyx
</code></pre>
<p>Keyword sscyx may be used to rename cysteine residues that form sulphur bridges to CYX.</p>
<pre><code>hbuild
</code></pre>
<p>Keyword hbuild may be used to add hydrogen atoms to the unknown segments of the structure found on the pdb file. Placement of hydrogen atoms is based on geometric criteria, and the resulting fragment and segment files should be carefully examined for correctness.</p>
<p>The database directories are used as specified in the file <code>.nwchemrc</code>. Specific definitions for the force field used may be changed in the input file using</p>
<pre><code>directory_(1-9) &lt;string ffdir&gt; [&lt;string parfile&gt;]
</code></pre>
<h2 id="prepare-e-file-generation">e file generation<a class="headerlink" href="#prepare-e-file-generation" title="Permanent link">&para;</a></h2>
<p>If no existing e file is present in the current directory, or if the new_seq keyword was specified in the prepare input deck, a new sequence file is generated from information from the pdb file, and the following input directives.</p>
<pre><code>maxscf &lt;integer maxscf default 20&gt;
</code></pre>
<p>Variable maxscf specifies the maximum number of atoms in a segment for which partial atomic charges will be determined from an SCF calculation followed by RESP charge fitting. For larger segments a crude partial charge guestimation will be done.</p>
<pre><code>qscale &lt;real qscale default 1.0&gt;
</code></pre>
<p>Variable qscale specifies the factor with which SCF/RESP determined charges will be multiplied.</p>
<pre><code>modify sequence { &lt;integer sgmnum&gt;:&lt;string sgmnam&gt; }
</code></pre>
<p>This command specifies that segment sgmnam should be used for segment with number sgmnum. This command can be used to specify a particular protonation state. For example, the following command specifies that residue 114 is a hystidine protonated at the N<sub>&epsilon;</sub> site and residue 202 is a hystidine protonated at the N<sub>&delta;</sub> site:</p>
<pre><code>modify sequence 114:HIE 202:HID
</code></pre>
<p>Links between atoms can be enforced with</p>
<pre><code>link &lt;string atomname&gt; &lt;string atomname&gt;
</code></pre>
<p>For example, to link atom SG in segment 20 with atom FE in segment 55, use:</p>
<pre><code>link 20:_SG 55:FE
</code></pre>
<p>The format of the sequence file is given in <a href="#format_of_md_files-format-of-sequence-file">this table</a>.
In addition to the list of segments this file also includes links between non-standard segments or other non-standard links. These links are generated based on distances found between atoms on the pdb file. When atoms are involved in such non-standard links that have not been identified in the fragment of segment files as a non-chain link atom, the prepare module will ignore these links and report them as skipped. If one or more of these links are required, the user has to include them with explicit link directives in the sequence file, making them forced links. Alternatively, these links can be made forced-links by changing link into LINK in the sequence file.</p>
<pre><code>fraction { &lt;integer imol&gt; }
</code></pre>
<p>Directive fraction can be used to separate solute molecules into fractions for which energies will be separately reported during molecular dynamics simulations. The listed molecules will be the last molecule in a fraction. Up to 10 molecules may be specified in this directive.</p>
<pre><code>counter &lt;integer num&gt; &lt;string ion&gt;
</code></pre>
<p>Directive counter adds num counter ions of type ion to the sequence file. Up to 10 counter directives may appear in the input block.</p>
<pre><code>counter &lt;real factor&gt;
</code></pre>
<p>This directive scales the counter ion charge by the specified factor in the determination of counter ions positions.</p>
<h2 id="prepare-topology-file-generation">Topology file generation<a class="headerlink" href="#prepare-topology-file-generation" title="Permanent link">&para;</a></h2>
<pre><code>new_top [ new_seq ]
</code></pre>
<p>Keyword new_top is used to force the generation of a new topology file. An existing topology file for the system in the current directory will be overwritten. If keyword new_seq is also specified, an existing sequence file will also be overwritten with a newly generated file.</p>
<pre><code>amber | charmm
</code></pre>
<p>The prepare module generates force field specific fragment, segment and topology files. The force field may be explicitly specified in the prepare input block by specifying its name. Currently AMBER and CHARMM are the supported force fields. A default force field may be specified in the file <code>$HOME/.nwchemrc</code>.</p>
<pre><code>standard &lt;string dir_s&gt;[&lt;string par_s&gt;]
extensions &lt;string dir_x&gt;[&lt;string par_x&gt;]
contributed &lt;string dir_q&gt;[&lt;string par_q&gt;]
user &lt;string dir_u&gt;[&lt;string par_u&gt;]
temporary &lt;string dir_t&gt;[&lt;string par_t&gt;]
current &lt;string dir_c&gt;[&lt;string par_c&gt;]
</code></pre>
<p>The user can explicitly specify the directories where force field specific databases can be found. These include force field standards, extensions, quality assurance tests, user preferences, temporary , and current database files. Defaults for the directories where database files reside may be specified in the file <code>$HOME/.nwchemrc</code> for each of the supported force fields. Fragment, segment and sequence files generated by the prepare module are written in the temporary directory. When not specified, the current directory will be used. Topology and restart files are always created in the current directory.</p>
<p>The following directives control the modifications of a topology file. These directives are executed in the order in which they appear in the prepare input deck. The topology modifying commands are not stored on the run-time database and are, therefor, not persistent.</p>
<pre><code>modify atom &lt;string atomname&gt; [set &lt;integer mset&gt; | initial | final] \
   ( type &lt;string atomtyp&gt; |  charge &lt;real atomcharge&gt; |  \
   polar &lt;real atompolar&gt; | dummy | self | quantum | quantum_high )
</code></pre>
<p>These modify commands change the atom type, partial atomic charge, atomic polarizability, specify a dummy, self-interaction and quantum atom, respectively. If mset is specified, the modification will only apply to the specified set, which has to be 1, 2 or 3. If not specified, the modification will be applied to all three sets. The quantum region in QM/MM simulations is defined by specifying atoms with the quantum or quantum_high label. For atoms defined quantum_high basis sets labeled X_H will be used. The atomnam should be specified as <integer isgm>:<string name>, where isgm is the segment number, and name is the atom name. A leading blank in an atom name should be substituted with an underscore. The modify commands may be combined. For example, the following directive changes for the specified atom the charge and atom type in set 2 and specifies the atom to be a dummy in set 3.</p>
<pre><code>modify atom 12:_C1 set 2 charge 0.12 type CA set 3 dummy
</code></pre>
<p>With the following directives modifications can be made for entire segments.</p>
<pre><code>modify segment &lt;integer isgm&gt; \
  [protonation &lt;integer iprot&gt; | set &lt;integer mset&gt; | initial | final] \
  ( dummy | self | uncharged | quantum | quantum_high )
</code></pre>
<p>where protonation specifies a modification of the default protonation state of the segment as specified in the segment file. This option only applies to Q-HOP simulations.</p>
<p>Modifications to bonded interaction parameters can be made with the following modify commands.</p>
<pre><code>modify ( bond &lt;string atomtyp&gt; &lt;string atomtyp&gt; |  \
    angle &lt;string atomtyp&gt; &lt;string atomtyp&gt; &lt;string atomtyp&gt; |        \ 
    torsion &lt;string atomtyp&gt; &lt;string atomtyp&gt; &lt;string atomtyp&gt;        \
        &lt;string atomtyp&gt; [ multiplicity &lt;integer multip&gt; ] |      \
    plane &lt;string atomtyp&gt; &lt;string atomtyp&gt; &lt;string atomtyp&gt;          \
        &lt;string atomtyp&gt; ) [set &lt;integer mset&gt; | initial | final] \
    &lt;real value&gt; &lt;real forcon&gt;
</code></pre>
<p>where atomtyp and mset are defined as above, multip is the torsion ultiplicity for which the modification is to be applied, value is the reference bond, angle, torsion angle of out-of-plane angle value respectively, and forcon is the force constant for bond, angle, torsion angle of out-of-plane angle. When multip or mset are not defined the modification will be applied to all multiplicities and sets, respectively, for the identified bonded interaction.</p>
<p>After modifying atoms to quantum atoms the bonded interactions in which only quantum atoms are involved are removed from the bonded lists using</p>
<pre><code>update lists
</code></pre>
<p>Error messages resulting from parameters not being defined for bonded interaction in which only quantum atoms are involved are ignored using</p>
<pre><code>ignore
</code></pre>
<p>To specify that a free energy calculation will be carried out using the topology file, the following keyword needs to be specified,</p>
<pre><code>free
</code></pre>
<p>To specify that a Q-HOP simulation will be carried out using the topology file, the following keyword needs to be specified,</p>
<pre><code>qhop
</code></pre>
<p>To specify that only the first set of parameters should be used, even if multiple sets have been defined in the fragment or segment files, the following keyword needs to be specified,</p>
<pre><code>first
</code></pre>
<p>Note that keywords free, qhop and qhop are mutually exclusive.</p>
<h2 id="prepare-appending-to-an-existing-topology-file">Appending to an existing topology file<a class="headerlink" href="#prepare-appending-to-an-existing-topology-file" title="Permanent link">&para;</a></h2>
<pre><code>noe &lt;string atom1&gt; &lt;string atom3&gt; \

 &lt;real dist1&gt; &lt;real dist2&gt;  &lt;real dist3&gt; &lt;real forc1&gt; &lt;real forc2&gt;
</code></pre>
<p>This directive specifies a distance restraint potential between atoms <em>atom1</em> and <em>atom2</em>, with a harmonic function with force constant <em>forc1</em> between <em>dist1</em> and <em>dist2</em>, and a harmonic function with force constant <em>forc2</em> between <em>dist2</em> and <em>dist3</em>. For distances shorter than <em>dist1</em> or larger than <em>dist3</em>, a constant force is applied such that force and energy are continuous at <em>dist1</em> and <em>dist3</em>, respectively. Distances are given in nm, force constants in <span class="arithmatex">\(kJ mol^{-1} nm^{-2}\)</span>.</p>
<pre><code>select &lt;integer isel&gt; { &lt;string atoms&gt; }
</code></pre>
<p>Directive select specifies a group of atoms used in the definition of potential of mean force potentials.</p>
<p>The selected atoms are specified by the string atoms which takes the form</p>
<pre><code>[{isgm [ - jsgm ] [,]} [:] [{aname[,]}]
</code></pre>
<p>For example, all carbon and oxygen atoms in segments 3 and 6 through 12 are selected for group 1 by</p>
<pre><code>3,6-12:_C????,_O????
pmf [all] [bias] zalign &lt;integer isel&gt; &lt;real forcon1&gt; &lt;real forcon2&gt;
pmf [combine] [bias] xyplane &lt;integer isel&gt; &lt;real forcon1&gt; &lt;real forcon2&gt;
pmf [constraint] [bias] (distance | zdistance) &lt;integer isel&gt; &lt;integer jsel&gt; \
            &lt;real dist1&gt; &lt;real dist2&gt; &lt;real forcon1&gt; &lt;real forcon2&gt;
pmf [bias] angle &lt;integer isel&gt; &lt;integer jsel&gt; &lt;integer ksel&gt; \
            &lt;real angle1&gt; &lt;real angle2&gt; &lt;real forcon1&gt; &lt;real forcon2&gt;
pmf [bias] torsion &lt;integer isel&gt; &lt;integer jsel&gt; &lt;integer ksel&gt; &lt;integer lsel&gt; \
            &lt;real angle1&gt; &lt;real angle2&gt; &lt;real forcon1&gt; &lt;real forcon2&gt;
pmf [bias] basepair &lt;integer isel&gt; &lt;integer jsel&gt; \
            &lt;real dist1&gt; &lt;real dist2&gt; &lt;real forcon1&gt; &lt;real forcon2&gt;
pmf [bias] (zaxis | zaxis-cog) &lt;integer isel&gt; &lt;integer jsel&gt; &lt;integer ksel&gt; \
            &lt;real dist1&gt; &lt;real dist2&gt; &lt;real forcon1&gt; &lt;real forcon2&gt;
</code></pre>
<p>Directive pmf specifies a potential of mean force potential in terms of the specified atom selection. Option zalign specifies the atoms in the selection to be restrained to a line parallel to the z-axis. Option xyplane specifies the atoms in the selection to be restrained to a plane perpendicular to the z-axis. Options distance, angle and torsion, are defined in terms of the center of geometry of the specified atom selections. Keyword basepair is used to specify a harmonic potential between residues isel and jsel. Keywords zaxis and zaxis-cog can be used to pull atoms toward the z-axis. Option all may be specified to apply an equivalent pmf to each of the equivalent solute molecules in the system. Option combine may be specified to apply the specified pmf to the atoms in all of the equivalent solute molecules. Option constraint may be specified to a distance pmf to treat the distance as a constraint. Option bias may be specified to indicate that this function should be treated as a biasing potential. Appropriate corrections to free energy results will be evaluated.</p>
<h2 id="prepare-generating-a-restart-file">Generating a restart file<a class="headerlink" href="#prepare-generating-a-restart-file" title="Permanent link">&para;</a></h2>
<pre><code>new_rst
</code></pre>
<p>Keyword new_rst will cause an existing restart file to be overwritten with a new file.</p>
<p>The follwing directives control the manipulation of restart files, and are executed in the order in which they appear in the prepare input deck.</p>
<pre><code> solvent name &lt;string*3 slvnam default 'HOH'&gt; \  
         model &lt;string slvmdl default 'spce'&gt;  
</code></pre>
<p>The solvent keyword can be used to specify the three letter solvent name as expected on the PDB formatted file, and the name of the solvent model for which solvent coordinates will be used.</p>
<pre><code>solvate   [ &lt; real rshell default 1.2 &gt; ] \
        ( [ cube [ &lt;real edge&gt; ]] |  \
          [ box [ &lt;real xedge&gt; [ &lt;real xedge&gt; [ &lt;real xedge&gt; ]]]] | \
          [ sphere &lt;real radius&gt; ] |
          [ troct &lt;real edge&gt; ])
</code></pre>
<p>Solvation can be specified to be in a cubic box with specified edge, rectangular box with specified edges, or in a sphere with specified radius. Solvation in a cube or rectangular box will automatically also set periodic boundary conditions. Solvation in a sphere will only allow simulations without periodic boundary conditions. The size of the cubic and rectangular boxes will be expanded by a length specified by the expand variable. If no shape is specified, solvation will be done for a cubic box with an edge that leaves rshell nm between any solute atom and a periodic image of any solute atom after the solute has been centered. An explicit write is not needed to write the restart file. The solvate will write out a file sys_calc.rst. If not specified, the dimension of the solvation cell will be as large as to have at least a distance of rshell nm between any solute atom and the edge of the cell. The experimental troct directive generates a truncated octrahedral box.</p>
<pre><code>touch &lt;real touch default 0.23&gt;
</code></pre>
<p>The variable touch specifies the minimum distance between a solvent and solute atom for which a solvent molecule will be accepted for solvation.</p>
<pre><code>envelope &lt;real xpndw default 0.0&gt;
</code></pre>
<p>sets the expand vealues to be used in solvate operations.</p>
<pre><code>expand &lt;real xpndw default 0.1&gt;
</code></pre>
<p>The variable xpndw specifies the size in nm with which the simulation volume will be increased after solvation.</p>
<pre><code>read [rst | rst_old | pdb] &lt;string filename&gt;
write [rst | [solute [&lt;integer nsolvent&gt;]] ( [large] pdb | xyz)] &lt;string filename&gt;
</code></pre>
<p>These directives read and write the file filename in the specified format. The solute option instructs to write out the coordinates for solute and all, or if specified the first nsolvent, crystal solvent molecules only. If no format is specified, it will be derived from the extension of the filename. Recognized extensions are rst, rst_old (read only), pdb, xyz (write only) and pov (write only). Reading and then writing the same restart file will cause the sub-block size information to be lost. If this information needs to be retained a shell copy command needs to be used. The large keyword allows PDB files to be written with more than 9999 residues. Since the PDB file will not conform to the PDB convention, this option should only be used if required. NWChem will be able to read the resulting PDB file, but other codes may not.</p>
<pre><code>scale &lt;real scale default -1.0&gt;
</code></pre>
<p>This directive scales the volume and coordinates written to povray files. A negative value of scale (default) scales the coordinates to lie in [-1:1].</p>
<pre><code>cpk [&lt;real cpk default 1.0&gt;]
</code></pre>
<p>This directive causes povray files to contain cpk model output. The optional value is used to scale the atomic radii. A neagtive value of cpk resets the rendering to stick.</p>
<pre><code>center | centerx | centery | centerz
</code></pre>
<p>These directives center the solute center of geometry at the origin, in the y-z plane, in the x-z plane or in the x-y plane, respectively.</p>
<pre><code>orient
</code></pre>
<p>This directive orients the solute principal axes.</p>
<pre><code>translate [atom | segment | molecule] \
    &lt;integer itran&gt; &lt;integer itran&gt; &lt;real xtran(3)&gt;
</code></pre>
<p>This directive translates solute atoms in the indicated range by xtran, without checking for bad contacts in the resulting structure.</p>
<pre><code>rotate [atom | segment | molecule] \

    &lt;integer itran&gt; &lt;integer itran&gt; &lt;real angle&gt; &lt;real xrot(3)&gt;
</code></pre>
<p>This directive rotates solute atoms in the indicated range by angle around the vector given by xrot,, without checking for bad contacts in the resulting structure.</p>
<pre><code>remove solvent [inside | outside] [x &lt;real xmin&gt; &lt;real xmax&gt;] \
[y &lt;real ymin&gt; &lt;real ymax&gt;] [z &lt;real zmin&gt; &lt;real zmax&gt;]
</code></pre>
<p>This directive removes solvent molecules inside or outside the specified coordinate range.</p>
<pre><code>periodic
</code></pre>
<p>This directive enables periodic boundary conditions.</p>
<pre><code>vacuo
</code></pre>
<p>This directive disables periodic boundary conditions.</p>
<pre><code>grid &lt;integer mgrid default 24&gt; &lt;real rgrid default 0.2&gt;
</code></pre>
<p>This directive specifies the grid size of trial counter-ion positions and minimum distance between an atom in the system and a counter-ion.</p>
<pre><code>crop
</code></pre>
<p>prints minimum and maximum solute coordinates.</p>
<pre><code>boxsize
</code></pre>
<p>specifies to redetermine the box size.</p>
<pre><code>cube
</code></pre>
<p>specifies to redetermine the smallest cubic box size.</p>
<pre><code>box &lt;real xsize&gt; &lt;real ysize&gt;  &lt;real zsize&gt;
</code></pre>
<p>The box directive resets the box size.</p>
<pre><code>align &lt;string atomi&gt; &lt;string atomj&gt; &lt;string atomk&gt;
</code></pre>
<p>The align directive orients the system such that atomi and atomj are on the z-axis, and atomk in the x=y plane.</p>
<pre><code>repeat [randomx | randomy | randomz] [chains | molecules | fractions ] \
 &lt;integer nx&gt; &lt;integer ny&gt; &lt;integer nz&gt; [&lt;real dist&gt;] [&lt;real zdist&gt;]
</code></pre>
<p>The repeat directive causes a subsequent write pdb directive to write out multiple copies of the system, with nx copies in the x, ny copies in the y, and nz copies in the z-direction, with a minimum distance of dist between any pair of atoms from different copies. If nz is -2, an inverted copy is placed in the z direction, with a separation of zdist nm. If dist is negative, the box dimensions will be used. For systems with solvent, this directive should be used with a negative dist. Optional keywords chains, molecules and fractions specify to write each repeating solute unit as a chain, to repeat each solute molecule, or each solute fraction separately. Optional keywords randomx, randomy, and randomz can be used to apply random rotations for each repeat unit around a vector through the center of geometry of the solute in the x, y or z direction.</p>
<pre><code>skip &lt;integer ix&gt; &lt;integer iy&gt; &lt;integer iz&gt;
</code></pre>
<p>The skip directive can be used to skip single repeat unit from the repeat directive. Up to 100 skip directives may be specified, and will only apply to the previously specified repeat directive.</p>
<pre><code>(collapsexy | collapsez) [ &lt;integer nmoves&gt;]
</code></pre>
<p>specifies to move all solute molecules toward the z-axis or x=y-plane, respectively, to within a distance of touch nm between any pair of atoms from different solute molecules. Parameter nmoves specifies the number of collapse moves that will be made. Monatomic ions will move with the nearest multi-atom molecule.</p>
<pre><code>collapse_group &lt;integer imol&gt; &lt;integer jmol&gt;
</code></pre>
<p>specifies that molecule jmol will move together with molecule imol in collapse operations.</p>
<pre><code>merge &lt;real xtran(3)&gt; &lt;string pdbfile&gt;
</code></pre>
<p>specifies to merge the coordinates found on the specified pdb file into the current structure after translation by xtran(3).</p></section><section class="print-page" id="md" heading-number="17.2"><h1 id="md-molecular-dynamics">Molecular dynamics<a class="headerlink" href="#md-molecular-dynamics" title="Permanent link">&para;</a></h1>
<h2 id="md-introduction">Introduction<a class="headerlink" href="#md-introduction" title="Permanent link">&para;</a></h2>
<h3 id="md-spacial-decomposition">Spacial decomposition<a class="headerlink" href="#md-spacial-decomposition" title="Permanent link">&para;</a></h3>
<p>The molecular dynamics module of NWChem uses a distribution of data based on a spacial decomposition of the molecular system, offering an efficient parallel implementation in terms of both memory requirements and communication costs, especially for simulations of large molecular systems.</p>
<p>Inter-processor communication using the global array tools and the design of a data structure allowing distribution based on spacial decomposition are the key elements in taking advantage of the distribution of memory requirements and computational work with minimal communication.</p>
<p>In the spacial decomposition approach, the physical simulation volume is divided into rectangular cells, each of which is assigned to a processor. Depending on the conditions of the calculation and the number of available processors, each processor contains one or more of these spacially grouped cells. The most important aspects of this decomposition are the dependence of the cell sizes and communication cost on the number of processors and the shape of the cells, the frequent reassignment of atoms to cells leading to a fluctuating number of atoms per cell, and the locality of communication which is the main reason for the efficiency of this approach for very large molecular systems.</p>
<p>To improve efficiency, molecular systems are broken up into separately treated solvent and solute parts. Solvent molecules are assigned to the domains according to their center of geometry and are always owned by a one node. This avoids solvent-solvent bonded interactions crossing node boundaries. Solute molecules are broken up into segments, with each segment assigned to a processor based on its center of geometry. This limits the number of solute bonded interactions that cross node boundaries. The processor to which a particular cell is assigned is responsible for the calculation of all interactions between atoms within that cell. For the calculation of forces and energies in which atoms in cells assigned to different processors are involved, data are exchanged between processors. The number of neighboring cells is determined by the size and shape of the cells and the range of interaction. The data exchange that takes place every simulation time step represents the main communication requirements. Consequently, one of the main efforts is to design algorithms and data structures to minimize the cost of this communication. However, for very large molecular systems, memory requirements also need to be taken into account.</p>
<p>To compromise between these requirements exchange of data is performed in successive point to point communications rather than using the shift algorithm which reduces the number of communication calls for the same amount of communicated data.</p>
<p>For inhomogeneous systems, the computational load of evaluating atomic interactions will generally differ between cell pairs. This will lead to load imbalance between processors. Two algorithms have been implemented that allow for dynamically balancing the workload of each processor. One method is the dynamic resizing of cells such that cells gradually become smaller on the busiest node, thereby reducing the computational load of that node. Disadvantages of this method are that the efficiency depends on the solute distribution in the simulation volume and the redistribution of work depends on the number of nodes which could lead to results that depend on the number of nodes used. The second method is based on the dynamic redistribution of intra-node cell-cell interactions. This method represents a more coarse load balancing scheme, but does not have the disadvantages of the cell resizing algorithm. For most molecular systems the cell pair redistribution is the more efficient and preferred method.</p>
<p>The description of a molecular system consists of static and dynamic information. The static information does not change during a simulation and includes items such as connectivity, excluded and third neighbor lists, equilibrium values and force constants for all bonded and non-bonded interactions. The static information is called the topology of the molecular system, and is kept on a separate topology file. The dynamic information includes coordinates and velocities for all atoms in the molecular system, and is kept in a so-called restart file.</p>
<h3 id="md-topology">Topology<a class="headerlink" href="#md-topology" title="Permanent link">&para;</a></h3>
<p>The static information about a molecular system that is needed for a molecular simulation is provided to the simulation module in a topology file. Items in this file include, among many other things, a list of atoms, their non-bonded parameters for van der Waals and electrostatic interactions, and the complete connectivity in terms of bonds, angles and dihedrals.</p>
<p>In molecular systems, a distinction is made between solvent and solute, which are treated separately. A solvent molecule is defined only once in the topology file, even though many solvent molecules usually are included in the actual molecular system. In the current implementation only one solvent can be defined. Everything that is not solvent in the molecular system is solute. Each solute atom in the system must be explicitly defined in the topology.</p>
<p>Molecules are defined in terms of one or more segments. Typically, repetitive parts of a molecule are each defined as a single segment, such as the amino acid residues in a protein. Segments can be quite complicated to define and are, therefore, collected in a set of database files. The definition of a molecular system in terms of segments is a sequence.</p>
<p>Topology files are created using the prepare module.</p>
<h3 id="md-files">Files<a class="headerlink" href="#md-files" title="Permanent link">&para;</a></h3>
<p>File names used have the form <em>system</em>_<em>calc</em>.<em>ext</em>, with exception of the <a href="#md-topology">topology file</a>, which is named <em>system</em>.top. Anything that refers to the definition of the chemical system can be used for <em>system</em>, as long as no periods or underlines are used. The identifier <em>calc</em> can be anything that refers to the type of calculation to be performed for the system with the topology defined. This file naming convention allows for the creation of a single topology file <em>system</em>.top that can be used for a number of different calculations, each identified with a different <em>calc</em>. For example, if crown.top is the name of the topology file for a crown ether, crown_em, crown_md, crown_ti could be used with appropriate extensions for the filenames for energy minimization, molecular dynamics simulation and multi-configuration thermodynamic integration, respectively. All of these calculations would use the same topology file crown.top.</p>
<p>The extensions <code>&lt;ext&gt;</code> identify the kind of information on a file, and are pre-determined.</p>
<table>
<thead>
<tr>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>dbg</strong></td>
<td>debug file</td>
</tr>
<tr>
<td><strong>frg</strong></td>
<td>fragment file</td>
</tr>
<tr>
<td><strong>gib</strong></td>
<td>free energy data file</td>
</tr>
<tr>
<td><strong>mri</strong></td>
<td>free energy multiple run input file</td>
</tr>
<tr>
<td><strong>mro</strong></td>
<td>free energy multiple run output file</td>
</tr>
<tr>
<td><strong>nw</strong></td>
<td><strong>NWChem</strong> input file</td>
</tr>
<tr>
<td><strong>nwout</strong></td>
<td><strong>NWChem</strong> output file</td>
</tr>
<tr>
<td><strong>out</strong></td>
<td>molecular dynamics output file</td>
</tr>
<tr>
<td><strong>pdb</strong></td>
<td>PDB formatted coordinate file</td>
</tr>
<tr>
<td><strong>prp</strong></td>
<td>property file</td>
</tr>
<tr>
<td><strong>qrs</strong></td>
<td>quenched restart file, resulting from an energy minimization</td>
</tr>
<tr>
<td><strong>rst</strong></td>
<td>restart file, used to start or restart a simulation</td>
</tr>
<tr>
<td><strong>seq</strong></td>
<td>sequence file, describing the system in segments</td>
</tr>
<tr>
<td><strong>sgm</strong></td>
<td>segment file, describing segments</td>
</tr>
<tr>
<td><strong>syn</strong></td>
<td>synchronization time file</td>
</tr>
<tr>
<td><strong>tst</strong></td>
<td>test file</td>
</tr>
<tr>
<td><strong>tim</strong></td>
<td>timing analysis file</td>
</tr>
<tr>
<td><strong>top</strong></td>
<td>topology file, contains the static description of a system</td>
</tr>
<tr>
<td><strong>trj</strong></td>
<td>trajectory file</td>
</tr>
<tr>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<h3 id="md-databases">Databases<a class="headerlink" href="#md-databases" title="Permanent link">&para;</a></h3>
<p>Database file supplied with NWChem and used by the prepare module are found in directories with name ffield_<em>level</em>, where <em>ffield</em> is any of the supported <a href="#md-force-fields">force fields</a>. The source of the data is identified by <em>level</em>, and can be</p>
<table>
<thead>
<tr>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td>level</td>
<td>Description</td>
</tr>
<tr>
<td><strong>s</strong></td>
<td>original published data</td>
</tr>
<tr>
<td><strong>x</strong></td>
<td>additional published data</td>
</tr>
<tr>
<td><strong>q</strong></td>
<td>contributed data</td>
</tr>
<tr>
<td><strong>u</strong></td>
<td>user preferred data</td>
</tr>
<tr>
<td><strong>t</strong></td>
<td>user defined temporary data</td>
</tr>
<tr>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<p>The user is can replace these directories or add additional database files by specifying them in the <em>.nwchemrc</em> file. or in the prepare input file.</p>
<p>The extension 1-9 defines the priority of database file.</p>
<table>
<thead>
<tr>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>frg</strong></td>
<td>fragments</td>
</tr>
<tr>
<td><strong>par</strong></td>
<td>parameters</td>
</tr>
<tr>
<td><strong>seq</strong></td>
<td>sequences</td>
</tr>
<tr>
<td><strong>sgm</strong></td>
<td>segments</td>
</tr>
<tr>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<h3 id="md-force-fields">Force fields<a class="headerlink" href="#md-force-fields" title="Permanent link">&para;</a></h3>
<p>Force fields recognized are</p>
<table>
<thead>
<tr>
<th></th>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td>Keyword</td>
<td>Force field</td>
<td>Status</td>
</tr>
<tr>
<td>amber</td>
<td>AMBER99</td>
<td>AMBER95,GLYCAM also available</td>
</tr>
<tr>
<td>charmm</td>
<td>CHARMM</td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<h2 id="md-format-of-fragment-files">Format of fragment files<a class="headerlink" href="#md-format-of-fragment-files" title="Permanent link">&para;</a></h2>
<p>Fragment files contain the basic information needed to specify all interactions that need to be considered in a molecular simulation. Normally these files are created by the prepare module. Manual editing is needed when, for example, the prepare module could not complete atom typing, or when modified charges are required.</p>
<p>The formats of files used in NWChem are listed <a href="#format_of_md_files"> here</a>.</p>
<h2 id="md-creating-segment-files">Creating segment files<a class="headerlink" href="#md-creating-segment-files" title="Permanent link">&para;</a></h2>
<p>The prepare module is used to generate segment files from corresponding fragment files. A segment file contains all information for the calculation of bonded and non-bonded interactions for a given chemical system using a specific force field.</p>
<p>Which atoms form a fragment is specified in the coordinate file, currently only in PDB format. The segment entries define three sets of parameters for each interaction.</p>
<p>Free energy perturbations can be performed using set 1 for the generation of the ensemble while using sets 2 and/or 3 as perturbations. Free energy multiconfiguration thermodynamic integration and multistep thermodynamic perturbation calculations are performed by gradually changing the interactions in the system from parameter set 2 to parameter set 3. These modifications can be edited into the segment files manually, or introduced directly into the topology file using the modify commands in the input for the prepare module.</p>
<h2 id="md-creating-sequence-files">Creating sequence files<a class="headerlink" href="#md-creating-sequence-files" title="Permanent link">&para;</a></h2>
<p>A sequence file describes a molecular system in terms of segments. This file is generated by the prepare module for the molecular system provided on a PDB-formatted coordinate file</p>
<h2 id="md-creating-topology-files">Creating topology files<a class="headerlink" href="#md-creating-topology-files" title="Permanent link">&para;</a></h2>
<p>The <a href="#md-topology">topology</a> describes all static information that describes a molecular system. This includes the connectivity in terms of bond-stretching, angle-bending and torsional interactions, as well as the non-bonded van der Waals and Coulombic interactions.</p>
<p>The topology of a molecular system is generated by the prepare module from the sequence in terms of segments as specified on the PDB file. For each unique segment specified in this file the segment database directories are searched for the segment definition. For segments not found in one of the database directories a segment definition is generated in the temporary directory if a fragment file was found. If a fragment file could not be found, it is generated by the prepare module base on what is found on the PDB file.</p>
<p>When all segments are found or created, the parameter substitutions are performed, using force field parameters taken from the parameter databases. After all lists have been generated the topology is written to a local topology file <code>&lt;system&gt;</code>.top.</p>
<h2 id="md-creating-restart-files">Creating restart files<a class="headerlink" href="#md-creating-restart-files" title="Permanent link">&para;</a></h2>
<p>Restart files contain all dynamical information about a molecular system and are created by the prepare module if a topology file is available. The prepare module will automatically generate coordinates for hydrogen atoms and monatomic counter ions not found on the PDB formatted coordinate file, if no fragment or segment files were generated using that PDB file.</p>
<p>The prepare module has a number of other optional input command, including solvation.</p>
<h2 id="md-molecular-simulations">Molecular simulations<a class="headerlink" href="#md-molecular-simulations" title="Permanent link">&para;</a></h2>
<p>The type of molecular dynamics simulation is specified by the NWChem task directive.</p>
<pre><code>task md [ energy | optimize | dynamics | thermodynamics ]
</code></pre>
<p>where the theory keyword md specifies use of the molecular dynamics module, and the operation keyword is one of</p>
<ul>
<li><em>energy</em> for single configuration energy evaluation</li>
<li><em>optimize</em> for energy minimization</li>
<li><em>dynamics</em> for molecular dynamics simulations and single step thermodynamic perturbation free energy molecular dynamics simulations</li>
<li><em>thermodynamics</em> for combined multi-configuration thermodynamic integration and multiple step thermodynamic perturbation free energy molecular dynamics simulations.</li>
</ul>
<h2 id="md-system-specification">System specification<a class="headerlink" href="#md-system-specification" title="Permanent link">&para;</a></h2>
<p>The chemical system for a calculation is specified in the topology and restart files. These files should be created using the utilities nwtop and nwrst before a simulation can be performed. The names of these files are determined from the required system directive.</p>
<pre><code>system &lt;string systemid&gt;_&lt;string calcid&gt;
</code></pre>
<p>where the strings systemid and calcid are user defined names for the chemical system and the type of calculation to ber performed, respectively. These names are used to derive the filenames used for the calculation. The topoly file used will be systemid.top, while all other files are named systemid_calcid.ext.</p>
<h2 id="md-restarting-and-continuing-simulations">Restarting and continuing simulations<a class="headerlink" href="#md-restarting-and-continuing-simulations" title="Permanent link">&para;</a></h2>
<pre><code>   finish
</code></pre>
<p>Specifies that the current job will finish a previous, incomplete simulation, using the input data that have been recorded by that previous run in the restart file. Most of the input in the current md input block will be ignored.</p>
<pre><code>   resume
</code></pre>
<p>Specifies that the current job will be an extension of a previous simulation, using most of the input data that have been recorded by that previous run in the restart file. Typically the input in the current md input block defines a larger number of steps than the previous job.</p>
<h2 id="md-parameter-set">Parameter set<a class="headerlink" href="#md-parameter-set" title="Permanent link">&para;</a></h2>
<pre><code>set &lt;integer iset&gt;
</code></pre>
<p>Specifies the use of parameter set <code>&lt;iset&gt;</code> for the molecular dynamics simulation. The topology file contains three separate parameters sets that can be used. The default for <code>&lt;iset&gt;</code> is 1.</p>
<pre><code>lambda &lt;integer ilambda&gt; &lt;integer ilambda&gt;
</code></pre>
<p>Specifies the use of parameter set for the ilambda-th of mlambda steps.</p>
<pre><code>pset &lt;integer isetp1&gt; [&lt;integer isetp2&gt;]
</code></pre>
<p>Specifies the parameter sets to be used as perturbation potentials in single step thermodynamic perturbation free energy evaluations, where <code>&lt;isetp1&gt;</code> specifies the first perturbation parameter set and <code>&lt;isetp2&gt;</code> specifies the second perturbation parameter set. Legal values for <code>&lt;isetp1&gt;</code> are 2 and 3. Legal value for <code>&lt;isetp2&gt;</code> is 3, in which case <code>&lt;isetp1&gt;</code> can only be 2. If specified, <code>&lt;iset&gt;</code> is automatically set to 1.</p>
<pre><code>pmf [ equilharm &lt;integer npmfc&gt; | scale &lt;real facpmf&gt;]
</code></pre>
<p>Specifies that any potential of mean force functions defined in the topology files are to be used. If equilharm is specified, the first npmfc dynamics steps will use a harmonic potential in stead of any pmf constraint. If scale is specified, all pmf force constants are scaled by a factor facpmf.</p>
<pre><code>distar [draver [&lt;integer ndaver default 1&gt;]] 
  [scale &lt;real drsscl&gt;]
  [after &lt;integer nfdrss&gt;]
</code></pre>
<p>Specifies that any distance restraint functions defined in the topology files are to be used.</p>
<pre><code>qhop [&lt;integer nfhop default 10&gt;] 
  [&lt;real rhop default 0.35&gt;]
  [&lt;real thop default 0.02&gt;]
</code></pre>
<p>Specifies that a Q-HOP simulation is to be carried out with attempted proton hops every <em>nfhop</em> steps, a cutoff for the donor-acceptor pair distance of <em>rhop</em> nm, and a minimum time before back hopping can occur of <em>thop</em> ps.</p>
<h2 id="md-energy-minimization-algorithms">Energy minimization algorithms<a class="headerlink" href="#md-energy-minimization-algorithms" title="Permanent link">&para;</a></h2>
<p>The energy minimization of the system as found in the restart file is performed with the following directives. If both are specified, steepest descent energy minimization precedes conjugate gradient minimization.</p>
<pre><code>   sd &lt;integer msdit&gt; [init &lt;real dx0sd&gt;] [min &lt;real dxsdmx&gt;] \
                      [max &lt;real dxmsd&gt;]
</code></pre>
<p>Specifies the variables for steepest descent energy minimizations, where <code>&lt;msdit&gt;</code> is the maximum number of steepest descent steps taken, for which the default is 100, <code>&lt;dx0sd&gt;</code> is the initial step size in nm for which the default is 0.001, <code>&lt;dxsdmx&gt;</code> is the threshold for the step size in nm for which the default is 0.0001, and <code>&lt;dxmsd&gt;</code> is the maximum allowed step size in nm for which the default is 0.05.</p>
<pre><code>   cg &lt;integer mcgit&gt; [init &lt;real dx0cg&gt;] [min &lt;real dxcgmx&gt;] \
                      [cy &lt;integer ncgcy&gt;]
</code></pre>
<p>Specifies the variables for conjugate gradient energy minimizations, where <code>&lt;mcgit&gt;</code> is the maximum number of conjugate gradient steps taken, for which the default is 100, <code>&lt;dx0cg&gt;</code> is the initial search interval size in nm for which the default is 0.001, <code>&lt;dxcgmx&gt;</code> is the threshold for the step size in nm for which the default is 0.0001, and <code>&lt;ncgcy&gt;</code> is the number of conjugate gradient steps after which the gradient history is discarded for which the default is 10. If conjugate gradient energy minimization is preceded by steepest descent energy minimization, the search interval is set to twice the final step of the steepest descent energy minimization.</p>
<h2 id="md-multi-configuration-thermodynamic-integration">Multi-configuration thermodynamic integration<a class="headerlink" href="#md-multi-configuration-thermodynamic-integration" title="Permanent link">&para;</a></h2>
<p>The following keywords control free energy difference simulations. Multi-configuration thermodynamic integrations are always combined with multiple step thermodynamic perturbations.</p>
<pre><code>   (forward | reverse) [[&lt;integer mrun&gt; of] &lt;integer maxlam&gt;]
</code></pre>
<p>Specifies the direction and number of integration steps in free energy evaluations, with forward being the default direction. <code>&lt;mrun&gt;</code> is the number of ensembles that will be generated in this calculation, and <code>&lt;maxlam&gt;</code> is the total number of ensembles to complete the thermodynamic integration. The default value for <code>&lt;maxlam&gt;</code> is 21. The default value of <code>&lt;mrun&gt;</code> is the value of <code>&lt;maxlam&gt;</code>.</p>
<pre><code>   error &lt;real edacq&gt;
</code></pre>
<p>Specifies the maximum allowed statistical error in each generated ensemble, where <code>&lt;edacq&gt;</code> is the maximum error allowed in the ensemble average derivative of the Hamiltonian with respect to  with a default of 5.0 kJ mol<sup>-1</sup>.</p>
<pre><code>   drift &lt;real ddacq&gt;
</code></pre>
<p>Specifies the maximum allowed drift in the free energy result, where <ddacq> is the maximum drift allowed in the ensemble average derivative of the Hamiltonian with respect to  with a default of 5.0 kJ mol<sup>-1</sup> ps<sup>-1</sup>.</p>
<pre><code>   factor &lt;real fdacq&gt;
</code></pre>
<p>Specifies the maximum allowed change in ensemble size where <code>&lt;fdacq&gt;</code> is the minimum size of an ensemble relative to the previous ensemble in the calculation with a default value of 0.75.</p>
<pre><code>   decomp
</code></pre>
<p>Specifies that a free energy decomposition is to be carried out. Since free energy contributions are path dependent, results from a decomposition analysis can no be unambiguously interpreted, and the default is not to perform this decomposition.</p>
<pre><code>   sss [delta &lt;real delta&gt;]
</code></pre>
<p>Specifies that atomic non-bonded interactions describe a dummy atom in either the initial or final state of the thermodynamic calculation will be calculated using separation-shifted scaling, where <code>&lt;delta&gt;</code> is the separation-shifted scaling factor with a default of 0.075 nm<sup>2</sup>. This scaling method prevents problems associated with singularities in the interaction potentials.</p>
<pre><code>   new | renew | extend
</code></pre>
<p>Specifies the initial conditions for thermodynamic calculations. new indicates that this is an initial mcti calculation, which is the default. renew instructs to obtain the initial conditions for each  from the mro-file from a previous mcti calculation, which has to be renamed to an mri-file. The keyword extend will extend a previous mcti calculation from the data read from an mri-file.</p>
<h2 id="md-time-and-integration-algorithm-directives">Time and integration algorithm directives<a class="headerlink" href="#md-time-and-integration-algorithm-directives" title="Permanent link">&para;</a></h2>
<p>Following directives control the integration of the equations of motion.</p>
<pre><code>   leapfrog | leapfrog_bc
</code></pre>
<p>Specifies the integration algorithm, where leapfrog specifies the default leap frog integration, and leapfrog_bc specifies the Brown-Clarke leap frog integrator.</p>
<pre><code>   guided [&lt;real fguide default 0.2&gt; [&lt;real tguide default 0.2&gt;]]
</code></pre>
<p>Specifies the use of the guided molecular dynamics simulation technique. Variable <em>fguide</em> defines the fraction of the averaged forces <em>g</em> to be added to the forces <em>f<sup>f</sup></em> evaluated using the force field functions to obtain the forces <em>f</em> used to advance the coordinates.</p>
<div class="arithmatex">\[f_i=f^{f}_i+fguide * g_{i-1}\]</div>
<p>Variable tguide defines the length of the averaging relative to the timestep  <em>t</em>.</p>
<div class="arithmatex">\[g_i = {\Delta t\over tguide} f_i + \left(1- {\Delta t\over tguide}\right) g_{i-1}\]</div>
<p>The current implementation is still under development.</p>
<pre><code>   equil &lt;integer mequi&gt;
</code></pre>
<p>Specifies the number of equilibration steps <code>&lt;mequi&gt;</code>, with a default of 100.</p>
<pre><code>   data &lt;integer mdacq&gt; [over &lt;integer ldacq&gt;]]
</code></pre>
<p>Specifies the number of data gathering steps <code>&lt;mdacq&gt;</code> with a default of 500. In multi-configuration thermodynamic integrations <code>&lt;mequi&gt;</code> and <code>&lt;mdacq&gt;</code> are for each of the ensembles, and variable <code>&lt;ldacq&gt;</code> specifies the minimum number of data gathering steps in each ensemble. In regular molecular dynamics simulations <code>&lt;ldacq&gt;</code> is not used. The default value for <code>&lt;ldacq&gt;</code> is the value of <code>&lt;mdacq&gt;</code>.</p>
<pre><code>   time &lt;real stime&gt;
</code></pre>
<p>Specifies the initial time <code>&lt;stime&gt;</code> of a molecular simulation in ps, with a default of 0.0.</p>
<pre><code>   step &lt;real tstep&gt;
</code></pre>
<p>Specifies the time step <code>&lt;tstep&gt;</code> in ps, with 0.001 as the default value.</p>
<h2 id="md-ensemble-selection">Ensemble selection<a class="headerlink" href="#md-ensemble-selection" title="Permanent link">&para;</a></h2>
<p>Following directives control the ensemble type.</p>
<pre><code>   isotherm [&lt;real tmpext&gt; [&lt;real tmpext2&gt;]] [trelax &lt;real tmprlx&gt; [&lt;real tmsrlx&gt;]] \
            [anneal [&lt;real tann1&gt;] &lt;real tann2&gt;]
</code></pre>
<p>Specifies a constant temperature ensemble using Berendsen&rsquo;s thermostat, where <code>&lt;tmpext&gt;</code> is the external temperature with a default of 298.15 K, and <code>&lt;tmprlx&gt;</code> and <code>&lt;tmsrlx&gt;</code> are temperature relaxation times in ps with a default of 0.1. If only <code>&lt;tmprlx&gt;</code> is given the complete system is coupled to the heat bath with relaxation time <code>&lt;tmprlx&gt;</code>. If both relaxation times are supplied, solvent and solute are independently coupled to the heat bath with relaxation times <code>&lt;tmprlx&gt;</code> and <code>&lt;tmsrlx&gt;</code>, respectively. If keyword anneal is specified, the external temperature will change from tmpext to tempext2 between simulation time tann1 and tann2</p>
<pre><code>   isobar [&lt;real prsext&gt;] [trelax &lt;real prsrlx&gt; ] \
          [compress &lt;real compr&gt;] [anisotropic] [xy | z | xy-z]
</code></pre>
<p>Specifies a constant pressure ensemble using Berendsen&rsquo;s piston, where <code>&lt;prsext&gt;</code> is the external pressure with a default of 1.025 10<sup>5</sup> Pa, <code>&lt;prsrlx&gt;</code> is the pressure relaxation time in ps with a default of 0.5, and <code>&lt;compr&gt;</code> is the system compressibility in <span class="arithmatex">\(m{^2}N^{-1}\)</span> with a default of 4.53E-10. Optional keywords xy, z and xy-z may be used to specify that pressure scaling is to be applied in the x and y dimension only, the z dimension only, or, in all three dimensions with identical scaling in the x and y dimension. The last option requires that anisotropic is also specified.</p>
<h2 id="md-velocity-reassignments">Velocity reassignments<a class="headerlink" href="#md-velocity-reassignments" title="Permanent link">&para;</a></h2>
<p>Velocities can be periodically reassigned to reflect a certain temperature.</p>
<pre><code>   vreass &lt;integer nfgaus&gt; &lt;real tgauss&gt;
          [fraction [&lt;real frgaus default 0.5]]
          [once]
          [(first | initial)] [(last | final)]
</code></pre>
<p>Specifies that velocities will be reassigned every <code>&lt;nfgaus&gt;</code> molecular dynamics steps, reflecting a temperature of <code>&lt;tgauss&gt;</code> K. The default is not to reassign velocities, i.e. <code>&lt;nfgaus&gt;</code> is 0. Keyword fraction allows the specification of the fraction of the new velocities are random. Keyword once specifies that velocity reassignment only should be done in the first step. Keywords first or initial and last or final specify that velocity reassigment should only be applied in the first and last window of multiple run simulations.</p>
<h2 id="md-cutoff-radii">Cutoff radii<a class="headerlink" href="#md-cutoff-radii" title="Permanent link">&para;</a></h2>
<p>Cutoff radii can be specified for short range and long range interactions.</p>
<pre><code>   cutoff [short] &lt;real rshort&gt; [long &lt;real rlong&gt;] \
          [qmmm &lt;real rqmmm&gt;]
</code></pre>
<p>Specifies the short range cutoff radius <code>&lt;rshort&gt;</code>, and the long range cutoff radius <code>&lt;rlong&gt;</code> in nm. If the long range cutoff radius is larger than the short range cutoff radius the twin range method will be used, in which short range forces and energies are evaluated every molecular dynamics step, and long range forces and energies with a frequency of <code>&lt;nflong&gt;</code> molecular dynamics steps. Keyword qmmm specifies the radius of the zone around quantum atoms defining the QM/MM bare charges. The default value for <code>&lt;rshort&gt;</code>, <code>&lt;rlong&gt;</code> and <code>&lt;rqmmm&gt;</code> is 0.9 nm.</p>
<h2 id="md-polarization">Polarization<a class="headerlink" href="#md-polarization" title="Permanent link">&para;</a></h2>
<p>First order and self consistent electronic polarization models have been implemented.</p>
<pre><code>   polar (first | scf [[&lt;integer mpolit&gt;] &lt;real ptol&gt;])
</code></pre>
<p>Specifies the use of polarization potentials, where the keyword first specifies the first order polarization model, and scf specifies the self consistent polarization field model, iteratively determined with a maximum of <code>&lt;mpolit&gt;</code> iterations to within a tolerance of <code>&lt;ptol&gt;</code> D in the generated induced dipoles. The default is not to use polarization models.</p>
<h2 id="md-external-electrostatic-field">External electrostatic field<a class="headerlink" href="#md-external-electrostatic-field" title="Permanent link">&para;</a></h2>
<pre><code>   field &lt;real xfield&gt; [freq &lt;real xffreq&gt;] [vector &lt;real xfvect(1:3)&gt;]
</code></pre>
<p>Specifies an external electrostatic field, where <code>&lt;xfield&gt;</code> is the field strength, <code>&lt;xffreq&gt;</code> is the frequency in MHz and <code>&lt;xfvect&gt;</code> is the external field vector.</p>
<h2 id="md-constraints">Constraints<a class="headerlink" href="#md-constraints" title="Permanent link">&para;</a></h2>
<p>Constraints are satisfied using the SHAKE coordinate resetting procedure.</p>
<pre><code>   shake [&lt;integer mshitw&gt; [&lt;integer mshits&gt;]]  \
         [&lt;real tlwsha&gt; [&lt;real tlssha&gt;]]
</code></pre>
<p>Specifies the use of SHAKE constraints, where <code>&lt;mshitw&gt;</code> is the maximum number of solvent SHAKE iterations, and <code>&lt;mshits&gt;</code> is the maximum number of solute SHAKE iterations. If only <code>&lt;mshitw&gt;</code> is specified, the value will also be used for <code>&lt;mshits&gt;</code>. The default maximum number of iterations is 100 for both. <code>&lt;tlwsha&gt;</code> is the solvent SHAKE tolerance in nm, and <code>&lt;tlssha&gt;</code> is the solute SHAKE tolerance in nm. If only <code>&lt;tlwsha&gt;</code> is specified, the value given will also be used for <code>&lt;tlssha&gt;</code>. The default tolerance is 0.001 nm for both.</p>
<pre><code>   noshake (solvent | solute)
</code></pre>
<p>Disables SHAKE and treats the bonded interaction according to the force field.</p>
<h2 id="md-long-range-interaction-corrections">Long range interaction corrections<a class="headerlink" href="#md-long-range-interaction-corrections" title="Permanent link">&para;</a></h2>
<p>Long range electrostatic interactions are implemented using the smooth particle mesh Ewald technique, for neutral periodic cubic systems in the constant volume ensemble, using pair interaction potentials. Particle-mesh Ewald long range interactions can only be used in molecular dynamics simulations using effective pair potentials, and not in free energy simulations, QMD or QM/MM simulations.</p>
<pre><code>   pme [grid &lt;integer ng&gt;] [alpha &lt;real ealpha&gt;] \
       [order &lt;integer morder&gt;] [fft &lt;integer imfft&gt;]\
       [procs &lt;integer nprocs&gt;] [solvent]
</code></pre>
<p>Specifies the use of smooth particle-mesh Ewald long range interaction treatment, where ng is the number of grid points per dimension, ealpha is the Ewald coefficient in <span class="arithmatex">\(nm^{-1}\)</span>, with a default that leads to a tolerance of <span class="arithmatex">\(10^{-4}\)</span> at the short range cutoff radius, and morder is order of the Cardinal B-spline interpolation which must be an even number and at least 4 (default value). A platform specific 3D fast Fourier transform is used, if available, when imfft is set to 2. nprocs can be used to define a subset of processors to be used to do the FFT calculations. If solvent is specified, the charge grid will be calculated from the solvent charges only.</p>
<pre><code>   react [&lt;real dielec default 80.0&gt;]
</code></pre>
<p>Specifies that a simple reaction field correction is used with a dielectric constant dielec. This is an experimental option that has not been well tested.</p>
<h2 id="md-fixing-coordinates">Fixing coordinates<a class="headerlink" href="#md-fixing-coordinates" title="Permanent link">&para;</a></h2>
<p>Solvent or solute may be fixed using the following keywords.</p>
<pre><code>   ( fix | free ) 
    solvent ( [&lt;integer idfirst&gt; [&lt;integer idlast&gt;]] | 
               ( within | beyond) &lt;real rfix&gt; &lt;string atomname&gt; ) | \
    solute  ( [&lt;integer idfirst&gt; [&lt;integer idlast&gt;]] [ heavy | {&lt;string atomname&gt;}] |
               ( within | beyond) &lt;real rfix&gt; &lt;string atomname&gt; )
    [permanent]
</code></pre>
<p>For solvent the molecule numbers idfirst and idlastmay be specified to be the first and last molecule to which the directive applies. If omitted, the directive applies to all molecules. For solute, the segment numbers idfirst and idlastmay be specified to be the first and last segment to which the directive applies. If omitted, the directive applies to all segments. In addition, the keyword heavy may be specified to apply to all non hydrogen atoms in the solute, or a set of atom names may be specified in which a wildcard character ? may be used. Keyword permanent is used to keep the specification on the restart file for subsequent simulations.</p>
<h2 id="md-special-options">Special options<a class="headerlink" href="#md-special-options" title="Permanent link">&para;</a></h2>
<pre><code>   import [&lt;integer impfr default 1&gt; [&lt;integer impto default impfr&gt; \
          [&lt;integer nftri default 1&gt;]]]
</code></pre>
<p>Specifies the import of frames impfr to impto with frequency nftri from a trajectory file with extension tri for which energies and forces are to be recalculated. This option only applied to task md energy.</p>
<pre><code>   detail
</code></pre>
<p>Specifies that moments of inertia and radii of gyration will be part of the recorded properties.</p>
<pre><code>   profile
</code></pre>
<p>Specifies that execution time profiling data will be part of the recorded properties.</p>
<pre><code>   scale &lt;real scaleq&gt;
</code></pre>
<p>Specifies that all charges will be scaled by the factro scaleq.</p>
<pre><code>   collapse [&lt;real fcoll default 10.0&gt; [ segment | z | xy ]
</code></pre>
<p>Specifies that additional forces directed to the origin of the simulation cell with strength fcoll will be applied to all solute molecules. If z or xy is specified, these forces will only apply in the specified dimension(s).</p>
<pre><code>   include fixed
</code></pre>
<p>Specifies that energies will be evaluated between fixed atoms. Normally these interactions are excluded from the pairlists.</p>
<pre><code>   eqm &lt;real eqm&gt;
</code></pre>
<p>Specifies the zero point of energy in QMD simulations.</p>
<pre><code>   atomlist
</code></pre>
<p>Specifies that pairlists will be atom based. Normally pairlist are charge group based.</p>
<h2 id="md-autocorrelation-function">Autocorrelation function<a class="headerlink" href="#md-autocorrelation-function" title="Permanent link">&para;</a></h2>
<p>For the evaluation of the statistical error of multi-configuration thermodynamic integration free energy results a correlated data analysis is carried out, involving the calculation of the autocorrelation function of the derivative of the Hamiltonian with respect to the control variable .</p>
<pre><code>   auto &lt;integer lacf&gt; [fit &lt;integer nfit&gt;] [weight &lt;real weight&gt;]
</code></pre>
<p>Controls the calculation of the autocorrelation, where <code>&lt;lacf&gt;</code> is the length of the autocorrelation function, with a default of 1000, <code>&lt;nfit&gt;</code> is the number of functions used in the fit of the autocorrelation function, with a default of 15, and <code>&lt;weight&gt;</code> is the weight factor for the autocorrelation function, with a default value of 0.0.</p>
<h2 id="md-print-options">Print options<a class="headerlink" href="#md-print-options" title="Permanent link">&para;</a></h2>
<p>Keywords that control print to the output file, with extension out. Print directives may be combined to a single directive.</p>
<pre><code>   print [topol [nonbond] [solvent] [solute]]     \
         [step &lt;integer nfoutp&gt; [extra] [energy]] \
         [stat &lt;integer nfstat&gt;]                  \
         [energies [&lt;integer nfener&gt;]]            \
         [forces [&lt;integer nfforce&gt;]]             \
         [matrix]                                 \
         [expect &lt;integer npxpct&gt;]                \
         [timing]                                 \
         [pmf [&lt;integer iprpmf&gt;]]                 \
         [out6]                                   \
         [dayout]
</code></pre>
<ul>
<li>Keyword topol specifies printing the topology information, where nonbond refers to the non-bonded interaction parameters, solvent to the solvent bonded parameters, and solute to the solute bonded parameters. If only topol is specified, all topology information will be printed to the output file.</li>
</ul>
<!-- -->

<ul>
<li>Keyword step specifies the frequency nfoutp of printing molecular dynamics step information to the output file. If the keyword extra is specified additional energetic data are printed for solvent and solute separately. If the keyword energy is specified, information is printed for all bonded solute interactions. The default for nfoutp is 0. For molecular dynamics simulations this frequency is in time steps, and for multi-configuration thermodynamic integration in -steps.</li>
</ul>
<!-- -->

<ul>
<li>Keyword stat specifies the frequency <code>&lt;nfstat&gt;</code> of printing statistical information of properties that are calculated during the simulation. For molecular dynamics simulation this frequency is in time steps, for multi-configuration thermodynamic integration in -steps.</li>
</ul>
<!-- -->

<ul>
<li>Keyword energies specifies the frequency nfener of printing solute bonded energies the output file for energy/import calculations. The default for nfener is 0.</li>
</ul>
<!-- -->

<ul>
<li>Keyword forces specifies the frequency nfforc of printing solute forces the output file for energy/import calculations. The default for nfforc is 0.</li>
</ul>
<!-- -->

<ul>
<li>Keyword matrix specifies that a solute distance matrix is to be printed.</li>
</ul>
<!-- -->

<ul>
<li>Keyword expect is obsolete.</li>
</ul>
<!-- -->

<ul>
<li>Keyword timing specifies that timing data is printed.</li>
</ul>
<!-- -->

<ul>
<li>Keyword pmf specifies that pmf data is printed every iprpmf steps. Keyword out6 specifies that output is written to standard out in stead of the output file with extension out.</li>
</ul>
<!-- -->

<ul>
<li>Keyword dayout is obsolete.</li>
</ul>
<h2 id="md-periodic-updates">Periodic updates<a class="headerlink" href="#md-periodic-updates" title="Permanent link">&para;</a></h2>
<p>Following keywords control periodic events during a molecular dynamics or thermodynamic integration simulation. Update directives may be combined to a single directive.</p>
<pre><code>   update [pairs &lt;integer nfpair default 1&gt;]                  \

          [long &lt;integer nflong default 1&gt;]                   \

          [center &lt;integer nfcntr default 0&gt; [zonly | xyonly] \

                  [fraction &lt;integer idscb(1:5)&gt;]             \

          [motion &lt;integer nfslow default 0&gt;]                 \

          [analysis &lt;integer nfanal default 0&gt;]               \

          [rdf &lt;integer nfrdf default 0&gt;                      \

                  [range &lt;real rrdf&gt;] [bins &lt;integer ngl&gt;]    \
</code></pre>
<ul>
<li>Keyword pairs specifies the frequency <code>&lt;nfpair&gt;</code> in molecular dynamics steps of updating the pair lists. The default for the frequency is 1. In addition, pair lists are also updated after each step in which recording of the restart or trajectory files is performed. Updating the pair lists includes the redistribution of atoms that changed domain and load balancing, if specified.</li>
</ul>
<!-- -->

<ul>
<li>Keyword long specifies the frequency <code>&lt;nflong&gt;</code> in molecular dynamics steps of updating the long range forces. The default frequency is 1. The distinction of short range and long range forces is only made if the long range cutoff radius was specified to be larger than the short range cutoff radius. Updating the long range forces is also done in every molecular dynamics step in which the pair lists are regenerated.</li>
</ul>
<!-- -->

<ul>
<li>Keywrod center specifies the frequency <code>&lt;nfcntr&gt;</code> in molecular dynamics steps in which the center of geometry of the solute(s) is translated to the center of the simulation volume. Optional keyword zonly or xyonly can be used to specify that centering will take place in the z-direction or in the xy-plane only. The solute fractions determining the solutes that will be centered are specified by the keyword fraction and the vector <code>&lt;idscb&gt;</code>, with a maximum of 5 entries. This translation is implemented such that it has no effect on any aspect of the simulation. The default is not to center, i.e. nfcntr is 0. The default fraction used to center solute is 1.</li>
</ul>
<!-- -->

<ul>
<li>Keyword motion specifies the frequency <code>&lt;nfslow&gt;</code> in molecular dynamics steps of removing the overall rotational and center of mass translational motion.</li>
</ul>
<!-- -->

<ul>
<li>Keyword analysis specifies the frequency <code>&lt;nfanal&gt;</code> in molecular dynamics steps of invoking the analysis module. This option is obsolete.</li>
</ul>
<!-- -->

<ul>
<li>Keyword rdf specifies the frequency <code>&lt;nfrdf&gt;</code> in molecular dynamics steps of calculating contributions to the radial distribution functions. The default is 0. The range of the radial distribution functions is given by <code>&lt;rrdf&gt;</code> in nm, with a default of the short range cutoff radius. Note that radial distribution functions are not evaluated beyond the short range cutoff radius. The number of bins in each radial distribution function is given by <code>&lt;ngl&gt;</code>, with a default of 1000. This option is no longer supported. If radial distribution function are to be calculated, a rdi files needs to be available in which the contributions are specified as follows.</li>
</ul>
<table>
<thead>
<tr>
<th></th>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td>Card</td>
<td>Format</td>
<td>Description</td>
</tr>
<tr>
<td>I-1</td>
<td>i</td>
<td>Type, 1=solvent-solvent, 2=solvent-solute, 3-solute-solute</td>
</tr>
<tr>
<td>I-2</td>
<td>i</td>
<td>Number of the rdf for this contribution</td>
</tr>
<tr>
<td>I-3</td>
<td>i</td>
<td>First atom number</td>
</tr>
<tr>
<td>I-4</td>
<td>i</td>
<td>Second atom number</td>
</tr>
</tbody>
</table>
<h2 id="md-recording">Recording<a class="headerlink" href="#md-recording" title="Permanent link">&para;</a></h2>
<p>The following keywords control recording data to file. Record directives may be combined to a single directive.</p>
<pre><code>   record [rest &lt;integer nfrest&gt; [keep]]     \

          [coord &lt;integer nfcoor default 0&gt;] \

          [wcoor &lt;integer nfwcoo default 0&gt;] \

          [scoor &lt;integer nfscoo default 0&gt;] \

          [veloc &lt;integer nfvelo default 0&gt;] \

          [wvelo &lt;integer nfwvel default 0&gt;] \

          [svelo &lt;integer nfsvel default 0&gt;] \

          [force &lt;integer nfvelo default 0&gt;] \

          [wforc &lt;integer nfwvel default 0&gt;] \

          [sforc &lt;integer nfsvel default 0&gt;] \

          [(prop | prop_average) &lt;integer nfprop default 0&gt;]  \

          [free &lt;integer nffree default 1&gt;]  \

          [sync &lt;integer nfsync default 0&gt;]  \

          [times &lt;integer nftime default 0&gt;] \

          [acf] [cnv] [fet]

          [binary] [ascii] [ecce] [argos]
</code></pre>
<ul>
<li>Keyword rest specifies the frequency <code>&lt;nfrest&gt;</code> in molecular dynamics steps of rewriting the restart file, with extension rst. For multi-configuration thermodynamic integration simulations the frequency is in steps in . The default is not to record. The restart file is used to start or restart simulations. The keyword keep causes all restart files written to be kept on disk, rather than to be overwritten.</li>
</ul>
<!-- -->

<ul>
<li>Keyword coord specifies the frequency <code>&lt;nfcoor&gt;</code> in molecular dynamics steps of writing coordinates to the trajectory file. This directive redefines previous coord, wcoor and scoor directives. The default is not to record.</li>
</ul>
<!-- -->

<ul>
<li>Keyword wcoor specifies the frequency <code>&lt;nfcoor&gt;</code> in molecular dynamics steps of writing solvent coordinates to the trajectory file. This keyword takes precedent over coord. This directive redefines previous coord, wcoor and scoor directives. The default is not to record.</li>
</ul>
<!-- -->

<ul>
<li>Keyword scoor specifies the frequency <code>&lt;nfscoo&gt;</code> in molecular dynamics steps of writing solute coordinates to the trajectory file. This keyword takes precedent over coord. This directive redefines previous coord, wcoor and scoor directives. The default is not to record.</li>
</ul>
<!-- -->

<ul>
<li>Keyword veloc specifies the frequency <code>&lt;nfvelo&gt;</code> in molecular dynamics steps of writing velocities to the trajectory file. This directive redefines previous veloc, wvelo and svelo directives. The default is not to record.</li>
</ul>
<!-- -->

<ul>
<li>Keyword wvelo specifies the frequency <code>&lt;nfvelo&gt;</code> in molecular dynamics steps of writing solvent velocitiesto the trajectory file. This keyword takes precedent over veloc. This directive redefines previous veloc, wvelo and svelo directives. The default is not to record.</li>
</ul>
<!-- -->

<ul>
<li>Keyword svelo specifies the frequency <code>&lt;nfsvel&gt;</code> in molecular dynamics steps of writing solute velocities to the trajectory file. This keyword takes precedent over veloc. This directive redefines previous veloc, wvelo and svelo directives. The default is not to record.</li>
</ul>
<!-- -->

<ul>
<li>Keyword force specifies the frequency <code>&lt;nfvelo&gt;</code> in molecular dynamics steps of writing forces to the trajectory file. This directive redefines previous vforce, wforc and sforc directives. The default is not to record.</li>
</ul>
<!-- -->

<ul>
<li>Keyword wforc specifies the frequency <code>&lt;nfvelo&gt;</code> in molecular dynamics steps of writing solvent forcesto the trajectory file. This keyword takes precedent over force. This directive redefines previous vforce, wforc and sforc directives. The default is not to record.</li>
</ul>
<!-- -->

<ul>
<li>Keyword sforc specifies the frequency <code>&lt;nfsvel&gt;</code> in molecular dynamics steps of writing solute forces to the trajectory file. This keyword takes precedent over force. This directive redefines previous vforce, wforc and sforc directives. The default is not to record.</li>
</ul>
<!-- -->

<ul>
<li>Keyword prop specifies the frequency <code>&lt;nfprop&gt;</code> in molecular dynamics steps of writing information to the property file, with extension prp. The default is not to record.</li>
</ul>
<!-- -->

<ul>
<li>Keyword prop_average specifies the frequency <code>&lt;nfprop&gt;</code> in molecular dynamics steps of writing average information to the property file, with extension prp. The default is not to record.</li>
</ul>
<!-- -->

<ul>
<li>Keyword free specifies the frequency <code>&lt;nffree&gt;</code> in multi-configuration thermodynamic integration steps to record data to the free energy data file, with extension gib. The default is 1, i.e. to record at every . This option is obsolete. All data are required to do the final analysis.</li>
</ul>
<!-- -->

<ul>
<li>Keyword sync specifies the frequency <code>&lt;nfsync&gt;</code> in molecular dynamics steps of writing information to the synchronization file, with extension syn. The default is not to record. The information written is the simulation time, the wall clock time of the previous MD step, the wall clock time of the previous force evaluation, the total synchronization time, the largest synchronization time and the node on which the largest synchronization time was found. The recording of synchronization times is part of the load balancing algorithm. Since load balancing is only performed when pair-lists are updated, the frequency <code>&lt;nfsync&gt;</code> is correlated with the frequency of pair-list updates <code>&lt;nfpair&gt;</code>. This directive is only needed for analysis of the load balancing performance. For normal use this directive is not used.</li>
</ul>
<!-- -->

<ul>
<li>Keyword times specifies the frequency <code>&lt;nfsync&gt;</code> in molecular dynamics steps of writing information to the timings file, with extension tim. The default is not to record. The information written is wall clock time used by each of the processors for the different components in the force evaluation. This directive is only needed for analysis of the wall clock time distribution. For normal use this directive is not used.</li>
</ul>
<!-- -->

<ul>
<li>Keywords acf, cnv and fet are obsolete.</li>
</ul>
<!-- -->

<ul>
<li>Keywords binary, ascii, ecce and argos are obsolete.</li>
</ul>
<h2 id="md-program-control-options">Program control options<a class="headerlink" href="#md-program-control-options" title="Permanent link">&para;</a></h2>
<pre><code>   load [reset] 
        ( none | 
          size [&lt;real factld&gt;] |
          sizez [&lt;real factld&gt;] | pairs | 
         (pairs [&lt;integer ldpair&gt;] size [&lt;real factld&gt;]) )
        [last]
        [minimum]
        [average]
        [combination]
        [iotime]
        [experimental]
</code></pre>
<p>Determines the type of dynamic load balancing performed, where the default is none. Load balancing option size is resizing cells on a node, and pairs redistributes the cell-cell interactions over nodes. Keyword reset will reset the load balancing read from the restart file. The level of cell resizing can be influenced with <em>factld</em>. The cells on the busiest node are resized with a factor</p>
<p><span class="arithmatex">\(\left( 1 - factld * { {T_{sync} \over n_p} - t^{min}_{sync} \over t_{wall}} \right)^{1\over 3}\)</span></p>
<p>Where <span class="arithmatex">\(T_{sync}\)</span> is the accumulated synchronization time of all nodes, <span class="arithmatex">\(n_p\)</span> is the total number of nodes, <span class="arithmatex">\(t^{min}_{sync}\)</span> is the synchronization time of the busiest node, and <span class="arithmatex">\(t_{wall}\)</span> is the wall clock time of the molecular dynamics step. For the combined load balancing, ldpair is the number of successive pair redistribution load balancing steps in which the accumulated synchronization time increases, before a resizing load balancing step will be attempted. Load balancing is only performed in molecular dynamics steps in which the pair-list is updated. The default load balancing is equivalent to specifying</p>
<pre><code>   load pairs 10 size 0.75
</code></pre>
<p>Keyword last specifies that the load balancing is based on the synchronization times of the last step. This is the default. Keyword average specifies that the load balancing is based on the average synchronization times since the last load balancing step. Keyword minimum specifies that the load balancing is based on the minimum synchronization times since the last load balancing step. Keywords combination, iotime and experimental are experimental load balancing options that should not be used in production runs.</p>
<pre><code>   (pack | nopack)
</code></pre>
<p>Specifies if data are communicated in packed or unpacked form. The default is pack.</p>
<pre><code>   procs &lt;integer npx&gt; &lt;integer npy&gt; &lt;integer npz&gt;
</code></pre>
<p>Specifies the distribution of the available processors over the three Cartesian dimensions. The default distribution is chosen such that, <code>&lt;npx&gt;</code>*<code>&lt;npy&gt;</code>*<code>&lt;npz&gt;</code>=<code>&lt;np&gt;</code> and <code>&lt;npx&gt;</code> <code>&lt;=</code> <code>&lt;npy&gt;</code> <code>&lt;= &lt;npz&gt;</code>, where <code>&lt;npx&gt;</code>, <code>&lt;npy&gt;</code> and <code>&lt;npz&gt;</code> are the processors in the x, y and z dimension respectively, and <code>&lt;np&gt;</code> is the number of processors allocated for the calculation. Where more than one combination of <code>&lt;npx&gt;</code>, <code>&lt;npy&gt;</code> and <code>&lt;npz&gt;</code> are possible, the combination is chosen with the minimum value of <code>&lt;npx&gt;+&lt;npy&gt;+&lt;npz&gt;</code>. To change the default setting the following optional input option is provided.</p>
<pre><code>   cells &lt;integer nbx&gt; &lt;integer nby&gt; &lt;integer nbz&gt;
</code></pre>
<p>Specifies the distribution of cells, where <code>&lt;nbx&gt;</code>, <code>&lt;nby&gt;</code> and <code>&lt;nbz&gt;</code> are the number of cells in x, y and z direction, respectively. The molecular system is decomposed into cells that form the smallest unit for communication of atomic data between nodes. The size of the cells is per default set to the short-range cutoff radius. If long-range cutoff radii are used the cell size is set to half the long-range cutoff radius if it is larger than the short-range cutoff. If the number of cells in a dimension is less than the number of processors in that dimension, the number of cells is set to the number of processors.</p>
<pre><code>   extra &lt;integer madbox&gt;
</code></pre>
<p>Sets the number of additional cells for which memory is allocated. In rare events the amount of memory set aside per node is insufficient to hold all atomic coordinates assigned to that node. This leads to execution which aborts with the message that mwm or msa is too small. Jobs may be restarted with additional space allocated by where <code>&lt;madbox&gt;</code> is the number of additional cells that are allocated on each node. The default for <code>&lt;madbox&gt;</code> is 6. In some cases <code>&lt;madbox&gt;</code> can be reduced to 4 if memory usage is a concern. Values of 2 or less will almost certainly result in memory shortage.</p>
<pre><code>   mwm &lt;integer mwmreq&gt;
</code></pre>
<p>Sets the maximum number of solvent molecules <code>&lt;mwmreq&gt;</code> per node, allowing increased memory to be allocated for solvent molecules. This option can be used if execution aborted because mwm was too small.</p>
<pre><code>   msa &lt;integer msareq&gt;
</code></pre>
<p>Sets the maximum number of solute atoms <code>&lt;msareq&gt;</code> per node, allowing increased memory to be allocated for solute atoms. This option can be used if execution aborted because msa was too small.</p>
<pre><code>   mcells &lt;integer mbbreq&gt;
</code></pre>
<p>Sets the maximum number of cell pairs <code>&lt;mbbreq&gt;</code> per node, allowing increased memory to be allocated for the cell pair lists. This option can be used if execution aborted because mbbl was too small.</p>
<pre><code>   boxmin &lt;real rbox&gt;
</code></pre>
<p>Sets the minimum size of a cell. This directive is obsolete. The use of mcells is preferred.</p>
<pre><code>   segmentsize &lt;real rsgm&gt;
</code></pre>
<p>Sets the maximum size of a segment. This value is used to determine which segments at the boundary of the cutoff radius should be considered in the generation of the pairlists. This value is also determined by the prepare module and written to the restart file. Use of this directive is not needed for simulations that use the current prepare module to generate the restart file.</p>
<pre><code>   memory &lt;integer memlim&gt;
</code></pre>
<p>Sets a limit <code>&lt;memlim&gt;</code> in kB on the allocated amount of memory used by the molecular dynamics module. Per default all available memory is allocated. Use of this command is required for QM/MM simulations only.</p>
<pre><code>   expert
</code></pre>
<p>Enables the use of certain combinations of features that are considered unsafe. This directive should not be used for production runs.</p>
<pre><code>   develop &lt;integer idevel&gt;
</code></pre>
<p>Enables the use of certain development options specified by the integer idevel. This option is for development purposes only, and should not be used for production runs.</p>
<pre><code>   control &lt;integer icntrl&gt;
</code></pre>
<p>Enables the use of certain development options specified by the integer icntrl. This option is for development purposes only, and should not be used for production runs.</p>
<pre><code>   numerical
</code></pre>
<p>Writes out analytical and finite difference forces for test purposes.</p>
<pre><code>   server &lt;string servername&gt; &lt;integer serverport&gt;
</code></pre>
<p>Allows monitoring over a socket connection to the specified port on the named server of basic data as a simulation is running.</p>
<p>For development purposes debug information can be written to the debug file with extension dbg with</p>
<pre><code>   debug &lt;integer idebug&gt;
</code></pre>
<p>where <em>idebug</em> specifies the type of debug information being written.</p>
<p>For testing purposes test information can be written to the test file with extension tst with</p>
<pre><code>   test &lt;integer itest&gt;
</code></pre>
<p>where <em>itest</em> specifies the number of steps test information is written.</p>
<p>On some platforms prefetching of data can improve the efficiency. This feature can be turned on using</p>
<pre><code>   prefetch [&lt;integer nbget&gt;]
</code></pre>
<p>where <em>nbget</em> is the number of outstanding communication operations.</p>
<p>Application of periodic boundary conditions for the evaluation of forces can be controlled with</p>
<pre><code>   pbc ( atom | residue | molecule )
</code></pre>
<p>This option rarely needs to be used.</p>
<p>Autocorrelation functions for error analysis are controlled using</p>
<pre><code>   auto [ fit &lt;integer iapprx&gt; | weight &lt;real weight&gt; ]
</code></pre>
<p>This option is disabled in the current release.</p>
<p>Membrane system equilibration can be made more efficient using</p>
<pre><code>   membrane [ rotations ]
</code></pre>
<p>Constraining the center of mass of solute molecules in the xy plane is accomplished using</p>
<pre><code>   scmxy [&lt;integer icmopt default 1&gt;]
</code></pre>
<p>where <em>icmopt</em> determines if the constraint is mass weighted (2).</p>
<p>Radius of gyration calculations are enabled using</p>
<pre><code>   radius_gyration
</code></pre>
<p>Calculations of diffusion coefficients is enabled using</p>
<pre><code>   diffusion
</code></pre>
<p>This option is disabled in the current release.</p>
<pre><code>   comlim ( on | off )
</code></pre>
<p>is disabled</p>
<p>To limit the size of recoding files, new files are opened every <em>nfnewf</em> md steps using</p>
<pre><code>   batch &lt;integer nfnewf&gt;
</code></pre></section><section class="print-page" id="analysis" heading-number="17.3"><h1 id="analysis-analysis">Analysis<a class="headerlink" href="#analysis-analysis" title="Permanent link">&para;</a></h1>
<p>The analysis module is used to analyze molecular trajectories generated by the NWChem molecular dynamics module, or partial charges generated by the NWChem electrostatic potential fit module. This module should not de run in parallel mode.</p>
<p>Directives for the analysis module are read from an input deck,</p>
<pre><code>analysis
 ...
end
</code></pre>
<p>The analysis is performed as post-analysis of trajectory files through using the task directive</p>
<pre><code>task analysis
</code></pre>
<p>or</p>
<pre><code>task analyze
</code></pre>
<h2 id="analysis-system-specification">System specification<a class="headerlink" href="#analysis-system-specification" title="Permanent link">&para;</a></h2>
<pre><code>system &lt;string systemid&gt;_&lt;string calcid&gt;
</code></pre>
<p>where the strings systemid and calcid are user defined names for the chemical system and the type of calculation to ber performed, respectively. These names are used to derive the filenames used for the calculation. The topoly file used will be systemid.top, while all other files are named systemid_calcid.ext.</p>
<h2 id="analysis-reference-coordinates">Reference coordinates<a class="headerlink" href="#analysis-reference-coordinates" title="Permanent link">&para;</a></h2>
<p>Most analyses require a set of reference coordinates. These coordinates are read from a NWChem restart file by the directive,</p>
<pre><code>reference &lt;string filename&gt;
</code></pre>
<p>where filename is the name of an existing restart file. This input directive is required.</p>
<h2 id="analysis-file-specification">File specification<a class="headerlink" href="#analysis-file-specification" title="Permanent link">&para;</a></h2>
<p>The trajectory file(s) to be analyzed are specified with</p>
<pre><code>file  &lt;string filename&gt;  [ &lt;integer firstfile&gt;   &lt;integer lastfile&gt; ]
</code></pre>
<p>where filename is an existing trj trajectory file. If firstfile and lastfile are specified, the specified filename needs to have a <code>?</code> wildcard character that will be substituted by the 3-character integer number from firstfile to lastfile, and the analysis will be performed on the series of files. For example,</p>
<pre><code>file tr_md?.trj 3 6
</code></pre>
<p>will instruct the analysis to be performed on files tr_md003.trj, tr_md004.trj, tr_md005.trj and tr_md006.trj.</p>
<p>From the specified files the subset of frames to be analyzed is specified by</p>
<pre><code>frames [ &lt;integer firstframe default 1&gt; ]  &lt;integer lastframe&gt;  \

       [ &lt;integer frequency default 1&gt; ]
</code></pre>
<p>For example, to analyze the first 100 frames from the specified trajectory files, use</p>
<pre><code>frames 100
</code></pre>
<p>To analyze every 10-th frame between frames 200 and 400 recorded on the specified trajectory files, use</p>
<pre><code>frames 200 400 10
</code></pre>
<p>A time offset can be specified with</p>
<pre><code>time &lt;real timoff&gt;
</code></pre>
<p>Solute coordinates of the reference set and ech subsequent frame read from a trajectory file are translated to have the center of geometry of the specified solute molecule at the center of the simulation box. After this translation all molecules are folded back into the box according to the periodic boundary conditions. The directive for this operation is</p>
<pre><code>center  &lt;integer imol&gt;  [ &lt;integer jmol default imol&gt; ]
</code></pre>
<p>Coordinates of each frame read from a trajectory file can be rotated using</p>
<pre><code>rotate ( off | x | y | z ) &lt;real angle units degrees&gt;
</code></pre>
<p>If center was defined, rotation takes place after the system has been centered. The rotate directives only apply to frames read from the trajectory files, and not to the reference coordinates. Up to 100 rotate directives can be specified, which will be carried out in the order in which they appear in the input deck. rotate off cancels all previously defined rotate directives.</p>
<p>To perform a hydrogen bond analysis:</p>
<pre><code>hbond [distance [[&lt;real rhbmin default 0.0&gt;] &lt;real rhbmin&gt;]] \
      [angle [&lt;real hbdmin&gt; [ &lt;real hbdmax default pi&gt;]]] \
      [solvent [&lt;integer numwhb&gt;]]
</code></pre>
<h2 id="analysis-selection">Selection<a class="headerlink" href="#analysis-selection" title="Permanent link">&para;</a></h2>
<p>Analyses can be applied to a selection of solute atoms and solvent molecules. The selection is determined by</p>
<pre><code>select ( [ super ] [ { &lt;string atomlist&gt; } ] |
   solvent &lt;real range&gt; | save &lt;string filename&gt; | read &lt;string filename&gt; )
</code></pre>
<p>where {atomlist} is the set of atom names selected from the specified residues. By default all solute atoms are selected. When keyword <code>super</code> is specified the selecion applies to the superimposition option.</p>
<p>The selected atoms are specified by the string atomlist which takes the form</p>
<pre><code>[{isgm [ - jsgm ] [,]} [:] [{aname[,]}]
</code></pre>
<p>where isgm and jsgm are the first and last residue numbers, and aname is an atom name. In the atomname a question mark may be used as a wildcard character.</p>
<p>For example, all protein backbone atoms are selected by</p>
<pre><code>select _N,_CA,_C
</code></pre>
<p>To select the backbone atoms in residues 20 to 80 and 90 to 100 only, use</p>
<pre><code>select 20-80,90-100:_N,_CA,_C
</code></pre>
<p>This selection is reset to apply to all atoms after each file directive.</p>
<p>Solvent molecules within range nm from any selected solute atom are selected by</p>
<pre><code>select solvent &lt;real range&gt;
</code></pre>
<p>After solvent selection, the solute atom selection is reset to being all selected.</p>
<p>The current selection can be saved to, or read from a file using the save and read keywords, respectively.</p>
<p>Some analysis are performed on groups of atoms. These groups of atoms are defined by</p>
<pre><code>define &lt;integer igroup&gt; [&lt;real rsel&gt;] [solvent] { &lt;string atomlist&gt; }
</code></pre>
<p>The string atom in this definitions again takes the form</p>
<pre><code>[{isgm [ - jsgm ] [,]} [:] [{aname[,]}]
</code></pre>
<p>where isgm and jsgm are the first and last residue numbers, and aname is an atom name. In the atomname a question mark may be used as a wildcard character.</p>
<p>Multiple define directive can be used to define a single set of atoms.</p>
<h2 id="analysis-coordinate-analysis">Coordinate analysis<a class="headerlink" href="#analysis-coordinate-analysis" title="Permanent link">&para;</a></h2>
<p>To analyze the root mean square deviation from the specified reference coordinates:</p>
<pre><code>rmsd
</code></pre>
<p>To analyze protein - and backbone hydrogen bonding:</p>
<pre><code>ramachandran
</code></pre>
<p>To define a distance:</p>
<pre><code>distance &lt;integer ibond&gt; &lt;string atomi&gt; &lt;string atomj&gt;
</code></pre>
<p>To define an angle:</p>
<pre><code>angle &lt;integer iangle&gt; &lt;string atomi&gt; &lt;string atomj&gt; &lt;string atomk&gt;
</code></pre>
<p>To define a torsion:</p>
<pre><code>torsion &lt;integer itorsion&gt; &lt;string atomi&gt; &lt;string atomj&gt; \
                      &lt;string atomk&gt; &lt;string atoml&gt;
</code></pre>
<p>To define a vector:</p>
<pre><code>vector &lt;integer ivector&gt; &lt;string atomi&gt; &lt;string atomj&gt;
</code></pre>
<p>The atom string in these definitions takes the form</p>
<pre><code>&lt;integer segment&gt;:&lt;string atomname&gt; | w&lt;integer molecule&gt;:&lt;string atomname&gt;
</code></pre>
<p>for solute and solvent atom specification, respectively.</p>
<p>To define charge distribution in z-direction:</p>
<pre><code>charge_distribution &lt;integer bins&gt;
</code></pre>
<p>Analyses on atoms in a predefined group are specified by</p>
<pre><code>group [&lt;integer igroup&gt; [periodic &lt;integer ipbc&gt;] \
      ( local [&lt;real rsel default 0.0&gt;] [&lt;real rval default rsel&gt;]
        &lt;string function&gt; )
</code></pre>
<p>where igroup specifies the group of atoms defined with a define directive. Keyword <code>periodic</code> can be used to specify the periodicity, ipbc=1 for periodicity in z, ipbc=2 for periodicity in x and y, and ipbc=3 for periodicity in x, y and z. Currently the only option is local which prints all selected solute atom with a distance between rsel and rval from the atoms defined in igroup. The actual analysis is done by the scan deirective. A formatted report is printed from group analyses using</p>
<pre><code>report &lt;string filename&gt; local
</code></pre>
<p>Analyses on pairs of atoms in predefined groups are specified by</p>
<pre><code>groups [&lt;integer igroup&gt; [&lt;integer jgroup&gt;]] [periodic [&lt;integer ipbc default 3&gt;]] \ 
       &lt;string function&gt; [&lt;real value1&gt; [&lt;real value2&gt;]] [&lt;string filename&gt;]
</code></pre>
<p>where <em>igroup</em> and <em>jgroup</em> are groups of atoms defined with a define directive. Keyword <code>periodic</code> specifies that periodic boundary conditions need to be applied in <em>ipbc</em> dimensions. The type of analysis is define by <em>function</em>, <em>value1</em> and <em>value2</em>. If <em>filename</em> is specified, the analysis is applied to the reference coordinates and written to the specified file. If no filename is given, the analysis is applied to the specified trajectory and performed as part of the scan directive. Implemented analyses defined by <code>&lt;string function&gt; [&lt;real value1&gt; [&lt;real value2&gt;]]</code> include</p>
<ul>
<li>distance to calculate the distance between the centers of geometry of the two specified groups of atoms, and</li>
<li>distances to calculate all atomic distances between atoms in the specified groups that lie between <em>value1</em> and <em>value2</em>.</li>
</ul>
<p>Coordinate histograms are specified by</p>
<pre><code>histogram &lt;integer idef&gt; [&lt;integer length&gt;] zcoordinate &lt;string filename&gt;
</code></pre>
<p>where <em>idef</em> is the atom group definition number, <em>length</em> is the size of the histogram, zcoordinate is the currently only histogram option, and <em>filename</em> is the filname to which the histogram is written.</p>
<p>Order parameters are evalated using</p>
<pre><code>order &lt;integer isel&gt; &lt;integer jsel&gt; &lt;string atomi&gt; &lt;string atomj&gt;
</code></pre>
<p>This is an experimental feature.</p>
<p>To write the average coordinates of a trajectory</p>
<pre><code>average [super] &lt;string filename&gt;
</code></pre>
<p>To perform the coordinate analysis:</p>
<pre><code>scan [ super ] &lt;string filename&gt;
</code></pre>
<p>which will create, depending on the specified analysis options files filename.rms and filename.ana. After the scan directive previously defined coordinate analysis options are all reset. Optional keyword <code>super</code> specifies that frames read from the trajectory file(s) are superimposed to the reference structure before the analysis is performed.</p>
<h2 id="analysis-essential-dynamics-analysis">Essential dynamics analysis<a class="headerlink" href="#analysis-essential-dynamics-analysis" title="Permanent link">&para;</a></h2>
<p>Essential dynamics analysis is performed by</p>
<pre><code>essential
</code></pre>
<p>This can be followed by one or more</p>
<pre><code>project &lt;integer vector&gt; &lt;string filename&gt;
</code></pre>
<p>to project the trajectory onto the specified vector. This will create files filename with extensions frm or trj, val, vec, _min.pdb and _max.pdb, with the projected trajectory, the projection value, the eigenvector, and the minimum and maximum projection structure.</p>
<p>For example, an essential dynamics analysis with projection onto the first vector generating files firstvec.{trj, val, vec, _min.pdb, _max.pdb} is generated by</p>
<pre><code>essential
project 1 firstvec
</code></pre>
<h2 id="analysis-trajectory-format-conversion">Trajectory format conversion<a class="headerlink" href="#analysis-trajectory-format-conversion" title="Permanent link">&para;</a></h2>
<p>To write a single frame in PDB or XYZ format, use</p>
<pre><code>write [&lt;integer number default 1&gt;] [super] [solute] &lt;string filename&gt;
</code></pre>
<p>To copy the selected frames from the specified trejctory file(s), onto a new file, use</p>
<pre><code>copy  [solute] [rotate &lt;real tangle&gt;] &lt;string filename&gt;
</code></pre>
<p>To superimpose the selected atoms for each specified frame to the reference coordinates before copying onto a new file, use</p>
<pre><code>super [solute] [rotate &lt;real tangle&gt;] &lt;string filename&gt;
</code></pre>
<p>The rotate directive specifies that the structure will make a full ratation every tangle ps. This directive only has effect when writing povray files.</p>
<p>The format of the new file is determined from the extension, which can be one of</p>
<ul>
<li><code>amb</code>    AMBER formatted trajectory file (obsolete)</li>
<li><code>arc</code>   DISCOVER archive file</li>
<li><code>bam</code>   AMBER unformatted trajectory file</li>
<li><code>crd</code>   AMBER formatted trajectory file</li>
<li><code>dcd</code>   CHARMM formatted trajectory file</li>
<li><code>esp</code>   gOpenMol formatted electrostatic potential files</li>
<li><code>frm</code>   ecce frames file (obsolete)</li>
<li><code>pov</code>   povray input files</li>
<li><code>trj</code>   NWChem trajectory file</li>
<li><code>xyz</code>   NWChem trajectory in xyz format</li>
</ul>
<p>If no extension is specified, a trj formatted file will be written.</p>
<p>A special tag can be added to frm and pov formatted files using</p>
<pre><code>label &lt;integer itag&gt; &lt;string tag&gt;  [ &lt;real rval default 1.0&gt; ] \\
      [ &lt;integer iatag&gt; [ &lt;integer jatag default iatag&gt; ] [ &lt;real rtag default 0.0&gt; ] ]
      [ &lt;string anam&gt; ]
</code></pre>
<p>where tag number <em>itag</em> is set to the string <em>tag</em> for all atoms anam within a distance <em>rtag</em> from segments <em>iatag</em> through <em>jatag</em>. A question mark can be used in anam as a wild card character.</p>
<p>Atom rendering is specified using</p>
<pre><code>render ( cpk | stick )  [ &lt;real rval default 1.0&gt; ] \\
       [ &lt;integer iatag&gt; [ &lt;integer jatag default iatag&gt; ] [ &lt;real rtag default 0.0&gt; ] ]
       [ &lt;string anam&gt; ]
</code></pre>
<p>for all atoms anam within a distance <em>rtag</em> from segments <em>iatag</em> through <em>jatag</em>, and a scaling factor of <em>rval</em>. A question mark can be used in anam as a wild card character.</p>
<p>Atom color is specified using</p>
<pre><code>color ( &lt;string color&gt; | atom ) \\

      [ &lt;integer iatag&gt; [ &lt;integer jatag default iatag&gt; ] [ &lt;real rtag default 0.0&gt; ] ]

      [ &lt;string anam&gt; ]
</code></pre>
<p>for all atoms anam within a distance <em>rtag</em> from segments <em>iatag</em> through <em>jatag</em>. A question mark can be used in anam as a wildcard character.</p>
<p>For example, to display all carbon atoms in segments 34 through 45 in green and rendered cpk in povray files can be specified with</p>
<pre><code>render cpk 34 45 _C??
color green 34 45 _C??
</code></pre>
<p>Coordinates written to a pov file can be scaled using</p>
<pre><code>scale &lt;real factor&gt;
</code></pre>
<p>A zero or negative scaling factor will scale the coordinates to lie within [-1,1] in all dimensions.</p>
<p>The cpk rendering in povray files can be scaled by</p>
<pre><code>cpk &lt;real factor default 1.0&gt;
</code></pre>
<p>The stick rendering in povray files can be scaled by</p>
<pre><code>stick &lt;real factor default 1.0&gt;
</code></pre>
<p>The initial sequence number of esp related files is defined by</p>
<pre><code>index &lt;integer index default 1&gt;
</code></pre>
<p>A sequence of trajectory files with unequal lengths can be converted to files with all <em>nclean</em> frames using</p>
<pre><code>clean &lt;integer nclean&gt;
</code></pre>
<h2 id="analysis-electrostatic-potentials">Electrostatic potentials<a class="headerlink" href="#analysis-electrostatic-potentials" title="Permanent link">&para;</a></h2>
<p>A file in plt format of the electrostatic potential resulting from partial charges generated by the ESP module is generated by the command</p>
<pre><code>esp  [ &lt;integer spacing default 10&gt; ] \
     [ &lt;real rcut default 1.0&gt; ] [periodic [&lt;integer iper default 3&gt;]] \
     [ &lt;string xfile&gt; [ &lt;string pltfile&gt; ] ]
</code></pre>
<p>The input coordinates are taken from the xyzq file that can be generated from a rst by the prepare module. Parameter spacing specifies the number of gridpoints per nm, rcut specifies extent of the charge grid beyond the molecule. Periodic boundaries will be used if periodic is specified. If iper is set to 2, periodic boundary conditions are applied in x and y dimensions only. If periodic is specified, a negative value of rcut will extend the grid in the periodic dimensions by abs(rcut), otherwise this value will be ignored in the periodic dimensions. The resulting plt formatted file pltfile can be viewed with the gOpenMol program. The resulting electrostatic potential grid is in units of <span class="arithmatex">\(kJ mol^{-1} e^{-1}\)</span>. If no files are specified, only the parameters are set. This analysis applies to solute(s) only.  </p>
<p>The electrostatic potential at specific point are evaluated using  </p>
<pre><code>esp_points [&lt;string filpin&gt; [&lt;string filhol&gt; [&lt;string filpou&gt; [&lt;string filavg&gt;]]]]
</code></pre></section><section class="print-page" id="format_of_md_files" heading-number="17.4"><h1 id="format_of_md_files-format-of-md-files">Format of MD files<a class="headerlink" href="#format_of_md_files-format-of-md-files" title="Permanent link">&para;</a></h1>
<h2 id="format_of_md_files-format-of-fragment-file">Format of fragment file<a class="headerlink" href="#format_of_md_files-format-of-fragment-file" title="Permanent link">&para;</a></h2>
<table>
<thead>
<tr>
<th style="text-align: left;">Card</th>
<th style="text-align: left;">Format</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">I-1-1</td>
<td style="text-align: left;">a1</td>
<td style="text-align: left;">$ or # comments to describe fragment</td>
</tr>
<tr>
<td style="text-align: left;">I-2-1</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">number of atoms in the fragment</td>
</tr>
<tr>
<td style="text-align: left;">I-2-2</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">number of parameter sets</td>
</tr>
<tr>
<td style="text-align: left;">I-2-3</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">default parameter set</td>
</tr>
<tr>
<td style="text-align: left;">I-2-4</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">number of z-matrix definition</td>
</tr>
<tr>
<td style="text-align: left;"><strong>For each parameter</strong></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;"><strong>set one card I-3</strong></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">I-3-1</td>
<td style="text-align: left;">a</td>
<td style="text-align: left;">residue name for parameter set</td>
</tr>
<tr>
<td style="text-align: left;"><strong>For each atom one deck II</strong></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">II-1-1</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">atom sequence number</td>
</tr>
<tr>
<td style="text-align: left;">II-1-2</td>
<td style="text-align: left;">a6</td>
<td style="text-align: left;">atom name</td>
</tr>
<tr>
<td style="text-align: left;">II-1-3</td>
<td style="text-align: left;">a5</td>
<td style="text-align: left;">atom type</td>
</tr>
<tr>
<td style="text-align: left;">II-1-4</td>
<td style="text-align: left;">a1</td>
<td style="text-align: left;">dynamics type</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">blank : normal</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">D : dummy atom</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">S: solute interactions only</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Q : quantum atom</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">other : intramolecular solute interactions only</td>
</tr>
<tr>
<td style="text-align: left;">II-1-5</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">link number</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">0: no link</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">1: first atom in chain</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">2: second atom in chain</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">3 and up: other links</td>
</tr>
<tr>
<td style="text-align: left;">II-1-6</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">environment type</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">0: no special identifier</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">1: planar, using improper torsion</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">2: tetrahedral, using improper torsion</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">3: tetrahedral, using improper torsion</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">4: atom in aromatic ring</td>
</tr>
<tr>
<td style="text-align: left;">II-1-7</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">II-1-8</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">charge group</td>
</tr>
<tr>
<td style="text-align: left;">II-1-9</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">polarization group</td>
</tr>
<tr>
<td style="text-align: left;">II-1-10</td>
<td style="text-align: left;">f12.6</td>
<td style="text-align: left;">atomic partial charge</td>
</tr>
<tr>
<td style="text-align: left;">II-1-11</td>
<td style="text-align: left;">f12.6</td>
<td style="text-align: left;">atomic polarizability</td>
</tr>
<tr>
<td style="text-align: left;"><strong>For each additional</strong></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;"><strong>parameter set on card II-2</strong></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">II-2-1</td>
<td style="text-align: left;">11x,a6</td>
<td style="text-align: left;">atom type</td>
</tr>
<tr>
<td style="text-align: left;">II-2-2</td>
<td style="text-align: left;">a1</td>
<td style="text-align: left;">dynamics type</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">blank : normal</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">D : dummy atom</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">S : solute interactions only</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Q : quantum atom</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">other : intramolecular solute interactions only</td>
</tr>
<tr>
<td style="text-align: left;">II-2-7</td>
<td style="text-align: left;">25x,f12.6</td>
<td style="text-align: left;">atomic partial charge</td>
</tr>
<tr>
<td style="text-align: left;">II-2-8</td>
<td style="text-align: left;">f12.6</td>
<td style="text-align: left;">atomic polarizability</td>
</tr>
<tr>
<td style="text-align: left;"><strong>Any number of cards in deck III to</strong></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;"><strong>specify complete connectivity</strong></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">III-1-1</td>
<td style="text-align: left;">16i5</td>
<td style="text-align: left;">connectivity, duplication allowed</td>
</tr>
<tr>
<td style="text-align: left;"><strong>One blank card to signal</strong></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;"><strong>the end of the connectivity list</strong></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;"><strong>For each z-matrix definition one card IV</strong></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">IV-1-1</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">atom i</td>
</tr>
<tr>
<td style="text-align: left;">IV-1-2</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">atom j</td>
</tr>
<tr>
<td style="text-align: left;">IV-1-3</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">atom k</td>
</tr>
<tr>
<td style="text-align: left;">IV-1-4</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">atom l</td>
</tr>
<tr>
<td style="text-align: left;">IV-1-5</td>
<td style="text-align: left;">f12.6</td>
<td style="text-align: left;">bond length i-j</td>
</tr>
<tr>
<td style="text-align: left;">IV-1-6</td>
<td style="text-align: left;">f12.6</td>
<td style="text-align: left;">angle i-j-k</td>
</tr>
<tr>
<td style="text-align: left;">IV-1-7</td>
<td style="text-align: left;">f12.6</td>
<td style="text-align: left;">torson i-j-k-l</td>
</tr>
</tbody>
</table>
<h2 id="format_of_md_files-format-of-segment-file-1-of-7">Format of segment file (1 of 7)<a class="headerlink" href="#format_of_md_files-format-of-segment-file-1-of-7" title="Permanent link">&para;</a></h2>
<table>
<thead>
<tr>
<th style="text-align: left;">Card</th>
<th style="text-align: left;">Format</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">I-0-1</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"># lines at top are comments</td>
</tr>
<tr>
<td style="text-align: left;">I-1-1</td>
<td style="text-align: left;">a1</td>
<td style="text-align: left;">$ to identify the start of a segment</td>
</tr>
<tr>
<td style="text-align: left;">I-1-2</td>
<td style="text-align: left;">a10</td>
<td style="text-align: left;">name of the segment, the tenth character</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">N: identifies beginning of a chain</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">C: identifies end of a chain</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">blank: identifies chain fragment</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">M: identifies an integral molecule</td>
</tr>
<tr>
<td style="text-align: left;">I-2-1</td>
<td style="text-align: left;">f12.6</td>
<td style="text-align: left;">version number</td>
</tr>
<tr>
<td style="text-align: left;">I-3-1</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">number of atoms in the segment</td>
</tr>
<tr>
<td style="text-align: left;">I-3-2</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">number of bonds in the segment</td>
</tr>
<tr>
<td style="text-align: left;">I-3-3</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">number of angles in the segment</td>
</tr>
<tr>
<td style="text-align: left;">I-3-4</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">number of proper dihedrals in the segment</td>
</tr>
<tr>
<td style="text-align: left;">I-3-5</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">number of improper dihedrals in the segment</td>
</tr>
<tr>
<td style="text-align: left;">I-3-6</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">number of z-matrix definitions</td>
</tr>
<tr>
<td style="text-align: left;">I-3-7</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">number of parameter sets</td>
</tr>
<tr>
<td style="text-align: left;">1-3-8</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">default parameter set</td>
</tr>
<tr>
<td style="text-align: left;">For each parameter set one card I-4</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">I-4-1</td>
<td style="text-align: left;">f12.6</td>
<td style="text-align: left;">dipole correction energy</td>
</tr>
</tbody>
</table>
<h2 id="format_of_md_files-format-of-segment-file-2-of-7">Format of segment file (2 of 7)<a class="headerlink" href="#format_of_md_files-format-of-segment-file-2-of-7" title="Permanent link">&para;</a></h2>
<table>
<thead>
<tr>
<th style="text-align: left;">Card</th>
<th style="text-align: left;">Format</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">For each atom one deck II</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">II-1-1</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">atom sequence number</td>
</tr>
<tr>
<td style="text-align: left;">II-1-2</td>
<td style="text-align: left;">a6</td>
<td style="text-align: left;">atom name</td>
</tr>
<tr>
<td style="text-align: left;">II-1-3</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">link number</td>
</tr>
<tr>
<td style="text-align: left;">II-1-4</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">environment type</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">0: no special identifier</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">1: planar, using improper torsion</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">2: tetrahedral, using improper torsion</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">3: tetrahedral, using improper torsion</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">4: atom in aromatic ring</td>
</tr>
<tr>
<td style="text-align: left;">II-1-5</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">II-1-6</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">charge group</td>
</tr>
<tr>
<td style="text-align: left;">II-1-7</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">polarization group</td>
</tr>
<tr>
<td style="text-align: left;">For each parameter set one card II-2</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">II-2-1</td>
<td style="text-align: left;">5x,a5</td>
<td style="text-align: left;">atom type</td>
</tr>
<tr>
<td style="text-align: left;">II-2-2</td>
<td style="text-align: left;">a1</td>
<td style="text-align: left;">dynamics type</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">blank : normal</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">D : dummy atom</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">S : solute interactions only</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Q : quantum atom</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">other : intramolecular solute interactions only</td>
</tr>
<tr>
<td style="text-align: left;">II-2-3</td>
<td style="text-align: left;">f12.6</td>
<td style="text-align: left;">atomic partial charge in e</td>
</tr>
<tr>
<td style="text-align: left;">II-2-4</td>
<td style="text-align: left;">f12.6</td>
<td style="text-align: left;">atomic polarizability (4 &pi; &epsilon;<sub>o</sub> in nm<sup>3</sup>)</td>
</tr>
</tbody>
</table>
<h2 id="format_of_md_files-format-of-segment-file-3-of-7">Format of segment file (3 of 7)<a class="headerlink" href="#format_of_md_files-format-of-segment-file-3-of-7" title="Permanent link">&para;</a></h2>
<table>
<thead>
<tr>
<th style="text-align: left;">Card</th>
<th style="text-align: left;">Format</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">For each bond a deck III</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">III-1-1</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">bond sequence number</td>
</tr>
<tr>
<td style="text-align: left;">III-1-2</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">bond atom i</td>
</tr>
<tr>
<td style="text-align: left;">III-1-3</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">bond atom j</td>
</tr>
<tr>
<td style="text-align: left;">III-1-4</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">bond type</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">0: harmonic</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">1: constrained bond</td>
</tr>
<tr>
<td style="text-align: left;">III-1-5</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">bond parameter origin</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">0: from database, next card ignored</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">1: from next card</td>
</tr>
<tr>
<td style="text-align: left;">For each parameter set one card III-2</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">III-2-1</td>
<td style="text-align: left;">f12.6</td>
<td style="text-align: left;">bond length in nm</td>
</tr>
<tr>
<td style="text-align: left;">III-2-2</td>
<td style="text-align: left;">e12.5</td>
<td style="text-align: left;">bond force constant in (kJ nm<sup>2</sup> mol<sup>-1</sup>)</td>
</tr>
</tbody>
</table>
<h2 id="format_of_md_files-format-of-segment-file-4-of-7">Format of segment file (4 of 7)<a class="headerlink" href="#format_of_md_files-format-of-segment-file-4-of-7" title="Permanent link">&para;</a></h2>
<table>
<thead>
<tr>
<th style="text-align: left;">Card</th>
<th style="text-align: left;">Format</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">For each angle a deck IV</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">IV-1-1</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">angle sequence number</td>
</tr>
<tr>
<td style="text-align: left;">IV-1-2</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">angle atom i</td>
</tr>
<tr>
<td style="text-align: left;">IV-1-3</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">angle atom j</td>
</tr>
<tr>
<td style="text-align: left;">IV-1-4</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">angle atom k</td>
</tr>
<tr>
<td style="text-align: left;">IV-1-5</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">angle type</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">0: harmonic</td>
</tr>
<tr>
<td style="text-align: left;">IV-1-6</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">angle parameter origin</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">0: from database, next card ignored</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">1: from next card</td>
</tr>
<tr>
<td style="text-align: left;">For each parameter set one card IV-2</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">IV-2-1</td>
<td style="text-align: left;">f10.6</td>
<td style="text-align: left;">angle in radians</td>
</tr>
<tr>
<td style="text-align: left;">IV-2-2</td>
<td style="text-align: left;">e12.5</td>
<td style="text-align: left;">angle force constant in (kJ mol<sup>-1</sup>)</td>
</tr>
</tbody>
</table>
<h2 id="format_of_md_files-format-of-segment-file-5-of-7">Format of segment file (5 of 7)<a class="headerlink" href="#format_of_md_files-format-of-segment-file-5-of-7" title="Permanent link">&para;</a></h2>
<table>
<thead>
<tr>
<th style="text-align: left;">Card</th>
<th style="text-align: left;">Format</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">For each proper dihedral a deck V</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">V-1-1</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">proper dihedral sequence number</td>
</tr>
<tr>
<td style="text-align: left;">V-1-2</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">proper dihedral atom i</td>
</tr>
<tr>
<td style="text-align: left;">V-1-3</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">proper dihedral atom j</td>
</tr>
<tr>
<td style="text-align: left;">V-1-4</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">proper dihedral atom k</td>
</tr>
<tr>
<td style="text-align: left;">V-1-5</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">proper dihedral atom l</td>
</tr>
<tr>
<td style="text-align: left;">V-1-6</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">proper dihedral type</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">0: (<em>C</em>cos(m&phi;-&delta;)</td>
</tr>
<tr>
<td style="text-align: left;">V-1-7</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">proper dihedral parameter origin</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">0: from database, next card ignored</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">1: from next card</td>
</tr>
<tr>
<td style="text-align: left;">For each parameter set one card V-2</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">V-2-1</td>
<td style="text-align: left;">i3</td>
<td style="text-align: left;">multiplicity</td>
</tr>
<tr>
<td style="text-align: left;">V-2-2</td>
<td style="text-align: left;">f10.6</td>
<td style="text-align: left;">proper dihedral in radians</td>
</tr>
<tr>
<td style="text-align: left;">V-2-3</td>
<td style="text-align: left;">e12.5</td>
<td style="text-align: left;">proper dihedral force constant in (kJ mol<sup>-1</sup>)</td>
</tr>
</tbody>
</table>
<h2 id="format_of_md_files-format-of-segment-file-6-of-7">Format of segment file (6 of 7)<a class="headerlink" href="#format_of_md_files-format-of-segment-file-6-of-7" title="Permanent link">&para;</a></h2>
<table>
<thead>
<tr>
<th style="text-align: left;">Card</th>
<th style="text-align: left;">Format</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">For each improper dihedral a deck VI</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">VI-1-1</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">improper dihedral sequence number</td>
</tr>
<tr>
<td style="text-align: left;">VI-1-2</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">improper dihedral atom i</td>
</tr>
<tr>
<td style="text-align: left;">VI-1-3</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">improper dihedral atom j</td>
</tr>
<tr>
<td style="text-align: left;">VI-1-4</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">improper dihedral atom k</td>
</tr>
<tr>
<td style="text-align: left;">VI-1-5</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">improper dihedral atom l</td>
</tr>
<tr>
<td style="text-align: left;">VI-1-6</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">improper dihedral type</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">0: harmonic</td>
</tr>
<tr>
<td style="text-align: left;">VI-1-7</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">improper dihedral parameter origin</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">0: from database, next card ignored</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">1: from next card</td>
</tr>
<tr>
<td style="text-align: left;">For each parameter set one card VI-2</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">VI-2-1</td>
<td style="text-align: left;">3x,f10.6</td>
<td style="text-align: left;">improper dihedral in radians</td>
</tr>
<tr>
<td style="text-align: left;">VI-2-2</td>
<td style="text-align: left;">e12.5</td>
<td style="text-align: left;">improper dihedral force constant in (kJ mol<sup>-1</sup>)</td>
</tr>
</tbody>
</table>
<h2 id="format_of_md_files-format-of-segment-file-7-of-7">Format of segment file (7 of 7)<a class="headerlink" href="#format_of_md_files-format-of-segment-file-7-of-7" title="Permanent link">&para;</a></h2>
<table>
<thead>
<tr>
<th style="text-align: left;">Card</th>
<th style="text-align: left;">Format</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">For each z-matrix definition one card VII</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">VII-1-1</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">atom i</td>
</tr>
<tr>
<td style="text-align: left;">VII-1-2</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">atom j</td>
</tr>
<tr>
<td style="text-align: left;">VII-1-3</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">atom k</td>
</tr>
<tr>
<td style="text-align: left;">VII-1-4</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">atom l</td>
</tr>
<tr>
<td style="text-align: left;">VII-1-5</td>
<td style="text-align: left;">f12.6</td>
<td style="text-align: left;">bond length i-j</td>
</tr>
<tr>
<td style="text-align: left;">VII-1-6</td>
<td style="text-align: left;">f12.6</td>
<td style="text-align: left;">angle i-j-k</td>
</tr>
<tr>
<td style="text-align: left;">VII-1-7</td>
<td style="text-align: left;">f12.6</td>
<td style="text-align: left;">torson i-j-k-l</td>
</tr>
</tbody>
</table>
<h2 id="format_of_md_files-format-of-sequence-file">Format of sequence file<a class="headerlink" href="#format_of_md_files-format-of-sequence-file" title="Permanent link">&para;</a></h2>
<table>
<thead>
<tr>
<th style="text-align: left;">Card</th>
<th style="text-align: left;">Format</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">I-1-1</td>
<td style="text-align: left;">a1</td>
<td style="text-align: left;">$ to identify the start of a sequence</td>
</tr>
<tr>
<td style="text-align: left;">I-1-2</td>
<td style="text-align: left;">a10</td>
<td style="text-align: left;">name of the sequence</td>
</tr>
<tr>
<td style="text-align: left;">Any number of cards 1 and 2 in deck II</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">to specify the system</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">II-1-1</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">segment number</td>
</tr>
<tr>
<td style="text-align: left;">II-1-2</td>
<td style="text-align: left;">a10</td>
<td style="text-align: left;">segment name, last character will be determined from chain</td>
</tr>
<tr>
<td style="text-align: left;">II-2-1</td>
<td style="text-align: left;">a</td>
<td style="text-align: left;">break to identify a break in the molecule chain</td>
</tr>
<tr>
<td style="text-align: left;">II-2-1</td>
<td style="text-align: left;">a</td>
<td style="text-align: left;">molecule to identify the end of a solute molecule</td>
</tr>
<tr>
<td style="text-align: left;">II-2-1</td>
<td style="text-align: left;">a</td>
<td style="text-align: left;">fraction to identify the end of a solute fraction</td>
</tr>
<tr>
<td style="text-align: left;">II-2-1</td>
<td style="text-align: left;">a5</td>
<td style="text-align: left;">link to specify a link</td>
</tr>
<tr>
<td style="text-align: left;">II-2-2</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">segment number of first link atom</td>
</tr>
<tr>
<td style="text-align: left;">II-2-3</td>
<td style="text-align: left;">a4</td>
<td style="text-align: left;">name of first link atom</td>
</tr>
<tr>
<td style="text-align: left;">II-2-4</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">segment number of second link atom</td>
</tr>
<tr>
<td style="text-align: left;">II-2-5</td>
<td style="text-align: left;">a4</td>
<td style="text-align: left;">name of second link atom</td>
</tr>
<tr>
<td style="text-align: left;">II-2-1</td>
<td style="text-align: left;">a</td>
<td style="text-align: left;">solvent to identify solvent definition on next card</td>
</tr>
<tr>
<td style="text-align: left;">II-2-1</td>
<td style="text-align: left;">a</td>
<td style="text-align: left;">stop to identify the end of the sequence</td>
</tr>
<tr>
<td style="text-align: left;">II-2-1</td>
<td style="text-align: left;">a6</td>
<td style="text-align: left;">repeat to repeat next ncard cards ncount times</td>
</tr>
<tr>
<td style="text-align: left;">II-2-2</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">number of cards to repeat (ncards)</td>
</tr>
<tr>
<td style="text-align: left;">II-2-3</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">number of times to repeat cards (ncount)</td>
</tr>
<tr>
<td style="text-align: left;">Any number of cards in deck II</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">to specify the system</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
<h2 id="format_of_md_files-format-of-trajectory-file">Format of trajectory file<a class="headerlink" href="#format_of_md_files-format-of-trajectory-file" title="Permanent link">&para;</a></h2>
<table>
<thead>
<tr>
<th style="text-align: left;">Card</th>
<th style="text-align: left;">Format</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">I-1-1</td>
<td style="text-align: left;">a6</td>
<td style="text-align: left;">keyword header</td>
</tr>
<tr>
<td style="text-align: left;">I-2-1</td>
<td style="text-align: left;">i10</td>
<td style="text-align: left;">number of atoms per solvent molecule</td>
</tr>
<tr>
<td style="text-align: left;">I-2-2</td>
<td style="text-align: left;">i10</td>
<td style="text-align: left;">number of solute atoms</td>
</tr>
<tr>
<td style="text-align: left;">I-2-3</td>
<td style="text-align: left;">i10</td>
<td style="text-align: left;">number of solute bonds</td>
</tr>
<tr>
<td style="text-align: left;">I-2-4</td>
<td style="text-align: left;">i10</td>
<td style="text-align: left;">number of solvent bonds</td>
</tr>
<tr>
<td style="text-align: left;">I-2-5</td>
<td style="text-align: left;">i10</td>
<td style="text-align: left;">number of solvent molecules</td>
</tr>
<tr>
<td style="text-align: left;">I-2-6</td>
<td style="text-align: left;">i10</td>
<td style="text-align: left;">precision of the coordinates. 0=standard, 1=high</td>
</tr>
<tr>
<td style="text-align: left;">For each atom per solvent molecule one card I-3</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">I-3-1</td>
<td style="text-align: left;">a5</td>
<td style="text-align: left;">solvent name</td>
</tr>
<tr>
<td style="text-align: left;">I-3-2</td>
<td style="text-align: left;">a5</td>
<td style="text-align: left;">atom name</td>
</tr>
<tr>
<td style="text-align: left;">For each solute atom one card I-4</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">I-4-1</td>
<td style="text-align: left;">a5</td>
<td style="text-align: left;">segment name</td>
</tr>
<tr>
<td style="text-align: left;">I-4-2</td>
<td style="text-align: left;">a5</td>
<td style="text-align: left;">atom name</td>
</tr>
<tr>
<td style="text-align: left;">I-4-3</td>
<td style="text-align: left;">i6</td>
<td style="text-align: left;">segment number</td>
</tr>
<tr>
<td style="text-align: left;">I-4-4</td>
<td style="text-align: left;">i10</td>
<td style="text-align: left;">solute atom counter</td>
</tr>
<tr>
<td style="text-align: left;">I-4-5</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">integer 1</td>
</tr>
<tr>
<td style="text-align: left;">For each solvent bond one card I-5</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">I-5-1</td>
<td style="text-align: left;">i8</td>
<td style="text-align: left;">atom index i for bond between i and j</td>
</tr>
<tr>
<td style="text-align: left;">I-5-2</td>
<td style="text-align: left;">i8</td>
<td style="text-align: left;">atom index j for bond between i and j</td>
</tr>
<tr>
<td style="text-align: left;">For each solute bond one card I-6</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">I-6-1</td>
<td style="text-align: left;">i8</td>
<td style="text-align: left;">atom index i for bond between i and j</td>
</tr>
<tr>
<td style="text-align: left;">I-6-2</td>
<td style="text-align: left;">i8</td>
<td style="text-align: left;">atom index j for bond between i and j</td>
</tr>
<tr>
<td style="text-align: left;">For each frame one deck II</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">II-1-1</td>
<td style="text-align: left;">a5</td>
<td style="text-align: left;">keyword frame</td>
</tr>
<tr>
<td style="text-align: left;">II-2-1</td>
<td style="text-align: left;">f12.6</td>
<td style="text-align: left;">time of frame in ps</td>
</tr>
<tr>
<td style="text-align: left;">II-2-2</td>
<td style="text-align: left;">12.6</td>
<td style="text-align: left;">temperature of frame in K</td>
</tr>
<tr>
<td style="text-align: left;">II-2-3</td>
<td style="text-align: left;">e12.5</td>
<td style="text-align: left;">pressure of frame in Pa</td>
</tr>
<tr>
<td style="text-align: left;">II-2-4</td>
<td style="text-align: left;">a10</td>
<td style="text-align: left;">date</td>
</tr>
<tr>
<td style="text-align: left;">II-2-5</td>
<td style="text-align: left;">a10</td>
<td style="text-align: left;">time</td>
</tr>
<tr>
<td style="text-align: left;">II-3-1</td>
<td style="text-align: left;">f12.6</td>
<td style="text-align: left;">box dimension x</td>
</tr>
<tr>
<td style="text-align: left;">II-3-2</td>
<td style="text-align: left;">12x,f12.6</td>
<td style="text-align: left;">box dimension y</td>
</tr>
<tr>
<td style="text-align: left;">II-3-3</td>
<td style="text-align: left;">24x,f12.6</td>
<td style="text-align: left;">box dimension z</td>
</tr>
<tr>
<td style="text-align: left;">II-4-1</td>
<td style="text-align: left;">l1</td>
<td style="text-align: left;">logical lxw for solvent coordinates</td>
</tr>
<tr>
<td style="text-align: left;">II-4-2</td>
<td style="text-align: left;">l1</td>
<td style="text-align: left;">logical lvw for solvent velocities</td>
</tr>
<tr>
<td style="text-align: left;">II-4-3</td>
<td style="text-align: left;">l1</td>
<td style="text-align: left;">logical lfw for solvent forces</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;">I-4-4</td>
<td style="text-align: left;">l1</td>
</tr>
<tr>
<td style="text-align: left;">II-4-5</td>
<td style="text-align: left;">l1</td>
<td style="text-align: left;">logical lxs for solute coordinates</td>
</tr>
<tr>
<td style="text-align: left;">II-4-6</td>
<td style="text-align: left;">l1</td>
<td style="text-align: left;">logical lvs for solute velocities</td>
</tr>
<tr>
<td style="text-align: left;">II-4-7</td>
<td style="text-align: left;">l1</td>
<td style="text-align: left;">logical lfs for solute forces</td>
</tr>
<tr>
<td style="text-align: left;">II-4-8</td>
<td style="text-align: left;">l1</td>
<td style="text-align: left;">logical lps for solute induced dipoles</td>
</tr>
<tr>
<td style="text-align: left;">II-4-5</td>
<td style="text-align: left;">i10</td>
<td style="text-align: left;">number of solvent molecules</td>
</tr>
<tr>
<td style="text-align: left;">II-4-6</td>
<td style="text-align: left;">i10</td>
<td style="text-align: left;">number of solvent atoms</td>
</tr>
<tr>
<td style="text-align: left;">II-4-7</td>
<td style="text-align: left;">i10</td>
<td style="text-align: left;">number of solute atoms</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;"><strong>For each solvent molecule one card II-5</strong></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;"><strong>for each atom, if standard precision</strong></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">II-5-1</td>
<td style="text-align: left;">3f8.3</td>
<td style="text-align: left;">solvent atom coordinates, if lxw or lvw</td>
</tr>
<tr>
<td style="text-align: left;">II-5-4</td>
<td style="text-align: left;">3f8.3</td>
<td style="text-align: left;">solvent atom velocities, if lvw</td>
</tr>
<tr>
<td style="text-align: left;">II-5-7</td>
<td style="text-align: left;">3f8.1</td>
<td style="text-align: left;">solvent atom forces, if lfw</td>
</tr>
<tr>
<td style="text-align: left;"><strong>For each solute atom one card II-6</strong></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;"><strong>for each atom, if standard precision</strong></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">II-6-1</td>
<td style="text-align: left;">3f8.3</td>
<td style="text-align: left;">solute atom coordinates, if lxs or lvs</td>
</tr>
<tr>
<td style="text-align: left;">II-6-4</td>
<td style="text-align: left;">3f8.3</td>
<td style="text-align: left;">solute atom velocities, if lvs</td>
</tr>
<tr>
<td style="text-align: left;">II-6-7</td>
<td style="text-align: left;">3f8.1</td>
<td style="text-align: left;">solute atom forces, if lfs</td>
</tr>
<tr>
<td style="text-align: left;"><strong>For each solvent molecule one card II-5</strong></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;"><strong>for each atom, if high precision</strong></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">II-5-1</td>
<td style="text-align: left;">3e12.6</td>
<td style="text-align: left;">solvent atom coordinates, if lxw or lvw</td>
</tr>
<tr>
<td style="text-align: left;">II-5-4</td>
<td style="text-align: left;">3e12.6</td>
<td style="text-align: left;">solvent atom velocities, if lvw</td>
</tr>
<tr>
<td style="text-align: left;">II-5-7</td>
<td style="text-align: left;">3e12.6</td>
<td style="text-align: left;">solvent atom forces, if lfw (on new card if both lxw and lvw)</td>
</tr>
<tr>
<td style="text-align: left;"><strong>For each solute atom one card II-6</strong></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;"><strong>for each atom, if high precision</strong></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">II-6-1</td>
<td style="text-align: left;">3e12.6</td>
<td style="text-align: left;">solute atom coordinates, if lxs or lvs</td>
</tr>
<tr>
<td style="text-align: left;">II-6-4</td>
<td style="text-align: left;">3e12.6</td>
<td style="text-align: left;">solute atom velocities, if lvs</td>
</tr>
<tr>
<td style="text-align: left;">II-6-7</td>
<td style="text-align: left;">3e12.6</td>
<td style="text-align: left;">solute atom forces, if lfs (on new card if both lxs and lvs)</td>
</tr>
</tbody>
</table>
<h2 id="format_of_md_files-format-of-free-energy-file">Format of free energy file<a class="headerlink" href="#format_of_md_files-format-of-free-energy-file" title="Permanent link">&para;</a></h2>
<table>
<thead>
<tr>
<th style="text-align: left;">Card</th>
<th style="text-align: left;">Format</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">For each step in &lambda; one deck I</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">I-1-1</td>
<td style="text-align: left;">i7</td>
<td style="text-align: left;">number nderiv of data summed in derivative decomposition array deriv</td>
</tr>
<tr>
<td style="text-align: left;">I-1-2</td>
<td style="text-align: left;">i7</td>
<td style="text-align: left;">length ndata of total derivative array drf</td>
</tr>
<tr>
<td style="text-align: left;">I-1-3</td>
<td style="text-align: left;">f12.6</td>
<td style="text-align: left;">current value of &lambda;</td>
</tr>
<tr>
<td style="text-align: left;">I-1-4</td>
<td style="text-align: left;">f12.6</td>
<td style="text-align: left;">step size of &lambda;</td>
</tr>
<tr>
<td style="text-align: left;">I-2-1</td>
<td style="text-align: left;">4e12.12</td>
<td style="text-align: left;">derivative decomposition array deriv(1:24)</td>
</tr>
<tr>
<td style="text-align: left;">I-3-1</td>
<td style="text-align: left;">4e12.12</td>
<td style="text-align: left;">total derivative array dfr(1:nda)</td>
</tr>
<tr>
<td style="text-align: left;">I-4-1</td>
<td style="text-align: left;">i10</td>
<td style="text-align: left;">size of ensemble at current &lambda;</td>
</tr>
<tr>
<td style="text-align: left;">I-4-2</td>
<td style="text-align: left;">e20.12</td>
<td style="text-align: left;">average temperature at current &lambda;</td>
</tr>
<tr>
<td style="text-align: left;">I-4-3</td>
<td style="text-align: left;">e20.12</td>
<td style="text-align: left;">average exponent reverse perturbation energy at current &lambda;</td>
</tr>
<tr>
<td style="text-align: left;">I-4-4</td>
<td style="text-align: left;">e20.12</td>
<td style="text-align: left;">average exponent forward perturbation energy at current &lambda;</td>
</tr>
</tbody>
</table>
<h2 id="format_of_md_files-format-of-root-mean-square-deviation-file">Format of root mean square deviation file<a class="headerlink" href="#format_of_md_files-format-of-root-mean-square-deviation-file" title="Permanent link">&para;</a></h2>
<table>
<thead>
<tr>
<th style="text-align: left;">Card</th>
<th style="text-align: left;">Format</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">For each analyzed time step one card I-1</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">I-1-1</td>
<td style="text-align: left;">f12.6</td>
<td style="text-align: left;">time in ps</td>
</tr>
<tr>
<td style="text-align: left;">I-1-2</td>
<td style="text-align: left;">f12.6</td>
<td style="text-align: left;">total rms deviation of the selected atoms before superimposition</td>
</tr>
<tr>
<td style="text-align: left;">I-1-3</td>
<td style="text-align: left;">f12.6</td>
<td style="text-align: left;">total rms deviation of the selected atoms after superimposition</td>
</tr>
<tr>
<td style="text-align: left;">II-1-1</td>
<td style="text-align: left;">a8</td>
<td style="text-align: left;">keyword analysis</td>
</tr>
<tr>
<td style="text-align: left;">For each solute atom one card II-2</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">II-2-1</td>
<td style="text-align: left;">a5</td>
<td style="text-align: left;">segment name</td>
</tr>
<tr>
<td style="text-align: left;">II-2-2</td>
<td style="text-align: left;">a5</td>
<td style="text-align: left;">atom name</td>
</tr>
<tr>
<td style="text-align: left;">II-2-3</td>
<td style="text-align: left;">i6</td>
<td style="text-align: left;">segment number</td>
</tr>
<tr>
<td style="text-align: left;">II-2-4</td>
<td style="text-align: left;">i10</td>
<td style="text-align: left;">atom number</td>
</tr>
<tr>
<td style="text-align: left;">II-2-5</td>
<td style="text-align: left;">i5</td>
<td style="text-align: left;">selected if 1</td>
</tr>
<tr>
<td style="text-align: left;">II-2-6</td>
<td style="text-align: left;">f12.6</td>
<td style="text-align: left;">average atom rms deviation after superimposition</td>
</tr>
<tr>
<td style="text-align: left;">III-1-1</td>
<td style="text-align: left;">a8</td>
<td style="text-align: left;">keyword analysis</td>
</tr>
<tr>
<td style="text-align: left;">For each solute segment one card III-2</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">III-2-1</td>
<td style="text-align: left;">a5</td>
<td style="text-align: left;">segment name</td>
</tr>
<tr>
<td style="text-align: left;">III-2-2</td>
<td style="text-align: left;">i6</td>
<td style="text-align: left;">segment number</td>
</tr>
<tr>
<td style="text-align: left;">III-2-3</td>
<td style="text-align: left;">f12.6</td>
<td style="text-align: left;">average segment rms deviation after superimposition</td>
</tr>
</tbody>
</table>
<h2 id="format_of_md_files-format-of-property-file">Format of property file<a class="headerlink" href="#format_of_md_files-format-of-property-file" title="Permanent link">&para;</a></h2>
<table>
<thead>
<tr>
<th style="text-align: left;">Card</th>
<th style="text-align: left;">Format</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">1</td>
<td style="text-align: left;">i7</td>
<td style="text-align: left;">number nprop of recorded properties</td>
</tr>
<tr>
<td style="text-align: left;">I-1-2</td>
<td style="text-align: left;">1x,2a10</td>
<td style="text-align: left;">date and time</td>
</tr>
<tr>
<td style="text-align: left;">For each of the nprop properties one card I-2</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">I-2-1</td>
<td style="text-align: left;">a50</td>
<td style="text-align: left;">description of recorded property</td>
</tr>
<tr>
<td style="text-align: left;">For each recorded step one deck II</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;">II-1-1</td>
<td style="text-align: left;">4(1pe12.5)</td>
<td style="text-align: left;">value of property</td>
</tr>
</tbody>
</table></section></section><section class="print-page" id="python" heading-number="18"><h1 id="python-controlling-nwchem-with-python">Controlling NWChem with Python<a class="headerlink" href="#python-controlling-nwchem-with-python" title="Permanent link">&para;</a></h1>
<h2 id="python-overview">Overview<a class="headerlink" href="#python-overview" title="Permanent link">&para;</a></h2>
<p>Python programs may be embedded into the NWChem input
and used to control the execution of NWChem. Python is a very powerful
and widely used scripting language that provides useful things such as
variables, conditional branches and loops, and is also readily extended.
Example applications include scanning potential energy surfaces,
computing properties in a variety of basis sets, optimizing the energy
w.r.t. parameters in the basis set, computing polarizabilities with
finite field, and simple molecular dynamics.</p>
<p>Look in the NWChem contrib directory for useful scripts and examples.
Visit the Python web-site <a href="https://www.python.org">https://www.python.org</a> for a full manual and
lots of useful code and resources.</p>
<h2 id="python-how-to-input-and-run-a-python-program-inside-nwchem">How to input and run a Python program inside NWChem<a class="headerlink" href="#python-how-to-input-and-run-a-python-program-inside-nwchem" title="Permanent link">&para;</a></h2>
<p>A Python program is input into NWChem inside a Python compound
directive.</p>
<pre><code> python [print|noprint]
   ...
 end
</code></pre>
<p>The END directive must be flush against the left margin (see the
Troubleshooting section for the reason why).</p>
<p>The program is by default printed to standard output when read, but this
may be disabled with the noprint keyword. Python uses indentation to
indicate scope (and the initial level of indentation must be zero),
whereas NWChem uses optional indentation only to make the input more
readable. For example, in Python, the contents of a loop, or
conditionally-executed block of code must be indented further than the
surrounding code. Also, Python attaches special meaning to several
symbols also used by NWChem. For these reasons, the input inside a
PYTHON compound directive is read verbatim except that if the first line
of the Python program is indented, the same amount of indentation is
removed from all subsequent lines. This is so that a program may be
indented inside the PYTHON input block for improved readability of the
NWChem input, while satisfying the constraint that when given to Python
the first line has zero indentation.</p>
<p>E.g., the following two sets of input specify the same Python program.</p>
<pre><code>python  
  print (&quot;Hello&quot;)
  print (&quot;Goodbye&quot;)
end

python  
print (&quot;Hello&quot;)
print (&quot;Goodbye&quot;)
end
</code></pre>
<p>whereas this program is in error since the indentation of the second
line is less than that of the first.</p>
<pre><code>python  
  print (&quot;Hello&quot;)
print (&quot;Goodbye&quot;)
end
</code></pre>
<p>The Python program is not executed until the following directive is
encountered</p>
<pre><code> task python
</code></pre>
<p>which is to maintain consistency with the behavior of NWChem in general.
The program is executed by all nodes. This enables the full
functionality and speed of NWChem to be accessible from Python, but
there are some gotchas</p>
<ul>
<li>Print statements and other output will be executed by all nodes so
    you will get a lot more output than probably desired unless the
    output is restricted to just one node (by convention node zero).</li>
<li>The calls to NWChem functions are all collective (i.e., all nodes
    must execute them). If these calls are not made collectively your
    program may deadlock (i.e., cease to make progress).</li>
<li>When writing to the database (<code>rtdb_put()</code>) it is the data from node
    zero that is written.</li>
<li>NWChem overrides certain default signal handlers so care must be
    taken when creating processes (see Section describing <a href="#python-scaning-a-basis-exponent-yet-again-plotting-and-handling-child-processes">a scanning example</a>).</li>
</ul>
<h2 id="python-nwchem-extensions">NWChem extensions<a class="headerlink" href="#python-nwchem-extensions" title="Permanent link">&para;</a></h2>
<p>Since we have little experience using Python, the NWChem-Python
interface might change in a non-backwardly compatible fashion as we
discover better ways of providing useful functionality. We would
appreciate suggestions about useful things that can be added to the
NWChem-Python interface. In principle, nearly any Fortran or C routine
within NWChem can be extended to Python, but we are also interested in
ideas that will enable users to build completely new things. For
instance, how about being able to define your own energy functions that
can be used with the existing optimizers or dynamics package?</p>
<p>Python has been extended with a module named &ldquo;nwchem&rdquo; which is
automatically imported and contains the following NWChem-specific
commands. They all handle NWChem-related errors by raising the exception
&ldquo;NWChemError&rdquo;, which may be handled in the standard Python manner (see
Section describing <a href="#python-handling-exceptions-from-nwchem">handling exception</a>).</p>
<ul>
<li><code>input_parse(string)</code> &ndash; invokes the standard NWChem input parser
    with the data in string as input. Note that the usual behavior of
    NWChem will apply &ndash; the parser only reads input up to either end of
    input or until a <code>TASK</code> directive is encountered (the task directive
    is not executed by the parser).</li>
<li><code>task_energy(theory)</code> &ndash; returns the energy as if computed with the
    NWChem directive <code>TASK ENERGY &lt;THEORY&gt;</code>.</li>
<li><code>task_gradient(theory)</code> &ndash; returns a tuple (energy,gradient) as if
    computed with the NWChem directive <code>TASK GRADIENT &lt;THEORY&gt;</code>.</li>
<li><code>task_optimize(theory)</code> &ndash; returns a tuple (energy,gradient) as if
    computed with the NWChem directive <code>TASK OPTIMIZE &lt;THEORY&gt;</code>. The
    energy and gradient will be those at the last point in the
    optimization and consistent with the current geometry in the
    database.</li>
<li><code>ga_nodeid()</code> &ndash; returns the number of the parallel process.</li>
<li><code>rtdb_print(print_values)</code> &ndash; prints the contents of the RTDB. If
    <code>print_values</code> is 0, only the keys are printed, if it is 1 then the
    values are also printed.</li>
<li><code>rtdb_put(name, values)</code> or <code>rtdb_put(name, values, type)</code> &ndash; puts the
    values into the database with the given name. In the first form, the
    type is inferred from the first value, and in the second form the
    type is specified using the last argument as one of INT, DBL,
    LOGICAL, or CHAR.</li>
<li><code>rtdb_get(name)</code> &ndash; returns the data from the database associated
    with the given name.</li>
</ul>
<p>An
<a href="#python-accessing-geometry-information-a-temporary-hack">example</a>
below explains, in lieu of a Python wrapper for the geometry object, how
to obtain the Cartesian molecular coordinates directly from the
database.</p>
<h2 id="python-examples">Examples<a class="headerlink" href="#python-examples" title="Permanent link">&para;</a></h2>
<p>Several examples will provide the best explanation of how the extensions
are used, and how Python might prove useful.</p>
<h3 id="python-hello-world">Hello world<a class="headerlink" href="#python-hello-world" title="Permanent link">&para;</a></h3>
<pre><code>python
  print ('Hello world from process %i'  % ga_nodeid())
end

task python
</code></pre>
<p>This input prints the traditional greeting from each parallel
process.</p>
<h3 id="python-scanning-a-basis-exponent">Scanning a basis exponent<a class="headerlink" href="#python-scanning-a-basis-exponent" title="Permanent link">&para;</a></h3>
<pre><code>geometry units au
 O 0 0 0; H 0 1.430 -1.107; H 0 -1.430 -1.107
end

python
exponent = 0.1
while (exponent &lt;= 2.01):
    input_parse('''                                                                                                
    basis noprint                                                                                                  
     H library 3-21g; O library 3-21g; O d; %f 1.0                                                                 
    end                                                                                                            
    ''' % (exponent))
    print (' exponent = %5.4f,' % exponent, ', energy = %16.10f' % task_energy('scf'))
    exponent = exponent + 0.1
end

print none

task python
</code></pre>
<p>This program augments a 3-21g basis for water with a d-function on
oxygen and varies the exponent from 0.1 to 2.0 in steps of 0.1, printing
the exponent and energy at each step.</p>
<p>The geometry is input as usual, but the basis set input is embedded
inside a call to <code>input_parse()</code> in the Python program. The standard
Python string substitution is used to put the current value of the
exponent into the basis set (replacing the %f) before being parsed by
NWChem. The energy is returned by <code>task_energy('scf')</code> and printed out.
The print none in the NWChem input switches off all NWChem output so all
you will see is the output from your Python program.</p>
<p>Note that execution in parallel may produce unwanted output since all
process execute the print statement inside the Python program.</p>
<p>Look in the NWChem contrib directory for a routine that makes the above
task easier.</p>
<h3 id="python-scanning-a-basis-exponent-revisited">Scanning a basis exponent revisited<a class="headerlink" href="#python-scanning-a-basis-exponent-revisited" title="Permanent link">&para;</a></h3>
<pre><code>geometry units au
O 0 0 0; H 0 1.430 -1.107; H 0 -1.430 -1.107
end


python
  if (ga_nodeid() == 0): plotdata = open(&quot;plotdata&quot;,'w')

  def energy_at_exponent(exponent):
    input_parse('''                                                                                                
    basis noprint                                                                                                  
      H library 3-21g; O library 3-21g; O d; %f 1.0                                                                
    end                                                                                                            
    ''' % (exponent))
    return task_energy('scf')

  exponent = 0.1
  while exponent &lt;= 2.01:
    energy = energy_at_exponent(exponent)
    if (ga_nodeid() == 0):
        print (' exponent = %5.4f,' % exponent, ', energy = %16.10f' % energy)
        plotdata.write('%f %f\n' % (exponent , energy))
    exponent = exponent + 0.1

  if (ga_nodeid() == 0): plotdata.close()
end

print none
task python
</code></pre>
<p>This input performs exactly the same calculation as the previous one,
but uses a slightly more sophisticated Python program, also writes the
data out to a file for easy visualization with a package such as
gnuplot, and protects write statements to prevent duplicate output in a
parallel job. The only significant differences are in the Python
program. A file called &ldquo;plotdata&rdquo; is opened, and then a procedure is
defined which given an exponent returns the energy. Next comes the main
loop that scans the exponent through the desired range and prints the
results to standard output and to the file. When the loop is finished
the additional output file is closed.</p>
<h3 id="python-scanning-a-geometric-variable">Scanning a geometric variable<a class="headerlink" href="#python-scanning-a-geometric-variable" title="Permanent link">&para;</a></h3>
<pre><code>python
   geometry = '''
     geometry noprint; symmetry d2h
        C 0 0 %f; H 0  0.916 1.224
     end
   '''
   x = 0.6
   while (x &lt; 0.721):
     input_parse(geometry % x)
     energy = task_energy('scf')
     print (' x = %5.2f   energy = %10.6f' % (x, energy))
     x = x + 0.01
end

basis; C library 6-31g*; H library 6-31g*; end

print none

task python
</code></pre>
<p>This scans the bond length in ethene from 1.2 to 1.44 in steps of 0.2
computing the energy at each geometry. Since it is using D<sub>2h</sub>
symmetry the program actually uses a variable (x) that is half the bond
length.</p>
<p>Look in the NWChem contrib directory for a routine that makes the above
task
easier.</p>
<h3 id="python-scan-using-the-bsse-counterpoise-corrected-energy">Scan using the BSSE counterpoise corrected energy<a class="headerlink" href="#python-scan-using-the-bsse-counterpoise-corrected-energy" title="Permanent link">&para;</a></h3>
<pre><code>basis spherical
  Ne library cc-pvdz; BqNe library Ne cc-pvdz
  He library cc-pvdz; BqHe library He cc-pvdz
end

mp2; tight; freeze core atomic; end

print none

python
  supermolecule = 'geometry noprint;   Ne 0 0 0;   He 0 0 %f; end\n'
  fragment1     = 'geometry noprint;   Ne 0 0 0; BqHe 0 0 %f; end\n'
  fragment2     = 'geometry noprint; BqNe 0 0 0;   He 0 0 %f; end\n'

  def energy(geometry):
    input_parse(geometry + 'scf; vectors atomic; end\n')
    return task_energy('mp2')

  def bsse_energy(z):
    return energy(supermolecule % z) - \
        energy(fragment1 % z) - \
        energy(fragment2 % z)
  z = 3.3
  while (z &lt; 4.301):
    e = bsse_energy(z)
    if (ga_nodeid() == 0): print (' z = %5.2f   energy = %10.7f ' % (z, e))
    z = z + 0.1
end

task python
</code></pre>
<p>This example scans the He&ndash;Ne bond-length from 3.3 to 4.3 and prints out
the BSSE counterpoise corrected MP2 energy.</p>
<p>The basis set is specified as usual, noting that we will need functions
on ghost centers to do the counterpoise correction. The Python program
commences by defining strings containing the geometry of the
super-molecule and two fragments, each having one variable to be
substituted. Next, a function is defined to compute the energy given a
geometry, and then a function is defined to compute the counterpoise
corrected energy at a given bond length. Finally, the bond length is
scanned and the energy printed. When computing the energy, the atomic
guess has to be forced in the SCF since by default it will attempt to
use orbitals from the previous calculation which is not appropriate
here.</p>
<p>Since the counterpoise corrected energy is a linear combination of other
standard energies, it is possible to compute the analytic derivatives
term by term. Thus, combining this example and the next could yield the
foundation of a BSSE corrected geometry optimization
package.</p>
<h3 id="python-scan-the-geometry-and-compute-the-energy-and-gradient">Scan the geometry and compute the energy and gradient<a class="headerlink" href="#python-scan-the-geometry-and-compute-the-energy-and-gradient" title="Permanent link">&para;</a></h3>
<pre><code>basis noprint; H library sto-3g; O library sto-3g; end

python
 print ('   y     z     energy                gradient')
 print (' ----- ----- ---------- ------------------------------------')
 y = 1.2
 while y &lt;= 1.61:
     z = 1.0
     while z &lt;= 1.21:
         input_parse('''
           geometry noprint units atomic
             O 0   0   0
             H 0  %f -%f
             H 0 -%f -%f
           end
         ''' % (y, z, y, z))

         (energy,gradient) = task_gradient('scf')

         print (' %5.2f %5.2f %9.6f' % (y, z, energy)),
         i = 0
         while (i &lt; len(gradient)):
             print ('%5.2f' % gradient[i]),
             i = i + 1
         print ('')
         z = z + 0.1
     y = y + 0.1
end

print none

task python
</code></pre>
<p>This program illustrates evaluating the energy and gradient by calling
task_gradient(). A water molecule is scanned through several C<sub>2v</sub>
geometries by varying the y and z coordinates of the two hydrogen atoms.
At each geometry the coordinates, energy and gradient are printed.</p>
<p>The basis set (sto-3g) is input as usual. The two while loops vary the y
and z coordinates. These are then substituted into a geometry which is
parsed by NWChem using <code>input_parse()</code>. The energy and gradient are then
evaluated by calling <code>task_gradient()</code> which returns a tuple containing
the energy (a scalar) and the gradient (a vector or list). These are
printed out exploiting the Python convention that a print statement
ending in a comma does not print end-of-line.</p>
<h3 id="python-reaction-energies-varying-the-basis-set">Reaction energies varying the basis set<a class="headerlink" href="#python-reaction-energies-varying-the-basis-set" title="Permanent link">&para;</a></h3>
<pre><code>mp2; freeze atomic; end

print none

python
  energies = {}
  c2h4 = 'geometry noprint; symmetry d2h; \
          C 0 0 0.672; H 0 0.935 1.238; end\n'
  ch4  = 'geometry noprint; symmetry td; \
          C 0 0 0; H 0.634 0.634 0.634; end\n'
  h2   = 'geometry noprint; H 0 0 0.378; H 0 0 -0.378; end\n'

  def energy(basis, geometry):
    input_parse('''
      basis spherical noprint
        c library %s ; h library %s 
      end
    ''' % (basis, basis))
    input_parse(geometry)
    return task_energy('mp2')

  for basis in ('sto-3g', '6-31g', '6-31g*', 'cc-pvdz', 'cc-pvtz'):
     energies[basis] =   2*energy(basis, ch4) \
                       - 2*energy(basis, h2) - energy(basis, c2h4)
     if (ga_nodeid() == 0): print (basis, ' %8.6f' % energies[basis])
end 

task python
</code></pre>
<p>In this example the reaction energy for  2H<sub>2</sub> + C<sub>2</sub>H<sub>4</sub> &rarr;
2CH<sub>4</sub> is evaluated using MP2 in several basis sets. The geometries are
fixed, but could be re-optimized in each basis. To illustrate the useful
associative arrays in Python, the reaction energies are put into the
associative array energies &ndash; note its declaration at the top of the
program.</p>
<h3 id="python-using-the-database">Using the database<a class="headerlink" href="#python-using-the-database" title="Permanent link">&para;</a></h3>
<pre><code> python  
   rtdb_put(&quot;test_int2&quot;, 22)  
   rtdb_put(&quot;test_int&quot;, [22, 10, 3],    INT)  
   rtdb_put(&quot;test_dbl&quot;, [22.9, 12.4, 23.908],  DBL)  
   rtdb_put(&quot;test_str&quot;, &quot;hello&quot;, CHAR)  
   rtdb_put(&quot;test_logic&quot;, [0,1,0,1,0,1], LOGICAL)  
   rtdb_put(&quot;test_logic2&quot;, 0, LOGICAL)  

   rtdb_print(1)  

   print &quot;test_str    = &quot;, rtdb_get(&quot;test_str&quot;)  
   print &quot;test_int    = &quot;, rtdb_get(&quot;test_int&quot;)  
   print &quot;test_in2    = &quot;, rtdb_get(&quot;test_int2&quot;)  
   print &quot;test_dbl    = &quot;, rtdb_get(&quot;test_dbl&quot;)  
   print &quot;test_logic  = &quot;, rtdb_get(&quot;test_logic&quot;)  
   print &quot;test_logic2 = &quot;, rtdb_get(&quot;test_logic2&quot;)  
 end  

 task python
</code></pre>
<p>This example illustrates how to access the database from Python.</p>
<h3 id="python-handling-exceptions-from-nwchem">Handling exceptions from NWChem<a class="headerlink" href="#python-handling-exceptions-from-nwchem" title="Permanent link">&para;</a></h3>
<pre><code> geometry; he 0 0 0; he 0 0 2; end  
 basis; he library 3-21g; end  
 scf; maxiter 1; end  

 python  
   try:  
     task_energy('scf')  
   except NWChemError, message:  
     print 'Error from NWChem ... ', message  
 end  

 task python
</code></pre>
<p>The above test program shows how to handle exceptions generated by
NWChem by forcing an SCF calculation on He<sub>2&lt;\sub&gt; to fail due to
insufficient iterations.</p>
<p>If an NWChem command fails it will raise the exception &ldquo;NWChemError&rdquo;
(case sensitive) unless the error was fatal. If the exception is not
caught, then it will cause the entire Python program to terminate with
an error. This Python program catches the exception, prints out the
message, and then continues as if all was well since the exception has
been handled.</p>
<p>If your Python program detects an error, raise an unhandled exception.
Do not call exit(1) since this may circumvent necessary clean-up of the
NWChem execution environment.</p>
<h3 id="python-accessing-geometry-information-a-temporary-hack">Accessing geometry information: a temporary hack<a class="headerlink" href="#python-accessing-geometry-information-a-temporary-hack" title="Permanent link">&para;</a></h3>
<p>In an ideal world the geometry and basis set objects would have full
Python wrappers, but until then a back-door solution will have to
suffice. We&rsquo;ve already seen how to use <code>input_parse()</code> to put geometry
(and basis) data into NWChem, so it only remains to get the geometry
data back after it has been updated by a geometry optimzation or some
other operation.</p>
<p>The following Python procedure retrieves the coordinates in the same
units as initially input for a geometry of a given name. Its full source
is included in the NWChem contrib directory.</p>
<pre><code> def geom_get_coords(name):  
   try:  
     actualname = rtdb_get(name)  
   except NWChemError:  
     actualname = name  
   coords = rtdb_get('geometry:' + actualname + ':coords')  
   units  = rtdb_get('geometry:' + actualname + ':user units')  
   if (units == 'a.u.'):  
     factor = 1.0  
   elif (units == 'angstroms'):  
     factor = rtdb_get('geometry:'+actualname+':angstrom_to_au')  
   else:  
     raise NWChemError,'unknown units'  
   i = 0  
   while (i &lt; len(coords)):  
     coords[i] = coords[i] / factor  
     i = i + 1  
   return coords
</code></pre>
<p>A <a href="#geometry">geometry</a> with name NAME has its coordinates (in
atomic units) stored in the database entry <code>geometry:NAME:coords</code>. A minor
wrinkle here is that indirection is possible (and used by the
optimizers) so that we must first check if NAME actually points to
another name. In the program this is done in the first try&hellip;except
sequence. With the actual name of the geometry, we can get the
coordinates. Any exceptions are passed up to the caller. The rest of the
code is just to convert back into the initial input units &ndash; only atomic
units or Angstrms are handled in this simple example. Returned is a
list of the atomic coordinates in the same units as your initial input.</p>
<p>The routine is used as follows</p>
<pre><code>   coords = geom_get_coords('geometry')
</code></pre>
<p>or, if you want better error handling</p>
<pre><code>   try:  
     coords = geom_get_coords('geometry')  
   except NWChemError,message:  
     print 'Coordinates for geometry not found ', message  
   else:  
     print coords
</code></pre>
<p>This is very dirty and definitely not supported from one release to
another, but, browsing the output of <code>rtdb_print()</code> at the end of a
calculation is a good way to find stuff. To be on safer ground, look in
the programmers manual since some of the high-level routines do pass
data via the database in a well-defined and supported manner. Be warned
&ndash; you must be very careful if you try to modify data in the database.
The input parser does many important things that are not immediately
apparent (e.g., ensure the geometry is consistent with the point group,
mark the SCF as not converged if the SCF options are changed, &hellip;).
Where at all possible your Python program should generate standard
NWChem input and pass it to <code>input_parse()</code> rather than setting
parameters directly in the
database.</p>
<h3 id="python-scaning-a-basis-exponent-yet-again-plotting-and-handling-child-processes">Scaning a basis exponent yet again: plotting and handling child processes<a class="headerlink" href="#python-scaning-a-basis-exponent-yet-again-plotting-and-handling-child-processes" title="Permanent link">&para;</a></h3>
<pre><code> geometry units au  
   O 0 0 0; H 0 1.430 -1.107; H 0 -1.430 -1.107  
 end  

 print none  

 python  
   import Gnuplot, time, signal  

   def energy_at_exponent(exponent):  
      input_parse('''  
         basis noprint  
            H library 3-21g; O library 3-21g; O d; %f 1.0  
         end  
      ''' % (exponent))  
      return task_energy('scf')  

   data = []  
   exponent = 0.5  
   while exponent &lt;= 0.6:  
      energy = energy_at_exponent(exponent)  
      print ' exponent = ', exponent, ' energy = ', energy  
      data = data + [exponent,energy](exponent,energy.md)  
      exponent = exponent + 0.02  

   if (ga_nodeid() == 0):  
      signal.signal(signal.SIGCHLD, signal.SIG_DFL)  
      g = Gnuplot.Gnuplot()  
      g('set data style linespoints')  
      g.plot(data)  
      time.sleep(30)  # 30s to look at the plot  

 end  

 task python
</code></pre>
<p>This illustrates how to handle signals from terminating child processes
and how to generate simple plots on UNIX systems. The <a href="#python-scanning-a-basis-exponent-revisited">scanning
example</a> is modified so
that instead of writing the data to a file for subsequent visualization,
it is saved for subsequent visualization with Gnuplot (you&rsquo;ll need both
Gnuplot and the corresponding package for Python in your PYTHONPATH.
Look at <a href="https://web.archive.org/web/20010717060625/http://monsoon.harvard.edu/~mhagger/download/">https://web.archive.org/web/20010717060625/http://monsoon.harvard.edu/~mhagger/download/</a>).</p>
<p>The issue is that NWChem traps various signals from the O/S that usually
indicate bad news in order to provide better error handling and reliable
clean-up of shared, parallel resources. One of these signals is SIGCHLD
which is generated whenever a child process terminates. If you want to
create child processes within Python, then the NWChem handler for
SIGCHLD must be replaced with the default handler. There seems to be no
easy way to restore the NWChem handler after the child has completed,
but this should have no serious side effect.</p>
<h2 id="python-troubleshooting">Troubleshooting<a class="headerlink" href="#python-troubleshooting" title="Permanent link">&para;</a></h2>
<p>Common problems with Python programs inside NWChem.</p>
<ol>
<li>
<p>You get the message<br />
<code>0:python_input: indentation must be &gt;= that of first line: 4</code>    <br />
   This indicates that NWChem thinks that a line is less indented than
   the first line. If this is not the case then perhaps there is a tab
    in your input which NWChem treats as a single space character but
    appears to you as more spaces. Try running untabify in Emacs. The
    problem could also be the <code>END</code> directive that terminates the <code>PYTHON</code>
    compound directive, since Python also has an end statement. To
    avoid confusion the <code>END</code> directive for NWChem must be at the start of
    the line.  </p>
</li>
<li>
<p>Your program hangs or deadlocks &ndash; most likely you have a piece of
    code that is restricted to executing on a subset of the processors
    (perhaps just node 0) but is calling (perhaps indirectly) a function
    that must execute on all nodes.</p>
</li>
</ol></section>
                    <section class='print-page md-section' id='section-19' heading-number='19'>
                        <h1>NWChem Examples<a class='headerlink' href='#section-19' title='Permanent link'></a>
                        </h1>
                    <section class="print-page" id="sample" heading-number="19.1"><h1 id="sample-sample-input-files">Sample input files<a class="headerlink" href="#sample-sample-input-files" title="Permanent link">&para;</a></h1>
<h2 id="sample-water-scf-calculation-and-geometry-optimization-in-a-6-31g-basis">Water SCF calculation and geometry optimization in a 6-31g basis<a class="headerlink" href="#sample-water-scf-calculation-and-geometry-optimization-in-a-6-31g-basis" title="Permanent link">&para;</a></h2>
<p>The <a href="#getting-started">Getting Started</a> input file performs a
geometry optimization in a single task. A single point SCF energy
calculation is performed and then restarted to perform the optimization
(both could of course be performed in a single task).</p>
<h3 id="sample-job-1-single-point-scf-energy">Job 1. Single point SCF energy<a class="headerlink" href="#sample-job-1-single-point-scf-energy" title="Permanent link">&para;</a></h3>
<pre><code>starth2o 
title&quot;Waterin6-31gbasisset&quot; 

geometryunitsau  
O0.000000000.000000000.00000000  
H0.000000001.43042809-1.10715266  
H0.00000000-1.43042809-1.10715266 
end  
basis  
Hlibrary6-31g  
Olibrary6-31g  
end
taskscf
</code></pre>
<p>The final energy should be -75.983998.</p>
<h3 id="sample-job-2-restarting-and-perform-a-geometry-optimization">Job 2. Restarting and perform a geometry optimization<a class="headerlink" href="#sample-job-2-restarting-and-perform-a-geometry-optimization" title="Permanent link">&para;</a></h3>
<pre><code>restarth2o
title&quot;Watergeometryoptimization&quot;

taskscfoptimize
</code></pre>
<p>There is no need to specify anything that has not changed from the
previous input deck, though it will do no harm to repeat it.</p>
<h2 id="sample-compute-the-polarizability-of-ne-using-finite-field">Compute the polarizability of Ne using finite field<a class="headerlink" href="#sample-compute-the-polarizability-of-ne-using-finite-field" title="Permanent link">&para;</a></h2>
<h3 id="sample-job-1-compute-the-atomic-energy">Job 1. Compute the atomic energy<a class="headerlink" href="#sample-job-1-compute-the-atomic-energy" title="Permanent link">&para;</a></h3>
<pre><code>startne
title&quot;Neon&quot;
geometry;ne000;end
basisspherical
nelibraryaug-cc-pvdz
end
scf;thresh1e-10;end
taskscf
</code></pre>
<p>The final energy should be -128.496350.</p>
<h3 id="sample-job-2-compute-the-energy-with-applied-field">Job 2. Compute the energy with applied field<a class="headerlink" href="#sample-job-2-compute-the-energy-with-applied-field" title="Permanent link">&para;</a></h3>
<p>An external field may be simulated with point charges. The charges here
apply a field of magnitude 0.01 atomic units to the atom at the origin.
Since the basis functions have not been reordered by the additional
centers we can also restart from the previous vectors, which is the
default for a restart job.</p>
<pre><code>restartne  
title&quot;Neoninelectricfield&quot;  
geometryunitsatomic  
bq100100charge50  
ne000  
bq200-100charge-50  
end  
taskscf
</code></pre>
<p>The final energy should be -128.496441, which together with the previous
field-free result yields an estimate for the polarizability of 1.83
atomic units. Note that by <a href="#geometry">default</a> NWChem does
not include the interaction between the two point charges in the total
energy.</p>
<h2 id="sample-scf-energy-of-h2co-using-ecps-for-c-and-o">SCF energy of H<sub>2</sub>CO using ECPs for C and O<a class="headerlink" href="#sample-scf-energy-of-h2co-using-ecps-for-c-and-o" title="Permanent link">&para;</a></h2>
<p>The following will compute the SCF energy for formaldehyde with ECPs on
the Carbon and Oxygen centers.</p>
<pre><code>title&quot;formaldehydeECPdeck&quot;  

startecpchho  

geometryunitsau  
C0.0000000.000000-1.025176  
O0.0000000.0000001.280289  
H0.0000001.767475-2.045628  
H0.000000-1.767475-2.045628  
end  

basis  
CSP  
0.1675097360D+02-0.7812840500D-010.3088908800D-01  
0.2888377460D+01-0.3741108860D+000.2645728130D+00  
0.6904575040D+000.1229059640D+010.8225024920D+00  
CSP  
0.1813976910D+000.1000000000D+010.1000000000D+01  
CD  
0.8000000000D+000.1000000000D+01  
CF  
0.1000000000D+010.1000000000D+01  
OSP  
0.1842936330D+02-0.1218775590D+000.5975796600D-01  
0.4047420810D+01-0.1962142380D+000.3267825930D+00  
0.1093836980D+010.1156987900D+010.7484058930D+00  
OSP  
0.2906290230D+000.1000000000D+010.1000000000D+01  
OD  
0.8000000000D+000.1000000000D+01  
OF  
0.1100000000D+010.1000000000D+01  
HS  
0.1873113696D+020.3349460434D-01  
0.2825394365D+010.2347269535D+00  
0.6401216923D+000.8137573262D+00  
HS  
0.1612777588D+000.1000000000D+01  
end  

ecp  
Cnelec2  
Cul  
180.0000000-1.60000000  
130.0000000-0.40000000  
20.5498205-0.03990210  
Cs  
00.73747600.63810832  
0135.235483211.00916230  
28.560556920.13797020  
Cp  
210.6863587-3.24684280  
223.49798970.78505765  
Onelec2  
Oul  
180.0000000-1.60000000  
130.0000000-0.40000000  
21.0953760-0.06623814  
Os  
00.92129520.39552179  
028.64819712.51654843  
29.303350017.04478500  
Op  
252.342701927.97790770  
230.7220233-16.49630500  
end  

scf  
vectorsinputhcore  
maxiter20  
end  

taskscf
</code></pre>
<p>This should produce the following output:</p>
<pre><code>FinalRHFresults  
------------------  
  
TotalSCFenergy=-22.507927218024  
Oneelectronenergy=-71.508730162974  
Twoelectronenergy=31.201960019808  
Nuclearrepulsionenergy=17.798842925142
</code></pre>
<h2 id="sample-mp2-optimization-and-ccsdt-on-nitrogen">MP2 optimization and CCSD(T) on nitrogen<a class="headerlink" href="#sample-mp2-optimization-and-ccsdt-on-nitrogen" title="Permanent link">&para;</a></h2>
<p>The following performs an MP2 geometry optimization followed by a
CCSD(T) energy evaluation at the converged geometry. A Dunning
correlation-consistent triple-zeta basis is used. The default of
Cartesian basis functions must be overridden using the keyword spherical
on the BASIS directive. The 1s core orbitals are frozen in both the MP2
and coupled-cluster calculations (note that these must separately
specified). The final MP2 energy is -109.383276, and the CCSD(T) energy
is -109.399662.</p>
<pre><code>startn2  

geometry  
symmetryd2h  
n000.542  
end  

basisspherical  
nlibrarycc-pvtz  
end  

mp2  
freezecore  
end  

taskmp2optimize  

ccsd  
freezecore  
end  

taskccsd(t)
</code></pre></section><section class="print-page" id="geometry-examples" heading-number="19.2"><h1 id="geometry-examples-examples-of-geometries-using-symmetry">Examples of geometries using symmetry<a class="headerlink" href="#geometry-examples-examples-of-geometries-using-symmetry" title="Permanent link">&para;</a></h1>
<p>Below are examples of the use of the <a href="#symmetry----symmetry-group-input">SYMMETRY directive</a> in the compound
<a href="#geometry">GEOMETRY directive</a>. The <em>z</em> axis is
always the primary rotation axis. When in doubt about which axes and
planes are used for the group elements, the keyword <code>print</code> may be added
to the <code>SYMMETRY</code> directive to obtain this information.</p>
<h2 id="geometry-examples-cs-methanol"><em>C<sub>s</sub></em> &nbsp; methanol<a class="headerlink" href="#geometry-examples-cs-methanol" title="Permanent link">&para;</a></h2>
<p>The <em>&sigma;<sub>h</sub></em> plane is the <em>xy</em> plane.</p>
<pre><code>geometry units angstroms
   C    0.11931097    -0.66334875     0.00000000
   H    1.20599017    -0.87824237     0.00000000
   H   -0.32267592    -1.15740001     0.89812652
   O   -0.01716588     0.78143468     0.00000000
   H   -1.04379735     0.88169812     0.00000000
   symmetry cs
 end
</code></pre>
<h2 id="geometry-examples-c2v-water"><em>C<sub>2v</sub></em> &nbsp; water<a class="headerlink" href="#geometry-examples-c2v-water" title="Permanent link">&para;</a></h2>
<p>The z axis is the <em>C<sub>2</sub></em> axis and the <em>&sigma;<sub>v</sub></em> may be either the <em>xz</em>
or the <em>yz</em> planes.</p>
<pre><code> geometry units au
   O     0.00000000    0.00000000    0.00000000
   H     0.00000000    1.43042809   -1.10715266
   symmetry group c2v
 end
</code></pre>
<h2 id="geometry-examples-d2h-acetylene"><em>D<sub>2h</sub></em> &nbsp; acetylene<a class="headerlink" href="#geometry-examples-d2h-acetylene" title="Permanent link">&para;</a></h2>
<p>Although acetylene has symmetry <em>D<sub>&infin;h</sub></em> the subgroup <em>D<sub>2h</sub></em> 
includes all operations that interchange equivalent atoms which is what
determines how much speedup you gain from using symmetry in building a
Fock matrix.</p>
<p>The <em>C<sub>2</sub></em> axes are the x, y, and z axes. The  planes are the xy, xz
and yz planes. Generally, the unique atoms are placed to use the z as
the primary rotational axis and use the xz or yz planes as the  plane.</p>
<pre><code> geometry units au
   symmetry group d2h
   C      0.000000000    0.000000000   -1.115108538
   H      0.000000000    0.000000000   -3.106737425
 end
</code></pre>
<h2 id="geometry-examples-d2h-ethylene"><em>D<sub>2h</sub></em> &nbsp; ethylene<a class="headerlink" href="#geometry-examples-d2h-ethylene" title="Permanent link">&para;</a></h2>
<p>The <em>C<sub>2</sub></em> axes are the x, y, and z axes. The  planes are the xy, xz
and yz planes. Generally, the unique atoms are placed to use the z as
the primary rotational axis and use the xz or yz planes as the  plane.</p>
<pre><code> geometry units angstroms
   C 0 0 0.659250 
   H 0 0.916366 1.224352 
   symmetry d2h
 end
</code></pre>
<h2 id="geometry-examples-td-methane"><em>T<sub>d</sub></em> &nbsp; methane<a class="headerlink" href="#geometry-examples-td-methane" title="Permanent link">&para;</a></h2>
<p>For ease of use, the primary <em>C<sub>3</sub></em> axis should be the x=y=z axis. The
3 <em>C<sub>2</sub></em> axes are the x, y, and z.</p>
<pre><code> geometry units au
   c   0.0000000      0.0000000      0.0000000
   h   1.1828637      1.1828637      1.1828637
   symmetry group Td
 end
</code></pre>
<h2 id="geometry-examples-ih-buckminsterfullerene"><em>I<sub>h</sub></em> &nbsp; buckminsterfullerene<a class="headerlink" href="#geometry-examples-ih-buckminsterfullerene" title="Permanent link">&para;</a></h2>
<p>One of the <em>C<sub>5</sub></em> axes is the z axis and the point of inversion is the
origin.</p>
<pre><code> geometry units angstroms # Bonds = 1.4445, 1.3945
   symmetry group Ih
   c   -1.2287651   0.0   3.3143121
 end
</code></pre>
<h2 id="geometry-examples-s4-porphyrin"><em>S<sub>4</sub></em> &nbsp; porphyrin<a class="headerlink" href="#geometry-examples-s4-porphyrin" title="Permanent link">&para;</a></h2>
<p>The <em>S<sub>4</sub></em> and <em>C<sub>2</sub></em> rotation axis is the z axis. The reflection
plane for the <em>S<sub>4</sub></em> operation is the xy plane.</p>
<pre><code> geometry units angstroms
    symmetry group s4

    fe                0.000  0.000  0.000         
    h                 2.242  6.496 -3.320   
    h                 1.542  4.304 -2.811
    c                 1.947  6.284 -2.433
    c                 1.568  4.987 -2.084
    h                 2.252  8.213 -1.695
    c                 1.993  7.278 -1.458
    h                 5.474 -1.041 -1.143
    c                 1.234  4.676 -0.765
    h                 7.738 -1.714 -0.606
    c                 0.857  3.276 -0.417
    h                 1.380 -4.889 -0.413
    c                 1.875  2.341 -0.234
    h                 3.629  3.659 -0.234
    c                 0.493 -2.964 -0.229
    c                 1.551 -3.933 -0.221
    c                 5.678 -1.273 -0.198
    c                 1.656  6.974 -0.144
    c                 3.261  2.696 -0.100
    n                 1.702  0.990 -0.035
 end
</code></pre>
<h2 id="geometry-examples-d3h-iron-penta-carbonyl"><em>D<sub>3h</sub></em> &nbsp; iron penta-carbonyl<a class="headerlink" href="#geometry-examples-d3h-iron-penta-carbonyl" title="Permanent link">&para;</a></h2>
<p>The <em>C<sub>3</sub></em> axis is the z axis. The <em>&sigma;<sub>h</sub></em> plane is the xy plane.
One of the perpendicular <em>C<sub>2</sub></em> axes is the x=y axis. One of the
<em>&sigma;<sub>v</sub></em> planes is the plane containing the x=y axis and the z axis.
(The other axes and planes are generated by the <em>C<sub>3</sub></em> operation.)</p>
<pre><code> geometry units au
   symmetry group d3h

   fe        0.0         0.0         0.0

   c         0.0         0.0         3.414358
   o         0.0         0.0         5.591323

   c         2.4417087   2.4417087   0.0
   o         3.9810552   3.9810552   0.0
 end
</code></pre>
<h2 id="geometry-examples-d3d-sodium-crown-ether"><em>D<sub>3d</sub></em> &nbsp; sodium crown ether<a class="headerlink" href="#geometry-examples-d3d-sodium-crown-ether" title="Permanent link">&para;</a></h2>
<p>The <em>C<sub>3</sub></em> axis is the z axis. The point of inversion is the origin.
One of the perpendicular <em>C<sub>2</sub></em> axes is the x=y axis. One of the
<em>&sigma;<sub>d</sub></em> planes is the plane containing the -x=y axis and the z
axis.</p>
<p>Note that the oxygen atom is rotated in the x-y plane 15 degrees away
from the y-axis so that it lies in a mirror plane. There is a total of
six atoms generated from the unique oxygen, in contrast to twelve from
each of the carbon and hydrogen atoms.</p>
<pre><code> geometry units au
   symmetry D3d

  NA     .0000000000    .0000000000   .0000000000
  O     1.3384771885   4.9952647969   .1544089284
  H     6.7342048019  -0.6723850379  2.6581562148
  C     6.7599180056  -0.4844977035   .6136583870
  H     8.6497577017   0.0709194071   .0345361934

 end
</code></pre>
<h2 id="geometry-examples-c3v-ammonia"><em>C<sub>3v</sub></em> &nbsp; ammonia<a class="headerlink" href="#geometry-examples-c3v-ammonia" title="Permanent link">&para;</a></h2>
<p>The <em>C<sub>3</sub></em> axis is the z axis. One of the <em>&sigma;<sub>v</sub></em> planes is the
plane containing the x=y axis and the z axis.</p>
<pre><code> geometry units angstroms
   N 0     0     -0.055 
   H 0.665 0.665 -0.481 
   symmetry c3v
 end
</code></pre>
<h2 id="geometry-examples-d6h-benzene"><em>D<sub>6h</sub></em> benzene<a class="headerlink" href="#geometry-examples-d6h-benzene" title="Permanent link">&para;</a></h2>
<p>The <em>C<sub>6</sub></em> axis is the z axis. The point of inversion is the origin.
One of the 6 perpendicular <em>C<sub>2</sub>&lsquo;</em> axes is the x=y axis. (-x=y works
as a <em>C<sub>2</sub>&lsquo;&rsquo;</em> axis.) The <em>&sigma;<sub>h</sub></em> plane is the xy plane. The
<em>&sigma;<sub>d</sub></em> planes contain the <em>C<sub>2</sub>&lsquo;&rsquo;</em> axis and the z axis. The
<em>&sigma;<sub>v</sub></em> planes contain the <em>C<sub>2</sub>&lsquo;</em> axis and the z axis.</p>
<pre><code> geometry units au
   C 1.855 1.855 0 
   H 3.289 3.289 0 
   symmetry D6h
 end
</code></pre>
<h2 id="geometry-examples-c3h-h3bo3"><em>C<sub>3h</sub></em> &nbsp; H<sub>3</sub>BO<sub>3</sub><a class="headerlink" href="#geometry-examples-c3h-h3bo3" title="Permanent link">&para;</a></h2>
<p>The <em>C<sub>3</sub></em> axis is the z axis. The <em>&sigma;<sub>h</sub></em> plane is the xy plane.</p>
<pre><code> geometry units au
   b 0 0 0 
   o 2.27238285 1.19464491 0.00000000 
   h 2.10895420 2.97347707 0.00000000 
  symmetry C3h
 end
</code></pre>
<h2 id="geometry-examples-d5d-ferrocene"><em>D<sub>5d</sub></em> &nbsp; ferrocene<a class="headerlink" href="#geometry-examples-d5d-ferrocene" title="Permanent link">&para;</a></h2>
<p>The <em>C<sub>5</sub></em> axis is the z axis. The center of inversion is the origin.
One of the perpendicular <em>C<sub>2</sub></em> axes is the x axis. One of the
<em>&sigma;<sub>d</sub></em> planes is the yz plane.</p>
<pre><code> geometry units angstroms
   symmetry d5d

   fe 0 0     0 
   c  0 1.194 1.789 
   h  0 2.256 1.789 
 end
</code></pre>
<h2 id="geometry-examples-c4v-sf5cl"><em>C<sub>4v</sub></em>  &nbsp; SF<sub>5</sub>Cl<a class="headerlink" href="#geometry-examples-c4v-sf5cl" title="Permanent link">&para;</a></h2>
<p>The <em>C<sub>4</sub></em> axis is the z axis. The <em>&sigma;<sub>v</sub></em> planes are the yz and
the xz planes. The <em>&sigma;<sub>d</sub></em> planes are: 1) the plane containing the
x=y axis and the z axis and 2) the plane containing the -x=y axis and
the z axis.</p>
<pre><code> geometry units au
   S  0.00000000 0.00000000 -0.14917600 
   Cl 0.00000000 0.00000000  4.03279700 
   F  3.13694200 0.00000000 -0.15321800 
   F  0.00000000 0.00000000 -3.27074500 

   symmetry C4v
 end
</code></pre>
<h2 id="geometry-examples-c2h-trans-dichloroethylene"><em>C<sub>2h</sub></em> &nbsp; trans-dichloroethylene<a class="headerlink" href="#geometry-examples-c2h-trans-dichloroethylene" title="Permanent link">&para;</a></h2>
<p>The <em>C<sub>2</sub></em> axis is the z axis. The origin is the inversion center. The
<em>&sigma;<sub>h</sub></em> plane is the xy plane.</p>
<pre><code> geometry units angstroms
   C  0.65051239 -0.08305064 0 
   Cl 1.75249381  1.30491767 0 
   H  1.14820954 -1.04789741 0 
   symmetry C2h
 end
</code></pre>
<h2 id="geometry-examples-d2d-ch2cch2"><em>D<sub>2d</sub></em>  &nbsp;   CH<sub>2</sub>CCH<sub>2</sub><a class="headerlink" href="#geometry-examples-d2d-ch2cch2" title="Permanent link">&para;</a></h2>
<p>The <em>C<sub>2</sub></em> axis is the z axis (z is also the <em>S<sub>4</sub></em> axis). The x and y
axes are the perpendicular <em>C<sub>2</sub>&rsquo;s</em>. The <em>&sigma;<sub>d</sub></em> planes are: 1)
the plane containing the x=y axis and the z axis and 2) the plane
containing the -x=y axis and the z axis.</p>
<pre><code> geometry units angstroms
   symmetry d2d
   c 0     0     0 
   c 0     0     1.300 
   h 0.656 0.656 1.857 
 end
</code></pre>
<h2 id="geometry-examples-d5h-cyclopentadiene-anion"><em>D<sub>5h</sub></em> &nbsp; cyclopentadiene anion<a class="headerlink" href="#geometry-examples-d5h-cyclopentadiene-anion" title="Permanent link">&para;</a></h2>
<p>The <em>C<sub>5</sub></em> axis is the z axis (z is also the <em>S<sub>5</sub></em> axis). The y axis
is one of the perpendicular <em>C<sub>2</sub></em> axes. The <em>&sigma;<sub>h</sub></em> plane is the
xy plane and one of the <em>&sigma;<sub>d</sub></em> planes is the yz plane.</p>
<pre><code> charge -1
 geometry units angstroms
   symmetry d5h
   c 0 1.1853 0 
   h 0 2.2654 0 
 end
</code></pre>
<h2 id="geometry-examples-d4h-gold-tetrachloride"><em>D<sub>4h</sub></em> &nbsp; gold tetrachloride<a class="headerlink" href="#geometry-examples-d4h-gold-tetrachloride" title="Permanent link">&para;</a></h2>
<p>The <em>C<sub>4</sub></em> axis is the z axis (z is also the <em>S<sub>4</sub></em> axis). The
<em>C<sub>2</sub>&lsquo;</em> axes are the x and y axes and the <em>C<sub>2</sub>&lsquo;&rsquo;</em> axes are the
x=y axis and the x=-y axis. The inversion center is the origin. The
<em>&sigma;<sub>h</sub></em> plane is the xy plane. The <em>&sigma;<sub>v</sub></em> planes are the xz
and yz planes and the <em>&sigma;<sub>d</sub></em> planes are 1) the plane containing
the x=-y axis and the z axis and 2) the plane containing the x=y axis
and the z axis.</p>
<pre><code> geometry units au
   Au 0 0     0 
   Cl 0 4.033 0
   symmetry D4h
 end
</code></pre></section></section><section class="print-page" id="benchmarks" heading-number="20"><h1 id="benchmarks-benchmarks-performed-with-nwchem">Benchmarks performed with NWChem<a class="headerlink" href="#benchmarks-benchmarks-performed-with-nwchem" title="Permanent link">&para;</a></h1>
<p>This page contains a suite of benchmarks performed with NWChem. The
benchmarks include a variety of computational chemistry methods on a
variety of high performance computing platforms. The list of benchmarks
available will evolve continuously as new data becomes available. If you
have benchmark information you would like to add for your computing
system, please contact one of the developers.</p>
<h1 id="benchmarks-hybrid-density-functional-calculation-on-the-c240-buckyball">Hybrid density functional calculation on the C<sub>240</sub> Buckyball<a class="headerlink" href="#benchmarks-hybrid-density-functional-calculation-on-the-c240-buckyball" title="Permanent link">&para;</a></h1>
<p>Performance of the Gaussian basis set DFT module in NWChem. This
calculation involved performing a PBE0 calculation (in direct mode) on
the on C<sub>240</sub> system with the 6-31G* basis set (3600 basis
functions) without symmetry. These calculations were performed on the
Cascade supercomputer located at PNNL. <a href="c240_631gs.nw">Input</a> and
<a href="c240_631gs.output">output</a> files are available.</p>
<p><img alt="c240_web10.png" src="c240_web10.png" /></p>
<h1 id="benchmarks-parallel-performance-of-ab-initio-molecular-dynamics-using-plane-waves">Parallel performance of <em>Ab initio</em> Molecular Dynamics using plane waves<a class="headerlink" href="#benchmarks-parallel-performance-of-ab-initio-molecular-dynamics-using-plane-waves" title="Permanent link">&para;</a></h1>
<p>AIMD Parallel timings for <span class="arithmatex">\(UO_2^{2+}\)</span>+122<span class="arithmatex">\(H_2\)</span>O.
These calculations were performed on the Franklin Cray-XT4 computer
system at NERSC.<br />
<img alt="PW" src="PWScaling.png" />  </p>
<p>AIMD and AIMD/MM Parallel Timings for <span class="arithmatex">\(Zn^{2+}\)</span>+64<span class="arithmatex">\(H_2\)</span>O (unit
cell parameters SC=12.4 Angs. and cutoff energy =100Ry). These
calculations were performed on the Chinook HP computer system at MSCF
EMSL, PNNL.<br />
<img alt="PWMD" src="PWMDScaling.png" /><br />
Exact exchange timings  80 atom cell of hematite (cutoff energy=100Ry). These
calculations were performed on the Franklin Cray-XT4 computer system at
NERSC.<br />
<img alt="" src="PWEXScaling.png" /><br />
Exact
exchange timings  576 atom cell of water (cutoff energy=100Ry). These
calculations were performed on the Hopper Cray-XE6 computer system at
NERSC.<br />
<img alt="" src="water192b.png" />  </p>
<h1 id="benchmarks-parallel-performance-of-the-cr-eomccsdt-method-triples-part">Parallel performance of the CR-EOMCCSD(T) method (triples part)<a class="headerlink" href="#benchmarks-parallel-performance-of-the-cr-eomccsdt-method-triples-part" title="Permanent link">&para;</a></h1>
<p>An example of the scalability of the triples part of the <a href="http://dx.doi.org/10.1063/1.3385315">CR-EOMCCSD(T)
approach</a> for Green Fluorescent
Protein Chromophore (GFPC) described by cc-pVTZ basis set (648 basis
functions) as obtained from NWChem. Timings were determined from
calculations on the Franklin Cray-XT4 computer system at NERSC. See the
<a href="input_gfpc.nw"> input file</a> for details.</p>
<p><img alt=" " src="creomccsd_t.png" /></p>
<p>And more recent scalability test of the CR-EOMCCSD(T) formalism (Jaguar
Cray XT5 at ORNL, see K. Kowalski, S. Krishnamoorthy, R.M. Olson, V.
Tipparaju, E. Apr , SC2011, for details).</p>
<p><img alt="curve_bg_lumo.png" src="curve_bg_lumo.png" /></p>
<h1 id="benchmarks-parallel-performance-of-the-multireference-coupled-cluster-mrcc-methods">Parallel performance of the multireference coupled cluster (MRCC) methods<a class="headerlink" href="#benchmarks-parallel-performance-of-the-multireference-coupled-cluster-mrcc-methods" title="Permanent link">&para;</a></h1>
<p>In collaboration with Dr. Jiri Pittner&rsquo;s group from Heyrovsky Institute
of Physical Chemistry implementations of two variants of state-specific
MRCC approaches have been developed. During his internship at PNNL Jirka
Brabec, using novel processor-group-based algorithms, implemented
Brillouin-Wigner and Mukherjee MRCC models with singles and doubles. The
scalabililty tests for the Brillouin-Wigner MRCCSD approach have been
performed on Jaguar XT5 system at ORNL for <span class="arithmatex">\(\beta\)</span>-carotene in 6-31
basis set (472 orbitals, 216 correlated electrons, 20 reference
functions; see J.Brabec, J. Pittner, H.J.J. van Dam, E. Apr, K.
Kowalski, JCTC 2012, 8(2), pp 487497). 
The <a href="betacarotene.nw">input file</a> and output files for runs
at <a href="betacarotene.out.6000procs">6000 cores</a>,
at <a href="betacarotene.out.12000procs">12000 cores</a>
and at <a href="betacarotene.out.24000procs">24000 cores</a>
are available.  </p>
<p><img alt="mrccsd_scalability.png" src="mrccsd_scalability.png" /></p>
<p>Former PNNL postdoctoral
fellow Dr. Kiran Bhaskaran Nair developed perturbative MRCCSD(T)
approaches, which accounts for the effect of triple excitations.
Scaling of the triples part of the BW-MRCCSD(T) method for
<span class="arithmatex">\(\beta\)</span>&rdquo; -carotene in 6-31 basis set (JCP 137, 094112 (2012)). The
scalability tests of the BW-MRCCSD(T) implementation of NWChem have been
performed on the Jaguar Cray-XK6 computer system of the National Center
for Computational Sciences at Oak Ridge National Laboratory.</p>
<p><img alt="mrccsd_t_titan.png" src="mrccsd_t_titan.png" /></p>
<h1 id="benchmarks-timings-of-ccsdeomccsd-for-the-oligoporphyrin-dimer">Timings of CCSD/EOMCCSD for the oligoporphyrin dimer<a class="headerlink" href="#benchmarks-timings-of-ccsdeomccsd-for-the-oligoporphyrin-dimer" title="Permanent link">&para;</a></h1>
<p>CCSD/EOMCCSD timings for oligoporphyrin dimer (942 basis functions, 270
correlated electrons, D2h symmetry, excited-state calculations were
performed for state of b1g symmetry, in all test calculation convergence
threshold was relaxed, 1024 cores were used). See the <a href="input_p2ta.nw"> input
file</a> for
details.</p>
<pre><code>--------------------------------------------------------
Iter          Residuum       Correlation     Cpu    Wall
 --------------------------------------------------------
   1   0.7187071521175  -7.9406033677717   640.9   807.7
   ......
 MICROCYCLE DIIS UPDATE: 10 5
  11   0.0009737920958  -7.9953441809574   691.1   822.2
 --------------------------------------------------------
 Iterations converged
 CCSD correlation energy / hartree =        -7.995344180957357
 CCSD total energy / hartree       =     -2418.570838364838890

 EOM-CCSD right-hand side iterations
 --------------------------------------------------------------
      Residuum       Omega / hartree  Omega / eV    Cpu    Wall
 --------------------------------------------------------------
......
Iteration   2 using    6 trial vectors
  0.1584284659595   0.0882389635508    2.40111   865.3  1041.2
Iteration   3 using    7 trial vectors
  0.0575982107592   0.0810948687618    2.20670   918.0  1042.2
</code></pre>
<h1 id="benchmarks-performance-tests-of-the-gpu-implementation-of-non-iterative-part-of-the-ccsdt-approach">Performance tests of the GPU implementation of non-iterative part of the CCSD(T) approach<a class="headerlink" href="#benchmarks-performance-tests-of-the-gpu-implementation-of-non-iterative-part-of-the-ccsdt-approach" title="Permanent link">&para;</a></h1>
<p>Recent tests of the GPU CCSD(T) implementation performed on Titan Cray
XK7 <a href="https://www.olcf.ornl.gov/olcf-resources/compute-systems/titan">1</a>
system at ORNL (C<sub>22</sub>H<sub>14</sub>, 378 basis set functions,
C1 symmetry; 98 nodes: 8 cores per node + 1GPU)</p>
<p><strong>Using 8 CPU cores</strong></p>
<pre><code>Using CUDA CCSD(T) code 
Using 0 device(s) per node 
CCSD[T] correction energy / hartree = -0.150973754992986 
CCSD[T] correlation energy / hartree = -3.067917061062492 
CCSD[T] total energy / hartree = -844.403376796441080 
CCSD(T) correction energy / hartree = -0.147996460406684 
CCSD(T) correlation energy / hartree = -3.064939766476190 
CCSD(T) total energy / hartree = -844.400399501854849 
Cpu &amp; wall time / sec 9229.9 9240.3 
</code></pre>
<p><strong>Using 7 CPU cores and one GPU</strong></p>
<pre><code>Using CUDA CCSD(T) code 
Using 1 device(s) per node  
CCSD[T] correction energy / hartree = -0.150973754993019 
CCSD[T] correlation energy / hartree = -3.067917061062597 
CCSD[T] total energy / hartree = -844.403376796441307 
CCSD(T) correction energy / hartree = -0.147996460406693 
CCSD(T) correlation energy / hartree = -3.064939766476270 
CCSD(T) total energy / hartree = -844.400399501854963 
Cpu &amp; wall time / sec 1468.0 1630.7 
</code></pre>
<p><strong>Using 1 CPU core and one GPU</strong></p>
<pre><code>Using CUDA CCSD(T) code
Using 1 device(s) per node
CCSD[T] correction energy / hartree = -0.150973754993069
CCSD[T] correlation energy / hartree = -3.067917061063028
CCSD[T] total energy / hartree =  -844.***************
CCSD(T) correction energy / hartree = -0.147996460406749
CCSD(T) correlation energy / hartree = -3.064939766476708
CCSD(T) total energy / hartree = -844.400399501861216
Cpu &amp; wall time / sec 1410.9 1756.5
</code></pre>
<p>Without GPU 9240.3 sec. With GPU 1630.7 sec.</p>
<p>Next release: GPU implementation of non-iterative part of the MRCCSD(T)
approach (K. Bhaskarsan-Nair, W. Ma, S. Krishnamoorthy, O. Villa, H. van
Dam, E. Apr, K. Kowalski, J. Chem. Theory Comput. 9, 1949 (2013))</p>
<p><img alt="GPU_MRCCSD_T.png" src="GPU_MRCCSD_T.png" /></p>
<h1 id="benchmarks-performance-tests-of-the-xeon-phi-implementation-of-non-iterative-part-of-the-ccsdt-approach">Performance tests of the Xeon Phi implementation of non-iterative part of the CCSD(T) approach<a class="headerlink" href="#benchmarks-performance-tests-of-the-xeon-phi-implementation-of-non-iterative-part-of-the-ccsdt-approach" title="Permanent link">&para;</a></h1>
<p>Tests  of the Xeon Phi CCSD(T) implementation
performed on the EMSL <a href="https://web.archive.org/web/20170731230327/http://www.emsl.pnl.gov/emslweb/instruments/computing-cascade-atipa-1440-intel-xeon-phi-node-fdr-infiniband-linux-cluster">cascade</a>
system at PNNL </p>
<p><img alt="Triplet_updated_feb2015.png" src="Triplet_updated_feb2015.png" /></p>
<p>Apr, E.; Klemm, M.; Kowalski, K., &ldquo;Efficient Implementation of
Many-Body Quantum Chemical Methods on the Intel Xeon Phi Coprocessor,&rdquo;
High Performance Computing, Networking, Storage and Analysis, SC14:
International Conference for , vol., no., pp.674-684, 16-21 Nov. 2014 <a href="http://dx.doi.org/10.1109/SC.2014.60">http://dx.doi.org/10.1109/SC.2014.60</a></p>
<p>(Triplet state of
Si<sub>4</sub>C<sub>3</sub>N<sub>2</sub>H<sub>12</sub>, 706 basis set
functions, C1
symmetry)</p>
<h1 id="benchmarks-non-iterative-part-of-the-ccsdt-approach-comparing-xeon-phi-and-nvidia-k20x-performance">Non-iterative part of the CCSD(T) approach: Comparing Xeon Phi and NVidia K20X performance<a class="headerlink" href="#benchmarks-non-iterative-part-of-the-ccsdt-approach-comparing-xeon-phi-and-nvidia-k20x-performance" title="Permanent link">&para;</a></h1>
<p>Wall time to solution (in seconds) of non-iterative triples part of the
single-reference CCSD(T) approach for the pentacene molecule using Intel
MIC and Nvidia GPU implementations. Tests were performed using 96
compute nodes on the Cascade system at EMSL (Intel Xeon Phi 5110P) and
Titan system at ORNL (NVIDIA Tesla K20X).</p>
<p>(<a href="pentacene_ccsdt.nw"> input file</a>)</p>
<table>
<thead>
<tr>
<th>Tilesize</th>
<th>Intel Xeon Phi 5110P</th>
<th>Nvidia K20X</th>
</tr>
</thead>
<tbody>
<tr>
<td>18</td>
<td>1806.4</td>
<td>1824.9</td>
</tr>
<tr>
<td>21</td>
<td>1652.2</td>
<td>1699.3</td>
</tr>
<tr>
<td>24</td>
<td>1453.3</td>
<td>1554.4</td>
</tr>
</tbody>
</table>
<h1 id="benchmarks-current-developments-for-high-accuracy-alternative-task-schedulers-ats">Current developments for high accuracy: alternative task schedulers (ATS)<a class="headerlink" href="#benchmarks-current-developments-for-high-accuracy-alternative-task-schedulers-ats" title="Permanent link">&para;</a></h1>
<p>Currently various development efforts are underway for high accuracy
methods that will be available in future releases of NWChem. The
examples below shows the first results of the performance of the triples
part of Reg-CCSD(T) on GPGPUs (left two examples) and of using
alternative task schedules for the iterative CCSD and EOMCCSD.</p>
<p><img alt="" src="gpu_scaling_spiro.png" />Scalability of the triples part of
the Reg-CCSD(T) approach for Spiro cation described by the Sadlej&rsquo;s TZ
basis set (POL1). The calculations were performed using Barracuda
cluster at EMSL.
<img alt="" src="gpu_speedup_uracil.png" />
Speedup
of GPU over CPU of the (T) part of the (T) part of the Reg-CCSD(T)
approach as a function of the tile size for the uracil molecule. The
calculations were performed using Barracuda cluster at EMSL.<br />
<img alt="" src="ccsd_eomccsd_new.png" />
*Comparison of the CCSD/EOMCCSD
iteration times for BacterioChlorophyll (BChl, Mg O6 N4 C 36 H38) for
various tile sizes. Calculations were performed for 3-21G basis set (503
basis functions, C1 symmetry, 240 correlated electrons, 1020
cores).</p>
<p><img alt="bchl_6_311G_ccsd.png" src="bchl_6_311G_ccsd.png" />
*Time per CCSD
iteration for BChl in 6-311G basis set (733 basis functions, C1
symmetry, 240 correlated electrons, 1020 cores) as a function of tile
size.
<img alt="" src="eomccsd_scaling_ic.png" />
Scalability of the
CCSD/EOMCCSD codes for BChl in 6-311G basis set (733 basis functions;
tilesize=40, C1 symmetry, 240 correlated electrons).</p>
<p>Other tests:</p>
<p>The impact of the tilesize on the CCSD(ATS) timings: All tests have been
performed for uracil trimer (6-31G* basis set; all core electrons
frozen) on Hopper using 25 nodes (600 cores). One can observe almost
10-fold speedup of the CCSD(ATS) code for tilesize=40 compared to
standard TCE CCSD implementation using tilesize=12.</p>
<p><img alt="uracil_trimer.png" src="uracil_trimer.png" /></p>
<p>Performance tests for water clusters</p>
<p><img alt="CTF.png" src="CTF.png" /></p>
<p>Luciferin (aug-cc-pVDZ basis set; RHF reference; frozen core) - time per
CCSD iteration (<a href="luciferin.nw"> input file</a>)</p>
<pre><code> tilesize = 30 
   256 cores      644 sec.
   512            378 sec.
   664            314 sec.
  1020            278 sec.
  1300            237 sec.
</code></pre>
<pre><code> tilesize = 40
    128             998 sec.
    256             575 sec.
</code></pre>
<p>Sucrose (6-311G** basis set; RHF reference; frozen core) - time per
CCSD iteration (<a href="sucrose.nw"> input file</a>)</p>
<pre><code>tilesize = 40
   256 cores   1486 sec. 
   512          910 sec.
  1024          608 sec.
</code></pre>
<p>Cytosine-OH (POL1; UHF reference; frozen core) - time per EOMCCSD
iteration (<a href="cytosine_oh_pol1.nw"> input file</a>)</p>
<pre><code> tilesize = 30
 256 cores    44.5 sec.
</code></pre>
<pre><code> tilesize = 40
 128 cores    55.6 sec.
</code></pre>
<h1 id="benchmarks-density-functional-calculation-of-a-zeolite-fragment">Density functional calculation of a zeolite fragment<a class="headerlink" href="#benchmarks-density-functional-calculation-of-a-zeolite-fragment" title="Permanent link">&para;</a></h1>
<p>Benchmark results with NWChem 7.0.0 for
LDA calculations (energy plus gradient) on a 533 atoms <a href="benchmarks/siosi8.nw">siosi8</a> zeolite fragment.
The input uses an atomic orbital basis set with 7108 functions
and a charge density fitting basis with 16501 functions.
The input file is available at this <a href="benchmarks/siosi8.nw">link</a>.</p>
<table>
<thead>
<tr>
<th>computer</th>
<th style="text-align: right;"># nodes</th>
<th style="text-align: right;">cores/node</th>
<th style="text-align: right;">total # cores</th>
<th style="text-align: right;">Wall time (seconds)</th>
</tr>
</thead>
<tbody>
<tr>
<td>cascade</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">16</td>
<td style="text-align: right;">144</td>
<td style="text-align: right;">1247</td>
</tr>
<tr>
<td>cascade</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">16</td>
<td style="text-align: right;">320</td>
<td style="text-align: right;">703</td>
</tr>
<tr>
<td>tahoma</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">36</td>
<td style="text-align: right;">144</td>
<td style="text-align: right;">927</td>
</tr>
<tr>
<td>tahoma</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">36</td>
<td style="text-align: right;">324</td>
<td style="text-align: right;">524</td>
</tr>
</tbody>
</table>
<p><br>
Hardware used:</p>
<ul>
<li>EMSL cascade <a href="https://web.archive.org/web/20210331015028/https://www.emsl.pnnl.gov/MSC/UserGuide/cascade/cascade_overview.html#nodes">specifications</a>  </li>
<li>EMSL tahoma <a href="https://www.emsl.pnnl.gov/MSC/UserGuide/tahoma/tahoma_overview.html#nodes">specifications</a></li>
</ul>
<p><img alt="benchmarks/siosi8.png" src="benchmarks/siosi8.png" /></p></section><section class="print-page" id="known-bugs" heading-number="21"><h1 id="known-bugs-known-bugs">Known Bugs<a class="headerlink" href="#known-bugs-known-bugs" title="Permanent link">&para;</a></h1>
<h3 id="known-bugs-how-to-report-bugs">How to report bugs<a class="headerlink" href="#known-bugs-how-to-report-bugs" title="Permanent link">&para;</a></h3>
<p>Use our <a href="https://github.com/nwchemgit/nwchem/issues">Github issue tracker</a> to report new bugs. </p>
<p>Also check there for other issues that may not have made it into these lists. </p>
<p>See the <a href="#faq">FAQ page</a> for solutions to common issues.</p>
<h3 id="known-bugs-bugs-present-in-nwchem-binary-packages">Bugs present in NWChem binary packages<a class="headerlink" href="#known-bugs-bugs-present-in-nwchem-binary-packages" title="Permanent link">&para;</a></h3>
<ul>
<li>The packages shipped with Ubuntu Bionic 18.04 are buggy. Calculations  stop with the error</li>
</ul>
<pre><code>spcart_bra2etran: nbf_xj.ne.nbf_sj  (xj-sj) =                   5
</code></pre>
<p><a href="https://groups.google.com/g/nwchem-forum/c/RA0tYxdfvaw">https://groups.google.com/g/nwchem-forum/c/RA0tYxdfvaw</a>
<a href="https://nwchemgit.github.io/Special_AWCforum/st/id3386/Ubuntu_18.html">https://nwchemgit.github.io/Special_AWCforum/st/id3386/Ubuntu_18.html</a>
<a href="https://bugs.launchpad.net/ubuntu/+source/nwchem/+bug/1675817">https://bugs.launchpad.net/ubuntu/+source/nwchem/+bug/1675817</a></p>
<p>Please use commands to install an updated version (as described
at the <a href="https://github.com/nwchemgit/nwchem/releases/tag/v7.0.0-release">NWChem 7.0.0 release page</a>)</p>
<pre><code>sudo apt -y install curl  python3-dev gfortran  mpi-default-bin mpi-default-dev libopenblas-dev ssh

curl -LJO https://github.com/nwchemgit/nwchem/releases/download/v7.0.0-release/nwchem-data_7.0.0-3_all.ubuntu_bionic.deb
curl -LJO https://github.com/nwchemgit/nwchem/releases/download/v7.0.0-release/nwchem_7.0.0-3_amd64.ubuntu_bionic.deb

sudo dpkg -i nwchem-data_7.0.0-3*_bionic.deb  nwchem_7.0.0-3*_bionic.deb
</code></pre>
<h3 id="known-bugs-known-bugs-for-nwchem-68">Known bugs for NWChem 6.8<a class="headerlink" href="#known-bugs-known-bugs-for-nwchem-68" title="Permanent link">&para;</a></h3>
<ul>
<li><em>AR comppilation failure on Mac OSX</em> <br />
<a href="https://github.com/nwchemgit/nwchem/issues/5">https://github.com/nwchemgit/nwchem/issues/5</a>
Temporary fix: set the env. variable USE_ARUR=n, e.g.</li>
</ul>
<pre><code>make USE_ARUR=n
</code></pre>
<p>Fix   available in  the branches master and hotfix/release-6-8</p>
<ul>
<li><em>Moldenfile property bug when symmetry and linear dependencies are present</em> <br />
<a href="https://github.com/nwchemgit/nwchem/issues/7">https://github.com/nwchemgit/nwchem/issues/7</a>
Workaround described in the issue entry</li>
</ul></section><section class="print-page" id="faq" heading-number="22"><h1 id="faq-general-information-about-nwchem">General information about NWChem<a class="headerlink" href="#faq-general-information-about-nwchem" title="Permanent link">&para;</a></h1>
<h2 id="faq-where-is-the-users-manual">Where is the User&rsquo;s Manual?<a class="headerlink" href="#faq-where-is-the-users-manual" title="Permanent link">&para;</a></h2>
<p>The NWChem User&rsquo;s Manual is now at
<a href="https://nwchemgit.github.io/Home.html">https://nwchemgit.github.io/Home.html</a></p>
<h2 id="faq-where-do-i-go-for-help-with-a-global-arrays-problem">Where do I go for help with a Global Arrays problem?<a class="headerlink" href="#faq-where-do-i-go-for-help-with-a-global-arrays-problem" title="Permanent link">&para;</a></h2>
<p>If you have problems with compiling the tools directory, please visit
the Global Arrays Google group at
<a href="http://groups.google.com/g/hpctools/">http://groups.google.com/g/hpctools/</a> or visit the Global Arrays
website at <a href="http://hpc.pnl.gov/globalarrays/">http://hpc.pnl.gov/globalarrays/</a></p>
<h2 id="faq-where-do-i-go-for-help-with-nwchem-problems">Where do I go for help with NWChem problems?<a class="headerlink" href="#faq-where-do-i-go-for-help-with-nwchem-problems" title="Permanent link">&para;</a></h2>
<p>Please post your NWChem issue to the NWChem forum hosted on Google Groups at
<a href="https://groups.google.com/g/nwchem-forum">https://groups.google.com/g/nwchem-forum</a></p>
<h2 id="faq-where-do-i-find-the-installation-instructions">Where do I find the installation instructions?<a class="headerlink" href="#faq-where-do-i-find-the-installation-instructions" title="Permanent link">&para;</a></h2>
<p>For updated instructions for compiling NWChem please visit the following
URL <a href="https://nwchemgit.github.io/Compiling-NWChem.html">https://nwchemgit.github.io/Compiling-NWChem.html</a></p>
<h2 id="faq-installation-problem-for-the-tools-directory">Installation Problem for the tools directory<a class="headerlink" href="#faq-installation-problem-for-the-tools-directory" title="Permanent link">&para;</a></h2>
<p>When compiling the tools directory, you might see the compilation
stopping with the message</p>
<pre><code>configure: error: could not compile simple C MPI program
</code></pre>
<p>This is most likely due to incorrect settings for the
<code>MPI_LIB</code>, <code>MPI_INCLUDE</code> and <code>LIBMPI</code> environment variables. The
suggested course of action is to unset all of the
three variables above and point your <code>PATH</code> env. variable to the
location of <code>mpif90</code>. If bash is your shell choice, this can be
accomplished by typing</p>
<pre><code>unset MPI_LIB
unset MPI_INCLUDE
unset LIBMPI
export PATH=&quot;directory where mpif90 is located&quot;:$PATH
</code></pre>
<h2 id="faq-what-are-armci-and-armci_network">What are ARMCI and ARMCI_NETWORK?<a class="headerlink" href="#faq-what-are-armci-and-armci_network" title="Permanent link">&para;</a></h2>
<p>ARMCI is a library used by Global Arrays (both ARMCI and GA source code
is located in NWChem&rsquo;s tools directory). More information can be found
at the following URL <a href="http://hpc.pnl.gov/armci">http://hpc.pnl.gov/armci</a><br />
If your installation
uses a fast network and you are aiming to get optimal communication
performance, you might want to assign a non-default value to
<code>ARMCI_NETWORK</code>.<br />
The following links contained useful information about
<code>ARMCI_NETWORK</code>:  </p>
<ul>
<li><a href="#armci">Choosing the ARMCI library</a></li>
<li><a href="#compiling-nwchem-setting-up-the-proper-environment-variables">Choosing the proper environment variables when compiling NWChem</a>  </li>
</ul>
<h2 id="faq-input-problem-no-output">Input Problem: no output<a class="headerlink" href="#faq-input-problem-no-output" title="Permanent link">&para;</a></h2>
<p>You might encounter the following error message:</p>
<pre><code>! warning: processed input with no task
</code></pre>
<p>Have you used emacs to create your input file? Emacs usually does not
put and an end-of-line as a last character of the file, therefore the
NWChem input parser ignores the last line of your input (the one
containing the task directive). To fix the problem, add one more blank
line after the task line and your task directive will be executed.</p>
<h2 id="faq-input-problem-autoz-fails-to-generate-valid-internal-coordinates">Input problem: AUTOZ fails to generate valid internal coordinates<a class="headerlink" href="#faq-input-problem-autoz-fails-to-generate-valid-internal-coordinates" title="Permanent link">&para;</a></h2>
<p>If <a href="#keywords-for-the-geometry-directive-autoznoautoz">AUTOZ</a> fails, NWChem will default to using Cartesian coordinates (and
ignore any zcoord data) so you don&rsquo;t have to do anything unless you
really need to use internal coordinates. An exception are certain cases
where we have a molecule that contains a linear chain of 4 or more
atoms, in which case the code will fail (see item 2. for work arounds).
For small systems you can easily construct a Z-matrix, but for larger
systems this can be quite hard.</p>
<p>First check your input. Are you using the correct units? The default is
Angstroms. If you input atomic units but did not tell NWChem, then it&rsquo;s
no wonder things are breaking. Also, is the geometry physically
sensible? If atoms are too close to each other you&rsquo;ll get many
unphysical bonds, whereas if they are too far apart AUTOZ will not be
able to figure out how to connect things.</p>
<p>Once the obvious has been checked, there are several possible modes of
failure, some of which may be worked around in the input.</p>
<ol>
<li>
<p>Strictly linear molecules with 3 or more atoms. AUTOZ does not
generate linear bend coordinates, but, just as in a real Z-matrix, you
can specify a dummy center that is not co-linear. There are two relevant
tips:</p>
</li>
<li>
<p>constrain the dummy center to be not co-linear otherwise the
center could become co-linear. Also, the inevitable small forces on the
dummy center can confuse the optimizer.</p>
</li>
<li>put the dummy center far
enough away so that only one connection is generated.</li>
</ol>
<p>E.g., this input for acetylene will not use internals</p>
<pre><code> geometry
   h  0  0  0
   c  0  0  1
   c  0  0  2.2
   h  0  0  3.2
 end

 but this one will

 geometry
   zcoord
     bond    2 3  3.0  cx constant
     angle 1 2 3 90.0 hcx constant
   end
   h  0  0  0
   c  0  0  1
   x  3  0  1
   c  0  0  2.2
   h  0  0  3.2
 end
</code></pre>
<ol>
<li>
<p>Larger molecules that contain a strictly linear chain of four or
more atoms (that ends in a free atom). For these molecules the autoz
will fail and the code can currently not recover by using cartesians.
One has to explicitly define noautoz in the geometry input to make it
work. If internal coordinates are required one can fix it in the same
manner as described above. However, you can also force a connection to a
real nearby atom.</p>
</li>
<li>
<p>Very highly connected systems generate too many internal coordinates
which can make optimization in redundant internals less efficient than
in Cartesians. For systems such as clusters of atoms or small molecules,
try using a smaller value of the scaling factor for covalent radii</p>
</li>
</ol>
<pre><code>zcoord; cvr_scaling 0.9; end
</code></pre>
<p>In addition to this you can also try specifying a minimal set of bonds
to connect the fragments.</p>
<p>If these together don&rsquo;t work, then you&rsquo;re out of luck. Use either 
Cartesians coordinates (by using the geometry <a href="#keywords-for-the-geometry-directive-autoznoautoz">NOAUTOZ</a> keyword) or
supply your own Z-matrix (using the <a href="#zmatrix">ZMATRIX</a> input option).</p>
<h2 id="faq-how-do-i-restart-a-geometry-optimization">How do I restart a geometry optimization?<a class="headerlink" href="#faq-how-do-i-restart-a-geometry-optimization" title="Permanent link">&para;</a></h2>
<p>If you have saved the restart information that is kept in the permanent
directory, then you can restart a calculation, as long as it did not
crash while writing to the data base.</p>
<p>Following are two input files. The first starts a geometry optimization
for ammonia. If this stops for nearly any reason such as it was
interrupted, ran out of time or disk space, or exceeded the maximum
number of iterations, then it may be restarted with the second job.</p>
<p>The key points are</p>
<ul>
<li>The first job contains a START directive with a name for the
    calculation.</li>
<li>All subsequent jobs should contain a RESTART directive with the same
    name for the calculation.</li>
<li>All jobs must specify the same permanent directory. The default
    permanent directory is the current directory.</li>
<li>If you want to change anything in the restart job, just put the data
    before the task directive. Otherwise, all options will be the same
    as in the original job.</li>
</ul>
<p>Job 1.</p>
<pre><code> start ammonia
 permanent_dir /u/myfiles

 geometry
   zmatrix
     n
     h 1 nh
     h 1 nh 2 hnh
     h 1 nh 2 hnh 3 hnh -1
   variables
     nh 1.
     hnh 115.
   end
 end

 basis
   n library 3-21g; h library 3-21g
 end

 task scf optimize
</code></pre>
<p>Job 2.</p>
<pre><code> restart ammonia
 permanent_dir /u/myfiles

 task scf optimize
</code></pre>
<h2 id="faq-execution-problem-how-do-i-set-the-value-of-armci_default_shmmax">Execution Problem: How do I set the value of ARMCI_DEFAULT_SHMMAX?<a class="headerlink" href="#faq-execution-problem-how-do-i-set-the-value-of-armci_default_shmmax" title="Permanent link">&para;</a></h2>
<p>Some ARMCI_NETWORK values (e.g. OPENIB) depend on the
<code>ARMCI_DEFAULT_SHMMAX</code> value for large allocations of Global memory. We
recommend a value of &ndash; at least &ndash; 2048, e.g. in bash shell parlance</p>
<pre><code>export ARMCI_DEFAULT_SHMMAX=2048
</code></pre>
<p>A value of 2048 for ARMCI_DEFAULT_SHMMAX corresponds to 2048 GBytes,
equal to 2048<em>1024</em>1024=2147483648 bytes. For
ARMCI_DEFAULT_SHMMAX=2048 to work, it is necessary that kernel
parameter <code>kernel.shmmax</code> to be greater than 2147483648. You can check
the current value of <code>kernel.shmmax</code> on your system by typing</p>
<pre><code>sysctl kernel.shmmax
</code></pre>
<p>More detail about kernel.shmmax can be found at this <a href="https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/5/html/tuning_and_optimizing_red_hat_enterprise_linux_for_oracle_9i_and_10g_databases/sect-oracle_9i_and_10g_tuning_guide-setting_shared_memory-setting_shmall_parameter">link</a></p>
<h2 id="faq-wsl-execution-problems">WSL execution problems<a class="headerlink" href="#faq-wsl-execution-problems" title="Permanent link">&para;</a></h2>
<p>NWChem runs on Windows Subsystem for Linux (WSL) can crash with the error message</p>
<pre><code>--------------------------------------------------------------------------
WARNING: Linux kernel CMA support was requested via the
btl_vader_single_copy_mechanism MCA variable, but CMA support is
not available due to restrictive ptrace settings.

The vader shared memory BTL will fall back on another single-copy
mechanism if one is available. This may result in lower performance.

  Local host: hostabc
--------------------------------------------------------------------------
[hostabc:16805] 1 more process has sent help message help-btl-vader.txt / cma-permission-denied
[hostabc:16805] Set MCA parameter &quot;orte_base_help_aggregate&quot; to 0 to see all help / error messages
</code></pre>
<p>The error can be fixed with the following command</p>
<pre><code>echo 0 | sudo tee /proc/sys/kernel/yama/ptrace_scope
</code></pre>
<p>More details at</p>
<ul>
<li><a href="https://github.com/Microsoft/WSL/issues/3397#issuecomment-417876710">https://github.com/Microsoft/WSL/issues/3397#issuecomment-417876710</a>  </li>
<li><a href="https://nwchemgit.github.io/Special_AWCforum/st/id2939/mpirun_nwchem_on_Windows_Subsyst....html">https://nwchemgit.github.io/Special_AWCforum/st/id2939/mpirun_nwchem_on_Windows_Subsyst....html</a>  </li>
</ul>
<h2 id="faq-how-do-i-increase-the-number-of-digits-of-the-s-matrix-printout">How do I increase the number of digits of the S matrix printout<a class="headerlink" href="#faq-how-do-i-increase-the-number-of-digits-of-the-s-matrix-printout" title="Permanent link">&para;</a></h2>
<p>The only way to increase the number of digits of the AO overlap matrix printout is by modify the source code of
the <code>ga_print()</code> function.</p>
<p>For example, in the cagse NWChem 7.0.2, you can do this by editing the C source code in
$NWCHEM_TOP/src/tools/ga-5.7.2/global/src/global.util.c by increaseing the number of digits from 5 to 7</p>
<pre><code>--- global.util.c.org 1969-07-20 15:50:45.000000000 -0700
+++ global.util.c 1969-07-20 15:51:19.000000000 -0700
@@ -122,22 +122,22 @@
             case C_DBL:
               pnga_get(g_a, lo, hi, dbuf, &amp;ld);
               for(jj=0; jj&lt;(jmax-j+1); jj++)
-                fprintf(file,&quot; %11.5f&quot;,dbuf[jj]);
+                fprintf(file,&quot; %11.7f&quot;,dbuf[jj]);
               break;
             case C_DCPL:
               pnga_get(g_a, lo, hi, dbuf, &amp;ld);
               for(jj=0; jj&lt;(jmax-j+1); jj+=2)
-                fprintf(file,&quot; %11.5f,%11.5f&quot;,dbuf[jj], dbuf[jj+1]);
+                fprintf(file,&quot; %11.7f,%11.7f&quot;,dbuf[jj], dbuf[jj+1]);
               break;
             case C_SCPL:
               pnga_get(g_a, lo, hi, dbuf, &amp;ld);
               for(jj=0; jj&lt;(jmax-j+1); jj+=2)
-                fprintf(file,&quot; %11.5f,%11.5f&quot;,dbuf[jj], dbuf[jj+1]);
+                fprintf(file,&quot; %11.7f,%11.7f&quot;,dbuf[jj], dbuf[jj+1]);
               break;
             case C_FLOAT:
               pnga_get(g_a, lo, hi, fbuf, &amp;ld);
               for(jj=0; jj&lt;(jmax-j+1); jj++)
-                fprintf(file,&quot; %11.5f&quot;,fbuf[jj]);
+                fprintf(file,&quot; %11.7f&quot;,fbuf[jj]);
               break;       
             case C_LONG:
               pnga_get(g_a, lo, hi, lbuf, &amp;ld);
@@ -229,22 +229,22 @@
             case C_DBL:
               pnga_get(g_a, lo, hi, dbuf, &amp;ld);
               for(jj=0; jj&lt;(jmax-j+1); jj++)
-                fprintf(file,&quot; %11.5f&quot;,dbuf[jj]);
+                fprintf(file,&quot; %11.7f&quot;,dbuf[jj]);
               break;
             case C_FLOAT:
               pnga_get(g_a, lo, hi, dbuf, &amp;ld);
               for(jj=0; jj&lt;(jmax-j+1); jj++)
-                fprintf(file,&quot; %11.5f&quot;,fbuf[jj]);
+                fprintf(file,&quot; %11.7f&quot;,fbuf[jj]);
               break;     
             case C_DCPL:
               pnga_get(g_a, lo, hi, dbuf, &amp;ld);
               for(jj=0; jj&lt;(jmax-j+1); jj+=2)
-                fprintf(file,&quot; %11.5f,%11.5f&quot;,dbuf[jj], dbuf[jj+1]);
+                fprintf(file,&quot; %11.7f,%11.7f&quot;,dbuf[jj], dbuf[jj+1]);
               break;
             case C_SCPL:
               pnga_get(g_a, lo, hi, dbuf, &amp;ld);
               for(jj=0; jj&lt;(jmax-j+1); jj+=2)
-                fprintf(file,&quot; %11.5f,%11.5f&quot;,dbuf[jj], dbuf[jj+1]);
+                fprintf(file,&quot; %11.7f,%11.7f&quot;,dbuf[jj], dbuf[jj+1]);
               break;
             default: pnga_error(&quot;ga_print: wrong type&quot;,0);
           }
@@ -761,28 +761,28 @@
                             if(ndim &gt; 1)
                                 for(j=0; j&lt;(hip[1]-lop[1]+1); j++)
                                     if((double)dbuf_2d[j*bufsize+i]&lt;100000.0)
-                                        fprintf(file,&quot; %11.5f&quot;,
+                                        fprintf(file,&quot; %11.7f&quot;,
                                                 dbuf_2d[j*bufsize+i]);
                                     else
                                         fprintf(file,&quot; %.5e&quot;,
                                                 dbuf_2d[j*bufsize+i]);
                             else
                                 if((double)dbuf_2d[i]&lt;100000.0)
-                                    fprintf(file,&quot; %11.5f&quot;,dbuf_2d[i]);
+                                    fprintf(file,&quot; %11.7f&quot;,dbuf_2d[i]);
                                 else
                                     fprintf(file,&quot; %.5e&quot;,dbuf_2d[i]);
                             break;
                         case C_FLOAT:
                             if(ndim &gt; 1)
                                 for(j=0; j&lt;(hip[1]-lop[1]+1); j++)
-                                    fprintf(file,&quot; %11.5f&quot;, fbuf_2d[j*bufsize+i]);
-                            else fprintf(file,&quot; %11.5f&quot;, fbuf_2d[i]);
+                                    fprintf(file,&quot; %11.7f&quot;, fbuf_2d[j*bufsize+i]);
+                            else fprintf(file,&quot; %11.7f&quot;, fbuf_2d[i]);
                             break;           
                         case C_DCPL:
                             if(ndim &gt; 1)
                                 for(j=0; j&lt;(hip[1]-lop[1]+1); j++)
                                     if(((double)dcbuf_2d[(j*bufsize+i)*2]&lt;100000.0)&amp;&amp;((double)dcbuf_2d[(j*bufsize+i)*2+1]&lt;100000.0))
-                                        fprintf(file,&quot; %11.5f,%11.5f&quot;,
+                                        fprintf(file,&quot; %11.7f,%11.7f&quot;,
                                                 dcbuf_2d[(j*bufsize+i)*2],
                                                 dcbuf_2d[(j*bufsize+i)*2+1]);
                                     else
@@ -792,7 +792,7 @@
                             else
                                 if(((double)dcbuf_2d[i*2]&lt;100000.0) &amp;&amp;
                                    ((double)dcbuf_2d[i*2+1]&lt;100000.0))
-                                    fprintf(file,&quot; %11.5f,%11.5f&quot;,
+                                    fprintf(file,&quot; %11.7f,%11.7f&quot;,
                                             dcbuf_2d[i*2], dcbuf_2d[i*2+1]);
                                 else
                                     fprintf(file,&quot; %.5e,%.5e&quot;,
@@ -802,7 +802,7 @@
                             if(ndim &gt; 1)
                                 for(j=0; j&lt;(hip[1]-lop[1]+1); j++)
                                     if(((float)fcbuf_2d[(j*bufsize+i)*2]&lt;100000.0)&amp;&amp;((float)fcbuf_2d[(j*bufsize+i)*2+1]&lt;100000.0))
-                                        fprintf(file,&quot; %11.5f,%11.5f&quot;,
+                                        fprintf(file,&quot; %11.7f,%11.7f&quot;,
                                                 fcbuf_2d[(j*bufsize+i)*2],
                                                 fcbuf_2d[(j*bufsize+i)*2+1]);
                                     else
@@ -812,7 +812,7 @@
                             else
                                 if(((float)fcbuf_2d[i*2]&lt;100000.0) &amp;&amp;
                                    ((float)fcbuf_2d[i*2+1]&lt;100000.0))
-                                    fprintf(file,&quot; %11.5f,%11.5f&quot;,
+                                    fprintf(file,&quot; %11.7f,%11.7f&quot;,
                                             fcbuf_2d[i*2], fcbuf_2d[i*2+1]);
                                 else
                                     fprintf(file,&quot; %.5e,%.5e&quot;,
</code></pre>
<p><a href="https://nwchemgit.github.io/Special_AWCforum/sp/id3358.html">https://nwchemgit.github.io/Special_AWCforum/sp/id3358.html</a></p>
<h2 id="faq-linear-dependencies">Linear Dependencies<a class="headerlink" href="#faq-linear-dependencies" title="Permanent link">&para;</a></h2>
<p>Two or more basis functions can be consider linearly dependent when they span the same region of space. This can result in SCF converge problems. Analysis of the eigenvectors of the S<sup>-1/2</sup> matrix (where S is the overlap matrix) is used to detect linear dependencies: if there are eigenvalues  close to zero, the basis set goes through the process of canonical orthogonalization (as described in Section 3.4.5 of <a href="https://store.doverpublications.com/0486691861.html">Szabo &amp; Ostlund &ldquo;Modern Quantum Chemistry&rdquo; book</a>). This has net effect of a reduction of number of basis function used, compared to the original number set by input.
By setting</p>
<pre><code>set lindep:n_dep 0
</code></pre>
<p>this orthogonalization process is skipped.</p>
<h2 id="faq-discrepancy-on-the-number-of-basis-functions-spherical-vs-cartesian-functions">Discrepancy on the number of basis functions: spherical vs cartesian functions<a class="headerlink" href="#faq-discrepancy-on-the-number-of-basis-functions-spherical-vs-cartesian-functions" title="Permanent link">&para;</a></h2>
<p>If you are comparing NWChem results with the ones obtained from  other codes and you believe  there is a
discrepancy in the number of basis functions, keep in mind that NWChem uses cartesian functions by default, while other codes could be using spherical functions, instead.<br />
If you need to use spherical functions, the beginning of the basis input field needs to be</p>
<pre><code>basis spherical
</code></pre>
<p>More details in the documentation at the link <a href="https://nwchemgit.github.io/Basis.html#spherical-or-cartesian">https://nwchemgit.github.io/Basis.html#spherical-or-cartesian</a>.</p>
<p>See also the following <a href="https://groups.google.com/g/nwchem-forum/search?q=default%20cartesian">forum entries</a>.</p>
<h2 id="faq-starting-nwchem-with-mpirun-np-1-crashes">Starting NWChem with <code>mpirun -np 1</code> crashes<a class="headerlink" href="#faq-starting-nwchem-with-mpirun-np-1-crashes" title="Permanent link">&para;</a></h2>
<p>This is most likely due to the fact that NWChem was compiled with the setting <code>ARMCI_NETWORK=MPI-PR</code>.<br />
This is the expected behavior, since <code>ARMCI_NETWORK=MPI-PR</code> requires asking for  for n+1 processes. In other words, a serial run (with a single computing process) is triggered by executing <code>mpirun -np 2</code>.<br />
If you would prefer  <code>mpirun -np 1</code> to work, other choice of <code>ARMCI_NETWORK</code> are possible as described in the <a href="#armci">ARMCI documentation</a>.</p>
<h2 id="faq-nb_wait_for_handle-error">nb_wait_for_handle Error<a class="headerlink" href="#faq-nb_wait_for_handle-error" title="Permanent link">&para;</a></h2>
<p>If you get the following error </p>
<pre><code>{1} nb_wait_for_handle Error: all user-level nonblocking handles have been exhausted
application called MPI_Abort(comm=0x84000002, -1)
</code></pre>
<p>you can fix it by executing the following command</p>
<pre><code>export COMEX_MAX_NB_OUTSTANDING=16
</code></pre>
<h2 id="faq-memory-errors">Memory errors<a class="headerlink" href="#faq-memory-errors" title="Permanent link">&para;</a></h2>
<p>If you get the following error</p>
<pre><code>[0] Received an Error in Communication: (-1) 0: ptsalloc: increase memory in input line:
</code></pre>
<p>you can fix it by either  <br />
<em> increasing the memory line in the input file using the syntax described in the <a href="#memory">Memory section</a> (e.g. <code>memory total 1000 mb</code>), or by<br />
</em> recompiling the NWChem binary with the <code>getmem.nwchem</code> script as described in the section avaible at this 
<a href="#compiling-nwchem-setting-the-default-memory-values">link</a>  </p></section><section class="print-page" id="tutorials" heading-number="23"><h1 id="tutorials-tutorials">Tutorials<a class="headerlink" href="#tutorials-tutorials" title="Permanent link">&para;</a></h1>
<h2 id="tutorials-links-to-material-of-past-nwchem-tutorials">Links to material of past NWChem Tutorials<a class="headerlink" href="#tutorials-links-to-material-of-past-nwchem-tutorials" title="Permanent link">&para;</a></h2>
<ul>
<li>
<p>EMSL Integration 2018 meeting
<a href="#tutorial-slides">Tutorial EMSL 2018</a></p>
</li>
<li>
<p>V-SEEM NAT-GR LS+ : 2018 NWChem Workshop  (10-11 September 2018, Athens, Greece)
<a href="#tutorial-athens2018">Tutorial Athens 2018</a></p>
</li>
<li>
<p><a href="#tut2019-index">Tutorial 2019 EMSL/ARM Aerosol Summer School</a></p>
</li>
<li>
<p><a href="#tutorial_singapore2012-index">Tutorial 2012 Singapore A*STAR</a></p>
</li>
</ul>
<h2 id="tutorials-websites-containing-nwchem-material">Websites containing NWChem material<a class="headerlink" href="#tutorials-websites-containing-nwchem-material" title="Permanent link">&para;</a></h2>
<ul>
<li>
<p>Simple DFT calculations and processing with NWChem (Youtube Videos&rsquo; playlist) from the Department of Materials and Optoelectronic Science at the National Sun Yat-sen University, Taiwan
<a href="https://www.youtube.com/playlist?list=PLkbV4NC7mZIXvouZ4vhWeQIEsw4GJ3v6I">Link for Youtube playlist</a></p>
</li>
<li>
<p>PRACE Spring School in Computational Chemistry 2019<br />
<a href="https://web.archive.org/web/20221103195703/https://events.prace-ri.eu/event/786/attachments/840/1256/QC-workshop-advanced.pdf">https://web.archive.org/web/20221103195703/https://events.prace-ri.eu/event/786/attachments/840/1256/QC-workshop-advanced.pdf</a></p>
</li>
<li>
<p>Introduction To NWChem by B.J. Lynch (UMN) 2006
<a href="https://web.archive.org/web/20230418140258/https://www.msi.umn.edu/sites/default/files/IntroNWChem.pdf">https://web.archive.org/web/20230418140258/https://www.msi.umn.edu/sites/default/files/IntroNWChem.pdf</a></p>
</li>
<li>
<p>Quantum Chemistry Course at Radboud University - Nijmegen
<a href="https://www.theochem.ru.nl/quantumchemistry/">https://www.theochem.ru.nl/quantumchemistry</a></p>
</li>
<li>NWChem tips from C.J. O&rsquo;Brien (NCSU)<br />
<a href="https://sites.google.com/a/ncsu.edu/cjobrien-nwchem-tips/">https://sites.google.com/a/ncsu.edu/cjobrien-nwchem-tips/</a></li>
</ul>
<!--# Useful websites for Computational Chemistry-->
<!--* [https://www.researchgate.net](https://www.researchgate.net/)-->
<!--* [https://chemistry.stackexchange.com](https://chemistry.stackexchange.com)-->
<!--* [http://www.ccl.net](http://www.ccl.net/)--></section><section class="print-page" id="forum" heading-number="24"><h1 id="forum-forum">Forum<a class="headerlink" href="#forum-forum" title="Permanent link">&para;</a></h1>
<p>We use google groups for the users&rsquo; forum.  </p>
<p>The group is open to registered members.</p>
<p>Please do mention your real name as display  name during the registration process.</p>
<p>The group can be accessed from the following link<br />
<a href="https://groups.google.com/g/nwchem-forum">https://groups.google.com/g/nwchem-forum</a></p>
<p>If you have trouble registering to this group, please send an email to <a href="mailto:nwchemgit@gmail.com?subject=nwchem-forum - Google Groups Registration">nwchemgit@gmail.com</a></p>
<p>You can use the form below to search the entries of the forum</p>
<!-- Load icon library -->
<!--link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"-->

<form action="https://groups.google.com/g/nwchem-forum/search" class="searchform" method="get" name="searchform" target="_blank">
<input autocomplete="on" class="form-control search" name="forum-q" placeholder="Search forum" required="required"  type="text">
<button type="submit"><i class="fa fa-search"></i></button>
</form></section>
                    <section class='print-page md-section' id='section-27' heading-number='27'>
                        <h1>Supplementary Information<a class='headerlink' href='#section-27' title='Permanent link'></a>
                        </h1>
                    <section class="print-page" id="armci" heading-number="27.2"><h1 id="armci-choosing-the-armci-library">Choosing the ARMCI Library<a class="headerlink" href="#armci-choosing-the-armci-library" title="Permanent link">&para;</a></h1>
<h2 id="armci-overview">Overview<a class="headerlink" href="#armci-overview" title="Permanent link">&para;</a></h2>
<p>The Global Arrays parallel environment relies upon a one-sided
communication runtime system. There are at least three options currently
available:</p>
<ul>
<li>ARMCI - This is the original one-sided library developed with Global
    Arrays by PNNL. It supports the widest range of platforms. See
    <a href="http://hpc.pnl.gov/globalarrays/support.shtml">http://hpc.pnl.gov/globalarrays/support.shtml</a> for details.</li>
<li>ComEx - This library is being developed by PNNL to replace ARMCI. It
    supports the fewer platforms than ARMCI but may be more robust on
    modern ones like InfiniBand. See
    <a href="http://hpc.pnl.gov/globalarrays/support.shtml">http://hpc.pnl.gov/globalarrays/support.shtml</a> for details.</li>
<li>ARMCI-MPI - This library is a completely separate implementation of
    the ARMCI interface by Argonne and Intel. It uses the one-sided
    communication features of MPI, rather than a platform-specific
    conduit. It currents provides both MPI-2 and MPI-3 implementations;
    the MPI-3 one is supported by nearly all platforms. See
    <a href="https://github.com/pmodels/mpich/blob/main/doc/wiki/Index.md">https://github.com/pmodels/mpich/blob/main/doc/wiki/Index.md</a> for details.</li>
</ul>
<p>By default, Global Arrays will choose ARMCI or ComEx, based upon the
environment variables selected by the user (see GA documentation for
details). When a native implementation is available and works reliably,
this is the best option for the NWChem user. However, in cases where an
native implementation of ARMCI is not available or is not reliable, the
user should consider using one of the MPI-based implementations.</p>
<p>There are many different ways to use MPI as the communication runtime of
Global Arrays:</p>
<table>
<thead>
<tr>
<th style="text-align: left;">ARMCI_NETWORK</th>
<th style="text-align: left;">Result</th>
<th style="text-align: left;">Notes</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><code>MPI-PR</code></td>
<td style="text-align: left;">ARMCI with progress rank</td>
<td style="text-align: left;">Recommended, except on Blue Gene/Q.</td>
</tr>
<tr>
<td style="text-align: left;"><code>MPI-PT</code></td>
<td style="text-align: left;">ARMCI with progress thread</td>
<td style="text-align: left;">Appropriate for Blue Gene/Q.</td>
</tr>
<tr>
<td style="text-align: left;"><code>MPI-MT</code></td>
<td style="text-align: left;">ARMCI over multi-threaded MPI</td>
<td style="text-align: left;"><a href="https://github.com/open-mpi/ompi/issues/157">Do not use Open-MPI 1.x.</a></td>
</tr>
<tr>
<td style="text-align: left;"><code>MPI-TS</code></td>
<td style="text-align: left;">ARMCI over MPI without data server</td>
<td style="text-align: left;"></td>
</tr>
<tr>
<td style="text-align: left;"><code>MPI-SPAWN</code></td>
<td style="text-align: left;">ARMCI using MPI dynamic processes</td>
<td style="text-align: left;">Requires MPI_Comm_spawn support.</td>
</tr>
<tr>
<td style="text-align: left;"><code>ARMCI</code></td>
<td style="text-align: left;">Uses ARMCI-MPI</td>
<td style="text-align: left;">See <a href="https://github.com/pmodels/mpich/blob/main/doc/wiki/Index.md">the ARMCI-MPI NWChem page</a> for details.</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;">(please use <em>mpi3rma</em> branch)</td>
<td style="text-align: left;">requires <code>EXTERNAL_ARMCI_PATH</code></td>
</tr>
</tbody>
</table>
<p>It is difficult to provide complete guidance to the user as to which
option to choose. However, we observe the following:</p>
<ul>
<li><code>ARMCI_NETWORK=MPI-PR</code> is stable and performs well on many platforms
    (including Cray XC platforms, e.g. NERSC Cori). This port will use
    one processes on each node for communication, therefore subtracting
    one process (again on each node) for NWChem. Therefore, when
    executing on a single node (i.e. the case of desktop execution) you
    would need to ask for n+1 processes; in other words, a serial
    execution would require the following mpirun invocation <code>mpirun -np 2 ...</code>  </li>
<li>On Intel True Scale and Omni Path systems, <code>MPI-PR</code> is more reliable
    than <code>OPENIB</code> or <code>MPI-SPAWN</code>. <code>ARMCI-MPI</code> with Casper and Intel MPI is
    also recommended. See <a href="https://github.com/jeffhammond/HPCInfo/blob/master/ofi/NWChem-OPA.md">this page</a>
    for details. Contact <a href="mailto:jeff.science@gmail.com&gt;">Jeff Hammond</a>
    for assistance.</li>
<li>On IBM Blue Gene/Q, one must use the <em>mpi2rma</em> branch of <code>ARMCI-MPI</code>
    because MPI-3 is not fully supported. On all other platforms, the
    <code>ARMCI-MPI</code> branch <em>mpi3rma</em> is recommended.</li>
<li>The performance of <code>ARMCI-MPI</code> is greatly enhanced by
    <a href="https://pmodels.github.io/casper-www/">Casper</a>. See <a href="https://pmodels.github.io/casper-www/publication.html">this
    link</a> 
    for design details and <a href="https://wiki.mpich.org/armci-mpi/index.php/Casper">this
    page</a> for
    instructions on how to use it.</li>
</ul>
<p>When using ARMCI-MPI, please make sure to use the most recent version of
MPI (MPICH 3.2+, Cray MPI 7.2+, MVAPICH2 2.0+, Intel MPI 5.1+, Open-MPI
2.0+). Older versions of MPI are known to have bugs in the MPI-3 RMA
features that affect the correctness of NWChem.</p>
<h2 id="armci-support">Support<a class="headerlink" href="#armci-support" title="Permanent link">&para;</a></h2>
<p>Global Arrays, ARMCI and ComEx are developed and supported by PNNL. The
user list for support is <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#104;&#112;&#99;&#116;&#111;&#111;&#108;&#115;&#64;&#103;&#111;&#111;&#103;&#108;&#101;&#103;&#114;&#111;&#117;&#112;&#115;&#46;&#99;&#111;&#109;">&#104;&#112;&#99;&#116;&#111;&#111;&#108;&#115;&#64;&#103;&#111;&#111;&#103;&#108;&#101;&#103;&#114;&#111;&#117;&#112;&#115;&#46;&#99;&#111;&#109;</a>.</p>
<p>ARMCI-MPI is developed by Argonne and Intel. All ARMCI-MPI questions
should be directed to <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#97;&#114;&#109;&#99;&#105;&#45;&#100;&#105;&#115;&#99;&#117;&#115;&#115;&#64;&#108;&#105;&#115;&#116;&#115;&#46;&#109;&#112;&#105;&#99;&#104;&#46;&#111;&#114;&#103;">&#97;&#114;&#109;&#99;&#105;&#45;&#100;&#105;&#115;&#99;&#117;&#115;&#115;&#64;&#108;&#105;&#115;&#116;&#115;&#46;&#109;&#112;&#105;&#99;&#104;&#46;&#111;&#114;&#103;</a>. <em>ARMCI-MPI is not
an Intel product.</em></p>
<h2 id="armci-automated-installation-of-armci-mpi">Automated installation of ARMCI-MPI<a class="headerlink" href="#armci-automated-installation-of-armci-mpi" title="Permanent link">&para;</a></h2>
<p>If you wish to use ARMCI-MPI, a script is available to automatically
install it:</p>
<pre><code>cd$NWCHEM_TOP/tools&amp;&amp;./install-armci-mpi
</code></pre></section></section><section class="print-page" id="interfaces-with-external-software" heading-number="28"><h1 id="interfaces-with-external-software-interfaces-with-external-software">Interfaces with External Software<a class="headerlink" href="#interfaces-with-external-software-interfaces-with-external-software" title="Permanent link">&para;</a></h1>
<h2 id="interfaces-with-external-software-overview">Overview<a class="headerlink" href="#interfaces-with-external-software-overview" title="Permanent link">&para;</a></h2>
<p>NWChem can be interfaced with external software packages by following the instructions below.</p>
<p>The interface can be set either using a pre-exisiting software package or
by downloading and compiling the software from the NWChem makefile infrastructure.  </p>
<h2 id="interfaces-with-external-software-simint-integrals-library">Simint integrals library<a class="headerlink" href="#interfaces-with-external-software-simint-integrals-library" title="Permanent link">&para;</a></h2>
<p>To generate the <a href="https://github.com/simint-chem/simint-generator">Simint</a>)
library and enable the NWChem interface (only for energy and first derivative code),
you need to define the following environment variables at compile time:  </p>
<ul>
<li><code>USE_SIMINT=y</code> (mandatory)</li>
<li><code>SIMINT_MAXAM=</code>&ldquo;Maximum angular momentum&rdquo; (optional, default is 3, therefore up to f orbitals)</li>
</ul>
<p>The following set directives are required in the input file to trigger use of Simint</p>
<pre><code>set int:cando_txs f
set int:cando_nw f
</code></pre>
<h2 id="interfaces-with-external-software-openblas">OpenBLAS<a class="headerlink" href="#interfaces-with-external-software-openblas" title="Permanent link">&para;</a></h2>
<p>To build NWChem with the optimized BLAS and Lapack <a href="https://github.com/xianyi/OpenBLAS">OpenBLAS</a> library,
you need to define the following environment variables at compile time:  </p>
<pre><code>BUILD_OPENBLAS=1
BLAS_SIZE=8
</code></pre>
<p>This procedure requires an internet connection to dowload the OpenBLAS source.  <br />
Instead, to use a pre-compiled OpenBLAS library, the <code>BLASOPT</code>, <code>LAPACK_LIB</code> and <code>BLAS_SIZE</code> environment variable need to be set.</p>
<h2 id="interfaces-with-external-software-scalapack">ScaLAPACK<a class="headerlink" href="#interfaces-with-external-software-scalapack" title="Permanent link">&para;</a></h2>
<p>To build NWChem with the <a href="https://github.com/Reference-ScaLAPACK/scalapack">ScaLAPACK</a> library,
you need to define the following environment variables at compile time:  </p>
<pre><code>BUILD_SCALAPACK=1
SCALAPACK_SIZE=8
</code></pre>
<p>This procedure requires an internet connection to dowload the OpenBLAS source.  <br />
Instead, to use a pre-compiled ScaLAPACK library, the <code>SCALAPACK_LIB</code>  and <code>SCALAPACK_SIZE</code> environment variable need to be set, 
as described in the related <a href="#compiling-nwchem-optimized-math-libraries">Compiling NWChem</a> section</p>
<h2 id="interfaces-with-external-software-elpa">ELPA<a class="headerlink" href="#interfaces-with-external-software-elpa" title="Permanent link">&para;</a></h2>
<p>To build NWChem with the <a href="https://gitlab.mpcdf.mpg.de/elpa/elpa">ELPA</a> eigensolver library,
you need first to set the ScaLAPACK settings as described in the previous <a href="#interfaces-with-external-software-scalapack">section</a> and
then you need to define the following environment variables at compile time:  </p>
<pre><code>BUILD_ELPA=1
</code></pre>
<p>This procedure requires an internet connection to dowload the OpenBLAS source. <br />
Instead, to use a pre-compiled ELPA library, the <code>ELPA</code>  and <code>ELPA_SIZE</code> environment variable need to be set.</p>
<h2 id="interfaces-with-external-software-plumed">Plumed<a class="headerlink" href="#interfaces-with-external-software-plumed" title="Permanent link">&para;</a></h2>
<p>The <code>BUILD_PLUMED</code> environment variable installs <a href="https://www.plumed.org/">Plumed</a> and
interfaces it with the <a href="#gaussian-basis-aimd">qmd</a> module.
This procedure requires an internet connection to dowload the Plumed source.</p>
<p>Instead, if you wish to use an existing Plumed installation, the following environment variables must be set (after having unset <code>BUILD_PLUMED</code>): </p>
<pre><code>USE_PLUMED=1
</code></pre>
<p>Requirements:
* The environment variable <code>PATH</code> should to point to the location of the <code>plumed</code> command  and <code>LD_LIBRARY_PATH</code> should point to the location of the plumed libraries.
* <code>BLAS_SIZE</code> and <code>SCALAPACK_SIZE</code> must be equal to 4 (this is not a requirement when using <code>BUILD_PLUMED</code>)</p>
<h2 id="interfaces-with-external-software-libxc">Libxc<a class="headerlink" href="#interfaces-with-external-software-libxc" title="Permanent link">&para;</a></h2>
<p>Building NWChem with the <a href="https://libxc.gitlab.io/">libxc</a> DFT library requires
setting the environment variable <code>USE_LIBXC=1</code>. </p>
<p>This procedure requires an internet connection to dowload the Libxc source.</p>
<p>Instead, if you wish to use an existing libxc library, the following environment variables must be set, after having unset <code>USE_LIBXC</code>:   </p>
<ul>
<li><code>LIBXC_INCLUDE</code>   location of the libxc C header files</li>
<li><code>LIBXC_MODDIR</code>   location of the libxc fortran90 module files</li>
<li><code>LIBXC_LIB</code>     location of the libxc libraries files</li>
</ul>
<p>For example, for Debian/Ubuntu systems, the following is needed after having installed the <code>libxc-dev</code> package</p>
<pre><code>    unset USE_LIBXC
    export LIBXC_LIB=/usr/lib/x86_64-linux-gnu
    export LIBXC_INCLUDE=/usr/include
</code></pre>
<p>For example, for Fedora systems, the following is needed after having installed the <code>libxc-devel</code> package</p>
<pre><code>    unset USE_LIBXC
    export LIBXC_LIB=/usr/lib64
    export LIBXC_INCLUDE=/usr/include 
    export LIBXC_MODDIR=/usr/lib64/gfortran/modules
</code></pre>
<h2 id="interfaces-with-external-software-xtb">XTB<a class="headerlink" href="#interfaces-with-external-software-xtb" title="Permanent link">&para;</a></h2>
<p>Building NWChem with the Light-weight tight-binding framework <a href="https://tblite.readthedocs.io">tblite</a>
requires  </p>
<ul>
<li>setting the environment variable <code>USE_TBLITE=1</code></li>
<li>adding <code>xtb</code> to list of <code>NWCHEM_MODULES</code></li>
</ul>
<p>Example:</p>
<pre><code>make nwchem_config NWCHEM_MODULES='tinyqmpw xtb'
export USE_TBLITE=1
make
</code></pre></section><section class="print-page" id="software-supporting-nwchem" heading-number="29"><h1 id="software-supporting-nwchem-software-supporting-nwchem">Software supporting NWChem<a class="headerlink" href="#software-supporting-nwchem-software-supporting-nwchem" title="Permanent link">&para;</a></h1>
<h2 id="software-supporting-nwchem-overview">Overview<a class="headerlink" href="#software-supporting-nwchem-overview" title="Permanent link">&para;</a></h2>
<p>While we have done our best to compile an exhaustive list of software
using NWChem, we might have missed packages and/or incorrectly described
some software features. Please use the <a href="https://github.com/nwchemgit/nwchem/issues">Github Issue feature</a> to provide
feedback on this page content.</p>
<h2 id="software-supporting-nwchem-user-interface-software">User interface software<a class="headerlink" href="#software-supporting-nwchem-user-interface-software" title="Permanent link">&para;</a></h2>
<ul>
<li><strong>ECCE</strong> Extensible Computational Chemistry Environment
    <a href="https://github.com/FriendsofECCE/ECCE">https://github.com/FriendsofECCE/ECCE</a> <sup id="fnref_footnote_1"><a class="footnote-ref" href="#software-supporting-nwchem-fn_footnote_1">1</a></sup></li>
<li><strong>EMSL Arrows</strong> Evolution of Chemical and Materials Computation
    <a href="https://nwchemgit.github.io/EMSL_Arrows.html">https://nwchemgit.github.io/EMSL_Arrows.html</a></li>
<li><strong>Avogadro</strong> reads cube files, generates NWChem input files,
    analyzes output files (including frequencies) <a href="https://avogadro.cc">https://avogadro.cc</a></li>
<li><strong>WebMO</strong> World Wide Web-based interface to computational chemistry
    packages <a href="https://www.webmo.net/">https://www.webmo.net/</a> <sup id="fnref_footnote_2"><a class="footnote-ref" href="#software-supporting-nwchem-fn_footnote_2">2</a></sup></li>
<li><strong>Jmol</strong> analyzes output and cube files
    <a href="https://wiki.jmol.org/index.php/NWChem">https://wiki.jmol.org/index.php/NWChem</a></li>
<li><strong>Scienomics MAPS</strong> platform has a NWChem Plugin that will allow
    users to easily create NWChem input files. Since MAPS platform also
    has complex builders available, users can create complex models and
    then submit NWChem simulations to HPCs. MAPS also allows easy
    analysis of NWChem output files
    <a href="https://www.scienomics.com/maps-platform/simulate/quantum/nwchem-plugin/">https://www.scienomics.com/maps-platform/simulate/quantum/nwchem-plugin/</a></li>
<li><strong>CULGI</strong> computational platform
    <a href="https://www.plm.automation.siemens.com/global/en/products/simcenter/culgi.html">https://www.plm.automation.siemens.com/global/en/products/simcenter/culgi.html</a></li>
<li><strong>Chemcraft</strong> <a href="https://www.chemcraftprog.com">https://www.chemcraftprog.com</a></li>
<li><strong>ASE</strong> Atomic Simulation Environment <a href="https://wiki.fysik.dtu.dk/ase">https://wiki.fysik.dtu.dk/ase</a></li>
<li><strong>Ascalaph</strong>
    <a href="http://www.biomolecular-modeling.com/Ascalaph/index.html">http://www.biomolecular-modeling.com/Ascalaph/index.html</a></li>
<li><strong>MoCalc2012</strong> <a href="https://mocalc2012.sourceforge.net/">https://mocalc2012.sourceforge.net/</a></li>
<li><strong>Chemissian</strong> <a href="https://www.chemissian.com/">https://www.chemissian.com/</a></li>
<li><strong>Gausssum</strong> a GUI application that can analyze the output since
    version 3.0 using the <a href="https://cclib.github.io/">cclib</a> library
    <a href="https://gausssum.sf.net">https://gausssum.sf.net</a></li>
<li><strong>OpenDFT</strong>  brings cutting edge solid state research to the people <a href="https://github.com/JannickWeisshaupt/OpenDFT">https://github.com/JannickWeisshaupt/OpenDFT</a></li>
<li><strong>STREAMM</strong> generates structures and input files for quantum chemical and molecular dynamics codes <a href="https://github.com/NREL/streamm-tools">https://github.com/NREL/streamm-tools</a></li>
<li><strong>SparcleQC</strong> is a Python package that creates QM/MM-like input files for quantum chemistry codes <a href="https://sparcle-qc.readthedocs.io">https://sparcle-qc.readthedocs.io</a></li>
</ul>
<p><references/></p>
<h2 id="software-supporting-nwchem-codes-using-nwchem-wavefunctions-andor-post-processing-nwchem-output-files">Codes using NWChem wavefunctions and/or post-processing NWChem output files<a class="headerlink" href="#software-supporting-nwchem-codes-using-nwchem-wavefunctions-andor-post-processing-nwchem-output-files" title="Permanent link">&para;</a></h2>
<ul>
<li><strong>KiSThelP</strong> is a program to predict thermodynamic properties and rate constants from quantum chemistry results
 <a href="http://kisthelp.univ-reims.fr/">http://kisthelp.univ-reims.fr/</a></li>
<li><strong>Fiesta</strong> is a Gaussian-basis GW and Bethe-Salpeter code
    <a href="http://perso.neel.cnrs.fr/xavier.blase/fiesta/">http://perso.neel.cnrs.fr/xavier.blase/fiesta/</a></li>
<li><strong>JANPA</strong> performs Natural Population Analysis <a href="https://janpa.sf.net">https://janpa.sf.net</a></li>
<li><strong>CamCASP</strong> Cambridge package for Calculation of Anisotropic Site
    Properties <a href="https://gitlab.com/anthonyjs/camcasp">https://gitlab.com/anthonyjs/camcasp</a> <a href="https://gitlab.com/anthonyjs/camcasp/-/wikis/home">https://gitlab.com/anthonyjs/camcasp/-/wikis/home</a></li>
<li><strong>ChemShell</strong> is a computational chemistry environment for standard
    quantum chemical or force field calculations
    <a href="https://www.chemshell.org">https://www.chemshell.org</a></li>
<li><strong>PUPIL</strong> allows developers to perform multi-scale simulations
    <a href="https://pupil.sf.net">https://pupil.sf.net</a></li>
<li><strong>LICHEM</strong> interfaces between QM and MM software
    <a href="https://github.com/kratman/LICHEM_QMMM">https://github.com/kratman/LICHEM_QMMM</a></li>
<li><strong>VENUS</strong> interfaces NWChem with chemical dynamics
    <a href="https://www.depts.ttu.edu/chemistry/Venus/index.php">https://www.depts.ttu.edu/chemistry/Venus/index.php</a>
    <a href="https://www.sciencedirect.com/science/article/pii/S0010465513004049">https://www.sciencedirect.com/science/article/pii/S0010465513004049</a></li>
<li><strong>VOTCA-XTP</strong> is a GW-BSE code to calculate excited state properties
    <a href="http://www.votca.org">http://www.votca.org</a> </li>
<li><strong>DP4-AI</strong> integrates NMR-AI, software for automatic processing, assignment and visualisation of raw NMR data
    <a href="https://github.com/KristapsE/DP4-AI">https://github.com/KristapsE/DP4-AI</a></li>
<li><strong>Fafoom</strong> Flexible algorithm for optimization of molecules
    <a href="https://github.com/adrianasupady/fafoom">https://github.com/adrianasupady/fafoom</a></li>
<li><strong>Pymatgen</strong> Python Materials Genomics open-source Python library for materials analysis
    <a href="http://pymatgen.org">http://pymatgen.org</a></li>
<li><strong>PVSCF</strong> A parallel vibrational self-consistent field program <a href="http://pvscf.org">http://pvscf.org</a></li>
<li><strong>ResLibCal</strong> A tool to compute triple-axis neutron spectrometer resolution <a href="http://ifit.mccode.org/Applications/ResLibCal/doc/ResLibCal.html">http://ifit.mccode.org/Applications/ResLibCal/doc/ResLibCal.html</a></li>
<li><strong>SMFA</strong>  General program package for performing quantum chemistry calculations on large molecules using an energy-based fragmentation approach <a href="https://github.com/mickcollins/SMFAPAC">https://github.com/mickcollins/SMFAPAC</a></li>
<li><strong>OCLIMAX</strong> Free program for simulation of inelastic neutron scattering <a href="https://sites.google.com/site/ornliceman/download">https://sites.google.com/site/ornliceman/download</a></li>
<li><strong>Artaios</strong> A code for calculating spin-dependent electron transport properties for molecular junctions in the coherent tunneling regime <a href="https://www.chemie.uni-hamburg.de/institute/ac/arbeitsgruppen/herrmann/software/artaios.html">https://www.chemie.uni-hamburg.de/institute/ac/arbeitsgruppen/herrmann/software/artaios.html</a></li>
<li><strong>Cuby</strong>  A computational chemistry framework that provides  access to various computational methods available in different software package <a href="http://cuby.molecular.cz/?page=Interfaces">http://cuby.molecular.cz/?page=Interfaces</a></li>
<li><strong>autoDIAS</strong> A python tool for an automated Distortion/Interaction Activation Strain Analysis <a href="https://github.com/dsvatunek/autoDIAS">https://github.com/dsvatunek/autoDIAS</a></li>
<li><strong>PolyParGen</strong> provides OPLS-AA and Amber force field parameters for polymers or large molecules <a href="http://polypargen.com">http://polypargen.com</a></li>
<li><strong>BiKi Life Sciences</strong> is a suite for Molecular Dynamics and related methods in Drug Discovery <a href="http://www.bikitech.com/products/">http://www.bikitech.com/products/</a></li>
<li><strong>Shermo</strong> is a general code for calculating molecular thermodynamic properties 
<a href="http://sobereva.com/soft/shermo/">http://sobereva.com/soft/shermo/</a></li>
<li><strong>QCengine</strong> is a program executor and IO standardizer for quantum chemistry.<a href="https://github.com/MolSSI/QCEngine">https://github.com/MolSSI/QCEngine</a></li>
<li><strong>AiiDA plugin for NWChem</strong> AiiDA is a Python infrastructure to deal with complex scientific workflows <a href="https://aiida-nwchem.readthedocs.io/">https://aiida-nwchem.readthedocs.io/</a></li>
<li><strong>cclib</strong> is a Python library for parsing and interpreting the results of computational chemistry packages <a href="https://cclib.github.io/">https://cclib.github.io/</a></li>
<li><strong>CHARMM</strong> can perform combined Quantum Mechanical and Molecular Mechanics simulations using NWChem <a href="https://www.charmm.org/archive/charmm/documentation/by-version/c45b1/nwchem.html">https://www.charmm.org/archive/charmm/documentation/by-version/c45b1/nwchem.html</a></li>
<li><strong>kMap</strong> is a program for simulation and data analysis in photoemission tomography <a href="https://github.com/brands-d/kMap">https://github.com/brands-d/kMap</a></li>
<li><strong>QMCube</strong> is a Python suite focused on multiscale QM/MM simulations of biological systems <a href="https://github.com/sergio-marti/qm3">https://github.com/sergio-marti/qm3</a></li>
<li><strong>autodE</strong> is a Python module designed for the automated generation of reaction profiles <a href="https://duartegroup.github.io/autodE">https://duartegroup.github.io/autodE</a></li>
<li><strong>GoodVibes</strong> is a Python program to compute thermochemical data from electronic structure calculations <a href="https://github.com/patonlab/GoodVibes">https://github.com/patonlab/GoodVibes</a></li>
<li><strong>ChemDyME</strong> is a Kinetically Steered, Automated Mechanism Generation Through Combined Molecular Dynamics and Master Equation Calculations <a href="https://github.com/RobinShannon/ChemDyME">https://github.com/RobinShannon/ChemDyME</a></li>
<li><strong>xtbdft</strong> is a wrapper script for multi-level molecular modelling powered by CREST/GFN2-XTB and NWChem (DFT) <a href="https://github.com/sibo/xtbdft">https://github.com/sibo/xtbdft</a></li>
<li><strong>OctaDist</strong> is an inorganic chemistry and crystallography program for computing the distortion parameters in coordination complexes <a href="https://octadist.github.io">https://octadist.github.io</a></li>
<li><strong>PyADF</strong> is a scripting framework for multiscale quantum chemistry <a href="https://github.com/chjacob-tubs/pyadf-releases">https://github.com/chjacob-tubs/pyadf-releases</a></li>
<li><strong>UniMoVib</strong> is a unified interface for molecular harmonic vibrational frequency calculations <a href="https://github.com/zorkzou/UniMoVib">https://github.com/zorkzou/UniMoVib</a></li>
<li><strong>MoBioTools</strong> is a toolkit to automatically setup QM/MM calculations <a href="https://github.com/mobiochem/MoBioTools">https://github.com/mobiochem/MoBioTools</a></li>
<li><strong>Fragmet</strong> is a framework that makes it easy to prototype, implement, and benchmark fragmentation methods <a href="https://gitlab.com/john-herbert-group/fragment">https://gitlab.com/john-herbert-group/fragment</a></li>
<li><strong>pyVPT2</strong> is a program to perform second-order vibrational perturbation theory (VPT2) computations to obtain anharmonic vibrational frequencies <a href="https://github.com/philipmnel/pyvpt2">https://github.com/philipmnel/pyvpt2</a></li>
<li><strong>Quick-QM-Spectra</strong>  creates a IR, UV-VIS or Raman spectrum from a NWChem output file <a href="https://hanlev.github.io/QQMS">https://hanlev.github.io/QQMS</a></li>
</ul>
<h2 id="software-supporting-nwchem-programs-that-can-display-or-manipulate-cube-andor-molden-files">Programs that can display or manipulate cube and/or Molden files<a class="headerlink" href="#software-supporting-nwchem-programs-that-can-display-or-manipulate-cube-andor-molden-files" title="Permanent link">&para;</a></h2>
<p>The following programs can display cube files from
<a href="#dplot-gaussian-gaussian-cube-format">charge density</a> and
<a href="#properties-gaussian-cube-files">ESP</a> and/or use
<a href="#properties-moldenfile">Molden</a> files</p>
<ul>
<li><strong>gOpenMol</strong> <a href="https://web.archive.org/web/20090518024059/http://www.csc.fi/english/pages/g0penMol">https://web.archive.org/web/20090518024059/http://www.csc.fi/english/pages/g0penMol</a></li>
<li><strong>Molden</strong> <a href="https://www.theochem.ru.nl/molden/">https://www.theochem.ru.nl/molden/</a></li>
<li><strong>Molekel</strong> <a href="http://ugovaretto.github.io/molekel/">http://ugovaretto.github.io/molekel/</a></li>
<li><strong>GaussView</strong> <a href="http://www.gaussian.com/g_prod/gv5.htm">http://www.gaussian.com/g_prod/gv5.htm</a></li>
<li><strong>VMD</strong> <a href="http://www.ks.uiuc.edu/Research/vmd">http://www.ks.uiuc.edu/Research/vmd</a></li>
<li><strong>VESTA</strong> <a href="http://jp-minerals.org/vesta/en/">http://jp-minerals.org/vesta/en/</a></li>
<li><strong>Jamberoo</strong>
    <a href="http://www.jamberoo.org/">http://www.jamberoo.org/</a></li>
<li><strong>Molden2AIM</strong> is a utility program which can be used to create AIM-WFN, AIM-WFX, and NBO-47 files from a Molden file
     <a href="https://github.com/zorkzou/Molden2AIM">https://github.com/zorkzou/Molden2AIM</a></li>
<li><strong>Multiwfn</strong> is a wavefunction analysis program <a href="http://sobereva.com/multiwfn">http://sobereva.com/multiwfn</a></li>
<li><strong>CrystalExplorer</strong> is a program for Hirshfeld surface analysis, visualization and quantitative analysis of molecular crystals that can read NWChem Molden Files <a href="https://crystalexplorer.net/">https://crystalexplorer.net/</a></li>
</ul>
<h2 id="software-supporting-nwchem-programs-post-processing-aim-files">Programs post-processing AIM files<a class="headerlink" href="#software-supporting-nwchem-programs-post-processing-aim-files" title="Permanent link">&para;</a></h2>
<p>NWChem can generate <a href="#properties-aimfile">AIM</a><sup id="fnref_footnote_3"><a class="footnote-ref" href="#software-supporting-nwchem-fn_footnote_3">3</a></sup> wavefunction
files (.wfn/.wfx) can be post-processed with a variety of codes, e.g.</p>
<ul>
<li><strong>XAIM</strong> is a graphical user interface to several programs based on some aspects of the Theory of Atoms in Molecules <a href="http://www.quimica.urv.es/XAIM">http://www.quimica.urv.es/XAIM</a></li>
<li><strong>NCIPLOT</strong> computates and visualizes inter- and intra-molecular non-covalent interactions <a href="https://github.com/aoterodelaroza/nciplot">https://github.com/aoterodelaroza/nciplot</a></li>
<li><strong>Multiwfn</strong> performs a variety of electronic wavefunction analysis <a href="http://sobereva.com/multiwfn/">http://sobereva.com/multiwfn/</a></li>
<li><strong>Postg</strong> calculates the dispersion energy and related quantities in gas-phase systems using the exchange-hole dipole moment (XDM) model <a href="https://github.com/aoterodelaroza/postg">https://github.com/aoterodelaroza/postg</a></li>
<li><strong>GPUAM</strong> computes Molecular Electrostatic Potential over Graphics Processing Units <a href="http://www.fqt.izt.uam.mx/Profes/JGO/GPUAM/GPUAM.html">http://www.fqt.izt.uam.mx/Profes/JGO/GPUAM/GPUAM.html</a></li>
<li><strong>CHARGEMOL</strong>  computes Density Derived Electrostatic and Chemical (DDEC) net atomic charges and atomic multipoles  <a href="https://sourceforge.net/projects/ddec/">https://sourceforge.net/projects/ddec/</a></li>
<li><strong>PAMoC</strong> is a program for the analysis of experimental and theoretical electron charge density distributions <a href="https://www.pamoc.it/">https://www.pamoc.it/</a></li>
<li><strong>DensToolKit2</strong> is a suite of programs for analyzing electronic densities <a href="https://github.com/jmsolano/denstoolkit">https://github.com/jmsolano/denstoolkit</a>  </li>
</ul>
<div class="footnote">
<hr />
<ol>
<li id="software-supporting-nwchem-fn_footnote_1">
<p>No longer been actively developed at PNNL. New development effort at
<a href="https://github.com/FriendsofECCE/ECCE/releases">https://github.com/FriendsofECCE/ECCE/releases</a>&#160;<a class="footnote-backref" href="#software-supporting-nwchem-fnref_footnote_1" title="Jump back to footnote 1 in the text">&#8617;</a></p>
</li>
<li id="software-supporting-nwchem-fn_footnote_2">
<p>The WebMo interface might not be compatible with NWChem 6.0 and
later versions&#160;<a class="footnote-backref" href="#software-supporting-nwchem-fnref_footnote_2" title="Jump back to footnote 2 in the text">&#8617;</a></p>
</li>
<li id="software-supporting-nwchem-fn_footnote_3">
<p><strong>WARNING:</strong> Since we have discovered issues in generating .WFN files with this module (e.g. systems with ECPs), the recommended method for generating .WFN file is to first generate a Molden file with the <a href="#properties-moldenfile">Moldenfile</a> option, then convert the Molden file into a WFN file by using the <a href="https://github.com/zorkzou/Molden2AIM">Molden2AIM</a> program.&#160;<a class="footnote-backref" href="#software-supporting-nwchem-fnref_footnote_3" title="Jump back to footnote 3 in the text">&#8617;</a></p>
</li>
</ol>
</div></section><section class="print-page" id="containers" heading-number="30"><h1 id="containers-containers">Containers<a class="headerlink" href="#containers-containers" title="Permanent link">&para;</a></h1>
<h2 id="containers-docker">Docker<a class="headerlink" href="#containers-docker" title="Permanent link">&para;</a></h2>
<p>Instruction for using the NWChem Docker container with the <a href="https://docs.docker.com/compose">docker compose</a> command</p>
<ol>
<li>
<p>Install docker  as described in <a href="https://docs.docker.com/engine/install">https://docs.docker.com/engine/install</a></p>
</li>
<li>
<p>Download the <a href="https://raw.githubusercontent.com/nwchemgit/nwchem-dockerfiles/refs/heads/master/nwchem-dev.mpipr/compose.yaml">compose.yaml</a> file</p>
</li>
</ol>
<pre><code> wget https://raw.githubusercontent.com/nwchemgit/nwchem-dockerfiles/refs/heads/master/nwchem-dev.mpipr/compose.yaml
</code></pre>
<ol>
<li>Create the nwchem service</li>
</ol>
<pre><code>docker compose up -d
</code></pre>
<p>In some installation, <code>docker compose</code> is available as the command <code>docker-compose</code>, therefore in those cases, this command becomes</p>
<pre><code>docker-compose up -d
</code></pre>
<ol>
<li>Run NWChem</li>
</ol>
<pre><code> docker compose run --rm nwchem h2o.nw
</code></pre>
<p>In some installation, <code>docker compose</code> is available as the command <code>docker-compose</code>, therefore in those cases, this command becomes</p>
<pre><code> docker-compose run --rm nwchem h2o.nw
</code></pre>
<p>In the example above the input file name is <code>h2o.nw</code>. <br />
The default setting will run NWChem using 2 processes.
If you would like to use more processes,
then you would have to set the environment variable <code>MYNPROC</code> to the number of processes plus one.
For example, if you wish to use 4 processes, then you will execute the following (bash syntax)</p>
<pre><code>export MYNPROC=5
</code></pre>
<h2 id="containers-singularityapptainer">Singularity/Apptainer<a class="headerlink" href="#containers-singularityapptainer" title="Permanent link">&para;</a></h2>
<p>Singularity/Apptainer recipes for NWChem are available at.<br />
<a href="https://github.com/edoapra/nwchem-singularity">https://github.com/edoapra/nwchem-singularity</a></p>
<p>Singularity images are available at 
<a href="https://ghcr.io/edoapra/nwchem-singularity/nwchem-dev.ompi41x">ghcr.io/edoapra/nwchem-singularity/nwchem-dev.ompi41x</a><br />
and 
<a href="https://ghcr.io/edoapra/nwchem-singularity/nwchem-dev.mpich3.4.2">ghcr.io/edoapra/nwchem-singularity/nwchem-dev.mpich3.4.2</a></p>
<h3 id="containers-instruction-for-running-on-emsl-tahoma">Instruction for running on EMSL Tahoma<a class="headerlink" href="#containers-instruction-for-running-on-emsl-tahoma" title="Permanent link">&para;</a></h3>
<p>Slurm script for running NWChem Singularity/Apptainer images on <a href="https://www.emsl.pnnl.gov/MSC/UserGuide/tahoma/tahoma_overview.html">EMSL tahoma</a></p>
<pre><code>#!/bin/bash
#SBATCH -N 2
#SBATCH -t 00:29:00
#SBATCH -A allocation_name
#SBATCH --ntasks-per-node 36
#SBATCH -o singularity_library.output.%j
#SBATCH -e ./singularity_library.err.%j
#SBATCH -J singularity_library
#SBATCH --export ALL
source /etc/profile.d/modules.sh
export https_proxy=http://proxy.emsl.pnl.gov:3128
module purge
module load gcc/9.3.0
module load openmpi/4.1.4
SCRATCH=/big_scratch
# pull new image to the current directory
singularity pull -F --name ./nwchems_`id -u`.img oras://ghcr.io/edoapra/nwchem-singularity/nwchem-dev.ompi41x:latest
# copy image from current directory to local /big_scratch/ on compute nodes
srun -N $SLURM_NNODES -n $SLURM_NNODES cp ./nwchems_`id -u`.img $SCRATCH/nwchems.img
# basis library files
export SINGULARITYENV_NWCHEM_BASIS_LIBRARY=/cluster/apps/nwchem/nwchem/src/basis/libraries/
# use /big_scratch as scratch_dir
export SINGULARITYENV_SCRATCH_DIR=$SCRATCH
# bind local file system
MYFS=$(findmnt -r -T . | tail -1 |cut -d ' ' -f 1)
# run
srun --mpi=pmi2 -N $SLURM_NNODES -n $SLURM_NPROCS singularity exec --bind $SCRATCH,$MYFS $SCRATCH/nwchems.img nwchem  &quot;file name&quot;
</code></pre>
<h3 id="containers-instruction-for-running-on-olcf-frontier">Instruction for running on OLCF Frontier<a class="headerlink" href="#containers-instruction-for-running-on-olcf-frontier" title="Permanent link">&para;</a></h3>
<p>Slurm script for running NWChem Singularity/Apptainer images on <a href="https://docs.olcf.ornl.gov/systems/frontier_user_guide.html">OLCF Frontier</a></p>
<pre><code>#!/bin/bash
#SBATCH -t 00:29:00
#SBATCH -A ABCXYZ123
#SBATCH -N 4
#SBATCH  --tasks-per-node 56
#SBATCH  --cpus-per-task 1
#SBATCH -J siosi6
#SBATCH -o siosi6.err.%j
module swap libfabric libfabric/1.15.2.0
module  load cray-mpich-abi
export APPTAINERENV_LD_LIBRARY_PATH=&quot;$CRAY_MPICH_DIR/lib-abi-mpich:$CRAY_MPICH_ROOTDIR/gtl/lib:/opt/rocm/lib:/opt/rocm/lib64:$CRAY_LD_LIBRARY_PATH:$LD_LIBRARY_PATH:/opt/cray/pe/lib64:$HIP_LIB_PATH:/opt/cray/pe/gcc/11.2.0/snos/lib64:/opt/cray/xpmem/default/lib64&quot;
export APPTAINER_CONTAINLIBS=&quot;/usr/lib64/libcxi.so.1,/usr/lib64/libjson-c.so.3,/lib64/libtinfo.so.6,/usr/lib64/libnl-3.so.200,/usr/lib64/libgfortran.so.5,/usr/lib64/libjansson.so.4&quot;
MYFS=$(findmnt -r -T . | tail -1 |cut -d ' ' -f 1)
export http_proxy=http://proxy.ccs.ornl.gov:3128/
export https_proxy=http://proxy.ccs.ornl.gov:3128/
export no_proxy='localhost,127.0.0.0/8,*.ccs.ornl.gov'
export PERMANENT_DIR=$MEMBERWORK/$SLURM_JOB_ACCOUNT
export BINDS=/usr/share/libdrm,/var/spool/slurmd,/opt/cray,${MYFS},${PERMANENT_DIR}
export FI_CXI_RX_MATCH_MODE=hybrid
export COMEX_EAGER_THRESHOLD=32768
export MPICH_SMP_SINGLE_COPY_MODE=NONE
export OMP_NUM_THREADS=1
MYIMG=oras://ghcr.io/edoapra/nwchem-singularity/nwchem-dev.mpich3.4.2:
export APPTAINER_CACHEDIR=$MEMBERWORK/$SLURM_JOB_ACCOUNT/cache
mkdir -p $APPTAINER_CACHEDIR
srun  -N $SLURM_NNODES  -l  apptainer exec --bind $BINDS --workdir `pwd` $MYIMG   nwchem siosi6.nw &gt;&amp; siosi6.out.$SLURM_JOBID
</code></pre>
<h2 id="containers-shifter">Shifter<a class="headerlink" href="#containers-shifter" title="Permanent link">&para;</a></h2>
<p>Instructions for running the NWChem <a href="https://docs.nersc.gov/applications/nwchem/">Shifter images</a> are available at <a href="https://docs.nersc.gov/applications/nwchem">https://docs.nersc.gov/applications/nwchem</a>  </p>
<h2 id="containers-podman">Podman<a class="headerlink" href="#containers-podman" title="Permanent link">&para;</a></h2>
<p>Docker images could be run using podman commands</p>
<pre><code>podman run --rm --shm-size 256m --volume /tmp:/data -i -t ghcr.io/nwchemgit/nwchem-dev/amd64 xvdw.nw
</code></pre></section></div><style>.print-site-enumerate-headings #index > h1:before { content: '1 ' }

                .print-site-enumerate-headings #index h2:before { content: '1.' counter(counter-index-2) ' ' }
                .print-site-enumerate-headings #index h2 {  counter-reset: counter-index-3  counter-index-4  counter-index-5  counter-index-6 ;  counter-increment: counter-index-2 }
            
                .print-site-enumerate-headings #index h3:before { content: '1.' counter(counter-index-2) '.' counter(counter-index-3) ' ' }
                .print-site-enumerate-headings #index h3 {  counter-reset: counter-index-4  counter-index-5  counter-index-6 ;  counter-increment: counter-index-3 }
            
                .print-site-enumerate-headings #index h4:before { content: '1.' counter(counter-index-2) '.' counter(counter-index-3) '.' counter(counter-index-4) ' ' }
                .print-site-enumerate-headings #index h4 {  counter-reset: counter-index-5  counter-index-6 ;  counter-increment: counter-index-4 }
            
                .print-site-enumerate-headings #index h5:before { content: '1.' counter(counter-index-2) '.' counter(counter-index-3) '.' counter(counter-index-4) '.' counter(counter-index-5) ' ' }
                .print-site-enumerate-headings #index h5 {  counter-reset: counter-index-6 ;  counter-increment: counter-index-5 }
            
                .print-site-enumerate-headings #index h6:before { content: '1.' counter(counter-index-2) '.' counter(counter-index-3) '.' counter(counter-index-4) '.' counter(counter-index-5) '.' counter(counter-index-6) ' ' }
                .print-site-enumerate-headings #index h6 {  counter-increment: counter-index-6 }
            
.print-site-enumerate-headings #section-3 > h1:before { content: '3 ' }
.print-site-enumerate-headings #download > h1:before { content: '3.1 ' }

                .print-site-enumerate-headings #download h2:before { content: '3.1.' counter(counter-download-2) ' ' }
                .print-site-enumerate-headings #download h2 {  counter-reset: counter-download-3  counter-download-4  counter-download-5 ;  counter-increment: counter-download-2 }
            
                .print-site-enumerate-headings #download h3:before { content: '3.1.' counter(counter-download-2) '.' counter(counter-download-3) ' ' }
                .print-site-enumerate-headings #download h3 {  counter-reset: counter-download-4  counter-download-5 ;  counter-increment: counter-download-3 }
            
                .print-site-enumerate-headings #download h4:before { content: '3.1.' counter(counter-download-2) '.' counter(counter-download-3) '.' counter(counter-download-4) ' ' }
                .print-site-enumerate-headings #download h4 {  counter-reset: counter-download-5 ;  counter-increment: counter-download-4 }
            
                .print-site-enumerate-headings #download h5:before { content: '3.1.' counter(counter-download-2) '.' counter(counter-download-3) '.' counter(counter-download-4) '.' counter(counter-download-5) ' ' }
                .print-site-enumerate-headings #download h5 {  counter-increment: counter-download-5 }
            
.print-site-enumerate-headings #compiling-nwchem > h1:before { content: '3.2 ' }

                .print-site-enumerate-headings #compiling-nwchem h2:before { content: '3.2.' counter(counter-compiling-nwchem-2) ' ' }
                .print-site-enumerate-headings #compiling-nwchem h2 {  counter-reset: counter-compiling-nwchem-3  counter-compiling-nwchem-4  counter-compiling-nwchem-5 ;  counter-increment: counter-compiling-nwchem-2 }
            
                .print-site-enumerate-headings #compiling-nwchem h3:before { content: '3.2.' counter(counter-compiling-nwchem-2) '.' counter(counter-compiling-nwchem-3) ' ' }
                .print-site-enumerate-headings #compiling-nwchem h3 {  counter-reset: counter-compiling-nwchem-4  counter-compiling-nwchem-5 ;  counter-increment: counter-compiling-nwchem-3 }
            
                .print-site-enumerate-headings #compiling-nwchem h4:before { content: '3.2.' counter(counter-compiling-nwchem-2) '.' counter(counter-compiling-nwchem-3) '.' counter(counter-compiling-nwchem-4) ' ' }
                .print-site-enumerate-headings #compiling-nwchem h4 {  counter-reset: counter-compiling-nwchem-5 ;  counter-increment: counter-compiling-nwchem-4 }
            
                .print-site-enumerate-headings #compiling-nwchem h5:before { content: '3.2.' counter(counter-compiling-nwchem-2) '.' counter(counter-compiling-nwchem-3) '.' counter(counter-compiling-nwchem-4) '.' counter(counter-compiling-nwchem-5) ' ' }
                .print-site-enumerate-headings #compiling-nwchem h5 {  counter-increment: counter-compiling-nwchem-5 }
            
.print-site-enumerate-headings #section-4 > h1:before { content: '4 ' }
.print-site-enumerate-headings #getting-started > h1:before { content: '4.1 ' }

                .print-site-enumerate-headings #getting-started h2:before { content: '4.1.' counter(counter-getting-started-2) ' ' }
                .print-site-enumerate-headings #getting-started h2 {  counter-reset: counter-getting-started-3  counter-getting-started-4  counter-getting-started-5 ;  counter-increment: counter-getting-started-2 }
            
                .print-site-enumerate-headings #getting-started h3:before { content: '4.1.' counter(counter-getting-started-2) '.' counter(counter-getting-started-3) ' ' }
                .print-site-enumerate-headings #getting-started h3 {  counter-reset: counter-getting-started-4  counter-getting-started-5 ;  counter-increment: counter-getting-started-3 }
            
                .print-site-enumerate-headings #getting-started h4:before { content: '4.1.' counter(counter-getting-started-2) '.' counter(counter-getting-started-3) '.' counter(counter-getting-started-4) ' ' }
                .print-site-enumerate-headings #getting-started h4 {  counter-reset: counter-getting-started-5 ;  counter-increment: counter-getting-started-4 }
            
                .print-site-enumerate-headings #getting-started h5:before { content: '4.1.' counter(counter-getting-started-2) '.' counter(counter-getting-started-3) '.' counter(counter-getting-started-4) '.' counter(counter-getting-started-5) ' ' }
                .print-site-enumerate-headings #getting-started h5 {  counter-increment: counter-getting-started-5 }
            
.print-site-enumerate-headings #nwchem-architecture > h1:before { content: '4.2 ' }

                .print-site-enumerate-headings #nwchem-architecture h2:before { content: '4.2.' counter(counter-nwchem-architecture-2) ' ' }
                .print-site-enumerate-headings #nwchem-architecture h2 {  counter-reset: counter-nwchem-architecture-3  counter-nwchem-architecture-4  counter-nwchem-architecture-5 ;  counter-increment: counter-nwchem-architecture-2 }
            
                .print-site-enumerate-headings #nwchem-architecture h3:before { content: '4.2.' counter(counter-nwchem-architecture-2) '.' counter(counter-nwchem-architecture-3) ' ' }
                .print-site-enumerate-headings #nwchem-architecture h3 {  counter-reset: counter-nwchem-architecture-4  counter-nwchem-architecture-5 ;  counter-increment: counter-nwchem-architecture-3 }
            
                .print-site-enumerate-headings #nwchem-architecture h4:before { content: '4.2.' counter(counter-nwchem-architecture-2) '.' counter(counter-nwchem-architecture-3) '.' counter(counter-nwchem-architecture-4) ' ' }
                .print-site-enumerate-headings #nwchem-architecture h4 {  counter-reset: counter-nwchem-architecture-5 ;  counter-increment: counter-nwchem-architecture-4 }
            
                .print-site-enumerate-headings #nwchem-architecture h5:before { content: '4.2.' counter(counter-nwchem-architecture-2) '.' counter(counter-nwchem-architecture-3) '.' counter(counter-nwchem-architecture-4) '.' counter(counter-nwchem-architecture-5) ' ' }
                .print-site-enumerate-headings #nwchem-architecture h5 {  counter-increment: counter-nwchem-architecture-5 }
            
.print-site-enumerate-headings #capabilities > h1:before { content: '5 ' }

                .print-site-enumerate-headings #capabilities h2:before { content: '5.' counter(counter-capabilities-2) ' ' }
                .print-site-enumerate-headings #capabilities h2 {  counter-reset: counter-capabilities-3  counter-capabilities-4  counter-capabilities-5  counter-capabilities-6 ;  counter-increment: counter-capabilities-2 }
            
                .print-site-enumerate-headings #capabilities h3:before { content: '5.' counter(counter-capabilities-2) '.' counter(counter-capabilities-3) ' ' }
                .print-site-enumerate-headings #capabilities h3 {  counter-reset: counter-capabilities-4  counter-capabilities-5  counter-capabilities-6 ;  counter-increment: counter-capabilities-3 }
            
                .print-site-enumerate-headings #capabilities h4:before { content: '5.' counter(counter-capabilities-2) '.' counter(counter-capabilities-3) '.' counter(counter-capabilities-4) ' ' }
                .print-site-enumerate-headings #capabilities h4 {  counter-reset: counter-capabilities-5  counter-capabilities-6 ;  counter-increment: counter-capabilities-4 }
            
                .print-site-enumerate-headings #capabilities h5:before { content: '5.' counter(counter-capabilities-2) '.' counter(counter-capabilities-3) '.' counter(counter-capabilities-4) '.' counter(counter-capabilities-5) ' ' }
                .print-site-enumerate-headings #capabilities h5 {  counter-reset: counter-capabilities-6 ;  counter-increment: counter-capabilities-5 }
            
                .print-site-enumerate-headings #capabilities h6:before { content: '5.' counter(counter-capabilities-2) '.' counter(counter-capabilities-3) '.' counter(counter-capabilities-4) '.' counter(counter-capabilities-5) '.' counter(counter-capabilities-6) ' ' }
                .print-site-enumerate-headings #capabilities h6 {  counter-increment: counter-capabilities-6 }
            
.print-site-enumerate-headings #top-level > h1:before { content: '6 ' }

                .print-site-enumerate-headings #top-level h2:before { content: '6.' counter(counter-top-level-2) ' ' }
                .print-site-enumerate-headings #top-level h2 {  counter-reset: counter-top-level-3  counter-top-level-4  counter-top-level-5  counter-top-level-6 ;  counter-increment: counter-top-level-2 }
            
                .print-site-enumerate-headings #top-level h3:before { content: '6.' counter(counter-top-level-2) '.' counter(counter-top-level-3) ' ' }
                .print-site-enumerate-headings #top-level h3 {  counter-reset: counter-top-level-4  counter-top-level-5  counter-top-level-6 ;  counter-increment: counter-top-level-3 }
            
                .print-site-enumerate-headings #top-level h4:before { content: '6.' counter(counter-top-level-2) '.' counter(counter-top-level-3) '.' counter(counter-top-level-4) ' ' }
                .print-site-enumerate-headings #top-level h4 {  counter-reset: counter-top-level-5  counter-top-level-6 ;  counter-increment: counter-top-level-4 }
            
                .print-site-enumerate-headings #top-level h5:before { content: '6.' counter(counter-top-level-2) '.' counter(counter-top-level-3) '.' counter(counter-top-level-4) '.' counter(counter-top-level-5) ' ' }
                .print-site-enumerate-headings #top-level h5 {  counter-reset: counter-top-level-6 ;  counter-increment: counter-top-level-5 }
            
                .print-site-enumerate-headings #top-level h6:before { content: '6.' counter(counter-top-level-2) '.' counter(counter-top-level-3) '.' counter(counter-top-level-4) '.' counter(counter-top-level-5) '.' counter(counter-top-level-6) ' ' }
                .print-site-enumerate-headings #top-level h6 {  counter-increment: counter-top-level-6 }
            
.print-site-enumerate-headings #geometry > h1:before { content: '7 ' }

                .print-site-enumerate-headings #geometry h2:before { content: '7.' counter(counter-geometry-2) ' ' }
                .print-site-enumerate-headings #geometry h2 {  counter-reset: counter-geometry-3  counter-geometry-4  counter-geometry-5  counter-geometry-6 ;  counter-increment: counter-geometry-2 }
            
                .print-site-enumerate-headings #geometry h3:before { content: '7.' counter(counter-geometry-2) '.' counter(counter-geometry-3) ' ' }
                .print-site-enumerate-headings #geometry h3 {  counter-reset: counter-geometry-4  counter-geometry-5  counter-geometry-6 ;  counter-increment: counter-geometry-3 }
            
                .print-site-enumerate-headings #geometry h4:before { content: '7.' counter(counter-geometry-2) '.' counter(counter-geometry-3) '.' counter(counter-geometry-4) ' ' }
                .print-site-enumerate-headings #geometry h4 {  counter-reset: counter-geometry-5  counter-geometry-6 ;  counter-increment: counter-geometry-4 }
            
                .print-site-enumerate-headings #geometry h5:before { content: '7.' counter(counter-geometry-2) '.' counter(counter-geometry-3) '.' counter(counter-geometry-4) '.' counter(counter-geometry-5) ' ' }
                .print-site-enumerate-headings #geometry h5 {  counter-reset: counter-geometry-6 ;  counter-increment: counter-geometry-5 }
            
                .print-site-enumerate-headings #geometry h6:before { content: '7.' counter(counter-geometry-2) '.' counter(counter-geometry-3) '.' counter(counter-geometry-4) '.' counter(counter-geometry-5) '.' counter(counter-geometry-6) ' ' }
                .print-site-enumerate-headings #geometry h6 {  counter-increment: counter-geometry-6 }
            
.print-site-enumerate-headings #basis > h1:before { content: '8 ' }

                .print-site-enumerate-headings #basis h2:before { content: '8.' counter(counter-basis-2) ' ' }
                .print-site-enumerate-headings #basis h2 {  counter-reset: counter-basis-3  counter-basis-4  counter-basis-5  counter-basis-6 ;  counter-increment: counter-basis-2 }
            
                .print-site-enumerate-headings #basis h3:before { content: '8.' counter(counter-basis-2) '.' counter(counter-basis-3) ' ' }
                .print-site-enumerate-headings #basis h3 {  counter-reset: counter-basis-4  counter-basis-5  counter-basis-6 ;  counter-increment: counter-basis-3 }
            
                .print-site-enumerate-headings #basis h4:before { content: '8.' counter(counter-basis-2) '.' counter(counter-basis-3) '.' counter(counter-basis-4) ' ' }
                .print-site-enumerate-headings #basis h4 {  counter-reset: counter-basis-5  counter-basis-6 ;  counter-increment: counter-basis-4 }
            
                .print-site-enumerate-headings #basis h5:before { content: '8.' counter(counter-basis-2) '.' counter(counter-basis-3) '.' counter(counter-basis-4) '.' counter(counter-basis-5) ' ' }
                .print-site-enumerate-headings #basis h5 {  counter-reset: counter-basis-6 ;  counter-increment: counter-basis-5 }
            
                .print-site-enumerate-headings #basis h6:before { content: '8.' counter(counter-basis-2) '.' counter(counter-basis-3) '.' counter(counter-basis-4) '.' counter(counter-basis-5) '.' counter(counter-basis-6) ' ' }
                .print-site-enumerate-headings #basis h6 {  counter-increment: counter-basis-6 }
            
.print-site-enumerate-headings #ecp > h1:before { content: '9 ' }

                .print-site-enumerate-headings #ecp h2:before { content: '9.' counter(counter-ecp-2) ' ' }
                .print-site-enumerate-headings #ecp h2 {  counter-reset: counter-ecp-3  counter-ecp-4  counter-ecp-5  counter-ecp-6 ;  counter-increment: counter-ecp-2 }
            
                .print-site-enumerate-headings #ecp h3:before { content: '9.' counter(counter-ecp-2) '.' counter(counter-ecp-3) ' ' }
                .print-site-enumerate-headings #ecp h3 {  counter-reset: counter-ecp-4  counter-ecp-5  counter-ecp-6 ;  counter-increment: counter-ecp-3 }
            
                .print-site-enumerate-headings #ecp h4:before { content: '9.' counter(counter-ecp-2) '.' counter(counter-ecp-3) '.' counter(counter-ecp-4) ' ' }
                .print-site-enumerate-headings #ecp h4 {  counter-reset: counter-ecp-5  counter-ecp-6 ;  counter-increment: counter-ecp-4 }
            
                .print-site-enumerate-headings #ecp h5:before { content: '9.' counter(counter-ecp-2) '.' counter(counter-ecp-3) '.' counter(counter-ecp-4) '.' counter(counter-ecp-5) ' ' }
                .print-site-enumerate-headings #ecp h5 {  counter-reset: counter-ecp-6 ;  counter-increment: counter-ecp-5 }
            
                .print-site-enumerate-headings #ecp h6:before { content: '9.' counter(counter-ecp-2) '.' counter(counter-ecp-3) '.' counter(counter-ecp-4) '.' counter(counter-ecp-5) '.' counter(counter-ecp-6) ' ' }
                .print-site-enumerate-headings #ecp h6 {  counter-increment: counter-ecp-6 }
            
.print-site-enumerate-headings #relativistic-all-electron-approximations > h1:before { content: '10 ' }

                .print-site-enumerate-headings #relativistic-all-electron-approximations h2:before { content: '10.' counter(counter-relativistic-all-electron-approximations-2) ' ' }
                .print-site-enumerate-headings #relativistic-all-electron-approximations h2 {  counter-reset: counter-relativistic-all-electron-approximations-3  counter-relativistic-all-electron-approximations-4  counter-relativistic-all-electron-approximations-5  counter-relativistic-all-electron-approximations-6 ;  counter-increment: counter-relativistic-all-electron-approximations-2 }
            
                .print-site-enumerate-headings #relativistic-all-electron-approximations h3:before { content: '10.' counter(counter-relativistic-all-electron-approximations-2) '.' counter(counter-relativistic-all-electron-approximations-3) ' ' }
                .print-site-enumerate-headings #relativistic-all-electron-approximations h3 {  counter-reset: counter-relativistic-all-electron-approximations-4  counter-relativistic-all-electron-approximations-5  counter-relativistic-all-electron-approximations-6 ;  counter-increment: counter-relativistic-all-electron-approximations-3 }
            
                .print-site-enumerate-headings #relativistic-all-electron-approximations h4:before { content: '10.' counter(counter-relativistic-all-electron-approximations-2) '.' counter(counter-relativistic-all-electron-approximations-3) '.' counter(counter-relativistic-all-electron-approximations-4) ' ' }
                .print-site-enumerate-headings #relativistic-all-electron-approximations h4 {  counter-reset: counter-relativistic-all-electron-approximations-5  counter-relativistic-all-electron-approximations-6 ;  counter-increment: counter-relativistic-all-electron-approximations-4 }
            
                .print-site-enumerate-headings #relativistic-all-electron-approximations h5:before { content: '10.' counter(counter-relativistic-all-electron-approximations-2) '.' counter(counter-relativistic-all-electron-approximations-3) '.' counter(counter-relativistic-all-electron-approximations-4) '.' counter(counter-relativistic-all-electron-approximations-5) ' ' }
                .print-site-enumerate-headings #relativistic-all-electron-approximations h5 {  counter-reset: counter-relativistic-all-electron-approximations-6 ;  counter-increment: counter-relativistic-all-electron-approximations-5 }
            
                .print-site-enumerate-headings #relativistic-all-electron-approximations h6:before { content: '10.' counter(counter-relativistic-all-electron-approximations-2) '.' counter(counter-relativistic-all-electron-approximations-3) '.' counter(counter-relativistic-all-electron-approximations-4) '.' counter(counter-relativistic-all-electron-approximations-5) '.' counter(counter-relativistic-all-electron-approximations-6) ' ' }
                .print-site-enumerate-headings #relativistic-all-electron-approximations h6 {  counter-increment: counter-relativistic-all-electron-approximations-6 }
            
.print-site-enumerate-headings #section-11 > h1:before { content: '11 ' }
.print-site-enumerate-headings #hartree-fock-theory-for-molecules > h1:before { content: '11.1 ' }

                .print-site-enumerate-headings #hartree-fock-theory-for-molecules h2:before { content: '11.1.' counter(counter-hartree-fock-theory-for-molecules-2) ' ' }
                .print-site-enumerate-headings #hartree-fock-theory-for-molecules h2 {  counter-reset: counter-hartree-fock-theory-for-molecules-3  counter-hartree-fock-theory-for-molecules-4  counter-hartree-fock-theory-for-molecules-5 ;  counter-increment: counter-hartree-fock-theory-for-molecules-2 }
            
                .print-site-enumerate-headings #hartree-fock-theory-for-molecules h3:before { content: '11.1.' counter(counter-hartree-fock-theory-for-molecules-2) '.' counter(counter-hartree-fock-theory-for-molecules-3) ' ' }
                .print-site-enumerate-headings #hartree-fock-theory-for-molecules h3 {  counter-reset: counter-hartree-fock-theory-for-molecules-4  counter-hartree-fock-theory-for-molecules-5 ;  counter-increment: counter-hartree-fock-theory-for-molecules-3 }
            
                .print-site-enumerate-headings #hartree-fock-theory-for-molecules h4:before { content: '11.1.' counter(counter-hartree-fock-theory-for-molecules-2) '.' counter(counter-hartree-fock-theory-for-molecules-3) '.' counter(counter-hartree-fock-theory-for-molecules-4) ' ' }
                .print-site-enumerate-headings #hartree-fock-theory-for-molecules h4 {  counter-reset: counter-hartree-fock-theory-for-molecules-5 ;  counter-increment: counter-hartree-fock-theory-for-molecules-4 }
            
                .print-site-enumerate-headings #hartree-fock-theory-for-molecules h5:before { content: '11.1.' counter(counter-hartree-fock-theory-for-molecules-2) '.' counter(counter-hartree-fock-theory-for-molecules-3) '.' counter(counter-hartree-fock-theory-for-molecules-4) '.' counter(counter-hartree-fock-theory-for-molecules-5) ' ' }
                .print-site-enumerate-headings #hartree-fock-theory-for-molecules h5 {  counter-increment: counter-hartree-fock-theory-for-molecules-5 }
            
.print-site-enumerate-headings #density-functional-theory-for-molecules > h1:before { content: '11.2 ' }

                .print-site-enumerate-headings #density-functional-theory-for-molecules h2:before { content: '11.2.' counter(counter-density-functional-theory-for-molecules-2) ' ' }
                .print-site-enumerate-headings #density-functional-theory-for-molecules h2 {  counter-reset: counter-density-functional-theory-for-molecules-3  counter-density-functional-theory-for-molecules-4  counter-density-functional-theory-for-molecules-5 ;  counter-increment: counter-density-functional-theory-for-molecules-2 }
            
                .print-site-enumerate-headings #density-functional-theory-for-molecules h3:before { content: '11.2.' counter(counter-density-functional-theory-for-molecules-2) '.' counter(counter-density-functional-theory-for-molecules-3) ' ' }
                .print-site-enumerate-headings #density-functional-theory-for-molecules h3 {  counter-reset: counter-density-functional-theory-for-molecules-4  counter-density-functional-theory-for-molecules-5 ;  counter-increment: counter-density-functional-theory-for-molecules-3 }
            
                .print-site-enumerate-headings #density-functional-theory-for-molecules h4:before { content: '11.2.' counter(counter-density-functional-theory-for-molecules-2) '.' counter(counter-density-functional-theory-for-molecules-3) '.' counter(counter-density-functional-theory-for-molecules-4) ' ' }
                .print-site-enumerate-headings #density-functional-theory-for-molecules h4 {  counter-reset: counter-density-functional-theory-for-molecules-5 ;  counter-increment: counter-density-functional-theory-for-molecules-4 }
            
                .print-site-enumerate-headings #density-functional-theory-for-molecules h5:before { content: '11.2.' counter(counter-density-functional-theory-for-molecules-2) '.' counter(counter-density-functional-theory-for-molecules-3) '.' counter(counter-density-functional-theory-for-molecules-4) '.' counter(counter-density-functional-theory-for-molecules-5) ' ' }
                .print-site-enumerate-headings #density-functional-theory-for-molecules h5 {  counter-increment: counter-density-functional-theory-for-molecules-5 }
            
.print-site-enumerate-headings #excited-state-calculations > h1:before { content: '11.3 ' }

                .print-site-enumerate-headings #excited-state-calculations h2:before { content: '11.3.' counter(counter-excited-state-calculations-2) ' ' }
                .print-site-enumerate-headings #excited-state-calculations h2 {  counter-reset: counter-excited-state-calculations-3  counter-excited-state-calculations-4  counter-excited-state-calculations-5 ;  counter-increment: counter-excited-state-calculations-2 }
            
                .print-site-enumerate-headings #excited-state-calculations h3:before { content: '11.3.' counter(counter-excited-state-calculations-2) '.' counter(counter-excited-state-calculations-3) ' ' }
                .print-site-enumerate-headings #excited-state-calculations h3 {  counter-reset: counter-excited-state-calculations-4  counter-excited-state-calculations-5 ;  counter-increment: counter-excited-state-calculations-3 }
            
                .print-site-enumerate-headings #excited-state-calculations h4:before { content: '11.3.' counter(counter-excited-state-calculations-2) '.' counter(counter-excited-state-calculations-3) '.' counter(counter-excited-state-calculations-4) ' ' }
                .print-site-enumerate-headings #excited-state-calculations h4 {  counter-reset: counter-excited-state-calculations-5 ;  counter-increment: counter-excited-state-calculations-4 }
            
                .print-site-enumerate-headings #excited-state-calculations h5:before { content: '11.3.' counter(counter-excited-state-calculations-2) '.' counter(counter-excited-state-calculations-3) '.' counter(counter-excited-state-calculations-4) '.' counter(counter-excited-state-calculations-5) ' ' }
                .print-site-enumerate-headings #excited-state-calculations h5 {  counter-increment: counter-excited-state-calculations-5 }
            
.print-site-enumerate-headings #rt-tddft > h1:before { content: '11.4 ' }

                .print-site-enumerate-headings #rt-tddft h2:before { content: '11.4.' counter(counter-rt-tddft-2) ' ' }
                .print-site-enumerate-headings #rt-tddft h2 {  counter-reset: counter-rt-tddft-3  counter-rt-tddft-4  counter-rt-tddft-5 ;  counter-increment: counter-rt-tddft-2 }
            
                .print-site-enumerate-headings #rt-tddft h3:before { content: '11.4.' counter(counter-rt-tddft-2) '.' counter(counter-rt-tddft-3) ' ' }
                .print-site-enumerate-headings #rt-tddft h3 {  counter-reset: counter-rt-tddft-4  counter-rt-tddft-5 ;  counter-increment: counter-rt-tddft-3 }
            
                .print-site-enumerate-headings #rt-tddft h4:before { content: '11.4.' counter(counter-rt-tddft-2) '.' counter(counter-rt-tddft-3) '.' counter(counter-rt-tddft-4) ' ' }
                .print-site-enumerate-headings #rt-tddft h4 {  counter-reset: counter-rt-tddft-5 ;  counter-increment: counter-rt-tddft-4 }
            
                .print-site-enumerate-headings #rt-tddft h5:before { content: '11.4.' counter(counter-rt-tddft-2) '.' counter(counter-rt-tddft-3) '.' counter(counter-rt-tddft-4) '.' counter(counter-rt-tddft-5) ' ' }
                .print-site-enumerate-headings #rt-tddft h5 {  counter-increment: counter-rt-tddft-5 }
            
.print-site-enumerate-headings #plane-wave-density-functional-theory > h1:before { content: '11.5 ' }

                .print-site-enumerate-headings #plane-wave-density-functional-theory h2:before { content: '11.5.' counter(counter-plane-wave-density-functional-theory-2) ' ' }
                .print-site-enumerate-headings #plane-wave-density-functional-theory h2 {  counter-reset: counter-plane-wave-density-functional-theory-3  counter-plane-wave-density-functional-theory-4  counter-plane-wave-density-functional-theory-5 ;  counter-increment: counter-plane-wave-density-functional-theory-2 }
            
                .print-site-enumerate-headings #plane-wave-density-functional-theory h3:before { content: '11.5.' counter(counter-plane-wave-density-functional-theory-2) '.' counter(counter-plane-wave-density-functional-theory-3) ' ' }
                .print-site-enumerate-headings #plane-wave-density-functional-theory h3 {  counter-reset: counter-plane-wave-density-functional-theory-4  counter-plane-wave-density-functional-theory-5 ;  counter-increment: counter-plane-wave-density-functional-theory-3 }
            
                .print-site-enumerate-headings #plane-wave-density-functional-theory h4:before { content: '11.5.' counter(counter-plane-wave-density-functional-theory-2) '.' counter(counter-plane-wave-density-functional-theory-3) '.' counter(counter-plane-wave-density-functional-theory-4) ' ' }
                .print-site-enumerate-headings #plane-wave-density-functional-theory h4 {  counter-reset: counter-plane-wave-density-functional-theory-5 ;  counter-increment: counter-plane-wave-density-functional-theory-4 }
            
                .print-site-enumerate-headings #plane-wave-density-functional-theory h5:before { content: '11.5.' counter(counter-plane-wave-density-functional-theory-2) '.' counter(counter-plane-wave-density-functional-theory-3) '.' counter(counter-plane-wave-density-functional-theory-4) '.' counter(counter-plane-wave-density-functional-theory-5) ' ' }
                .print-site-enumerate-headings #plane-wave-density-functional-theory h5 {  counter-increment: counter-plane-wave-density-functional-theory-5 }
            
.print-site-enumerate-headings #tce > h1:before { content: '11.6 ' }

                .print-site-enumerate-headings #tce h2:before { content: '11.6.' counter(counter-tce-2) ' ' }
                .print-site-enumerate-headings #tce h2 {  counter-reset: counter-tce-3  counter-tce-4  counter-tce-5 ;  counter-increment: counter-tce-2 }
            
                .print-site-enumerate-headings #tce h3:before { content: '11.6.' counter(counter-tce-2) '.' counter(counter-tce-3) ' ' }
                .print-site-enumerate-headings #tce h3 {  counter-reset: counter-tce-4  counter-tce-5 ;  counter-increment: counter-tce-3 }
            
                .print-site-enumerate-headings #tce h4:before { content: '11.6.' counter(counter-tce-2) '.' counter(counter-tce-3) '.' counter(counter-tce-4) ' ' }
                .print-site-enumerate-headings #tce h4 {  counter-reset: counter-tce-5 ;  counter-increment: counter-tce-4 }
            
                .print-site-enumerate-headings #tce h5:before { content: '11.6.' counter(counter-tce-2) '.' counter(counter-tce-3) '.' counter(counter-tce-4) '.' counter(counter-tce-5) ' ' }
                .print-site-enumerate-headings #tce h5 {  counter-increment: counter-tce-5 }
            
.print-site-enumerate-headings #mp2 > h1:before { content: '11.7 ' }

                .print-site-enumerate-headings #mp2 h2:before { content: '11.7.' counter(counter-mp2-2) ' ' }
                .print-site-enumerate-headings #mp2 h2 {  counter-reset: counter-mp2-3  counter-mp2-4  counter-mp2-5 ;  counter-increment: counter-mp2-2 }
            
                .print-site-enumerate-headings #mp2 h3:before { content: '11.7.' counter(counter-mp2-2) '.' counter(counter-mp2-3) ' ' }
                .print-site-enumerate-headings #mp2 h3 {  counter-reset: counter-mp2-4  counter-mp2-5 ;  counter-increment: counter-mp2-3 }
            
                .print-site-enumerate-headings #mp2 h4:before { content: '11.7.' counter(counter-mp2-2) '.' counter(counter-mp2-3) '.' counter(counter-mp2-4) ' ' }
                .print-site-enumerate-headings #mp2 h4 {  counter-reset: counter-mp2-5 ;  counter-increment: counter-mp2-4 }
            
                .print-site-enumerate-headings #mp2 h5:before { content: '11.7.' counter(counter-mp2-2) '.' counter(counter-mp2-3) '.' counter(counter-mp2-4) '.' counter(counter-mp2-5) ' ' }
                .print-site-enumerate-headings #mp2 h5 {  counter-increment: counter-mp2-5 }
            
.print-site-enumerate-headings #ccsd > h1:before { content: '11.8 ' }

                .print-site-enumerate-headings #ccsd h2:before { content: '11.8.' counter(counter-ccsd-2) ' ' }
                .print-site-enumerate-headings #ccsd h2 {  counter-reset: counter-ccsd-3  counter-ccsd-4  counter-ccsd-5 ;  counter-increment: counter-ccsd-2 }
            
                .print-site-enumerate-headings #ccsd h3:before { content: '11.8.' counter(counter-ccsd-2) '.' counter(counter-ccsd-3) ' ' }
                .print-site-enumerate-headings #ccsd h3 {  counter-reset: counter-ccsd-4  counter-ccsd-5 ;  counter-increment: counter-ccsd-3 }
            
                .print-site-enumerate-headings #ccsd h4:before { content: '11.8.' counter(counter-ccsd-2) '.' counter(counter-ccsd-3) '.' counter(counter-ccsd-4) ' ' }
                .print-site-enumerate-headings #ccsd h4 {  counter-reset: counter-ccsd-5 ;  counter-increment: counter-ccsd-4 }
            
                .print-site-enumerate-headings #ccsd h5:before { content: '11.8.' counter(counter-ccsd-2) '.' counter(counter-ccsd-3) '.' counter(counter-ccsd-4) '.' counter(counter-ccsd-5) ' ' }
                .print-site-enumerate-headings #ccsd h5 {  counter-increment: counter-ccsd-5 }
            
.print-site-enumerate-headings #multiconfiguration_scf > h1:before { content: '11.9 ' }

                .print-site-enumerate-headings #multiconfiguration_scf h2:before { content: '11.9.' counter(counter-multiconfiguration_scf-2) ' ' }
                .print-site-enumerate-headings #multiconfiguration_scf h2 {  counter-reset: counter-multiconfiguration_scf-3  counter-multiconfiguration_scf-4  counter-multiconfiguration_scf-5 ;  counter-increment: counter-multiconfiguration_scf-2 }
            
                .print-site-enumerate-headings #multiconfiguration_scf h3:before { content: '11.9.' counter(counter-multiconfiguration_scf-2) '.' counter(counter-multiconfiguration_scf-3) ' ' }
                .print-site-enumerate-headings #multiconfiguration_scf h3 {  counter-reset: counter-multiconfiguration_scf-4  counter-multiconfiguration_scf-5 ;  counter-increment: counter-multiconfiguration_scf-3 }
            
                .print-site-enumerate-headings #multiconfiguration_scf h4:before { content: '11.9.' counter(counter-multiconfiguration_scf-2) '.' counter(counter-multiconfiguration_scf-3) '.' counter(counter-multiconfiguration_scf-4) ' ' }
                .print-site-enumerate-headings #multiconfiguration_scf h4 {  counter-reset: counter-multiconfiguration_scf-5 ;  counter-increment: counter-multiconfiguration_scf-4 }
            
                .print-site-enumerate-headings #multiconfiguration_scf h5:before { content: '11.9.' counter(counter-multiconfiguration_scf-2) '.' counter(counter-multiconfiguration_scf-3) '.' counter(counter-multiconfiguration_scf-4) '.' counter(counter-multiconfiguration_scf-5) ' ' }
                .print-site-enumerate-headings #multiconfiguration_scf h5 {  counter-increment: counter-multiconfiguration_scf-5 }
            
.print-site-enumerate-headings #gw > h1:before { content: '11.10 ' }

                .print-site-enumerate-headings #gw h2:before { content: '11.10.' counter(counter-gw-2) ' ' }
                .print-site-enumerate-headings #gw h2 {  counter-reset: counter-gw-3  counter-gw-4  counter-gw-5 ;  counter-increment: counter-gw-2 }
            
                .print-site-enumerate-headings #gw h3:before { content: '11.10.' counter(counter-gw-2) '.' counter(counter-gw-3) ' ' }
                .print-site-enumerate-headings #gw h3 {  counter-reset: counter-gw-4  counter-gw-5 ;  counter-increment: counter-gw-3 }
            
                .print-site-enumerate-headings #gw h4:before { content: '11.10.' counter(counter-gw-2) '.' counter(counter-gw-3) '.' counter(counter-gw-4) ' ' }
                .print-site-enumerate-headings #gw h4 {  counter-reset: counter-gw-5 ;  counter-increment: counter-gw-4 }
            
                .print-site-enumerate-headings #gw h5:before { content: '11.10.' counter(counter-gw-2) '.' counter(counter-gw-3) '.' counter(counter-gw-4) '.' counter(counter-gw-5) ' ' }
                .print-site-enumerate-headings #gw h5 {  counter-increment: counter-gw-5 }
            
.print-site-enumerate-headings #xtb > h1:before { content: '11.11 ' }

                .print-site-enumerate-headings #xtb h2:before { content: '11.11.' counter(counter-xtb-2) ' ' }
                .print-site-enumerate-headings #xtb h2 {  counter-reset: counter-xtb-3  counter-xtb-4  counter-xtb-5 ;  counter-increment: counter-xtb-2 }
            
                .print-site-enumerate-headings #xtb h3:before { content: '11.11.' counter(counter-xtb-2) '.' counter(counter-xtb-3) ' ' }
                .print-site-enumerate-headings #xtb h3 {  counter-reset: counter-xtb-4  counter-xtb-5 ;  counter-increment: counter-xtb-3 }
            
                .print-site-enumerate-headings #xtb h4:before { content: '11.11.' counter(counter-xtb-2) '.' counter(counter-xtb-3) '.' counter(counter-xtb-4) ' ' }
                .print-site-enumerate-headings #xtb h4 {  counter-reset: counter-xtb-5 ;  counter-increment: counter-xtb-4 }
            
                .print-site-enumerate-headings #xtb h5:before { content: '11.11.' counter(counter-xtb-2) '.' counter(counter-xtb-3) '.' counter(counter-xtb-4) '.' counter(counter-xtb-5) ' ' }
                .print-site-enumerate-headings #xtb h5 {  counter-increment: counter-xtb-5 }
            
.print-site-enumerate-headings #section-12 > h1:before { content: '12 ' }
.print-site-enumerate-headings #plane-wave-density-functional-theory > h1:before { content: '12.1 ' }

                .print-site-enumerate-headings #plane-wave-density-functional-theory h2:before { content: '12.1.' counter(counter-plane-wave-density-functional-theory-2) ' ' }
                .print-site-enumerate-headings #plane-wave-density-functional-theory h2 {  counter-reset: counter-plane-wave-density-functional-theory-3  counter-plane-wave-density-functional-theory-4  counter-plane-wave-density-functional-theory-5 ;  counter-increment: counter-plane-wave-density-functional-theory-2 }
            
                .print-site-enumerate-headings #plane-wave-density-functional-theory h3:before { content: '12.1.' counter(counter-plane-wave-density-functional-theory-2) '.' counter(counter-plane-wave-density-functional-theory-3) ' ' }
                .print-site-enumerate-headings #plane-wave-density-functional-theory h3 {  counter-reset: counter-plane-wave-density-functional-theory-4  counter-plane-wave-density-functional-theory-5 ;  counter-increment: counter-plane-wave-density-functional-theory-3 }
            
                .print-site-enumerate-headings #plane-wave-density-functional-theory h4:before { content: '12.1.' counter(counter-plane-wave-density-functional-theory-2) '.' counter(counter-plane-wave-density-functional-theory-3) '.' counter(counter-plane-wave-density-functional-theory-4) ' ' }
                .print-site-enumerate-headings #plane-wave-density-functional-theory h4 {  counter-reset: counter-plane-wave-density-functional-theory-5 ;  counter-increment: counter-plane-wave-density-functional-theory-4 }
            
                .print-site-enumerate-headings #plane-wave-density-functional-theory h5:before { content: '12.1.' counter(counter-plane-wave-density-functional-theory-2) '.' counter(counter-plane-wave-density-functional-theory-3) '.' counter(counter-plane-wave-density-functional-theory-4) '.' counter(counter-plane-wave-density-functional-theory-5) ' ' }
                .print-site-enumerate-headings #plane-wave-density-functional-theory h5 {  counter-increment: counter-plane-wave-density-functional-theory-5 }
            
.print-site-enumerate-headings #gaussian-basis-aimd > h1:before { content: '12.2 ' }

                .print-site-enumerate-headings #gaussian-basis-aimd h2:before { content: '12.2.' counter(counter-gaussian-basis-aimd-2) ' ' }
                .print-site-enumerate-headings #gaussian-basis-aimd h2 {  counter-reset: counter-gaussian-basis-aimd-3  counter-gaussian-basis-aimd-4  counter-gaussian-basis-aimd-5 ;  counter-increment: counter-gaussian-basis-aimd-2 }
            
                .print-site-enumerate-headings #gaussian-basis-aimd h3:before { content: '12.2.' counter(counter-gaussian-basis-aimd-2) '.' counter(counter-gaussian-basis-aimd-3) ' ' }
                .print-site-enumerate-headings #gaussian-basis-aimd h3 {  counter-reset: counter-gaussian-basis-aimd-4  counter-gaussian-basis-aimd-5 ;  counter-increment: counter-gaussian-basis-aimd-3 }
            
                .print-site-enumerate-headings #gaussian-basis-aimd h4:before { content: '12.2.' counter(counter-gaussian-basis-aimd-2) '.' counter(counter-gaussian-basis-aimd-3) '.' counter(counter-gaussian-basis-aimd-4) ' ' }
                .print-site-enumerate-headings #gaussian-basis-aimd h4 {  counter-reset: counter-gaussian-basis-aimd-5 ;  counter-increment: counter-gaussian-basis-aimd-4 }
            
                .print-site-enumerate-headings #gaussian-basis-aimd h5:before { content: '12.2.' counter(counter-gaussian-basis-aimd-2) '.' counter(counter-gaussian-basis-aimd-3) '.' counter(counter-gaussian-basis-aimd-4) '.' counter(counter-gaussian-basis-aimd-5) ' ' }
                .print-site-enumerate-headings #gaussian-basis-aimd h5 {  counter-increment: counter-gaussian-basis-aimd-5 }
            
.print-site-enumerate-headings #section-13 > h1:before { content: '13 ' }
.print-site-enumerate-headings #solvation-models > h1:before { content: '13.1 ' }

                .print-site-enumerate-headings #solvation-models h2:before { content: '13.1.' counter(counter-solvation-models-2) ' ' }
                .print-site-enumerate-headings #solvation-models h2 {  counter-reset: counter-solvation-models-3  counter-solvation-models-4  counter-solvation-models-5 ;  counter-increment: counter-solvation-models-2 }
            
                .print-site-enumerate-headings #solvation-models h3:before { content: '13.1.' counter(counter-solvation-models-2) '.' counter(counter-solvation-models-3) ' ' }
                .print-site-enumerate-headings #solvation-models h3 {  counter-reset: counter-solvation-models-4  counter-solvation-models-5 ;  counter-increment: counter-solvation-models-3 }
            
                .print-site-enumerate-headings #solvation-models h4:before { content: '13.1.' counter(counter-solvation-models-2) '.' counter(counter-solvation-models-3) '.' counter(counter-solvation-models-4) ' ' }
                .print-site-enumerate-headings #solvation-models h4 {  counter-reset: counter-solvation-models-5 ;  counter-increment: counter-solvation-models-4 }
            
                .print-site-enumerate-headings #solvation-models h5:before { content: '13.1.' counter(counter-solvation-models-2) '.' counter(counter-solvation-models-3) '.' counter(counter-solvation-models-4) '.' counter(counter-solvation-models-5) ' ' }
                .print-site-enumerate-headings #solvation-models h5 {  counter-increment: counter-solvation-models-5 }
            
.print-site-enumerate-headings #vem-model > h1:before { content: '13.2 ' }

                .print-site-enumerate-headings #vem-model h2:before { content: '13.2.' counter(counter-vem-model-2) ' ' }
                .print-site-enumerate-headings #vem-model h2 {  counter-reset: counter-vem-model-3  counter-vem-model-4  counter-vem-model-5 ;  counter-increment: counter-vem-model-2 }
            
                .print-site-enumerate-headings #vem-model h3:before { content: '13.2.' counter(counter-vem-model-2) '.' counter(counter-vem-model-3) ' ' }
                .print-site-enumerate-headings #vem-model h3 {  counter-reset: counter-vem-model-4  counter-vem-model-5 ;  counter-increment: counter-vem-model-3 }
            
                .print-site-enumerate-headings #vem-model h4:before { content: '13.2.' counter(counter-vem-model-2) '.' counter(counter-vem-model-3) '.' counter(counter-vem-model-4) ' ' }
                .print-site-enumerate-headings #vem-model h4 {  counter-reset: counter-vem-model-5 ;  counter-increment: counter-vem-model-4 }
            
                .print-site-enumerate-headings #vem-model h5:before { content: '13.2.' counter(counter-vem-model-2) '.' counter(counter-vem-model-3) '.' counter(counter-vem-model-4) '.' counter(counter-vem-model-5) ' ' }
                .print-site-enumerate-headings #vem-model h5 {  counter-increment: counter-vem-model-5 }
            
.print-site-enumerate-headings #oniom > h1:before { content: '13.3 ' }

                .print-site-enumerate-headings #oniom h2:before { content: '13.3.' counter(counter-oniom-2) ' ' }
                .print-site-enumerate-headings #oniom h2 {  counter-reset: counter-oniom-3  counter-oniom-4  counter-oniom-5 ;  counter-increment: counter-oniom-2 }
            
                .print-site-enumerate-headings #oniom h3:before { content: '13.3.' counter(counter-oniom-2) '.' counter(counter-oniom-3) ' ' }
                .print-site-enumerate-headings #oniom h3 {  counter-reset: counter-oniom-4  counter-oniom-5 ;  counter-increment: counter-oniom-3 }
            
                .print-site-enumerate-headings #oniom h4:before { content: '13.3.' counter(counter-oniom-2) '.' counter(counter-oniom-3) '.' counter(counter-oniom-4) ' ' }
                .print-site-enumerate-headings #oniom h4 {  counter-reset: counter-oniom-5 ;  counter-increment: counter-oniom-4 }
            
                .print-site-enumerate-headings #oniom h5:before { content: '13.3.' counter(counter-oniom-2) '.' counter(counter-oniom-3) '.' counter(counter-oniom-4) '.' counter(counter-oniom-5) ' ' }
                .print-site-enumerate-headings #oniom h5 {  counter-increment: counter-oniom-5 }
            
.print-site-enumerate-headings #section-13-4 > h1:before { content: '13.4 ' }
.print-site-enumerate-headings #qmmm_introduction > h1:before { content: '13.4.1 ' }

                .print-site-enumerate-headings #qmmm_introduction h2:before { content: '13.4.1.' counter(counter-qmmm_introduction-2) ' ' }
                .print-site-enumerate-headings #qmmm_introduction h2 {  counter-reset: counter-qmmm_introduction-3  counter-qmmm_introduction-4 ;  counter-increment: counter-qmmm_introduction-2 }
            
                .print-site-enumerate-headings #qmmm_introduction h3:before { content: '13.4.1.' counter(counter-qmmm_introduction-2) '.' counter(counter-qmmm_introduction-3) ' ' }
                .print-site-enumerate-headings #qmmm_introduction h3 {  counter-reset: counter-qmmm_introduction-4 ;  counter-increment: counter-qmmm_introduction-3 }
            
                .print-site-enumerate-headings #qmmm_introduction h4:before { content: '13.4.1.' counter(counter-qmmm_introduction-2) '.' counter(counter-qmmm_introduction-3) '.' counter(counter-qmmm_introduction-4) ' ' }
                .print-site-enumerate-headings #qmmm_introduction h4 {  counter-increment: counter-qmmm_introduction-4 }
            
.print-site-enumerate-headings #qmmm > h1:before { content: '13.4.2 ' }

                .print-site-enumerate-headings #qmmm h2:before { content: '13.4.2.' counter(counter-qmmm-2) ' ' }
                .print-site-enumerate-headings #qmmm h2 {  counter-reset: counter-qmmm-3  counter-qmmm-4 ;  counter-increment: counter-qmmm-2 }
            
                .print-site-enumerate-headings #qmmm h3:before { content: '13.4.2.' counter(counter-qmmm-2) '.' counter(counter-qmmm-3) ' ' }
                .print-site-enumerate-headings #qmmm h3 {  counter-reset: counter-qmmm-4 ;  counter-increment: counter-qmmm-3 }
            
                .print-site-enumerate-headings #qmmm h4:before { content: '13.4.2.' counter(counter-qmmm-2) '.' counter(counter-qmmm-3) '.' counter(counter-qmmm-4) ' ' }
                .print-site-enumerate-headings #qmmm h4 {  counter-increment: counter-qmmm-4 }
            
.print-site-enumerate-headings #section-13-4-3 > h1:before { content: '13.4.3 ' }
.print-site-enumerate-headings #qmmm_restart_and_topology_files > h1:before { content: '13.4.3.1 ' }

                .print-site-enumerate-headings #qmmm_restart_and_topology_files h2:before { content: '13.4.3.1.' counter(counter-qmmm_restart_and_topology_files-2) ' ' }
                .print-site-enumerate-headings #qmmm_restart_and_topology_files h2 {  counter-reset: counter-qmmm_restart_and_topology_files-3 ;  counter-increment: counter-qmmm_restart_and_topology_files-2 }
            
                .print-site-enumerate-headings #qmmm_restart_and_topology_files h3:before { content: '13.4.3.1.' counter(counter-qmmm_restart_and_topology_files-2) '.' counter(counter-qmmm_restart_and_topology_files-3) ' ' }
                .print-site-enumerate-headings #qmmm_restart_and_topology_files h3 {  counter-increment: counter-qmmm_restart_and_topology_files-3 }
            
.print-site-enumerate-headings #qmmm_preparation_prerequisites > h1:before { content: '13.4.3.2 ' }

                .print-site-enumerate-headings #qmmm_preparation_prerequisites h2:before { content: '13.4.3.2.' counter(counter-qmmm_preparation_prerequisites-2) ' ' }
                .print-site-enumerate-headings #qmmm_preparation_prerequisites h2 {  counter-reset: counter-qmmm_preparation_prerequisites-3 ;  counter-increment: counter-qmmm_preparation_prerequisites-2 }
            
                .print-site-enumerate-headings #qmmm_preparation_prerequisites h3:before { content: '13.4.3.2.' counter(counter-qmmm_preparation_prerequisites-2) '.' counter(counter-qmmm_preparation_prerequisites-3) ' ' }
                .print-site-enumerate-headings #qmmm_preparation_prerequisites h3 {  counter-increment: counter-qmmm_preparation_prerequisites-3 }
            
.print-site-enumerate-headings #qmmm_preparation_basic > h1:before { content: '13.4.3.3 ' }

                .print-site-enumerate-headings #qmmm_preparation_basic h2:before { content: '13.4.3.3.' counter(counter-qmmm_preparation_basic-2) ' ' }
                .print-site-enumerate-headings #qmmm_preparation_basic h2 {  counter-reset: counter-qmmm_preparation_basic-3 ;  counter-increment: counter-qmmm_preparation_basic-2 }
            
                .print-site-enumerate-headings #qmmm_preparation_basic h3:before { content: '13.4.3.3.' counter(counter-qmmm_preparation_basic-2) '.' counter(counter-qmmm_preparation_basic-3) ' ' }
                .print-site-enumerate-headings #qmmm_preparation_basic h3 {  counter-increment: counter-qmmm_preparation_basic-3 }
            
.print-site-enumerate-headings #qmmm_preparation_solvation > h1:before { content: '13.4.3.4 ' }

                .print-site-enumerate-headings #qmmm_preparation_solvation h2:before { content: '13.4.3.4.' counter(counter-qmmm_preparation_solvation-2) ' ' }
                .print-site-enumerate-headings #qmmm_preparation_solvation h2 {  counter-reset: counter-qmmm_preparation_solvation-3 ;  counter-increment: counter-qmmm_preparation_solvation-2 }
            
                .print-site-enumerate-headings #qmmm_preparation_solvation h3:before { content: '13.4.3.4.' counter(counter-qmmm_preparation_solvation-2) '.' counter(counter-qmmm_preparation_solvation-3) ' ' }
                .print-site-enumerate-headings #qmmm_preparation_solvation h3 {  counter-increment: counter-qmmm_preparation_solvation-3 }
            
.print-site-enumerate-headings #qmmm_preparation_constraints > h1:before { content: '13.4.3.5 ' }

                .print-site-enumerate-headings #qmmm_preparation_constraints h2:before { content: '13.4.3.5.' counter(counter-qmmm_preparation_constraints-2) ' ' }
                .print-site-enumerate-headings #qmmm_preparation_constraints h2 {  counter-reset: counter-qmmm_preparation_constraints-3 ;  counter-increment: counter-qmmm_preparation_constraints-2 }
            
                .print-site-enumerate-headings #qmmm_preparation_constraints h3:before { content: '13.4.3.5.' counter(counter-qmmm_preparation_constraints-2) '.' counter(counter-qmmm_preparation_constraints-3) ' ' }
                .print-site-enumerate-headings #qmmm_preparation_constraints h3 {  counter-increment: counter-qmmm_preparation_constraints-3 }
            
.print-site-enumerate-headings #section-13-4-4 > h1:before { content: '13.4.4 ' }
.print-site-enumerate-headings #qmmm_input_file > h1:before { content: '13.4.4.1 ' }

                .print-site-enumerate-headings #qmmm_input_file h2:before { content: '13.4.4.1.' counter(counter-qmmm_input_file-2) ' ' }
                .print-site-enumerate-headings #qmmm_input_file h2 {  counter-reset: counter-qmmm_input_file-3 ;  counter-increment: counter-qmmm_input_file-2 }
            
                .print-site-enumerate-headings #qmmm_input_file h3:before { content: '13.4.4.1.' counter(counter-qmmm_input_file-2) '.' counter(counter-qmmm_input_file-3) ' ' }
                .print-site-enumerate-headings #qmmm_input_file h3 {  counter-increment: counter-qmmm_input_file-3 }
            
.print-site-enumerate-headings #qmmm_qm_parameters > h1:before { content: '13.4.4.2 ' }

                .print-site-enumerate-headings #qmmm_qm_parameters h2:before { content: '13.4.4.2.' counter(counter-qmmm_qm_parameters-2) ' ' }
                .print-site-enumerate-headings #qmmm_qm_parameters h2 {  counter-reset: counter-qmmm_qm_parameters-3 ;  counter-increment: counter-qmmm_qm_parameters-2 }
            
                .print-site-enumerate-headings #qmmm_qm_parameters h3:before { content: '13.4.4.2.' counter(counter-qmmm_qm_parameters-2) '.' counter(counter-qmmm_qm_parameters-3) ' ' }
                .print-site-enumerate-headings #qmmm_qm_parameters h3 {  counter-increment: counter-qmmm_qm_parameters-3 }
            
.print-site-enumerate-headings #qmmm_mm_parameters > h1:before { content: '13.4.4.3 ' }

                .print-site-enumerate-headings #qmmm_mm_parameters h2:before { content: '13.4.4.3.' counter(counter-qmmm_mm_parameters-2) ' ' }
                .print-site-enumerate-headings #qmmm_mm_parameters h2 {  counter-reset: counter-qmmm_mm_parameters-3 ;  counter-increment: counter-qmmm_mm_parameters-2 }
            
                .print-site-enumerate-headings #qmmm_mm_parameters h3:before { content: '13.4.4.3.' counter(counter-qmmm_mm_parameters-2) '.' counter(counter-qmmm_mm_parameters-3) ' ' }
                .print-site-enumerate-headings #qmmm_mm_parameters h3 {  counter-increment: counter-qmmm_mm_parameters-3 }
            
.print-site-enumerate-headings #qmmm_qmmm_parameters > h1:before { content: '13.4.4.4 ' }

                .print-site-enumerate-headings #qmmm_qmmm_parameters h2:before { content: '13.4.4.4.' counter(counter-qmmm_qmmm_parameters-2) ' ' }
                .print-site-enumerate-headings #qmmm_qmmm_parameters h2 {  counter-reset: counter-qmmm_qmmm_parameters-3 ;  counter-increment: counter-qmmm_qmmm_parameters-2 }
            
                .print-site-enumerate-headings #qmmm_qmmm_parameters h3:before { content: '13.4.4.4.' counter(counter-qmmm_qmmm_parameters-2) '.' counter(counter-qmmm_qmmm_parameters-3) ' ' }
                .print-site-enumerate-headings #qmmm_qmmm_parameters h3 {  counter-increment: counter-qmmm_qmmm_parameters-3 }
            
.print-site-enumerate-headings #section-13-4-5 > h1:before { content: '13.4.5 ' }
.print-site-enumerate-headings #qmmm_sp_energy > h1:before { content: '13.4.5.1 ' }

                .print-site-enumerate-headings #qmmm_sp_energy h2:before { content: '13.4.5.1.' counter(counter-qmmm_sp_energy-2) ' ' }
                .print-site-enumerate-headings #qmmm_sp_energy h2 {  counter-reset: counter-qmmm_sp_energy-3 ;  counter-increment: counter-qmmm_sp_energy-2 }
            
                .print-site-enumerate-headings #qmmm_sp_energy h3:before { content: '13.4.5.1.' counter(counter-qmmm_sp_energy-2) '.' counter(counter-qmmm_sp_energy-3) ' ' }
                .print-site-enumerate-headings #qmmm_sp_energy h3 {  counter-increment: counter-qmmm_sp_energy-3 }
            
.print-site-enumerate-headings #qmmm_excited_states > h1:before { content: '13.4.5.2 ' }

                .print-site-enumerate-headings #qmmm_excited_states h2:before { content: '13.4.5.2.' counter(counter-qmmm_excited_states-2) ' ' }
                .print-site-enumerate-headings #qmmm_excited_states h2 {  counter-reset: counter-qmmm_excited_states-3 ;  counter-increment: counter-qmmm_excited_states-2 }
            
                .print-site-enumerate-headings #qmmm_excited_states h3:before { content: '13.4.5.2.' counter(counter-qmmm_excited_states-2) '.' counter(counter-qmmm_excited_states-3) ' ' }
                .print-site-enumerate-headings #qmmm_excited_states h3 {  counter-increment: counter-qmmm_excited_states-3 }
            
.print-site-enumerate-headings #qmmm_sp_property > h1:before { content: '13.4.5.3 ' }

                .print-site-enumerate-headings #qmmm_sp_property h2:before { content: '13.4.5.3.' counter(counter-qmmm_sp_property-2) ' ' }
                .print-site-enumerate-headings #qmmm_sp_property h2 {  counter-reset: counter-qmmm_sp_property-3 ;  counter-increment: counter-qmmm_sp_property-2 }
            
                .print-site-enumerate-headings #qmmm_sp_property h3:before { content: '13.4.5.3.' counter(counter-qmmm_sp_property-2) '.' counter(counter-qmmm_sp_property-3) ' ' }
                .print-site-enumerate-headings #qmmm_sp_property h3 {  counter-increment: counter-qmmm_sp_property-3 }
            
.print-site-enumerate-headings #qmmm_esp > h1:before { content: '13.4.5.4 ' }

                .print-site-enumerate-headings #qmmm_esp h2:before { content: '13.4.5.4.' counter(counter-qmmm_esp-2) ' ' }
                .print-site-enumerate-headings #qmmm_esp h2 {  counter-reset: counter-qmmm_esp-3 ;  counter-increment: counter-qmmm_esp-2 }
            
                .print-site-enumerate-headings #qmmm_esp h3:before { content: '13.4.5.4.' counter(counter-qmmm_esp-2) '.' counter(counter-qmmm_esp-3) ' ' }
                .print-site-enumerate-headings #qmmm_esp h3 {  counter-increment: counter-qmmm_esp-3 }
            
.print-site-enumerate-headings #section-13-4-6 > h1:before { content: '13.4.6 ' }
.print-site-enumerate-headings #qmmm_optimization > h1:before { content: '13.4.6.1 ' }

                .print-site-enumerate-headings #qmmm_optimization h2:before { content: '13.4.6.1.' counter(counter-qmmm_optimization-2) ' ' }
                .print-site-enumerate-headings #qmmm_optimization h2 {  counter-reset: counter-qmmm_optimization-3 ;  counter-increment: counter-qmmm_optimization-2 }
            
                .print-site-enumerate-headings #qmmm_optimization h3:before { content: '13.4.6.1.' counter(counter-qmmm_optimization-2) '.' counter(counter-qmmm_optimization-3) ' ' }
                .print-site-enumerate-headings #qmmm_optimization h3 {  counter-increment: counter-qmmm_optimization-3 }
            
.print-site-enumerate-headings #qmmm_transition_states > h1:before { content: '13.4.6.2 ' }

                .print-site-enumerate-headings #qmmm_transition_states h2:before { content: '13.4.6.2.' counter(counter-qmmm_transition_states-2) ' ' }
                .print-site-enumerate-headings #qmmm_transition_states h2 {  counter-reset: counter-qmmm_transition_states-3 ;  counter-increment: counter-qmmm_transition_states-2 }
            
                .print-site-enumerate-headings #qmmm_transition_states h3:before { content: '13.4.6.2.' counter(counter-qmmm_transition_states-2) '.' counter(counter-qmmm_transition_states-3) ' ' }
                .print-site-enumerate-headings #qmmm_transition_states h3 {  counter-increment: counter-qmmm_transition_states-3 }
            
.print-site-enumerate-headings #qmmm_freq > h1:before { content: '13.4.6.3 ' }

                .print-site-enumerate-headings #qmmm_freq h2:before { content: '13.4.6.3.' counter(counter-qmmm_freq-2) ' ' }
                .print-site-enumerate-headings #qmmm_freq h2 {  counter-reset: counter-qmmm_freq-3 ;  counter-increment: counter-qmmm_freq-2 }
            
                .print-site-enumerate-headings #qmmm_freq h3:before { content: '13.4.6.3.' counter(counter-qmmm_freq-2) '.' counter(counter-qmmm_freq-3) ' ' }
                .print-site-enumerate-headings #qmmm_freq h3 {  counter-increment: counter-qmmm_freq-3 }
            
.print-site-enumerate-headings #qmmm_neb_calculations > h1:before { content: '13.4.6.4 ' }

                .print-site-enumerate-headings #qmmm_neb_calculations h2:before { content: '13.4.6.4.' counter(counter-qmmm_neb_calculations-2) ' ' }
                .print-site-enumerate-headings #qmmm_neb_calculations h2 {  counter-reset: counter-qmmm_neb_calculations-3 ;  counter-increment: counter-qmmm_neb_calculations-2 }
            
                .print-site-enumerate-headings #qmmm_neb_calculations h3:before { content: '13.4.6.4.' counter(counter-qmmm_neb_calculations-2) '.' counter(counter-qmmm_neb_calculations-3) ' ' }
                .print-site-enumerate-headings #qmmm_neb_calculations h3 {  counter-increment: counter-qmmm_neb_calculations-3 }
            
.print-site-enumerate-headings #qmmm_dynamics > h1:before { content: '13.4.7 ' }

                .print-site-enumerate-headings #qmmm_dynamics h2:before { content: '13.4.7.' counter(counter-qmmm_dynamics-2) ' ' }
                .print-site-enumerate-headings #qmmm_dynamics h2 {  counter-reset: counter-qmmm_dynamics-3  counter-qmmm_dynamics-4 ;  counter-increment: counter-qmmm_dynamics-2 }
            
                .print-site-enumerate-headings #qmmm_dynamics h3:before { content: '13.4.7.' counter(counter-qmmm_dynamics-2) '.' counter(counter-qmmm_dynamics-3) ' ' }
                .print-site-enumerate-headings #qmmm_dynamics h3 {  counter-reset: counter-qmmm_dynamics-4 ;  counter-increment: counter-qmmm_dynamics-3 }
            
                .print-site-enumerate-headings #qmmm_dynamics h4:before { content: '13.4.7.' counter(counter-qmmm_dynamics-2) '.' counter(counter-qmmm_dynamics-3) '.' counter(counter-qmmm_dynamics-4) ' ' }
                .print-site-enumerate-headings #qmmm_dynamics h4 {  counter-increment: counter-qmmm_dynamics-4 }
            
.print-site-enumerate-headings #qmmm_free_energy > h1:before { content: '13.4.8 ' }

                .print-site-enumerate-headings #qmmm_free_energy h2:before { content: '13.4.8.' counter(counter-qmmm_free_energy-2) ' ' }
                .print-site-enumerate-headings #qmmm_free_energy h2 {  counter-reset: counter-qmmm_free_energy-3  counter-qmmm_free_energy-4 ;  counter-increment: counter-qmmm_free_energy-2 }
            
                .print-site-enumerate-headings #qmmm_free_energy h3:before { content: '13.4.8.' counter(counter-qmmm_free_energy-2) '.' counter(counter-qmmm_free_energy-3) ' ' }
                .print-site-enumerate-headings #qmmm_free_energy h3 {  counter-reset: counter-qmmm_free_energy-4 ;  counter-increment: counter-qmmm_free_energy-3 }
            
                .print-site-enumerate-headings #qmmm_free_energy h4:before { content: '13.4.8.' counter(counter-qmmm_free_energy-2) '.' counter(counter-qmmm_free_energy-3) '.' counter(counter-qmmm_free_energy-4) ' ' }
                .print-site-enumerate-headings #qmmm_free_energy h4 {  counter-increment: counter-qmmm_free_energy-4 }
            
.print-site-enumerate-headings #qmmm_appendix > h1:before { content: '13.4.9 ' }

                .print-site-enumerate-headings #qmmm_appendix h2:before { content: '13.4.9.' counter(counter-qmmm_appendix-2) ' ' }
                .print-site-enumerate-headings #qmmm_appendix h2 {  counter-reset: counter-qmmm_appendix-3  counter-qmmm_appendix-4 ;  counter-increment: counter-qmmm_appendix-2 }
            
                .print-site-enumerate-headings #qmmm_appendix h3:before { content: '13.4.9.' counter(counter-qmmm_appendix-2) '.' counter(counter-qmmm_appendix-3) ' ' }
                .print-site-enumerate-headings #qmmm_appendix h3 {  counter-reset: counter-qmmm_appendix-4 ;  counter-increment: counter-qmmm_appendix-3 }
            
                .print-site-enumerate-headings #qmmm_appendix h4:before { content: '13.4.9.' counter(counter-qmmm_appendix-2) '.' counter(counter-qmmm_appendix-3) '.' counter(counter-qmmm_appendix-4) ' ' }
                .print-site-enumerate-headings #qmmm_appendix h4 {  counter-increment: counter-qmmm_appendix-4 }
            
.print-site-enumerate-headings #bq > h1:before { content: '13.5 ' }

                .print-site-enumerate-headings #bq h2:before { content: '13.5.' counter(counter-bq-2) ' ' }
                .print-site-enumerate-headings #bq h2 {  counter-reset: counter-bq-3  counter-bq-4  counter-bq-5 ;  counter-increment: counter-bq-2 }
            
                .print-site-enumerate-headings #bq h3:before { content: '13.5.' counter(counter-bq-2) '.' counter(counter-bq-3) ' ' }
                .print-site-enumerate-headings #bq h3 {  counter-reset: counter-bq-4  counter-bq-5 ;  counter-increment: counter-bq-3 }
            
                .print-site-enumerate-headings #bq h4:before { content: '13.5.' counter(counter-bq-2) '.' counter(counter-bq-3) '.' counter(counter-bq-4) ' ' }
                .print-site-enumerate-headings #bq h4 {  counter-reset: counter-bq-5 ;  counter-increment: counter-bq-4 }
            
                .print-site-enumerate-headings #bq h5:before { content: '13.5.' counter(counter-bq-2) '.' counter(counter-bq-3) '.' counter(counter-bq-4) '.' counter(counter-bq-5) ' ' }
                .print-site-enumerate-headings #bq h5 {  counter-increment: counter-bq-5 }
            
.print-site-enumerate-headings #1d-rism > h1:before { content: '13.6 ' }

                .print-site-enumerate-headings #1d-rism h2:before { content: '13.6.' counter(counter-1d-rism-2) ' ' }
                .print-site-enumerate-headings #1d-rism h2 {  counter-reset: counter-1d-rism-3  counter-1d-rism-4  counter-1d-rism-5 ;  counter-increment: counter-1d-rism-2 }
            
                .print-site-enumerate-headings #1d-rism h3:before { content: '13.6.' counter(counter-1d-rism-2) '.' counter(counter-1d-rism-3) ' ' }
                .print-site-enumerate-headings #1d-rism h3 {  counter-reset: counter-1d-rism-4  counter-1d-rism-5 ;  counter-increment: counter-1d-rism-3 }
            
                .print-site-enumerate-headings #1d-rism h4:before { content: '13.6.' counter(counter-1d-rism-2) '.' counter(counter-1d-rism-3) '.' counter(counter-1d-rism-4) ' ' }
                .print-site-enumerate-headings #1d-rism h4 {  counter-reset: counter-1d-rism-5 ;  counter-increment: counter-1d-rism-4 }
            
                .print-site-enumerate-headings #1d-rism h5:before { content: '13.6.' counter(counter-1d-rism-2) '.' counter(counter-1d-rism-3) '.' counter(counter-1d-rism-4) '.' counter(counter-1d-rism-5) ' ' }
                .print-site-enumerate-headings #1d-rism h5 {  counter-increment: counter-1d-rism-5 }
            
.print-site-enumerate-headings #section-14 > h1:before { content: '14 ' }
.print-site-enumerate-headings #geometry-optimization > h1:before { content: '14.1 ' }

                .print-site-enumerate-headings #geometry-optimization h2:before { content: '14.1.' counter(counter-geometry-optimization-2) ' ' }
                .print-site-enumerate-headings #geometry-optimization h2 {  counter-reset: counter-geometry-optimization-3  counter-geometry-optimization-4  counter-geometry-optimization-5 ;  counter-increment: counter-geometry-optimization-2 }
            
                .print-site-enumerate-headings #geometry-optimization h3:before { content: '14.1.' counter(counter-geometry-optimization-2) '.' counter(counter-geometry-optimization-3) ' ' }
                .print-site-enumerate-headings #geometry-optimization h3 {  counter-reset: counter-geometry-optimization-4  counter-geometry-optimization-5 ;  counter-increment: counter-geometry-optimization-3 }
            
                .print-site-enumerate-headings #geometry-optimization h4:before { content: '14.1.' counter(counter-geometry-optimization-2) '.' counter(counter-geometry-optimization-3) '.' counter(counter-geometry-optimization-4) ' ' }
                .print-site-enumerate-headings #geometry-optimization h4 {  counter-reset: counter-geometry-optimization-5 ;  counter-increment: counter-geometry-optimization-4 }
            
                .print-site-enumerate-headings #geometry-optimization h5:before { content: '14.1.' counter(counter-geometry-optimization-2) '.' counter(counter-geometry-optimization-3) '.' counter(counter-geometry-optimization-4) '.' counter(counter-geometry-optimization-5) ' ' }
                .print-site-enumerate-headings #geometry-optimization h5 {  counter-increment: counter-geometry-optimization-5 }
            
.print-site-enumerate-headings #hessians-and-vibrational-frequencies > h1:before { content: '14.2 ' }

                .print-site-enumerate-headings #hessians-and-vibrational-frequencies h2:before { content: '14.2.' counter(counter-hessians-and-vibrational-frequencies-2) ' ' }
                .print-site-enumerate-headings #hessians-and-vibrational-frequencies h2 {  counter-reset: counter-hessians-and-vibrational-frequencies-3  counter-hessians-and-vibrational-frequencies-4  counter-hessians-and-vibrational-frequencies-5 ;  counter-increment: counter-hessians-and-vibrational-frequencies-2 }
            
                .print-site-enumerate-headings #hessians-and-vibrational-frequencies h3:before { content: '14.2.' counter(counter-hessians-and-vibrational-frequencies-2) '.' counter(counter-hessians-and-vibrational-frequencies-3) ' ' }
                .print-site-enumerate-headings #hessians-and-vibrational-frequencies h3 {  counter-reset: counter-hessians-and-vibrational-frequencies-4  counter-hessians-and-vibrational-frequencies-5 ;  counter-increment: counter-hessians-and-vibrational-frequencies-3 }
            
                .print-site-enumerate-headings #hessians-and-vibrational-frequencies h4:before { content: '14.2.' counter(counter-hessians-and-vibrational-frequencies-2) '.' counter(counter-hessians-and-vibrational-frequencies-3) '.' counter(counter-hessians-and-vibrational-frequencies-4) ' ' }
                .print-site-enumerate-headings #hessians-and-vibrational-frequencies h4 {  counter-reset: counter-hessians-and-vibrational-frequencies-5 ;  counter-increment: counter-hessians-and-vibrational-frequencies-4 }
            
                .print-site-enumerate-headings #hessians-and-vibrational-frequencies h5:before { content: '14.2.' counter(counter-hessians-and-vibrational-frequencies-2) '.' counter(counter-hessians-and-vibrational-frequencies-3) '.' counter(counter-hessians-and-vibrational-frequencies-4) '.' counter(counter-hessians-and-vibrational-frequencies-5) ' ' }
                .print-site-enumerate-headings #hessians-and-vibrational-frequencies h5 {  counter-increment: counter-hessians-and-vibrational-frequencies-5 }
            
.print-site-enumerate-headings #constraints > h1:before { content: '14.3 ' }

                .print-site-enumerate-headings #constraints h2:before { content: '14.3.' counter(counter-constraints-2) ' ' }
                .print-site-enumerate-headings #constraints h2 {  counter-reset: counter-constraints-3  counter-constraints-4  counter-constraints-5 ;  counter-increment: counter-constraints-2 }
            
                .print-site-enumerate-headings #constraints h3:before { content: '14.3.' counter(counter-constraints-2) '.' counter(counter-constraints-3) ' ' }
                .print-site-enumerate-headings #constraints h3 {  counter-reset: counter-constraints-4  counter-constraints-5 ;  counter-increment: counter-constraints-3 }
            
                .print-site-enumerate-headings #constraints h4:before { content: '14.3.' counter(counter-constraints-2) '.' counter(counter-constraints-3) '.' counter(counter-constraints-4) ' ' }
                .print-site-enumerate-headings #constraints h4 {  counter-reset: counter-constraints-5 ;  counter-increment: counter-constraints-4 }
            
                .print-site-enumerate-headings #constraints h5:before { content: '14.3.' counter(counter-constraints-2) '.' counter(counter-constraints-3) '.' counter(counter-constraints-4) '.' counter(counter-constraints-5) ' ' }
                .print-site-enumerate-headings #constraints h5 {  counter-increment: counter-constraints-5 }
            
.print-site-enumerate-headings #nudged-elastic-band-and-zero-temperature-string-methods > h1:before { content: '14.4 ' }

                .print-site-enumerate-headings #nudged-elastic-band-and-zero-temperature-string-methods h2:before { content: '14.4.' counter(counter-nudged-elastic-band-and-zero-temperature-string-methods-2) ' ' }
                .print-site-enumerate-headings #nudged-elastic-band-and-zero-temperature-string-methods h2 {  counter-reset: counter-nudged-elastic-band-and-zero-temperature-string-methods-3  counter-nudged-elastic-band-and-zero-temperature-string-methods-4  counter-nudged-elastic-band-and-zero-temperature-string-methods-5 ;  counter-increment: counter-nudged-elastic-band-and-zero-temperature-string-methods-2 }
            
                .print-site-enumerate-headings #nudged-elastic-band-and-zero-temperature-string-methods h3:before { content: '14.4.' counter(counter-nudged-elastic-band-and-zero-temperature-string-methods-2) '.' counter(counter-nudged-elastic-band-and-zero-temperature-string-methods-3) ' ' }
                .print-site-enumerate-headings #nudged-elastic-band-and-zero-temperature-string-methods h3 {  counter-reset: counter-nudged-elastic-band-and-zero-temperature-string-methods-4  counter-nudged-elastic-band-and-zero-temperature-string-methods-5 ;  counter-increment: counter-nudged-elastic-band-and-zero-temperature-string-methods-3 }
            
                .print-site-enumerate-headings #nudged-elastic-band-and-zero-temperature-string-methods h4:before { content: '14.4.' counter(counter-nudged-elastic-band-and-zero-temperature-string-methods-2) '.' counter(counter-nudged-elastic-band-and-zero-temperature-string-methods-3) '.' counter(counter-nudged-elastic-band-and-zero-temperature-string-methods-4) ' ' }
                .print-site-enumerate-headings #nudged-elastic-band-and-zero-temperature-string-methods h4 {  counter-reset: counter-nudged-elastic-band-and-zero-temperature-string-methods-5 ;  counter-increment: counter-nudged-elastic-band-and-zero-temperature-string-methods-4 }
            
                .print-site-enumerate-headings #nudged-elastic-band-and-zero-temperature-string-methods h5:before { content: '14.4.' counter(counter-nudged-elastic-band-and-zero-temperature-string-methods-2) '.' counter(counter-nudged-elastic-band-and-zero-temperature-string-methods-3) '.' counter(counter-nudged-elastic-band-and-zero-temperature-string-methods-4) '.' counter(counter-nudged-elastic-band-and-zero-temperature-string-methods-5) ' ' }
                .print-site-enumerate-headings #nudged-elastic-band-and-zero-temperature-string-methods h5 {  counter-increment: counter-nudged-elastic-band-and-zero-temperature-string-methods-5 }
            
.print-site-enumerate-headings #section-15 > h1:before { content: '15 ' }
.print-site-enumerate-headings #properties > h1:before { content: '15.1 ' }

                .print-site-enumerate-headings #properties h2:before { content: '15.1.' counter(counter-properties-2) ' ' }
                .print-site-enumerate-headings #properties h2 {  counter-reset: counter-properties-3  counter-properties-4  counter-properties-5 ;  counter-increment: counter-properties-2 }
            
                .print-site-enumerate-headings #properties h3:before { content: '15.1.' counter(counter-properties-2) '.' counter(counter-properties-3) ' ' }
                .print-site-enumerate-headings #properties h3 {  counter-reset: counter-properties-4  counter-properties-5 ;  counter-increment: counter-properties-3 }
            
                .print-site-enumerate-headings #properties h4:before { content: '15.1.' counter(counter-properties-2) '.' counter(counter-properties-3) '.' counter(counter-properties-4) ' ' }
                .print-site-enumerate-headings #properties h4 {  counter-reset: counter-properties-5 ;  counter-increment: counter-properties-4 }
            
                .print-site-enumerate-headings #properties h5:before { content: '15.1.' counter(counter-properties-2) '.' counter(counter-properties-3) '.' counter(counter-properties-4) '.' counter(counter-properties-5) ' ' }
                .print-site-enumerate-headings #properties h5 {  counter-increment: counter-properties-5 }
            
.print-site-enumerate-headings #esp > h1:before { content: '15.2 ' }

                .print-site-enumerate-headings #esp h2:before { content: '15.2.' counter(counter-esp-2) ' ' }
                .print-site-enumerate-headings #esp h2 {  counter-reset: counter-esp-3  counter-esp-4  counter-esp-5 ;  counter-increment: counter-esp-2 }
            
                .print-site-enumerate-headings #esp h3:before { content: '15.2.' counter(counter-esp-2) '.' counter(counter-esp-3) ' ' }
                .print-site-enumerate-headings #esp h3 {  counter-reset: counter-esp-4  counter-esp-5 ;  counter-increment: counter-esp-3 }
            
                .print-site-enumerate-headings #esp h4:before { content: '15.2.' counter(counter-esp-2) '.' counter(counter-esp-3) '.' counter(counter-esp-4) ' ' }
                .print-site-enumerate-headings #esp h4 {  counter-reset: counter-esp-5 ;  counter-increment: counter-esp-4 }
            
                .print-site-enumerate-headings #esp h5:before { content: '15.2.' counter(counter-esp-2) '.' counter(counter-esp-3) '.' counter(counter-esp-4) '.' counter(counter-esp-5) ' ' }
                .print-site-enumerate-headings #esp h5 {  counter-increment: counter-esp-5 }
            
.print-site-enumerate-headings #dplot > h1:before { content: '15.3 ' }

                .print-site-enumerate-headings #dplot h2:before { content: '15.3.' counter(counter-dplot-2) ' ' }
                .print-site-enumerate-headings #dplot h2 {  counter-reset: counter-dplot-3  counter-dplot-4  counter-dplot-5 ;  counter-increment: counter-dplot-2 }
            
                .print-site-enumerate-headings #dplot h3:before { content: '15.3.' counter(counter-dplot-2) '.' counter(counter-dplot-3) ' ' }
                .print-site-enumerate-headings #dplot h3 {  counter-reset: counter-dplot-4  counter-dplot-5 ;  counter-increment: counter-dplot-3 }
            
                .print-site-enumerate-headings #dplot h4:before { content: '15.3.' counter(counter-dplot-2) '.' counter(counter-dplot-3) '.' counter(counter-dplot-4) ' ' }
                .print-site-enumerate-headings #dplot h4 {  counter-reset: counter-dplot-5 ;  counter-increment: counter-dplot-4 }
            
                .print-site-enumerate-headings #dplot h5:before { content: '15.3.' counter(counter-dplot-2) '.' counter(counter-dplot-3) '.' counter(counter-dplot-4) '.' counter(counter-dplot-5) ' ' }
                .print-site-enumerate-headings #dplot h5 {  counter-increment: counter-dplot-5 }
            
.print-site-enumerate-headings #section-16 > h1:before { content: '16 ' }
.print-site-enumerate-headings #electron-transfer > h1:before { content: '16.1 ' }

                .print-site-enumerate-headings #electron-transfer h2:before { content: '16.1.' counter(counter-electron-transfer-2) ' ' }
                .print-site-enumerate-headings #electron-transfer h2 {  counter-reset: counter-electron-transfer-3  counter-electron-transfer-4  counter-electron-transfer-5 ;  counter-increment: counter-electron-transfer-2 }
            
                .print-site-enumerate-headings #electron-transfer h3:before { content: '16.1.' counter(counter-electron-transfer-2) '.' counter(counter-electron-transfer-3) ' ' }
                .print-site-enumerate-headings #electron-transfer h3 {  counter-reset: counter-electron-transfer-4  counter-electron-transfer-5 ;  counter-increment: counter-electron-transfer-3 }
            
                .print-site-enumerate-headings #electron-transfer h4:before { content: '16.1.' counter(counter-electron-transfer-2) '.' counter(counter-electron-transfer-3) '.' counter(counter-electron-transfer-4) ' ' }
                .print-site-enumerate-headings #electron-transfer h4 {  counter-reset: counter-electron-transfer-5 ;  counter-increment: counter-electron-transfer-4 }
            
                .print-site-enumerate-headings #electron-transfer h5:before { content: '16.1.' counter(counter-electron-transfer-2) '.' counter(counter-electron-transfer-3) '.' counter(counter-electron-transfer-4) '.' counter(counter-electron-transfer-5) ' ' }
                .print-site-enumerate-headings #electron-transfer h5 {  counter-increment: counter-electron-transfer-5 }
            
.print-site-enumerate-headings #python > h1:before { content: '16.2 ' }

                .print-site-enumerate-headings #python h2:before { content: '16.2.' counter(counter-python-2) ' ' }
                .print-site-enumerate-headings #python h2 {  counter-reset: counter-python-3  counter-python-4  counter-python-5 ;  counter-increment: counter-python-2 }
            
                .print-site-enumerate-headings #python h3:before { content: '16.2.' counter(counter-python-2) '.' counter(counter-python-3) ' ' }
                .print-site-enumerate-headings #python h3 {  counter-reset: counter-python-4  counter-python-5 ;  counter-increment: counter-python-3 }
            
                .print-site-enumerate-headings #python h4:before { content: '16.2.' counter(counter-python-2) '.' counter(counter-python-3) '.' counter(counter-python-4) ' ' }
                .print-site-enumerate-headings #python h4 {  counter-reset: counter-python-5 ;  counter-increment: counter-python-4 }
            
                .print-site-enumerate-headings #python h5:before { content: '16.2.' counter(counter-python-2) '.' counter(counter-python-3) '.' counter(counter-python-4) '.' counter(counter-python-5) ' ' }
                .print-site-enumerate-headings #python h5 {  counter-increment: counter-python-5 }
            
.print-site-enumerate-headings #quantum > h1:before { content: '16.3 ' }

                .print-site-enumerate-headings #quantum h2:before { content: '16.3.' counter(counter-quantum-2) ' ' }
                .print-site-enumerate-headings #quantum h2 {  counter-reset: counter-quantum-3  counter-quantum-4  counter-quantum-5 ;  counter-increment: counter-quantum-2 }
            
                .print-site-enumerate-headings #quantum h3:before { content: '16.3.' counter(counter-quantum-2) '.' counter(counter-quantum-3) ' ' }
                .print-site-enumerate-headings #quantum h3 {  counter-reset: counter-quantum-4  counter-quantum-5 ;  counter-increment: counter-quantum-3 }
            
                .print-site-enumerate-headings #quantum h4:before { content: '16.3.' counter(counter-quantum-2) '.' counter(counter-quantum-3) '.' counter(counter-quantum-4) ' ' }
                .print-site-enumerate-headings #quantum h4 {  counter-reset: counter-quantum-5 ;  counter-increment: counter-quantum-4 }
            
                .print-site-enumerate-headings #quantum h5:before { content: '16.3.' counter(counter-quantum-2) '.' counter(counter-quantum-3) '.' counter(counter-quantum-4) '.' counter(counter-quantum-5) ' ' }
                .print-site-enumerate-headings #quantum h5 {  counter-increment: counter-quantum-5 }
            
.print-site-enumerate-headings #vscf > h1:before { content: '16.4 ' }

                .print-site-enumerate-headings #vscf h2:before { content: '16.4.' counter(counter-vscf-2) ' ' }
                .print-site-enumerate-headings #vscf h2 {  counter-reset: counter-vscf-3  counter-vscf-4  counter-vscf-5 ;  counter-increment: counter-vscf-2 }
            
                .print-site-enumerate-headings #vscf h3:before { content: '16.4.' counter(counter-vscf-2) '.' counter(counter-vscf-3) ' ' }
                .print-site-enumerate-headings #vscf h3 {  counter-reset: counter-vscf-4  counter-vscf-5 ;  counter-increment: counter-vscf-3 }
            
                .print-site-enumerate-headings #vscf h4:before { content: '16.4.' counter(counter-vscf-2) '.' counter(counter-vscf-3) '.' counter(counter-vscf-4) ' ' }
                .print-site-enumerate-headings #vscf h4 {  counter-reset: counter-vscf-5 ;  counter-increment: counter-vscf-4 }
            
                .print-site-enumerate-headings #vscf h5:before { content: '16.4.' counter(counter-vscf-2) '.' counter(counter-vscf-3) '.' counter(counter-vscf-4) '.' counter(counter-vscf-5) ' ' }
                .print-site-enumerate-headings #vscf h5 {  counter-increment: counter-vscf-5 }
            
.print-site-enumerate-headings #ccca_method > h1:before { content: '16.5 ' }

                .print-site-enumerate-headings #ccca_method h2:before { content: '16.5.' counter(counter-ccca_method-2) ' ' }
                .print-site-enumerate-headings #ccca_method h2 {  counter-reset: counter-ccca_method-3  counter-ccca_method-4  counter-ccca_method-5 ;  counter-increment: counter-ccca_method-2 }
            
                .print-site-enumerate-headings #ccca_method h3:before { content: '16.5.' counter(counter-ccca_method-2) '.' counter(counter-ccca_method-3) ' ' }
                .print-site-enumerate-headings #ccca_method h3 {  counter-reset: counter-ccca_method-4  counter-ccca_method-5 ;  counter-increment: counter-ccca_method-3 }
            
                .print-site-enumerate-headings #ccca_method h4:before { content: '16.5.' counter(counter-ccca_method-2) '.' counter(counter-ccca_method-3) '.' counter(counter-ccca_method-4) ' ' }
                .print-site-enumerate-headings #ccca_method h4 {  counter-reset: counter-ccca_method-5 ;  counter-increment: counter-ccca_method-4 }
            
                .print-site-enumerate-headings #ccca_method h5:before { content: '16.5.' counter(counter-ccca_method-2) '.' counter(counter-ccca_method-3) '.' counter(counter-ccca_method-4) '.' counter(counter-ccca_method-5) ' ' }
                .print-site-enumerate-headings #ccca_method h5 {  counter-increment: counter-ccca_method-5 }
            
.print-site-enumerate-headings #interface > h1:before { content: '16.6 ' }

                .print-site-enumerate-headings #interface h2:before { content: '16.6.' counter(counter-interface-2) ' ' }
                .print-site-enumerate-headings #interface h2 {  counter-reset: counter-interface-3  counter-interface-4  counter-interface-5 ;  counter-increment: counter-interface-2 }
            
                .print-site-enumerate-headings #interface h3:before { content: '16.6.' counter(counter-interface-2) '.' counter(counter-interface-3) ' ' }
                .print-site-enumerate-headings #interface h3 {  counter-reset: counter-interface-4  counter-interface-5 ;  counter-increment: counter-interface-3 }
            
                .print-site-enumerate-headings #interface h4:before { content: '16.6.' counter(counter-interface-2) '.' counter(counter-interface-3) '.' counter(counter-interface-4) ' ' }
                .print-site-enumerate-headings #interface h4 {  counter-reset: counter-interface-5 ;  counter-increment: counter-interface-4 }
            
                .print-site-enumerate-headings #interface h5:before { content: '16.6.' counter(counter-interface-2) '.' counter(counter-interface-3) '.' counter(counter-interface-4) '.' counter(counter-interface-5) ' ' }
                .print-site-enumerate-headings #interface h5 {  counter-increment: counter-interface-5 }
            
.print-site-enumerate-headings #section-17 > h1:before { content: '17 ' }
.print-site-enumerate-headings #prepare > h1:before { content: '17.1 ' }

                .print-site-enumerate-headings #prepare h2:before { content: '17.1.' counter(counter-prepare-2) ' ' }
                .print-site-enumerate-headings #prepare h2 {  counter-reset: counter-prepare-3  counter-prepare-4  counter-prepare-5 ;  counter-increment: counter-prepare-2 }
            
                .print-site-enumerate-headings #prepare h3:before { content: '17.1.' counter(counter-prepare-2) '.' counter(counter-prepare-3) ' ' }
                .print-site-enumerate-headings #prepare h3 {  counter-reset: counter-prepare-4  counter-prepare-5 ;  counter-increment: counter-prepare-3 }
            
                .print-site-enumerate-headings #prepare h4:before { content: '17.1.' counter(counter-prepare-2) '.' counter(counter-prepare-3) '.' counter(counter-prepare-4) ' ' }
                .print-site-enumerate-headings #prepare h4 {  counter-reset: counter-prepare-5 ;  counter-increment: counter-prepare-4 }
            
                .print-site-enumerate-headings #prepare h5:before { content: '17.1.' counter(counter-prepare-2) '.' counter(counter-prepare-3) '.' counter(counter-prepare-4) '.' counter(counter-prepare-5) ' ' }
                .print-site-enumerate-headings #prepare h5 {  counter-increment: counter-prepare-5 }
            
.print-site-enumerate-headings #md > h1:before { content: '17.2 ' }

                .print-site-enumerate-headings #md h2:before { content: '17.2.' counter(counter-md-2) ' ' }
                .print-site-enumerate-headings #md h2 {  counter-reset: counter-md-3  counter-md-4  counter-md-5 ;  counter-increment: counter-md-2 }
            
                .print-site-enumerate-headings #md h3:before { content: '17.2.' counter(counter-md-2) '.' counter(counter-md-3) ' ' }
                .print-site-enumerate-headings #md h3 {  counter-reset: counter-md-4  counter-md-5 ;  counter-increment: counter-md-3 }
            
                .print-site-enumerate-headings #md h4:before { content: '17.2.' counter(counter-md-2) '.' counter(counter-md-3) '.' counter(counter-md-4) ' ' }
                .print-site-enumerate-headings #md h4 {  counter-reset: counter-md-5 ;  counter-increment: counter-md-4 }
            
                .print-site-enumerate-headings #md h5:before { content: '17.2.' counter(counter-md-2) '.' counter(counter-md-3) '.' counter(counter-md-4) '.' counter(counter-md-5) ' ' }
                .print-site-enumerate-headings #md h5 {  counter-increment: counter-md-5 }
            
.print-site-enumerate-headings #analysis > h1:before { content: '17.3 ' }

                .print-site-enumerate-headings #analysis h2:before { content: '17.3.' counter(counter-analysis-2) ' ' }
                .print-site-enumerate-headings #analysis h2 {  counter-reset: counter-analysis-3  counter-analysis-4  counter-analysis-5 ;  counter-increment: counter-analysis-2 }
            
                .print-site-enumerate-headings #analysis h3:before { content: '17.3.' counter(counter-analysis-2) '.' counter(counter-analysis-3) ' ' }
                .print-site-enumerate-headings #analysis h3 {  counter-reset: counter-analysis-4  counter-analysis-5 ;  counter-increment: counter-analysis-3 }
            
                .print-site-enumerate-headings #analysis h4:before { content: '17.3.' counter(counter-analysis-2) '.' counter(counter-analysis-3) '.' counter(counter-analysis-4) ' ' }
                .print-site-enumerate-headings #analysis h4 {  counter-reset: counter-analysis-5 ;  counter-increment: counter-analysis-4 }
            
                .print-site-enumerate-headings #analysis h5:before { content: '17.3.' counter(counter-analysis-2) '.' counter(counter-analysis-3) '.' counter(counter-analysis-4) '.' counter(counter-analysis-5) ' ' }
                .print-site-enumerate-headings #analysis h5 {  counter-increment: counter-analysis-5 }
            
.print-site-enumerate-headings #format_of_md_files > h1:before { content: '17.4 ' }

                .print-site-enumerate-headings #format_of_md_files h2:before { content: '17.4.' counter(counter-format_of_md_files-2) ' ' }
                .print-site-enumerate-headings #format_of_md_files h2 {  counter-reset: counter-format_of_md_files-3  counter-format_of_md_files-4  counter-format_of_md_files-5 ;  counter-increment: counter-format_of_md_files-2 }
            
                .print-site-enumerate-headings #format_of_md_files h3:before { content: '17.4.' counter(counter-format_of_md_files-2) '.' counter(counter-format_of_md_files-3) ' ' }
                .print-site-enumerate-headings #format_of_md_files h3 {  counter-reset: counter-format_of_md_files-4  counter-format_of_md_files-5 ;  counter-increment: counter-format_of_md_files-3 }
            
                .print-site-enumerate-headings #format_of_md_files h4:before { content: '17.4.' counter(counter-format_of_md_files-2) '.' counter(counter-format_of_md_files-3) '.' counter(counter-format_of_md_files-4) ' ' }
                .print-site-enumerate-headings #format_of_md_files h4 {  counter-reset: counter-format_of_md_files-5 ;  counter-increment: counter-format_of_md_files-4 }
            
                .print-site-enumerate-headings #format_of_md_files h5:before { content: '17.4.' counter(counter-format_of_md_files-2) '.' counter(counter-format_of_md_files-3) '.' counter(counter-format_of_md_files-4) '.' counter(counter-format_of_md_files-5) ' ' }
                .print-site-enumerate-headings #format_of_md_files h5 {  counter-increment: counter-format_of_md_files-5 }
            
.print-site-enumerate-headings #python > h1:before { content: '18 ' }

                .print-site-enumerate-headings #python h2:before { content: '18.' counter(counter-python-2) ' ' }
                .print-site-enumerate-headings #python h2 {  counter-reset: counter-python-3  counter-python-4  counter-python-5  counter-python-6 ;  counter-increment: counter-python-2 }
            
                .print-site-enumerate-headings #python h3:before { content: '18.' counter(counter-python-2) '.' counter(counter-python-3) ' ' }
                .print-site-enumerate-headings #python h3 {  counter-reset: counter-python-4  counter-python-5  counter-python-6 ;  counter-increment: counter-python-3 }
            
                .print-site-enumerate-headings #python h4:before { content: '18.' counter(counter-python-2) '.' counter(counter-python-3) '.' counter(counter-python-4) ' ' }
                .print-site-enumerate-headings #python h4 {  counter-reset: counter-python-5  counter-python-6 ;  counter-increment: counter-python-4 }
            
                .print-site-enumerate-headings #python h5:before { content: '18.' counter(counter-python-2) '.' counter(counter-python-3) '.' counter(counter-python-4) '.' counter(counter-python-5) ' ' }
                .print-site-enumerate-headings #python h5 {  counter-reset: counter-python-6 ;  counter-increment: counter-python-5 }
            
                .print-site-enumerate-headings #python h6:before { content: '18.' counter(counter-python-2) '.' counter(counter-python-3) '.' counter(counter-python-4) '.' counter(counter-python-5) '.' counter(counter-python-6) ' ' }
                .print-site-enumerate-headings #python h6 {  counter-increment: counter-python-6 }
            
.print-site-enumerate-headings #section-19 > h1:before { content: '19 ' }
.print-site-enumerate-headings #sample > h1:before { content: '19.1 ' }

                .print-site-enumerate-headings #sample h2:before { content: '19.1.' counter(counter-sample-2) ' ' }
                .print-site-enumerate-headings #sample h2 {  counter-reset: counter-sample-3  counter-sample-4  counter-sample-5 ;  counter-increment: counter-sample-2 }
            
                .print-site-enumerate-headings #sample h3:before { content: '19.1.' counter(counter-sample-2) '.' counter(counter-sample-3) ' ' }
                .print-site-enumerate-headings #sample h3 {  counter-reset: counter-sample-4  counter-sample-5 ;  counter-increment: counter-sample-3 }
            
                .print-site-enumerate-headings #sample h4:before { content: '19.1.' counter(counter-sample-2) '.' counter(counter-sample-3) '.' counter(counter-sample-4) ' ' }
                .print-site-enumerate-headings #sample h4 {  counter-reset: counter-sample-5 ;  counter-increment: counter-sample-4 }
            
                .print-site-enumerate-headings #sample h5:before { content: '19.1.' counter(counter-sample-2) '.' counter(counter-sample-3) '.' counter(counter-sample-4) '.' counter(counter-sample-5) ' ' }
                .print-site-enumerate-headings #sample h5 {  counter-increment: counter-sample-5 }
            
.print-site-enumerate-headings #geometry-examples > h1:before { content: '19.2 ' }

                .print-site-enumerate-headings #geometry-examples h2:before { content: '19.2.' counter(counter-geometry-examples-2) ' ' }
                .print-site-enumerate-headings #geometry-examples h2 {  counter-reset: counter-geometry-examples-3  counter-geometry-examples-4  counter-geometry-examples-5 ;  counter-increment: counter-geometry-examples-2 }
            
                .print-site-enumerate-headings #geometry-examples h3:before { content: '19.2.' counter(counter-geometry-examples-2) '.' counter(counter-geometry-examples-3) ' ' }
                .print-site-enumerate-headings #geometry-examples h3 {  counter-reset: counter-geometry-examples-4  counter-geometry-examples-5 ;  counter-increment: counter-geometry-examples-3 }
            
                .print-site-enumerate-headings #geometry-examples h4:before { content: '19.2.' counter(counter-geometry-examples-2) '.' counter(counter-geometry-examples-3) '.' counter(counter-geometry-examples-4) ' ' }
                .print-site-enumerate-headings #geometry-examples h4 {  counter-reset: counter-geometry-examples-5 ;  counter-increment: counter-geometry-examples-4 }
            
                .print-site-enumerate-headings #geometry-examples h5:before { content: '19.2.' counter(counter-geometry-examples-2) '.' counter(counter-geometry-examples-3) '.' counter(counter-geometry-examples-4) '.' counter(counter-geometry-examples-5) ' ' }
                .print-site-enumerate-headings #geometry-examples h5 {  counter-increment: counter-geometry-examples-5 }
            
.print-site-enumerate-headings #benchmarks > h1:before { content: '20 ' }

                .print-site-enumerate-headings #benchmarks h2:before { content: '20.' counter(counter-benchmarks-2) ' ' }
                .print-site-enumerate-headings #benchmarks h2 {  counter-reset: counter-benchmarks-3  counter-benchmarks-4  counter-benchmarks-5  counter-benchmarks-6 ;  counter-increment: counter-benchmarks-2 }
            
                .print-site-enumerate-headings #benchmarks h3:before { content: '20.' counter(counter-benchmarks-2) '.' counter(counter-benchmarks-3) ' ' }
                .print-site-enumerate-headings #benchmarks h3 {  counter-reset: counter-benchmarks-4  counter-benchmarks-5  counter-benchmarks-6 ;  counter-increment: counter-benchmarks-3 }
            
                .print-site-enumerate-headings #benchmarks h4:before { content: '20.' counter(counter-benchmarks-2) '.' counter(counter-benchmarks-3) '.' counter(counter-benchmarks-4) ' ' }
                .print-site-enumerate-headings #benchmarks h4 {  counter-reset: counter-benchmarks-5  counter-benchmarks-6 ;  counter-increment: counter-benchmarks-4 }
            
                .print-site-enumerate-headings #benchmarks h5:before { content: '20.' counter(counter-benchmarks-2) '.' counter(counter-benchmarks-3) '.' counter(counter-benchmarks-4) '.' counter(counter-benchmarks-5) ' ' }
                .print-site-enumerate-headings #benchmarks h5 {  counter-reset: counter-benchmarks-6 ;  counter-increment: counter-benchmarks-5 }
            
                .print-site-enumerate-headings #benchmarks h6:before { content: '20.' counter(counter-benchmarks-2) '.' counter(counter-benchmarks-3) '.' counter(counter-benchmarks-4) '.' counter(counter-benchmarks-5) '.' counter(counter-benchmarks-6) ' ' }
                .print-site-enumerate-headings #benchmarks h6 {  counter-increment: counter-benchmarks-6 }
            
.print-site-enumerate-headings #known-bugs > h1:before { content: '21 ' }

                .print-site-enumerate-headings #known-bugs h2:before { content: '21.' counter(counter-known-bugs-2) ' ' }
                .print-site-enumerate-headings #known-bugs h2 {  counter-reset: counter-known-bugs-3  counter-known-bugs-4  counter-known-bugs-5  counter-known-bugs-6 ;  counter-increment: counter-known-bugs-2 }
            
                .print-site-enumerate-headings #known-bugs h3:before { content: '21.' counter(counter-known-bugs-2) '.' counter(counter-known-bugs-3) ' ' }
                .print-site-enumerate-headings #known-bugs h3 {  counter-reset: counter-known-bugs-4  counter-known-bugs-5  counter-known-bugs-6 ;  counter-increment: counter-known-bugs-3 }
            
                .print-site-enumerate-headings #known-bugs h4:before { content: '21.' counter(counter-known-bugs-2) '.' counter(counter-known-bugs-3) '.' counter(counter-known-bugs-4) ' ' }
                .print-site-enumerate-headings #known-bugs h4 {  counter-reset: counter-known-bugs-5  counter-known-bugs-6 ;  counter-increment: counter-known-bugs-4 }
            
                .print-site-enumerate-headings #known-bugs h5:before { content: '21.' counter(counter-known-bugs-2) '.' counter(counter-known-bugs-3) '.' counter(counter-known-bugs-4) '.' counter(counter-known-bugs-5) ' ' }
                .print-site-enumerate-headings #known-bugs h5 {  counter-reset: counter-known-bugs-6 ;  counter-increment: counter-known-bugs-5 }
            
                .print-site-enumerate-headings #known-bugs h6:before { content: '21.' counter(counter-known-bugs-2) '.' counter(counter-known-bugs-3) '.' counter(counter-known-bugs-4) '.' counter(counter-known-bugs-5) '.' counter(counter-known-bugs-6) ' ' }
                .print-site-enumerate-headings #known-bugs h6 {  counter-increment: counter-known-bugs-6 }
            
.print-site-enumerate-headings #faq > h1:before { content: '22 ' }

                .print-site-enumerate-headings #faq h2:before { content: '22.' counter(counter-faq-2) ' ' }
                .print-site-enumerate-headings #faq h2 {  counter-reset: counter-faq-3  counter-faq-4  counter-faq-5  counter-faq-6 ;  counter-increment: counter-faq-2 }
            
                .print-site-enumerate-headings #faq h3:before { content: '22.' counter(counter-faq-2) '.' counter(counter-faq-3) ' ' }
                .print-site-enumerate-headings #faq h3 {  counter-reset: counter-faq-4  counter-faq-5  counter-faq-6 ;  counter-increment: counter-faq-3 }
            
                .print-site-enumerate-headings #faq h4:before { content: '22.' counter(counter-faq-2) '.' counter(counter-faq-3) '.' counter(counter-faq-4) ' ' }
                .print-site-enumerate-headings #faq h4 {  counter-reset: counter-faq-5  counter-faq-6 ;  counter-increment: counter-faq-4 }
            
                .print-site-enumerate-headings #faq h5:before { content: '22.' counter(counter-faq-2) '.' counter(counter-faq-3) '.' counter(counter-faq-4) '.' counter(counter-faq-5) ' ' }
                .print-site-enumerate-headings #faq h5 {  counter-reset: counter-faq-6 ;  counter-increment: counter-faq-5 }
            
                .print-site-enumerate-headings #faq h6:before { content: '22.' counter(counter-faq-2) '.' counter(counter-faq-3) '.' counter(counter-faq-4) '.' counter(counter-faq-5) '.' counter(counter-faq-6) ' ' }
                .print-site-enumerate-headings #faq h6 {  counter-increment: counter-faq-6 }
            
.print-site-enumerate-headings #tutorials > h1:before { content: '23 ' }

                .print-site-enumerate-headings #tutorials h2:before { content: '23.' counter(counter-tutorials-2) ' ' }
                .print-site-enumerate-headings #tutorials h2 {  counter-reset: counter-tutorials-3  counter-tutorials-4  counter-tutorials-5  counter-tutorials-6 ;  counter-increment: counter-tutorials-2 }
            
                .print-site-enumerate-headings #tutorials h3:before { content: '23.' counter(counter-tutorials-2) '.' counter(counter-tutorials-3) ' ' }
                .print-site-enumerate-headings #tutorials h3 {  counter-reset: counter-tutorials-4  counter-tutorials-5  counter-tutorials-6 ;  counter-increment: counter-tutorials-3 }
            
                .print-site-enumerate-headings #tutorials h4:before { content: '23.' counter(counter-tutorials-2) '.' counter(counter-tutorials-3) '.' counter(counter-tutorials-4) ' ' }
                .print-site-enumerate-headings #tutorials h4 {  counter-reset: counter-tutorials-5  counter-tutorials-6 ;  counter-increment: counter-tutorials-4 }
            
                .print-site-enumerate-headings #tutorials h5:before { content: '23.' counter(counter-tutorials-2) '.' counter(counter-tutorials-3) '.' counter(counter-tutorials-4) '.' counter(counter-tutorials-5) ' ' }
                .print-site-enumerate-headings #tutorials h5 {  counter-reset: counter-tutorials-6 ;  counter-increment: counter-tutorials-5 }
            
                .print-site-enumerate-headings #tutorials h6:before { content: '23.' counter(counter-tutorials-2) '.' counter(counter-tutorials-3) '.' counter(counter-tutorials-4) '.' counter(counter-tutorials-5) '.' counter(counter-tutorials-6) ' ' }
                .print-site-enumerate-headings #tutorials h6 {  counter-increment: counter-tutorials-6 }
            
.print-site-enumerate-headings #forum > h1:before { content: '24 ' }

                .print-site-enumerate-headings #forum h2:before { content: '24.' counter(counter-forum-2) ' ' }
                .print-site-enumerate-headings #forum h2 {  counter-reset: counter-forum-3  counter-forum-4  counter-forum-5  counter-forum-6 ;  counter-increment: counter-forum-2 }
            
                .print-site-enumerate-headings #forum h3:before { content: '24.' counter(counter-forum-2) '.' counter(counter-forum-3) ' ' }
                .print-site-enumerate-headings #forum h3 {  counter-reset: counter-forum-4  counter-forum-5  counter-forum-6 ;  counter-increment: counter-forum-3 }
            
                .print-site-enumerate-headings #forum h4:before { content: '24.' counter(counter-forum-2) '.' counter(counter-forum-3) '.' counter(counter-forum-4) ' ' }
                .print-site-enumerate-headings #forum h4 {  counter-reset: counter-forum-5  counter-forum-6 ;  counter-increment: counter-forum-4 }
            
                .print-site-enumerate-headings #forum h5:before { content: '24.' counter(counter-forum-2) '.' counter(counter-forum-3) '.' counter(counter-forum-4) '.' counter(counter-forum-5) ' ' }
                .print-site-enumerate-headings #forum h5 {  counter-reset: counter-forum-6 ;  counter-increment: counter-forum-5 }
            
                .print-site-enumerate-headings #forum h6:before { content: '24.' counter(counter-forum-2) '.' counter(counter-forum-3) '.' counter(counter-forum-4) '.' counter(counter-forum-5) '.' counter(counter-forum-6) ' ' }
                .print-site-enumerate-headings #forum h6 {  counter-increment: counter-forum-6 }
            
.print-site-enumerate-headings #section-27 > h1:before { content: '27 ' }
.print-site-enumerate-headings #armci > h1:before { content: '27.2 ' }

                .print-site-enumerate-headings #armci h2:before { content: '27.2.' counter(counter-armci-2) ' ' }
                .print-site-enumerate-headings #armci h2 {  counter-reset: counter-armci-3  counter-armci-4  counter-armci-5 ;  counter-increment: counter-armci-2 }
            
                .print-site-enumerate-headings #armci h3:before { content: '27.2.' counter(counter-armci-2) '.' counter(counter-armci-3) ' ' }
                .print-site-enumerate-headings #armci h3 {  counter-reset: counter-armci-4  counter-armci-5 ;  counter-increment: counter-armci-3 }
            
                .print-site-enumerate-headings #armci h4:before { content: '27.2.' counter(counter-armci-2) '.' counter(counter-armci-3) '.' counter(counter-armci-4) ' ' }
                .print-site-enumerate-headings #armci h4 {  counter-reset: counter-armci-5 ;  counter-increment: counter-armci-4 }
            
                .print-site-enumerate-headings #armci h5:before { content: '27.2.' counter(counter-armci-2) '.' counter(counter-armci-3) '.' counter(counter-armci-4) '.' counter(counter-armci-5) ' ' }
                .print-site-enumerate-headings #armci h5 {  counter-increment: counter-armci-5 }
            
.print-site-enumerate-headings #interfaces-with-external-software > h1:before { content: '28 ' }

                .print-site-enumerate-headings #interfaces-with-external-software h2:before { content: '28.' counter(counter-interfaces-with-external-software-2) ' ' }
                .print-site-enumerate-headings #interfaces-with-external-software h2 {  counter-reset: counter-interfaces-with-external-software-3  counter-interfaces-with-external-software-4  counter-interfaces-with-external-software-5  counter-interfaces-with-external-software-6 ;  counter-increment: counter-interfaces-with-external-software-2 }
            
                .print-site-enumerate-headings #interfaces-with-external-software h3:before { content: '28.' counter(counter-interfaces-with-external-software-2) '.' counter(counter-interfaces-with-external-software-3) ' ' }
                .print-site-enumerate-headings #interfaces-with-external-software h3 {  counter-reset: counter-interfaces-with-external-software-4  counter-interfaces-with-external-software-5  counter-interfaces-with-external-software-6 ;  counter-increment: counter-interfaces-with-external-software-3 }
            
                .print-site-enumerate-headings #interfaces-with-external-software h4:before { content: '28.' counter(counter-interfaces-with-external-software-2) '.' counter(counter-interfaces-with-external-software-3) '.' counter(counter-interfaces-with-external-software-4) ' ' }
                .print-site-enumerate-headings #interfaces-with-external-software h4 {  counter-reset: counter-interfaces-with-external-software-5  counter-interfaces-with-external-software-6 ;  counter-increment: counter-interfaces-with-external-software-4 }
            
                .print-site-enumerate-headings #interfaces-with-external-software h5:before { content: '28.' counter(counter-interfaces-with-external-software-2) '.' counter(counter-interfaces-with-external-software-3) '.' counter(counter-interfaces-with-external-software-4) '.' counter(counter-interfaces-with-external-software-5) ' ' }
                .print-site-enumerate-headings #interfaces-with-external-software h5 {  counter-reset: counter-interfaces-with-external-software-6 ;  counter-increment: counter-interfaces-with-external-software-5 }
            
                .print-site-enumerate-headings #interfaces-with-external-software h6:before { content: '28.' counter(counter-interfaces-with-external-software-2) '.' counter(counter-interfaces-with-external-software-3) '.' counter(counter-interfaces-with-external-software-4) '.' counter(counter-interfaces-with-external-software-5) '.' counter(counter-interfaces-with-external-software-6) ' ' }
                .print-site-enumerate-headings #interfaces-with-external-software h6 {  counter-increment: counter-interfaces-with-external-software-6 }
            
.print-site-enumerate-headings #software-supporting-nwchem > h1:before { content: '29 ' }

                .print-site-enumerate-headings #software-supporting-nwchem h2:before { content: '29.' counter(counter-software-supporting-nwchem-2) ' ' }
                .print-site-enumerate-headings #software-supporting-nwchem h2 {  counter-reset: counter-software-supporting-nwchem-3  counter-software-supporting-nwchem-4  counter-software-supporting-nwchem-5  counter-software-supporting-nwchem-6 ;  counter-increment: counter-software-supporting-nwchem-2 }
            
                .print-site-enumerate-headings #software-supporting-nwchem h3:before { content: '29.' counter(counter-software-supporting-nwchem-2) '.' counter(counter-software-supporting-nwchem-3) ' ' }
                .print-site-enumerate-headings #software-supporting-nwchem h3 {  counter-reset: counter-software-supporting-nwchem-4  counter-software-supporting-nwchem-5  counter-software-supporting-nwchem-6 ;  counter-increment: counter-software-supporting-nwchem-3 }
            
                .print-site-enumerate-headings #software-supporting-nwchem h4:before { content: '29.' counter(counter-software-supporting-nwchem-2) '.' counter(counter-software-supporting-nwchem-3) '.' counter(counter-software-supporting-nwchem-4) ' ' }
                .print-site-enumerate-headings #software-supporting-nwchem h4 {  counter-reset: counter-software-supporting-nwchem-5  counter-software-supporting-nwchem-6 ;  counter-increment: counter-software-supporting-nwchem-4 }
            
                .print-site-enumerate-headings #software-supporting-nwchem h5:before { content: '29.' counter(counter-software-supporting-nwchem-2) '.' counter(counter-software-supporting-nwchem-3) '.' counter(counter-software-supporting-nwchem-4) '.' counter(counter-software-supporting-nwchem-5) ' ' }
                .print-site-enumerate-headings #software-supporting-nwchem h5 {  counter-reset: counter-software-supporting-nwchem-6 ;  counter-increment: counter-software-supporting-nwchem-5 }
            
                .print-site-enumerate-headings #software-supporting-nwchem h6:before { content: '29.' counter(counter-software-supporting-nwchem-2) '.' counter(counter-software-supporting-nwchem-3) '.' counter(counter-software-supporting-nwchem-4) '.' counter(counter-software-supporting-nwchem-5) '.' counter(counter-software-supporting-nwchem-6) ' ' }
                .print-site-enumerate-headings #software-supporting-nwchem h6 {  counter-increment: counter-software-supporting-nwchem-6 }
            
.print-site-enumerate-headings #containers > h1:before { content: '30 ' }

                .print-site-enumerate-headings #containers h2:before { content: '30.' counter(counter-containers-2) ' ' }
                .print-site-enumerate-headings #containers h2 {  counter-reset: counter-containers-3  counter-containers-4  counter-containers-5  counter-containers-6 ;  counter-increment: counter-containers-2 }
            
                .print-site-enumerate-headings #containers h3:before { content: '30.' counter(counter-containers-2) '.' counter(counter-containers-3) ' ' }
                .print-site-enumerate-headings #containers h3 {  counter-reset: counter-containers-4  counter-containers-5  counter-containers-6 ;  counter-increment: counter-containers-3 }
            
                .print-site-enumerate-headings #containers h4:before { content: '30.' counter(counter-containers-2) '.' counter(counter-containers-3) '.' counter(counter-containers-4) ' ' }
                .print-site-enumerate-headings #containers h4 {  counter-reset: counter-containers-5  counter-containers-6 ;  counter-increment: counter-containers-4 }
            
                .print-site-enumerate-headings #containers h5:before { content: '30.' counter(counter-containers-2) '.' counter(counter-containers-3) '.' counter(counter-containers-4) '.' counter(counter-containers-5) ' ' }
                .print-site-enumerate-headings #containers h5 {  counter-reset: counter-containers-6 ;  counter-increment: counter-containers-5 }
            
                .print-site-enumerate-headings #containers h6:before { content: '30.' counter(counter-containers-2) '.' counter(counter-containers-3) '.' counter(counter-containers-4) '.' counter(counter-containers-5) '.' counter(counter-containers-6) ' ' }
                .print-site-enumerate-headings #containers h6 {  counter-increment: counter-containers-6 }
            </style>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "/", "features": ["navigation.top", "navigation.instant", "navigation.tracking", "navigation.indexes", "navigation.footer", "search.highlight", "search.share", "content.action.edit", "content.action.view", "content.code.copy"], "search": "assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="assets/javascripts/bundle.ad660dcc.min.js"></script>
      
        <script src="js/print-site.js"></script>
      
        <script src="css/mathjax.js"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@4/tex-mml-chtml.js"></script>
      
    
  </body>
</html>